---
title: "01_General_Analysis"
author: "Vincent Somerville"
date: "11/01/2021"
output:
  html_document:
    number_sections: yes
    theme: readable
    highlight: tango
    df_print: paged
    toc: yes
    toc_float:
      collapsed: yes
      smooth_scroll: yes
  pdf_document:
    toc: yes
  #   number_sections: true
  #   theme: readable
  #  html_document:
  #   # code_folding: hide
  #   df_print: paged
  #   toc: true
  #   toc_float:
  #     collapsed: true
  #     smooth_scroll: true
---



```{r setup ,echo=FALSE}
#Rmakrdown setup
knitr::opts_chunk$set(message=FALSE, echo=TRUE, eval=FALSE)

```


```{bash ,echo=FALSE}
#Here, I setup the location were I stored the data and where I will plot my results to. 

mkdir -p /home/vincent/Desktop/Manuscripts/2019_RMK202
cd /home/vincent/Desktop/Manuscripts/2019_RMK202
mkdir -p 02_data
mkdir -p 03_results

```

```{r ,echo=FALSE}
#Here, I setup the location were I stored the data and where I will plot my results to. 

setwd("/home/vincent/Desktop/Manuscripts/2019_RMK202")
knitr::opts_knit$set(root.dir = '/home/vincent/Desktop/Manuscripts/2019_RMK202')
```

```{r ,echo=FALSE}
#load libraries
library(ggbiplot)
library(officer)
library(openxlsx)
library(readr)
library(ggbiplot)
library(tidyverse)
library(patchwork)
library(plotly)
library(readr)
library(tsibble)

library(readr)
library(ggplot2)
library(plyr)
library(plotly)
library(tidyverse)
library(lubridate)
library(growthrates)
library(broom)
library(drc)
library(patchwork)
library(dplyr)
require(zoo)

library(genoPlotR)
library(tidyverse)
library(alluvial)
library(ggalluvial)

library(genoPlotR)

library(igraph)

```

This script includes the complete workflow how the data for the following publication was analysed:

*Functional strain redundancy and persistent phage infection in a Swiss hard cheese starter culture*

The scripts are divided into three subscripts. Each on is saved and compiled as a Rmarkdown with extensive comments. The three parts consist of:

1. the general analysis including all genomic and non-genomic data-analysis
2. the plotting and analysis for the main body figures
3. the plotting and analysis for the supplemental figures

Here, we look at the the first part: **General analysis**

# General analysis

Here, I go through the basic analysis on which the figures are the created upon. 

## Genomic data analysis

This is the analysis of the sequenced isolate data. 

### FastQC

In a first step we creat [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) reports for all the libraries. 
These are then visualized with [MultiQC](https://multiqc.info/)

```{bash fastqc}
threads=8

mkdir -p $mainPath/01_rawData/IlluminaFastQC/


for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/names_changed.txt)
  do
  echo ${id}e

    
    fastqc $mainPath/01_rawData/fastq/gz/${id}_R1.fastq.gz -t ${threads} -o $mainPath/01_rawData/IlluminaFastQC/
    fastqc $mainPath/01_rawData/fastq/gz/${id}_R2.fastq.gz -t ${threads} -o $mainPath/01_rawData/IlluminaFastQC/
    
  done
  mkdir -p $mainPath/01_rawData/IlluminaFastQC/Multiqc,output_file='test.html')

  
  multiqc $mainPath/01_rawData/IlluminaFastQC/ -o $mainPath/01_rawData/IlluminaFastQC/Multiqc


```

### Trim-galore
In the following we clip adaptors with [trim-galore]()

```{bash }
source /archiv/Projects/2018_Culturomics/02_script/fileShortcuts.sh
threads=30
id=RMK202_withOld

for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/names_changed_new.txt)
  do
  echo "##=============="
  echo ${id}
  echo "##=============="
      rm -r $mainPath/01_rawData/trimmed_Galore/gz/${id}
      mkdir -p $mainPath/01_rawData/trimmed_Galore/gz/${id}
      trim_galore -paired -o $mainPath/01_rawData/trimmed_Galore/gz/${id} \
        $mainPath/01_rawData/fastq/gz/${id}_R1.fastq.gz  \
        $mainPath/01_rawData/fastq/gz/${id}_R2.fastq.gz 
        
  echo "##=============="
  echo ${id} "fastQC of trimmed"
  echo "##=============="
      
    mkdir -p $mainPath/01_rawData/trimmed_Galore/fastQC/
      
    fastqc $mainPath/01_rawData/trimmed_Galore/gz/${id}/${id}_R1_val_1.fq.gz -t ${threads} -o $mainPath/01_rawData/trimmed_Galore/fastQC/
    
    fastqc $mainPath/01_rawData/trimmed_Galore/gz/${id}/${id}_R2_val_2.fq.gz -t ${threads} -o $mainPath/01_rawData/trimmed_Galore/fastQC/
    
        
  done 
  

  mkdir -p $mainPath/01_rawData/trimmed_Galore/fastQC/Multiqc,output_file='test.html')

  multiqc $mainPath/01_rawData/trimmed_Galore/fastQC/ -o $mainPath/01_rawData/trimmed_Galore/fastQC//Multiqc



```

### SPAdes

In this step we assemble the raw reads of the isolate with [spades](http://bioinf.spbau.ru/en/spades3.7).

```{bash spades}
source /archiv/Projects/2018_Culturomics/02_script/fileShortcuts.sh

threads=35

for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/names_changed.txt)
  do
  echo ==========================
   echo ${id}
   echo ==========================
 
 
       mkdir -p $mainPath/06_Spades/${id}/assembly/
 
 
     /home/vincent/miniconda2/pkgs/spades-3.13.0-0/bin/spades.py --pe1-1 $mainPath/01_rawData/trimmed_Galore/gz/${id}/${id}_R1_val_1.fq.gz \
       --pe1-2 $mainPath/01_rawData/trimmed_Galore/gz/${id}/${id}_R2_val_2.fq.gz \
       -o $mainPath/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta -t ${threads} -m 100

   
    done

#---------------
##short info
##---------------

for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/isolates_names_changed.txt)
  do
  echo ==========================
   echo ${id}
   echo ==========================
 
   grep ">" -c $mainPath/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/contigs.fasta 
    done
##---------------
##PostAssemblyAnalysis
##---------------

for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/isolates_names_changed.txt)
  do
  echo ==========================
   echo ${id}
   echo ==========================
    
    #tail -3 $mainPath/03_metaphlan2/${id}_trimmed/metaphlan_profile/profiled_metagenome_${id}_trimmed.txt
    PostAssemblyAnalysis.R -a $mainPath/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/scaffolds.fasta \
      -F $mainPath/01_rawData/trimmed_Galore/gz/${id}/${id}_R1_val_1.fq.gz \
      -R $mainPath/01_rawData/trimmed_Galore/gz/${id}/${id}_R2_val_2.fq.gz \
      -o $mainPath/06_Spades/${id}/assembly/PostaAssembly \
      -t 35
    
    done

```

## Metagenomic data analysis

### Illumina

### FastQC

In a first step we creat [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) reports for all the libraries. 
These are then visualized with [MultiQC](https://multiqc.info/)

```{bash}
threads=4

rm -r /archiv/Projects/2019_pilotplant/01_rawData/03_rawFastQCs/allSamples_fastqcs

mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/03_rawFastQCs/allSamples_fastqcs

num=1
for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name.txt )
  do
  echo ${num}"/37  :" ${names}
  num=$((num+1))
    
    fastqc /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/all_fastq_gz/${names}_R1.fastq.gz -t ${threads} -o /archiv/Projects/2019_pilotplant/01_rawData/03_rawFastQCs/allSamples_fastqcs
    fastqc /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/all_fastq_gz/${names}_R2.fastq.gz -t ${threads} -o /archiv/Projects/2019_pilotplant/01_rawData/03_rawFastQCs/allSamples_fastqcs
    
  done
  mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/Multiqc,output_file='test.html')

  
  multiqc /archiv/Projects/2019_pilotplant/01_rawData/03_rawFastQCs/allSamples_fastqcs -o /archiv/Projects/2019_pilotplant/01_rawData/Multiqc

```




### Adapter removal

In the following we clip adaptors with [trim-galore](https://github.com/FelixKrueger/TrimGalore)

```{bash trimGalore}
threads=35

num=1
for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/log/20191031_logDownload_changeName.txt)
  do
  echo ${num}"/3  :" ${names}
  num=$((num+1))
  
  
  
  #    rm -r /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq//trimmed_Galore/gz/
      mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq//trimmed_Galore/gz/
      trim_galore --fastqc -paired -o /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/ \
        /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/all_fastq_gz/${names}_R1.fastq.gz   \
        /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/all_fastq_gz/${names}_R2.fastq.gz 
        
  echo "##=============="
  echo ${id} "fastQC of trimmed"
  echo "##=============="
      
    mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC/
      
    fastqc /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R1_val_1.fq.gz -t ${threads} -o /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore//fastQC/
    #names=th_K2_8h
    fastqc /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R2_val_2.fq.gz -t ${threads} -o /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC/
    
        
  done 
  

  mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC//Multiqc,output_file='test.html')

  multiqc /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC/ -o /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC/Multiqc


  for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/log/20191031_logDownload_changeName.txt)
  do
 mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC/log
echo "/archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC/" >>  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC/log/multiqc_logfile.txt
   
     done 
  
  mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC//Multiqc_02
  
  multiqc --config /archiv/logs/multiQC_config.txt --file-list /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/log/multiqc_logfile.txt \
  -o /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC//Multiqc_02


```


### KneadData

Here, we filter out the cow genome contamination reads with [KneadData](https://huttenhower.sph.harvard.edu/kneaddata/)

```{bash}

##prepare reference DB
mkdir -p /archiv/cowDB/mixed/20190520/

cat /archiv/cowDB/mito/20190520/cowDB_renamed.fasta \
  /archiv/cowDB/genome/20190520/GCF_002263795.1_ARS-UCD1.2_genomic.fna > \
  /archiv/cowDB/mixed/20190520/cow_genomeANDmito_DB.fna


bowtie2-build /archiv/cowDB/mixed/20190520/cow_genomeANDmito_DB.fna /archiv/cowDB/mixed/20190520/cow_genomeANDmito_DB

##===================
##remove reads
##===================

##--------------
##experiment
##--------------

threads=37

num=1
for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name_pilotplant.txt |grep "_K2")
#for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/log/20191031_logDownload_changeName.txt)
  do
  

  echo ${num}"/37  :" ${names}
  num=$((num+1))
rm -r /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}
mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}

#gunzip /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R1_val_1.fq.gz
#gunzip /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R2_val_2.fq.gz


kneaddata --max-memory 90000m --no-discordant -t 37 -p 37 --input /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R1_val_1.fq \
  --input /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R2_val_2.fq \
  -db /archiv/cowDB/mixed/20190520/cow_genomeANDmito_DB  \
  --trimmomatic /home/vincent/Joanito/brain/Software/Trimmomatic-0.39/ \
  --output /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}

done


##--------------
##strains
##--------------


threads=37

for names in $(cat /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name_allOfExperiment.txt |grep -v "K1"|grep "Lyo" -v)
  do
  echo ==========================
   echo ${id}
   echo ==========================
 
rm -r /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}
mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}



#kneaddata --input /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${names}/${names}_R1*fq \
#  --input /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${names}/${names}_R2*fq \
#  -db /archiv/cowDB/mixed/20190520/cow_genomeANDmito_DB  \
#  --trimmomatic /home/vincent/Joanito/brain/Software/Trimmomatic-0.39/ \
#  --output /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}

done



##--------------
##evolution experiment
##--------------

names=G4_6_18
threads=37

for names in $(ls /home/vincent/Projects/2020_StarterEvolution/01_data/20200929_Novogene/X204SC20090774-Z01-F001/trimm_Galore/gz/ | grep "^G")
  do
  echo ==========================
   echo ${id}
   echo ==========================
 
rm -r /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}
mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}



kneaddata --input /home/vincent/Projects/2020_StarterEvolution/01_data/20200929_Novogene/X204SC20090774-Z01-F001/trimm_Galore/gz/${names}/${names}-R1*fq.gz \
  --input /home/vincent/Projects/2020_StarterEvolution/01_data/20200929_Novogene/X204SC20090774-Z01-F001/trimm_Galore/gz/${names}/${names}-R2*fq.gz \
  -db /archiv/cowDB/mixed/20190520/cow_genomeANDmito_DB  -t 37 \
  --trimmomatic /home/vincent/Joanito/brain/Software/Trimmomatic-0.39/ \
  --output /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}


mv /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/*_paired_clean_1.fastq  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}_R1_kneaddata_paired_1.fastq

mv /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/*_paired_clean_2.fastq  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}_R2_kneaddata_paired_2.fastq

for delss in $(ls /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/|grep -v "_kneaddata_paired_" )
do

rm /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/$delss


done #delss

done




```

###mOTUs

For taxon identifcation we use [mOTU](https://motu-tool.org/). 

```{bash}


num=1
  for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/names.txt)
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))
  rm -r /archiv/Projects/2019_pilotplant/mOTU/${names}/

  mkdir -p /archiv/Projects/2019_pilotplant/mOTU/${names}/
 
motus profile -f /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_1.fastq \
  -r /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_2.fastq -o /archiv/Projects/2019_pilotplant/mOTU/${names}/ -t 37 -n ${names}_mOTU

done

rm /archiv/Projects/2019_pilotplant/mOTU/all_mOTUs_prepforR.txt
  for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/names_onlymeta.txt)
  do
#  echo "-----------------------------------------"
#  echo ${names}

grep "^#" -v /archiv/Projects/2019_pilotplant/mOTU/${names}/* |grep -v "0.0000000000" |grep -v "sp." |awk -F "\t" -v sampless="${names}" 'BEGIN{OFS="\t"} {print $1, $2, sampless}' >> /archiv/Projects/2019_pilotplant/mOTU/all_mOTUs_prepforR.txt

done

```


### ONT
### Read filtering

In a first step I filter the fastq Nanopore reads for the pooled replicates with [filtlong](https://github.com/rrwick/Filtlong). 
I use a filter for min read length of 8kb and the best quality reads. 


```{bash}

##2AB
/home/vincent/apps/Filtlong/bin/filtlong --min_length 7000 --keep_percent 90 --target_bases 1000000000 --mean_q_weight 10 /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_2A2B/fastq_pass/all_2AB_raw.fq | gzip > /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_2A2B/fastq_pass/all_2AB_qualityFiltered_min7000_Q90.fq.gz

mv /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_2A2B/fastq_pass/all_2AB_qualityFiltered_min7000_Q90.fq /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_2A2B/fastq_pass/all_2AB_qualityFiltered_min7000_Q90.fq.gz

##1AB
/home/vincent/apps/Filtlong/bin/filtlong --min_length 7000 --keep_percent 90 --target_bases 1000000000 --mean_q_weight 10 /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_1A1B/fastq_pass/all_1AB_raw.fq | gzip > /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_1A1B/fastq_pass/all_1AB_qualityFiltered_min7000_Q90.fq.gz

mv /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_1A1B/fastq_pass/all_1AB_qualityFiltered_min7000_Q90.fq /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_1A1B/fastq_pass/all_1AB_qualityFiltered_min7000_Q90.fq.gz


##===============
##Demultiplexed data
##===============

mkdir -p /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered

for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
  do

  echo "=============="
  echo ${sample}
  echo "=============="

    /home/vincent/apps/Filtlong/bin/filtlong --min_length 7000 --keep_percent 90 --target_bases 1000000000 --mean_q_weight 10 /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_renamed/${sample}_nanopore_raw.fastq.gz | gzip > /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/${sample}_nanopore_filtered.fastq.gz


  done


scp -r vincent@130.223.51.151:/archiv/Projects/2019_pilotplant/01_rawData/Multiqc /home/vincent/Desktop/Projects/2019_Nano_Meta/Illumina_analysis


```


### Genome Assembly

Here, we create a [flye](https://github.com/fenderglass/Flye/blob/flye/docs/graph_example.png) assembly of the following:
1. demultiplexed samples with Qualtiy filtering max 100000000 and minLength 7000 reads
2. Than I look at the Assembly graph with Bandage. 
3. Map the short contigs to blastN online to id them

```{bash}

##===========
##Demultiplexed
##===========

##===========
##Assembly plasmid
##===========
for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
  do
  
    sample=th_K2_8h
    
    rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${sample}/assembly/
    mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${sample}/assembly/
  echo "=============="
  echo ${sample}
  echo "=============="

    flye --threads 15  --iterations 5 --genome-size 4500000 --plasmids --nano-raw \
      /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/${sample}_nanopore_filtered.fastq.gz \
      --out-dir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${sample}/assembly/

  done



###---Bandage
sample=th_K2_8h


mkdir -p ~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/assembly/

scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${sample}/assembly/assembly_graph.gfa ~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/assembly/${sample}_assembly_Graph_plasmid.gfa



##-------assembly length

seq_length.py /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/assembly.fasta
seq_length.py /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/th_K2_8h/assembly/assembly.fasta

##-------cd -hit
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/cd_hit
#cd-hit-est -i  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/assembly.fasta \
#    -o  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/cd_hit/assembly_Clustering.txt -c 0.85 -M 30000 -T 37 -sc 1 -sf 1
  
/home/vincent/apps/cd-hit-v4.8.1-2019-0228/cd-hit-est -i   /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/assembly.fasta \
    -o  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/cd_hit/assembly_Clustering.txt -c 0.85
  
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/th_K2_8h/assembly/cd_hit
/home/vincent/apps/cd-hit-v4.8.1-2019-0228/cd-hit-est -i  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/th_K2_8h/assembly/assembly.fasta     -o  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/th_K2_8h/assembly/cd_hit/assembly_Clustering.txt -c 0.85


```




### circlator
as circlator did not work. I will do a prokka annotation and annotate manually at the DnaA

```{bash}

###-----------devide all contigs
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs
for contigs in $(grep ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/di_rmk202_MAG_reference_renamed.fasta |sed 's/>//g' )
  do
  echo ${contigs}
samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/di_rmk202_MAG_reference_renamed.fasta ${contigs} > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/${contigs}.fasta
  done
###-----------PRokka
##------------------------------------Streptococcus thermophilus
contigs=S_thermophilus_mag_rmk202
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/unpolished/${contigs}
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/unpolished/${contigs}
    
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/unpolished/${contigs}/ --force --usegenus --genus Streptococcus --locustag Sterm_202 --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/${contigs}.fasta
##------------------------------------Lactoabillus delbrueckii
contigs=L_delbrueckii_mag_rmk202
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/unpolished/${contigs}
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/unpolished/${contigs}
    
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/unpolished/${contigs}/ --force --usegenus --genus Lactobacillus --locustag Ldel_202 --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/${contigs}.fasta
  
  ##=================  
###-----------rearrang contigs
  ##=================  
  
  ##===================================================Lactobacillus delbrueckii
  grep "DNAa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/unpolished/L_delbrueckii_mag_rmk202/PROKKA_11142019.gff
  
  #37825
  contigs=L_delbrueckii_mag_rmk202


###---------------extract fragemtent 1-------------------
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/logs/
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/logs/

echo -e ${contigs}"\t1\t37000" > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/circlaring_ldel/logs/${contigs}_fragment1.bed

samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/${contigs}.fasta

bedtools getfasta -fi /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/${contigs}.fasta \
  -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/circlaring_ldel/logs/${contigs}_fragment1.bed \
  -fo /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/circlaring_ldel/logs/${contigs}_fragment1.fasta

grep -v ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/circlaring_ldel/logs/${contigs}_fragment1.fasta > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/circlaring_ldel/logs/${contigs}_fragment1_megerable.fasta


###---------------extract fragemtent 3-------------------


#seq_length.py /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/${contigs}.fasta

echo -e ${contigs}"\t37001\t2193739" > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/logs//${contigs}_fragment2.bed

#samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring_ldel/input/contig_1.fasta

bedtools getfasta -fi /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/${contigs}.fasta \
  -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/logs//${contigs}_fragment2.bed \
  -fo /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/logs//${contigs}_fragment2.fasta

grep -v ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/logs//${contigs}_fragment2.fasta > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/logs//${contigs}_fragment2_mergeable.fasta


##-------merging fragments-------
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/final

echo ">"${contigs} > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/final/${contigs}_unformated.fasta

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/circlaring_ldel/logs/${contigs}_fragment2_mergeable.fasta \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/logs//${contigs}_fragment1_megerable.fasta | sed 's/\n//g' >>  \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/final/${contigs}_unformated.fasta
  

fasta_formatter -w 50 -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/final/${contigs}_unformated.fasta -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/final/${contigs}.fasta

###------------------checking with annotation
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/final/annoted/${contigs}
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/final/annoted/${contigs}
    
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/final/annoted/${contigs} --force --usegenus --genus Lactobacillus --locustag Ldel_202 --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/final/${contigs}.fasta
  
  grep "DNAa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/final/annoted/${contigs}/PROKKA_*.gff
  ##===================================================Streptococcus_thermophilus


  ##===================================================Lactobacillus delbrueckii
  contigs=S_thermophilus_mag_rmk202
  grep "DNAa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/unpolished/${contigs}/PROKKA_*.gff
  #345399
  
###---------------extract fragemtent 1-------------------
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/logs/
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/logs/

echo -e ${contigs}"\t1\t344900" > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/circlaring_sterm/logs/${contigs}_fragment1.bed

samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/${contigs}.fasta

bedtools getfasta -fi /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/${contigs}.fasta \
  -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/circlaring_sterm/logs/${contigs}_fragment1.bed \
  -fo /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/circlaring_sterm/logs/${contigs}_fragment1.fasta

grep -v ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/circlaring_sterm/logs/${contigs}_fragment1.fasta > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/circlaring_sterm/logs/${contigs}_fragment1_megerable.fasta


###---------------extract fragemtent 3-------------------


#seq_length.py /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/${contigs}.fasta

echo -e ${contigs}"\t344900\t1895563" > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/logs//${contigs}_fragment2.bed

#samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring_sterm/input/contig_1.fasta

bedtools getfasta -fi /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/${contigs}.fasta \
  -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/logs//${contigs}_fragment2.bed \
  -fo /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/logs//${contigs}_fragment2.fasta

grep -v ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/logs//${contigs}_fragment2.fasta > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/logs//${contigs}_fragment2_mergeable.fasta


##-------merging fragments-------
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final

echo ">"${contigs} > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/${contigs}_unformated.fasta

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/circlaring_sterm/logs/${contigs}_fragment2_mergeable.fasta \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/logs//${contigs}_fragment1_megerable.fasta | sed 's/\n//g' >>  \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/${contigs}_unformated.fasta
  

fasta_formatter -w 50 -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/${contigs}_unformated.fasta -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/${contigs}.fasta

###------------------checking with annotation
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/${contigs}
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/${contigs}
    
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/${contigs} --force --usegenus --genus Streptococcus --locustag Sterm_202 --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/${contigs}.fasta
  
  grep "DNAa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/${contigs}/PROKKA_*.gff
  


##--------------------------------merge all again
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/final/L_delbrueckii_mag_rmk202.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/S_thermophilus_mag_rmk202.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/L_del_plasmid_01.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/S_thermophilus_repA.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/L_del_phage_01.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/Bos_taurus_mito.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/Bos_taurus_genome_01.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/Bos_taurus_genome_02.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/Bos_taurus_genome_03.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/Bos_taurus_genome_04.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/Bos_taurus_genome_05.fasta > \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned_unformated.fasta


fasta_formatter -w 50 -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned_unformated.fasta -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta



#/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/Unknown_01.fasta \

###------------------checking with annotation
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all
    
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all --force --usegenus --genus Streptococcus --locustag Sterm_202 --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta
  
  grep "DNAa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all/PROKKA_*.gff
  

```



### ONT polishing Racon

After cirlarication we do several rounds of ONT-based polishing with [Racon](https://github.com/isovic/racon)
This is then check by continious Quality control steps with the postAssemblyAnylsis tool. 


```{bash}

###===========================
###-----------------
##Racon polishing with graphmap
###-----------------
###===========================    
     
##------------------define variables------------------
#for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt |tail -2)
#do


name=di_K2_6h
sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta 
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/di_K2_6h_nanopore_filtered.fastq.gz
#reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta
echo "========================================="
echo $sample
echo "========================================="

echo $reference_fasta
echo "---------------------------"
     
###===========================
###-----------------
##graphmap mapping
###-----------------
###===========================      

name=first
num=1

for name in first second third fourth
do 
rm -r $Overall_output_directory/0${num}_${name}Racon_graphmap/
mkdir -p $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/bamqc

#name=first

graphmap align -t ${threads} \
  -r $reference_fasta \
  -d $ont_reads \
  -o $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam 

#  -d $ont_reads | samtools sort -@${threads} -O BAM -o $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam - 

  
###===========================
###-----------------
##racon polishing
###-----------------
###===========================   

#rm -r$Overall_output_directory/0${num}_${name}Racon_graphmap/assembly

mkdir -p $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly

samtools view -bS $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam | samtools sort -@${threads} -O BAM -o $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam -

#samtools sort -@${threads} -O SAM $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam -

#samtools view -S -b $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam > $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam 

/home/vincent/apps/racon/new/racon/build/bin/racon \
 -t ${threads} \
  $ont_reads \
  $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam \
  $reference_fasta > \
  $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly/${name}Racon_NWC_2Assembly.fasta
  


samtools index $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam 

qualimap bamqc -bam $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam \
  -outdir $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/bamqc --java-mem-size=40G 

#rm $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam

  reference_fasta=$(echo $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly/${name}Racon_NWC_2Assembly.fasta)
echo $reference_fasta
num=$((num+1)) 

done


##--------------------cleanup


name=di_K2_6h
sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta 
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/di_K2_6h_nanopore_filtered.fastq.gz
#reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta
name=first
num=1

for name in first second third fourth
do 


rm -r $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/

  reference_fasta=$(echo $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly/${name}Racon_NWC_2Assembly.fasta)
echo $reference_fasta
num=$((num+1)) 

done
#done # loop for 1AB and 2AB samples


##--------------------test PROKKA
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
name=fourth
num=4

rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all_0${num}_Racon
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all_0${num}_Racon
    
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all_0${num}_Racon --force --usegenus --genus Streptococcus --locustag Sterm_202 --proteins proteins.faa --evalue 0.001 --addgenes   $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly/${name}Racon_NWC_2Assembly.fasta
  
  grep "DNAa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all_0${num}_Racon/PROKKA_*.gff
  
  echo "=======================================================ALT==============================================================="
  grep "DNAa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all/PROKKA_*.gff


  


###===========================
###-----------------
##quast
###-----------------
###===========================  

#for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
#do


name=di_K2_6h
sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta 
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/di_K2_6h_nanopore_filtered.fastq.gz


echo "========================================="
echo $sample
echo "========================================="

echo $reference_fasta
echo "---------------------------"
     
    rm -r $Overall_output_directory/quastwithRaconONT/
    mkdir -p $Overall_output_directory/quastwithRaconONT
    quast.py \
      -t ${threads} -l "FlyeAssembly,firstONTracon,secondONTracon,thirdONTracon" \
      -R $Overall_output_directory/04_fourthRacon_graphmap/assembly/fourthRacon_NWC_2Assembly.fasta \
      $reference_fasta \
      $Overall_output_directory/01_firstRacon_graphmap/assembly/firstRacon_NWC_2Assembly.fasta \
      $Overall_output_directory/02_secondRacon_graphmap/assembly/secondRacon_NWC_2Assembly.fasta \
      $Overall_output_directory/03_thirdRacon_graphmap/assembly/thirdRacon_NWC_2Assembly.fasta \
      -o $Overall_output_directory/quastwithRaconONT
 
 
 ##-------------------making MultiQC output
 rm $Overall_output_directory/logs_MultiQC_raconONT.txt
 
 echo $Overall_output_directory/quastwithRaconONT/report.tsv >> $Overall_output_directory/logs_MultiQC_raconONT.txt
 
 num=1

for name in first second third fourth
do 

echo $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/bamqc >> $Overall_output_directory/logs_MultiQC_raconONT.txt
num=$((num+1)) 

done ##loop for name
 
#done #loop for sample
###===========================
###-----------------
##MultiQC
###-----------------
###===========================  


#for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
#do
name=di_K2_6h
sample=di_K2_6h
threads=37
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${sample}/
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${sample}/assembly/assembly.fasta 
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/${sample}_nanopore_filtered.fastq.gz

echo "========================================="
echo $sample
echo "========================================="



#done #loop for sample
 
rm -r $Overall_output_directory/MultiQCRaconONT/
mkdir -p $Overall_output_directory/MultiQCRaconONT/
  /home/vincent/.local/bin/multiqc --config /archiv/logs/multiQC_config.txt --file-list   $Overall_output_directory/logs_MultiQC_raconONT.txt \
    -o $Overall_output_directory/MultiQCRaconONT/



###--------------on local

name=di_K2_6h
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/quastwithRaconONT/report.html ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC/${name}_quast_report.html

scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/MultiQCRaconONT/multiqc_report.html ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC/${name}_multiqc_report.html



```


### Illumina polishing freebayes 

Now we polish with the high quality Illumina reads. 

```{bash}


###-----------------
##Illuminamapping rawReads to circlator genome
###-----------------

   
##------------------define variables------------------
#for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
#do

name=di_K2_6h
sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/04_fourthRacon_graphmap/assembly/fourthRacon_NWC_2Assembly.fasta
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/di_K2_6h_nanopore_filtered.fastq.gz

num=5


for run in first second third fourth
   do 

name=run
##--------------create directories----------------
rm -r $Overall_output_directory/0${num}_${run}Freebayes/
mkdir -p $Overall_output_directory/0${num}_${run}Freebayes/{rawIlluminaMapped,freebayesOuput}
mkdir -p $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/bamqc

#mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fq

# gunzip -c /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R1_val_1.fq.gz > \
#  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fq/${sample}_R1_val_1.fq
# gunzip -c /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R2_val_2.fq.gz > \
#  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fq/${sample}_R2_val_2.fq


bwa index $reference_fasta

bwa mem -t ${threads}  $reference_fasta /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R1_val_1.fq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R2_val_2.fq |samtools sort -@8 -O BAM \
-o $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam - 
     

qualimap bamqc -bam $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam \
        -outdir $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/bamqc --java-mem-size=40G
    
samtools index $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam

###-----------------
##freebayesAfterCirclator
###-----------------
echo freebayes

    SECONDS=0

    freebayes -F 0.5 --min-coverage 4 -p 1 -f \
      $reference_fasta \
      $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam > \
      $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf
      
      
    echo "RUNTIME (h:min:s)= $(($SECONDS / 3600)):$((($SECONDS / 60) % 60)):$(($SECONDS % 60))"
    echo END = `date +"%Y-%m-%d %H:%M:%S"`

##-------   
##bcftools variant integration of Illumina data into Pacbio polished genome
##-------

    SECONDS=0
    echo "bgzip"
bgzip -c  $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf > \
       $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf.gz
      echo "tabix"
tabix -p vcf  $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf.gz
    echo "bcftools"
bcftools consensus -f $reference_fasta \
      -o $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${num}_${run}Freebayes.fasta \
       $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf.gz
      
    echo "RUNTIME (h:min:s)= $(($SECONDS / 3600)):$((($SECONDS / 60) % 60)):$(($SECONDS % 60))"
    echo END = `date +"%Y-%m-%d %H:%M:%S"`
  
  bwa index $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${num}_${run}Freebayes.fasta 
  
   ##-------------------new variable name for next polishing steps-----------
  reference_fasta=$Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${num}_${run}Freebayes.fasta  
  num=$((num+1)) 
done
    
 ##------------------------------------------clean-------------------------  


name=di_K2_6h
sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/04_fourthRacon_graphmap/assembly/fourthRacon_NWC_2Assembly.fasta
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/di_K2_6h_nanopore_filtered.fastq.gz

num=5


for run in first second third fourth
   do 

name=run

rm -r $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped

 num=$((num+1)) 

done

##--------------indexing


bwa index $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${num}_${run}Freebayes.fasta  




##--------------------test PROKKA
#Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
name=second
num=6

rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all_0${num}
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all_0${num}
    
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all_0${num} --force --usegenus --genus Streptococcus --locustag Sterm_202 --proteins proteins.faa --evalue 0.001 --addgenes  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/0${num}_${name}Freebayes/freebayesOuput/${num}_${name}Freebayes.fasta
  
  grep "DNAa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all_0${num}/PROKKA_*.gff
  
  echo "=======================================================ALT==============================================================="
  grep "DNAa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all/PROKKA_*.gff



#done #samples
###===========================
###-----------------
##quast
###-----------------
###===========================  

#for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
#do
#sample=1
name=di_K2_6h
sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta 
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/di_K2_6h_nanopore_filtered.fastq.gz

rm $Overall_output_directory/logs_MultiQC_raconFreebayesONT.txt

echo "========================================="
echo $sample
echo "========================================="

echo $reference_fasta
echo "---------------------------"
     
    rm -r $Overall_output_directory/quastwithRacon_FreebayesONT/
    mkdir -p $Overall_output_directory/quastwithRacon_FreebayesONT
    quast.py \
      -t ${threads} -l "FlyeAssembly,firstONTracon,secondONTracon,thirdONTracon,fourthONTracon,firstFreebayes,secondFreebayes,thirdFreebayes" \
      -R $Overall_output_directory/08_fourthFreebayes/freebayesOuput/8_fourthFreebayes.fasta \
      $reference_fasta \
      $Overall_output_directory/01_firstRacon_graphmap/assembly/firstRacon_NWC_2Assembly.fasta \
      $Overall_output_directory/02_secondRacon_graphmap/assembly/secondRacon_NWC_2Assembly.fasta \
      $Overall_output_directory/03_thirdRacon_graphmap/assembly/thirdRacon_NWC_2Assembly.fasta \
      $Overall_output_directory/04_fourthRacon_graphmap/assembly/fourthRacon_NWC_2Assembly.fasta \
      $Overall_output_directory/05_firstFreebayes/freebayesOuput/5_firstFreebayes.fasta \
      $Overall_output_directory/06_secondFreebayes/freebayesOuput/6_secondFreebayes.fasta  \
      $Overall_output_directory/07_thirdFreebayes/freebayesOuput/7_thirdFreebayes.fasta  \
            -o $Overall_output_directory/quastwithRacon_FreebayesONT
 
      
     
    
      
###===========================
###-----------------
##MultiQC
###-----------------
###===========================  


name=di_K2_6h
sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta 
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/di_K2_6h_nanopore_filtered.fastq.gz

echo "========================================="
echo $sample
echo "========================================="


  rm -r $Overall_output_directory/MultiQCRaconFreebayesONT
    
  mkdir -p $Overall_output_directory/MultiQCRaconFreebayesONT
  
  /home/vincent/.local/bin/multiqc --config /archiv/logs/multiQC_config.txt --file-list  $Overall_output_directory/logs_MultiQC_raconFreebayesONT.txt \
    -o $Overall_output_directory/MultiQCRaconFreebayesONT/




```


### remove cow

In some metagenoms we noticed some cow contamination originating from the milk. We manually filtered this out by remove the assembled mitocondrial cow contig. 

```{bash}

###----------------------------------devide all contigs
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/08_fourthFreebayes/splitContigs/
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/08_fourthFreebayes/splitContigs/
for contigs in $(grep ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/08_fourthFreebayes/freebayesOuput/di_rmk202_MAG_reference_polished_renamed.fasta |sed 's/>//g' )
  do
  echo ${contigs}
samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/08_fourthFreebayes/freebayesOuput/di_rmk202_MAG_reference_polished_renamed.fasta ${contigs} > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/08_fourthFreebayes/splitContigs//${contigs}.fasta
  done

##---------------------------------------------only keep non cow

mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/Final_Polished_withoutCow/

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/08_fourthFreebayes/splitContigs//L_delbrueckii_mag_rmk202.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/08_fourthFreebayes/splitContigs//S_thermophilus_mag_rmk202.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/08_fourthFreebayes/splitContigs//L_del_plasmid_01.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/08_fourthFreebayes/splitContigs//S_thermophilus_repA.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/08_fourthFreebayes/splitContigs//L_del_phage_01.fasta > \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/Final_Polished_withoutCow/di_rmk202_MAG_reference_polished_unformated.fasta


fasta_formatter -w 50 -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/Final_Polished_withoutCow/di_rmk202_MAG_reference_polished_unformated.fasta -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/Final_Polished_withoutCow/di_rmk202_MAG_reference_polished.fasta


##---------------------------------------------BRING LOCAL

mkdir -p ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_FINALgENOMEassembly/
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/08_fourthFreebayes/splitContigs/S_thermophilus_mag_rmk202.fasta ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_FINALgENOMEassembly/

scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/08_fourthFreebayes/splitContigs/L_delbrueckii_mag_rmk202.fasta ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_FINALgENOMEassembly/


##-------------phage for RAST
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/08_fourthFreebayes/splitContigs/L_del_phage_01.fasta ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_FINALgENOMEassembly/


```


## phage disentangle


### plasmid spades of non-mapping

```{bash}

##===============
##merge all non-mapping reads
##===============
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/nonMappingReads_merged.fq
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/
for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt)
do


cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq >> /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/nonMappingReads_merged.fq

done


##===============
##spades plasmid assembly
##===============

mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly

/home/vincent/miniconda2/pkgs/spades-3.13.0-0/bin/spades.py --plasmid -t 37 -m 100 \
  -s /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/nonMappingReads_merged.fq \
  -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly
  
##--------------meta  

mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly_meta

/home/vincent/miniconda2/pkgs/spades-3.13.0-0/bin/spades.py --meta -t 37 -m 100 \
  -s /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/nonMappingReads_merged.fq \
  -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly_meta
  
###--------------------------------move local


names=di_K2_6h
mkdir -p ~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_demupliplex/assembly/
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly/assembly_graph_with_scaffolds.gfa ~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_demupliplex/assembly/${names}_graph.gfa


###======================================single sample unmapped assembly

for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt|grep "_K")
do
echo ${names}
##--------------plasmid spades
rm - r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly_everySample_indiviually_plasmid/${names}/
mkdir -p  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly_everySample_indiviually_plasmid/${names}/
/home/vincent/miniconda2/pkgs/spades-3.13.0-0/bin/spades.py --plasmid -t 37 -m 100 \
  -s /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq \
  -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly_everySample_indiviually_plasmid/${names}/

##--------------meta  
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly_everySample_indiviually_meta/${names}/
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly_everySample_indiviually_meta/${names}/
/home/vincent/miniconda2/pkgs/spades-3.13.0-0/bin/spades.py --plasmid -t 37 -m 100 \
  -s /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq \
  -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly_everySample_indiviually_meta/${names}/


done



```



```{bash}
rm - r /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/log/
mkidr -p /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/log/

 num=1
  for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt )
    do
    
    echo ${num}"/37  :" ${names} 
    num=$((num+1))

##extract from qualimap

grep -A 100 ">>>> Coverage per contig" /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/${names}/bwaMapping2DB/bamqc/genome_results.txt | sed '1,2d' | sed '/^$/d' | sed -e 's/^[ \t]*//'  |grep -v "CP" |awk -F "\t" -v namess="$names"  'BEGIN{OFS="\t"}{print $0, namess}' >> /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/log/mappingcoverage.txt

done


scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/log/mappingcoverage.txt ~/Desktop/Projects/2019_Pilotplan/PhagemappingCoverage.txt

```


```{r}

library(readr)
library(plyr)
library(tidyverse)


PhagemappingCoverage <- read_delim("Desktop/Projects/2019_Pilotplan/PhagemappingCoverage.txt",  "\t", escape_double = FALSE, col_names = c("contig","size","reads","coverage","stdv","sample"),  trim_ws = TRUE)

tmp1 <- data.frame(str_split_fixed(PhagemappingCoverage$contig, fixed("_"), 7))[,1:4]
PhagemappingCoverage$genome <- paste(tmp1$X1,tmp1$X2,tmp1$X3,tmp1$X4,sep="_")

PhagemappingCoverage$coverage <- PhagemappingCoverage$coverage+0.1

ggplot(PhagemappingCoverage,aes(x=coverage,y=size,color=genome,fill=genome))+geom_point()+theme_classic()


PhagemappingCoverage$sample <- revalue(PhagemappingCoverage$sample, c("lyo202_96"="Lyo\n1996","Lyo_202_2012"="Lyo\n2012", "Lyo_202_2014"="Lyo\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018","di_K2_6h"="cheesemaking\nday1","th_K2_8h"="cheesemaking\nday2"))
PhagemappingCoverage$sample = factor(PhagemappingCoverage$sample, levels=c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2"))
levels(PhagemappingCoverage$sample)
# PhagemappingCoverage$sample <- as.factor(PhagemappingCoverage$sample)

Phage_mapping_coverage <- ggplot(PhagemappingCoverage,aes(x=sample,y=coverage,group=contig,color=genome,fill=genome))+ geom_line(size=0.5, alpha=.8)+ 
       facet_wrap(genome~.,scales = "free")+
   labs("",
         x="",
         # y="",
          y="all")+
       # scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  coord_trans( y="log2")+
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
Phage_mapping_coverage


```

I had to manually disentangle the two genomes manually. 
By looking at the coverages of the shared and divergent regions. 
In order to plot this I had to manually get the coverage of the different components out of bandage. 
The components are not the contigs. After disentengling the genomes I merge and extracted the path

```{r}
library(readr)
library(ggplot2)
library(plyr)
library(dplyr)
library(tidyverse)
library(patchwork)

phages_sterm_contigs_mapping <- read_delim("~/Desktop/Projects/2019_Pilotplan/11_phages/phages_sterm_contigs_mapping/phages_sterm_contigs_mapping.csv", "\t", escape_double = FALSE, trim_ws = TRUE)
  phages_sterm_contigs_mapping$contigFraction <- factor(phages_sterm_contigs_mapping$contigFraction, levels=c("A","B","C"))

  
phages_sterm_contigs_mapping$contigFraction <- revalue(phages_sterm_contigs_mapping$contigFraction, c("A"="High\ncoverage\ncontigs","B"="Low\ncoverage\ncontigs","C"="Shared\ncontigs"))



# phages_sterm_contigs_mapping$contigNumber

plotPoints <- ggplot(phages_sterm_contigs_mapping,aes(x=depth,y=length,group=contigNumber,color=contigFraction,fill=contigFraction))+geom_point(size=2)+geom_line()+labs(x="Contig coverage",y="Contig length")+theme_classic()+coord_trans( y="log2")+scale_y_continuous(breaks =c(0,100,250,500,1000,5000,10000,20000,25000),labels=c(0,100,250,500,1000,5000,10000,20000,25000),limits = c(100,25000))+theme(legend.position = "none")

plot_box <- ggplot(phages_sterm_contigs_mapping,aes(x=contigFraction,y=depth,color=contigFraction,fill=contigFraction))+geom_boxplot(outlier.size =2)+scale_y_continuous(limits=c(0,2100))+coord_flip()+theme_classic()+theme(legend.position = "none",axis.title = element_blank(),axis.text.x=element_blank())

(plot_box/plotPoints)+plot_layout(heights =c(1,2))


  svg("~/Desktop/Projects/2019_Pilotplan/11_phages/phages_sterm_contigs_mapping/phage_contigsPlot.svg",width=6,height=7)
 # png("~/Desktop/Projects/2019_Pilotplan/11_phages/phages_sterm_contigs_mapping/phage_contigsPlot.png", width = 1400, height = 1600,res=300)
(plot_box/plotPoints)+plot_layout(heights =c(1,2))
 dev.off()

```




###polish disentangled genomes



```{bash}
###------------------------
#rename
###------------------------
cd /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/02_blast_nonBacterialReadsAssembled/PlasmidSpades/AssemblyGraphs

sed 's/>.*$/>Streptococcus_phage_1/g' lyo96_phage_sterm_high_path > Phages_rmk202.fasta
sed 's/>.*$/>Streptococcus_phage_2/g' Sterm_low_coverage_lyo20296.fasta >> Phages_rmk202.fasta
sed 's/>.*$/>Lactobacillus_phage_2/g' Sterm_3_lyo20296.fasta >> Phages_rmk202.fasta
sed  's/>.*$/>Lactobacillus_phage_1/g' Lactobacillus_phage_rmk202.fasta >> Phages_rmk202.fasta


##==================
##cd-hit
##==================
mkdir -p cd_hit_phages
cd-hit-est -i Phages_rmk202.fasta\
    -o cd_hit_phages/phages.txt -c 0.90 -M 30000 -T 37 -sc 1 -sf 1


###------------------------
#bring  bigmachine
###------------------------


scp /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/02_blast_nonBacterialReadsAssembled/PlasmidSpades/AssemblyGraphs/Phages_rmk202.fasta vincent@130.223.51.116:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/


###------------------------
#merge with bacterial genomes
###------------------------
cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished_onlyBACTERIA_PLASMID.fasta /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/Phages_rmk202.fasta > \
 /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/rmk202_bacteria_phage_mag_ref.fasta


###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt  ##the file with all sample names
logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_withPhageAssembly
Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/rmk202_bacteria_phage_mag_ref.fasta

##!!!!!!!!!!!!!change names of Forward and Reverse reads

###================
##script
###================
##============
##mapping to reference
##============
bwa index $Assembly
# name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 

mkdir -p ${logFilelocation}

num=1

for names in $(cat ${samplesss})
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))
  rm -r ${BaseLocation}/${names}

  mkdir -p ${BaseLocation}/${names}/bwaMapping2DB/
  
 # bwa mem -t ${threads} /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids/di_rmk202_MAG_reference_polished.fasta
  bwa mem -t ${threads} ${Assembly} \
    /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_1.fastq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_2.fastq | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 

samtools view -b -f 4 ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

bedtools bamtofastq -i ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam -fq ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq

rm ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam &

##-----Qualimap

mkdir -p ${BaseLocation}/${names}/bwaMapping2DB//bamqc

qualimap bamqc -bam ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir ${BaseLocation}/${names}/bwaMapping2DB//bamqc --java-mem-size=80G


 mkdir -p ${BaseLocation}/log
echo ${BaseLocation}"/"${names}"/bwaMapping2DB/bamqc" >> ${BaseLocation}/log/multiqc_logfile_finalGenome.txt
   done 
   
   ##----------------------------------
   ##bamqc
      ##----------------------------------
for names in $(cat ${samplesss})
  do
  
  cp -r ${BaseLocation}/${names}/bwaMapping2DB//bamqc ${BaseLocation}/log/bamQC_${names}
  
  done
  mkdir -p ~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/bamQCs
```


```{bash}
scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_withPhageAssembly/log  ~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/bamQCs
```


```{bash}
###-----------------
##Illuminamapping rawReads to circlator genome
###-----------------

   
##------------------define variables------------------
#for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
#do

name=di_K2_6h
sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_withPhageAssembly/polishing
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/rmk202_bacteria_phage_mag_ref.fasta
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/di_K2_6h_nanopore_filtered.fastq.gz


for name in $(echo "RMK202 lyo202_96 di_K2_6h")
do

Overall_output_directory=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_withPhageAssembly/polishing/$name
sample=$name
num=1



for run in first second third fourth
   do 

name=run
##--------------create directories----------------
rm -r $Overall_output_directory/0${num}_${run}Freebayes/
mkdir -p $Overall_output_directory/0${num}_${run}Freebayes/{rawIlluminaMapped,freebayesOuput}
mkdir -p $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/bamqc

#mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fq

# gunzip -c /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R1_val_1.fq.gz > \
#  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fq/${sample}_R1_val_1.fq
# gunzip -c /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R2_val_2.fq.gz > \
#  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fq/${sample}_R2_val_2.fq


bwa index $reference_fasta

bwa mem -t ${threads}  $reference_fasta /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_1.fastq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_2.fastq |samtools sort -@8 -O BAM \
-o $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam - 
     

qualimap bamqc -bam $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam \
        -outdir $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/bamqc --java-mem-size=40G
    
samtools index $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam

###-----------------
##freebayesAfterCirclator
###-----------------
echo freebayes

    SECONDS=0

    freebayes -F 0.5 --min-coverage 4 -p 1 -f \
      $reference_fasta \
      $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam > \
      $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf
      
      
    echo "RUNTIME (h:min:s)= $(($SECONDS / 3600)):$((($SECONDS / 60) % 60)):$(($SECONDS % 60))"
    echo END = `date +"%Y-%m-%d %H:%M:%S"`

##-------   
##bcftools variant integration of Illumina data into Pacbio polished genome
##-------

    SECONDS=0
    echo "bgzip"
bgzip -c  $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf > \
       $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf.gz
      echo "tabix"
tabix -p vcf  $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf.gz
    echo "bcftools"
bcftools consensus -f $reference_fasta \
      -o $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${num}_${run}Freebayes.fasta \
       $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf.gz
      
    echo "RUNTIME (h:min:s)= $(($SECONDS / 3600)):$((($SECONDS / 60) % 60)):$(($SECONDS % 60))"
    echo END = `date +"%Y-%m-%d %H:%M:%S"`
  
  bwa index $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${num}_${run}Freebayes.fasta 
  
   ##-------------------new variable name for next polishing steps-----------
  reference_fasta=$Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${num}_${run}Freebayes.fasta  
  num=$((num+1)) 
done
   
done # samples


###===========================
###-----------------
##quast
###-----------------
###===========================  

#for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
#do


for name in $(echo "RMK202 lyo202_96 di_K2_6h")
do

Overall_output_directory=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_withPhageAssembly/polishing/$name
sample=$name
num=1



#sample=1
threads=37
Overall_output_directory=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_withPhageAssembly/polishing/$name
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/rmk202_bacteria_phage_mag_ref.fasta
#ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/di_K2_6h_nanopore_filtered.fastq.gz
OUPUT=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_withPhageAssembly/polishing/Quast/$name


echo "========================================="
echo $sample
echo "========================================="

echo $reference_fasta
echo "---------------------------"
     
    rm -r $Overall_output_directory/quastwithRaconONT/
    mkdir -p $Overall_output_directory/quastwithRaconONT
    quast.py \
      -t ${threads} -l "FlyeAssembly,firstONTracon,secondONTracon" \
      -R $Overall_output_directory/03_thirdFreebayes/freebayesOuput/3_thirdFreebayes.fasta \
      $reference_fasta \
      $Overall_output_directory/01_firstFreebayes/freebayesOuput/1_firstFreebayes.fasta \
  $Overall_output_directory/02_secondFreebayes/freebayesOuput/2_secondFreebayes.fasta \
      -o $OUPUT
 
 done
```
look at Quast 

```{bash}

###------------------------
##local
###------------------------
mkdir -p ~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/Quast
scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_withPhageAssembly/polishing/Quast/  ~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/Quast


```
extract polished contigs and merge with Bacterial contigs for Final Reference file
for Streptococcus_phage_1 and Streptococcus_phage_2 take lyo202_96
for Streptococcus_phage_3 and di_K2_6h
for Lactococcus_phage_1 take RMK202

```{bash}

###------------------------
##extract contigs
###------------------------
##------------------------- bacteria

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished_onlyBACTERIA_PLASMID.fasta > \
 /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta
##------------------------- Streptococcus_phage_1
name=lyo202_96
i=Streptococcus_phage_1

samtools faidx /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_withPhageAssembly/polishing/$name/03_thirdFreebayes/freebayesOuput/3_thirdFreebayes.fasta ${i} >>  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta

##------------------------- Streptococcus_phage_2

name=lyo202_96
i=Streptococcus_phage_2

samtools faidx /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_withPhageAssembly/polishing/$name/03_thirdFreebayes/freebayesOuput/3_thirdFreebayes.fasta ${i} >>  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta

##------------------------- Lactobacillus_phage_1

name=RMK202
i=Lactobacillus_phage_1

samtools faidx /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_withPhageAssembly/polishing/$name/03_thirdFreebayes/freebayesOuput/3_thirdFreebayes.fasta ${i} >>  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta

##------------------------- Lactobacillus_phage_2

name=di_K2_6h
i=Streptococcus_phage_3

samtools faidx /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_withPhageAssembly/polishing/$name/03_thirdFreebayes/freebayesOuput/3_thirdFreebayes.fasta ${i} |sed 's/Streptococcus_phage_3/Lactobacillus_phage_2/g'  >>  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta

```

## QC
### Quast 

Here, I do the analysis of the Quast output

```{r}

library(readr)
library(ggplot2)
library(gridExtra)

quast_K1 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC/quastwithRacon_FreebayesONT/report.tsv","\t", escape_double = FALSE, trim_ws = TRUE)

rownames(quast_K1) <- quast_K1$Assembly
# quast_K1 <- quast_K1[,-1]

quast_K1_analysis <- t(quast_K1)
View(quast_K1_analysis)
quast_K1_analysis <- as.data.frame(quast_K1_analysis[-1,])
quast_K1_analysis$nameSample <- rownames(quast_K1_analysis)

quast_K1_analysis$`# misassemblies` <- as.double(as.character(quast_K1_analysis$`# misassemblies`))
quast_K1_analysis$`Duplication ratio` <- as.double(as.character(quast_K1_analysis$`Duplication ratio`))
quast_K1_analysis$`# mismatches per 100 kbp` <- as.double(as.character(quast_K1_analysis$`# mismatches per 100 kbp`))
quast_K1_analysis$`# indels per 100 kbp` <- as.double(as.character(quast_K1_analysis$`# indels per 100 kbp`))

#table(quast_K1_analysis$nameSample)
#quast_K1_analysis$nameSample <- revalue(quast_K1_analysis$nameSample, c("S50"="mst1", "S72"="mst2", "Streptococcus_thermophilus_RMK202"="RMK202"))
quast_K1_analysis$nameSample_f = factor(quast_K1_analysis$nameSample, levels=c("FlyeAssembly","firstONTracon","secondONTracon","thirdONTracon","fourthONTracon","firstFreebayes","secondFreebayes","thirdFreebayes"))


##Misassemblies
p1 <- ggplot(quast_K1_analysis,aes(x=nameSample_f,y=`# misassemblies`,group=1))+geom_point(size=2)+geom_line()+
  theme_classic()+
  labs(x="",
       y="# Misassemblies")+
  theme(axis.text.x = element_blank())


##Misassemblies
p2 <- ggplot(quast_K1_analysis,aes(x=nameSample_f,y=`Duplication ratio`,group=1))+geom_point(size=2)+geom_line()+
  theme_classic()+
  labs(x="",
       y="Duplication ratio")+
  theme(axis.text.x =element_blank())


##Mismatches
p3 <- ggplot(quast_K1_analysis,aes(x=nameSample_f,y=`# mismatches per 100 kbp`,group=1))+geom_point(size=2)+geom_line()+
  theme_classic()+
  labs(x="",
       y="# Mismatches per 100 kbp")+
  theme(axis.text.x = element_text(angle = 75, hjust = 1))

##INdels
p4 <- ggplot(quast_K1_analysis,aes(x=nameSample_f,y=`# indels per 100 kbp`,group=1))+geom_point(size=2)+geom_line()+
  theme_classic()+
  labs(x="",
       y="# INDELS per 100 kbp")+
  theme(axis.text.x = element_text(angle = 75, hjust = 1))


grid.arrange(p1,p2,p3,p4, ncol=2, heights=c(1,2))

  svg("~/Desktop/presenations/my_presentations/20190708_groupmeeting/figures/polishing_K1.svg",width=6,height=5)
grid.arrange(p1,p2,p3,p4, ncol=2, heights=c(2,3))
 dev.off()


```


### Ideel

In the following we run [ideel](https://github.com/phiweger/ideel).
Ideel should be run after all polishing steps in order to see the changes after the single steps. 

1. In order to run Ideel we first need to download the (tremble protein database)[https://www.uniprot.org/downloads]. 
...and after clone from github I had to change the location of the tremble.database in the SnakeFile before getting it to work. 

2. change tremble.database location in script

3. change .fa to .fasta in script

Runs incredibly long (days!)

```{bash}

##download Database
mkdir -p /archiv/TREMBL_database/20190710
cd /archiv/TREMBL_database/20190710 
wget ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_trembl.fasta.gz

##make database
cd /archiv/TREMBL_database/20190710 

pigz -d /archiv/TREMBL_database/20190710/uniprot_trembl.fasta.gz

diamond makedb --in uniprot_trembl.fasta.gz  -d uniprot_trembl

##create ideal
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta 

mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/IDEEL
cd /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/IDEEL

git clone https://github.com/mw55309/ideel

cd ideel 
mkdir -p genomes
cd genomes



cp $Overall_output_directory/08_fourthFreebayes/freebayesOuput/8_fourthFreebayes.fasta .
cp $reference_fasta .
cp $Overall_output_directory/01_firstRacon_graphmap/assembly/firstRacon_NWC_2Assembly.fasta .
cp $Overall_output_directory/02_secondRacon_graphmap/assembly/secondRacon_NWC_2Assembly.fasta .
cp $Overall_output_directory/03_thirdRacon_graphmap/assembly/thirdRacon_NWC_2Assembly.fasta .
cp $Overall_output_directory/04_fourthRacon_graphmap/assembly/fourthRacon_NWC_2Assembly.fasta .
cp $Overall_output_directory/05_firstFreebayes/freebayesOuput/5_firstFreebayes.fasta .
cp $Overall_output_directory/06_secondFreebayes/freebayesOuput/6_secondFreebayes.fasta .
cp $Overall_output_directory/07_thirdFreebayes/freebayesOuput/7_thirdFreebayes.fasta .
cd ..

snakemake

###________________________move local

scp -r vincent@130.223.51.116:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/IDEEL/ideel/hists/ /home/vincent/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/

scp -r vincent@130.223.51.116:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/IDEEL/ideel/lengths/ /home/vincent/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/


```

```{r}

library(readr)
library(ggplot2)

assembly_01 <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/lengths/K1_ONT_assembly_circlor.data",  "\t", escape_double = FALSE, col_names = c("length","expected"), trim_ws = TRUE)
assembly_02 <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/lengths/firstRacon_NWC_2Assembly.data",  "\t", escape_double = FALSE, col_names = c("length","expected"), trim_ws = TRUE)
assembly_03 <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/lengths/secondRacon_NWC_2Assembly.data",  "\t", escape_double = FALSE, col_names = c("length","expected"), trim_ws = TRUE)
assembly_04 <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/lengths/thirdRacon_NWC_2Assembly.data",  "\t", escape_double = FALSE, col_names = c("length","expected"), trim_ws = TRUE)
assembly_05 <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/lengths/fourthRacon_NWC_2Assembly.data",  "\t", escape_double = FALSE, col_names = c("length","expected"), trim_ws = TRUE)
assembly_06 <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/lengths/_fifthFreebayes.data",  "\t", escape_double = FALSE, col_names = c("length","expected"), trim_ws = TRUE)
assembly_07 <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/lengths/_sixthFreebayes.data",  "\t", escape_double = FALSE, col_names = c("length","expected"), trim_ws = TRUE)
assembly_08 <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/lengths/_seventhFreebayes.data",  "\t", escape_double = FALSE, col_names = c("length","expected"), trim_ws = TRUE)
assembly_09 <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/lengths/_eightFreebayes.data",  "\t", escape_double = FALSE, col_names = c("length","expected"), trim_ws = TRUE)


pseudoPlotData <- data.frame(pseudo=c(100-(sum((assembly_01$length/assembly_01$expected)>0.9)/nrow(assembly_01)*100),
100-(sum((assembly_02$length/assembly_02$expected)>0.9)/nrow(assembly_02)*100),
100-(sum((assembly_03$length/assembly_03$expected)>0.9)/nrow(assembly_03)*100),
100-(sum((assembly_04$length/assembly_04$expected)>0.9)/nrow(assembly_04)*100),
100-(sum((assembly_05$length/assembly_05$expected)>0.9)/nrow(assembly_05)*100),
100-(sum((assembly_06$length/assembly_06$expected)>0.9)/nrow(assembly_06)*100),
100-(sum((assembly_07$length/assembly_07$expected)>0.9)/nrow(assembly_07)*100),
100-(sum((assembly_08$length/assembly_08$expected)>0.9)/nrow(assembly_08)*100),
100-(sum((assembly_09$length/assembly_09$expected)>0.9)/nrow(assembly_09)*100)),names=c("FlyeAssembly","firstONTracon","secondONTracon","thirdONTracon","fourthONTracon","firstFreebayes","secondFreebayes","thirdFreebayes","fourthFreebayes"))

pseudoPlotData$nameSample_f = factor(pseudoPlotData$names, levels=c("FlyeAssembly","firstONTracon","secondONTracon","thirdONTracon","fourthONTracon","firstFreebayes","secondFreebayes","thirdFreebayes","fourthFreebayes"))


p5 <- ggplot(pseudoPlotData,aes(x=nameSample_f,y=pseudo,group=1))+geom_point(size=2)+geom_line()+
  theme_classic()+
  ylim(10,100)+
  scale_y_continuous(breaks = c(12,25,50,75,96),labels=c(12,25,50,75,96))+
  labs(x="",
       y="Pseudogenes [%]")+
  theme(axis.text.x = element_text(angle = 75, hjust = 1))
p5


  svg("~/Desktop/presenations/my_presentations/20190708_groupmeeting/figures/polishing_K1_pseudo.svg",width=6,height=3)
p5
 dev.off()


```


### BUSCO

In order to see the assembly completness we use [BUSCO](https://busco.ezlab.org/) 
```{bash}


##-----------------------------------
##move faa files to the directory
##-----------------------------------

cp /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/vincent2/output/L_delbrueckii_RMK202/L_delbrueckii_RMK202_mag.faa /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/


cp /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/vincent2/output/S_thermophilus_RMK202/S_thermophilus_RMK202_mag.faa /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/faa/


##-----------------------------------
##script
##-----------------------------------
rm /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/all_analysis.txt

species=Sterm
species=Ldel
speciesss=Ldel
for speciesss in $(echo "Sterm Ldel")
do
echo "##------------------------"
echo ${speciesss}
echo "##------------------------"

for genemess in $(ls /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${speciesss}/faa/ )
  do
  #genemess=L_delbrueckii_RMK202_mag.faa
  genomesssss=$(echo ${genemess}|cut -d '.' -f 1)
  
  rm -r /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/${genemess}
  rm -r /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/${genomesssss}

  mkdir -p /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/${genomesssss}
  cd /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/${genomesssss}
  
  python /home/vincent/apps/busco/scripts/run_BUSCO.py -c 32 -i /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${speciesss}/faa/${genemess} -o ${genomesssss} -l /home/vincent/apps/busco/lactobacillales_odb9 -m prot

  #grep -v "^#" run_${genomesssss}/short_summary_${genomesssss}.txt| grep "C:" -A 7 
  all_INFO=$(grep -v "^#" run_${genomesssss}/short_summary_${genomesssss}.txt| grep "C:" | sed "s/^[ \t]*//"  )
  complet=$(echo ${all_INFO} | awk -F "[,[]" '{print $1}'|sed 's/%//g'|sed 's/C://g' )
  single=$(echo ${all_INFO} | awk -F "[,[]" '{print $2}'|sed 's/%//g'|sed 's/S://g')
  dupli=$(echo ${all_INFO} | awk -F "[,[]" '{print $3}'|sed 's/%]//g'|sed 's/D://g')
  fragm=$(echo ${all_INFO} | awk -F "[,[]" '{print $4}'|sed 's/%//g'|sed 's/F://g')
  Missi=$(echo ${all_INFO} | awk -F "[,[]" '{print $5}'|sed 's/%//g'|sed 's/M://g')
  numbi=$(echo ${all_INFO} | awk -F "[,[]" '{print $6}'|sed 's/n://g')
  
 echo -e ${speciesss}"\t"${genomesssss}"\t"${complet}"\t"${single}"\t"${dupli}"\t"${fragm}"\t"${Missi}"\t"${numbi}  \
 >> /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/all_analysis.txt

  done
done ## species

sed -i  's/Streptococcus_thermophilus_RMK202_pgap/S_thermophilus_mag_rmk202/g' /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/all_analysis.txt
sed -i  's/Lactobacillus_delbrueckii_RMK202_pgap/L_delbrueckii_mag_rmk202/g' /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/all_analysis.txt

```
### mapping

We also look at the raw read mapping. 

```{bash}

for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt)
do
threads=38

#names=di_K2_6h

  echo ${num}"/37  :" ${names}
  num=$((num+1))
  rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/

  mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/

  # bwa index /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/Final_Polished_withoutCow/di_rmk202_MAG_reference_polished.fasta
  
  bwa mem -t ${threads} /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/Final_Polished_withoutCow/di_rmk202_MAG_reference_polished.fasta /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_1.fastq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*kneaddata_paired_2.fastq | samtools sort -@${threads} -O BAM -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 

#cd /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/
#samstat /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam

samtools view -b -f 4 /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

bedtools bamtofastq -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam -fq /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq

rm /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

##-----Qualimap

mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/bamqc

qualimap bamqc -bam /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/bamqc --java-mem-size=80G



 mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/log
echo "/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/bamqc" >> /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/log/multiqc_mapping_logfile.txt
   

samtools index /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam &


done

###--------------------------------multiQC

rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL//MultiQC/
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL//MultiQC/


  multiqc --config /archiv/logs/multiQC_config.txt --file-list /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/log/multiqc_mapping_logfile.txt \
  -o archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL//MultiQC/

```


## Final annoation

The final annoation will be done with:
- PGAP for Bacteria, Plasmids and the cow mito
- phanotate for the phages
- preliminary prokka

### PGAP

make PGAP annotation by submitting in the [submission portal](https://submit.ncbi.nlm.nih.gov/).

```{bash}


##-------------------------
##prepare files for PGAP
##-------------------------
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/PGAP_prep/
##----------------------Sterm

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_sterm/final/S_thermophilus_mag_rmk202.fasta \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//splitContigs/S_thermophilus_repA.fasta > \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/PGAP_prep/S_thermophilus_RMK202.fasta
  
##----------------------Ldel

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_ldel/final/L_delbrueckii_mag_rmk202.fasta \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//splitContigs/L_del_plasmid_01.fasta \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/splitContigs/L_del_plasmid_02.fasta > \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/PGAP_prep/L_delbrueckii_RMK202.fasta
  
##------------------------------------BRING LOCAL


mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/assembly/  

scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/PGAP_prep/L_delbrueckii_RMK202.fasta /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/assembly/


mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/assembly/  

scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/PGAP_prep/S_thermophilus_RMK202.fasta /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/assembly/


```

getting files from PGAG NCBI submission portal and moving to bigmachine for further analysis. 





### Prokka

First we however do PROKKA for all the genomes indiviually.

```{bash}


###================
##description file
###================

Outputlocation=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/03_PROKKA_annotation_polishedGenome
Assembly_sterm=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_sterm/final/S_thermophilus_mag_rmk202.fasta
Assembly_ldel=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_ldel/final/L_delbrueckii_mag_rmk202.fasta
BaseFILNAME_ldel=L_delbrueckii_mag_rmk202
BaseFILNAME_sterm=S_thermophilus_mag_rmk202
locusTagSterm=Sterm_202
locusTagLdel=Ldel_202
Genus_Sterm=Streptococcus
Genus_Ldel=Lactobacillus
threads=37


###================
##script
###================



##===============
##annotaion
##===============


##====================================================================================-Streptococcus_thermophilus================

rm -r ${Outputlocation}/${BaseFILNAME_sterm}
    mkdir -p ${Outputlocation}/${BaseFILNAME_sterm}
    
prokka --outdir ${Outputlocation}/${BaseFILNAME_sterm}/ --force --usegenus --genus ${Genus_Sterm} --locustag ${locusTagSterm} --proteins proteins.faa --evalue 0.001 --addgenes ${Assembly_sterm}


#grep "dnaa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/03_PROKKA_annotation_polishedGenome/${names}/*.gff


##====================================================================================-Lactobacillus delbrueckii================

rm -r ${Outputlocation}/${BaseFILNAME_ldel}
    mkdir -p ${Outputlocation}/${BaseFILNAME_ldel}
    
prokka --outdir ${Outputlocation}/${BaseFILNAME_ldel}/ --force --usegenus --genus ${Genus_Ldel} --locustag ${locusTagLdel} --proteins proteins.faa --evalue 0.001 --addgenes ${Assembly_ldel}


#grep "dnaa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/03_PROKKA_annotation_polishedGenome/${names}/*.gff

```

looking into Prokka annoation

```{bash}

ll /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/

##Lactobacillus_delbrueckii_RMK202
gff_location=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_delbrueckii_RMK202/*gff

grep "dnaa" -i ${gff_location}
grep "pseudogene" -i ${gff_location}
grep "transposase" -i ${gff_location}
grep "16S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "23S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "5S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep " ribosomal RNA" -i ${gff_location} 


##Lactobacillus_delbrueckii_RMK202
gff_location=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Streptococcus_thermophilus_RMK202/*gff

grep "dnaa" -i ${gff_location}
grep "pseudogene" -i ${gff_location}
grep "transposase" -i ${gff_location}
grep "16S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "23S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "5S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep " ribosomal RNA" -i ${gff_location} 


##S72
gff_location=/archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/*gff

grep "dnaa" -i ${gff_location}
grep "pseudogene" -i ${gff_location}
grep "transposase" -i ${gff_location}
grep "16S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "23S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "5S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep " ribosomal RNA" -i ${gff_location} 


##S50
gff_location=/archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/*gff

grep "dnaa" -i ${gff_location}
grep "pseudogene" -i ${gff_location}
grep "transposon" -i ${gff_location}
grep "16S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "23S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "5S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep " ribosomal RNA" -i ${gff_location} 



#L104
gff_location=/archiv/Projects/2018_Culturomics/06_Spades/L104/Prokka/20190203/*gff

grep "dnaa" -i ${gff_location}
grep "pseudogene" -i ${gff_location}
grep "transposon" -i ${gff_location}
grep "16S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "23S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "5S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep " ribosomal RNA" -i ${gff_location} 


```



### eggnog annoation

All assemblies were annotated on the [eggnog homepage](http://eggnog-mapper.embl.de/)
After the annoation the files are downloaded locally and analysed in R.

genes of interest

Transporter: GO:0140104 (molecular carrier activity), GO:0005215 (transporter activity)

The different annotation options:
1. [KEGG](https://www.genome.jp/kegg/kegg1a.html) has many different annotations schems:
  - Probably the most interesting one is [KO (KEGG ORTHOLOGY) Database](https://www.genome.jp/kegg/ko.html).
  - [KEGG modules](https://www.genome.jp/kegg/module.html) less grained the Brite
  - Brite
  
  
Do the mapping of the annotations [here](https://www.genome.jp/kegg/tool/map_pathway.html)
  


```{bash}

##=================
##create mappable files
##=================

##Ldel  
awk -F "\t" 'BEGIN{OFS="\t"} {print $1, $9}' /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Lactobacillus_delbrueckii_RMK202_pgap_query_seqs.fa.emapper.annotations |sed 's/ko://g' |grep "^#" -v > /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog_Kegg/Lactobacillus_delbrueckii_RMK202_pgap_query_seqs.fa.emapper.annotations

grep -v "^#" /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Lactobacillus_delbrueckii_RMK202_pgap_query_seqs.fa.emapper.annotations | awk -F "\t" 'BEGIN{OFS="\t"} {print $9}'  |sed 's/ko://g' > /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog_Kegg/Lactobacillus_delbrueckii_RMK202_pgap_query_seqs.fa.emapper.annotations

```


I noticed that i first have to seperate the Brite columns with multiple annoations in order to get their annoation

```{r}
library(thacklr)
library(readr)
library(dplyr)
library(tidyverse)
library(tidyr)

S50 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/S50_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3)) %>% add_column(.,sample="S50")

Lactobacillus_delbrueckii_RMK202 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Lactobacillus_delbrueckii_RMK202_pgap_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3)) %>% add_column(.,sample="Lactobacillus_delbrueckii_RMK202")
colnames(S50) <- c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description","genome")


colnames(Lactobacillus_delbrueckii_RMK202) <- c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description","genome")


## The ";\\s+" means that the separator is a ";" followed by one or more spaces


Lactobacillus_delbrueckii_RMK202_subset4Brite <- Lactobacillus_delbrueckii_RMK202[,c("query_name","KEGG_ko")]
# Lactobacillus_delbrueckii_RMK202_subset4Brite$KEGG_ko <- gsub("ko:","",Lactobacillus_delbrueckii_RMK202_subset4Brite$KEGG_ko)
Lactobacillus_delbrueckii_RMK202_subset4Brite <- separate_rows(Lactobacillus_delbrueckii_RMK202_subset4Brite,KEGG_ko,sep=",")

write.table(Lactobacillus_delbrueckii_RMK202_subset4Brite, "~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog_Kegg/Lactobacillus_delbrueckii_RMK202_subset4Brite_pgap_query_seqs.fa.emapper.annotations",na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names = FALSE)

```

### Pandaroo

Here, we identify accessory genes. 

first I combine only the completely sequenced genomes (ONT)
```{bash}
##----------------------------------
#primer

base=/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all/
format=fna
combineOutput=/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_complete


##----------------------
##do

mkdir -p ${combineOutput}

cp ${base}/202-13494.${format}  ${combineOutput}
cp ${base}/202-13496.${format} ${combineOutput}
cp ${base}/202-13498.${format} ${combineOutput}
cp ${base}/202-13499.${format} ${combineOutput}
cp ${base}/202-24737.${format} ${combineOutput}
cp ${base}/202-24738.${format} ${combineOutput}
cp ${base}/202-24739.${format} ${combineOutput}
cp ${base}/202-24740.${format} ${combineOutput}
cp ${base}/202-24853.${format} ${combineOutput}
cp ${base}/202-24854.${format} ${combineOutput}
cp ${base}/202-24855.${format} ${combineOutput}
cp ${base}/202-SMAG.${format} ${combineOutput}


##----------------------
##only Bac contig
##----------------------

rm -r ${combineOutput}/onlyBac/
mkdir -p ${combineOutput}/onlyBac/
for genomesss in $(ls ${combineOutput}|grep ".fna$")
do
echo ${genomesss}
contig_name=$(seq_length.py ${combineOutput}/${genomesss} |head -1 |cut -f 1)
rm ${combineOutput}/${genomesss}.fai
samtools faidx ${combineOutput}/${genomesss} ${contig_name} > ${combineOutput}/onlyBac/${genomesss}

seq_length.py ${combineOutput}/onlyBac/${genomesss}
done

##----------------------
##Prokka
##----------------------
species=Streptococcus

for genomesss in $(ls ${combineOutput}|grep ".fna$"|sed 's/.fna//g')
do
rm -r ${combineOutput}/onlyBac/PROKKA/${genomesss}
mkdir -p  ${combineOutput}/onlyBac/PROKKA/${genomesss}

echo ${genomesss}


/usr/bin/perl /usr/bin/prokka --cpu 37 --outdir ${combineOutput}/onlyBac/PROKKA/${genomesss}/ --force --usegenus --genus ${species} --locustag ${genomesss} --proteins proteins.faa --evalue 0.001 --addgenes ${combineOutput}/onlyBac/${genomesss}.fna

done

###---------------------------------
##gather
###---------------------------------

rm -r ${combineOutput}/onlyBac/PROKKA/all_GFF/
mkdir -p  ${combineOutput}/onlyBac/PROKKA/all_GFF/

for genomesss in $(ls ${combineOutput}|grep ".fna$"|sed 's/.fna//g')
do
echo ${genomesss}

cat ${combineOutput}/onlyBac/PROKKA/${genomesss}/*PROKKA*.gff > ${combineOutput}/onlyBac/PROKKA/all_GFF/${genomesss}.gff

done


##==========================================
###-----------------panaroo
##==========================================

##-----sterm

mkdir -p ${combineOutput}/onlyBac/01_Panarooo/

panaroo -t 37 -i  ${combineOutput}/onlyBac/PROKKA/all_GFF//*.gff -o ${combineOutput}/onlyBac/01_Panarooo/ --clean-mode strict

##-------------------------------------
##post Panaroo analysis
##-------------------------------------

species=Sterm
cd ${combineOutput}/onlyBac/01_Panarooo/
#sudo python3 /home/vincent/apps/Roary/contrib/roary_plots/roary_plots.py --format eps /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/analysis/parsnp.tree_final gene_presence_absence.csv

sed '1d' gene_presence_absence_roary.csv | awk -F "\",\"" '{OFS=","}{print $5}'  |sort|uniq -c
#sudo chmod 777 ../analysis_1588600355
#sudo chmod 777 pan_genome_reference.fa

rm accesoryGenes.fasta

sed  's/ /-/g'  pan_genome_reference.fa >  pan_genome_reference_cleaned.fa
for genesss in $(sed '1d' gene_presence_absence_roary.csv  |awk -F "," '{OFS=","}{if($4<12)print $1}' |sed 's/"//g')
do
echo -e ${genesss}
longNamezzz=$(grep "${genesss}$" pan_genome_reference_cleaned.fa |sed 's/>//g')

samtools faidx pan_genome_reference_cleaned.fa ${longNamezzz} >>  accesoryGenes.fasta
done


grep ">" -c accesoryGenes.fasta

transeq  -sequence accesoryGenes.fasta > \
  -outseq  accesoryGenes.faa
  
  
###----------------------------

awk -F "," '{if($4<12)print $1}' gene_presence_absence_roary.csv > accessoryNames.txt


grep ">" accesoryGenes.faa |sed 's/>//g' > accesoryGenes.names

```


2. find core genomes
```{bash}
##==========================================
###-----------------panaroo
##==========================================

#conda create -n panaroo
conda activate panaroo
conda install -c conda-forge -c bioconda -c defaults panaroo
species=Ldel
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/01_Panarooo/${species}
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/01_Panarooo/${species}

panaroo -t 37 -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/GFF_all/*.gff -o  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/01_Panarooo/${species} --clean-mode strict


grep ">" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FNA_all/*.fna


cut -d ',' -f 4 /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/01_Panarooo/${species}/gene_presence_absence_roary.csv|sort|uniq -c
awk -F "," '{if($4=="17" && $5=="17") print $0}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/01_Panarooo/${species}/gene_presence_absence_roary.csv|wc -l
awk -F "," '{if($4<"16") print $0}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/01_Panarooo/${species}/gene_presence_absence_roary.csv|wc -l
grep ">" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/01_Panarooo/${species}/pan_genome_reference.fa
wc -l /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/01_Panarooo/${species}/gene_presence_absence_roary.csv
samtools faidx /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/01_Panarooo/${species}/pan_genome_reference.fa

awk -F "," '{if($4<"16") print $3}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/01_Panarooo/${species}/gene_presence_absence_roary.csv |sort|uniq -c



##==========================================
###-----------------panaroo
##==========================================

##-----sterm

species=Sterm

mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/01_Panarooo/${species}

panaroo -t 37 -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/GFF_all/*.gff -o  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/01_Panarooo/${species} --clean-mode strict


##==========================================
###-----------------analysis panaroo
##==========================================

ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/GFF/ |grep ".gff" -c

cut -d ',' -f 4 /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/gene_presence_absence_roary.csv|sort|uniq -c
awk -F "," '{if($4=="65" && $5=="65") print $0}' /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/gene_presence_absence_roary.csv|wc -l

##==========================================
###-----------------loop through single copy core genes and gather single copy core faa and ffn
##==========================================


roaryOUT=/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/gene_presence_absence_roary.csv

awk -F "," '{if($4=="65" && $5=="65") print $0}' /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/gene_presence_absence_roary.csv|wc -l

rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FAA/SCCG
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FAA/SCCG

rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FFN/SCCG
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FFN/SCCG

for SCCG in $(awk -F "," '{if($4=="65" && $5=="65") print $1}' ${roaryOUT})
do
echo "================================================="
echo ${SCCG}
colzz=$(head -1 ${roaryOUT}  |tr ',' '\n' |wc -l)
for genomesss in $(seq 15 1 $colzz)
do

genesss=$(grep "^${SCCG}" ${roaryOUT} |cut -d ',' -f ${genomesss})
echo ${genesss}
samtools faidx /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FAA/complete_Sterm_aminoAcid_gene.fasta ${genesss} >> /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FAA/SCCG/${SCCG}.faa

samtools faidx /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FFN/complete_Sterm_nucleotide_gene.fasta ${genesss} >> /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FFN/SCCG/${SCCG}.ffn

done #genomesss
done #SCCG


```
## CRISPR ANALYSIS

## Extract CRISPRs

Here, I used two different appraoches to extract the CRISPR spacers. 

### CRISPRcasFinder

[CRISPRcasFinder](https://crisprcas.i2bc.paris-saclay.fr/) is a recent and very useful tool to extract the CRISPRcas arrays of assembled genomes. 

```{bash}

#for species in $(echo "Ldel Sterm")
for species in $(echo "Sterm")

do
echo ==========================
     echo "working on: " ${species}
     echo ==========================
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}
for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/ |grep "_202_"|grep ".fasta$" |sed 's/.fasta//g')
do
      echo ==========================
     echo -e "extracting strain: " ${genomes}
     echo ==========================
     cd /home/vincent/apps/CRISPRCasFinder/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/${genomes}_cas
/usr/bin/perl ./CRISPRCasFinder.pl -in /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}//${genomes}.fasta \
  -cf -cas CasFinder-2.0.3 -def G -keep -html -log  -cpuM 37 -ccc \
  -out /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/${genomes}_cas/


  done #genomes
done #species

##-------------------------
#some analsis
##-------------------------
genomes=S_O_202_24853

for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/ |grep "_202_"|grep ".fasta$" |sed 's/.fasta//g')
do
echo "-------------------------------------------------------------------------------"
echo $genomes
echo "-----------"
grep "lead" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/${genomes}_cas//GFF/*.gff
#grep "^Un" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/${genomes}_cas/TSV/Crisprs_REPORT.tsv|awk -F "\t" -v vasrss="$genomes" '{OFS="\n"}{print vasrss"-"$9,$11}'

#grep "^Un" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/${genomes}_cas/TSV/Crisprs_REPORT.tsv|awk -F "\t" -v vasrss="$genomes" '{OFS="\t"}{print vasrss,$15}'

done

##-------------------------
#find orientation
##-------------------------
genomes=S_O_202_24853

rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes/ARRAY_orientation.txt
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes/
for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/ |grep "_202_"|grep ".fasta$" |sed 's/.fasta//g')
do
echo "-------------------------------------------------------------------------------"
echo $genomes
echo "-----------"
grep "^Un" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/${genomes}_cas/TSV/Crisprs_REPORT.tsv|awk -F "\t" -v vasrss="$genomes" '{OFS="\n"}{print vasrss"-"$9,$11}'

grep "^Un" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/${genomes}_cas/TSV/Crisprs_REPORT.tsv|awk -F "\t" -v vasrss="$genomes" '{OFS="\n"}{print ">"vasrss"-"$9,$11}' >>  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes/ARRAY_orientation.txt

done
###----add the reference repeats from chunk CRISPR-repeat
cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/cdHIT/CRISPR_Clustering.txt >>  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes/ARRAY_orientation.txt

###----cluster

/home/vincent/apps/cd-hit-v4.8.1-2019-0228/cd-hit-est -i  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes/ARRAY_orientation.txt \
    -o  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes/CRISPR_Clustering.txt -c 0.80
    
###----only use big clusters

grep -B 1000 ">Cluster 3"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes/CRISPR_Clustering.txt.clstr

###----loop through arrays
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes/orientation_CRISPRarrays.txt
for species in $(echo "Sterm")
do
echo ==========================
     echo "working on: " ${species}
     echo ==========================
species=Sterm
count=3

##----------prime
start_num=0
i=1
###-----------------
##extract information
###-----------------
for i in $(seq 1 $count)
  do 
  echo $i
  

  sed -n "/>Cluster $start_num$/,/>Cluster $i$/p" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes/CRISPR_Clustering.txt.clstr > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes//tmpCRISPRS.txt

ClusterName=$(head -1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}//tmpCRISPRS.txt |sed 's/>//g'|sed 's/ /_/g')

grep -v "^>Clu" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes//tmpCRISPRS.txt > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes//tmpCRISPRS_02.txt

numSpacers=$(wc -l /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes//tmpCRISPRS_02.txt|cut -d ' ' -f 1)

arrayNum=$(grep ">ARRAY" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes//tmpCRISPRS_02.txt |awk -F "[>...]" '{print $2}'|sed 's/ARRAY//g')

for spacersss in $(awk -F "[>...]" '{print $2}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes//tmpCRISPRS_02.txt|grep -v "^ARRAY")
do

sample=$(echo ${spacersss}| cut -d '-' -f 1|cut -d '_' -f 4)
orientation=$(echo ${spacersss}| cut -d '-' -f 2|head -c 3)


 echo -e ${sample}"\t"${orientation}"\tCRISPR array "${arrayNum}
 echo -e ${sample}"\t"${orientation}"\tCRISPR array "${arrayNum} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes/orientation_CRISPRarrays.txt


done #spacersss

start_num=$((start_num+1))
done #i

done #species

```



### PilerCR crispr 

Futerh I identify the CRISPR spacers with [PilerCR](https://www.drive5.com/pilercr/).

```{bash}

##=============
##pilerCR
##=============

for species in $(echo "Ldel Sterm")
do
echo ==========================
     echo "working on: " ${species}
     echo ==========================
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}
for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/ |grep "_202_"|grep ".fasta$" |sed 's/.fasta//g')
do
      echo ==========================
     echo -e "extracting strain: " ${genomes}
     echo ==========================
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${genomes}



  ~/apps/PilerCR/pilercr1.06/pilercr -in /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}//${genomes}.fasta \
    -out /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${genomes}/${genomes}.pilarCR_out -noinfo\
    -seq /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${genomes}/${genomes}.pilarCR_out.fasta

  
  perl ~/apps/PilerCR/PilerCR_parser_new_vincent /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${genomes}/${genomes}.pilarCR_out > \
    /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${genomes}/${genomes}.pilarCR_out_final
  done #genomes
  #  grep ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/${id}.pilarCR_out_final_new | sed 's/>//g' |awk -F "__" '{OFS="\t"} {print $4,$2,$3,$1}' > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/${id}.pilarCR_out_spacer.bed

done #species

##------------------------------------------------------------
##check directionality of CRISPRS
##------------------------------------------------------------

species=Sterm
for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/ |grep "_202_"|grep ".fasta$" |sed 's/.fasta//g')
do
echo "--------------------------------------------------------------------------------------"
echo  ${genomes}
grep "SUMMARY BY SIMILARITY" -A 15   /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${genomes}/${genomes}.pilarCR_out
done
```


## CRISPR structure

### unique CRISPR repeats

Here, I identify the number of different/unique CRISPR repeats.
Thereupon I relabel the genome arrays according to the common repeat number.
Then I calculate the repeat nucleotide identity the different CRISPR arrays. 
I plot this with Weblogo

Because

make a multiSeq alignment file of all CRISPR arrays. 
We have multiple CRISPR arrays in every species but they all come from the same

```{bash}

###------------
##change array names
###every strain has a different array labelling according to the CRISPR locations 
##here I want to make them all the same (because the CRISPRs are within one of three arrays)
###------------
for species in $(echo "Ldel Sterm")
do
echo "##=========================="
echo $species
echo "##=========================="
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}
#for genomess in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/ |grep ".fna$" |sed 's/.fna//g')
for genomess in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/ |grep "_202_"|grep ".fasta$" |sed 's/.fasta//g')
do

sed 's/\[Array/-a/' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${genomess}/${genomess}.pilarCR_out.fasta |sed 's/\;.*\]//' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/all_repeats.fasta


done #genomes


done # species

#======================================
##---------------------------cd -hit
#======================================

for species in $(echo "Ldel Sterm")
do
echo "##=========================="
echo $species
echo "##=========================="

/home/vincent/apps/cd-hit-v4.8.1-2019-0228/cd-hit-est -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/all_repeats.fasta \
    -o  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/CRISPR_Clustering.txt -c 0.80
    

done # species

##------------manaully assign the Clusters to the arrays

echo -e ">Cluster_0\ta3\n>Cluster_1\ta1\n>Cluster_2\ta2" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/Sterm/Arrays.log
echo -e ">Cluster_0\ta1\n>Cluster_1\ta2" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/Ldel/Arrays.log


#======================================
##---------------------------prepare multi-seq alignment file
#======================================


for species in $(echo "Ldel Sterm")
do
echo "##=========================="
echo $species
echo "##=========================="
 rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/cdHIT/
 mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/cdHIT/
for clustersss in $(cut -f 1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/Arrays.log)
do
aryName=$(grep "${clustersss}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/Arrays.log |cut -f 2)
ClusterNumber=$(echo $clustersss|awk -F "_" '{print $2}')
EndCluster=$((ClusterNumber+1))

sed -n "/>Cluster $ClusterNumber$/,/>Cluster $EndCluster$/p" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/CRISPR_Clustering.txt.clstr | grep -v "^>Cl" |awk -F "[>...]" '{print $2}' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/cdHIT//tmpCRISPRS_repeats.txt 

num=1
for samplesss in $(cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/cdHIT//tmpCRISPRS_repeats.txt)
do

grep "${samplesss}" -A 1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/all_repeats.fasta >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/cdHIT/${species}_${aryName}_CRISPRrepeats_all.fasta

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/cdHIT/${species}_${aryName}_CRISPRrepeats_all.fasta
#-----------------
done #samplesss
done # clustersss
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/cdHIT//tmpCRISPRS_repeats.txt
done # species

##--------------------
##clustalw
##creat weblogo of alignment
##--------------------
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/WebLogo
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/WebLogo
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/ClustalW
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/ClustalW
for species in $(echo "Ldel Sterm")
do

for aryName in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/cdHIT/ |grep ".fasta$" |cut -d '_' -f 2)
do

clustalw -INFILE=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/cdHIT/${species}_${aryName}_CRISPRrepeats_all.fasta -ALIGN -OUTFILE=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/ClustalW/${species}_${aryName}_CRISPRrepeats_all
/home/vincent/apps/weblogo/seqlogo -f /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/ClustalW/${species}_${aryName}_CRISPRrepeats_all -F EPS -k 1 -o /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/WebLogo/${species}_${aryName}_CRISPRrepeats

done #aryName
done # species
```


#####tracrRNA

Here, I identity the tracrRNA.

trans-activating crRNA (tracrRNA) is important to form a RNA duplexes by the base-pairing of tracrRNA with the pre-crRNA repeats in the presence of Cas9.

Identify the tracrRNA by looking for the anti CRISPR repeats on the genome

traccRNA prediction (http://www.tandfonline.com/doi/full/10.1080/15476286.2018.1498281)
https://www.ncbi.nlm.nih.gov/pubmed/24811454

this can be done with the motif search tool directly in IGV. 
Therefore we need to also search for the reverse complement
```{bash}
## tracrRNA
grep -i "tracrRNA"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/*.gff
grep -i "crRNA"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/*.gff
grep -i "crRNA"  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/S_thermophilus_RMK202.current.gb

###-------------
## reverse complement
###-------------
revseq /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Sterm_CRISPR_repeat.txt /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Sterm_CRISPR_repeat_reverse_complement.txt 


revseq -complement false /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Sterm_CRISPR_repeat.txt /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Sterm_CRISPR_repeat_reverse.txt 

```

As the the tracrRNA finder tools such as [traccRNA prediction](http://www.tandfonline.com/doi/full/10.1080/15476286.2018.1498281) did not work very well I will no try to identify them manually. 

IGV has a usueful motif finder function but it does not work with a single error from the sequence therefore I extract the CAS-gene casette and blast the repeat (and possibly also the rev-complemement) to this extracted regions. 

Therefore I first mask the CRISPRrepeat regions and then map the CRISPR repeats to the remaining genome to identify the tracrRNAs. 

```{bash}

###--------------------------------
##mask CRISPR repeat region for bwa mapping of repeat to identify anti CRISPRs (tracrRNAs)
###--------------------------------
grep "repeat" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/CRISPR_repeats_SMAG.bed


bedtools maskfasta -fi /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/Sterm/renamedContigs/StartAligned/StartAligned/202-SMAG.fna -bed /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/CRISPR_repeats_SMAG.bed -fo /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/masked_CRISPR_repeats_SMAG.fasta


###--------------------------------
#Blast repeats to CAS locations
###--------------------------------

makeblastdb -max_file_sz 1GB -in /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/masked_CRISPR_repeats_SMAG.fasta \
 -out /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/masked_CRISPR_repeats_SMAG -parse_seqids -dbtype nucl


  blastn -num_threads 35 -max_hsps 40 -max_target_seqs 40 -task megablast -show_gis \
  -query /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Sterm_CRISPR_repeat.txt \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen sgi sacc" \
  -db /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/masked_CRISPR_repeats_SMAG \
  -out /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/CAS_repeat_matches.blast \
  -evalue 0.01 -word_size 16


awk -F "\t" '{OFS="\t"}{print "CP046134",$9,$10}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/CAS_repeat_matches.blast > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/CAS_repeat_matches.bed


#awk -F "\t" '{OFS="\t"}{print $2,$9,$10}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/CAS_repeat_matches.blast > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/CAS_repeat_matches.bed
```

#### geneIdentity CAS
here I want to do a multi-seq alignment of all Cas-genes to identify the similarity. 

```{bash}


#for species in $(echo "Ldel Sterm"|tail -1)
for species in $(echo "Sterm")
do
echo ==========================
     echo "working on: " ${species}
     echo ==========================
#rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}//GeneID/{ary1,ary2,ary3}
#for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/|grep ".fna$" |sed 's/.fna//g')
for genomess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/${species}/ |grep "_202_"|grep ".fasta$" |sed 's/.fasta//g'|cut -d '_' -f 4)
do
genomes=$(echo -e "202-"${genomess})
      echo ==========================
     echo -e "extracting strain: " ${genomes}
     echo ==========================

##-------------------ARY1
grep "cas9" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}'| awk -F "\t" '{OFS="\t"}{if($7=="+") print $0}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary1/cas9.gff
grep "cas2" -B 6 -A 2 -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}'|grep "cas9" -A 4|grep "cas1"|awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}'| awk -F "\t" '{OFS="\t"}{if($7=="+") print $0}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary1/cas1.gff
grep "cas2" -B 6 -A 2 -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}'|grep "cas9" -A 4|grep "cas2"| awk -F "\t" '{OFS="\t"}{if($7=="+") print $0}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary1/cas2.gff
grep "cas2" -B 6 -A 2 -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}'|grep "cas9" -A 4|grep "csn2" |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}'| awk -F "\t" '{OFS="\t"}{if($7=="+") print $0}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary1/csn2.gff

##-------------------ARY3
grep "cas2" -A 6 -B 2 -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}'|grep "cas9" -B 3|grep "cas9"|awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}'| awk -F "\t" '{OFS="\t"}{if($7=="-") print $0}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary3/cas9.gff
grep "cas2" -A 6 -B 2 -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}'|grep "cas9" -B 3|grep "cas1" |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}'| awk -F "\t" '{OFS="\t"}{if($7=="-") print $0}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary3/cas1.gff
grep "cas2" -A 6 -B 2 -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}'|grep "cas9" -B 3|grep "cas2"|awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}'| awk -F "\t" '{OFS="\t"}{if($7=="-") print $0}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary3/cas2.gff

grep "cas2" -A 6 -B 2 -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}'|grep "cas9" -B 3|head -1 >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary3/csn2-like.gff

##-------------------ARY2

grep "csm6" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary2/csm6.gff
grep "csm5" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary2/csm5.gff
grep "csm4" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary2/csm4.gff
grep "csm3" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary2/csm3.gff
grep "cas10" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="CDS") print $0}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary2/cas10.gff
grep "cas6" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary2/cas6.gff
grep "cas6" -A 6 /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |grep "cas1" |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary2/cas1.gff
grep "cas6" -A 6 /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |grep "cas2"| awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary2/cas2.gff


cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.fna >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/AllGenomes.fna
  done #genomes
  #  grep ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/${id}.pilarCR_out_final_new | sed 's/>//g' |awk -F "__" '{OFS="\t"} {print $4,$2,$3,$1}' > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/${id}.pilarCR_out_spacer.bed

done #species

###------------get fasta

#for species in $(echo "Ldel Sterm"|tail -1)
for species in $(echo "Sterm")
do
echo ==========================
     echo "working on: " ${species}
     echo ==========================
#rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}
#rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}//GeneID/alignmentScore.txt

mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}//GeneID/{ary1,ary2,ary3}
#for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/|grep ".fna$" |sed 's/.fna//g')
#do
   

for arrayzzz in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}//GeneID/ |grep "^ary")
do

   echo ==========================
     echo -e "extracting arrayzzz: " ${arrayzzz}
     echo ==========================
for geness in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}//GeneID/${arrayzzz} |grep ".gff"|sed 's/.gff//g')
do


   echo ==========================
     echo -e "extracting geness: " ${geness}
     echo ==========================

#bedtools getfasta -fi /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/AllGenomes.fna \
#  -bed /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}//GeneID/${arrayzzz}/${geness}.gff \
#  -fo  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}//GeneID/${arrayzzz}/${geness}.fna

#muscle -clw -in /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}//GeneID/${arrayzzz}/${geness}.fna -out /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}//GeneID/${arrayzzz}/${geness}.msa


#clustalw -INFILE=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}//GeneID/${arrayzzz}/${geness}.fna -ALIGN -OUTFILE=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}//GeneID/${arrayzzz}/${geness}_clustalwAlignement | grep "Aligned. Score:" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}//GeneID/${arrayzzz}/${geness}_clustalwAlignement_Score.txt


sed 's/) Aligned. Score:  /\t/g' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}//GeneID/${arrayzzz}/${geness}_clustalwAlignement_Score.txt | sed 's/Sequences (//g' |awk -F "\t" -v namesss="$species" -v arraysss="$arrayzzz" -v genezzz="$geness" '{OFS="\t"}{print $1,$2,namesss,arraysss,genezzz}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}//GeneID/alignmentScore.txt


done #geness
done #arrayzzz
#  done #genomes

done #species
```
file of interest is the .dnd file which contains the alignment score.

```{bash}

mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/Sterm//GeneID/
scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/Sterm//GeneID/alignmentScore.txt /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/Sterm//GeneID/

```



```{r}
library(readr)
library(ggplot2)
library(tidyverse)
alignmentScore_casGenes <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/Sterm/GeneID/alignmentScore.txt", "\t", escape_double = FALSE, col_names = c("comparison","alignentScore","species","array","gene"),  trim_ws = TRUE)

ggplot(alignmentScore_casGenes,aes(x=gene,y=alignentScore,fill=gene))+geom_boxplot()+theme_classic()+facet_wrap(~array)+lims(y=c(80,100))


 d2 <- alignmentScore_casGenes %>%
  group_by(interaction(gene,array)) %>% 
  dplyr::summarise(median=median(alignentScore),mean=mean(alignentScore)) 
# d2 <- cbind(d2,d2,d2)

 d2$gene <- str_split_fixed(as.character(d2$`interaction(gene, array)`), ".ary", 3)[,1]
 d2$array <- str_split_fixed(as.character(d2$`interaction(gene, array)`), ".ary", 3)[,2]
   d2

##===============================================
##heatmap
##===============================================

ggheatmap_ANI <- ggplot(d2, aes(gene, array, fill = median))+
 geom_tile(color = "blue",size=1.1)+
 scale_fill_gradient2(low = "white", high = "black",
  midpoint = 96, limit = c(95,100), space = "Lab",
  name="alignmentscore") +
  scale_y_discrete(position = "right")+
  labs(legend="ANI",x="",y="")+
  # scale_fill_distiller(name = "SNPs", palette = "Blues", direction = -1)+
  # scale_fill_distiller(name = "ANI", palette = "Reds", direction = 1)+
  theme_minimal()+ # minimal theme
 theme(axis.text.y = element_text(size = 12),
      axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1))+
 coord_fixed()
# Print the heatmap
 svg("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/Sterm/GeneID/alignmentScore_heatmap.svg",width=8,height=4)

print(ggheatmap_ANI)

dev.off()

```



#### making the plot

making the CRISPR-cas region plot with genoplotR. 
First I need to prepare the data
important is to make three files for every ary one

```{bash}
##=============================
##extract the location for all necessary componenets of the three arrays
##=============================

##------------------
##prepare files for every ary
##------------------
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles

echo -e "name\tstart\tend\tstrand\tcol\tfill\tgene_type" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary1_genoPlotR.txt
echo -e "name\tstart\tend\tstrand\tcol\tfill\tgene_type" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary2_genoPlotR.txt
echo -e "name\tstart\tend\tstrand\tcol\tfill\tgene_type" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary3_genoPlotR.txt

##------------------
##tracrRNA
##------------------
#tracrRNA="red"

head -1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/CAS_repeat_matches.blast |awk -F "\t" '{OFS="\t"}{print "tracrRNA",$9,$10,"1","black","red","arrows"}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary1_genoPlotR.txt

tail -1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/CAS_repeat_matches.blast |awk -F "\t" '{OFS="\t"}{print "tracrRNA",$9,$10,"1","black","red","arrows"}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary3_genoPlotR.txt

##------------------
##ary 1,3
##------------------

##ary1
grep "cas9" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |head -1|awk -F "\t" '{OFS="\t"}{if($7=="+") print "cas9",$4,$5,"1","black","#440154FF","arrows"; else print "cas9",$4,$5,"-1","black","#440154FF","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary1_genoPlotR.txt
grep "cas1" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |head -1|awk -F "\t" '{OFS="\t"}{if($7=="+") print "cas1",$4,$5,"1","black","#39568CFF","arrows"; else print "cas1",$4,$5,"-1","black","#39568CFF","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary1_genoPlotR.txt
grep "cas2" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |head -1|awk -F "\t" '{OFS="\t"}{if($7=="+") print "cas2",$4,$5,"1","black","#1F968BFF","arrows"; else print "cas2",$4,$5,"-1","black","#1F968BFF","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary1_genoPlotR.txt
grep "csn2" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |head -1|awk -F "\t" '{OFS="\t"}{if($7=="+") print "csn2",$4,$5,"1","black","#55C667FF","arrows"; else print "csn2",$4,$5,"-1","black","#55C667FF","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary1_genoPlotR.txt

#csn2="#55C667FF"
##ary3
grep "cas9" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |tail -1|awk -F "\t" '{OFS="\t"}{if($7=="+") print "cas9",$4,$5,"1","black","#440154FF","arrows"; else print "cas9",$4,$5,"-1","black","#440154FF","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary3_genoPlotR.txt
grep "cas1" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |tail -1|awk -F "\t" '{OFS="\t"}{if($7=="+") print "cas1",$4,$5,"1","black","#39568CFF","arrows"; else print "cas1",$4,$5,"-1","black","#39568CFF","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary3_genoPlotR.txt
grep "cas2" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |tail -1|awk -F "\t" '{OFS="\t"}{if($7=="+") print "cas2",$4,$5,"1","black","#1F968BFF","arrows"; else print "cas2",$4,$5,"-1","black","#1F968BFF","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary3_genoPlotR.txt
grep "202-SMAG_01302_gene" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |awk -F "\t" '{OFS="\t"}{if($7=="+") print "Stu0660like Csn2",$4,$5,"1","black","grey","arrows"; else print "Stu0660like Csn2",$4,$5,"-1","black","grey","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary3_genoPlotR.txt

##------------------
##ary 2
##------------------
grep "csm6" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |awk -F "\t" '{OFS="\t"}{if($7=="+") print "csm6",$4,$5,"1","black","#6B4596FF","arrows"; else print "csm6",$4,$5,"-1","black","#6B4596FF","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary2_genoPlotR.txt

grep "csm5" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |awk -F "\t" '{OFS="\t"}{if($7=="+") print "csm5",$4,$5,"1","black","#90548bff","arrows"; else print "csm5",$4,$5,"-1","black","#90548bff","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary2_genoPlotR.txt

grep "csm4" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |awk -F "\t" '{OFS="\t"}{if($7=="+") print "csm4",$4,$5,"1","black","#b8627dff","arrows"; else print "csm4",$4,$5,"-1","black","#b8627dff","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary2_genoPlotR.txt

grep "csm3" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |awk -F "\t" '{OFS="\t"}{if($7=="+") print "csm3",$4,$5,"1","black","#de7065ff","arrows"; else print "csm3",$4,$5,"-1","black","#de7065ff","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary2_genoPlotR.txt

grep "csm2" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |awk -F "\t" '{OFS="\t"}{if($7=="+") print "csm2",$4,$5,"1","black","#f68f46ff","arrows"; else print "csm2",$4,$5,"-1","black","#f68f46ff","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary2_genoPlotR.txt

grep "cas10" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="CDS") print $0}' |awk -F "\t" '{OFS="\t"}{if($7=="+") print "cas10",$4,$5,"1","black","#f9a242ff","arrows"; else print "cas10",$4,$5,"-1","black","#f9a242ff","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary2_genoPlotR.txt

grep "cas6" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |awk -F "\t" '{OFS="\t"}{if($7=="+") print "cas6",$4,$5,"1","black","#f7cb44ff","arrows"; else print "cas6",$4,$5,"-1","black","#f7cb44ff","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary2_genoPlotR.txt


grep "cas1" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |grep "202-SMAG_01043_gene" |awk -F "\t" '{OFS="\t"}{if($7=="+") print "cas1",$4,$5,"1","black","#39568CFF","arrows"; else print "cas1",$4,$5,"-1","black","#39568CFF","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary2_genoPlotR.txt

grep "cas2" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |grep "202-SMAG_01042_gene" |awk -F "\t" '{OFS="\t"}{if($7=="+") print "cas2",$4,$5,"1","black","#1F968BFF","arrows"; else print "cas2",$4,$5,"-1","black","#1F968BFF","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary2_genoPlotR.txt

##------------------
##repeats
##------------------
##ary 1
sed -n "/^Array 1$/,/^Array 2$/p"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/Sterm/202-SMAG/202-SMAG.pilarCR_out | sed -n "/^==========/,/^==========/p" |grep "==" -v | sed 's/^ *//g' |sed 's/ \+ /\t/g'|awk -F "\t" '{OFS="\t"}{print "repeat",$1,$1+$2,"1","black","black","blocks"}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary1_genoPlotR.txt

##ary 2
sed -n "/^Array 2$/,/^Array 3$/p"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/Sterm/202-SMAG/202-SMAG.pilarCR_out | sed -n "/^==========/,/^==========/p" |grep "==" -v | sed 's/^ *//g' |sed 's/ \+ /\t/g'|awk -F "\t" '{OFS="\t"}{print "repeat",$1,$1+$2,"1","black","black","blocks"}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary2_genoPlotR.txt

##ary 3
sed -n "/^Array 3$/,/^Array 4$/p"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/Sterm/202-SMAG/202-SMAG.pilarCR_out | sed -n "/^==========/,/^==========/p" |grep "==" -v | sed 's/^ *//g' |sed 's/ \+ /\t/g'|awk -F "\t" '{OFS="\t"}{print "repeat",$1,$1+$2,"1","black","black","blocks"}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary3_genoPlotR.txt

sed -n "/^Array 4$/,/^SUMMARY BY SIMILARITY/p"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/Sterm/202-SMAG/202-SMAG.pilarCR_out | sed -n "/^==========/,/^==========/p" |grep "==" -v | sed 's/^ *//g' |sed 's/ \+ /\t/g'|awk -F "\t" '{OFS="\t"}{print "repeat",$1,$1+$2,"1","black","black","blocks"}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary3_genoPlotR.txt


##------------------
##spacers
##------------------

##ary 1
sed -n "/^Array 1$/,/^Array 2$/p"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/Sterm/202-SMAG/202-SMAG.pilarCR_out | sed -n "/^==========/,/^==========/p" |grep "==" -v | sed 's/^ *//g' |sed 's/ \+ /\t/g'|awk -F "\t" '{OFS="\t"}{if($4>10) print "spacer",$1+$2,$1+$2+$4,"1","black","green","exons"}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary1_genoPlotR.txt

##ary 2
sed -n "/^Array 2$/,/^Array 3$/p"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/Sterm/202-SMAG/202-SMAG.pilarCR_out | sed -n "/^==========/,/^==========/p" |grep "==" -v | sed 's/^ *//g' |sed 's/ \+ /\t/g'|awk -F "\t" '{OFS="\t"}{if($4>10) print "spacer",$1+$2,$1+$2+$4,"1","black","green","exons"}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary2_genoPlotR.txt


##ary 3
sed -n "/^Array 3$/,/^Array 4$/p"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/Sterm/202-SMAG/202-SMAG.pilarCR_out | sed -n "/^==========/,/^==========/p" |grep "==" -v | sed 's/^ *//g' |sed 's/ \+ /\t/g'|awk -F "\t" '{OFS="\t"}{if($4<40) print "spacer",$1+$2,$1+$2+$4,"1","black","green","exons"}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary3_genoPlotR.txt

sed -n "/^Array 4$/,/^SUMMARY BY SIMILARITY/p"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/Sterm/202-SMAG/202-SMAG.pilarCR_out | sed -n "/^==========/,/^==========/p" |grep "==" -v | sed 's/^ *//g' |sed 's/ \+ /\t/g'|awk -F "\t" '{OFS="\t"}{if($4<40) print "spacer",$1+$2,$1+$2+$4,"1","black","green","exons"}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary3_genoPlotR.txt

```

now  plot the regions

```{r}

##-------------------------
##ary1
##-------------------------
library(readr)
ary1_genoPlotR <- read_delim("02_data/ary1_genoPlotR.txt", "\t", escape_double = FALSE, trim_ws = TRUE)  %>% as.data.frame() #%>% filter(name!="spacer") #%>% select(-fill)
dna_seg1 <- dna_seg(ary1_genoPlotR)
dna_segs <- list(dna_seg1)
plot_gene_map(dna_segs=dna_segs)

##-------------------------
##ary3
##-------------------------
library(readr)
ary3_genoPlotR <- read_delim("02_data/ary3_genoPlotR.txt", "\t", escape_double = FALSE, trim_ws = TRUE)  %>% as.data.frame() #%>% filter(name!="spacer") #%>% select(-fill)
# ary1_genoPlotR$gene_type <- revalue(ary1_genoPlotR$gene_type, c("triangleGrob"="blocks"))
# ary1_genoPlotR$lty <- 0.5 #necessary depending on plotting size
# ary1_genoPlotR <- head(ary1_genoPlotR,n = 4)
dna_seg3 <- dna_seg(ary3_genoPlotR)
dna_segs3 <- list(dna_seg3)
plot_gene_map(dna_segs=dna_segs3)


##-------------------------
##ary2
##-------------------------
library(readr)
ary2_genoPlotR <- read_delim("02_data/ary2_genoPlotR.txt", "\t", escape_double = FALSE, trim_ws = TRUE)  %>% as.data.frame() #%>% filter(name!="spacer") #%>% select(-fill)
# ary1_genoPlotR$gene_type <- revalue(ary1_genoPlotR$gene_type, c("triangleGrob"="blocks"))
# ary1_genoPlotR$lty <- 0.5 #necessary depending on plotting size
# ary1_genoPlotR <- head(ary1_genoPlotR,n = 4)
dna_seg2 <- dna_seg(ary2_genoPlotR)
dna_segs2 <- list(dna_seg2)
plot_gene_map(dna_segs=dna_segs2)

##-------------------------
##merged
##-------------------------


dna_segs <- list(dna_seg1, dna_seg3, dna_seg2)

# annots <- lapply(dna_segs, function(x){mid <- middle(x)annot <- annotation(x1=mid, text=x$name, rot=30)

annots <- lapply(dna_segs, function(x){
  mid <- middle(x)
  annot <- annotation(x1=mid, text=x$name, rot=30)
idx <- which(!annot$text %in% c("repeat","spacer"))
# annot[idx[idx %% 4 == 0],]
annot[idx[idx ],]
})

plot_gene_map(dna_segs=dna_segs,annotations=annots)

##-------------------------
##plot
##-------------------------


    # png("~/Desktop/Projects/2019_RMK202_analysis/10_assembly/heatmap_all_Sterm_strains.png", width = 4000, height = 4000,res=300)
svg("03_results/all_spacersPlot.svg",width=6,height=6)
plot_gene_map(dna_segs=dna_segs,annotations=annots)

dev.off()
```

## CRISPR-repeats

The CRISPR repeats are conserved. Therefore in the assemblies we should have the same repeats occuring again and again. 


```{bash}
###================
##description file
###================
threads=37
BaseLocation=/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/04_withOlDsamples/repeats


#===============================
##merge the repeats all together
#===============================

for species in $(echo "Ldel Sterm")
do
echo ==========================
     echo "working on: " ${species}
     echo ==========================
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/allRepeatsAssembled.fasta
for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/ |grep "_202_"|grep ".fasta$" |sed 's/.fasta//g')
do
      echo ==========================
     echo -e "extracting repeats of strain: " ${genomes}
     echo ==========================

genomes_short=$(echo ${genomes}|sed 's/S_O_202_//g'|sed 's/S_I_202_//g'|sed 's/L_I_202_//g')

sed "s/.*\[Array/>_a/g" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${genomes}/${genomes}.pilarCR_out.fasta|sed 's/;.*$//g'|sed "s/>_/>${genomes_short}_/g" >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/allRepeatsAssembled.fasta


done #genomes

done #species


#===============================
##cd-hit the repeats
#===============================

for species in $(echo "Ldel Sterm")
do
echo ==========================
     echo "working on: " ${species}
     echo ==========================
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/cdHIT/
  mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/cdHIT/
/home/vincent/apps/cd-hit-v4.8.1-2019-0228/cd-hit-est -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/allRepeatsAssembled.fasta \
    -o  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/cdHIT/CRISPR_Clustering.txt -c 0.80
 
done #species

##!!!!!!!!!!!!!!!!!!!!!this step has to be done manually
##by looking in which Cluster the Reference (circular assembled) arrays lie

species=Sterm
genomes=202-SMAG
cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/cdHIT/CRISPR_Clustering.txt.clstr

mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/logs/
echo -e ">Cluster_0\ta3\n>Cluster_1\ta1\n>Cluster_2\ta2" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/logs/Arrays.log

species=Ldel
genomes=202-LMAG
cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/cdHIT/CRISPR_Clustering.txt.clstr

mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/logs/
echo -e ">Cluster_0\ta1\n>Cluster_1\ta2" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/logs/Arrays.log

#===============================
##assign the arrays of all strain to the reference array
#===============================

for species in $(echo "Ldel Sterm")
do
echo ==========================
     echo "working on: " ${species}
     echo ==========================
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final.txt
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final_WithouAccumulation.txt

for clustersss in $(cut -f 1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/logs/Arrays.log)
do
aryName=$(grep "${clustersss}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/logs/Arrays.log |cut -f 2)
ClusterNumber=$(echo $clustersss|awk -F "_" '{print $2}')
EndCluster=$((ClusterNumber+1))

sed -n "/>Cluster $ClusterNumber$/,/>Cluster $EndCluster$/p" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/cdHIT/CRISPR_Clustering.txt.clstr | grep -v "^>Cl" |awk -F "[>...]" '{print $2}' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt

#for samplesss in $(cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt)
num=1
for samplesss in $(cut -f 3 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt|sort|uniq|grep -v "sample")
do

echo -e ${num}
NumberCRISPRarrays=$(grep -c "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt)


if [ ${NumberCRISPRarrays} == "1" ]; then
              WhichCRISPRarrays=$(grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt|cut -d '_' -f 2|sed 's/a//g')
              grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,$5}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final.txt
              grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,$5}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final_WithouAccumulation.txt
            elif [[ ${NumberCRISPRarrays} == "0"  ]]
            then
            echo -e "No arrays in this Cluster "${aryName} "for "${samplesss} 
            elif [[ ${NumberCRISPRarrays} == "2"  ]]
            then
            echo -e "Two Arrays matcht for Cluster" ${aryName} "for "${samplesss}
            ##C----Array_ONe
            WhichCRISPRarrays=$(grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt|head -1|cut -d '_' -f 2|sed 's/a//g')
            grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,$5}' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats_Spacers.txt
            spacerCount=$(cut -f 5 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats_Spacers.txt |sort|tail -1)
            cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats_Spacers.txt >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final.txt
            cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats_Spacers.txt >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final_WithouAccumulation.txt
            ##C----Array_TWO
            WhichCRISPRarrays=$(grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt|tail -1|cut -d '_' -f 2|sed 's/a//g')
            grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Countss="${spacerCount}" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,Countss+$5}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final.txt
            grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Countss="${spacerCount}" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,$5}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final_WithouAccumulation.txt
            elif [[ ${NumberCRISPRarrays} == "3"  ]]
            then
            echo -e "Three Arrays matcht for Cluster" ${aryName} "for "${samplesss}
            ##C----Array_ONe
            WhichCRISPRarrays=$(grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt|head -1|cut -d '_' -f 2|sed 's/a//g')
            grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,$5}' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats_Spacers.txt
            spacerCount=$(cut -f 5 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats_Spacers.txt |sort|tail -1)
            cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats_Spacers.txt >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final.txt
            cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats_Spacers.txt >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final_WithouAccumulation.txt
            ##C----Array_TWO
            WhichCRISPRarrays=$(grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt|head -2|tail -1|cut -d '_' -f 2|sed 's/a//g')
            grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Countss="${spacerCount}" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,Countss+$5}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final.txt
            grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Countss="${spacerCount}" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,$5}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final_WithouAccumulation.txt
            ##C----Array_THREE
            WhichCRISPRarrays=$(grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt|head -3|tail -1|cut -d '_' -f 2|sed 's/a//g')
            grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Countss="${spacerCount}" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,Countss+$5}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final.txt
            grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Countss="${spacerCount}" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,$5}' >> /archiv/Projects/2019_RMK202_an
            else
               echo "more than three arrays-----Currently nothing is done with these"
               
               
            fi
num=$((num+1))
done #samplesss

done #clusterssss

done #species


```


## CRISPR spacers

#### merge spacers

Here, I want to bring all annotated CRISPR spacers into one file.


```{bash}

#===============================
##merge the spacers all together
#===============================

for species in $(echo "Ldel Sterm")
do

#specsss=$(echo ${species}|head -c 1)
echo ==========================
     echo "working on: " ${species}
     echo ==========================
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/CRISPR_spacer_merged_short.fasta
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/
for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/ |grep "_202_"|grep ".fasta$" |sed 's/.fasta//g')
#for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned| grep ".fna$" |sed 's/.fna//g')
do
      echo ==========================
     echo -e "extracting strain: " ${genomes}
     echo ==========================

 cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${genomes}/${genomes}.pilarCR_out_final >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/CRISPR_spacer_merged.fasta


genomes_short=$(echo ${genomes}|sed 's/S_O_202_//g'|sed 's/S_I_202_//g'|sed 's/L_I_202_//g')
sed 's/__.*$//g' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${genomes}/${genomes}.pilarCR_out_final | sed "s/>/>${genomes_short}_/g"| sed 's/ary0/a/g' |sed 's/spc0/s/g' |sed 's/202-//g' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/CRISPR_spacer_merged_short.fasta
done #genomes

grep ">" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}//CRISPR_spacer_merged_short.fasta
grep ">"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}//CRISPR_spacer_merged_short.fasta| awk -F "[_]" '{OFS="-"}{print $1}'|sort|uniq -c
done #species

```
This will be used later in the CRISPR spacer section

### pairwise alignement

Here, I want to check for pairwise CRISPR diversity. 
We do this to see how many and which spacers are shared. 
First we align the spacers witha needle -munsch algorithm. 

This is currently not a crucial analysis

```{bash}
##====================================
##split  every CRIPSR into indiviual files
##====================================

for species in $(echo "Ldel Sterm")
do

samtools faidx /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}//CRISPR_spacer_merged_short.fasta

rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/seperated
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/seperated
for spacer in $(grep "^>" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}//CRISPR_spacer_merged_short.fasta|sed 's/>//g')
  do

samtools faidx /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}//CRISPR_spacer_merged_short.fasta ${spacer} > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/seperated/${spacer}.fasta

  done
  
done #species
##-------------
##pairwise needle-wunsch alignment
##-------------

##-------------do parallel
###------define TASK


task(){ 
    spacer2_final=$(echo ${spacer2} )
    needle -asequence /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/seperated/${spacer1_final}.fasta \
      -bsequence /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/seperated/${spacer2_final}.fasta -gapopen 20 -gapextend 0 -outfile /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/Pairwise_needle/${spacer1_final}__${spacer2_final}.msa

      length=$(grep "Length" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/Pairwise_needle/${spacer1_final}__${spacer2_final}.msa | cut -d ':' -f 2 | sed -e 's/^[ \t]*//' )
      Identity=$(grep "Identity" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/Pairwise_needle/${spacer1_final}__${spacer2_final}.msa | cut -d ':' -f 2 | cut -d '/' -f 1| sed -e 's/^[ \t]*//' )
      Gaps=$(grep "Gaps" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/Pairwise_needle/${spacer1_final}__${spacer2_final}.msa | cut -d ':' -f 2 | cut -d '/' -f 1| sed -e 's/^[ \t]*//' )
      Score=$(grep "Score" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/Pairwise_needle/${spacer1_final}__${spacer2_final}.msa | cut -d ':' -f 2 | sed -e 's/^[ \t]*//' )
      
     echo -e ${spacer1_final}"\t"${spacer2_final}"\t"${length}"\t"${Identity}"\t"${Gaps}"\t"${Score} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/Pairwise_needle/final_pairwise_alignment.txt
     
    rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/Pairwise_needle/${spacer1_final}__${spacer2_final}.msa

}

###------Run parallel in batches

for species in $(echo "Sterm Ldel")
do

rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/Pairwise_needle
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/Pairwise_needle
#counts=$(wc -l /data/Project/CRISPRspacers/02_PilerCR/20200217/all/pilerCR_out_repeats_short|cut -d ' ' -f 1)
N=300
for spacer1 in $(grep "^>" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}//CRISPR_spacer_merged_short.fasta|sed 's/>//g')
  do
  
  spacer1_final=$(echo ${spacer1})

  for spacer2 in $(grep "^>" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}//CRISPR_spacer_merged_short.fasta|sed 's/>//g')
    do
    
      spacer2_final=$(echo ${spacer2})

    ((i=i%N)); ((i++==0)) && wait
    ##--------------run task
    task &
    
    done #spacer2
    
  done #spacer1
  
  done #species

```


```{r}
##======================
##Sterm
##======================

spacer_pairs <- read_delim("02_data/Sterm_final_pairwise_alignment_new.txt",  "\t", escape_double = FALSE, col_names = c("spacer1","spacer2","length","matching","gaps","score"),trim_ws = TRUE)

spacer_pairs$id <- 100*(spacer_pairs$matching/spacer_pairs$length)

##----------subset
100*(sum(spacer_pairs$id>90)/nrow(spacer_pairs))
100*(sum(spacer_pairs$id>90)/nrow(spacer_pairs))

##----------histogramm

hist(spacer_pairs$id)

spacer_pairs_similiar <- spacer_pairs[(spacer_pairs$id>80),]
hist(spacer_pairs_similiar$id)


##---------------export the shared spacers
 
unique_spacers <- unique(c(spacer_pairs_similiar$spacer1,spacer_pairs_similiar$spacer2))

write.table(unique_spacers, "02_data/unique_Sterm_spacers_new.txt",na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names = FALSE)

pairwise_spacer_Matrix <- read_delim("02_data/pairwise_spacer_Matrix_sterm.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)

rownames(pairwise_spacer_Matrix) <- colnames(pairwise_spacer_Matrix)

##-----------heatmap 2
library(stringr)
library(rafalib)
library(RColorBrewer)
names_genome <- str_split_fixed(names, "_", 2)[,1]
cols <- palette(brewer.pal(3, "Dark2"))[as.fumeric(names_genome)]


library(gplots) ##Available from CRAN
pairwise_spacer_Matrix_ready <- as.matrix(pairwise_spacer_Matrix)
image <- heatmap.2(pairwise_spacer_Matrix_ready,
          trace="none",
          # Rowv = "none",
          Colv=NA,
          ColSideColors=cols,
          # density.info = "none") ##historgram in col scale
          tracecol="black",
          )

##-----------plot histogramm

svg("03_results/Streptococcus_thermophilus_SPACERS_histo.svg",width=5,height=3)

hist(spacer_pairs$id)
          
 dev.off()
 
 
```

Based on this analysis (especially the histogramm), we see that we have a very distinct biomodial distribution. By setting a clustering cutoff of 80% we should be able to distinguish common CRISPR spacers and newly aquired

### make CRISPR clusters plot

now we try to see if there are differences in spacers between the CRISPR spacers of all genomes. 
This is the old way, I used cd-hit-est to see how many clusters I have within a certain similiartity threshold


```{bash}
date=20190312

##==================
##cd-hit
##==================
  

for species in $(echo "Ldel Sterm")
do
echo ==========================
     echo "working on: " ${species}
     echo ==========================


  
  rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers
  mkdir -p  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}

  
   cd-hit-est -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/CRISPR_spacer_merged_short.fasta \
    -o /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results.txt -c 0.9 -M 30000 -T 37
  
done #species
  

  ###======================
  ##assign representaive Spacers to Cluster
  ###======================
  
  for species in $(echo "Ldel Sterm")
do
echo ==========================
     echo "working on: " ${species}
     echo ==========================

   cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results.txt > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results_ClusterName.txt
   
   for spacersss in $(grep ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results.txt)
   do
   echo "--------------------------"

   newNAME=$(grep -B 30 "${spacersss}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results.txt.clstr |grep "^>Cluster" |tail -1 |sed 's/ /_/g')
   
   sed -i "s/${spacersss}/${newNAME}/g" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results_ClusterName.txt
   
   done
   
   rm archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}//tmp_spacer.txt
   
done #species
```

### CRISPR spacers prep

Here, I want ot make an alluvial plot to trace the CRISPR spacers through the samples.
I can directly do this instead of of the last approach
```{bash}

for species in $(echo "Ldel Sterm")
do
echo ==========================
     echo "working on: " ${species}
     echo ==========================
species=Sterm
count=$(grep ">Cluster" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results.txt.clstr)

##----------prime
start_num=0
i=1
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}
 mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}
 
#  grep ">" /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_new_withOLDsamples_newNAMED_short |sed 's/>//g' | cut -d '_' -f 1 |sort| uniq
#rm /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledSpacers/Alluvialplot/
   #echo -e "Clustername\tMAGRMK202\tmst1\tmst2\tF13491\tF13492\tF13493\tF13494\tF13495\tF13496\tF13497\tF13498\tF13499c1\tF13499c2\tF13500" > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledSpacers/CD-hit_new/count//clusterCRISPRs.txt

 echo -e "ClusterName\tnumSpacers\tsample\tarray\tspc" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_Sterm.txt

###-----------------
##extract information
###-----------------
for i in $(seq 1 $count)
  do 
  echo $i
  

  sed -n "/>Cluster $start_num$/,/>Cluster $i$/p" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results.txt.clstr > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}//tmpCRISPRS.txt

ClusterName=$(head -1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}//tmpCRISPRS.txt |sed 's/>//g'|sed 's/ /_/g')

grep -v "^>Clu" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}//tmpCRISPRS.txt > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}//tmpCRISPRS_02.txt

numSpacers=$(wc -l /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}//tmpCRISPRS_02.txt|cut -d ' ' -f 1)

for spacersss in $(awk -F "[>...]" '{print $2}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}//tmpCRISPRS_02.txt)
do

sample=$(echo ${spacersss}| cut -d '_' -f 1)
array=$(echo ${spacersss}| cut -d '_' -f 2|sed 's/a//g' |sed 's/s/_/g' |cut -d '_' -f 1)
spc=$(echo ${spacersss}| cut -d '_' -f 2|sed 's/a//g' |sed 's/s/_/g' |cut -d '_' -f 2)

 echo -e ${ClusterName}"\t"${numSpacers}"\t"${sample}"\t"${array}"\t"${spc} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt


done #spacersss

start_num=$((start_num+1))
done #i

done #species

```


### Annotate shared CRISPRs 

Here, we count how many and which spacers are in every Cluster from all different samples
If a new sample (whole genome sequence) is added it needs to be added to the list...
This is used for the heatmap of the CRISPR spacer identity.

```{bash}

species=sTERM
count=$(grep ">Cluster" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results.txt.clstr)

grep ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results.txt |cut -d _ -f 1 |sed 's/>//g'|sort|uniq -c|wc -l

##----------prime
start_num=0
i=1
 mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count/
 
  echo -e "Clustername\tSMAG\tS50\tS72\t13491\t13492\t13493\t13494\t13495\t13496\t13497\t13498\t13499c1\t13499c2\t13500\t13499\t24737\t24738\t24739\t24740\t24853\t24854\t24855" >  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//clusterCRISPRs.txt

  
##---------start analysis 
for i in $(seq 1 $count)
  do 
  echo $i
#  done
#start_num=0
#i=169
  sed -n "/>Cluster $start_num$/,/>Cluster $i$/p" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results.txt.clstr > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt

REF_mst1=$(grep ">S50" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt)  
REF_mst2=$(grep ">S72" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt)  
REF_RMK202=$(grep ">SMAG" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
  
F13491=$(grep ">13491" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F13492=$(grep ">13492" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F13493=$(grep ">13493" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F13494=$(grep ">13494" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F13495=$(grep ">13495" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F13496=$(grep ">13496" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F13497=$(grep ">13497" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F13498=$(grep ">13498" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F13499c1=$(grep ">13499c1" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F13499c2=$(grep ">13499c2" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F13500=$(grep ">13500" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 

F13499_old=$(grep ">13499_" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24737=$(grep ">24737" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24738=$(grep ">24738" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24740=$(grep ">24740" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24739=$(grep ">24739" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24853=$(grep ">24853" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24854=$(grep ">24854" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24855=$(grep ">24855" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 

 echo -e "Cluster_"${start_num}"\t"${REF_RMK202}"\t"${REF_mst1}"\t"${REF_mst2}"\t"${F13491}"\t"${F13492}"\t"${F13493}"\t"${F13494}"\t"${F13495}"\t"${F13496}"\t"${F13497}"\t"${F13498}"\t"${F13499c1}"\t"${F13499c2}"\t"${F13500}"\t"${F13499_old}"\t"${F24737}"\t"${F24738}"\t"${F24739}"\t"${F24740}"\t"${F24853}"\t"${F24854}"\t"${F24855} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count///clusterCRISPRs.txt

  
  start_num=$((start_num+1))
  done
```

LDEL-------------------------------

```{bash}

species=Ldel
count=$(grep ">Cluster" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results.txt.clstr)

grep ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results.txt |cut -d _ -f 1 |sed 's/>//g'|sort|uniq -c|wc -l

##----------prime
start_num=0
i=1
 mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count/
 
  #grep ">" /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_new_withOLDsamples_newNAMED_short |sed 's/>//g' | cut -d '_' -f 1 |sort| uniq

 #  echo -e "Clustername\tMAGRMK202\tmst1\tmst2\tF13491\tF13492\tF13493\tF13494\tF13495\tF13496\tF13497\tF13498\tF13499c1\tF13499c2\tF13500" > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledSpacers/CD-hit_new/count//clusterCRISPRs.txt
# echo -e "Clustername\tSMAG\tS50\tS72\tF13491\tF13492\tF13493\tF13494\tF13495\tF13496\tF13497\tF13498\tF13499c1\tF13499c2\tF13500\t13499\t24737\t24738\t24739\t24740\t24853\t24854\t24855" >  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//clusterCRISPRs.txt
  echo -e "Clustername\tLMAG\t11141\t11142\t11143\t12104\t12105\t12107\t12109\t24776\t24777\t13498\t13499c1\t13499c2\t13500\t13499\t24737\t24738\t24739\t24740\t24853\t24854\t24855" >  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//clusterCRISPRs.txt

  
##---------start analysis 
for i in $(seq 1 $count)
  do 
  echo $i
#  done
#start_num=0
#i=169
  sed -n "/>Cluster $start_num$/,/>Cluster $i$/p" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results.txt.clstr > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt

REF_RMK202=$(grep ">LMAG" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
REF_11141=$(grep ">11141" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt)  
F11142=$(grep ">11142" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F11143=$(grep ">11143" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F12104=$(grep ">12104" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F12105=$(grep ">12105" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F12107=$(grep ">12107" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F12109=$(grep ">12109" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24776=$(grep ">24776" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24777=$(grep ">24777" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24778=$(grep ">24778" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 

F13499c2=$(grep ">13499c2" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F13500=$(grep ">13500" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F13499_old=$(grep ">13499_" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24737=$(grep ">24737" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24738=$(grep ">24738" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24740=$(grep ">24740" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24739=$(grep ">24739" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24853=$(grep ">24853" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24854=$(grep ">24854" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24855=$(grep ">24855" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 

 echo -e "Cluster_"${start_num}"\t"${REF_RMK202}"\t"${REF_mst1}"\t"${REF_mst2}"\t"${F13491}"\t"${F13492}"\t"${F13493}"\t"${F13494}"\t"${F13495}"\t"${F13496}"\t"${F13497}"\t"${F13498}"\t"${F13499c1}"\t"${F13499c2}"\t"${F13500}"\t"${F13499_old}"\t"${F24737}"\t"${F24738}"\t"${F24739}"\t"${F24740}"\t"${F24853}"\t"${F24854}"\t"${F24855} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count///clusterCRISPRs.txt

  
  start_num=$((start_num+1))
  done
```


Now I look at how many SPACERS ARE SHARED amongst strains

```{r}
clusterCRISPRs <- read_delim("02_data/clusterCRISPRs_sterm.txt", "\t", escape_double = FALSE, trim_ws = TRUE)


pairwiseSharedSpacers <- data.frame()

for (rowss in colnames(clusterCRISPRs)[-1]) {
  for (Colss in colnames(clusterCRISPRs)[-1]) {
  
    tmp <- clusterCRISPRs[,c(rowss,Colss)]
    SpacersTotal <- as.numeric(sum(rowSums(tmp>0) >0))
    SpacersShared <- as.numeric(sum(rowSums(tmp>0) >1))
 
    
    ID <- (SpacersShared/ SpacersTotal)*100
    

    tmp_05 <- data.frame(Ref1=rowss,Ref2=Colss,totalSpacer=SpacersTotal,sharedSpacers=SpacersShared,PercentShared=round(ID,digits=2))
  
  pairwiseSharedSpacers <- rbind(pairwiseSharedSpacers,tmp_05)
    
}
}

write.table(pairwiseSharedSpacers, "02_data/clusterCRISPRs_curated_sterm.txt",na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names = TRUE)
```


```{r}
##----------------------
#Ldel
##----------------------

library(readr)
clusterCRISPRs <- read_delim("02_data/clusterCRISPRs_ldel.txt", "\t", escape_double = FALSE, trim_ws = TRUE)

# rowss="mst1" 
# Colss="F13493"

pairwiseSharedSpacers <- data.frame()

for (rowss in colnames(clusterCRISPRs)[-1]) {
  for (Colss in colnames(clusterCRISPRs)[-1]) {
  
    tmp <- clusterCRISPRs[,c(rowss,Colss)]
    SpacersTotal <- as.numeric(sum(rowSums(tmp>0) >0))
    SpacersShared <- as.numeric(sum(rowSums(tmp>0) >1))
    
    # tmp_02  <- tmp[which(rowSums(tmp>0)==1),]
    # Ref1_spacersize <- sum(tmp_02[,1]>0&tmp_02[,2]<1)
    # Ref2_spacersize <- sum(tmp_02[,2]>0&tmp_02[,1]<1)
   
    
    ID <- (SpacersShared/ SpacersTotal)*100
    
    # tmp_05 <- data.frame(Ref1=rowss,Ref2=Colss,totalSpacer=SpacersTotal,sharedSpacers=SpacersShared,UniqueREF1=Ref1_spacersize,UniqueREF2=Ref2_spacersize,PercentShared=round(ID,digits=2))
    
    tmp_05 <- data.frame(Ref1=rowss,Ref2=Colss,totalSpacer=SpacersTotal,sharedSpacers=SpacersShared,PercentShared=round(ID,digits=2))
  
  pairwiseSharedSpacers <- rbind(pairwiseSharedSpacers,tmp_05)
    
}
}

write.table(pairwiseSharedSpacers, "02_data/clusterCRISPRs_curated_ldel.txt",na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names = TRUE)

```

### MAKE HEATMAPS
here I combine the fastANI heatmap and the CRISPR spacer heatmap to make one big heatmap
```{r}


##--------------------------
 #Sterm
##--------------------------
 
library(readr)
library(plyr)
library(tidyverse)
pairwiseSharedSpacers <- read_delim("02_data/clusterCRISPRs_curated_sterm.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)
final_ANI <- read_delim("02_data/fastaANI_comparision_curated_sterm.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)

unique(final_ANI$GenomeA)
unique(pairwiseSharedSpacers$Ref1)

unique(final_ANI$GenomeA) %in% unique(pairwiseSharedSpacers$Ref1)
unique(pairwiseSharedSpacers$Ref1)%in%unique(final_ANI$GenomeA)


##===================================================
##dENDROGRAMM
#this is to order them according to the phylogeny
##=================================================== 
##------------
##make matrix
##------------

names<- unique(final_ANI$GenomeA)

heatmap_prep_matrix <- matrix(NA,nrow=length(names),ncol = length(names))
rownames(heatmap_prep_matrix) <- names
colnames(heatmap_prep_matrix) <- names
# heatmap_prep_matrix <- heatmap_prep_matrix[!duplicated(heatmap_prep_matrix)]

final_ANI_02 <- final_ANI[!duplicated(final_ANI[,]), ]

for (rowsss in names){
  for (colsss in names){
  
heatmap_prep_matrix[rowsss,colsss]  <-  final_ANI_02[which(final_ANI_02$GenomeA==rowsss & final_ANI_02$GenomeB==colsss),"ANI"] %>% as.numeric
  
  
  }  
}

##------------
##make dendro of all snps sites
##------------
library(ggdendro)

otter.dendro <- as.dendrogram(hclust(d = dist(x = heatmap_prep_matrix)))

orderSterm <- labels(otter.dendro)

# Create dendro
dendro.plot <- ggdendrogram(data = otter.dendro, rotate = FALSE)
# Preview the plot
print(dendro.plot)

##===================================================
##FASTani heatmap
#this is the upper heatmap
##=================================================== 
final_ANI$GenomeA = factor(final_ANI$GenomeA, levels=orderSterm)
final_ANI$GenomeB = factor(final_ANI$GenomeB, levels=orderSterm)
levels(final_ANI$GenomeA)
levels(final_ANI$GenomeB)

  final_ANI$GenomeA = factor(final_ANI$GenomeA, levels=c("13492","13491","24854","24737","13500","13493","13498","SMAG","24853","13494","24855","S72","13499c1","13499","13499c2","24738","24740","S50", "13497", "24739","13495","13496"))
final_ANI$GenomeB = factor(final_ANI$GenomeB, levels=c("13492","13491","24854","24737","13500","13493","13498","SMAG","24853","13494","24855","S72","13499c1","13499","13499c2","24738","24740","S50", "13497", "24739","13495","13496"))


##3-----------------
#try to ouptu only top or bottom
##3-----------------


final_ANI_02 <- final_ANI
numberColumnsss <- 1
for (rowsss in levels(final_ANI_02$GenomeA)[-1]) {
  
  for (numberColumnsss_02 in 1:numberColumnsss) {
    
    Columnss <- levels(final_ANI_02$GenomeA)[numberColumnsss_02]
    
    final_ANI_02 <- final_ANI_02 %>% filter(!(GenomeA==rowsss&GenomeB==Columnss))
    
  }
  
  numberColumnsss=numberColumnsss+1
  
}

# densityANI <- ggplot(final_ANI_02,aes(x=ANI))+geom_density()+theme_classic()+lims(x=c(95,100))

ggheatmap_ANI <- ggplot(final_ANI_02, aes(GenomeA, GenomeB, fill = ANI))+
 geom_tile(color = "white",size=1.1)+
 scale_fill_gradient2(low = "grey", high = "red", midpoint=99,limit = c(99,100), space = "Lab",
  name="ANI") +
  scale_y_discrete(position = "right")+
  labs(legend="ANI",x="",y="")+
  # scale_fill_distiller(name = "SNPs", palette = "Blues", direction = -1)+
  # scale_fill_distiller(name = "ANI", palette = "Reds", direction = 1)+
  theme_minimal()+ # minimal theme
 theme(legend.position = "top",axis.text.y = element_text(size = 12),
      axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1))+
 coord_fixed()
# Print the heatmap
print(ggheatmap_ANI)


final_ANI_03 <- final_ANI_02 %>% filter(GenomeA!=GenomeB)

densityANI <- ggplot(final_ANI_03,aes(x=ANI))+geom_density(color="#EB4D4D",size=1.5)+theme_classic()+lims(x=c(99,100))
densityANI

##===================================================
##CRISPR ID
#this is the LOWER heatmap
##=================================================== 


library(viridis)

pairwiseSharedSpacers_tmp_02 <-  data.frame(Ref1=c(as.character(pairwiseSharedSpacers$Ref2),as.character(pairwiseSharedSpacers$Ref1)),Ref2=c(as.character(pairwiseSharedSpacers$Ref1),as.character(pairwiseSharedSpacers$Ref2)),PercentShared=c(as.numeric(pairwiseSharedSpacers$PercentShared),as.numeric(pairwiseSharedSpacers$PercentShared)))

pairwiseSharedSpacers_tmp_02$Ref1 = factor(pairwiseSharedSpacers_tmp_02$Ref1, levels=orderSterm)
pairwiseSharedSpacers_tmp_02$Ref2 = factor(pairwiseSharedSpacers_tmp_02$Ref2, levels=orderSterm)
levels(final_ANI$GenomeA)
levels(final_ANI$GenomeB)

pairwiseSharedSpacers_tmp_02$Ref2 = factor(pairwiseSharedSpacers_tmp_02$Ref2, levels=c("13492","13491","24854","24737","13500","13493","13498","SMAG","24853","13494","24855","S72","13499c1","13499","13499c2","24738","24740","S50", "13497", "24739","13495","13496"))
pairwiseSharedSpacers_tmp_02$Ref1 = factor(pairwiseSharedSpacers_tmp_02$Ref1, levels=c("13492","13491","24854","24737","13500","13493","13498","SMAG","24853","13494","24855","S72","13499c1","13499","13499c2","24738","24740","S50", "13497", "24739","13495","13496"))


##3-----------------
#try to ouptu only top or bottom
##3-----------------


pairwiseSharedSpacers_tmp_03 <- pairwiseSharedSpacers_tmp_02
numberColumnsss <- 1
for (rowsss in levels(pairwiseSharedSpacers_tmp_02$Ref1)[-1]) {
  
  for (numberColumnsss_02 in 1:numberColumnsss) {
    
    Columnss <- levels(pairwiseSharedSpacers_tmp_02$Ref1)[numberColumnsss_02]
    
    pairwiseSharedSpacers_tmp_03 <- pairwiseSharedSpacers_tmp_03 %>% filter(!(Ref1==rowsss&Ref2==Columnss))
    
  }
  
  numberColumnsss=numberColumnsss+1
  
}

ggheatmap_CRISPR <- ggplot(pairwiseSharedSpacers_tmp_03, aes(Ref2, Ref1, fill = PercentShared))+
 geom_tile(color = "white",size=1.1)+
  # theme_classic()+
 scale_fill_gradient2(low = "grey", high = "blue",
  midpoint = 25, limit = c(0,100), space = "Lab",
  name="shared CRISPR") +
    labs(legend="shared CRISPR",x="",y="")+
  theme_minimal()+ # minimal theme
 theme(legend.position = "top",axis.text.y = element_text(size = 12),
      axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1))+
 coord_fixed()
# Print the heatmap
print(ggheatmap_CRISPR)

pairwiseSharedSpacers_tmp_04 <- pairwiseSharedSpacers_tmp_03 %>% filter(Ref1!=Ref2)

densityCRISPR <- ggplot(pairwiseSharedSpacers_tmp_04,aes(x=PercentShared))+geom_density(color="#EB4D4D",size=1.5)+theme_classic()+lims(x=c(0,100))
densityCRISPR



library(patchwork)

##===================================================
##plot
##=================================================== 

  svg("03_results/HEATMAP_ani_crispr_density.svg",width=20,height=12)

(densityANI+densityCRISPR) / (ggheatmap_ANI+ggheatmap_CRISPR)

 dev.off()

 
  svg("03_results/HEATMAP_ani_crispr.svg",width=15,height=7.5)

ggheatmap_ANI+ggheatmap_CRISPR+dendro.plot

 dev.off()

 
  svg("03_results/HEATMAP_ani_crispr_woLegend.svg",width=9,height=7.5)

(ggheatmap_ANI+theme(legend.position = "none"))+(ggheatmap_CRISPR+theme(legend.position = "none"))

 dev.off()
 
 ##-----------------------
 #bring together
 ##-----------------------
all_together_sterm <-  merge(pairwiseSharedSpacers_tmp_03,final_ANI_02,by.x=c("Ref1","Ref2"),by.y=c("GenomeA","GenomeB"))%>% add_column(species="S.thermophilus")

 ggplot(all_together_sterm,aes(x=PercentShared,y=coverage_ANI))+geom_point()+theme_classic()
 
```


```{r}
##--------------------------
 #Ldel
##--------------------------
 
library(readr)
library(plyr)
library(tidyverse)
pairwiseSharedSpacers <- read_delim("02_data/clusterCRISPRs_curated_ldel.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)
final_ANI <- read_delim("02_data/fastaANI_comparision_curated_ldel.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)

unique(final_ANI$GenomeA)
unique(pairwiseSharedSpacers$Ref1)

unique(final_ANI$GenomeA) %in% unique(pairwiseSharedSpacers$Ref1)
unique(pairwiseSharedSpacers$Ref1)%in%unique(final_ANI$GenomeA)


##===================================================
##dENDROGRAMM
#this is to order them according to the phylogeny
##=================================================== 
##------------
##make matrix
##------------

names<- unique(final_ANI$GenomeA)

heatmap_prep_matrix <- matrix(NA,nrow=length(names),ncol = length(names))
rownames(heatmap_prep_matrix) <- names
colnames(heatmap_prep_matrix) <- names
# heatmap_prep_matrix <- heatmap_prep_matrix[!duplicated(heatmap_prep_matrix)]

final_ANI_02 <- final_ANI[!duplicated(final_ANI[,]), ]

for (rowsss in names){
  for (colsss in names){
  
heatmap_prep_matrix[rowsss,colsss]  <-  final_ANI_02[which(final_ANI_02$GenomeA==rowsss & final_ANI_02$GenomeB==colsss),"ANI"] %>% as.numeric
  
  
  }  
}

##------------
##make dendro of all snps sites
##------------
library(ggdendro)

otter.dendro <- as.dendrogram(hclust(d = dist(x = heatmap_prep_matrix)))

orderSterm <- labels(otter.dendro)

# Create dendro
dendro.plot <- ggdendrogram(data = otter.dendro, rotate = FALSE)
# Preview the plot
print(dendro.plot)

##===================================================
##FASTani heatmap
#this is the upper heatmap
##=================================================== 
final_ANI$GenomeA = factor(final_ANI$GenomeA, levels=orderSterm)
final_ANI$GenomeB = factor(final_ANI$GenomeB, levels=orderSterm)
levels(final_ANI$GenomeA)
levels(final_ANI$GenomeB)
##3-----------------
#try to ouptu only top or bottom
##3-----------------


final_ANI_02 <- final_ANI
numberColumnsss <- 1
for (rowsss in levels(final_ANI_02$GenomeA)[-1]) {
  
  for (numberColumnsss_02 in 1:numberColumnsss) {
    
    Columnss <- levels(final_ANI_02$GenomeA)[numberColumnsss_02]
    
    final_ANI_02 <- final_ANI_02 %>% filter(!(GenomeA==rowsss&GenomeB==Columnss))
    
  }
  
  numberColumnsss=numberColumnsss+1
  
}

 final_ANI_02 <- final_ANI_02 %>% filter(GenomeA!="11142") %>% filter(GenomeB!="11142")

ggheatmap_ANI <- ggplot(final_ANI_02, aes(GenomeA, GenomeB, fill = ANI))+
 geom_tile(color = "white",size=1.1)+
 scale_fill_gradient2(low = "grey", high = "red",
  midpoint = 99, limit = c(99,100), space = "Lab",
  name="ANI") +
  scale_y_discrete(position = "right")+
  labs(legend="ANI",x="",y="")+
  theme_minimal()+ # minimal theme
 theme(legend.position = "top",axis.text.y = element_text(size = 12),
      axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1))+
 coord_fixed()
# Print the heatmap
print(ggheatmap_ANI)



final_ANI_03 <- final_ANI_02 %>% filter(GenomeA!=GenomeB)

densityANI <- ggplot(final_ANI_03,aes(x=ANI))+geom_density(color="#10B552",size=1.5)+theme_classic()+lims(x=c(99,100))
densityANI


##===================================================
##CRISPR ID
#this is the LOWER heatmap
##=================================================== 


library(viridis)

pairwiseSharedSpacers_tmp_02 <-  data.frame(Ref1=c(as.character(pairwiseSharedSpacers$Ref2),as.character(pairwiseSharedSpacers$Ref1)),Ref2=c(as.character(pairwiseSharedSpacers$Ref1),as.character(pairwiseSharedSpacers$Ref2)),PercentShared=c(as.numeric(pairwiseSharedSpacers$PercentShared),as.numeric(pairwiseSharedSpacers$PercentShared)))

pairwiseSharedSpacers_tmp_02$Ref1 = factor(pairwiseSharedSpacers_tmp_02$Ref1, levels=orderSterm)
pairwiseSharedSpacers_tmp_02$Ref2 = factor(pairwiseSharedSpacers_tmp_02$Ref2, levels=orderSterm)
levels(final_ANI$GenomeA)
levels(final_ANI$GenomeB)


##3-----------------
#try to ouptu only top or bottom
##3-----------------


pairwiseSharedSpacers_tmp_03 <- pairwiseSharedSpacers_tmp_02
numberColumnsss <- 1
for (rowsss in levels(pairwiseSharedSpacers_tmp_02$Ref1)[-1]) {
  
  for (numberColumnsss_02 in 1:numberColumnsss) {
    
    Columnss <- levels(pairwiseSharedSpacers_tmp_02$Ref1)[numberColumnsss_02]
    
    pairwiseSharedSpacers_tmp_03 <- pairwiseSharedSpacers_tmp_03 %>% filter(!(Ref1==rowsss&Ref2==Columnss))
    
  }
  
  numberColumnsss=numberColumnsss+1
  
}
##3-----------------
#plot
##3-----------------
 ###the strain 11142 is strange because it was divergent from the others and no additional SNV was found for it in the metagenome. 
 #we assume there was a mistake in the strain collection
 
 # all_together_all$Ref1
 pairwiseSharedSpacers_tmp_03 <- pairwiseSharedSpacers_tmp_03 %>% filter(Ref1!="11142")%>% filter(Ref2!="11142")
 
ggheatmap_CRISPR <- ggplot(pairwiseSharedSpacers_tmp_03, aes(Ref2, Ref1, fill = PercentShared))+
 geom_tile(color = "white",size=1.1)+
  # theme_classic()+
 scale_fill_gradient2(low = "grey", high = "blue",
   limit = c(0,100), space = "Lab",
  name="shared CRISPR") +
  labs(legend="shared CRISPR",x="",y="")+
  theme_minimal()+ # minimal theme
 theme(legend.position = "top",axis.text.y = element_text(size = 12),
      axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1))+
 coord_fixed()
# Print the heatmap
print(ggheatmap_CRISPR)


pairwiseSharedSpacers_tmp_04 <- pairwiseSharedSpacers_tmp_03 %>% filter(Ref1!=Ref2)

densityCRISPR <- ggplot(pairwiseSharedSpacers_tmp_04,aes(x=PercentShared))+geom_density(color="#10B552",size=1.5)+theme_classic()+lims(x=c(0,100))
densityCRISPR



##===================================================
##plot
##=================================================== 

  svg("03_results/HEATMAP_ani_crispr_density_ldel.svg",width=20,height=12)

(densityANI+densityCRISPR) / (ggheatmap_ANI+ggheatmap_CRISPR)

 dev.off()



 
  svg("03_results/HEATMAP_ani_crispr_ldel.svg",width=15,height=7.5)

ggheatmap_ANI+ggheatmap_CRISPR+dendro.plot

 dev.off()

 
  svg("03_resultsHEATMAP_ani_crispr_woLegend_ldel.svg",width=9,height=7.5)

(ggheatmap_ANI+theme(legend.position = "none"))+(ggheatmap_CRISPR+theme(legend.position = "none"))

 dev.off()
 ##-----------------------
 #bring together
 ##-----------------------
all_together_ldel <-  merge(pairwiseSharedSpacers_tmp_03,final_ANI_02,by.x=c("Ref1","Ref2"),by.y=c("GenomeA","GenomeB")) %>% add_column(species="L.delbrueckii")

 ggplot(all_together_ldel,aes(x=PercentShared,y=coverage_ANI))+geom_point()+theme_classic()
 
 colorsss <- c("#fec3b7ff","#0081a7")
 # colorsss_01 <- c("#0081a7ff","#fec3b7ff")
all_together_all <- rbind(all_together_sterm,all_together_ldel)
plots_correlation <- ggplot(all_together_all,aes(x=PercentShared,y=ANI,color=species,fill=species))+geom_point(alpha=0.6)+theme_classic()+scale_color_manual(values=colorsss)+scale_fill_manual(values=colorsss)+theme(legend.position="none")+labs(x="shared CRISPR","ANI")+lims(y=c(99,100))
# all_together_all$ANI
  plots_correlation
  
  all_together_all <- rbind(all_together_sterm,all_together_ldel)
plots_correlation <- ggplot(all_together_all,aes(x=ANI,y=PercentShared,color=species,fill=species))+geom_point(alpha=0.6)+theme_classic()+scale_color_manual(values=colorsss)+scale_fill_manual(values=colorsss)+theme(legend.position="none")+labs(x="shared CRISPR","ANI")+lims(x=c(99,100))
# all_together_all$ANI
  plots_correlation
  
    svg("03_results/correaltion_all.svg",width=3,height=3)

plots_correlation

 dev.off()
 
 ###the strain 11142 is strange because it was divergent from the others and no additional SNV was found for it in the metagenome. 
 #we assume there was a mistake in the strain collection
 
 # all_together_all$Ref1
 all_together_all <- all_together_all %>% filter(Ref1!="11142")%>% filter(Ref2!="11142")
 
  plots_correlation <- ggplot(all_together_all,aes(x=PercentShared,y=ANI,color=species,fill=species))+geom_point(alpha=0.6)+theme_classic()+scale_color_manual(values=colorsss)+scale_fill_manual(values=colorsss)+theme(legend.position="none")+labs(x="shared CRISPR","ANI")+lims(y=c(99,100))
# all_together_all$ANI
  plots_correlation
  
    svg("03_results/correaltion_all.svg",width=3,height=3)

plots_correlation

 dev.off()
  
```


### Reoccuring Spacers

Here, we check which spacers are common. Then we check who shares them (same genotype) and where they occur (topology)

```{r}
clusterCRISPRs <- read_delim("02_data/clusterCRISPRs_sterm.txt", "\t", escape_double = FALSE, trim_ws = TRUE)
length(names) #from previous chunk
occurence <- data.frame(ClusterName=clusterCRISPRs$Clustername,occurrencecounts=rowSums(clusterCRISPRs[,-1]))
# occurence <- data.frame(ClusterName=clusterCRISPRs$Clustername,occurrencecounts=rowSums(clusterCRISPRs[,-1]>0))

table(occurence$occurrencecounts)
table(occurence$occurrencecounts)[-1] %>% sum()
table(occurence$occurrencecounts) %>% sum()
occurence <- occurence %>% add_column(species="S.thermophilus")
plotspacerCOUNT <- ggplot(occurence,aes(x=occurrencecounts))+geom_bar()+theme_classic()+labs(x="number of genomes containing the same spacer",y="spacer count")+scale_x_continuous(breaks =c(1:length(names)),labels=c(1:length(names)),limits = c(0,length(names)))+scale_y_continuous(breaks =c(0,7,31,60,154),labels=c(0,7,31,60,154))
plotspacerCOUNT
svg("03_results/spacerUniquenessCount.svg",width=4.5,height=2.5)
plotspacerCOUNT
 dev.off()
```

### Spacers analysis

Here, I do various spacer analysis with the data I gathered in the previous chunks.

```{r}

spacer_Infos_Sterm_final <- read_delim("02_data/spacer_Infos_Sterm_final.txt",  "\t", escape_double = FALSE, col_names = c("ClusterName","numSpacers_in_CLUSTER","sample","array","spc","ARRAY","SPACER"),  col_types = cols(SPACER = col_number()),  trim_ws = TRUE)

##------------------------------
##clean
##------------------------------
spacer_Infos_Sterm_final$spacerUniqueness  <- ifelse(spacer_Infos_Sterm_final$numSpacers_in_CLUSTER<2,"unique","shared")
table(spacer_Infos_Sterm_final$spacerUniqueness)
spacer_Infos_Sterm_final$ARRAY <- revalue(spacer_Infos_Sterm_final$ARRAY, c("a2"="CRISPR array 2","a1"="CRISPR array 1","a3"="CRISPR array 3"))


# spacer_Infos_Sterm_final$sample = factor(spacer_Infos_Sterm_final$sample, levels=orderSterm)
##------------------------------
###How many spacers in each strain
##------------------------------
spacerCount1 <- ggplot(spacer_Infos_Sterm_final,aes(x=sample,group=spacerUniqueness,fill=spacerUniqueness))+geom_bar()+theme_classic()+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9))+labs(x="",y="spacer count",fill="Uniqueness")

spacerCount2 <- ggplot(spacer_Infos_Sterm_final,aes(x=sample,group=spacerUniqueness,fill=spacerUniqueness))+geom_bar()+theme_classic()+facet_wrap(~ARRAY)+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),legend.position = "none")+labs(x="",y="spacer count",fill="Uniqueness")

# svg("~/Desktop/Projects/2019_RMK202_analysis/plot/CRISPR_spacer_count.svg",width=5.5,height=5.5)
((spacerCount1+plot_spacer() +plot_layout(widths = c(2,1.5)))/spacerCount2+plot_layout(nrow = 2))
 # dev.off()

table(spacer_Infos_Sterm_final$ARRAY,spacer_Infos_Sterm_final$sample)
range(table(spacer_Infos_Sterm_final$sample))

spacer_Infos_Sterm_final$sample = factor(spacer_Infos_Sterm_final$sample, levels=c("13492","13491","24854","24737","13500","13493","13498","SMAG","24853","13494","24855","S72","13499c1","13499","13499c2","24738","24740","S50", "13497", "24739","13495","13496"))
colors <- c("#fd8d3cff","#807dbaff","#ef3b2cff")

spacerCount3 <- ggplot(spacer_Infos_Sterm_final,aes(x=sample,group=ARRAY,fill=ARRAY))+geom_bar()+theme_classic()+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9))+labs(x="",y="spacer count",fill="")+scale_fill_manual(values=colors)
spacerCount3
##------------------------------
###spacer location
##------------------------------ 
 

max(uniqueSpacersss$SPACER)

uniqueSpacersss <- spacer_Infos_Sterm_final[which(spacer_Infos_Sterm_final$numSpacers_in_CLUSTER==1),]
range(spacer_Infos_Sterm_final$SPACER)
pfirst <- ggplot(uniqueSpacersss,aes(x=SPACER,group=ARRAY,fill=ARRAY))+geom_bar()+theme_classic()+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7))+scale_fill_manual(values=colorsss)+labs(x="spacer location",y="number of unique spacers",fill="Uniqueness")+scale_x_continuous(breaks =c(1:max(uniqueSpacersss$SPACER)),labels=c(1:max(uniqueSpacersss$SPACER)))

plocation <- (pfirst+plot_spacer()+plot_layout(widths = c(2,0.5)))


# svg("~/Desktop/Projects/2019_RMK202_analysis/plot/CRISPR_spacer_count_location.svg",width=6.5,height=8.5)
((spacerCount1+plot_spacer() +plot_layout(widths = c(2,1.5)))/spacerCount2+plocation+plot_layout(nrow = 3))
 # dev.off()

 
##----------------------------------------------
#number of spacers
##----------------------------------------------

spacer_Infos_Sterm_final_sub <- spacer_Infos_Sterm_final %>% select(ClusterName,ARRAY) 
spacer_Infos_Sterm_final_sub_uniques <- spacer_Infos_Sterm_final_sub[!duplicated(spacer_Infos_Sterm_final_sub[,]), ]
namesSingleArrays <- which(table(spacer_Infos_Sterm_final_sub_uniques$ClusterName)==1) %>% names()
tmp1 <- spacer_Infos_Sterm_final_sub_uniques %>% filter(ClusterName %in% namesSingleArrays)
# namesMultipleArrays <- which(table(spacer_Infos_Sterm_final_sub_uniques$ClusterName)==2) %>% names()
# tmp2 <- data.frame(ClusterName=namesMultipleArrays,ARRAY="multiple Arrays")
# Arrayasignment <- rbind(tmp1,tmp2)
  Arrayasignment <- tmp1
# clusterCRISPRs <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/clusterCRISPRs_sterm_new.txt", "\t", escape_double = FALSE, trim_ws = TRUE)
clusterCRISPRs <- read_delim("02_data/clusterCRISPRs_sterm.txt", "\t", escape_double = FALSE, trim_ws = TRUE)
length(names) #from previous chunk
occurence <- data.frame(ClusterName=clusterCRISPRs$Clustername,occurrencecounts=rowSums(clusterCRISPRs[,-1]))
# occurence <- data.frame(ClusterName=clusterCRISPRs$Clustername,occurrencecounts=rowSums(clusterCRISPRs[,-1]>0))
  occurence_final <- merge(occurence,Arrayasignment,by="ClusterName")

colors <- c("#fd8d3cff","#807dbaff","#ef3b2cff","grey")
occurence_final_sterm <- occurence_final %>% add_column(species="S.thermophilus")
table(occurence$occurrencecounts)
plotspacerCOUNT <- ggplot(occurence_final,aes(x=occurrencecounts,fill=ARRAY))+geom_bar(alpha=0.8)+theme_classic()+labs(x="number of genomes containing the same spacer",y="spacer count") #+scale_x_continuous(breaks =c(1:length(names)),labels=c(1:length(names)),limits = c(0,length(names)))+scale_fill_manual(values = colors)+theme(legend.title = element_blank())

# +scale_y_continuous(breaks =c(0,7,31,60,154),labels=c(0,7,31,60,154))
plotspacerCOUNT
# svg("~/Desktop/Projects/2019_RMK202_analysis/plot/spacerUniquenessCount.svg",width=4.5,height=2.5)
svg("03_results/spacerUniquenessCount.svg",width=6,height=2.5)

plotspacerCOUNT
 dev.off()
 

 
  svg("03_results/StrainSpacercount.svg",width=8,height=3.5)

spacerCount3
 dev.off()
```




## Metagenome CRISPR spacers

The way I try to identify CRISPR spacers in teh metagenome is by mapping the CRISPR repeats to the raw metagenomic reads and clipping of teh repeats to only keep the spacers. 
leave genomic spacers away and check in the end if the undemultplexed spacers occur int the genomes with cd-hit.

```{bash}

#manually extract the CRISPR repeats
mkdir -p /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/

###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples_withEvolution.txt  ##the file with all sample names
BaseLocation=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract_Sterm/
species=Sterm
repeatsFile=/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt

###================
##script
###================
##test
pair=1
names=G4_6_18
adapName=$(grep ">" ${repeatsFile} |head -1)
###----end

for names in $(cat ${samplesss} )
  do
 
for pair in $(echo "1 2")
  do
#pair=1

read=/archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_${pair}.fastq


for adapName in $(grep ">" ${repeatsFile} )
  do
adapSeq=$(grep "${adapName}" -A 1 ${repeatsFile}|tail -1)
adap=$(echo ${adapName} | sed 's/>//g'|cut -d '_' -f 1)
cleanRound=1

echo "------------------------------------------------"
echo $adap
echo $adapSeq
echo $names
echo $read
echo $species
echo "------------------------------------------------"

mkdir -p ${BaseLocation}/${names}/ReadPair_${pair}//tmp/

cutadapt  ${read} -b ${adapSeq}  -e  0.1 -O 15 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}.txt
rm ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}.txt &

## reads with match
awk '{if($2!="-1") print $0}' ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt

rm ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt & 

##make a list of current (potentially still containing 5' adapters)
awk -F "\t" '{OFS="\n"} {print "@"$1, $5,"+", $9}' ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt  > \
${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp.fastq


##trim 5' ends of these (with relaxed matching errors)
cutadapt ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp.fastq -g ${adapSeq}  -e  0.1 -O 5 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_.fastq.txt

rm ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_.fastq.txt &

## finalise first spacers (we will not be able to keep reads that don't have a small (>5bp) of repeat left)
##removing potential 5' adapters of first trim
mkdir -p ${BaseLocation}/${names}/final/

awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($7)>"25"  && length($7)<"46" && length($7)==length($11)) print "@"$1, $7 ,"+", $11}' ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}/${names}_${adap}_final_round1_R${pair}.fastq

cat ${BaseLocation}/${names}/ReadPair_${pair}/${names}_${adap}_final_round1_R${pair}.fastq > ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R${pair}.fastq


##take remaining reads and remove 3' adapters again --> do this 5 times
awk -F "\t" '{OFS="\n"} {print "@"$1, $7,"+", $11}' ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt  > \
${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round2_matchestmp_timmed5_readyForRound2.fastq

#grep --color "GTTTTAGAGCTGTGTTGTTTCGAATGGTTCCAAAAC" /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/readswithrepeats_${adap}_intraReads_info_trimmed_matches_readyForRound2.fastq  |head
cleanRound=2
for cleanRound in $(echo "2 3 4 5 6")
  do
  echo ${cleanRound}
  echo "------------------------------------"
cleanRound_next=$((cleanRound+1))
  
cutadapt ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_readyForRound${cleanRound}.fastq -b ${adapSeq}  -e  0.1 -O 5 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.fasta
rm ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.fasta &
##finalised spacers
awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($5)>"25"  && length($5)<"46"&& length($5)==length($9)) print "@"$1, $5 ,"+", $9}' ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}/${names}_${adap}_final_round${cleanRound}_R${pair}.fastq


cat ${BaseLocation}/${names}/ReadPair_${pair}/${names}_${adap}_final_round${cleanRound}_R${pair}.fastq >> ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R${pair}.fastq


##finalised prepered
awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($7)>"25" && length($7)<"46" && length($7)==length($11)) print "@"$1, $7,"+", $11}' ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound_next}_matchestmp_timmed5_readyForRound${cleanRound_next}.fastq

  done #cleanRound
  cat ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R1.fastq \
  ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R2.fastq > \
  ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads.fastq
  done #adap (spacer)
  
  ###-------------====
  ##reverse
    ###-------------====
#adapName=$(grep ">" /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut_reversed.txt| cut -d ' ' -f 1 |head -1)
for adapName in $(grep ">" ${repeatsFile}| cut -d ' ' -f 1)
  do
adapSeq=$(grep "${adapName}" -A 1 ${repeatsFile}|tail -1)
adap=$(echo ${adapName} | sed 's/>//g'|cut -d '_' -f 1)
cleanRound=1

echo "------------------------------------------------"
echo $adap
echo $adapSeq
echo $names
echo $read
echo $species
echo "------------------------------------------------"

mkdir -p ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/

cutadapt ${read} -b ${adapSeq}  -e  0.1 -O 15 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}.txt
rm ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}.txt &

## reads with match
awk '{if($2!="-1") print $0}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt

rm ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt & 

##make a list of current (potentially still containing 5' adapters)
awk -F "\t" '{OFS="\n"} {print "@"$1, $5,"+", $9}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt  > \
${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp.fastq


##trim 5' ends of these (with relaxed matching errors)
cutadapt ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp.fastq -g ${adapSeq}  -e  0.1 -O 5 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_.fastq.txt

rm ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_.fastq.txt &

## finalise first spacers (we will not be able to keep reads that don't have a small (>5bp) of repeat left)
##removing potential 5' adapters of first trim
mkdir -p ${BaseLocation}/${names}/final/
awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($7)>"25" && length($7)<"46" && length($7)==length($11)) print "@"$1, $7 ,"+", $11}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_info.txt > \
${BaseLocation}/${names}/ReadPair_${pair}_reversed/${names}_${adap}_final_round1_R${pair}_reversed.fastq

cat ${BaseLocation}/${names}/ReadPair_${pair}_reversed/${names}_${adap}_final_round1_R${pair}_reversed.fastq > ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R${pair}_reversed.fastq


##take remaining reads and remove 3' adapters again --> do this 5 times
awk -F "\t" '{OFS="\n"} {print "@"$1, $7,"+", $11}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt  > \
${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round2_matchestmp_timmed5_readyForRound2.fastq

#grep --color "GTTTTAGAGCTGTGTTGTTTCGAATGGTTCCAAAAC" /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/readswithrepeats_${adap}_intraReads_info_trimmed_matches_readyForRound2.fastq  |head
cleanRound=2
for cleanRound in $(echo "2 3 4 5 6")
  do
  echo ${cleanRound}
  echo "------------------------------------"
cleanRound_next=$((cleanRound+1))
  
cutadapt ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_readyForRound${cleanRound}.fastq -b ${adapSeq}  -e  0.1 -O 5 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.fasta
rm ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.fasta &
##finalised spacers
awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($5)>"25" && length($5)<"46" && length($5)==length($9)) print "@"$1, $5 ,"+", $9}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}_reversed/${names}_${adap}_final_round${cleanRound}_R${pair}_reversed.fastq

cat ${BaseLocation}/${names}/ReadPair_${pair}_reversed/${names}_${adap}_final_round${cleanRound}_R${pair}_reversed.fastq >> ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R${pair}_reversed.fastq


##finalised prepered
awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($7)>"25" && length($7)<"46" && length($7)==length($11)) print "@"$1, $7,"+", $11}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound_next}_matchestmp_timmed5_readyForRound${cleanRound_next}.fastq



  done #cleanRound
  cat ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R1_reversed.fastq \
  ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R2_reversed.fastq > \
  ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed.fastq
  #revseq ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed.fastq ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular.fastq
  seqtk seq -r  ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed.fastq > ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular.fastq

  done #adap (spacer)
  
  done #pair (readPair)
  done #names (sampleName)

##==============================================
##-----------------clean up array labels
##==============================================

repeatsFile_old=/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut_old.txt
for names in $(cat ${samplesss} |grep "^G")
  do
for adapName in $(grep ">" ${repeatsFile_old}  )
  do
  
Arraynum=$(echo ${adapName} |tail -c2)
adap=$(grep "${Arraynum}" ${repeatsFile_old} |sed 's/>//g')
#cleanRound=1
adapSeqNew=$(grep "${Arraynum}" ${repeatsFile} |sed 's/>//g')

#echo -e ${BaseLocation}/${names}"/final/"${names}_${adap}"_final_allRounds_allReads.fastq"
#echo -e ${BaseLocation}/${names}"/final/"${names}_${adapSeqNew}"_final_allRounds_allReads.fastq"

cp ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads.fastq ${BaseLocation}/${names}/final/${names}_${adapSeqNew}_final_allRounds_allReads.fastq
cp ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed.fastq ${BaseLocation}/${names}/final/${names}_${adapSeqNew}_final_allRounds_allReads_reversed.fastq


done

done


##==============================================
##-----------------merge all samples spacers
##==============================================
rm ${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_${adap}_Sterm.fastq
rm -r ${BaseLocation}/FinalSpacers/ForDADA2_Sterm/
rm -r ${BaseLocation}/FinalSpacers/ForDADA2_woREverse_Sterm
mkdir -p ${BaseLocation}/FinalSpacers/ForDADA2_woREverse_Sterm
mkdir -p ${BaseLocation}/FinalSpacers/ForDADA2_Sterm/
for names in $(cat ${samplesss} )
  do
for adapName in $(grep ">" ${repeatsFile}  )
  do
adapSeq=$(grep "${adapName}" -A 1 ${repeatsFile})
adap=$(echo ${adapName} | sed 's/>//g'|cut -d '_' -f 1)
cleanRound=1

seqtk seq -r  ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed.fastq > ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular.fastq


#-----------change name



adapShor=$(echo $adap |sed 's/Array/a/g')
awk -v sampless="$names" -v arrayss="$adapShor"  '{if(NR%4==1) $0=sprintf("@"sampless":"arrayss":s"++k,(1+i++)); print;}' ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads.fastq > ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_renamed.fastq

adapShor=$(echo $adap |sed 's/Array/a/g')
awk -v sampless="$names" -v arrayss="$adapShor"  '{if(NR%4==1) $0=sprintf("@"sampless":R"arrayss":s"++k,(1+i++)); print;}' ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular.fastq > ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular_renamed.fastq


###----------------------without Reversing
adapShor=$(echo $adap |sed 's/Array/a/g')
awk -v sampless="$names" -v arrayss="$adapShor"  '{if(NR%4==1) $0=sprintf("@"sampless":"arrayss":s"++k,(1+i++)); print;}' ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads.fastq >> ${BaseLocation}/FinalSpacers/ForDADA2_woREverse_Sterm/${names}_allSpacermerged_S_${adap}_Sterm.fastq

adapShor=$(echo $adap |sed 's/Array/a/g')
awk -v sampless="$names" -v arrayss="$adapShor"  '{if(NR%4==1) $0=sprintf("@"sampless":"arrayss":s"++k,(1+i++)); print;}' ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed.fastq >> ${BaseLocation}/FinalSpacers/ForDADA2_woREverse_Sterm/${names}_allSpacermerged_${adap}_Sterm.fastq


#-----------merge samples
mkdir -p  ${BaseLocation}/FinalSpacers/mergeSpacersFile
cat ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_renamed.fastq >> ${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_S_${adap}.fastq
cat ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular_renamed.fastq >> ${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_S_${adap}.fastq


##------ make a folder for dada2
cat ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_renamed.fastq >> ${BaseLocation}/FinalSpacers/ForDADA2_Sterm/${names}_allSpacermerged_S_${adap}.fastq
cat ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular_renamed.fastq >> ${BaseLocation}/FinalSpacers/ForDADA2_Sterm/${names}_allSpacermerged_S_${adap}.fastq


done #adap (spacer)
  done #names (sampleName)
  
  echo ${BaseLocation}/FinalSpacers/ForDADA2_woREverse_Sterm/
echo ${BaseLocation}/FinalSpacers/ForDADA2_Sterm/

```


### Dada2

Here, I try out the [Dada2](https://benjjneb.github.io/dada2/tutorial.html) tutorial to call true positive spacers. 

```{r}
library(dada2)
library(plyr)
library(tidyverse)

path <- "02_data/ForDADA2_woREverse_sterm/" 

list.files(path)

fnFs <- sort(list.files(path, pattern=".fastq", full.names = TRUE)) 
sample.names <- str_split_fixed(basename(fnFs), ".fastq",2)[,1]

seq.file <- fnFs[1]
foo <- readLines(seq.file)
seqlens <- sapply(foo[seq(2,length(foo),4)], nchar)
quallens <- sapply(foo[seq(4,length(foo),4)], nchar)
table(seqlens); table(quallens)
table(seqlens) == table(quallens)

plotQualityProfile(fnFs[1:2])
plotQualityProfile(fnFs[3:4])

# Place filtered files in filtered/ subdirectory
filtFs <- file.path(path, "filtered", paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(path, "filtered", paste0(sample.names, "_R_filt.fastq.gz"))
names(filtFs) <- sample.names
names(filtRs) <- sample.names

out <- filterAndTrim(fnFs, filtFs, truncQ=2, rm.phix=TRUE,
              compress=TRUE, multithread=TRUE) # On Windows set multithread=FALSE
##--------------
##plot number of CRISPR spacer reads
##--------------

plotREads <- as.data.frame(out)
plotREads$reads.in
plotREads$longName <- rownames(plotREads)
plotREads$shortName <- gsub("_allSpacermerged_","-",plotREads$longName) %>% gsub(".fastq","",.)
plotREads$sample <- str_split_fixed(plotREads$shortName, "-", 2)[,1]
plotREads$array <- str_split_fixed(plotREads$shortName, "-", 2)[,2]
plotREads$sample <- revalue(plotREads$sample, c("lyo202_96"="Lyo 1996","Lyo_202_2012"="Lyo 2012", "Lyo_202_2014"="Lyo 2014", "Konserve_202"="working stock", "RMK202"="starter culture 2012", "Versand_202"="starter culture 2018","di_K2_6h"="cheesemaking day1","th_K2_8h"="cheesemaking day2","G1_6_18"="experiment_A","G2_6_18"="experiment_B","G3_6_18"="experiment_C","G4_6_18"="experiment_D","G5_6_18"="experiment_E"))
plotREads$sample = factor(plotREads$sample, levels=c("Lyo 1996","Lyo 2012","Lyo 2014","working stock","starter culture 2012","starter culture 2018","cheesemaking day1","cheesemaking day2","experiment_A","experiment_B","experiment_C","experiment_D","experiment_E"))

# plotREads %>%  group_by(sample) %>% 
colors <- c("#fd8d3cff","#807dbaff","#ef3b2cff")
rawspacerReads <- ggplot(plotREads,aes(sample,y=reads.in,fill=array))+geom_bar(stat="identity")+theme_classic()+facet_wrap(~array)+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),legend.position = "none")+labs(x="",y="CRISPR spacer reads")+scale_fill_manual(values = colors)
rawspacerReads
  svg("03_results/rawCRISPRspacerReads_sterm.svg",width=4,height=4)
  rawspacerReads
   dev.off()



##--------------
##error learn
##--------------
errF <- learnErrors(filtFs, multithread=TRUE)
plotErrors(errF, nominalQ=TRUE)
dadaFs <- dada(filtFs, err=errF, multithread=TRUE)
dadaFs[[1]]

###fasta table

seqtab <- makeSequenceTable(dadaFs)
dim(seqtab)
table(nchar(getSequences(seqtab)))


##do not remove Chimeras because they are all on the same read and most likely not technical errors but truth biological variation
# seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=TRUE, verbose=TRUE)
# dim(seqtab.nochim)
# sum(seqtab.nochim)/sum(seqtab)


##----phyloseq

library(phyloseq); packageVersion("phyloseq")
library(Biostrings); packageVersion("Biostrings")
library(ggplot2); packageVersion("ggplot2")

samples.out <- rownames(seqtab)
sample <- str_split_fixed(samples.out, "_allSpacermerged_", 2)[,1]
array <- str_split_fixed(samples.out, "_allSpacermerged_", 2)[,2]
# subject <- sapply(strsplit(samples.out, "D"), `[`, 1)
# gender <- substr(subject,1,1)
# subject <- substr(subject,2,999)
# day <- as.integer(sapply(strsplit(samples.out, "D"), `[`, 2))
samdf <- data.frame(Sample=sample, Array=array)
# samdf$When <- "Early"
# samdf$When[samdf$Day>100] <- "Late"
rownames(samdf) <- samples.out


# seq
ps <- phyloseq(otu_table(seqtab, taxa_are_rows=FALSE),sample_data(samdf))

sequences <- Biostrings::DNAStringSet(taxa_names(ps))

names(sequences) <- taxa_names(ps)
ps <- merge_phyloseq(ps, sequences)
taxa_names(ps) <- paste0("spacer_",seq(ntaxa(ps)))

writeXStringSet(refseq(ps), "02_data/ForDADA2_woREverse_sterm//all_unique_spacers.fasta", append=FALSE, format="fasta") 
refseq(ps)
# plot_richness(ps, x="Sample",measures=c("Observed","Shannon", "Simpson"),color="Array")


colors <- c("#fd8d3cff","#807dbaff","#ef3b2cff")

ShannonCRISPR <- plot_richness(ps, x="Array", measures=c( "Shannon"),color="Array") + geom_boxplot()+scale_fill_manual(values=colors)+scale_color_manual(values=colors)+theme_classic()+labs(x="",y="Shannon diversity of CRISPR spacers")+theme(legend.title = element_blank(),legend.position="top",axis.text.x =element_text(angle = 75, hjust = 1,size=9))

AbsorvedCRISPR <- plot_richness(ps, x="Array", measures=c( "Observed"),color="Array") + geom_boxplot()+scale_fill_manual(values=colors)+scale_color_manual(values=colors)+theme_classic()+labs(x="",y="Number of CRISPR spacers observed")+theme(legend.position = "none" ,axis.text.x =element_text(angle = 75, hjust = 1,size=9))

  svg("03_results/rarefaction_spacers_Shannon_sterm.svg",width=4,height=8)
  AbsorvedCRISPR+ShannonCRISPR
   dev.off()

##------rarefaction

library(ranacapa)
# rarefaction_spacers <- ggrare(ps, step = 10, label = NULL, color = "Sample",
#   plot = TRUE, parallel = FALSE, se = TRUE) +theme_classic()
sample_variables(ps)
colors <- c("#fd8d3cff","#807dbaff","#ef3b2cff")

sample_names(ps) <- gsub("_allSpacermerged_Array","_a",sample_names(ps))


# rarefaction_spacers <- ggrare(ps, step = 10, label = "Sample", color = "Array",
  # plot = TRUE, parallel = FALSE, se = TRUE) +theme_classic()+scale_fill_manual(values=colors)+scale_color_manual(values=colors)+scale_x_continuous(breaks =c(0,10000,20000,30000,40000,45000),labels=c(0,10000,20000,30000,40000,45000))+labs(x="number of CRISPR spacer reads",y="number of CRISPR spacers detected")+theme(legend.title = element_blank())
# rarefaction_spacers

  
rarefaction_spacers <- ggrare(ps, step = 10, color = "Array",
  plot = TRUE, parallel = FALSE, se = TRUE) +theme_classic()+scale_fill_manual(values=colors)+scale_color_manual(values=colors)+scale_x_continuous(breaks =c(0,10000,20000,30000,40000,45000),labels=c(0,10000,20000,30000,40000,45000))+labs(x="Number of CRISPR spacer reads",y="Number of CRISPR spacers observed")+theme(legend.title = element_blank())
rarefaction_spacers
  
  svg("03_results/rarefaction_spacers_sterm.svg",width=5,height=3)
  rarefaction_spacers
   dev.off()
   
   library(patchwork)
   
  svg("03_results/rarefaction_spacers_together_sterm.svg",width=15,height=8)
  rarefaction_spacers+theme(legend.position = "none")+AbsorvedCRISPR+ShannonCRISPR+plot_layout(widths =c(4,1,1))
   dev.off()  
##------pca
ps.prop <- transform_sample_counts(ps, function(otu) otu/sum(otu))
ord.nmds.bray <- ordinate(ps.prop, method="NMDS", distance="bray")

plot_ordination(ps.prop, ord.nmds.bray, color="Array", title="Bray NMDS")

##-----------------------
##export OTU table
##-----------------------

normfunction <- function(x) {return(100000*x/sum(x))}
d.norm1 <- transform_sample_counts(physeq = ps, fun = normfunction)
otu_table(d.norm1)




otu_table_df <- otu_table(d.norm1) %>% as.data.frame()%>% rownames_to_column() %>% separate(rowname, c("sample","Array"), sep = "_allSpacermerged_") %>% select(-Array) %>%  group_by(sample) %>% summarise_all(.funs = c(sum="sum")) %>% column_to_rownames(var="sample") %>% t() %>% as.data.frame() %>% rownames_to_column("spacer") %>% 
  mutate(spacer = str_replace(spacer, "_sum", "")) 
otu_table_df
write.table(otu_table_df,"02_data/ForDADA2_woREverse_sterm//dada_spacer_count.txt",na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names =TRUE)

##-----------------------
##count number of OTUs larger then 0 in every sample
##-----------------------
nrow(otu_table(ps))
otu_counts_samples <- NA
for (samplesss in rownames(otu_table(ps))) {
  tmp_2 <- sum(otu_table(ps)[samplesss]>0)
  tmp_print <- c(sample=as.character(samplesss), otus=as.numeric(tmp_2))
  print(tmp_print)
  otu_counts_samples <- rbind(otu_counts_samples,tmp_print)
}
rownames(otu_table(ps)[1])
sum(otu_table(ps)[1]>0)

otu_counts_samples <- as.data.frame(otu_counts_samples[-1,]) %>% remove_rownames()
otu_counts_samples$otus <- as.numeric(as.character(otu_counts_samples$otus))

otu_counts_samples$sample_short <- gsub("_a\\d","",otu_counts_samples$sample)

 otu_counts_samples %>%
  group_by(sample_short) %>% 
  dplyr::summarise(summedcount=sum(otus))
##-----------------------
##spaccerss belonging to sample
##-----------------------

ncol(otu_table(ps))
spacer_counts_samples <- data.frame()
# samplesss <- "spacer_1"
for (samplesss in colnames(otu_table(ps))) {
  tmp_2 <- sum(otu_table(ps)[,samplesss]>0)

tmp1 <- sample_names(otu_table(ps)[which(otu_table(ps)[,samplesss]>0)])
# countArray1 <- length(grep("Array1",tmp1))
# countArray2 <- length(grep("Array2",tmp1))
# countArray3 <- length(grep("Array3",tmp1))
countArray1 <- length(grep("_a1",tmp1))
countArray2 <- length(grep("_a2",tmp1))
countArray3 <- length(grep("_a3",tmp1))

numberArrays <-(countArray1>0)+(countArray2>0)+(countArray3>0)

ArrayAssigned <- ifelse(countArray1>0,"Array1",ifelse(countArray2>0,"Array2","Array3"))

  tmp_print <- data.frame(spacer=as.character(samplesss), samplesCount=as.numeric(tmp_2),Array1count=countArray1,Array2count=countArray2,Array3count=countArray3,totalArrays=numberArrays,Array=ArrayAssigned)
  
  print(tmp_print)
  spacer_counts_samples <- rbind(spacer_counts_samples,tmp_print)
}
# rownames(spacer_counts_samples(ps)[1])
sum(otu_table(ps)[1]>0)

# spacer_counts_samples <- as.data.frame(spacer_counts_samples[-1,]) %>% remove_rownames()
# spacer_counts_samples$otus <- as.numeric(as.character(spacer_counts_samples$otus))

spacer_counts_samples[which(spacer_counts_samples$Array2count > 0),]
table(spacer_counts_samples$totalArrays)
table(spacer_counts_samples$Array)

```


### match strains to spacers

this output can be used for the spacer heatmap
We check which clusters are only metagenomic and which only strains

```{bash}
###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt  ##the file with all sample names
BaseLocation=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract/
species=Sterm
repeatsFile=/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt
MetaspacerFile=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers/ForDADA2_woREverse//all_unique_spacers.fasta

###---------------------------
##merge CRISPR spacer files

mkdir -p ${BaseLocation}/FinalSpacers/cdhit_withReference

cat ${MetaspacerFile} \
 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results_ClusterName.txt |sed 's/Cluster/cl/g' >  \
${BaseLocation}/FinalSpacers/cdhit_withReference/all_spacers_withRef_and_oldStrains.fasta
##--------------------------------------------------
##-----------cd -hit
##--------------------------------------------------
mkdir -p ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain

/home/vincent/apps/cd-hit-v4.8.1-2019-0228/cd-hit-est -i ${BaseLocation}/FinalSpacers/cdhit_withReference/all_spacers_withRef_and_oldStrains.fasta \
    -o  /${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered.txt -c 0.995


##--Qualtiy control

grep ">" -c ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered.txt


  
###=====================================================0  
##extract via spacers
###----------new approach 
###=====================================================
#spacersss=$( echo ">spacer_330")


sed 's/>Clu/>META_Clu/g' ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered.txt.clstr |sed 's/META_Cluster /META_Cluster_/g' > ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed.txt.clstr


echo -e "METAClusterName\tdadaSpacer\tStrains\tARRAYINFO\tClusterINFO\tspacer" > ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain//uniqueSpacers_count_Ref_wOLDstrains.txt

for spacersssClusts in $(grep "^>" ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed.txt.clstr)
do
echo ${spacersssClusts}


ClusterEND=$(grep -A 10 "${spacersssClusts}$(printf '$')" ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed.txt.clstr |grep ">META_C"|head -2|tail -1)


  sed -n "/$spacersssClusts$/,/$ClusterEND$/p" ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed.txt.clstr > ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/tmp.txt

tmpfile=${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/tmp.txt


spacer=$(grep ">spacer_" -c ${tmpfile})  
strains=$(grep ">cl_" -c ${tmpfile})
spacerName=$(grep ">spacer_" ${tmpfile} |head -1|awk -F "[>...]" '{print $2}')
if [ ${strains} == "0" ]; then
            arrayINFO=$(grep "${spacerName}$(printf '\t')" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers/ForDADA2_woREverse/spacersAssigned.txt |cut -f 2|sed 's/Array/a/g')
            strains_INFO=$(echo "MetaGenome")
            else
              strains_INFO=$(grep ">cl_" ${tmpfile} |head -1|awk -F "[>...]" '{print $2}'|sed 's/cl_/Cluster_/g' )
            arrayINFO=$(grep "${strains_INFO}$(printf '\t')" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final.txt |cut -f 6|head -1)
               
            fi

clusterNAME=$(echo -e "${spacersssClusts}" | sed 's/ /_/g'|sed 's/>//g')

 echo -e ${clusterNAME}"\t"${spacer}"\t"${strains}"\t"${arrayINFO}"\t"${strains_INFO}"\t"${spacerName} >> ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/uniqueSpacers_count_Ref_wOLDstrains.txt

done #spacers
cut -f 2 ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/uniqueSpacers_count_Ref_wOLDstrains.txt|sort|uniq -c
cut -f 3 ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/uniqueSpacers_count_Ref_wOLDstrains.txt|sort|uniq -c
cut -f 4 ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/uniqueSpacers_count_Ref_wOLDstrains.txt|sort|uniq -c


awk '{if($2==2)print $0}' ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/uniqueSpacers_count_Ref_wOLDstrains.txt

###----------------------------------------
##change name of SPACERS
###----------------------------------------

 cat ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered.txt > ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed_ClusterName.txt
 rm ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/tmp_spacer.txt
   for spacersss in $(grep ">" ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed_ClusterName.txt)
   do
   echo "--------------------------"
   #spacersss=spacer_32
   
   newNAME=$(grep -B 20 "${spacersss}\." ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed.txt.clstr |grep "^>META" |tail -1 )
   
   echo -e ${newNAME}"\t"${spacersss} >> ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain//tmp_spacer.txt
   
   sed -i "s/${spacersss}$/${newNAME}/g" ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed_ClusterName.txt
   
   done
   rm ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain//tmp_spacer.txt
   grep ">" ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed_ClusterName.txt |sort|uniq |wc -l
   
  echo ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed_ClusterName.txt 
  echo ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/uniqueSpacers_count_Ref_wOLDstrains.txt
```



```{r}

uniqueSpacers_count_Ref_wOLDstrains <- read_delim("02_data/uniqueSpacers_count_Ref_wOLDstrains_sterm.txt","\t", escape_double = FALSE, trim_ws = TRUE)

uniqueSpacers_count_Ref_wOLDstrains$explained <- ifelse(uniqueSpacers_count_Ref_wOLDstrains$dadaSpacer>0&uniqueSpacers_count_Ref_wOLDstrains$Strains>0,"both Meta & strains",ifelse(
  uniqueSpacers_count_Ref_wOLDstrains$dadaSpacer>0&uniqueSpacers_count_Ref_wOLDstrains$Strains==0,"explained only by meta","explained only by strains"
))


##-------------plot explained

colorsss <- rev(c("darkcyan","darkturquoise","lightblue"))
uniqueSpacers_count_Ref_wOLDstrains$explained = factor(uniqueSpacers_count_Ref_wOLDstrains$explained, levels=c("explained only by meta","both Meta & strains","explained only by strains"))
uniqueSpacers_count_Ref_wOLDstrains$ARRAYINFO <- revalue(uniqueSpacers_count_Ref_wOLDstrains$ARRAYINFO, c("a2"="CRISPR array 2","a1"="CRISPR array 1","a3"="CRISPR array 3"))
table(uniqueSpacers_count_Ref_wOLDstrains$explained,uniqueSpacers_count_Ref_wOLDstrains$ARRAYINFO)

 
 ##new and wider
svg("03_results/spacerexplained_wide.svg",width=4,height=3)
ggplot(uniqueSpacers_count_Ref_wOLDstrains,aes(x=ARRAYINFO,group=explained,fill=explained))+geom_bar()+theme_classic()+scale_fill_manual(values=colorsss)+theme(axis.title.x = element_blank(),axis.text.x =element_text(angle = 45, hjust = 1,size=9) )+labs(y="spacer counts",fill="")+scale_y_continuous(breaks =c(0,10,27,39,92,117,139,163),labels=c(0,10,27,39,92,117,139,163))
 dev.off()
 ##------------------------
 #what are the only strains CRISPRs
  ##------------------------
ONlyIsolateSPACERS <- uniqueSpacers_count_Ref_wOLDstrains %>% filter(explained=="explained only by strains")
 

merge(ONlyIsolateSPACERS,spacer_Infos_Sterm_final,by.x="ClusterINFO",by.y="ClusterName")
 
merge(ONlyIsolateSPACERS,uniqueSpacersss_extended,by.x="ClusterINFO",by.y="ClusterName")


```

## extended Analysis
### blast Spacers

Here a look which spacers map to the phages or bacteria. 


```{bash}
###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt  ##the file with all sample names
BaseLocation=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract/
species=Sterm
repeatsFile=/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt
MetaspacerFile=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers/ForDADA2_woREverse//all_unique_spacers.fasta
mappingFile=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed_ClusterName.txt

##--------------------------blast to local phages 
  
#rm -r /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast
mkdir -p ${BaseLocation}/CRISPRspacerBLAST/blast
mkdir -p ${BaseLocation}/CRISPRspacerBLAST/starterPhagesPlasmid
mkdir -p ${BaseLocation}/CRISPRspacerBLAST/genomes/
  
##---------------------
##get local phages
##---------------------
rm ${BaseLocation}/CRISPRspacerBLAST/genomes/phagenGenome.fasta
for phageesss in $(grep ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta |grep "CP046134" -v | grep -v "CP046131"|sed 's/>//g')
do

samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta ${phageesss} > ${BaseLocation}/CRISPRspacerBLAST/genomes/${phageesss}.fasta
cat ${BaseLocation}/CRISPRspacerBLAST/genomes/${phageesss}.fasta >> ${BaseLocation}/CRISPRspacerBLAST/genomes/phagenGenome.fasta
done


makeblastdb -max_file_sz 1GB -in ${BaseLocation}/CRISPRspacerBLAST/genomes/phagenGenome.fasta \
 -out ${BaseLocation}/CRISPRspacerBLAST/genomes/phagenGenome -parse_seqids -dbtype nucl


  blastn -num_threads 35 -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query ${mappingFile} \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen sgi sacc sblastnames staxids sscinames" \
  -db ${BaseLocation}/CRISPRspacerBLAST/genomes/phagenGenome \
  -out ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages.txt \
  -evalue 0.01 -word_size 16



##---------------------
##one phage at the time
##---------------------
#phageesss=Streptococcus_phage_2

for phageesss in $(echo "Streptococcus_phage_1 Streptococcus_phage_2")
do
rm ${BaseLocation}/CRISPRspacerBLAST/genomes/${phageesss}.fasta

samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta ${phageesss} > ${BaseLocation}/CRISPRspacerBLAST/genomes/${phageesss}.fasta


makeblastdb -max_file_sz 1GB -in ${BaseLocation}/CRISPRspacerBLAST/genomes/${phageesss}.fasta \
 -out ${BaseLocation}/CRISPRspacerBLAST/genomes/${phageesss} -parse_seqids -dbtype nucl


  blastn -num_threads 35 -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query ${mappingFile} \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen sgi sacc sblastnames staxids sscinames" \
  -db ${BaseLocation}/CRISPRspacerBLAST/genomes/${phageesss} \
  -out ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_${phageesss}.txt \
  -evalue 0.01 -word_size 16

awk -F "\t" '{OFS="\t"}{if($5<"2")print $0}' ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_${phageesss}.txt > ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_${phageesss}_cleaned.txt
done
phageesss=Streptococcus_phage_1
rm ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both.txt
#rm ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both_allINfo.txt
for phageesss in $(echo "Streptococcus_phage_1 Streptococcus_phage_2")
do
echo ${phageesss}
wc -l ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_${phageesss}.txt
cut -f 3 ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_${phageesss}_cleaned.txt |sort|uniq -c
cut -f 1,2 ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_${phageesss}_cleaned.txt >>  ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both.txt

cat ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_${phageesss}_cleaned.txt >> ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both_allINfo.txt
done

wc -l ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both.txt
cut -f 1 ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both.txt|sort|uniq|wc -l
cut -f 1 ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both.txt|sort|uniq -c |sed 's/^ *//g'


rm ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits.txt
cut -f 1 ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both.txt|sort|uniq -c |sed 's/^ *//g'| while read genenomes 
do
#echo ${genenomes}
#echo "--------------------"
num=$(echo ${genenomes}|cut -d ' ' -f 1)
spacer=$(echo ${genenomes}|cut -d ' ' -f 2)

if [ "$num" = "1" ]
then
genom=$(grep ${spacer} ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both.txt|cut -f 2|head -1)
echo -e "localPHAGE\t"${spacer}"\t"${genom} >> ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits.txt
else
echo -e "localPHAGE\t"${spacer}"\tbothPhages" >> ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits.txt
fi
done


###--------------------where do the mutations lie 
##prepare for genoPlotR
rm -r ${BaseLocation}/CRISPRspacerBLAST/blast/GenotplotR/
mkdir -p ${BaseLocation}/CRISPRspacerBLAST/blast/GenotplotR/
for phageesss in $(echo "Streptococcus_phage_1 Streptococcus_phage_2")
do
echo ${phageesss}
#wc -l ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_${phageesss}.txt

echo -e "name\tstart\tend\tstrand\tcol\tfill" > ${BaseLocation}/CRISPRspacerBLAST/blast/GenotplotR/CRISPRblast_rmk202phages_${phageesss}_genoplotR.txt

awk -F "\t" '{OFS="\t"}{print $1,$9,$10,"1","black",$5}' ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_${phageesss}.txt >> ${BaseLocation}/CRISPRspacerBLAST/blast/GenotplotR/CRISPRblast_rmk202phages_${phageesss}_genoplotR.txt

awk -F "\t" -v genomsss="$phageesss" '{OFS="\t"}{print genomsss,$9,$10}' ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_${phageesss}.txt >> ${BaseLocation}/CRISPRspacerBLAST/blast/GenotplotR/CRISPRblast_rmk202phages_${phageesss}_Circos.txt

done

###--------------------where do the mutations lie for circos
rm ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits_bothGenomes_circos_protospacers.txt
cut -f 1 ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both.txt|sort|uniq -c |sed 's/^ *//g'| while read genenomes 
do
#echo ${genenomes}
#echo "--------------------"
num=$(echo ${genenomes}|cut -d ' ' -f 1)
spacer=$(echo ${genenomes}|cut -d ' ' -f 2)

if [ "$num" = "1" ]
then
grep "${spacer}" ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both_allINfo.txt |awk -F "\t" '{OFS="\t"}{if($9>$10)print $2,$9,$10,"black_a1";else print $2,$9,$10,"black_a1"}' >> ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits_bothGenomes_circos_protospacers.txt
else
grep "${spacer}" ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both_allINfo.txt|head -1 |awk -F "\t" '{OFS="\t"}{if($9>$10)print $2,$9,$10,"chr5_a1";else print $2,$9,$10,"chr5_a1"}' >> ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits_bothGenomes_circos_protospacers.txt
grep "${spacer}" ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both_allINfo.txt|tail -1 |awk -F "\t" '{OFS="\t"}{if($9>$10)print $2,$9,$10,"chr5_a1";else print $2,$9,$10,"chr5_a1"}' >> ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits_bothGenomes_circos_protospacers.txt
fi
done

###--------------------where do the mutations lie for protospacer vs. genes intersect
rm ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits_bothGenomes_interesect_protospacers.txt
cut -f 1 ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both.txt|sort|uniq -c |sed 's/^ *//g'| while read genenomes 
do
#echo ${genenomes}
#echo "--------------------"
num=$(echo ${genenomes}|cut -d ' ' -f 1)
spacer=$(echo ${genenomes}|cut -d ' ' -f 2)

if [ "$num" = "1" ]
then
grep "${spacer}" ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both_allINfo.txt |awk -F "\t" '{OFS="\t"}{if($9>$10)print $2,$10,$9,$1;else print $2,$9,$10,$1}' >> ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits_bothGenomes_interesect_protospacers.txt
else
grep "${spacer}" ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both_allINfo.txt|head -1 |awk -F "\t" '{OFS="\t"}{if($9>$10)print $2,$10,$9,$1;else print $2,$9,$10,$1}' >> ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits_bothGenomes_interesect_protospacers.txt
grep "${spacer}" ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both_allINfo.txt|tail -1 |awk -F "\t" '{OFS="\t"}{if($9>$10)print $2,$10,$9,$1;else print $2,$9,$10,$1}' >> ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits_bothGenomes_interesect_protospacers.txt
fi
done

```


secondly we map against a complete bacteria collection.

```{bash}
##--------------------------------  
##--------------------------blast to local bacteria
##preperation have ot do once
##--------------------------------   
 
###----------------
 ##CRISPR locations
 ##and CRISPR MASK
###----------------
species=Sterm
 rm ${BaseLocation}/CRISPRspacerBLAST/blast/all_Genomes_masked.fna
 for genomes in $(echo "202-SMAG 202-S50 202-S72")
do
echo $genomes

 grep "repeat" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA_*.gff |cut -f 1,4,5 > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/CRISPRrepeat_${genomes}.bed


bedtools maskfasta -fi /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned//${genomes}.fna  -bed /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/CRISPRrepeat_${genomes}.bed -fo /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned//${genomes}_CRISPRmasked.fna 

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned//${genomes}_CRISPRmasked.fna  >> ${BaseLocation}/CRISPRspacerBLAST/blast/all_Genomes_masked.fna
 
done

 
#----------ldel 
 
species=Ldel
genomes=202-LMAG
 grep "repeat" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA_*.gff |cut -f 1,4,5 > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/CRISPRrepeat_${genomes}.bed

grep "repeat" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA_*.gff |awk -F "\t" '{OFS="\t"}{print $1,$4-500,$5+500}' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/CRISPRrepeat_${genomes}_extended.bed



bedtools maskfasta -fi /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned//${genomes}.fna  -bed /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/CRISPRrepeat_${genomes}.bed -fo /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned//${genomes}_CRISPRmasked.fna 

bedtools maskfasta -fi /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned//${genomes}.fna  -bed /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/CRISPRrepeat_${genomes}_extended.bed -fo /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned//${genomes}_CRISPRmasked_extended.fna 


cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned//${genomes}_CRISPRmasked.fna >> ${BaseLocation}/CRISPRspacerBLAST/blast/all_Genomes_masked.fna
 
 
 
  ##create bacteria DB
  ##---------rmk202
#bedtools maskfasta -fi /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_sterm/final/S_thermophilus_mag_rmk202.fasta -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/allStreptococcus_CRISPRlocation.bed \
#-fo /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/Streptococcus_CRISPR.masked_rmk202Genome.fa

##---------S50
#bedtools maskfasta -fi /archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/S50.fna -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/allStreptococcus_CRISPRlocation.bed \
#-fo /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_S50_Genome.fa

##---------rmk202
#bedtools maskfasta -fi /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/S72.fna -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/allStreptococcus_CRISPRlocation.bed \
#-fo /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_S72_Genome.fa

##_rename contigs for easy search

#sed -i 's/>/>REF_R202:/g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/Streptococcus_CRISPR.masked_rmk202Genome.fa
#sed -i 's/>/>REF_mst1:/g' /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_S50_Genome.fa
#sed -i 's/>/>REF_mst2:/g' /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_S72_Genome.fa

#cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/Streptococcus_CRISPR.masked_rmk202Genome.fa \
#/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_ldel/final/L_delbrueckii_mag_rmk202.fasta \
#/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_S50_Genome.fa \
#/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_S72_Genome.fa > \
#/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_allStrepGenomesrmk202_Genome.fa


#makeblastdb -max_file_sz 1GB -in /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/starterPhagesPlasmid/rmk202_phages_plasmids.fasta \
# -out /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/starterPhagesPlasmid/rmk202_phages_plasmids -parse_seqids -dbtype nucl




makeblastdb -max_file_sz 1GB -in ${BaseLocation}/CRISPRspacerBLAST/blast/all_Genomes_masked.fna \
 -out ${BaseLocation}/CRISPRspacerBLAST/blast/all_Genomes_masked -parse_seqids -dbtype nucl


  blastn -num_threads 35 -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query ${mappingFile} \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen sgi sacc sblastnames staxids sscinames" \
  -db ${BaseLocation}/CRISPRspacerBLAST/blast/all_Genomes_masked \
  -out ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Bacteria_02.txt \
  -evalue 0.01 -word_size 16
```

We also blast against a phage Reference database. 

```{bash}
##---------------------------------------------------------------
##--------------------------blast to viral DB
#this is the vContact database
##---------------------------------------------------------------

makeblastdb -max_file_sz 1GB -in /archiv/Phage_DB/BLAST/20190519/Vcontact/Database/alltogether/data/merged_all_phages.fna \
 -out /archiv/Phage_DB/BLAST/20190519/Vcontact/Database/alltogether/data/merged_all_phages -parse_seqids -dbtype nucl


  blastn -num_threads 35 -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query ${mappingFile} \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen sgi sacc sblastnames staxids sscinames" \
  -db /archiv/Phage_DB/BLAST/20190519/Vcontact/Database/alltogether/data/merged_all_phages \
  -out ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_ViralDB.txt \
  -evalue 0.01 -word_size 16
  


##--------------------------blast to old viral DB
#this is the RefSeq database

  blastn -num_threads 35 -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query ${mappingFile} \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen sgi sacc sblastnames staxids sscinames" \
  -db /archiv/BLAST/blast_db/20190204/viral_nucleotide/viral \
  -out ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_ViralDB_old.txt \
  -evalue 0.01 -word_size 16
  
  
  
##--------------------------blast unmapped to nucleotide DB


blastn -num_threads 37 -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query ${mappingFile} \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen" \
  -db /archiv/BLAST/blast_db/20190204/nucleotide/prokaryotes \
  -out  ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_nucleotideDB.txt \
  -evalue 0.01 -word_size 16




##-----------------------merge all blasts


 # cut -f 1,2 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages.txt |sed 's/_contig[[:digit:]]\+//g'  | awk -F "[\t]" '{OFS="\t"} {print "localPHAGE",$1,$2}' > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/all_differentBlastHits.txt

#  cut -f 1,2 ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages.txt | awk -F "[\t]" '{OFS="\t"} {print "localPHAGE",$1,$2}' > ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits.txt

  awk -F "\t" '{OFS="\t"} {print $1 ,$2 }' ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Bacteria_02.txt | awk -F "[\t]" '{OFS="\t"} {print "localBAC",$1,$2}' >>  ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits.txt


  cut -f 1,13 ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_ViralDB.txt |cut -d ',' -f 1 | sed 's/ complete prophage genome//g' |sed 's/ complete genome//g'  | awk -F "[\t]" '{OFS="\t"} {print "phageDB",$1,$2}' >>  ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits.txt

cut -f 1,13 ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_nucleotideDB.txt |cut -d ',' -f 1 | awk -F "[\t]" '{OFS="\t"} {print "BactDB",$1,$2}' >>  ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits.txt

cut -f 1 ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits.txt | sort|uniq -c
#cut -f 1 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/all_differentBlastHits.txt|sort|uniq -c
```



### map meta

Here we map all raw reads against the Pan-CRISPR-array and follow the mapping coverage over time.

What we do:
1. extend the spacer with the appropiate spacer 
2. make a bwa index all all pan-CRISPR-arrays
3. map all timepoints against the pan-CRISPR-array

```{bash}
###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples_withEvolution.txt  ##the file with all sample names
BaseLocation=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract/
species=Sterm
repeatsFile=/home/vincent/Projects/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt
MetaspacerFile=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers/ForDADA2_woREverse//all_unique_spacers.fasta
mappingFile=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed_ClusterName.txt
BaseLocation_mapping=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//bwaMapping
metaSpacerFile=${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/uniqueSpacers_count_Ref_wOLDstrains.txt #inlcudes in column four the name of array



  
 # ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable_final.fasta
  
  
##=============================================
##-----------create mappable and blastable file
##=============================================
  
  
#scp ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/ForDADA2/spacersAssigned.txt vincent@130.223.51.116:/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//FinalSpacers/ForDADA2/


##--------------------------------------------------
##--------------------------------------------------
rm ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable.fasta
rm ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable.bed
rm ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable_final.fasta
#rm ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_blastable_final.fasta
mkdir -p ${BaseLocation}/FinalSpacers/cdHit_spacers/

for spacerss in $(grep "^META_" ${metaSpacerFile} |cut -f 1)
do


#grep ">" ${BaseLocation}/FinalSpacers/cdHit_spacers/final/clusterCRISPRs_allArrays_keeps.txt|cut -d ':' -f 2|cut -d 's' -f 1|sed 's/0//g' |sed 's/R//g' | sed 's/a/Array/g'
adap=$(grep -P "${spacerss}\t" ${metaSpacerFile} |cut -f 4|sed 's/a/Array/g' )
adapSeq=$(grep "${adap}" -A 1 /home/vincent/Projects/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt|tail -1)
startSpacer=$(echo $adapSeq|wc -c)


echo "=========================="
echo $adap
echo $adapSeq
echo "=========================="



echo $spacerss

seq=$(grep "$spacerss" -A 1 ${mappingFile} |tail -1)

echo -e ">"$spacerss"\n"$adapSeq"\n"$seq"\n"$adapSeq >> ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable.fasta


LENGTHpacer=$(echo $seq|wc -c)
endspacer=$((startSpacer+LENGTHpacer))
spacershort=$(echo $spacerss |sed 's/>//g')


echo -e $spacershort"\t"$startSpacer"\t"$endspacer"\t"$spacershort >> ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable.bed



done #spacerss (spacer)


fasta_formatter -w 50 -i ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable.fasta -o ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable_final.fasta

## Qualtiy control

grep ">" -c ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable_final.fasta
grep ">" ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable_final.fasta | cut -d ':' -f 1  |sort|uniq -c
grep ">" ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable_final.fasta | cut -d ':' -f 2  |sort|uniq -c
grep ">" -A 1 ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable.fasta |grep ">" -v|grep "-" -v|sort|uniq -c

  mappingFile=${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable_final.fasta
    ##=============================================
  ##mapping to reference
    ##=============================================
  ##-----------------------------------------map rawReads
  rm -r $BaseLocation_mapping
  mkdir -p $BaseLocation_mapping

  bwa index $mappingFile
  # name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 
  
  mkdir -p ${logFilelocation}
  rm ${BaseLocation_mapping}/log/MappingQualityLength.txt
  
  num=1
  rm ${BaseLocation_mapping}/log/flagstat_logfile_CRISPRmappingGenome.txt
  rm ${BaseLocation_mapping}/log/multiqc_logfile_CRISPRmappingGenome.txt
  
  names=G4_6_18
r1=/home/vincent/Projects/2020_StarterEvolution/01_data/20200929_Novogene/X204SC20090774-Z01-F001/trimm_Galore/gz/${names}/${names}-R1_val_1.fq.gz
r2=/home/vincent/Projects/2020_StarterEvolution/01_data/20200929_Novogene/X204SC20090774-Z01-F001/trimm_Galore/gz/${names}/${names}-R2_val_2.fq.gz



cat ${BaseLocation_mapping}/final/CRISPR_spacer_coverage.bed > ${BaseLocation_mapping}/final/CRISPR_spacer_coverage_backup.bed
cat ${BaseLocation_mapping}/final/CRISPR_spacer_coverage_protospacer.bed > ${BaseLocation_mapping}/final/CRISPR_spacer_coverage_protospacer_backup.bed

awk -F "\t" '{OFS="\t"}{if($9!="G4_6_18") print $0}'  ${BaseLocation_mapping}/final/CRISPR_spacer_coverage_protospacer_backup.bed > ${BaseLocation_mapping}/final/CRISPR_spacer_coverage_protospacer.bed
awk -F "\t" '{OFS="\t"}{if($9!="G4_6_18") print $0}'  ${BaseLocation_mapping}/final/CRISPR_spacer_coverage_backup.bed > ${BaseLocation_mapping}/final/CRISPR_spacer_coverage.bed


  for names in $(cat ${samplesss})
    do
    echo ${num}"/16  :" ${names}
    num=$((num+1))
    rm -r ${BaseLocation_mapping}/${names}/bwaMapping2DB/
  
    mkdir -p ${BaseLocation_mapping}/${names}/bwaMapping2DB/
    
    ##------only for g4_6_18 because the bam file is corrupted
    #names=G4_6_18
    #bwa mem -t 37 $mappingFile \
    #/home/vincent/Projects/2020_StarterEvolution/01_data/20200929_Novogene/X204SC20090774-Z01-F001/trimm_Galore/gz/${names}/${names}-R1_val_1.fq /home/vincent/Projects/2020_StarterEvolution/01_data/20200929_Novogene/X204SC20090774-Z01-F001/trimm_Galore/gz/${names}/${names}-R2_val_2.fq | samtools sort -@${threads} -O BAM -o ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 
  
    
    bwa mem -t ${threads} $mappingFile \
    /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_1.fastq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_2.fastq | samtools sort -@${threads} -O BAM -o ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 
  
  samtools view -b -F 4 ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped.bam
  
  ##-------------------coverage
  
  
    samtools view -h -q 8 -b ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped.bam > ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30.bam
   
  ##----take only mappings longer than 42 
  
  mkdir -p ${BaseLocation_mapping}/log/
  
  samtools view ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30.bam |cut -f 13 |sed 's/MD:Z://g'  | awk -F "[CTAG]" '{print $1+$2+$3+$4+$5+$6+$7+$8}' |sort -rn |uniq -c | sed -e 's/^[[:space:]]*//' |awk -F " " -v var="$names" '{OFS=" "} {print $0,var}' >> ${BaseLocation_mapping}/log/MappingQualityLength.txt
  
  
  #samtools view ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped.bam |grep "M02102:390:000000000-C7G6T:1:1102:13014:15583:N:0:GGACTCCT+GTAAGGAG#0"
  
  ###-----------------ONLY spacer mapped reads --> reads mapping to protospacer
    cd ${BaseLocation_mapping}/${names}/bwaMapping2DB/

  samtools view ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30.bam > ${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub.sam

  /home/vincent/scripts/mappingLengthfiltering_protospacer.R -i ${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub.sam -o ${names}_rawIllumina_bwamem_sorted_mapped_protospacer.sam

  
  samtools view -H ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped.bam  > ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_header.sam  # extract header only
  
  awk '{OFS="\t"} {print $1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14}' ${names}_rawIllumina_bwamem_sorted_mapped_protospacer.sam > ${names}_rawIllumina_bwamem_sorted_mapped_protospacer_02.sam
  cat ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_header.sam \
   ${names}_rawIllumina_bwamem_sorted_mapped_protospacer_02.sam |samtools view -S -b > ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_protospacer_final.bam
  
  
  bedtools coverage -bed -a ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable.bed -b ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_protospacer_final.bam  >  ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_protospacer_coverage.bed 
    
    mkdir -p ${BaseLocation_mapping}/final
    
  awk -F "\t" -v samplessss="$names" '{OFS="\t"} {print $0,samplessss}' ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_protospacer_coverage.bed   >> ${BaseLocation_mapping}/final/CRISPR_spacer_coverage_protospacer.bed
  


  ###-----------------properly mapped reads --> reads mapping to CRISPR SPACER
  cd ${BaseLocation_mapping}/${names}/bwaMapping2DB/
  
  /home/vincent/scripts/mappingLengthfiltering.R -i ${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub.sam -o ${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub_out.sam
  rm ${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub.sam
  
  #samtools view ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30.bam |cut -f 13 |sed 's/MD:Z://g'  | awk -F "[CTAG]" '{if($1+$2+$3+$4+$5+$6+$7+$8>42) print $0}' |sort -rn |uniq -c | sed -e 's/^[[:space:]]*//' |awk -F " " -v var="$names" '{OFS=" "} {print $0,var}' >> ${BaseLocation_mapping}/log/MappingQualityLength.txt
  
  
  samtools view -H ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped.bam > ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_header.sam  # extract header only
  
  awk '{OFS="\t"} {print $1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14}' ${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub_out.sam > ${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub_out_02.sam
  cat ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_header.sam \
   ${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub_out_02.sam |samtools view -S -b > ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub_out_final.bam
  
  
  bedtools coverage -bed -a ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable.bed -b ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub_out_final.bam  >  ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_coverage.bed 
    
    mkdir -p ${BaseLocation_mapping}/final
    
  awk -F "\t" -v samplessss="$names" '{OFS="\t"} {print $0,samplessss}' ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_coverage.bed  >> ${BaseLocation_mapping}/final/CRISPR_spacer_coverage.bed
  
  ##-----Qualimap
  
  mkdir -p ${BaseLocation_mapping}/${names}/bwaMapping2DB//bamqc
  
  qualimap bamqc -bam ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir ${BaseLocation_mapping}/${names}/bwaMapping2DB//bamqc --java-mem-size=80G
  
  echo ${names} >>  ${BaseLocation_mapping}/log/flagstat_logfile_CRISPRmappingGenome.txt
  samtools flagstat ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam >>  ${BaseLocation_mapping}/log/flagstat_logfile_CRISPRmappingGenome.txt
  
  rm ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam &
   mkdir -p ${BaseLocation_mapping}/log
  echo ${BaseLocation_mapping}"/"${names}"/bwaMapping2DB/bamqc" >> ${BaseLocation_mapping}/log/multiqc_logfile_CRISPRmappingGenome.txt
  
  done
  
  
  
     rm -r ${BaseLocation}/log/MultiQC
   
    mkdir -p ${BaseLocation}/log/MultiQC
    
    /home/vincent/.local/bin/multiqc --config /archiv/logs/multiQC_config.txt --file-list  ${BaseLocation_mapping}/log/multiqc_logfile_CRISPRmappingGenome.txt \
      -o ${BaseLocation}/log/MultiQC

##-------------------control
 awk -F "\t" '{OFS="\t"}{if($9=="G4_6_18") print $8}'  ${BaseLocation_mapping}/final/CRISPR_spacer_coverage_protospacer.bed |sort|uniq -c


##-------------------calculate mapping reads for normalization
##in order to normalize I have to take the total number of raw reads. 
#this is used to do cpm
##-------------------

rm ${BaseLocation_mapping}/log/reads4normalization.txt
num=1
names=RMK202
for names in $(cat ${samplesss}|grep "_G4_" -v)
  do
  echo ${num}"/16  :" ${names}

#reads=$(grep "CP046134" /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/${names}/bwaMapping2DB/bamqc/genome_results.txt | sed -e 's/^[[:space:]]*//'|cut -f 3)

# names=G4_6_18
 #reads=$(wc -l /home/vincent/Projects/2020_StarterEvolution/01_data/20200929_Novogene/X204SC20090774-Z01-F001/trimm_Galore/gz/${names}/${names}-R1_val_1.fq|awk '{print $1/2}') 

reads=$(wc -l /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_1.fastq|awk '{print $1/2}') 

#reads=$(samtools view ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub_out_final.bam |wc -l)

echo -e ${names}"\t"${reads}"\tStreptococcus thermophilus"  >> ${BaseLocation_mapping}/log/reads4normalization.txt

done
echo ${BaseLocation_mapping}/final/CRISPR_spacer_coverage.bed
echo ${BaseLocation_mapping}/log/reads4normalization.txt
```

##Vcontact with metagenome

Here, I want to add my metagenomes to the current vContact analysis.
First we try to add the entire metagenome at once. (next time add contig by contig)

First we have to do the Phage annotation. For this project I did it with Phaster. 


```{bash}
##================
##files
##================

date=20190519
species=Sterm
date_new=20200424

mkdir -p /archiv/Phage_DB/BLAST/${date_new}/downloads/${species}_proteins/Renamed/merged/

cat  /archiv/Phage_DB/BLAST/${date}/downloads/${species}_proteins/Renamed/merged/${species}_phage_protein_db_gene2genome.txt > \
/archiv/Phage_DB/BLAST/${date_new}/downloads/${species}_proteins/Renamed/merged/${species}_phage_protein_db_gene2genome_oldsamples.txt


mkdir -p /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/data

###----------------------
##add phages
###----------------------
rm -r /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/ForVContact2/
mkdir -p /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/ForVContact2/
for phageesss in $(cut -f 1 /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/gene_CLEANED.gff|sort|uniq )
do

echo -e "wokring on: "${phageesss}
echo "..."

###----faa
#grep "," /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/gene_CLEANED.gff

sed 's/,//g' /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/gene_CLEANED.gff |grep "${phageesss}" | cut -f 9 | awk -F ";locusTag= " -v Genomesss="$phageesss" '{OFS=","}{print $2,"RMK202_"Genomesss,$1}'  >>  /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/ForVContact2/RMK202_phage_protein_db_gene2genome.txt

cat /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/faa_for_virfam/${phageesss}.faa >>  /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/ForVContact2/all_Genes_merged.faa

done

##---------------------------
##merge with existing data
##---------------------------

mkdir -p /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/data/

  cat /archiv/Phage_DB/BLAST/${date}/Vcontact/Database/alltogether/data/all_phage_protein_db_gene2genome.csv \
  /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/ForVContact2/RMK202_phage_protein_db_gene2genome.txt > \
  /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/data/all_phage_protein_db_gene2genome.csv
  
##prep fasta  


cat /archiv/Phage_DB/BLAST/${date}/Vcontact/Database/alltogether/data/merged_all_phages.faa \
/archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/ForVContact2/all_Genes_merged.faa > /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/data/merged_all_phages.faa
  
##================
##run vcontact
##================

##before running activate environoment
/home/vincent/anaconda3/bin/conda init  bash
conda activate /home/vincent/miniconda2/envs/vContact2

##if it doesnt work just try another screen ...has been playing up alot

cd /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/data

#rm -r /archiv/Phage_DB/BLAST/${date}/Vcontact/Database/alltogether/database
rm -r /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/database
mkdir -p /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/database

/home/vincent/apps/vcontact2/bin/vcontact --raw-proteins merged_all_phages.faa --rel-mode Diamond --proteins-fp all_phage_protein_db_gene2genome.csv --db ProkaryoticViralRefSeq88-Merged --pcs-mode MCL --vcs-mode ClusterONE --c1-bin /home/vincent/apps/cluster_one-1.0.jar -t 30 --output-dir /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/database


###===========
##move important files to one folder
###===========
mkdir -p /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/database/allPhages_vContact
cd /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/database/

cp c1.clusters allPhages_vContact/
cp c1.ntw allPhages_vContact/
cp genome_by_genome_overview.csv allPhages_vContact/
cp modules.ntwk allPhages_vContact/
cp viral_cluster_overview.csv allPhages_vContact/
```



```{bash}
##================
##files
##================

date=20190519
species=Ldel
date_new=20201106

mkdir -p /archiv/Phage_DB/BLAST/${date_new}/downloads/${species}_proteins/Renamed/merged/

cat  /archiv/Phage_DB/BLAST/${date}/downloads/${species}_proteins/Renamed/merged/${species}_phage_protein_db_gene2genome.txt > \
/archiv/Phage_DB/BLAST/${date_new}/downloads/${species}_proteins/Renamed/merged/${species}_phage_protein_db_gene2genome_oldsamples.txt


mkdir -p /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/data

###----------------------
##add phages
###----------------------
rm -r /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/${date_new}_ForVContact2/
mkdir -p /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/${date_new}_ForVContact2/
#for phageesss in $(cut -f 1 /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/gene_CLEANED.gff|sort|uniq )
for phageesss in $(echo "Lactobacillus_phage_1")
do

echo -e "wokring on: "${phageesss}
echo "..."

###----faa
#grep "," /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/gene_CLEANED.gff

sed 's/,//g' /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/gene_CLEANED.gff |grep "${phageesss}" | cut -f 9 | awk -F ";locusTag= " -v Genomesss="$phageesss" '{OFS=","}{print $2,"RMK202_"Genomesss,$1}'  >>  /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/${date_new}_ForVContact2/RMK202_phage_protein_db_gene2genome.txt

cat /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/faa_for_virfam/${phageesss}.faa >>  /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/${date_new}_ForVContact2/all_Genes_merged.faa

done

##---------------------------
##merge with existing data
##---------------------------

mkdir -p /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/data/

  cat /archiv/Phage_DB/BLAST/${date}/Vcontact/Database/alltogether/data/all_phage_protein_db_gene2genome.csv \
  /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/${date_new}_ForVContact2/RMK202_phage_protein_db_gene2genome.txt > \
  /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/data/all_phage_protein_db_gene2genome.csv
  
##prep fasta  


cat /archiv/Phage_DB/BLAST/${date}/Vcontact/Database/alltogether/data/merged_all_phages.faa \
/archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/${date_new}_ForVContact2/all_Genes_merged.faa > /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/data/merged_all_phages.faa
  
##================
##run vcontact
##================

##before running activate environoment
/home/vincent/anaconda3/bin/conda init  bash
conda activate /home/vincent/miniconda2/envs/vContact2

##if it doesnt work just try another screen ...has been playing up alot

cd /archiv/Phage_DB/BLAST/20200424/Vcontact/Database/alltogether/data

#rm -r /archiv/Phage_DB/BLAST/${date}/Vcontact/Database/alltogether/database
rm -r /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/database
mkdir -p /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/database

/home/vincent/apps/vcontact2/bin/vcontact --raw-proteins merged_all_phages.faa --rel-mode Diamond --proteins-fp all_phage_protein_db_gene2genome.csv --db ProkaryoticViralRefSeq88-Merged --pcs-mode MCL --vcs-mode ClusterONE --c1-bin /home/vincent/apps/cluster_one-1.0.jar -t 30 --output-dir /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/database


###===========
##move important files to one folder
###===========
mkdir -p /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/database/allPhages_vContact
cd /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/database/

cp c1.clusters allPhages_vContact/
cp c1.ntw allPhages_vContact/
cp genome_by_genome_overview.csv allPhages_vContact/
cp modules.ntwk allPhages_vContact/
cp viral_cluster_overview.csv allPhages_vContact/
```


#### Vcontact cluster analysis
Here, we do the clusteranylasis of the Vcontact output
##### Ldel
```{r}


#---------------
##import data
#---------------
###---------All phages

Sterm_cluster <- read_table2("02_data/c1_ldel.clusters", 
    col_names = c("from","to","power"))
Phages_names <- read_csv("02_data/genome_by_genome_overview_ldel.csv")
Sterm_clustering_complete <- read_csv("02_data/genome_by_genome_overview_ldel.csv")
Sterm_clustering_new <- read_csv("02_data/c1_ldel.clusters") %>% as.data.frame()


Sterm_clustering_forFiltering_new <- Sterm_clustering_new  %>% filter(., grepl('Lactobacillus_phage_1', Members,ignore.case = TRUE))

Sterm_clustering_unnested <- Sterm_clustering_forFiltering_new %>% 
    mutate(Members = strsplit(as.character(Members), " ")) %>% 
    unnest(Members)

str_split_fixed(Sterm_clustering_unnested$Members, "~", 3)[,1]  %>% table()


###---------------1. filter:
##clusters that contain a Streptococcus


nrow(Sterm_clustering_unnested)
nrow(Sterm_cluster)
Sterm_cluster_sub <- Sterm_cluster[which(Sterm_cluster$from %in% Sterm_clustering_unnested$Members | Sterm_cluster$to %in% Sterm_clustering_unnested$Members),]
nrow(Sterm_cluster_sub)
# Sterm_cluster_sub <- Sterm_cluster_sub[which(Sterm_cluster_sub$to %in% Sterm_clustering_unnested$Members),]
# nrow(Sterm_cluster_sub)

###---------------2. filter:
##connections that have a higher qualtiy than 50 (decided on distribution plot)

ggplot(Sterm_cluster,aes(x=power))+geom_histogram(binwidth=1)
Sterm_cluster_sub <- Sterm_cluster_sub[Sterm_cluster_sub$power>1,]
nrow(Sterm_cluster_sub)

###---------------3. filter:
##cluster containing more than 2 phages 

Sterm_cluster_sub_uniques <- unique(c(Sterm_cluster_sub$from,Sterm_cluster_sub$to))

Sterm_clustering_unnested_which <- 
  Sterm_clustering_complete[which(Sterm_clustering_complete$Genome %in% Sterm_cluster_sub_uniques),] %>% select(VC) %>% table( )
  # select_if(~n_distinct(.) > 2 & is.numeric(.)) %>% 
  names(Sterm_clustering_unnested_which)
  Sterm_clustering_unnested_which_keep <- Sterm_clustering_unnested_which[which(Sterm_clustering_unnested_which>1)] %>% names()
  Sterm_clustering_unnested_which_keep_genome <- Sterm_clustering_complete[which(Sterm_clustering_complete$VC %in% Sterm_clustering_unnested_which_keep),"Genome"] %>% as.data.frame()
  
  
  # sum(Sterm_cluster_sub$from %in% Sterm_clustering_unnested_which_keep_genome$Genome) 
  nrow(Sterm_cluster_sub)
  Sterm_cluster_sub <- Sterm_cluster_sub[which(Sterm_cluster_sub$from %in% Sterm_clustering_unnested_which_keep_genome$Genome|Sterm_cluster_sub$to %in% Sterm_clustering_unnested_which_keep_genome$Genome),]
  nrow(Sterm_cluster_sub)
  
  unique(Sterm_cluster_sub$from)
# own_cluster_sub_new<- Sterm_cluster_sub
 

 ##===============================
 #old plots
  ##===============================
#---------------
##plot interaction
#---------------
ggplot(Sterm_cluster,aes(x=power))+geom_histogram(binwidth=1)
dim(Sterm_cluster_sub)
# 
#---------------
##creat graph
#---------------
# write_csv(Sterm_cluster_sub,path="~/Desktop/Projects/2019_PhageDB/Vcontact//allPhages_vContact/Streptococcus_phage_betwork_connections.txt",col_names = TRUE)

write_csv(Sterm_cluster_sub,path="03_results/Ldel_phage_betwork_connections.txt",col_names = TRUE)

Sterm_net <- graph_from_data_frame(Sterm_cluster_sub,direct=F)
# Ldel_net <- graph_from_data_frame(Ldel_cluster_sub,direct=F)
# Llac_net <- graph_from_data_frame(Llac_cluster_sub,direct=F)

# dim(Sterm_net)

##==================================
 ##new analysis
  ##==================================
 library(ggnetwork)
 library(ITNr)
 library(intergraph)
 
 net<-asNetwork(Sterm_net) #ggnetwork requires a network object
 n<-ggnetwork(net)

 

###--------
##coloring
#---------------
# n[grep("RMK202_",n$vertex.names),"vertex.names"] %>% table()
 n$color <- as.character("Viral DB")
 n$color[grep("RMK202_",n$vertex.names)] <- "RMK202 phages"

 
 
 n$color = factor(n$color, levels=c("RMK202 phages","Viral DB"))
 colorzzz <- c("red","grey50")
 size <- c("4","1.5")
 ##------------------
 ##extend
 ##------------------
 
 nExtended <-  merge(n, Phages_names, by.y="Genome",by.x="vertex.names",all.x = TRUE)
 # nrow(n)
 # nExtended <-  merge(n, Phages_names, by.y="Genome",by.x="vertex.names",all.x = TRUE) 
 # nrow(nExtended)
 # nExtended <- n
 # ccolnames(nExtended)
colnames(nExtended)
colnames(nExtended) <- revalue(colnames(nExtended), c("Adj P-value"="Adj_P_value","Genera in VC"="Genera_in_VC","Topology Confidence Score"="Topology_Confidence_Score","Genus Confidence Score"="Genus_Confidence_Score"))

# nExtended$ad
 ##------------------
 ##plot
 ##------------------
 networkplot <- ggplot(nExtended, aes(x = x, y = y, xend = xend, yend = yend,text =paste("genome:", vertex.names,"\npower: ",power,"\nOrder: ",Order,"\nFamily: ",Family,"\nGenus: ",Genus,"\nVC: ",VC,"\nSize: ",Size,"\nAdj P-value: ",Adj_P_value,"\nGenera in VC: ",Genera_in_VC,"\nGenus_Confidence_ScoreC: ",Genus_Confidence_Score,"\nTopology_Confidence_Score: ",Topology_Confidence_Score))) + geom_edges(color = "lightgray") +
   geom_nodes(aes(color = as.factor(color),size=as.factor(color))) +
   # scale_fill_manual(values="gray41")+
    scale_color_manual(values=colorzzz)+
    scale_size_manual(values=as.numeric(size))+
   # guides(color=FALSE,size=FALSE)+ #Remove legends
 theme_blank()+theme(legend.title = element_blank())
 networkplot
 
  svg("03_results/networkPlot_subset_ldel_02.svg",width=7,height=7)
 # png("~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03//allPhages_vContact//networkPlot_subset.png", width = 2800, height = 2800,res=300)
  networkplot
   dev.off()
 
 library(plotly)
 ggp <- ggplotly(networkplot)
 htmlwidgets::saveWidget(ggp, "03_results/allPhages_vContact//networkPlot_ldel.html")
 # View(nExtended)
 
tmp <-   nExtended %>% filter(color=="RMK202 phages") 
table(tmp$`VC Subcluster`)
 
 
 nExtended %>% filter(color!="Viral DB") %>% filter()
 sort(table(nExtended$VC),decreasing = TRUE)
 
 count_perViralCluster <- nExtended %>%
  group_by(VC,color) %>% 
  dplyr::summarise(count=n()) %>% spread(.,color,count)
 
  ##==================================
 ##keep only phages in our s. thermophilus phage cluster
  ##==================================
 ###---------------------identify phages in our cluster

 
# table(nExtended$VC) 

tmp_phages <- nExtended[grep("RMK202_",nExtended$vertex.names),] 
ourPhageCluster <- c(unique(tmp_phages$VC))
nExtended_ourphages <- nExtended %>% filter(VC %in% ourPhageCluster)

phagesinClusters <- Phages_names %>% filter(VC %in% ourPhageCluster)
phagesinClusters$Genome

###---------------------clustering
# Sterm_clustering_new <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/c1.clusters") %>% as.data.frame()
Sterm_clustering_new <- read_csv("02_data/c1_ldel.clusters") %>% as.data.frame()
Sterm_clustering_forFiltering_new <- Sterm_clustering_new  %>% filter(., grepl('Streptococcus', Members,ignore.case = TRUE))

Sterm_clustering_unnested <- Sterm_clustering_forFiltering_new %>% 
    mutate(Members = strsplit(as.character(Members), " ")) %>% 
    unnest(Members)

# str_split_fixed(Sterm_clustering_unnested$Members, "~", 3)[,1]  %>% table()


###---------------1. filter:
##clusters that contain a Streptococcus


# nrow(Sterm_clustering_unnested)
nrow(Sterm_cluster)
Sterm_cluster_sub <- Sterm_cluster[which(Sterm_cluster$from %in% phagesinClusters$Genome | Sterm_cluster$to %in% phagesinClusters$Genome),]
nrow(Sterm_cluster_sub)
Sterm_net_sub <- graph_from_data_frame(Sterm_cluster_sub,direct=F)

 

 net<-asNetwork(Sterm_net_sub) #ggnetwork requires a network object
 n<-ggnetwork(net)
  ##------------------
 ##color and size change
  ##------------------
 
 
###--------
##coloring
#---------------
# n[grep("RMK202_",n$vertex.names),"vertex.names"] %>% table()
 n$color <- as.character("Viral DB")
 n$color[grep("RMK202_",n$vertex.names)] <- "RMK202 phages"


 
 n$color = factor(n$color, levels=c("RMK202 phages","Viral DB"))
 colorzzz <- c("red","grey50")
 size <- c("4","1.5")
 
 
 # n$color = factor(n$color, levels=c("RMK202 phages","cos","987","Viral DB"))
 # colorzzz <- c("red","blue","#808000","grey50")
 # size <- c("4","2","2","1.5")
 
 table(n$color)
 
 n$color_new <- as.character("Viral DB")
n$color_new[grep("RMK202_Lactobacillus_phage_1",n$vertex.names)[1]] <- "RMK202 phages"
# n$color_new[grep("RMK202_Streptococcus_phage_2",n$vertex.names)[1]] <- "RMK202 phages"


 ##------------------
 ##extend
 ##------------------
 
 nExtended <-  merge(n, Phages_names, by.y="Genome",by.x="vertex.names",all.x = TRUE)
 # nrow(n)
 # nExtended <-  merge(n, Phages_names, by.y="Genome",by.x="vertex.names",all.x = TRUE) 
 # nrow(nExtended)
 # nExtended <- n
 # ccolnames(nExtended)
colnames(nExtended)
colnames(nExtended) <- revalue(colnames(nExtended), c("Adj P-value"="Adj_P_value","Genera in VC"="Genera_in_VC","Topology Confidence Score"="Topology_Confidence_Score","Genus Confidence Score"="Genus_Confidence_Score"))



 nExtended$alpha <- as.character("Viral DB")
nExtended$alpha[nExtended$VC %in% ourPhageCluster] <- "RMK202 phages"
table(nExtended$alpha)
  alphaCol <- c(1,0.03)
  
tmp <- nExtended[nExtended$alpha=="RMK202 phages",]
table(tmp$VC)
  
nExtended$fillings <- as.character("Viral DB")
# nExtended$fillings[nExtended$VC %in% ourPhageCluster[1]] <- ourPhageCluster[1]
# nExtended$fillings[nExtended$VC %in% ourPhageCluster[2]] <- ourPhageCluster[2]
# # nExtended$fillings[nExtended$vertex.names %in% Mahony_genomes_cos$namez] <- "cos"
# nExtended$fillings[nExtended$vertex.names %in% Mahony_genomes_987$namez] <- "987"
# nExtended$fillings[nExtended$vertex.names=="RMK202_Streptococcus_phage_1"] <- "Streptococcus_phage_1"
# nExtended$fillings[nExtended$vertex.names=="RMK202_Streptococcus_phage_2"] <- "Streptococcus_phage_2"
as.factor(nExtended$fillings)
# library("RColorBrewer")

# three_reds <- c(brewer.pal(9,name="YlOrRd")[c(4,6)],
# 							 brewer.pal(3,name="Reds")[3])
# three_browns <- brewer.pal(9,name="YlOrBr")[c(7,8,9)]
# three_blues <- brewer.pal(9,name="YlGnBu")[c(5,6,8)]

filling <- c("red")
# filling <- c(three_reds[1],three_browns[1],three_reds[2],three_browns[2],three_reds[3],three_browns[3],"grey")

# filling <- c("black","red","darkgrey","darkred","grey")
# table(nExtended$alpha)

  



# nExtended$ad
 ##------------------
 ##plot
 ##------------------
table(nExtended$alpha)
table(nExtended$power)
# nExtended$Family
 networkplot_sub <- ggplot(nExtended, aes(x = x, y = y, xend = xend, yend = yend,text =paste("genome:", vertex.names,"\npower: ",power,"\nOrder: ",Order,"\nFamily: ",Family,"\nGenus: ",Genus,"\nVC: ",VC,"\nSize: ",Size,"\nAdj P-value: ",Adj_P_value,"\nGenera in VC: ",Genera_in_VC,"\nGenus_Confidence_ScoreC: ",Genus_Confidence_Score,"\nTopology_Confidence_Score: ",Topology_Confidence_Score))) + geom_edges(color = "lightgray",aes(size=power),alpha=1) +
   geom_nodes(aes(color = as.factor(fillings),alpha=as.factor(alpha))) +
   # scale_fill_manual(values="gray41")+
    scale_color_manual(values=filling)+
     scale_alpha_manual(values=alphaCol)+
    scale_fill_manual(values=colorzzz)+
   geom_nodelabel_repel(aes(label = vertex.names),
                       box.padding = unit(1, "lines"),
                       data = function(x) { x[ x$color_new == "RMK202 phages", ]})+
    # scale_size_manual(values=as.numeric(c(2,1.25)))+
   # guides(color=FALSE,size=FALSE)+ #Remove legends
 theme_blank()+theme(legend.title = element_blank())
 networkplot_sub
 table(nExtended$color_new)
 # RMK202 phages
#  x[grep("RMK202_",x$vertex.names),]
#  n$color[grep("RMK202_",n$vertex.names)]
 svg("03_results/networkPlot_subset_ldel.svg",width=7,height=7)
 # png("~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03//allPhages_vContact//networkPlot_subset.png", width = 2800, height = 2800,res=300)
  networkplot_sub
   dev.off()
 
 library(plotly)
 # ggp <- ggplotly(networkplot_sub)
 # htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03//allPhages_vContact//networkPlot_subset.html")
 # View(nExtended)


 ##==================================
 ##old analysis
  ##==================================



###--------
##grep fo Mahony phages
#---------------
Mahony_genomes$namez <- paste(Mahony_genomes$Genus,Mahony_genomes$phage,Mahony_genomes$strain,sep="~")
Mahony_genomes_cos <- Mahony_genomes[which(Mahony_genomes$Type=="cos"),]
Mahony_genomes_pac <- Mahony_genomes[which(Mahony_genomes$Type=="pac"),]
Mahony_genomes_987 <- Mahony_genomes[which(Mahony_genomes$Type=="987"),]
Mahony_genomes_5093 <- Mahony_genomes[which(Mahony_genomes$Type=="5093"),]

##colour

vcolour_Sterm <- rep("grey",length(V(Sterm_net)$name))
# vcolour_Sterm_ori <- V(Sterm_net)$name
# vcolour_Sterm <- "grey"
vcolour_Sterm[which(V(Sterm_net)$name %in% paste(Mahony_genomes_cos$Genus,Mahony_genomes_cos$phage,Mahony_genomes_cos$strain,sep="~"))] <- "green"
vcolour_Sterm[which(V(Sterm_net)$name %in% paste(Mahony_genomes_pac$Genus,Mahony_genomes_pac$phage,Mahony_genomes_pac$strain,sep="~"))] <- "blue"
vcolour_Sterm[which(V(Sterm_net)$name %in% paste(Mahony_genomes_987$Genus,Mahony_genomes_987$phage,Mahony_genomes_987$strain,sep="~"))] <- "pink"
vcolour_Sterm[which(V(Sterm_net)$name %in% paste(Mahony_genomes_5093$Genus,Mahony_genomes_5093$phage,Mahony_genomes_5093$strain,sep="~"))] <- "brown"

# vcolour_Sterm[which(!vcolour_Sterm_ori %in% paste(Mahony_genomes$Genus,Mahony_genomes$phage,Mahony_genomes$strain,sep="~"))] <- "grey"
table(vcolour_Sterm)

V(Sterm_net)$color <- vcolour_Sterm
table(V(Sterm_net)$color)
# V(Sterm_net)$color[which(V(Sterm_net)$name=='Streptococcus~phage~SW19')] <- "grey"
# V(Sterm_net)$color[which(V(Sterm_net)$name=='Streptococcus~phage~SW24')] <- "grey"
##size

vsize_Sterm <- V(Sterm_net)$name
vsize_Sterm_ori <- V(Sterm_net)$name

grep("S_t",V(Sterm_net)$name)
vsize_Sterm[which(!vsize_Sterm_ori %in% paste(Mahony_genomes$Genus,Mahony_genomes$phage,Mahony_genomes$strain,sep="~"))] <- "1.5"
vsize_Sterm[which(vsize_Sterm_ori %in% paste(Mahony_genomes$Genus,Mahony_genomes$phage,Mahony_genomes$strain,sep="~"))] <- "5"

table(vsize_Sterm)

vsize_Sterm <- as.numeric(vsize_Sterm)

#---------------
#plot
#---------------

set.seed(8051)

 V(Sterm_net)$name[grep("RMK202",V(Sterm_net)$name)]
 length( V(Sterm_net)$name)
 # V(Llac_net)$name
 # V(Ldel_net)$name
 # length(V(Ldel_net)$name)


# create vertices
fg_lay_Sterm <- layout_with_fr(Sterm_net,dim=2,niter=99)
# fg_lay_Ldel <- layout_with_fr(Ldel_net,dim=2,niter=99)
# fg_lay_Llac <- layout_with_fr(Llac_net,dim=2,niter=99)

# plot  Sterm
 # plot(Sterm_net,vertex.label=NA,layout=fg_lay_Sterm,
 #     vertex.size=vsize_Sterm,
 #       main="Sterm phages network"
 #     )
     # png("~/Desktop/Projects/2019_RMK202_analysis/plot/vContact_sterm.png", width = 1800, height = 1200,res=300)

 plot(Sterm_net,vertex.label=NA,layout=fg_lay_Sterm,vertex.size=vsize_Sterm,
       main="Sterm phages network"
     )
 legend("topright", 
       legend=c("cos","pac","987","5093","unknown"), pch=19,
      col=c("green","blue","pink","brown","grey"), bty="n")

```


##### Sterm

```{r}

#---------------
##import data
#---------------
Sterm_cluster <- read_table2("02_data/c1_ldel.clusters", 
    col_names = c("from","to","power"))
Phages_names <- read_csv("02_data/genome_by_genome_overview_ldel.csv")
Sterm_clustering_complete <- read_csv("02_data/genome_by_genome_overview_ldel.csv")
Sterm_clustering_new <- read_csv("02_data/c1_ldel.clusters") %>% as.data.frame()


Mahony_genomes <- read_csv("02_data/Mahony_table_ncbi_genomes.txt",
    col_names = c("Genus","phage","strain","NCBI","Type"))

#---------------
##reduce to only Strep clusters
#---------------

#----keep cluster that contain one sterm!!

# Sterm_clustering_complete <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03///allPhages_vContact/genome_by_genome_overview.csv")
# Sterm_clustering_complete <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/genome_by_genome_overview.csv")


Sterm_clustering_forFiltering_new <- Sterm_clustering_new  %>% filter(., grepl('Streptococcus', Members,ignore.case = TRUE))

Sterm_clustering_unnested <- Sterm_clustering_forFiltering_new %>% 
    mutate(Members = strsplit(as.character(Members), " ")) %>% 
    unnest(Members)

str_split_fixed(Sterm_clustering_unnested$Members, "~", 3)[,1]  %>% table()


###---------------1. filter:
##clusters that contain a Streptococcus


nrow(Sterm_clustering_unnested)
nrow(Sterm_cluster)
Sterm_cluster_sub <- Sterm_cluster[which(Sterm_cluster$from %in% Sterm_clustering_unnested$Members | Sterm_cluster$to %in% Sterm_clustering_unnested$Members),]
nrow(Sterm_cluster_sub)
# Sterm_cluster_sub <- Sterm_cluster_sub[which(Sterm_cluster_sub$to %in% Sterm_clustering_unnested$Members),]
# nrow(Sterm_cluster_sub)

###---------------2. filter:
##connections that have a higher qualtiy than 50 (decided on distribution plot)

ggplot(Sterm_cluster,aes(x=power))+geom_histogram(binwidth=1)
Sterm_cluster_sub <- Sterm_cluster_sub[Sterm_cluster_sub$power>4,]
nrow(Sterm_cluster_sub)

###---------------3. filter:
##cluster containing more than 2 phages 

Sterm_cluster_sub_uniques <- unique(c(Sterm_cluster_sub$from,Sterm_cluster_sub$to))

Sterm_clustering_unnested_which <- 
  Sterm_clustering_complete[which(Sterm_clustering_complete$Genome %in% Sterm_cluster_sub_uniques),] %>% select(VC) %>% table( )
  # select_if(~n_distinct(.) > 2 & is.numeric(.)) %>% 
  names(Sterm_clustering_unnested_which)
  Sterm_clustering_unnested_which_keep <- Sterm_clustering_unnested_which[which(Sterm_clustering_unnested_which>2)] %>% names()
  Sterm_clustering_unnested_which_keep_genome <- Sterm_clustering_complete[which(Sterm_clustering_complete$VC %in% Sterm_clustering_unnested_which_keep),"Genome"] %>% as.data.frame()
  
  
  # sum(Sterm_cluster_sub$from %in% Sterm_clustering_unnested_which_keep_genome$Genome) 
  nrow(Sterm_cluster_sub)
  Sterm_cluster_sub <- Sterm_cluster_sub[which(Sterm_cluster_sub$from %in% Sterm_clustering_unnested_which_keep_genome$Genome|Sterm_cluster_sub$to %in% Sterm_clustering_unnested_which_keep_genome$Genome),]
  nrow(Sterm_cluster_sub)
  
  unique(Sterm_cluster_sub$from)
# own_cluster_sub_new<- Sterm_cluster_sub
 
 ##===============================
 #old plots
  ##===============================
#---------------
##plot interaction
#---------------
ggplot(Sterm_cluster,aes(x=power))+geom_histogram(binwidth=1)
dim(Sterm_cluster_sub)
# 
# ggplot(Ldel_cluster_sub,aes(x=power))+geom_histogram(binwidth=1)
# dim(Ldel_cluster_sub)
# 
# ggplot(Llac_cluster_sub,aes(x=power))+geom_histogram(binwidth=1)
# dim(Llac_cluster_sub)

#---------------
##subset for only highest interactions
#---------------
#Sterm_cluster_sub <- Sterm_cluster_sub[Sterm_cluster_sub$power>1,]
#dim(Sterm_cluster_sub)

#Sterm_cluster_sub <- Sterm_cluster_sub[Sterm_cluster_sub$power>10,]
#dim(Sterm_cluster_sub)

# length(unique(Sterm_cluster_sub$from))

#---------------
##creat graph
#---------------
# write_csv(Sterm_cluster_sub,path="~/Desktop/Projects/2019_PhageDB/Vcontact//allPhages_vContact/Streptococcus_phage_betwork_connections.txt",col_names = TRUE)

write_csv(Sterm_cluster_sub,path="03_results/Streptococcus_phage_betwork_connections.txt",col_names = TRUE)

Sterm_net <- graph_from_data_frame(Sterm_cluster_sub,direct=F)
# Ldel_net <- graph_from_data_frame(Ldel_cluster_sub,direct=F)
# Llac_net <- graph_from_data_frame(Llac_cluster_sub,direct=F)

# dim(Sterm_net)

##==================================
 ##new analysis
  ##==================================
 library(ggnetwork)
 library(ITNr)
 library(intergraph)
 
 net<-asNetwork(Sterm_net) #ggnetwork requires a network object
 n<-ggnetwork(net)
  ##------------------
 ##color and size change
  ##------------------
 
 
###--------
##grep fo Mahony phages
#---------------
Mahony_genomes$namez <- paste(Mahony_genomes$Genus,Mahony_genomes$phage,Mahony_genomes$strain,sep="~")
Mahony_genomes_cos <- Mahony_genomes[which(Mahony_genomes$Type=="cos"),]
Mahony_genomes_pac <- Mahony_genomes[which(Mahony_genomes$Type=="pac"),]
Mahony_genomes_987 <- Mahony_genomes[which(Mahony_genomes$Type=="987"),]
Mahony_genomes_5093 <- Mahony_genomes[which(Mahony_genomes$Type=="5093"),]

###--------
##coloring
#---------------
# n[grep("RMK202_",n$vertex.names),"vertex.names"] %>% table()
 n$color <- as.character("Viral DB")
 n$color[grep("RMK202_",n$vertex.names)] <- "RMK202 phages"
 n$color[n$vertex.names %in% Mahony_genomes_cos] <- "cos"
 n$color[n$vertex.names %in% Mahony_genomes_pac] <- "pac"
 n$color[n$vertex.names %in% Mahony_genomes_987] <- "987"
 n$color[n$vertex.names %in% Mahony_genomes_5093] <- "5093"
 
 
 n$color = factor(n$color, levels=c("RMK202 phages","cos","pac","987","5093","Viral DB"))
 colorzzz <- c("red","blue","green","yellow","purple","grey50")
 size <- c("4","2","2","2","2","1.5")
 ##------------------
 ##extend
 ##------------------
 
 nExtended <-  merge(n, Phages_names, by.y="Genome",by.x="vertex.names",all.x = TRUE)
 # nrow(n)
 # nExtended <-  merge(n, Phages_names, by.y="Genome",by.x="vertex.names",all.x = TRUE) 
 # nrow(nExtended)
 # nExtended <- n
 # ccolnames(nExtended)
colnames(nExtended)
colnames(nExtended) <- revalue(colnames(nExtended), c("Adj P-value"="Adj_P_value","Genera in VC"="Genera_in_VC","Topology Confidence Score"="Topology_Confidence_Score","Genus Confidence Score"="Genus_Confidence_Score"))

# nExtended$ad
 ##------------------
 ##plot
 ##------------------
 networkplot <- ggplot(nExtended, aes(x = x, y = y, xend = xend, yend = yend,text =paste("genome:", vertex.names,"\npower: ",power,"\nOrder: ",Order,"\nFamily: ",Family,"\nGenus: ",Genus,"\nVC: ",VC,"\nSize: ",Size,"\nAdj P-value: ",Adj_P_value,"\nGenera in VC: ",Genera_in_VC,"\nGenus_Confidence_ScoreC: ",Genus_Confidence_Score,"\nTopology_Confidence_Score: ",Topology_Confidence_Score))) + geom_edges(color = "lightgray") +
   geom_nodes(aes(color = as.factor(color),size=as.factor(color))) +
   # scale_fill_manual(values="gray41")+
    scale_color_manual(values=colorzzz)+
    scale_size_manual(values=as.numeric(size))+
   # guides(color=FALSE,size=FALSE)+ #Remove legends
 theme_blank()+theme(legend.title = element_blank())
 networkplot
 
 library(plotly)
 ggp <- ggplotly(networkplot)
 #htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03//allPhages_vContact//networkPlot.html")
 # View(nExtended)
 
tmp <-   nExtended %>% filter(color=="RMK202 phages") 
table(tmp$`VC Subcluster`)
 
 
 nExtended %>% filter(color!="Viral DB") %>% filter()
 sort(table(nExtended$VC),decreasing = TRUE)
 
 count_perViralCluster <- nExtended %>%
  group_by(VC,color) %>% 
  dplyr::summarise(count=n()) %>% spread(.,color,count)
 
  ##==================================
 ##keep only phages in our s. thermophilus phage cluster
  ##==================================
 ###---------------------identify phages in our cluster

 
# table(nExtended$VC) 

tmp_phages <- nExtended[grep("RMK202_",nExtended$vertex.names),] 
ourPhageCluster <- c(unique(tmp_phages$VC))
nExtended_ourphages <- nExtended %>% filter(VC %in% ourPhageCluster)

phagesinClusters <- Phages_names %>% filter(VC %in% ourPhageCluster)
phagesinClusters$Genome

###---------------------clustering
# Sterm_clustering_new <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/c1.clusters") %>% as.data.frame()
Sterm_clustering_new <- read_csv("02_data/c1_ldel.clusters") %>% as.data.frame()
Sterm_clustering_forFiltering_new <- Sterm_clustering_new  %>% filter(., grepl('Streptococcus', Members,ignore.case = TRUE))

Sterm_clustering_unnested <- Sterm_clustering_forFiltering_new %>% 
    mutate(Members = strsplit(as.character(Members), " ")) %>% 
    unnest(Members)

# str_split_fixed(Sterm_clustering_unnested$Members, "~", 3)[,1]  %>% table()


###---------------1. filter:
##clusters that contain a Streptococcus


# nrow(Sterm_clustering_unnested)
nrow(Sterm_cluster)
Sterm_cluster_sub <- Sterm_cluster[which(Sterm_cluster$from %in% phagesinClusters$Genome | Sterm_cluster$to %in% phagesinClusters$Genome),]
nrow(Sterm_cluster_sub)
Sterm_net_sub <- graph_from_data_frame(Sterm_cluster_sub,direct=F)

 

 net<-asNetwork(Sterm_net_sub) #ggnetwork requires a network object
 n<-ggnetwork(net)
  ##------------------
 ##color and size change
  ##------------------
 
 
###--------
##coloring
#---------------
# n[grep("RMK202_",n$vertex.names),"vertex.names"] %>% table()
 n$color <- as.character("Viral DB")
 n$color[grep("RMK202_",n$vertex.names)] <- "RMK202 phages"
 n$color[n$vertex.names %in% Mahony_genomes_cos$namez] <- "cos"
 n$color[n$vertex.names %in% Mahony_genomes_pac$namez] <- "pac"
 n$color[n$vertex.names %in% Mahony_genomes_987$namez] <- "987"
 n$color[n$vertex.names %in% Mahony_genomes_5093$namez] <- "5093"
 
 
 n$color = factor(n$color, levels=c("RMK202 phages","cos","pac","987","5093","Viral DB"))
 colorzzz <- c("red","blue","green","#808000","purple","grey50")
 size <- c("4","2","2","2","2","1.5")
 
 
 # n$color = factor(n$color, levels=c("RMK202 phages","cos","987","Viral DB"))
 # colorzzz <- c("red","blue","#808000","grey50")
 # size <- c("4","2","2","1.5")
 
 table(n$color)
 
 n$color_new <- as.character("Viral DB")
n$color_new[grep("RMK202_Streptococcus_phage_1",n$vertex.names)[1]] <- "RMK202 phages"
n$color_new[grep("RMK202_Streptococcus_phage_2",n$vertex.names)[1]] <- "RMK202 phages"


 ##------------------
 ##extend
 ##------------------
 
 nExtended <-  merge(n, Phages_names, by.y="Genome",by.x="vertex.names",all.x = TRUE)
 # nrow(n)
 # nExtended <-  merge(n, Phages_names, by.y="Genome",by.x="vertex.names",all.x = TRUE) 
 # nrow(nExtended)
 # nExtended <- n
 # ccolnames(nExtended)
colnames(nExtended)
colnames(nExtended) <- revalue(colnames(nExtended), c("Adj P-value"="Adj_P_value","Genera in VC"="Genera_in_VC","Topology Confidence Score"="Topology_Confidence_Score","Genus Confidence Score"="Genus_Confidence_Score"))



 nExtended$alpha <- as.character("Viral DB")
nExtended$alpha[nExtended$VC %in% ourPhageCluster] <- "RMK202 phages"
table(nExtended$alpha)
  alphaCol <- c(1,0.03)
  
tmp <- nExtended[nExtended$alpha=="RMK202 phages",]
table(tmp$VC)
  
nExtended$fillings <- as.character("Viral DB")
nExtended$fillings[nExtended$VC %in% ourPhageCluster[1]] <- ourPhageCluster[1]
nExtended$fillings[nExtended$VC %in% ourPhageCluster[2]] <- ourPhageCluster[2]
nExtended$fillings[nExtended$vertex.names %in% Mahony_genomes_cos$namez] <- "cos"
nExtended$fillings[nExtended$vertex.names %in% Mahony_genomes_987$namez] <- "987"
nExtended$fillings[nExtended$vertex.names=="RMK202_Streptococcus_phage_1"] <- "Streptococcus_phage_1"
nExtended$fillings[nExtended$vertex.names=="RMK202_Streptococcus_phage_2"] <- "Streptococcus_phage_2"
as.factor(nExtended$fillings)
library("RColorBrewer")

three_reds <- c(brewer.pal(9,name="YlOrRd")[c(4,6)],
							 brewer.pal(3,name="Reds")[3])
three_browns <- brewer.pal(9,name="YlOrBr")[c(7,8,9)]
three_blues <- brewer.pal(9,name="YlGnBu")[c(5,6,8)]

filling <- c(three_reds[1],three_blues[1],three_reds[2],three_blues[2],three_reds[3],three_blues[3],"grey")
# filling <- c(three_reds[1],three_browns[1],three_reds[2],three_browns[2],three_reds[3],three_browns[3],"grey")

# filling <- c("black","red","darkgrey","darkred","grey")
# table(nExtended$alpha)

  



# nExtended$ad
 ##------------------
 ##plot
 ##------------------
table(nExtended$alpha)
# nExtended$Family
 networkplot_sub <- ggplot(nExtended, aes(x = x, y = y, xend = xend, yend = yend,text =paste("genome:", vertex.names,"\npower: ",power,"\nOrder: ",Order,"\nFamily: ",Family,"\nGenus: ",Genus,"\nVC: ",VC,"\nSize: ",Size,"\nAdj P-value: ",Adj_P_value,"\nGenera in VC: ",Genera_in_VC,"\nGenus_Confidence_ScoreC: ",Genus_Confidence_Score,"\nTopology_Confidence_Score: ",Topology_Confidence_Score))) + geom_edges(color = "lightgray",alpha=0.1) +
   geom_nodes(aes(color = as.factor(fillings),size=as.factor(alpha),alpha=as.factor(alpha))) +
   # scale_fill_manual(values="gray41")+
    scale_color_manual(values=filling)+
     scale_alpha_manual(values=alphaCol)+
    scale_fill_manual(values=colorzzz)+
   geom_nodelabel_repel(aes(label = vertex.names),
                       box.padding = unit(1, "lines"),
                       data = function(x) { x[ x$color_new == "RMK202 phages", ]})+
    scale_size_manual(values=as.numeric(c(2,1.25)))+
   # guides(color=FALSE,size=FALSE)+ #Remove legends
 theme_blank()+theme(legend.title = element_blank())
 networkplot_sub
 table(nExtended$color_new)
 # RMK202 phages
#  x[grep("RMK202_",x$vertex.names),]
#  n$color[grep("RMK202_",n$vertex.names)]
 # svg("~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03//allPhages_vContact//networkPlot_subset.svg",width=7,height=7)
 # png("~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03//allPhages_vContact//networkPlot_subset.png", width = 2800, height = 2800,res=300)
  # networkplot_sub
   # dev.off()
 
 # library(plotly)
 # ggp <- ggplotly(networkplot_sub)
 # htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03//allPhages_vContact//networkPlot_subset.html")
 # View(nExtended)


 ##==================================
 ##old analysis
  ##==================================



###--------
##grep fo Mahony phages
#---------------
Mahony_genomes$namez <- paste(Mahony_genomes$Genus,Mahony_genomes$phage,Mahony_genomes$strain,sep="~")
Mahony_genomes_cos <- Mahony_genomes[which(Mahony_genomes$Type=="cos"),]
Mahony_genomes_pac <- Mahony_genomes[which(Mahony_genomes$Type=="pac"),]
Mahony_genomes_987 <- Mahony_genomes[which(Mahony_genomes$Type=="987"),]
Mahony_genomes_5093 <- Mahony_genomes[which(Mahony_genomes$Type=="5093"),]

##colour

vcolour_Sterm <- rep("grey",length(V(Sterm_net)$name))
# vcolour_Sterm_ori <- V(Sterm_net)$name
# vcolour_Sterm <- "grey"
vcolour_Sterm[which(V(Sterm_net)$name %in% paste(Mahony_genomes_cos$Genus,Mahony_genomes_cos$phage,Mahony_genomes_cos$strain,sep="~"))] <- "green"
vcolour_Sterm[which(V(Sterm_net)$name %in% paste(Mahony_genomes_pac$Genus,Mahony_genomes_pac$phage,Mahony_genomes_pac$strain,sep="~"))] <- "blue"
vcolour_Sterm[which(V(Sterm_net)$name %in% paste(Mahony_genomes_987$Genus,Mahony_genomes_987$phage,Mahony_genomes_987$strain,sep="~"))] <- "pink"
vcolour_Sterm[which(V(Sterm_net)$name %in% paste(Mahony_genomes_5093$Genus,Mahony_genomes_5093$phage,Mahony_genomes_5093$strain,sep="~"))] <- "brown"

# vcolour_Sterm[which(!vcolour_Sterm_ori %in% paste(Mahony_genomes$Genus,Mahony_genomes$phage,Mahony_genomes$strain,sep="~"))] <- "grey"
table(vcolour_Sterm)

V(Sterm_net)$color <- vcolour_Sterm
table(V(Sterm_net)$color)
# V(Sterm_net)$color[which(V(Sterm_net)$name=='Streptococcus~phage~SW19')] <- "grey"
# V(Sterm_net)$color[which(V(Sterm_net)$name=='Streptococcus~phage~SW24')] <- "grey"
##size

vsize_Sterm <- V(Sterm_net)$name
vsize_Sterm_ori <- V(Sterm_net)$name

grep("S_t",V(Sterm_net)$name)
vsize_Sterm[which(!vsize_Sterm_ori %in% paste(Mahony_genomes$Genus,Mahony_genomes$phage,Mahony_genomes$strain,sep="~"))] <- "1.5"
vsize_Sterm[which(vsize_Sterm_ori %in% paste(Mahony_genomes$Genus,Mahony_genomes$phage,Mahony_genomes$strain,sep="~"))] <- "5"

table(vsize_Sterm)

vsize_Sterm <- as.numeric(vsize_Sterm)

#---------------
#plot
#---------------

set.seed(8051)

 V(Sterm_net)$name[grep("RMK202",V(Sterm_net)$name)]
 length( V(Sterm_net)$name)
 # V(Llac_net)$name
 # V(Ldel_net)$name
 # length(V(Ldel_net)$name)


# create vertices
fg_lay_Sterm <- layout_with_fr(Sterm_net,dim=2,niter=99)
# fg_lay_Ldel <- layout_with_fr(Ldel_net,dim=2,niter=99)
# fg_lay_Llac <- layout_with_fr(Llac_net,dim=2,niter=99)

# plot  Sterm
 # plot(Sterm_net,vertex.label=NA,layout=fg_lay_Sterm,
 #     vertex.size=vsize_Sterm,
 #       main="Sterm phages network"
 #     )
     # png("~/Desktop/Projects/2019_RMK202_analysis/plot/vContact_sterm.png", width = 1800, height = 1200,res=300)

 plot(Sterm_net,vertex.label=NA,layout=fg_lay_Sterm,vertex.size=vsize_Sterm,
       main="Sterm phages network"
     )
 legend("topright", 
       legend=c("cos","pac","987","5093","unknown"), pch=19,
      col=c("green","blue","pink","brown","grey"), bty="n")

# dev.off()
 
# # plot  Ldel
#  plot(Ldel_net,vertex.label=NA,layout=fg_lay_Ldel,
#        main="Ldel phages network"
#      )
#  
#  plot(Llac_net,vertex.label=NA,layout=fg_lay_Llac,
#        main="Llac phages network"
#      )
 
 #---------------
 ##creat modules
 #---------------
 # ceb_sterm_high25 <- ceb_sterm
# ceb_sterm <- cluster_edge_betweenness(Sterm_net)
#dendPlot(ceb_ara,mode="hclust")
# plot(ceb_sterm,Sterm_net,vertex.label=NA,vertex.size=vsize_Sterm,vertex.frame.color="red",
#      main="arabidospsis network")

 
#  ##---------Sterm
# ceb_sterm<- cluster_edge_betweenness(Sterm_net)
#   mem_cab_sterm <- membership(ceb_sterm)
#   df_mem_cab_sterml <- split(names(mem_cab_sterm),mem_cab_sterm)
# 
# memberss_sterm <- data.frame(names(mem_cab_sterm),as.double(mem_cab_sterm))
# # dendPlot(ceb_ldel,mode="hclust",hang = 0.0001)
# plot(ceb_sterm,Sterm_net,vertex.label=NA,vertex.frame.color="red",
#      main="sterm with modules network") 
# 
# 
# table(memberss_sterm$as.double.mem_cab_sterm.)
# which(table(memberss_sterm$as.double.mem_cab_sterm.)>2)
# 
# memberss_sterm[which(memberss_sterm$names.mem_cab_sterm. %in% Mahony_genomes_pac$namez),]
# memberss_sterm[which(memberss_sterm$names.mem_cab_sterm. %in% Mahony_genomes_987$namez),]
# memberss_sterm[which(memberss_sterm$names.mem_cab_sterm. %in% Mahony_genomes_cos$namez),]
# memberss_sterm[which(memberss_sterm$names.mem_cab_sterm. %in% Mahony_genomes_5093$namez),]
# 
# keepers_names <- as.double(which(table(memberss_sterm$as.double.mem_cab_sterm.)>2))
# length(as.double(which(table(memberss_sterm$as.double.mem_cab_sterm.)>2)))
# 
# memberss_sterm_subset <- memberss_sterm[which(memberss_sterm$as.double.mem_cab_sterm. %in% keepers_names),]
#   table(memberss_sterm_subset$as.double.mem_cab_sterm.)
# 
# 
#   
# write_csv(memberss_sterm,path="~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/Streptococcus_phage_cutoff50.members",col_names = FALSE)
# write_csv(memberss_sterm_subset,path="~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/Sterm_phage_cutoff55_subset.members",col_names = FALSE)
# 


# 
# memberss_sterm$names.mem_cab_sterm. 
# Mahony_genomes_pac$namez
# 
#   ##---------ldel
#   ceb_ldel <- cluster_edge_betweenness(Ldel_net)
#   mem_cab_ldel <- membership(ceb_ldel)
#   df_mem_cab_Ldel <- split(names(mem_cab_ldel),mem_cab_ldel)
# 
# memberss_ldel <- data.frame(names(mem_cab_ldel),as.double(mem_cab_ldel))
# # dendPlot(ceb_ldel,mode="hclust",hang = 0.0001)
# plot(ceb_ldel,Ldel_net,vertex.label=NA,vertex.frame.color="red",
#      main="Ldel with modules network") 
# 
# #write_csv(memberss_ldel,path="~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/Ldel_phage.members",col_names = FALSE)
# write_csv(memberss_ldel,path="~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/Ldel_phage_cutoff25.members",col_names = FALSE)
# 
# 
# table(memberss_ldel$as.double.mem_cab_ldel.)
#   ##---------llac
#   ceb_Llac <- cluster_edge_betweenness(Llac_net)
#   mem_cab_llac <- membership(ceb_Llac)
#   df_mem_cab_Llac <- split(names(mem_cab_llac),mem_cab_llac)
# 
# memberss_llac <- data.frame(names(mem_cab_llac),as.double(mem_cab_llac))
# # dendPlot(ceb_ldel,mode="hclust",hang = 0.0001)
# plot(ceb_Llac,Llac_net,vertex.label=NA,vertex.frame.color="red",
#      main="llac with modules network") 
# 
# 
# table(memberss_llac$as.double.mem_cab_llac.)
# which(table(memberss_llac$as.double.mem_cab_llac.)>2)
# 
# 
# keepers_names_llac <- as.double(which(table(memberss_llac$as.double.mem_cab_llac.)>2))
# #length(as.double(which(table(memberss_sterm$as.double.mem_cab_sterm.)>2)))
# 
# memberss_llac_subset <- memberss_llac[which(memberss_llac$as.double.mem_cab_llac. %in% keepers_names_llac),]
#   table(memberss_llac_subset$as.double.mem_cab_llac.)
# 
# 
# write_csv(memberss_llac,path="~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/Lactococcus_phage_cutoff50.members",col_names = FALSE)
# write_csv(memberss_llac_subset,path="~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/Lactococcus_phage_cutoff50_subset.members",col_names = FALSE)
# 
# 
# 
# 
#   ##----------hubscore ==High number of connections
#   hub_ara <- hub_score(Sterm_net)$vector
#   
#   #table(hub_ara)
#   
#   
#   mem_cab_ara <- membership(ceb_ara)
#   df_mem_cab_ara <- split(names(mem_cab_ara),mem_cab_ara)
#   names(df_mem_cab_ara) <- paste0("module_",names(df_mem_cab_ara))
#   
#   membersdata <- melt(df_mem_cab_ara)
#   return(membersdata)

 
  ###-----------------------------------
  ##Clustering analysis
  ###-----------------------------------

library(tidyverse)
library(readr)
Sterm_clustering <- read_csv("02_data/c1_ldel.clusters")
  
Sterm_clustering_unnested <- Sterm_clustering %>% 
    mutate(Members = strsplit(as.character(Members), " ")) %>% 
    unnest(Members) %>% filter(., grepl('Streptococcus', Members))

Sterm_clustering_forFiltering <- Sterm_clustering  %>% filter(., grepl('Streptococcus', Members))

ggplot(Sterm_clustering_forFiltering,aes(x=Quality))+geom_density()+theme_classic()
ggplot(Sterm_clustering_forFiltering,aes(x=Quality,y=Size))+geom_point()+theme_classic()
nrow(Sterm_clustering_forFiltering)
Sterm_clustering_forFiltering %>% filter(Size>3) %>% filter(Quality>0.2) %>% nrow

ggplot(Sterm_clustering_forFiltering,aes(x=Size))+geom_density()+theme_classic()

 Sterm_clustering_forFiltering  %>% filter(., grepl('SW', Members))

  ###-----------------------------------
  ##Clustering analysis 02
  ###-----------------------------------
# ~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new//allPhages_vContact
library(tidyverse)
library(readr)
Sterm_clustering <- read_csv("02_data/genome_by_genome_overview_ldel.csv")
Sterm_clustering[grep("rmk",Sterm_clustering$Genome),"Genome"]
#   
# Sterm_clustering_unnested <- Sterm_clustering %>% 
#     mutate(Members = strsplit(as.character(Members), " ")) %>% 
#     unnest(Members) %>% filter(., grepl('Streptococcus', Members))

Sterm_clustering$Genome <- Sterm_clustering$Genome %>% gsub("~"," ",.)
Sterm_clustering_filtered <- Sterm_clustering %>% select(Genome,Order,Family,Genus,VC,Size,Quality)
# Sterm_clustering_filtered$known <- 

Mahony_genomes$Genome <- paste(Mahony_genomes$Genus,Mahony_genomes$phage,Mahony_genomes$strain,sep=" ")
Mahony_genomes_prep <- Mahony_genomes %>% select(Genome,Type)

Sterm_clustering_filtered_2 <- merge(Sterm_clustering_filtered,Mahony_genomes_prep,by="Genome",all.x = TRUE)

clustersKnown <- Sterm_clustering_filtered_2[which(!is.na(Sterm_clustering_filtered_2$Type)),] %>% select(VC,Type) %>% unique() 

Sterm_clustering_filtered_final <- merge(Sterm_clustering_filtered_2,clustersKnown,by = "VC",all.x = TRUE)


table(Sterm_clustering_filtered_final$Type.y)
table(Sterm_clustering_filtered_final$VC) %>% sort()

# which(Sterm_clustering_filtered_final$Type.y=="cos")

# Sterm_clustering_filtered_final[which(!is.na(Sterm_clustering_filtered_final$Type.y)),]


write_csv(Sterm_clustering_filtered_final,path="03_results/Streptococcus_phage_Cluster_assignment.txt",col_names = TRUE)

# ggplot(Sterm_clustering_forFiltering,aes(x=Quality))+geom_density()+theme_classic()
# ggplot(Sterm_clustering_forFiltering,aes(x=Quality,y=Size))+geom_point()+theme_classic()

```


