---
title: "RMK202_cheese_experiment_20190128"
author: "Vincent Somerville"
date: "4 January 2019"
output: html_document
editor_options: 
  chunk_output_type: console
---
## Setup

```{r setup ,echo=FALSE}
knitr::opts_chunk$set(message=FALSE, echo=FALSE, eval=FALSE)

```

##setting up plotting window
Infos from  [here](https://stackoverflow.com/questions/46534643/make-r-studio-plots-only-show-up-in-new-window)
```{r}
#to setup in extra window
options(device = "X11")
dev.new()
#to setup in Rstudio
options(device = "RStudioGD")
dev.new()
```

##Evolution experiment

#####load data
The data was stored and extracted from google.sheets

```{r}
library(readr)
library(tidyverse)

rmk202_test_cellCounts <- read_csv("~/Desktop/Projects/2020_StarterEvolutionExperiment/01_cellCount/rmk202_Evol_experiment_data.csv",skip = 2) %>% filter(sample!="sample") %>% select(c("sample","step","pH","average BM","average SR.9.3","calculated Ldel","generations_all" ,"generations_sterm","generations_ldel","cummulative_generation_all"  ,"cummulative_generation_sterm","cummulative_generation_ldel","doublingTime_all","doublingTime_sterm","doublingTime_ldel","survival_rate_all","survival_rate_sterm","survival_rate_ldel" ))

rmk202_test_cellCounts$sampleNAME <- str_split_fixed(rmk202_test_cellCounts$sample, fixed("_"), 3)[,1]
rmk202_test_cellCounts$week <- str_split_fixed(rmk202_test_cellCounts$sample, fixed("_"), 3)[,2]
rmk202_test_cellCounts$passage <- str_split_fixed(rmk202_test_cellCounts$sample, fixed("_"), 3)[,3]

###------------------------cumulative generation per species at end
cumGenData <- rmk202_test_cellCounts %>% filter(passage==18) %>% select(sample, cummulative_generation_sterm, cummulative_generation_ldel) %>% gather(.,feature, generations,c("cummulative_generation_sterm", "cummulative_generation_ldel"), factor_key=TRUE,na.rm = TRUE)

cumGenData$feature  <- plyr::revalue(cumGenData$feature, c("cummulative_generation_sterm"="S.thermophilus","cummulative_generation_ldel"="L. delbrueckii"))

colorsSpecies <- (c("#EB4D4D","#10B552") ) 

cumGenPLOT <- ggplot(cumGenData,aes(x=feature,group=feature,y=generations,fill=feature))+geom_boxplot()+theme_classic()+labs(y="number of generations\n at the end",x="")+theme(axis.text.x=element_blank(),legend.position = "none")+scale_fill_manual(values=colorsSpecies)
cumGenPLOT


###------------------------number of generations per species 

GenData <- rmk202_test_cellCounts %>% filter(passage!=1) %>% select(sample, generations_sterm, generations_ldel) %>% gather(.,feature, generations,c("generations_sterm", "generations_ldel"), factor_key=TRUE,na.rm = TRUE)

GenData$feature  <- plyr::revalue(GenData$feature, c("generations_sterm"="S.thermophilus","generations_ldel"="L. delbrueckii"))


GenPLOT <- ggplot(GenData,aes(x=feature,group=feature,y=generations,fill=feature))+geom_boxplot()+theme_classic()+labs(y="number of generations\n per propagation step",x="")+theme(axis.text.x=element_blank(),legend.position = "none")+scale_fill_manual(values=colorsSpecies)
GenPLOT

###------------------------number of generations per species 

GenData <- rmk202_test_cellCounts %>% filter(passage!=1) %>% select(sample, generations_sterm, generations_ldel) %>% gather(.,feature, generations,c("generations_sterm", "generations_ldel"), factor_key=TRUE,na.rm = TRUE)

GenData$feature  <- plyr::revalue(GenData$feature, c("generations_sterm"="S.thermophilus","generations_ldel"="L. delbrueckii"))
mean(GenData$generations)
GenPLOT <- ggplot(GenData,aes(x=feature,group=feature,y=generations,fill=feature))+geom_boxplot()+theme_classic()+labs(y="Number of generations\nRelative abundance step",x="")+theme(axis.text.x=element_blank(),legend.position = "none")+scale_fill_manual(values=colorsSpecies)
GenPLOT

 GenData %>% 
    group_by(feature) %>% 
    dplyr::summarize(median = mean(generations))
  
 ###------------------------average CFU after second passage
colnames(rmk202_test_cellCounts)
CountData <- rmk202_test_cellCounts %>% filter(step=="second_passage") %>% select("sample", "average SR.9.3", "calculated Ldel") %>% gather(.,feature, generations,c("average SR.9.3", "calculated Ldel"), factor_key=TRUE,na.rm = TRUE)

CountData$feature  <- plyr::revalue(CountData$feature, c("average SR.9.3"="S.thermophilus","calculated Ldel"="L. delbrueckii"))

CountPLOT <- ggplot(CountData,aes(x=feature,group=feature,y=generations,fill=feature))+geom_boxplot()+theme_classic()+labs(y="Bacterial count of\nworking stock [CFU/ml]",x="")+theme(axis.text.x=element_blank(),legend.position = "none")+scale_fill_manual(values=colorsSpecies)
CountPLOT

 CountData %>% 
    group_by(feature) %>% 
    dplyr::summarize(median = mean(generations))
  
 CountData %>% 
    group_by(feature) %>% 
    dplyr::summarize(median = mean(generations),sd = sd(generations))
  
 ###------------------------death rate after freeze drying
 rmk202_test_cellCounts$step
SurvivalData <- rmk202_test_cellCounts %>% filter(step=="freeze_dry") %>% select("sample", "survival_rate_all") %>% add_column(species="all")

# CountData$feature  <- plyr::revalue(CountData$feature, c("average SR.9.3"="S.thermophilus","calculated Ldel"="L. delbrueckii"))

SurvivalPLOT <- ggplot(SurvivalData,aes(x=species,group=species,y=survival_rate_all,fill=species))+geom_boxplot()+theme_classic()+labs(y="Survivial rate\n [%]",x="")+theme(axis.text.x=element_blank(),legend.position = "none")+scale_fill_manual(values="grey88")
SurvivalPLOT


 SurvivalData %>% 
    group_by(species) %>% 
    dplyr::summarize(median = mean(survival_rate_all))
  
 
 ###------------------------CFU over time
 rmk202_test_cellCounts$cummulative_generation_all
CFUData <- rmk202_test_cellCounts  %>% filter(passage!="17") %>% filter(step!="freeze_dry") %>% select("sample","step","cummulative_generation_all","sampleNAME","passage", "average SR.9.3", "calculated Ldel") %>% add_column(species="all")%>% gather(.,feature, generations,c("average SR.9.3", "calculated Ldel"), factor_key=TRUE,na.rm = TRUE)
# CFUData[which(is.na(CFUData$cummulative_generation_all)),]
CFUData[which(is.na(CFUData$cummulative_generation_all)),"cummulative_generation_all"] <- 0



CFUData$feature  <- plyr::revalue(CFUData$feature, c("average SR.9.3"="S.thermophilus","calculated Ldel"="L. delbrueckii"))
CFUData$passage <- as.double(CFUData$passage)
CFUData$sampleNAME <- as.factor(CFUData$sampleNAME)


CFUPLOT <- ggplot(CFUData,aes(x=cummulative_generation_all,group=sampleNAME,y=generations))+facet_wrap(~feature,scales="free_y",ncol=1)+geom_point(aes(color=step),size=2)+geom_line(aes(color=feature),alpha=0.4,size=1)+theme_classic()+labs(y="Bacterial count\n[CFU/ml]",x="generations")+theme(axis.text.x = element_text(size=9),legend.position = "top",legend.title = element_blank())+scale_fill_manual(values=colorsSpecies)+scale_color_manual(values=c("start"="black","first_passage"="#d9d9d9","second_passage"="#b3b3ff","freeze_dry"="#ff0101","S.thermophilus"="#EB4D4D","L. delbrueckii"="#10B552"))
CFUPLOT

# 
# CFUData$passage <- as.character(CFUData$passage)
# 
# CFUData$passage = factor(CFUData$passage, levels=as.character(1:18))
# 
# 
# CFUPLOT <- ggplot(CFUData,aes(x=passage,group=passage,y=generations,fill=step))+facet_wrap(~feature,scales="free_y",ncol=1)+geom_boxplot()+theme_classic()+labs(y="count of bacteria [CFU/ml]",x="passages")+theme(axis.text.x = element_text(size=9),legend.position = "none")+scale_fill_manual(values=c("start"="blue","first_passage"="pink","second_passage"="orange","freeze_dry"="gold","S.thermophilus"="#EB4D4D","L. delbrueckii"="#10B552"))
# CFUPLOT

  ###------------------------pH over time

pHData <- rmk202_test_cellCounts  %>% filter(step!="freeze_dry")%>% select("sample","step","sampleNAME","cummulative_generation_all", "pH") 
pHData[which(is.na(pHData$cummulative_generation_all)),"cummulative_generation_all"] <- 0

# pHData$passage <- as.double(pHData$passage)
pHData$sampleNAME <- as.factor(pHData$sampleNAME)


pHPLOT <- ggplot(pHData,aes(x=cummulative_generation_all,group=sampleNAME,y=pH))+geom_point(aes(color=step),size=2)+geom_line(color="grey",alpha=0.4)+theme_classic()+labs(y="pH",x="generations")+theme(axis.text.x = element_text(size=9),legend.position = "none")+scale_color_manual(values=c("start"="black","first_passage"="#d9d9d9","second_passage"="#b3b3ff","freeze_dry"="#ff0101","S.thermophilus"="#EB4D4D","L. delbrueckii"="#10B552"))
pHPLOT

 pHData %>% 
    dplyr::summarize(median = mean(pH),sd = sd(pH))
  
###------------------------put figurers together
 
 library(patchwork)
 
 plot1 <- SurvivalPLOT+CountPLOT+GenPLOT+cumGenPLOT+plot_layout(ncol = 4,widths =  c(0.5,1,1,1))

plotFinal <- CFUPLOT+plot1+pHPLOT+plot_layout(nrow = 3,heights = c(1,1,0.5))
plotFinal

svg("/home/vincent/Desktop/Projects/2019_RMK202_analysis/plot/evolutionExperiment_plot.svg",width=6,height=8)
plotFinal
dev.off()
 
```


##prepare for NCBI submission

This is actually a very late stage. Here, I gather the genomes for NCBI submission.

```{bash}

##----------------------Ldel

ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Ldel/ |grep "_202_" |grep ".fasta$" 

for ldelsss in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Ldel/ |grep "_202_" |grep ".fasta$" )
do

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMK202_only/Ldel

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Ldel/${ldelsss} /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMK202_only/Ldel/

done

cp /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/Ldel/genome/202-LMAG.fasta /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMK202_only/Ldel/

##----------------------Sterm

ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree_with_References/Sterm/ |grep "202-" |grep ".fasta$"  |sed 's/.fasta//g'|sed 's/202-//g'

for stermss in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree_with_References/Sterm/ |grep "202-" |grep ".fasta$"  )
do

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMK202_only/Sterm

cp /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree_with_References/Sterm//${stermss} /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMK202_only/Sterm/

done


##which are ONT
ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Sterm/ |grep "202-" |grep ".fasta$" 


###----------------scp files

mkdir -p ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/

scp -r vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMK202_only/ ~/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/


```


#metabolomics
Here, I analyse the metabolomics data I received from the GC/MS of pascal an yihelen on 13.8.2020.
```{r}
library(openxlsx)
library(readr)
library(ggbiplot)
library(tidyverse)
library(patchwork)
library(plotly)
X20200810_Vincent_cultures_profinder_untargeted <- read.xlsx("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/metabolomics/20200810_Vincent_cultures_profinder_untargeted.xlsx")
# pca_res <- prcomp(df, scale. = TRUE)

library(readr)
Metabolites_information_reduced <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/metabolomics/Metabolites_information_reduced.csv",  "\t", escape_double = FALSE, trim_ws = TRUE)

nrow(Metabolites_information_reduced)
par(mfrow=c(1,1))
hist(Metabolites_information_reduced$`RT (avg)`)
Metabolites_information_reduced_goodOnes <- Metabolites_information_reduced %>% filter(`RT (avg)`<46.6)
nrow(Metabolites_information_reduced_goodOnes)
# colnames(X20200810_Vincent_cultures_profinder_untargeted)[1:4]
X20200810_Vincent_cultures_profinder_untargeted <-   X20200810_Vincent_cultures_profinder_untargeted %>% dplyr::select(c(colnames(X20200810_Vincent_cultures_profinder_untargeted)[1:4],Metabolites_information_reduced_goodOnes$compounds))
# Metabolites_information_reduced_goodOnes


colnames(X20200810_Vincent_cultures_profinder_untargeted)[1:3,1:10]
X20200810_Vincent_cultures_profinder_untargeted[1:3,1:10]
##------------------remove zero columns
X20200810_Vincent_cultures_profinder_untargeted_new <- X20200810_Vincent_cultures_profinder_untargeted[,!(colnames(X20200810_Vincent_cultures_profinder_untargeted)%in% names(which(colSums(X20200810_Vincent_cultures_profinder_untargeted[,5:ncol(X20200810_Vincent_cultures_profinder_untargeted)])==0)))]

dim(X20200810_Vincent_cultures_profinder_untargeted)
dim(X20200810_Vincent_cultures_profinder_untargeted_new)

# which(colSums(X20200810_Vincent_cultures_profinder_untargeted_new[,5:ncol(X20200810_Vincent_cultures_profinder_untargeted_new)])==0)
ploting_data

metabolo.pca <- prcomp(X20200810_Vincent_cultures_profinder_untargeted_new[,5:ncol(X20200810_Vincent_cultures_profinder_untargeted_new)], center = TRUE,scale. = TRUE)
metabolo.sum <- summary(metabolo.pca)
metabolo.sum
metabolo.sum <- metabolo.sum$importance %>% as.data.frame()



ploting_data <- metabolo.pca$x %>% as.data.frame()
ploting_data$sample <- X20200810_Vincent_cultures_profinder_untargeted$sample
ploting_data$biological_duplicate <- X20200810_Vincent_cultures_profinder_untargeted$biological_duplicate
ploting_data$analytical_duplicate <- X20200810_Vincent_cultures_profinder_untargeted$analytical_duplicate
ploting_data$grouping <- substring(X20200810_Vincent_cultures_profinder_untargeted$sample, 1, 1)
table(ploting_data$grouping)
ploting_data$grouping_long  <- plyr::revalue(ploting_data$grouping, c("A"="All strains combined","L"="L. delbrueckii only","S"="S. thermophilus only","R"="original RMK202 starter culture","M"="pairwise strain mix"))

# pca1 <- ggbiplot(metabolo.pca,var.axes=FALSE,groups=ploting_data$grouping_long,ellipse=TRUE,obs.scale = 1,text =paste("sample: ",ploting_data$sample,"\nbio. dupli.: ",ploting_data$biological_duplicate, "\ntec.dupli.: ",ploting_data$analytical_duplicate))+theme_classic()
# 
# ggp <- ggplotly(pca1)
# ggp
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/metabolomics/20200810_Vincent_cultures_pca.html")


pca1 <- ggbiplot(metabolo.pca,var.axes=FALSE,groups=ploting_data$biological_duplicate,ellipse=TRUE,obs.scale = 1)+theme_classic()+labs(title="grouping by biological replicates")+theme(legend.position = "none")
pca2 <- ggbiplot(metabolo.pca,var.axes=FALSE,groups=ploting_data$grouping_long,ellipse=TRUE,obs.scale = 1)+theme_classic()+labs(title="grouping by treatment")+theme(legend.position = "none")
pca3 <- ggbiplot(metabolo.pca,var.axes=TRUE,groups=ploting_data$grouping_long,ellipse=TRUE,obs.scale = 1)+theme_classic()+labs(title="grouping by treatment")+theme(legend.position = "none")

pca1+pca2+plot_layout(nrow = 2)


png("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/metabolomics/20200810_Vincent_cultures_pca_01.png", width = 1500, height = 2100,res=300)
 # svg("Desktop/Projects/2019_RMK202_analysis/00_FINAL/04_AA_annotation/GapMind/plots/AminoAcidBiosynthesis_bothSpecies_predicted.svg",width=10,height=7)
pca1+pca2+plot_layout(nrow = 2)
dev.off()
##-----------------------
##----------------------NON-ADJUSTED SCALES
##-----------------------
metabolo.pca <- prcomp(X20200810_Vincent_cultures_profinder_untargeted_new[,5:ncol(X20200810_Vincent_cultures_profinder_untargeted_new)], center = FALSE,scale. = FALSE)
metabolo.sum <- summary(metabolo.pca)
metabolo.sum
metabolo.sum <- metabolo.sum$importance %>% as.data.frame()


# pca1 <- ggbiplot(metabolo.pca,var.axes=FALSE,groups=ploting_data$grouping_long,ellipse=TRUE,obs.scale = 1,text =paste("sample: ",ploting_data$sample,"\nbio. dupli.: ",ploting_data$biological_duplicate, "\ntec.dupli.: ",ploting_data$analytical_duplicate))+theme_classic()
# 
# ggp <- ggplotly(pca1)
# ggp
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/metabolomics/20200810_Vincent_cultures_pca.html")


pca4 <- ggbiplot(metabolo.pca,var.axes=FALSE,groups=ploting_data$biological_duplicate,ellipse=TRUE,obs.scale = 1)+theme_classic()+labs(title="grouping by biological replicates")
pca5 <- ggbiplot(metabolo.pca,var.axes=FALSE,groups=ploting_data$grouping_long,ellipse=TRUE,obs.scale = 1)+theme_classic()+labs(title="grouping by treatment")
pca6 <- ggbiplot(metabolo.pca,var.axes=TRUE,groups=ploting_data$grouping_long,ellipse=TRUE,obs.scale = 1)+theme_classic()+labs(title="grouping by treatment")



pca7 <- ggbiplot(metabolo.pca,var.axes=FALSE,groups=ploting_data$sample,ellipse=TRUE,obs.scale = 1)+theme_classic()+labs(title="grouping by sample")
pca7
 ggp <- ggplotly(pca7)
ggp

 pca1+pca2+plot_layout(nrow = 2)


png("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/metabolomics/20200810_Vincent_cultures_pca_02.png", width = 1500, height = 2100,res=300)
 # svg("Desktop/Projects/2019_RMK202_analysis/00_FINAL/04_AA_annotation/GapMind/plots/AminoAcidBiosynthesis_bothSpecies_predicted.svg",width=10,height=7)
pca1+pca2+plot_layout(nrow = 2)
dev.off()



png("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/metabolomics/20200810_Vincent_cultures_pca_explained.png", width = 2500, height = 1800,res=300)
 # svg("Desktop/Projects/2019_RMK202_analysis/00_FINAL/04_AA_annotation/GapMind/plots/AminoAcidBiosynthesis_bothSpecies_predicted.svg",width=10,height=7)
pca6
dev.off()

pca1+pca4+pca2+pca5+pca3+pca6+plot_layout(nrow = 3)

png("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/metabolomics/20200810_Vincent_cultures_pca_04.png", width = 3000, height = 4200,res=300)
 # svg("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/metabolomics/20200810_Vincent_cultures_pca_04.svg",width=15,height=21)
pca1+pca4+pca2+pca5+pca3+pca6+plot_layout(nrow = 3)
dev.off()







pcaFINAL <- ggbiplot(metabolo.pca,var.axes=FALSE,groups=ploting_data$grouping_long,ellipse=TRUE,obs.scale = 1)+theme_classic()
pcaFINAL

svg("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/metabolomics/20200810_Vincent_cultures_pca_FINAL.svg",width=8,height=5)
pcaFINAL
dev.off()


###------------------------------------
##remove all isolates mixed
###------------------------------------









##-----------------------
##----------------------unnormalised metabolo
##-----------------------
unnormalized_metabolo <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/metabolomics/20200810_Vincent_cultures_profinder_untargeted_paraldehyde-normalised.csv", ";", escape_double = FALSE, trim_ws = TRUE)

unnormalized_metabolo_new <- unnormalized_metabolo[,!(colnames(unnormalized_metabolo)%in% names(which(colSums(unnormalized_metabolo[,5:ncol(unnormalized_metabolo)])==0)))]

dim(unnormalized_metabolo)
dim(unnormalized_metabolo_new)

# which(colSums(X20200810_Vincent_cultures_profinder_untargeted_new[,5:ncol(X20200810_Vincent_cultures_profinder_untargeted_new)])==0)


metabolo.pca <- prcomp(unnormalized_metabolo_new[,5:ncol(unnormalized_metabolo_new)], center = TRUE,scale. = TRUE)
metabolo.sum <- summary(metabolo.pca)
metabolo.sum
metabolo.sum <- metabolo.sum$importance %>% as.data.frame()

pca1 <- ggbiplot(metabolo.pca,var.axes=FALSE,groups=ploting_data$grouping_long,ellipse=TRUE,obs.scale = 1,text =paste("sample: ",ploting_data$sample,"\nbio. dupli.: ",ploting_data$biological_duplicate, "\ntec.dupli.: ",ploting_data$analytical_duplicate))+theme_classic()
# 
# ggp <- ggplotly(pca1)
# ggp
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/metabolomics/20200810_Vincent_cultures_pca.html")
# 

pca1 <- ggbiplot(metabolo.pca,var.axes=FALSE,groups=ploting_data$biological_duplicate,ellipse=TRUE,obs.scale = 1)+theme_classic()+labs(title="grouping by biological replicates")
pca2 <- ggbiplot(metabolo.pca,var.axes=FALSE,groups=ploting_data$grouping_long,ellipse=TRUE,obs.scale = 1)+theme_classic()+labs(title="grouping by treatment")
pca3 <- ggbiplot(metabolo.pca,choices=c(2,3),var.axes=FALSE,groups=ploting_data$grouping_long,ellipse=TRUE,obs.scale = 1)+theme_classic()+labs(title="grouping by treatment")

pca1+pca2+plot_layout(nrow = 2)


png("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/metabolomics/20200810_Vincent_cultures_pca_03.png", width = 1500, height = 2100,res=300)
 # svg("Desktop/Projects/2019_RMK202_analysis/00_FINAL/04_AA_annotation/GapMind/plots/AminoAcidBiosynthesis_bothSpecies_predicted.svg",width=10,height=7)
pca1+pca2+plot_layout(nrow = 2)
dev.off()


# ggp <- ggplotly(pca1)
# ggp
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/metabolomics/20200810_Vincent_cultures_pca.html")
# 
# pca1Prop <- metabolo.sum["Proportion of Variance","PC1"]*100 
# pca1Prop <- round(pca1Prop,digits =1 )
# pca2Prop <- metabolo.sum["Proportion of Variance","PC2"]*100
# pca2Prop <- round(pca2Prop,digits =1 )
# 

# autoplot(metabolo.pca)
# 
# 
# pca1 <- ggplot(ploting_data,aes(x=PC1,y=PC2,color=grouping_long,fill=grouping_long,text =paste("sample: ",ploting_data$sample,"\nbio. dupli.: ",ploting_data$biological_duplicate, "\ntec.dupli.: ",ploting_data$analytical_duplicate)))+geom_point(size=2)+theme_classic()+labs(x=paste0("PC1 [explained=",pca1Prop,"%]"),y=paste0("PC1 [explained=",pca2Prop,"%]"))
# 
# ggp <- ggplotly(pca1)
# ggp
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/metabolomics/20200810_Vincent_cultures_pca.html")
# 
# 
# ploting_data$PC18
# plot.new()
# par(mfrow=c(1,1))
# plot(metabolo.pca$x[,1],metabolo.pca$x[,2],)
##========================================================================================================
##------------------------------------------------
##z-Score calculation
##------------------------------------------------

tmpss <- t(X20200810_Vincent_cultures_profinder_untargeted_new[5:3085])

dim(X20200810_Vincent_cultures_profinder_untargeted_new)
for (metaolsss in colnames(X20200810_Vincent_cultures_profinder_untargeted_new)[5:3085]) {
  tmp <- X20200810_Vincent_cultures_profinder_untargeted_new %>% mutate(zscore = (metaolsss - mean(metaolsss))/sd(metaolsss)) %>% select(zscore)
}


dat %>% mutate(zscore = (BMI - mean(BMI))/sd(BMI))


##--------------------------------------------------------------------
##test
##--------------------------------------------------------------------

##multiple T-test function
ttestRat <- function(df, grp1, grp2) {
  x = df[grp1]
  y = df[grp2]
  x = as.numeric(x)
  y = as.numeric(y)  
  results = t.test(x, y)
  results$p.value
}


ttestmaxMEAN <- function(df, grp1, grp2) {
  x = df[grp1]
  y = df[grp2]
  x = as.numeric(x)
  y = as.numeric(y)  
  results = t.test(x, y)
  max(results$estimate)
}

##prepare for function
X20200810_Vincent_cultures_profinder_untargeted_new[,2]

X20200810_Vincent_cultures_profinder_untargeted_new[,2][c(1:6,13:34,51:56,63:84)]
X20200810_Vincent_cultures_profinder_untargeted_new[,2][c(7:12,35:50,57:62,85:100)]

# metabolite=10
# results2 <- t.test(X20200810_Vincent_cultures_profinder_untargeted_new[ c(1:6,13:34,51:56,63:84),metabolite],X20200810_Vincent_cultures_profinder_untargeted_new[ c(7:12,35:50,57:62,85:100),metabolite])
# max(results2$estimate)
metagData <- t(X20200810_Vincent_cultures_profinder_untargeted_new[,5:ncol(X20200810_Vincent_cultures_profinder_untargeted_new)])
dim(metagData)
###aply function
rawpvalue = apply(metagData, 1, ttestRat, grp1 = c(1:6,13:34,51:56,63:84), grp2 =c(7:12,35:50,57:62,85:100))
hist(rawpvalue)

###----fold change
##transform our data into log2 base.
metagData_log = log2(metagData)

#calculate the mean of each gene per control group
control = apply(metagData_log[,c(1:6,13:34,51:56,63:84)], 1, mean)

#calcuate the mean of each gene per test group
test = apply(metagData_log[, c(7:12,35:50,57:62,85:100)], 1, mean) 

#confirming that we have a vector of numbers
class(control) 
foldchange <- control - test 

hist(foldchange, xlab = "log2 Fold Change (Control vs Test)")


###maxMEANvalue

rawpmaxMean = apply(metagData, 1, ttestmaxMEAN, grp1 = c(1:6,13:34,51:56,63:84), grp2 =c(7:12,35:50,57:62,85:100))
hist(log10(rawpmaxMean))

##----volcano

results = cbind(foldchange, rawpvalue,MeanExpression=log10(rawpmaxMean))
results = as.data.frame(results)
results$probename <- rownames(results)

results$fill <- ifelse(results$foldchange>2&-log10(results$rawpvalue)>10,"significant","NS")
table(results$fill)
colorsss <- c("grey","red")

library(ggplot2)
volcano = ggplot(data = results, aes(x = foldchange, y = -1*log10(rawpvalue),color=fill,size=MeanExpression))+theme_classic()+geom_point()+theme(legend.position = "none")+scale_color_manual(values=colorsss)
volcano

namessSignificant <- results %>% filter(fill=="significant") %>% rownames()


significant_samples <- X20200810_Vincent_cultures_profinder_untargeted_new %>%  dplyr::select(namessSignificant)
significant_samples$groups <- substr(X20200810_Vincent_cultures_profinder_untargeted_new$sample, 1, 1)
significant_samples$sample <- X20200810_Vincent_cultures_profinder_untargeted_new$sample
# gather(significant_samples,)

significant_samples_long <- gather(significant_samples, feature, intensity, colnames(significant_samples)[1:10], factor_key=TRUE,na.rm = TRUE) 
ggplot(significant_samples_long,aes(x=groups,intensity,fill=groups))+theme_classic()+geom_boxplot()+facet_wrap(~feature,scales="free")

# str_split_fixed(X20200810_Vincent_cultures_profinder_untargeted_new$sample, fixed("*"), 3)[,1]

png("~/Desktop/Projects/2019_RMK202_analysis/plot/GC_MS_ttest_coculture-vs-isolates.png", width = 2500, height = 2500,res=300)
 # svg("Desktop/Projects/2019_RMK202_analysis/00_FINAL/04_AA_annotation/GapMind/plots/AminoAcidBiosynthesis_bothSpecies_predicted.svg",width=10,height=7)
volcano
dev.off()
```


#starter culture stability


```{r}
library(readr)
library(tsibble)
library(tidyverse)
# frmAuswert_Flüssigkulturen <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/starterCultureProduction/frmAuswert_Flüssigkulturen.csv",  "\t", escape_double = FALSE, col_types = cols(qCharge = col_date(format = "%d/%m/%y")),  trim_ws = TRUE)

# ph_workingStock_rmk202 <- read_delim("Desktop/Projects/2019_RMK202_analysis/workingStock_pH/ph_workingStock_rmk202.csv", "\t", escape_double = FALSE, trim_ws = TRUE, col_types = cols(mixDate = col_date(format = "%W/%y")))

ph_workingStock_rmk202 <- read_delim("Desktop/Projects/2019_RMK202_analysis/workingStock_pH/ph_workingStock_rmk202.csv", "\t", escape_double = FALSE, trim_ws = TRUE)

# Path$New.Week <-
  # strftime(ph_workingStock_rmk202$mixDate, format = "%W/%Y")

  yearweek(ph_workingStock_rmk202$yearweekcol)
ph_workingStock_rmk202$week
 ph_workingStock_rmk202$date <-  MMWRweek::MMWRweek2Date(MMWRyear = ph_workingStock_rmk202$year,
                        MMWRweek = ph_workingStock_rmk202$week,
                        MMWRday = 3)
  
  ph_workingStock_rmk202$Versand
acidicationPlot <- ggplot(ph_workingStock_rmk202,aes(x=date,y=Versand))+geom_point()+theme_classic()+scale_y_continuous(limits = c(0,150))+labs(y="acid-base titration value of starter culture",x="time [year]")+theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9))+geom_hline(yintercept = 110)
 
 acidicationPlot
 is.num(ph_workingStock_rmk202$`D(-)%`)
 ph_workingStock_rmk202 %>% filter(!is.na("D(-)%"))
lactatePlot <- ggplot(ph_workingStock_rmk202,aes(x=date,y=`D(-)%`))+geom_point()+theme_classic()+scale_y_continuous(limits = c(0,150))+labs(y="acid-base titration value of starter culture",x="time [year]")+theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9))+geom_hline(yintercept = 110)
 
 lactatePlot
 
# colnames(frmAuswert_Flüssigkulturen)
# table(frmAuswert_Flüssigkulturen$tKulturBezeichnung)


# subsetAcidification <- frmAuswert_Flüssigkulturen %>% filter(tKulturBezeichnung=="RMK 202")
# subsetAcidification$`BK38Th 18 h` <- as.Date(subsetAcidification$`BK38Th 18 h`,format = "%d/%m/%y")
# str(subsetAcidification)
# subsetAcidification$qCharge <- as.Date(subsetAcidification$qCharge)
# subsetAcidification$qWoche
# acidicationPlot <- ggplot(subsetAcidification,aes(x=qCharge,`BK38Th 18 h`))+geom_point()+theme_classic()+scale_y_continuous(limits = c(0,150))+labs(y="acid-base titration value after 18 h",x="time [month-year]")+theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9))+scale_x_date(date_labels = "%m-%y")
# ss <- subset(subsetAcidification, qWoche > as.Date("2006-1-1"))
# ggplot(subsetAcidification,aes(x=qWoche,`BK38Th 18 h`))+geom_point()+theme_classic()+scale_y_continuous(limits = c(0,150))+labs(y="acid-base titration value after 18 h",x="")+theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9))

png("~/Desktop/mid_thesis/report/figures/20200430/chapter1/acidificationPlot.png", width = 2100, height = 1500,res=300)
 # svg("Desktop/Projects/2019_RMK202_analysis/00_FINAL/04_AA_annotation/GapMind/plots/AminoAcidBiosynthesis_bothSpecies_predicted.svg",width=10,height=7)
acidicationPlot
dev.off()
```


#acidification and growth

Here, I analysis the acidification and growth rates measured on the 13.7.2020

here, I measured platte one which was pippetted with the robo, first ph 4.6 than milk than +culture (total 175ul)

take homes:
- too much shaking
- too much volume
--> potential contame
--> maybe ph4.6 measurment at the end
--> see if calibration value in both output can be used
--> I have to copy paste the necessary rows out of the file (Why!!??)


```{bash}
cut -f 1 ~/Desktop/Projects/2020_StarterCultureDiversity/02_ph_measurment/hydroplates_measurment/platte1/'200701_plate_01_rmk_r01_ph4.6 (copy).txt'


/home/vincent/Desktop/Projects/2019_RMK202_analysis/growth_acidification/20200714/20200712_rmk202_strains_curated
#cut -f 1 ~/Desktop/Projects/2020_StarterCultureDiversity/02_ph_measurment/hydroplates_measurment/platte1/200701_plate_01_rmk_r01_final_copypaste.txt > ~/Desktop/Projects/2020_StarterCultureDiversity/02_ph_measurment/hydroplates_measurment/platte1/200701_plate_01_rmk_r01_final_copypaste_onlyTime.txt

#awk -F "[.:]" '{if($1=="1") print $0;else print "0."$0}' ~/Desktop/Projects/2020_StarterCultureDiversity/02_ph_measurment/hydroplates_measurment/platte1/200701_plate_01_rmk_r01_final_copypaste_onlyTime.txt

awk -F "." '{if($1=="1") print $0;else print "0."$0}' /home/vincent/Desktop/Projects/2019_RMK202_analysis/growth_acidification/20200714/20200712_rmk202_strains_curated.txt |sed 's/\#SAT/NA/g'> /home/vincent/Desktop/Projects/2019_RMK202_analysis/growth_acidification/20200714/20200712_rmk202_strains_curated_02.txt

```


```{r}
###-----------------------------
##variables
###-----------------------------

location="~/Desktop/Projects/2019_RMK202_analysis/growth_acidification/20200714/"
plateName="20200712_rmk202_strains_curated_02.txt"
NamesWell="200714_names.csv"
SampleNamesWell="200714_samples_names.csv"

sampleName="20200712_rmk202_strains"
# replicate=03
# plateNumber=1

CALIBRATION_PH4_5=6000 #UNTILL WHICH SECOND IS PH 4.5
CALIBRATION_start_PH6_5=13000 #UNTILL WHICH SECOND IS PH 4.5
CALIBRATION_PH6_5=16000 #UNTILL WHICH SECOND IS PH 6.5
MEASURMENT_START=14000 #UNTILL WHICH SECOND IS THE MEASURMENT START

###-----------------------------
##contaminated rows
##this has to be added after the first run. see if large outliers or fermented blanks are occuring
###-----------------------------
exclude <- c("B5","A12","A8","A4")


###-----------------------------
##libraries
###-----------------------------
library(readr)
library(ggplot2)
library(plyr)
library(plotly)
library(tidyverse)
library(lubridate)
library(growthrates)
library(broom)
library(drc)
library(patchwork)
library(dplyr)

###-----------------------------
##import
###-----------------------------

    # X20200616_test_hydro_final_02 <- read_delim("~/Desktop/Projects/2020_StarterCultureDiversity/02_ph_measurment/hydroplates_measurment/platte1/200701_plate_01_rmk_r01_final_copypaste_02.txt",  "\t", escape_double = FALSE, col_types = cols(Time = col_time(format = "%d.%H:%M:%S")), trim_ws = TRUE) %>%dplyr::select(-X99)

 X20200616_test_hydro_final_02 <- read_delim(paste0(location,plateName),  "\t", escape_double = FALSE, trim_ws = TRUE) %>% dplyr::select(-"X99")
colnames(X20200616_test_hydro_final_02)
    

    ###-----------------------------
##prep date
#unfortunately the molecular devices machine gives a weird date format (after 24h it adds 1. to the hour column)
#in order to correct it we have to run also the previous bash chunk
###-----------------------------
    
X20200616_test_hydro_final_02$days <- as.numeric(str_split_fixed(X20200616_test_hydro_final_02$`0.Time`, fixed("."), 2)[,1])
X20200616_test_hydro_final_02$time <- str_split_fixed(X20200616_test_hydro_final_02$`0.Time`, fixed("."), 2)[,2]
X20200616_test_hydro_final_02$hourOld <- as.numeric(str_split_fixed(X20200616_test_hydro_final_02$time, fixed(":"), 3)[,1])
X20200616_test_hydro_final_02$min <-  as.numeric(str_split_fixed(X20200616_test_hydro_final_02$time, fixed(":"), 3)[,2])
# X20200616_test_hydro_final_02$sec <- str_split_fixed(X20200616_test_hydro_final_02$time, fixed(":"), 3)[,3]
# X20200616_test_hydro_final_02[,90:103]
# X20200616_test_hydro_final_02$hourNew <- as.character(X20200616_test_hydro_final_02$hourOld+(24*X20200616_test_hydro_final_02$days))

X20200616_test_hydro_final_03 <- X20200616_test_hydro_final_02 %>% 
  mutate(
    days = duration(days, 'day'),
    hourOld = duration(hourOld, 'hour'),
    min = duration(min, 'minute'),
    # sec = duration(sec, 'second'),
    TIMEfinal = hourOld + days + min
  )  %>% dplyr::select(-c("days","time","hourOld",`0.Time`,`Temperature(¡C)`,"min"))

# test[1:20,90:108]
# X20200616_test_hydro_final_03[1:20,90:97]

# hydroplate_wide_ph_calibration_02

# X20200616_test_hydro_final_03 <- unite(X20200616_test_hydro_final_02, TIMEfinal, c(hourNew,min,sec),sep = ":") %>%dplyr::select(-c("days","time","hourOld",`0.Time`,`Temperature(¡C)`))
# X20200616_test_hydro_final_03[,90:97]
# X20200616_test_hydro_final_03$TIMEfinal

# X20200616_test_hydro_final_03$TIMEfinal <- hms(X20200616_test_hydro_final_03$TIMEfinal)
# str(hydroplate_wide_prep)
hydroplate_wide_prep <- X20200616_test_hydro_final_03

# ggplot(hydroplate_wide_prep,aes(x=TIMEfinal,y=A1))+geom_point()+theme_classic()+geom_vline(xintercept = c(CALIBRATION_PH4_5,CALIBRATION_PH6_5,MEASURMENT_START))


plotForCalibration <-ggplot(hydroplate_wide_prep,aes(x=TIMEfinal,y=A1))+geom_point()+theme_classic()+geom_vline(xintercept = c(CALIBRATION_PH4_5,CALIBRATION_PH6_5,MEASURMENT_START,CALIBRATION_start_PH6_5))
plotForCalibration

ggp <- ggplotly(plotForCalibration)
ggp
# htmlwidgets::saveWidget(ggp, paste0(location,sampleName,"_cleaned.html"))


# test <-  X20200616_test_hydro_final_03 %>% filter(TIMEfinal >= ('3000s') & TIMEfinal <= ('22840s'))
###-----------------------------
##pH_calibration
###-----------------------------


# hydroplate_wide_ph_6.4 <- X20200616_test_hydro_final_03 %>% filter(TIMEfinal >= ('3000s') & TIMEfinal <= ('22840s')) %>% dplyr::select(-"TIMEfinal") %>% colMeans(na.rm = TRUE) %>% as.data.frame() %>% rownames_to_column(var="well")
hydroplate_wide_ph_6.4 <- X20200616_test_hydro_final_03 %>%  filter(TIMEfinal >= (paste0(CALIBRATION_start_PH6_5,"s")) & TIMEfinal <= (paste0(CALIBRATION_PH6_5,"s"))) %>% dplyr::select(-"TIMEfinal") %>% colMeans(na.rm = TRUE) %>% as.data.frame() %>% rownames_to_column(var="well")


# hydroplate_wide_ph_6.4 <- X20200616_test_hydro_final_03 %>% filter(TIMEfinal >= ('00H 23M 00S') & TIMEfinal <= ('02H 00M 00S')) %>%dplyr::select(-"TIMEfinal") %>% colMeans() %>% as.data.frame() %>% rownames_to_column(var="well")

colnames(hydroplate_wide_ph_6.4)[2] <- "pH_6.4"

# ###ph_tief
# test <-  X20200616_test_hydro_final_03 %>% filter(TIMEfinal < ('3000s') ) %>% dplyr::select(-"TIMEfinal") 
# test <-  X20200616_test_hydro_final_03 %>% filter(TIMEfinal < (paste0(CALIBRATION_PH4_5,"s"))) %>% dplyr::select(-"TIMEfinal") 
# test[,80:97]

# str(test)
hydroplate_wide_ph_4_66 <- X20200616_test_hydro_final_03 %>% filter(TIMEfinal < (paste0(CALIBRATION_PH4_5,"s"))) %>% dplyr::select(-"TIMEfinal") %>% colMeans(na.rm = TRUE) %>% as.data.frame() %>% rownames_to_column(var="well")

# hydroplate_wide_ph_4_66 <- tail(X20200616_test_hydro_final_03,n = 10) %>% dplyr::select(-"TIMEfinal") %>% colMeans() %>% as.data.frame() %>% rownames_to_column(var="well")
colnames(hydroplate_wide_ph_4_66)[2] <- "pH_4.66"

# hydroplate_wide[90,]
hydroplate_wide_ph_calibration <- merge(hydroplate_wide_ph_6.4,hydroplate_wide_ph_4_66,by="well")
hydroplate_wide_ph_calibration_curve <- hydroplate_wide_ph_calibration
###---------------calculate ph_regression
ph_high <- as.numeric(6.5)
ph_low <- as.numeric(4.6)
# hydroplate_wide_ph_calibration_curve$slope
# hydroplate_wide_ph_calibration_curve$slope <- ((hydroplate_wide_ph_calibration_curve$pH_6.4- hydroplate_wide_ph_calibration_curve$pH_4.66)/(ph_high-ph_low))
hydroplate_wide_ph_calibration_curve$slope <- ((ph_high-ph_low)/(as.integer(hydroplate_wide_ph_calibration_curve$pH_6.4)- as.integer(hydroplate_wide_ph_calibration_curve$pH_4.66)))
# hydroplate_wide_ph_calibration_curve$intersect <- hydroplate_wide_ph_calibration_curve$pH_6.4/(hydroplate_wide_ph_calibration_curve$slope*ph_high)
hydroplate_wide_ph_calibration_curve$intersect <- ph_high-(hydroplate_wide_ph_calibration_curve$slope*as.integer(hydroplate_wide_ph_calibration_curve$pH_6.4))
# hydroplate_wide_ph_calibration_curve$intersect <- ph_low/(hydroplate_wide_ph_calibration_curve$slope*hydroplate_wide_ph_calibration_curve$pH_4.66)

# hydroplate_wide_ph_calibration_curve$test <- (as.integer(hydroplate_wide_ph_calibration_curve$pH_6.4)*hydroplate_wide_ph_calibration_curve$slope)+hydroplate_wide_ph_calibration_curve$intersect

# test_intenstiy <- 4045145
# hydroplate_wide_ph_calibration_curve$slope*test_intenstiy+hydroplate_wide_ph_calibration_curve$intersect

hydroplate_calbration_test <- hydroplate_wide_ph_calibration_curve %>% dplyr::select(well,slope,intersect)

###-----------------------------
##merge samples
###-----------------------------
hydroplate_long <- gather(hydroplate_wide_prep, sample, pH, colnames(hydroplate_wide_prep)[1:96], factor_key=TRUE,na.rm = TRUE) 
hydroplate_wide_ph_calibration_02 <- merge(hydroplate_long,hydroplate_calbration_test,by.x = "sample",by.y = "well")
hydroplate_wide_ph_calibration_02$intensity_calibrated <- hydroplate_wide_ph_calibration_02$slope*hydroplate_wide_ph_calibration_02$pH+hydroplate_wide_ph_calibration_02$intersect

# ggplot(hydroplate_wide_ph_calibration_02,aes(x=TIMEfinal,y=intensity_calibrated,group=sample,fill=sample,color=sample))+geom_point()+theme_classic()

hydroplate_wide_ph_calibration_03 <-hydroplate_wide_ph_calibration_02
# hydroplate_wide_ph_calibration_03 <- hydroplate_wide_ph_calibration_02 %>% filter(TIMEfinal >= ('32000s'))
# hydroplate_wide_ph_calibration_03 <- hydroplate_wide_ph_calibration_02 %>% filter(TIMEfinal >= ('32000s')&TIMEfinal < ('150'))

# hydroplate_wide_ph_calibration_03 <- hydroplate_wide_ph_calibration_02 %>% filter(TIMEfinal < ('40000s'))

ggplot(hydroplate_wide_ph_calibration_03,aes(x=TIMEfinal,y=intensity_calibrated,group=sample,fill=sample,color=sample))+geom_point()+theme_classic()

# ###-----------------------------
# ##plot
# ###-----------------------------
# ##RMK202
# 
# hydroplate_wide_ph_calibration_rmk202 <- dplyr::filter(hydroplate_wide_ph_calibration_03, grepl('B|C', sample))
# ggplot(hydroplate_wide_ph_calibration_rmk202,aes(x=Time,y=intensity_calibrated,group=sample,fill=sample,color=sample))+geom_point()+theme_classic()
# 
# ##RMK203
# 
# hydroplate_wide_ph_calibration_rmk202 <- dplyr::filter(hydroplate_wide_ph_calibration_03, grepl('D|E', sample))
# ggplot(hydroplate_wide_ph_calibration_rmk202,aes(x=Time,y=intensity_calibrated,group=sample,fill=sample,color=sample))+geom_point()+theme_classic()
# 
# 
# ##RMK101
# 
# hydroplate_wide_ph_calibration_rmk202 <- dplyr::filter(hydroplate_wide_ph_calibration_03, grepl('F|G', sample))
# ggplot(hydroplate_wide_ph_calibration_rmk202,aes(x=Time,y=intensity_calibrated,group=sample,fill=sample,color=sample))+geom_point()+theme_classic()
# 
# ##RMK190
# 
# hydroplate_wide_ph_calibration_rmk202 <- dplyr::filter(hydroplate_wide_ph_calibration_03, grepl('H', sample))
# ggplot(hydroplate_wide_ph_calibration_rmk202,aes(x=Time,y=intensity_calibrated,group=sample,fill=sample,color=sample))+geom_point()+theme_classic()

###-----------------------------
##add names
###-----------------------------

# hydroplate_wide_ph_calibration_03

hydroplates_names <- read_delim(paste0(location,NamesWell),  "\t", escape_double = FALSE,  trim_ws = TRUE)
hydroplates_names_samples <- read_delim(paste0(location,SampleNamesWell),  "\t", escape_double = FALSE,  trim_ws = TRUE)


hydroplates_names_final <- merge(hydroplates_names,hydroplates_names_samples,by="SAMPLE",all = TRUE)


hydroplate_wide_ph_calibration_04 <- merge(hydroplate_wide_ph_calibration_03,hydroplates_names_final,by.x = "sample",by.y = "well") %>% filter(SAMPLE!="blank")
# table(hydroplate_wide_ph_calibration_04$no_growth)
table(hydroplate_wide_ph_calibration_04$SAMPLE)
table(hydroplate_wide_ph_calibration_04$sample)
# hydroplate_wide_ph_calibration_04$name <- un

# dplyr::select(well,slope,intersect)
# htmlPrep <- ggplot(hydroplate_wide_ph_calibration_04,aes(x=Time,y=intensity_calibrated,group=sample,fill=sample,color=sample))+geom_point()+theme_classic()+facet_wrap(~name)
# ggp <- ggplotly(htmlPrep)
# htmlwidgets::saveWidget(ggp, paste0(location,sampleName,"_nothingExcluded.html"))

###-----------------------------
##exclude if necessary
###-----------------------------
# require(scales)
# as.Time(hydroplate_wide_ph_calibration_05$Time,"%H:%M:%S")

# hydroplate_wide_ph_calibration_04 

# hydroplate_wide_ph_calibration_04 <- hydroplate_wide_ph_calibration_03
hydroplate_wide_ph_calibration_05 <- hydroplate_wide_ph_calibration_04 %>% filter(!sample %in% exclude)

# hydroplate_wide_ph_calibration_05$Time <- hms(hydroplate_wide_ph_calibration_05$Time) 
# hydroplate_wide_ph_calibration_05$Time <-  round(as.numeric(hydroplate_wide_ph_calibration_05$Time,"hours"),digits = 3)
# hydroplate_wide_ph_calibration_05$Time <- hydroplate_wide_ph_calibration_05$Time -1.5
htmlPrep <- ggplot(hydroplate_wide_ph_calibration_05,aes(x=TIMEfinal,y=intensity_calibrated,group=sample,fill=SAMPLE,color=SAMPLE))+geom_point()+theme_classic()+facet_wrap(~SAMPLE)+labs(y="pH",x="")+geom_vline(xintercept = c(CALIBRATION_PH4_5,CALIBRATION_PH6_5,MEASURMENT_START))
# htmlPrep <- ggplot(hydroplate_wide_ph_calibration_05,aes(x=TIMEfinal,y=intensity_calibrated,group=sample,fill=sample,color=sample))+geom_point()+theme_classic()+labs(y="pH",x="")
# htmlPrep
ggp <- ggplotly(htmlPrep)
ggp
htmlwidgets::saveWidget(ggp, paste0(location,sampleName,"_cleaned.html"))

# hydroplate_wide_ph_calibration_05 <- hydroplate_wide_ph_calibration_05 %>% filter(TIMEfinal > (paste0(MEASURMENT_START,"s")))

hydroplate_wide_ph_calibration_05 <- hydroplate_wide_ph_calibration_05 %>% filter(TIMEfinal > (paste0(CALIBRATION_start_PH6_5,"s")))



##-----------------------------------------------------------
#modelling
##-----------------------------------------------------------
Sys.sleep(10)

p     <- c(y0 = 6.608016, mumax = 0.0003549404, K = 4.703572,h0=11.044679)
# 
# lower   <- c(y0 = 5, mumax = 0.5, K = 3.5, h0 = 1)
# upper   <- c(y0 = 8,   mumax = 2.5,    K = 7,   h0 = 10)

hydroplate_wide_ph_calibration_05$TIME <- as.numeric(hydroplate_wide_ph_calibration_05$TIMEfinal)

many_baranyi_sub <- all_growthmodels(
                   intensity_calibrated ~ grow_baranyi(TIMEfinal, parms) | sample+NAME+grouping,
                   data = hydroplate_wide_ph_calibration_05, p=p,ncores = 8 )

# results(many_baranyi_sub)

par(mfrow = c(12, 8))
par(mar = c(1, 1, 1, 1))
plot(many_baranyi_sub)




many_baranyi2_res <- results(many_baranyi_sub)
many_baranyi2_res$lagPhase <- many_baranyi2_res$h0/log(2)
many_baranyi2_res
many_baranyi2_res$lagPhase <- many_baranyi2_res$h0/many_baranyi2_res$mumax
many_baranyi2_res_preped <- many_baranyi2_res
# many_baranyi2_res_preped <- many_baranyi2_res  %>% filter(name!="blank")
# many_baranyi2_res_preped <- many_baranyi2_res %>% filter(r2>0.98) %>% filter(name!="blank")
many_baranyi2_res_preped$mumax <- -many_baranyi2_res_preped$mumax
summary(many_baranyi2_res_preped)
table(many_baranyi2_res_preped$NAME)
many_baranyi2_res_preped$grouping = factor(many_baranyi2_res_preped$grouping, levels=c("lactobacillus","streptococcus","pairwise","complex"))
mumax_plot <- ggplot(many_baranyi2_res_preped,aes(x=NAME,y=mumax))+geom_boxplot()+theme_classic()+labs(x="",y="maximum pH decrease")+facet_wrap(~grouping,ncol = 5,scales = "free_x")
# ggplot(many_baranyi2_res_preped,aes(x=name,y=y0))+geom_boxplot()+theme_classic()+labs(x="","intitial pH")
k0plot <- ggplot(many_baranyi2_res_preped,aes(x=NAME,y=K))+geom_boxplot()+theme_classic()+labs(x="",y="lowest pH")+facet_wrap(~grouping,ncol = 5,scales = "free_x")
# ggplot(many_baranyi2_res_preped,aes(x=name,y=h0))+geom_boxplot()+theme_classic()


###-------------------------------------------------
##plot
###-------------------------------------------------

png(paste0(location,sampleName,"_all_baranyi.png"),width=4000,height=2000,res=300)
# svg("~/Desktop/Projects/2020_strainDelineation/04_pH_measurments/191127_pH_PLOT.svg",width=4,height=3)
par(mfrow = c(12, 8))
par(mar = c(1, 1, 1, 1))
plot(many_baranyi_sub)
dev.off()

finalSamplesPlot <- ggplot(hydroplate_wide_ph_calibration_05,aes(x=TIMEfinal,y=intensity_calibrated,group=NAME,fill=NAME,color=NAME))+geom_point()+theme_classic()+facet_wrap(~NAME)+labs(y="pH",x="")+theme(legend.position = "none")

svg(paste0(location,sampleName,"_all_samples_pH.svg"),width=12,height=8)
finalSamplesPlot
dev.off()


svg(paste0(location,sampleName,"_all_boxplots_summary.svg"),width=4,height=4)
mumax_plot + k0plot+plot_layout(nrow = 2)
dev.off()

mumax_plot + k0plot+plot_layout(nrow = 2)

###-------------------------------------------------
##output data
###-------------------------------------------------
many_baranyi2_res_preped$plate <- as.character(plateNumber)
many_baranyi2_res_preped$replicate <- as.character(replicate)

write.table(many_baranyi2_res_preped,paste0(location,sampleName,"_model.txt"),na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names = TRUE)


###-------------------------------------------------
##lagPHase calcuation
###-------------------------------------------------
# 
# hydroplate_wide_ph_calibration_05$intensity_calibrated_inverted <- 7.5 -hydroplate_wide_ph_calibration_05$intensity_calibrated
# 
# p     <- c(y0 = 6.608016, mumax = 0.0003549404, K = 4.703572,h0=11.044679)
# # 
# # lower   <- c(y0 = 5, mumax = 0.5, K = 3.5, h0 = 1)
# # upper   <- c(y0 = 8,   mumax = 2.5,    K = 7,   h0 = 10)
# 
# # hydroplate_wide_ph_calibration_05$TIME <- as.numeric(hydroplate_wide_ph_calibration_05$TIMEfinal)
# 
# many_baranyi_sub_inverted <- all_growthmodels(
#                    intensity_calibrated_inverted ~ grow_baranyi(TIMEfinal, parms) | sample+NAME+grouping,
#                    data = hydroplate_wide_ph_calibration_05, p=p,ncores = 8 )
# 
# # results(many_baranyi_sub)
# 
# par(mfrow = c(12, 8))
# par(mar = c(1, 1, 1, 1))
# plot(many_baranyi_sub_inverted)
# 
# 
# 
# many_baranyi2_res_inverted <- results(many_baranyi_sub_inverted)
# many_baranyi2_res_inverted$lagPhase <- many_baranyi2_res$h0/many_baranyi2_res$mumax
# many_baranyi2_res_inverted

###-------------------------------------------------
##lagPHase phased
##h0 parameter specifying the initial physiological state of organisms (e.g. cells) and in consequence the lag phase (h0 = max growth rate * lag phase).
###-------------------------------------------------
str(many_baranyi2_res)
library(tidyverse)
many_baranyi2_res$sample
preped <- many_baranyi2_res %>% dplyr::select(c("sample","lagPhase")) %>% remove_rownames()

hydroplate_wide_ph_calibration_06 <- merge(hydroplate_wide_ph_calibration_05,preped,by="sample")

 hydroplate_wide_ph_calibration_06 <- hydroplate_wide_ph_calibration_06 %>% 
  mutate(lagSeconds = duration(lagPhase, 'second'))  #%>% dplyr::select(-c("days","time","hourOld",`0.Time`,`Temperature(¡C)`,"min"))

 hydroplate_wide_ph_calibration_06$curatedTime <- hydroplate_wide_ph_calibration_06$TIMEfinal-hydroplate_wide_ph_calibration_06$lagSeconds
 
  hydroplate_wide_ph_calibration_06$curatedTime <- ifelse(hydroplate_wide_ph_calibration_06$grouping=="lactobacillus",hydroplate_wide_ph_calibration_06$TIMEfinal-(0.75*hydroplate_wide_ph_calibration_06$lagSeconds),hydroplate_wide_ph_calibration_06$TIMEfinal-hydroplate_wide_ph_calibration_06$lagSeconds)

 
htmlPrep <- ggplot(hydroplate_wide_ph_calibration_06,aes(x=curatedTime,y=intensity_calibrated,group=sample,fill=grouping,color=grouping))+geom_point(alpha=0.1)+theme_classic()
 
ggp <- ggplotly(htmlPrep)
ggp
# hydroplate_wide_ph_calibration_06$grouping
htmlPrep <- ggplot(hydroplate_wide_ph_calibration_06,aes(x=curatedTime,y=intensity_calibrated,group=sample,fill=grouping,color=grouping))+geom_point()+theme_classic()+facet_wrap(.~grouping,ncol = 4)
 
ggp <- ggplotly(htmlPrep)
ggp


# htmlPrep <- ggplot(hydroplate_wide_ph_calibration_06,aes(x=curatedTime,y=intensity_calibrated,group=sample,fill=SAMPLE,color=SAMPLE))+geom_boxplot()+theme_classic()+facet_wrap(.~grouping,ncol = 4)
#  
# ggp <- ggplotly(htmlPrep)
# ggp



hydroplate_wide_ph_calibration_06_sub <- hydroplate_wide_ph_calibration_06 %>%
 filter(row_number() %% 20 == 1)

htmlPrep <- ggplot(hydroplate_wide_ph_calibration_06_sub,aes(x=curatedTime,y=intensity_calibrated,group=sample,fill=grouping,color=grouping))+geom_point(alpha=0.1)+theme_classic()
 
ggp <- ggplotly(htmlPrep)
ggp

# htmlPrep <- ggplot(hydroplate_wide_ph_calibration_06,aes(x=curatedTime,y=intensity_calibrated,group=sample,fill=SAMPLE,color=SAMPLE))+geom_point()+theme_classic()+facet_wrap(~SAMPLE)+labs(y="pH",x="")+geom_vline(xintercept = c(CALIBRATION_PH4_5,CALIBRATION_PH6_5,MEASURMENT_START))
# htmlPrep <- ggplot(hydroplate_wide_ph_calibration_05,aes(x=TIMEfinal,y=intensity_calibrated,group=sample,fill=sample,color=sample))+geom_point()+theme_classic()+labs(y="pH",x="")
# htmlPrep
# ggp <- ggplotly(htmlPrep)
# ggp
# htmlwidgets::saveWidget(ggp, paste0(location,sampleName,"_cleaned.html"))

###-------------------------------------
##make a subsetting and averaging
###-------------------------------------

require(zoo)
# TS <- zoo(c(4, 5, 7, 3, 9, 8))
# rollapply(TS, width = 3, by = 2, FUN = mean, align = "left")
final_subsetting <- data_frame()
for (groupsss  in unique(hydroplate_wide_ph_calibration_06$grouping)) {
  # hydroplate_wide_ph_calibration_06$
  tmp <- hydroplate_wide_ph_calibration_06 %>% filter(grouping==groupsss)%>% arrange(curatedTime)
  
  min <- rollapply(tmp$intensity_calibrated, width = 10, by = 5, FUN = min, align = "left")
  mean <- rollapply(tmp$intensity_calibrated, width = 10, by = 5, FUN = mean, align = "left")
  max <- rollapply(tmp$intensity_calibrated, width = 10, by = 5, FUN = max, align = "left")
 namesss <- tmp %>% filter(row_number() %% 5 == 1) 
 namesssss <- namesss[1:length(max),"curatedTime"]
  length(max)

 tmp02 <- data_frame(time=namesssss,grouping=groupsss,minum=min,maxum=max,median=mean)
  final_subsetting <- rbind(final_subsetting,tmp02)
}

table(final_subsetting$grouping)


 final_subsetting_subset <- final_subsetting %>% filter(time<64800)

final_subsetting_subset$timeFinal <- final_subsetting_subset$time/3600

final_subsetting_subset_final <- final_subsetting_subset %>% filter(grouping!="complex")

minTime <- min(final_subsetting_subset_final$timeFinal)

low_phased_ribbon <- ggplot()+
    geom_ribbon(aes(x=timeFinal,ymin = minum , ymax = maxum,group=grouping,fill=grouping),data = final_subsetting_subset_final, alpha=.4)+
       # scale_x_discrete( expand = c(0, 0)) +
  # facet_wrap(~grouping,scales = "free")+
    labs(y="pH",x="phased incubation time")+
    scale_x_continuous(breaks =c(0,6,12,18),labels=c(0,6,12,18))+
  theme_classic()+
 theme(rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="bottom",
       legend.title = element_blank(),
       # axis.text.x = element_blank(),
       axis.text.y = element_text(size=8),
       # axis.title = element_text(size=9),
       
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )


  low_phased_ribbon
  
  

  
  # pdf(file = "~/Desktop/Projects/2019_RMK202_analysis/plot/acidifaction_curve.pdf",   width = 6,  height = 6) 
     svg("~/Desktop/Projects/2019_RMK202_analysis/plot/acidifaction_curve.svg",width=6,height=4)
    # png("~/Desktop/Projects/2019_RMK202_analysis/plot/acidifaction_curve.png", width = 2000, height = 2000,res=150)

  low_phased_ribbon

dev.off()
 


##------------------
##look at offset
##------------------
colnames(hydroplate_wide_ph_calibration_06)
lagsss <- hydroplate_wide_ph_calibration_06 %>% dplyr::select(c("grouping","lagPhase"))  %>% remove_rownames() %>% unique()
lagsss_mean <- aggregate(. ~grouping, data=lagsss, median, na.rm=TRUE)
lagsss_mean$hours <- lagsss_mean$lagPhase/3600
lagsss_mean$hours_corrected <- lagsss_mean$hours-2

```


CFU count of the same time series
```{r}
library(readr)
rmk202_strain_timeseries_Sheet1 <- read_csv("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/rmk202_strain_timeseries - Sheet1.csv", skip = 1)


rmk202_strain_timeseries_Sheet1_long <- rmk202_strain_timeseries_Sheet1 %>% gather(.,"sampless","CFU",`0_BM`:`24_MR`)
 rmk202_strain_timeseries_Sheet1_long$time <- as.numeric(str_split_fixed(rmk202_strain_timeseries_Sheet1_long$sampless, "_", 2)[,1])
 rmk202_strain_timeseries_Sheet1_long$plate <- str_split_fixed(rmk202_strain_timeseries_Sheet1_long$sampless, "_", 2)[,2]

 table(rmk202_strain_timeseries_Sheet1_long$plate)
 rmk202_strain_timeseries_Sheet1_long$species  <- plyr::revalue(rmk202_strain_timeseries_Sheet1_long$plate, c("BM"="total","M17X"="S. thermophilus","MR"="L. delbrueckii"))

 rmk202_strain_timeseries_Sheet1_long$samplecount <- 1:nrow(rmk202_strain_timeseries_Sheet1_long)
 
plot_overtime <-  ggplot(rmk202_strain_timeseries_Sheet1_long,aes(x=time,y=CFU,color=species,fill=species,group=sample,text =paste("sampleID=",samplecount)))+geom_line()+theme_classic()+facet_wrap(~species+group,ncol=5)


 library(plotly)
 ggp <- ggplotly(plot_overtime)
ggp


##----------------------
##exlude
##----------------------
excludes <- c("228","276","277","282","286","288")

rmk202_strain_timeseries_Sheet1_long_cleaned <- rmk202_strain_timeseries_Sheet1_long %>% filter(!samplecount %in%excludes)


plot_overtime <-  ggplot(rmk202_strain_timeseries_Sheet1_long_cleaned,aes(x=time,y=CFU,color=species,fill=species,group=sample,text =paste("sampleID=",samplecount)))+geom_line()+theme_classic()+facet_wrap(~species+group,ncol=5,scales = "free")

 ggp <- ggplotly(plot_overtime)
ggp


 ggplot(rmk202_strain_timeseries_Sheet1_long_cleaned,aes(x=time,y=CFU,color=species,fill=species,group=sample,text =paste("sampleID=",samplecount)))+geom_boxplot()+theme_classic()+facet_wrap(~species+group,ncol=5)+coord_trans(y="log2")

##----------------------
##mean , max ,median
##----------------------
 
 group_growth_cfu <- data.frame()
for (typess in unique(rmk202_strain_timeseries_Sheet1_long_cleaned$group)) {
  
  for (timesss in unique(rmk202_strain_timeseries_Sheet1_long_cleaned$time)) {
    
    
    for (speccc in unique(rmk202_strain_timeseries_Sheet1_long_cleaned$species)) {
      
    
    maxxx <- rmk202_strain_timeseries_Sheet1_long_cleaned %>% filter(group==typess) %>% filter(time==timesss) %>% filter(species==speccc)%>% select(CFU) %>% max(na.rm = TRUE)
    minnn <- rmk202_strain_timeseries_Sheet1_long_cleaned %>% filter(group==typess) %>% filter(time==timesss) %>% filter(species==speccc)%>% select(CFU)%>% min(na.rm = TRUE)
    meannn <- rmk202_strain_timeseries_Sheet1_long_cleaned %>% filter(group==typess) %>% filter(time==timesss)%>% filter(species==speccc)%>% select(CFU) %>% colMeans(na.rm = TRUE)

    tmps <- data.frame(sample=typess,species=speccc,time=timesss,min=minnn,max=maxxx,mean=meannn)
    
    group_growth_cfu <- rbind(group_growth_cfu,tmps)
    
    }
  }
} 
 
 group_growth_cfu$time <- as.numeric(group_growth_cfu$time)
 
 group_growth_cfu <- group_growth_cfu %>%  filter(mean!="NaN")
plot_overtime <-  ggplot(group_growth_cfu,aes(x=time,y=mean,color=sample,fill=sample,group=interaction(sample,species)))+geom_line()+theme_classic()+facet_wrap(~species,nrow=3)
plot_overtime
#  ggp <- ggplotly(plot_overtime)
# ggp

##----------------------
##polish
##----------------------

group_growth_cfu_mean_species <- group_growth_cfu %>% filter(species!="total") %>% filter(sample!="all_strains")#%>% filter(!is.na(mean))

plot_overtime <-  ggplot(group_growth_cfu_mean_species,aes(x=time,y=mean,color=sample,fill=sample,group=interaction(sample,species)))+geom_line()+theme_classic()+facet_wrap(~species,nrow=2)
plot_overtime


plot_overtime_RIBBON <-  ggplot()+geom_ribbon(aes(x=time,ymin = min , ymax = max,group=interaction(sample,species),fill=sample),data = group_growth_cfu_mean_species, alpha=.1)+theme_classic()+facet_wrap(~species,nrow=2)+geom_line(aes(x=time,y=mean,color=sample,fill=sample,group=interaction(sample,species)),data = group_growth_cfu_mean_species)
plot_overtime_RIBBON


##----------------------
##polish
##----------------------
rmk202_strain_timeseries_Sheet1_long_cleaned_box <- rmk202_strain_timeseries_Sheet1_long_cleaned %>% filter(species!="total") %>% filter(group!="all_strains")#%>% filter(!is.na(mean))

rmk202_strain_timeseries_Sheet1_long_cleaned_box$time <- as.factor(rmk202_strain_timeseries_Sheet1_long_cleaned_box$time)
rmk202_strain_timeseries_Sheet1_long_cleaned_box$species = factor(rmk202_strain_timeseries_Sheet1_long_cleaned_box$species, levels=c("S. thermophilus" ,"L. delbrueckii"))


plot_overtime_box <-  ggplot(rmk202_strain_timeseries_Sheet1_long_cleaned_box,aes(x=time,y=CFU,fill=group))+geom_boxplot(alpha=.3,color="grey")+theme_classic()+facet_wrap(~species,nrow=2)
plot_overtime_box


rmk202_strain_timeseries_Sheet1_long_cleaned_box$groupingFinal <- as.factor(paste0(rmk202_strain_timeseries_Sheet1_long_cleaned_box$species,"_",rmk202_strain_timeseries_Sheet1_long_cleaned_box$group))
levels(rmk202_strain_timeseries_Sheet1_long_cleaned_box$groupingFinal)
rmk202_strain_timeseries_Sheet1_long_cleaned_box$groupingFinal = factor(rmk202_strain_timeseries_Sheet1_long_cleaned_box$groupingFinal, levels=c("S. thermophilus_rmk" ,"S. thermophilus_pairwise","S. thermophilus_strepto" ,"S. thermophilus_lacto","L. delbrueckii_rmk","L. delbrueckii_pairwise","L. delbrueckii_lacto","L. delbrueckii_strepto"))



plot_overtime_box <-  ggplot(rmk202_strain_timeseries_Sheet1_long_cleaned_box,aes(x=time,y=CFU,fill=groupingFinal))+geom_boxplot(alpha=.3,color="grey")+theme_classic()
plot_overtime_box


##----------------------
##model the data with bayani
##----------------------

library(growthrates)
library(broom)
library(drc)
library(patchwork)

Sys.sleep(5)
p     <- c(y0 = 500000, mumax = 1, K = 800000000,h0=8.044679)

# rmk202_strain_timeseries_Sheet1_long_cleaned_box$time <- as.numeric(rmk202_strain_timeseries_Sheet1_long_cleaned_box$time)
rmk202_strain_timeseries_Sheet1_long_cleaned_model <- rmk202_strain_timeseries_Sheet1_long_cleaned %>% filter(species!="total") %>% filter(group!="all_strains")
rmk202_strain_timeseries_Sheet1_long_cleaned_model  <- rmk202_strain_timeseries_Sheet1_long_cleaned_model %>% filter(!is.na(CFU))
rmk202_strain_timeseries_Sheet1_long_cleaned_model$groupingFinal <- as.factor(paste0(rmk202_strain_timeseries_Sheet1_long_cleaned_model$species,"_",rmk202_strain_timeseries_Sheet1_long_cleaned_model$group))


##----------model1
many_baranyi_sub <- all_growthmodels(
                   CFU ~ grow_baranyi(time, parms) | groupingFinal,
                   data = rmk202_strain_timeseries_Sheet1_long_cleaned_model, p=p,ncores = 8 )

results(many_baranyi_sub)

par(mfrow = c(12, 8))
par(mar = c(1, 1, 1, 1))
plot(many_baranyi_sub)


many_baranyi2_res <- results(many_baranyi_sub)
many_baranyi2_res
many_baranyi2_res_preped <- many_baranyi2_res
many_baranyi2_res_preped$mumax <- -many_baranyi2_res_preped$mumax
summary(many_baranyi2_res_preped)
table(many_baranyi2_res_preped$name)
mumax_plot <- ggplot(many_baranyi2_res_preped,aes(x=groupingFinal,y=mumax))+geom_boxplot()+theme_classic()+labs(x="",y="maximum pH decrease")
k0plot <- ggplot(many_baranyi2_res_preped,aes(x=groupingFinal,y=K))+geom_boxplot()+theme_classic()+labs(x="",y="lowest pH")

mumax_plot + k0plot+plot_layout(nrow = 2)


###---------------------
#boxplot
###---------------------
library(growthcurver)

fill_colers <- c("grey77","grey55","red","grey88","grey55","orange")
colorr_colers <- c("red","red","red","orange","orange","orange")

plot_overtime_box <-  ggplot()+geom_boxplot(data=rmk202_strain_timeseries_Sheet1_long_cleaned_box,aes(x=time,y=CFU,fill=groupingFinal,color=groupingFinal),alpha=.2)+scale_fill_manual(values=fill_colers)+scale_color_manual(values=colorr_colers)+theme_classic()+theme(legend.position = "bottom")
plot_overtime_box

fill_colers <- c("grey77","grey55","red","grey88","grey55","orange")
colorr_colers <- c("grey77","grey77","grey77","grey55","grey55","grey55")



plot_overtime_box <-  ggplot()+geom_boxplot(data=rmk202_strain_timeseries_Sheet1_long_cleaned_box,aes(x=time,y=CFU,fill=groupingFinal,color=groupingFinal),alpha=.3)+scale_fill_manual(values=colorr_colers)+scale_color_manual(values=fill_colers)+theme_classic()+theme(legend.position = "bottom")+facet_wrap(~time+species,ncol=14,scales = "free_x")
plot_overtime_box


rmk202_strain_timeseries_Sheet1_long_cleaned_box$time


##----------------------
##ribbon of interquartile range
##----------------------


group_growth_cfu_quantiles <- data.frame()
for (typess in unique(rmk202_strain_timeseries_Sheet1_long_cleaned$group)) {
  
  for (timesss in unique(rmk202_strain_timeseries_Sheet1_long_cleaned$time)) {
    
    
    for (speccc in unique(rmk202_strain_timeseries_Sheet1_long_cleaned$species)) {
      
      tmp <- rmk202_strain_timeseries_Sheet1_long_cleaned %>% filter(group==typess) %>% filter(time==timesss) %>% filter(species==speccc)%>% dplyr::select(CFU) %>% quantile(na.rm=TRUE) %>% as.data.frame()
      
      
      s25ss <- tmp[2,1]
      s75ss <- tmp[4,1]
meannn <- tmp[3,1]
      # quantile(x)
    # 
    # maxxx <- rmk202_strain_timeseries_Sheet1_long_cleaned %>% filter(group==typess) %>% filter(time==timesss) %>% filter(species==speccc)%>% select(CFU) %>% max(na.rm = TRUE)
    # minnn <- rmk202_strain_timeseries_Sheet1_long_cleaned %>% filter(group==typess) %>% filter(time==timesss) %>% filter(species==speccc)%>% select(CFU)%>% min(na.rm = TRUE)
    # meannn <- rmk202_strain_timeseries_Sheet1_long_cleaned %>% filter(group==typess) %>% filter(time==timesss)%>% filter(species==speccc)%>% select(CFU) %>% colMeans(na.rm = TRUE)
    # 
    tmps <- data.frame(sample=typess,species=speccc,time=timesss,twentyfifths=s25ss,seventhts=s75ss,mean=meannn)
    # 
    group_growth_cfu_quantiles <- rbind(group_growth_cfu_quantiles,tmps)
    
    }
  }
} 

# group_growth_cfu_quantiles


group_growth_cfu_quantiles_spec <- group_growth_cfu_quantiles %>% filter(species!="total") %>% filter(sample!="all_strains")%>% filter(!is.na(mean))

# plot_overtime <-  ggplot(group_growth_cfu_quantiles_spec,aes(x=time,y=mean,color=sample,fill=sample,group=interaction(sample,species)))+geom_line()+theme_classic()+facet_wrap(~species,nrow=2)
# plot_overtime
group_growth_cfu_quantiles_spec$groupingFinal <- as.factor(paste0(group_growth_cfu_quantiles_spec$species,"_",group_growth_cfu_quantiles_spec$sample))
group_growth_cfu_quantiles_spec$groupingFinal = factor(group_growth_cfu_quantiles_spec$groupingFinal, levels=c("S. thermophilus_rmk" ,"S. thermophilus_pairwise","S. thermophilus_strepto" ,"S. thermophilus_lacto","L. delbrueckii_rmk","L. delbrueckii_pairwise","L. delbrueckii_lacto","L. delbrueckii_strepto"))


# fill_colers <- c("grey77","grey55","red","grey88","grey55","orange")
fill_colers <- c("grey77","#97BC62FF","red","grey88","#339E66FF","orange")
# colorr_colers <- c("grey77","grey77","grey77","grey55","grey55","grey55")
fill_colers <- c("#FC766AFF","#783937FF","#F1AC88FF","#339E66FF","#078282FF","#95DBE5FF")

maxValue <- max(group_growth_cfu_quantiles_spec$seventhts)
plot_overtime_RIBBON <-  ggplot()+geom_ribbon(aes(x=time,ymin = twentyfifths , ymax = seventhts,group=interaction(sample,species),fill=groupingFinal),data = group_growth_cfu_quantiles_spec, alpha=.2)+theme_classic()+lims(x=c(0,24),y=c(0,maxValue))+scale_fill_manual(values=fill_colers)+scale_color_manual(values=fill_colers)+scale_x_continuous(breaks =c(0,6,12,18,24),labels=c(0,6,12,18,24))
plot_overtime_RIBBON


###models--------------quartile
group_growth_cfu_quantiles_spec$twentyfifths
df.predicted_model_upper <- data.frame()
df.predicted_model_lower <- data.frame()

for (samplezzz in unique(group_growth_cfu_quantiles_spec$groupingFinal)){
  
  tmp <- group_growth_cfu_quantiles_spec %>% filter(groupingFinal==samplezzz)
  
  # tmp$time <- as.double(tmp$time)
  model.wt <- SummarizeGrowth(tmp$time, tmp$seventhts)
  model.wt_lower <- SummarizeGrowth(tmp$time, tmp$twentyfifths)

# predict(model.wt$model)
# model.wt$data
tt <- seq(0,24, length=50)
# predict(model.wt$model,newdata=list(t=tt))

# data(model.wt$model)
# df.predicted <- data.frame(time = tmp$time, pred.wt = predict(model.wt$model,))
tmp <- data.frame(group=samplezzz,time = tt, pred.wt_upper = predict(model.wt$model,newdata=list(t=tt)), pred.wt_lower = predict(model.wt_lower$model,newdata=list(t=tt)))


df.predicted_model_upper <- rbind(df.predicted_model_upper,tmp)

# plot_overtime_box + geom_line(data=df.predicted, aes(x=time,y=pred.wt), color="red")
}

logCFU <- ggplot()+geom_ribbon(aes(x=time,ymin = pred.wt_lower , ymax = pred.wt_upper,group=group,fill=group),data = df.predicted_model_upper, alpha=.2)+theme_classic()+scale_y_continuous(trans = 'log10')+labs(y="CFU/ml")#+coord_trans(y="log10")

nonLOG <- ggplot()+geom_ribbon(aes(x=time,ymin = pred.wt_lower , ymax = pred.wt_upper,group=group,fill=group),data = df.predicted_model_upper, alpha=.2)+theme_classic()+labs(y="CFU/ml")

logCFU+nonLOG+plot_layout(nrow = 2)


# svg("~/Desktop/Projects/2019_RMK202_analysis/plot/growth_curve.svg",width=8,height=8)
    png("~/Desktop/Projects/2019_RMK202_analysis/plot/growth_curve_CFU.png", width = 1800, height = 1400,res=300)

logCFU+nonLOG+plot_layout(nrow = 2)

 
dev.off()




##----------------------------------
##modelling
##----------------------------------
##----------model2
df.predicted_model <- data.frame()
for (samplezzz in unique(rmk202_strain_timeseries_Sheet1_long_cleaned_model$groupingFinal)){
  
  tmp <- rmk202_strain_timeseries_Sheet1_long_cleaned_model %>% filter(groupingFinal==samplezzz)
  # tmp$time <- as.double(tmp$time)
  model.wt <- SummarizeGrowth(tmp$time, tmp$CFU)
# predict(model.wt$model)
# model.wt$data
tt <- seq(0,24, length=50)
# predict(model.wt$model,newdata=list(t=tt))

# data(model.wt$model)
# df.predicted <- data.frame(time = tmp$time, pred.wt = predict(model.wt$model,))
tmp <- data.frame(group=samplezzz,time = tt, pred.wt = predict(model.wt$model,newdata=list(t=tt)))


df.predicted_model <- rbind(df.predicted_model,tmp)

# plot_overtime_box + geom_line(data=df.predicted, aes(x=time,y=pred.wt), color="red")
}

df.predicted_model$group = factor(df.predicted_model$group, levels=c("S. thermophilus_rmk" ,"S. thermophilus_pairwise","S. thermophilus_strepto" ,"S. thermophilus_lacto","L. delbrueckii_rmk","L. delbrueckii_pairwise","L. delbrueckii_lacto","L. delbrueckii_strepto"))

 df.predicted_model$species <- str_split_fixed(df.predicted_model$group, "_", 2)[,1]


plot_overtime_model <- ggplot()+ geom_line(data=df.predicted_model, aes(x=time,y=pred.wt,group=group,color=group,linetype=species),size =1.25)+theme_classic()+lims(x=c(0,24),y=c(0,maxValue))+scale_color_manual(values=fill_colers)+scale_linetype_manual(values=c("dashed", "dotted"))+scale_x_continuous(breaks =c(0,6,12,18,24),labels=c(0,6,12,18,24))
plot_overtime_model
# 
plot_overtime_RIBBON + plot_overtime_model+plot_layout(nrow = 2)
svg("~/Desktop/Projects/2019_RMK202_analysis/plot/growth_curve.svg",width=8,height=8)
# 
 plot_overtime_RIBBON + plot_overtime_model+plot_layout(nrow = 2)

 
dev.off()

##--------------------------------
##phase
##--------------------------------

 df.predicted_model$groupsss <- str_split_fixed(df.predicted_model$group, "_", 2)[,2]


df.predicted_model$group
lagsss_mean

lagsss_mean$grouping_new <- plyr::revalue(lagsss_mean$grouping, c("lactobacillus"="lacto", "starterCulture"="rmk", "streptococcus"="strepto"))
  lagsss_mean_prepped <- lagsss_mean %>%  dplyr::select(c("grouping_new","hours","hours_corrected"))

  df.predicted_model_extended <- merge(df.predicted_model,lagsss_mean_prepped,by.x="groupsss",by.y="grouping_new") 
  # df.predicted_model_extended$timeCurated <- df.predicted_model_extended$time -df.predicted_model_extended$hours_corrected
  
  df.predicted_model_extended$timeCurated <-  ifelse(df.predicted_model_extended$groupsss=="rmk",df.predicted_model_extended$time -df.predicted_model_extended$hours,df.predicted_model_extended$time -df.predicted_model_extended$hours_corrected)
  
  
df.predicted_model_extended <- df.predicted_model_extended %>% filter(timeCurated<=18) %>% filter(timeCurated>=minTime)
  
plot_overtime_model <- ggplot()+ geom_line(data=df.predicted_model_extended, aes(x=timeCurated,y=pred.wt,group=group,color=group,linetype=species),size =1.25)+theme_classic()+lims(x=c(minTime,18),y=c(0,maxValue))+labs(y="CFU/ml")+scale_color_manual(values=fill_colers)+scale_linetype_manual(values=c("dashed", "dotted"))+scale_x_continuous(breaks =c(0,6,12,18),labels=c(0,6,12,18)) +theme(legend.position = "none")#+coord_trans(y="log2")
plot_overtime_model
  
##-------------------Ribbon

group_growth_cfu_quantiles_spec$groupingFinal <- as.factor(paste0(group_growth_cfu_quantiles_spec$species,"_",group_growth_cfu_quantiles_spec$sample))
group_growth_cfu_quantiles_spec$groupingFinal = factor(group_growth_cfu_quantiles_spec$groupingFinal, levels=c("S. thermophilus_rmk" ,"S. thermophilus_pairwise","S. thermophilus_strepto" ,"S. thermophilus_lacto","L. delbrueckii_rmk","L. delbrueckii_pairwise","L. delbrueckii_lacto","L. delbrueckii_strepto"))
 # df.predicted_model$species <- str_split_fixed(group_growth_cfu_quantiles_spec$group, "_", 2)[,1]


# fill_colers <- c("grey77","grey55","red","grey88","grey55","orange")
fill_colers <- c("grey77","#97BC62FF","red","grey88","#339E66FF","orange")
# colorr_colers <- c("grey77","grey77","grey77","grey55","grey55","grey55")
fill_colers <- c("#FC766AFF","#783937FF","#F1AC88FF","#339E66FF","#078282FF","#95DBE5FF")
maxValue <- max(group_growth_cfu_quantiles_spec$seventhts)
minnsValue <- min(group_growth_cfu_quantiles_spec$seventhts)

group_growth_cfu_quantiles_spec_curated <- merge(group_growth_cfu_quantiles_spec,lagsss_mean_prepped,by.x="sample",by.y="grouping_new") 
# group_growth_cfu_quantiles_spec_curated$timeCurated <- group_growth_cfu_quantiles_spec_curated$time -group_growth_cfu_quantiles_spec_curated$hours_corrected

group_growth_cfu_quantiles_spec_curated$timeCurated <- ifelse(group_growth_cfu_quantiles_spec_curated$sample=="rmk",group_growth_cfu_quantiles_spec_curated$time -group_growth_cfu_quantiles_spec_curated$hours,group_growth_cfu_quantiles_spec_curated$time -group_growth_cfu_quantiles_spec_curated$hours_corrected)
group_growth_cfu_quantiles_spec_curated <- group_growth_cfu_quantiles_spec_curated  %>% filter(timeCurated>=minTime)#%>% filter(timeCurated<=18) %>% filter(timeCurated>=minTime)

plot_overtime_RIBBON <-  ggplot()+geom_ribbon(aes(x=timeCurated,ymin = twentyfifths , ymax = seventhts,group=interaction(sample,species),fill=groupingFinal),data = group_growth_cfu_quantiles_spec_curated, alpha=.2)+theme_classic()+lims(x=c(minTime,18),y=c(minnsValue,maxValue))+scale_fill_manual(values=fill_colers)+scale_color_manual(values=fill_colers)+scale_x_continuous(breaks =c(0,6,12,18),labels=c(0,6,12,18))+theme(legend.position = "none")#+coord_trans(y="log2")
plot_overtime_RIBBON
  # ,y=c(1,maxValue)


# plot_overtime_RIBBON + plot_overtime_model+plot_layout(nrow = 2)
 plot_overtime_RIBBON + plot_overtime_model+(low_phased_ribbon+theme(legend.position = "none"))+plot_layout(nrow = 3)

svg("~/Desktop/Projects/2019_RMK202_analysis/plot/growth_curve.svg",width=6,height=9)
# 
 plot_overtime_RIBBON + plot_overtime_model+(low_phased_ribbon+theme(legend.position = "none"))+plot_layout(nrow = 3)

 
dev.off()

##--------------------------------
##stats
##compare final growth value
##--------------------------------
table(rmk202_strain_timeseries_Sheet1_long_cleaned_model$group)
table(rmk202_strain_timeseries_Sheet1_long_cleaned_model$Method)
table(rmk202_strain_timeseries_Sheet1_long_cleaned_model$groupingFinal)
table(rmk202_strain_timeseries_Sheet1_long_cleaned_model$time)


ldel_vergleich_final_01 <- rmk202_strain_timeseries_Sheet1_long_cleaned_model %>% filter(groupingFinal=="L. delbrueckii_lacto") %>% filter(time=="18")
ldel_vergleich_final_02 <- rmk202_strain_timeseries_Sheet1_long_cleaned_model %>% filter(groupingFinal %in% c("L. delbrueckii_pairwise","L. delbrueckii_rmk")) %>% filter(time=="18")

t.test(ldel_vergleich_final_01$CFU,ldel_vergleich_final_02$CFU)

###between Sterm and Ldel

Sterm_vergleich_final_01 <- rmk202_strain_timeseries_Sheet1_long_cleaned_model %>% filter(groupingFinal %in% c("S. thermophilus_pairwise","S. thermophilus_rmk")) %>% filter(time=="18")
# ldel_vergleich_final_02 <- rmk202_strain_timeseries_Sheet1_long_cleaned_model %>% filter(groupingFinal %in% c("L. delbrueckii_pairwise","L. delbrueckii_rmk")) %>% filter(time=="18")

t.test(Sterm_vergleich_final_01$CFU,ldel_vergleich_final_02$CFU)


```




#isolates

##### Amino acid auxotrophy
In order to detect amino acid auxotrophies I use [gapmind](http://papers.genomics.lbl.gov/cgi-bin/gapView.cgi)
```{bash}
/home/vincent/apps/genix/scripts/genbank2faa.py /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.current.gb /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.current.faa

sed -i 's/\|//g' /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.current.faa

```

##### Phaster analysis

Here, I analyse the zipped output I downloaded from phaster online. 
I want to now which genomes have a prophages (from category questionable or higher).
I first have to download and rename all folders. 
I do this locally on my laptop and manually.

```{bash}


grep "^gi" -A 10 summary.txt  |grep "^gi" -v | sed 's/^[ \t]*//;s/[ \t]*$//' |grep "^-" -v
```





#metagenome Names

```{r}
library(RColorBrewer)

metasample_colors <- data.frame(sample=c("lyo202_96", "L_202_2014","RMK202","L_202_2012","th_K2_6h" ,"Konserve_202","di_K2_6h","Versand_202"),colors=
c(brewer.pal(9,name="YlGnBu")[c(5,4,3,2)],"lightblue","darkturquoise","darkcyan","#0062B4FF"))

metasample_colors <- metasample_colors %>% mutate(shortfinal = gsub("L_", "Lyo_", sample)) %>% mutate(shortfinal = gsub("th_K2_6h", "th_K2_8h", shortfinal))

metasample_colors$samplefinal_multipleLines <- plyr::revalue(metasample_colors$shortfinal, c("lyo202_96"="Lyo\n1996","Lyo_202_2012"="Lyo\n2012", "Lyo_202_2014"="Lyo\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018","di_K2_6h"="cheesemaking\nday1","th_K2_8h"="cheesemaking\nday2"))
  
metasample_colors$samplefinal_oneLines <- plyr::revalue(metasample_colors$shortfinal, c("lyo202_96"="Lyo 1996","Lyo_202_2012"="Lyo 2012", "Lyo_202_2014"="Lyo 2014", "Konserve_202"="working stock", "RMK202"="starter culture 2012", "Versand_202"="starter culture 2018","di_K2_6h"="cheesemaking day1","th_K2_8h"="cheesemaking day2"))
  
  
metasample_colors$shortfinal_2 <- plyr::revalue(metasample_colors$shortfinal, c("lyo202_96"="L_96","Lyo_202_2012"="L_2012", "Lyo_202_2014"="L_2014", "Konserve_202"="K_", "RMK202"="R202", "Versand_202"="V__202","di_K2_6h"="d2_6h","th_K2_8h"="t2_8h"))

metasample_colors$final <- plyr::revalue(metasample_colors$shortfinal, c("lyo202_96"="starter culture 1","Lyo_202_2012"="starter culture 2", "Lyo_202_2014"="starter culture 3", "Konserve_202"="starter culture 4", "RMK202"="starter culture 5", "Versand_202"="starter culture 6","di_K2_6h"="starter culture 7","th_K2_8h"="starter culture 8"))
  
##-----------------colorschemas


palePurple <- brewer.pal(3,name="Set2")[3] ## low
paleOrange <- brewer.pal(3,name="Set2")[2] ## medium
paleGreen<-brewer.pal(3,name="Set2")[1] ##  High

Pcolors <- c(palePurple , paleOrange, paleGreen)

three_blues <- c(brewer.pal(9,name="YlGnBu")[c(4,6,8)],"lightgray")
three_reds <-c(brewer.pal(9,name="YlOrRd")[c(4,6)],
							 brewer.pal(3,name="Reds")[3],"lightgray")
three_browns <-c(brewer.pal(9,name="YlOrBr")[c(7,8,9)],"lightgray")

```


### renaming

```{bash}

#old labels
cat /archiv/Projects/2018_Culturomics/02_script/names_changed.txt
#new labels
181213_3	L70	mst3
181213_4	L44	mst10
181213_5	S50	mst1
181213_6	L108	mst7
181213_7	S72	mst2
181213_8	L99	mst8
181213_10	L104	mst6
181213_11	L80	mst5
181213_12	L35	mst9
181213_13	L71	mst4

```

###old strains

```{bash}

mkdir -p /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz
cd /archiv/Projects/2018_Culturomics/archiv/Projects/2018_Culturomics/01_rawData/OldStrains/fq/gz

##check md5sum
rm sum1
for file in $(awk '{print $2}' md5sum.txt)
do

md5sum ${file} >> sum1

done
```


###FastQC
In a first step we creat [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) reports for all the libraries. 
These are then visualized with [MultiQC](https://multiqc.info/)

```{bash fastqc}
source /archiv/Projects/2018_Culturomics/02_script/fileShortcuts.sh
threads=8

mkdir -p $mainPath/01_rawData/IlluminaFastQC/


for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/names_changed.txt)
  do
  echo ${id}

    
    fastqc $mainPath/01_rawData/fastq/gz/${id}_R1.fastq.gz -t ${threads} -o $mainPath/01_rawData/IlluminaFastQC/
    fastqc $mainPath/01_rawData/fastq/gz/${id}_R2.fastq.gz -t ${threads} -o $mainPath/01_rawData/IlluminaFastQC/
    
  done
  mkdir -p $mainPath/01_rawData/IlluminaFastQC/Multiqc,output_file='test.html')

  
  multiqc $mainPath/01_rawData/IlluminaFastQC/ -o $mainPath/01_rawData/IlluminaFastQC/Multiqc

##Merging both RMK202

zcat $mainPath/01_rawData/fastq/gz/Konserve_202_R1.fastq.gz $mainPath/01_rawData/fastq/gz/Versand_202_R1.fastq.gz | gzip > $mainPath/01_rawData/fastq/gz/RMK202_R1.fastq.gz

zcat $mainPath/01_rawData/fastq/gz/Konserve_202_R2.fastq.gz $mainPath/01_rawData/fastq/gz/Versand_202_R2.fastq.gz | gzip > $mainPath/01_rawData/fastq/gz/RMK202_R2.fastq.gz

##Merging all isolates

zcat $mainPath/01_rawData/fastq/gz/L104_R1.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L108_R1.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L35_R1.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L39_R1.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L44_R1.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L70_R1.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L71_R1.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L80_R1.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L99_R1.fastq.gz \
  $mainPath/01_rawData/fastq/gz/S50_R1.fastq.gz \
  $mainPath/01_rawData/fastq/gz/S72_R1.fastq.gz \
  | gzip > $mainPath/01_rawData/fastq/gz/isolates_R1.fastq.gz

zcat $mainPath/01_rawData/fastq/gz/L104_R2.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L108_R2.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L35_R2.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L39_R2.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L44_R2.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L70_R2.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L71_R2.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L80_R2.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L99_R2.fastq.gz \
  $mainPath/01_rawData/fastq/gz/S50_R2.fastq.gz \
  $mainPath/01_rawData/fastq/gz/S72_R2.fastq.gz \
  | gzip > $mainPath/01_rawData/fastq/gz/isolates_R2.fastq.gz

##Merging both RMK202

base=$mainPath/01_rawData/fastq/gz/

zcat $base/Konserve_202_R1.fastq.gz $base/Versand_202_R1.fastq.gz $base/RMK202_old_R1.fastq.gz  | gzip > $base/RMK202_withOld_R1.fastq.gz

zcat $base/Konserve_202_R2.fastq.gz $base/Versand_202_R2.fastq.gz $base/RMK202_old_R2.fastq.gz $ | gzip > $base/RMK202_withOld_R2.fastq.gz


```

### Trim-galore
In the following we clip adaptors with [trim-galore]()

```{bash trimGalore}
source /archiv/Projects/2018_Culturomics/02_script/fileShortcuts.sh
threads=30
id=RMK202_withOld

for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/names_changed_new.txt)
  do
  echo "##=============="
  echo ${id}
  echo "##=============="
      rm -r $mainPath/01_rawData/trimmed_Galore/gz/${id}
      mkdir -p $mainPath/01_rawData/trimmed_Galore/gz/${id}
      trim_galore -paired -o $mainPath/01_rawData/trimmed_Galore/gz/${id} \
        $mainPath/01_rawData/fastq/gz/${id}_R1.fastq.gz  \
        $mainPath/01_rawData/fastq/gz/${id}_R2.fastq.gz 
        
  echo "##=============="
  echo ${id} "fastQC of trimmed"
  echo "##=============="
      
    mkdir -p $mainPath/01_rawData/trimmed_Galore/fastQC/
      
    fastqc $mainPath/01_rawData/trimmed_Galore/gz/${id}/${id}_R1_val_1.fq.gz -t ${threads} -o $mainPath/01_rawData/trimmed_Galore/fastQC/
    
    fastqc $mainPath/01_rawData/trimmed_Galore/gz/${id}/${id}_R2_val_2.fq.gz -t ${threads} -o $mainPath/01_rawData/trimmed_Galore/fastQC/
    
        
  done 
  

  mkdir -p $mainPath/01_rawData/trimmed_Galore/fastQC/Multiqc,output_file='test.html')

  multiqc $mainPath/01_rawData/trimmed_Galore/fastQC/ -o $mainPath/01_rawData/trimmed_Galore/fastQC//Multiqc




```

###spades

In this step we assemble the raw reads of the isolate with [spades](http://bioinf.spbau.ru/en/spades3.7).

```{bash spades}
source /archiv/Projects/2018_Culturomics/02_script/fileShortcuts.sh

threads=35

for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/names_changed.txt)
  do
  echo ==========================
   echo ${id}
   echo ==========================
 
 
       mkdir -p $mainPath/06_Spades/${id}/assembly/
 
 
     /home/vincent/miniconda2/pkgs/spades-3.13.0-0/bin/spades.py --pe1-1 $mainPath/01_rawData/trimmed_Galore/gz/${id}/${id}_R1_val_1.fq.gz \
       --pe1-2 $mainPath/01_rawData/trimmed_Galore/gz/${id}/${id}_R2_val_2.fq.gz \
       -o $mainPath/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta -t ${threads} -m 100

   
    done

#---------------
##short info
##---------------

for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/isolates_names_changed.txt)
  do
  echo ==========================
   echo ${id}
   echo ==========================
 
   grep ">" -c $mainPath/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/contigs.fasta 
    done
##---------------
##PostAssemblyAnalysis
##---------------

for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/isolates_names_changed.txt)
  do
  echo ==========================
   echo ${id}
   echo ==========================
    
    #tail -3 $mainPath/03_metaphlan2/${id}_trimmed/metaphlan_profile/profiled_metagenome_${id}_trimmed.txt
    PostAssemblyAnalysis.R -a $mainPath/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/scaffolds.fasta \
      -F $mainPath/01_rawData/trimmed_Galore/gz/${id}/${id}_R1_val_1.fq.gz \
      -R $mainPath/01_rawData/trimmed_Galore/gz/${id}/${id}_R2_val_2.fq.gz \
      -o $mainPath/06_Spades/${id}/assembly/PostaAssembly \
      -t 35
    
    done


```



### megabat2

here, we try to remove the Lactobacillus delbrueckii contamination from the Streptococcus thermophilus isolates with [metabat2](https://bitbucket.org/berkeleylab/metabat)

```{bash metabat}
threads=10
id=S50
date=20190203
##-------
##S50
##-------
mkdir -p /archiv/Projects/2018_Culturomics/06_Spades//${id}/metabat2/${date}/

metabat2 -t ${threads} \
  -i /archiv/Projects/2018_Culturomics/06_Spades//${id}/assembly/assemblyMetaSpades_20190121.fasta/scaffolds.fasta \
  /archiv/Projects/2018_Culturomics/06_Spades/S50/assembly/PostaAssembly/Allillumina2assembly/rawIllumina_bwamem_Scaffolds_sorted.bam \
  -o /archiv/Projects/2018_Culturomics/06_Spades//${id}/metabat2/${date}/binning_contigs


id=S72
##-------
##S72
##-------
mkdir -p /archiv/Projects/2018_Culturomics/06_Spades//${id}/metabat2/${date}/

metabat2 -t ${threads} \
  -i /archiv/Projects/2018_Culturomics/06_Spades//${id}/assembly/assemblyMetaSpades_20190121.fasta/scaffolds.fasta \
  /archiv/Projects/2018_Culturomics/06_Spades/S50/assembly/PostaAssembly/Allillumina2assembly/rawIllumina_bwamem_Scaffolds_sorted.bam \
  -o /archiv/Projects/2018_Culturomics/06_Spades//${id}/metabat2/${date}/binning_contigs
  
  
##-------
##BLAST QC
##-------

for genome in $(ls /archiv/Projects/2018_Culturomics/06_Spades//${id}/metabat2/${date}/)
  do
  
  genome_renamed=$(echo $genome |sed 's/.fa//g'  )
  
  #mkdir -p /archiv/Projects/2018_Culturomics/06_Spades//${id}/metabat2/${date}/${genome_renamed}_metaphlan
  
  
   infoseq /archiv/Projects/2018_Culturomics/06_Spades//${id}/metabat2/${date}/$genome 
   
  
 
  done

```
###Gather all strain for analysis

Importantly gather all strains first.
The strains are from the following occassions:
1. Metagenome assembled genomes (MAG)
2. Reference strains picked from the genotyping
3. old Strains extracted out of the same strarter culture

```{bash}
species=Sterm
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}

##metagenome
samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/PGAP_prep/S_thermophilus_RMK202.fasta S_thermophilus_mag_rmk202 > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/202-SMAG.fna
##picked strains (older)
cat /archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/S50.fna > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/202-S50.fna
cat /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/S72.fna > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/202-S72.fna
##old strains isolated from RMK202
for genomesss in $(ls /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Dialact/Sterm/ |grep ".fna")
do
newGenomss=$(echo ${genomesss}|cut -d '_' -f 1|sed 's/FAM//g')
cat /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Dialact/Sterm/${genomesss} > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/202-${newGenomss}.fna
done
##-------------------------------------------------
species=Ldel
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}

##metagenome
samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/PGAP_prep/L_delbrueckii_RMK202_02.fasta L_delbrueckii_mag_rmk202 > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/202-LMAG.fna
##picked strains (older)
#cp /archiv/Projects/2018_Culturomics/genomes/L*.fna  /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/
for genomesss in $(ls /archiv/Projects/2018_Culturomics/genomes/ |grep "^L"|sed 's/.fna//g')
do
cat /archiv/Projects/2018_Culturomics/genomes//${genomesss}.fna > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/202-${genomesss}.fna
done
##old strains isolated from RMK202
for genomesss in $(ls /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Dialact/${species}/ |grep ".fna")
do
newGenomss=$(echo ${genomesss}|cut -d '_' -f 1|sed 's/FAM//g')
cat /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Dialact/${species}/${genomesss} > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/202-${newGenomss}.fna
done

```

rename contig names
```{bash}
for species in $(echo "Ldel Sterm")
do
mkdir -p  /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/
for genomesss in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/ |grep ".fna" |sed 's/.fna//g')
do
num=1
cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/${genomesss}.fna > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/${genomesss}.fna
for contigzz in $(grep "^>" /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/${genomesss}.fna)
do
newContigName=$(echo -e ${genomesss}"-"${num})

sed -i "s/${contigzz}/>${newContigName}/g" /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/${genomesss}.fna

num=$((num+1))

done #contigzz

done #genomesss
done #species
```

###start align

```{bash}

threads=37
for species in $(echo "Ldel Sterm")
do
rm -r  /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned
REF=$(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/ |grep "MAG"|grep ".fna$")
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs//${REF}.*
bwa index /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs//${REF}
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/tmp/

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs//${REF}  > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/${REF}
for genomess in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/ |grep ".fna$"|cut -d '_' -f 1|grep "MAG" -v |sed 's/.fna//g')
do

#/archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs//202-SMAG.fna

mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/startaligned/Aligned/${species}/${genomess}

bwa mem -x intractg /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs//${REF} /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs//${genomess}.fna | samtools sort -@${threads} -O BAM -o /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/tmp/${genomess}_allcontigsMapped2CRISPRs.bam -

samtools view /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/tmp/${genomess}_allcontigsMapped2CRISPRs.bam |awk -F "\t" '{OFS="\t"}{if($2=="0" ||$2=="16"||$2=="4")print $0}'|uniq  > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/tmp/${genomess}_allcontigsMapped2CRISPRs.cleaned

#testcase
#species=Ldel
#genomess=202-11141
#numfile=1
#rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/${genomess}.fna

num=1
numfile=$(wc -l /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/tmp/${genomess}_allcontigsMapped2CRISPRs.cleaned| cut -d ' ' -f 1)
#cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/tmp/${genomess}_allcontigsMapped2CRISPRs.cleaned | 
for linezzzzz in $(seq -s' ' 1 $numfile)
do
#liyyye=$(sed -n "${linezzzzz}p" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/startaligned/Aligned/${species}/${genomess}/allcontigsMapped2CRISPRs.cleaned)
#echo -e ${liyyye}
 


lineQual=$(sed -n "${linezzzzz}p" /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/tmp/${genomess}_allcontigsMapped2CRISPRs.cleaned |cut -f 2)
contigName=$(sed -n "${linezzzzz}p" /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/tmp/${genomess}_allcontigsMapped2CRISPRs.cleaned |cut -d '-' -f 1,2)
echo -e ${lineQual} ${contigName}
#sed -n "${linezzzzz}p" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/startaligned/Aligned/${species}/${genomess}/allcontigsMapped2CRISPRs.cleaned |cut -f 8

if [ $lineQual == "0" ]; then
              echo -e $lineQual" equals 0"
            echo -e ">"${contigName}"-"${num} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/${genomess}.fna
           sed -n "${linezzzzz}p" //archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/tmp/${genomess}_allcontigsMapped2CRISPRs.cleaned  | cut -f 10  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/${genomess}.fna
            elif [[ $lineQual == "16" ]]
            then
           echo -e $lineQual" equals 16"
            echo -e ">"${contigName}"-"${num} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/${genomess}.fna
            sed -n "${linezzzzz}p" //archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/tmp/${genomess}_allcontigsMapped2CRISPRs.cleaned  |cut -f 10   >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/${genomess}.fna
            #revseq -sequence /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/tmp/tmpfile.fasta -outseq /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/tmp/tmpfile_rev_compement.fasta -notag 
           
           # cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/tmp/tmpfile_rev_compement.fasta >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/${genomess}.fna
            
            else
            echo -e $lineQual" equals 4"
               echo -e ">"${contigName}"-"${num} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/${genomess}.fna
         sed -n "${linezzzzz}p" //archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/tmp/${genomess}_allcontigsMapped2CRISPRs.cleaned  |cut -f 10    >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/${genomess}.fna
               
            fi
  
 num=$((num+1)) 
  
done

fasta_formatter -w 50 -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/${genomess}.fna -o /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/${genomess}_cleaned.fna
mv  /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/${genomess}.fna  /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/tmp/${genomess}_finalsed.fna
mv /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/${genomess}_cleaned.fna /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/${genomess}.fna

done #genomes
done # species


samtools view /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/tmp/${genomess}_allcontigsMapped2CRISPRs.bam |grep -P "202-12105-1\t" |cut -f 1,2,3,4,5,6

samtools view /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/startaligned/Aligned/${species}/${genomess}/allcontigsMapped2CRISPRs.bam |grep -P "202-12105-6\t" |cut -f 1,2,3,4,5,6

##test

seq_length.py /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/202-11141.fna |head
seq_length.py /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/tmp/202-11141_finalsed.fna|head
cut -f 1,2 /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/tmp/202-11141_allcontigsMapped2CRISPRs.cleaned |head


```
start align again


```{bash}
threads=37
for species in $(echo "Ldel Sterm")
do
rm -r  /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned
REF=$(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs//StartAligned |grep "MAG"|grep ".fna$")
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs///StartAligned/${REF}.*
bwa index /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs//StartAligned/${REF}
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/tmp/

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs//StartAligned/${REF}  > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/${REF}
for genomess in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/ |grep ".fna$"|cut -d '_' -f 1|grep "MAG" -v |sed 's/.fna//g')
do

#/archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs//202-SMAG.fna


bwa mem -x intractg /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned//${REF} /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs//StartAligned/${genomess}.fna | samtools sort -@${threads} -O BAM -o /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/tmp/${genomess}_allcontigsMapped2CRISPRs.bam -

samtools view /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/tmp/${genomess}_allcontigsMapped2CRISPRs.bam |awk -F "\t" '{OFS="\t"}{if($2=="0" ||$2=="16"||$2=="4")print $0}' |uniq > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/tmp/${genomess}_allcontigsMapped2CRISPRs.cleaned


num=1
numfile=$(wc -l /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/tmp/${genomess}_allcontigsMapped2CRISPRs.cleaned| cut -d ' ' -f 1)
#cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/tmp/${genomess}_allcontigsMapped2CRISPRs.cleaned | 

for linezzzzz in $(seq -s' ' 1 $numfile)
do
#liyyye=$(sed -n "${linezzzzz}p" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/startaligned/Aligned/${species}/${genomess}/allcontigsMapped2CRISPRs.cleaned)
#echo -e ${liyyye}
 


lineQual=$(sed -n "${linezzzzz}p" /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/tmp/${genomess}_allcontigsMapped2CRISPRs.cleaned |cut -f 2)
contigName=$(sed -n "${linezzzzz}p" /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/tmp/${genomess}_allcontigsMapped2CRISPRs.cleaned |cut -d '-' -f 1,2)
echo -e ${lineQual} ${contigName}
#sed -n "${linezzzzz}p" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/startaligned/Aligned/${species}/${genomess}/allcontigsMapped2CRISPRs.cleaned |cut -f 8

if [ $lineQual == "0" ]; then
           #   echo -e $lineQual" equals 0"
            echo -e ">"${contigName}"-"${num} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/${genomess}.fna
           sed -n "${linezzzzz}p" //archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/tmp/${genomess}_allcontigsMapped2CRISPRs.cleaned  | cut -f 10  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/${genomess}.fna
            elif [[ $lineQual == "16" ]]
            then
          #  echo -e $lineQual" equals 16"
            echo -e ">"${contigName}"-"${num} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/${genomess}.fna
            sed -n "${linezzzzz}p" //archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/tmp/${genomess}_allcontigsMapped2CRISPRs.cleaned  |cut -f 10   >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/${genomess}.fna
            #revseq -sequence /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/tmp/tmpfile.fasta -outseq /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/tmp/tmpfile_rev_compement.fasta -notag 
           
            #cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/tmp/tmpfile_rev_compement.fasta >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned//StartAligned/${genomess}.fna
            
            else
            #echo -e $lineQual" equals 4"
               echo -e ">"${contigName}"-"${num} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/${genomess}.fna
         sed -n "${linezzzzz}p" //archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/tmp/${genomess}_allcontigsMapped2CRISPRs.cleaned  |cut -f 10    >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/${genomess}.fna
               
            fi
  
 num=$((num+1)) 
  
done

fasta_formatter -w 50 -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/${genomess}.fna -o /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/${genomess}_cleaned.fna

mv  /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/${genomess}.fna  /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/tmp/${genomess}_finalsed.fna

mv /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/${genomess}_cleaned.fna /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/${genomess}.fna

done #genomes
done # species


##test old
species=Ldel
genomess=202-11141
#numfile=1
seq_length.py /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/202-11141.fna |head
seq_length.py /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/tmp/202-11141_finalsed.fna|head
cut -f 1,2 /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/tmp/202-11141_allcontigsMapped2CRISPRs.cleaned |head
```

this chunk is to test the previous stuff
```{bash}
##test new

seq_length.py /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/202-11141.fna |head
seq_length.py /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/tmp/202-11141_finalsed.fna|head

cut -f 1,2 /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/tmp/202-11141_allcontigsMapped2CRISPRs.cleaned |head
cut -f 1,2 /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/tmp/202-11141_allcontigsMapped2CRISPRs.cleaned |head
cut -f 2 /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/tmp/202-11141_allcontigsMapped2CRISPRs.cleaned |sort|uniq -c

head -c 100 /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/tmp/202-11141_allcontigsMapped2CRISPRs.cleaned
head -c 100 /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/tmp/202-11141_allcontigsMapped2CRISPRs.cleaned


cut -f 1,2,6 /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/tmp/202-11141_allcontigsMapped2CRISPRs.cleaned |head
cut -f 1,2,6 /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/tmp/202-11141_allcontigsMapped2CRISPRs.cleaned |head


cut -f 1,2 /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/Ldel/renamedContigs/StartAligned/tmp/${genomess}_allcontigsMapped2CRISPRs.cleaned |head
cut -f 1,2 /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/Ldel/renamedContigs/StartAligned/StartAligned/tmp/${genomess}_allcontigsMapped2CRISPRs.cleaned |head
grep ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/Ldel/renamedContigs/StartAligned/${genomess}.fna |head
samtools view /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/tmp/${genomess}_allcontigsMapped2CRISPRs.bam |grep -P "202-12105-1\t" |cut -f 1,2,3,4,5,6

samtools view /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/startaligned/Aligned/${species}/${genomess}/allcontigsMapped2CRISPRs.bam |grep -P "202-12105-6\t" |cut -f 1,2,3,4,5,6

```
Also after two start alignment some contigs still map in reverse complement. this is because they are duplicates or low complexity and map either way. 

move local

```{bash}
mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Sterm/
scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/Sterm/renamedContigs/StartAligned/StartAligned/ ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Sterm/

```

##include ONT
Here, I include the seperately calculated ONT genomes

```{bash}

species=Sterm
starterculture=202

ll /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/ 

for sample in $(echo "13494
13496
13498
13499
24737
24738
24739
24740
24853
24854
24855")
do

echo ${sample}
cat /data/Project/2020_StarterCultureDiversity/09_flye/StartAlignAfterPolishr/${sample}/aligned/8_fourthFreebayes_startAligned.fasta > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/${starterculture}-${sample}.fna

done


ll /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/ |grep ".fna"



##and others

species=Ldel
#ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/${species}/ 
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}
for genomess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/${species}/ |grep "_202_"|grep ".fasta$" |sed 's/.fasta//g')
do
cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/${species}/${genomess}.fasta /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/

done

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/202-LMAG.fna >  /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/L_I_202_LMAG.fasta

species=Sterm
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}
for genomess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/${species}/ |grep "_202_"|grep ".fasta$" |sed 's/.fasta//g')
do
cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/${species}/${genomess}.fasta /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/

done

ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/ |grep ".fna$"|grep -v "CRISPR" -i

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/202-13491.fna >  /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/S_I_202_13491.fasta

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/202-13494.fna >  /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/S_O_202_13494.fasta

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/202-13496.fna >  /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/S_O_202_13496.fasta

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/202-13499c1.fna >  /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/S_I_202_13499c1.fasta

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/202-13499c2.fna >  /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/S_I_202_13499c2.fasta

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/202-S50.fna >  /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/S_I_202_S50.fasta

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/202-S72.fna >  /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/S_I_202_S72.fasta

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/202-SMAG.fna >  /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/S_O_202_SMAG.fasta

```




##Annotation

Prokka
```{bash}
for species in $(echo "Ldel Sterm")
do
species=Ldel
for genomess in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/ |grep "_202_"|grep ".fasta$" |sed 's/.fasta//g')
#for genomess in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/ |grep ".fna$"|grep -v "CRISPR" -i|cut -d '_' -f 1|sed 's/.fna//g')
do
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/

##I have to change "_" to "-"
locusTag=$(echo ${genomess}|sed 's/_/-/g')

/usr/bin/perl /usr/bin/prokka --cpus 38 --outdir /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/ --force --usegenus --genus ${species} --locustag ${locusTag} --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}//${genomess}.fasta

#grep "${id}"  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PROKKA_plasmids/$id/PROKKA*.gff |sed 's/Parent.*product=//g' |awk -F "[\t;]" '{OFS="\t"}{if($3=="CDS") print $9,$10}'

done #genomes
done # species

##-------------------------
###move
##-------------------------

for species in $(echo "Ldel Sterm")
do
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FNA_all/
mkdir -p //archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FNA_all
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FAA_all/
mkdir -p //archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FAA_all
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FFN_all/
mkdir -p //archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FFN_all
#rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/GFF_all/
mkdir -p //archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/GFF_all
#species=Sterm
#for genomess in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/ |grep ".fna$"|grep -v "CRISPR" -i|cut -d '_' -f 1|sed 's/.fna//g')
for genomess in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/ |grep "_202_"|grep ".fasta$" |sed 's/.fasta//g')
do
echo "====================================================================="
echo ${genomess}
#cut -d ' ' -f 1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.faa > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FAA_all/${genomess}.faa

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.faa > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FAA_all/${genomess}.faa

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.fna > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FNA_all/${genomess}.fna

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.ffn > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FFN_all/${genomess}.ffn

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.faa > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FAA//${genomess}.faa

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.gff > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/GFF_all/${genomess}.gff

grep ">" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FNA_all/${genomess}.fna


done #genomes
done # species
##-------------------------
###merge
##-------------------------

#mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/LOGS/
#for species in $(echo "Ldel Sterm")
#do
#cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FAA_all/*.faa > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/LOGS/all_FAA_for_eggnog_${speices}.faa
#done # species



```

#### Eggnog
move local for Eggnog
```{bash}

mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/LOGS/
scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/LOGS/all_FAA_for_eggnog_.faa ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/LOGS/


```


#### AA biosynthesis

For the Amino acid biosynthesis annotation of the strain I will use [Gapmind](http://papers.genomics.lbl.gov/cgi-bin/gapView.cgi).
Therefor I have to upload all FAA files first to the GapMind server

```{bash}

for species in $(echo "Ldel Sterm")
do

rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FAA_all/
mkdir -p //archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FAA_all

#for genomess in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/ |grep ".fna$"|grep -v "CRISPR" -i|cut -d '_' -f 1|sed 's/.fna//g')
for genomess in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/ |grep "_202_"|grep ".fasta$" |sed 's/.fasta//g')
do
#cut -d ' ' -f 1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.faa > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FAA_all/${genomess}.faa

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.faa > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FAA_all/${genomess}.faa


done #genomes
done # species

```

move local

```{bash}
species=Sterm
mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/${species}/
scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FAA_all/ ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/${species}/


species=Ldel
mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/${species}/
scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FAA_all/ ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/${species}/

```

extract the scores for all strains from the rules file of GapMind

in order to make a complete score we have to consider the possible range of the final score for every pathway. 
Gapminder calculates the score accordingly:
considers a secondary score which gives weights of -2, -0.1, and +1 to low-, medium-, and
high-confidence steps. 

My own score goes as following:
High_quality steps=2
medium-quality steps=1.9
low-quality steps=0

In order to get it a relative (comparable between the different aa) I divided by the number of steps

```{bash, eval=FALSE}

echo -e "species\tstrain\tpathway\trule\tnHi\tnMed\tnLo\tscore\tpercentScore" > ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/04_AA_annotation/GapMind/all_species_strains.rules

for species in $(echo "Ldel Sterm")
do

for genomess in $(ls ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/04_AA_annotation/GapMind/${species}/ |grep ".rules$"|cut -d '.' -f 1)
do

sed '1d'  ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/04_AA_annotation/GapMind/${species}/${genomess}.rules | awk -F "\t" -v SPECIES="$species" -v NAME="$genomess" '{OFS="\t"}{if($5=="all") print SPECIES,NAME, $4,$5,$6,$7,$8,$9,((($6*2)+($7*1.9))/($6+$7+$8))/2}' >> ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/04_AA_annotation/GapMind/all_species_strains.rules

done #genomes
done # species

```

```{r}
library(ggplot2)
library(gplots)
library(RColorBrewer)
library(readr)
library(plyr)
library(tidyverse)
all_species_strains <- read_delim("Desktop/Projects/2019_RMK202_analysis/00_FINAL/04_AA_annotation/GapMind/all_species_strains.rules","\t", escape_double = FALSE, trim_ws = TRUE) 
all_species_strains <- all_species_strains %>% select(-c("rule" ,  "nHi" ,  "nMed" ,  "nLo" , "score", "species" ))

unique(all_species_strains$pathway)

levels(all_species_strains$strain)
all_species_strains$strain = factor(all_species_strains$strain, levels=c("202-11141","202-11142","202-11143","202-12104","202-12105","202-12107","202-12109","202-L104","202-L108","202-L35","202-L39","202-L44","202-L70","202-L71" , "202-L80","202-L99","202-LMAG","202-13494","202-S72","202-13498","202-13493","202-13500","202_SMAG","202-13491","202-13492","202-13495","202-13496","202-13497","202-13499c1","202-13499c2","202-S50"))
levels(all_species_strains$strain)

all_species_strains_wide <- spread(all_species_strains, strain, percentScore) %>% column_to_rownames(var="pathway") %>% as.matrix()

##----------------------
#heatmap
##----------------------

mat_data <- all_species_strains_wide # transform column 2-5 into a matrix
# mat_data[upper.tri(mat_data, diag = TRUE)] <- NA
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 89)
# col_breaks = c(seq(99.3,100,length=300))

col_breaks = c(seq(1,0.9,length=30),  # for red
  seq(0.89,0.6,length=30),           # for yellow
  seq(0.59,0,length=30))             # for green

col_breaks = c(seq(0,0.59,length=30),seq(0.6,0.89,length=30),seq(0.9,1,length=30))

dev.off()
    # png("~/Desktop/Projects/2019_RMK202_analysis/10_assembly/heatmap_all_Sterm_strains.png", width = 4000, height = 4000,res=300)
 svg("Desktop/Projects/2019_RMK202_analysis/00_FINAL/04_AA_annotation/GapMind/plots/AminoAcidBiosynthesis_bothSpecies_predicted.svg",width=10,height=7)
heatmap.2(mat_data,
  # cellnote = mat_data,  # same data set for cell labels
  main = "strains Amino acid biosynthesis", # heat map title
  notecol="black",      # change font color of cell labels to black
  density.info="none",
  trace="none",         # turns off trace lines inside the heat map
 # margins = c(12, 12),     # widens margins around plot
  col=my_palette,   
  breaks=col_breaks,    # enable color transition at specified limits
  sepcolor="white",
  sepwidth=c(0.05,0.05),
  colsep=1:ncol(mat_data),
  rowsep=1:nrow(mat_data),
 Rowv = FALSE,
 Colv = FALSE,
  dendrogram="none")
dev.off()

```

##Phylogeny

####core genome snp tree
Here, I maka a core genome snp tree. In order to do this I use the tool parsnp. 
take the species [type strains](https://bacdive.dsmz.de/search?search=Lactobacillus+delbrueckii+subsp.+lactis&submit=) as outgroups. 
```{bash}
##==========================================
###-----------------type strains including
##==========================================
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree

curl 'ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/download.file

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/download.file |grep "lactobacillus" -i|grep "DSM"|grep "20072"|tail -1|awk 'BEGIN{FS="\t";OFS="\t"}  {print $20}'| sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/download_ldel.file

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/download.file |grep "streptococcus" -i|grep "19258"|grep "representative genome"|awk 'BEGIN{FS="\t";OFS="\t"}  {print $20}'| sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/download_sterm.file


##..............--------------------------------
##all sterm complete
##..............--------------------------------
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree_with_References/Sterm
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree_with_References/Sterm
cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200529_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' >    /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree_with_References/download_sterm.file
    

cd /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree_with_References/Sterm
wget -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree_with_References/download_sterm.file
gunzip *.fna.gz


##==========================================
###-----------------gather all strains
##==========================================

for species in $(echo "Ldel Sterm")
do

rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/{genome,analysis}

species=Sterm
for genomess in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/ |grep "_202_"|grep ".fasta$" |sed 's/.fasta//g')
do

#cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned//${genomess}.fna > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/genome/${genomess}.fasta
cat  /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/${genomess}.fasta > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree_with_References/Sterm/${genomess}.fasta



done #genomes
done # species

##==========================================
###-----------------run parsnp
##==========================================
species=Sterm
genomess=/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/GCF_010120595.1_ASM1012059v1_genomic.fna

parsnp -r ${genomess} -v -c  -d /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/genome/ -p 5 -o  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/analysis/

sed -i 's/.fasta//g' /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/analysis/parsnp.tree
sed -i 's/GCF_010120595.1_ASM1012059v1_genomic.fna/S. thermophilus type strain ATCC 19258/g' /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/analysis/parsnp.tree

##------------Ldel

species=Ldel
genomess=/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/GCF_002278095.1_ASM227809v1_genomic.fna

parsnp -r ${genomess} -c -d /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/genome/ -p 37 -o  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/analysis/


sed -i 's/.fasta//g' /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/analysis/parsnp.tree
sed -i 's/GCF_002278095.1_ASM227809v1_genomic.fna/L. delbrueckii type strain ATCC 12315/g' /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/analysis/parsnp.tree


##------------Sterm with Ref

rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree_with_References/Sterm/GCF_010120595.1_ASM1012059v1_genomic.fna
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree_with_References/Sterm//analysis/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree_with_References/Sterm//analysis/

species=Sterm_ref
genomess=/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/GCF_010120595.1_ASM1012059v1_genomic.fna

parsnp -r ${genomess} -v -c  -d /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree_with_References/Sterm/ -p 5 -o  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree_with_References/Sterm//analysis/


```

move local

```{bash}

species=Sterm

mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/
scp  -r vincent@130.223.49.145:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/analysis/ ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/

species=Ldel

mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/
scp  -r vincent@130.223.49.145:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/analysis/ ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/

species=Ldel

mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/
scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/202-LMAG.fna ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/202-LMAG.fna

##===================================
species=Sterm

scp  ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/analysis/parsnp.tree_final vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/analysis/ 

##===================================
##all Sterm with references
##===================================

species=Sterm

scp  vincent@130.223.49.145:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree_with_References/Sterm//analysis/parsnp.tree ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/analysis/parsnp_with_References.tree



/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree_with_References/Sterm//analysis/

##----------------------make vcf

/home/vincent/apps/harvesttools-Linux64-v1.2/harvesttools -i /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/Ldel/analysis/parsnp.ggr -V /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/Ldel/analysis/parsnp.vcf

```

####own nucleotide reference tree

1. get genomes (.gff files)
```{bash}

curl 'ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt' > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200711_NCBI_log.txt
##---------------------change name

species_short=Sterm
cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200711_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $1, "S_thermophilus_"$9}'|sed 's/strain=//g' > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download_sterm_names.txt

cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200711_NCBI_log.txt |grep -i "NCTC" |grep "8618"|grep "representative genome"  |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $1, "S_salivarius_"$9}' |sed 's/strain=//g' >> /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download_sterm_names.txt



sed -i 's/ //g' /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download_sterm_names.txt

for genomesss in $(cut -f 1 /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download_sterm_names.txt)
do
shortnamesss=$(grep "${genomesss}" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download_sterm_names.txt | cut -f 2)
echo -e ${genomesss} " to " ${shortnamesss}

mv /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/FAA/${genomesss}*.faa /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/FAA/${shortnamesss}.faa 

done


##------------------------all complete streptos
species=$(echo "Streptococcus thermophilus")
species_short=Sterm_references
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log
curl 'ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt' > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200711_NCBI_log.txt
rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/FNA

mkdir -p //home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/FNA

###----------------------complete genomes

cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200711_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' >    \
    /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/download_sterm.txt
    species_short=Sterm_references

###----------------------outgroup

cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200711_NCBI_log.txt |grep -i "NCTC" |grep "8618"|grep "representative genome" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' >>    \
    /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/download_sterm.txt
    species_short=Sterm_references

   cd //home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/FNA
    wget -i /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/download_sterm.txt
gunzip *.fna.gz

###----------------------PROKKA annotate these
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/GFF
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/FFN
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/FAA

for genome in $(cut -f 1 /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download_sterm_names.txt)
#for genome in $(ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/FNA |grep "fna")
do

shortnamesss=$(grep "${genome}" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download_sterm_names.txt | cut -f 2)
echo -e ${genome} " to " ${shortnamesss}

locustagss=$(echo ${shortnamesss}|cut -d '_' -f 3)

#rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/PROKKA/${shortnamesss}
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/PROKKA/${shortnamesss}

#/usr/bin/perl /usr/bin/prokka --outdir /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/PROKKA/${shortnamesss} --force --usegenus --genus streptococcus --locustag ${locustagss} --proteins proteins.faa --evalue 0.001 --addgenes /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/FNA/${genome}*.fna

#cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/PROKKA/${shortnamesss}/PROKKA*.gff > \
#/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/GFF/${shortnamesss}.gff

cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/PROKKA/${shortnamesss}/PROKKA*.ffn > \
/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/FFN/${shortnamesss}.ffn

cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/PROKKA/${shortnamesss}/PROKKA*.faa > \
/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/${species_short}/FAA/${shortnamesss}.faa


done


###----------------------gather rmk202 genomes

species=Sterm
for genomess in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/ |grep ".fna$"|grep -v "CRISPR" -i|cut -d '_' -f 1|sed 's/.fna//g')
do
#cut -d ' ' -f 1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.faa > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FAA_all/${genomess}.faa

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.gff > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/GFF//${genomess}.gff


cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.faa > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FAA//${genomess}.faa

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.ffn > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FFN//${genomess}.ffn


done #genomes


###----------------------CREATE A single faa and ffn file


cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FFN/*.ffn > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FFN/complete_Sterm_nucleotide_gene.fasta

cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FAA/*.faa > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FAA/complete_Sterm_aminoAcid_gene.fasta

```


2. find core genomes
```{bash}
##==========================================
###-----------------panaroo
##==========================================

##-----sterm

species=Sterm

mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/10_comparative_genomics/01_Panarooo/onlyStarterCulture/${species}/

panaroo -t 37 -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/GFF_all//*.gff -o /archiv/Projects/2019_RMK202_analysis/00_FINAL/10_comparative_genomics/01_Panarooo/onlyStarterCulture/${species}/ --clean-mode strict


##-----sterm

species=Ldel

mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/10_comparative_genomics/01_Panarooo/onlyStarterCulture/${species}/

panaroo -t 37 -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/GFF_all//*.gff -o /archiv/Projects/2019_RMK202_analysis/00_FINAL/10_comparative_genomics/01_Panarooo/onlyStarterCulture/${species}/ --clean-mode strict 


##==========================================
###-----------------analysis panaroo
##==========================================


ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/GFF/ |grep ".gff" -c

cut -d ',' -f 4 /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/gene_presence_absence_roary.csv|sort|uniq -c
awk -F "," '{if($4=="65" && $5=="65") print $0}' /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/gene_presence_absence_roary.csv|wc -l

##==========================================
###-----------------loop through single copy core genes and gather single copy core faa and ffn
##==========================================


roaryOUT=/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/gene_presence_absence_roary.csv

awk -F "," '{if($4=="65" && $5=="65") print $0}' /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/gene_presence_absence_roary.csv|wc -l

rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FAA/SCCG
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FAA/SCCG

rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FFN/SCCG
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FFN/SCCG

for SCCG in $(awk -F "," '{if($4=="65" && $5=="65") print $1}' ${roaryOUT})
do
echo "================================================="
echo ${SCCG}
colzz=$(head -1 ${roaryOUT}  |tr ',' '\n' |wc -l)
for genomesss in $(seq 15 1 $colzz)
do

genesss=$(grep "^${SCCG}" ${roaryOUT} |cut -d ',' -f ${genomesss})
echo ${genesss}
samtools faidx /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FAA/complete_Sterm_aminoAcid_gene.fasta ${genesss} >> /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FAA/SCCG/${SCCG}.faa

samtools faidx /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FFN/complete_Sterm_nucleotide_gene.fasta ${genesss} >> /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FFN/SCCG/${SCCG}.ffn

done #genomesss
done #SCCG


```


####REference trees

```{bash}
###-----------------------------------------
##get references
###-----------------------------------------
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log
curl 'ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt' > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200529_NCBI_log.txt

##ldel
species=$(echo "Lactobacillus delbrueckii")
species_short=Ldel
typeStrain=$(echo "equicursoris")

rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}


cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200529_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >    \
    /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download.txt


cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200529_NCBI_log.txt |grep -e "${typeStrain}" -i |grep "type strain"|head -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
    /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download.txt
    
    
mkdir -p //home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/FAA
    cd //home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/FAA
    wget -i /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download.txt
    gunzip *.faa.gz

#-------------------------------Sterm

species=$(echo "Streptococcus thermophilus")
species_short=Sterm
typeStrain=$(echo "S735")
typeStrain=$(echo "vestibularis")

rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}

cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200529_NCBI_log.txt |grep -e "${species}"|cut -f 12|sort|uniq -c
#cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200529_NCBI_log.txt |grep -e "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'

cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200529_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >    \
    /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download.txt
    

cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200529_NCBI_log.txt |grep -e "${typeStrain}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|head -1|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
    /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download.txt
    

mkdir -p //home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/FAA
    cd //home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/FAA
    wget -i /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download.txt
gunzip *.faa.gz


##------------------------all complete streptos
species=$(echo "Streptococcus thermophilus")
species_short=Sterm_references
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log
#curl 'ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt' > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200711_NCBI_log.txt

mkdir -p //home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/FAA

cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200711_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >    \
    /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download_sterm.txt
    species_short=Sterm_references


###type strain

cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200711_NCBI_log.txt |grep -i "NCTC" |grep "8618"|grep "representative genome" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >>    \
    /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download_sterm.txt
    species_short=Sterm_references

   cd //home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/FAA
    wget -i /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download_sterm.txt
gunzip *.faa.gz

##---------------------change name


cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200711_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $1, "S_thermophilus_"$9}'|sed 's/strain=//g' > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download_sterm_names.txt

cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200711_NCBI_log.txt |grep -i "NCTC" |grep "8618"|grep "representative genome"  |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $1, "S_salivarius_"$9}' |sed 's/strain=//g' >> /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download_sterm_names.txt

sed -i 's/ //g' /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download_sterm_names.txt

for genomesss in $(cut -f 1 /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download_sterm_names.txt)
do
shortnamesss=$(grep "${genomesss}" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download_sterm_names.txt | cut -f 2)
echo -e ${genomesss} " to " ${shortnamesss}

mv /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/FAA/${genomesss}*.faa /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/FAA/${shortnamesss}.faa 

done

grep "7710"  /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download_sterm_names.txt

###-----------------------------------------
##only Metagenome
###-----------------------------------------
##--------------ldel
species=Ldel

mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/FAA_withMETARMK202

cp /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/FAA/*.faa /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/FAA_withMETARMK202/

cp /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FAA_all/*MAG.faa /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/FAA_withMETARMK202/
##--------------Sterm
species=Sterm

mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/FAA_withMETARMK202

cp /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/FAA/*.faa /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/FAA_withMETARMK202/

cp /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FAA_all//*MAG.faa /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/FAA_withMETARMK202/

###-----------------------------------------
##orthorindfer
###-----------------------------------------
for species in $(echo "Ldel Sterm")
do

rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/orthofinder_onlyMETA


    orthofinder -f /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/FAA_withMETARMK202// \
      -t 37 \
      -o /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/orthofinder_onlyMETA/ \
      -a 37

done # species


###-----------------------------------------
##all RMK202 strains
###-----------------------------------------
##--------------ldel
species=Ldel

mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/FAA_withRMK202

cp /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/FAA/*.faa /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/FAA_withRMK202/

cp /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FAA_all//*.faa /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/FAA_withRMK202/
##--------------Sterm
species=Sterm

mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/FAA_withRMK202

cp /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/FAA/*.faa /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/FAA_withRMK202/

cp /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FAA_all//*.faa /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/FAA_withRMK202/

###-----------------------------------------
##orthorindfer
###-----------------------------------------
for species in $(echo "Ldel Sterm")
do

rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/orthofinder_allRMK202

    orthofinder -f /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/FAA_withRMK202// \
      -t 37 \
      -o /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/orthofinder_allRMK202 \
      -a 37

done # species

###-----------------------------------------
##new
###-----------------------------------------
##--------------ldel

for species in $(echo "Ldel Sterm")
do

for genomess in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/ |grep ".fna$"|grep -v "CRISPR" -i|cut -d '_' -f 1|sed 's/.fna//g')
do


/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/

done #genomes
done # species


###-----------------------------------------
##REferences with Metagenome + isolates
###-----------------------------------------
species=Sterm
species_short=Sterm_references

   cd //home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Sterm_references/FAA
/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FAA_all

#for species in $(echo "Ldel Sterm")
#do

rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Sterm/orthofinder_Ref_Meta_isolates/


    orthofinder -f //home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Sterm_references/FAA/ \
      -t 37 \
      -o /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Sterm/orthofinder_Ref_Meta_isolates/ \
      -a 37

#done # species

```


```{bash}
rm -r ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree
mkdir ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree

#/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/orthofinder/Results_May30/Species_Tree/SpeciesTree_rooted_node_labels.txt

##===================================
species=Sterm

scp  vincent@130.223.51.116:/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/orthofinder_allRMK202//Results_May30/Species_Tree/SpeciesTree_rooted_node_labels.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_allRMK_SpeciesTree_rooted_node_labels.txt
##===================================
species=Ldel

scp  vincent@130.223.51.116:/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/orthofinder_allRMK202//Results_May30/Species_Tree/SpeciesTree_rooted_node_labels.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_allRMK_SpeciesTree_rooted_node_labels.txt


##===================================
species=Sterm

scp  vincent@130.223.51.116:/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/orthofinder_onlyMETA///Results_May30/Species_Tree/SpeciesTree_rooted_node_labels.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_onylMe_SpeciesTree_rooted_node_labels.txt
##===================================
species=Ldel

scp  vincent@130.223.51.116:/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/orthofinder_onlyMETA///Results_May30/Species_Tree/SpeciesTree_rooted_node_labels.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_onylMeta_SpeciesTree_rooted_node_labels.txt

##===================================new

species=Sterm

scp  vincent@130.223.49.145:/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Sterm/orthofinder_Ref_Meta_isolates/Results_Jul15/Species_Tree/SpeciesTree_rooted.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_onylMe_SpeciesTree_rooted_node_labels.txt

```
####bcgtree

```{bash}
##----------------------------------------
##prep file
##----------------------------------------
rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Sterm_BcgTree_all
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Sterm_BcgTree_all
rm /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Sterm_BcgTree_all/log.txt
for genomes in $(ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Sterm_references/FAA/ |grep ".faa")
do
shortName=$(echo ${genomes} |sed 's/.faa//g')
echo -e "--proteome "${shortName}"="${genomes} >> /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Sterm_BcgTree_all/log.txt

done
echo -e "--outdir /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Sterm_BcgTree_all/" >> /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Sterm_BcgTree_all/log.txt

##----------------------------------------
##run bcgTree
##----------------------------------------
cd /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Sterm_references/FAA/
bcgTree.pl @/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Sterm_BcgTree_all/log.txt --threads 37

```

```{bash}



##===================================

mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Sterm_BcgTree_all/

scp   vincent@130.223.49.145:/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Sterm_BcgTree_all/RAxML_* ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Sterm_BcgTree_all/
##===================================
```


##### 1.orthofinder

```{bash}
species=Sterm
species=Ldel

for species in $(echo "Ldel Sterm")
do
rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FAA
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FAA

cp /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FAA_all/* /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FAA/

#cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/* /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FAA/

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/L-DSM-2007.faa /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FAA/
echo "==========================================================="
ll /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FAA/
done


###-----------------------------------------
##orthorfinder
###-----------------------------------------

species=Ldel
for species in $(echo "Ldel Sterm")
do
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/
rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}
#mkdir -p  /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/


    orthofinder -f /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FAA/ \
      -t 35 \
      -o /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species} \
      -a 35

done
```

##### 2.Kirstin's approach


```{bash}
species=Ldel
for species in $(echo "Ldel Sterm")
do
#date=Results_Aug28/

#/data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/Results_Jul28/
#for species in $(echo "Ldel Sterm")
#do
echo "==========================================="
date=$(ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species})
echo -e ${species}
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/get_singlecp_orthologs.pl /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}/${date}/Orthogroups/Orthogroups.txt > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}/${date}/Orthogroups/Orthogroups_SCOG.txt

wc -l /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}/${date}/Orthogroups/Orthogroups_SCOG.txt
wc -l /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}/${date}/Orthogroups/Orthogroups.txt
done
#done
```

/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}/${date}/Species_Tree/
```{bash}
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/Ldel/Results_Aug28/

scp -r vincent@130.223.51.116:/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/Ldel/Results_Aug28//Species_Tree/ /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/Ldel/Results_Aug28/

```



Here, I make a directory containing the fna and faa files of all genomes with the file name corresponding to the locus tag. 

```{bash}





#cp /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FAA_all/* /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Sterm_references/FAA/
#cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/* /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Sterm_references/FAA/

for species in $(echo "Ldel Sterm")
#for species in $(echo "Sterm")
do 
date=$(ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species})


#rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Sterm_references/aligned_combined_ncbi_own/FAA/
#mkdir -p  /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Sterm_references/aligned_combined_ncbi_own/FAA/

rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FFN/
mkdir -p  /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FFN/

leterzzz=$(echo $species |head -c 1)

#cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

#cp /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FAA_all/* /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/combined_ncbi_own/FAA/${species}/

#cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FFN/* /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FFN/
cp  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FFN/L-DSM-2007.ffn /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FFN/
cp /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FFN_all/* /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FFN/


done

###--------------------
for species in $(echo "Ldel Sterm")
#for species in $(echo "Sterm")
do
rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}/${date}/Genomes_FAA_FNA/
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}/${date}/Genomes_FAA_FNA/
for genomesss in $(ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FAA/ |grep ".faa$"|sed 's/.faa//g')
do
#locusTagsss=$(head -1 /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FAA/${genomesss}.faa|cut -d '_' -f 1|sed 's/>//g')
locusTagsss=$(head -1 /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FAA/${genomesss}.faa|cut -d ' ' -f 1| sed 's/_000.*$//g'|sed 's/>//g')
echo -e "This  genome : " ${genomesss} " has the following locus Tag : "${locusTagsss}

cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FAA//${genomesss}.faa > \
/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}/${date}/Genomes_FAA_FNA//${locusTagsss}.faa

cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FFN//${genomesss}.ffn > \
/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}/${date}/Genomes_FAA_FNA/${locusTagsss}.ffn


done
done

##------=================================================================
#extract orthogroups
##------=================================================================

for species in $(echo "Ldel Sterm")
#for species in $(echo "Sterm")
do
cd /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}/${date}/Genomes_FAA_FNA/
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/extract_orthologs.pl  /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}/${date}/Orthogroups/Orthogroups_SCOG.txt --folder ${species}

done
```

align protein families

```{bash}


for species in $(echo "Ldel Sterm")
#for species in $(echo "Sterm")
do
date=$(ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species})

cd /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}/${date}/Genomes_FAA_FNA/${species}
for genesss in $(ls |grep ".faa$" |sed 's/.faa//g' )
do
rm /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}/${date}/Genomes_FAA_FNA/${species}/${genesss}.aln.fasta
mafft --thread 37 --maxiterate 1000  --auto /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}/${date}/Genomes_FAA_FNA/${species}/${genesss}.faa > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}/${date}/Genomes_FAA_FNA/${species}/${genesss}_aln.fasta

done 

done #species
```
here I back translate the faa alignment to dna.

```{bash}

for species in $(echo "Ldel Sterm")
#for species in $(echo "Sterm")
do
cd /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}/${date}/Genomes_FAA_FNA/${species}
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/aln_aa_to_dna.pl

done

```

With perl: prune alignments (removing columns represented by less than 50% of the sequences)


```{bash}


for species in $(echo "Ldel Sterm")
#for species in $(echo "Sterm")
do
date=$(ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species})

cd /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}/${date}/Genomes_FAA_FNA/${species}

rm *_prune.fasta
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/prune_aln.pl

done

sed -i 's/L-I-202-//g' /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}/${date}/Genomes_FAA_FNA/${species}/*_aln_prune.fasta

##================================================================
#remove all infomration (except genome info) from multifasta header
##================================================================

for species in $(echo "Ldel Sterm"|tail -1)
do
date=$(ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species})

cd /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}/${date}/Genomes_FAA_FNA/${species}
for genesss in $(ls |grep "_aln_prune.fasta$" |sed 's/_aln_prune.fasta//g' )
do
sed -i "s/_0.*//" ${genesss}_aln_prune.fasta 

done
done

##================================================================
#shorten names
##================================================================

species=Sterm
cd /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}/${date}/Genomes_FAA_FNA/${species}

for genesss in $(ls |grep "_aln_prune.fasta$" |sed 's/_aln_prune.fasta//g' )
do
sed -i 's/202-13499c/202-13499/g' ${genesss}_aln_prune.fasta 

done


```
With perl: Concatenate pruned alignments
```{bash}
  
 for species in $(echo "Ldel Sterm")
#  for species in $(echo "Sterm")
  do
  echo -e ${species}
  date=$(ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species})
rm /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}/${date}/${species}.phylip
  cd /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}/${date}/Genomes_FAA_FNA/${species}
  perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/cat_align.pl > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}/${date}/${species}.phylip
  
  done
```



With RAxML: infer the phylogeny
```{bash}
rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/raxml_all_combined_NCBI_own/
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/raxml_all_combined_NCBI_own/

for species in $(echo "Ldel Sterm")
#for species in $(echo "Sterm")
  do
  echo -e ${species}
  rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/raxml_all_combined_NCBI_own//${species}
  mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/raxml_all_combined_NCBI_own//${species}
cd /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/raxml_all_combined_NCBI_own//${species}
  date=$(ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species})

#rm -r *${species}_all*

/home/vincent/anaconda3/bin/raxmlHPC-PTHREADS-AVX2 -f a -x 12345 -p 12345 -# 100 -m GTRCAT -s /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}/${date}/${species}.phylip -n ${species}_all -T 37
done

##---------------------
##change some names
##---------------------

sed 's/(S/(S-/g' /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/raxml_all_combined_NCBI_own/RAxML_bipartitions.Sterm_all |sed 's/,S/,S-/g'|sed 's/S--/S-/g' > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/raxml_all_combined_NCBI_own/RAxML_bipartitions.Sterm_all_new


#/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/raxml_all_combined_NCBI_own/Ldel/RAxML_bipartitions.Ldel_all

```

find close samples

```{bash}
species=$(echo "Streptococcus thermophilus")
cat /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'| grep "zlw" -i

```

make pyhlip to aligned multifasta file
```{python}
from Bio import SeqIO

records = SeqIO.parse("/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/Sterm/Results_Aug06/Sterm.phylip", "phylip")
count = SeqIO.write(records, "/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/Sterm/Results_Aug06/Sterm.fasta", "fasta")
print("Converted %i records" % count)



records = SeqIO.parse("/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/Ldel/Results_Aug06/Ldel.phylip", "phylip")
count = SeqIO.write(records, "/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/Ldel/Results_Aug06/Ldel.fasta", "fasta")
print("Converted %i records" % count)
```


snp-sites (could also be interesting for BEAST) and the raxml

```{bash}

for species in $(echo "Ldel Sterm")
#  for species in $(echo "Sterm")
  do
  echo -e ${species}
  date=$(ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species})
  cd /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}/${date}/
  rm  ${species}_ONLYsnpSites.fasta
  snp-sites -cb -o ${species}_ONLYsnpSites.fasta ${species}.fasta
  
  seq_length.py ${species}.fasta |head -1
  seq_length.py ${species}_ONLYsnpSites.fasta |head -1
  done


##------------------------------
##RAXML
##------------------------------
#rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/raxml_all_combined_NCBI_own/
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/raxml_all_combined_NCBI_own/

for species in $(echo "Ldel Sterm")
#for species in $(echo "Sterm")
  do
  echo -e ${species}
cd /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/raxml_all_combined_NCBI_own/
  date=$(ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species})

#rm -r *${species}_all*

/home/vincent/anaconda3/bin/raxmlHPC-PTHREADS-AVX2 -f a -x 12345 -p 12345 -# 100 -m GTRCAT -s /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}/${date}/${species}.phylip -n ${species}_all_SNP_SITES -T 37
done


ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Sterm_references/FAA/|grep "202" |sed 's/.faa//g'|sed 's/c//g' > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Sterm_references/202_strainNames.txt
```


move local

```{bash}

mkdir -p /home/vincent/Desktop/Projects//2019_RMK202_analysis/00_FINAL/09_referenceTree/

scp -r vincent@130.223.51.116:/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/raxml_all_combined_NCBI_own/ /home/vincent/Desktop/Projects//2019_RMK202_analysis/00_FINAL/09_referenceTree/

scp vincent@130.223.51.116:/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Sterm_references/202_strainNames.txt /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/202_strainNames.txt


```
##### 3.iTol

make a tree with Itol

prepare the tree datasets.
1. import the raxml tree
2. prepare the origin dataset
```{bash}
mkdir -p /home/vincent/Desktop/Projects//2019_RMK202_analysis/00_FINAL/09_referenceTree/Itol/Sterm
echo -e "DATASET_COLORSTRIP\nSEPARATOR TAB\nDATASET_LABEL\tsource\nCOLOR\t#ff0000\nDATA" > /home/vincent/Desktop/Projects//2019_RMK202_analysis/00_FINAL/09_referenceTree/Itol/Sterm/origin_dataset.txt


###-----------------------------------------
##NCBi strains
###-----------------------------------------
#less /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_sterm_edited.csv

cut -f 7  /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_sterm_edited.csv|sed '1d' |sort |uniq -c

awk -F "\t" 'BEGIN{OFS="\t"} {if($7=="Cheese") 
print $1,"#DD0181",$7;
else if($7=="host")
print $1,"red",$7;
else if($7=="milk")
print $1,"green",$7;
else if($7=="yogurt")
print $1,"#1974D3",$7;
else print $1, "white","NA"}' /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_sterm_edited.csv |sed 's/3.1/3/g' |sed 's/SMQ-3/SMQ-301/g'>> /home/vincent/Desktop/Projects//2019_RMK202_analysis/00_FINAL/09_referenceTree/Itol/Sterm/origin_dataset.txt 


###-----------------------------------------
##own strains
###-----------------------------------------

grep -v "MAG" /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/202_strainNames.txt |
awk -F "\t" 'BEGIN{OFS="\t"} {print $1,"#DD0181","Cheese"}'  >> /home/vincent/Desktop/Projects//2019_RMK202_analysis/00_FINAL/09_referenceTree/Itol/Sterm/origin_dataset.txt

grep "MAG" /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/202_strainNames.txt |
awk -F "\t" 'BEGIN{OFS="\t"} {print $1,"#53CCEC","Metagenome"}' >> /home/vincent/Desktop/Projects//2019_RMK202_analysis/00_FINAL/09_referenceTree/Itol/Sterm/origin_dataset.txt

```

3. prepare the location dataset
```{bash}
mkdir -p /home/vincent/Desktop/Projects//2019_RMK202_analysis/00_FINAL/09_referenceTree/Itol/Sterm
echo -e "DATASET_COLORSTRIP\nSEPARATOR TAB\nDATASET_LABEL\tlocation\nCOLOR\t#ff0000\nDATA" > /home/vincent/Desktop/Projects//2019_RMK202_analysis/00_FINAL/09_referenceTree/Itol/Sterm/location_dataset.txt


###-----------------------------------------
##NCBi strains
###-----------------------------------------
#less /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_sterm_edited.csv

cut -f 10 /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_sterm_edited.csv|sed '1d' |sort |uniq -c

awk -F "\t" 'BEGIN{OFS="\t"} {if($10=="Asia") 
print $1,"#FEA655",$10;
else if($10=="Australia")
print $1,"red",$10;
else if($10=="Europe")
print $1,"#FFD98E",$10;
else if($10=="USA")
print $1,"pink",$10;
else print $1, "white","NA"}' /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_sterm_edited.csv |sed 's/3.1/3/g' |sed 's/SMQ-3/SMQ-301/g' >> /home/vincent/Desktop/Projects//2019_RMK202_analysis/00_FINAL/09_referenceTree/Itol/Sterm/location_dataset.txt


###-----------------------------------------
##own strains
###-----------------------------------------

cat /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/202_strainNames.txt |
awk -F "\t" 'BEGIN{OFS="\t"} {print $1,"#FFD98E","Europe"}'  >> /home/vincent/Desktop/Projects//2019_RMK202_analysis/00_FINAL/09_referenceTree/Itol/Sterm/location_dataset.txt

```
3. changing names
```{bash}
mkdir -p /home/vincent/Desktop/Projects//2019_RMK202_analysis/00_FINAL/09_referenceTree/Itol/Sterm
echo -e "DATASET_TEXT\nSEPARATOR TAB\nDATASET_LABEL\tNamesChange\nCOLOR\t#ff0000\nMARGIN\t0\nSHOW_INTERNAL\t0\nLABEL_ROTATION\t0\nALIGN_TO_TREE\t0\nSIZE_FACTOR\t1\nDATA" > /home/vincent/Desktop/Projects//2019_RMK202_analysis/00_FINAL/09_referenceTree/Itol/Sterm/names_dataset.txt


###-----------------------------------------
##NCBi strains
###-----------------------------------------
#less /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_sterm_edited.csv

cut -f 5 /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_sterm_edited.csv|sed '1d' |sort |uniq -c

awk -F "\t" 'BEGIN{OFS="\t"} {print $1,$5,"-1","black","italic","0.2","0"}' /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/20200728_NCBI_log_shortNames_final_sterm_edited.csv |sed 's/S-/S./g'|sed 's/^S./S-/g'|sed 's/ilus-/ilus /g'  >> /home/vincent/Desktop/Projects//2019_RMK202_analysis/00_FINAL/09_referenceTree/Itol/Sterm/names_dataset.txt


###-----------------------------------------
##own strains
###-----------------------------------------

cat /home/vincent/Desktop/Projects/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/log/202_strainNames.txt |
awk -F "-" 'BEGIN{OFS="\t"} {print $1"-"$2,"FAM"$2,"-1","#ff0000","italic","0.2","0"}'  >> /home/vincent/Desktop/Projects//2019_RMK202_analysis/00_FINAL/09_referenceTree/Itol/Sterm/names_dataset.txt

```





####making a smaller tree 
here I make a reduced tree of only the closer related strains around the RMK202 strains


```{bash}
species=Sterm

mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FAA_subset/



cp /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FAA_all/* /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FAA_subset/


cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/S-I-1046.faa  /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FAA_subset/
cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/Sterm_references/PROKKA_FAA/S-MN-BM-A0.faa /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FAA_subset/
cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/S-MN-ZLW-0.faa /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FAA_subset/
cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/S-I-1051.faa /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FAA_subset/
cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/S-I-1035.faa /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FAA_subset/
cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/S-I-1058.faa /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FAA_subset/
cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/NCBI_Refs/${species}_references/PROKKA_FAA/S-I-19.faa /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FAA_subset/


###-----------------------------------------
##orthorfinder
###-----------------------------------------

species=Sterm
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/
#mkdir -p  /data/Project/2020_StarterCultureDiversity/11_referenceTree/Orthofinder_combined_NCBI_own/${species}/


    orthofinder -f /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FAA_subset/ \
      -t 35 \
      -o /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}_subset \
      -a 35

```

```{bash}
species=Sterm
echo "==========================================="
date=$(ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}_subset)
echo -e ${species}
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/get_singlecp_orthologs.pl /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}_subset/${date}/Orthogroups/Orthogroups.txt > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}_subset/${date}/Orthogroups/Orthogroups_SCOG.txt

wc -l /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}_subset/${date}/Orthogroups/Orthogroups_SCOG.txt
wc -l /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}_subset/${date}/Orthogroups/Orthogroups.txt
```

Here, I make a directory containing the fna and faa files of all genomes with the file name corresponding to the locus tag. 

```{bash}


date=$(ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}_subset)


rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}_subset/${date}/Genomes_FAA_FNA/
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}_subset/${date}/Genomes_FAA_FNA/
for genomesss in $(ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FAA_subset/ |grep ".faa$"|sed 's/.faa//g')
do
locusTagsss=$(head -1 /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FAA/${genomesss}.faa|cut -d '_' -f 1|sed 's/>//g')
echo -e "This  genome : " ${genomesss} " has the following locus Tag : "${locusTagsss}

cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FAA//${genomesss}.faa > \
/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}_subset/${date}/Genomes_FAA_FNA//${locusTagsss}.faa

cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}_references/FFN//${genomesss}.ffn > \
/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}_subset/${date}/Genomes_FAA_FNA/${locusTagsss}.ffn

done


##------=================================================================
#extract orthogroups
##------=================================================================

date=$(ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}_subset)

cd /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}_subset/${date}/Genomes_FAA_FNA/
rm -r ${species}
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/extract_orthologs.pl  /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}_subset/${date}/Orthogroups/Orthogroups_SCOG.txt --folder ${species}



```

align protein families

```{bash}
species=Sterm

date=$(ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}_subset)

cd /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}_subset/${date}/Genomes_FAA_FNA/${species}
for genesss in $(ls |grep ".faa$" |sed 's/.faa//g' )
do
rm /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}_subset/${date}/Genomes_FAA_FNA/${species}/${genesss}.aln.fasta
mafft --thread 37 --maxiterate 1000  --auto /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}_subset/${date}/Genomes_FAA_FNA/${species}/${genesss}.faa > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}_subset/${date}/Genomes_FAA_FNA/${species}/${genesss}_aln.fasta

done 

```
here I back translate the faa alignment to dna.

```{bash}

cd /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}_subset/${date}/Genomes_FAA_FNA/${species}
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/aln_aa_to_dna.pl


```

With perl: prune alignments (removing columns represented by less than 50% of the sequences)


```{bash}


cd /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}_subset/${date}/Genomes_FAA_FNA/${species}

rm *_prune.fasta
perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/prune_aln.pl


##================================================================
#remove all infomration (except genome info) from multifasta header
##================================================================

cd /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}_subset/${date}/Genomes_FAA_FNA/${species}
for genesss in $(ls |grep "_aln_prune.fasta$" |sed 's/_aln_prune.fasta//g' )
do
sed -i "s/_0.*//" ${genesss}_aln_prune.fasta 

done


##================================================================
#shorten names
##================================================================

species=Sterm
cd /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}_subset/${date}/Genomes_FAA_FNA/${species}

for genesss in $(ls |grep "_aln_prune.fasta$" |sed 's/_aln_prune.fasta//g' )
do
sed -i 's/202-13499c/202-13499/g' ${genesss}_aln_prune.fasta 

done


```
With perl: Concatenate pruned alignments
```{bash}
  
rm /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}_subset/${date}/${species}.phylip
  cd /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}_subset/${date}/Genomes_FAA_FNA/${species}
  perl /data/Project/2020_StarterCultureDiversity/99_log/aln_aa_to_dna/vincent/cat_align.pl > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}_subset/${date}/${species}.phylip


```



With RAxML: infer the phylogeny
```{bash}
#rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/raxml_all_combined_NCBI_own/
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/raxml_all_combined_NCBI_own/

  echo -e ${species}
cd /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/raxml_all_combined_NCBI_own/
#  date=$(ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species})

#rm -r *${species}_all*

/home/vincent/anaconda3/bin/raxmlHPC-PTHREADS-AVX2 -f a -x 12345 -p 12345 -# 100 -m GTRCAT -s /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Orthofinder/${species}_subset/${date}/${species}.phylip -n ${species}_reduced -T 37

```


##Fastani with RMK202
```{bash}

###-----------------------------------------
##make list
###-----------------------------------------

for species in $(echo "Ldel Sterm")
do
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/
rm /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/fastani_rmk202.txt

for genomssss in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FNA_all/ |grep ".fna$")
do
echo "/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/"${species}"/FNA_all/"${genomssss} >> /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/fastani_rmk202.txt


done #genomes
done # species


###-----------------------------------------
##fastani
###-----------------------------------------



for species in $(echo "Ldel Sterm")
do

fastANI --fragLen 1000  -t 37 --ql /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/fastani_rmk202.txt \
  --rl /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/fastani_rmk202.txt \
  -o /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/fastaANI_comparision.txt

done # species


```
move local

```{bash}
species=Sterm
mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/
scp  vincent@130.223.51.116:/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/fastaANI_comparision.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/fastaANI_comparision.txt


species=Ldel
mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/
scp  vincent@130.223.51.116:/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/fastaANI_comparision.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/fastaANI_comparision.txt
```



```{r}
##==============================
##Sterm
##==============================

library(readr)
library(plyr)
library(tidyverse)
final_ANI <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Sterm/fastaANI_comparision.txt",  "\t", escape_double = FALSE, col_names = c("GenomeA","GenomeB","ANI","mappedFragemnts","totFragemnts"),  trim_ws = TRUE)
final_ANI$coverage_ANI <- 100*(final_ANI$mappedFragemnts/final_ANI$totFragemnts)



final_ANI$GenomeA <- str_remove(final_ANI$GenomeA,"/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all/") %>% str_remove(.,".fna") %>% str_remove(.,".fasta")%>% str_remove(.,"L_I_202_")%>% str_remove(.,"S_O_202_")%>% str_remove(.,"S_I_202_")

final_ANI$GenomeB <- str_remove(final_ANI$GenomeB,"/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all/") %>% str_remove(.,".fna") %>% str_remove(.,".fasta")%>% str_remove(.,"L_I_202_")%>% str_remove(.,"S_O_202_")%>% str_remove(.,"S_I_202_")


# final_ANI$GenomeB <- revalue(final_ANI$GenomeB, c("S50"="mst1","S72"="mst2"))
# final_ANI$GenomeA <- revalue(final_ANI$GenomeA, c("S50"="mst1","S72"="mst2"))

final_ANI$ANI <- round(final_ANI$ANI,digits = 2)

write.table(final_ANI,"~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Sterm/fastaANI_comparision_curated.txt",sep = "\t",quote = FALSE,row.names = FALSE,col.names = TRUE)

##==============================
##Ldel
##==============================

library(readr)
library(plyr)
library(tidyverse)
final_ANI <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Ldel/fastaANI_comparision.txt",  "\t", escape_double = FALSE, col_names = c("GenomeA","GenomeB","ANI","mappedFragemnts","totFragemnts"),  trim_ws = TRUE)
final_ANI$coverage_ANI <- 100*(final_ANI$mappedFragemnts/final_ANI$totFragemnts)



final_ANI$GenomeA <- str_remove(final_ANI$GenomeA,"/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Ldel/FNA_all/") %>% str_remove(.,".fna") %>% str_remove(.,".fasta")%>% str_remove(.,"L_I_202_")%>% str_remove(.,"S_O_202_")%>% str_remove(.,"S_I_202_")

final_ANI$GenomeB <- str_remove(final_ANI$GenomeB,"/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Ldel/FNA_all/") %>% str_remove(.,".fna") %>% str_remove(.,".fasta")%>% str_remove(.,"L_I_202_")%>% str_remove(.,"S_O_202_")%>% str_remove(.,"S_I_202_")


# final_ANI$GenomeB <- revalue(final_ANI$GenomeB, c("S50"="mst1","S72"="mst2"))
# final_ANI$GenomeA <- revalue(final_ANI$GenomeA, c("S50"="mst1","S72"="mst2"))

final_ANI$ANI <- round(final_ANI$ANI,digits = 2)
final_ANI
write.table(final_ANI,"~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Ldel/fastaANI_comparision_curated.txt",sep = "\t",quote = FALSE,row.names = FALSE,col.names = TRUE)
##------------
```


## Pandaroo

first I combine only the completely sequenced genomes (ONT)
```{bash}
##----------------------------------
#primer

base=/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all/
format=fna
combineOutput=/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_complete


##----------------------
##do

mkdir -p ${combineOutput}

cp ${base}/202-13494.${format}  ${combineOutput}
cp ${base}/202-13496.${format} ${combineOutput}
cp ${base}/202-13498.${format} ${combineOutput}
cp ${base}/202-13499.${format} ${combineOutput}
cp ${base}/202-24737.${format} ${combineOutput}
cp ${base}/202-24738.${format} ${combineOutput}
cp ${base}/202-24739.${format} ${combineOutput}
cp ${base}/202-24740.${format} ${combineOutput}
cp ${base}/202-24853.${format} ${combineOutput}
cp ${base}/202-24854.${format} ${combineOutput}
cp ${base}/202-24855.${format} ${combineOutput}
cp ${base}/202-SMAG.${format} ${combineOutput}


##----------------------
##only Bac contig
##----------------------

rm -r ${combineOutput}/onlyBac/
mkdir -p ${combineOutput}/onlyBac/
for genomesss in $(ls ${combineOutput}|grep ".fna$")
do
echo ${genomesss}
contig_name=$(seq_length.py ${combineOutput}/${genomesss} |head -1 |cut -f 1)
rm ${combineOutput}/${genomesss}.fai
samtools faidx ${combineOutput}/${genomesss} ${contig_name} > ${combineOutput}/onlyBac/${genomesss}

seq_length.py ${combineOutput}/onlyBac/${genomesss}
done


##----------------------
##Prokka
##----------------------
species=Streptococcus

for genomesss in $(ls ${combineOutput}|grep ".fna$"|sed 's/.fna//g')
do
rm -r ${combineOutput}/onlyBac/PROKKA/${genomesss}
mkdir -p  ${combineOutput}/onlyBac/PROKKA/${genomesss}

echo ${genomesss}


/usr/bin/perl /usr/bin/prokka --cpu 37 --outdir ${combineOutput}/onlyBac/PROKKA/${genomesss}/ --force --usegenus --genus ${species} --locustag ${genomesss} --proteins proteins.faa --evalue 0.001 --addgenes ${combineOutput}/onlyBac/${genomesss}.fna

done

###---------------------------------
##gather
###---------------------------------

rm -r ${combineOutput}/onlyBac/PROKKA/all_GFF/
mkdir -p  ${combineOutput}/onlyBac/PROKKA/all_GFF/

for genomesss in $(ls ${combineOutput}|grep ".fna$"|sed 's/.fna//g')
do
echo ${genomesss}

cat ${combineOutput}/onlyBac/PROKKA/${genomesss}/*PROKKA*.gff > ${combineOutput}/onlyBac/PROKKA/all_GFF/${genomesss}.gff

done


##==========================================
###-----------------panaroo
##==========================================

##-----sterm

mkdir -p ${combineOutput}/onlyBac/01_Panarooo/

panaroo -t 37 -i  ${combineOutput}/onlyBac/PROKKA/all_GFF//*.gff -o ${combineOutput}/onlyBac/01_Panarooo/ --clean-mode strict

##-------------------------------------
##post Panaroo analysis
##-------------------------------------

species=Sterm
cd ${combineOutput}/onlyBac/01_Panarooo/
#sudo python3 /home/vincent/apps/Roary/contrib/roary_plots/roary_plots.py --format eps /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/analysis/parsnp.tree_final gene_presence_absence.csv

sed '1d' gene_presence_absence_roary.csv | awk -F "\",\"" '{OFS=","}{print $5}'  |sort|uniq -c
#sudo chmod 777 ../analysis_1588600355
#sudo chmod 777 pan_genome_reference.fa

rm accesoryGenes.fasta

sed  's/ /-/g'  pan_genome_reference.fa >  pan_genome_reference_cleaned.fa
for genesss in $(sed '1d' gene_presence_absence_roary.csv  |awk -F "," '{OFS=","}{if($4<12)print $1}' |sed 's/"//g')
do
echo -e ${genesss}
longNamezzz=$(grep "${genesss}$" pan_genome_reference_cleaned.fa |sed 's/>//g')

samtools faidx pan_genome_reference_cleaned.fa ${longNamezzz} >>  accesoryGenes.fasta
done


grep ">" -c accesoryGenes.fasta

transeq  -sequence accesoryGenes.fasta > \
  -outseq  accesoryGenes.faa
  
  
###----------------------------

awk -F "," '{if($4<12)print $1}' gene_presence_absence_roary.csv > accessoryNames.txt


grep ">" accesoryGenes.faa |sed 's/>//g' > accesoryGenes.names

```

move local


```{bash}

mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_complete/onlyBac/01_Panarooo/
scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_complete/onlyBac/01_Panarooo/accesoryGenes.faa ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_complete/onlyBac/01_Panarooo/accesoryGenes.faa

scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_complete/onlyBac/01_Panarooo/accessoryNames.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_complete/onlyBac/01_Panarooo/accessoryNames.txt

cd ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_complete/onlyBac/01_Panarooo/
grep ">" accesoryGenes.faa |sed 's/>//g' > accesoryGenes.names


scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_complete/onlyBac/01_Panarooo/gene_presence_absence.Rtab ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_complete/onlyBac/01_Panarooo/gene_presence_absence.Rtab



```



2. find core genomes
```{bash}

grep ">" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FNA_all/*.fna

##==========================================
###-----------------panaroo
##==========================================

##-----sterm

species=Sterm

mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/01_Panarooo/${species}

panaroo -t 37 -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/GFF_all/*.gff -o  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/01_Panarooo/${species} --clean-mode strict


##==========================================
###-----------------analysis panaroo
##==========================================


ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/GFF/ |grep ".gff" -c

cut -d ',' -f 4 /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/gene_presence_absence_roary.csv|sort|uniq -c
awk -F "," '{if($4=="65" && $5=="65") print $0}' /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/gene_presence_absence_roary.csv|wc -l

##==========================================
###-----------------loop through single copy core genes and gather single copy core faa and ffn
##==========================================


roaryOUT=/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/gene_presence_absence_roary.csv

awk -F "," '{if($4=="65" && $5=="65") print $0}' /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/01_Panarooo/gene_presence_absence_roary.csv|wc -l

rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FAA/SCCG
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FAA/SCCG

rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FFN/SCCG
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FFN/SCCG

for SCCG in $(awk -F "," '{if($4=="65" && $5=="65") print $1}' ${roaryOUT})
do
echo "================================================="
echo ${SCCG}
colzz=$(head -1 ${roaryOUT}  |tr ',' '\n' |wc -l)
for genomesss in $(seq 15 1 $colzz)
do

genesss=$(grep "^${SCCG}" ${roaryOUT} |cut -d ',' -f ${genomesss})
echo ${genesss}
samtools faidx /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FAA/complete_Sterm_aminoAcid_gene.fasta ${genesss} >> /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FAA/SCCG/${SCCG}.faa

samtools faidx /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FFN/complete_Sterm_nucleotide_gene.fasta ${genesss} >> /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/10_ownreferenceTree/Sterm_references/FFN/SCCG/${SCCG}.ffn

done #genomesss
done #SCCG


```

eggnog of pandarooo results
```{r}
library(readr)
library(plyr)
library(ggplot2)
library(tidyverse)
# genes_soacer <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/eggnog/accessoryGenes_eggnog.annotations", "\t", escape_double = FALSE, trim_ws = TRUE,  skip = 3)

# accesoryGenesNames <- read_csv("/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/eggnog/query_seqs.fa.emapper.annotations", col_names = "names")

# accesoryGenesNames <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_complete/onlyBac/01_Panarooo/accessoryNames.txt", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 3)

accesoryGenesNames <- read_csv("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_complete/onlyBac/01_Panarooo/accesoryGenes.names",  col_names = "names")

# intersect_genes_prep_prepped <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/intersect_genes_prep_prepped.txt", "\t", escape_double = FALSE, col_names = c("query_name","spacerName"),   trim_ws = TRUE)


# S50 <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3)) 

genes_soacer <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/eggnog/query_seqs.fa.emapper_accessory.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3))


colnames(genes_soacer) <- c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description")

COG_functional_categories <- read_delim("~/Desktop/ScriptRepository/COG_functional_categories.csv",  ",", escape_double = FALSE, col_names = c("short","long","divided"),trim_ws = TRUE, na = "NA")

# genes_soacer

# accessoryNames$names %in% genes_soacer$query_name

# merge(accessoryNames,genes_soacer,by.x = "names",by.y = "query_name",all.x = TRUE)
##----------------COG

ALL_sterm <- genes_soacer
nrow(ALL_sterm)

ALL_sterm <- genes_soacer %>% 
    mutate(COG_Functional_Category = strsplit(as.character(COG_Functional_Category), "")) %>% 
    unnest(COG_Functional_Category)


ALL_sterm <- merge(ALL_sterm,COG_functional_categories,by.y="short",by.x="COG_Functional_Category",all.x=TRUE)

# table(ALL_sterm$long)
# ALL_sterm %>% filter(long=="multiple COGs")

# nrow(ALL_sterm)
# # ALL_sterm$co
# #unique(sterm_all_unique$COG_Functional_Category)[match(COG_functional_categories$short , unique(sterm_all_unique$COG_Functional_Category))]
# unique(ALL_sterm$COG_Functional_Category)[which(!unique(ALL_sterm$COG_Functional_Category) %in% COG_functional_categories$short )]
# # ALL_sterm$
# # table(ALL_sterm$COG_Functional_Category)
# 
# i=4
# ALL_sterm$longCOG <- NA
# for (i in 1:nrow(ALL_sterm)) {
#   ALL_sterm[i,"longCOG"] <- COG_functional_categories[which(COG_functional_categories$short==as.character(ALL_sterm[i,"COG_Functional_Category"])),"long"] %>% as.character()
#   
# }

colnames(ALL_sterm)
table(ALL_sterm$long)
ALL_sterm_aa <- ALL_sterm %>% filter(long=="Amino acid metabolism and transport")

ALL_sterm_transcriptioon <- ALL_sterm %>% filter(long=="Transcription")
ALL_sterm_interesting <- ALL_sterm %>% filter(!long %in% c("Replication and repair","Function Unknown")) %>% filter(!is.na(COG_Functional_Category))
# ALL_sterm$COG_Functional_Category
grep("transpor",ignore.case = FALSE,ALL_sterm_interesting$`eggNOG free text description`) %>% length()

##---------------which genes do not cluster in eggnog

genes <- accesoryGenesNames[which(!accesoryGenesNames$names %in% genes_soacer$query_name),"names"]
unclustering <- data.frame("query_name"=genes,"longCOG"="Function Unknown")
colnames(unclustering) <- c("query_name","long")
# table(ALL_sterm$longCOG)
ALL_sterm_extended <- ALL_sterm %>% select(c("query_name","long"))
ALL_sterm_extended <- rbind(ALL_sterm_extended,unclustering)
# table(ALL_sterm_extended$longCOG)
ALL_sterm_extended$long  <- revalue(ALL_sterm_extended$long, c("character(0)"="Function Unknown"))
100*(sort(table(ALL_sterm_extended$long ))/nrow(ALL_sterm_extended))

table(ALL_sterm_extended$long ) %>% sort()
##---------------plot

CogPlotSterm <- ggplot(ALL_sterm_extended,aes(x=forcats::fct_infreq(long)))+geom_bar()+
  theme_classic()+
  # facet_wrap(array ~ .)+
  labs(x="",y="Accessory genes")+
    # scale_fill_manual(values=three_reds)+
   # scale_color_manual(values=three_reds)+
  scale_y_continuous(breaks =c(0,25,80,192),labels=c(0,25,80,192))+
  theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9),
          # # axis.text.y.left = element_text(color = "blue"),
          # axis.text.y.right = element_text(color="red"),
          # axis.title.y.left = element_text(color="blue"),
          # axis.title.y.right = element_text(color="red"),
          # rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          #legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank(),
          plot.margin=unit(c(0,0,0,2), "cm"),
          # axis.text.x.bottom = element_text(size=10)
          #legend.position = c(0.02, 0.98),
          #legend.justification = c("left", "top")
          )
CogPlotSterm
library(patchwork)


# png("~/Desktop/Projects/2019_RMK202_analysis/plot/MappingDB_mappingtoBac_eggnog.png", width = 1000, height = 2000,res=300)
svg("~/Desktop/Projects/2019_RMK202_analysis/plot/accessory_genes_eggnog.svg",width=6,height=5)

CogPlotSterm

dev.off()

###-----------------------
##amino acid accessory genes
###-----------------------
table(ALL_sterm$longCOG)

aa_genes <- ALL_sterm %>% filter(longCOG=="Amino acid metabolism and transport")


###-----------------------
##amino acid accessory genes
###-----------------------
colnames(ALL_sterm)

ALL_sterm_sub <- ALL_sterm %>% select("query_name","Preferred_name","eggNOG free text description","COG_Functional_Category","long","CAZy","EC","KEGG_ko","GOs")
dim(ALL_sterm_sub)
write.table(ALL_sterm_sub, "~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/accessoryGenes_rmk202_from_Sterm.txt",na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names = TRUE)


ALL_sterm_sub$Gene <- gsub("_1$","",ALL_sterm_sub$query_name)
##---import presence absence data

gene_presence_absence <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_complete/onlyBac/01_Panarooo/gene_presence_absence.Rtab", "\t", escape_double = FALSE, trim_ws = TRUE)

ALL_sterm_sub$Gene %in% gene_presence_absence$Gene    
ALL_sterm_sub_complete <- merge(ALL_sterm_sub,gene_presence_absence,by.y="Gene",by.x="Gene")
dim(ALL_sterm_sub_complete)
dim(ALL_sterm_sub)

ALL_sterm_sub_complete$numberOFisolates <- rowSums(ALL_sterm_sub_complete[,11:22])

ggplot(ALL_sterm_sub_complete,aes(x=numberOFisolates))+geom_histogram()+theme_classic()

c("HUtH","hutU","Glycosyl transferases group 1")
# ALL_sterm_sub_complete[grep("Glycosyl transferases group 1",ALL_sterm_sub_complete$Gene,ignore.case = TRUE),]

genesOfinterest <- c("HUtH","hutU","Glycosyl transferases group 1","Bacterial sugar transferase","Glycosyltransferase like family 2","Oligosaccharide biosynthesis protein Alg14 like","LPS","Bacteriocin class II","Interbacterial warfare")
genesOfinterest <- c("HUtH","hutU","Glycosyl transferases group 1","Bacterial sugar transferase","Glycosyltransferase like family 2","Oligosaccharide biosynthesis protein Alg14 like")

genesDF <- data.frame()
for (genesss in genesOfinterest) {
  print(genesss)
  numbersss <- grep(genesss,ALL_sterm_sub_complete,ignore.case = TRUE)[1]
tmp <- ALL_sterm_sub_complete[grep(genesss,ALL_sterm_sub_complete[,numbersss],ignore.case = TRUE),]
  
genesDF <- rbind(genesDF,tmp)
  
}


```
##roary

Here, we make a roary annotation in order to see the size of the pangenome. 


first, we need to get the created parsnp tree  and then delete the outgroup manually (with VIM).

```{bash}

##===================================
species=Sterm

scp  ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/analysis/parsnp.tree_final vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/analysis/ 
##===================================
species=Ldel

scp  ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/analysis/parsnp.tree_final vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/analysis/ 


```

```{bash}




##==========================================
###-----------------gather all strains
##==========================================

for species in $(echo "Ldel Sterm")
do

rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/01_roaryStrains/${species}
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/01_roaryStrains/${species}/{analysis,genomes}

for genomess in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/ |grep ".fna$"|cut -d '_' -f 1|sed 's/.fna//g' )
do

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.gff > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/01_roaryStrains/${species}/genomes/${genomess}.gff


done #genomes
done # species


##==========================================
###-----------------roary
##==========================================

##-----sterm

species=Ldel

sudo roary -f /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/01_roaryStrains/${species}/analysis -p 5 -e -n -v /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/01_roaryStrains/${species}/genomes/*.gff

species=Sterm
sudo roary -f /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/01_roaryStrains/${species}/analysis -p 5 -e -n -v /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/01_roaryStrains/${species}/genomes/*.gff

##==========================================
###-----------------roary
##==========================================

##-----sterm

species=Ldel

mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/01_Panarooo_Strains/${species}/analysis

panaroo -t 37 -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/01_roaryStrains/${species}/genomes/*.gff -o /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/01_Panarooo_Strains/${species}/analysis/ --clean-mode strict

species=Sterm

mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/01_Panarooo_Strains/${species}/analysis

panaroo -t 37 -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/01_roaryStrains/${species}/genomes/*.gff -o /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/01_Panarooo_Strains/${species}/analysis/ --clean-mode strict



##==========================================
###-----------------pangenome
##==========================================
species=Sterm
cd /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/01_Panarooo_Strains/${species}/analysis/
#sudo python3 /home/vincent/apps/Roary/contrib/roary_plots/roary_plots.py --format eps /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/analysis/parsnp.tree_final gene_presence_absence.csv

sed '1d' gene_presence_absence_roary.csv | awk -F "\",\"" '{OFS=","}{print $5}'  |sort|uniq -c
#sudo chmod 777 ../analysis_1588600355
#sudo chmod 777 pan_genome_reference.fa

rm accesoryGenes.fasta

sed  's/ /-/g'  pan_genome_reference.fa >  pan_genome_reference_cleaned.fa
for genesss in $(sed '1d' gene_presence_absence_roary.csv  |awk -F "," '{OFS=","}{if($4<14)print $1}' |sed 's/"//g')
do
echo -e ${genesss}
longNamezzz=$(grep "${genesss}$" pan_genome_reference_cleaned.fa |sed 's/>//g')

samtools faidx pan_genome_reference_cleaned.fa ${longNamezzz} >>  accesoryGenes.fasta
done


grep ">" -c accesoryGenes.fasta


transeq  -sequence accesoryGenes.fasta > \
  -outseq  accesoryGenes.faa

###----------------------ldel
species=Ldel
cd /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/01_Panarooo_Strains/${species}/analysis/
#sudo python3 /home/vincent/apps/Roary/contrib/roary_plots/roary_plots.py --format png /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/analysis_1589128141/parsnp.tree_final gene_presence_absence.csv

rm accesoryGenes.fasta

sed  's/ /-/g'  pan_genome_reference.fa >  pan_genome_reference_cleaned.fa
for genesss in $(sed '1d' gene_presence_absence_roary.csv  |awk -F "," '{OFS=","}{if($4<17)print $1}' |sed 's/"//g')
do
echo -e ${genesss}
longNamezzz=$(grep "${genesss}$" pan_genome_reference_cleaned.fa |sed 's/>//g')

samtools faidx pan_genome_reference_cleaned.fa ${longNamezzz} >>  accesoryGenes.fasta
done
#awk -F "\t" -v namezzz="$genesss" '{OFS="\t"}{if($1==namezzz)print $3}' gene_presence_absence.csv >>  accesoryGenes.annotations
grep "^\"${genesss}\"" gene_presence_absence.csv |cut -d ',' -f 3 >>  accesoryGenes.annotations


#longNamezzz=$(grep "${genesss}$" pan_genome_reference_cleaned.fa |sed 's/>//g')

#samtools faidx pan_genome_reference_cleaned.fa ${longNamezzz} >>  accesoryGenes.fasta
#done

wc -l accesoryGenes.annotations
grep -c -i "hypothe" accesoryGenes.annotations
grep -c -i "IS" accesoryGenes.annotations
grep -c -i "RNA" accesoryGenes.annotations


grep ">" -c accesoryGenes.fasta


transeq  -sequence accesoryGenes.fasta > \
  -outseq  accesoryGenes.faa
```

move images back local

```{bash}

/archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/01_roaryStrains/${species}/analysis_1588600355/

species=Sterm

scp   vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/01_roaryStrains/${species}/analysis_1588600355/*.pdf ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/analysis/


species=Sterm

scp   vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/01_Panarooo_Strains/${species}/analysis/accesoryGenes.faa ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/analysis/accesoryGenes.faa

species=Sterm

scp   vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/01_Panarooo_Strains/${species}/analysis/gene_presence_absence.csv ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/analysis/

###-----------------eggnog prep
species=Sterm

grep ">" ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/analysis/accesoryGenes.faa |sed 's/>//g' > ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/${species}/analysis/accesoryGenes.names
```

######lineage specific genes
```{r}
library(readr)
gene_presence_absence <- read_csv("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/Sterm/analysis/gene_presence_absence.csv")

###----------------------lineage 1
gene_presence_absence$`202-13494`
LINEAGE_1 <- gene_presence_absence %>% filter(`No. isolates`==2) %>% filter(!is.na(`202-S72`) & !is.na(`202-13494`)) %>% nrow()
LINEAGE_2 <- gene_presence_absence %>% filter(`No. isolates`==6) %>% filter(!is.na(`202-13491`) & !is.na(`202-13500`) &!is.na(`202-13492`) &!is.na(`202-13493`) &!is.na(`202-13498`) &!is.na(`202-SMAG`)) %>% nrow()

LINEAGE_3 <- gene_presence_absence %>% filter(`No. isolates`==3) %>% filter(!is.na(`202-13495`) & !is.na(`202-13496`)& !is.na(`202-13497`)) %>% nrow()
LINEAGE_4 <- gene_presence_absence %>% filter(`No. isolates`==3) %>% filter(!is.na(`202-S50`) & !is.na(`202-13499c1`)& !is.na(`202-13499c2`)) %>% nrow()


# two overlaps
LINEAGE_3_4 <- gene_presence_absence %>% filter(`No. isolates`==6) %>% filter(!is.na(`202-13495`) & !is.na(`202-13496`)& !is.na(`202-13497`)&!is.na(`202-S50`) & !is.na(`202-13499c1`)& !is.na(`202-13499c2`)) %>% nrow()

LINEAGE_1_3 <- gene_presence_absence %>% filter(`No. isolates`==5) %>% filter(!is.na(`202-S72`) & !is.na(`202-13494`)&!is.na(`202-13495`) & !is.na(`202-13496`)& !is.na(`202-13497`)) %>% nrow()

LINEAGE_1_4 <- gene_presence_absence %>% filter(`No. isolates`==5) %>% filter(!is.na(`202-S72`) & !is.na(`202-13494`)&!is.na(`202-S50`) & !is.na(`202-13499c1`)& !is.na(`202-13499c2`)) %>% nrow()

LINEAGE_1_2 <- gene_presence_absence %>% filter(`No. isolates`==8) %>% filter(!is.na(`202-S72`) & !is.na(`202-13494`)&!is.na(`202-13491`) & !is.na(`202-13500`) &!is.na(`202-13492`) &!is.na(`202-13493`) &!is.na(`202-13498`) &!is.na(`202-SMAG`)) %>% nrow()

LINEAGE_2_3 <- gene_presence_absence %>% filter(`No. isolates`==9) %>% filter(!is.na(`202-13491`) & !is.na(`202-13500`) &!is.na(`202-13492`) &!is.na(`202-13493`) &!is.na(`202-13498`) &!is.na(`202-SMAG`)&!is.na(`202-13495`) & !is.na(`202-13496`)& !is.na(`202-13497`)) %>% nrow()

LINEAGE_2_4 <- gene_presence_absence %>% filter(`No. isolates`==9) %>% filter(!is.na(`202-13491`) & !is.na(`202-13500`) &!is.na(`202-13492`) &!is.na(`202-13493`) &!is.na(`202-13498`) &!is.na(`202-SMAG`)&!is.na(`202-S50`) & !is.na(`202-13499c1`)& !is.na(`202-13499c2`)) %>% nrow()

# thre overlaps

LINEAGE_1_2_3 <- gene_presence_absence %>% filter(`No. isolates`==11) %>% filter(!is.na(`202-S72`) & !is.na(`202-13494`)&!is.na(`202-13491`) & !is.na(`202-13500`) &!is.na(`202-13492`) &!is.na(`202-13493`) &!is.na(`202-13498`) &!is.na(`202-SMAG`)&!is.na(`202-13495`) & !is.na(`202-13496`)& !is.na(`202-13497`)) %>% nrow()

LINEAGE_2_3_4 <- gene_presence_absence %>% filter(`No. isolates`==12) %>% filter(!is.na(`202-13491`) & !is.na(`202-13500`) &!is.na(`202-13492`) &!is.na(`202-13493`) &!is.na(`202-13498`) &!is.na(`202-SMAG`)&!is.na(`202-13495`) & !is.na(`202-13496`)& !is.na(`202-13497`)&!is.na(`202-S50`) & !is.na(`202-13499c1`)& !is.na(`202-13499c2`)) %>% nrow()

LINEAGE_1_3_4 <- gene_presence_absence %>% filter(`No. isolates`==3) %>% filter(!is.na(`202-S72`) & !is.na(`202-13494`)&!is.na(`202-13495`) & !is.na(`202-13496`)& !is.na(`202-13497`)&!is.na(`202-S50`) & !is.na(`202-13499c1`)& !is.na(`202-13499c2`)) %>% nrow()

LINEAGE_1_2_4 <- gene_presence_absence %>% filter(`No. isolates`==12) %>% filter(!is.na(`202-S72`) & !is.na(`202-13494`)&!is.na(`202-13491`) & !is.na(`202-13500`) &!is.na(`202-13492`) &!is.na(`202-13493`) &!is.na(`202-13498`) &!is.na(`202-SMAG`)&!is.na(`202-S50`) & !is.na(`202-13499c1`)& !is.na(`202-13499c2`)) %>% nrow()

LINEAGE_1_2_3_4 <- gene_presence_absence %>% filter(`No. isolates`==14) %>% filter(!is.na(`202-S72`) & !is.na(`202-13494`)&!is.na(`202-13491`) & !is.na(`202-13500`) &!is.na(`202-13492`) &!is.na(`202-13493`) &!is.na(`202-13498`) &!is.na(`202-SMAG`)&!is.na(`202-S50`) & !is.na(`202-13499c1`)& !is.na(`202-13499c2`)&!is.na(`202-13495`) & !is.na(`202-13496`)& !is.na(`202-13497`)) %>% nrow()


LINEAGE_1
LINEAGE_2
LINEAGE_3
LINEAGE_4
LINEAGE_3_4
LINEAGE_1_3
LINEAGE_1_4
LINEAGE_1_2
LINEAGE_2_3
LINEAGE_2_4

```



eggnog of lineage specific genes
```{r}
library(readr)
library(plyr)
library(tidyverse)
# genes_soacer <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/eggnog/accessoryGenes_eggnog.annotations", "\t", escape_double = FALSE, trim_ws = TRUE,  skip = 3)

accesoryGenesNames <- read_csv("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/Sterm/analysis/accesoryGenes.names", col_names = "names")

# intersect_genes_prep_prepped <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/intersect_genes_prep_prepped.txt", "\t", escape_double = FALSE, col_names = c("query_name","spacerName"),   trim_ws = TRUE)


# S50 <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3)) 

genes_soacer <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/eggnog/accessoryGenes_eggnog.annotations_02", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3))
colnames(genes_soacer) <- c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description")

COG_functional_categories <- read_delim("~/Desktop/ScriptRepository/COG_functional_categories.csv",  ",", escape_double = FALSE, col_names = c("short","long","divided"),trim_ws = TRUE, na = "NA")


##----------------COG

ALL_sterm <- genes_soacer
nrow(ALL_sterm)

ALL_sterm <- ALL_sterm %>% 
    mutate(COG_Functional_Category = strsplit(as.character(COG_Functional_Category), "")) %>% 
    unnest(COG_Functional_Category)

nrow(ALL_sterm)
# ALL_sterm$co
#unique(sterm_all_unique$COG_Functional_Category)[match(COG_functional_categories$short , unique(sterm_all_unique$COG_Functional_Category))]
unique(ALL_sterm$COG_Functional_Category)[which(!unique(ALL_sterm$COG_Functional_Category) %in% COG_functional_categories$short )]
# ALL_sterm$
# table(ALL_sterm$COG_Functional_Category)

i=4
ALL_sterm$longCOG <- NA
for (i in 1:nrow(ALL_sterm)) {
  ALL_sterm[i,"longCOG"] <- COG_functional_categories[which(COG_functional_categories$short==as.character(ALL_sterm[i,"COG_Functional_Category"])),"long"] %>% as.character()
  
}

ALL_sterm_aa <- ALL_sterm %>% filter(longCOG=="Amino acid metabolism and transport")

ALL_sterm_transcriptioon <- ALL_sterm %>% filter(longCOG=="Transcription")
ALL_sterm_interesting <- ALL_sterm %>% filter(!longCOG %in% c("Replication and repair","Function Unknown")) %>% filter(!is.na(COG_Functional_Category))
# ALL_sterm$COG_Functional_Category
grep("transpor",ignore.case = FALSE,ALL_sterm_interesting$`eggNOG free text description`) %>% length()

##---------------which genes do not cluster in eggnog

genes <- accesoryGenesNames[which(!accesoryGenesNames$names %in% genes_soacer$query_name),"names"]
unclustering <- data.frame("query_name"=genes,"longCOG"="Function Unknown")
colnames(unclustering) <- c("query_name","longCOG")
# table(ALL_sterm$longCOG)
ALL_sterm_extended <- ALL_sterm %>% select(c("query_name","longCOG"))
ALL_sterm_extended <- rbind(ALL_sterm_extended,unclustering)
# table(ALL_sterm_extended$longCOG)
ALL_sterm_extended$longCOG  <- revalue(ALL_sterm_extended$longCOG, c("character(0)"="Function Unknown"))
100*(sort(table(ALL_sterm_extended$longCOG ))/nrow(ALL_sterm_extended))

table(ALL_sterm_extended$longCOG )
##---------------plot

CogPlotSterm <- ggplot(ALL_sterm_extended,aes(x=forcats::fct_infreq(longCOG)))+geom_bar()+
  theme_classic()+
  # facet_wrap(array ~ .)+
  labs(x="",y="Accessory genes")+
    # scale_fill_manual(values=three_reds)+
   # scale_color_manual(values=three_reds)+
  theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9),
          # # axis.text.y.left = element_text(color = "blue"),
          # axis.text.y.right = element_text(color="red"),
          # axis.title.y.left = element_text(color="blue"),
          # axis.title.y.right = element_text(color="red"),
          # rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          #legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank(),
          plot.margin=unit(c(0,0,0,2), "cm"),
          # axis.text.x.bottom = element_text(size=10)
          #legend.position = c(0.02, 0.98),
          #legend.justification = c("left", "top")
          )
CogPlotSterm
library(patchwork)


# png("~/Desktop/Projects/2019_RMK202_analysis/plot/MappingDB_mappingtoBac_eggnog.png", width = 1000, height = 2000,res=300)
svg("~/Desktop/Projects/2019_RMK202_analysis/plot/accessory_genes_eggnog.svg",width=6,height=5)

CogPlotSterm

dev.off()

###-----------------------
##amino acid accessory genes
###-----------------------
table(ALL_sterm$longCOG)

aa_genes <- ALL_sterm %>% filter(longCOG=="Amino acid metabolism and transport")
```

##orthofinder

Here I run Orthofinder. 
I do this for multiple reasons:
1. core genome phylogeny
2. core genes

```{bash}



for species in $(echo "Ldel Sterm")
do

rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}


    orthofinder -f /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FAA_all/ \
      -t 37 \
      -o /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species} \
      -a 37

done # species

###-----------------------------------------------
##extract all orthogroups fro eggnog
###-----------------------------------------------
for species in $(echo "Ldel Sterm")
do
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/final_OG_${species}.fa
for genesss in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08//Orthogroup_Sequences/ |grep ".fa")
do

genesss_short=$(echo ${genesss} |sed 's/.fa//g')

namezzz=$(grep ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08//Orthogroup_Sequences/${genesss} |head -1|sed 's/>//g')

samtools faidx /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08//Orthogroup_Sequences/${genesss} ${namezzz} > /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/tmp.fa

sed "s/${namezzz}/${genesss_short}/g" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/tmp.fa >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/final_OG_${species}.fa

done #geneesss
done # species

```

move local

```{bash}
species=Sterm

mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08/Species_Tree/
scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08/Species_Tree/SpeciesTree_rooted_node_labels.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08/Species_Tree/


species=Ldel
mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Apr02/Species_Tree/
scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Apr02/Species_Tree/SpeciesTree_rooted_node_labels.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Apr02/Species_Tree/

##------------------------------------

species=Ldel
scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/final_OG_Ldel.fa ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/

scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/final_OG_Sterm.fa ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/

```
##### eggnog orthofinder

bring Bigmachine

```{bash}
#/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder
scp -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/ vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder

```

PREPARE PGAP .FAA FILES FOR EGGNOG AND AFTER THAT FUNCTIONAL ANNOTATION
```{bash}
preample=CP046134_TRANSCRIPT

cp /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/faa/S_M_202_SMAG.faa /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/faa/S_M_202_SMAG_forEggnog.faa

for genesss in $(grep ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/faa/S_M_202_SMAG.faa | cut -d ' ' -f 1)
do

shortNamess=$(echo ${genesss}|cut -d '|' -f 3)
sed -i "s/${genesss}.*$/>${preample}_${shortNamess}/g" /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/faa/S_M_202_SMAG_forEggnog.faa


#sed "s/${genesss}.*$/>${preample}_${shortNamess}/g" /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/faa/S_M_202_SMAG_forEggnog.faa |grep "${shortNamess}"


done



preample=CP046131_TRANSCRIPT

cp /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/faa/L_M_202_LMAG.faa /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/faa/L_M_202_LMAG_forEggnog.faa

for genesss in $(grep ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/faa/L_M_202_LMAG.faa| cut -d ' ' -f 1)
do

shortNamess=$(echo ${genesss}|cut -d '|' -f 3)
sed -i "s/${genesss}.*$/>${preample}_${shortNamess}/g" /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/faa/L_M_202_LMAG_forEggnog.faa


#sed "s/${genesss}.*$/>${preample}_${shortNamess}/g" /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/faa/S_M_202_SMAG_forEggnog.faa |grep "${shortNamess}"


done

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/faa/L_M_202_LMAG_forEggnog.faa /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/faa/S_M_202_SMAG_forEggnog.faa > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/faa/bothMAG_forEggnog.faa



```

```{bash}
/archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/faa/bothMAG_forEggnog.faa

scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/faa/bothMAG_forEggnog.faa ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/bothMAG_forEggnog_fromPGAP.faa

```


make a OG list for all MAG genes

```{bash}
cd /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/
species=Sterm

for species in $(echo "Ldel Sterm")
do
####================
##OGs
####================
rm OGs_${species}.txt
for OGsss in $(ls ../${species}/Results_Jun08/Orthogroup_Sequences/ |grep ".fa$" |sed 's/.fa//g' )
do

grep ">" ../${species}/Results_Jun08/Orthogroup_Sequences/${OGsss}.fa | grep "MAG_"|sed 's/>//g' |awk -F "\t" -v OGzzz="$OGsss" '{OFS="\t"}{print OGzzz,$1}' >> OGs_${species}.txt
echo -e ${OGsss}

done #species
done #OGsss
```

iron uptake
mainly ABC transporters:

```{bash}
fatA
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PGAP


cd /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/

species=Sterm
for species in $(echo "Ldel Sterm")
do
echo "-----------------------"
echo ${species}
#awk -F "\t" '{OFS="\t"}{if($6=="fatA")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1
grep "fata" -i eggnog_orthifinder_${species}_annotations.tsv
grep "fatb" -i eggnog_orthifinder_${species}_annotations.tsv
grep "fatc" -i eggnog_orthifinder_${species}_annotations.tsv
grep "fatd" -i eggnog_orthifinder_${species}_annotations.tsv
#grep "fur" -i eggnog_orthifinder_${species}_annotations.tsv
grep "dpr" -i eggnog_orthifinder_${species}_annotations.tsv
grep "pox1" -i eggnog_orthifinder_${species}_annotations.tsv
grep "pyrD" -i eggnog_orthifinder_${species}_annotations.tsv



done
```


```{bash}

cd /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/
species=Sterm
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt
for species in $(echo "Ldel Sterm")
do
####================
##OGs
####================

##-----------prt
prtSsss=$(grep "prtS" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/eggnog_orthifinder_${species}_annotations.tsv | cut -f 1)
##-----------pep
#prtSsss=$(grep "pep" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/eggnog_orthifinder_${species}_annotations.tsv | cut -f 1)
pepA=$(awk -F "\t" '{OFS="\t"}{if($6=="pepA")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepC=$(awk -F "\t" '{OFS="\t"}{if($6=="pepC")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepD=$(awk -F "\t" '{OFS="\t"}{if($6=="pepD")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepF=$(awk -F "\t" '{OFS="\t"}{if($6=="pepF")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepI=$(awk -F "\t" '{OFS="\t"}{if($6=="pepI")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepL=$(awk -F "\t" '{OFS="\t"}{if($6=="pepL")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepN=$(awk -F "\t" '{OFS="\t"}{if($6=="pepN")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepP=$(awk -F "\t" '{OFS="\t"}{if($6=="pepP")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepQ=$(awk -F "\t" '{OFS="\t"}{if($6=="pepQ")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepS=$(awk -F "\t" '{OFS="\t"}{if($6=="pepS")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepT=$(awk -F "\t" '{OFS="\t"}{if($6=="pepT")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepT2=$(awk -F "\t" '{OFS="\t"}{if($6=="pepT2")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepV=$(awk -F "\t" '{OFS="\t"}{if($6=="pepV")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepX=$(awk -F "\t" '{OFS="\t"}{if($6=="pepX")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)


##-----------Urea
UreCCC=$(grep "3\.5\.1\.5" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/eggnog_orthifinder_${species}_annotations.tsv | grep "ureC" | cut -f 1)
UreBBB=$(grep "3\.5\.1\.5" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/eggnog_orthifinder_${species}_annotations.tsv | grep "ureB" | cut -f 1)
UreAAA=$(grep "3\.5\.1\.5" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/eggnog_orthifinder_${species}_annotations.tsv | grep "ureA" | cut -f 1)
##-----------formate
formmsss=$(grep "2\.3\.1\.54" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/eggnog_orthifinder_${species}_annotations.tsv | grep "pflB" | cut -f 1)
##-----------folicAcid
FolBBB=$(grep "4\.1\.2\.25" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/eggnog_orthifinder_${species}_annotations.tsv | grep "folB" | cut -f 1)
Folkkk=$(grep "2\.7\.6\.3" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/eggnog_orthifinder_${species}_annotations.tsv | grep "folK" | cut -f 1)
Folppp=$(grep "2\.5\.1\.15" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/eggnog_orthifinder_${species}_annotations.tsv | grep "folP" | cut -f 1)
FolCCC=$(grep "6\.3\.2\.12" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/eggnog_orthifinder_${species}_annotations.tsv | grep "folC"|head -1 | cut -f 1)
FolAAA=$(grep "1\.5\.1\.3" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/eggnog_orthifinder_${species}_annotations.tsv | grep "folA"|head -1 | cut -f 1)

####================
##number of Genes
####================
num_UreCCC=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08/Orthogroup_Sequences/${UreCCC}.fa)
num_UreBBB=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08/Orthogroup_Sequences/${UreBBB}.fa)
num_UreAAA=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08/Orthogroup_Sequences/${UreAAA}.fa)
num_formmsss=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08/Orthogroup_Sequences/${formmsss}.fa)
num_FolBBB=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08/Orthogroup_Sequences/${FolBBB}.fa)
num_Folkkk=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08/Orthogroup_Sequences/${Folkkk}.fa)
num_Folppp=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08/Orthogroup_Sequences/${Folppp}.fa)
num_FolCCC=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08/Orthogroup_Sequences/${FolCCC}.fa)
num_FolAAA=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08/Orthogroup_Sequences/${FolAAA}.fa)
num_prtSsss=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08/Orthogroup_Sequences/${prtSsss}.fa)


num_pepA=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08/Orthogroup_Sequences/${pepA}.fa)
num_pepC=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08/Orthogroup_Sequences/${pepC}.fa)
num_pepD=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08/Orthogroup_Sequences/${pepD}.fa)
num_pepF=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08/Orthogroup_Sequences/${pepF}.fa)
num_pepI=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08/Orthogroup_Sequences/${pepI}.fa)
num_pepL=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08/Orthogroup_Sequences/${pepL}.fa)
num_pepN=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08/Orthogroup_Sequences/${pepN}.fa)
num_pepP=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08/Orthogroup_Sequences/${pepP}.fa)
num_pepQ=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08/Orthogroup_Sequences/${pepQ}.fa)
num_pepS=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08/Orthogroup_Sequences/${pepS}.fa)
num_pepT=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08/Orthogroup_Sequences/${pepT}.fa)
num_pepT2=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08/Orthogroup_Sequences/${pepT2}.fa)
num_pepV=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08/Orthogroup_Sequences/${pepV}.fa)
num_pepX=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08/Orthogroup_Sequences/${pepX}.fa)


####================
#output
####================
echo -e ${species}"\tUreC\t"${UreCCC}"\t"${num_UreCCC} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt
echo -e ${species}"\tUreB\t"${UreBBB}"\t"${num_UreBBB} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt
echo -e ${species}"\tUreA\t"${UreAAA}"\t"${num_UreAAA} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt
echo -e ${species}"\tformate\t"${formmsss}"\t"${num_formmsss} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt
echo -e ${species}"\tfolB\t"${FolBBB}"\t"${num_FolBBB} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt
echo -e ${species}"\tfolK\t"${Folkkk}"\t"${num_Folkkk} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt
echo -e ${species}"\tfolP\t"${Folppp}"\t"${num_Folppp} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt
echo -e ${species}"\tfolC\t"${FolCCC}"\t"${num_FolCCC} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt
echo -e ${species}"\tfolA\t"${FolAAA}"\t"${num_FolAAA} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt
echo -e ${species}"\tprotease\t"${prtSsss}"\t"${num_prtSsss} >> all_interaction_functioning_Genes.txt


echo -e ${species}"\tpepA\t"${pepA}"\t"${num_pepA} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepC\t"${pepC}"\t"${num_pepC} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepD\t"${pepD}"\t"${num_pepD} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepF\t"${pepF}"\t"${num_pepF} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepI\t"${pepI}"\t"${num_pepI} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepL\t"${pepL}"\t"${num_pepL} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepN\t"${pepN}"\t"${num_pepN} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepP\t"${pepP}"\t"${num_pepP} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepQ\t"${pepQ}"\t"${num_pepQ} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepS\t"${pepS}"\t"${num_pepS} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepT\t"${pepT}"\t"${num_pepT} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepT2\t"${pepT2}"\t"${num_pepT2} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepV\t"${pepV}"\t"${num_pepV} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepX\t"${pepX}"\t"${num_pepX} >> all_interaction_functioning_Genes.txt



awk -F "\t" -v nameesss="${UreCCC}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv > MAG_genes_of_Interes.txt
awk -F "\t" -v nameesss="${UreBBB}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
awk -F "\t" -v nameesss="${UreAAA}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
awk -F "\t" -v nameesss="${formmsss}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
awk -F "\t" -v nameesss="${FolBBB}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
awk -F "\t" -v nameesss="${Folkkk}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
awk -F "\t" -v nameesss="${Folppp}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
awk -F "\t" -v nameesss="${FolCCC}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
awk -F "\t" -v nameesss="${FolAAA}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
awk -F "\t" -v nameesss="${prtSsss}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
awk -F "\t" -v nameesss="${pepA}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
awk -F "\t" -v nameesss="${pepC}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
awk -F "\t" -v nameesss="${pepD}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
awk -F "\t" -v nameesss="${pepF}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
awk -F "\t" -v nameesss="${pepI}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
awk -F "\t" -v nameesss="${pepL}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
awk -F "\t" -v nameesss="${pepN}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
awk -F "\t" -v nameesss="${pepP}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
awk -F "\t" -v nameesss="${pepQ}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
awk -F "\t" -v nameesss="${pepS}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
awk -F "\t" -v nameesss="${pepT}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
awk -F "\t" -v nameesss="${pepT2}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
awk -F "\t" -v nameesss="${pepV}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
awk -F "\t" -v nameesss="${pepX}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt



done # species

grep "_" MAG_genes_of_Interes.txt > MAG_genes_of_Interes_cleaned.txt
####================
#msa of OGs
####================
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/all_clustalwAlignement_final.txt
for species in $(echo "Ldel Sterm")
do
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/InteractionGenesAlignment/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/InteractionGenesAlignment/
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}_clustalwAlignement_final.txt
for OGsss in $(grep "${species}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt |cut -f 3)
do

namesss=$(grep "${species}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt |grep "${OGsss}"|cut -f 2)
numbersss=$(grep "${species}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt |grep "${OGsss}"|cut -f 4)



   echo ==========================
     echo -e "extracting geness: " ${OGsss}
     echo ==========================

clustalw -TYPE=PROTEIN -INFILE=/archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Jun08/Orthogroup_Sequences/${OGsss}.fa -ALIGN -OUTFILE=/archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/InteractionGenesAlignment/${OGsss}_clustalwAlignement | grep "Aligned. Score:" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/InteractionGenesAlignment/${OGsss}_clustalwAlignement.txt


sed 's/) Aligned. Score:  /\t/g' /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/InteractionGenesAlignment/${OGsss}_clustalwAlignement.txt | sed 's/Sequences (//g' |awk -F "\t" -v namesss="$species" -v arraysss="$namesss" -v genezzz="$numbersss" -v OGsssssss="$OGsss" '{OFS="\t"}{sum+=$2} END {print namesss,arraysss,OGsssssss,genezzz,sum/NR}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}_clustalwAlignement_final.txt

done #OGsss
cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}_clustalwAlignement_final.txt >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/all_clustalwAlignement_final.txt
done # species
awk -F "\t" '{OFS="\t"}{if($3=="")print $0,"NA","NA","NA"}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/all_clustalwAlignement_final.txt
```

autolysin
```{bash}

cd /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/
species=Sterm
for species in $(echo "Ldel Sterm")
do
echo "========================="
echo ${species}
echo "========================="

#grep "lysi" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/eggnog_orthifinder_${species}_annotations.tsv |grep "auto"
grep "Acm" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/eggnog_orthifinder_${species}_annotations.tsv 
grep "Lytr" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/eggnog_orthifinder_${species}_annotations.tsv 
#grep "Acm" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/eggnog_orthifinder_${species}_annotations.tsv 



done

```


peps
```{bash}
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/peps.txt
for species in $(echo "Ldel Sterm")
do
awk -F "\t" '{OFS="\t"}{if($6~"pep")print $0}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/eggnog_orthifinder_${species}_annotations.tsv |cut -f 6 |sort|uniq>> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/peps.txt
done
sort /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/peps.txt|uniq -c
```
move local for r

```{bash}

mkdir -p /home/vincent/bin/apps/circos-0.69-6/data/CRISPR
scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/all_clustalwAlignement_final.txt   ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/all_clustalwAlignement_final.txt


###----------------for DN/dS ratio

scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt  ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/

scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/OGs_Sterm.txt  ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/

scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/OGs_Ldel.txt  ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/



```

!!!!!!!!!!!!!!!!!!!!!!!!!!!
NEWNEWNEW PGAP ANNOTATION
!!!!!!!!!!!!!!!!!!!!!!!!!!!


```{bash}

scp ~/Downloads/query_seqs.fa.emapper.annotations vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PGAP/


scp ~ /home/vincent/Downloads/*annotations.tsv vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA_new/
```

iron uptake
mainly ABC transporters:

```{bash}

mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA_new/
cd /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA_new/

#grep "^CP046131" query_seqs.fa.emapper.annotations > eggnog_orthifinder_Ldel_annotations.tsv
#grep "^CP046134" query_seqs.fa.emapper.annotations > eggnog_orthifinder_Sterm_annotations.tsv


#cd /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PGAP/

species=Sterm
for species in $(echo "Ldel Sterm")
do
echo "-----------------------"
echo ${species}
#awk -F "\t" '{OFS="\t"}{if($6=="fatA")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1
grep "fata" -i eggnog_orthifinder_${species}_annotations.tsv
grep "fatb" -i eggnog_orthifinder_${species}_annotations.tsv
grep "fatc" -i eggnog_orthifinder_${species}_annotations.tsv
grep "fatd" -i eggnog_orthifinder_${species}_annotations.tsv
#grep "fur" -i eggnog_orthifinder_${species}_annotations.tsv
grep "dpr" -i eggnog_orthifinder_${species}_annotations.tsv
grep "pox1" -i eggnog_orthifinder_${species}_annotations.tsv
grep "pyrD" -i eggnog_orthifinder_${species}_annotations.tsv



done
```


```{bash}

#cd /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PGAP/
species=Sterm
rm all_interaction_functioning_Genes.txt
for species in $(echo "Ldel Sterm")
do
####================
##OGs
####================

##-----------prt
prtSsss=$(grep "prtS" -i eggnog_orthifinder_${species}_annotations.tsv | cut -f 1)
##-----------pep
#prtSsss=$(grep "pep" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/eggnog_orthifinder_${species}_annotations.tsv | cut -f 1)
pepA=$(awk -F "\t" '{OFS="\t"}{if($6=="pepA")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepC=$(awk -F "\t" '{OFS="\t"}{if($6=="pepC")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepD=$(awk -F "\t" '{OFS="\t"}{if($6=="pepD")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepF=$(awk -F "\t" '{OFS="\t"}{if($6=="pepF")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepI=$(awk -F "\t" '{OFS="\t"}{if($6=="pepI")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepL=$(awk -F "\t" '{OFS="\t"}{if($6=="pepL")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepN=$(awk -F "\t" '{OFS="\t"}{if($6=="pepN")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepP=$(awk -F "\t" '{OFS="\t"}{if($6=="pepP")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepQ=$(awk -F "\t" '{OFS="\t"}{if($6=="pepQ")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepS=$(awk -F "\t" '{OFS="\t"}{if($6=="pepS")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepT=$(awk -F "\t" '{OFS="\t"}{if($6=="pepT")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepT2=$(awk -F "\t" '{OFS="\t"}{if($6=="pepT2")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepV=$(awk -F "\t" '{OFS="\t"}{if($6=="pepV")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepX=$(awk -F "\t" '{OFS="\t"}{if($6=="pepX")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)


##-----------Urea
UreCCC=$(grep "3\.5\.1\.5" eggnog_orthifinder_${species}_annotations.tsv | grep "ureC" | cut -f 1)
UreBBB=$(grep "3\.5\.1\.5" eggnog_orthifinder_${species}_annotations.tsv | grep "ureB" | cut -f 1)
UreAAA=$(grep "3\.5\.1\.5" eggnog_orthifinder_${species}_annotations.tsv | grep "ureA" | cut -f 1)
##-----------formate
formmsss=$(grep "2\.3\.1\.54" eggnog_orthifinder_${species}_annotations.tsv | grep "pflB" | cut -f 1)
##-----------folicAcid
FolBBB=$(grep "4\.1\.2\.25" eggnog_orthifinder_${species}_annotations.tsv | grep "folB" | cut -f 1)
Folkkk=$(grep "2\.7\.6\.3" eggnog_orthifinder_${species}_annotations.tsv | grep "folK" | cut -f 1)
Folppp=$(grep "2\.5\.1\.15" eggnog_orthifinder_${species}_annotations.tsv | grep "folP" | cut -f 1)
FolCCC=$(grep "6\.3\.2\.12" eggnog_orthifinder_${species}_annotations.tsv | grep "folC"|head -1 | cut -f 1)
FolAAA=$(grep "1\.5\.1\.3" eggnog_orthifinder_${species}_annotations.tsv | grep "folA"|head -1 | cut -f 1)


####================
#output
####================
echo -e ${species}"\tUreC\t"${UreCCC} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tUreB\t"${UreBBB} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tUreA\t"${UreAAA} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tformate\t"${formmsss} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tfolB\t"${FolBBB} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tfolK\t"${Folkkk} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tfolP\t"${Folppp} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tfolC\t"${FolCCC} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tfolA\t"${FolAAA} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tprotease\t"${prtSsss} >> all_interaction_functioning_Genes.txt


echo -e ${species}"\tpepA\t"${pepA} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepC\t"${pepC} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepD\t"${pepD} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepF\t"${pepF} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepI\t"${pepI} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepL\t"${pepL} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepN\t"${pepN} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepP\t"${pepP} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepQ\t"${pepQ} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepS\t"${pepS} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepT\t"${pepT} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepT2\t"${pepT2} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepV\t"${pepV} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepX\t"${pepX} >> all_interaction_functioning_Genes.txt



#awk -F "\t" -v nameesss="${UreCCC}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv > MAG_genes_of_Interes.txt
#awk -F "\t" -v nameesss="${UreBBB}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
#awk -F "\t" -v nameesss="${UreAAA}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
#awk -F "\t" -v nameesss="${formmsss}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
#awk -F "\t" -v nameesss="${FolBBB}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
#awk -F "\t" -v nameesss="${Folkkk}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
#awk -F "\t" -v nameesss="${Folppp}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
#awk -F "\t" -v nameesss="${FolCCC}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
#awk -F "\t" -v nameesss="${FolAAA}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
#awk -F "\t" -v nameesss="${prtSsss}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
#awk -F "\t" -v nameesss="${pepA}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
#awk -F "\t" -v nameesss="${pepC}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
#awk -F "\t" -v nameesss="${pepD}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
#awk -F "\t" -v nameesss="${pepF}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
#awk -F "\t" -v nameesss="${pepI}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
#awk -F "\t" -v nameesss="${pepL}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
#awk -F "\t" -v nameesss="${pepN}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
#awk -F "\t" -v nameesss="${pepP}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
#awk -F "\t" -v nameesss="${pepQ}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
#awk -F "\t" -v nameesss="${pepS}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
#awk -F "\t" -v nameesss="${pepT}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
#awk -F "\t" -v nameesss="${pepT2}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
#awk -F "\t" -v nameesss="${pepV}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt
#awk -F "\t" -v nameesss="${pepX}" '{OFS="\t"}{if($1==nameesss)print $15}' ../${species}/Results_Jun08/Orthogroups/Orthogroups.tsv >> MAG_genes_of_Interes.txt



done # species


awk '{OFS="\t"}{if($3!="") print $1"_"$2,$3}' all_interaction_functioning_Genes.txt > all_interaction_functioning_Genes_cleaned.txt

```


autolysin
```{bash}

cd /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/
species=Sterm
for species in $(echo "Ldel Sterm")
do
echo "========================="
echo ${species}
echo "========================="

#grep "lysi" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/eggnog_orthifinder_${species}_annotations.tsv |grep "auto"
grep "Acm" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/eggnog_orthifinder_${species}_annotations.tsv 
grep "Lytr" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/eggnog_orthifinder_${species}_annotations.tsv 
#grep "Acm" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/eggnog_orthifinder_${species}_annotations.tsv 



done

```


peps
```{bash}
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/peps.txt
for species in $(echo "Ldel Sterm")
do
awk -F "\t" '{OFS="\t"}{if($6~"pep")print $0}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/eggnog_orthifinder_${species}_annotations.tsv |cut -f 6 |sort|uniq>> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/peps.txt
done
sort /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/peps.txt|uniq -c
```
move local for r

```{bash}

mkdir -p /home/vincent/bin/apps/circos-0.69-6/data/CRISPR
scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/all_clustalwAlignement_final.txt   ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/all_clustalwAlignement_final.txt


###----------------for DN/dS ratio

scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt  ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/

scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/OGs_Sterm.txt  ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/

scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/OGs_Ldel.txt  ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/





scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PGAP/all_interaction_functioning_Genes_cleaned.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/


```

```{r}
library(readr)
interaction_interest_genes <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/all_clustalwAlignement_final.txt",  "\t", escape_double = FALSE, col_names = c("species","gene","OGss","numberGenomes","proteinID"),  trim_ws = TRUE)

interaction_interest_genes_wide <- interaction_interest_genes %>% select(species,gene,proteinID) %>% spread(.,species,proteinID) %>% column_to_rownames(var="gene")



library(viridis)

# pairwiseSharedSpacers_tmp_02 <-  data.frame(Ref1=c(as.character(pairwiseSharedSpacers$Ref2),as.character(pairwiseSharedSpacers$Ref1)),Ref2=c(as.character(pairwiseSharedSpacers$Ref1),as.character(pairwiseSharedSpacers$Ref2)),PercentShared=c(as.numeric(pairwiseSharedSpacers$PercentShared),as.numeric(pairwiseSharedSpacers$PercentShared)))
# 
# pairwiseSharedSpacers_tmp_02$Ref1 = factor(pairwiseSharedSpacers_tmp_02$Ref1, levels=orderSterm)
# pairwiseSharedSpacers_tmp_02$Ref2 = factor(pairwiseSharedSpacers_tmp_02$Ref2, levels=orderSterm)
# levels(final_ANI$GenomeA)
# levels(final_ANI$GenomeB)
# pairwiseSharedSpacers_final <- rbind(pairwiseSharedSpacers_tmp_02[,c("Ref2","Ref1","PercentShared")],pairwiseSharedSpacers[,c("Ref1","Ref2","PercentShared")])
# ggheatmap_CRISPR <- ggplot(pairwiseSharedSpacers_tmp_02, aes(Ref2, Ref1, fill = PercentShared))+
#  geom_tile(color = "white",size=1.1)+
#   # theme_classic()+
#  #scale_fill_gradient2(low = "blue", high = "red", mid = "white", 
#   # midpoint = 99.5, limit = c(98.9,100), space = "Lab", 
#   # name="ANI") +
#   # scale_fill_distiller(name = "SNPs", palette = "Blues", direction = -1)+
#   scale_fill_viridis(alpha=0.8)+
#   labs(legend="CRISPR ID",x="",y="")+
#   # scale_fill_distiller(name = "SNPs", palette = "Viridis", direction = -1)+
#   # scale_fill_distiller(name = "ANI", palette = "Reds", direction = 1)+
#   theme_minimal()+ # minimal theme
#  theme(axis.text.y = element_text(size = 12),
#       axis.text.x = element_text(angle = 45, vjust = 1, 
#     size = 12, hjust = 1))+
#  coord_fixed()
# # Print the heatmap
# print(ggheatmap_CRISPR)


ggheatmap_CRISPR <- ggplot(interaction_interest_genes, aes(species, gene,color = proteinID))+
 # geom_tile(color = "white",size=1.1,shape="circle")+
 geom_point(size=8,shape="circle")+
  geom_point(shape = 1,size = 8,colour = "black")+
  # theme_classic()+
 # scale_fill_gradient2(low = "grey", high = "red",
  # midpoint = 90, limit = c(80,100), space = "Lab",
  # name="protein id",na.value = 'white',colour="black",pch=21) +
   scale_color_gradient2(low = "grey", high = "red",
  midpoint = 90, limit = c(80,100), space = "Lab",
  name="protein id",na.value = 'white') +
  # scale_fill_distiller(name = "SNPs", palette = "Blues", direction = -1)+
  # scale_fill_viridis(alpha=0.8)+
  labs(legend="CRISPR ID",x="",y="")+
  # scale_fill_distiller(name = "SNPs", palette = "Viridis", direction = -1)+
  # scale_fill_distiller(name = "ANI", palette = "Reds", direction = 1)+
  theme_minimal()+ # minimal theme
 theme(axis.text.y = element_text(size = 12),
      axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1))+
 coord_fixed()
# Print the heatmap
print(ggheatmap_CRISPR)


##===================================================
##plot
##=================================================== 

 
  svg("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder//HEATMAP_interactionGenes.svg",width=3,height=9)

print(ggheatmap_CRISPR)

 dev.off()
```

eps

```{bash}

cd /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/
species=Sterm
for species in $(echo "Ldel Sterm")
do
####================
##OGs
####================

##-----------prt
prtSsss=$(grep "prtS" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/eggnog_orthifinder_${species}_annotations.tsv | cut -f 1)
##-----------pep
#prtSsss=$(grep "pep" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/eggnog_orthifinder_${species}_annotations.tsv | cut -f 1)
pepA=$(awk -F "\t" '{OFS="\t"}{if($6~"eps")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
awk -F "\t" '{OFS="\t"}{if($6~"eps")print $0}' eggnog_orthifinder_${species}_annotations.tsv
awk -F "\t" '{OFS="\t"}{if($6~"cpo")print $0}' eggnog_orthifinder_${species}_annotations.tsv
grep "Polysaccharide" -i eggnog_orthifinder_${species}_annotations.tsv
done


```

##CRISPR ANALYSIS

#### CRISPRcasFinder
added 28.9.2020
```{bash}



#for species in $(echo "Ldel Sterm")
for species in $(echo "Sterm")

do
echo ==========================
     echo "working on: " ${species}
     echo ==========================
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}
for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/ |grep "_202_"|grep ".fasta$" |sed 's/.fasta//g')
do
      echo ==========================
     echo -e "extracting strain: " ${genomes}
     echo ==========================
     cd /home/vincent/apps/CRISPRCasFinder/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/${genomes}_cas
/usr/bin/perl ./CRISPRCasFinder.pl -in /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}//${genomes}.fasta \
  -cf -cas CasFinder-2.0.3 -def G -keep -html -log  -cpuM 37 -ccc \
  -out /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/${genomes}_cas/


  done #genomes
done #species

##-------------------------
#some analsis
##-------------------------
genomes=S_O_202_24853

for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/ |grep "_202_"|grep ".fasta$" |sed 's/.fasta//g')
do
echo "-------------------------------------------------------------------------------"
echo $genomes
echo "-----------"
grep "lead" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/${genomes}_cas//GFF/*.gff
#grep "^Un" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/${genomes}_cas/TSV/Crisprs_REPORT.tsv|awk -F "\t" -v vasrss="$genomes" '{OFS="\n"}{print vasrss"-"$9,$11}'

#grep "^Un" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/${genomes}_cas/TSV/Crisprs_REPORT.tsv|awk -F "\t" -v vasrss="$genomes" '{OFS="\t"}{print vasrss,$15}'

done

##-------------------------
#find orientation
##-------------------------
genomes=S_O_202_24853

rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes/ARRAY_orientation.txt
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes/
for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/ |grep "_202_"|grep ".fasta$" |sed 's/.fasta//g')
do
echo "-------------------------------------------------------------------------------"
echo $genomes
echo "-----------"
grep "^Un" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/${genomes}_cas/TSV/Crisprs_REPORT.tsv|awk -F "\t" -v vasrss="$genomes" '{OFS="\n"}{print vasrss"-"$9,$11}'

grep "^Un" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/${genomes}_cas/TSV/Crisprs_REPORT.tsv|awk -F "\t" -v vasrss="$genomes" '{OFS="\n"}{print ">"vasrss"-"$9,$11}' >>  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes/ARRAY_orientation.txt

done
###----add the reference repeats from chunk CRISPR-repeat
cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/cdHIT/CRISPR_Clustering.txt >>  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes/ARRAY_orientation.txt

###----cluster

/home/vincent/apps/cd-hit-v4.8.1-2019-0228/cd-hit-est -i  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes/ARRAY_orientation.txt \
    -o  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes/CRISPR_Clustering.txt -c 0.80
    
###----only use big clusters

grep -B 1000 ">Cluster 3"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes/CRISPR_Clustering.txt.clstr

###----loop through arrays
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes/orientation_CRISPRarrays.txt
for species in $(echo "Sterm")
do
echo ==========================
     echo "working on: " ${species}
     echo ==========================
species=Sterm
count=3

##----------prime
start_num=0
i=1
###-----------------
##extract information
###-----------------
for i in $(seq 1 $count)
  do 
  echo $i
  

  sed -n "/>Cluster $start_num$/,/>Cluster $i$/p" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes/CRISPR_Clustering.txt.clstr > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes//tmpCRISPRS.txt

ClusterName=$(head -1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}//tmpCRISPRS.txt |sed 's/>//g'|sed 's/ /_/g')

grep -v "^>Clu" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes//tmpCRISPRS.txt > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes//tmpCRISPRS_02.txt

numSpacers=$(wc -l /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes//tmpCRISPRS_02.txt|cut -d ' ' -f 1)

arrayNum=$(grep ">ARRAY" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes//tmpCRISPRS_02.txt |awk -F "[>...]" '{print $2}'|sed 's/ARRAY//g')

for spacersss in $(awk -F "[>...]" '{print $2}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes//tmpCRISPRS_02.txt|grep -v "^ARRAY")
do

sample=$(echo ${spacersss}| cut -d '-' -f 1|cut -d '_' -f 4)
orientation=$(echo ${spacersss}| cut -d '-' -f 2|head -c 3)


 echo -e ${sample}"\t"${orientation}"\tCRISPR array "${arrayNum}
 echo -e ${sample}"\t"${orientation}"\tCRISPR array "${arrayNum} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/${species}/allGenomes/orientation_CRISPRarrays.txt


done #spacersss

start_num=$((start_num+1))
done #i

done #species

```
move local
```{bash}
mkdir -p /home/vincent/Desktop/Projects//2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/Sterm/allGenomes/

scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/Sterm/allGenomes/orientation_CRISPRarrays.txt /home/vincent/Desktop/Projects//2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/Sterm/allGenomes/orientation_CRISPRarrays.txt



```


####PilerCR crispr 

Importantly gather all strains first

I identify the CRISPR spacers with [PilerCR](https://www.drive5.com/pilercr/).

```{bash}


##=============
##pilerCR
##=============

for species in $(echo "Ldel Sterm")
do
echo ==========================
     echo "working on: " ${species}
     echo ==========================
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}
for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/ |grep "_202_"|grep ".fasta$" |sed 's/.fasta//g')
do
      echo ==========================
     echo -e "extracting strain: " ${genomes}
     echo ==========================
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${genomes}



  ~/apps/PilerCR/pilercr1.06/pilercr -in /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}//${genomes}.fasta \
    -out /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${genomes}/${genomes}.pilarCR_out -noinfo\
    -seq /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${genomes}/${genomes}.pilarCR_out.fasta

  
  perl ~/apps/PilerCR/PilerCR_parser_new_vincent /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${genomes}/${genomes}.pilarCR_out > \
    /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${genomes}/${genomes}.pilarCR_out_final
  done #genomes
  #  grep ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/${id}.pilarCR_out_final_new | sed 's/>//g' |awk -F "__" '{OFS="\t"} {print $4,$2,$3,$1}' > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/${id}.pilarCR_out_spacer.bed

done #species

##------------------------------------------------------------
##check directionality of CRISPRS
##------------------------------------------------------------

species=Sterm
for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/ |grep "_202_"|grep ".fasta$" |sed 's/.fasta//g')
do
echo "--------------------------------------------------------------------------------------"
echo  ${genomes}
grep "SUMMARY BY SIMILARITY" -A 15   /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${genomes}/${genomes}.pilarCR_out
done
```
all CRISPR arrays are in the same orientation.


###CRISPR structure

#####unique CRISPR repeats

Here, I identify the number of different/unique CRISPR repeats.
Thereupon I relabel the genome arrays according to the common repeat number.
Then I calculate the repeat nucleotide identity the different CRISPR arrays. 
I plot this with Weblogo

Because

make a multiSeq alignment file of all CRISPR arrays. 
We have multiple CRISPR arrays in every species but they all come from the same

```{bash}

###------------
##change array names
###every strain has a different array labelling according to the CRISPR locations 
##here I want to make them all the same (because the CRISPRs are within one of three arrays)
###------------
for species in $(echo "Ldel Sterm")
do
echo "##=========================="
echo $species
echo "##=========================="
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}
#for genomess in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/ |grep ".fna$" |sed 's/.fna//g')
for genomess in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/ |grep "_202_"|grep ".fasta$" |sed 's/.fasta//g')
do

sed 's/\[Array/-a/' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${genomess}/${genomess}.pilarCR_out.fasta |sed 's/\;.*\]//' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/all_repeats.fasta


done #genomes


done # species

#======================================
##---------------------------cd -hit
#======================================

for species in $(echo "Ldel Sterm")
do
echo "##=========================="
echo $species
echo "##=========================="

/home/vincent/apps/cd-hit-v4.8.1-2019-0228/cd-hit-est -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/all_repeats.fasta \
    -o  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/CRISPR_Clustering.txt -c 0.80
    

done # species

##------------manaully assign the Clusters to the arrays

echo -e ">Cluster_0\ta3\n>Cluster_1\ta1\n>Cluster_2\ta2" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/Sterm/Arrays.log
echo -e ">Cluster_0\ta1\n>Cluster_1\ta2" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/Ldel/Arrays.log


#======================================
##---------------------------prepare multi-seq alignment file
#======================================


for species in $(echo "Ldel Sterm")
do
echo "##=========================="
echo $species
echo "##=========================="
 rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/cdHIT/
 mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/cdHIT/
for clustersss in $(cut -f 1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/Arrays.log)
do
aryName=$(grep "${clustersss}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/Arrays.log |cut -f 2)
ClusterNumber=$(echo $clustersss|awk -F "_" '{print $2}')
EndCluster=$((ClusterNumber+1))

sed -n "/>Cluster $ClusterNumber$/,/>Cluster $EndCluster$/p" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/CRISPR_Clustering.txt.clstr | grep -v "^>Cl" |awk -F "[>...]" '{print $2}' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/cdHIT//tmpCRISPRS_repeats.txt 

num=1
for samplesss in $(cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/cdHIT//tmpCRISPRS_repeats.txt)
do

grep "${samplesss}" -A 1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/all_repeats.fasta >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/cdHIT/${species}_${aryName}_CRISPRrepeats_all.fasta

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/cdHIT/${species}_${aryName}_CRISPRrepeats_all.fasta
#-----------------
done #samplesss
done # clustersss
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/cdHIT//tmpCRISPRS_repeats.txt
done # species

##--------------------
##clustalw
##creat weblogo of alignment
##--------------------
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/WebLogo
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/WebLogo
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/ClustalW
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/ClustalW
for species in $(echo "Ldel Sterm")
do

for aryName in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/cdHIT/ |grep ".fasta$" |cut -d '_' -f 2)
do

clustalw -INFILE=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/${species}/cdHIT/${species}_${aryName}_CRISPRrepeats_all.fasta -ALIGN -OUTFILE=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/ClustalW/${species}_${aryName}_CRISPRrepeats_all
/home/vincent/apps/weblogo/seqlogo -f /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/ClustalW/${species}_${aryName}_CRISPRrepeats_all -F EPS -k 1 -o /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/WebLogo/${species}_${aryName}_CRISPRrepeats

done #aryName
done # species
```


move local 

```{bash}

scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/WebLogo/ /home/vincent/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/


scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/REPEATS/ClustalW/ /home/vincent/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/

```

#####tracrRNA

Here, I identity the tracrRNA.

trans-activating crRNA (tracrRNA) is important to form a RNA duplexes by the base-pairing of tracrRNA with the pre-crRNA repeats in the presence of Cas9.

Identify the tracrRNA by looking for the anti CRISPR repeats on the genome

traccRNA prediction (http://www.tandfonline.com/doi/full/10.1080/15476286.2018.1498281)
https://www.ncbi.nlm.nih.gov/pubmed/24811454

this can be done with the motif search tool directly in IGV. 
Therefore we need to also search for the reverse complement
```{bash}
## tracrRNA
grep -i "tracrRNA"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/*.gff
grep -i "crRNA"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/*.gff
grep -i "crRNA"  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/S_thermophilus_RMK202.current.gb

###-------------
## reverse complement
###-------------
revseq /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Sterm_CRISPR_repeat.txt /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Sterm_CRISPR_repeat_reverse_complement.txt 


revseq -complement false /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Sterm_CRISPR_repeat.txt /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Sterm_CRISPR_repeat_reverse.txt 

```

move local

```{bash}
species=Sterm
mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/${species}/
scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/GBF_all/ ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/${species}/


species=Ldel
mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/${species}/
scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/GBF_all/ ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/${species}/

```

As the the tracrRNA finder tools such as [traccRNA prediction](http://www.tandfonline.com/doi/full/10.1080/15476286.2018.1498281) did not work very well I will no try to identify them manually. 

IGV has a usueful motif finder function but it does not work with a single error from the sequence therefore I extract the CAS-gene casette and blast the repeat (and possibly also the rev-complemement) to this extracted regions. 

Therefore I first mask the CRISPRrepeat regions and then map the CRISPR repeats to the remaining genome to identify the tracrRNAs. 

```{bash}

###--------------------------------
##mask CRISPR repeat region for bwa mapping of repeat to identify anti CRISPRs (tracrRNAs)
###--------------------------------
grep "repeat" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/CRISPR_repeats_SMAG.bed


bedtools maskfasta -fi /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/Sterm/renamedContigs/StartAligned/StartAligned/202-SMAG.fna -bed /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/CRISPR_repeats_SMAG.bed -fo /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/masked_CRISPR_repeats_SMAG.fasta


###--------------------------------
#Blast repeats to CAS locations
###--------------------------------

makeblastdb -max_file_sz 1GB -in /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/masked_CRISPR_repeats_SMAG.fasta \
 -out /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/masked_CRISPR_repeats_SMAG -parse_seqids -dbtype nucl


  blastn -num_threads 35 -max_hsps 40 -max_target_seqs 40 -task megablast -show_gis \
  -query /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Sterm_CRISPR_repeat.txt \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen sgi sacc" \
  -db /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/masked_CRISPR_repeats_SMAG \
  -out /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/CAS_repeat_matches.blast \
  -evalue 0.01 -word_size 16


awk -F "\t" '{OFS="\t"}{print "CP046134",$9,$10}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/CAS_repeat_matches.blast > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/CAS_repeat_matches.bed


#awk -F "\t" '{OFS="\t"}{print $2,$9,$10}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/CAS_repeat_matches.blast > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/CAS_repeat_matches.bed
```

move local

```{bash}
mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/CAS_repeat_matches.bed
scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/CAS_repeat_matches.bed ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/CAS_repeat_matches.bed


```

#####geneIdentity CAS
here I want to do a multi-seq alignment of all Cas-genes to identify the similarity. 

```{bash}


#for species in $(echo "Ldel Sterm"|tail -1)
for species in $(echo "Sterm")
do
echo ==========================
     echo "working on: " ${species}
     echo ==========================
#rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}//GeneID/{ary1,ary2,ary3}
#for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/|grep ".fna$" |sed 's/.fna//g')
for genomess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/${species}/ |grep "_202_"|grep ".fasta$" |sed 's/.fasta//g'|cut -d '_' -f 4)
do
genomes=$(echo -e "202-"${genomess})
      echo ==========================
     echo -e "extracting strain: " ${genomes}
     echo ==========================

##-------------------ARY1
grep "cas9" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}'| awk -F "\t" '{OFS="\t"}{if($7=="+") print $0}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary1/cas9.gff
grep "cas2" -B 6 -A 2 -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}'|grep "cas9" -A 4|grep "cas1"|awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}'| awk -F "\t" '{OFS="\t"}{if($7=="+") print $0}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary1/cas1.gff
grep "cas2" -B 6 -A 2 -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}'|grep "cas9" -A 4|grep "cas2"| awk -F "\t" '{OFS="\t"}{if($7=="+") print $0}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary1/cas2.gff
grep "cas2" -B 6 -A 2 -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}'|grep "cas9" -A 4|grep "csn2" |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}'| awk -F "\t" '{OFS="\t"}{if($7=="+") print $0}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary1/csn2.gff

##-------------------ARY3
grep "cas2" -A 6 -B 2 -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}'|grep "cas9" -B 3|grep "cas9"|awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}'| awk -F "\t" '{OFS="\t"}{if($7=="-") print $0}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary3/cas9.gff
grep "cas2" -A 6 -B 2 -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}'|grep "cas9" -B 3|grep "cas1" |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}'| awk -F "\t" '{OFS="\t"}{if($7=="-") print $0}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary3/cas1.gff
grep "cas2" -A 6 -B 2 -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}'|grep "cas9" -B 3|grep "cas2"|awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}'| awk -F "\t" '{OFS="\t"}{if($7=="-") print $0}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary3/cas2.gff

grep "cas2" -A 6 -B 2 -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}'|grep "cas9" -B 3|head -1 >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary3/csn2-like.gff

##-------------------ARY2

grep "csm6" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary2/csm6.gff
grep "csm5" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary2/csm5.gff
grep "csm4" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary2/csm4.gff
grep "csm3" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary2/csm3.gff
grep "cas10" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="CDS") print $0}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary2/cas10.gff
grep "cas6" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary2/cas6.gff
grep "cas6" -A 6 /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |grep "cas1" |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary2/cas1.gff
grep "cas6" -A 6 /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.gff |grep "cas2"| awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/ary2/cas2.gff


cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA*.fna >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/AllGenomes.fna
  done #genomes
  #  grep ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/${id}.pilarCR_out_final_new | sed 's/>//g' |awk -F "__" '{OFS="\t"} {print $4,$2,$3,$1}' > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/${id}.pilarCR_out_spacer.bed

done #species

###------------get fasta

#for species in $(echo "Ldel Sterm"|tail -1)
for species in $(echo "Sterm")
do
echo ==========================
     echo "working on: " ${species}
     echo ==========================
#rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}
#rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}//GeneID/alignmentScore.txt

mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}//GeneID/{ary1,ary2,ary3}
#for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/|grep ".fna$" |sed 's/.fna//g')
#do
   

for arrayzzz in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}//GeneID/ |grep "^ary")
do

   echo ==========================
     echo -e "extracting arrayzzz: " ${arrayzzz}
     echo ==========================
for geness in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}//GeneID/${arrayzzz} |grep ".gff"|sed 's/.gff//g')
do


   echo ==========================
     echo -e "extracting geness: " ${geness}
     echo ==========================

#bedtools getfasta -fi /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}/GeneID/AllGenomes.fna \
#  -bed /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}//GeneID/${arrayzzz}/${geness}.gff \
#  -fo  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}//GeneID/${arrayzzz}/${geness}.fna

#muscle -clw -in /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}//GeneID/${arrayzzz}/${geness}.fna -out /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}//GeneID/${arrayzzz}/${geness}.msa


#clustalw -INFILE=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}//GeneID/${arrayzzz}/${geness}.fna -ALIGN -OUTFILE=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}//GeneID/${arrayzzz}/${geness}_clustalwAlignement | grep "Aligned. Score:" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}//GeneID/${arrayzzz}/${geness}_clustalwAlignement_Score.txt


sed 's/) Aligned. Score:  /\t/g' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}//GeneID/${arrayzzz}/${geness}_clustalwAlignement_Score.txt | sed 's/Sequences (//g' |awk -F "\t" -v namesss="$species" -v arraysss="$arrayzzz" -v genezzz="$geness" '{OFS="\t"}{print $1,$2,namesss,arraysss,genezzz}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/${species}//GeneID/alignmentScore.txt


done #geness
done #arrayzzz
#  done #genomes

done #species
```
file of interest is the .dnd file which contains the alignment score.

```{bash}

mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/Sterm//GeneID/
scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/Sterm//GeneID/alignmentScore.txt /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/Sterm//GeneID/

```



```{r}
library(readr)
library(ggplot2)
library(tidyverse)
alignmentScore_casGenes <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/Sterm/GeneID/alignmentScore.txt", "\t", escape_double = FALSE, col_names = c("comparison","alignentScore","species","array","gene"),  trim_ws = TRUE)

ggplot(alignmentScore_casGenes,aes(x=gene,y=alignentScore,fill=gene))+geom_boxplot()+theme_classic()+facet_wrap(~array)+lims(y=c(80,100))


 d2 <- alignmentScore_casGenes %>%
  group_by(interaction(gene,array)) %>% 
  dplyr::summarise(median=median(alignentScore),mean=mean(alignentScore)) 
# d2 <- cbind(d2,d2,d2)

 d2$gene <- str_split_fixed(as.character(d2$`interaction(gene, array)`), ".ary", 3)[,1]
 d2$array <- str_split_fixed(as.character(d2$`interaction(gene, array)`), ".ary", 3)[,2]
   d2

##===============================================
##heatmap
##===============================================

ggheatmap_ANI <- ggplot(d2, aes(gene, array, fill = median))+
 geom_tile(color = "blue",size=1.1)+
 scale_fill_gradient2(low = "white", high = "black",
  midpoint = 96, limit = c(95,100), space = "Lab",
  name="alignmentscore") +
  scale_y_discrete(position = "right")+
  labs(legend="ANI",x="",y="")+
  # scale_fill_distiller(name = "SNPs", palette = "Blues", direction = -1)+
  # scale_fill_distiller(name = "ANI", palette = "Reds", direction = 1)+
  theme_minimal()+ # minimal theme
 theme(axis.text.y = element_text(size = 12),
      axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1))+
 coord_fixed()
# Print the heatmap
 svg("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/Sterm/GeneID/alignmentScore_heatmap.svg",width=8,height=4)

print(ggheatmap_ANI)

dev.off()

```



#####making the plot

making the CRISPR-cas region plot with genoplotR. 
First I need to prepare the data
important is to make three files for every ary one
```{bash}
##=============================
##extract the location for all necessary componenets of the three arrays
##=============================

##------------------
##prepare files for every ary
##------------------
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles

echo -e "name\tstart\tend\tstrand\tcol\tfill\tgene_type" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary1_genoPlotR.txt
echo -e "name\tstart\tend\tstrand\tcol\tfill\tgene_type" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary2_genoPlotR.txt
echo -e "name\tstart\tend\tstrand\tcol\tfill\tgene_type" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary3_genoPlotR.txt

##------------------
##tracrRNA
##------------------
#tracrRNA="red"

head -1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/CAS_repeat_matches.blast |awk -F "\t" '{OFS="\t"}{print "tracrRNA",$9,$10,"1","black","red","arrows"}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary1_genoPlotR.txt

tail -1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/tracrRNA/manual_search/CAS_repeat_matches.blast |awk -F "\t" '{OFS="\t"}{print "tracrRNA",$9,$10,"1","black","red","arrows"}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary3_genoPlotR.txt

##------------------
##ary 1,3
##------------------

##ary1
grep "cas9" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |head -1|awk -F "\t" '{OFS="\t"}{if($7=="+") print "cas9",$4,$5,"1","black","#440154FF","arrows"; else print "cas9",$4,$5,"-1","black","#440154FF","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary1_genoPlotR.txt
grep "cas1" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |head -1|awk -F "\t" '{OFS="\t"}{if($7=="+") print "cas1",$4,$5,"1","black","#39568CFF","arrows"; else print "cas1",$4,$5,"-1","black","#39568CFF","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary1_genoPlotR.txt
grep "cas2" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |head -1|awk -F "\t" '{OFS="\t"}{if($7=="+") print "cas2",$4,$5,"1","black","#1F968BFF","arrows"; else print "cas2",$4,$5,"-1","black","#1F968BFF","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary1_genoPlotR.txt
grep "csn2" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |head -1|awk -F "\t" '{OFS="\t"}{if($7=="+") print "csn2",$4,$5,"1","black","#55C667FF","arrows"; else print "csn2",$4,$5,"-1","black","#55C667FF","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary1_genoPlotR.txt

#csn2="#55C667FF"
##ary3
grep "cas9" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |tail -1|awk -F "\t" '{OFS="\t"}{if($7=="+") print "cas9",$4,$5,"1","black","#440154FF","arrows"; else print "cas9",$4,$5,"-1","black","#440154FF","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary3_genoPlotR.txt
grep "cas1" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |tail -1|awk -F "\t" '{OFS="\t"}{if($7=="+") print "cas1",$4,$5,"1","black","#39568CFF","arrows"; else print "cas1",$4,$5,"-1","black","#39568CFF","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary3_genoPlotR.txt
grep "cas2" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |tail -1|awk -F "\t" '{OFS="\t"}{if($7=="+") print "cas2",$4,$5,"1","black","#1F968BFF","arrows"; else print "cas2",$4,$5,"-1","black","#1F968BFF","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary3_genoPlotR.txt
grep "202-SMAG_01302_gene" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |awk -F "\t" '{OFS="\t"}{if($7=="+") print "Stu0660‐like Csn2",$4,$5,"1","black","grey","arrows"; else print "Stu0660‐like Csn2",$4,$5,"-1","black","grey","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary3_genoPlotR.txt

##------------------
##ary 2
##------------------
grep "csm6" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |awk -F "\t" '{OFS="\t"}{if($7=="+") print "csm6",$4,$5,"1","black","#6B4596FF","arrows"; else print "csm6",$4,$5,"-1","black","#6B4596FF","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary2_genoPlotR.txt

grep "csm5" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |awk -F "\t" '{OFS="\t"}{if($7=="+") print "csm5",$4,$5,"1","black","#90548bff","arrows"; else print "csm5",$4,$5,"-1","black","#90548bff","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary2_genoPlotR.txt

grep "csm4" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |awk -F "\t" '{OFS="\t"}{if($7=="+") print "csm4",$4,$5,"1","black","#b8627dff","arrows"; else print "csm4",$4,$5,"-1","black","#b8627dff","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary2_genoPlotR.txt

grep "csm3" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |awk -F "\t" '{OFS="\t"}{if($7=="+") print "csm3",$4,$5,"1","black","#de7065ff","arrows"; else print "csm3",$4,$5,"-1","black","#de7065ff","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary2_genoPlotR.txt

grep "csm2" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |awk -F "\t" '{OFS="\t"}{if($7=="+") print "csm2",$4,$5,"1","black","#f68f46ff","arrows"; else print "csm2",$4,$5,"-1","black","#f68f46ff","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary2_genoPlotR.txt

grep "cas10" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="CDS") print $0}' |awk -F "\t" '{OFS="\t"}{if($7=="+") print "cas10",$4,$5,"1","black","#f9a242ff","arrows"; else print "cas10",$4,$5,"-1","black","#f9a242ff","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary2_genoPlotR.txt

grep "cas6" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |awk -F "\t" '{OFS="\t"}{if($7=="+") print "cas6",$4,$5,"1","black","#f7cb44ff","arrows"; else print "cas6",$4,$5,"-1","black","#f7cb44ff","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary2_genoPlotR.txt


grep "cas1" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |grep "202-SMAG_01043_gene" |awk -F "\t" '{OFS="\t"}{if($7=="+") print "cas1",$4,$5,"1","black","#39568CFF","arrows"; else print "cas1",$4,$5,"-1","black","#39568CFF","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary2_genoPlotR.txt

grep "cas2" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |grep "202-SMAG_01042_gene" |awk -F "\t" '{OFS="\t"}{if($7=="+") print "cas2",$4,$5,"1","black","#1F968BFF","arrows"; else print "cas2",$4,$5,"-1","black","#1F968BFF","arrows"}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary2_genoPlotR.txt

##------------------
##repeats
##------------------
##ary 1
sed -n "/^Array 1$/,/^Array 2$/p"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/Sterm/202-SMAG/202-SMAG.pilarCR_out | sed -n "/^==========/,/^==========/p" |grep "==" -v | sed 's/^ *//g' |sed 's/ \+ /\t/g'|awk -F "\t" '{OFS="\t"}{print "repeat",$1,$1+$2,"1","black","black","blocks"}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary1_genoPlotR.txt

##ary 2
sed -n "/^Array 2$/,/^Array 3$/p"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/Sterm/202-SMAG/202-SMAG.pilarCR_out | sed -n "/^==========/,/^==========/p" |grep "==" -v | sed 's/^ *//g' |sed 's/ \+ /\t/g'|awk -F "\t" '{OFS="\t"}{print "repeat",$1,$1+$2,"1","black","black","blocks"}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary2_genoPlotR.txt

##ary 3
sed -n "/^Array 3$/,/^Array 4$/p"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/Sterm/202-SMAG/202-SMAG.pilarCR_out | sed -n "/^==========/,/^==========/p" |grep "==" -v | sed 's/^ *//g' |sed 's/ \+ /\t/g'|awk -F "\t" '{OFS="\t"}{print "repeat",$1,$1+$2,"1","black","black","blocks"}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary3_genoPlotR.txt

sed -n "/^Array 4$/,/^SUMMARY BY SIMILARITY/p"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/Sterm/202-SMAG/202-SMAG.pilarCR_out | sed -n "/^==========/,/^==========/p" |grep "==" -v | sed 's/^ *//g' |sed 's/ \+ /\t/g'|awk -F "\t" '{OFS="\t"}{print "repeat",$1,$1+$2,"1","black","black","blocks"}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary3_genoPlotR.txt


##------------------
##spacers
##------------------

##ary 1
sed -n "/^Array 1$/,/^Array 2$/p"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/Sterm/202-SMAG/202-SMAG.pilarCR_out | sed -n "/^==========/,/^==========/p" |grep "==" -v | sed 's/^ *//g' |sed 's/ \+ /\t/g'|awk -F "\t" '{OFS="\t"}{if($4>10) print "spacer",$1+$2,$1+$2+$4,"1","black","green","exons"}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary1_genoPlotR.txt

##ary 2
sed -n "/^Array 2$/,/^Array 3$/p"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/Sterm/202-SMAG/202-SMAG.pilarCR_out | sed -n "/^==========/,/^==========/p" |grep "==" -v | sed 's/^ *//g' |sed 's/ \+ /\t/g'|awk -F "\t" '{OFS="\t"}{if($4>10) print "spacer",$1+$2,$1+$2+$4,"1","black","green","exons"}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary2_genoPlotR.txt


##ary 3
sed -n "/^Array 3$/,/^Array 4$/p"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/Sterm/202-SMAG/202-SMAG.pilarCR_out | sed -n "/^==========/,/^==========/p" |grep "==" -v | sed 's/^ *//g' |sed 's/ \+ /\t/g'|awk -F "\t" '{OFS="\t"}{if($4<40) print "spacer",$1+$2,$1+$2+$4,"1","black","green","exons"}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary3_genoPlotR.txt

sed -n "/^Array 4$/,/^SUMMARY BY SIMILARITY/p"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/Sterm/202-SMAG/202-SMAG.pilarCR_out | sed -n "/^==========/,/^==========/p" |grep "==" -v | sed 's/^ *//g' |sed 's/ \+ /\t/g'|awk -F "\t" '{OFS="\t"}{if($4<40) print "spacer",$1+$2,$1+$2+$4,"1","black","green","exons"}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ary3_genoPlotR.txt

```

move local

```{bash}
mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/
scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/ ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/
```


now  plot
```{r}
library(genoPlotR)
library(tidyverse)

##-------------------------
##ary1
##-------------------------
library(readr)
ary1_genoPlotR <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/arrayFiles/ary1_genoPlotR.txt", "\t", escape_double = FALSE, trim_ws = TRUE)  %>% as.data.frame() #%>% filter(name!="spacer") #%>% select(-fill)
dna_seg1 <- dna_seg(ary1_genoPlotR)
dna_segs <- list(dna_seg1)
plot_gene_map(dna_segs=dna_segs)

##-------------------------
##ary3
##-------------------------
library(readr)
ary3_genoPlotR <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/arrayFiles/ary3_genoPlotR.txt", "\t", escape_double = FALSE, trim_ws = TRUE)  %>% as.data.frame() #%>% filter(name!="spacer") #%>% select(-fill)
# ary1_genoPlotR$gene_type <- revalue(ary1_genoPlotR$gene_type, c("triangleGrob"="blocks"))
# ary1_genoPlotR$lty <- 0.5 #necessary depending on plotting size
# ary1_genoPlotR <- head(ary1_genoPlotR,n = 4)
dna_seg3 <- dna_seg(ary3_genoPlotR)
dna_segs3 <- list(dna_seg3)
plot_gene_map(dna_segs=dna_segs3)


##-------------------------
##ary2
##-------------------------
library(readr)
ary2_genoPlotR <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/arrayFiles/ary2_genoPlotR.txt", "\t", escape_double = FALSE, trim_ws = TRUE)  %>% as.data.frame() #%>% filter(name!="spacer") #%>% select(-fill)
# ary1_genoPlotR$gene_type <- revalue(ary1_genoPlotR$gene_type, c("triangleGrob"="blocks"))
# ary1_genoPlotR$lty <- 0.5 #necessary depending on plotting size
# ary1_genoPlotR <- head(ary1_genoPlotR,n = 4)
dna_seg2 <- dna_seg(ary2_genoPlotR)
dna_segs2 <- list(dna_seg2)
plot_gene_map(dna_segs=dna_segs2)

##-------------------------
##merged
##-------------------------


dna_segs <- list(dna_seg1, dna_seg3, dna_seg2)

# annots <- lapply(dna_segs, function(x){mid <- middle(x)annot <- annotation(x1=mid, text=x$name, rot=30)

annots <- lapply(dna_segs, function(x){
  mid <- middle(x)
  annot <- annotation(x1=mid, text=x$name, rot=30)
idx <- which(!annot$text %in% c("repeat","spacer"))
# annot[idx[idx %% 4 == 0],]
annot[idx[idx ],]
})

plot_gene_map(dna_segs=dna_segs,annotations=annots)

##-------------------------
##plot
##-------------------------


    # png("~/Desktop/Projects/2019_RMK202_analysis/10_assembly/heatmap_all_Sterm_strains.png", width = 4000, height = 4000,res=300)
svg("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/plot/all_spacersPlot.svg",width=6,height=6)
plot_gene_map(dna_segs=dna_segs,annotations=annots)

dev.off()
```

the two putative csn2 protein from ary3 is not annotated. therefore I want to check if it is an annotation problem or if it is different from csn2 from array 1 with a multiseq- allignment

```{bash}

mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/csn2/
##csn2 from ary 1

grep "202-SMAG_00535_gene" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff  |head -1
grep "202-SMAG_00535" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.ffn
samtools faidx /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.ffn 202-SMAG_00535 > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/csn2/geneExtracted_ary1.fasta
##csn2 from ary3
grep "202-SMAG_01302_gene" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.gff |head -1
grep "202-SMAG_01302" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.ffn
samtools faidx /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.ffn 202-SMAG_01302 > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/csn2/geneExtracted_ary3.fasta
##--------------
##get faa for paperblast
##--------------
samtools faidx /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA*.faa 202-SMAG_01302 > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/csn2/geneExtracted_ary3.faa
##--------------
##alignement
##--------------

needle -asequence /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/csn2/geneExtracted_ary1.fasta \
      -bsequence /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/csn2/geneExtracted_ary3.fasta -gapopen 20 -gapextend 0 -outfile /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/csn2/aligment.msa
```

```{bash}


mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/
scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/csn2/ ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/


```

### CRISPR-repeats
The CRISPR repeats are conserved. Therefore in the assemblies we should have the same repeats occuring again and again. 
working list:
-add sample name
-add REFs
-cd-hit
-if multiple arrays annotated to the same repeat and location the array is extended

```{bash}
###================
##description file
###================
threads=37
#samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt  ##the file with all sample names
#logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/04_withOlDsamples/repeats
#Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/FASTA/S_thermophilus_alltogether.CRISPRlocation.fasta



#===============================
##merge the repeats all together
#===============================

for species in $(echo "Ldel Sterm")
do
echo ==========================
     echo "working on: " ${species}
     echo ==========================
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/allRepeatsAssembled.fasta
#mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/
#for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned| grep ".fna$" |sed 's/.fna//g')
for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/ |grep "_202_"|grep ".fasta$" |sed 's/.fasta//g')
do
      echo ==========================
     echo -e "extracting repeats of strain: " ${genomes}
     echo ==========================

genomes_short=$(echo ${genomes}|sed 's/S_O_202_//g'|sed 's/S_I_202_//g'|sed 's/L_I_202_//g')

sed "s/.*\[Array/>_a/g" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${genomes}/${genomes}.pilarCR_out.fasta|sed 's/;.*$//g'|sed "s/>_/>${genomes_short}_/g" >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/allRepeatsAssembled.fasta



#sed 's/__.*$//g' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${genomes}/${genomes}.pilarCR_out.fasta | sed "s/>/>${genomes_short}_/g"| sed 's/ary0/a/g' |sed 's/spc0/s/g' |sed 's/202-//g' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/CRISPR_spacer_merged_short.fasta

done #genomes

done #species


#sed 's/\:.*\[/:/' ${BaseLocation}/assembledRepeats/allRepeatsAssembled.fasta |sed 's/Array/a0/' |sed 's/\;.*\]//' |sed 's/_0.*\:/:/' > ${BaseLocation}/assembledRepeats/allRepeatsAssembled_short.fasta

#===============================
##cd-hit the repeats
#===============================

for species in $(echo "Ldel Sterm")
do
echo ==========================
     echo "working on: " ${species}
     echo ==========================
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/cdHIT/
  mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/cdHIT/
/home/vincent/apps/cd-hit-v4.8.1-2019-0228/cd-hit-est -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/allRepeatsAssembled.fasta \
    -o  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/cdHIT/CRISPR_Clustering.txt -c 0.80
    
    #grep ">" ${BaseLocation}/assembledSpacers/allSpacersAssembled_short.fasta|cut -d ':' -f 1|sort|uniq -c

#cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/cdHIT/CRISPR_Clustering.txt

###------------
##change array names
###every strain has a different array labelling according to the CRISPR locations 
##here I want to make them all the same (because the CRISPRs are within one of three arrays)
###------------
##-------------------check!!
##this only runs if all strains have zero, one or two annotated CRISPR arrays for all of the three arrays
##all other scenario it cannot cope with
done #species

##!!!!!!!!!!!!!!!!!!!!!this step has to be done manually
##by looking in which Cluster the Reference (circular assembled) arrays lie
species=Sterm
genomes=202-SMAG
cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/cdHIT/CRISPR_Clustering.txt.clstr

mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/logs/
echo -e ">Cluster_0\ta3\n>Cluster_1\ta1\n>Cluster_2\ta2" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/logs/Arrays.log

species=Ldel
genomes=202-LMAG
cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/cdHIT/CRISPR_Clustering.txt.clstr

mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/logs/
echo -e ">Cluster_0\ta1\n>Cluster_1\ta2" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/logs/Arrays.log

#===============================
##assign the arrays of all strain to the reference array
#===============================

for species in $(echo "Ldel Sterm")
do
echo ==========================
     echo "working on: " ${species}
     echo ==========================
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final.txt
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final_WithouAccumulation.txt

for clustersss in $(cut -f 1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/logs/Arrays.log)
do
aryName=$(grep "${clustersss}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/logs/Arrays.log |cut -f 2)
ClusterNumber=$(echo $clustersss|awk -F "_" '{print $2}')
EndCluster=$((ClusterNumber+1))

sed -n "/>Cluster $ClusterNumber$/,/>Cluster $EndCluster$/p" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/cdHIT/CRISPR_Clustering.txt.clstr | grep -v "^>Cl" |awk -F "[>...]" '{print $2}' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt

#for samplesss in $(cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt)
num=1
for samplesss in $(cut -f 3 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt|sort|uniq|grep -v "sample")
do

echo -e ${num}
NumberCRISPRarrays=$(grep -c "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt)


if [ ${NumberCRISPRarrays} == "1" ]; then
              WhichCRISPRarrays=$(grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt|cut -d '_' -f 2|sed 's/a//g')
              grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,$5}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final.txt
              grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,$5}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final_WithouAccumulation.txt
            elif [[ ${NumberCRISPRarrays} == "0"  ]]
            then
            echo -e "No arrays in this Cluster "${aryName} "for "${samplesss} 
            elif [[ ${NumberCRISPRarrays} == "2"  ]]
            then
            echo -e "Two Arrays matcht for Cluster" ${aryName} "for "${samplesss}
            ##C----Array_ONe
            WhichCRISPRarrays=$(grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt|head -1|cut -d '_' -f 2|sed 's/a//g')
            grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,$5}' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats_Spacers.txt
            spacerCount=$(cut -f 5 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats_Spacers.txt |sort|tail -1)
            cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats_Spacers.txt >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final.txt
            cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats_Spacers.txt >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final_WithouAccumulation.txt
            ##C----Array_TWO
            WhichCRISPRarrays=$(grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt|tail -1|cut -d '_' -f 2|sed 's/a//g')
            grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Countss="${spacerCount}" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,Countss+$5}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final.txt
            grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Countss="${spacerCount}" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,$5}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final_WithouAccumulation.txt
            elif [[ ${NumberCRISPRarrays} == "3"  ]]
            then
            echo -e "Three Arrays matcht for Cluster" ${aryName} "for "${samplesss}
            ##C----Array_ONe
            WhichCRISPRarrays=$(grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt|head -1|cut -d '_' -f 2|sed 's/a//g')
            grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,$5}' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats_Spacers.txt
            spacerCount=$(cut -f 5 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats_Spacers.txt |sort|tail -1)
            cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats_Spacers.txt >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final.txt
            cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats_Spacers.txt >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final_WithouAccumulation.txt
            ##C----Array_TWO
            WhichCRISPRarrays=$(grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt|head -2|tail -1|cut -d '_' -f 2|sed 's/a//g')
            grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Countss="${spacerCount}" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,Countss+$5}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final.txt
            grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Countss="${spacerCount}" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,$5}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final_WithouAccumulation.txt
            ##C----Array_THREE
            WhichCRISPRarrays=$(grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt|head -3|tail -1|cut -d '_' -f 2|sed 's/a//g')
            grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Countss="${spacerCount}" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,Countss+$5}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final.txt
            grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Countss="${spacerCount}" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,$5}' >> /archiv/Projects/2019_RMK202_an
            else
               echo "more than three arrays-----Currently nothing is done with these"
               
               
            fi
num=$((num+1))
done #samplesss

done #clusterssss

done #species


```
Here, we can created the list of in which one of the the three Streptococci CRISPR arrays the spacer lies and at which postion.

```{bash}
##bring local
species=Sterm
mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/
scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/

scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final_WithouAccumulation.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final_WithouAccumulation.txt

##bring local
species=Ldel
mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/
scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final_WithouAccumulation.txt

scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final_WithouAccumulation.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final_WithouAccumulation.txt

```



### CRISPR spacers

######merge spacers
Here, I want to bring all annotated CRISPR spacers into one file.

However I also want to change the name of the spacer header to the following:
>"genomess"_a(number)_s(number)
I do this later

```{bash}

#===============================
##merge the spacers all together
#===============================

for species in $(echo "Ldel Sterm")
do

#specsss=$(echo ${species}|head -c 1)
echo ==========================
     echo "working on: " ${species}
     echo ==========================
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/CRISPR_spacer_merged_short.fasta
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/
for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/ |grep "_202_"|grep ".fasta$" |sed 's/.fasta//g')
#for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned| grep ".fna$" |sed 's/.fna//g')
do
      echo ==========================
     echo -e "extracting strain: " ${genomes}
     echo ==========================

 cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${genomes}/${genomes}.pilarCR_out_final >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/CRISPR_spacer_merged.fasta


genomes_short=$(echo ${genomes}|sed 's/S_O_202_//g'|sed 's/S_I_202_//g'|sed 's/L_I_202_//g')
sed 's/__.*$//g' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${genomes}/${genomes}.pilarCR_out_final | sed "s/>/>${genomes_short}_/g"| sed 's/ary0/a/g' |sed 's/spc0/s/g' |sed 's/202-//g' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/CRISPR_spacer_merged_short.fasta
done #genomes

grep ">" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}//CRISPR_spacer_merged_short.fasta
grep ">"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}//CRISPR_spacer_merged_short.fasta| awk -F "[_]" '{OFS="-"}{print $1}'|sort|uniq -c
done #species

```
This will be used later in the CRISPR spacer section

#####pairwise alignement
Here, I want to check for pairwise CRISPR diversity. 
We do this to see how many and which spacers are shared. 
First we align the spacers witha needle -munsch algorithm. 

This is currently not a crucial analysis
```{bash}
##====================================
##split  every CRIPSR into indiviual files
##====================================

for species in $(echo "Ldel Sterm")
do

samtools faidx /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}//CRISPR_spacer_merged_short.fasta

rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/seperated
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/seperated
for spacer in $(grep "^>" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}//CRISPR_spacer_merged_short.fasta|sed 's/>//g')
  do

samtools faidx /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}//CRISPR_spacer_merged_short.fasta ${spacer} > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/seperated/${spacer}.fasta

  done
  
done #species
##-------------
##pairwise needle-wunsch alignment
##-------------

##-------------do parallel
###------define TASK


task(){ 
    spacer2_final=$(echo ${spacer2} )
    needle -asequence /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/seperated/${spacer1_final}.fasta \
      -bsequence /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/seperated/${spacer2_final}.fasta -gapopen 20 -gapextend 0 -outfile /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/Pairwise_needle/${spacer1_final}__${spacer2_final}.msa

      length=$(grep "Length" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/Pairwise_needle/${spacer1_final}__${spacer2_final}.msa | cut -d ':' -f 2 | sed -e 's/^[ \t]*//' )
      Identity=$(grep "Identity" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/Pairwise_needle/${spacer1_final}__${spacer2_final}.msa | cut -d ':' -f 2 | cut -d '/' -f 1| sed -e 's/^[ \t]*//' )
      Gaps=$(grep "Gaps" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/Pairwise_needle/${spacer1_final}__${spacer2_final}.msa | cut -d ':' -f 2 | cut -d '/' -f 1| sed -e 's/^[ \t]*//' )
      Score=$(grep "Score" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/Pairwise_needle/${spacer1_final}__${spacer2_final}.msa | cut -d ':' -f 2 | sed -e 's/^[ \t]*//' )
      
     echo -e ${spacer1_final}"\t"${spacer2_final}"\t"${length}"\t"${Identity}"\t"${Gaps}"\t"${Score} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/Pairwise_needle/final_pairwise_alignment.txt
     
    rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/Pairwise_needle/${spacer1_final}__${spacer2_final}.msa

}



###------Run parallel in batches

for species in $(echo "Sterm Ldel")
do

rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/Pairwise_needle
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/Pairwise_needle
#counts=$(wc -l /data/Project/CRISPRspacers/02_PilerCR/20200217/all/pilerCR_out_repeats_short|cut -d ' ' -f 1)
N=300
for spacer1 in $(grep "^>" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}//CRISPR_spacer_merged_short.fasta|sed 's/>//g')
  do
  
  spacer1_final=$(echo ${spacer1})

  for spacer2 in $(grep "^>" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}//CRISPR_spacer_merged_short.fasta|sed 's/>//g')
    do
    
      spacer2_final=$(echo ${spacer2})

    ((i=i%N)); ((i++==0)) && wait
    ##--------------run task
    task &
    
    done #spacer2
    
  done #spacer1
  
  done #species

```
move local

```{bash}
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/04_CRISPR_spacer/

scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/Sterm/Pairwise_needle/final_pairwise_alignment.txt /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/04_CRISPR_spacer/Sterm_final_pairwise_alignment_new.txt

scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/Ldel/Pairwise_needle/final_pairwise_alignment.txt /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/04_CRISPR_spacer/Ldel_final_pairwise_alignment_new.txt

```

```{r}

##======================
##Sterm
##======================
library(readr)
library(ggplot2)
spacer_pairs <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/04_CRISPR_spacer/Sterm_final_pairwise_alignment_new.txt",  "\t", escape_double = FALSE, col_names = c("spacer1","spacer2","length","matching","gaps","score"),trim_ws = TRUE)

# spacer_pairs <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/final_pairwise_alignment_new.txt",  "\t", escape_double = FALSE, col_names = c("spacer1","spacer2","length","matching","gaps","score"),trim_ws = TRUE)

spacer_pairs$id <- 100*(spacer_pairs$matching/spacer_pairs$length)

##----------subset
100*(sum(spacer_pairs$id>90)/nrow(spacer_pairs))
100*(sum(spacer_pairs$id>90)/nrow(spacer_pairs))

##----------histogramm

hist(spacer_pairs$id)

spacer_pairs_similiar <- spacer_pairs[(spacer_pairs$id>80),]
hist(spacer_pairs_similiar$id)


##---------------export the shared spacers
 
unique_spacers <- unique(c(spacer_pairs_similiar$spacer1,spacer_pairs_similiar$spacer2))

write.table(unique_spacers, "~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/04_CRISPR_spacer/unique_Sterm_spacers_new.txt",na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names = FALSE)

##----------create matrix

# names <- levels(as.factor(spacer_pairs$spacer1))
# 
# pairwise_spacer_Matrix <- data.frame(matrix(NA, nrow = length(names), ncol = length(names)))
# colnames(pairwise_spacer_Matrix) <- names
# rownames(pairwise_spacer_Matrix) <- names
# 
# for (i in 1:length(names)) {
#    rowName <- names[i]
#     print(rowName)
#   for (a in 1:length(names)) {
#    rowName <- names[i]
#     print(rowName)
#     
#     colNames <- names[a]
#     print(colNames)
#     
#     pairwise_spacer_Matrix[rowName,colNames] <- spacer_pairs[spacer_pairs$spacer1==rowName & spacer_pairs$spacer2==colNames,"id"]
#     
#     }
#         
# # count <- count +1 
#   
# }
# dim(pairwise_spacer_Matrix)

# write.table(pairwise_spacer_Matrix, "~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/04_CRISPR_spacer/pairwise_spacer_Matrix_sterm.txt",na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names = TRUE)

pairwise_spacer_Matrix <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/04_CRISPR_spacer/pairwise_spacer_Matrix_sterm.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)

rownames(pairwise_spacer_Matrix) <- colnames(pairwise_spacer_Matrix)

##-----------heatmap 2
library(stringr)
library(rafalib)
library(RColorBrewer)
names_genome <- str_split_fixed(names, "_", 2)[,1]
cols <- palette(brewer.pal(3, "Dark2"))[as.fumeric(names_genome)]


library(gplots) ##Available from CRAN
pairwise_spacer_Matrix_ready <- as.matrix(pairwise_spacer_Matrix)
image <- heatmap.2(pairwise_spacer_Matrix_ready,
          trace="none",
          # Rowv = "none",
          Colv=NA,
          ColSideColors=cols,
          # density.info = "none") ##historgram in col scale
          tracecol="black",
          )

# 
# svg("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/Streptococcus_thermophilus_SPACERS.svg",width=5,height=5)
#  heatmap.2(pairwise_spacer_Matrix_ready,
#           trace="none",
#           # Rowv = "none",
#           Colv=NA,
#           cexRow=0.1,
#           ColSideColors=cols,
#           # density.info = "none") ##historgram in col scale
#           tracecol="black",
#           )
#  dev.off()

 
 
library(plotly)
# 
# library(orca)
# p <- heatmaply(pairwise_spacer_Matrix_ready, k_row = NA, dendrogram = "row",ColSideColors=cols)
# plotly::orca(p, "Streptococcus_thermophilus_SPACERS_plotly.svg")

##-----------plot histogramm

svg("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/Streptococcus_thermophilus_SPACERS_histo.svg",width=5,height=3)

hist(spacer_pairs$id)
          
 dev.off()
 
 ##---------------export the shared spacers
 
sharedspacers <- sterm_curated[which(sterm_curated$id>80),]

unique_spacers <- unique(c(sharedspacers$spacer1,sharedspacers$spacer2))

write.table(unique_spacers, "~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/unique_Sterm_spacers.txt",na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names = FALSE)
```
Based on this analysis (especially the histogramm), we see that we have a very distinct biomodial distribution. By setting a clustering cutoff of 80% we should be able to distinguish common CRISPR spacers and newly aquired

#####make CRISPR clusters plot

now we try to see if there are differences in spacers between the CRISPR spacers of all genomes. 
This is the old way, I used cd-hit-est to see how many clusters I have within a certain similiartity threshold


```{bash}
date=20190312
source /archiv/Projects/2018_Culturomics/02_script/fileShortcuts.sh

##==================
##cd-hit
##==================
  
  #--------------
  ##Streptococcus thermophilus
  #--------------
  

   
   
for species in $(echo "Ldel Sterm")
do
echo ==========================
     echo "working on: " ${species}
     echo ==========================


  
  rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers
  mkdir -p  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}

#sed 's/FAM//g' /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_new_withOLDsamples_newNAMED |sed 's/_0.*__/_/g'  |sed 's/__/_/g' > /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_new_withOLDsamples_newNAMED_short

  
   
  
   cd-hit-est -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/CRISPR_spacer_merged_short.fasta \
    -o /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results.txt -c 0.9 -M 30000 -T 37
  
  
  #grep -c ">" -c /archiv/Projects/2018_Culturomics/06_Spades/S72/PilerCR/${date}/S72.pilarCR_out_final
  #grep -c ">" -c /archiv/Projects/2018_Culturomics/06_Spades/S50/PilerCR/${date}/S50.pilarCR_out_final
  #grep -c ">" -c /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/S_thermophilus_RMK202.pilarCR_out_final


done #species
  

  ###======================
  ##assign representaive Spacers to Cluster
  ###======================
  
  for species in $(echo "Ldel Sterm")
do
echo ==========================
     echo "working on: " ${species}
     echo ==========================

   cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results.txt > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results_ClusterName.txt
   
   for spacersss in $(grep ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results.txt)
   do
   echo "--------------------------"
   #grep "${spacersss}" -A 1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results.txt  > archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}//tmp_spacer.txt
   
   newNAME=$(grep -B 30 "${spacersss}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results.txt.clstr |grep "^>Cluster" |tail -1 |sed 's/ /_/g')
   
   sed -i "s/${spacersss}/${newNAME}/g" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results_ClusterName.txt
   
   done
   
   rm archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}//tmp_spacer.txt
   
done #species
```

#####CRISPR spacers prep
Here, I want ot make an alluvial plot to trace the CRISPR spacers through the samples.
I can directly do this instead of of the last approach
```{bash}

for species in $(echo "Ldel Sterm")
do
echo ==========================
     echo "working on: " ${species}
     echo ==========================
species=Sterm
count=$(grep ">Cluster" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results.txt.clstr)

##----------prime
start_num=0
i=1
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}
 mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}
 
#  grep ">" /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_new_withOLDsamples_newNAMED_short |sed 's/>//g' | cut -d '_' -f 1 |sort| uniq
#rm /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledSpacers/Alluvialplot/
   #echo -e "Clustername\tMAGRMK202\tmst1\tmst2\tF13491\tF13492\tF13493\tF13494\tF13495\tF13496\tF13497\tF13498\tF13499c1\tF13499c2\tF13500" > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledSpacers/CD-hit_new/count//clusterCRISPRs.txt

 echo -e "ClusterName\tnumSpacers\tsample\tarray\tspc" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_Sterm.txt

###-----------------
##extract information
###-----------------
for i in $(seq 1 $count)
  do 
  echo $i
  

  sed -n "/>Cluster $start_num$/,/>Cluster $i$/p" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results.txt.clstr > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}//tmpCRISPRS.txt

ClusterName=$(head -1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}//tmpCRISPRS.txt |sed 's/>//g'|sed 's/ /_/g')

grep -v "^>Clu" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}//tmpCRISPRS.txt > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}//tmpCRISPRS_02.txt

numSpacers=$(wc -l /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}//tmpCRISPRS_02.txt|cut -d ' ' -f 1)

for spacersss in $(awk -F "[>...]" '{print $2}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}//tmpCRISPRS_02.txt)
do

sample=$(echo ${spacersss}| cut -d '_' -f 1)
array=$(echo ${spacersss}| cut -d '_' -f 2|sed 's/a//g' |sed 's/s/_/g' |cut -d '_' -f 1)
spc=$(echo ${spacersss}| cut -d '_' -f 2|sed 's/a//g' |sed 's/s/_/g' |cut -d '_' -f 2)

 echo -e ${ClusterName}"\t"${numSpacers}"\t"${sample}"\t"${array}"\t"${spc} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt


done #spacersss

start_num=$((start_num+1))
done #i

done #species

```


#####Annotate shared CRISPRs 
Here, we count how many and which spacers are in every Cluster from all different samples

If a new sample (whole genome sequence) is added it needs to be added to the list...

This is used for the heatmap of the CRISPR spacer identity.
```{bash}

species=sTERM
count=$(grep ">Cluster" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results.txt.clstr)

grep ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results.txt |cut -d _ -f 1 |sed 's/>//g'|sort|uniq -c|wc -l

##----------prime
start_num=0
i=1
 mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count/
 
  #grep ">" /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_new_withOLDsamples_newNAMED_short |sed 's/>//g' | cut -d '_' -f 1 |sort| uniq

 #  echo -e "Clustername\tMAGRMK202\tmst1\tmst2\tF13491\tF13492\tF13493\tF13494\tF13495\tF13496\tF13497\tF13498\tF13499c1\tF13499c2\tF13500" > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledSpacers/CD-hit_new/count//clusterCRISPRs.txt
# echo -e "Clustername\tSMAG\tS50\tS72\tF13491\tF13492\tF13493\tF13494\tF13495\tF13496\tF13497\tF13498\tF13499c1\tF13499c2\tF13500\t13499\t24737\t24738\t24739\t24740\t24853\t24854\t24855" >  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//clusterCRISPRs.txt
  echo -e "Clustername\tSMAG\tS50\tS72\t13491\t13492\t13493\t13494\t13495\t13496\t13497\t13498\t13499c1\t13499c2\t13500\t13499\t24737\t24738\t24739\t24740\t24853\t24854\t24855" >  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//clusterCRISPRs.txt

  
##---------start analysis 
for i in $(seq 1 $count)
  do 
  echo $i
#  done
#start_num=0
#i=169
  sed -n "/>Cluster $start_num$/,/>Cluster $i$/p" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results.txt.clstr > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt

REF_mst1=$(grep ">S50" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt)  
REF_mst2=$(grep ">S72" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt)  
REF_RMK202=$(grep ">SMAG" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
  
F13491=$(grep ">13491" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F13492=$(grep ">13492" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F13493=$(grep ">13493" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F13494=$(grep ">13494" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F13495=$(grep ">13495" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F13496=$(grep ">13496" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F13497=$(grep ">13497" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F13498=$(grep ">13498" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F13499c1=$(grep ">13499c1" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F13499c2=$(grep ">13499c2" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F13500=$(grep ">13500" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 

F13499_old=$(grep ">13499_" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24737=$(grep ">24737" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24738=$(grep ">24738" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24740=$(grep ">24740" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24739=$(grep ">24739" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24853=$(grep ">24853" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24854=$(grep ">24854" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24855=$(grep ">24855" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 

 echo -e "Cluster_"${start_num}"\t"${REF_RMK202}"\t"${REF_mst1}"\t"${REF_mst2}"\t"${F13491}"\t"${F13492}"\t"${F13493}"\t"${F13494}"\t"${F13495}"\t"${F13496}"\t"${F13497}"\t"${F13498}"\t"${F13499c1}"\t"${F13499c2}"\t"${F13500}"\t"${F13499_old}"\t"${F24737}"\t"${F24738}"\t"${F24739}"\t"${F24740}"\t"${F24853}"\t"${F24854}"\t"${F24855} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count///clusterCRISPRs.txt

  
  start_num=$((start_num+1))
  done
```

LDEL-------------------------------
```{bash}

species=Ldel
count=$(grep ">Cluster" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results.txt.clstr)

grep ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results.txt |cut -d _ -f 1 |sed 's/>//g'|sort|uniq -c|wc -l

##----------prime
start_num=0
i=1
 mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count/
 
  #grep ">" /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_new_withOLDsamples_newNAMED_short |sed 's/>//g' | cut -d '_' -f 1 |sort| uniq

 #  echo -e "Clustername\tMAGRMK202\tmst1\tmst2\tF13491\tF13492\tF13493\tF13494\tF13495\tF13496\tF13497\tF13498\tF13499c1\tF13499c2\tF13500" > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledSpacers/CD-hit_new/count//clusterCRISPRs.txt
# echo -e "Clustername\tSMAG\tS50\tS72\tF13491\tF13492\tF13493\tF13494\tF13495\tF13496\tF13497\tF13498\tF13499c1\tF13499c2\tF13500\t13499\t24737\t24738\t24739\t24740\t24853\t24854\t24855" >  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//clusterCRISPRs.txt
  echo -e "Clustername\tLMAG\t11141\t11142\t11143\t12104\t12105\t12107\t12109\t24776\t24777\t13498\t13499c1\t13499c2\t13500\t13499\t24737\t24738\t24739\t24740\t24853\t24854\t24855" >  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//clusterCRISPRs.txt

  
##---------start analysis 
for i in $(seq 1 $count)
  do 
  echo $i
#  done
#start_num=0
#i=169
  sed -n "/>Cluster $start_num$/,/>Cluster $i$/p" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results.txt.clstr > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt

REF_RMK202=$(grep ">LMAG" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
REF_11141=$(grep ">11141" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt)  
F11142=$(grep ">11142" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F11143=$(grep ">11143" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F12104=$(grep ">12104" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F12105=$(grep ">12105" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F12107=$(grep ">12107" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F12109=$(grep ">12109" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24776=$(grep ">24776" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24777=$(grep ">24777" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24778=$(grep ">24778" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 

F13499c2=$(grep ">13499c2" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F13500=$(grep ">13500" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F13499_old=$(grep ">13499_" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24737=$(grep ">24737" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24738=$(grep ">24738" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24740=$(grep ">24740" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24739=$(grep ">24739" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24853=$(grep ">24853" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24854=$(grep ">24854" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 
F24855=$(grep ">24855" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count//tmpCRISPRS.txt) 

 echo -e "Cluster_"${start_num}"\t"${REF_RMK202}"\t"${REF_mst1}"\t"${REF_mst2}"\t"${F13491}"\t"${F13492}"\t"${F13493}"\t"${F13494}"\t"${F13495}"\t"${F13496}"\t"${F13497}"\t"${F13498}"\t"${F13499c1}"\t"${F13499c2}"\t"${F13500}"\t"${F13499_old}"\t"${F24737}"\t"${F24738}"\t"${F24739}"\t"${F24740}"\t"${F24853}"\t"${F24854}"\t"${F24855} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/count///clusterCRISPRs.txt

  
  start_num=$((start_num+1))
  done
```

move local 
```{bash}
mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/Sterm/count///
scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/Sterm/count///clusterCRISPRs.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/Sterm/count///clusterCRISPRs.txt

```

Now I look at how many SPACERS ARE SHARED amongst strains

```{r}
library(readr)
clusterCRISPRs <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/Sterm/count///clusterCRISPRs.txt", "\t", escape_double = FALSE, trim_ws = TRUE)

# rowss="mst1" 
# Colss="F13493"

pairwiseSharedSpacers <- data.frame()

for (rowss in colnames(clusterCRISPRs)[-1]) {
  for (Colss in colnames(clusterCRISPRs)[-1]) {
  
    tmp <- clusterCRISPRs[,c(rowss,Colss)]
    SpacersTotal <- as.numeric(sum(rowSums(tmp>0) >0))
    SpacersShared <- as.numeric(sum(rowSums(tmp>0) >1))
    
    # tmp_02  <- tmp[which(rowSums(tmp>0)==1),]
    # Ref1_spacersize <- sum(tmp_02[,1]>0&tmp_02[,2]<1)
    # Ref2_spacersize <- sum(tmp_02[,2]>0&tmp_02[,1]<1)
   
    
    ID <- (SpacersShared/ SpacersTotal)*100
    
    # tmp_05 <- data.frame(Ref1=rowss,Ref2=Colss,totalSpacer=SpacersTotal,sharedSpacers=SpacersShared,UniqueREF1=Ref1_spacersize,UniqueREF2=Ref2_spacersize,PercentShared=round(ID,digits=2))
    
    tmp_05 <- data.frame(Ref1=rowss,Ref2=Colss,totalSpacer=SpacersTotal,sharedSpacers=SpacersShared,PercentShared=round(ID,digits=2))
  
  pairwiseSharedSpacers <- rbind(pairwiseSharedSpacers,tmp_05)
    
}
}

write.table(pairwiseSharedSpacers, "~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/Sterm/count///clusterCRISPRs_curated.txt",na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names = TRUE)
```


```{r}
##----------------------
#Ldel
##----------------------


library(readr)
clusterCRISPRs <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/Ldel/count///clusterCRISPRs.txt", "\t", escape_double = FALSE, trim_ws = TRUE)

# rowss="mst1" 
# Colss="F13493"

pairwiseSharedSpacers <- data.frame()

for (rowss in colnames(clusterCRISPRs)[-1]) {
  for (Colss in colnames(clusterCRISPRs)[-1]) {
  
    tmp <- clusterCRISPRs[,c(rowss,Colss)]
    SpacersTotal <- as.numeric(sum(rowSums(tmp>0) >0))
    SpacersShared <- as.numeric(sum(rowSums(tmp>0) >1))
    
    # tmp_02  <- tmp[which(rowSums(tmp>0)==1),]
    # Ref1_spacersize <- sum(tmp_02[,1]>0&tmp_02[,2]<1)
    # Ref2_spacersize <- sum(tmp_02[,2]>0&tmp_02[,1]<1)
   
    
    ID <- (SpacersShared/ SpacersTotal)*100
    
    # tmp_05 <- data.frame(Ref1=rowss,Ref2=Colss,totalSpacer=SpacersTotal,sharedSpacers=SpacersShared,UniqueREF1=Ref1_spacersize,UniqueREF2=Ref2_spacersize,PercentShared=round(ID,digits=2))
    
    tmp_05 <- data.frame(Ref1=rowss,Ref2=Colss,totalSpacer=SpacersTotal,sharedSpacers=SpacersShared,PercentShared=round(ID,digits=2))
  
  pairwiseSharedSpacers <- rbind(pairwiseSharedSpacers,tmp_05)
    
}
}

write.table(pairwiseSharedSpacers, "~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/Ldel/count///clusterCRISPRs_curated.txt",na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names = TRUE)

```

#####MAKE HEATMAPS
here I combine the fastANI heatmap and the CRISPR spacer heatmap to make one big heatmap
```{r}


##--------------------------
 #Sterm
##--------------------------
 
library(readr)
library(plyr)
library(tidyverse)
pairwiseSharedSpacers <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/Sterm/count///clusterCRISPRs_curated.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)
final_ANI <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Sterm/fastaANI_comparision_curated.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)

unique(final_ANI$GenomeA)
unique(pairwiseSharedSpacers$Ref1)

unique(final_ANI$GenomeA) %in% unique(pairwiseSharedSpacers$Ref1)
unique(pairwiseSharedSpacers$Ref1)%in%unique(final_ANI$GenomeA)

# pairwiseSharedSpacers$Ref1 <- revalue(pairwiseSharedSpacers$Ref1, c("SMAG"="Sterm:MAG","S50"="mst1","S72"="mst2","F13491"="FAM13491","F13492"="FAM13492","F13493"="FAM13493","F13494"="FAM13494","F13495"="FAM13495","F13496"="FAM13496","F13497"="FAM13497","F13498"="FAM13498","F13499c1"="FAM13499c1","F13499c2"="FAM13499c2","F13500"="FAM13500"))
# pairwiseSharedSpacers$Ref2 <- revalue(pairwiseSharedSpacers$Ref2, c("SMAG"="Sterm:MAG","S50"="mst1","S72"="mst2","F13491"="FAM13491","F13492"="FAM13492","F13493"="FAM13493","F13494"="FAM13494","F13495"="FAM13495","F13496"="FAM13496","F13497"="FAM13497","F13498"="FAM13498","F13499c1"="FAM13499c1","F13499c2"="FAM13499c2","F13500"="FAM13500"))
# 
# final_ANI$GenomeA <- revalue(final_ANI$GenomeA, c("FAM13491_03292016"="FAM13491","FAM13492_03292016"="FAM13492","FAM13493_03292016"="FAM13493","FAM13494_03292016"="FAM13494","FAM13495_03292016"="FAM13495","FAM13496_03292016"="FAM13496","FAM13497_03292016"="FAM13497","FAM13498_03292016"="FAM13498","FAM13499c1_03292016"="FAM13499c1","FAM13499c2_03292016"="FAM13499c2","FAM13500_03292016"="FAM13500","Sterm_MAG"="Sterm:MAG"))
# 
# final_ANI$GenomeB <- revalue(final_ANI$GenomeB, c("FAM13491_03292016"="FAM13491","FAM13492_03292016"="FAM13492","FAM13493_03292016"="FAM13493","FAM13494_03292016"="FAM13494","FAM13495_03292016"="FAM13495","FAM13496_03292016"="FAM13496","FAM13497_03292016"="FAM13497","FAM13498_03292016"="FAM13498","FAM13499c1_03292016"="FAM13499c1","FAM13499c2_03292016"="FAM13499c2","FAM13500_03292016"="FAM13500","Sterm_MAG"="Sterm:MAG"))
# 
# final_ANI$GenomeA <- final_ANI$GenomeA %>% gsub("_MAG",":MAG",.) %>% gsub("_.*","",.)
# final_ANI$GenomeB <- final_ANI$GenomeB %>% gsub("_MAG",":MAG",.) %>% gsub("_.*","",.)
# 
# pairwiseSharedSpacers$Ref1 <- pairwiseSharedSpacers$Ref1  %>% gsub("F","FAM",.) %>% gsub("MAGRMK202","Sterm:MAG",.)
# pairwiseSharedSpacers$Ref2 <- pairwiseSharedSpacers$Ref2  %>% gsub("F","FAM",.) %>% gsub("MAGRMK202","Sterm:MAG",.)


##===================================================
##dENDROGRAMM
#this is to order them according to the phylogeny
##=================================================== 
##------------
##make matrix
##------------

names<- unique(final_ANI$GenomeA)

heatmap_prep_matrix <- matrix(NA,nrow=length(names),ncol = length(names))
rownames(heatmap_prep_matrix) <- names
colnames(heatmap_prep_matrix) <- names
# heatmap_prep_matrix <- heatmap_prep_matrix[!duplicated(heatmap_prep_matrix)]

final_ANI_02 <- final_ANI[!duplicated(final_ANI[,]), ]

for (rowsss in names){
  for (colsss in names){
  
heatmap_prep_matrix[rowsss,colsss]  <-  final_ANI_02[which(final_ANI_02$GenomeA==rowsss & final_ANI_02$GenomeB==colsss),"ANI"] %>% as.numeric
  
  
  }  
}

##------------
##make dendro of all snps sites
##------------
library(ggdendro)

otter.dendro <- as.dendrogram(hclust(d = dist(x = heatmap_prep_matrix)))

orderSterm <- labels(otter.dendro)

# Create dendro
dendro.plot <- ggdendrogram(data = otter.dendro, rotate = FALSE)
# Preview the plot
print(dendro.plot)

##===================================================
##FASTani heatmap
#this is the upper heatmap
##=================================================== 
final_ANI$GenomeA = factor(final_ANI$GenomeA, levels=orderSterm)
final_ANI$GenomeB = factor(final_ANI$GenomeB, levels=orderSterm)
levels(final_ANI$GenomeA)
levels(final_ANI$GenomeB)
# ggheatmap_ANI <- ggplot(final_ANI, aes(GenomeA, GenomeB, fill = ANI))+
#  geom_tile(color = "white",size=1.1)+
#  scale_fill_gradient2(low = "blue", high = "red", mid = "white",
#   midpoint = 99.5, limit = c(98.9,100), space = "Lab",
#   name="ANI") +
#   scale_y_discrete(position = "right")+
#   labs(legend="ANI",x="",y="")+
#   # scale_fill_distiller(name = "SNPs", palette = "Blues", direction = -1)+
#   # scale_fill_distiller(name = "ANI", palette = "Reds", direction = 1)+
#   theme_minimal()+ # minimal theme
#  theme(axis.text.y = element_text(size = 12),
#       axis.text.x = element_text(angle = 45, vjust = 1, 
#     size = 12, hjust = 1))+
#  coord_fixed()
# # Print the heatmap
# print(ggheatmap_ANI)
# 

  final_ANI$GenomeA = factor(final_ANI$GenomeA, levels=c("13492","13491","24854","24737","13500","13493","13498","SMAG","24853","13494","24855","S72","13499c1","13499","13499c2","24738","24740","S50", "13497", "24739","13495","13496"))
final_ANI$GenomeB = factor(final_ANI$GenomeB, levels=c("13492","13491","24854","24737","13500","13493","13498","SMAG","24853","13494","24855","S72","13499c1","13499","13499c2","24738","24740","S50", "13497", "24739","13495","13496"))


##3-----------------
#try to ouptu only top or bottom
##3-----------------


final_ANI_02 <- final_ANI
numberColumnsss <- 1
for (rowsss in levels(final_ANI_02$GenomeA)[-1]) {
  
  for (numberColumnsss_02 in 1:numberColumnsss) {
    
    Columnss <- levels(final_ANI_02$GenomeA)[numberColumnsss_02]
    
    final_ANI_02 <- final_ANI_02 %>% filter(!(GenomeA==rowsss&GenomeB==Columnss))
    
  }
  
  numberColumnsss=numberColumnsss+1
  
}

# densityANI <- ggplot(final_ANI_02,aes(x=ANI))+geom_density()+theme_classic()+lims(x=c(95,100))

ggheatmap_ANI <- ggplot(final_ANI_02, aes(GenomeA, GenomeB, fill = ANI))+
 geom_tile(color = "white",size=1.1)+
 scale_fill_gradient2(low = "grey", high = "red", midpoint=99,limit = c(99,100), space = "Lab",
  name="ANI") +
  scale_y_discrete(position = "right")+
  labs(legend="ANI",x="",y="")+
  # scale_fill_distiller(name = "SNPs", palette = "Blues", direction = -1)+
  # scale_fill_distiller(name = "ANI", palette = "Reds", direction = 1)+
  theme_minimal()+ # minimal theme
 theme(legend.position = "top",axis.text.y = element_text(size = 12),
      axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1))+
 coord_fixed()
# Print the heatmap
print(ggheatmap_ANI)


final_ANI_03 <- final_ANI_02 %>% filter(GenomeA!=GenomeB)

densityANI <- ggplot(final_ANI_03,aes(x=ANI))+geom_density(color="#EB4D4D",size=1.5)+theme_classic()+lims(x=c(99,100))
densityANI

##===================================================
##CRISPR ID
#this is the LOWER heatmap
##=================================================== 


library(viridis)

pairwiseSharedSpacers_tmp_02 <-  data.frame(Ref1=c(as.character(pairwiseSharedSpacers$Ref2),as.character(pairwiseSharedSpacers$Ref1)),Ref2=c(as.character(pairwiseSharedSpacers$Ref1),as.character(pairwiseSharedSpacers$Ref2)),PercentShared=c(as.numeric(pairwiseSharedSpacers$PercentShared),as.numeric(pairwiseSharedSpacers$PercentShared)))

pairwiseSharedSpacers_tmp_02$Ref1 = factor(pairwiseSharedSpacers_tmp_02$Ref1, levels=orderSterm)
pairwiseSharedSpacers_tmp_02$Ref2 = factor(pairwiseSharedSpacers_tmp_02$Ref2, levels=orderSterm)
levels(final_ANI$GenomeA)
levels(final_ANI$GenomeB)
# pairwiseSharedSpacers_final <- rbind(pairwiseSharedSpacers_tmp_02[,c("Ref2","Ref1","PercentShared")],pairwiseSharedSpacers[,c("Ref1","Ref2","PercentShared")])
# ggheatmap_CRISPR <- ggplot(pairwiseSharedSpacers_tmp_02, aes(Ref2, Ref1, fill = PercentShared))+
#  geom_tile(color = "white",size=1.1)+
#   # theme_classic()+
#  #scale_fill_gradient2(low = "blue", high = "red", mid = "white", 
#   # midpoint = 99.5, limit = c(98.9,100), space = "Lab", 
#   # name="ANI") +
#   # scale_fill_distiller(name = "SNPs", palette = "Blues", direction = -1)+
#   scale_fill_viridis(alpha=0.8)+
#   labs(legend="CRISPR ID",x="",y="")+
#   # scale_fill_distiller(name = "SNPs", palette = "Viridis", direction = -1)+
#   # scale_fill_distiller(name = "ANI", palette = "Reds", direction = 1)+
#   theme_minimal()+ # minimal theme
#  theme(axis.text.y = element_text(size = 12),
#       axis.text.x = element_text(angle = 45, vjust = 1, 
#     size = 12, hjust = 1))+
#  coord_fixed()
# # Print the heatmap
# print(ggheatmap_CRISPR)

pairwiseSharedSpacers_tmp_02$Ref2 = factor(pairwiseSharedSpacers_tmp_02$Ref2, levels=c("13492","13491","24854","24737","13500","13493","13498","SMAG","24853","13494","24855","S72","13499c1","13499","13499c2","24738","24740","S50", "13497", "24739","13495","13496"))
pairwiseSharedSpacers_tmp_02$Ref1 = factor(pairwiseSharedSpacers_tmp_02$Ref1, levels=c("13492","13491","24854","24737","13500","13493","13498","SMAG","24853","13494","24855","S72","13499c1","13499","13499c2","24738","24740","S50", "13497", "24739","13495","13496"))


##3-----------------
#try to ouptu only top or bottom
##3-----------------


pairwiseSharedSpacers_tmp_03 <- pairwiseSharedSpacers_tmp_02
numberColumnsss <- 1
for (rowsss in levels(pairwiseSharedSpacers_tmp_02$Ref1)[-1]) {
  
  for (numberColumnsss_02 in 1:numberColumnsss) {
    
    Columnss <- levels(pairwiseSharedSpacers_tmp_02$Ref1)[numberColumnsss_02]
    
    pairwiseSharedSpacers_tmp_03 <- pairwiseSharedSpacers_tmp_03 %>% filter(!(Ref1==rowsss&Ref2==Columnss))
    
  }
  
  numberColumnsss=numberColumnsss+1
  
}

ggheatmap_CRISPR <- ggplot(pairwiseSharedSpacers_tmp_03, aes(Ref2, Ref1, fill = PercentShared))+
 geom_tile(color = "white",size=1.1)+
  # theme_classic()+
 scale_fill_gradient2(low = "grey", high = "blue",
  midpoint = 25, limit = c(0,100), space = "Lab",
  name="shared CRISPR") +
  # scale_fill_distiller(name = "SNPs", palette = "Blues", direction = -1)+
  # scale_fill_viridis(alpha=0.8)+
  labs(legend="shared CRISPR",x="",y="")+
  # scale_fill_distiller(name = "SNPs", palette = "Viridis", direction = -1)+
  # scale_fill_distiller(name = "ANI", palette = "Reds", direction = 1)+
  theme_minimal()+ # minimal theme
 theme(legend.position = "top",axis.text.y = element_text(size = 12),
      axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1))+
 coord_fixed()
# Print the heatmap
print(ggheatmap_CRISPR)

pairwiseSharedSpacers_tmp_04 <- pairwiseSharedSpacers_tmp_03 %>% filter(Ref1!=Ref2)

densityCRISPR <- ggplot(pairwiseSharedSpacers_tmp_04,aes(x=PercentShared))+geom_density((color="#EB4D4D",size=1.5)+theme_classic()+lims(x=c(0,100))
densityCRISPR



library(patchwork)

##===================================================
##plot
##=================================================== 

  svg("~/Desktop/Projects/2019_RMK202_analysis/plot/HEATMAP_ani_crispr_density.svg",width=20,height=12)

(densityANI+densityCRISPR) / (ggheatmap_ANI+ggheatmap_CRISPR)

 dev.off()



 
  svg("~/Desktop/Projects/2019_RMK202_analysis/plot/HEATMAP_ani_crispr.svg",width=15,height=7.5)

ggheatmap_ANI+ggheatmap_CRISPR+dendro.plot

 dev.off()

 
  svg("~/Desktop/Projects/2019_RMK202_analysis/plot/HEATMAP_ani_crispr_woLegend.svg",width=9,height=7.5)

(ggheatmap_ANI+theme(legend.position = "none"))+(ggheatmap_CRISPR+theme(legend.position = "none"))

 dev.off()
```


```{r}
##--------------------------
 #Ldel
##--------------------------
 
 
library(readr)
library(plyr)
library(tidyverse)
pairwiseSharedSpacers <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/Ldel/count///clusterCRISPRs_curated.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)
final_ANI <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/Ldel/fastaANI_comparision_curated.txt",  "\t", escape_double = FALSE, trim_ws = TRUE)

unique(final_ANI$GenomeA)
unique(pairwiseSharedSpacers$Ref1)

unique(final_ANI$GenomeA) %in% unique(pairwiseSharedSpacers$Ref1)
unique(pairwiseSharedSpacers$Ref1)%in%unique(final_ANI$GenomeA)

# pairwiseSharedSpacers$Ref1 <- revalue(pairwiseSharedSpacers$Ref1, c("SMAG"="Sterm:MAG","S50"="mst1","S72"="mst2","F13491"="FAM13491","F13492"="FAM13492","F13493"="FAM13493","F13494"="FAM13494","F13495"="FAM13495","F13496"="FAM13496","F13497"="FAM13497","F13498"="FAM13498","F13499c1"="FAM13499c1","F13499c2"="FAM13499c2","F13500"="FAM13500"))
# pairwiseSharedSpacers$Ref2 <- revalue(pairwiseSharedSpacers$Ref2, c("SMAG"="Sterm:MAG","S50"="mst1","S72"="mst2","F13491"="FAM13491","F13492"="FAM13492","F13493"="FAM13493","F13494"="FAM13494","F13495"="FAM13495","F13496"="FAM13496","F13497"="FAM13497","F13498"="FAM13498","F13499c1"="FAM13499c1","F13499c2"="FAM13499c2","F13500"="FAM13500"))
# 
# final_ANI$GenomeA <- revalue(final_ANI$GenomeA, c("FAM13491_03292016"="FAM13491","FAM13492_03292016"="FAM13492","FAM13493_03292016"="FAM13493","FAM13494_03292016"="FAM13494","FAM13495_03292016"="FAM13495","FAM13496_03292016"="FAM13496","FAM13497_03292016"="FAM13497","FAM13498_03292016"="FAM13498","FAM13499c1_03292016"="FAM13499c1","FAM13499c2_03292016"="FAM13499c2","FAM13500_03292016"="FAM13500","Sterm_MAG"="Sterm:MAG"))
# 
# final_ANI$GenomeB <- revalue(final_ANI$GenomeB, c("FAM13491_03292016"="FAM13491","FAM13492_03292016"="FAM13492","FAM13493_03292016"="FAM13493","FAM13494_03292016"="FAM13494","FAM13495_03292016"="FAM13495","FAM13496_03292016"="FAM13496","FAM13497_03292016"="FAM13497","FAM13498_03292016"="FAM13498","FAM13499c1_03292016"="FAM13499c1","FAM13499c2_03292016"="FAM13499c2","FAM13500_03292016"="FAM13500","Sterm_MAG"="Sterm:MAG"))
# 
# final_ANI$GenomeA <- final_ANI$GenomeA %>% gsub("_MAG",":MAG",.) %>% gsub("_.*","",.)
# final_ANI$GenomeB <- final_ANI$GenomeB %>% gsub("_MAG",":MAG",.) %>% gsub("_.*","",.)
# 
# pairwiseSharedSpacers$Ref1 <- pairwiseSharedSpacers$Ref1  %>% gsub("F","FAM",.) %>% gsub("MAGRMK202","Sterm:MAG",.)
# pairwiseSharedSpacers$Ref2 <- pairwiseSharedSpacers$Ref2  %>% gsub("F","FAM",.) %>% gsub("MAGRMK202","Sterm:MAG",.)


##===================================================
##dENDROGRAMM
#this is to order them according to the phylogeny
##=================================================== 
##------------
##make matrix
##------------

names<- unique(final_ANI$GenomeA)

heatmap_prep_matrix <- matrix(NA,nrow=length(names),ncol = length(names))
rownames(heatmap_prep_matrix) <- names
colnames(heatmap_prep_matrix) <- names
# heatmap_prep_matrix <- heatmap_prep_matrix[!duplicated(heatmap_prep_matrix)]

final_ANI_02 <- final_ANI[!duplicated(final_ANI[,]), ]

for (rowsss in names){
  for (colsss in names){
  
heatmap_prep_matrix[rowsss,colsss]  <-  final_ANI_02[which(final_ANI_02$GenomeA==rowsss & final_ANI_02$GenomeB==colsss),"ANI"] %>% as.numeric
  
  
  }  
}

##------------
##make dendro of all snps sites
##------------
library(ggdendro)

otter.dendro <- as.dendrogram(hclust(d = dist(x = heatmap_prep_matrix)))

orderSterm <- labels(otter.dendro)

# Create dendro
dendro.plot <- ggdendrogram(data = otter.dendro, rotate = FALSE)
# Preview the plot
print(dendro.plot)

##===================================================
##FASTani heatmap
#this is the upper heatmap
##=================================================== 
final_ANI$GenomeA = factor(final_ANI$GenomeA, levels=orderSterm)
final_ANI$GenomeB = factor(final_ANI$GenomeB, levels=orderSterm)
levels(final_ANI$GenomeA)
levels(final_ANI$GenomeB)
# ggheatmap_ANI <- ggplot(final_ANI, aes(GenomeA, GenomeB, fill = ANI))+
#  geom_tile(color = "white",size=1.1)+
#  scale_fill_gradient2(low = "blue", high = "red", mid = "white",
#   midpoint = 99.5, limit = c(98.9,100), space = "Lab",
#   name="ANI") +
#   scale_y_discrete(position = "right")+
#   labs(legend="ANI",x="",y="")+
#   # scale_fill_distiller(name = "SNPs", palette = "Blues", direction = -1)+
#   # scale_fill_distiller(name = "ANI", palette = "Reds", direction = 1)+
#   theme_minimal()+ # minimal theme
#  theme(axis.text.y = element_text(size = 12),
#       axis.text.x = element_text(angle = 45, vjust = 1, 
#     size = 12, hjust = 1))+
#  coord_fixed()
# # Print the heatmap
# print(ggheatmap_ANI)
# 

  # final_ANI$GenomeA = factor(final_ANI$GenomeA, levels=c("13492","13491","24854","24737","13500","13493","13498","SMAG","24853","13494","24855","S72","13499c1","13499","13499c2","24738","24740","S50", "13497", "24739","13495","13496"))
# final_ANI$GenomeB = factor(final_ANI$GenomeB, levels=c("13492","13491","24854","24737","13500","13493","13498","SMAG","24853","13494","24855","S72","13499c1","13499","13499c2","24738","24740","S50", "13497", "24739","13495","13496"))


##3-----------------
#try to ouptu only top or bottom
##3-----------------


final_ANI_02 <- final_ANI
numberColumnsss <- 1
for (rowsss in levels(final_ANI_02$GenomeA)[-1]) {
  
  for (numberColumnsss_02 in 1:numberColumnsss) {
    
    Columnss <- levels(final_ANI_02$GenomeA)[numberColumnsss_02]
    
    final_ANI_02 <- final_ANI_02 %>% filter(!(GenomeA==rowsss&GenomeB==Columnss))
    
  }
  
  numberColumnsss=numberColumnsss+1
  
}

# densityANI <- ggplot(final_ANI_02,aes(x=ANI))+geom_density()+theme_classic()+lims(x=c(99,100))
# densityANI

ggheatmap_ANI <- ggplot(final_ANI_02, aes(GenomeA, GenomeB, fill = ANI))+
 geom_tile(color = "white",size=1.1)+
 scale_fill_gradient2(low = "grey", high = "red",
  midpoint = 99, limit = c(99,100), space = "Lab",
  name="ANI") +
  scale_y_discrete(position = "right")+
  labs(legend="ANI",x="",y="")+
  # scale_fill_distiller(name = "SNPs", palette = "Blues", direction = -1)+
  # scale_fill_distiller(name = "ANI", palette = "Reds", direction = 1)+
  theme_minimal()+ # minimal theme
 theme(legend.position = "top",axis.text.y = element_text(size = 12),
      axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1))+
 coord_fixed()
# Print the heatmap
print(ggheatmap_ANI)



final_ANI_03 <- final_ANI_02 %>% filter(GenomeA!=GenomeB)

densityANI <- ggplot(final_ANI_03,aes(x=ANI))+geom_density(color="#10B552",size=1.5)+theme_classic()+lims(x=c(99,100))
densityANI


##===================================================
##CRISPR ID
#this is the LOWER heatmap
##=================================================== 


library(viridis)

pairwiseSharedSpacers_tmp_02 <-  data.frame(Ref1=c(as.character(pairwiseSharedSpacers$Ref2),as.character(pairwiseSharedSpacers$Ref1)),Ref2=c(as.character(pairwiseSharedSpacers$Ref1),as.character(pairwiseSharedSpacers$Ref2)),PercentShared=c(as.numeric(pairwiseSharedSpacers$PercentShared),as.numeric(pairwiseSharedSpacers$PercentShared)))

pairwiseSharedSpacers_tmp_02$Ref1 = factor(pairwiseSharedSpacers_tmp_02$Ref1, levels=orderSterm)
pairwiseSharedSpacers_tmp_02$Ref2 = factor(pairwiseSharedSpacers_tmp_02$Ref2, levels=orderSterm)
levels(final_ANI$GenomeA)
levels(final_ANI$GenomeB)



##3-----------------
#try to ouptu only top or bottom
##3-----------------


pairwiseSharedSpacers_tmp_03 <- pairwiseSharedSpacers_tmp_02
numberColumnsss <- 1
for (rowsss in levels(pairwiseSharedSpacers_tmp_02$Ref1)[-1]) {
  
  for (numberColumnsss_02 in 1:numberColumnsss) {
    
    Columnss <- levels(pairwiseSharedSpacers_tmp_02$Ref1)[numberColumnsss_02]
    
    pairwiseSharedSpacers_tmp_03 <- pairwiseSharedSpacers_tmp_03 %>% filter(!(Ref1==rowsss&Ref2==Columnss))
    
  }
  
  numberColumnsss=numberColumnsss+1
  
}
# pairwiseSharedSpacers_final <- rbind(pairwiseSharedSpacers_tmp_02[,c("Ref2","Ref1","PercentShared")],pairwiseSharedSpacers[,c("Ref1","Ref2","PercentShared")])
# ggheatmap_CRISPR <- ggplot(pairwiseSharedSpacers_tmp_02, aes(Ref2, Ref1, fill = PercentShared))+
#  geom_tile(color = "white",size=1.1)+
#   # theme_classic()+
#  #scale_fill_gradient2(low = "blue", high = "red", mid = "white", 
#   # midpoint = 99.5, limit = c(98.9,100), space = "Lab", 
#   # name="ANI") +
#   # scale_fill_distiller(name = "SNPs", palette = "Blues", direction = -1)+
#   scale_fill_viridis(alpha=0.8)+
#   labs(legend="CRISPR ID",x="",y="")+
#   # scale_fill_distiller(name = "SNPs", palette = "Viridis", direction = -1)+
#   # scale_fill_distiller(name = "ANI", palette = "Reds", direction = 1)+
#   theme_minimal()+ # minimal theme
#  theme(axis.text.y = element_text(size = 12),
#       axis.text.x = element_text(angle = 45, vjust = 1, 
#     size = 12, hjust = 1))+
#  coord_fixed()
# # Print the heatmap
# print(ggheatmap_CRISPR)

# pairwiseSharedSpacers_tmp_02$Ref2 = factor(pairwiseSharedSpacers_tmp_02$Ref2, levels=c("13492","13491","24854","24737","13500","13493","13498","SMAG","24853","13494","24855","S72","13499c1","13499","13499c2","24738","24740","S50", "13497", "24739","13495","13496"))
# pairwiseSharedSpacers_tmp_02$Ref1 = factor(pairwiseSharedSpacers_tmp_02$Ref1, levels=c("13492","13491","24854","24737","13500","13493","13498","SMAG","24853","13494","24855","S72","13499c1","13499","13499c2","24738","24740","S50", "13497", "24739","13495","13496"))

##3-----------------
#plot
##3-----------------


ggheatmap_CRISPR <- ggplot(pairwiseSharedSpacers_tmp_03, aes(Ref2, Ref1, fill = PercentShared))+
 geom_tile(color = "white",size=1.1)+
  # theme_classic()+
 scale_fill_gradient2(low = "grey", high = "blue",
   limit = c(0,100), space = "Lab",
  name="shared CRISPR") +
  # scale_fill_distiller(name = "SNPs", palette = "Blues", direction = -1)+
  # scale_fill_viridis(alpha=0.8)+
  labs(legend="shared CRISPR",x="",y="")+
  # scale_fill_distiller(name = "SNPs", palette = "Viridis", direction = -1)+
  # scale_fill_distiller(name = "ANI", palette = "Reds", direction = 1)+
  theme_minimal()+ # minimal theme
 theme(legend.position = "top",axis.text.y = element_text(size = 12),
      axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1))+
 coord_fixed()
# Print the heatmap
print(ggheatmap_CRISPR)


pairwiseSharedSpacers_tmp_04 <- pairwiseSharedSpacers_tmp_03 %>% filter(Ref1!=Ref2)

densityCRISPR <- ggplot(pairwiseSharedSpacers_tmp_04,aes(x=PercentShared))+geom_density(color="#10B552",size=1.5)+theme_classic()+lims(x=c(0,100))
densityCRISPR



library(patchwork)


##===================================================
##plot
##=================================================== 

  svg("~/Desktop/Projects/2019_RMK202_analysis/plot/HEATMAP_ani_crispr_density_ldel.svg",width=20,height=12)

(densityANI+densityCRISPR) / (ggheatmap_ANI+ggheatmap_CRISPR)

 dev.off()



 
  svg("~/Desktop/Projects/2019_RMK202_analysis/plot/HEATMAP_ani_crispr_ldel.svg",width=15,height=7.5)

ggheatmap_ANI+ggheatmap_CRISPR+dendro.plot

 dev.off()

 
  svg("~/Desktop/Projects/2019_RMK202_analysis/plot/HEATMAP_ani_crispr_woLegend_ldel.svg",width=9,height=7.5)

(ggheatmap_ANI+theme(legend.position = "none"))+(ggheatmap_CRISPR+theme(legend.position = "none"))

 dev.off()
```

calculate average ANI and CRISPR within the different groups.

```{r}

##==============
##groups
##==============
group1 <- c("13492","13491","24854","24737","13500","13493","13498","SMAG","24853")
group2 <- c("13494","24855","S72")
group3 <- c("13499c1","13499","13499c2","24738","24740","S50")
group4 <- c("13497", "24739","13495","13496")


##==============
##CRISPRid
##==============

pairwiseSharedSpacers_tmp_02 %>% filter(Ref1 %in% group1 & Ref2 %in% group1) %>% filter(Ref1!=Ref2) %>%  summarise(mean = mean(PercentShared))
pairwiseSharedSpacers_tmp_02 %>% filter(Ref1 %in% group2 & Ref2 %in% group2) %>% filter(Ref1!=Ref2) %>%  summarise(mean = mean(PercentShared))
pairwiseSharedSpacers_tmp_02 %>% filter(Ref1 %in% group3 & Ref2 %in% group3) %>% filter(Ref1!=Ref2) %>%  summarise(mean = mean(PercentShared))
pairwiseSharedSpacers_tmp_02 %>% filter(Ref1 %in% group4 & Ref2 %in% group4) %>% filter(Ref1!=Ref2) %>%  summarise(mean = mean(PercentShared))

##==============
##ANI
##==============
final_ANI %>% filter(GenomeA %in% group1 & GenomeB %in% group1) %>% filter(GenomeA!=GenomeB) %>%  summarise(mean = round(mean(ANI),digits = 6)) 
final_ANI %>% filter(GenomeA %in% group2 & GenomeB %in% group2) %>% filter(GenomeA!=GenomeB) %>%  summarise(mean = mean(ANI))
final_ANI %>% filter(GenomeA %in% group3 & GenomeB %in% group3) %>% filter(GenomeA!=GenomeB) %>%  summarise(mean = mean(ANI))
final_ANI %>% filter(GenomeA %in% group4 & GenomeB %in% group4) %>% filter(GenomeA!=GenomeB) %>%  summarise(mean = mean(ANI))

```
make a table of this with the [latex maker](https://www.tablesgenerator.com/latex_tables#) and then convert it with overleaf. 

#####reoccuring Spacers
Here, we check which spacers are common. Then we check who shares them (same genotype?) and where they occur (typology?)

```{r}
library(readr)
# clusterCRISPRs <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/clusterCRISPRs_sterm_new.txt", "\t", escape_double = FALSE, trim_ws = TRUE)
clusterCRISPRs <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/Sterm/count///clusterCRISPRs.txt", "\t", escape_double = FALSE, trim_ws = TRUE)
length(names) #from previous chunk
occurence <- data.frame(ClusterName=clusterCRISPRs$Clustername,occurrencecounts=rowSums(clusterCRISPRs[,-1]))
# occurence <- data.frame(ClusterName=clusterCRISPRs$Clustername,occurrencecounts=rowSums(clusterCRISPRs[,-1]>0))

table(occurence$occurrencecounts)
table(occurence$occurrencecounts)[-1] %>% sum()
table(occurence$occurrencecounts) %>% sum()

plotspacerCOUNT <- ggplot(occurence,aes(x=occurrencecounts))+geom_bar()+theme_classic()+labs(x="number of genomes containing the same spacer",y="spacer count")+scale_x_continuous(breaks =c(1:length(names)),labels=c(1:length(names)),limits = c(0,length(names)))+scale_y_continuous(breaks =c(0,7,31,60,154),labels=c(0,7,31,60,154))
plotspacerCOUNT
svg("~/Desktop/Projects/2019_RMK202_analysis/plot/spacerUniquenessCount.svg",width=4.5,height=2.5)
plotspacerCOUNT
 dev.off()
```

#####Spacers analysis

Here, I do various spacer analysis with the data I gathered in the previous chunks.
I do:
1. 


```{r}
library(readr)
library(plyr)
library(tidyverse)
library(ggplot2)
library(patchwork)
library(alluvial)
library(ggalluvial)
#spacer_Infos_Sterm_final <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/spacer_Infos_Sterm_final.txt",  "\t", escape_double = FALSE, col_names = c("ClusterName","numSpacers_in_CLUSTER","sample","array","spc","ARRAY","SPACER"),  col_types = cols(SPACER = col_number()),  trim_ws = TRUE)

spacer_Infos_Sterm_final <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/Sterm/spacer_Infos_Sterm_final.txt",  "\t", escape_double = FALSE, col_names = c("ClusterName","numSpacers_in_CLUSTER","sample","array","spc","ARRAY","SPACER"),  col_types = cols(SPACER = col_number()),  trim_ws = TRUE)

##------------------------------
##clean
##------------------------------
spacer_Infos_Sterm_final$spacerUniqueness  <- ifelse(spacer_Infos_Sterm_final$numSpacers_in_CLUSTER<2,"unique","shared")
table(spacer_Infos_Sterm_final$spacerUniqueness)
spacer_Infos_Sterm_final$ARRAY <- revalue(spacer_Infos_Sterm_final$ARRAY, c("a2"="CRISPR array 2","a1"="CRISPR array 1","a3"="CRISPR array 3"))


# spacer_Infos_Sterm_final$sample = factor(spacer_Infos_Sterm_final$sample, levels=orderSterm)
##------------------------------
###How many spacers in each strain
##------------------------------
spacerCount1 <- ggplot(spacer_Infos_Sterm_final,aes(x=sample,group=spacerUniqueness,fill=spacerUniqueness))+geom_bar()+theme_classic()+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9))+labs(x="",y="spacer count",fill="Uniqueness")

spacerCount2 <- ggplot(spacer_Infos_Sterm_final,aes(x=sample,group=spacerUniqueness,fill=spacerUniqueness))+geom_bar()+theme_classic()+facet_wrap(~ARRAY)+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),legend.position = "none")+labs(x="",y="spacer count",fill="Uniqueness")

# svg("~/Desktop/Projects/2019_RMK202_analysis/plot/CRISPR_spacer_count.svg",width=5.5,height=5.5)
((spacerCount1+plot_spacer() +plot_layout(widths = c(2,1.5)))/spacerCount2+plot_layout(nrow = 2))
 # dev.off()

table(spacer_Infos_Sterm_final$ARRAY,spacer_Infos_Sterm_final$sample)
range(table(spacer_Infos_Sterm_final$sample))

spacer_Infos_Sterm_final$sample = factor(spacer_Infos_Sterm_final$sample, levels=c("13492","13491","24854","24737","13500","13493","13498","SMAG","24853","13494","24855","S72","13499c1","13499","13499c2","24738","24740","S50", "13497", "24739","13495","13496"))
colors <- c("#fd8d3cff","#807dbaff","#ef3b2cff")

spacerCount3 <- ggplot(spacer_Infos_Sterm_final,aes(x=sample,group=ARRAY,fill=ARRAY))+geom_bar()+theme_classic()+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9))+labs(x="",y="spacer count",fill="")+scale_fill_manual(values=colors)
spacerCount3
##------------------------------
###spacer location
##------------------------------ 
 

max(uniqueSpacersss$SPACER)

uniqueSpacersss <- spacer_Infos_Sterm_final[which(spacer_Infos_Sterm_final$numSpacers_in_CLUSTER==1),]
range(spacer_Infos_Sterm_final$SPACER)
pfirst <- ggplot(uniqueSpacersss,aes(x=SPACER,group=ARRAY,fill=ARRAY))+geom_bar()+theme_classic()+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7))+scale_fill_manual(values=colorsss)+labs(x="spacer location",y="number of unique spacers",fill="Uniqueness")+scale_x_continuous(breaks =c(1:max(uniqueSpacersss$SPACER)),labels=c(1:max(uniqueSpacersss$SPACER)))

plocation <- (pfirst+plot_spacer()+plot_layout(widths = c(2,0.5)))


# svg("~/Desktop/Projects/2019_RMK202_analysis/plot/CRISPR_spacer_count_location.svg",width=6.5,height=8.5)
((spacerCount1+plot_spacer() +plot_layout(widths = c(2,1.5)))/spacerCount2+plocation+plot_layout(nrow = 3))
 # dev.off()

##------------------------------
###number of uniq spacers
##------------------------------ 
colorsss <- rev(c("darkcyan","darkturquoise","red"))
 
spacer_Infos_Sterm_final$ARRAY
uniqueSpacersss_tmp1 <- spacer_Infos_Sterm_final[which(spacer_Infos_Sterm_final$numSpacers_in_CLUSTER==1),] %>% select(ClusterName,ARRAY) %>% unique() %>% add_column(uniqueness="unique") 

uniqueSpacersss_tmp2 <- spacer_Infos_Sterm_final[which(spacer_Infos_Sterm_final$numSpacers_in_CLUSTER>1),] %>% select(ClusterName,ARRAY) %>% unique()%>% add_column(uniqueness="total")
uniqueSpacersss <- rbind(uniqueSpacersss_tmp1,uniqueSpacersss_tmp2)
uniqueSpacersss$ARRAY
# range(spacer_Infos_Sterm_final$SPACER)
# uniqueSpacersss$ARRAY <- factor(uniqueSpacersss$ARRAY)
uniqueSpacersss$ARRAY <- revalue(uniqueSpacersss$ARRAY, c("a2"="CRISPR array 2","a1"="CRISPR array 1","a3"="CRISPR array 3"))

colorsss <- c("lightgray","lightgray","lightgray","#fd8d3c","#ef3b2c")
 uniqueSpac <- ggplot(uniqueSpacersss,aes(x=ARRAY,fill=interaction(ARRAY,uniqueness),color=interaction(ARRAY,uniqueness)))+geom_bar()+theme_classic()+ scale_x_discrete(drop=FALSE)+theme(axis.text.x = element_text(angle = 45, hjust = 1,size=7),legend.position = "right")+labs(x="",y="number of spacers")+scale_fill_manual(values=colorsss)+scale_color_manual(values=colorsss)+scale_y_continuous(breaks =c(0,8,24,29,60,86),labels=c(0,8,24,29,60,86))

colorsss <- c("lightgray","darkgray")
 uniqueSpac <- ggplot(uniqueSpacersss,aes(x=ARRAY,fill=interaction(uniqueness),color=interaction(uniqueness)))+geom_bar()+theme_classic()+ scale_x_discrete(drop=FALSE)+theme(axis.text.x = element_text(angle = 45, hjust = 1,size=7),legend.position = "right",legend.title = element_blank())+labs(x="",y="number of spacers")+scale_fill_manual(values=colorsss)+scale_color_manual(values=colorsss)#+scale_y_continuous(breaks =c(0,8,24,29,60,86),labels=c(0,8,24,29,60,86))
 uniqueSpac
 
table(uniqueSpacersss$uniqueness)
table(uniqueSpacersss$uniqueness,uniqueSpacersss$ARRAY)
uniqueSpac
svg("~/Desktop/Projects/2019_RMK202_analysis/plot/UniqueSpacers.svg",width=2.5,height=3)
uniqueSpac
 dev.off()


##------------------------------
###spacer location relative distance
##------------------------------ 
orientation_CRISPRarrays <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/Sterm/allGenomes/orientation_CRISPRarrays.txt", "\t", escape_double = FALSE, col_names = c("sample","orientation","ARRAY"),    trim_ws = TRUE)
 
 
uniqueSpacersss_again <- spacer_Infos_Sterm_final %>% filter(spacerUniqueness=="unique")
 
TotSpacer <- spacer_Infos_Sterm_final %>% group_by(sample,ARRAY) %>%  summarize(totalSpacersARRAY=max(SPACER))
uniqueSpacersss_extended_tmp001 <-  merge(uniqueSpacersss_again, TotSpacer, by=c("sample", "ARRAY"))  %>%filter((sample!="24853")) # filter(sample!=24853&ARRAY!="CRISPR array 2")

uniqueSpacersss_extended_tmp002 <-  merge(uniqueSpacersss_again, TotSpacer, by=c("sample", "ARRAY"))  %>%filter((ARRAY=="CRISPR array 2")) # filter(sample!=24853&ARRAY!="CRISPR array 2")
uniqueSpacersss_extended_tmp02 <- rbind(uniqueSpacersss_extended_tmp001,uniqueSpacersss_extended_tmp002)

uniqueSpacersss_extended_tmp2 <-  merge(uniqueSpacersss_extended_tmp02, orientation_CRISPRarrays, by=c("sample", "ARRAY")) %>% mutate(relDistanceLeader = ifelse(orientation=="For", SPACER/totalSpacersARRAY,1-(SPACER/totalSpacersARRAY)))

uniqueSpacersss_extended <- uniqueSpacersss_extended_tmp2 %>% mutate(relDistanceLeader = ifelse(ARRAY=="CRISPR array 3", relDistanceLeader,1-(relDistanceLeader)))



# 

colors <- c("#fd8d3cff","#807dbaff","#ef3b2cff")
spacerRelDistance <- ggplot(uniqueSpacersss_extended,aes(x=relDistanceLeader,group=ARRAY,fill=ARRAY))+geom_density(alpha=0.3)+theme_classic()+labs(x="relative proximity to CRISPR leader")+theme(legend.title = element_blank(),legend.position = "right",axis.ticks.y=element_blank(),axis.text.y=element_blank())+scale_fill_manual(values = colors)
# 
# # ggplot(uniqueSpacersss_extended,aes(x=relDistanceLeader,group=ARRAY,color=ARRAY,fill=ARRAY))+geom_histogram(alpha=0.3)+theme_classic()+labs(x="relative proximity to CRISPR leader")
spacerRelDistance
svg("~/Desktop/Projects/2019_RMK202_analysis/plot/CRISPR_spacer_count_location_rel.svg",width=4.5,height=2.5)
spacerRelDistance
 dev.off()
 
 
##----------------------------------------------
#number of spacers
##----------------------------------------------



 library(readr)
spacer_Infos_Sterm_final_sub <- spacer_Infos_Sterm_final %>% select(ClusterName,ARRAY) 
spacer_Infos_Sterm_final_sub_uniques <- spacer_Infos_Sterm_final_sub[!duplicated(spacer_Infos_Sterm_final_sub[,]), ]
namesSingleArrays <- which(table(spacer_Infos_Sterm_final_sub_uniques$ClusterName)==1) %>% names()
tmp1 <- spacer_Infos_Sterm_final_sub_uniques %>% filter(ClusterName %in% namesSingleArrays)
# namesMultipleArrays <- which(table(spacer_Infos_Sterm_final_sub_uniques$ClusterName)==2) %>% names()
# tmp2 <- data.frame(ClusterName=namesMultipleArrays,ARRAY="multiple Arrays")
# Arrayasignment <- rbind(tmp1,tmp2)
  Arrayasignment <- tmp1
# clusterCRISPRs <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/clusterCRISPRs_sterm_new.txt", "\t", escape_double = FALSE, trim_ws = TRUE)
clusterCRISPRs <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/Sterm/count///clusterCRISPRs.txt", "\t", escape_double = FALSE, trim_ws = TRUE)
length(names) #from previous chunk
occurence <- data.frame(ClusterName=clusterCRISPRs$Clustername,occurrencecounts=rowSums(clusterCRISPRs[,-1]))
# occurence <- data.frame(ClusterName=clusterCRISPRs$Clustername,occurrencecounts=rowSums(clusterCRISPRs[,-1]>0))
  occurence_final <- merge(occurence,Arrayasignment,by="ClusterName")

colors <- c("#fd8d3cff","#807dbaff","#ef3b2cff","grey")

table(occurence$occurrencecounts)
plotspacerCOUNT <- ggplot(occurence_final,aes(x=occurrencecounts,fill=ARRAY))+geom_bar(alpha=0.8)+theme_classic()+labs(x="number of genomes containing the same spacer",y="spacer count") #+scale_x_continuous(breaks =c(1:length(names)),labels=c(1:length(names)),limits = c(0,length(names)))+scale_fill_manual(values = colors)+theme(legend.title = element_blank())

# +scale_y_continuous(breaks =c(0,7,31,60,154),labels=c(0,7,31,60,154))
plotspacerCOUNT
# svg("~/Desktop/Projects/2019_RMK202_analysis/plot/spacerUniquenessCount.svg",width=4.5,height=2.5)
svg("~/Desktop/Projects/2019_RMK202_analysis/plot/spacerUniquenessCount.svg",width=6,height=2.5)

plotspacerCOUNT
 dev.off()
 
 
 svg("~/Desktop/Projects/2019_RMK202_analysis/plot/spacerUniquenessCount_distance.svg",width=13,height=3)

plotspacerCOUNT+spacerRelDistance
 dev.off()

 
  svg("~/Desktop/Projects/2019_RMK202_analysis/plot/StrainSpacercount.svg",width=8,height=3.5)

spacerCount3
 dev.off()
```




##Metagenome CRISPR spacers
The way I try to identify CRISPR spacers in teh metagenome is by mapping the CRISPR repeats to the raw metagenomic reads and clipping of teh repeats to only keep the spacers. 


leave genomic spacers away and check in the end if the undemultplexed spacers occur int the genomes with cd-hit.


!!!!Important I did not have to reverse complement the reverse reads. 
```{bash}
#---test
adap=GTTTTAGAGCTGTGTTGTTTCGAATGGTTCCAAAAC
reverse=GTTTTGGAACCATTCGAAACAACACAGCTCTAAAAC

cat /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt

head -1000000 /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/RMK202/RMK202/RMK202_R1_kneaddata_paired_1.fastq |grep -c ${adap} 
grep -c ${reverse} /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/RMK202/RMK202/RMK202_R1_kneaddata_paired_1.fastq

echo $adap
echo $reverse


#manually extract the CRISPR repeats
mkdir -p /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/
#cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/cdHIT/CRISPR_Clustering.txt > /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt

###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples_withEvolution.txt  ##the file with all sample names
#BaseLocation=/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/
BaseLocation=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract/
species=Sterm
#BaseLocation_bam=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP
#BaseLocation_bam=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP
repeatsFile=/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt

#revseq /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut_reversed.txt

###================
##script
###================
##test
pair=1
names=G4_6_18
adapName=$(grep ">" ${repeatsFile} |head -1)
###----end

for names in $(cat ${samplesss} |grep "^G")
  do
 
for pair in $(echo "1 2")
  do
#pair=1

read=/archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_${pair}.fastq


for adapName in $(grep ">" ${repeatsFile} )
  do
adapSeq=$(grep "${adapName}" -A 1 ${repeatsFile}|tail -1)
adap=$(echo ${adapName} | sed 's/>//g'|cut -d '_' -f 1)
cleanRound=1

echo "------------------------------------------------"
echo $adap
echo $adapSeq
echo $names
echo $read
echo $species
echo "------------------------------------------------"

mkdir -p ${BaseLocation}/${names}/ReadPair_${pair}//tmp/

cutadapt  ${read} -b ${adapSeq}  -e  0.1 -O 15 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}.txt
rm ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}.txt &

## reads with match
awk '{if($2!="-1") print $0}' ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt

rm ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt & 

##make a list of current (potentially still containing 5' adapters)
awk -F "\t" '{OFS="\n"} {print "@"$1, $5,"+", $9}' ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt  > \
${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp.fastq


##trim 5' ends of these (with relaxed matching errors)
cutadapt ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp.fastq -g ${adapSeq}  -e  0.1 -O 5 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_.fastq.txt

rm ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_.fastq.txt &

## finalise first spacers (we will not be able to keep reads that don't have a small (>5bp) of repeat left)
##removing potential 5' adapters of first trim
mkdir -p ${BaseLocation}/${names}/final/
#awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($7)>"25" && length($7)<"46") print ">"$1, $7}' #${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_info.txt > \
#${BaseLocation}/${names}/ReadPair_${pair}/${names}_${adap}_final_round1_R${pair}.fasta


awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($7)>"25"  && length($7)<"46" && length($7)==length($11)) print "@"$1, $7 ,"+", $11}' ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}/${names}_${adap}_final_round1_R${pair}.fastq

cat ${BaseLocation}/${names}/ReadPair_${pair}/${names}_${adap}_final_round1_R${pair}.fastq > ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R${pair}.fastq


##take remaining reads and remove 3' adapters again --> do this 5 times
awk -F "\t" '{OFS="\n"} {print "@"$1, $7,"+", $11}' ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt  > \
${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round2_matchestmp_timmed5_readyForRound2.fastq

#grep --color "GTTTTAGAGCTGTGTTGTTTCGAATGGTTCCAAAAC" /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/readswithrepeats_${adap}_intraReads_info_trimmed_matches_readyForRound2.fastq  |head
cleanRound=2
for cleanRound in $(echo "2 3 4 5 6")
  do
  echo ${cleanRound}
  echo "------------------------------------"
cleanRound_next=$((cleanRound+1))
  
cutadapt ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_readyForRound${cleanRound}.fastq -b ${adapSeq}  -e  0.1 -O 5 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.fasta
rm ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.fasta &
##finalised spacers
awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($5)>"25"  && length($5)<"46"&& length($5)==length($9)) print "@"$1, $5 ,"+", $9}' ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}/${names}_${adap}_final_round${cleanRound}_R${pair}.fastq


cat ${BaseLocation}/${names}/ReadPair_${pair}/${names}_${adap}_final_round${cleanRound}_R${pair}.fastq >> ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R${pair}.fastq


##finalised prepered
awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($7)>"25" && length($7)<"46" && length($7)==length($11)) print "@"$1, $7,"+", $11}' ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound_next}_matchestmp_timmed5_readyForRound${cleanRound_next}.fastq

  done #cleanRound
  cat ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R1.fastq \
  ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R2.fastq > \
  ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads.fastq
  done #adap (spacer)
  
  ###-------------====
  ##reverse
    ###-------------====
#adapName=$(grep ">" /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut_reversed.txt| cut -d ' ' -f 1 |head -1)
for adapName in $(grep ">" ${repeatsFile}| cut -d ' ' -f 1)
  do
adapSeq=$(grep "${adapName}" -A 1 ${repeatsFile}|tail -1)
adap=$(echo ${adapName} | sed 's/>//g'|cut -d '_' -f 1)
cleanRound=1

echo "------------------------------------------------"
echo $adap
echo $adapSeq
echo $names
echo $read
echo $species
echo "------------------------------------------------"

mkdir -p ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/

cutadapt ${read} -b ${adapSeq}  -e  0.1 -O 15 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}.txt
rm ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}.txt &

## reads with match
awk '{if($2!="-1") print $0}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt

rm ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt & 

##make a list of current (potentially still containing 5' adapters)
awk -F "\t" '{OFS="\n"} {print "@"$1, $5,"+", $9}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt  > \
${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp.fastq


##trim 5' ends of these (with relaxed matching errors)
cutadapt ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp.fastq -g ${adapSeq}  -e  0.1 -O 5 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_.fastq.txt

rm ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_.fastq.txt &

## finalise first spacers (we will not be able to keep reads that don't have a small (>5bp) of repeat left)
##removing potential 5' adapters of first trim
mkdir -p ${BaseLocation}/${names}/final/
awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($7)>"25" && length($7)<"46" && length($7)==length($11)) print "@"$1, $7 ,"+", $11}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_info.txt > \
${BaseLocation}/${names}/ReadPair_${pair}_reversed/${names}_${adap}_final_round1_R${pair}_reversed.fastq

cat ${BaseLocation}/${names}/ReadPair_${pair}_reversed/${names}_${adap}_final_round1_R${pair}_reversed.fastq > ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R${pair}_reversed.fastq


##take remaining reads and remove 3' adapters again --> do this 5 times
awk -F "\t" '{OFS="\n"} {print "@"$1, $7,"+", $11}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt  > \
${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round2_matchestmp_timmed5_readyForRound2.fastq

#grep --color "GTTTTAGAGCTGTGTTGTTTCGAATGGTTCCAAAAC" /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/readswithrepeats_${adap}_intraReads_info_trimmed_matches_readyForRound2.fastq  |head
cleanRound=2
for cleanRound in $(echo "2 3 4 5 6")
  do
  echo ${cleanRound}
  echo "------------------------------------"
cleanRound_next=$((cleanRound+1))
  
cutadapt ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_readyForRound${cleanRound}.fastq -b ${adapSeq}  -e  0.1 -O 5 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.fasta
rm ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.fasta &
##finalised spacers
awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($5)>"25" && length($5)<"46" && length($5)==length($9)) print "@"$1, $5 ,"+", $9}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}_reversed/${names}_${adap}_final_round${cleanRound}_R${pair}_reversed.fastq

cat ${BaseLocation}/${names}/ReadPair_${pair}_reversed/${names}_${adap}_final_round${cleanRound}_R${pair}_reversed.fastq >> ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R${pair}_reversed.fastq


##finalised prepered
awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($7)>"25" && length($7)<"46" && length($7)==length($11)) print "@"$1, $7,"+", $11}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound_next}_matchestmp_timmed5_readyForRound${cleanRound_next}.fastq



  done #cleanRound
  cat ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R1_reversed.fastq \
  ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R2_reversed.fastq > \
  ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed.fastq
  #revseq ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed.fastq ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular.fastq
  seqtk seq -r  ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed.fastq > ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular.fastq

  done #adap (spacer)
  
  done #pair (readPair)
  done #names (sampleName)

##==============================================
##-----------------clean up array labels
##==============================================

repeatsFile_old=/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut_old.txt
for names in $(cat ${samplesss} |grep "^G")
  do
for adapName in $(grep ">" ${repeatsFile_old}  )
  do
  
Arraynum=$(echo ${adapName} |tail -c2)
adap=$(grep "${Arraynum}" ${repeatsFile_old} |sed 's/>//g')
#cleanRound=1
adapSeqNew=$(grep "${Arraynum}" ${repeatsFile} |sed 's/>//g')

#echo -e ${BaseLocation}/${names}"/final/"${names}_${adap}"_final_allRounds_allReads.fastq"
#echo -e ${BaseLocation}/${names}"/final/"${names}_${adapSeqNew}"_final_allRounds_allReads.fastq"

cp ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads.fastq ${BaseLocation}/${names}/final/${names}_${adapSeqNew}_final_allRounds_allReads.fastq
cp ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed.fastq ${BaseLocation}/${names}/final/${names}_${adapSeqNew}_final_allRounds_allReads_reversed.fastq


done

done




##==============================================
##-----------------merge all samples spacers
##==============================================
rm ${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_${adap}.fastq
rm -r ${BaseLocation}/FinalSpacers/ForDADA2/
rm -r ${BaseLocation}/FinalSpacers/ForDADA2_woREverse
mkdir -p ${BaseLocation}/FinalSpacers/ForDADA2_woREverse
mkdir -p ${BaseLocation}/FinalSpacers/ForDADA2/
for names in $(cat ${samplesss} )
  do
for adapName in $(grep ">" ${repeatsFile}  )
  do
adapSeq=$(grep "${adapName}" -A 1 ${repeatsFile})
adap=$(echo ${adapName} | sed 's/>//g'|cut -d '_' -f 1)
cleanRound=1

seqtk seq -r  ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed.fastq > ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular.fastq


#-----------change name



#awk -v sampless="$names" -v arrayss="$adapShor"  '{if(NR%4==1) $0=sprintf("@"sampless":"arrayss":s"++k,(1+i++)); print;}' ${BaseLocation}/FinalSpacers/ForDADA2/di_K2_6h_allSpacermerged_Array1.fastq


adapShor=$(echo $adap |sed 's/Array/a/g')
awk -v sampless="$names" -v arrayss="$adapShor"  '{if(NR%4==1) $0=sprintf("@"sampless":"arrayss":s"++k,(1+i++)); print;}' ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads.fastq > ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_renamed.fastq

adapShor=$(echo $adap |sed 's/Array/a/g')
awk -v sampless="$names" -v arrayss="$adapShor"  '{if(NR%4==1) $0=sprintf("@"sampless":R"arrayss":s"++k,(1+i++)); print;}' ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular.fastq > ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular_renamed.fastq


###----------------------without Reversing
adapShor=$(echo $adap |sed 's/Array/a/g')
awk -v sampless="$names" -v arrayss="$adapShor"  '{if(NR%4==1) $0=sprintf("@"sampless":"arrayss":s"++k,(1+i++)); print;}' ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads.fastq >> ${BaseLocation}/FinalSpacers/ForDADA2_woREverse/${names}_allSpacermerged_${adap}.fastq

adapShor=$(echo $adap |sed 's/Array/a/g')
awk -v sampless="$names" -v arrayss="$adapShor"  '{if(NR%4==1) $0=sprintf("@"sampless":"arrayss":s"++k,(1+i++)); print;}' ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed.fastq >> ${BaseLocation}/FinalSpacers/ForDADA2_woREverse/${names}_allSpacermerged_${adap}.fastq


#-----------merge samples
mkdir -p  ${BaseLocation}/FinalSpacers/mergeSpacersFile
cat ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_renamed.fastq >> ${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_${adap}.fastq
cat ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular_renamed.fastq >> ${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_${adap}.fastq


##------ make a folder for dada2
cat ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_renamed.fastq >> ${BaseLocation}/FinalSpacers/ForDADA2/${names}_allSpacermerged_${adap}.fastq
cat ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular_renamed.fastq >> ${BaseLocation}/FinalSpacers/ForDADA2/${names}_allSpacermerged_${adap}.fastq


done #adap (spacer)
  done #names (sampleName)
  
  echo ${BaseLocation}/FinalSpacers/ForDADA2_woREverse/
echo ${BaseLocation}/FinalSpacers/ForDADA2/



```
move local for Dada2

```{bash}
##----------------move local for Dada2
mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers/
scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers/ForDADA2/ ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers/

##----------------2ithout reversed
mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers/
scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers/ForDADA2_woREverse/ ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers/
```


Remarks:
1. I remove the following sections
  - making fastqs of the CRISPR spacers from the genome assemblies (they were all lost and it makes more sense to cd-hit them later)
  - cd-hit clustering (this was the old way I did the clustering new I changed to DADA2)


####Dada2

Here, I try out the [Dada2](https://benjjneb.github.io/dada2/tutorial.html) tutorial to call true positive spacers. 

```{r}
library(dada2)
library(plyr)
library(tidyverse)

#path <- "~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/ForDADA2/" # CHANGE ME to the directory containing the fastq files after unzipping.
# path <- "~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers//ForDADA2/" # CHANGE ME to the directory containing the fastq files after unzipping.
path <- "~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers//ForDADA2_woREverse/" 

list.files(path)

fnFs <- sort(list.files(path, pattern=".fastq", full.names = TRUE)) 
sample.names <- str_split_fixed(basename(fnFs), ".fastq",2)[,1]

seq.file <- fnFs[1]
foo <- readLines(seq.file)
seqlens <- sapply(foo[seq(2,length(foo),4)], nchar)
quallens <- sapply(foo[seq(4,length(foo),4)], nchar)
table(seqlens); table(quallens)
table(seqlens) == table(quallens)

plotQualityProfile(fnFs[1:2])
plotQualityProfile(fnFs[3:4])

# Place filtered files in filtered/ subdirectory
filtFs <- file.path(path, "filtered", paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(path, "filtered", paste0(sample.names, "_R_filt.fastq.gz"))
names(filtFs) <- sample.names
names(filtRs) <- sample.names

out <- filterAndTrim(fnFs, filtFs, truncQ=2, rm.phix=TRUE,
              compress=TRUE, multithread=TRUE) # On Windows set multithread=FALSE
##--------------
##plot number of CRISPR spacer reads
##--------------

plotREads <- as.data.frame(out)
plotREads$reads.in
plotREads$longName <- rownames(plotREads)
plotREads$shortName <- gsub("_allSpacermerged_","-",plotREads$longName) %>% gsub(".fastq","",.)
plotREads$sample <- str_split_fixed(plotREads$shortName, "-", 2)[,1]
plotREads$array <- str_split_fixed(plotREads$shortName, "-", 2)[,2]
plotREads$sample <- revalue(plotREads$sample, c("lyo202_96"="Lyo 1996","Lyo_202_2012"="Lyo 2012", "Lyo_202_2014"="Lyo 2014", "Konserve_202"="working stock", "RMK202"="starter culture 2012", "Versand_202"="starter culture 2018","di_K2_6h"="cheesemaking day1","th_K2_8h"="cheesemaking day2","G1_6_18"="experiment_A","G2_6_18"="experiment_B","G3_6_18"="experiment_C","G4_6_18"="experiment_D","G5_6_18"="experiment_E"))
plotREads$sample = factor(plotREads$sample, levels=c("Lyo 1996","Lyo 2012","Lyo 2014","working stock","starter culture 2012","starter culture 2018","cheesemaking day1","cheesemaking day2","experiment_A","experiment_B","experiment_C","experiment_D","experiment_E"))

# plotREads %>%  group_by(sample) %>% 
colors <- c("#fd8d3cff","#807dbaff","#ef3b2cff")
rawspacerReads <- ggplot(plotREads,aes(sample,y=reads.in,fill=array))+geom_bar(stat="identity")+theme_classic()+facet_wrap(~array)+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),legend.position = "none")+labs(x="",y="CRISPR spacer reads")+scale_fill_manual(values = colors)
rawspacerReads
  svg("~/Desktop/Projects/2019_RMK202_analysis/plot/rawCRISPRspacerReads.svg",width=4,height=4)
  rawspacerReads
   dev.off()



##--------------
##error learn
##--------------
errF <- learnErrors(filtFs, multithread=TRUE)
plotErrors(errF, nominalQ=TRUE)
dadaFs <- dada(filtFs, err=errF, multithread=TRUE)
dadaFs[[1]]

###fasta table

seqtab <- makeSequenceTable(dadaFs)
dim(seqtab)
table(nchar(getSequences(seqtab)))


##do not remove Chimeras because they are all on the same read and most likely not technical errors but truth biological variation
# seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=TRUE, verbose=TRUE)
# dim(seqtab.nochim)
# sum(seqtab.nochim)/sum(seqtab)


##----phyloseq

library(phyloseq); packageVersion("phyloseq")
library(Biostrings); packageVersion("Biostrings")
library(ggplot2); packageVersion("ggplot2")

samples.out <- rownames(seqtab)
sample <- str_split_fixed(samples.out, "_allSpacermerged_", 2)[,1]
array <- str_split_fixed(samples.out, "_allSpacermerged_", 2)[,2]
# subject <- sapply(strsplit(samples.out, "D"), `[`, 1)
# gender <- substr(subject,1,1)
# subject <- substr(subject,2,999)
# day <- as.integer(sapply(strsplit(samples.out, "D"), `[`, 2))
samdf <- data.frame(Sample=sample, Array=array)
# samdf$When <- "Early"
# samdf$When[samdf$Day>100] <- "Late"
rownames(samdf) <- samples.out


# seq
ps <- phyloseq(otu_table(seqtab, taxa_are_rows=FALSE),sample_data(samdf))

sequences <- Biostrings::DNAStringSet(taxa_names(ps))

names(sequences) <- taxa_names(ps)
ps <- merge_phyloseq(ps, sequences)
taxa_names(ps) <- paste0("spacer_",seq(ntaxa(ps)))

writeXStringSet(refseq(ps), "~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers//ForDADA2_woREverse//all_unique_spacers.fasta", append=FALSE, format="fasta") 
refseq(ps)
# plot_richness(ps, x="Sample",measures=c("Observed","Shannon", "Simpson"),color="Array")


colors <- c("#fd8d3cff","#807dbaff","#ef3b2cff")

ShannonCRISPR <- plot_richness(ps, x="Array", measures=c( "Shannon"),color="Array") + geom_boxplot()+scale_fill_manual(values=colors)+scale_color_manual(values=colors)+theme_classic()+labs(x="",y="Shannon diversity of CRISPR spacers")+theme(legend.title = element_blank(),legend.position="top",axis.text.x =element_text(angle = 75, hjust = 1,size=9))

AbsorvedCRISPR <- plot_richness(ps, x="Array", measures=c( "Observed"),color="Array") + geom_boxplot()+scale_fill_manual(values=colors)+scale_color_manual(values=colors)+theme_classic()+labs(x="",y="Number of CRISPR spacers observed")+theme(legend.position = "none" ,axis.text.x =element_text(angle = 75, hjust = 1,size=9))

  svg("~/Desktop/Projects/2019_RMK202_analysis/plot/rarefaction_spacers_Shannon.svg",width=4,height=8)
  AbsorvedCRISPR+ShannonCRISPR
   dev.off()

##------rarefaction

library(ranacapa)
# rarefaction_spacers <- ggrare(ps, step = 10, label = NULL, color = "Sample",
#   plot = TRUE, parallel = FALSE, se = TRUE) +theme_classic()
sample_variables(ps)
colors <- c("#fd8d3cff","#807dbaff","#ef3b2cff")

sample_names(ps) <- gsub("_allSpacermerged_Array","_a",sample_names(ps))


# rarefaction_spacers <- ggrare(ps, step = 10, label = "Sample", color = "Array",
  # plot = TRUE, parallel = FALSE, se = TRUE) +theme_classic()+scale_fill_manual(values=colors)+scale_color_manual(values=colors)+scale_x_continuous(breaks =c(0,10000,20000,30000,40000,45000),labels=c(0,10000,20000,30000,40000,45000))+labs(x="number of CRISPR spacer reads",y="number of CRISPR spacers detected")+theme(legend.title = element_blank())
# rarefaction_spacers

  
rarefaction_spacers <- ggrare(ps, step = 10, color = "Array",
  plot = TRUE, parallel = FALSE, se = TRUE) +theme_classic()+scale_fill_manual(values=colors)+scale_color_manual(values=colors)+scale_x_continuous(breaks =c(0,10000,20000,30000,40000,45000),labels=c(0,10000,20000,30000,40000,45000))+labs(x="Number of CRISPR spacer reads",y="Number of CRISPR spacers observed")+theme(legend.title = element_blank())
rarefaction_spacers
  
  svg("~/Desktop/Projects/2019_RMK202_analysis/plot/rarefaction_spacers.svg",width=5,height=3)
  rarefaction_spacers
   dev.off()
   
   library(patchwork)
   
  svg("~/Desktop/Projects/2019_RMK202_analysis/plot/rarefaction_spacers_together.svg",width=15,height=8)
  rarefaction_spacers+theme(legend.position = "none")+AbsorvedCRISPR+ShannonCRISPR+plot_layout(widths =c(4,1,1))
   dev.off()  
##------pca
ps.prop <- transform_sample_counts(ps, function(otu) otu/sum(otu))
ord.nmds.bray <- ordinate(ps.prop, method="NMDS", distance="bray")

plot_ordination(ps.prop, ord.nmds.bray, color="Array", title="Bray NMDS")

##-----------------------
##export OTU table
##-----------------------

normfunction <- function(x) {return(100000*x/sum(x))}
d.norm1 <- transform_sample_counts(physeq = ps, fun = normfunction)
otu_table(d.norm1)




otu_table_df <- otu_table(d.norm1) %>% as.data.frame()%>% rownames_to_column() %>% separate(rowname, c("sample","Array"), sep = "_allSpacermerged_") %>% select(-Array) %>%  group_by(sample) %>% summarise_all(.funs = c(sum="sum")) %>% column_to_rownames(var="sample") %>% t() %>% as.data.frame() %>% rownames_to_column("spacer") %>% 
  mutate(spacer = str_replace(spacer, "_sum", "")) 
otu_table_df
write.table(otu_table_df,"~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers//ForDADA2_woREverse//dada_spacer_count.txt",na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names =TRUE)

##-----------------------
##count number of OTUs larger then 0 in every sample
##-----------------------
nrow(otu_table(ps))
otu_counts_samples <- NA
for (samplesss in rownames(otu_table(ps))) {
  tmp_2 <- sum(otu_table(ps)[samplesss]>0)
  tmp_print <- c(sample=as.character(samplesss), otus=as.numeric(tmp_2))
  print(tmp_print)
  otu_counts_samples <- rbind(otu_counts_samples,tmp_print)
}
rownames(otu_table(ps)[1])
sum(otu_table(ps)[1]>0)

otu_counts_samples <- as.data.frame(otu_counts_samples[-1,]) %>% remove_rownames()
otu_counts_samples$otus <- as.numeric(as.character(otu_counts_samples$otus))


##-----------------------
##spaccerss belonging to sample
##-----------------------

ncol(otu_table(ps))
spacer_counts_samples <- data.frame()
# samplesss <- "spacer_1"
for (samplesss in colnames(otu_table(ps))) {
  tmp_2 <- sum(otu_table(ps)[,samplesss]>0)

tmp1 <- sample_names(otu_table(ps)[which(otu_table(ps)[,samplesss]>0)])
# countArray1 <- length(grep("Array1",tmp1))
# countArray2 <- length(grep("Array2",tmp1))
# countArray3 <- length(grep("Array3",tmp1))
countArray1 <- length(grep("_a1",tmp1))
countArray2 <- length(grep("_a2",tmp1))
countArray3 <- length(grep("_a3",tmp1))

numberArrays <-(countArray1>0)+(countArray2>0)+(countArray3>0)

ArrayAssigned <- ifelse(countArray1>0,"Array1",ifelse(countArray2>0,"Array2","Array3"))

  tmp_print <- data.frame(spacer=as.character(samplesss), samplesCount=as.numeric(tmp_2),Array1count=countArray1,Array2count=countArray2,Array3count=countArray3,totalArrays=numberArrays,Array=ArrayAssigned)
  
  print(tmp_print)
  spacer_counts_samples <- rbind(spacer_counts_samples,tmp_print)
}
# rownames(spacer_counts_samples(ps)[1])
sum(otu_table(ps)[1]>0)

# spacer_counts_samples <- as.data.frame(spacer_counts_samples[-1,]) %>% remove_rownames()
# spacer_counts_samples$otus <- as.numeric(as.character(spacer_counts_samples$otus))

spacer_counts_samples[which(spacer_counts_samples$Array2count > 0),]
table(spacer_counts_samples$totalArrays)
table(spacer_counts_samples$Array)
spacer_counts_samples_final <- spacer_counts_samples %>% select(-c(Array1count,Array2count,Array3count,totalArrays))
spacer_counts_samples_out <- spacer_counts_samples %>% select(c(spacer,Array))

write.table(spacer_counts_samples_out,"~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers//ForDADA2_woREverse//spacersAssigned.txt",na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names =FALSE)
write.table(spacer_counts_samples_final,"~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers//ForDADA2_woREverse//spacers_info.txt",na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names =TRUE)

ncol(otu_table_df)

##the following spacers only have coverage in the experiment samples
otu_table_df[which(rowSums(otu_table_df[,-c(1,grep("_6_18",colnames(otu_table_df)))])==0),]



```

spacers only occur in one array. They never change array. 

```{bash}
###move to big machine

scp ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers//ForDADA2_woREverse//all_unique_spacers.fasta vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers/ForDADA2_woREverse/

scp ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers//ForDADA2_woREverse//spacersAssigned.txt vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers/ForDADA2_woREverse/

##--------------------spacer count


scp ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers//ForDADA2_woREverse//dada_spacer_count.txt vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers/ForDADA2_woREverse/




```





####match strains to spacers

this output can be used for the spacer heatmap


```{bash}
###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt  ##the file with all sample names
#logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract/
#Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/FASTA/S_thermophilus_alltogether.CRISPRlocation.fasta
species=Sterm
#BaseLocation_bam=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP
repeatsFile=/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt
MetaspacerFile=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers/ForDADA2_woREverse//all_unique_spacers.fasta

###---------------------------
##merge CRISPR spacer files

mkdir -p ${BaseLocation}/FinalSpacers/cdhit_withReference

#cat /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//FinalSpacers/ForDADA2/all_unique_spacers.fasta \
# /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/S_thermophilus_RMK202.pilarCR_out_final_new_short \
#/archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out_final_newshort \
#/archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out_final_newshort >  \
#/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//FinalSpacers/cdhit_withReference/all_spacers_withRef.fasta


cat ${MetaspacerFile} \
 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results_ClusterName.txt |sed 's/Cluster/cl/g' >  \
${BaseLocation}/FinalSpacers/cdhit_withReference/all_spacers_withRef_and_oldStrains.fasta
#/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//FinalSpacers/cdhit_withReference/all_spacers_withRef_and_oldStrains.fasta
##--------------------------------------------------
##-----------cd -hit
##--------------------------------------------------
mkdir -p ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain

/home/vincent/apps/cd-hit-v4.8.1-2019-0228/cd-hit-est -i ${BaseLocation}/FinalSpacers/cdhit_withReference/all_spacers_withRef_and_oldStrains.fasta \
    -o  /${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered.txt -c 0.995


##--Qualtiy control

grep ">" -c ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered.txt


  
###=====================================================0  
##extract via spacers
###----------new approach 
###=====================================================
#spacersss=$( echo ">spacer_330")


sed 's/>Clu/>META_Clu/g' ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered.txt.clstr |sed 's/META_Cluster /META_Cluster_/g' > ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed.txt.clstr


echo -e "METAClusterName\tdadaSpacer\tStrains\tARRAYINFO\tClusterINFO\tspacer" > ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain//uniqueSpacers_count_Ref_wOLDstrains.txt

for spacersssClusts in $(grep "^>" ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed.txt.clstr)
do
#spacersssClusts=$(echo ">META_Cluster_3")
echo ${spacersssClusts}

#spacerextend=$(echo ${spacersss}"..")
#spacersssClusts=$(echo ">META_Cluster_331")

ClusterEND=$(grep -A 10 "${spacersssClusts}$(printf '$')" ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed.txt.clstr |grep ">META_C"|head -2|tail -1)

#ClusterSTART=329
#ClusterEND=330

  sed -n "/$spacersssClusts$/,/$ClusterEND$/p" ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed.txt.clstr > ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/tmp.txt

tmpfile=${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/tmp.txt

#spacerName=spacer_16

spacer=$(grep ">spacer_" -c ${tmpfile})  
strains=$(grep ">cl_" -c ${tmpfile})
spacerName=$(grep ">spacer_" ${tmpfile} |head -1|awk -F "[>...]" '{print $2}')
if [ ${strains} == "0" ]; then
            arrayINFO=$(grep "${spacerName}$(printf '\t')" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers/ForDADA2_woREverse/spacersAssigned.txt |cut -f 2|sed 's/Array/a/g')
            strains_INFO=$(echo "MetaGenome")
            else
              strains_INFO=$(grep ">cl_" ${tmpfile} |head -1|awk -F "[>...]" '{print $2}'|sed 's/cl_/Cluster_/g' )
            arrayINFO=$(grep "${strains_INFO}$(printf '\t')" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final.txt |cut -f 6|head -1)
               
            fi

clusterNAME=$(echo -e "${spacersssClusts}" | sed 's/ /_/g'|sed 's/>//g')

 echo -e ${clusterNAME}"\t"${spacer}"\t"${strains}"\t"${arrayINFO}"\t"${strains_INFO}"\t"${spacerName} >> ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/uniqueSpacers_count_Ref_wOLDstrains.txt

done #spacers
cut -f 2 ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/uniqueSpacers_count_Ref_wOLDstrains.txt|sort|uniq -c
cut -f 3 ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/uniqueSpacers_count_Ref_wOLDstrains.txt|sort|uniq -c
cut -f 4 ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/uniqueSpacers_count_Ref_wOLDstrains.txt|sort|uniq -c


awk '{if($2==2)print $0}' ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/uniqueSpacers_count_Ref_wOLDstrains.txt

###----------------------------------------
##change name of SPACERS
###----------------------------------------
#/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed.txt

 cat ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered.txt > ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed_ClusterName.txt
 rm ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/tmp_spacer.txt
   for spacersss in $(grep ">" ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed_ClusterName.txt)
   do
   echo "--------------------------"
   #grep "${spacersss}$(printf '\n')" -A 1 /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed_ClusterName.txt  #> /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/sterm/tmp_spacer.txt
   
   #spacersss=spacer_32
   
   newNAME=$(grep -B 20 "${spacersss}\." ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed.txt.clstr |grep "^>META" |tail -1 )
   
   echo -e ${newNAME}"\t"${spacersss} >> ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain//tmp_spacer.txt
   
   sed -i "s/${spacersss}$/${newNAME}/g" ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed_ClusterName.txt
   
   done
   rm ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain//tmp_spacer.txt
   grep ">" ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed_ClusterName.txt |sort|uniq |wc -l
   
  echo ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed_ClusterName.txt 
  echo ${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/uniqueSpacers_count_Ref_wOLDstrains.txt
```


```{bash}
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain
scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/uniqueSpacers_count_Ref_wOLDstrains.txt /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/uniqueSpacers_count_Ref_wOLDstrains.txt

###--------------------all CRISPR spacers to use with CRISPR map
scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed_ClusterName.txt /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/
```
Than we check which clusters are only metagenomic and which only strains

```{r}
library(readr)
library(ggplot2)
# uniqueSpacers_count_Ref_wOLDstrains <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/ForDADA2/uniqueSpacers_count_Ref_wOLDstrains.txt","\t", escape_double = FALSE, trim_ws = TRUE)
uniqueSpacers_count_Ref_wOLDstrains <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/uniqueSpacers_count_Ref_wOLDstrains.txt","\t", escape_double = FALSE, trim_ws = TRUE)

uniqueSpacers_count_Ref_wOLDstrains$explained <- ifelse(uniqueSpacers_count_Ref_wOLDstrains$dadaSpacer>0&uniqueSpacers_count_Ref_wOLDstrains$Strains>0,"both Meta & strains",ifelse(
  uniqueSpacers_count_Ref_wOLDstrains$dadaSpacer>0&uniqueSpacers_count_Ref_wOLDstrains$Strains==0,"explained only by meta","explained only by strains"
))

# table(uniqueSpacers_count_Ref_wOLDstrains$explained)

##-------------plot explained
# uniqueSpacers_count_Ref_wOLDstrains$ARRAYINFO

colorsss <- rev(c("darkcyan","darkturquoise","lightblue"))
uniqueSpacers_count_Ref_wOLDstrains$explained = factor(uniqueSpacers_count_Ref_wOLDstrains$explained, levels=c("explained only by meta","both Meta & strains","explained only by strains"))
uniqueSpacers_count_Ref_wOLDstrains$ARRAYINFO <- revalue(uniqueSpacers_count_Ref_wOLDstrains$ARRAYINFO, c("a2"="CRISPR array 2","a1"="CRISPR array 1","a3"="CRISPR array 3"))
# uniqueSpacers_count_Ref_wOLDstrains$explained <- revalue(uniqueSpacers_count_Ref_wOLDstrains$ARRAYINFO, c("a2"="CRISPR array 2","a1"="CRISPR array 1","a3"="CRISPR array 3"))
table(uniqueSpacers_count_Ref_wOLDstrains$explained,uniqueSpacers_count_Ref_wOLDstrains$ARRAYINFO)

# svg("~/Desktop/Projects/2019_RMK202_analysis/plot/spacerexplained.svg",width=4,height=3)
# ggplot(uniqueSpacers_count_Ref_wOLDstrains,aes(x=ARRAYINFO,group=explained,fill=explained))+geom_bar()+theme_classic()+scale_fill_manual(values=colorsss)+theme(axis.title.x = element_blank(),axis.text.x =element_text(angle = 45, hjust = 1,size=9) )+labs(y="spacer counts",fill="")+scale_y_continuous(breaks =c(0,39,50,100,142,158),labels=c(0,39,50,100,142,158))
#  dev.off()
 
 ##new and wider
svg("~/Desktop/Projects/2019_RMK202_analysis/plot/spacerexplained_wide.svg",width=4,height=3)
ggplot(uniqueSpacers_count_Ref_wOLDstrains,aes(x=ARRAYINFO,group=explained,fill=explained))+geom_bar()+theme_classic()+scale_fill_manual(values=colorsss)+theme(axis.title.x = element_blank(),axis.text.x =element_text(angle = 45, hjust = 1,size=9) )+labs(y="spacer counts",fill="")+scale_y_continuous(breaks =c(0,10,27,39,92,117,139,163),labels=c(0,10,27,39,92,117,139,163))
 dev.off()
 ##------------------------
 #what are the only strains CRISPRs
  ##------------------------
ONlyIsolateSPACERS <- uniqueSpacers_count_Ref_wOLDstrains %>% filter(explained=="explained only by strains")
 

merge(ONlyIsolateSPACERS,spacer_Infos_Sterm_final,by.x="ClusterINFO",by.y="ClusterName")
 
merge(ONlyIsolateSPACERS,uniqueSpacersss_extended,by.x="ClusterINFO",by.y="ClusterName")


```

##extended Analysis
#####blast Spacers
Here a look which spacers map to the phages or bacteria. 

I could also use the [CRISPRCasdb](https://crisprcas.i2bc.paris-saclay.fr/) tool for this. 
this however did not work ideally. 

```{bash}


mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/genomes/
scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/genomes/ ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/genomes/

```


```{bash}
###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt  ##the file with all sample names
#logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract/
#Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/FASTA/S_thermophilus_alltogether.CRISPRlocation.fasta
species=Sterm
#BaseLocation_bam=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP
repeatsFile=/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt
MetaspacerFile=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers/ForDADA2_woREverse//all_unique_spacers.fasta
mappingFile=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed_ClusterName.txt

##--------------------------blast to local phages 
  
#rm -r /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast
mkdir -p ${BaseLocation}/CRISPRspacerBLAST/blast
mkdir -p ${BaseLocation}/CRISPRspacerBLAST/starterPhagesPlasmid
mkdir -p ${BaseLocation}/CRISPRspacerBLAST/genomes/
  
##---------------------
##get local phages
##---------------------
rm ${BaseLocation}/CRISPRspacerBLAST/genomes/phagenGenome.fasta
for phageesss in $(grep ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta |grep "CP046134" -v | grep -v "CP046131"|sed 's/>//g')
do

samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta ${phageesss} > ${BaseLocation}/CRISPRspacerBLAST/genomes/${phageesss}.fasta
cat ${BaseLocation}/CRISPRspacerBLAST/genomes/${phageesss}.fasta >> ${BaseLocation}/CRISPRspacerBLAST/genomes/phagenGenome.fasta
done


makeblastdb -max_file_sz 1GB -in ${BaseLocation}/CRISPRspacerBLAST/genomes/phagenGenome.fasta \
 -out ${BaseLocation}/CRISPRspacerBLAST/genomes/phagenGenome -parse_seqids -dbtype nucl


  blastn -num_threads 35 -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query ${mappingFile} \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen sgi sacc sblastnames staxids sscinames" \
  -db ${BaseLocation}/CRISPRspacerBLAST/genomes/phagenGenome \
  -out ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages.txt \
  -evalue 0.01 -word_size 16



##---------------------
##one phage at the time
##---------------------
#phageesss=Streptococcus_phage_2

for phageesss in $(echo "Streptococcus_phage_1 Streptococcus_phage_2")
do
rm ${BaseLocation}/CRISPRspacerBLAST/genomes/${phageesss}.fasta

samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta ${phageesss} > ${BaseLocation}/CRISPRspacerBLAST/genomes/${phageesss}.fasta


makeblastdb -max_file_sz 1GB -in ${BaseLocation}/CRISPRspacerBLAST/genomes/${phageesss}.fasta \
 -out ${BaseLocation}/CRISPRspacerBLAST/genomes/${phageesss} -parse_seqids -dbtype nucl


  blastn -num_threads 35 -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query ${mappingFile} \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen sgi sacc sblastnames staxids sscinames" \
  -db ${BaseLocation}/CRISPRspacerBLAST/genomes/${phageesss} \
  -out ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_${phageesss}.txt \
  -evalue 0.01 -word_size 16

awk -F "\t" '{OFS="\t"}{if($5<"2")print $0}' ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_${phageesss}.txt > ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_${phageesss}_cleaned.txt
done
phageesss=Streptococcus_phage_1
rm ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both.txt
#rm ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both_allINfo.txt
for phageesss in $(echo "Streptococcus_phage_1 Streptococcus_phage_2")
do
echo ${phageesss}
wc -l ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_${phageesss}.txt
cut -f 3 ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_${phageesss}_cleaned.txt |sort|uniq -c
cut -f 1,2 ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_${phageesss}_cleaned.txt >>  ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both.txt

cat ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_${phageesss}_cleaned.txt >> ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both_allINfo.txt
done

wc -l ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both.txt
cut -f 1 ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both.txt|sort|uniq|wc -l
cut -f 1 ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both.txt|sort|uniq -c |sed 's/^ *//g'


rm ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits.txt
cut -f 1 ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both.txt|sort|uniq -c |sed 's/^ *//g'| while read genenomes 
do
#echo ${genenomes}
#echo "--------------------"
num=$(echo ${genenomes}|cut -d ' ' -f 1)
spacer=$(echo ${genenomes}|cut -d ' ' -f 2)

if [ "$num" = "1" ]
then
genom=$(grep ${spacer} ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both.txt|cut -f 2|head -1)
echo -e "localPHAGE\t"${spacer}"\t"${genom} >> ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits.txt
else
echo -e "localPHAGE\t"${spacer}"\tbothPhages" >> ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits.txt
fi
done


###--------------------where do the mutations lie 
##prepare for genoPlotR
rm -r ${BaseLocation}/CRISPRspacerBLAST/blast/GenotplotR/
mkdir -p ${BaseLocation}/CRISPRspacerBLAST/blast/GenotplotR/
for phageesss in $(echo "Streptococcus_phage_1 Streptococcus_phage_2")
do
echo ${phageesss}
#wc -l ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_${phageesss}.txt

echo -e "name\tstart\tend\tstrand\tcol\tfill" > ${BaseLocation}/CRISPRspacerBLAST/blast/GenotplotR/CRISPRblast_rmk202phages_${phageesss}_genoplotR.txt

awk -F "\t" '{OFS="\t"}{print $1,$9,$10,"1","black",$5}' ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_${phageesss}.txt >> ${BaseLocation}/CRISPRspacerBLAST/blast/GenotplotR/CRISPRblast_rmk202phages_${phageesss}_genoplotR.txt

awk -F "\t" -v genomsss="$phageesss" '{OFS="\t"}{print genomsss,$9,$10}' ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_${phageesss}.txt >> ${BaseLocation}/CRISPRspacerBLAST/blast/GenotplotR/CRISPRblast_rmk202phages_${phageesss}_Circos.txt

done

###--------------------where do the mutations lie for circos
rm ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits_bothGenomes_circos_protospacers.txt
cut -f 1 ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both.txt|sort|uniq -c |sed 's/^ *//g'| while read genenomes 
do
#echo ${genenomes}
#echo "--------------------"
num=$(echo ${genenomes}|cut -d ' ' -f 1)
spacer=$(echo ${genenomes}|cut -d ' ' -f 2)

if [ "$num" = "1" ]
then
grep "${spacer}" ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both_allINfo.txt |awk -F "\t" '{OFS="\t"}{if($9>$10)print $2,$9,$10,"black_a1";else print $2,$9,$10,"black_a1"}' >> ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits_bothGenomes_circos_protospacers.txt
else
grep "${spacer}" ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both_allINfo.txt|head -1 |awk -F "\t" '{OFS="\t"}{if($9>$10)print $2,$9,$10,"chr5_a1";else print $2,$9,$10,"chr5_a1"}' >> ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits_bothGenomes_circos_protospacers.txt
grep "${spacer}" ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both_allINfo.txt|tail -1 |awk -F "\t" '{OFS="\t"}{if($9>$10)print $2,$9,$10,"chr5_a1";else print $2,$9,$10,"chr5_a1"}' >> ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits_bothGenomes_circos_protospacers.txt
fi
done

###--------------------where do the mutations lie for protospacer vs. genes intersect
rm ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits_bothGenomes_interesect_protospacers.txt
cut -f 1 ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both.txt|sort|uniq -c |sed 's/^ *//g'| while read genenomes 
do
#echo ${genenomes}
#echo "--------------------"
num=$(echo ${genenomes}|cut -d ' ' -f 1)
spacer=$(echo ${genenomes}|cut -d ' ' -f 2)

if [ "$num" = "1" ]
then
grep "${spacer}" ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both_allINfo.txt |awk -F "\t" '{OFS="\t"}{if($9>$10)print $2,$10,$9,$1;else print $2,$9,$10,$1}' >> ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits_bothGenomes_interesect_protospacers.txt
else
grep "${spacer}" ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both_allINfo.txt|head -1 |awk -F "\t" '{OFS="\t"}{if($9>$10)print $2,$10,$9,$1;else print $2,$9,$10,$1}' >> ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits_bothGenomes_interesect_protospacers.txt
grep "${spacer}" ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_both_allINfo.txt|tail -1 |awk -F "\t" '{OFS="\t"}{if($9>$10)print $2,$10,$9,$1;else print $2,$9,$10,$1}' >> ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits_bothGenomes_interesect_protospacers.txt
fi
done

```


```{bash}
mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/GenotplotR/
scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/GenotplotR/ ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/GenotplotR/

scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/all_differentBlastHits_bothGenomes_circos_protospacers.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/all_differentBlastHits_bothGenomes_circos_protospacers.txt
    
```

now bacteria

```{bash}


##--------------------------------  
##--------------------------blast to local bacteria
##preperation have ot do once
##--------------------------------   
 
###----------------
 ##CRISPR locations
 ##and CRISPR MASK
###----------------
species=Sterm
 rm ${BaseLocation}/CRISPRspacerBLAST/blast/all_Genomes_masked.fna
 for genomes in $(echo "202-SMAG 202-S50 202-S72")
do
echo $genomes

 grep "repeat" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA_*.gff |cut -f 1,4,5 > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/CRISPRrepeat_${genomes}.bed


bedtools maskfasta -fi /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned//${genomes}.fna  -bed /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/CRISPRrepeat_${genomes}.bed -fo /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned//${genomes}_CRISPRmasked.fna 

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned//${genomes}_CRISPRmasked.fna  >> ${BaseLocation}/CRISPRspacerBLAST/blast/all_Genomes_masked.fna
 
done

 
#----------ldel 
 
species=Ldel
genomes=202-LMAG
 grep "repeat" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA_*.gff |cut -f 1,4,5 > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/CRISPRrepeat_${genomes}.bed

grep "repeat" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA_*.gff |awk -F "\t" '{OFS="\t"}{print $1,$4-500,$5+500}' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/CRISPRrepeat_${genomes}_extended.bed



bedtools maskfasta -fi /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned//${genomes}.fna  -bed /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/CRISPRrepeat_${genomes}.bed -fo /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned//${genomes}_CRISPRmasked.fna 

bedtools maskfasta -fi /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned//${genomes}.fna  -bed /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/CRISPRrepeat_${genomes}_extended.bed -fo /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned//${genomes}_CRISPRmasked_extended.fna 


cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned//${genomes}_CRISPRmasked.fna >> ${BaseLocation}/CRISPRspacerBLAST/blast/all_Genomes_masked.fna
 
 
 
  ##create bacteria DB
  ##---------rmk202
#bedtools maskfasta -fi /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_sterm/final/S_thermophilus_mag_rmk202.fasta -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/allStreptococcus_CRISPRlocation.bed \
#-fo /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/Streptococcus_CRISPR.masked_rmk202Genome.fa

##---------S50
#bedtools maskfasta -fi /archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/S50.fna -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/allStreptococcus_CRISPRlocation.bed \
#-fo /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_S50_Genome.fa

##---------rmk202
#bedtools maskfasta -fi /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/S72.fna -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/allStreptococcus_CRISPRlocation.bed \
#-fo /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_S72_Genome.fa

##_rename contigs for easy search

#sed -i 's/>/>REF_R202:/g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/Streptococcus_CRISPR.masked_rmk202Genome.fa
#sed -i 's/>/>REF_mst1:/g' /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_S50_Genome.fa
#sed -i 's/>/>REF_mst2:/g' /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_S72_Genome.fa

#cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/Streptococcus_CRISPR.masked_rmk202Genome.fa \
#/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_ldel/final/L_delbrueckii_mag_rmk202.fasta \
#/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_S50_Genome.fa \
#/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_S72_Genome.fa > \
#/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_allStrepGenomesrmk202_Genome.fa


#makeblastdb -max_file_sz 1GB -in /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/starterPhagesPlasmid/rmk202_phages_plasmids.fasta \
# -out /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/starterPhagesPlasmid/rmk202_phages_plasmids -parse_seqids -dbtype nucl




makeblastdb -max_file_sz 1GB -in ${BaseLocation}/CRISPRspacerBLAST/blast/all_Genomes_masked.fna \
 -out ${BaseLocation}/CRISPRspacerBLAST/blast/all_Genomes_masked -parse_seqids -dbtype nucl


  blastn -num_threads 35 -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query ${mappingFile} \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen sgi sacc sblastnames staxids sscinames" \
  -db ${BaseLocation}/CRISPRspacerBLAST/blast/all_Genomes_masked \
  -out ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Bacteria_02.txt \
  -evalue 0.01 -word_size 16
  


  ####------------exkursion
  ##------check if the matches are not undetected CRISPR regions
  ##currently not doing orphan CRISPR prediction
  ##-----------------------------------
  
#  grep "NODE_" //archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Bacteria.txt |awk -F "\t" '{OFS="\t"} {if ($9<$10) print $2,$9-50,$10+50,$1 ; else print $2,$10-50,$9+50,$1 }' > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.txt

  
#samtools faidx /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_allStrepGenomesrmk202_Genome.fa
  
#bedtools getfasta -fi /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_allStrepGenomesrmk202_Genome.fa\
#  -bed /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.txt \
#  -fo /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.fasta

#grep -B 1 --color "GAGCTGTGTTGTTTCGAATGGT" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.fasta |grep ">" > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_filtered.fasta
#grep -B 1 --color "GTCCCCTCTCGAGGTAATTAGG" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.fasta|grep ">" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_filtered.fasta
#grep -B 1  --color "ACAGTTACTTAAATCTTGAGAGTA" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.fasta |grep ">"  >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_filtered.fasta



#grep -B 1 --color "GAACCATTCGAAACAACACAGCT" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.fasta |grep ">" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_filtered.fasta
#grep -B 1 --color "CCTAATTACCTCGAGAGGGGACG" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.fasta|grep ">" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_filtered.fasta
#grep -B 1  --color "GTACTCTCAAGATTTAAGTAACTG" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.fasta |grep ">"  >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_filtered.fasta

#sed -i 's/>//g' /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_filtered.fasta

#remove the CRISPR repeats
#rm /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/all_differentBlastHits.txt

#for removess in $(cat /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_filtered.fasta)
#do

#contigss=$(echo $removess | cut -d ':' -f 1,2)
#start=$(echo $removess | cut -d ':' -f 3|cut -d '-' -f 1)
#end=$(echo $removess | cut -d ':' -f 3|cut -d '-' -f 2)
#hit=$(echo $removess | cut -d ':' -f 1)
#echo $removess

#explained=$(grep "${contigss}" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.txt | grep "${start}" |grep "${end}" |awk -F "\t" '{print $4}')

#echo -e "OrphanCRISPR""\t"${explained}"\t"${hit} >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/all_differentBlastHits.txt

#done




#rm /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_spacers.txt
#for removess in $(cat /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_filtered.fasta)
#do
#contigss=$(echo $removess | cut -d ':' -f 1)
#start=$(echo $removess | cut -d ':' -f 2|cut -d '-' -f 1)
#end=$(echo $removess | cut -d ':' -f 2|cut -d '-' -f 2)

#grep "${contigss}" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.txt | grep "${start}" |grep "${end}" |awk -F "\t" '{print $4}' >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_spacers.txt
#done
#cat /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages.txt > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_new.txt
#for spazerss in $(cat /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_spacers.txt)
#do
#echo ${spazerss}
#grep "${spazerss}" -v /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_new.txt > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_tmp.txt 
#cat /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_tmp.txt > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_new.txt

#done

#wc -l /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages.txt
#wc -l /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_new.txt

 ##-------------continue

  
# cut -f 1 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_viralREF.txt > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/mapped_spacersNames_onlyNames.txt
  

 # diff /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/mapped_spacersNames_onlyNames.txt \
#  /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/unmapped2local_spacersNames_onlyNames.txt | grep ">" | sed 's/> //g' >\
#  /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/unmapped_spacersNames_onlyNames.txt
##---------------------------------------------------------------
##--------------------------blast to viral DB
#this is the vContact database
##---------------------------------------------------------------
#cat /archiv/Phage_DB/BLAST/20190519/downloads/Sterm/Renamed/RenamedContigs/*.fna \
#/archiv/Phage_DB/BLAST/20190519/downloads/Ldel/Renamed/RenamedContigs/*.fna \
#/archiv/Phage_DB/BLAST/20190519/downloads/Lactococcus/Renamed/RenamedContigs/*.fna > \
#/archiv/Phage_DB/BLAST/20190519/Vcontact/Database/alltogether/data/merged_all_phages.fna


makeblastdb -max_file_sz 1GB -in /archiv/Phage_DB/BLAST/20190519/Vcontact/Database/alltogether/data/merged_all_phages.fna \
 -out /archiv/Phage_DB/BLAST/20190519/Vcontact/Database/alltogether/data/merged_all_phages -parse_seqids -dbtype nucl


  blastn -num_threads 35 -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query ${mappingFile} \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen sgi sacc sblastnames staxids sscinames" \
  -db /archiv/Phage_DB/BLAST/20190519/Vcontact/Database/alltogether/data/merged_all_phages \
  -out ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_ViralDB.txt \
  -evalue 0.01 -word_size 16
  


##--------------------------blast to old viral DB
#this is the RefSeq database

  blastn -num_threads 35 -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query ${mappingFile} \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen sgi sacc sblastnames staxids sscinames" \
  -db /archiv/BLAST/blast_db/20190204/viral_nucleotide/viral \
  -out ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_ViralDB_old.txt \
  -evalue 0.01 -word_size 16
  
  
  
##--------------------------blast unmapped to nucleotide DB

#rm /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/spacers/unmapped_spacers4blast.fasta
#for i in $(cat /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/unmapped_spacersNames_onlyNames.txt)
#do
#grep -A 1 "${i}" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/spacers/all_spacers4blast.fasta >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/spacers/unmapped_spacers4blast.fasta
#done


blastn -num_threads 37 -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query ${mappingFile} \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen" \
  -db /archiv/BLAST/blast_db/20190204/nucleotide/prokaryotes \
  -out  ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_nucleotideDB.txt \
  -evalue 0.01 -word_size 16




##-----------------------merge all blasts


 # cut -f 1,2 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages.txt |sed 's/_contig[[:digit:]]\+//g'  | awk -F "[\t]" '{OFS="\t"} {print "localPHAGE",$1,$2}' > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/all_differentBlastHits.txt

#  cut -f 1,2 ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages.txt | awk -F "[\t]" '{OFS="\t"} {print "localPHAGE",$1,$2}' > ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits.txt

  awk -F "\t" '{OFS="\t"} {print $1 ,$2 }' ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Bacteria_02.txt | awk -F "[\t]" '{OFS="\t"} {print "localBAC",$1,$2}' >>  ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits.txt


  cut -f 1,13 ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_ViralDB.txt |cut -d ',' -f 1 | sed 's/ complete prophage genome//g' |sed 's/ complete genome//g'  | awk -F "[\t]" '{OFS="\t"} {print "phageDB",$1,$2}' >>  ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits.txt

cut -f 1,13 ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_nucleotideDB.txt |cut -d ',' -f 1 | awk -F "[\t]" '{OFS="\t"} {print "BactDB",$1,$2}' >>  ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits.txt

cut -f 1 ${BaseLocation}/CRISPRspacerBLAST/blast/all_differentBlastHits.txt | sort|uniq -c
#cut -f 1 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/all_differentBlastHits.txt|sort|uniq -c
```


```{bash}
##-------------------------transfer local

  mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast
   scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/all_differentBlastHits.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/all_differentBlastHits.txt
  
```

##### identify protospacers
I do this after the phage circos chunk. 

```{bash}

 # mkdir -p /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/
   scp  /home/vincent/Desktop/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/gene_CLEANED_FINAL_ALL.gff vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/gene_CLEANED_FINAL_ALL.gff
  
```

```{bash}
awk -F "\t" '{OFS="\t"}{print $1,$2,$3,$4,$5,$6,$7,$8,$9";"$10}' /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/gene_CLEANED_FINAL_ALL.gff > /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/gene_CLEANED_FINAL_ALL_ready.gff



##------------------------------
##intersect with protospacers
##------------------------------

mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/protospacer/intersectingWithGENES
rm  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/protospacer/intersectingWithGENES/geneIntersect.txt
bedtools intersect -a /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/gene_CLEANED_FINAL_ALL_ready.gff  \
  -b /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/all_differentBlastHits_bothGenomes_interesect_protospacers.txt -wb > \
   /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/protospacer/intersectingWithGENES/geneIntersect.txt
wc -l  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/protospacer/intersectingWithGENES/geneIntersect.txt


awk -F "[\t;]" '{OFS="\t"}{print $1,$13,$14,$10,$11,$15}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/protospacer/intersectingWithGENES/geneIntersect.txt > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/protospacer/intersectingWithGENES/geneIntersect_cleaned_wo_nomatch.txt

awk -F "[\t]" '{OFS="\t"}{print $5}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/protospacer/intersectingWithGENES/geneIntersect_cleaned_wo_nomatch.txt |sort|uniq -c


###------------no matching protospacers

awk -F "[\t;]" '{OFS="\t"}{print $12,$13,$14,$15}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/protospacer/intersectingWithGENES/geneIntersect.txt |sort|uniq > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/protospacer/intersectingWithGENES/geneIntersect_present.txt 

sort /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/all_differentBlastHits_bothGenomes_interesect_protospacers.txt > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/all_differentBlastHits_bothGenomes_interesect_protospacers_sorted.txt 

comm -3 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/all_differentBlastHits_bothGenomes_interesect_protospacers_sorted.txt /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/protospacer/intersectingWithGENES/geneIntersect_present.txt |awk -F "[\t]" '{OFS="\t"}{print $1,$2,$3,"intergenic","intergenic",$4}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/protospacer/intersectingWithGENES/geneIntersect_cleaned_wo_nomatch.txt

awk -F "[\t]" '{OFS="\t"}{print $5}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/protospacer/intersectingWithGENES/geneIntersect_cleaned_wo_nomatch.txt |sort|uniq -c

```

```{bash}
mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/protospacer/intersectingWithGENES/
 scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/protospacer/intersectingWithGENES/geneIntersect_cleaned_wo_nomatch.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/protospacer/intersectingWithGENES/geneIntersect_cleaned_wo_nomatch.txt
  
```

```{r}
library(readr)
geneIntersect_cleaned_wo_nomatch <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/protospacer/intersectingWithGENES/geneIntersect_cleaned_wo_nomatch.txt",   "\t", escape_double = FALSE, col_names = c("Genome","Start","End","locusTAG","geneType","Cluster"),    trim_ws = TRUE)

histo_protspacerGenes <- ggplot(geneIntersect_cleaned_wo_nomatch,aes(x=forcats::fct_infreq(geneType),fill=geneType))+geom_bar()+theme_classic()+facet_wrap(~Genome)+theme(legend.title = element_blank(),axis.text.x = element_text(angle = 75, hjust = 1,size=9))+labs(x="",y="count")

histo_protspacerGenes
 
  svg("~/Desktop/Projects/2019_RMK202_analysis/plot/histo_protspacerGenes.svg",width=6,height=4)
histo_protspacerGenes
 dev.off()
```



#####spacer location

Here, I want to look where the Protospacers are located

```{r}


library(genoPlotR)
library(tidyverse)

##-------------------------
##ary1
##-------------------------
# library(readr)
# ary1_genoPlotR <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/genoPlotR/arrayFiles/arrayFiles/ary1_genoPlotR.txt", "\t", escape_double = FALSE, trim_ws = TRUE)  %>% as.data.frame() #%>% filter(name!="spacer") #%>% select(-fill)
# dna_seg1 <- dna_seg(ary1_genoPlotR)
# dna_segs <- list(dna_seg1)
# plot_gene_map(dna_segs=dna_segs)

##-------------------------
##Streptococcus_phage_1
##-------------------------

PROTOSPACER_Streptococcus_phage_1_genoplotR <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract/CRISPRspacerBLAST/blast/GenotplotR/GenotplotR/CRISPRblast_rmk202phages_Streptococcus_phage_1_genoplotR.txt",  "\t", escape_double = FALSE,  trim_ws = TRUE)

colors <- colorRampPalette(c("red", "white"))(n = 10)
PROTOSPACER_Streptococcus_phage_1_genoplotR$gene_type <- "blocks"
table(PROTOSPACER_Streptococcus_phage_1_genoplotR$fill)
PROTOSPACER_Streptococcus_phage_1_genoplotR$fill <- PROTOSPACER_Streptococcus_phage_1_genoplotR$fill %>% as.character()
PROTOSPACER_Streptococcus_phage_1_genoplotR$fill <- revalue(PROTOSPACER_Streptococcus_phage_1_genoplotR$fill, c("0"=colors[1], "1"=colors[2], "2"=colors[3], "3"=colors[4]))

dna_seg1 <- dna_seg(PROTOSPACER_Streptococcus_phage_1_genoplotR)
dna_segs <- list(dna_seg1)
plot_gene_map(dna_segs=dna_segs)

ggplot(PROTOSPACER_Streptococcus_phage_1_genoplotR,aes(x=start))+geom_density()+theme_classic()

##-------------------------
##Streptococcus_phage_2
##-------------------------

PROTOSPACER_Streptococcus_phage_2_genoplotR <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract/CRISPRspacerBLAST/blast/GenotplotR/GenotplotR/CRISPRblast_rmk202phages_Streptococcus_phage_2_genoplotR.txt",  "\t", escape_double = FALSE,  trim_ws = TRUE)

colors <- colorRampPalette(c("red", "white"))(n = 10)
PROTOSPACER_Streptococcus_phage_2_genoplotR$gene_type <- "blocks"
table(PROTOSPACER_Streptococcus_phage_2_genoplotR$fill)
PROTOSPACER_Streptococcus_phage_2_genoplotR$fill <- PROTOSPACER_Streptococcus_phage_2_genoplotR$fill %>% as.character()
PROTOSPACER_Streptococcus_phage_2_genoplotR$fill <- revalue(PROTOSPACER_Streptococcus_phage_2_genoplotR$fill, c("0"=colors[1], "1"=colors[2], "2"=colors[3], "3"=colors[4]))

dna_seg2 <- dna_seg(PROTOSPACER_Streptococcus_phage_2_genoplotR)
dna_segs2 <- list(dna_seg2)
plot_gene_map(dna_segs=dna_segs2)

ggplot(PROTOSPACER_Streptococcus_phage_2_genoplotR,aes(x=start))+geom_density()+theme_classic()

```



####map meta

Here we map all raw reads against the Pan-CRISPR-array and follow the mapping coverage over time.

What we do:
1. extend the spacer with the appropiate spacer 
2. make a bwa index all all pan-CRISPR-arrays
3. map all timepoints against the pan-CRISPR-array

```{bash}
###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples_withEvolution.txt  ##the file with all sample names
#logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract/
#Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/FASTA/S_thermophilus_alltogether.CRISPRlocation.fasta
species=Sterm
repeatsFile=/home/vincent/Projects/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt
MetaspacerFile=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers/ForDADA2_woREverse//all_unique_spacers.fasta
mappingFile=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed_ClusterName.txt
  #BaseLocation_mapping=/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion_wOLDStrains/bwaMapping
  BaseLocation_mapping=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//bwaMapping
metaSpacerFile=${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/uniqueSpacers_count_Ref_wOLDstrains.txt #inlcudes in column four the name of array



  
 # ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable_final.fasta
  
  
##=============================================
##-----------create mappable and blastable file
##=============================================
  
  
#scp ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/ForDADA2/spacersAssigned.txt vincent@130.223.51.116:/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//FinalSpacers/ForDADA2/


##--------------------------------------------------
##--------------------------------------------------
rm ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable.fasta
rm ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable.bed
rm ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable_final.fasta
#rm ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_blastable_final.fasta
mkdir -p ${BaseLocation}/FinalSpacers/cdHit_spacers/
#spacerss=$(echo "spacer_2")
#for spacerss in $(cut -f 1 /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//FinalSpacers/ForDADA2/spacersAssigned.txt)
#do
for spacerss in $(grep "^META_" ${metaSpacerFile} |cut -f 1)
do


#grep ">" ${BaseLocation}/FinalSpacers/cdHit_spacers/final/clusterCRISPRs_allArrays_keeps.txt|cut -d ':' -f 2|cut -d 's' -f 1|sed 's/0//g' |sed 's/R//g' | sed 's/a/Array/g'
adap=$(grep -P "${spacerss}\t" ${metaSpacerFile} |cut -f 4|sed 's/a/Array/g' )
adapSeq=$(grep "${adap}" -A 1 /home/vincent/Projects/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt|tail -1)
startSpacer=$(echo $adapSeq|wc -c)


echo "=========================="
echo $adap
echo $adapSeq
echo "=========================="



echo $spacerss

seq=$(grep "$spacerss" -A 1 ${mappingFile} |tail -1)

echo -e ">"$spacerss"\n"$adapSeq"\n"$seq"\n"$adapSeq >> ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable.fasta


LENGTHpacer=$(echo $seq|wc -c)
endspacer=$((startSpacer+LENGTHpacer))
spacershort=$(echo $spacerss |sed 's/>//g')


echo -e $spacershort"\t"$startSpacer"\t"$endspacer"\t"$spacershort >> ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable.bed



done #spacerss (spacer)


fasta_formatter -w 50 -i ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable.fasta -o ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable_final.fasta

## Qualtiy control

grep ">" -c ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable_final.fasta
grep ">" ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable_final.fasta | cut -d ':' -f 1  |sort|uniq -c
grep ">" ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable_final.fasta | cut -d ':' -f 2  |sort|uniq -c
grep ">" -A 1 ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable.fasta |grep ">" -v|grep "-" -v|sort|uniq -c

  mappingFile=${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable_final.fasta
    ##=============================================
  ##mapping to reference
    ##=============================================
  ##-----------------------------------------map rawReads
  rm -r $BaseLocation_mapping
  mkdir -p $BaseLocation_mapping

  bwa index $mappingFile
  # name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 
  
  mkdir -p ${logFilelocation}
  rm ${BaseLocation_mapping}/log/MappingQualityLength.txt
  
  num=1
  rm ${BaseLocation_mapping}/log/flagstat_logfile_CRISPRmappingGenome.txt
  rm ${BaseLocation_mapping}/log/multiqc_logfile_CRISPRmappingGenome.txt
  
  names=G4_6_18
r1=/home/vincent/Projects/2020_StarterEvolution/01_data/20200929_Novogene/X204SC20090774-Z01-F001/trimm_Galore/gz/${names}/${names}-R1_val_1.fq.gz
r2=/home/vincent/Projects/2020_StarterEvolution/01_data/20200929_Novogene/X204SC20090774-Z01-F001/trimm_Galore/gz/${names}/${names}-R2_val_2.fq.gz

  for names in $(cat ${samplesss})
    do
    echo ${num}"/16  :" ${names}
    num=$((num+1))
    rm -r ${BaseLocation_mapping}/${names}/bwaMapping2DB/
  
    mkdir -p ${BaseLocation_mapping}/${names}/bwaMapping2DB/
    
    ##------only for g4_6_18 because the bam file is corrupted
    names=G4_6_18
    bwa mem -t 37 $mappingFile \
    /home/vincent/Projects/2020_StarterEvolution/01_data/20200929_Novogene/X204SC20090774-Z01-F001/trimm_Galore/gz/${names}/${names}-R1_val_1.fq /home/vincent/Projects/2020_StarterEvolution/01_data/20200929_Novogene/X204SC20090774-Z01-F001/trimm_Galore/gz/${names}/${names}-R2_val_2.fq | samtools sort -@${threads} -O BAM -o ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 
  
    
    #bwa mem -t ${threads} $mappingFile \
   # /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_1.fastq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_2.fastq | samtools sort -@${threads} -O BAM -o ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 
  
  samtools view -b -F 4 ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped.bam
  
  ##-------------------coverage
  
  
    samtools view -h -q 30 -b ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped.bam > ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30.bam
   
  ##----take only mappings longer than 42 
  
  mkdir -p ${BaseLocation_mapping}/log/
  
  samtools view ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30.bam |cut -f 13 |sed 's/MD:Z://g'  | awk -F "[CTAG]" '{print $1+$2+$3+$4+$5+$6+$7+$8}' |sort -rn |uniq -c | sed -e 's/^[[:space:]]*//' |awk -F " " -v var="$names" '{OFS=" "} {print $0,var}' >> ${BaseLocation_mapping}/log/MappingQualityLength.txt
  
  
  #samtools view ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped.bam |grep "M02102:390:000000000-C7G6T:1:1102:13014:15583:N:0:GGACTCCT+GTAAGGAG#0"
  
  ###-----------------ONLY spacer mapped reads --> reads mapping to protospacer
    cd ${BaseLocation_mapping}/${names}/bwaMapping2DB/

  samtools view ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30.bam > ${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub.sam

  /home/vincent/scripts/mappingLengthfiltering_protospacer.R -i ${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub.sam -o ${names}_rawIllumina_bwamem_sorted_mapped_protospacer.sam

  
  samtools view -H ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped.bam  > ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_header.sam  # extract header only
  
  awk '{OFS="\t"} {print $1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14}' ${names}_rawIllumina_bwamem_sorted_mapped_protospacer.sam > ${names}_rawIllumina_bwamem_sorted_mapped_protospacer_02.sam
  cat ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_header.sam \
   ${names}_rawIllumina_bwamem_sorted_mapped_protospacer_02.sam |samtools view -S -b > ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_protospacer_final.bam
  
  
  bedtools coverage -bed -a ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable.bed -b ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_protospacer_final.bam  >  ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_protospacer_coverage.bed 
    
    mkdir -p ${BaseLocation_mapping}/final
    
  awk -F "\t" -v samplessss="$names" '{OFS="\t"} {print $0,samplessss}' ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_protospacer_coverage.bed   >> ${BaseLocation_mapping}/final/CRISPR_spacer_coverage_protospacer.bed
  


  ###-----------------properly mapped reads --> reads mapping to CRISPR SPACER
  cd ${BaseLocation_mapping}/${names}/bwaMapping2DB/
  
  /home/vincent/scripts/mappingLengthfiltering.R -i ${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub.sam -o ${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub_out.sam
  rm ${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub.sam
  
  #samtools view ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30.bam |cut -f 13 |sed 's/MD:Z://g'  | awk -F "[CTAG]" '{if($1+$2+$3+$4+$5+$6+$7+$8>42) print $0}' |sort -rn |uniq -c | sed -e 's/^[[:space:]]*//' |awk -F " " -v var="$names" '{OFS=" "} {print $0,var}' >> ${BaseLocation_mapping}/log/MappingQualityLength.txt
  
  
  samtools view -H ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped.bam > ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_header.sam  # extract header only
  
  awk '{OFS="\t"} {print $1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14}' ${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub_out.sam > ${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub_out_02.sam
  cat ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_header.sam \
   ${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub_out_02.sam |samtools view -S -b > ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub_out_final.bam
  
  
  bedtools coverage -bed -a ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable.bed -b ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub_out_final.bam  >  ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_coverage.bed 
    
    mkdir -p ${BaseLocation_mapping}/final
    
  awk -F "\t" -v samplessss="$names" '{OFS="\t"} {print $0,samplessss}' ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_coverage.bed  >> ${BaseLocation_mapping}/final/CRISPR_spacer_coverage.bed
  
  ##-----Qualimap
  
  mkdir -p ${BaseLocation_mapping}/${names}/bwaMapping2DB//bamqc
  
  qualimap bamqc -bam ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir ${BaseLocation_mapping}/${names}/bwaMapping2DB//bamqc --java-mem-size=80G
  
  echo ${names} >>  ${BaseLocation_mapping}/log/flagstat_logfile_CRISPRmappingGenome.txt
  samtools flagstat ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam >>  ${BaseLocation_mapping}/log/flagstat_logfile_CRISPRmappingGenome.txt
  
  rm ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam &
   mkdir -p ${BaseLocation_mapping}/log
  echo ${BaseLocation_mapping}"/"${names}"/bwaMapping2DB/bamqc" >> ${BaseLocation_mapping}/log/multiqc_logfile_CRISPRmappingGenome.txt
  
  done
  
  
  
     rm -r ${BaseLocation}/log/MultiQC
   
    mkdir -p ${BaseLocation}/log/MultiQC
    
    /home/vincent/.local/bin/multiqc --config /archiv/logs/multiQC_config.txt --file-list  ${BaseLocation_mapping}/log/multiqc_logfile_CRISPRmappingGenome.txt \
      -o ${BaseLocation}/log/MultiQC

##-------------------calculate mapping reads for normalization
rm ${BaseLocation_mapping}/log/reads4normalization.txt
num=1
names=RMK202
for names in $(cat ${samplesss})
  do
  echo ${num}"/16  :" ${names}

#reads=$(grep "CP046134" /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/${names}/bwaMapping2DB/bamqc/genome_results.txt | sed -e 's/^[[:space:]]*//'|cut -f 3)

reads=$(samtools view ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub_out_final.bam |wc -l)

echo -e ${names}"\t"${reads}"\tStreptococcus thermophilus"  >> ${BaseLocation_mapping}/log/reads4normalization.txt

done
echo ${BaseLocation_mapping}/final/CRISPR_spacer_coverage.bed
echo ${BaseLocation_mapping}/log/reads4normalization.txt
```


```{bash}
###------------------transfer local

#scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion_wOLDStrains/bwaMapping/final/CRISPR_spacer_coverage.bed ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/CRISPR_spacer_coverage_wOldStrains.bed
mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//bwaMapping/final/
scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//bwaMapping/final/CRISPR_spacer_coverage.bed ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//bwaMapping/final/CRISPR_spacer_coverage.bed


#scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion_wOLDStrains/bwaMapping/final/CRISPR_spacer_coverage_protospacer.bed ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/CRISPR_spacer_coverage_protospacer_wOldStrains.bed

scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//bwaMapping/final//CRISPR_spacer_coverage_protospacer.bed ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//bwaMapping/final

#scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/log/MultiQC/multiqc_report.html ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/

#scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion_wOLDStrains/bwaMapping/log/reads4normalization.txt ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/
mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//bwaMapping/log
scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//bwaMapping/log/reads4normalization.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//bwaMapping/log

scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//bwaMapping/log/MappingQualityLength.txt ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/


#scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/bwaMapping/RMK202/bwaMapping2DB/RMK202_rawIllumina_bwamem_sorted_mapped_mapQ_30.bam ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/

#scp  /home/vincent/apps/mappingLengthfiltering.R vincent@130.223.51.116:/home/vincent/scripts/
```

#####Map length evaluation
Here, we check out the read length mapping. This is only QC.
Based on this I filtered for reads only mapping longer than the spacer in order to avoid count reads that map to the phage and not the genome. 

all spacers that are represented by the local phages are very highly abundant
- most likely we have also protospacers read mappings
we need to find a method to split protospacer and spacer hits 
--> therefore I only take mappings that are longer than 42 (which is the longest spacer length)
with this i ensure that I do not have protospacer mappings but also some repeat mapping


```{r}
#samtools view ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30.bam |cut -f 13 |sed 's/MD:Z://g'  | awk -F "[CTAG]" '{if($1+$2+$3+$4+$5+$6+$7+$8>42) print $0}' |sort -rn |uniq -c | sed -e 's/^[[:space:]]*//' |awk -F " " -v var="$names" '{OFS=" "} {print $0,var}' >> ${BaseLocation_mapping}/log/MappingQualityLength.txt
###-------------------------look at mapping length
library(readr)
library(patchwork)
  
MappingQualityLength <- read_table2("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/MappingQualityLength.txt",  col_names = c("count","length","sample"))
  
range(MappingQualityLength$length)
mappinglengthPlot <- ggplot(MappingQualityLength,aes(x=length,y=count,color=sample,fill=sample))+geom_bar(stat="identity")+theme_classic()+geom_vline(xintercept = 42)
  mappinglengthPlot
MappingQualityLength_long <- MappingQualityLength[which(MappingQualityLength$length>42),]
ggplot(MappingQualityLength_long,aes(x=length,y=count,color=sample,fill=sample))+geom_bar(stat="identity")+theme_classic()

table(MappingQualityLength_long$sample)

MappingQualityLength$keeping <- ifelse(MappingQualityLength$length>42,"keep","remove")
MappingQualityLength_keeping <- aggregate(. ~sample+keeping, data=MappingQualityLength[,c("sample","count","keeping")], sum, na.rm=TRUE)

MappingQualityLength_keeping$keeping = factor(MappingQualityLength_keeping$keeping, levels=c("remove","keep"))
# colorsss <- (c("darkcyan","darkturquoise","lightblue","lightgray"))


keepingPlot <- ggplot(MappingQualityLength_keeping,aes(x=sample,y=count,color=keeping,fill=keeping))+geom_bar(stat="identity")+theme_classic()+labs(title = "42bp mapping thresehold",x="")+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9))
keepingPlot


 
  svg("~/Desktop/Projects/2019_RMK202_analysis/plot/mappinglengthplot.svg",width=6,height=3)

(mappinglengthPlot|keepingPlot)
# (p1|p2|p3)+plot_layout(widths =c(2,4,2))

 dev.off()

# nrow(CRISPR_spacer_coverage_extended_wide)
```


##Vcontact with metagenome

Here, I want to add my metagenomes to the current vContact analysis.
First we try to add the entire metagenome at once. (next time add contig by contig)

First we have to do the Phage annotation. For this project I did it with Phaster. 

```{bash}

mkdir -p /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/
scp -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/ vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/

```


```{bash}
##================
##files
##================

date=20190519
species=Sterm
date_new=20200424

mkdir -p /archiv/Phage_DB/BLAST/${date_new}/downloads/${species}_proteins/Renamed/merged/

cat  /archiv/Phage_DB/BLAST/${date}/downloads/${species}_proteins/Renamed/merged/${species}_phage_protein_db_gene2genome.txt > \
/archiv/Phage_DB/BLAST/${date_new}/downloads/${species}_proteins/Renamed/merged/${species}_phage_protein_db_gene2genome_oldsamples.txt


mkdir -p /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/data

###----------------------
##add phages
###----------------------
rm -r /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/ForVContact2/
mkdir -p /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/ForVContact2/
for phageesss in $(cut -f 1 /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/gene_CLEANED.gff|sort|uniq )
do

echo -e "wokring on: "${phageesss}
echo "..."

###----faa
#grep "," /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/gene_CLEANED.gff

sed 's/,//g' /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/gene_CLEANED.gff |grep "${phageesss}" | cut -f 9 | awk -F ";locusTag= " -v Genomesss="$phageesss" '{OFS=","}{print $2,"RMK202_"Genomesss,$1}'  >>  /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/ForVContact2/RMK202_phage_protein_db_gene2genome.txt

cat /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/faa_for_virfam/${phageesss}.faa >>  /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/ForVContact2/all_Genes_merged.faa

done

##---------------------------
##merge with existing data
##---------------------------

mkdir -p /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/data/

  cat /archiv/Phage_DB/BLAST/${date}/Vcontact/Database/alltogether/data/all_phage_protein_db_gene2genome.csv \
  /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/ForVContact2/RMK202_phage_protein_db_gene2genome.txt > \
  /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/data/all_phage_protein_db_gene2genome.csv
  
##prep fasta  


cat /archiv/Phage_DB/BLAST/${date}/Vcontact/Database/alltogether/data/merged_all_phages.faa \
/archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/ForVContact2/all_Genes_merged.faa > /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/data/merged_all_phages.faa
  
##================
##run vcontact
##================

##before running activate environoment
/home/vincent/anaconda3/bin/conda init  bash
conda activate /home/vincent/miniconda2/envs/vContact2

##if it doesnt work just try another screen ...has been playing up alot

cd /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/data

#rm -r /archiv/Phage_DB/BLAST/${date}/Vcontact/Database/alltogether/database
rm -r /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/database
mkdir -p /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/database

/home/vincent/apps/vcontact2/bin/vcontact --raw-proteins merged_all_phages.faa --rel-mode Diamond --proteins-fp all_phage_protein_db_gene2genome.csv --db ProkaryoticViralRefSeq88-Merged --pcs-mode MCL --vcs-mode ClusterONE --c1-bin /home/vincent/apps/cluster_one-1.0.jar -t 30 --output-dir /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/database


###===========
##move important files to one folder
###===========
mkdir -p /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/database/allPhages_vContact
cd /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/database/

cp c1.clusters allPhages_vContact/
cp c1.ntw allPhages_vContact/
cp genome_by_genome_overview.csv allPhages_vContact/
cp modules.ntwk allPhages_vContact/
cp viral_cluster_overview.csv allPhages_vContact/
```



```{bash}
##================
##files
##================

date=20190519
species=Ldel
date_new=20201106

mkdir -p /archiv/Phage_DB/BLAST/${date_new}/downloads/${species}_proteins/Renamed/merged/

cat  /archiv/Phage_DB/BLAST/${date}/downloads/${species}_proteins/Renamed/merged/${species}_phage_protein_db_gene2genome.txt > \
/archiv/Phage_DB/BLAST/${date_new}/downloads/${species}_proteins/Renamed/merged/${species}_phage_protein_db_gene2genome_oldsamples.txt


mkdir -p /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/data

###----------------------
##add phages
###----------------------
rm -r /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/${date_new}_ForVContact2/
mkdir -p /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/${date_new}_ForVContact2/
#for phageesss in $(cut -f 1 /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/gene_CLEANED.gff|sort|uniq )
for phageesss in $(echo "Lactobacillus_phage_1")
do

echo -e "wokring on: "${phageesss}
echo "..."

###----faa
#grep "," /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/gene_CLEANED.gff

sed 's/,//g' /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/gene_CLEANED.gff |grep "${phageesss}" | cut -f 9 | awk -F ";locusTag= " -v Genomesss="$phageesss" '{OFS=","}{print $2,"RMK202_"Genomesss,$1}'  >>  /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/${date_new}_ForVContact2/RMK202_phage_protein_db_gene2genome.txt

cat /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/faa_for_virfam/${phageesss}.faa >>  /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/${date_new}_ForVContact2/all_Genes_merged.faa

done

##---------------------------
##merge with existing data
##---------------------------

mkdir -p /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/data/

  cat /archiv/Phage_DB/BLAST/${date}/Vcontact/Database/alltogether/data/all_phage_protein_db_gene2genome.csv \
  /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/${date_new}_ForVContact2/RMK202_phage_protein_db_gene2genome.txt > \
  /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/data/all_phage_protein_db_gene2genome.csv
  
##prep fasta  


cat /archiv/Phage_DB/BLAST/${date}/Vcontact/Database/alltogether/data/merged_all_phages.faa \
/archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/${date_new}_ForVContact2/all_Genes_merged.faa > /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/data/merged_all_phages.faa
  
##================
##run vcontact
##================

##before running activate environoment
/home/vincent/anaconda3/bin/conda init  bash
conda activate /home/vincent/miniconda2/envs/vContact2

##if it doesnt work just try another screen ...has been playing up alot

cd /archiv/Phage_DB/BLAST/20200424/Vcontact/Database/alltogether/data

#rm -r /archiv/Phage_DB/BLAST/${date}/Vcontact/Database/alltogether/database
rm -r /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/database
mkdir -p /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/database

/home/vincent/apps/vcontact2/bin/vcontact --raw-proteins merged_all_phages.faa --rel-mode Diamond --proteins-fp all_phage_protein_db_gene2genome.csv --db ProkaryoticViralRefSeq88-Merged --pcs-mode MCL --vcs-mode ClusterONE --c1-bin /home/vincent/apps/cluster_one-1.0.jar -t 30 --output-dir /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/database


###===========
##move important files to one folder
###===========
mkdir -p /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/database/allPhages_vContact
cd /archiv/Phage_DB/BLAST/${date_new}/Vcontact/Database/alltogether/database/

cp c1.clusters allPhages_vContact/
cp c1.ntw allPhages_vContact/
cp genome_by_genome_overview.csv allPhages_vContact/
cp modules.ntwk allPhages_vContact/
cp viral_cluster_overview.csv allPhages_vContact/
```


```{bash}
###===========
##move local
###===========

##Mahony phages (only once)
#scp vincent@130.223.51.151:/archiv/Phage_DB/Mahony/table_ncbi_genomes.txt /home/vincent/Desktop/Projects/2019_PhageDB/Vcontact/cluster/Mahony_table_ncbi_genomes.txt


##taking from vcontact folder
mkdir -p /home/vincent/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03/
scp -r vincent@130.223.51.116:/archiv/Phage_DB/BLAST/20200122/Vcontact/Database/alltogether/database/allPhages_vContact /home/vincent/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03/

```


#### Vcontact cluster analysis
Here, we do the clusteranylasis of the Vcontact output
##### Ldel
```{r}


library(ggplot2)
library(readr)
library(igraph)
library(plyr)
library(tidyverse)
#---------------
##import data
#---------------
###---------All phages
 Phages_names <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03//allPhages_vContact/genome_by_genome_overview.csv")

Sterm_cluster <- read_table2("~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03//allPhages_vContact/c1.ntw", 
    col_names = c("from","to","power"))

# Sterm_cluster <- read_table2("~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/c1.ntw", 
#     col_names = c("from","to","power"))

# 
# ##all contig names
# Sterm_contigs <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact/cluster/Sterm_contigs.ntw", 
#     col_names = "contigs")
# ##all clusterings
# Sterm_cluster <- read_table2("~/Desktop/Projects/2019_PhageDB/Vcontact/cluster/Sterm_cluster.ntw", 
#     col_names = c("from","to","power"))
# Mahony_genomes <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact/cluster/Mahony_table_ncbi_genomes.txt",
    # col_names = c("Genus","phage","strain","NCBI","Type"))

#---------------
##reduce to only Strep clusters
#---------------

#----keep cluster that contain one sterm!!

Sterm_clustering_complete <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03///allPhages_vContact/genome_by_genome_overview.csv")
# Sterm_clustering_complete <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/genome_by_genome_overview.csv")


# Sterm_clustering_new <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/c1.clusters") %>% as.data.frame()
Sterm_clustering_new <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03//allPhages_vContact/c1.clusters") %>% as.data.frame()
Sterm_clustering_forFiltering_new <- Sterm_clustering_new  %>% filter(., grepl('Lactobacillus_phage_1', Members,ignore.case = TRUE))

Sterm_clustering_unnested <- Sterm_clustering_forFiltering_new %>% 
    mutate(Members = strsplit(as.character(Members), " ")) %>% 
    unnest(Members)

str_split_fixed(Sterm_clustering_unnested$Members, "~", 3)[,1]  %>% table()


###---------------1. filter:
##clusters that contain a Streptococcus


nrow(Sterm_clustering_unnested)
nrow(Sterm_cluster)
Sterm_cluster_sub <- Sterm_cluster[which(Sterm_cluster$from %in% Sterm_clustering_unnested$Members | Sterm_cluster$to %in% Sterm_clustering_unnested$Members),]
nrow(Sterm_cluster_sub)
# Sterm_cluster_sub <- Sterm_cluster_sub[which(Sterm_cluster_sub$to %in% Sterm_clustering_unnested$Members),]
# nrow(Sterm_cluster_sub)

###---------------2. filter:
##connections that have a higher qualtiy than 50 (decided on distribution plot)

ggplot(Sterm_cluster,aes(x=power))+geom_histogram(binwidth=1)
Sterm_cluster_sub <- Sterm_cluster_sub[Sterm_cluster_sub$power>1,]
nrow(Sterm_cluster_sub)

###---------------3. filter:
##cluster containing more than 2 phages 

Sterm_cluster_sub_uniques <- unique(c(Sterm_cluster_sub$from,Sterm_cluster_sub$to))

Sterm_clustering_unnested_which <- 
  Sterm_clustering_complete[which(Sterm_clustering_complete$Genome %in% Sterm_cluster_sub_uniques),] %>% select(VC) %>% table( )
  # select_if(~n_distinct(.) > 2 & is.numeric(.)) %>% 
  names(Sterm_clustering_unnested_which)
  Sterm_clustering_unnested_which_keep <- Sterm_clustering_unnested_which[which(Sterm_clustering_unnested_which>1)] %>% names()
  Sterm_clustering_unnested_which_keep_genome <- Sterm_clustering_complete[which(Sterm_clustering_complete$VC %in% Sterm_clustering_unnested_which_keep),"Genome"] %>% as.data.frame()
  
  
  # sum(Sterm_cluster_sub$from %in% Sterm_clustering_unnested_which_keep_genome$Genome) 
  nrow(Sterm_cluster_sub)
  Sterm_cluster_sub <- Sterm_cluster_sub[which(Sterm_cluster_sub$from %in% Sterm_clustering_unnested_which_keep_genome$Genome|Sterm_cluster_sub$to %in% Sterm_clustering_unnested_which_keep_genome$Genome),]
  nrow(Sterm_cluster_sub)
  
  unique(Sterm_cluster_sub$from)
# own_cluster_sub_new<- Sterm_cluster_sub
 

 ##===============================
 #old plots
  ##===============================
#---------------
##plot interaction
#---------------
ggplot(Sterm_cluster,aes(x=power))+geom_histogram(binwidth=1)
dim(Sterm_cluster_sub)
# 
# ggplot(Ldel_cluster_sub,aes(x=power))+geom_histogram(binwidth=1)
# dim(Ldel_cluster_sub)
# 
# ggplot(Llac_cluster_sub,aes(x=power))+geom_histogram(binwidth=1)
# dim(Llac_cluster_sub)

#---------------
##subset for only highest interactions
#---------------
#Sterm_cluster_sub <- Sterm_cluster_sub[Sterm_cluster_sub$power>1,]
#dim(Sterm_cluster_sub)

#Sterm_cluster_sub <- Sterm_cluster_sub[Sterm_cluster_sub$power>10,]
#dim(Sterm_cluster_sub)

# length(unique(Sterm_cluster_sub$from))

#---------------
##creat graph
#---------------
# write_csv(Sterm_cluster_sub,path="~/Desktop/Projects/2019_PhageDB/Vcontact//allPhages_vContact/Streptococcus_phage_betwork_connections.txt",col_names = TRUE)

write_csv(Sterm_cluster_sub,path="~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03///allPhages_vContact/Ldel_phage_betwork_connections.txt",col_names = TRUE)

Sterm_net <- graph_from_data_frame(Sterm_cluster_sub,direct=F)
# Ldel_net <- graph_from_data_frame(Ldel_cluster_sub,direct=F)
# Llac_net <- graph_from_data_frame(Llac_cluster_sub,direct=F)

# dim(Sterm_net)

##==================================
 ##new analysis
  ##==================================
 library(ggnetwork)
 library(ITNr)
 library(intergraph)
 
 net<-asNetwork(Sterm_net) #ggnetwork requires a network object
 n<-ggnetwork(net)
  ##------------------
 ##color and size change
  ##------------------
 
 

###--------
##coloring
#---------------
# n[grep("RMK202_",n$vertex.names),"vertex.names"] %>% table()
 n$color <- as.character("Viral DB")
 n$color[grep("RMK202_",n$vertex.names)] <- "RMK202 phages"

 
 
 n$color = factor(n$color, levels=c("RMK202 phages","Viral DB"))
 colorzzz <- c("red","grey50")
 size <- c("4","1.5")
 ##------------------
 ##extend
 ##------------------
 
 nExtended <-  merge(n, Phages_names, by.y="Genome",by.x="vertex.names",all.x = TRUE)
 # nrow(n)
 # nExtended <-  merge(n, Phages_names, by.y="Genome",by.x="vertex.names",all.x = TRUE) 
 # nrow(nExtended)
 # nExtended <- n
 # ccolnames(nExtended)
colnames(nExtended)
colnames(nExtended) <- revalue(colnames(nExtended), c("Adj P-value"="Adj_P_value","Genera in VC"="Genera_in_VC","Topology Confidence Score"="Topology_Confidence_Score","Genus Confidence Score"="Genus_Confidence_Score"))

# nExtended$ad
 ##------------------
 ##plot
 ##------------------
 networkplot <- ggplot(nExtended, aes(x = x, y = y, xend = xend, yend = yend,text =paste("genome:", vertex.names,"\npower: ",power,"\nOrder: ",Order,"\nFamily: ",Family,"\nGenus: ",Genus,"\nVC: ",VC,"\nSize: ",Size,"\nAdj P-value: ",Adj_P_value,"\nGenera in VC: ",Genera_in_VC,"\nGenus_Confidence_ScoreC: ",Genus_Confidence_Score,"\nTopology_Confidence_Score: ",Topology_Confidence_Score))) + geom_edges(color = "lightgray") +
   geom_nodes(aes(color = as.factor(color),size=as.factor(color))) +
   # scale_fill_manual(values="gray41")+
    scale_color_manual(values=colorzzz)+
    scale_size_manual(values=as.numeric(size))+
   # guides(color=FALSE,size=FALSE)+ #Remove legends
 theme_blank()+theme(legend.title = element_blank())
 networkplot
 
 library(plotly)
 ggp <- ggplotly(networkplot)
 #htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03//allPhages_vContact//networkPlot.html")
 # View(nExtended)
 
tmp <-   nExtended %>% filter(color=="RMK202 phages") 
table(tmp$`VC Subcluster`)
 
 
 nExtended %>% filter(color!="Viral DB") %>% filter()
 sort(table(nExtended$VC),decreasing = TRUE)
 
 count_perViralCluster <- nExtended %>%
  group_by(VC,color) %>% 
  dplyr::summarise(count=n()) %>% spread(.,color,count)
 
  ##==================================
 ##keep only phages in our s. thermophilus phage cluster
  ##==================================
 ###---------------------identify phages in our cluster

 
# table(nExtended$VC) 

tmp_phages <- nExtended[grep("RMK202_",nExtended$vertex.names),] 
ourPhageCluster <- c(unique(tmp_phages$VC))
nExtended_ourphages <- nExtended %>% filter(VC %in% ourPhageCluster)

phagesinClusters <- Phages_names %>% filter(VC %in% ourPhageCluster)
phagesinClusters$Genome

###---------------------clustering
# Sterm_clustering_new <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/c1.clusters") %>% as.data.frame()
Sterm_clustering_new <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03//allPhages_vContact/c1.clusters") %>% as.data.frame()
Sterm_clustering_forFiltering_new <- Sterm_clustering_new  %>% filter(., grepl('Streptococcus', Members,ignore.case = TRUE))

Sterm_clustering_unnested <- Sterm_clustering_forFiltering_new %>% 
    mutate(Members = strsplit(as.character(Members), " ")) %>% 
    unnest(Members)

# str_split_fixed(Sterm_clustering_unnested$Members, "~", 3)[,1]  %>% table()


###---------------1. filter:
##clusters that contain a Streptococcus


# nrow(Sterm_clustering_unnested)
nrow(Sterm_cluster)
Sterm_cluster_sub <- Sterm_cluster[which(Sterm_cluster$from %in% phagesinClusters$Genome | Sterm_cluster$to %in% phagesinClusters$Genome),]
nrow(Sterm_cluster_sub)
Sterm_net_sub <- graph_from_data_frame(Sterm_cluster_sub,direct=F)

 

 net<-asNetwork(Sterm_net_sub) #ggnetwork requires a network object
 n<-ggnetwork(net)
  ##------------------
 ##color and size change
  ##------------------
 
 
###--------
##coloring
#---------------
# n[grep("RMK202_",n$vertex.names),"vertex.names"] %>% table()
 n$color <- as.character("Viral DB")
 n$color[grep("RMK202_",n$vertex.names)] <- "RMK202 phages"


 
 n$color = factor(n$color, levels=c("RMK202 phages","Viral DB"))
 colorzzz <- c("red","grey50")
 size <- c("4","1.5")
 
 
 # n$color = factor(n$color, levels=c("RMK202 phages","cos","987","Viral DB"))
 # colorzzz <- c("red","blue","#808000","grey50")
 # size <- c("4","2","2","1.5")
 
 table(n$color)
 
 n$color_new <- as.character("Viral DB")
n$color_new[grep("RMK202_Lactobacillus_phage_1",n$vertex.names)[1]] <- "RMK202 phages"
# n$color_new[grep("RMK202_Streptococcus_phage_2",n$vertex.names)[1]] <- "RMK202 phages"


 ##------------------
 ##extend
 ##------------------
 
 nExtended <-  merge(n, Phages_names, by.y="Genome",by.x="vertex.names",all.x = TRUE)
 # nrow(n)
 # nExtended <-  merge(n, Phages_names, by.y="Genome",by.x="vertex.names",all.x = TRUE) 
 # nrow(nExtended)
 # nExtended <- n
 # ccolnames(nExtended)
colnames(nExtended)
colnames(nExtended) <- revalue(colnames(nExtended), c("Adj P-value"="Adj_P_value","Genera in VC"="Genera_in_VC","Topology Confidence Score"="Topology_Confidence_Score","Genus Confidence Score"="Genus_Confidence_Score"))



 nExtended$alpha <- as.character("Viral DB")
nExtended$alpha[nExtended$VC %in% ourPhageCluster] <- "RMK202 phages"
table(nExtended$alpha)
  alphaCol <- c(1,0.03)
  
tmp <- nExtended[nExtended$alpha=="RMK202 phages",]
table(tmp$VC)
  
nExtended$fillings <- as.character("Viral DB")
# nExtended$fillings[nExtended$VC %in% ourPhageCluster[1]] <- ourPhageCluster[1]
# nExtended$fillings[nExtended$VC %in% ourPhageCluster[2]] <- ourPhageCluster[2]
# # nExtended$fillings[nExtended$vertex.names %in% Mahony_genomes_cos$namez] <- "cos"
# nExtended$fillings[nExtended$vertex.names %in% Mahony_genomes_987$namez] <- "987"
# nExtended$fillings[nExtended$vertex.names=="RMK202_Streptococcus_phage_1"] <- "Streptococcus_phage_1"
# nExtended$fillings[nExtended$vertex.names=="RMK202_Streptococcus_phage_2"] <- "Streptococcus_phage_2"
as.factor(nExtended$fillings)
# library("RColorBrewer")

# three_reds <- c(brewer.pal(9,name="YlOrRd")[c(4,6)],
# 							 brewer.pal(3,name="Reds")[3])
# three_browns <- brewer.pal(9,name="YlOrBr")[c(7,8,9)]
# three_blues <- brewer.pal(9,name="YlGnBu")[c(5,6,8)]

filling <- c("red")
# filling <- c(three_reds[1],three_browns[1],three_reds[2],three_browns[2],three_reds[3],three_browns[3],"grey")

# filling <- c("black","red","darkgrey","darkred","grey")
# table(nExtended$alpha)

  



# nExtended$ad
 ##------------------
 ##plot
 ##------------------
table(nExtended$alpha)
table(nExtended$power)
# nExtended$Family
 networkplot_sub <- ggplot(nExtended, aes(x = x, y = y, xend = xend, yend = yend,text =paste("genome:", vertex.names,"\npower: ",power,"\nOrder: ",Order,"\nFamily: ",Family,"\nGenus: ",Genus,"\nVC: ",VC,"\nSize: ",Size,"\nAdj P-value: ",Adj_P_value,"\nGenera in VC: ",Genera_in_VC,"\nGenus_Confidence_ScoreC: ",Genus_Confidence_Score,"\nTopology_Confidence_Score: ",Topology_Confidence_Score))) + geom_edges(color = "lightgray",aes(size=power),alpha=1) +
   geom_nodes(aes(color = as.factor(fillings),alpha=as.factor(alpha))) +
   # scale_fill_manual(values="gray41")+
    scale_color_manual(values=filling)+
     scale_alpha_manual(values=alphaCol)+
    scale_fill_manual(values=colorzzz)+
   geom_nodelabel_repel(aes(label = vertex.names),
                       box.padding = unit(1, "lines"),
                       data = function(x) { x[ x$color_new == "RMK202 phages", ]})+
    # scale_size_manual(values=as.numeric(c(2,1.25)))+
   # guides(color=FALSE,size=FALSE)+ #Remove legends
 theme_blank()+theme(legend.title = element_blank())
 networkplot_sub
 table(nExtended$color_new)
 # RMK202 phages
#  x[grep("RMK202_",x$vertex.names),]
#  n$color[grep("RMK202_",n$vertex.names)]
 svg("~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03//allPhages_vContact//networkPlot_subset_ldel.svg",width=7,height=7)
 # png("~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03//allPhages_vContact//networkPlot_subset.png", width = 2800, height = 2800,res=300)
  networkplot_sub
   dev.off()
 
 library(plotly)
 # ggp <- ggplotly(networkplot_sub)
 # htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03//allPhages_vContact//networkPlot_subset.html")
 # View(nExtended)


 ##==================================
 ##old analysis
  ##==================================



###--------
##grep fo Mahony phages
#---------------
Mahony_genomes$namez <- paste(Mahony_genomes$Genus,Mahony_genomes$phage,Mahony_genomes$strain,sep="~")
Mahony_genomes_cos <- Mahony_genomes[which(Mahony_genomes$Type=="cos"),]
Mahony_genomes_pac <- Mahony_genomes[which(Mahony_genomes$Type=="pac"),]
Mahony_genomes_987 <- Mahony_genomes[which(Mahony_genomes$Type=="987"),]
Mahony_genomes_5093 <- Mahony_genomes[which(Mahony_genomes$Type=="5093"),]

##colour

vcolour_Sterm <- rep("grey",length(V(Sterm_net)$name))
# vcolour_Sterm_ori <- V(Sterm_net)$name
# vcolour_Sterm <- "grey"
vcolour_Sterm[which(V(Sterm_net)$name %in% paste(Mahony_genomes_cos$Genus,Mahony_genomes_cos$phage,Mahony_genomes_cos$strain,sep="~"))] <- "green"
vcolour_Sterm[which(V(Sterm_net)$name %in% paste(Mahony_genomes_pac$Genus,Mahony_genomes_pac$phage,Mahony_genomes_pac$strain,sep="~"))] <- "blue"
vcolour_Sterm[which(V(Sterm_net)$name %in% paste(Mahony_genomes_987$Genus,Mahony_genomes_987$phage,Mahony_genomes_987$strain,sep="~"))] <- "pink"
vcolour_Sterm[which(V(Sterm_net)$name %in% paste(Mahony_genomes_5093$Genus,Mahony_genomes_5093$phage,Mahony_genomes_5093$strain,sep="~"))] <- "brown"

# vcolour_Sterm[which(!vcolour_Sterm_ori %in% paste(Mahony_genomes$Genus,Mahony_genomes$phage,Mahony_genomes$strain,sep="~"))] <- "grey"
table(vcolour_Sterm)

V(Sterm_net)$color <- vcolour_Sterm
table(V(Sterm_net)$color)
# V(Sterm_net)$color[which(V(Sterm_net)$name=='Streptococcus~phage~SW19')] <- "grey"
# V(Sterm_net)$color[which(V(Sterm_net)$name=='Streptococcus~phage~SW24')] <- "grey"
##size

vsize_Sterm <- V(Sterm_net)$name
vsize_Sterm_ori <- V(Sterm_net)$name

grep("S_t",V(Sterm_net)$name)
vsize_Sterm[which(!vsize_Sterm_ori %in% paste(Mahony_genomes$Genus,Mahony_genomes$phage,Mahony_genomes$strain,sep="~"))] <- "1.5"
vsize_Sterm[which(vsize_Sterm_ori %in% paste(Mahony_genomes$Genus,Mahony_genomes$phage,Mahony_genomes$strain,sep="~"))] <- "5"

table(vsize_Sterm)

vsize_Sterm <- as.numeric(vsize_Sterm)

#---------------
#plot
#---------------

set.seed(8051)

 V(Sterm_net)$name[grep("RMK202",V(Sterm_net)$name)]
 length( V(Sterm_net)$name)
 # V(Llac_net)$name
 # V(Ldel_net)$name
 # length(V(Ldel_net)$name)


# create vertices
fg_lay_Sterm <- layout_with_fr(Sterm_net,dim=2,niter=99)
# fg_lay_Ldel <- layout_with_fr(Ldel_net,dim=2,niter=99)
# fg_lay_Llac <- layout_with_fr(Llac_net,dim=2,niter=99)

# plot  Sterm
 # plot(Sterm_net,vertex.label=NA,layout=fg_lay_Sterm,
 #     vertex.size=vsize_Sterm,
 #       main="Sterm phages network"
 #     )
     # png("~/Desktop/Projects/2019_RMK202_analysis/plot/vContact_sterm.png", width = 1800, height = 1200,res=300)

 plot(Sterm_net,vertex.label=NA,layout=fg_lay_Sterm,vertex.size=vsize_Sterm,
       main="Sterm phages network"
     )
 legend("topright", 
       legend=c("cos","pac","987","5093","unknown"), pch=19,
      col=c("green","blue","pink","brown","grey"), bty="n")

# dev.off()
 
# # plot  Ldel
#  plot(Ldel_net,vertex.label=NA,layout=fg_lay_Ldel,
#        main="Ldel phages network"
#      )
#  
#  plot(Llac_net,vertex.label=NA,layout=fg_lay_Llac,
#        main="Llac phages network"
#      )
 
 #---------------
 ##creat modules
 #---------------
 # ceb_sterm_high25 <- ceb_sterm
# ceb_sterm <- cluster_edge_betweenness(Sterm_net)
#dendPlot(ceb_ara,mode="hclust")
# plot(ceb_sterm,Sterm_net,vertex.label=NA,vertex.size=vsize_Sterm,vertex.frame.color="red",
#      main="arabidospsis network")

 
#  ##---------Sterm
# ceb_sterm<- cluster_edge_betweenness(Sterm_net)
#   mem_cab_sterm <- membership(ceb_sterm)
#   df_mem_cab_sterml <- split(names(mem_cab_sterm),mem_cab_sterm)
# 
# memberss_sterm <- data.frame(names(mem_cab_sterm),as.double(mem_cab_sterm))
# # dendPlot(ceb_ldel,mode="hclust",hang = 0.0001)
# plot(ceb_sterm,Sterm_net,vertex.label=NA,vertex.frame.color="red",
#      main="sterm with modules network") 
# 
# 
# table(memberss_sterm$as.double.mem_cab_sterm.)
# which(table(memberss_sterm$as.double.mem_cab_sterm.)>2)
# 
# memberss_sterm[which(memberss_sterm$names.mem_cab_sterm. %in% Mahony_genomes_pac$namez),]
# memberss_sterm[which(memberss_sterm$names.mem_cab_sterm. %in% Mahony_genomes_987$namez),]
# memberss_sterm[which(memberss_sterm$names.mem_cab_sterm. %in% Mahony_genomes_cos$namez),]
# memberss_sterm[which(memberss_sterm$names.mem_cab_sterm. %in% Mahony_genomes_5093$namez),]
# 
# keepers_names <- as.double(which(table(memberss_sterm$as.double.mem_cab_sterm.)>2))
# length(as.double(which(table(memberss_sterm$as.double.mem_cab_sterm.)>2)))
# 
# memberss_sterm_subset <- memberss_sterm[which(memberss_sterm$as.double.mem_cab_sterm. %in% keepers_names),]
#   table(memberss_sterm_subset$as.double.mem_cab_sterm.)
# 
# 
#   
# write_csv(memberss_sterm,path="~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/Streptococcus_phage_cutoff50.members",col_names = FALSE)
# write_csv(memberss_sterm_subset,path="~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/Sterm_phage_cutoff55_subset.members",col_names = FALSE)
# 


# 
# memberss_sterm$names.mem_cab_sterm. 
# Mahony_genomes_pac$namez
# 
#   ##---------ldel
#   ceb_ldel <- cluster_edge_betweenness(Ldel_net)
#   mem_cab_ldel <- membership(ceb_ldel)
#   df_mem_cab_Ldel <- split(names(mem_cab_ldel),mem_cab_ldel)
# 
# memberss_ldel <- data.frame(names(mem_cab_ldel),as.double(mem_cab_ldel))
# # dendPlot(ceb_ldel,mode="hclust",hang = 0.0001)
# plot(ceb_ldel,Ldel_net,vertex.label=NA,vertex.frame.color="red",
#      main="Ldel with modules network") 
# 
# #write_csv(memberss_ldel,path="~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/Ldel_phage.members",col_names = FALSE)
# write_csv(memberss_ldel,path="~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/Ldel_phage_cutoff25.members",col_names = FALSE)
# 
# 
# table(memberss_ldel$as.double.mem_cab_ldel.)
#   ##---------llac
#   ceb_Llac <- cluster_edge_betweenness(Llac_net)
#   mem_cab_llac <- membership(ceb_Llac)
#   df_mem_cab_Llac <- split(names(mem_cab_llac),mem_cab_llac)
# 
# memberss_llac <- data.frame(names(mem_cab_llac),as.double(mem_cab_llac))
# # dendPlot(ceb_ldel,mode="hclust",hang = 0.0001)
# plot(ceb_Llac,Llac_net,vertex.label=NA,vertex.frame.color="red",
#      main="llac with modules network") 
# 
# 
# table(memberss_llac$as.double.mem_cab_llac.)
# which(table(memberss_llac$as.double.mem_cab_llac.)>2)
# 
# 
# keepers_names_llac <- as.double(which(table(memberss_llac$as.double.mem_cab_llac.)>2))
# #length(as.double(which(table(memberss_sterm$as.double.mem_cab_sterm.)>2)))
# 
# memberss_llac_subset <- memberss_llac[which(memberss_llac$as.double.mem_cab_llac. %in% keepers_names_llac),]
#   table(memberss_llac_subset$as.double.mem_cab_llac.)
# 
# 
# write_csv(memberss_llac,path="~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/Lactococcus_phage_cutoff50.members",col_names = FALSE)
# write_csv(memberss_llac_subset,path="~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/Lactococcus_phage_cutoff50_subset.members",col_names = FALSE)
# 
# 
# 
# 
#   ##----------hubscore ==High number of connections
#   hub_ara <- hub_score(Sterm_net)$vector
#   
#   #table(hub_ara)
#   
#   
#   mem_cab_ara <- membership(ceb_ara)
#   df_mem_cab_ara <- split(names(mem_cab_ara),mem_cab_ara)
#   names(df_mem_cab_ara) <- paste0("module_",names(df_mem_cab_ara))
#   
#   membersdata <- melt(df_mem_cab_ara)
#   return(membersdata)

 
  ###-----------------------------------
  ##Clustering analysis
  ###-----------------------------------

library(tidyverse)
library(readr)
Sterm_clustering <- read_csv("Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03/allPhages_vContact/c1.clusters")
  
Sterm_clustering_unnested <- Sterm_clustering %>% 
    mutate(Members = strsplit(as.character(Members), " ")) %>% 
    unnest(Members) %>% filter(., grepl('Streptococcus', Members))

Sterm_clustering_forFiltering <- Sterm_clustering  %>% filter(., grepl('Streptococcus', Members))

ggplot(Sterm_clustering_forFiltering,aes(x=Quality))+geom_density()+theme_classic()
ggplot(Sterm_clustering_forFiltering,aes(x=Quality,y=Size))+geom_point()+theme_classic()
nrow(Sterm_clustering_forFiltering)
Sterm_clustering_forFiltering %>% filter(Size>3) %>% filter(Quality>0.2) %>% nrow

ggplot(Sterm_clustering_forFiltering,aes(x=Size))+geom_density()+theme_classic()

 Sterm_clustering_forFiltering  %>% filter(., grepl('SW', Members))

  ###-----------------------------------
  ##Clustering analysis 02
  ###-----------------------------------
# ~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new//allPhages_vContact
library(tidyverse)
library(readr)
Sterm_clustering <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03/allPhages_vContact/genome_by_genome_overview.csv")
Sterm_clustering[grep("rmk",Sterm_clustering$Genome),"Genome"]
#   
# Sterm_clustering_unnested <- Sterm_clustering %>% 
#     mutate(Members = strsplit(as.character(Members), " ")) %>% 
#     unnest(Members) %>% filter(., grepl('Streptococcus', Members))

Sterm_clustering$Genome <- Sterm_clustering$Genome %>% gsub("~"," ",.)
Sterm_clustering_filtered <- Sterm_clustering %>% select(Genome,Order,Family,Genus,VC,Size,Quality)
# Sterm_clustering_filtered$known <- 

Mahony_genomes$Genome <- paste(Mahony_genomes$Genus,Mahony_genomes$phage,Mahony_genomes$strain,sep=" ")
Mahony_genomes_prep <- Mahony_genomes %>% select(Genome,Type)

Sterm_clustering_filtered_2 <- merge(Sterm_clustering_filtered,Mahony_genomes_prep,by="Genome",all.x = TRUE)

clustersKnown <- Sterm_clustering_filtered_2[which(!is.na(Sterm_clustering_filtered_2$Type)),] %>% select(VC,Type) %>% unique() 

Sterm_clustering_filtered_final <- merge(Sterm_clustering_filtered_2,clustersKnown,by = "VC",all.x = TRUE)


table(Sterm_clustering_filtered_final$Type.y)
table(Sterm_clustering_filtered_final$VC) %>% sort()

# which(Sterm_clustering_filtered_final$Type.y=="cos")

# Sterm_clustering_filtered_final[which(!is.na(Sterm_clustering_filtered_final$Type.y)),]


write_csv(Sterm_clustering_filtered_final,path="~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03///allPhages_vContact/Streptococcus_phage_Cluster_assignment.txt",col_names = TRUE)

# ggplot(Sterm_clustering_forFiltering,aes(x=Quality))+geom_density()+theme_classic()
# ggplot(Sterm_clustering_forFiltering,aes(x=Quality,y=Size))+geom_point()+theme_classic()


```


##### Sterm

```{r}
library(ggplot2)
library(readr)
library(igraph)
library(plyr)
library(tidyverse)
#---------------
##import data
#---------------
###---------All phages
 Phages_names <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03//allPhages_vContact/genome_by_genome_overview.csv")

Sterm_cluster <- read_table2("~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03//allPhages_vContact/c1.ntw", 
    col_names = c("from","to","power"))

# Sterm_cluster <- read_table2("~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/c1.ntw", 
#     col_names = c("from","to","power"))

# 
# ##all contig names
# Sterm_contigs <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact/cluster/Sterm_contigs.ntw", 
#     col_names = "contigs")
# ##all clusterings
# Sterm_cluster <- read_table2("~/Desktop/Projects/2019_PhageDB/Vcontact/cluster/Sterm_cluster.ntw", 
#     col_names = c("from","to","power"))
Mahony_genomes <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact/cluster/Mahony_table_ncbi_genomes.txt",
    col_names = c("Genus","phage","strain","NCBI","Type"))

#---------------
##reduce to only Strep clusters
#---------------

#----keep cluster that contain one sterm!!

Sterm_clustering_complete <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03///allPhages_vContact/genome_by_genome_overview.csv")
# Sterm_clustering_complete <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/genome_by_genome_overview.csv")


# Sterm_clustering_new <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/c1.clusters") %>% as.data.frame()
Sterm_clustering_new <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03//allPhages_vContact/c1.clusters") %>% as.data.frame()
Sterm_clustering_forFiltering_new <- Sterm_clustering_new  %>% filter(., grepl('Streptococcus', Members,ignore.case = TRUE))

Sterm_clustering_unnested <- Sterm_clustering_forFiltering_new %>% 
    mutate(Members = strsplit(as.character(Members), " ")) %>% 
    unnest(Members)

str_split_fixed(Sterm_clustering_unnested$Members, "~", 3)[,1]  %>% table()


###---------------1. filter:
##clusters that contain a Streptococcus


nrow(Sterm_clustering_unnested)
nrow(Sterm_cluster)
Sterm_cluster_sub <- Sterm_cluster[which(Sterm_cluster$from %in% Sterm_clustering_unnested$Members | Sterm_cluster$to %in% Sterm_clustering_unnested$Members),]
nrow(Sterm_cluster_sub)
# Sterm_cluster_sub <- Sterm_cluster_sub[which(Sterm_cluster_sub$to %in% Sterm_clustering_unnested$Members),]
# nrow(Sterm_cluster_sub)

###---------------2. filter:
##connections that have a higher qualtiy than 50 (decided on distribution plot)

ggplot(Sterm_cluster,aes(x=power))+geom_histogram(binwidth=1)
Sterm_cluster_sub <- Sterm_cluster_sub[Sterm_cluster_sub$power>4,]
nrow(Sterm_cluster_sub)

###---------------3. filter:
##cluster containing more than 2 phages 

Sterm_cluster_sub_uniques <- unique(c(Sterm_cluster_sub$from,Sterm_cluster_sub$to))

Sterm_clustering_unnested_which <- 
  Sterm_clustering_complete[which(Sterm_clustering_complete$Genome %in% Sterm_cluster_sub_uniques),] %>% select(VC) %>% table( )
  # select_if(~n_distinct(.) > 2 & is.numeric(.)) %>% 
  names(Sterm_clustering_unnested_which)
  Sterm_clustering_unnested_which_keep <- Sterm_clustering_unnested_which[which(Sterm_clustering_unnested_which>2)] %>% names()
  Sterm_clustering_unnested_which_keep_genome <- Sterm_clustering_complete[which(Sterm_clustering_complete$VC %in% Sterm_clustering_unnested_which_keep),"Genome"] %>% as.data.frame()
  
  
  # sum(Sterm_cluster_sub$from %in% Sterm_clustering_unnested_which_keep_genome$Genome) 
  nrow(Sterm_cluster_sub)
  Sterm_cluster_sub <- Sterm_cluster_sub[which(Sterm_cluster_sub$from %in% Sterm_clustering_unnested_which_keep_genome$Genome|Sterm_cluster_sub$to %in% Sterm_clustering_unnested_which_keep_genome$Genome),]
  nrow(Sterm_cluster_sub)
  
  unique(Sterm_cluster_sub$from)
# own_cluster_sub_new<- Sterm_cluster_sub
 
 ##===============================
 #old plots
  ##===============================
#---------------
##plot interaction
#---------------
ggplot(Sterm_cluster,aes(x=power))+geom_histogram(binwidth=1)
dim(Sterm_cluster_sub)
# 
# ggplot(Ldel_cluster_sub,aes(x=power))+geom_histogram(binwidth=1)
# dim(Ldel_cluster_sub)
# 
# ggplot(Llac_cluster_sub,aes(x=power))+geom_histogram(binwidth=1)
# dim(Llac_cluster_sub)

#---------------
##subset for only highest interactions
#---------------
#Sterm_cluster_sub <- Sterm_cluster_sub[Sterm_cluster_sub$power>1,]
#dim(Sterm_cluster_sub)

#Sterm_cluster_sub <- Sterm_cluster_sub[Sterm_cluster_sub$power>10,]
#dim(Sterm_cluster_sub)

# length(unique(Sterm_cluster_sub$from))

#---------------
##creat graph
#---------------
# write_csv(Sterm_cluster_sub,path="~/Desktop/Projects/2019_PhageDB/Vcontact//allPhages_vContact/Streptococcus_phage_betwork_connections.txt",col_names = TRUE)

write_csv(Sterm_cluster_sub,path="~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03///allPhages_vContact/Streptococcus_phage_betwork_connections.txt",col_names = TRUE)

Sterm_net <- graph_from_data_frame(Sterm_cluster_sub,direct=F)
# Ldel_net <- graph_from_data_frame(Ldel_cluster_sub,direct=F)
# Llac_net <- graph_from_data_frame(Llac_cluster_sub,direct=F)

# dim(Sterm_net)

##==================================
 ##new analysis
  ##==================================
 library(ggnetwork)
 library(ITNr)
 library(intergraph)
 
 net<-asNetwork(Sterm_net) #ggnetwork requires a network object
 n<-ggnetwork(net)
  ##------------------
 ##color and size change
  ##------------------
 
 
###--------
##grep fo Mahony phages
#---------------
Mahony_genomes$namez <- paste(Mahony_genomes$Genus,Mahony_genomes$phage,Mahony_genomes$strain,sep="~")
Mahony_genomes_cos <- Mahony_genomes[which(Mahony_genomes$Type=="cos"),]
Mahony_genomes_pac <- Mahony_genomes[which(Mahony_genomes$Type=="pac"),]
Mahony_genomes_987 <- Mahony_genomes[which(Mahony_genomes$Type=="987"),]
Mahony_genomes_5093 <- Mahony_genomes[which(Mahony_genomes$Type=="5093"),]

###--------
##coloring
#---------------
# n[grep("RMK202_",n$vertex.names),"vertex.names"] %>% table()
 n$color <- as.character("Viral DB")
 n$color[grep("RMK202_",n$vertex.names)] <- "RMK202 phages"
 n$color[n$vertex.names %in% Mahony_genomes_cos] <- "cos"
 n$color[n$vertex.names %in% Mahony_genomes_pac] <- "pac"
 n$color[n$vertex.names %in% Mahony_genomes_987] <- "987"
 n$color[n$vertex.names %in% Mahony_genomes_5093] <- "5093"
 
 
 n$color = factor(n$color, levels=c("RMK202 phages","cos","pac","987","5093","Viral DB"))
 colorzzz <- c("red","blue","green","yellow","purple","grey50")
 size <- c("4","2","2","2","2","1.5")
 ##------------------
 ##extend
 ##------------------
 
 nExtended <-  merge(n, Phages_names, by.y="Genome",by.x="vertex.names",all.x = TRUE)
 # nrow(n)
 # nExtended <-  merge(n, Phages_names, by.y="Genome",by.x="vertex.names",all.x = TRUE) 
 # nrow(nExtended)
 # nExtended <- n
 # ccolnames(nExtended)
colnames(nExtended)
colnames(nExtended) <- revalue(colnames(nExtended), c("Adj P-value"="Adj_P_value","Genera in VC"="Genera_in_VC","Topology Confidence Score"="Topology_Confidence_Score","Genus Confidence Score"="Genus_Confidence_Score"))

# nExtended$ad
 ##------------------
 ##plot
 ##------------------
 networkplot <- ggplot(nExtended, aes(x = x, y = y, xend = xend, yend = yend,text =paste("genome:", vertex.names,"\npower: ",power,"\nOrder: ",Order,"\nFamily: ",Family,"\nGenus: ",Genus,"\nVC: ",VC,"\nSize: ",Size,"\nAdj P-value: ",Adj_P_value,"\nGenera in VC: ",Genera_in_VC,"\nGenus_Confidence_ScoreC: ",Genus_Confidence_Score,"\nTopology_Confidence_Score: ",Topology_Confidence_Score))) + geom_edges(color = "lightgray") +
   geom_nodes(aes(color = as.factor(color),size=as.factor(color))) +
   # scale_fill_manual(values="gray41")+
    scale_color_manual(values=colorzzz)+
    scale_size_manual(values=as.numeric(size))+
   # guides(color=FALSE,size=FALSE)+ #Remove legends
 theme_blank()+theme(legend.title = element_blank())
 networkplot
 
 library(plotly)
 ggp <- ggplotly(networkplot)
 #htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03//allPhages_vContact//networkPlot.html")
 # View(nExtended)
 
tmp <-   nExtended %>% filter(color=="RMK202 phages") 
table(tmp$`VC Subcluster`)
 
 
 nExtended %>% filter(color!="Viral DB") %>% filter()
 sort(table(nExtended$VC),decreasing = TRUE)
 
 count_perViralCluster <- nExtended %>%
  group_by(VC,color) %>% 
  dplyr::summarise(count=n()) %>% spread(.,color,count)
 
  ##==================================
 ##keep only phages in our s. thermophilus phage cluster
  ##==================================
 ###---------------------identify phages in our cluster

 
# table(nExtended$VC) 

tmp_phages <- nExtended[grep("RMK202_",nExtended$vertex.names),] 
ourPhageCluster <- c(unique(tmp_phages$VC))
nExtended_ourphages <- nExtended %>% filter(VC %in% ourPhageCluster)

phagesinClusters <- Phages_names %>% filter(VC %in% ourPhageCluster)
phagesinClusters$Genome

###---------------------clustering
# Sterm_clustering_new <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/c1.clusters") %>% as.data.frame()
Sterm_clustering_new <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03//allPhages_vContact/c1.clusters") %>% as.data.frame()
Sterm_clustering_forFiltering_new <- Sterm_clustering_new  %>% filter(., grepl('Streptococcus', Members,ignore.case = TRUE))

Sterm_clustering_unnested <- Sterm_clustering_forFiltering_new %>% 
    mutate(Members = strsplit(as.character(Members), " ")) %>% 
    unnest(Members)

# str_split_fixed(Sterm_clustering_unnested$Members, "~", 3)[,1]  %>% table()


###---------------1. filter:
##clusters that contain a Streptococcus


# nrow(Sterm_clustering_unnested)
nrow(Sterm_cluster)
Sterm_cluster_sub <- Sterm_cluster[which(Sterm_cluster$from %in% phagesinClusters$Genome | Sterm_cluster$to %in% phagesinClusters$Genome),]
nrow(Sterm_cluster_sub)
Sterm_net_sub <- graph_from_data_frame(Sterm_cluster_sub,direct=F)

 

 net<-asNetwork(Sterm_net_sub) #ggnetwork requires a network object
 n<-ggnetwork(net)
  ##------------------
 ##color and size change
  ##------------------
 
 
###--------
##coloring
#---------------
# n[grep("RMK202_",n$vertex.names),"vertex.names"] %>% table()
 n$color <- as.character("Viral DB")
 n$color[grep("RMK202_",n$vertex.names)] <- "RMK202 phages"
 n$color[n$vertex.names %in% Mahony_genomes_cos$namez] <- "cos"
 n$color[n$vertex.names %in% Mahony_genomes_pac$namez] <- "pac"
 n$color[n$vertex.names %in% Mahony_genomes_987$namez] <- "987"
 n$color[n$vertex.names %in% Mahony_genomes_5093$namez] <- "5093"
 
 
 n$color = factor(n$color, levels=c("RMK202 phages","cos","pac","987","5093","Viral DB"))
 colorzzz <- c("red","blue","green","#808000","purple","grey50")
 size <- c("4","2","2","2","2","1.5")
 
 
 # n$color = factor(n$color, levels=c("RMK202 phages","cos","987","Viral DB"))
 # colorzzz <- c("red","blue","#808000","grey50")
 # size <- c("4","2","2","1.5")
 
 table(n$color)
 
 n$color_new <- as.character("Viral DB")
n$color_new[grep("RMK202_Streptococcus_phage_1",n$vertex.names)[1]] <- "RMK202 phages"
n$color_new[grep("RMK202_Streptococcus_phage_2",n$vertex.names)[1]] <- "RMK202 phages"


 ##------------------
 ##extend
 ##------------------
 
 nExtended <-  merge(n, Phages_names, by.y="Genome",by.x="vertex.names",all.x = TRUE)
 # nrow(n)
 # nExtended <-  merge(n, Phages_names, by.y="Genome",by.x="vertex.names",all.x = TRUE) 
 # nrow(nExtended)
 # nExtended <- n
 # ccolnames(nExtended)
colnames(nExtended)
colnames(nExtended) <- revalue(colnames(nExtended), c("Adj P-value"="Adj_P_value","Genera in VC"="Genera_in_VC","Topology Confidence Score"="Topology_Confidence_Score","Genus Confidence Score"="Genus_Confidence_Score"))



 nExtended$alpha <- as.character("Viral DB")
nExtended$alpha[nExtended$VC %in% ourPhageCluster] <- "RMK202 phages"
table(nExtended$alpha)
  alphaCol <- c(1,0.03)
  
tmp <- nExtended[nExtended$alpha=="RMK202 phages",]
table(tmp$VC)
  
nExtended$fillings <- as.character("Viral DB")
nExtended$fillings[nExtended$VC %in% ourPhageCluster[1]] <- ourPhageCluster[1]
nExtended$fillings[nExtended$VC %in% ourPhageCluster[2]] <- ourPhageCluster[2]
nExtended$fillings[nExtended$vertex.names %in% Mahony_genomes_cos$namez] <- "cos"
nExtended$fillings[nExtended$vertex.names %in% Mahony_genomes_987$namez] <- "987"
nExtended$fillings[nExtended$vertex.names=="RMK202_Streptococcus_phage_1"] <- "Streptococcus_phage_1"
nExtended$fillings[nExtended$vertex.names=="RMK202_Streptococcus_phage_2"] <- "Streptococcus_phage_2"
as.factor(nExtended$fillings)
library("RColorBrewer")

three_reds <- c(brewer.pal(9,name="YlOrRd")[c(4,6)],
							 brewer.pal(3,name="Reds")[3])
three_browns <- brewer.pal(9,name="YlOrBr")[c(7,8,9)]
three_blues <- brewer.pal(9,name="YlGnBu")[c(5,6,8)]

filling <- c(three_reds[1],three_blues[1],three_reds[2],three_blues[2],three_reds[3],three_blues[3],"grey")
# filling <- c(three_reds[1],three_browns[1],three_reds[2],three_browns[2],three_reds[3],three_browns[3],"grey")

# filling <- c("black","red","darkgrey","darkred","grey")
# table(nExtended$alpha)

  



# nExtended$ad
 ##------------------
 ##plot
 ##------------------
table(nExtended$alpha)
# nExtended$Family
 networkplot_sub <- ggplot(nExtended, aes(x = x, y = y, xend = xend, yend = yend,text =paste("genome:", vertex.names,"\npower: ",power,"\nOrder: ",Order,"\nFamily: ",Family,"\nGenus: ",Genus,"\nVC: ",VC,"\nSize: ",Size,"\nAdj P-value: ",Adj_P_value,"\nGenera in VC: ",Genera_in_VC,"\nGenus_Confidence_ScoreC: ",Genus_Confidence_Score,"\nTopology_Confidence_Score: ",Topology_Confidence_Score))) + geom_edges(color = "lightgray",alpha=0.1) +
   geom_nodes(aes(color = as.factor(fillings),size=as.factor(alpha),alpha=as.factor(alpha))) +
   # scale_fill_manual(values="gray41")+
    scale_color_manual(values=filling)+
     scale_alpha_manual(values=alphaCol)+
    scale_fill_manual(values=colorzzz)+
   geom_nodelabel_repel(aes(label = vertex.names),
                       box.padding = unit(1, "lines"),
                       data = function(x) { x[ x$color_new == "RMK202 phages", ]})+
    scale_size_manual(values=as.numeric(c(2,1.25)))+
   # guides(color=FALSE,size=FALSE)+ #Remove legends
 theme_blank()+theme(legend.title = element_blank())
 networkplot_sub
 table(nExtended$color_new)
 # RMK202 phages
#  x[grep("RMK202_",x$vertex.names),]
#  n$color[grep("RMK202_",n$vertex.names)]
 svg("~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03//allPhages_vContact//networkPlot_subset.svg",width=7,height=7)
 # png("~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03//allPhages_vContact//networkPlot_subset.png", width = 2800, height = 2800,res=300)
  networkplot_sub
   dev.off()
 
 library(plotly)
 # ggp <- ggplotly(networkplot_sub)
 # htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03//allPhages_vContact//networkPlot_subset.html")
 # View(nExtended)


 ##==================================
 ##old analysis
  ##==================================



###--------
##grep fo Mahony phages
#---------------
Mahony_genomes$namez <- paste(Mahony_genomes$Genus,Mahony_genomes$phage,Mahony_genomes$strain,sep="~")
Mahony_genomes_cos <- Mahony_genomes[which(Mahony_genomes$Type=="cos"),]
Mahony_genomes_pac <- Mahony_genomes[which(Mahony_genomes$Type=="pac"),]
Mahony_genomes_987 <- Mahony_genomes[which(Mahony_genomes$Type=="987"),]
Mahony_genomes_5093 <- Mahony_genomes[which(Mahony_genomes$Type=="5093"),]

##colour

vcolour_Sterm <- rep("grey",length(V(Sterm_net)$name))
# vcolour_Sterm_ori <- V(Sterm_net)$name
# vcolour_Sterm <- "grey"
vcolour_Sterm[which(V(Sterm_net)$name %in% paste(Mahony_genomes_cos$Genus,Mahony_genomes_cos$phage,Mahony_genomes_cos$strain,sep="~"))] <- "green"
vcolour_Sterm[which(V(Sterm_net)$name %in% paste(Mahony_genomes_pac$Genus,Mahony_genomes_pac$phage,Mahony_genomes_pac$strain,sep="~"))] <- "blue"
vcolour_Sterm[which(V(Sterm_net)$name %in% paste(Mahony_genomes_987$Genus,Mahony_genomes_987$phage,Mahony_genomes_987$strain,sep="~"))] <- "pink"
vcolour_Sterm[which(V(Sterm_net)$name %in% paste(Mahony_genomes_5093$Genus,Mahony_genomes_5093$phage,Mahony_genomes_5093$strain,sep="~"))] <- "brown"

# vcolour_Sterm[which(!vcolour_Sterm_ori %in% paste(Mahony_genomes$Genus,Mahony_genomes$phage,Mahony_genomes$strain,sep="~"))] <- "grey"
table(vcolour_Sterm)

V(Sterm_net)$color <- vcolour_Sterm
table(V(Sterm_net)$color)
# V(Sterm_net)$color[which(V(Sterm_net)$name=='Streptococcus~phage~SW19')] <- "grey"
# V(Sterm_net)$color[which(V(Sterm_net)$name=='Streptococcus~phage~SW24')] <- "grey"
##size

vsize_Sterm <- V(Sterm_net)$name
vsize_Sterm_ori <- V(Sterm_net)$name

grep("S_t",V(Sterm_net)$name)
vsize_Sterm[which(!vsize_Sterm_ori %in% paste(Mahony_genomes$Genus,Mahony_genomes$phage,Mahony_genomes$strain,sep="~"))] <- "1.5"
vsize_Sterm[which(vsize_Sterm_ori %in% paste(Mahony_genomes$Genus,Mahony_genomes$phage,Mahony_genomes$strain,sep="~"))] <- "5"

table(vsize_Sterm)

vsize_Sterm <- as.numeric(vsize_Sterm)

#---------------
#plot
#---------------

set.seed(8051)

 V(Sterm_net)$name[grep("RMK202",V(Sterm_net)$name)]
 length( V(Sterm_net)$name)
 # V(Llac_net)$name
 # V(Ldel_net)$name
 # length(V(Ldel_net)$name)


# create vertices
fg_lay_Sterm <- layout_with_fr(Sterm_net,dim=2,niter=99)
# fg_lay_Ldel <- layout_with_fr(Ldel_net,dim=2,niter=99)
# fg_lay_Llac <- layout_with_fr(Llac_net,dim=2,niter=99)

# plot  Sterm
 # plot(Sterm_net,vertex.label=NA,layout=fg_lay_Sterm,
 #     vertex.size=vsize_Sterm,
 #       main="Sterm phages network"
 #     )
     # png("~/Desktop/Projects/2019_RMK202_analysis/plot/vContact_sterm.png", width = 1800, height = 1200,res=300)

 plot(Sterm_net,vertex.label=NA,layout=fg_lay_Sterm,vertex.size=vsize_Sterm,
       main="Sterm phages network"
     )
 legend("topright", 
       legend=c("cos","pac","987","5093","unknown"), pch=19,
      col=c("green","blue","pink","brown","grey"), bty="n")

# dev.off()
 
# # plot  Ldel
#  plot(Ldel_net,vertex.label=NA,layout=fg_lay_Ldel,
#        main="Ldel phages network"
#      )
#  
#  plot(Llac_net,vertex.label=NA,layout=fg_lay_Llac,
#        main="Llac phages network"
#      )
 
 #---------------
 ##creat modules
 #---------------
 # ceb_sterm_high25 <- ceb_sterm
# ceb_sterm <- cluster_edge_betweenness(Sterm_net)
#dendPlot(ceb_ara,mode="hclust")
# plot(ceb_sterm,Sterm_net,vertex.label=NA,vertex.size=vsize_Sterm,vertex.frame.color="red",
#      main="arabidospsis network")

 
#  ##---------Sterm
# ceb_sterm<- cluster_edge_betweenness(Sterm_net)
#   mem_cab_sterm <- membership(ceb_sterm)
#   df_mem_cab_sterml <- split(names(mem_cab_sterm),mem_cab_sterm)
# 
# memberss_sterm <- data.frame(names(mem_cab_sterm),as.double(mem_cab_sterm))
# # dendPlot(ceb_ldel,mode="hclust",hang = 0.0001)
# plot(ceb_sterm,Sterm_net,vertex.label=NA,vertex.frame.color="red",
#      main="sterm with modules network") 
# 
# 
# table(memberss_sterm$as.double.mem_cab_sterm.)
# which(table(memberss_sterm$as.double.mem_cab_sterm.)>2)
# 
# memberss_sterm[which(memberss_sterm$names.mem_cab_sterm. %in% Mahony_genomes_pac$namez),]
# memberss_sterm[which(memberss_sterm$names.mem_cab_sterm. %in% Mahony_genomes_987$namez),]
# memberss_sterm[which(memberss_sterm$names.mem_cab_sterm. %in% Mahony_genomes_cos$namez),]
# memberss_sterm[which(memberss_sterm$names.mem_cab_sterm. %in% Mahony_genomes_5093$namez),]
# 
# keepers_names <- as.double(which(table(memberss_sterm$as.double.mem_cab_sterm.)>2))
# length(as.double(which(table(memberss_sterm$as.double.mem_cab_sterm.)>2)))
# 
# memberss_sterm_subset <- memberss_sterm[which(memberss_sterm$as.double.mem_cab_sterm. %in% keepers_names),]
#   table(memberss_sterm_subset$as.double.mem_cab_sterm.)
# 
# 
#   
# write_csv(memberss_sterm,path="~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/Streptococcus_phage_cutoff50.members",col_names = FALSE)
# write_csv(memberss_sterm_subset,path="~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/Sterm_phage_cutoff55_subset.members",col_names = FALSE)
# 


# 
# memberss_sterm$names.mem_cab_sterm. 
# Mahony_genomes_pac$namez
# 
#   ##---------ldel
#   ceb_ldel <- cluster_edge_betweenness(Ldel_net)
#   mem_cab_ldel <- membership(ceb_ldel)
#   df_mem_cab_Ldel <- split(names(mem_cab_ldel),mem_cab_ldel)
# 
# memberss_ldel <- data.frame(names(mem_cab_ldel),as.double(mem_cab_ldel))
# # dendPlot(ceb_ldel,mode="hclust",hang = 0.0001)
# plot(ceb_ldel,Ldel_net,vertex.label=NA,vertex.frame.color="red",
#      main="Ldel with modules network") 
# 
# #write_csv(memberss_ldel,path="~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/Ldel_phage.members",col_names = FALSE)
# write_csv(memberss_ldel,path="~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/Ldel_phage_cutoff25.members",col_names = FALSE)
# 
# 
# table(memberss_ldel$as.double.mem_cab_ldel.)
#   ##---------llac
#   ceb_Llac <- cluster_edge_betweenness(Llac_net)
#   mem_cab_llac <- membership(ceb_Llac)
#   df_mem_cab_Llac <- split(names(mem_cab_llac),mem_cab_llac)
# 
# memberss_llac <- data.frame(names(mem_cab_llac),as.double(mem_cab_llac))
# # dendPlot(ceb_ldel,mode="hclust",hang = 0.0001)
# plot(ceb_Llac,Llac_net,vertex.label=NA,vertex.frame.color="red",
#      main="llac with modules network") 
# 
# 
# table(memberss_llac$as.double.mem_cab_llac.)
# which(table(memberss_llac$as.double.mem_cab_llac.)>2)
# 
# 
# keepers_names_llac <- as.double(which(table(memberss_llac$as.double.mem_cab_llac.)>2))
# #length(as.double(which(table(memberss_sterm$as.double.mem_cab_sterm.)>2)))
# 
# memberss_llac_subset <- memberss_llac[which(memberss_llac$as.double.mem_cab_llac. %in% keepers_names_llac),]
#   table(memberss_llac_subset$as.double.mem_cab_llac.)
# 
# 
# write_csv(memberss_llac,path="~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/Lactococcus_phage_cutoff50.members",col_names = FALSE)
# write_csv(memberss_llac_subset,path="~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/Lactococcus_phage_cutoff50_subset.members",col_names = FALSE)
# 
# 
# 
# 
#   ##----------hubscore ==High number of connections
#   hub_ara <- hub_score(Sterm_net)$vector
#   
#   #table(hub_ara)
#   
#   
#   mem_cab_ara <- membership(ceb_ara)
#   df_mem_cab_ara <- split(names(mem_cab_ara),mem_cab_ara)
#   names(df_mem_cab_ara) <- paste0("module_",names(df_mem_cab_ara))
#   
#   membersdata <- melt(df_mem_cab_ara)
#   return(membersdata)

 
  ###-----------------------------------
  ##Clustering analysis
  ###-----------------------------------

library(tidyverse)
library(readr)
Sterm_clustering <- read_csv("Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03/allPhages_vContact/c1.clusters")
  
Sterm_clustering_unnested <- Sterm_clustering %>% 
    mutate(Members = strsplit(as.character(Members), " ")) %>% 
    unnest(Members) %>% filter(., grepl('Streptococcus', Members))

Sterm_clustering_forFiltering <- Sterm_clustering  %>% filter(., grepl('Streptococcus', Members))

ggplot(Sterm_clustering_forFiltering,aes(x=Quality))+geom_density()+theme_classic()
ggplot(Sterm_clustering_forFiltering,aes(x=Quality,y=Size))+geom_point()+theme_classic()
nrow(Sterm_clustering_forFiltering)
Sterm_clustering_forFiltering %>% filter(Size>3) %>% filter(Quality>0.2) %>% nrow

ggplot(Sterm_clustering_forFiltering,aes(x=Size))+geom_density()+theme_classic()

 Sterm_clustering_forFiltering  %>% filter(., grepl('SW', Members))

  ###-----------------------------------
  ##Clustering analysis 02
  ###-----------------------------------
# ~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new//allPhages_vContact
library(tidyverse)
library(readr)
Sterm_clustering <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03/allPhages_vContact/genome_by_genome_overview.csv")
Sterm_clustering[grep("rmk",Sterm_clustering$Genome),"Genome"]
#   
# Sterm_clustering_unnested <- Sterm_clustering %>% 
#     mutate(Members = strsplit(as.character(Members), " ")) %>% 
#     unnest(Members) %>% filter(., grepl('Streptococcus', Members))

Sterm_clustering$Genome <- Sterm_clustering$Genome %>% gsub("~"," ",.)
Sterm_clustering_filtered <- Sterm_clustering %>% select(Genome,Order,Family,Genus,VC,Size,Quality)
# Sterm_clustering_filtered$known <- 

Mahony_genomes$Genome <- paste(Mahony_genomes$Genus,Mahony_genomes$phage,Mahony_genomes$strain,sep=" ")
Mahony_genomes_prep <- Mahony_genomes %>% select(Genome,Type)

Sterm_clustering_filtered_2 <- merge(Sterm_clustering_filtered,Mahony_genomes_prep,by="Genome",all.x = TRUE)

clustersKnown <- Sterm_clustering_filtered_2[which(!is.na(Sterm_clustering_filtered_2$Type)),] %>% select(VC,Type) %>% unique() 

Sterm_clustering_filtered_final <- merge(Sterm_clustering_filtered_2,clustersKnown,by = "VC",all.x = TRUE)


table(Sterm_clustering_filtered_final$Type.y)
table(Sterm_clustering_filtered_final$VC) %>% sort()

# which(Sterm_clustering_filtered_final$Type.y=="cos")

# Sterm_clustering_filtered_final[which(!is.na(Sterm_clustering_filtered_final$Type.y)),]


write_csv(Sterm_clustering_filtered_final,path="~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03///allPhages_vContact/Streptococcus_phage_Cluster_assignment.txt",col_names = TRUE)

# ggplot(Sterm_clustering_forFiltering,aes(x=Quality))+geom_density()+theme_classic()
# ggplot(Sterm_clustering_forFiltering,aes(x=Quality,y=Size))+geom_point()+theme_classic()

```



!!This is already implemented in the /home/vincent/scripts/mappingLengthfiltering.R script which runs in the previous loop chunk
## R 

add the following information for spacers:
- array assignment
- number of samples mapped to
- blast
- mapping coverage spacer and protospacer (see correlation between spacer coverage and otu call)


#####basic analysis
```{r }
library(ggplot2)
library(plyr)
library(readr)
library(tidyverse)
###-----------------------
##grouping
###-----------------------
groupingRMK202Strains <- data.frame(strain=c("13492","13491","24854","24837","13500","13493","13498","SMAG","24853","13494","24855","S72","13499c1","13499","13499c2","24838","24840","S50","13497","24839","13495","13496"),group=c(rep("lineage 1",9),rep("lineage 2",3),rep("lineage 3",6),rep("lineage 4",4)),colors=c(rep("#0000FF",9),rep("#6699FF",3),rep("#99CCFF",6),rep("#00FFFF",4)))


write.table(groupingRMK202Strains, "~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/log/ReferenceGenomeGrouping.txt",na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names =FALSE)

#error with wrong number as genome labels e.g. 24839 instead of the right 24739
#ReferenceGenomeGrouping <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/log/ReferenceGenomeGrouping.txt<",  "\t", escape_double = FALSE, col_names = c("strain","lineage","colors"),  trim_ws = TRUE)

ReferenceGenomeGrouping <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/log/ReferenceGenomeGrouping2.txt",  "\t", escape_double = FALSE, col_names = c("strain","lineage","colors"),  trim_ws = TRUE)

##=========================
##new

##-------------spacer count DADA over metagenomic samples

dada_spacer_count <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract/FinalSpacers/ForDADA2_woREverse/dada_spacer_count.txt", "\t", escape_double = FALSE, trim_ws = TRUE)


##=========================
##old
##the old analysis is pretty cool because I can distinguish between protospacer and spacer mapping

##-----------------------------------------------
##-------------------------bwa spacer count normalised
##-----------------------------------------------
#CRISPR_spacer_coverage <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/CRISPR_spacer_coverage.bed",   "\t", escape_double = FALSE, col_names = c("Name","startSpacer","endSpacer","ClusterName","numReads","basesCoverd","basesTotal","numcov","sample"),  trim_ws = TRUE)
#reads4normalization <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/reads4normalization.txt", "\t", escape_double = FALSE, col_names = c("sample","readNumber","species"), trim_ws = TRUE) ;  reads4normalization <- reads4normalization[,-3]

CRISPR_spacer_coverage <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//bwaMapping/final/CRISPR_spacer_coverage.bed",   "\t", escape_double = FALSE, col_names = c("Name","startSpacer","endSpacer","ClusterName","numReads","basesCoverd","basesTotal","numcov","sample"),  trim_ws = TRUE)
reads4normalization <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//bwaMapping/log/reads4normalization.txt", "\t", escape_double = FALSE, col_names = c("sample","readNumber","species"), trim_ws = TRUE) ;  reads4normalization <- reads4normalization[,-3]
###----------------normalize with reads mapped to the CRISPR spacers
CRISPR_spacer_coverage <- merge(CRISPR_spacer_coverage,reads4normalization,by="sample",all.x = TRUE)
CRISPR_spacer_coverage$CPM <- 100000*(CRISPR_spacer_coverage$numReads/CRISPR_spacer_coverage$readNumber)

# table(CRISPR_spacer_coverage$sample)
# CRISPR_spacer_coverage %>% filter(!is.na(CPM)) %>% select(sample) %>% table()
CRISPR_spacer_coverage <- CRISPR_spacer_coverage  %>% filter(!is.na(CPM))


# CRISPR_spacer_coverage <- CRISPR_spacer_coverage[!duplicated(CRISPR_spacer_coverage[,c("sample","ClusterName")]), ]

bwaSPACERCount_wide <- CRISPR_spacer_coverage %>% dplyr::select(sample,Name,CPM) %>% spread(.,sample,CPM) #%>% rename(spacer=Name)
##-----------------------------------------------
##-------------------------bwa protospacer count normalised
##-----------------------------------------------
CRISPR_protospacer_coverage <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//bwaMapping/final//CRISPR_spacer_coverage_protospacer.bed",   "\t", escape_double = FALSE, col_names = c("Name","startSpacer","endSpacer","ClusterName","numReads","basesCoverd","basesTotal","numcov","sample"),  trim_ws = TRUE)
# reads4normalization <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/reads4normalization.txt", "\t", escape_double = FALSE, col_names = c("sample","readNumber","species"), trim_ws = TRUE) ;  reads4normalization <- reads4normalization[,-3]
###----------------normalize with reads mapped to the CRISPR spacers
CRISPR_protospacer_coverage <- merge(CRISPR_protospacer_coverage,reads4normalization,by="sample",all.x = TRUE)
CRISPR_protospacer_coverage$CPM <- 100000*(CRISPR_protospacer_coverage$numReads/CRISPR_protospacer_coverage$readNumber)

table(CRISPR_protospacer_coverage$sample)
table(table(CRISPR_protospacer_coverage$ClusterName))
CRISPR_protospacer_coverage <- CRISPR_protospacer_coverage  %>% filter(!is.na(CPM))

# CRISPR_protospacer_coverage <- CRISPR_protospacer_coverage[!duplicated(CRISPR_protospacer_coverage[,c("sample","ClusterName")]), ]
# table(CRISPR_spacer_coverage_02$sample)


bwaPROTOSPACERCount_wide <- CRISPR_protospacer_coverage %>% dplyr::select(sample,Name,CPM) %>% spread(sample,CPM) #%>% rename(spacer=Name) #rename(Name="spacer")
# bwaPROTOSPACERCount_wide <- CRISPR_spacer_coverage %>% dplyr::select(sample,Name,CPM) %>% unique()%>% spread(.,Name,CPM) #%>% rename(spacer=Name) #rename(Name="spacer")
 # CRISPR_spacer_coverage %>% dplyr::select(sample,Name) %>% nrow()
 # CRISPR_spacer_coverage %>% dplyr::select(sample,Name) %>% unique() %>% nrow()

# CRISPR_spacer_coverage %>% dplyr::select(sample,ClusterName)
###===========================================
##spacers only in experiment
###===========================================
ncol(bwaSPACERCount_wide)

bwaSPACERCount_wide[which(rowSums(bwaSPACERCount_wide[,-c(1,grep("_6_18",colnames(bwaSPACERCount_wide)))])==0),]

dada_spacer_count

##the following spacers only have coverage in the experiment samples
onlyExperimentSpacers <- dada_spacer_count[which(rowSums(dada_spacer_count[,-c(1,grep("_6_18",colnames(dada_spacer_count)))])==0),]

onlyExperimentSpacers



# bwaPROTOSPACERCount_wide[which(rowSums(bwaPROTOSPACERCount_wide[,-c(1,grep("_6_18",colnames(bwaPROTOSPACERCount_wide)))])==0),]

##!!!!!!!!!!!!!!!sample file
  sampleDF <- metasample_colors
  
##!!!!!!!!!!!!!!!spacer file
library(readr)
##-------information array assignment
  
metaspacer_info <- read_delim("/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/uniqueSpacers_count_Ref_wOLDstrains.txt", "\t", escape_double = FALSE, trim_ws = TRUE)  
metaspacer_info$METAClusterName <- gsub(">","",metaspacer_info$METAClusterName)
 spacers_info <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers//ForDADA2_woREverse//spacers_info.txt", "\t", escape_double = FALSE, trim_ws = TRUE)
##-------information spacer blast
# spacers_info_blast <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/all_differentBlastHits.txt", "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("DB","spacer","assigned")) %>% spread(., DB, assigned)
spacers_info_blast <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/all_differentBlastHits.txt", "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("DB","spacer","assigned")) %>% spread(., DB, assigned)

##-------clustering vContact blast
# Streptococcus_phage_Cluster_assignment <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new//allPhages_vContact/Streptococcus_phage_Cluster_assignment.txt",col_types = cols(.default = "c")) %>% mutate(Size = as.double(Size), Quality = as.double(Quality)) %>% select(-c(Order,Family,Genus,Quality,Type.x))
Streptococcus_phage_Cluster_assignment <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03///allPhages_vContact/Streptococcus_phage_Cluster_assignment.txt",col_types = cols(.default = "c")) %>% mutate(Size = as.double(Size), Quality = as.double(Quality)) %>% select(-c(Order,Family,Genus,Quality,Type.x))

# Streptococcus_phage_Cluster_assignment[grep("rmk",Streptococcus_phage_Cluster_assignment$Genome),"Genome"]

# Streptococcus_phage_Cluster_assignment %>% filter(!is.na(Type.y)) %>% select(VC) %>% table()
# sort(table(Streptococcus_phage_Cluster_assignment$VC),decreasing=T) %>% head(n=30)
# sort(table(Streptococcus_phage_Cluster_assignment$Type.y),decreasing=T)
##!!!!!!!!!!!!!!!spacer matrix FROM DADA
  # CRISPR_spacer_coverage_extended_wide <-  CRISPR_spacer_coverage %>% select(sample,Name,CPM) %>% spread(sample, CPM)
  
library(readr)
dada_spacer_count <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract/FinalSpacers/ForDADA2_woREverse/dada_spacer_count.txt","\t", escape_double = FALSE, trim_ws = TRUE)

metaspacer_info_tmp <- metaspacer_info %>% select(METAClusterName,spacer)

dada_spacer_count <- merge(dada_spacer_count,metaspacer_info_tmp,by.x="spacer",by.y="spacer",all.x = TRUE) %>% select(-spacer) #%>% rename( spacer = METAClusterName) 
##-------information strains explained
# library(readr)
# uniqueSpacers_count_Ref <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/ForDADA2/uniqueSpacers_count_Ref.txt","\t", escape_double = FALSE, trim_ws = TRUE)
# uniqueSpacers_count_Ref$Num_ref_explained <- (uniqueSpacers_count_Ref$REF_mst1>0)+(uniqueSpacers_count_Ref$REF_mst2>0)+(uniqueSpacers_count_Ref$REF_RMK202>0)
# uniqueSpacers_count_Ref$RefStrain_explained <- ifelse(uniqueSpacers_count_Ref$Num_ref_explained>1,"multiple Strains",ifelse(uniqueSpacers_count_Ref$REF_RMK202>0,"Meta_RMK202",ifelse(uniqueSpacers_count_Ref$REF_mst1>0,"mst 1",ifelse(uniqueSpacers_count_Ref$REF_mst2>0,"mst2","not explained")))) 
# table(uniqueSpacers_count_Ref$RefStrain_explained)
# uniqueSpacers_count_Ref_final <- uniqueSpacers_count_Ref %>% mutate(spacer = str_replace( spacerName, ">","")) %>% select(spacer,RefStrain_explained) 

#----new----

# unique(spacer_Infos_Sterm_final$sample)
spacer_Infos_Sterm_final <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/Sterm/spacer_Infos_Sterm_final.txt",  "\t", escape_double = FALSE, col_names = c("ClusterName","numSpacers_in_CLUSTER","sample","array","spc","ARRAY","SPACER"),  col_types = cols(SPACER = col_number()),  trim_ws = TRUE)
# spacer_Infos_Sterm_final <- unique(spacer_Infos_Sterm_final)

# spacer_Infos_Sterm_final_tmp <- spacer_Infos_Sterm_final %>% select(ClusterName,numSpacers_in_CLUSTER)
spacer_Infos_Sterm_final_tmp <- spacer_Infos_Sterm_final %>% select(ClusterName,numSpacers_in_CLUSTER,SPACER)

spacer_Infos_Sterm_final_tmp <- spacer_Infos_Sterm_final_tmp[!duplicated(spacer_Infos_Sterm_final_tmp[-3]),]

# spacer_Infos_Sterm_final %>% filter(ClusterName=="Cluster_1")
# spacer_Infos_Sterm_final_tmp<- spacer_Infos_Sterm_final_tmp %>% filter(ClusterName=="Cluster_1")

spacer_Infos_Sterm_final_new <- spacer_Infos_Sterm_final  %>% select(c(ClusterName,sample)) %>%  table() %>% as.data.frame() %>% spread(sample,Freq)
spacer_Infos_Sterm_final_new <- merge(spacer_Infos_Sterm_final_new,spacer_Infos_Sterm_final_tmp,by="ClusterName") %>% unique()
# table(spacer_Infos_Sterm_final_new$numSpacers_in_CLUSTER)
# spacer_Infos_Sterm_final_new2 <- merge(spacer_Infos_Sterm_final_new,spacer_Infos_Sterm_final_tmp,by="ClusterName") %>% unique()
# sum(is.na(spacer_Infos_Sterm_final_new$SPACER))
# uniqueSpacers_count_Ref$RefStrain_explained <- ifelse(spacer_Infos_Sterm_final_new$numSpacers_in_CLUSTER>1,"multiple Strains",ifelse(uniqueSpacers_count_Ref$REF_RMK202>0,"Meta_RMK202",ifelse(uniqueSpacers_count_Ref$REF_mst1>0,"mst 1",ifelse(uniqueSpacers_count_Ref$REF_mst2>0,"mst2","not explained"))))

# range(table(spacer_Infos_Sterm_final$sample))

uniqueSpacers_count_Ref_final <- spacer_Infos_Sterm_final_new
nrow(spacer_Infos_Sterm_final_new)
##-------merge

#   spacers_final <- merge(spacers_info,spacers_info_blast,by="spacer",all=TRUE)
#   spacers_final$explainedBLAST <- ifelse(!is.na(spacers_final$localPHAGE),"localPHAGE",ifelse(!is.na(spacers_final$localBAC),"localBAC",ifelse(!is.na(spacers_final$phageDB),"phageDB",ifelse(!is.na(spacers_final$BactDB),"phageDB","No-match"))))
# # table(spacers_final$explainedBLAST)
#   spacers_final <- merge(spacers_final,uniqueSpacers_count_Ref_final,by="spacer",all=TRUE)
#   spacers_final <- merge(spacers_final,Streptococcus_phage_Cluster_assignment,by.x="phageDB",by.y="Genome",all=TRUE)
#   spacers_final <- merge(spacers_final,spacer_Infos_Sterm_final_new,by.x="phageDB",by.y="Genome",all=TRUE)

##-------merge new
# sort(table(spacers_info_blast$phageDB),decreasing=TRUE)
# sort(table(spacers_info_blast$localPHAGE),decreasing=TRUE)

    spacers_final <- merge(metaspacer_info,spacers_info_blast,by.x="METAClusterName",by.y="spacer",all=TRUE)
  spacers_final$explainedBLAST <- ifelse(!is.na(spacers_final$localPHAGE),"localPHAGE",ifelse(!is.na(spacers_final$localBAC),"localBAC",ifelse(!is.na(spacers_final$phageDB),"phageDB",ifelse(!is.na(spacers_final$BactDB),"BactDB","No-match"))))
  spacers_final[which(spacers_final$explainedBLAST=="No-match"),]
  
#spacers_final$explainedBLAST <- ifelse(!is.na(spacers_info_blast$localPHAGE),"localPHAGE",ifelse(!is.na(spacers_info_blast$localBAC),"localBAC",ifelse(!is.na(spacers_info_blast$phageDB),"phageDB",ifelse(!is.na(spacers_info_blast$BactDB),"BacDB","No-match"))))
table(spacers_final$explainedBLAST)

spacers_final[which(spacers_final$explainedBLAST=="BactDB"),]
sum(is.na(spacers_final$explainedBLAST))


 # table(spacers_final$VC)
  
  # dim(uniqueSpacers_count_Ref_final)
  # uniqueSpacers_count_Ref_final$numSpacers_in_CLUSTER==0
# # nrow(spacers_final)
# nrow(uniqueSpacers_count_Ref_final)
# sum(is.na(spacers_final$phageDB))
# spacers_final$phageDB

  spacers_final$phageDB <- revalue(spacers_final$phageDB,c("Lactococcus lactis phage BK5-T"="Lactococcus phage BK5-T","Streptococcus thermophilus bacteriophage 7201"="Streptococcus virus 7201","Streptococcus thermophilus bacteriophage Sfi19"="Streptococcus virus Sfi19","Streptococcus thermophilus temperate bacteriophage O1205"="Streptococcus virus O1205"))
  spacers_final <- merge(spacers_final,uniqueSpacers_count_Ref_final,by.x="ClusterINFO",by.y="ClusterName",all=TRUE)
   spacers_final <- merge(spacers_final,Streptococcus_phage_Cluster_assignment,by.x="phageDB",by.y="Genome",all.x=TRUE)
   
   spacers_final[which(is.na(spacers_final$VC)&!is.na(spacers_final$phageDB)),]
 #   
  # tmp <- spacers_final[which(is.na(spacers_final$VC)&!is.na(spacers_final$phageDB)),] %>% select(phageDB)
 #

  #  for (i in 1:nrow(tmp)) { print(grep(paste0(" ",tmp[i,"phageShort"],"$"),Streptococcus_phage_Cluster_assignment$Genome))}
  # 
  # i=9
  # tmp[i,"phageDB"]
  # Streptococcus_phage_Cluster_assignment[grep(paste0(" ",tmp[i,"phageShort"],"$"),Streptococcus_phage_Cluster_assignment$Genome),"Genome"]
  # 

  
  # grep(tmp$phageShort,spacers_final$phageDB)
  
  # tmp
   
      # Streptococcus_phage_Cluster_assignment[grep("7201$",Streptococcus_phage_Cluster_assignment$Genome),]

  # spacers_final <- merge(spacers_final,spacer_Infos_Sterm_final_new,by.x="ClusterINFO",by.y="ClusterName",all=TRUE)
  
##------------------add unique
spacers_final$RefStrain_explained <- ifelse(is.na(spacers_final$numSpacers_in_CLUSTER),"only Metagenome",ifelse(spacers_final$numSpacers_in_CLUSTER>1,"multiple Strains",ifelse(spacers_final$numSpacers_in_CLUSTER==1,"unique spacer","not explained")))
# table(spacers_final$RefStrain_explained )

# hist(spacers_final$numSpacers_in_CLUSTER)
##--------------quick analysis  
table(spacers_final$VC) %>% sort()
table(spacers_final$Type.y) %>% sort()
# table(spacers_final$Type.x) %>% sort()
Streptococcus_phage_Cluster_assignment %>% filter(VC=="151_0")
Streptococcus_phage_Cluster_assignment %>% filter(VC=="92_0")
Streptococcus_phage_Cluster_assignment %>% filter(VC=="385_0")
Streptococcus_phage_Cluster_assignment %>% filter(VC=="316_1")
Streptococcus_phage_Cluster_assignment %>% filter(VC=="251_1")

# Streptococcus_phage_Cluster_assignment[grep("Javan63",Streptococcus_phage_Cluster_assignment$Genome),]
spacers_final %>% filter(!is.na(phageDB)) %>% filter(is.na(VC))

  write.table(spacers_final, "~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/log/Clusters_spacers_final.txt",na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names =FALSE)


###-----------------------------------
##num spacer in cluster
 ###----------------------------------- 
  
nrow(spacers_final)

table(spacers_final$numSpacers_in_CLUSTER)
sum(is.na(spacers_final$numSpacers_in_CLUSTER))
table(spacers_final$SPACER)
spacers_final$ARRAYINFO

ggplot(spacers_final,aes(x=SPACER,fill=explainedBLAST,color=explainedBLAST))+geom_bar()+facet_wrap(~ARRAYINFO)+theme_classic()

DF <- spacers_final %>% select(explainedBLAST,SPACER,ARRAYINFO) %>% filter(!is.na(SPACER))


DF$explainedBLAST = factor(DF$explainedBLAST, levels=c("No-match" ,"BactDB","localBAC" ,  "phageDB","localPHAGE"))

colorsssss <- rev(c("darkcyan","darkturquoise","goldenrod3","yellow","lightgray"))
  

spacerlocation_explained <- ggplot(DF,aes(x = SPACER,  fill = explainedBLAST)) + 
  geom_bar(position = "fill") +labs(x="Spacer position",y="percent of spacers")+
  scale_y_continuous(labels = scales::percent)+facet_wrap(~ARRAYINFO,scales = "free")+theme_classic()+scale_fill_manual(values = colorsssss)+theme(legend.position = "none")
spacerlocation_explained



 svg("~/Desktop/Projects/2019_RMK202_analysis/plot/spacerlocation_explained.svg",width=5,height=3)

spacerlocation_explained#   
   dev.off()

```

```{bash}
##------------------for PAm idenfitfaction
scp ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/log/Clusters_spacers_final.txt vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/log/.
```

###### find location of oldest phage spacers
```{r}

metaspacer_info <- read_delim("/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/uniqueSpacers_count_Ref_wOLDstrains.txt", "\t", escape_double = FALSE, trim_ws = TRUE)  
metaspacer_info$METAClusterName <- gsub(">","",metaspacer_info$METAClusterName)
 spacers_info <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers//ForDADA2_woREverse//spacers_info.txt", "\t", escape_double = FALSE, trim_ws = TRUE)
##-------information spacer blast
# spacers_info_blast <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/all_differentBlastHits.txt", "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("DB","spacer","assigned")) %>% spread(., DB, assigned)
spacers_info_blast <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/all_differentBlastHits.txt", "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("DB","spacer","assigned")) %>% spread(., DB, assigned)

combined_spacer_01 <- merge(spacers_info_blast,metaspacer_info,by.x="spacer",by.y="METAClusterName",all.y=TRUE)
nrow(spacers_info_blast)
nrow(metaspacer_info)
nrow(combined_spacer_01)
# unique(metaspacer_info$METAClusterName) %>% length()
# unique(spacers_info_blast$METAClusterName) %>% length()

###---------------------
##spacer location
###---------------------

orientation_CRISPRarrays <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/Sterm/allGenomes/orientation_CRISPRarrays.txt", "\t", escape_double = FALSE, col_names = c("sample","orientation","ARRAY"),    trim_ws = TRUE)
 
 
# uniqueSpacersss_again <- spacer_Infos_Sterm_final %>% filter(spacerUniqueness=="unique")
 
TotSpacer <- spacer_Infos_Sterm_final %>% group_by(sample,ARRAY) %>%  summarize(totalSpacersARRAY=max(SPACER))
uniqueSpacersss_extended_tmp001_large <-  merge(spacer_Infos_Sterm_final, TotSpacer, by=c("sample", "ARRAY")) # %>%filter((sample!="24853")) # filter(sample!=24853&ARRAY!="CRISPR array 2")

# uniqueSpacersss_extended_tmp002 <-  merge(uniqueSpacersss_again, TotSpacer, by=c("sample", "ARRAY"))  %>%filter((ARRAY=="CRISPR array 2")) # filter(sample!=24853&ARRAY!="CRISPR array 2")
# uniqueSpacersss_extended_tmp02 <- rbind(uniqueSpacersss_extended_tmp001_large,uniqueSpacersss_extended_tmp002)

uniqueSpacersss_extended_tmp2 <-  merge(uniqueSpacersss_extended_tmp001_large, orientation_CRISPRarrays, by=c("sample", "ARRAY")) %>% mutate(absDistanceLeader = ifelse(orientation=="For", SPACER,totalSpacersARRAY-SPACER))

uniqueSpacersss_extended <- uniqueSpacersss_extended_tmp2 %>% mutate(absDistanceLeader = ifelse(ARRAY=="CRISPR array 3", absDistanceLeader,totalSpacersARRAY-absDistanceLeader))

##---------------merge-----

FINAL_SPACER_DISTANCE <- merge(uniqueSpacersss_extended,combined_spacer_01,by.x="ClusterName",by.y="ClusterINFO",all.x=TRUE) #%>% filter(!is.na(localPHAGE)) 
oldestspacers <- FINAL_SPACER_DISTANCE%>% filter(!is.na(localPHAGE))  %>% group_by(sample) %>% dplyr::summarise(oldestSpacer=max(absDistanceLeader)) 

min(oldestspacers$oldestSpacer)/0.034 
max(oldestspacers$oldestSpacer)/0.034 
oldestspacers$oldestSpacer_time <- oldestspacers$oldestSpacer/0.034 
 oldestspacers$sample = factor(oldestspacers$sample, levels=c("13492","13491","24854","24737","13500","13493","13498","SMAG","24853","13494","24855","S72","13499c1","13499","13499c2","24738","24740","S50", "13497", "24739","13495","13496"))

 colorsss <- rev(c("#99CCFF","#00FFFF","#0000FF","#6699FF"))
 
oldestspacers$lineages <-  c(rep("lineage 1",9),rep("lineage 2",3),rep("lineage 3",6),rep("lineage 4",4))


spacerOrigin <- ggplot(oldestspacers,aes(x=sample,y=oldestSpacer_time,fill=lineages))+geom_bar(stat="identity")+theme_classic()+scale_fill_manual(values=colorsss)+labs(x="",y="oldest spacers against\nthe co-existing phages [# generations]")+theme(legend.title = element_blank(),axis.text.x = element_text(angle = 75, hjust = 1,size=9))

spacerOrigin

   svg("~/Desktop/mid_thesis/report/figures/20200430/chapter1/20201006/supplement/Oldestspacers_perStrain.svg",width=5,height=2.5)

  spacerOrigin

  dev.off()

```



######bwa check plots

```{r}
library(ggplot2)
##---------------------------
###correlation bwa and dada
##---------------------------
# head(dada_spacer_count)
# dada_ggprep <- bwaPROTOSPACERCount_wide %>% gather(.,key="sample",value="DadaCount",-Name) %>% add_column(method="protospacer")
# bwaSPACERCount_ggprep <- bwaSPACERCount_wide %>% gather(.,key="sample",value="bwaCount",-Name) %>% add_column(method="spacer") 
# countComparison <- merge(dada_ggprep,bwaSPACERCount_ggprep,by.x=c("METAClusterName","sample"),by.y=c("Name","sample")) %>% select(-method.x,-method.y)
# ggplot(countComparison,aes(x=bwaCount,y=DadaCount))+geom_point()+theme_classic()

##---------------------------
###correlation protospacer and spacer
# ##---------------------------
bwaSPACERCount_ggprep <- bwaSPACERCount_wide %>% gather(.,key="sample",value="bwaCount",-Name) %>% add_column(method="bwa") 

# dada_ggprep <- dada_spacer_count %>% gather(.,key="sample",value="DadaCount",-spacer) %>% add_column(method="dada")
 bwaPROTOSPACERCount_ggprep <- bwaPROTOSPACERCount_wide %>% gather(.,key="sample",value="protospacer",-Name) %>% add_column(method="bwaProto")
countComparison <- merge(bwaPROTOSPACERCount_ggprep,bwaSPACERCount_ggprep,by=c("Name","sample")) %>% select(-method.x,-method.y)
countComparison_ggprep <- merge(countComparison,spacers_final,by.x="Name",by.y="METAClusterName",all=TRUE)
countComparison_ggprep$protospacer <- countComparison_ggprep$protospacer+0.1
countComparison_ggprep$bwaCount <- countComparison_ggprep$bwaCount+0.1

ggplot(countComparison_ggprep,aes(x=protospacer,y=bwaCount,color=RefStrain_explained))+geom_point()+theme_classic()+coord_trans(y="log2",x="log2")#+facet_wrap(~explainedBLAST)

ggplot(countComparison_ggprep,aes(x=protospacer,y=bwaCount,color=explainedBLAST))+geom_point()+theme_classic()+coord_trans(y="log2",x="log2")+ geom_smooth(method = "lm", fill = NA,se = TRUE)
#+facet_wrap(~explainedBLAST)
# countComparison_ggprep$explainedBLAST

##---------------------------
#REMOVE low samples
##---------------------------
table(countComparison_ggprep$explainedBLAST)
countComparison_ggprep_02 <- countComparison_ggprep %>% filter(!is.na(protospacer)) %>% filter(!is.na(bwaCount)) %>% filter(protospacer>0.5)%>% filter(bwaCount>0.5)%>% filter(explainedBLAST %in% c("localPHAGE","phageDB"))
table(countComparison_ggprep_02$explainedBLAST)


countComparison_ggprep_02$explainedBLAST = factor(countComparison_ggprep_02$explainedBLAST, levels=c("localPHAGE" ,"phageDB"))

colorsssss <- c("darkcyan","darkturquoise")

# myplot <- ggplot(countComparison_ggprep_02,aes(x=protospacer,y=bwaCount,color=explainedBLAST))+geom_point()+theme_classic()+ geom_smooth(method = "lm", fill = NA)
#+facet_wrap(~explainedBLAST)+coord_trans(y="log2",x="log2")
#
# myplot
# my.formula <- bwaCount ~ protospacer
my.formula <- y ~ x
library(ggpmisc)
plot <- ggplot(countComparison_ggprep_02,aes(x=protospacer,y=bwaCount,color=explainedBLAST))+geom_point()+theme_classic()+
  scale_fill_manual(values = colorsssss)+
  scale_color_manual(values = colorsssss)+
labs(x="protospacer [cpm]",y="spacers [cpm]")+
geom_point() +
scale_y_log10(breaks=c(1,10,100,1000,5000)) +
scale_x_log10(breaks=c(1,10,100,1000,10000,100000)) +
geom_smooth(method="lm")+
  stat_poly_eq(formula = my.formula, 
               aes(label = paste(..eq.label.., ..rr.label.., sep = "~~~")), 
               parse = TRUE,)   
plot

  # svg("~/Desktop/Manuscripts/2019_RMK202/Figures/S05_rawReadMapping.svg",width=6,height=4)
    png("~/Desktop/Projects/2019_RMK202_analysis/plot/spacer_protospacer_regression.png", width = 2800, height = 2500,res=300)


plot
 dev.off()

 ##--------------------
 ##histogramm
 
 plot_density_x <- ggplot(countComparison_ggprep_02,aes(x=protospacer,color=explainedBLAST,fill=explainedBLAST))+geom_density(alpha=0.5)+theme_classic()+
  scale_fill_manual(values = colorsssss)+
  scale_color_manual(values = colorsssss)+
labs(x="",y="density")+scale_x_log10(breaks=c(1,10,100,1000,10000,100000))  +theme(legend.position = "none",axis.ticks = element_blank(),axis.text = element_blank(),axis.title.x=element_blank())
plot_density_x 

 plot_density_y <- ggplot(countComparison_ggprep_02,aes(x=bwaCount,color=explainedBLAST,fill=explainedBLAST))+geom_density(alpha=0.5)+theme_classic()+
  scale_fill_manual(values = colorsssss)+
  scale_color_manual(values = colorsssss)+
labs(x="",y="density")+scale_x_log10(breaks=c(1,10,100,1000,5000)) +theme(legend.position = "none",axis.ticks = element_blank(),axis.text = element_blank(),axis.title.y=element_blank())+coord_flip()
plot_density_y 





###---------------------------
##make mix plot
###---------------------------

library(ggpubr)
library(patchwork)


# (plot+theme(legend.position = "none"))+plot_density_y+plot_layout(widths =c(3,1))
(plot_density_x+plot_spacer()+plot_layout(widths = c(3,1)))/((plot+theme(legend.position = "none")+plot_density_y)+plot_layout(widths =c(3,1)))+plot_layout(heights = c(1,3))



  svg("~/Desktop/Projects/2019_RMK202_analysis/plot/spacer_vs_proto_01.svg",width=8,height=6)
(plot+theme(legend.position = "none"))+plot_density_y+plot_layout(widths =c(3,1))
 dev.off()
 
   svg("~/Desktop/Projects/2019_RMK202_analysis/plot/spacer_vs_proto_02.svg",width=6,height=8)
plot_density_x/(plot+theme(legend.position = "none"))+plot_layout(heights = c(1,3))
 dev.off()
    svg("~/Desktop/Projects/2019_RMK202_analysis/plot/spacer_vs_proto_03.svg",width=6,height=6)
(plot_density_x+plot_spacer()+plot_layout(widths = c(3,1)))/((plot+theme(legend.position = "none")+plot_density_y)+plot_layout(widths =c(3,1)))+plot_layout(heights = c(1,3))
 dev.off()


 ##--------------------
 ##boxpot
 ##--------------------

 ggplot(countComparison_ggprep_02,aes(x=explainedBLAST,y=protospacer,color=explainedBLAST,fill=explainedBLAST))+geom_boxplot(alpha=0.5)

library(ggpubr)
library(patchwork)
 plot_box_x <- ggplot(countComparison_ggprep_02,aes(x=explainedBLAST,group=explainedBLAST,y=protospacer,color=explainedBLAST,fill=explainedBLAST))+geom_boxplot(alpha=0.5)+theme_classic()+
labs(x="",y="protospacers [cpm]")+scale_y_log10(breaks=c(1,10,100,1000,10000,100000))  +theme(legend.position = "none")+ stat_compare_means(method = "wilcox.test")
plot_box_x 

 plot_box_y <- ggplot(countComparison_ggprep_02,aes(x=explainedBLAST,group=explainedBLAST,y=bwaCount,color=explainedBLAST,fill=explainedBLAST))+geom_boxplot(alpha=0.5)+theme_classic()+
labs(x="",y="spacers [cpm]")+scale_y_log10(breaks=c(1,10,100,1000,10000,100000))  +theme(legend.position = "none")+ stat_compare_means(method = "wilcox.test")
plot_box_y 

plot_box_x+plot_box_y

 png("~/Desktop/Projects/2019_RMK202_analysis/plot/spacer_protospacer_regression_boxplot.png", width = 2000, height = 2500,res=300)

plot_box_x+plot_box_y
 dev.off()

 
 source("/home/vincent/Desktop/ScriptRepository/R_functions-master/R_functions/g_legend.R") ##for plotting only legends
# legend_1 <- g_legend(p_mst1_sterm_woModi_clustered)
 
 library(gridExtra)
grid.arrange(p1,p2,p3,p4, ncol=2, heights=c(1,2))

  svg("~/Desktop/presenations/my_presentations/20190708_groupmeeting/figures/polishing_K1.svg",width=6,height=5)
grid.arrange(p1,p2,p3,p4, ncol=2, heights=c(2,3))
 dev.off()
 
library(gridExtra)
grid.arrange(p1,p2,p3,p4, ncol=2, heights=c(1,2))

  svg("~/Desktop/presenations/my_presentations/20190708_groupmeeting/figures/polishing_K1.svg",width=6,height=5)
grid.arrange(p1,p2,p3,p4, ncol=2, heights=c(2,3))
 dev.off()

```




here I have created the spacer matrix and the sample file and spacer file
these will continiously be propagated


```{r}
##---------------put infomration of location into plot
  # clusterCRISPRs_keep_extended$REFspacer
# CRISPR_spacer_coverage$ClusterName %in%  clusterCRISPRs_keep_extended$REFspacer

##---------------------I do this to not get mixed up between the actual cpm coverage gotten by mapping and the grep within the clusters which I am removing here
# clusterCRISPRs_keep_extended_new <- clusterCRISPRs_keep_extended %>% select(-lyo202_96,-L_202_2012,-L_202_2014,-RMK202,-Konserve_202,-Versand_202,-di_K2_6h,-th_K2_6h)
# colnames(clusterCRISPRs_keep_extended)
  # CRISPR_spacer_coverage_extended <- merge(CRISPR_spacer_coverage,clusterCRISPRs_keep_extended_new,by.y="REFspacer",by.x = "ClusterName",all.x = TRUE)
# CRISPR_spacer_coverage_extended$grouping
# CRISPR_spacer_coverage_extended$grouping
  


 
  # CRISPR_spacer_coverage_extended_wide <-  spread(CRISPR_spacer_coverage_extended, Name, CPM)

 ##---------------add blast information
# clusterCRISPRs_keep_extended[,c(1,3)] %>% duplicate()

# 
# longCrispr_new_tmp <- longCrispr_new
# longCrispr_new_tmp$array <- longCrispr_new_tmp$array %>% gsub("a","",.) %>% as.numeric() %>% as.character()
# longCrispr_new_tmp$Name <- paste0(longCrispr_new_tmp$sample,":a",longCrispr_new_tmp$array,"s",as.character(as.numeric(longCrispr_new_tmp$spacer)))
# longCrispr_new_tmp <- longCrispr_new_tmp[,c(1,10)]
# # longCrispr_new_tmp$spacerName <- longCrispr_new_tmp$spacerName %>% gsub(":a",":ary",.) %>% gsub("s",":ary",.)
# CRISPRblast_all_extended_forPlotting <- CRISPRblast_all_extended[!duplicated(CRISPRblast_all_extended[,c(1,3)]),c(1,3)]
# CRISPRblast_all_extended_forPlotting$Name <- CRISPRblast_all_extended_forPlotting$Name %>% gsub(":ary",":a",.) %>% gsub("spc","s",.)  %>% gsub("Versand","Ver",.) %>% gsub("Konserve","Kon",.) %>% gsub("Lyo","L",.)
# 
# CRISPRblast_all_extended_forPlotting$Name %in% longCrispr_new_tmp$Name
# 
# CRISPRblast_all_extended_forPlotting_02 <- merge(CRISPRblast_all_extended_forPlotting,longCrispr_new_tmp,by.y="Name",by.x = "Name",all.x = TRUE)
# CRISPRblast_all_extended_forPlotting_02 <- merge(CRISPRblast_all_extended_forPlotting,longCrispr_new_tmp,by.y="Name",by.x = "Name",all.x = TRUE)
#   CRISPR_spacer_coverage_02 <- merge(CRISPR_spacer_coverage_extended,CRISPRblast_all_extended_forPlotting_02,by.y="clusterName",by.x = "ClusterName",all.x = TRUE)


# CRISPRblast_all_extended_forPlotting_02$clusterName %>% unique() %>% length()
  ##---------------plot
  # 
  # CRISPR_spacer_coverage_extended[which(CRISPR_spacer_coverage_extended$CPM>1.5),]
  #   CRISPR_spacer_coverage_extended <- CRISPR_spacer_coverage_extended[which(CRISPR_spacer_coverage_extended$Cluster!="Cluster_115"),]
  # 
  # CRISPR_spacer_coverage_extended$RefStrain <- ifelse(CRISPR_spacer_coverage_extended$all_REF>1,"multiple Strains",ifelse(CRISPR_spacer_coverage_extended$REF_RMK202>0,"Meta RMK202",ifelse(CRISPR_spacer_coverage_extended$REF_mst1>0,"mst 1",ifelse(CRISPR_spacer_coverage_extended$REF_mst2>1,"mst2","not explained"))))
  # 
##-----------------------------normalize with number of reads mapping to spacers in that array
# CRISPR_spacer_coverage_extended$numReads
# normalizeWithCRISPRmapping <- aggregate(. ~sample+adap, data=CRISPR_spacer_coverage_extended[,c("sample","numReads","adap")], sum, na.rm=TRUE)
# 
# 
# CRISPR_spacer_coverage_extended <- merge(CRISPR_spacer_coverage_extended,normalizeWithCRISPRmapping,by=c("sample","adap"),all.x = TRUE)
# CRISPR_spacer_coverage_extended$normalizedCount <- 1000000*(CRISPR_spacer_coverage_extended$numReads.x/CRISPR_spacer_coverage_extended$numReads.y)
```

#####SPACERS overtime

```{r}
##--------------------------
##gather data from plotting
##--------------------------
# head(dada_spacer_count)
# dada_ggprep <- dada_spacer_count %>% gather(.,key="sample",value="DadaCount",-METAClusterName) %>% group_by(METAClusterName) %>% dplyr::summarise(DadaCount=median(DadaCount)) 
# # dada_ggprep <- dada_spacer_count %>% gather(.,key="sample",value="DadaCount",-METAClusterName) %>% group_by(METAClusterName) %>% dplyr::summarise(DadaCount=median(DadaCount),mean=mean(DadaCount)) 
# 
# 
# # dada_ggprep <- dada_spacer_count %>% gather(.,key="sample",value="DadaCount",-METAClusterName) 
# 
dada_ggprep_plot_time <- merge(dada_ggprep,spacers_final,by.x="METAClusterName",by.y="METAClusterName")
# 
dada_ggprep_plot_time$RefStrain_explained
# # CRISPR_spacer_long <- gather(CRISPR_spacer_coverage_extended_wide, sample, CPM,c("Lyo\n1996", "Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2"), factor_key=TRUE,na.rm = TRUE) %>% filter(Name!="d2_6h:Ra2:s289")
# ##create a named vector for revalue
# # test <- setNames(sampleDF$final,sampleDF$shortfinal)
# ##revalue
# # dada_ggprep_plot_time$sample  <- plyr::revalue(dada_ggprep_plot_time$sample, test)
# dada_ggprep_plot_time$DadaCount <- dada_ggprep_plot_time$DadaCount+10
# 
#       p3 <- ggplot(dada_ggprep_plot_time,aes(x=sample,y=DadaCount,group=spacer,color=RefStrain_explained))+ geom_line(size=0.4, alpha=.4) +theme_classic()+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9))+facet_wrap(~RefStrain_explained)+coord_trans(y="log2")
#  
#   p3
# 
#     
      # ggplot(dada_ggprep_plot_time,aes(x=origin,y=mean))+geom_violin()+theme_classic()+coord_trans(y="log2")

    plotSpacerAbundance <- ggplot(dada_ggprep_plot_time,aes(x=origin,y=mean))+geom_boxplot()+labs(y="mean spacer count")+theme_classic()+coord_trans(y="log2")+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),axis.title.x = element_blank())

        t.test(mean ~ origin, data = dada_ggprep_plot_time)$p.value
        t.test(mean ~ origin, data = dada_ggprep_plot_time)

#    svg("~/Desktop/mid_thesis/report/figures/20200430/chapter1/data_spacerCoverage.svg",width=2,height=4)
# 
#   plotSpacerAbundance
#   
#   dev.off()
        
        
  ###-------------------------
 ## spacer count
    ###-------------------------
  # dada_ggprep <- dada_spacer_count %>% gather(.,key="sample",value="DadaCount",-spacer) %>% add_column(method="dada")
bwaSPACERCount_ggprep <- bwaSPACERCount_wide %>% gather(.,key="sample",value="bwaCount",-Name) %>% add_column(method="bwa") #%>% rename(METAClusterName="Name") # %>% rename(Name="METAClusterName") 
bwaSPACERCount_ggprep_plot_time <- merge(bwaSPACERCount_ggprep,spacers_final,by.x="Name",by.y="METAClusterName") %>% filter(!sample %in% c("di_K2_6h","th_K2_8h"))

# table(bwaSPACERCount_ggprep_plot_time$sample)

  
  bwaSPACERCount_ggprep_plot_time$bwaCount <- bwaSPACERCount_ggprep_plot_time$bwaCount+10

  
       p4 <- ggplot(bwaSPACERCount_ggprep_plot_time,aes(x=sample,y=bwaCount,group=spacer,color=RefStrain_explained))+ geom_line(size=0.4, alpha=.4) +theme_classic()+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9))+facet_wrap(~RefStrain_explained)+coord_trans(y="log10")
 
  p4
  
       p3 <- ggplot(bwaSPACERCount_ggprep_plot_time,aes(x=sample,y=bwaCount,group=spacer,color=explainedBLAST))+ geom_line(size=0.4, alpha=.4) +theme_classic()+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9))+facet_wrap(~explainedBLAST)+coord_trans(y="log10")
 
  p3
  
  ###--------------------label group-specific spacers
  
  ##-------------------------
##only shared between strains in cluster
##-------------------------

tmp1 <- bwaSPACERCount_ggprep_plot_time %>% filter(numSpacers_in_CLUSTER=="2"& get("13494")>"0"& get("S72")>"0") %>%select(METAClusterName) %>%  add_column(explaines="group 1") %>% unique()# %>% select(METAClusterName) %>% unique() %>% nrow()

tmp2 <- bwaSPACERCount_ggprep_plot_time %>% filter(numSpacers_in_CLUSTER=="6"& get("13498")>"0"& get("13500")>"0"& get("SMAG")>"0"& get("13491")>"0"& get("13492")>"0"& get("13493")>"0")  %>%select(METAClusterName) %>%  add_column(explaines="group 2") %>% unique()

tmp3 <- bwaSPACERCount_ggprep_plot_time %>% filter(numSpacers_in_CLUSTER=="3"& get("13495")>"0"& get("13496")>"0"& get("13497")>"0") %>%select(METAClusterName) %>%  add_column(explaines="group 3") %>% unique()

tmp4 <- bwaSPACERCount_ggprep_plot_time %>% filter(numSpacers_in_CLUSTER=="3"& get("13499c1")>"0"& get("13499c2")>"0"& get("S50")>"0") %>%select(METAClusterName) %>%  add_column(explaines="group 4") %>% unique()

tmpAll <- rbind(tmp1,tmp2,tmp3,tmp4)
  table(tmpAll$explaines)
bwaSPACERCount_ggprep_plot_time$Refs_explained <- bwaSPACERCount_ggprep_plot_time$RefStrain_explained
bwaSPACERCount_ggprep_plot_time <- merge(bwaSPACERCount_ggprep_plot_time,tmpAll,by="METAClusterName",all.x=TRUE)
groupingRMK202Strains <- data.frame(strain=c("13494","S72","13498","13500","SMAG","13491","13492","13493","13495","13496","13497","13499c1","13499c2","S50"),group=c(rep("group 1",2),rep("group 2",6),rep("group 3",3),rep("group 4",3)))

# bwaSPACERCount_ggprep_plot_time <- merge(bwaSPACERCount_ggprep_plot_time,tmpAll,by="METAClusterName",all.x=TRUE)


# bwaSPACERCount_ggprep_plot_time$Refs_explained_final <- NA
table(bwaSPACERCount_ggprep_plot_time$RefStrain_explained)
#   i <- 205
table(bwaSPACERCount_ggprep_plot_time$explaines)
for (i in which(bwaSPACERCount_ggprep_plot_time$numSpacers_in_CLUSTER=="1")) {
  # colnames(bwaSPACERCount_ggprep_plot_time)
  tmpName <- as.character(colnames(bwaSPACERCount_ggprep_plot_time)[17+which(bwaSPACERCount_ggprep_plot_time[i,18:31]==1)])
  bwaSPACERCount_ggprep_plot_time[i,"explaines"] <- as.character(groupingRMK202Strains[which(groupingRMK202Strains$strain==tmpName),"group"])

#   
}
table(bwaSPACERCount_ggprep_plot_time$explaines)

for (i in which(is.na(bwaSPACERCount_ggprep_plot_time$explaines)&(bwaSPACERCount_ggprep_plot_time$RefStrain_explained=="multiple Strains"))) {
  # colnames(bwaSPACERCount_ggprep_plot_time)
  # tmpName <- as.character(colnames(bwaSPACERCount_ggprep_plot_time)[17+which(bwaSPACERCount_ggprep_plot_time[i,18:31]==1)])
  bwaSPACERCount_ggprep_plot_time[i,"explaines"] <- "multiple groups"

#   
}
table(bwaSPACERCount_ggprep_plot_time$explaines)


# bwaSPACERCount_ggprep_plot_time$explaines <- revalue(bwaSPACERCount_ggprep_plot_time$explaines,NA="only Meta")

bwaSPACERCount_ggprep_plot_time$explaines <- fct_explicit_na(bwaSPACERCount_ggprep_plot_time$explaines, na_level = "only Meta")
table(bwaSPACERCount_ggprep_plot_time$explaines)


###------------Rename  
  
bwaSPACERCount_ggprep_plot_time <- bwaSPACERCount_ggprep_plot_time  %>% mutate(sample=recode(sample, lyo202_96="Lyo 1996",Lyo_202_2012="Lyo 2012", Lyo_202_2014="Lyo 2014", Konserve_202="working stock", RMK202="starter culture\n2012", Versand_202="starter culture\n2018","di_K2_6h"="cheesemaking\nday1","th_K2_8h"="cheesemaking\nday2"))

bwaSPACERCount_ggprep_plot_time$sample = factor(bwaSPACERCount_ggprep_plot_time$sample, levels=c("Lyo 1996","Lyo 2012","Lyo 2014","working stock","starter culture\n2012","starter culture\n2018","cheesemaking\nday1","cheesemaking\nday2"))

###------------plot  

colorzzz <- c("#0000FF","#6699FF","#99CCFF","#00FFFF","brown","#b8b8b8")

       plines <- ggplot(bwaSPACERCount_ggprep_plot_time,aes(x=sample,y=bwaCount,group=spacer,color=explaines))+ geom_line(size=0.4, alpha=.8) +theme_classic()+theme(axis.text.x = element_text(angle = 45, hjust = 1,size=5.5),axis.text.y = element_text(size=7),legend.position = "none",axis.title.x = element_blank(),strip.text = element_text(size=7))+facet_wrap(~explaines,ncol = 6,fill=colorzzz)+coord_trans(y="log10")+scale_y_continuous(breaks =c(100,500,1000,2000,4000,5000),labels=c(100,500,1000,2000,4000,5000))+labs(y="normalized coverage")+scale_fill_manual(values=colorzzz)+
   scale_color_manual(values=colorzzz)

 
  plines
  
        pboxes <- ggplot(bwaSPACERCount_ggprep_plot_time,aes(x=explaines,y=bwaCount,fill=explaines))+ geom_boxplot() +theme_classic()+coord_trans(y="log10")+scale_y_continuous(breaks =c(100,500,1000,2000,4000,5000),labels=c(100,500,1000,2000,4000,5000))+theme(legend.position = "none",axis.title.x = element_blank(),axis.title.y = element_blank(),axis.text.x = element_text(angle = 45, hjust = 1,size=5.5))+scale_fill_manual(values=colorzzz)+
   scale_color_manual(values=colorzzz)

 
  pboxes 
  
   svg("~/Desktop/Projects/2019_RMK202_analysis/plot/Coverage_plots.svg",width=10,height=5.5)

  plines+pboxes+plot_layout(widths =c(6,1))
  
  dev.off()
bwaSPACERCount_ggprep_plot_time$ARRAYINFO

         p3 <- ggplot(bwaSPACERCount_ggprep_plot_time,aes(x=sample,y=bwaCount,group=spacer,color=explaines))+ geom_line(size=0.4, alpha=.8) +theme_classic()+theme(axis.text.x = element_text(angle = 45, hjust = 1,size=5.5),legend.position = "none",axis.title.x = element_blank(),strip.text = element_text(size=7))+facet_wrap(~ARRAYINFO,ncol = 6)+coord_trans(y="log10")+scale_y_continuous(breaks =c(100,500,1000,2000,4000,5000),labels=c(100,500,1000,2000,4000,5000))+labs(y="normalized coverage")

 
  p3
  
 
         p3 <- ggplot(bwaSPACERCount_ggprep_plot_time,aes(x=sample,y=bwaCount,group=spacer,color=explaines))+ geom_line(size=0.4, alpha=.8) +theme_classic()+theme(axis.text.x = element_text(angle = 45, hjust = 1,size=5.5),legend.position = "right",axis.title.x = element_blank(),strip.text = element_text(size=7))+facet_wrap(~explainedBLAST,ncol = 6)+coord_trans(y="log10")+scale_y_continuous(breaks =c(100,500,1000,2000,4000,5000),labels=c(100,500,1000,2000,4000,5000))+labs(y="normalized coverage")

 
  p3
  bwaSPACERCount_ggprep_plot_time$Type.y
      p3 <- ggplot(bwaSPACERCount_ggprep_plot_time,aes(x=sample,y=bwaCount,group=spacer,color=Type.y))+ geom_line(size=0.4, alpha=.8) +theme_classic()+theme(axis.text.x = element_text(angle = 45, hjust = 1,size=5.5),legend.position = "right",axis.title.x = element_blank(),strip.text = element_text(size=7))+facet_wrap(~explainedBLAST,ncol = 6)+coord_trans(y="log10")+scale_y_continuous(breaks =c(100,500,1000,2000,4000,5000),labels=c(100,500,1000,2000,4000,5000))+labs(y="normalized coverage")

 
  p3
  
      p3 <- ggplot(bwaSPACERCount_ggprep_plot_time,aes(x=sample,y=bwaCount,group=spacer,color=Type.y))+ geom_line(size=0.4, alpha=.8) +theme_classic()+theme(axis.text.x = element_text(angle = 45, hjust = 1,size=5.5),legend.position = "right",axis.title.x = element_blank(),strip.text = element_text(size=7))+facet_wrap(~Type.y,ncol = 6)+coord_trans(y="log10")+scale_y_continuous(breaks =c(100,500,1000,2000,4000,5000),labels=c(100,500,1000,2000,4000,5000))+labs(y="normalized coverage")

 
  p3
  
  ####------------------
  ##add the local phage name 
  
  bwaSPACERCount_ggprep_plot_time$explainedBLAST_extended <- bwaSPACERCount_ggprep_plot_time$explainedBLAST
  table(bwaSPACERCount_ggprep_plot_time$explainedBLAST_extended)
  for (i in which(!is.na(bwaSPACERCount_ggprep_plot_time$localPHAGE))) {
    
    bwaSPACERCount_ggprep_plot_time[i,"explainedBLAST_extended"] <- bwaSPACERCount_ggprep_plot_time[i,"localPHAGE"] %>% as.character()
    
  }
  table(bwaSPACERCount_ggprep_plot_time$explainedBLAST_extended)
  
  bwaSPACERCount_ggprep_plot_time_sub <- bwaSPACERCount_ggprep_plot_time[grep("Strepto",bwaSPACERCount_ggprep_plot_time$explainedBLAST_extended),]
  # bwaSPACERCount_ggprep_plot_time_sub <- bwaSPACERCount_ggprep_plot_time[which(!is.na(bwaSPACERCount_ggprep_plot_time$Type.y)),]

  pspacers <- ggplot(bwaSPACERCount_ggprep_plot_time_sub,aes(x=sample,y=bwaCount,group=spacer,color=explainedBLAST_extended))+ geom_line(size=0.4, alpha=.8) +theme_classic()+theme(axis.text.y = element_text(size=7),axis.text.x = element_text(angle = 45, hjust = 1,size=5.5),legend.position = "right",axis.title.x = element_blank(),strip.background = element_blank(),strip.text.x = element_blank())+facet_wrap(~explainedBLAST_extended,ncol = 6)+coord_trans(y="log10")+scale_y_continuous(breaks =c(100,500,1000,2000,4000,5000),labels=c(100,500,1000,2000,4000,5000))+labs(y="normalized coverage")

 
  pspacers
  
  
# plotMerged
  ###-------------------------
 ## protospacer count
    ###-------------------------
  
  
  
    # dada_ggprep <- dada_spacer_count %>% gather(.,key="sample",value="DadaCount",-spacer) %>% add_column(method="dada")
bwaPROTOSPACERCount_ggprep <- bwaPROTOSPACERCount_wide %>% gather(.,key="sample",value="bwaCount",-spacer) %>% add_column(method="bwa") %>% rename(METAClusterName="spacer")# %>% rename(spacer="METAClusterName")
bwaPROTOSPACERCount_ggprep_plot_time <- merge(bwaPROTOSPACERCount_ggprep,spacers_final,by.x="METAClusterName",by.y="METAClusterName") %>% filter(!sample %in% c("di_K2_6h","th_K2_8h"))

# table(bwaSPACERCount_ggprep_plot_time$sample)

  
  bwaPROTOSPACERCount_ggprep_plot_time$bwaCount <- bwaPROTOSPACERCount_ggprep_plot_time$bwaCount+10

  
       p3 <- ggplot(bwaPROTOSPACERCount_ggprep_plot_time,aes(x=sample,y=bwaCount,group=spacer,color=RefStrain_explained))+ geom_line(size=0.4, alpha=.4) +theme_classic()+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9))+facet_wrap(~RefStrain_explained)+coord_trans(y="log10")
 
  p3
  
  bwaPROTOSPACERCount_ggprep_plot_time$Type.y
       p3 <- ggplot(bwaPROTOSPACERCount_ggprep_plot_time,aes(x=sample,y=bwaCount,group=spacer,color=Type.y))+ geom_line(size=0.4, alpha=.4) +theme_classic()+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9))+facet_wrap(~explainedBLAST)+coord_trans(y="log10")
 
  p3
  
###------------Rename  
  
bwaPROTOSPACERCount_ggprep_plot_time <- bwaPROTOSPACERCount_ggprep_plot_time  %>% mutate(sample=recode(sample, lyo202_96="Lyo 1996",Lyo_202_2012="Lyo 2012", Lyo_202_2014="Lyo 2014", Konserve_202="working stock", RMK202="starter culture\n2012", Versand_202="starter culture\n2018","di_K2_6h"="cheesemaking\nday1","th_K2_8h"="cheesemaking\nday2"))

bwaPROTOSPACERCount_ggprep_plot_time$sample = factor(bwaPROTOSPACERCount_ggprep_plot_time$sample, levels=c("Lyo 1996","Lyo 2012","Lyo 2014","working stock","starter culture\n2012","starter culture\n2018","cheesemaking\nday1","cheesemaking\nday2"))

   ####------------------
  ##add the local phage name 
  
  
  bwaPROTOSPACERCount_ggprep_plot_time$explainedBLAST_extended <- bwaPROTOSPACERCount_ggprep_plot_time$explainedBLAST
  table(bwaPROTOSPACERCount_ggprep_plot_time$explainedBLAST_extended)
  for (i in which(!is.na(bwaPROTOSPACERCount_ggprep_plot_time$localPHAGE))) {
    
    bwaPROTOSPACERCount_ggprep_plot_time[i,"explainedBLAST_extended"] <- bwaPROTOSPACERCount_ggprep_plot_time[i,"localPHAGE"] %>% as.character()
    
  }
  table(bwaPROTOSPACERCount_ggprep_plot_time$explainedBLAST_extended)
  
  bwaPROTOSPACERCount_ggprep_plot_time_sub <- bwaPROTOSPACERCount_ggprep_plot_time[grep("Strepto",bwaPROTOSPACERCount_ggprep_plot_time$explainedBLAST_extended),]
  # bwaSPACERCount_ggprep_plot_time_sub <- bwaSPACERCount_ggprep_plot_time[which(!is.na(bwaSPACERCount_ggprep_plot_time$Type.y)),]

  pProtoSpacers <- ggplot(bwaPROTOSPACERCount_ggprep_plot_time_sub,aes(x=sample,y=bwaCount,group=spacer,color=explainedBLAST_extended))+ geom_line(size=0.4, alpha=.8) +theme_classic()+theme(axis.text.y = element_text(size=7),axis.text.x =element_blank(),legend.position = "right",axis.title.x = element_blank(),strip.background = element_blank(),strip.text.x = element_blank())+facet_wrap(~explainedBLAST_extended,ncol = 6)+coord_trans(y="log10")+labs(y="normalized coverage")#+scale_y_continuous(breaks =c(100,500,1000,2000,4000,5000,10000,20000,50000,100000),labels=c(100,500,1000,2000,4000,5000,10000,20000,50000,100000))

 
  pProtoSpacers
##------------------------------merge two plots
#finally I will make a splitplot of both current streptococcus phages spacers and protospacers  
  
  plot2 <- pProtoSpacers/pspacers 
  
    plot1 <- plines+pboxes+plot_layout(widths =c(6,1))

  
   svg("~/Desktop/Projects/2019_RMK202_analysis/plot/Coverage_plots_all.svg",width=17,height=5.5)

  (plot1+plot2)+plot_layout(widths = c(3,0.5,1.5),ncol = 3)
  
  dev.off()
  
#       png("~/Desktop/Projects/2019_RMK202_analysis/plot/spacer_following_raw.png", width = 2800, height = 3000,res=300)
# 
# p3
# 
# dev.off()
```

#####first analysis
Here we add the blast results


```{r}
library(plyr)
library(tidyverse)

###-------------------How many spacers only map in experiment 

dada_spacer_count <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract/FinalSpacers/ForDADA2_woREverse/dada_spacer_count.txt", "\t", escape_double = FALSE, trim_ws = TRUE)
samplesss <- "lyo202_96"

dada_spacer_count_final <- dada_spacer_count$spacer
# colnames(dada_spacer_count_final) <- "spacer"
for (samplesss in c("lyo202_96","L_202_2012","L_202_2014","RMK202","Konserve_202","Versand_202","di_K2_6h","th_K2_6h","G1_6_18","G2_6_18","G3_6_18","G4_6_18","G5_6_18")) {
  
  tmp1 <- data.frame(rowSums(dada_spacer_count[,grep(samplesss,colnames(dada_spacer_count))]))
  colnames(tmp1) <- samplesss
  dada_spacer_count_final <- cbind(dada_spacer_count_final,tmp1)
  
}



spacers_final_plot_experiment <- spacers_final
spacers_experiment <- merge(spacers_final_plot_experiment,dada_spacer_count_final,by.x="spacer",by.y="dada_spacer_count_final")




ncol(spacers_experiment)

colnames(spacers_experiment)[40]

onlyExperimentSpacers <- spacers_experiment[which(rowSums(spacers_experiment[,-c(1:39,39+grep("_6_18",colnames(spacers_experiment)[40:52]))])==0),] %>% filter(ClusterINFO=="MetaGenome") 

##rowSums experiment that actually have a mapings
onlyExperimentSpacers_cleaned <- onlyExperimentSpacers[which(rowSums(onlyExperimentSpacers[,48:52]>0)!=0),]

#%>% select(c("13491", "13492","13493","13494","13495","13496","13497",13498,13499,13499c1,13499c2,13500,24737,24738,24739,24740,24853,24854,24855,S50,S72,SMAG))

onlyExperimentSpacers_cleaned
nrow(onlyExperimentSpacers_cleaned)

onlyExperimentSpacers_cleaned_final <- onlyExperimentSpacers_cleaned %>% select(c("spacer","G1_6_18","G2_6_18","G3_6_18","G4_6_18","G5_6_18")) %>%  gather(., sample, coverage, c("G1_6_18","G2_6_18","G3_6_18","G4_6_18","G5_6_18"), factor_key=TRUE,na.rm = TRUE) %>% filter(coverage>0)

table(onlyExperimentSpacers_cleaned_final$sample)

1/124
8/124

###-------------------How many SPACERS map to the known viruses

spacers_final_plot <- spacers_final


spacers_final_plot$explainedBLAST = factor(spacers_final_plot$explainedBLAST, levels=c("localPHAGE","localBAC","phageDB","No-match"))
# colorsss <- (c("#0062B4FF","darkcyan","darkturquoise","lightblue","lightgray"))
colorsss <- (c("darkcyan","darkturquoise","lightblue","lightgray"))

MAPPINGTO <- ggplot(spacers_final_plot,aes(x=explainedBLAST,fill=explainedBLAST,color=explainedBLAST))+geom_bar()+theme_classic()+labs(x="")+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),legend.position="none")+
   scale_fill_manual(values=colorsss)+
   scale_color_manual(values=colorsss)
MAPPINGTO
    png("~/Desktop/Projects/2019_RMK202_analysis/plot/MAPPINGTO.PNG", width = 1000, height = 1200,res=300)

MAPPINGTO

dev.off()

  
###-------------------To how many different viruses do they map

CRISPRblast_all_mapping_phages <- spacers_final[which(spacers_final$phageDB!="NA"),]

MAPPINGTO2 <- ggplot(CRISPRblast_all_mapping_phages, aes(x = forcats::fct_infreq(phageDB)))+geom_bar(stat = "count")+theme_classic()+labs(x="",y="Frequency",title = "CRISPR spacers mapping to which phages")+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9))+scale_fill_manual(values=colorsss)

MAPPINGTO2
    png("~/Desktop/Projects/2019_RMK202_analysis/plot/MAPPINGTO2.PNG", width = 2500, height = 1900,res=300)

MAPPINGTO2

dev.off()
  

###-------------------How many spacers are explained with the strains

##---------add how many REFstrains explain spacer


# spacers_final_plot <- spacers_final
# # spacers_final_plot$
# unique(spacers_final_plot$RefStrain_explained)
# 
# spacers_final_plot$RefStrain_explained = factor(spacers_final_plot$RefStrain_explained, levels=rev(c("multiple Strains","Meta_RMK202","mst2","mst 1","OrphanCRISPR","not explained")))
# # colorsss <- rev(c("darkcyan","darkturquoise","lightblue","lightgray"))
# colorsss <- rev(c("#0062B4FF","darkcyan","darkturquoise","lightblue","lightgray"))
# 
# # SPACERDF_sub <- SPACERDF[which(!is.na(SPACERDF$adap.x)),]
# 
# 
# MAPPINGTO_what <- ggplot(spacers_final_plot,aes(x=Array,fill=RefStrain_explained,color=RefStrain_explained))+geom_bar()+theme_classic()+labs(x="",y="spacer count")+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),legend.position="right",legend.title = element_blank() )+
# scale_fill_manual(values=colorsss)+
# scale_color_manual(values=colorsss)
# 
# 
# MAPPINGTO_what
#     png("~/Desktop/Projects/2019_RMK202_analysis/plot/MAPPINGTO_02.PNG", width = 1500, height = 1200,res=300)
# 
# MAPPINGTO_what
# 
# dev.off()

##-----------------------
##assign to types
##-----------------------
# spacers_final$explainedBLAST
SPACERDF <- spacers_final

ggplot(SPACERDF,aes(x=forcats::fct_infreq(VC)))+geom_bar()+theme_classic()
table(SPACERDF$VC) %>% sort() %>% sum()
table(SPACERDF$VC) %>% sort() 
SPACERDF$explainedType <- NA
table(SPACERDF$Type.y) %>% sort() 


for (i in 1:nrow(SPACERDF)) {
  
 SPACERDF[i,"explainedType"] <-  ifelse(!is.na(SPACERDF[i,"Type.y"]),SPACERDF[i,"Type.y"],ifelse(!is.na(SPACERDF[i,"VC"]),SPACERDF[i,"VC"],"no Phage mapping"))
  
}
table(SPACERDF$explainedType) %>% sort()
# 
# clusterCRISPRs_keep_extended_TYPEclustering_forDescription <- SPACERDF

# clusterCRISPRs_keep_extended_TYPE_PHAGEdb <- Streptococcus_phage_Cluster_assignment[which(Streptococcus_phage_Cluster_assignment$explainedType!="no Phage mapping"),]


#SPACERDF$explainedType <- revalue(SPACERDF$explainedType, c("316_1"="unknownCluster_316_1", "150_0"="unknownCluster_150_0", "pac"="Phage Type PAC",   "5093"="Phage Type 5093",  "cos"="Phage Type COS",   "987"="Phage Type 987",   "89_0"="unknownCluster_89_0",  "92_0"="unknownCluster_92_0",  "88_0"="unknownCluster_88_0",  "104_3"="unknownCluster_104_3", "327_0"="unknownCluster_327_0", "384_0"="unknownCluster_384_0", "nan"="unknownCluster",   "373_0"="unknownCluster_373_0"))

SPACERDF$explainedType <- revalue(SPACERDF$explainedType, c("134_0"="unknownCluster_134_0", "151_0"="unknownCluster_151_0", "pac"="Phage Type PAC",   "5093"="Phage Type 5093",  "cos"="Phage Type COS",   "987"="Phage Type 987",   "251_1"="unknownCluster_251_1",  "316_1"="unknownCluster_316_1",  "92_0"="unknownCluster_92_0",  "385_0"="unknownCluster_385_0"))
table(SPACERDF$explainedType) %>% sort()

clusterCRISPRs_keep_extended_TYPE_PHAGEdb <- SPACERDF[which(SPACERDF$explainedType!="no Phage mapping"),]

table(SPACERDF$explainedType)
clusterCRISPRs_keep_extended_TYPE_PHAGEdb$explainedTypeSimple <- clusterCRISPRs_keep_extended_TYPE_PHAGEdb$explainedType

# clusterCRISPRs_keep_extended_TYPE_PHAGEdb[which(grepl("unknownCluster_150_0",clusterCRISPRs_keep_extended_TYPE_PHAGEdb$explainedTypeSimple)),"explainedTypeSimple"] <-  "Cluster 150"
clusterCRISPRs_keep_extended_TYPE_PHAGEdb[which(grepl("unknownClu",clusterCRISPRs_keep_extended_TYPE_PHAGEdb$explainedTypeSimple)),"explainedTypeSimple"] <-  "Unknown"

table(clusterCRISPRs_keep_extended_TYPE_PHAGEdb$explainedTypeSimple)


clusterCRISPRs_keep_extended_TYPE_PHAGEdb$explainedTypeSimple %>% as.factor() %>% levels()
clusterCRISPRs_keep_extended_TYPE_PHAGEdb$explainedType %>% as.factor() %>% levels()

clusterCRISPRs_keep_extended_TYPE_PHAGEdb$explainedTypeSimple = factor(clusterCRISPRs_keep_extended_TYPE_PHAGEdb$explainedTypeSimple, levels=c("Phage Type COS" ,"Phage Type PAC" , "Phage Type 987" , "Phage Type 5093","Unknown"))
clusterCRISPRs_keep_extended_TYPE_PHAGEdb$explainedType = factor(clusterCRISPRs_keep_extended_TYPE_PHAGEdb$explainedType, levels=c("Phage Type COS" ,"Phage Type PAC" , "Phage Type 987" , "Phage Type 5093","unknownCluster_150_0", "unknownCluster_250_1", "unknownCluster_315_1", "unknownCluster_383_0","unknownCluster_92_0"))

# colorsss <- (c("lightblue","#0062B4FF","darkcyan","darkturquoise","lightgray"))
# colorsss <- (c("chocolate1","#0062B4FF","darkcyan","darkturquoise","lightblue","lightgray"))
colorsss <- (c("#0062B4FF","darkcyan","darkturquoise","lightblue","lightgray"))


PLOT_CLUSTERS_counts <- ggplot(clusterCRISPRs_keep_extended_TYPE_PHAGEdb,aes(x=forcats::fct_infreq(explainedTypeSimple),fill=explainedTypeSimple))+geom_bar()+theme_classic()+theme(axis.text.x = element_text(angle = 45, hjust = 1,size=9),legend.position = "right")+labs(x="",y="Number of spacers mapping to VC")+
   scale_fill_manual(values=colorsss)+
   scale_color_manual(values=colorsss)
PLOT_CLUSTERS_counts
    png("~/Desktop/Projects/2019_RMK202_analysis/plot/phagetypemapping_new.PNG", width = 2500, height = 1900,res=300)

PLOT_CLUSTERS_counts

dev.off()
clusterCRISPRs_keep_extended_TYPE_PHAGEdb$ARRAYINFO <- revalue(clusterCRISPRs_keep_extended_TYPE_PHAGEdb$ARRAYINFO, c("a1"="CRISPR Array 1", "a2"="CRISPR Array 2",   "a3"="CRISPR Array 3")) %>% as.factor
clusterCRISPRs_keep_extended_TYPE_PHAGEdb$explainedTypeSimple
PLOT_CLUSTERS_counts_short <- ggplot(clusterCRISPRs_keep_extended_TYPE_PHAGEdb,aes(x=forcats::fct_infreq(explainedTypeSimple),fill=explainedTypeSimple,color=explainedTypeSimple,group=ARRAYINFO))+geom_bar( position = "dodge")+theme_classic()+theme(axis.text.x = element_text(angle = 45, hjust = 1,size=7),legend.position = "right")+labs(x="",y="number of spacers mapping to cluster")+
   scale_fill_manual(values=colorsss)+
   scale_color_manual(values=colorsss)+ scale_x_discrete(drop=FALSE)
PLOT_CLUSTERS_counts_short


clusterCRISPRs_keep_extended_TYPE_PHAGEdb$explainedTypeSimple
PLOT_CLUSTERS_counts_short <- ggplot(clusterCRISPRs_keep_extended_TYPE_PHAGEdb,aes(x=forcats::fct_infreq(explainedTypeSimple),fill=explainedTypeSimple,color=explainedTypeSimple))+geom_bar(width=0.75)+scale_y_continuous(breaks = c(13,88,156),labels = c(13,88,156))+theme_classic()+theme(axis.text.x = element_text(angle = 45, hjust = 1,size=7),legend.position = "right")+labs(x="",y="Number of spacers mapping to VR")+
   scale_fill_manual(values=colorsss)+
   # scale_color_manual(values=c("#fd8d3c","#807dba","#f03c2c"))+ scale_x_discrete(drop=FALSE)
   scale_color_manual(values=colorsss)+ scale_x_discrete(drop=FALSE)

PLOT_CLUSTERS_counts_short


 svg("~/Desktop/Projects/2019_RMK202_analysis/plot/mergedPlots_ready_new02-_simple.svg",width=3,height=4)
 
# plotMerged
PLOT_CLUSTERS_counts_short
dev.off()
##---------------------------
##plot species phage mapping again with type assigment
##---------------------------

colorsss <- (c("#0062B4FF","darkcyan","darkturquoise","lightblue",rep("lightgray",5)))

# colorsss <- (c("#0062B4FF","darkcyan","darkturquoise","lightblue",rep("lightgray",2),"chocolate1",rep("lightgray",8)))

MAPPINGTO3 <- ggplot(clusterCRISPRs_keep_extended_TYPE_PHAGEdb, aes(x = forcats::fct_infreq(phageDB),fill=explainedType,color=explainedType))+geom_bar(stat = "count")+theme_classic()+labs(x="",y="number of spacer mapping to the phage",title = "")+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=4))+scale_fill_manual(values=colorsss)+scale_color_manual(values=colorsss)

MAPPINGTO3
    png("~/Desktop/Projects/2019_RMK202_analysis/plot/MAPPINGTO3_new.PNG", width = 2500, height = 1900,res=300)

MAPPINGTO3

dev.off()

##--------------------------------------------------------
##number of spacers per strain against the phages
is.na(spacers_final$localPHAGE)
library(tidyverse)
spaceragain <- spacers_final %>% filter(!is.na(localPHAGE)) %>% gather(., strain, count, "13491":"SMAG", factor_key=TRUE,na.rm = TRUE)  %>% filter(count!="0") 

table(table(spaceragain$strain)) 

spaceragain_extended <- merge(spaceragain,ReferenceGenomeGrouping,by="strain",all.x=TRUE)

table(spaceragain_extended$strain)
spaceragain_extended$strain <- as.character(droplevels(spaceragain_extended$strain))

unique(spaceragain_extended$strain) %in% ReferenceGenomeGrouping$strain
length(ReferenceGenomeGrouping$strain)
length(unique(spaceragain_extended$strain))

spaceragain_extended$strain = factor(spaceragain_extended$strain, levels=c(ReferenceGenomeGrouping$strain))

table(spaceragain_extended$strain)



mapping_to_rmk_phages <- ggplot(spaceragain_extended,aes(x=strain,fill=lineage))+geom_histogram(stat="count")+theme_classic()+facet_wrap(~localPHAGE)+scale_fill_manual(values =unique(ReferenceGenomeGrouping$colors) )+theme(axis.title.x = element_blank(),axis.text.x = element_text(angle = 75, hjust = 1,size=6))+labs(y="number of spacer")

mapping_to_rmk_phages


 # png("~/Desktop/Projects/2019_RMK202_analysis/plot/mapping_to_rmk_phagest.png", width = 1600, height = 100,res=300)
 png("~/Desktop/mid_thesis/report/figures/20200430/chapter1/20201006/supplement/mapping_to_rmk_phagest.png", width = 1800, height = 1000,res=300)

 
mapping_to_rmk_phages

 
 dev.off()

```

####phage Clustermapping

first run the Vcontact cluster chunk
```{r}


#---------------
##creat graph
#---------------
# write_csv(Sterm_cluster_sub,path="~/Desktop/Projects/2019_PhageDB/Vcontact//allPhages_vContact/Streptococcus_phage_betwork_connections.txt",col_names = TRUE)

# write_csv(Sterm_cluSterm_cluster_subster_sub,path="~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03///allPhages_vContact/Streptococcus_phage_betwork_connections.txt",col_names = TRUE)
Sterm_cluster_sub <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03/allPhages_vContact/Streptococcus_phage_betwork_connections.txt")
# unique(Sterm_cluster_sub$from)
# nrow(Sterm_cluster_sub)
 Sterm_cluster_sub <- Sterm_cluster_sub[!duplicated(Sterm_cluster_sub[,-3]), ]
  # Sterm_cluster_sub %>% filter(from=="Bacillus~phage~BCJA1c"&to=="Streptococcus~phage~315.4")
    # Sterm_cluster_sub 
    
  ##deduplicate
  Sterm_cluster_sub <-  Sterm_cluster_sub %>% 
      rowwise() %>%                             # Perform next step by row
  mutate(dup=paste0(sort(c(from,to)),collapse="")) %>%   # Sort and combine feedID and feedID2
  ungroup() %>%                           # Remove rowwise grouping
  mutate(dup=duplicated(dup)) %>%           # Find duplicated feedID:feedID2 pairs
  filter(dup==F) %>%                        # Remove duplicated pairs
  filter(!(from==to)) %>%            # Remove where feedID == feedID2
  select(-dup)   

   # n_subset <- nExtended %>%  filter(nExtended$VC %in% unique(spacers_final$VC))
# unique(n_subset$vertex.names)
nrow(Sterm_cluster_sub)
table(n_subset$VC)

nrow(Phages_names)


subsetGenomes <- Phages_names %>%  filter(VC %in% names(table(spacers_final$VC)))

# table(subsetGenomes$VC)

# names  <- n_subset %>% filter(!is.na(VC))
# table(names$VC)
Sterm_cluster_sub <- Sterm_cluster_sub[which(Sterm_cluster_sub$from %in% unique(subsetGenomes$Genome) & Sterm_cluster_sub$to %in% unique(subsetGenomes$Genome)),]
 

# Sterm_cluster_sub$from
nrow(Sterm_cluster_sub)

Sterm_net <- graph_from_data_frame(Sterm_cluster_sub,direct=F)
# Ldel_net <- graph_from_data_frame(Ldel_cluster_sub,direct=F)
# Llac_net <- graph_from_data_frame(Llac_cluster_sub,direct=F)

# dim(Sterm_net)

##==================================
 ##new analysis
  ##==================================
 library(ggnetwork)
 library(ITNr)
 library(intergraph)
 
 net<-asNetwork(Sterm_net) #ggnetwork requires a network object
 

 n<-ggnetwork(net)


 
###--------
##grep fo Mahony phages
#---------------
Mahony_genomes$namez <- paste(Mahony_genomes$Genus,Mahony_genomes$phage,Mahony_genomes$strain,sep="~")
Mahony_genomes_cos <- Mahony_genomes[which(Mahony_genomes$Type=="cos"),]
Mahony_genomes_pac <- Mahony_genomes[which(Mahony_genomes$Type=="pac"),]
Mahony_genomes_987 <- Mahony_genomes[which(Mahony_genomes$Type=="987"),]
Mahony_genomes_5093 <- Mahony_genomes[which(Mahony_genomes$Type=="5093"),]

###--------
##coloring
#---------------
# n[grep("RMK202_",n$vertex.names),"vertex.names"] %>% table()
 n$color <- as.character("Viral DB")
 n$color[grep("RMK202_",n$vertex.names)] <- "RMK202 phages"
 n$color[n$vertex.names %in% Mahony_genomes_cos$namez] <- "cos"
 n$color[n$vertex.names %in% Mahony_genomes_pac$namez] <- "pac"
 n$color[n$vertex.names %in% Mahony_genomes_987$namez] <- "987"
 n$color[n$vertex.names %in% Mahony_genomes_5093$namez] <- "5093"
 
 
 n$color = factor(n$color, levels=c("RMK202 phages","cos","pac","987","5093","Viral DB"))
 colorzzz <- c("red","blue","green","yellow","purple","grey50")
 size <- c("4","2","2","2","2","1.5")
 ##------------------
 ##extend
 ##------------------
 unique(n$vertex.names)
 nExtended <-  merge(n, Phages_names, by.y="Genome",by.x="vertex.names",all.x = TRUE)
  grep("Streptococcus", unique(nExtended$vertex.names))
table(nExtended$VC) 
 # nrow(n)
 # nExtended <-  merge(n, Phages_names, by.y="Genome",by.x="vertex.names",all.x = TRUE) 
 # nrow(nExtended)
 # nExtended <- n
 # ccolnames(nExtended)
colnames(nExtended)
colnames(nExtended) <- revalue(colnames(nExtended), c("Adj P-value"="Adj_P_value","Genera in VC"="Genera_in_VC","Topology Confidence Score"="Topology_Confidence_Score","Genus Confidence Score"="Genus_Confidence_Score"))

# nExtended$ad
table(nExtended$color)
table(nExtended$color)

 ##------------------
 ##plot
 ##------------------
 networkplot <- ggplot(nExtended, aes(x = x, y = y, xend = xend, yend = yend,text =paste("genome:", vertex.names,"\npower: ",power,"\nOrder: ",Order,"\nFamily: ",Family,"\nGenus: ",Genus,"\nVC: ",VC,"\nSize: ",Size,"\nAdj P-value: ",Adj_P_value,"\nGenera in VC: ",Genera_in_VC,"\nGenus_Confidence_ScoreC: ",Genus_Confidence_Score,"\nTopology_Confidence_Score: ",Topology_Confidence_Score))) + geom_edges(color = "lightgray") +
   geom_nodes(aes(color = as.factor(color),size=as.factor(color))) +
   # scale_fill_manual(values="gray41")+
    scale_color_manual(values=colorzzz)+
    scale_size_manual(values=as.numeric(size))+
   # guides(color=FALSE,size=FALSE)+ #Remove legends
 theme_blank()+theme(legend.title = element_blank())
 networkplot
 # 
 # library(plotly)
 # ggp <- ggplotly(networkplot)
 # 
 library(plotly)
 ggp <- ggplotly(networkplot)
 htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new_03//allPhages_vContact//networkPlot_onlyClusterGenomes.html")
 # View(nExtended)
 
 
 ##-------------------------------------------------
 ##subsetting
 ##color extensive
 ##-------------------------------------------------
 
 table(nExtended$color)
 
 Cos_VC <- nExtended %>% filter(color=="cos") %>% select(VC) %>% unique()
 Pac_VC <- nExtended %>% filter(color=="pac") %>% select(VC) %>% unique()
 VC_987 <- nExtended %>% filter(color=="987") %>% select(VC) %>% unique()
 VC_5093 <- nExtended %>% filter(color=="5093") %>% select(VC) %>% unique()

 
 nExtended$color <- as.character("Viral DB")
 nExtended$color[nExtended$VC %in% Cos_VC$VC] <- "cos"
 nExtended$color[nExtended$VC %in% Pac_VC$VC] <- "pac"
 nExtended$color[nExtended$VC %in% VC_987$VC] <- "987"
 nExtended$color[nExtended$VC %in% VC_5093$VC] <- "5093"
 nExtended$color[grep("RMK202_",nExtended$vertex.names)] <- "RMK202 phages"

 # nExtended$color
 table(nExtended$color)
 # nExtended$color = factor(nExtended$color, levels=c("RMK202 phages","cos","pac","987","5093","Viral DB"))
  nExtended$color = factor(nExtended$color, levels=c("Viral DB","cos","pac","987","5093","RMK202 phages"))

 # n$hit = 
spacers_final$phageDB
unique(spacers_hit$phageDB)
spacers_hit <- spacers_final %>% filter(!is.na(phageDB)) %>% select(phageDB) %>% add_column(hit="hit")
spacers_hit$phageDB <- gsub(" ","~",spacers_hit$phageDB)
unique(nExtended$vertex.names)

nExtended_02 <- merge(nExtended,spacers_hit,by.x="vertex.names",by.y="phageDB",all.x=TRUE) %>% dplyr::mutate(hit = replace_na(hit, "no-hit"))
table(nExtended_02$hit)

 nExtended_02$hit = factor(nExtended_02$hit, levels=c("hit","no-hit"))
  nExtended_02$hit = factor(nExtended_02$hit, levels=c("hit","no-hit"))
 nExtended_02$transparancy  <- as.numeric(as.character((revalue(nExtended_02$hit, c("hit"="1", "no-hit"="0.3")))))

  
table(nExtended_02$hit)



 colorzzz <- rev(c("red","blue","green","yellow","purple","grey50","pink"))
 size <- c("4","2")
  # dim(nExtended)

 # n_subset <- nExtended %>%  filter(nExtended$VC %in% unique(spacers_final$VC))
 # nExtended %>%  filter(!VC %in% unique(spacers_final$VC)) %>% select(vertex.names,VC)

 # dim(n_subset)
    nExtended_02$color <- as.character(nExtended_02$color)

 nExtended_02 %>% filter(color=="RMK202 phages")
 unique(nExtended_02$color)
   table(nExtended_02$color)
    ###------------change color of type genomes that were hit

 for (i in 1:nrow(nExtended_02)) { nExtended_02[i,"color"] <- ifelse(nExtended_02[i,"hit"]=="no-hit","Viral DB",as.character(nExtended_02[i,"color"]))}
  unique(nExtended_02$color)
  table(nExtended_02$color)
  
  
  ###------------change RMKstrains color
  for (i in 1:nrow(nExtended_02)) { nExtended_02[i,"color"] <- ifelse(as.character(nExtended_02[i,"vertex.names"])=="RMK202_Streptococcus_phage_1"|as.character(nExtended_02[i,"vertex.names"])=="RMK202_Streptococcus_phage_2","RMK202 phages",as.character(nExtended_02[i,"color"]))}
  ###------------change RMKstrains size
  for (i in 1:nrow(nExtended_02)) { nExtended_02[i,"hit"] <- ifelse(as.character(nExtended_02[i,"vertex.names"])=="RMK202_Streptococcus_phage_1"|as.character(nExtended_02[i,"vertex.names"])=="RMK202_Streptococcus_phage_2","hit",as.character(nExtended_02[i,"hit"]))}
    ###------------change name of hitted viral DB genomes

  table(nExtended_02$color)
 for (i in 1:nrow(nExtended_02)) { nExtended_02[i,"color"] <- ifelse(nExtended_02[i,"hit"]=="hit"& as.character(nExtended_02[i,"color"])=="Viral DB",as.character("unknown Cluster hit"),as.character(nExtended_02[i,"color"]))}

 unique(nExtended_02$color)
 
 
   nExtended_02$color = factor(nExtended_02$color, levels=c("Viral DB","unknown Cluster hit","cos","pac","987","5093","RMK202 phages"))
 # colorzzz <- rev(c("red","blue","green","yellow","purple","grey50","grey88"))
 colorzzz <- rev(c("red","#0062B4FF","darkcyan","darkturquoise","lightblue","grey50","grey70"))
    nExtended_02$color = factor(nExtended_02$color, levels=rev(c("Viral DB","unknown Cluster hit","cos","pac","987","5093","RMK202 phages")))

 colorzzz <- c("red","#0062B4FF","darkcyan","darkturquoise","lightblue","grey50","grey70")
# colorsss <- (c("#0062B4FF","darkcyan","darkturquoise","lightblue","lightgray"))

 networkplot_subset <- ggplot(nExtended_02, aes(x = x, y = y, xend = xend, yend = yend,text =paste("genome:", vertex.names,"\npower: ",power,"\nOrder: ",Order,"\nFamily: ",Family,"\nGenus: ",Genus,"\nVC: ",VC,"\nSize: ",Size,"\nAdj P-value: ",Adj_P_value,"\nGenera in VC: ",Genera_in_VC,"\nGenus_Confidence_ScoreC: ",Genus_Confidence_Score,"\nTopology_Confidence_Score: ",Topology_Confidence_Score))) + geom_edges(size=0.1,color = "grey60",alpha=0.4) +
   geom_nodes(aes(color = as.factor(color),fill = as.factor(color),size=as.factor(hit),alpha=transparancy)) +
   # scale_fill_manual(values="gray41")+
# scale_alpha_discrete(range = c(0.35, 0.9))+
   scale_color_manual(values=colorzzz)+
     scale_fill_manual(values=colorzzz)+
    scale_size_manual(values=as.numeric(size))+
      # scale_alpha_manual(values=as.numeric(c(1,0.3)))+
   # guides(color=FALSE,size=FALSE)+ #Remove legends
 theme_blank()+theme(legend.title = element_blank())
 networkplot_subset
 svg("~/Desktop/Projects/2019_RMK202_analysis/plot/network_subsetted.svg",width=16,height=13.5)
 
# plotMerged
 networkplot_subset
dev.off()
 

 
 library(plotly)
 ggp <- ggplotly(networkplot_subset)
  htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_RMK202_analysis/plot/networkPlot_onlyClusterGenomes_only.html")

 ##------------==========================
 ##old
 ##------------==========================

 
##------------------------------------------
## put clusters into network if they were mapped to
##------------------------------------------
library(igraph)
library(readr)
##first do phageDB.Rmd script

# Streptococcus_phage_betwork_connections <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact_with_meta_new//allPhages_vContact/Streptococcus_phage_betwork_connections.txt")

##-------make graph
Sterm_net <- graph_from_data_frame(Streptococcus_phage_betwork_connections,direct=F)

###-------color edges
vcolour_Sterm <- rep("lightgray",length(V(Sterm_net)$name))

colorsss <- (c("#0062B4FF","darkcyan","darkturquoise","lightblue",rep("lightgray",5)))

tmp_names_type <- as.factor(clusterCRISPRs_keep_extended_TYPE_PHAGEdb$explainedType) %>% levels()
i <- 1
for (i in 1:4) {
  tmp <- clusterCRISPRs_keep_extended_TYPE_PHAGEdb[which(clusterCRISPRs_keep_extended_TYPE_PHAGEdb$explainedType==tmp_names_type[i]),"phageDB"] %>% gsub(" ","~",.) %>% as.data.frame()
  vcolour_Sterm[which(V(Sterm_net)$name %in% tmp$.)] <- colorsss[i]

}


  # tmp <- clusterCRISPRs_keep_extended_TYPE_PHAGEdb[which(clusterCRISPRs_keep_extended_TYPE_PHAGEdb$explainedType=="unknownCluster_150_0"),"phageDB"] %>% gsub(" ","~",.) %>% as.data.frame()
  # vcolour_Sterm[which(V(Sterm_net)$name %in% tmp$.)] <- "chocolate1"


table(vcolour_Sterm)

V(Sterm_net)$color <- vcolour_Sterm
table(V(Sterm_net)$color)


###-------size edges

vsize_Sterm <- rep(as.numeric("1.5"),length(V(Sterm_net)$name))

vsize_Sterm[which(V(Sterm_net)$name %in% gsub(" ", "~",SPACERDF$phageDB))] <- "8"

table(vsize_Sterm)

vsize_Sterm <- as.numeric(vsize_Sterm)

###-------plot network

set.seed(8051)

fg_lay_Sterm <- layout_with_fr(Sterm_net,dim=2,niter=99)

     # png("~/Desktop/Projects/2019_RMK202_analysis/plot/vContact_sterm_withAssinment.png", width = 1800, height = 1200,res=300)

 plot(Sterm_net,vertex.label=NA,layout=fg_lay_Sterm,vertex.size=vsize_Sterm
     )
 legend("topright",  cex=0.5,
       legend=c(tmp_names_type[1:5],"cluster 150"), pch=19,
      col=c(colorsss[1:5],"chocolate1"), bty="n")

 
 # dev.off()
 
 ##--------cluster size
 
 
##--------------merge plots together
library(patchwork)
# plotMerged <-  MAPPINGTO_what / (MAPPINGTO3+theme(legend.position = "none") +PLOT_CLUSTERS_counts) 

 PLOT_CLUSTERS_counts
 # svg("~/Desktop/Projects/2019_RMK202_analysis/plot/mergedPlots_ready.svg", width = 1800, height = 1800)
 svg("~/Desktop/Projects/2019_RMK202_analysis/plot/mergedPlots_ready_new.svg",width=9,height=9)
 
# plotMerged
PLOT_CLUSTERS_counts
dev.off()

 svg("~/Desktop/Projects/2019_RMK202_analysis/plot/mergedPlots_ready_new02.svg",width=4,height=4)
 
# plotMerged
PLOT_CLUSTERS_counts
dev.off()


 # svg("~/Desktop/Projects/2019_RMK202_analysis/plot/mergedPlots_ready.svg", width = 1800, height = 1800)
 svg("~/Desktop/Projects/2019_RMK202_analysis/plot/mergedPlots_ready_network_new.svg",width=9,height=9)
 
 plot(Sterm_net,vertex.label=NA,layout=fg_lay_Sterm,vertex.size=vsize_Sterm
     )
 legend("topright",  cex=1,
       legend=c(tmp_names_type[1:5],"cluster 150"), pch=19,
      col=c(colorsss[1:5],"chocolate1"), bty="n")

dev.off()

##----------------------------
##put infomration of mapping into the abudance plot
##----------------------------
# tmp_clusterCRISPRs_keep_extended_TYPEclustering <- clusterCRISPRs_keep_extended_TYPEclustering %>% select(Cluster,explainedType)
# 
# CRISPR_spacer_coverage_extended_02 <- merge(CRISPR_spacer_coverage_extended,tmp_clusterCRISPRs_keep_extended_TYPEclustering,by="Cluster",all.x = TRUE)
# 
# # table(CRISPR_spacer_coverage_extended_02$explainedType)
# 
# # range(CRISPR_spacer_coverage$Cpm)
# # CRISPR_spacer_coverage_extended$explained
#   # p2 <- ggplot(CRISPR_spacer_coverage_02,aes(x=sample,y=CPM,group=ClusterName,color=assignedArray))+ geom_line(size=0.5, alpha=.5)+ 
#   p2 <- ggplot(CRISPR_spacer_coverage_extended_02,aes(x=sample,y=CPM,group=Cluster,color=RefStrain))+ geom_line(size=0.4, alpha=.4)+ 
#     # facet_grid(day~kessel~species)+
#      # facet_grid(DB~.)+
#     facet_wrap(adap~., scales="free")+
#     # facet_wrap(explainedType~., scales="free")+
#     # facet_wrap(explained~., scales="free")+
#     # coord_trans(y="log2")+
#     # facet_grid(treatment~species)+
#     # facet_grid(treatment~species)+
#     labs("",
#          x=" ",
#          y="counts per million")+
#     #scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
#     # scale_y_continuous(limits = c(0,10))+
#     theme_classic()+
#    # scale_fill_manual(values=all_colours)+
#    # scale_color_manual(values=all_colours)+
#     scale_x_discrete( expand = c(0, 0)) +
#     theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
#           rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
#       #axis.text.x = element_blank(),
#           legend.position="right"
#           #legend.justification=c(1,1), legend.position=c(1,1),
#           #legend.title = element_blank()
#           )
#   
#   p2
#     # png("~/Desktop/Projects/2019_RMK202_analysis/plot/spacer_following.png", width = 2800, height = 1400,res=300)
#  svg("~/Desktop/Projects/2019_RMK202_analysis/plot/mergedPlots_ready_network_distribution.svg",width=11,height=6)
# 
# p2
# 
# dev.off()

##----------------------------
##output genomes within the cluster names
##----------------------------
library(readr)
Sterm_namesAssigment <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/Sterm_namesAssigment", "\t", escape_double = FALSE, col_names = c("genomeName","contigname"), trim_ws = TRUE)

tmp_prepOutput <- clusterCRISPRs_keep_extended_TYPE_PHAGEdb[which(clusterCRISPRs_keep_extended_TYPE_PHAGEdb$explainedTypeSimple!="Unknown"),]

names(table(tmp_prepOutput$VC))
i="307_0"
for (i in names(table(tmp_prepOutput$VC))) {
 print(i)
  
  tmp_02 <- tmp_prepOutput %>% filter(VC==i) %>% select(phageDB) %>% unique()
  tmpfinal <- NA ##prim the tmp output file

  for (phagezz in tmp_02$phageDB) {
    print(phagezz)
    tmp_03 <- Sterm_namesAssigment[grep(paste0(phagezz,","),Sterm_namesAssigment$contigname),"genomeName"] %>% as.character
    tmpfinal <- c(tmpfinal,tmp_03)
    tmp_03
    
  }
  tmpfinal <- tmpfinal[-1]
  write.table(tmpfinal, paste0("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/CLUSTERS/phagesVcontact_",i,"_cluster.txt"),na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names =FALSE)

  
}

##------------lactococcus phages of VC=150_0

lacto_namesAssigment <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/lacto_namesAssigment", "\t", escape_double = FALSE, col_names = c("genomeName","contigname"), trim_ws = TRUE)


names(table(tmp_prepOutput$VC))
i="150_0"

  
  tmp_02 <- tmp_prepOutput %>% filter(VC==i) %>% select(phageDB) %>% unique()
  tmpfinal <- NA ##prim the tmp output file

  for (phagezz in tmp_02$phageDB) {
    print(phagezz)
    tmp_03 <- lacto_namesAssigment[grep(paste0(phagezz,","),lacto_namesAssigment$contigname),"genomeName"] %>% as.character
   lacto_namesAssigment[grep(phagezz,lacto_namesAssigment$contigname),] 

    tmpfinal <- c(tmpfinal,tmp_03)
    
  }
  tmpfinal <- tmpfinal[-1]
  write.table(tmpfinal, paste0("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/CLUSTERS/phagesVcontact_",i,"_cluster.txt"),na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names =FALSE)

  
##---------------------------
##amount of maps to explained by database to every array
##---------------------------
  clusterCRISPRs_keep_extended_TYPEclustering_forDescription <- SPACERDF

  table(clusterCRISPRs_keep_extended_TYPEclustering_forDescription$explainedType)
  clusterCRISPRs_keep_extended_TYPEclustering_forDescription$explainedTypeSimple <- clusterCRISPRs_keep_extended_TYPEclustering_forDescription$explainedType
  
    clusterCRISPRs_keep_extended_TYPEclustering_forDescription[grep("150_",clusterCRISPRs_keep_extended_TYPEclustering_forDescription$explainedType),"explainedTypeSimple"] <- "Cluster 150"

  table(clusterCRISPRs_keep_extended_TYPEclustering_forDescription$explainedTypeSimple)
  
  clusterCRISPRs_keep_extended_TYPEclustering_forDescription[grep("_",clusterCRISPRs_keep_extended_TYPEclustering_forDescription$explainedTypeSimple),"explainedTypeSimple"] <- "unknown Cluster"
    clusterCRISPRs_keep_extended_TYPEclustering_forDescription[grep("nan",clusterCRISPRs_keep_extended_TYPEclustering_forDescription$explainedType),"explainedTypeSimple"] <- "unknown Cluster"

  table(clusterCRISPRs_keep_extended_TYPEclustering_forDescription$explainedTypeSimple)
  colnames(clusterCRISPRs_keep_extended_TYPEclustering_forDescription)
  # library(dplyr)
  # clusterCRISPRs_keep_extended_TYPEclustering_forDescription$ARRAYINFO
  d2 <- clusterCRISPRs_keep_extended_TYPEclustering_forDescription %>%
  group_by(ARRAYINFO,is.na(BactDB)) %>% 
  dplyr::summarise(count=n()) %>% 
  mutate(perc=count/sum(count))
  
  
##BactDB
BactDB <-   clusterCRISPRs_keep_extended_TYPEclustering_forDescription %>% 
  group_by(ARRAYINFO,is.na(BactDB)) %>% 
  dplyr::summarise(count=n()) %>% 
  mutate(BactDB_percent=100*(count/sum(count))) %>% filter(`is.na(BactDB)`==FALSE)  %>% select(ARRAYINFO,BactDB_percent)
BactDB
  
  
##localBAC
localBAC <-   clusterCRISPRs_keep_extended_TYPEclustering_forDescription %>% 
  group_by(ARRAYINFO,is.na(localBAC)) %>% 
  dplyr::summarise(count=n()) %>% 
  mutate(localBAC_percent=100*(count/sum(count))) %>% filter(`is.na(localBAC)`==FALSE)  %>% select(ARRAYINFO,localBAC_percent)
localBAC
  
##localPHAGE
localPHAGE <-   clusterCRISPRs_keep_extended_TYPEclustering_forDescription %>% 
  group_by(ARRAYINFO,is.na(localPHAGE)) %>% 
  dplyr::summarise(count=n()) %>% 
  mutate(localPHAGE_percent=100*(count/sum(count))) %>% filter(`is.na(localPHAGE)`==FALSE)  %>% select(ARRAYINFO,localPHAGE_percent)
localPHAGE

##phageDB
phageDB <-   clusterCRISPRs_keep_extended_TYPEclustering_forDescription %>% 
  group_by(ARRAYINFO,is.na(phageDB)) %>% 
  dplyr::summarise(count=n()) %>% 
  mutate(phageDB_percent=100*(count/sum(count))) %>% filter(`is.na(phageDB)`==FALSE)  %>% select(ARRAYINFO,phageDB_percent)
phageDB

tmp1 <- merge(BactDB,localBAC,by="ARRAYINFO")  
tmp2 <- merge(tmp1,localPHAGE,by="ARRAYINFO")  
final_percentmapped <- merge(tmp2,phageDB,by="ARRAYINFO")  

final_percentmapped_long <- gather(final_percentmapped, DB, percent, "BactDB_percent", "localBAC_percent", "localPHAGE_percent", "phageDB_percent", factor_key=TRUE,na.rm = TRUE) 

final_percentmapped_long$DB <- gsub("_percent","",final_percentmapped_long$DB)
##plot

final_percentmapped_long$DB = factor(final_percentmapped_long$DB, levels=c("localPHAGE" ,"phageDB" , "localBAC" , "BactDB"))

colorsssss <- c("darkcyan","darkturquoise","goldenrod3","goldenrod1")

# colorsss <- (c("#0062B4FF","darkcyan","darkturquoise","lightblue",rep("lightgray",11)))


mappingWhichDBPlot <- ggplot(final_percentmapped_long,aes(x=ARRAYINFO,group=DB,color=DB,fill=DB,y=percent))+labs(x="",y="percent of spacers mapping")+geom_bar(stat="identity", position = "dodge")+theme_classic()+facet_wrap(~DB,ncol = 4)+theme(legend.position="none",axis.text.x = element_text(angle = 75, hjust = 1,size=9))+
    scale_fill_manual(values=colorsssss)+
   scale_color_manual(values=colorsssss)
mappingWhichDBPlot


phagemapping <-   clusterCRISPRs_keep_extended_TYPEclustering_forDescription %>% 
  group_by(ARRAYINFO,explainedTypeSimple) %>% 
  dplyr::summarise(count=n()) %>% 
  mutate(PhageMapping=100*(count/sum(count)))   %>% select(ARRAYINFO,explainedTypeSimple,PhageMapping)
phagemapping


phagemapping$explainedTypeSimple = factor(phagemapping$explainedTypeSimple, levels=c("Phage Type COS" ,"Phage Type PAC" , "Phage Type 987" , "Phage Type 5093","Unknown"))
# phagemapping$explainedTypeSimple = factor(phagemapping$explainedTypeSimple, levels=c("cos" ,"pac" , "987" , "5093","Cluster 150","unknown Cluster","no Phage mapping"))

# colorsssss <- c("darkcyan","darkturquoise","goldenrod3","goldenrod1")

colorsss <- (c("darkturquoise","lightblue","darkcyan","#0062B4FF","chocolate1","lightgray","grey"))

 phagemapping$ARRAYINFO  <- revalue( phagemapping$ARRAYINFO, c("a1"="CRISPR Array 1", "a2"="CRISPR Array 2",   "a3"="CRISPR Array 3"))


mappingWhichDBPlot_phagType <- ggplot(phagemapping,aes(x=ARRAYINFO,group=explainedTypeSimple,color=explainedTypeSimple,fill=explainedTypeSimple,y=PhageMapping))+labs(x="",y="percent of spacers mapping")+geom_bar(stat="identity", position = "dodge")+theme_classic()+facet_wrap(~ explainedTypeSimple)+theme(legend.position="none",axis.text.x = element_text(angle = 75, hjust = 1,size=9))+
    scale_fill_manual(values=colorsss)+
   scale_color_manual(values=colorsss)
mappingWhichDBPlot_phagType

###------explained by which DB

clusterCRISPRs_keep_extended_TYPEclustering_forDescription$explainedBLAST

explainedDB <-   clusterCRISPRs_keep_extended_TYPEclustering_forDescription %>% mutate(explained = gsub("orphanCRISPRspacers", "localPHAGE", explainedBLAST)) %>%
  group_by(ARRAYINFO,explainedBLAST) %>% 
  dplyr::summarise(count=n())
# %>% 
#   group_by(ARRAYINFO) %>%
#   dplyr::summarise(countArray=sum(count))
# ungroup() %>%
#   mutate(PhageMapping=100*(count/sum(count)),countArray=sum(count))   %>% select(ARRAYINFO,explainedBLAST,PhageMapping)
# table(explainedDB$explainedBLAST)


tmpCount <-   clusterCRISPRs_keep_extended_TYPEclustering_forDescription %>% mutate(explained = gsub("orphanCRISPRspacers", "localPHAGE", explainedBLAST)) %>%
  group_by(ARRAYINFO,explainedBLAST) %>% 
  dplyr::summarise(count=n()) %>% 
  group_by(ARRAYINFO) %>%
  dplyr::summarise(countArray=sum(count))

explainedDB <- merge(explainedDB,tmpCount,by="ARRAYINFO") %>% mutate(PhageMapping=100*(count/countArray))   %>% select(ARRAYINFO,explainedBLAST,PhageMapping)

explainedDB$explainedBLAST = factor(explainedDB$explainedBLAST, levels=c("No-match" ,"BactDB","localBAC" ,  "phageDB","localPHAGE"))

colorsssss <- rev(c("darkcyan","darkturquoise","goldenrod3","yellow","lightgray"))
  
explainedDB <- explainedDB[which(!is.na(explainedDB$ARRAYINFO)),]

 explainedDB$ARRAYINFO  <- revalue( explainedDB$ARRAYINFO, c("a1"="CRISPR Array 1", "a2"="CRISPR Array 2",   "a3"="CRISPR Array 3"))

 table(explainedDB$explainedBLAST)

mappingWhichDBPlot_DB <- ggplot(explainedDB,aes(x=ARRAYINFO,color=explainedBLAST,fill=explainedBLAST,y=PhageMapping))+labs(x="",y="Spacers mapping [%]")+geom_bar(stat="identity")+theme_classic()+theme(legend.title = element_blank(),legend.position="right",axis.text.x = element_text(angle = 45, hjust = 1,size=9))+
    scale_fill_manual(values=colorsssss)+
   scale_color_manual(values=colorsssss)
mappingWhichDBPlot_DB

# png("~/Desktop/Projects/2019_RMK202_analysis/plot/MappingDB_perArray_02.png", width = 1000, height = 1000,res=300)
 svg("~/Desktop/Projects/2019_RMK202_analysis/plot/MappingDB_perArray_02.svg",width=4,height=4)

mappingWhichDBPlot_DB


dev.off()

explainedDB

###------merge plots

mappingWhichDBPlot_DB+mappingWhichDBPlot+mappingWhichDBPlot_phagType
# png("~/Desktop/Projects/2019_RMK202_analysis/plot/MappingDB_perArray.png", width = 2800, height = 1900,res=300)
 svg("~/Desktop/Projects/2019_RMK202_analysis/plot/MappingDB_perArray_new.svg",width=11,height=8)

mappingWhichDBPlot_DB+mappingWhichDBPlot+mappingWhichDBPlot_phagType


dev.off()
###-----------------------------------
##average number of spacers
explainedDB_02 <-   clusterCRISPRs_keep_extended_TYPEclustering_forDescription %>% mutate(explained = gsub("orphanCRISPRspacers", "localPHAGE", explainedBLAST)) %>%
  group_by(explainedBLAST) %>% 
  dplyr::summarise(count=n())
# %>% 
#   group_by(ARRAYINFO) %>%
#   dplyr::summarise(countArray=sum(count))
# ungroup() %>%
#   mutate(PhageMapping=100*(count/sum(count)),countArray=sum(count))   %>% select(ARRAYINFO,explainedBLAST,PhageMapping)
# table(explainedDB$explainedBLAST)


tmpCount_02 <-   clusterCRISPRs_keep_extended_TYPEclustering_forDescription %>% mutate(explained = gsub("orphanCRISPRspacers", "localPHAGE", explainedBLAST)) %>%
  group_by(explainedBLAST) %>% 
  dplyr::summarise(count=n()) %>% 
  dplyr::summarise(countArray=sum(count))

explainedDB_02 <- merge(explainedDB_02,tmpCount_02,by="explainedBLAST") %>% mutate(PhageMapping=100*(count/countArray))   %>% select(ARRAYINFO,explainedBLAST,PhageMapping)

explainedDB_02$percent <- 100*(explainedDB_02$count/357)
###------do all strains have spacers against the local phages
colnames(spacers_final)
spacers_final_localPhage <- spacers_final %>% filter(!is.na(localPHAGE)) %>% gather(.,sample,spacerCount,"13491":"SMAG", factor_key=TRUE,na.rm = TRUE) %>% filter(spacerCount>0)
# nrow(spacers_final_localPhage)
table(spacers_final_localPhage$sample)
spacers_final_localPhage <- merge(spacers_final_localPhage,ReferenceGenomeGrouping,by.x="sample",by.y="strain")

spacers_final_localPhage$sample = factor(spacers_final_localPhage$sample, levels=(unique(ReferenceGenomeGrouping$strain)))

colorsss <- unique(ReferenceGenomeGrouping$colors)

length(table(spacers_final_localPhage$sample))

plotSpacers <- ggplot(spacers_final_localPhage,aes(x=sample,color=lineage,fill=lineage))+geom_bar()+theme_classic()+scale_fill_manual(values=colorsss)+scale_color_manual(values=colorsss)+theme(axis.text.x=element_text(angle=45, hjust=1,size=9),legend.title = element_blank(),legend.position = "top",axis.title.x = element_blank())+facet_wrap(~localPHAGE)
plotSpacers


svg("~/Desktop/Projects/2019_RMK202_analysis/plot/spacersCount_against_localPhage.svg",width=6,height=3.5)

plotSpacers


dev.off()


###---------------------------
##boxplot of spacers in reference genome per array
###---------------------------
ReferenceGenomeGrouping
spacers_final_Reference <- spacers_final %>% filter(RefStrain_explained!="only Metagenome")  %>% gather(.,sample,spacerCount,"13491":"SMAG", factor_key=TRUE,na.rm = TRUE) %>% filter(spacerCount>0)
spacers_final_Reference_ready <- table(spacers_final_Reference$sample,spacers_final_Reference$ARRAYINFO) %>% as.data.frame()
spacers_final_Reference_ready <- merge(spacers_final_Reference_ready,ReferenceGenomeGrouping,by.x="Var1",by.y="strain")

spacers_final_Reference_ready$Var1 = factor(spacers_final_Reference_ready$Var1, levels=(unique(ReferenceGenomeGrouping$strain)))
colorsss <- unique(ReferenceGenomeGrouping$colors)

ggplot(spacers_final_Reference_ready,aes(x=Var2,y=Freq,color=Var2))+geom_boxplot()
 spacers_final_Reference_ready$Var2  <- revalue( spacers_final_Reference_ready$Var2, c("a1"="CRISPR Array 1", "a2"="CRISPR Array 2",   "a3"="CRISPR Array 3"))

plotSpacerNum <- ggplot(spacers_final_Reference_ready,aes(x=Var1,y=Freq,fill=lineage))+geom_bar(stat="identity")+facet_wrap(~Var2)+theme_classic()+scale_fill_manual(values=colorsss)+labs(x="",y="number of CRISPR spacers")+theme(axis.text.x=element_text(angle=45, hjust=1,size=7),legend.title = element_blank())
plotSpacerNum


svg("~/Desktop/Projects/2019_RMK202_analysis/plot/spacersNum_perArray.svg",width=7,height=4)

plotSpacerNum


dev.off()

spacers_final_Reference_ready %>%   group_by(Var1) %>% dplyr::summarise(countspacers=sum(Freq))


clusterCRISPRs_keep_extended_TYPEclustering_forDescription %>% mutate(explained = gsub("orphanCRISPRspacers", "localPHAGE", explainedBLAST)) %>%
  group_by(ARRAYINFO,explainedBLAST) %>% 
  dplyr::summarise(count=n()) %>% 
  group_by(ARRAYINFO) %>%
  dplyr::summarise(countArray=sum(count))
```

##### lineage-specific spacers
```{r}
###--------------------------------------------------------------
##where do the lineage-specific strains lie
  ###--------------------------------------------------------------
 
tmp1 <- bwaSPACERCount_ggprep_plot_time %>% filter(numSpacers_in_CLUSTER>"7"& get("13491")>"0"& get("13492")>"0"& get("24854")>"0"& get("24737")>"0"& get("13500")>"0"& get("13493")>"0"& get("13498")>"0"& get("SMAG")>"0") %>%select(METAClusterName) %>%  add_column(explaines="lineage 1") %>% unique()# %>% select(METAClusterName) %>% unique() %>% nrow()

tmp2 <- bwaSPACERCount_ggprep_plot_time %>% filter(numSpacers_in_CLUSTER=="3"& get("13494")>"0"& get("24855")>"0"& get("S72")>"0")  %>%select(METAClusterName) %>%  add_column(explaines="lineage 2") %>% unique()

tmp3 <- bwaSPACERCount_ggprep_plot_time %>% filter(numSpacers_in_CLUSTER=="6"& get("13499")>"0"& get("13499c1")>"0"& get("13499c2")>"0"& get("24738")>"0"& get("24740")>"0"& get("S50")>"0") %>%select(METAClusterName) %>%  add_column(explaines="lineage 3") %>% unique()

tmp4 <- bwaSPACERCount_ggprep_plot_time %>% filter(numSpacers_in_CLUSTER=="4"& get("13495")>"0"& get("13496")>"0"& get("13497")>"0"& get("24739")>"0") %>%select(METAClusterName) %>%  add_column(explaines="lineage 4") %>% unique()

tmpAll <- rbind(tmp1,tmp2,tmp3,tmp4)
  table(tmpAll$explaines) 
  nrow(spacers_final)
 lineage_specific_spacers <-  merge(spacers_final,tmpAll,by="METAClusterName")
    nrow(lineage_specific_spacers)
table(lineage_specific_spacers$explaines)
table(lineage_specific_spacers$ARRAYINFO)

lineage_specific_spacers_subset <- lineage_specific_spacers %>% select(METAClusterName,ARRAYINFO,numSpacers_in_CLUSTER,explaines)
# lineage_specific_spacers_subset$explaines <- gsub("group","lineage",lineage_specific_spacers_subset$explaines)
spacerLocation <- spacers_final_Reference_ready %>% group_by(Var2,lineage) %>%  summarize(maxNumberSpacers=max(Freq)) %>% select(Var2,maxNumberSpacers,lineage)
spacerLocation$Var2 <- gsub("CRISPR Array ","a",spacerLocation$Var2)

all_spacers <- merge(lineage_specific_spacers_subset,spacerLocation,by.x=c("explaines","ARRAYINFO"),by.y=c("lineage","Var2"))
all_spacers$relativProx <-  1-all_spacers$numSpacers_in_CLUSTER/all_spacers$maxNumberSpacers
# 
# 
colors <- c("#fd8d3cff","#807dbaff","#ef3b2cff")
# ggplot(all_spacers,aes(x=relativProx,fill=ARRAYINFO))+geom_density(adjust = 5)+theme_classic()+ scale_x_continuous(limits = c(0,1))
# ggplot(all_spacers,aes(x=relativProx,fill=ARRAYINFO))+geom_histogram(color="#e9ecef",alpha=0.8, position = 'identity',bins = 16)+theme_classic()+ scale_x_continuous(limits = c(0,1))
# ggplot(all_spacers,aes(x=relativProx,fill=ARRAYINFO))+geom_histogram(color="#e9ecef",alpha=0.8, position = 'dodge',bins = 16)+theme_classic()+ scale_x_continuous(limits = c(0,1))
plothistospacerlineage <- ggplot(all_spacers,aes(x=relativProx,fill=ARRAYINFO))+geom_histogram(color="#e9ecef",alpha=0.8,bins = 16)+theme_classic()+ scale_x_continuous(limits = c(0,1))+scale_fill_manual(values =colors )+theme(legend.position = "none")+labs(x="Relative proximity to CRISPR leader",y="CRISPR spacer count")

###------------------------
spacers_final_Reference_ready %>% group_by(Var2,lineage) %>%  summarize(maxNumberSpacers=max(Freq))

tmp1 <- table(all_spacers$ARRAYINFO)%>% as.data.frame()%>% add_column(explained="lineag-specific")
tmp_spacer <- spacers_final %>% filter(RefStrain_explained!="only Metagenome") 
tmp2 <- table(tmp_spacer$ARRAYINFO) %>% as.data.frame()%>% add_column(explained="total")
allSpacers <- rbind(tmp2,tmp1)

allSpacers$explained = factor(allSpacers$explained, levels=(c("total","lineag-specific")))

colorsss <- c("lightgray","darkgray")
 uniqueSpac <- ggplot(allSpacers,aes(x=Var1,y=Freq,fill=explained,color=explained))+geom_bar(stat="identity",position ="identity")+theme_classic()+ scale_x_discrete(drop=FALSE)+theme(axis.text.x = element_text(angle = 45, hjust = 1,size=7),legend.position = "none",legend.title = element_blank())+labs(x="",y="CRISPR spacers")+scale_fill_manual(values=colorsss)+scale_color_manual(values=colorsss)+scale_y_continuous(breaks =c(0,8,24,29,60,86),labels=c(0,8,24,29,60,86))
 uniqueSpac
#  
 
 library(patchwork)

 uniqueSpac+plothistospacerlineage+plot_layout(widths = c(1,3))


# table(uniqueSpacersss$uniqueness)
# table(uniqueSpacersss$uniqueness,uniqueSpacersss$ARRAY)
svg("~/Desktop/Projects/2019_RMK202_analysis/plot/UniqueSpacers.svg",width=6,height=3)
 uniqueSpac+plothistospacerlineage+plot_layout(widths = c(1,3))
 dev.off()


# TotSpacer <- spacers_final %>% filter(!is.na(spacer)) %>% group_by(ARRAYINFO) %>%  summarize(totalSpacersARRAY=nrow)
table(TotSpacer$ARRAYINFO)

TotSpacer <- spacers_final %>% filter(!is.na(numSpacers_in_CLUSTER)) %>% group_by(ARRAYINFO) %>%  summarize(totalSpacersARRAY=max(numSpacers_in_CLUSTER))

TotSpacer <- uniqueSpacersss %>% group_by(sample,ARRAY) %>%  summarize(totalSpacersARRAY=max(SPACER))
uniqueSpacersss_extended <-  merge(uniqueSpacersss, TotSpacer, by=c("sample", "ARRAY")) %>% mutate(relDistanceLeader=SPACER/totalSpacersARRAY)

spacerRelDistance <- ggplot(uniqueSpacersss_extended,aes(x=relDistanceLeader,group=ARRAY,fill=ARRAY))+geom_density(alpha=0.3)+theme_classic()+labs(x="relative proximity to CRISPR leader")+theme(legend.position = "none")

# ggplot(uniqueSpacersss_extended,aes(x=relDistanceLeader,group=ARRAY,color=ARRAY,fill=ARRAY))+geom_histogram(alpha=0.3)+theme_classic()+labs(x="relative proximity to CRISPR leader")

svg("~/Desktop/Projects/2019_RMK202_analysis/plot/CRISPR_spacer_count_location_rel.svg",width=4.5,height=2.5)
spacerRelDistance
 dev.off()

```

#####BacBlast
Here, I want to disentangle to what the CRISPRs are mapping to on the bacteria. 
I hyptophesised that they might against parasitic elements, like:
-transposase
-prophages
-mobile elements


Annotated with new prokka to get IS-finder calling.
This step is already done in the annotation chunk. 
I do not know if it is at all necessary anymore.

```{bash}

##========================
##PROKKA annotation
##already done in annotation chunk
##========================
num=0
species=Sterm

##---------------------S50
genomes=202-S50
head -5 /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA_*.gff


num_tmp=$(grep -c "ISfinder" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA_*.gff)
num=$((num+num_tmp))
echo $num_tmp " in " $genome
echo $num
##---------------------S50
genome=202-S72

grep -c "ISfinder" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA_*.gff

num_tmp=$(grep -c "ISfinder" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA_*.gff)
num=$((num+num_tmp))
echo $num_tmp " in " $genome
echo $num
##---------------------S50
genome=202-SMAG

grep -c "ISfinder" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA_*.gff


num_tmp=$(grep -c "ISfinder" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA_*.gff)
num=$((num+num_tmp))
echo $num_tmp " in " $genome
echo $num

mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/

echo -e "transposase\ttotal\t"${num}"\tall" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest.txt

echo -e "transposase\tmatched\t0\tall" >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest.txt

```


######calculations
Where do they blast to on bacterial genome

```{bash}
###--------------------------
##prepare bed of protospacers
###--------------------------
#/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/CRISPRblast_rmk202Bacteria.txt

awk -F "\t" '{OFS="\t"}{if($9>$10) print $2,$10,$9,$1;else print $2,$9,$10,$1 }' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/CRISPRblast_rmk202Bacteria_02.txt |sed 's/REF_mst2://g' |sed 's/REF_mst1://g' |sed 's/REF_R202://g'| sed s'/S_thermophilus_mag_rmk202/CP046134/g' | sed s'/L_delbrueckii_mag_rmk202/CP046131/g' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/locations_bacteria.bed


###--------------------------
##prepare large gff of all bacteria used for protospacer mapping
###--------------------------
species=Sterm
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/all_gffs_toExtractfrom.gff
for genomess in $(echo "202-SMAG 202-S50 202-S72")
do

grep "$(printf '^')${genomess}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA_*.gff  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/all_gffs_toExtractfrom.gff

done

grep "^202-LMAG" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Ldel/202-LMAG/PROKKA_*.gff  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/all_gffs_toExtractfrom.gff

###===============================0
##also do this for the uniq bacteria DB matches
#n=5
##look this up in chunk 
###===============================
for gensss in $(echo "META_Cluster_106 META_Cluster_132 META_Cluster_51")
do
grep "${gensss}" ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_nucleotideDB.txt
done

###--------------------------
##intersect protspacers
##with bacteria Prokka genes
###--------------------------

bedtools intersect -a /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/all_gffs_toExtractfrom.gff  \
  -b /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/locations_bacteria.bed -wb > \
   /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast//intersect_genes.gff


###short analysis
awk '{if($3=="CDS") print $0}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast//intersect_genes.gff >  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast//intersect_genes_reduced.gff

wc -l /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast//intersect_genes_reduced.gff
#grep "hypothetical" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast//intersect_genes_reduced.gff
#  grep "hypothetical" -v /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast//intersect_genes_reduced.gff| awk -F "product=" '{print $2}'


cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast//intersect_genes_reduced.gff| awk -F "product=" '{print $2}' >  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast//intersect_genes_reduced_final.gff
```


```{bash}
##bring local
/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast//intersect_genes_reduced_final.gff

mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast//
scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast//intersect_genes_reduced_final.gff ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast//intersect_genes_reduced_final.gff


```

######analysis


```{r}
table(spacers_final$explainedBLAST)
bacterialMapping_disentagle <- spacers_final %>% filter(explainedBLAST %in% c("localBAC","BactDB")) %>% select(-explainedBLASTEXTENDED)

intersect_genes_reduced_final <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract/CRISPRspacerBLAST/blast/intersect_genes_reduced_final.gff","\t", escape_double = FALSE, col_names = c("explainedBLASTEXTENDED","genome","start","end","METAClusterName"), trim_ws = TRUE) %>% select(explainedBLASTEXTENDED,METAClusterName)

bacterialMapping_disentagle_FINAL <- merge(bacterialMapping_disentagle,intersect_genes_reduced_final,by="METAClusterName",all.x=TRUE) %>% select(METAClusterName,ARRAYINFO,explainedBLASTEXTENDED)

 ###for this part I did an additional blast search of the regions to elucidate what the localBac spacer map to
   bacterialMapping_disentagle_FINAL[which(bacterialMapping_disentagle_FINAL$METAClusterName %in% c("META_Cluster_111","META_Cluster_168")),"explainedBLASTEXTENDED"] <- "plasmid"
   bacterialMapping_disentagle_FINAL[which(bacterialMapping_disentagle_FINAL$METAClusterName %in% c("META_Cluster_106","META_Cluster_132","META_Cluster_51")),"explainedBLASTEXTENDED"] <- "CRISPR"
   table(bacterialMapping_disentagle_FINAL$explainedBLASTEXTENDED)
   
   bacterialMapping_disentagle_FINAL$ARRAYINFO  <- revalue( bacterialMapping_disentagle_FINAL$ARRAYINFO, c("a1"="CRISPR Array 1", "a2"="CRISPR Array 2",   "a3"="CRISPR Array 3"))
 
  
# ggplot(bacterialMapping_disentagle_FINAL,aes(x=ARRAYINFO,color=explainedBLASTEXTENDED,fill=explainedBLASTEXTENDED))+geom_bar(stat="identity")+theme_classic()
   mappingWhichDBPlot_DB_bacteriaONLY <- ggplot(bacterialMapping_disentagle_FINAL,aes(x=ARRAYINFO,color=explainedBLASTEXTENDED,fill=explainedBLASTEXTENDED))+geom_bar()+theme_classic()+labs(x="",y="percent of spacers mapping")+theme(legend.title = element_blank(),legend.position="right",axis.text.x = element_text(angle = 45, hjust = 1,size=9))#+
   #  scale_fill_manual(values=colorsssss)+
   # scale_color_manual(values=colorsssss)
mappingWhichDBPlot_DB_bacteriaONLY
   
   

# png("~/Desktop/Projects/2019_RMK202_analysis/plot/MappingDB_perArray_02.png", width = 1000, height = 1000,res=300)
 svg("~/Desktop/Projects/2019_RMK202_analysis/plot/MappingDB_perArray_BacteriaONLY.svg",width=5,height=4)

mappingWhichDBPlot_DB_bacteriaONLY


dev.off()

   
```



```{bash}
###extended analysis
num=0
genesCount=$(awk '{if($3=="CDS") print $0}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast//intersect_genes.gff |cut -f 13|sort|uniq |wc -l)
 num=$((num+genesCount))
 echo ${num}
 
 num_tot=0
 genesCount=$(grep "^CP046131" -v /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/all_gffs_toExtractfrom.gff|awk '{if($3=="gene") print $0}'|sort|uniq |wc -l)
 
 num_tot=$((num_tot+genesCount))
 echo ${num_tot}

echo -e "genes\ttotal\t"${num_tot}"\tall" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest.txt

echo -e "genes\tmatched" > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest_tmp.txt
echo -e "genes\tmatched" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest_tmp.txt
echo -e "genes\tmatched" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest_tmp.txt

awk '{if($3=="gene") print $0}' /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/intersect_genes.gff |cut -f 13|sort|uniq| sed 's/:R/:/g' |cut -d ':' -f 2 |sort|uniq -c| sed -e 's/^[[:space:]]*//' |sed 's/ /\t/g' > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest_tmp2.txt

paste -d ',' /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest_tmp.txt /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest_tmp2.txt |sed 's/,/\t/g' >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest.txt


###------------------------------
##extract genes
###------------------------------
awk '{if($3=="gene") print $0}' /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/intersect_genes.gff | sed 's/_gene;/\t/g' | sed 's/;Name=/\t/g' > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/intersect_genes_prep.gff
 
 
rm -r /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/genesMapped/
mkdir -p /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/genesMapped/
 
genesss=S72_00155
for genesss in $(cut -f 9 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/intersect_genes_prep.gff | sed 's/ID=//g')
do

genome=$(echo ${genesss} | awk -F "_0" '{print $1}')
 
  echo ${genesss}
samtools faidx /archiv/Projects/2019_RMK202_analysis/Prokka_1_14_withIS/${genome}/*.faa ${genesss} >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/genesMapped/mappedGenes_forKEGG.faa

done

cut -f 9,14 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/intersect_genes_prep.gff | sed 's/ID=//g' > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/intersect_genes_prep_prepped.txt

##-----------------

#is elements


###-----------------transfer local for eggnog


scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/genesMapped/mappedGenes_forKEGG.faa ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/


scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/intersect_genes_prep_prepped.txt ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/


```

#####Prophage

Detection of prophage with phaster.
The genomic location has to be manually extracted out of the output file. 
The output files are save.

```{bash}

#saved on local machine at
ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/06_Phaster


###------------------on bigmachine
mkdir -p /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/Prophages/

##rmk202_sterm (complete)
echo -e "CP046134\t850670\t877821" > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/Prophages/phasterAnnotatedProphage.bed

##S50 (incomplete)
echo -e "NODE_37_length_31935_cov_62.349252\t10121\t20946" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/Prophages/phasterAnnotatedProphage.bed

##S72 (incomplete)
echo -e "NODE_23_length_38834_cov_28.794430\t17059\t27885" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/Prophages/phasterAnnotatedProphage.bed

##ldel rmk202 (two incomplete)
echo -e "CP046131\t212504\t220802" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/Prophages/phasterAnnotatedProphage.bed
echo -e "CP046131\t1990687\t1997908" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/Prophages/phasterAnnotatedProphage.bed

sed 's/CP046134/S_thermophilus_mag_rmk202/g'  /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/Prophages/phasterAnnotatedProphage.bed | sed s'/CP046131/L_delbrueckii_mag_rmk202/g' > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/Prophages/phasterAnnotatedProphage_changedNames.bed

##--------------intersect

bedtools intersect -a /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/Prophages/phasterAnnotatedProphage_changedNames.bed  \
  -b /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/locations_bacteria.bed -wb > \
   /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/intersect_prophage.gff


num=0
num_tmp=$(wc -l /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/intersect_prophage.gff)
num=$((num+num_tmp))
echo $num_tmp " in " $genome
echo $num

num_total=0
num_tmp=$(wc -l //archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/Prophages/phasterAnnotatedProphage_changedNames.bed|cut -d ' ' -f 1)
num_total=$((num_total+num_tmp))
echo $num_tmp " in " $genome
echo $num_total

echo -e "prophage\ttotal\t"${num_total}"\tall" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest.txt

echo -e "prophage\tmatched\t"${num}"\tall" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest.txt


```

The spacers do not seem to be matching to the prophages!

#####mobile elements

prediction with SRID from the [ALM paper](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0223680#sec002)

```{bash}


 
###================
##description file
###================
threads=35
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt  ##the file with all sample names
logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_allRMKs202
#Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/assemblyFinalwithIsolates.fasta
#name_folder=02_againstpolished_allRMKs202

##------------------
#prep
##------------------


###================
##script
###================
##============
##mapping to reference
##============

mkdir -p ${logFilelocation}

num=1

for names in $(cat ${samplesss})
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))
  
  
#mean insertSize

meanInsert=$(grep "mean insert size" ${BaseLocation}/${names}/bwaMapping2DB/bamqc//genome_results.txt |awk -F "= " '{print $2}'|sed 's/,//g' |cut -d '.' -f 1)
stdInsert=$(grep "std insert size" ${BaseLocation}/${names}/bwaMapping2DB/bamqc//genome_results.txt |awk -F "= " '{print $2}'|sed 's/,//g'|cut -d '.' -f 1)


##average read length
ReadLength=$(samtools view ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam  |awk '{print length($10)}' | awk '{ total += $1; count++ } END { print total/count }'|cut -d '.' -f 1 )


echo "========================================"
echo "sample: "$names
echo "mean read length: " ${ReadLength}
echo "mean insert size: " ${meanInsert}
echo "std insert size: " ${stdInsert}
echo "========================================"

  rm -r ${BaseLocation}/02_mobile_elements/${names}

  mkdir -p ${BaseLocation}/02_mobile_elements/${names}
  
/home/vincent/scripts/SRID/SRID.py \
  --bam ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam \
  --out ${BaseLocation}/02_mobile_elements/${names} \
  --readlen ${ReadLength} \
  --mean ${meanInsert} \
  --sd ${stdInsert} \
  --threads ${threads} \
  

done


##----------extract sequences
rm ${BaseLocation}/02_mobile_elements/all_samples_mobileElements.bed
for names in $(cat ${samplesss})
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))
  
  
  awk -F "\t" -v samplezzz="$names" '{OFS="\t"} {print $1,$2,$3,samplezzz}' ${BaseLocation}/02_mobile_elements/${names}/*.tab | grep -v "^Scaffold"  >>  ${BaseLocation}/02_mobile_elements/all_samples_mobileElements.bed
  
done


##---------------------------
##see if spacers map to putative mobile elements
##---------------------------
cut -f 1,2,3 ${BaseLocation}/02_mobile_elements/all_samples_mobileElements.bed |sort|uniq -c

cut -f 1,2,3 ${BaseLocation}/02_mobile_elements/all_samples_mobileElements.bed |sort|uniq  > ${BaseLocation}/02_mobile_elements/all_samples_mobileElements_mapping.bed

bedtools intersect -a ${BaseLocation}/02_mobile_elements/all_samples_mobileElements_mapping.bed  \
  -b /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/locations_bacteria.bed -wb > \
   ${BaseLocation}/02_mobile_elements/mappedCRISPRs_mobileElements_mapping.bed


##------------------------count
#cut -f 7 ${BaseLocation}/02_mobile_elements/mappedCRISPRs_mobileElements_mapping.bed|sort|uniq|sed 's/:R/:/g' |cut -d ':' -f 2|sort|uniq -c | sed -e 's/^[[:space:]]*//' |sed 's/ /\t/g'


 
 num_tot=0
 genesCount=$(wc -l ${BaseLocation}/02_mobile_elements/all_samples_mobileElements_mapping.bed |cut -d ' ' -f 1)
 
 num_tot=$((num_tot+genesCount))
 echo ${num_tot}

echo -e "mobile elements\ttotal\t"${num_tot}"\tall" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest.txt

echo -e "mobile elements\tmatched" > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest_tmp.txt

cut -f 7 /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_allRMKs202/02_mobile_elements/mappedCRISPRs_mobileElements_mapping.bed|sort|uniq|sed 's/:R/:/g' |cut -d ':' -f 2|sort|uniq -c | sed -e 's/^[[:space:]]*//' |sed 's/ /\t/g' > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest_tmp2.txt

paste -d ',' /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest_tmp.txt /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest_tmp2.txt |sed 's/,/\t/g' >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest.txt





```

The spacers do not seem to be matching to the mobile elements!
only spacers that map to the CRISPR arrays

###remaining spacers

```{bash}

awk '{if($3=="gene") print $0}' /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/intersect_genes.gff |cut -f 13|sort|uniq >  /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/matchspacers.txt

cut -f 7 ${BaseLocation}/02_mobile_elements/mappedCRISPRs_mobileElements_mapping.bed|sort|uniq >>  /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/matchspacers.txt


sort /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/matchspacers.txt |uniq|sort > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/matchspacers_uniq.txt

##all spacers
cut -f 4 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/locations_bacteria.bed |sort> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/matchspacers_all.txt

diff /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/matchspacers_uniq.txt /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/matchspacers_all.txt |grep ">" |sed 's/> //g' |sed 's/:R/:/g' |cut -d ':' -f 2 |sort|uniq -c | sed -e 's/^[[:space:]]*//' |sed 's/ /\t/g' >  /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/matchspacers_all_remaining_tmp1.txt


echo -e "remaining\tmatched" > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/matchspacers_all_remaining_tmp2.txt
echo -e "remaining\tmatched" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/matchspacers_all_remaining_tmp2.txt
echo -e "remaining\tmatched" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/matchspacers_all_remaining_tmp2.txt

paste -d ',' /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/matchspacers_all_remaining_tmp2.txt /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/matchspacers_all_remaining_tmp1.txt |sed 's/,/\t/g' >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest.txt


##-----------------------------------------



scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest.txt ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/


```


```{r}
library(ggplot2)
library(readr)
library(RColorBrewer)
library(plyr)
library(tidyverse)


Number_genes_of_interest <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/Number_genes_of_interest.txt",  "\t", escape_double = FALSE, col_names = c("grouping","fraction","spacerCount","array"),  trim_ws = TRUE)

Number_genes_of_interest_01 <- Number_genes_of_interest[which(Number_genes_of_interest$fraction!="total"),]

Number_genes_of_interest_02 <- rbind(Number_genes_of_interest_01,c("mobile elements","mapped","0","a2"))
Number_genes_of_interest_03 <- rbind(Number_genes_of_interest_02,c("mobile elements","mapped","0","a3"))

Number_genes_of_interest_03$spacerCount <- as.numeric(Number_genes_of_interest_03$spacerCount)
 Number_genes_of_interest_03$array  <- revalue( Number_genes_of_interest_03$array, c("a1"="Array 1", "a2"="Array 2",   "a3"="Array 3"))

Number_genes_of_interest_03$grouping = factor(Number_genes_of_interest_03$grouping, levels=(c("transposase" ,"prophage" , "mobile elements" , "genes" , "remaining")))
Number_genes_of_interest_03$array = factor(Number_genes_of_interest_03$array, levels=(c("Array 1", "Array 2",   "Array 3","all")))

spacersmapped_genes_plot <-  ggplot(Number_genes_of_interest_03,aes(x=grouping,y=spacerCount,group=array,color=array,fill=array))+geom_bar(width=0.7,stat="identity",position = position_dodge(width=0.8))+theme_classic()+scale_y_discrete( expand = c(0, 0)) +lims(y=c(-0.5,50))+ theme(legend.title = element_blank(),axis.text.x = element_text(angle = 75, hjust = 1,size=9))+labs(x="",y="spacers mapped")+
    scale_fill_manual(values=three_reds)+
   scale_color_manual(values=three_reds)

spacersmapped_genes_plot
png("~/Desktop/Projects/2019_RMK202_analysis/plot/MappingDB_mappingtoBac.png", width = 1000, height = 1000,res=300)
 # svg("~/Desktop/Projects/2019_RMK202_analysis/plot/MappingDB_perArray.svg",width=11,height=8)

spacersmapped_genes_plot


dev.off()


```

######eggnog

```{r}
library(readr)
genes_soacer <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE,  skip = 3)

intersect_genes_prep_prepped <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/intersect_genes_prep_prepped.txt", "\t", escape_double = FALSE, col_names = c("query_name","spacerName"),   trim_ws = TRUE)


S50 <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3)) 

genes_soacer <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3)) 
colnames(genes_soacer) <- c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description")

COG_functional_categories <- read_delim("~/Desktop/ScriptRepository/COG_functional_categories.csv",  ",", escape_double = FALSE, col_names = c("short","long","divided"),trim_ws = TRUE, na = "NA")
##----------------COG

ALL_sterm <- genes_soacer
nrow(ALL_sterm)

ALL_sterm <- ALL_sterm %>% 
    mutate(COG_Functional_Category = strsplit(as.character(COG_Functional_Category), "")) %>% 
    unnest(COG_Functional_Category)

nrow(ALL_sterm)
# ALL_sterm$co
#unique(sterm_all_unique$COG_Functional_Category)[match(COG_functional_categories$short , unique(sterm_all_unique$COG_Functional_Category))]
unique(ALL_sterm$COG_Functional_Category)[which(!unique(ALL_sterm$COG_Functional_Category) %in% COG_functional_categories$short )]
# ALL_sterm$
# table(ALL_sterm$COG_Functional_Category)

i=4
ALL_sterm$longCOG <- NA
for (i in 1:nrow(ALL_sterm)) {
  ALL_sterm[i,"longCOG"] <- COG_functional_categories[which(COG_functional_categories$short==as.character(ALL_sterm[i,"COG_Functional_Category"])),"long"] %>% as.character()
  
}

# ALL_sterm[which(is.na(ALL_sterm$longCOG)),"COG_Functional_Category"] %>% table()


# ALL_sterm_unnested_not <- ALL_sterm %>% select(COG_Functional_Category)

# ALL_sterm <- ALL_sterm[which(!is.na(ALL_sterm$longCOG)),]
# unique(ALL_sterm$COG_Functional_Category)[which(!unique(ALL_sterm$COG_Functional_Category) %in% COG_functional_categories$short )]


   # svg("~/Desktop/presenations/my_presentations/20190807_committeeMeeting/figures/cogPlot_allSterm.svg",width=7,height=7)
# CogPlotSterm 
 # dev.off()

##-----------------------
##add array information
##-----------------------
ALL_sterm$query_name



ALL_sterm_sub <- ALL_sterm %>% drop_na(COG_Functional_Category) %>% select(query_name,longCOG)
# ALL_sterm[which(ALL_sterm$longCOG=="character(0)"),]%>% select(-1,-2,-3)
ALL_sterm_extended <- merge(ALL_sterm_sub,intersect_genes_prep_prepped,by="query_name") 

ALL_sterm_extended$spacerName <- gsub(":R",":",ALL_sterm_extended$spacerName) 

ALL_sterm_extended$array <- str_split_fixed(ALL_sterm_extended$spacerName, ":", 3)[,2]

 ALL_sterm_extended$array  <- revalue( ALL_sterm_extended$array, c("a1"="Array 1", "a2"="Array 2",   "a3"="Array 3"))

ALL_sterm_extended$array = factor(ALL_sterm_extended$array, levels=(c("Array 1", "Array 2",   "Array 3","all")))


CogPlotSterm <- ggplot(ALL_sterm_extended,aes(x=forcats::fct_infreq(longCOG),group=array,color=array,fill=array))+geom_bar()+
  theme_classic()+
  # facet_wrap(array ~ .)+
  labs(x="",y="spacers mapped")+
    scale_fill_manual(values=three_reds)+
   scale_color_manual(values=three_reds)+
  theme(axis.text.x = element_text(angle = 60, hjust = 1,size=6),
          # # axis.text.y.left = element_text(color = "blue"),
          # axis.text.y.right = element_text(color="red"),
          # axis.title.y.left = element_text(color="blue"),
          # axis.title.y.right = element_text(color="red"),
          # rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          #legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank(),
          plot.margin=unit(c(0,0,0,2), "cm"),
          # axis.text.x.bottom = element_text(size=10)
          #legend.position = c(0.02, 0.98),
          #legend.justification = c("left", "top")
          )

library(patchwork)

spacersmapped_genes_plot/CogPlotSterm

png("~/Desktop/Projects/2019_RMK202_analysis/plot/MappingDB_mappingtoBac_eggnog.png", width = 1000, height = 2000,res=300)

CogPlotSterm

dev.off()


```
### CRISPR spacer dating
Here I hope to see how high the turn over rate is of the CRISPR spacers. 
I aim to do this by looking at the branches in the phylogeny and the seeing how many CRISPR are shared. 

```{r}
colnames(spacers_final)
spacers_final_shared <- spacers_final %>% select(c("METAClusterName" ,"13491","13492" ,"13493", "13494" , "13495" , "13496","13497","13498","13499c1","13499c2" ,"13500","S50","S72" ,"SMAG","numSpacers_in_CLUSTER","ARRAYINFO"))

spacers_final_shared %>% filter(numSpacers_in_CLUSTER=="2"& get("13494")>"0"& get("S72")>"0") %>% nrow()

spacers_final_shared %>% filter(numSpacers_in_CLUSTER>"1"& get("13494")>"0"& get("S72")>"0") 
spacers_final_shared %>% filter(numSpacers_in_CLUSTER>"5"& get("13498")>"0"& get("13500")>"0"& get("SMAG")>"0"& get("13491")>"0"& get("13492")>"0"& get("13493")>"0") %>% nrow()


##-------------------------
##shared between strains in cluster and others
##-------------------------

spacers_final_shared %>% filter(numSpacers_in_CLUSTER>"1"& get("13494")>"0"& get("S72")>"0") %>% nrow()

spacers_final_shared %>% filter(numSpacers_in_CLUSTER>"1"& get("13494")>"0"& get("S72")>"0") 


spacers_final_shared %>% filter(numSpacers_in_CLUSTER>"5"& get("13498")>"0"& get("13500")>"0"& get("SMAG")>"0"& get("13491")>"0"& get("13492")>"0"& get("13493")>"0") %>% nrow()


##-------------------------
##only shared between strains in cluster
##-------------------------

spacers_final_shared %>% filter(numSpacers_in_CLUSTER=="2"& get("13494")>"0"& get("S72")>"0") %>% nrow()

spacers_final_shared %>% filter(numSpacers_in_CLUSTER=="6"& get("13498")>"0"& get("13500")>"0"& get("SMAG")>"0"& get("13491")>"0"& get("13492")>"0"& get("13493")>"0") %>% nrow()

spacers_final_shared %>% filter(numSpacers_in_CLUSTER=="3"& get("13495")>"0"& get("13496")>"0"& get("13497")>"0") %>% nrow()

spacers_final_shared %>% filter(numSpacers_in_CLUSTER=="3"& get("13499c1")>"0"& get("13499c2")>"0"& get("S50")>"0") %>% nrow()

##-------------------------
##only shared between strains in cluster per array
##-------------------------

spacers_final_shared %>% filter(numSpacers_in_CLUSTER=="2"& get("13494")>"0"& get("S72")>"0") %>% group_by(ARRAYINFO) %>% tally()

spacers_final_shared %>% filter(numSpacers_in_CLUSTER=="6"& get("13498")>"0"& get("13500")>"0"& get("SMAG")>"0"& get("13491")>"0"& get("13492")>"0"& get("13493")>"0")%>% group_by(ARRAYINFO) %>% tally()

spacers_final_shared %>% filter(numSpacers_in_CLUSTER=="3"& get("13495")>"0"& get("13496")>"0"& get("13497")>"0")%>% group_by(ARRAYINFO) %>% tally()

spacers_final_shared %>% filter(numSpacers_in_CLUSTER=="3"& get("13499c1")>"0"& get("13499c2")>"0"& get("S50")>"0") %>% group_by(ARRAYINFO) %>% tally()


tmp1 <- spacers_final_shared %>% filter(ARRAYINFO=="a2")
ggplot(tmp1,aes(x=numSpacers_in_CLUSTER))+geom_histogram()+theme_classic()
##-------------------------
##shared between larger clusters
##-------------------------

spacers_final_shared %>% filter(get("13494")>"0"& get("S72")>"0",get("13498")>"0"& get("13500")>"0"& get("SMAG")>"0"& get("13491")>"0"& get("13492")>"0"& get("13493")>"0") %>% nrow()



spacers_final_shared %>% filter(get("13495")>"0"& get("13496")>"0"& get("13497")>"0",get("13499c1")>"0"& get("13499c2")>"0"& get("S50")>"0") %>% nrow()

```


####rarefaction
Here we want to compute two rarefactions
1. the common rarefaction whre we hope to see if we sequenced deep enough and detecting probably all spacers
2. Estimate spacer turnover by seeing how many unique spacers we have. (traditonal pangenome graphs)


rarefaction from [here](https://www.fromthebottomoftheheap.net/2015/04/16/drawing-rarefaction-curves-with-custom-colours/)
```{r}
library(ggplot2)
library(readr)
library(forcats) 

clusterCRISPRs_allArrays <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/clusterCRISPRs_allArrays.txt", "\t", escape_double = FALSE, trim_ws = TRUE)
library(RColorBrewer)

# clusterCRISPRs_allArrays <-CRISPR_spacer_coverage_extended
library(tidyverse)
# rownames(clusterCRISPRs_allArrays) <- clusterCRISPRs_allArrays$REFspacer
# colnames(clusterCRISPRs_allArrays)
# clusterCRISPRs_allArrays_matrix <- clusterCRISPRs_allArrays %>% select (lyo202_96,L_202_2012,L_202_2014,RMK202,Konserve_202,Versand_202,di_K2_6h,th_K2_6h)%>% as.matrix() %>% t()

clusterCRISPRs_allArrays <- CRISPR_spacer_coverage_extended_wide
rownames(clusterCRISPRs_allArrays) <- clusterCRISPRs_allArrays$Name
colnames(clusterCRISPRs_allArrays)
clusterCRISPRs_allArrays_matrix <- clusterCRISPRs_allArrays %>% select ("Lyo\n1996", "Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2")%>% as.matrix() %>% t()

# clusterCRISPRs_allArrays_matrix <- CRISPR_spacer_coverage_extended_wide %>% column_to_rownames(var="Name") %>% round() %>% as.matrix() %>% t()
mode(clusterCRISPRs_allArrays_matrix) <- "integer"
# class(clusterCRISPRs_allArrays_matrix)
# dim(clusterCRISPRs_allArrays_matrix)
# str(clusterCRISPRs_allArrays_matrix)
# clusterCRISPRs_allArrays_matrix <- clusterCRISPRs_allArrays_matrix[which(rowSums(clusterCRISPRs_allArrays_matrix)>0),]
# dim(clusterCRISPRs_allArrays_matrix)
# typeof(clusterCRISPRs_allArrays_matrix)
library(vegan) 
# data(BCI)
S <- specnumber(clusterCRISPRs_allArrays_matrix) # observed number of species
# dim(BCI)
(raremax <- min(rowSums(clusterCRISPRs_allArrays_matrix))) %>% as.numeric()
Srare <- rarefy(as.data.frame(clusterCRISPRs_allArrays_matrix), raremax)
plot(S, Srare, xlab = "sampled", ylab = "Rarefied No. of Species")
abline(0, 1)
rarecurve(clusterCRISPRs_allArrays_matrix, step = 20, sample = raremax, col = "blue", cex = 0.6,xlab="CRISPR spacer reads",ylab="Number of CRISPR spacers rarefied", label = FALSE)

sampleNumber <- as.numeric(nrow(clusterCRISPRs_allArrays_matrix))

col <- as.character(metasample_colors[match(rownames(clusterCRISPRs_allArrays_matrix),metasample_colors$sample),"colors"])

# col <- c(brewer.pal(9,name="YlGnBu")[c(5,4,3,2)],"lightblue","darkturquoise","darkcyan","#0062B4FF")
 # col <- c("black", "darkred", "forestgreen", "orange", "blue", "yellow", "hotpink","green","green")
lty <- c("solid")
lwd <- c(3)
pars <- expand.grid(col = col, lty = lty, stringsAsFactors = FALSE)
out <- with(pars[1:sampleNumber, ],rarecurve(clusterCRISPRs_allArrays_matrix, step = 20, sample = raremax, col = col, label = FALSE))
pars




# svg("~/Desktop/Projects/2019_RMK202_analysis/plot/Rarefaction.svg",width=8,height=8)
# png("~/Desktop/Projects/2019_RMK202_analysis/plot/Pan_arrayPlot.png", width = 1600, height = 1200,res=300)

Nmax <- sapply(out, function(x) max(attr(x, "Subsample")))
Smax <- sapply(out, max)
plot(c(1, max(Nmax)), c(1, max(Smax)), xlab = "CRISPR spacer reads",
     ylab = "Number of CRISPR spacers rarefied", type = "n")
# abline(v = raremax)
for (i in seq_along(out)) {
    N <- attr(out[[i]], "Subsample")
    with(pars, lines(N, out[[i]], col =alpha(col[i], 0.8), lty = lty[i], lwd = lwd))
}
# legend("bottomright",  cex=0.5,
#        legend=rownames(clusterCRISPRs_allArrays_matrix), pch=19,
#       col=col, bty="n")


dev.off()
 

rowSums_count <- rowSums(clusterCRISPRs_allArrays_matrix) %>% as.data.frame()
rowSums_count$sample <- rownames(rowSums_count)
colnames(rowSums_count)[1] <- "count"


##create a named vector for revalue
test <- setNames(metasample_colors$samplefinal_oneLines,metasample_colors$samplefinal_multipleLines)
##revalue
rowSums_count$sampleNames  <- plyr::revalue( rowSums_count$sample, test)

rowSums_count$sampleNames = factor(rowSums_count$sampleNames, levels=rowSums_count[order(rowSums_count$count,decreasing = TRUE),"sampleNames"])




col <- as.character(metasample_colors[match(levels(rowSums_count$sampleNames),as.character(metasample_colors$samplefinal_oneLines)),"colors"])
CRISPRCOUNTplot <- ggplot(rowSums_count,aes(x=sampleNames,y=count,fill=sampleNames,color=sampleNames))+ geom_bar( stat="identity")+
    scale_fill_manual(values=col)+
   scale_color_manual(values=col)+
  labs(y="CRISPR spacer reads")+
  theme_classic()+theme(legend.title = element_blank(),axis.title.x = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),panel.border = element_rect(colour = "black", fill=NA, size=1))+
  scale_y_discrete( expand = c(0, 0)) 

svg("~/Desktop/Projects/2019_RMK202_analysis/plot/CRISPR_countplot.svg",width=6.5,height=4)
# png("~/Desktop/Projects/2019_RMK202_analysis/plot/Pan_arrayPlot.png", width = 1600, height = 1200,res=300)

CRISPRCOUNTplot


dev.off()



##-----------------------
##pan-CRISPR array
##-----------------------


clusterCRISPRs_allArrays <-CRISPR_spacer_coverage_extended_wide
library(tidyverse)
rownames(clusterCRISPRs_allArrays) <- clusterCRISPRs_allArrays$Name
colnames(clusterCRISPRs_allArrays)
# clusterCRISPRs_allArrays_matrix <- clusterCRISPRs_allArrays %>% select (lyo202_96,L_202_2012,L_202_2014,RMK202,Konserve_202,Versand_202,di_K2_6h,th_K2_6h)%>% as.matrix() %>% t()


# as.data.frame(clusterCRISPRs_allArrays_matrix)>0 %>% rowSums()
# clusterCRISPRs_allArrays_df <-  clusterCRISPRs_allArrays %>% select (lyo202_96,L_202_2012,L_202_2014,RMK202,Konserve_202,Versand_202,di_K2_6h,th_K2_6h,Name) %>% unique()%>% remove_rownames() %>%  column_to_rownames(var="Name")
clusterCRISPRs_allArrays_df <- clusterCRISPRs_allArrays %>% select ("Lyo\n1996", "Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2")%>% unique()


clusterCRISPRs_allArrays_df_zero <- clusterCRISPRs_allArrays_df>0 

clusterCRISPRs_allArrays_df_zero_summed <- rowSums(clusterCRISPRs_allArrays_df_zero) %>% as.data.frame()
# table(clusterCRISPRs_allArrays_df_zero_summed$.)
colnames(clusterCRISPRs_allArrays_df_zero_summed)[1] <- "sampleCount"
clusterCRISPRs_allArrays_df_zero_summed$spacerName <- rownames(clusterCRISPRs_allArrays_df_zero_summed)

clusterCRISPRs_allArrays_df_zero_summed$spacerName <- gsub(":R",":",clusterCRISPRs_allArrays_df_zero_summed$spacerName) %>% gsub("^>","",.)
clusterCRISPRs_allArrays_df_zero_summed$array <- str_split_fixed(clusterCRISPRs_allArrays_df_zero_summed$spacerName, ":", 3)[,2]
clusterCRISPRs_allArrays_df_zero_summed$sample <- str_split_fixed(clusterCRISPRs_allArrays_df_zero_summed$spacerName, ":", 3)[,1]

clusterCRISPRs_allArrays_df_zero_summed$grouping <- NA
# clusterCRISPRs_allArrays_df_zero_summed[which(clusterCRISPRs_allArrays_df_zero_summed$.=="1"),]

i=2
for (i in 1:nrow(clusterCRISPRs_allArrays_df_zero_summed)) {
  clusterCRISPRs_allArrays_df_zero_summed[i,"grouping"] <- ifelse(clusterCRISPRs_allArrays_df_zero_summed[i,"sampleCount"]=="1",as.character(clusterCRISPRs_allArrays_df_zero_summed[i,"sample"]),"multiple")
  
  
}
table(clusterCRISPRs_allArrays_df_zero_summed$grouping)
##----------------------------------------------
##create a named vector for revalue ( the second part has to be changed according to the name of the smaple to be changed)
test <- setNames(metasample_colors$samplefinal_oneLines,metasample_colors$shortfinal_2)
##revalue
clusterCRISPRs_allArrays_df_zero_summed$grouping <- plyr::revalue(clusterCRISPRs_allArrays_df_zero_summed$grouping, test)

clusterCRISPRs_allArrays_df_zero_summed$grouping = factor(clusterCRISPRs_allArrays_df_zero_summed$grouping, levels=(c("starter culture 2018", "Lyo 2012","cheesemaking day2", "cheesemaking day1","Lyo 2014","Lyo 1996","multiple")))

# col <- 
  
  clusterCRISPRs_allArrays_df_zero_summed <- clusterCRISPRs_allArrays_df_zero_summed %>% filter(sampleCount>0) %>% filter(!is.na(grouping))


col <- c(as.character(metasample_colors[match(levels(clusterCRISPRs_allArrays_df_zero_summed$grouping),metasample_colors$samplefinal_oneLines),"colors"][1:5]),"lightgray")

##----------------------------------------------

Pan_arrayPlot <- ggplot(clusterCRISPRs_allArrays_df_zero_summed,aes(x=sampleCount,fill=grouping,color=grouping,group=grouping))+geom_bar()+theme_classic()+labs(x="number of samples",y="spacer count")+theme(legend.title = element_blank())+
    scale_fill_manual(values=col)+
   scale_color_manual(values=col)
Pan_arrayPlot


colnames(clusterCRISPRs_allArrays_df_zero_summed)

# svg("~/Desktop/Projects/2019_RMK202_analysis/plot/Pan_arrayPlot.svg",width=8,height=6)
png("~/Desktop/Projects/2019_RMK202_analysis/plot/Pan_arrayPlot.png", width = 1800, height = 1200,res=300)

Pan_arrayPlot


dev.off()

##for plotting afterwards
clusterCRISPRs_allArrays_subset <- clusterCRISPRs_allArrays_df_zero_summed %>% select(spacerName,sampleCount)
```

##spacer coverage

has to been run after the blast Spacers + map meta chunks

here we want to plot the coverage of the spacers over time therefore we first need to filter very rare spacers
```{r}
##------------------------------lineplot 
##coverage of spacers over time

CRISPR_spacer_coverage_extended$Name_short <- gsub(":R",":",CRISPR_spacer_coverage_extended$Name)
CRISPR_spacer_coverage_extended <- merge(CRISPR_spacer_coverage_extended,clusterCRISPRs_allArrays_subset,by.y = "spacerName",by.x="Name_short",all.x = TRUE)
table(CRISPR_spacer_coverage_extended$sampleCount)

CRISPR_spacer_coverage_extended_remove_unique <- CRISPR_spacer_coverage_extended %>% filter(sampleCount>1)
table(CRISPR_spacer_coverage_extended_remove_unique$sampleCount)
    
CRISPR_spacer_coverage_extended_unique <- CRISPR_spacer_coverage_extended %>% filter(sampleCount==1)


CRISPR_spacer_coverage_extended[which(CRISPR_spacer_coverage_extended$adap=="Array2"&CRISPR_spacer_coverage_extended$sampleCount=="5"),]

# range(CRISPR_spacer_coverage$Cpm)
# CRISPR_spacer_coverage_extended$explained
  # p2 <- ggplot(CRISPR_spacer_coverage_02,aes(x=sample,y=CPM,group=ClusterName,color=assignedArray))+ geom_line(size=0.5, alpha=.5)+ 
  p3 <- ggplot(CRISPR_spacer_coverage_extended,aes(x=sample,y=normalizedCount,group=Cluster,color=RefStrain))+ geom_line(size=0.4, alpha=.4)+ 
    # facet_grid(day~kessel~species)+
     # facet_grid(DB~.)+
    facet_wrap(adap~sampleCount, scales="free")+
    # facet_wrap(explained~., scales="free")+
    # coord_trans(y="log2")+
    # facet_grid(treatment~species)+
    # facet_grid(treatment~species)+
    labs("",
         x=" ",
         y="counts per million")+
    #scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    # scale_y_continuous(limits = c(0,10))+
    theme_classic()+
   # scale_fill_manual(values=all_colours)+
   # scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
      #axis.text.x = element_blank(),
          legend.position="right"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          )
  
  p3

```



####Phage cluster ANI/AAI

calculate the ANI and AAI within the mapped VCs.

```{bash}

##prepare file to assign genome names

mkdir -p /archiv/Phage_DB/BLAST/20190519/downloads/log/
rm /archiv/Phage_DB/BLAST/20190519/downloads/log/Sterm_namesAssigment
for Filezz in $(ls /archiv/Phage_DB/BLAST/20190519/downloads/Sterm/Renamed/RenamedContigs/ |grep ".f"|sed 's/.fna//g')
do
echo ${Filezz}
contigsss=$(grep "^>" /archiv/Phage_DB/BLAST/20190519/downloads/Sterm/Renamed/RenamedContigs/${Filezz}.fna|head -1)

echo -e ${Filezz}"\t"${contigsss} >> /archiv/Phage_DB/BLAST/20190519/downloads/log/Sterm_namesAssigment

done


##-lactococcus phages
mkdir -p /archiv/Phage_DB/BLAST/20190519/downloads/log/
rm /archiv/Phage_DB/BLAST/20190519/downloads/log/lacto_namesAssigment
for Filezz in $(ls /archiv/Phage_DB/BLAST/20190519/downloads/Lactococcus//Renamed/RenamedContigs/ |grep ".f"|sed 's/.fna//g')
do
echo ${Filezz}
contigsss=$(grep "^>" /archiv/Phage_DB/BLAST/20190519/downloads/Lactococcus//Renamed/RenamedContigs/${Filezz}.fna|head -1)

echo -e ${Filezz}"\t"${contigsss} >> /archiv/Phage_DB/BLAST/20190519/downloads/log/lacto_namesAssigment

done

##--------bring local 
scp  vincent@130.223.51.116:/archiv/Phage_DB/BLAST/20190519/downloads/log/Sterm_namesAssigment ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/
scp  vincent@130.223.51.116:/archiv/Phage_DB/BLAST/20190519/downloads/log/lacto_namesAssigment ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/

##--------bring BIGMACHINE 
scp -r ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/CLUSTERS/* vincent@130.223.51.116:/archiv/Phage_DB/BLAST/20190519/downloads/log/

##----------------make a fasta path file for ANI calcualtions
path=/archiv/Phage_DB/BLAST/20190519/downloads/Sterm/Renamed/RenamedContigs/
#path=/archiv/Phage_DB/BLAST/20190519/downloads/Sterm/Renamed/RenamedContigs/${Filezz}.fna
i=150_0
mkdir -p /archiv/Phage_DB/BLAST/20190519/downloads/log/forANI/
for i in $(echo "154_0  307_0  309_0  359_0")
do
rm /archiv/Phage_DB/BLAST/20190519/downloads/log/forANI/phagesVcontact_${i}_cluster.txt
echo ${i} 
awk -v pathss="$path" '{OFS=""}{print pathss,$1,".fna"}' /archiv/Phage_DB/BLAST/20190519/downloads/log/phagesVcontact_${i}_cluster.txt >> /archiv/Phage_DB/BLAST/20190519/downloads/log/forANI/phagesVcontact_${i}_cluster.txt

done

##---lactococcus
path=/archiv/Phage_DB/BLAST/20190519/downloads/Lactococcus/Renamed/RenamedContigs/

i="150_0"
rm /archiv/Phage_DB/BLAST/20190519/downloads/log/forANI/phagesVcontact_${i}_cluster.txt
awk -v pathss="$path" '{OFS=""}{print pathss,$1,".fna"}' /archiv/Phage_DB/BLAST/20190519/downloads/log/phagesVcontact_${i}_cluster.txt >> /archiv/Phage_DB/BLAST/20190519/downloads/log/forANI/phagesVcontact_${i}_cluster.txt

##------------------------------
##---make and ANI AAI calculations
##------------------------------

i=359_0
rm /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/03_ANI_AAI_inClust/ANI_AAI/final_AAI.output
rm /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/03_ANI_AAI_inClust/ANI_AAI/final_ANI.output
for i in $(echo "150_0 154_0  307_0  309_0  359_0")
do
rm -r /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/03_ANI_AAI_inClust/ANI_AAI/${i}_clust_ANI//
mkdir -p   /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/03_ANI_AAI_inClust/ANI_AAI/${i}_clust_ANI//
echo ${i} 

grep "character(0).fna"  -v /archiv/Phage_DB/BLAST/20190519/downloads/log/forANI/phagesVcontact_${i}_cluster.txt > /archiv/Phage_DB/BLAST/20190519/downloads/log/forANI/phagesVcontact_${i}_cluster_corrected.txt
  
  comparem aai_wf /archiv/Phage_DB/BLAST/20190519/downloads/log/forANI/phagesVcontact_${i}_cluster_corrected.txt /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/03_ANI_AAI_inClust/ANI_AAI/${i}_clust_AAI/

awk -F "\t" -v samplezz="$i" '{OFS="\t"} {print samplezz,$0}' /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/03_ANI_AAI_inClust/ANI_AAI/${i}_clust_AAI/aai/aai_summary.tsv |grep "Genome A" -v >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/03_ANI_AAI_inClust/ANI_AAI/final_AAI.output

fastANI --fragLen 1000 --minFrag 10 -t 37 --ql /archiv/Phage_DB/BLAST/20190519/downloads/log/forANI/phagesVcontact_${i}_cluster_corrected.txt \
  --rl /archiv/Phage_DB/BLAST/20190519/downloads/log/forANI/phagesVcontact_${i}_cluster_corrected.txt \
  -o /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/03_ANI_AAI_inClust/ANI_AAI/${i}_clust_ANI/ANI_output.txt

awk -F "\t" -v samplezz="$i" '{OFS="\t"} {print samplezz,$0}' /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/03_ANI_AAI_inClust/ANI_AAI/${i}_clust_ANI/ANI_output.txt >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/03_ANI_AAI_inClust/ANI_AAI/final_ANI.output

done


#----------------move local
scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/03_ANI_AAI_inClust/ANI_AAI/final_* ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/

##========================================
##for all clusters
##========================================

###-----------prepare



for speciesss in $(echo "Sterm Ldel Lactococcus")
do

rm /archiv/Phage_DB/BLAST/20190519/downloads/log/${speciesss}_namesAssigment
for Filezz in $(ls /archiv/Phage_DB/BLAST/20190519/downloads/${speciesss}/Renamed/RenamedContigs/ |grep ".f"|sed 's/.fna//g')
do
echo ${Filezz}
contigsss=$(grep "^>" /archiv/Phage_DB/BLAST/20190519/downloads/${speciesss}/Renamed/RenamedContigs/${Filezz}.fna|head -1)

echo -e ${Filezz}"\t"${contigsss}"," >> /archiv/Phage_DB/BLAST/20190519/downloads/log/${speciesss}_namesAssigment

done
done  #speciesss



###--------------------get path to genome file
rm -r /archiv/Phage_DB/BLAST/20190519/downloads/log/ForANI_all/
mkdir -p /archiv/Phage_DB/BLAST/20190519/downloads/log/ForANI_all/


for speciesss in $(echo "Sterm Ldel Lactococcus")
do
if [ "$speciesss" = "Sterm" ]
then
echo $speciesss
species_2=Streptococcus
elif [ "$speciesss" = "Ldel" ]
then
species_2=Lactobacillus
else
	species_2=Lactococcus
fi
echo $speciesss $species_2


for genomess in $(grep "$species_2" /archiv/Phage_DB/BLAST/20190519/Vcontact/Database/alltogether/database/genome_by_genome_overview.csv |cut -d ',' -f 2|awk -F "\t" '{OFS=""}{print $1,","}')
do

echo $genomess 
genomess_2=$(echo $genomess | sed 's/~/ /g')

VCss=$(grep "$genomess" /archiv/Phage_DB/BLAST/20190519/Vcontact/Database/alltogether/database/genome_by_genome_overview.csv |cut -d ',' -f 6)

fileName=$(grep "$genomess_2" /archiv/Phage_DB/BLAST/20190519/downloads/log/${speciesss}_namesAssigment |cut -f 1)


echo "/archiv/Phage_DB/BLAST/20190519/downloads/"${speciesss}"/Renamed/RenamedContigs/"${fileName}".fna" >>  /archiv/Phage_DB/BLAST/20190519/downloads/log/ForANI_all/Cluster_${VCss}_namesAssigment_download.file


done #genomess

done #species


###--------------------------------------call ANI and AAI

clustersss=101_0
rm -r /archiv/Phage_DB/BLAST/20190519/downloads/log/ForANI_all/cleaned
mkdir -p /archiv/Phage_DB/BLAST/20190519/downloads/log/ForANI_all/cleaned

rm -r /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/03_ANI_AAI_inClust/ANI_AAI_all/
mkdir -p /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/03_ANI_AAI_inClust/ANI_AAI_all/

for clustersss in $(ls /archiv/Phage_DB/BLAST/20190519/downloads/log/ForANI_all/ |grep ".file"|grep "nan" -v |sed 's/Cluster_//g' | sed 's/_namesAssigment_download.file//g')
do
mkdir -p /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/03_ANI_AAI_inClust/ANI_AAI_all/${clustersss}_clust_AAI/

echo $clustersss


grep -v "/.fna" /archiv/Phage_DB/BLAST/20190519/downloads/log/ForANI_all/Cluster_${clustersss}_namesAssigment_download.file |grep "character(0).fna"  -v  > /archiv/Phage_DB/BLAST/20190519/downloads/log/ForANI_all/cleaned/Cluster_${clustersss}_namesAssigment_download.file

  comparem aai_wf /archiv/Phage_DB/BLAST/20190519/downloads/log/ForANI_all/cleaned/Cluster_${clustersss}_namesAssigment_download.file /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/03_ANI_AAI_inClust/ANI_AAI_all/${clustersss}_clust_AAI/

awk -F "\t" -v samplezz="$clustersss" '{OFS="\t"} {print samplezz,$0}' /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/03_ANI_AAI_inClust/ANI_AAI_all/${clustersss}_clust_AAI/aai/aai_summary.tsv |grep "Genome A" -v >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/03_ANI_AAI_inClust/ANI_AAI_all/final_AAI.output
mkdir -p /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/03_ANI_AAI_inClust/ANI_AAI_all/${clustersss}_clust_ANI

fastANI --fragLen 1000 --minFrag 5 -t 37 --ql /archiv/Phage_DB/BLAST/20190519/downloads/log/ForANI_all/cleaned/Cluster_${clustersss}_namesAssigment_download.file \
  --rl /archiv/Phage_DB/BLAST/20190519/downloads/log/ForANI_all/cleaned/Cluster_${clustersss}_namesAssigment_download.file \
  -o /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/03_ANI_AAI_inClust/ANI_AAI_all/${clustersss}_clust_ANI/ANI_output.txt

awk -F "\t" -v samplezz="$clustersss" '{OFS="\t"} {print samplezz,$0}' /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/03_ANI_AAI_inClust/ANI_AAI_all/${clustersss}_clust_ANI/ANI_output.txt >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/03_ANI_AAI_inClust/ANI_AAI_all/final_ANI.output


done

scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/03_ANI_AAI_inClust/ANI_AAI_all/fina* ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/


```

```{r}

library(readr)
final_ANI <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/final_ANI.output",  "\t", escape_double = FALSE, col_names = c("VC","GenomeA","GenomeB","ANI","mappedFragemnts","totFragemnts"),  trim_ws = TRUE)
final_ANI$coverage_ANI <- 100*(final_ANI$mappedFragemnts/final_ANI$totFragemnts)

final_AAI <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/final_AAI.output", "\t", escape_double = FALSE, col_names = c("VC","GenomeA","numGenesA","GenomeB","numGenesB","sharedGenes","AAI","std","coverage_AAI"), trim_ws = TRUE)


##------------calculate means

## of: ANI +coverage 
  ANI_mean <- aggregate(. ~VC, data=final_ANI[,c("VC","ANI")], mean, na.rm=TRUE)
  ANI_coverage <- aggregate(. ~VC, data=final_ANI[,c("VC","coverage_ANI")], mean, na.rm=TRUE)

  table(final_ANI$VC) %>%as.data.frame()
tmp <- merge(ANI_mean,ANI_coverage,by="VC")

  ## of: AAI +coverage 

  AAI_mean <- aggregate(. ~VC, data=final_AAI[,c("VC","AAI")], mean, na.rm=TRUE)
  AAI_coverage <- aggregate(. ~VC, data=final_AAI[,c("VC","coverage_AAI")], mean, na.rm=TRUE)
tmp2 <- merge(AAI_mean,AAI_coverage,by="VC")

  finalANI_AAI_tmp <- merge(tmp,tmp2,by="VC")

  ##add size 

  
Sterm_clustering_complete <- read_csv("~/Desktop/Projects/2019_PhageDB/Vcontact/allPhages_vContact/genome_by_genome_overview.csv")


  tmp3 <- Sterm_clustering_complete %>% select(VC,Size) %>% unique()

  
  finalANI_AAI <- merge(finalANI_AAI_tmp,tmp3,by="VC",all.x=TRUE)
  
   finalANI_AAI$explained  <- finalANI_AAI$VC
   
 finalANI_AAI$explained  <- revalue( finalANI_AAI$explained, c("150_0"="Cluster 150", "359_0"="Phage Type PAC",   "309_0"="Phage Type 5093",  "307_0"="Phage Type COS",   "154_0"="Phage Type 987"))

finalANI_AAI[grep("_",finalANI_AAI$explained),"explained"] <- "unknown Cluster"
  
  # clusterCRISPRs_keep_extended_TYPEclustering %>% select(VC,explainedType) %>% unique()
  
  ##PLOT
  # colorsss <- (c("#0062B4FF","darkcyan","darkturquoise","lightblue"))
colorsss <- rev(c("#0062B4FF","darkcyan","darkturquoise","lightblue","chocolate1","lightgray"))

finalANI_AAI$explained = factor(finalANI_AAI$explained, levels=rev(c("Phage Type 5093" ,"Phage Type 987" , "Phage Type COS" , "Phage Type PAC" , "Cluster 150","unknown Cluster")))

finalANI_AAI <- finalANI_AAI %>% arrange(explained)

  ANI_plot <- ggplot(finalANI_AAI,aes(x=ANI,y=AAI,size=Size,color=explained,fill=explained))+geom_point()+theme_classic()+labs(x="ANI [%]", y="AAI [%]")+
    scale_fill_manual(values=colorsss)+
   scale_color_manual(values=colorsss)
  ANI_plot
  
  coveraage_plot <- ggplot(finalANI_AAI,aes(x=coverage_ANI,y=coverage_AAI,size=Size,color=explained,fill=explained))+geom_point()+theme_classic()+labs(x="coverage ANI [%]", y="coverage AAI [%]")+theme(legend.title = element_blank())+
    scale_fill_manual(values=colorsss)+
   scale_color_manual(values=colorsss)
    
  coveraage_plot
  library(patchwork)
    plotMerged <-  (ANI_plot+theme(legend.position = "none") +coveraage_plot) 
    
    
    # png("~/Desktop/Projects/2019_RMK202_analysis/plot/MappingDB_perArray_ANI_AAI.png", width = 2100, height = 950,res=300)
 svg("~/Desktop/Projects/2019_RMK202_analysis/plot/MappingDB_perArray_ANI_AAI.svg",width=8,height=3.5)

    plotMerged


dev.off()
```



###spacer coverage
Here, we look at the coverage of every spacer over time in order to see if the spacers we see today are newly aquired. 

```{bash}

###================
##description file
###================
threads=37
BaseLocation_bed=/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_allRMKs202
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt  ##the file with all sample names

###------------------------------------
##merge spacer bed files
###------------------------------------

mkdir -p ${BaseLocation_bed}/CRISPRspacers/location/

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/S_thermophilus_RMK202.pilarCR_out_spacer.bed \
/archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out_spacer.bed \
/archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out_spacer.bed > \
${BaseLocation_bed}/CRISPRspacers/location/allSPACERS_sterm_location.bed


##============
##extract coverage
##============

# name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 


num=1
rm -r ${BaseLocation_bed}/CRISPRspacers/location/mappingCoverage
 mkdir -p ${BaseLocation_bed}/CRISPRspacers/location/mappingCoverage
for names in $(cat ${samplesss})
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))

  samtools view -h -q 30 -b ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapQ30.bam 
 
  
bedtools coverage -bed -a ${BaseLocation_bed}/CRISPRspacers/location/allSPACERS_sterm_location.bed -b ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapQ30.bam  >  ${BaseLocation_bed}/CRISPRspacers/location/mappingCoverage/${names}_coverage_CRISPRspacers.bed 
  
  
  
  awk -F "\t" -v name="$names" '{OFS="\t"}{print $0,name}' ${BaseLocation_bed}/CRISPRspacers/location/mappingCoverage/${names}_coverage_CRISPRspacers.bed >> ${BaseLocation_bed}/CRISPRspacers/location/mappingCoverage/all_coverage_CRISPRspacers.bed 
  
  echo ${BaseLocation_bed}/CRISPRspacers/location/mappingCoverage/all_coverage_CRISPRspacers.bed 
   done 
   
scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacers/location/mappingCoverage/all_coverage_CRISPRspacers.bed ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/

```
analysis of coverage in R
```{r}
library(readr)
library(stringr)
library(tidyverse)
library(plyr)
library(dplyr)
all_coverage_CRISPRspacers <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/all_coverage_CRISPRspacers.bed", "\t", escape_double = FALSE, col_names = c("contig","start","end","name","numreads","bases","basescovered","percentCovered","sample"),  trim_ws = TRUE)

all_coverage_CRISPRspacers$genome <- str_split_fixed(all_coverage_CRISPRspacers$name, ":", 2)[,1]
str_split_fixed(all_coverage_CRISPRspacers$name, ":", 2)[,2] %>% str_split_fixed(., "spc", 2)[,1]

str_split_fixed(all_coverage_CRISPRspacers$name, fixed(":","spc"), 3)[,2]

all_coverage_CRISPRspacers %>% separate(name, into = c("genome","array",), sep = '[+-]')
all_coverage_CRISPRspacers <- all_coverage_CRISPRspacers %>% 
  mutate(name = gsub("ary", "", name), # a with acute
         name = gsub("spc", ":", name) # a with acute
   )%>% separate(name, into = c("genome","array","spacer"), sep = '[:]')
all_coverage_CRISPRspacers$spacerID <- paste(all_coverage_CRISPRspacers$genome,":",all_coverage_CRISPRspacers$array,":",all_coverage_CRISPRspacers$spacer)
all_coverage_CRISPRspacers$arrayID <- paste(all_coverage_CRISPRspacers$genome," array :",all_coverage_CRISPRspacers$array)

##--------------------plot


all_coverage_CRISPRspacers$sample <- revalue(all_coverage_CRISPRspacers$sample, c("lyo202_96"="Lyo\n1996","Lyo_202_2012"="Lyo\n2012", "Lyo_202_2014"="Lyo\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018","di_K2_6h"="cheesemaking\nday1","th_K2_8h"="cheesemaking\nday2"))

all_coverage_CRISPRspacers$sample = factor(all_coverage_CRISPRspacers$sample, levels=c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2"))

  p3 <- ggplot(all_coverage_CRISPRspacers,aes(x=sample,y=numreads,group=spacerID,color=arrayID,fill=arrayID,))+ geom_line(size=0.5, alpha=.5)+ 
    # facet_grid(day~kessel~species)+
    # facet_grid(species~.)+
    # facet_grid(treatment~species)+
    # facet_grid(treatment~species)+
    labs("",
         x="",
         y="number of reads")+
    #scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
   scale_y_continuous(limits = c(0,500))+
    theme_classic()+
   # scale_fill_manual(values=all_colours[2])+
   # scale_color_manual(values=all_colours[2])+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
      #axis.text.x = element_blank(),
          legend.position="right"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          )
  
#   svg("~/Desktop/presenations/my_presentations/20190919_VUA/abundance_snps.png", width = 1900, height = 1200,res=300)
# # 
p3
```
The CRISPR do not seem to be lost over time
Let's see if mutations occured in the CRISPR spacers

#####CRISPRspacer mutations
first we have to call SNPs
```{bash}

###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt  ##the file with all sample names
logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_allRMKs202
Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/assemblyFinalwithIsolates.fasta

###--------special
FREEBAYES_out=freebayesOuput_WithONT_Parallel_default



###================
##script
###================

##--------------------------------------------------------------------------------------------------------------------------------------
##-----------change read group
##--------------------------------------------------------------------------------------------------------------------------------------

#name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 
names=Lyo_202_2014
###------------------adding readgroup to bam
rm ${BaseLocation}/log/multiqc_logfile_bams.txt
mkdir -p  ${BaseLocation}/log/
  #for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt)
  for names in $(cat ${samplesss})
do
echo ${names}

java -jar /home/vincent/apps/picard/picard.jar AddOrReplaceReadGroups \
       I=${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam \
       O=${BaseLocation}/${names}/bwaMapping2DB/${names}_mapping2ref.bam \
       RGID=${names} \
       RGLB=lib1 \
       RGPL=illumina \
       RGPU=unit1 \
       RGSM=${names}

#rm /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam

echo ${BaseLocation}"/"${names}"/bwaMapping2DB/"${names}"_mapping2ref.bam " >> ${BaseLocation}/log/multiqc_logfile_bams.txt

done




##remove trailing whitespaces

 sed -i 's/[ \t]*$//' ${BaseLocation}/log/multiqc_logfile_bams.txt

##--------------------------------------------------------------------------------------------------------------------------------------
##-----------check bam mapping
##--------------------------------------------------------------------------------------------------------------------------------------

 for map in $(cat ${BaseLocation}/log/multiqc_logfile_bams.txt)
    do
    echo "----------------------------------------------------"
    echo $map
    samtools flagstat $map |grep "mapped ("
    
    done

##--------------------------------------------------------------------------------------------------------------------------------------
##-----------indexing file
##--------------------------------------------------------------------------------------------------------------------------------------
##need before freebayes parallel

  for names in $(cat ${BaseLocation}/log/multiqc_logfile_bams.txt)
do

samtools index ${names}

done

##--------------------------------------------------------------------------------------------------------------------------------------
##-----------SNP calling
##--------------------------------------------------------------------------------------------------------------------------------------

#rm -r ${BaseLocation}/freebayesOuput_WithONT_notParallel_default
#mkdir -p ${BaseLocation}/freebayesOuput_WithONT_notParallel_default


  #  freebayes -F 0.05 -C 3 --min-coverage 30 --pooled-discrete \
  #    --bam-list ${BaseLocation}/log/multiqc_logfile_bams.txt -f \
  #    ${Assembly} > \
  #   ${BaseLocation}/freebayesOuput_WithONT_notParallel_default/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf
  
##--------------------------------------------------------------------------------------------------------------------------------------
##-----------SNP calling parallel
##--------------------------------------------------------------------------------------------------------------------------------------


  rm -r ${BaseLocation}/${FREEBAYES_out}
  mkdir -p ${BaseLocation}/${FREEBAYES_out}
 
  
    freebayes-parallel <(fasta_generate_regions.py ${Assembly} 100000) 36 -f ${Assembly} -C 5 --pooled-continuous --min-alternate-fraction 0.05 --min-coverage 10 --bam-list ${BaseLocation}/log/multiqc_logfile_bams.txt > ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf
  
  
  grep "^#" -v ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf |cut -f 1 |sort|uniq -c
  
  
vcftools --vcf ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf --recode --recode-INFO-all --keep ${samplesss} --bed ${BaseLocation_bed}/CRISPRspacers/location/allSPACERS_sterm_location.bed --out ${BaseLocation_bed}/CRISPRspacers/location/allSPACERS_sterm_location_snps.vcf

grep "^##" -v /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacers/location/allSPACERS_sterm_location_snps.vcf.recode.vcf > ${BaseLocation_bed}/CRISPRspacers/location/allSPACERS_sterm_location_snps.vcf.recode_allGenomes.vcf

echo ${BaseLocation_bed}/CRISPRspacers/location/allSPACERS_sterm_location_snps.vcf.recode_allGenomes.vcf


scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacers/location/allSPACERS_sterm_location_snps.vcf.recode_allGenomes.vcf ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/
```


```{bash}



###================
##description file
###================
threads=37
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP
BaseLocation_bed=/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion
FREEBAYES_out=freebayesOuput_WithONT_Parallel_default
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt  ##the file with all sample names

###--------only CRISPR

sed 's/S_thermophilus_mag_rmk202/CP046134/g' ${BaseLocation_bed}/CRISPRspacers/location/allSPACERS_sterm_location.bed > ${BaseLocation_bed}/CRISPRspacers/location/allSPACERS_sterm_location_renamed.bed


vcftools --vcf ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf --recode --recode-INFO-all --keep ${samplesss} --bed ${BaseLocation_bed}/CRISPRspacers/location/allSPACERS_sterm_location_renamed.bed --out ${BaseLocation_bed}/CRISPRspacers/location/allSPACERS_sterm_location_snps.vcf


grep "^##" -v /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacers/location/allSPACERS_sterm_location_snps.vcf.recode.vcf > ${BaseLocation_bed}/CRISPRspacers/location/allSPACERS_sterm_location_snps.vcf.recode_onlyRef.vcf

echo ${BaseLocation_bed}/CRISPRspacers/location/allSPACERS_sterm_location_snps.vcf.recode_onlyRef.vcf

scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacers/location/allSPACERS_sterm_location_snps.vcf.recode_onlyRef.vcf ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/
```


```{r}

library(readr)
library(stringr)
library(tidyverse)
library(plyr)
library(dplyr)
# all_coverage_CRISPRspacers <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/allSPACERS_sterm_location_snps.vcf.recode_allGenomes.vcf", "\t", escape_double = FALSE, col_names = c("contig","start","end","name","numreads","bases","basescovered","percentCovered","sample"),  trim_ws = TRUE)


##==================
##01_import data
##==================
#snps_freebayes <- read_delim("/home/vincent/Desktop/Projects/2019_RMK202_analysis/06_snpCalling2ONT/variantCallsfreebayes_all_f_005_min_20x_prepforR.vcf", "\t", escape_double = FALSE, trim_ws = TRUE)
snps_freebayes <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/allSPACERS_sterm_location_snps.vcf.recode_allGenomes.vcf", "\t", escape_double = FALSE, trim_ws = TRUE)
# snps_freebayes <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/allSPACERS_sterm_location_snps.vcf.recode_onlyRef.vcf", "\t", escape_double = FALSE, trim_ws = TRUE)

##==================
##02_long list and frequency calc
##==================

table(snps_freebayes$`#CHROM`)
##wide2long

 # snps_freebayes[,10]
snps_freebayes_long <- gather(snps_freebayes, sample, Snps, colnames(snps_freebayes)[10]:colnames(snps_freebayes)[ncol(snps_freebayes)], factor_key=TRUE,na.rm = TRUE) 
nrow(snps_freebayes_long)
table(snps_freebayes_long$sample)
###------------------------remove non-called snps
snps_freebayes_long_cleaned <- snps_freebayes_long[snps_freebayes_long$Snps!=".",]
nrow(snps_freebayes_long_cleaned)
# snps_freebayes_long[c(7438,7437,7439, 7440, 7441, 7442, 7443, 7444, 7445),]

###------------------------calculate allel frequency   

sample_relative_temp <-snps_freebayes_long_cleaned %>% separate("Snps", into=c("GT","DP","AD","RO","QR","AO","QA","GL"), sep = ":") %>% transform(.,allel_frequency=as.numeric(AO)/as.numeric(DP))

sample_relative_temp_withONT <- sample_relative_temp
sample_relative_temp_withONT$site <- paste(sample_relative_temp_withONT$X.CHROM,sample_relative_temp_withONT$POS,sample_relative_temp_withONT$REF,sample_relative_temp_withONT$ALT,sep="_")


##==================
##03_filter
##==================

##remove SNPs that have more than two allel frequencies
nrow(sample_relative_temp)
nas <- sample_relative_temp[which(is.na(sample_relative_temp$allel_frequency)),]
nrow(nas)
sample_relative_temp_cleaned <- sample_relative_temp[which(!is.na(sample_relative_temp$allel_frequency)),]

##remove low coverage (>30x)
cutoff <- 20
sample_relative_temp_cleaned$DP <- as.numeric(sample_relative_temp_cleaned$DP)
nrow(sample_relative_temp_cleaned)
sample_relative_safe <- sample_relative_temp_cleaned[which(sample_relative_temp_cleaned$DP>cutoff),]
nrow(sample_relative_safe)

##make all allele frequencies smaller than 0.05 to NA
# sample_relative_safe[sample_relative_safe$allel_frequency<0.05,"allel_frequency"] <- NA

#is.na(sample_relative_safe$allel_frequency)

##remove snps with less than 2x alternative allel frequency or zero
sample_relative_safe$AO <- as.numeric(sample_relative_safe$AO)
# nrow(sample_relative_safe[sample_relative_safe$AO<=2,])
# nrow(sample_relative_safe[sample_relative_safe$AO==0 | sample_relative_safe$AO>2,])
nrow(sample_relative_safe[sample_relative_safe$AO>0 & sample_relative_safe$AO<=2,])
sample_relative_safe <- (sample_relative_safe[sample_relative_safe$AO==0 | sample_relative_safe$AO>2,])
nrow(sample_relative_safe)
table(sample_relative_safe$sample)

##==================
##04_preperation and make wide
##==================

sample_relative_temp_cleaned <- sample_relative_safe
#ggplot(sample_relative_temp_cleaned, aes(y=DP))+geom_boxplot()+ylim(c(0,1000))


##make site name
sample_relative_temp_cleaned$site <- paste(sample_relative_temp_cleaned$X.CHROM,sample_relative_temp_cleaned$POS,sample_relative_temp_cleaned$REF,sample_relative_temp_cleaned$ALT,sep="_")


###spread the dataframe again

nrow(sample_relative_temp_cleaned)

sample_relative_safe_final_prep <- sample_relative_temp_cleaned[,c("X.CHROM","POS","site","allel_frequency","sample")]
#nrow(sample_relative_safe_final_prep[grep("S_thermophilus_RMK202",sample_relative_safe_final_prep$site),])
sample_relative_safe_final_tsne <- spread(sample_relative_safe_final_prep, sample, allel_frequency)
nrow(sample_relative_safe_final_tsne)

##==================
##05_filter rowsums!=0
##==================

##remove row sums ==0
nrow(sample_relative_safe_final_tsne)
sum(rowSums(sample_relative_safe_final_tsne[,4:ncol(sample_relative_safe_final_tsne)],na.rm = TRUE)==0)


sample_relative_wide <- sample_relative_safe_final_tsne[rowSums(sample_relative_safe_final_tsne[,4:ncol(sample_relative_safe_final_tsne)],na.rm = TRUE)>0,]
# nrow(sample_relative_wide)
sample_relative_wide <- subset(sample_relative_wide, select=-POS)
sample_relative_wide$species <- sample_relative_wide$X.CHROM
# sample_relative_wide$species <- revalue(sample_relative_wide$species, c("CP046131"="L. delbrueckii","CP046134"="S. thermophilus"))
sample_relative_wide$culture <- "RMK202"

sample_relative_wide <- sample_relative_wide %>%replace(is.na(.), 0)
snps_freebayes_long_CRISPR <- gather(sample_relative_wide, sample, Snps, c("lyo202_96" ,   "Versand_202" , "Konserve_202", "th_K2_8h"    , "di_K2_6h"  ,   "Lyo_202_2014", "Lyo_202_2012" ,"RMK202"), factor_key=TRUE,na.rm = TRUE) 


snps_freebayes_long_CRISPR$sample = factor(snps_freebayes_long_CRISPR$sample, levels=c("lyo202_96" ,    "Lyo_202_2012",  "Lyo_202_2014" ,"RMK202", "Konserve_202",  "Versand_202","th_K2_8h"    , "di_K2_6h"  ))
# snps_freebayes_long_CRISPR$sample = factor(snps_freebayes_long_CRISPR$sample, levels=c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2"))
levels(snps_freebayes_long_CRISPR$sample)



  ##============
### plot 
   ##============

# all_colours <- rev(c("#EB4D4D","#10B552") ) 

# colnames(sample_relative_safe_woSTRAINS)

# snps_freebayes_long_CRISPR <- snps_freebayes_long_CRISPR %>% filter(.,Snps>0.05 & mst2 <=0.9)


##plot
  p2 <- ggplot(snps_freebayes_long_CRISPR,aes(x=sample,y=Snps,group=site))+ geom_line(size=1, alpha=.5)+ 
    # facet_grid(day~kessel~species)+
    # facet_grid(species~.)+
    # facet_grid(treatment~species)+
    # facet_grid(treatment~species)+
    labs("",
         x=" ",
         y="Alternative allele frequency")+
    #scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    #scale_y_continuous(limits = c(0,0.3))+
    theme_classic()+
   # scale_fill_manual(values=all_colours)+
   # scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
      #axis.text.x = element_blank(),
          legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          )
  
  p2

```

##Additional analysis
Here, I do some very interesting addtional analysis

####Nucmer phage

Here, I use [nucmer](http://mummer.sourceforge.net/) to do the analysis of the largest repeats

```{bash}
###================
##description file
###================
#samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt  ##the file with all sample names
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples_withEvolution.txt


threads=37
logFilelocation=/archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/maskedReferenceMapping//01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/repeating_regions/
Assembly=/archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/maskedReferenceMapping/CRISPRmasked_RMK202_MAG_assembly.fasta
names=G1_6_18

###==========================
##get genomes
###==========================
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/repeating_regions/genomes

samtools faidx ${Assembly}  CP046134 > /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/repeating_regions/genomes/Sterm_genome_CRISPRmased.fasta
samtools faidx ${Assembly}  Streptococcus_phage_1 > /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/repeating_regions/genomes/Sterm_phage1.fasta
samtools faidx ${Assembly}  Streptococcus_phage_2 > /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/repeating_regions/genomes/Sterm_phage2.fasta

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/repeating_regions/genomes/Sterm_genome_CRISPRmased.fasta \
  /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/repeating_regions/genomes/Sterm_phage1.fasta > \
  /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/repeating_regions/genomes/Sterm_phage1_withGenome.fasta

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/repeating_regions/genomes/Sterm_genome_CRISPRmased.fasta \
  /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/repeating_regions/genomes/Sterm_phage2.fasta > \
  /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/repeating_regions/genomes/Sterm_phage2_withGenome.fasta
  
cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all/*.fna > /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/repeating_regions/genomes/all_Sterm_genomes.fasta
  
  
  
###==========================
##nucmer
###==========================
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/repeating_regions/{nucmer_phage1,nucmer_phage2}

cd /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/repeating_regions/nucmer_phage1

nucmer -b 100 -c 50 -maxmatch -nosimplify /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/repeating_regions/genomes/all_Sterm_genomes.fasta /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/repeating_regions/genomes/Sterm_phage1.fasta
cat out.delta

cd /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/repeating_regions/nucmer_phage2

nucmer -maxmatch -nosimplify /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/repeating_regions/genomes/all_Sterm_genomes.fasta /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/repeating_regions/genomes/Sterm_phage2.fasta

cat out.delta


###==========================
##delta filter
###==========================
cd /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/repeating_regions/nucmer_phage1

#mkdir -p $DATA_ROOT/11_nucmer_repeat/cottens_abruijn_delbruecki_streptococcus/{fasta,nucmer,deltaFiltered}
delta-filter -i 99 -l 5000  out.delta > phage1_filtered.txt
##split files
split -t ">" -l 1 phage1_filtered.txt


cd /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/repeating_regions/nucmer_phage2

#mkdir -p $DATA_ROOT/11_nucmer_repeat/cottens_abruijn_delbruecki_streptococcus/{fasta,nucmer,deltaFiltered}
delta-filter -i 99 -l 5000  out.delta > phage2_filtered.txt
##split files
split -t ">" -l 1 phage2_filtered.txt

##--------------
##extract Ldel repeats
##--------------
grep "[[:space:]]" xab | awk 'BEGIN {OFS ="\t"} {print "L_delbrueckii_RMK202" , $1 , $2 , "L_delbrueckii_RMK202" , $3 , $4}' > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/links_ldel.txt
cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/links_ldel.txt > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/links_ldel_curated.txt
##remove first row and reduant repeats

awk '{print $3-$2}' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/links_ldel_curated.txt |sort


```
It seems like they have no repeats so all mapping must be true

## Identify active Prophage reads

I try to identify active prophage reads. I want to do this by looking for paired end reads that the mates map to different contigs. 
The necessary columns in the Sam-file are the following:

$7 RNEXT String \*|=|[:rname:∧ *=][:rname:]* Reference name of the mate/next read
$8 PNEXT Int [0, 231 − 1] Position of the mate/next read

7. RNEXT: Reference sequence name of the primary alignment of the NEXT read in the template. For
the last read, the next read is the first read in the template. If @SQ header lines are present, RNEXT (if
not ‘*’ or ‘=’) must be present in one of the SQ-SN tag. This field is set as ‘*’ when the information is
unavailable, and set as ‘=’ if RNEXT is identical RNAME. If not ‘=’ and the next read in the template
has one primary mapping (see also bit 0x100 in FLAG), this field is identical to RNAME at the primary
line of the next read. If RNEXT is ‘*’, no assumptions can be made on PNEXT and bit 0x20.

8. PNEXT: 1-based Position of the primary alignment of the NEXT read in the template. Set as 0 when
the information is unavailable. This field equals POS at the primary line of the next read. If PNEXT
is 0, no assumptions can be made on RNEXT and bit 0x20.

Obviously, I need to first etract all reads that map to the phages. 

```{bash}




###================
##description file
###================
#samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt  ##the file with all sample names
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples_withEvolution.txt


threads=37
logFilelocation=/archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/maskedReferenceMapping//01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/maskedReferenceMapping/
Assembly=/archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/maskedReferenceMapping/CRISPRmasked_RMK202_MAG_assembly.fasta
names=G4_6_18

##--------------------------------------------------------------------------------------------------------------------------------------
##-----------make complete reference with CRISPR masked genomes
##--------------------------------------------------------------------------------------------------------------------------------------

rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/maskedReferenceMapping/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/maskedReferenceMapping/
sed 's/202-SMAG-1/CP046134/g' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/Sterm/maskedGenomes/202-SMAG_CRISPRmasked_extended.fna > /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/maskedReferenceMapping/CRISPRmasked_RMK202_MAG_assembly.fasta
sed 's/202-LMAG-1/CP046131/g' /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/Ldel/renamedContigs/StartAligned/StartAligned//202-LMAG_CRISPRmasked_extended.fna >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/maskedReferenceMapping/CRISPRmasked_RMK202_MAG_assembly.fasta


for remainsss in $(grep ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta |sed 's/>//g' |grep "CP046134" -v |grep -v "CP046131")
do
samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta ${remainsss} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/maskedReferenceMapping/CRISPRmasked_RMK202_MAG_assembly.fasta

done

###================
##description file
###================

~/apps/PilerCR/pilercr1.06/pilercr -in ${Assembly} \
  -out ${Assembly}_pilarTest -noinfo\
  -seq ${Assembly}_pilarTest.fasta

##============
##mapping to reference
##============
bwa index $Assembly
# name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 

mkdir -p ${logFilelocation}

num=1

for names in $(cut -f 1 ${samplesss})
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))
  rm -r ${BaseLocation}/${names}

  mkdir -p ${BaseLocation}/${names}/bwaMapping2DB/
  
  
 # bwa mem -t ${threads} ${Assembly} \
#    /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_1.fastq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_2.fastq | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 

##-------------special for G4
names=G4_6_18
  bwa mem -t ${threads} ${Assembly} \
     /home/vincent/Projects/2020_StarterEvolution/01_data/20200929_Novogene/X204SC20090774-Z01-F001/trimm_Galore/gz/${names}/${names}-R1_val_1.fq /home/vincent/Projects/2020_StarterEvolution/01_data/20200929_Novogene/X204SC20090774-Z01-F001/trimm_Galore/gz/${names}/${names}-R2_val_2.fq | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 




samtools view ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam |awk -F "\t" '{OFS="\t"}{if($7!="=")print $0}' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/${names}_reads_2_mappings.sam


samtools view -H ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/all_reads_2_mappings.sam
rm ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam &
   done 

mv /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/all_reads_2_mappings.sam /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/all_reads_2_mappings_all.sam

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/*_reads_2_mappings.sam >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/all_reads_2_mappings_all.sam

samtools sort /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/all_reads_2_mappings_all.sam |samtools view -S -b - > /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/all_reads_2_mappings_all.bam 

##--------------------------------------------------------------------------------------------------------------------------------------
##-----------extract reads that map to two different contigs
##--------------------------------------------------------------------------------------------------------------------------------------



mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/
  #for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt)
  for names in $(cut -f 1 ${samplesss})
do
echo ${names}


#samtools view ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam |awk -F "\t" '{OFS="\t"}{if($7!="=")print $0}' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/${names}_reads_2_mappings.sam


done



##--------------------------------------------------------------------------------------------------------------------------------------
##-----------extract reads that map to two different contigs
##--------------------------------------------------------------------------------------------------------------------------------------



mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/
  #for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt)
  for names in $(cut -f 1 ${samplesss} )
do
echo ${names}
echo "-----------------"
echo "number of phage reads:"
grep -c "phage" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/${names}_reads_2_mappings.sam
echo "S. phage and lactobacillus genome reads:"
grep "Streptococcus_phage" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/${names}_reads_2_mappings.sam |grep -c "CP046131"
echo "S. phage and Streptococcus genome reads:"
grep "Streptococcus_phage" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/${names}_reads_2_mappings.sam |grep -c "CP046134"
echo "L. phage and lactobacillus genome reads:"
grep "Lactobacillus_phage" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/${names}_reads_2_mappings.sam |grep -c "CP046131"
echo "L. phageand Streptococcus genome reads:"
grep "Lactobacillus_phage" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/${names}_reads_2_mappings.sam |grep -c "CP046134"


#grep  "phage" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/${names}_reads_2_mappings.sam |cut -f 7 |sort |uniq -c
#grep  "phage" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/${names}_reads_2_mappings.sam |cut -f 3 |sort |uniq -c


echo "=================================================================="
done



##--------------------------------------------------------------------------------------------------------------------------------------
##-----------creat circos file for mapping 
##--------------------------------------------------------------------------------------------------------------------------------------


rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/
  #for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt)
  for names in $(cut -f 1 ${samplesss} )
do
echo ${names}
echo "-----------------"


awk -F "\t" -v samplesss="$names" '{OFS="\t"}{if($3~"_phage_")print $3,$4,$4+length($10),$7,$8,$8+length($10),samplesss}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/${names}_reads_2_mappings.sam >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/allGenomes_mapping_2_genomes.txt


awk -F "\t" -v samplesss="$names" '{OFS="\t"}{if($7~"_phage_")print $7,$8,$8+length($10),$3,$4,$4+length($10),samplesss}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/${names}_reads_2_mappings.sam >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/allGenomes_mapping_2_genomes.txt



echo "=================================================================="
done
wc -l /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/allGenomes_mapping_2_genomes.txt



awk -F "\t" '{OFS="\t"}{if($4!~"_phage_")print $0}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/allGenomes_mapping_2_genomes.txt | awk -F "\t" '{OFS="\t"}{if($7!="")print $0}'> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/allGenomes_mapping_2_genomes_cleaned.txt
wc -l /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/allGenomes_mapping_2_genomes_cleaned.txt

cut -f 7 /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/allGenomes_mapping_2_genomes.txt|sort|uniq -c
cut -f 7 /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/allGenomes_mapping_2_genomes_cleaned.txt|sort|uniq -c



###--------------------------make bedfile for coverage

for names in $(cut -f 1 ${samplesss} )
do
echo ${names}
echo "-----------------"


awk -F "\t" -v samplesss="$names" '{OFS="\t"}{if($3~"_phage_")print $3,$4,$4+length($10),samplesss"\n"$7,$8,$8+length($10),samplesss}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/${names}_reads_2_mappings.sam  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/allGenomes_mapping_2_genomes.bed


awk -F "\t" -v samplesss="$names" '{OFS="\t"}{if($7~"_phage_")print $7,$8,$8+length($10),$3,$4,$4+length($10),samplesss}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/${names}_reads_2_mappings.sam >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/allGenomes_mapping_2_genomes.txt



echo "=================================================================="
done

seq_length.py ${Assembly} |cut -f 1,3 |sort > /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/genome.bed
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/*llGenomes_mapping_2_genomes_cleaned_*_coverage.bed

for names in $(cut -f 1 ${samplesss} )
do
echo ${names}
echo "-----------------"


awk  -F "\t" -v samplezzz="$names" '{OFS="\t"}{if($1~"Streptococcus_phage"&& $7==samplezzz) print $1,$2,$3,$7"\n"$4,$5,$6,$7}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/allGenomes_mapping_2_genomes_cleaned.txt |bedtools sort > /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/allGenomes_mapping_2_genomes_cleaned_StermPhages.bed

awk  -F "\t"  -v samplezzz="$names"  '{OFS="\t"}{if($1~"Streptococcus_phage"&& $7==samplezzz) print "Streptococcus_phage_1",$2,$3,$7"\n"$4,$5,$6,$7}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/allGenomes_mapping_2_genomes_cleaned.txt |bedtools sort > /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/allGenomes_mapping_2_genomes_cleaned_StermPhages_bothtoegther.bed


awk  -F "\t"  -v samplezzz="$names" '{OFS="\t"}{if($1=="Lactobacillus_phage_1"&& $7==samplezzz) print $1,$2,$3,$7"\n"$4,$5,$6,$7}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/allGenomes_mapping_2_genomes_cleaned.txt |bedtools sort > /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/allGenomes_mapping_2_genomes_cleaned_LdelPhages.bed

##----coverage

bedtools genomecov -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/allGenomes_mapping_2_genomes_cleaned_StermPhages.bed  -d -g /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/genome.bed  |awk  -v samplezzz="$names"  -F "\t" '{OFS="\t"}{if($3!=0) print $0,samplezzz}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/allGenomes_mapping_2_genomes_cleaned_StermPhages_coverage.bed

bedtools genomecov -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/allGenomes_mapping_2_genomes_cleaned_StermPhages_bothtoegther.bed  -d -g /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/genome.bed  |awk  -v samplezzz="$names"  -F "\t" '{OFS="\t"}{if($3!=0) print $0,samplezzz}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/allGenomes_mapping_2_genomes_cleaned_StermPhages_bothtoegther_coverage.bed

bedtools genomecov -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/allGenomes_mapping_2_genomes_cleaned_LdelPhages.bed  -d -g /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/genome.bed  |awk -v samplezzz="$names" -F "\t" '{OFS="\t"}{if($3!=0) print $0,samplezzz}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/allGenomes_mapping_2_genomes_cleaned_LdelPhages_coverage.bed

echo "=================================================================="
done




```

move local

```{bash}

mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/
scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/allGenomes_mapping_2_genome*.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/

scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/all_reads_2_mappings_all.bam  ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/

scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/maskedReferenceMapping/CRISPRmasked_RMK202_MAG_assembly.fasta ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/


mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/
scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/allGenomes_mapping_2_*.bed ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/


```

```{r}
library(readr)
library(tidyverse)
allGenomes_mapping_2_genomes <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/allGenomes_mapping_2_genomes_cleaned.txt", "\t", escape_double = FALSE, col_names = c("phageGenome","phageStart","phageEnd","SecondGenome","SecondStart","SecondEnd","sample"),trim_ws = TRUE) 


ggplot(allGenomes_mapping_2_genomes,aes(x=SecondStart,fill=phageGenome,color=phageGenome))+geom_histogram(alpha=0.5)+theme_classic()+facet_wrap(~SecondGenome+sample,scales = "free")

ggplot(allGenomes_mapping_2_genomes,aes(x=phageStart,fill=SecondGenome,color=SecondGenome))+geom_density(alpha=0.5)+theme_classic()+facet_wrap(~phageGenome,scales = "free")


allGenomes_mapping_2_genomes$SecondStart

ggplot(allGenomes_mapping_2_genomes,aes(x=SecondStart,fill=phageGenome,color=phageGenome))+geom_density(alpha=0.5)+theme_classic()+facet_wrap(~SecondGenome+phageGenome,scales = "free")


ggplot(allGenomes_mapping_2_genomes,aes(x=SecondStart,fill=phageGenome,color=phageGenome))+geom_histogram(alpha=0.5)+theme_classic()+facet_grid(vars(SecondGenome),vars(phageGenome),scales = "free")


ggplot(allGenomes_mapping_2_genomes,aes(x=phageStart,fill=phageGenome,color=phageGenome))+geom_histogram(alpha=0.5)+theme_classic()+facet_grid(vars(SecondGenome),vars(phageGenome),scales = "free")

##------------------------
###only streptococci
##------------------------
table(allGenomes_mapping_2_genomes$SecondGenome)
allGenomes_mapping_2_genomes_sterm <- allGenomes_mapping_2_genomes %>% filter(phageGenome %in% c("Streptococcus_phage_1","Streptococcus_phage_2")) %>% filter(SecondGenome=="CP046134")

ggplot(allGenomes_mapping_2_genomes_sterm,aes(x=phageStart,fill=phageGenome,color=phageGenome))+geom_histogram(alpha=0.5)+theme_classic()+facet_grid(vars(SecondGenome),vars(phageGenome),scales = "free")


ggplot(allGenomes_mapping_2_genomes_sterm,aes(x=SecondStart,fill=phageGenome,color=phageGenome))+geom_histogram(alpha=0.5)+theme_classic()+facet_grid(vars(SecondGenome),vars(phageGenome),scales = "free")


ggplot(allGenomes_mapping_2_genomes_sterm,aes(x=SecondStart,y=phageStart,fill=phageGenome,color=phageGenome))+geom_point(alpha=0.5)+theme_classic()

integrationSite <- ggplot(allGenomes_mapping_2_genomes_sterm,aes(x=SecondStart,y=phageStart))+stat_density_2d(aes(fill = ..level..), geom = "polygon")+theme_classic()+facet_grid(vars(phageGenome),scales = "free")+labs(x="location of mate mapping on S. thermophilus genome",y="location of mate mapping on phage genome")+theme(legend.position = "none")
integrationSite

integrationSite <- ggplot(allGenomes_mapping_2_genomes_sterm,aes(x=SecondStart,y=phageStart))+stat_density_2d(aes(fill = ..level..), geom = "polygon")+theme_classic()+labs(x="location of phage integration\non S. thermophilus genome",y="location of phage integration\non phage genome")+theme(legend.position = "right")#+geom_hline(yintercept=c(7500,29000,35000))
integrationSite

###=====================================
   ##number and percent of read supporting the integration 
###=====================================
   
##---------------------------------newappraoch with coverage

allGenomes_mapping_2_genomes_coverage <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/mappingLocation/allGenomes_mapping_2_genomes_cleaned_StermPhages_bothtoegther_coverage.bed", "\t", escape_double = FALSE, col_names = c("Genome","position","coverage","sample"),trim_ws = TRUE) 


# ggplot(allGenomes_mapping_2_genomes_coverage,aes(x=position,y=coverage,color=sample))+geom_bar(stat="identity",alpha=0.5)+theme_classic()+facet_wrap(~Genome+sample,scales = "free")
   

summary_mapping <- allGenomes_mapping_2_genomes_coverage %>%  group_by(interaction(sample,Genome)) %>% 
  dplyr::summarise(max=max(coverage)) 
 
summary_mapping$sample <- str_split_fixed(summary_mapping$`interaction(sample, Genome)`, fixed("."), 2)[,1]
summary_mapping$chr <- str_split_fixed(summary_mapping$`interaction(sample, Genome)`, fixed("."), 2)[,2]

summary_mapping <- summary_mapping %>% filter(chr=="CP046134") %>% select(c(-`interaction(sample, Genome)`,-"chr")) 

colnames(summary_mapping) <- plyr::revalue(colnames(summary_mapping), c("max"="phageCoverage"))

###-------------------------------coverage genome

  read_count <- read_delim("/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/07_abundances/mapping/all_2bacteria_and_phages_from_MAG.bed","\t", escape_double = FALSE, col_names = c("chr","start","end","count","length_mapped","geneLength","unknown","sample"),trim_ws = TRUE)

table(read_count$chr)

# read_count$species <- ifelse(read_count$chr=="L_del_phage_01","L_del_phage_01","S_term_phage_01")
  
  table(read_count$chr)
  
  read_count$geneCoverage  <- (read_count$count*600)/read_count$geneLength
  
  # ggplot(read_count,aes(y=geneCoverage,group=sort,color=sort,fill=sort))+geom_boxplot()+facet_grid(sample~species, scales="free")
  
  library(dplyr)
  
  
  all_final <- read_count %>% 
    group_by(sample,chr) %>% 
    dplyr::summarize(BacteriaCoverage = median(geneCoverage)) %>% filter(chr %in% c("CP046134","Streptococcus_phage_1","Streptococcus_phage_2")) %>% spread(., chr, BacteriaCoverage)
  

  
final_phage <- merge(all_final,summary_mapping,by="sample") %>% filter(!sample %in% c("di_K2_6h","th_K2_8h"))
   
final_phage$percent_Prophage <- 100*(final_phage$phageCoverage/final_phage$CP046134)
final_phage$abundance_phage1 <- 100*(final_phage$Streptococcus_phage_1/final_phage$CP046134)
final_phage$abundance_phage2 <- 100*(final_phage$Streptococcus_phage_2/final_phage$CP046134)

final_phage$sample <- plyr::revalue(final_phage$sample, c("lyo202_96"="Lyo 1996","Lyo_202_2012"="Lyo 2012", "Lyo_202_2014"="Lyo 2014", "Konserve_202"="working stock", "RMK202"="starter culture 2012", "Versand_202"="starter culture 2018", "di_K2_6h"="cheesemaking day1","th_K2_8h"="cheesemaking day2","G1_6_18"="experiment_A","G2_6_18"="experiment_B","G3_6_18"="experiment_C","G4_6_18"="experiment_D","G5_6_18"="experiment_E"))

final_phage$sample = factor(final_phage$sample, levels=c("Lyo 1996","Lyo 2012","Lyo 2014","working stock","starter culture 2012","starter culture 2018","cheesemaking day1","cheesemaking day2","experiment_A","experiment_B","experiment_C","experiment_D","experiment_E"))

NUMpLOT <- ggplot(final_phage,aes(x=sample,percent_Prophage))+geom_bar(stat="identity")+theme_classic()+lims(y=c(0,15))+labs(x="",y="Percent of S. thermophilus\nwith  prophage")+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9))+theme(axis.text.x = element_blank())
NUMpLOT

##----------


final_phage_long_phage <- final_phage %>% gather(., species, percent,c("abundance_phage1","abundance_phage2"), factor_key=TRUE,na.rm = TRUE) 
# Biolog_long <- gather(Biolog_all_plates, well, intensity, A01:H12, factor_key=TRUE,na.rm = TRUE) 
# final_phage$abundance_phage_both <- final_phage$abundance_phage2+final_phage$abundance_phage1

ggplot(final_phage_long_phage,aes(x=sample,y=percent,fill=species))+geom_bar(stat="identity")+theme_classic()+labs(x="",y="Percent of S. thermophilus MAG\nwith putative prophage")+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9))

##----------
##final plot
##----------
library(patchwork)
integrationSite+NUMpLOT +   plot_layout(nrow=2,heights = c(2, 1)) #

svg("~/Desktop/Projects/2019_RMK202_analysis/plot/IntegrationProphage.svg",width=6.5,height=5.5)
integrationSite+NUMpLOT +   plot_layout(nrow=2,heights = c(2, 1)) #
   dev.off()
```



#####PAM identification

Here, I try to identify the PAMs by taking 10bp up and dwonstream and looking at the nucleotide conservation at all positions with weblogo.

This [paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3890593/) reviews the two S. thermophilus IIA CRISPRs to have the following PAMs (5′–3′):
1. NNAAGAW
2. W
3. NGGNG

```{bash}

###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt  ##the file with all sample names
#logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract/
#Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/FASTA/S_thermophilus_alltogether.CRISPRlocation.fasta
species=Sterm
repeatsFile=/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt
MetaspacerFile=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers/ForDADA2_woREverse//all_unique_spacers.fasta
mappingFile=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/allspacersClustered_renamed_ClusterName.txt
  #BaseLocation_mapping=/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion_wOLDStrains/bwaMapping
  BaseLocation_mapping=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//bwaMapping
metaSpacerFile=${BaseLocation}/FinalSpacers//FinalSpacers/cdhit_withReference_oldStrain/uniqueSpacers_count_Ref_wOLDstrains.txt #inlcudes in column four the name of array

###---------------------------
##split 
##----------------------------

awk -F "\t" '{OFS="\t"}{if($6=="a1")print $0}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/log/Clusters_spacers_final.txt > /archiv/Projects/2019_RMK202_analysis/00_FINAL/log/Clusters_spacers_final_a1.txt

awk -F "\t" '{OFS="\t"}{if($6=="a2")print $0}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/log/Clusters_spacers_final.txt > /archiv/Projects/2019_RMK202_analysis/00_FINAL/log/Clusters_spacers_final_a2.txt

awk -F "\t" '{OFS="\t"}{if($6=="a3")print $0}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/log/Clusters_spacers_final.txt > /archiv/Projects/2019_RMK202_analysis/00_FINAL/log/Clusters_spacers_final_a3.txt


###---------------------------
##split 
##----------------------------
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/PAM_identifaction/weblogo/all/

for arrayzzz in $(echo "a1 a2 a3")
do
echo "=========================="
echo ${arrayzzz}
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/PAM_identifaction/extract/${arrayzzz}/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/PAM_identifaction/extract/${arrayzzz}/
for spacersss in $(cut -f 3 /archiv/Projects/2019_RMK202_analysis/00_FINAL/log/Clusters_spacers_final_${arrayzzz}.txt)
do

#cut -f 3 /archiv/Projects/2019_RMK202_analysis/00_FINAL/log/Clusters_spacers_final_${arrayzzz}.txt| while read genenomes 
#do

grep -e "${spacersss}$(printf '\t')" ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages.txt |awk -F "\t" '{OFS="\t"}{if(sqrt(($10-$9)^2)>27) print $0}' |grep "Streptococcus_phage_" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/log/tmp.txt


#contig=$(cut -f 2 /archiv/Projects/2019_RMK202_analysis/00_FINAL/log/tmp.txt)
#start=$(cut -f 9 /archiv/Projects/2019_RMK202_analysis/00_FINAL/log/tmp.txt)
#end=$(cut -f 10 /archiv/Projects/2019_RMK202_analysis/00_FINAL/log/tmp.txt)

#echo -e ${contig}"\t"${start}"\t"${end}

awk -F "\t"  '{OFS="\t"}{print $2,$9,$10,$1}'  /archiv/Projects/2019_RMK202_analysis/00_FINAL/log/tmp.txt

#awk -F "\t" -v contigsss="$contig" -v startss="$start" -v endss="$end" -v spacer="$spacersss" '{OFS="\t"}{if(startss>endss)print contigsss}'
##-------------------reverse start
awk -F "\t"  '{OFS="\t"}{if($9>$10)print $2,$9,$9+10,"reverseSTART;"$1}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/log/tmp.txt >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/PAM_identifaction/extract/${arrayzzz}/${arrayzzz}_reverseSTART.bed
##-------------------reverse end
awk -F "\t"  '{OFS="\t"}{if($9>$10)print $2,$10-10,$10,"reverseEND;"$1}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/log/tmp.txt >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/PAM_identifaction/extract/${arrayzzz}/${arrayzzz}_reverseEND.bed
#-------------------normal start
awk -F "\t"  '{OFS="\t"}{if($9<$10)print $2,$9-10,$9,"normalSTART;"$1}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/log/tmp.txt >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/PAM_identifaction/extract/${arrayzzz}/${arrayzzz}_normalSTART.bed
##-------------------normal end
awk -F "\t"  '{OFS="\t"}{if($9<$10)print $2,$10,$10+10,"normalEND;"$1}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/log/tmp.txt >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/PAM_identifaction/extract/${arrayzzz}/${arrayzzz}_normalEND.bed

done #spacersss 

for aryz in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/PAM_identifaction/extract/${arrayzzz}/ |grep ".bed" |sed 's/.bed//g')
do



bedtools getfasta -s -name -fi ${BaseLocation}/CRISPRspacerBLAST/genomes/phagenGenome.fasta \
  -bed /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/PAM_identifaction/extract/${arrayzzz}/${aryz}.bed \
  -fo /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/PAM_identifaction/extract/${arrayzzz}/${aryz}.fasta

rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/PAM_identifaction/weblogo/${arrayzzz}/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/PAM_identifaction/weblogo/${arrayzzz}/
  
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/PAM_identifaction/weblogo/all/
  
clustalw -INFILE=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/PAM_identifaction/extract/${arrayzzz}/${aryz}.fasta -ALIGN -OUTFILE=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/PAM_identifaction/weblogo/${arrayzzz}/${aryz}
/home/vincent/apps/weblogo/seqlogo -f /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/PAM_identifaction/weblogo/${arrayzzz}/${aryz} -F EPS -k 1 -o /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/PAM_identifaction/weblogo/all/${aryz}_weblogo

done

done #arrayzzz
```

```{bash}

 
 mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/PAM_identifaction/weblogo/
  scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/PAM_identifaction/weblogo/all ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/PAM_identifaction/weblogo/

```

#####Protospacer annotation
Here, I look in more depth into the phages that the CRISPR spacers where extracted from.
```{bash}



###-----------------
##-get genome annotation of these phages
###-----------------


rm -r /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/
mkdir -p /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/

cut -f 13 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_viralREF.txt|sort|uniq | sed 's/bacteriophage //g' |awk -F "[ ,]" '{print $3}' > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/phageNames.txt
for phage in $(cat /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/phageNames.txt)
  do
  echo "${phage}"
  grep -P "${phage}\t" /archiv/BLAST/blast_db/20190204/viral_genomic_file.txt |grep "Streptococc" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/CRISPRblast_viralREF_refseq.txt
  
  grep -P "${phage}\t" /archiv/BLAST/blast_db/20190204/viral_genomic_file.txt |grep "Streptococc"  |cut -f 8 |cut -d ' ' -f 3
  echo "--------"
  done
  
  ##-----some didnt work---> manual work
  
  grep -P " phiL47" /archiv/BLAST/blast_db/20190204/viral_genomic_file.txt  >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/CRISPRblast_viralREF_refseq.txt
  
  grep -P "ul36" /archiv/BLAST/blast_db/20190204/viral_genomic_file.txt  >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/CRISPRblast_viralREF_refseq.txt
  
  grep -P "ALQ132" /archiv/BLAST/blast_db/20190204/viral_genomic_file.txt  >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/CRISPRblast_viralREF_refseq.txt

  wc -l /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/phageNames.txt
  wc -l /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/CRISPRblast_viralREF_refseq.txt
 
 cut -f 13 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_viralREF.txt|sort|uniq | sed 's/bacteriophage //g' |awk -F "[ ,]" '{print $3}' > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/phageNames.txt
 
 #----------bacteria
 
#curl 'ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt' > /Desktop/bacteria_genomic_file.txt
#cut -f 13 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_unmapped2prokaryoteREF.txt|sort|uniq| sed 's/^.* strain//g' |cut -d ',' -f 1 > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/bacteriaNames.txt
 
 rm /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/CRISPRblast_bacteriaREF_refseq.txt

 grep -P "UC109" /archiv/BLAST/blast_db/20190204/bacteria_genomic_file.txt  >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/CRISPRblast_bacteriaREF_refseq.txt
  
  grep -P "UC06\t" /archiv/BLAST/blast_db/20190204/bacteria_genomic_file.txt  >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/CRISPRblast_bacteriaREF_refseq.txt
  
 grep -P "MN-BM-A01" /archiv/BLAST/blast_db/20190204/bacteria_genomic_file.txt  >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/CRISPRblast_bacteriaREF_refseq.txt
 
 grep -P "N4L" /archiv/BLAST/blast_db/20190204/bacteria_genomic_file.txt  >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/CRISPRblast_bacteriaREF_refseq.txt
 
 grep -P "NCTC12958" /archiv/BLAST/blast_db/20190204/bacteria_genomic_file.txt  >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/CRISPRblast_bacteriaREF_refseq.txt

 cat /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/CRISPRblast_bacteriaREF_refseq.txt \
 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/CRISPRblast_viralREF_refseq.txt > \
 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/CRISPRblast_all_refseq.txt
###-----------------
##-download gff file
###-----------------
rm -r /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/downloads
mkdir -p /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/downloads

#curl 'ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt' > /archiv/BLAST/blast_db/20190204/bacteria_genomic_file.txt
 awk 'BEGIN{FS="\t";OFS="\t"}  {print $8 , $20}' /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/CRISPRblast_all_refseq.txt | cut -f 2 | \
    sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gff.gz|' >    \
    /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/downloads/downloadFile.txt
    

mkdir -p /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/downloads/GFF
cd //archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/downloads/GFF
wget -i /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/downloads/downloadFile.txt 
gunzip *.gff.gz

rm /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/downloads/GFF/mergedAllMatching.gff

cat /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/downloads/GFF/*.gff > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/downloads/GFF/mergedAllMatching.gff

###-----------------
##-download faa file
###-----------------


 awk 'BEGIN{FS="\t";OFS="\t"}  {print $8 , $20}' /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/CRISPRblast_all_refseq.txt  | cut -f 2 | \
    sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >    \
    /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/downloads/downloadFile.txt
   

mkdir -p /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/downloads/FAA
cd /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/downloads/FAA
wget -i /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/downloads/downloadFile.txt 
gunzip *.faa.gz

cat /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/downloads/FAA/*faa > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/downloads/FAA/mergedAllMatching.faa

###---------------------------------------------- make bedfile for extracting genes

awk -F "\t" '{OFS="\t"} {print $2,$9,$10,$1}' /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_viralREF.txt | sed 's/|//g' | sed 's/|//g' |sed 's/ref//g' > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPR_blast_allREF.bed

awk -F "\t" '{OFS="\t"} {print $2,$9,$10,$1}' /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_unmapped2prokaryoteREF.txt | sed 's/|//g' | sed 's/|//g' |sed 's/ref//g' >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPR_blast_allREF.bed



 awk -F "\t" '{OFS="\t"} {
if ($2 > $3)
	print $1,$3,$2,$4 ;
else
	print $1,$2,$3,$4;
}' /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPR_blast_allREF.bed > \
/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPR_blast_allREF_final.bed

###---------------------------------------------- intersect protspacers with genes

bedtools intersect -a /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPR_blast_allREF_final.bed -b /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/downloads/GFF/mergedAllMatching.gff -wa -wb > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/intersected_genes_with_protospacers.gff

 awk -F "[\t;]" '{OFS="\t"} {
if ($7 == "CDS")
	print $4,$5,$13 ;}'  /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/intersected_genes_with_protospacers.gff |sed 's/ID=cds-//g' |sed 's/ //g' |sort|uniq | sed 's/ID=cds2357/WP_127084528.1/g' > \
/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/intersected_genes_with_protospacers_tmp1.gff


sed 's/^.*\(product=.*protein\).*$/\1/' /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/intersected_genes_with_protospacers_final.gff | sed 's/product=//g' |sed 's/;protein//g' > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/intersected_genes_with_protospacers_tmp2.gff

paste /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/intersected_genes_with_protospacers_tmp1.gff \
  /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/intersected_genes_with_protospacers_tmp2.gff > \
 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/intersected_genes_with_protospacers_final.gff


awk -F "[\t;]" '{OFS="\t"} {print $13}' /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/intersected_genes_with_protospacers_final.gff |sed 's/ID=cds-//g' |sed 's/ //g' |sort|uniq | sed 's/ID=cds2357/WP_127084528.1/g' >  /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/intersected_genes_with_protospacers_final_genes2query.txt

fasta_formatter -i /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/downloads/FAA/mergedAllMatching.faa -o /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/downloads/FAA/mergedAllMatching_endless.faa

rm /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/intersected_genes_with_protospacers_final_genes.faa
for id in $(cat /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/intersected_genes_with_protospacers_final_genes2query.txt)
  do
  echo $id
  grep "$id" -A 1 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/downloads/FAA/mergedAllMatching_endless.faa >> \
  /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/intersected_genes_with_protospacers_final_genes.faa
  
  done
  
  wc -l /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/intersected_genes_with_protospacers_final_genes2query.txt
  grep ">" -c /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/intersected_genes_with_protospacers_final_genes.faa


###-----------------
##blast to UNIPROT db
###-----------------
rm -r /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/UNIprot_protoGEnes
mkdir -p /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/UNIprot_protoGEnes
rm /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/UNIprot_protoGEnes/UNIPROT_DIAMOND_matching_CRISPRgenes.txt
diamond blastp --threads 32 --max-target-seqs 1 --db /archiv/TREMBL_database/20190710/uniprot_trembl.dmnd --query /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/intersected_genes_with_protospacers_final_genes.faa --out /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/UNIprot_protoGEnes/UNIPROT_DIAMOND_matching_CRISPRgenes.txt



 ##-------------------------transfer local
  scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/protospacer/intersected_genes_with_protospacers_final.gff ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/

```

```{r}

library(readr)
intersected_genes <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/intersected_genes_with_protospacers_final.gff", "\t", escape_double = FALSE, col_names = c("Name","mappingContig","geneName","geneProduct"),  trim_ws = TRUE); intersected_genes$Name <- paste(str_split_fixed(intersected_genes$Name, ":", 3)[,1],str_split_fixed(intersected_genes$Name, ":", 3)[,2],sep=":")

intersected_genes_prepped <- intersected_genes %>% select(Name,geneName,geneProduct)
CRISPRblast_all_extended <- merge(CRISPRblast_all,intersected_genes_prepped,by="Name",all.x = TRUE)

dim(CRISPRblast_all_extended)
dim(CRISPRblast_all)
###-------------------where do they map (intragenic)

CRISPRblast_all_extended <- CRISPRblast_all_extended %>% mutate(mapping =ifelse(mapping=="mapping", ifelse(is.na(geneName),"intergenic","genic"), "non-mapping"))

# CRISPRblast_all_extended$DB = factor(CRISPRblast_all_extended$DB, levels=c("BakteriaDB","PhageDB","rmk202Phages","non-mapping"))
colorsss <- (c("darkcyan","darkturquoise","lightgray"))

ggplot(CRISPRblast_all_extended, aes(x=mapping,fill=mapping))+geom_bar()+theme_classic()+labs(x="",y="Frequency",title = "CRISPR spacers mapping")+
   scale_fill_manual(values=colorsss)+
   scale_color_manual(values=colorsss)

###-------------------which genes (uniprotDB)




```

#Ldel CRISPR

####pilarCRISPR
call and merge CRISPR repeats
```{bash}

##-----------------
##call and merge pilar CR
##-----------------
  date=20190312
rm /data/Project/CRISPRspacers/02_PilerCR//assembledSpacers/Ldel_allRepeatsAssembled.fasta
rm /data/Project/CRISPRspacers/02_PilerCR//assembledSpacers/Ldel_allSpacersAssembled.fasta
mkdir -p /data/Project/CRISPRspacers/02_PilerCR//assembledSpacers/
for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/names_changed_isolates.txt |grep "^L")
  do
    echo ==========================
   echo ${id}
   echo ==========================
    cd /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/
    
    mkdir -p /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}
#~/apps/PilerCR/pilercr1.06/pilercr -in scaffolds.fasta \
#  -out /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}/${id}.pilarCR_out -noinfo\
#  -seq /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}/${id}.pilarCR_out.fasta

perl ~/apps/PilerCR/PilerCR_parser_new_vincent /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}/${id}.pilarCR_out > \
  /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}/${id}.pilarCR_out_final


cat /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}/${id}.pilarCR_out.fasta | sed "s/>/>${id}-/g" >> /data/Project/CRISPRspacers/02_PilerCR//assembledSpacers/Ldel_allRepeatsAssembled.fasta


  sed "s/>/>$names:/g" /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}/${id}.pilarCR_out_final > /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}/${id}.pilarCR_out_final
  cat /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}/${id}.pilarCR_out_final >> /data/Project/CRISPRspacers/02_PilerCR//assembledSpacers/Ldel_allSpacersAssembled.fasta

  done 
cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/L_delbrueckii_RMK202.pilarCR_out.fasta | sed "s/>/>rmk202_meta-/g" >> /data/Project/CRISPRspacers/02_PilerCR//assembledSpacers/Ldel_allRepeatsAssembled.fasta


##-------------------------
##dialact
##-------------------------


date=20190513
species=Lactobacillus_delbrueckii

rm -r /data/Project/CRISPRspacers/02_PilerCR/Dialact_Ldel/
mkdir -p /data/Project/CRISPRspacers/02_PilerCR/Dialact_Ldel/

cd /archiv/Dialact/ReferenceDatabase/20190513/Genomes/Merged/Lactobacillus_delbrueckii/
for genome in $(ls /archiv/Dialact/ReferenceDatabase/20190513/Genomes/Merged/Lactobacillus_delbrueckii/ |grep "L_del-la_" |cut -d '.' -f 1)
#for species in $(cat $mainPath/02_script/names.txt)
   do



~/apps/PilerCR/pilercr1.06/pilercr -in ${genome}.fna -out /data/Project/CRISPRspacers/02_PilerCR/Dialact_Ldel/$genome.pilarCR_out -noinfo\
  -seq /data/Project/CRISPRspacers/02_PilerCR/Dialact_Ldel/$genome.pilarCR_out.fasta


cat /data/Project/CRISPRspacers/02_PilerCR/Dialact_Ldel/$genome.pilarCR_out.fasta | sed "s/>/>${genome}-/g" >> /data/Project/CRISPRspacers/02_PilerCR//assembledSpacers/Ldel_allRepeatsAssembled.fasta


  done


```


####cd-hit of all repeats

```{bash}

mkdir -p  /data/Project/CRISPRspacers/02_PilerCR/cd_hit_ALl_ldel_repeats
cd-hit-est -i  /data/Project/CRISPRspacers/02_PilerCR//assembledSpacers/Ldel_allRepeatsAssembled.fasta \
    -o  //data/Project/CRISPRspacers/02_PilerCR/cd_hit_ALl_ldel_repeats/assembly_Clustering.txt -c 0.9
  
###take rmk_202_mag repeats

#cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/L_delbrueckii_RMK202.pilarCR_out.fasta > \
#/data/Project/CRISPRspacers/02_PilerCR/cd_hit_ALl_ldel_repeats/prototype_ldel_arrays.fasta

```

the two rmk_mag clusters are the common repeats


##### extrac rawRead spacers


as previously done for Sterm


```{bash}
#---test
###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt  ##the file with all sample names
#logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Ldel/
#Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/FASTA/S_thermophilus_alltogether.CRISPRlocation.fasta
species=Ldel
BaseLocation_bam=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP


revseq /data/Project/CRISPRspacers/02_PilerCR/cd_hit_ALl_ldel_repeats/prototype_ldel_arrays.fasta /data/Project/CRISPRspacers/02_PilerCR/cd_hit_ALl_ldel_repeats/prototype_ldel_arrays_reversed.fasta

###================
##script
###================
##test
pair=1
names=Konserve_202
adapName=$(grep ">" /data/Project/CRISPRspacers/02_PilerCR/cd_hit_ALl_ldel_repeats/prototype_ldel_arrays.fasta |head -1)
###----end

for names in $(cat ${samplesss})
  do
 
for pair in $(echo "1 2")
  do
#pair=1

read=/archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_${pair}.fastq

for adapName in $(grep ">" /data/Project/CRISPRspacers/02_PilerCR/cd_hit_ALl_ldel_repeats/prototype_ldel_arrays.fasta )
#for adapName in $(grep ">" /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt )
  do
adapSeq=$(grep "${adapName}" -A 1 /data/Project/CRISPRspacers/02_PilerCR/cd_hit_ALl_ldel_repeats/prototype_ldel_arrays.fasta|tail -1)
adap=$(echo ${adapName} | sed 's/>//g'|cut -d '_' -f 1)
cleanRound=1

echo "------------------------------------------------"
echo $adap
echo $adapSeq
echo $names
echo $read
echo $species
echo "------------------------------------------------"

mkdir -p ${BaseLocation}/${names}/ReadPair_${pair}//tmp/

cutadapt ${read} -b ${adapSeq}  -e  0.1 -O 15 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}.txt
rm ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}.txt &

## reads with match
awk '{if($2!="-1") print $0}' ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt

rm ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt & 

##make a list of current (potentially still containing 5' adapters)
awk -F "\t" '{OFS="\n"} {print "@"$1, $5,"+", $9}' ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt  > \
${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp.fastq


##trim 5' ends of these (with relaxed matching errors)
cutadapt ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp.fastq -g ${adapSeq}  -e  0.1 -O 5 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_.fastq.txt

rm ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_.fastq.txt &

## finalise first spacers (we will not be able to keep reads that don't have a small (>5bp) of repeat left)
##removing potential 5' adapters of first trim
mkdir -p ${BaseLocation}/${names}/final/
#awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($7)>"25" && length($7)<"46") print ">"$1, $7}' #${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_info.txt > \
#${BaseLocation}/${names}/ReadPair_${pair}/${names}_${adap}_final_round1_R${pair}.fasta


awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($7)>"25"  && length($7)<"46" && length($7)==length($11)) print "@"$1, $7 ,"+", $11}' ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}/${names}_${adap}_final_round1_R${pair}.fastq

cat ${BaseLocation}/${names}/ReadPair_${pair}/${names}_${adap}_final_round1_R${pair}.fastq > ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R${pair}.fastq


##take remaining reads and remove 3' adapters again --> do this 5 times
awk -F "\t" '{OFS="\n"} {print "@"$1, $7,"+", $11}' ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt  > \
${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round2_matchestmp_timmed5_readyForRound2.fastq

#grep --color "GTTTTAGAGCTGTGTTGTTTCGAATGGTTCCAAAAC" /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/readswithrepeats_${adap}_intraReads_info_trimmed_matches_readyForRound2.fastq  |head
cleanRound=2
for cleanRound in $(echo "2 3 4 5 6")
  do
  echo ${cleanRound}
  echo "------------------------------------"
cleanRound_next=$((cleanRound+1))
  
cutadapt ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_readyForRound${cleanRound}.fastq -b ${adapSeq}  -e  0.1 -O 5 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.fasta
rm ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.fasta &
##finalised spacers
awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($5)>"25"  && length($5)<"46"&& length($5)==length($9)) print "@"$1, $5 ,"+", $9}' ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}/${names}_${adap}_final_round${cleanRound}_R${pair}.fastq


cat ${BaseLocation}/${names}/ReadPair_${pair}/${names}_${adap}_final_round${cleanRound}_R${pair}.fastq >> ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R${pair}.fastq


##finalised prepered
awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($7)>"25" && length($7)<"46" && length($7)==length($11)) print "@"$1, $7,"+", $11}' ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound_next}_matchestmp_timmed5_readyForRound${cleanRound_next}.fastq

  done #cleanRound
  cat ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R1.fastq \
  ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R2.fastq > \
  ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads.fastq
  done #adap (spacer)
  
  ###-------------====
  ##reverse
    ###-------------====
#adapName=$(grep ">" /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut_reversed.txt| cut -d ' ' -f 1 |head -1)
for adapName in $(grep ">" /data/Project/CRISPRspacers/02_PilerCR/cd_hit_ALl_ldel_repeats/prototype_ldel_arrays_reversed.fasta| cut -d ' ' -f 1)
  do
adapSeq=$(grep "${adapName}" -A 1 /data/Project/CRISPRspacers/02_PilerCR/cd_hit_ALl_ldel_repeats/prototype_ldel_arrays_reversed.fasta|tail -1)
adap=$(echo ${adapName} | sed 's/>//g'|cut -d '_' -f 1)
cleanRound=1

echo "------------------------------------------------"
echo $adap
echo $adapSeq
echo $names
echo $read
echo $species
echo "------------------------------------------------"

mkdir -p ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/

cutadapt ${read} -b ${adapSeq}  -e  0.1 -O 15 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}.txt
rm ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}.txt &

## reads with match
awk '{if($2!="-1") print $0}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt

rm ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt & 

##make a list of current (potentially still containing 5' adapters)
awk -F "\t" '{OFS="\n"} {print "@"$1, $5,"+", $9}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt  > \
${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp.fastq


##trim 5' ends of these (with relaxed matching errors)
cutadapt ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp.fastq -g ${adapSeq}  -e  0.1 -O 5 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_.fastq.txt

rm ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_.fastq.txt &

## finalise first spacers (we will not be able to keep reads that don't have a small (>5bp) of repeat left)
##removing potential 5' adapters of first trim
mkdir -p ${BaseLocation}/${names}/final/
awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($7)>"25" && length($7)<"46" && length($7)==length($11)) print "@"$1, $7 ,"+", $11}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_info.txt > \
${BaseLocation}/${names}/ReadPair_${pair}_reversed/${names}_${adap}_final_round1_R${pair}_reversed.fastq

cat ${BaseLocation}/${names}/ReadPair_${pair}_reversed/${names}_${adap}_final_round1_R${pair}_reversed.fastq > ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R${pair}_reversed.fastq


##take remaining reads and remove 3' adapters again --> do this 5 times
awk -F "\t" '{OFS="\n"} {print "@"$1, $7,"+", $11}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt  > \
${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round2_matchestmp_timmed5_readyForRound2.fastq

#grep --color "GTTTTAGAGCTGTGTTGTTTCGAATGGTTCCAAAAC" /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/readswithrepeats_${adap}_intraReads_info_trimmed_matches_readyForRound2.fastq  |head
cleanRound=2
for cleanRound in $(echo "2 3 4 5 6")
  do
  echo ${cleanRound}
  echo "------------------------------------"
cleanRound_next=$((cleanRound+1))
  
cutadapt ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_readyForRound${cleanRound}.fastq -b ${adapSeq}  -e  0.1 -O 5 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.fasta
rm ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.fasta &
##finalised spacers
awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($5)>"25" && length($5)<"46" && length($5)==length($9)) print "@"$1, $5 ,"+", $9}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}_reversed/${names}_${adap}_final_round${cleanRound}_R${pair}_reversed.fastq

cat ${BaseLocation}/${names}/ReadPair_${pair}_reversed/${names}_${adap}_final_round${cleanRound}_R${pair}_reversed.fastq >> ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R${pair}_reversed.fastq


##finalised prepered
awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($7)>"25" && length($7)<"46" && length($7)==length($11)) print "@"$1, $7,"+", $11}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound_next}_matchestmp_timmed5_readyForRound${cleanRound_next}.fastq



  done #cleanRound
  cat ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R1_reversed.fastq \
  ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R2_reversed.fastq > \
  ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed.fastq
  #revseq ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed.fastq ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular.fastq
  seqtk seq -r  ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed.fastq > ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular.fastq

  done #adap (spacer)
  
  done #pair (readPair)
  done #names (sampleName)

##==============================================
##-----------------merge all samples spacers
##==============================================
rm ${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_${adap}.fastq
rm -r ${BaseLocation}/FinalSpacers/ForDADA2/
mkdir -p ${BaseLocation}/FinalSpacers/ForDADA2/
for names in $(cat ${samplesss} )
  do
for adapName in $(grep ">" /data/Project/CRISPRspacers/02_PilerCR/cd_hit_ALl_ldel_repeats/prototype_ldel_arrays.fasta  )
  do
adapSeq=$(grep "${adapName}" -A 1 /data/Project/CRISPRspacers/02_PilerCR/cd_hit_ALl_ldel_repeats/prototype_ldel_arrays.fasta)
adap=$(echo ${adapName} | sed 's/>//g'|cut -d '_' -f 1)
cleanRound=1

seqtk seq -r  ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed.fastq > ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular.fastq


#-----------change name



#awk -v sampless="$names" -v arrayss="$adapShor"  '{if(NR%4==1) $0=sprintf("@"sampless":"arrayss":s"++k,(1+i++)); print;}' ${BaseLocation}/FinalSpacers/ForDADA2/di_K2_6h_allSpacermerged_Array1.fastq


adapShor=$(echo $adap |sed 's/Array/a/g')
awk -v sampless="$names" -v arrayss="$adapShor"  '{if(NR%4==1) $0=sprintf("@"sampless":"arrayss":s"++k,(1+i++)); print;}' ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads.fastq > ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_renamed.fastq

adapShor=$(echo $adap |sed 's/Array/a/g')
awk -v sampless="$names" -v arrayss="$adapShor"  '{if(NR%4==1) $0=sprintf("@"sampless":R"arrayss":s"++k,(1+i++)); print;}' ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular.fastq > ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular_renamed.fastq



#-----------merge samples
mkdir -p  ${BaseLocation}/FinalSpacers/mergeSpacersFile
cat ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_renamed.fastq >> ${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_${adap}.fastq
cat ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular_renamed.fastq >> ${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_${adap}.fastq


##------ make a folder for dada2
cat ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_renamed.fastq >> ${BaseLocation}/FinalSpacers/ForDADA2/${names}_allSpacermerged_${adap}.fastq
cat ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular_renamed.fastq >> ${BaseLocation}/FinalSpacers/ForDADA2/${names}_allSpacermerged_${adap}.fastq


done #adap (spacer)
  done #names (sampleName)

##----------------move local for Dada2
mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/Ldel/
scp -r vincent@130.223.51.116:/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Ldel//FinalSpacers/ForDADA2/ ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/Ldel/



##----Qualtiy control
grep ">" ${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_Array1.fasta | cut -d ':' -f 1  |sort|uniq -c
grep ">" ${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_Array2.fasta | cut -d ':' -f 1  |sort|uniq -c
grep ">" ${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_Array3.fasta | cut -d ':' -f 1  |sort|uniq -c

grep ">" ${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_Array1.fasta | cut -d ':' -f 2  |sort|uniq -c
grep ">" ${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_Array2.fasta | cut -d ':' -f 2  |sort|uniq -c
grep ">" ${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_Array3.fasta | cut -d ':' -f 2  |sort|uniq -c


```



##Dada2
same as for Sterm

```{r}
library(dada2)
library(tidyverse)

path <- "~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/Ldel/ForDADA2/" # CHANGE ME to the directory containing the fastq files after unzipping.
list.files(path)

fnFs <- sort(list.files(path, pattern=".fastq", full.names = TRUE)) 
sample.names <- str_split_fixed(basename(fnFs), ".fastq",2)[,1]

seq.file <- fnFs[1]
foo <- readLines(seq.file)
seqlens <- sapply(foo[seq(2,length(foo),4)], nchar)
quallens <- sapply(foo[seq(4,length(foo),4)], nchar)
table(seqlens); table(quallens)
table(seqlens) == table(quallens)

plotQualityProfile(fnFs[1:2])
plotQualityProfile(fnFs[3:4])

# Place filtered files in filtered/ subdirectory
filtFs <- file.path(path, "filtered", paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(path, "filtered", paste0(sample.names, "_R_filt.fastq.gz"))
names(filtFs) <- sample.names
names(filtRs) <- sample.names

out <- filterAndTrim(fnFs, filtFs, truncQ=2, rm.phix=TRUE,
              compress=TRUE, multithread=TRUE) # On Windows set multithread=FALSE
head(out)


errF <- learnErrors(filtFs, multithread=TRUE)
plotErrors(errF, nominalQ=TRUE)
dadaFs <- dada(filtFs, err=errF, multithread=TRUE)
dadaFs[[1]]

###fasta table

seqtab <- makeSequenceTable(dadaFs)
dim(seqtab)
table(nchar(getSequences(seqtab)))


##do not remove Chimeras because they are all on the same read and most likely not technical errors but truth biological variation
# seqtab.nochim <- removeBimeraDenovo(seqtab, method="consensus", multithread=TRUE, verbose=TRUE)
# dim(seqtab.nochim)
# sum(seqtab.nochim)/sum(seqtab)


##----phyloseq

library(phyloseq); packageVersion("phyloseq")
library(Biostrings); packageVersion("Biostrings")
library(ggplot2); packageVersion("ggplot2")

samples.out <- rownames(seqtab)
sample <- str_split_fixed(samples.out, "_allSpacermerged_", 2)[,1]
array <- str_split_fixed(samples.out, "_allSpacermerged_", 2)[,2]
# subject <- sapply(strsplit(samples.out, "D"), `[`, 1)
# gender <- substr(subject,1,1)
# subject <- substr(subject,2,999)
# day <- as.integer(sapply(strsplit(samples.out, "D"), `[`, 2))
samdf <- data.frame(Sample=sample, Array=array)
# samdf$When <- "Early"
# samdf$When[samdf$Day>100] <- "Late"
rownames(samdf) <- samples.out


# seq
ps <- phyloseq(otu_table(seqtab, taxa_are_rows=FALSE),sample_data(samdf))

sequences <- Biostrings::DNAStringSet(taxa_names(ps))

names(sequences) <- taxa_names(ps)
ps <- merge_phyloseq(ps, sequences)
taxa_names(ps) <- paste0("spacer_",seq(ntaxa(ps)))

writeXStringSet(refseq(ps), "~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/Ldel/ForDADA2/all_unique_spacers.fasta", append=FALSE, format="fasta") 
refseq(ps)
plot_richness(ps, x="Sample",measures=c("Observed","Shannon", "Simpson"),color="Array")

##------rarefaction

library(ranacapa)
ggrare(ps, step = 10, label = NULL, color = "Sample",
  plot = TRUE, parallel = FALSE, se = TRUE)


##------pca
ps.prop <- transform_sample_counts(ps, function(otu) otu/sum(otu))
ord.nmds.bray <- ordinate(ps.prop, method="NMDS", distance="bray")

plot_ordination(ps.prop, ord.nmds.bray, color="Array", title="Bray NMDS")

##-----------------------
##export OTU table
##-----------------------

normfunction <- function(x) {return(100000*x/sum(x))}
d.norm1 <- transform_sample_counts(physeq = ps, fun = normfunction)
otu_table(d.norm1)




otu_table_df <- otu_table(d.norm1) %>% as.data.frame()%>% rownames_to_column() %>% separate(rowname, c("sample","Array"), sep = "_allSpacermerged_") %>% select(-Array) %>%  group_by(sample) %>% summarise_all(.funs = c(sum="sum")) %>% column_to_rownames(var="sample") %>% t() %>% as.data.frame() %>% rownames_to_column("spacer") %>% 
  mutate(spacer = str_replace(spacer, "_sum", "")) 
otu_table_df
write.table(otu_table_df,"~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/Ldel/ForDADA2/dada_spacer_count.txt",na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names =TRUE)

##-----------------------
##count number of OTUs larger then 0 in every sample
##-----------------------
nrow(otu_table(ps))
otu_counts_samples <- NA
for (samplesss in rownames(otu_table(ps))) {
  tmp_2 <- sum(otu_table(ps)[samplesss]>0)
  tmp_print <- c(sample=as.character(samplesss), otus=as.numeric(tmp_2))
  print(tmp_print)
  otu_counts_samples <- rbind(otu_counts_samples,tmp_print)
}
rownames(otu_table(ps)[1])
sum(otu_table(ps)[1]>0)

otu_counts_samples <- as.data.frame(otu_counts_samples[-1,]) %>% remove_rownames()
otu_counts_samples$otus <- as.numeric(as.character(otu_counts_samples$otus))


##-----------------------
##spaccerss belonging to sample
##-----------------------

ncol(otu_table(ps))
spacer_counts_samples <- data.frame()
# samplesss <- "spacer_1"
for (samplesss in colnames(otu_table(ps))) {
  tmp_2 <- sum(otu_table(ps)[,samplesss]>0)

tmp1 <- sample_names(otu_table(ps)[which(otu_table(ps)[,samplesss]>0)])
countArray1 <- length(grep("Array1",tmp1))
countArray2 <- length(grep("Array2",tmp1))
countArray3 <- length(grep("Array3",tmp1))

numberArrays <-(countArray1>0)+(countArray2>0)+(countArray3>0)

ArrayAssigned <- ifelse(countArray1>0,"Array1",ifelse(countArray2>0,"Array2","Array3"))

  tmp_print <- data.frame(spacer=as.character(samplesss), samplesCount=as.numeric(tmp_2),Array1count=countArray1,Array2count=countArray2,Array3count=countArray3,totalArrays=numberArrays,Array=ArrayAssigned)
  
  print(tmp_print)
  spacer_counts_samples <- rbind(spacer_counts_samples,tmp_print)
}
# rownames(spacer_counts_samples(ps)[1])
sum(otu_table(ps)[1]>0)

# spacer_counts_samples <- as.data.frame(spacer_counts_samples[-1,]) %>% remove_rownames()
# spacer_counts_samples$otus <- as.numeric(as.character(spacer_counts_samples$otus))

spacer_counts_samples[which(spacer_counts_samples$Array2count > 0),]
table(spacer_counts_samples$totalArrays)
table(spacer_counts_samples$Array)
spacer_counts_samples_final <- spacer_counts_samples %>% select(-c(Array1count,Array2count,Array3count,totalArrays))
spacer_counts_samples_out <- spacer_counts_samples %>% select(c(spacer,Array))

write.table(spacer_counts_samples_out,"~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/Ldel/ForDADA2/spacersAssigned.txt",na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names =FALSE)
write.table(spacer_counts_samples_final,"~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/Ldel/ForDADA2/spacers_info.txt",na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names =TRUE)

```

spacers only occur in one array. They never change array. 


####cd-hit of all strain CRISPR spacers


#Explore Ldel plasmid

```{bash}

sed '/##FAS/q' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/L_delbrueckii_RMK202.current.gff  > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/L_delbrueckii_RMK202.current_withoutFASTA.gff 

grep "^#" -v /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/L_delbrueckii_RMK202.current_withoutFASTA.gff |cut -f 1 |sort|uniq -c

##---------------both plasmids
grep "^#" -v /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/L_delbrueckii_RMK202.current_withoutFASTA.gff |grep "CP046131" -v |awk '{if($3=="CDS")print $0}'

###-------------------plasmid 1


grep "CP046132" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/L_delbrueckii_RMK202.current_withoutFASTA.gff |awk '{if($3=="CDS")print $0}'


###-------------------plasmid 2
grep "CP046133" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/L_delbrueckii_RMK202.current_withoutFASTA.gff |awk '{if($3=="CDS")print $0}'

##-------------------------------extract genes from plasmids---------------
rm /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/L_delbrueckii_RMK202_genes_onlyPlasmid.faa

for plasmidGenes in $(grep "^#" -v /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/L_delbrueckii_RMK202.current_withoutFASTA.gff |grep "CP046131" -v |awk -F "[\t.]" '{if($3=="CDS")print $10}' |sed 's/ID=//g')
do

samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/L_delbrueckii_RMK202_genes.faa ${plasmidGenes} >> /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/L_delbrueckii_RMK202_genes_onlyPlasmid.faa

done 



##-------------------------------protein blast---------------

  
  diamond blastp --threads 32 --max-target-seqs 1 --db /archiv/TREMBL_database/20190710/uniprot_trembl.dmnd --query /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/L_delbrueckii_RMK202_genes_onlyPlasmid.faa --out /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/L_delbrueckii_RMK202_genes_onlyPlasmid_diamond.txt

##---------------------------------------move local for eggnong

mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/05_LdelPlasmid/
scp  vincent@130.223.51.116:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/L_delbrueckii_RMK202_genes_onlyPlasmid.faa ~/Desktop/Projects/2019_RMK202_analysis/05_LdelPlasmid/

```


# Illumina analysis

#### download

1. create a file containing all paths to the files from [LIMS](https://uhts-lgtf.vital-it.ch/user/symlinkdataruns/1453/951)
2. download the file with wget -i [File] in the appropitate folder
```{bash}
http://uhts-lgtf.vital-it.ch/symlink/V1_L6_R1_001_eVimQCCGzvPR.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V1_L6_R2_001_X0NeL654UFhg.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V1_L7_R1_001_jvgJBy864KhI.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V1_L7_R2_001_kzfyKrQ5YJuO.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V1_L8_R1_001_M9EtbZWANTlH.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V1_L8_R2_001_6LCQu5kVhdme.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V2_L6_R1_001_D3J3VNGUO2sb.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V2_L6_R2_001_Uh5JXQhzM0Rt.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V2_L7_R1_001_JJDw9KIxeXvd.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V2_L7_R2_001_6a1Vrn1zZPPQ.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V2_L8_R1_001_lPTEu1ohnTPS.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V2_L8_R2_001_eyxyKJ20KL3i.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V3_L6_R1_001_hf6ISQBSU3vw.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V3_L6_R2_001_eXTnM0N6Wnps.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V3_L7_R1_001_dQNFjCHgsKmE.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V3_L7_R2_001_gfhhG5a76yw4.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V3_L8_R1_001_DrKMggM4S7Eo.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V3_L8_R2_001_4fB1hvNMF3Ts.fastq.gz

ls /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz |grep "^V"

```





###name change

create a file with the labels and the names of the file in each a column

```{bash}


for name in $(ls /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz |grep "^V")
do

  name_01=$(echo $name |cut -d '_' -f 1)
  Lane=$(echo $name |cut -d '_' -f 2)
  Read=$(echo $name |cut -d '_' -f 3)  
  
  newName=$(awk -F "\t" -v name="$name_01" '{if($1==name) {print $2}}' /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name.txt)
  
  echo ${newName}"-"${Lane}"-"${Read}".fastq.gz"
    
  mv /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/${name} /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/${newName}-${Lane}-${Read}.fastq.gz
    
    
done

```
the plasmids do not seem to contain any interesting genes. 

concatenenat the fastq.gz files

```{bash}

mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/all_fastq_gz/ 

for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name.txt)
  do
  
  echo ${names}
  
  forward_01=$(ls /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/| grep "$names"  |grep "\-L1\-R1.fastq.gz")
  forward_02=$(ls /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/| grep "$names"  |grep "\-L2\-R1.fastq.gz")
  
  cat /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/${forward_01} \
  /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/${forward_02} > \
  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/all_fastq_gz/${names}_R1.fastq.gz
  
  reverse_01=$(ls /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/| grep "$names"  |grep "\-L1\-R2.fastq.gz")
  reverse_02=$(ls /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/| grep "$names"  |grep "\-L2\-R2.fastq.gz")

  cat /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/${reverse_01} \
  /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/${reverse_02} > \
  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/all_fastq_gz/${names}_R2.fastq.gz
  
  done

##-----------added files

mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/all_fastq_gz/ 
names=V1
for names in $(cut -f 1 /archiv/Projects/2019_pilotplant/01_rawData/log/20191031_logDownload_changeName.txt)
  do
  
  echo ${names}
  
  newName=$(grep "${names}" /archiv/Projects/2019_pilotplant/01_rawData/log/20191031_logDownload_changeName.txt |cut -f 2)
  
  forward_01=$(ls /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/| grep "^${names}_L6_R1")
  forward_02=$(ls /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/| grep "^${names}_L7_R1")
  forward_03=$(ls /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/| grep "^${names}_L8_R1")
  

  cat /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/${forward_01} \
  /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/${forward_02}  \
  /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/${forward_03} > \
  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/all_fastq_gz/${newName}_R1.fastq.gz
  
  
  reverse_01=$(ls /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/| grep "^${names}_L6_R2")
  reverse_02=$(ls /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/| grep "^${names}_L7_R2")
  reverse_03=$(ls /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/| grep "^${names}_L8_R2")
  
  cat /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/${reverse_01} \
  /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/${reverse_02}  \
  /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/${reverse_03} > \
  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/all_fastq_gz/${newName}_R2.fastq.gz
  
   done



```


####FastQC

In a first step we creat [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) reports for all the libraries. 
These are then visualized with [MultiQC](https://multiqc.info/)

```{bash}
threads=4

rm -r /archiv/Projects/2019_pilotplant/01_rawData/03_rawFastQCs/allSamples_fastqcs

mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/03_rawFastQCs/allSamples_fastqcs

num=1
for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name.txt )
  do
  echo ${num}"/37  :" ${names}
  num=$((num+1))
    
    fastqc /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/all_fastq_gz/${names}_R1.fastq.gz -t ${threads} -o /archiv/Projects/2019_pilotplant/01_rawData/03_rawFastQCs/allSamples_fastqcs
    fastqc /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/all_fastq_gz/${names}_R2.fastq.gz -t ${threads} -o /archiv/Projects/2019_pilotplant/01_rawData/03_rawFastQCs/allSamples_fastqcs
    
  done
  mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/Multiqc,output_file='test.html')

  
  multiqc /archiv/Projects/2019_pilotplant/01_rawData/03_rawFastQCs/allSamples_fastqcs -o /archiv/Projects/2019_pilotplant/01_rawData/Multiqc

```




####Adapter removal


In the following we clip adaptors with [trim-galore](https://github.com/FelixKrueger/TrimGalore)

```{bash trimGalore}
threads=35

num=1
for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/log/20191031_logDownload_changeName.txt)
  do
  echo ${num}"/3  :" ${names}
  num=$((num+1))
  
  
  
  #    rm -r /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq//trimmed_Galore/gz/
      mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq//trimmed_Galore/gz/
      trim_galore --fastqc -paired -o /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/ \
        /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/all_fastq_gz/${names}_R1.fastq.gz   \
        /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/all_fastq_gz/${names}_R2.fastq.gz 
        
  echo "##=============="
  echo ${id} "fastQC of trimmed"
  echo "##=============="
      
    mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC/
      
    fastqc /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R1_val_1.fq.gz -t ${threads} -o /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore//fastQC/
    #names=th_K2_8h
    fastqc /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R2_val_2.fq.gz -t ${threads} -o /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC/
    
        
  done 
  

  mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC//Multiqc,output_file='test.html')

  multiqc /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC/ -o /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC/Multiqc


  for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/log/20191031_logDownload_changeName.txt)
  do
 mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC/log
echo "/archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC/" >>  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC/log/multiqc_logfile.txt
   
     done 
  
  mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC//Multiqc_02
  
  multiqc --config /archiv/logs/multiQC_config.txt --file-list /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/log/multiqc_logfile.txt \
  -o /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC//Multiqc_02


```


####KneadData

Here, we filter out the cow genome contamination reads

```{bash}

##prepare reference DB
mkdir -p /archiv/cowDB/mixed/20190520/

cat /archiv/cowDB/mito/20190520/cowDB_renamed.fasta \
  /archiv/cowDB/genome/20190520/GCF_002263795.1_ARS-UCD1.2_genomic.fna > \
  /archiv/cowDB/mixed/20190520/cow_genomeANDmito_DB.fna


bowtie2-build /archiv/cowDB/mixed/20190520/cow_genomeANDmito_DB.fna /archiv/cowDB/mixed/20190520/cow_genomeANDmito_DB

##===================
##remove reads
##===================

##--------------
##experiment
##--------------

threads=37

num=1
for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name_pilotplant.txt |grep "_K2")
#for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/log/20191031_logDownload_changeName.txt)
  do
  

  echo ${num}"/37  :" ${names}
  num=$((num+1))
rm -r /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}
mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}

#gunzip /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R1_val_1.fq.gz
#gunzip /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R2_val_2.fq.gz


kneaddata --max-memory 90000m --no-discordant -t 37 -p 37 --input /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R1_val_1.fq \
  --input /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R2_val_2.fq \
  -db /archiv/cowDB/mixed/20190520/cow_genomeANDmito_DB  \
  --trimmomatic /home/vincent/Joanito/brain/Software/Trimmomatic-0.39/ \
  --output /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}

done


##--------------
##strains
##--------------


threads=37

for names in $(cat /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name_allOfExperiment.txt |grep -v "K1"|grep "Lyo" -v)
  do
  echo ==========================
   echo ${id}
   echo ==========================
 
rm -r /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}
mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}



#kneaddata --input /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${names}/${names}_R1*fq \
#  --input /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${names}/${names}_R2*fq \
#  -db /archiv/cowDB/mixed/20190520/cow_genomeANDmito_DB  \
#  --trimmomatic /home/vincent/Joanito/brain/Software/Trimmomatic-0.39/ \
#  --output /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}

done



##--------------
##evolution experiment
##--------------

names=G4_6_18
threads=37

for names in $(ls /home/vincent/Projects/2020_StarterEvolution/01_data/20200929_Novogene/X204SC20090774-Z01-F001/trimm_Galore/gz/ | grep "^G")
  do
  echo ==========================
   echo ${id}
   echo ==========================
 
rm -r /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}
mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}



kneaddata --input /home/vincent/Projects/2020_StarterEvolution/01_data/20200929_Novogene/X204SC20090774-Z01-F001/trimm_Galore/gz/${names}/${names}-R1*fq.gz \
  --input /home/vincent/Projects/2020_StarterEvolution/01_data/20200929_Novogene/X204SC20090774-Z01-F001/trimm_Galore/gz/${names}/${names}-R2*fq.gz \
  -db /archiv/cowDB/mixed/20190520/cow_genomeANDmito_DB  -t 37 \
  --trimmomatic /home/vincent/Joanito/brain/Software/Trimmomatic-0.39/ \
  --output /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}


mv /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/*_paired_clean_1.fastq  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}_R1_kneaddata_paired_1.fastq

mv /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/*_paired_clean_2.fastq  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}_R2_kneaddata_paired_2.fastq

for delss in $(ls /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/|grep -v "_kneaddata_paired_" )
do

rm /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/$delss


done #delss

done




```

###mOTUs

```{bash}


num=1
  for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/names.txt)
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))
  rm -r /archiv/Projects/2019_pilotplant/mOTU/${names}/

  mkdir -p /archiv/Projects/2019_pilotplant/mOTU/${names}/
 
motus profile -f /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_1.fastq \
  -r /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_2.fastq -o /archiv/Projects/2019_pilotplant/mOTU/${names}/ -t 37 -n ${names}_mOTU

done

rm /archiv/Projects/2019_pilotplant/mOTU/all_mOTUs_prepforR.txt
  for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/names_onlymeta.txt)
  do
#  echo "-----------------------------------------"
#  echo ${names}

grep "^#" -v /archiv/Projects/2019_pilotplant/mOTU/${names}/* |grep -v "0.0000000000" |grep -v "sp." |awk -F "\t" -v sampless="${names}" 'BEGIN{OFS="\t"} {print $1, $2, sampless}' >> /archiv/Projects/2019_pilotplant/mOTU/all_mOTUs_prepforR.txt

done

scp -r vincent@130.223.51.151:/archiv/Projects/2019_pilotplant/mOTU/all_mOTUs_prepforR.txt /home/vincent/Desktop/Projects/2019_Nano_Meta/Illumina_analysis/


```

```{r}
library(readr)
library(ggplot2)

motus_abundance <- read_delim("/home/vincent/Desktop/Projects/2019_Nano_Meta/Illumina_analysis/all_mOTUs_prepforR.txt", "\t", escape_double = FALSE, col_names = c("species","abundance","sample"),trim_ws = TRUE)

motus_abundance$sample

motus_abundance$sample = factor(motus_abundance$sample, levels=c("Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202"))

Sterm_colours <- rev(c("#EB4D4D","#10B552"))
# Sterm_colours <- rev(c("#EB4D4D","#FAA0A0","indianred1","#10B552","#36E37B","#6CF5A3","cyan3","cyan"))

library(plyr)
motus_abundance$sample <- revalue(motus_abundance$sample, c("Lyo_202_2012"="Lyoampulle\n2012", "Lyo_202_2014"="Lyoampulle\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018"))

motus_abundance_meragable <- motus_abundance[,c("sample","species","abundance")]

motus_abundance_meragable$species <- revalue(motus_abundance_meragable$species, c("Streptococcus thermophilus [ref_mOTU_v25_01348]"="S. thermophilus", "Lactobacillus delbrueckii [ref_mOTU_v25_01298]"="L. delbrueckii\nspp. lactis"))
motus_abundance_meragable$abundance <- 100*motus_abundance_meragable$abundance
motus_abundance_meragable$method <- "mOTUs"

pMOTUs <-  ggplot( data = motus_abundance,aes(y = abundance, x = sample, group=species,fill = species))+ geom_bar( stat="identity")+
    labs("",
         x="",
         y="relative Abundance")+
    theme_classic()+
   # scale_color_viridis(discrete=TRUE)+
  scale_fill_manual(values=Sterm_colours)+
  # scale_fill_viridis(discrete=TRUE)+
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
      #axis.text.x = element_blank(),
          legend.position="right",
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank()
          )

  pMOTUs
  
###===========
##relative Abundance my method

  
 onlyBac <- bac_final[bac_final$species=="L_delbrueckii_RMK202"| bac_final$species=="S_thermophilus_RMK202",]
  
  
  

  onlyBac_sum <- aggregate(. ~sample, data=onlyBac[,c("sample","median")], sum, na.rm=TRUE)

onlyBac$total_coverage <- onlyBac_sum[match(onlyBac$sample,onlyBac_sum$sample),"median"]
onlyBac$percent_coverage <- 100*(onlyBac$median/onlyBac$total_coverage)

onlyBac$species <- as.factor(onlyBac$species)

levels(onlyBac$species)
onlyBac$species <- factor(onlyBac$species, levels=rev(c("S_thermophilus_RMK202","L_delbrueckii_RMK202")))
onlyBac$species <- revalue(onlyBac$species, c("S_thermophilus_RMK202"="S. thermophilus", "L_delbrueckii_RMK202"="L. delbrueckii\nspp. lactis"))

onlyBac$sample <- as.factor(onlyBac$sample)

levels(onlyBac$sample)
onlyBac$sample <- factor(onlyBac$sample, levels=(c("Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202")))

library(plyr)
onlyBac$sample <- revalue(onlyBac$sample, c("Lyo_202_2012"="Lyoampulle\n2012", "Lyo_202_2014"="Lyoampulle\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018"))

onlyBac_mergable <- onlyBac[,c("sample","species","percent_coverage")]
colnames(onlyBac_mergable)[3] <- "abundance"
onlyBac_mergable$method <- "mapping"

all_colours <- rev(c("#EB4D4D","#10B552") ) 



##----------------plot

PrelAbundance <-  ggplot( data = onlyBac,aes(y = percent_coverage, x = sample, group=interaction(species),fill = species))+ geom_bar( stat="identity")+
    labs("",
         x="",
         y="Relative\n  abundance [%]")+
    theme_classic()+
   # scale_color_viridis(discrete=TRUE)+
   scale_fill_manual(values=all_colours)+
  # scale_fill_viridis(discrete=TRUE)+
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
        #axis.text.x = element_blank(),
          legend.position="right",
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank()
          )

  PrelAbundance
  
    png("~/Desktop/presenations/my_presentations/20190919_VUA/abundance_species.png", width = 1900, height = 1200,res=300)

PrelAbundance

dev.off()
  
  
##-------merged methods
  motus_abundance_meragable
  onlyBac_mergable
all_methods <- rbind(as.data.frame(motus_abundance_meragable),as.data.frame(onlyBac_mergable))



PrelAbundance <-  ggplot( data = all_methods,aes(y = abundance, x = method, group=interaction(species),fill = species))+ geom_bar( stat="identity")+
  facet_grid(~sample)+
    labs("",
         x="",
         y="Relative\n  abundance [%]")+
    theme_classic()+
   # scale_color_viridis(discrete=TRUE)+
   scale_fill_manual(values=all_colours)+
  # scale_fill_viridis(discrete=TRUE)+
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
        # axis.text.x = element_blank(),
          legend.position="right",
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank()
          )

  PrelAbundance
  

#   svg("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.svg",width=7,height=4)

# 
# #  svg("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.svg",width=7,height=4)
png("~/Desktop/Manuscripts/2019_RMK202/Figures/sF01_relativeAbundance_motus.png", width = 2000, height = 1000,res=300)

PrelAbundance

dev.off()


```


##metaphlan2


```{bash}


num=1
  for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/names.txt)
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))
  rm -r /archiv/Projects/2019_pilotplant/metaphlan2/${names}/

  mkdir -p /archiv/Projects/2019_pilotplant/metaphlan2/${names}/{bowtie2ouput,metaphlan_profile}
 
    metaphlan2 --input_type fastq <(cat /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_1.fastq  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_2.fastq ) \
   --bowtie2out  /archiv/Projects/2019_pilotplant/metaphlan2/${names}//bowtie2ouput/metagenome_${names}_trimmed.bowtie2.bz2 --nproc 37 > \
   /archiv/Projects/2019_pilotplant/metaphlan2/${names}/metaphlan_profile/profiled_metagenome_${names}_trimmed.txt
   

done


    metaphlan2.py --input_type fastq <(zcat DSM20333_L1_R1_001_YgqGCufO9nNQ.fastq.gz  DSM20333_L1_R2_001_2lH3QoiYKwwW.fastq.gz ) \
   --bowtie2out  /metagenome_probe_trimmed.bowtie2.bz2 --nproc 8 > \
   profiled_metagenome_Probe_trimmed.txt
   


```


## Nanopore analysis

The data was basecalled with ont-guppy-for-minknow/now 2.0.5-1~xenial amd64. 

#### get DATA

download data

```{bash}

wget https://campuscloud.unibe.ch/ssf/a/do?p_name=ss_forum&p_action=1&binderId=2753552&action=view_permalink&entityType=folder&novl_url=1&euet=null#1557129336614/fast5_pass.tar.gz.md5

```



#### Read filtering
In a first step I filter the fastq Nanopore reads for the pooled replicates with [filtlong](https://github.com/rrwick/Filtlong). 
I use a filter for min read length of 8kb and the best quality reads. 

```{bash}

##2AB
/home/vincent/apps/Filtlong/bin/filtlong --min_length 7000 --keep_percent 90 --target_bases 1000000000 --mean_q_weight 10 /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_2A2B/fastq_pass/all_2AB_raw.fq | gzip > /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_2A2B/fastq_pass/all_2AB_qualityFiltered_min7000_Q90.fq.gz

mv /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_2A2B/fastq_pass/all_2AB_qualityFiltered_min7000_Q90.fq /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_2A2B/fastq_pass/all_2AB_qualityFiltered_min7000_Q90.fq.gz

##1AB
/home/vincent/apps/Filtlong/bin/filtlong --min_length 7000 --keep_percent 90 --target_bases 1000000000 --mean_q_weight 10 /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_1A1B/fastq_pass/all_1AB_raw.fq | gzip > /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_1A1B/fastq_pass/all_1AB_qualityFiltered_min7000_Q90.fq.gz

mv /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_1A1B/fastq_pass/all_1AB_qualityFiltered_min7000_Q90.fq /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_1A1B/fastq_pass/all_1AB_qualityFiltered_min7000_Q90.fq.gz


##===============
##Demultiplexed data
##===============

mkdir -p /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered

for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
  do

  echo "=============="
  echo ${sample}
  echo "=============="

    /home/vincent/apps/Filtlong/bin/filtlong --min_length 7000 --keep_percent 90 --target_bases 1000000000 --mean_q_weight 10 /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_renamed/${sample}_nanopore_raw.fastq.gz | gzip > /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/${sample}_nanopore_filtered.fastq.gz


  done


scp -r vincent@130.223.51.151:/archiv/Projects/2019_pilotplant/01_rawData/Multiqc /home/vincent/Desktop/Projects/2019_Nano_Meta/Illumina_analysis


```


## Genome Assembly


#### Flye
Here, we create a [flye](https://github.com/fenderglass/Flye/blob/flye/docs/graph_example.png) assembly of the following:
1. demultiplexed samples with Qualtiy filtering max 100000000 and minLength 7000 reads
2. Than I look at the Assembly graph with Bandage. 
3. Map the short contigs to blastN online to id them

```{bash}


##===========
##Demultiplexed
##===========

##===========
##Assembly plasmid
##===========
for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
  do
  
    sample=th_K2_8h
    
    rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${sample}/assembly/
    mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${sample}/assembly/
  echo "=============="
  echo ${sample}
  echo "=============="

    flye --threads 15  --iterations 5 --genome-size 4500000 --plasmids --nano-raw \
      /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/${sample}_nanopore_filtered.fastq.gz \
      --out-dir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${sample}/assembly/

  done



###---Bandage
sample=th_K2_8h


mkdir -p ~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/assembly/

scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${sample}/assembly/assembly_graph.gfa ~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/assembly/${sample}_assembly_Graph_plasmid.gfa



##-------assembly length

seq_length.py /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/assembly.fasta
seq_length.py /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/th_K2_8h/assembly/assembly.fasta

##-------cd -hit
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/cd_hit
#cd-hit-est -i  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/assembly.fasta \
#    -o  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/cd_hit/assembly_Clustering.txt -c 0.85 -M 30000 -T 37 -sc 1 -sf 1
  
/home/vincent/apps/cd-hit-v4.8.1-2019-0228/cd-hit-est -i   /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/assembly.fasta \
    -o  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/cd_hit/assembly_Clustering.txt -c 0.85
  
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/th_K2_8h/assembly/cd_hit
/home/vincent/apps/cd-hit-v4.8.1-2019-0228/cd-hit-est -i  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/th_K2_8h/assembly/assembly.fasta     -o  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/th_K2_8h/assembly/cd_hit/assembly_Clustering.txt -c 0.85




```


The rmk202 reference sample I will use di_K2_6h :
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/

#### PostAssembly analysis



```{bash}


##===========
##Assembly demultiplexed
##===========


for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
  do
  
    sample=di_K2_6h
    
    rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${sample}/postAssemblyAnalysis
    mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${sample}/postAssemblyAnalysis
  echo "=============="
  echo ${sample}
  echo "=============="


  PostAssemblyAnalysis.R \
  -i Y \
  -x ${sample} \
  -a /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${sample}/assembly/assembly.fasta \
  -l /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/${sample}_nanopore_filtered.fastq.gz \
  -F /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R1_val_1.fq \
  -R /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R2_val_2.fq \
  -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${sample}/postAssemblyAnalysis \
  -t 35 \
  -s Y \
  --BlastDB /archiv/BLAST/blast_db/20190204/Bacteria_and_Phages/Bac_Phage


  done
  
```

###looking into read filtering with the edit distance
Good recources are the (SAM description)[https://samtools.github.io/hts-specs/SAMv1.pdf] and (additional file)[https://samtools.github.io/hts-specs/SAMtags.pdf] and the (blog post of WouterDeCoster)[https://gigabaseorgigabyte.wordpress.com/2017/04/14/getting-the-edit-distance-from-a-bam-alignment-a-journey/].

Interesting BWA output columns
- The edit distance is the NM tag ($12) in the sam file. The edit distance is the number of changes to the query to get the reference
- The CIGAR string is $6
- The read alignment length is the AS tag ($15)
- The MD tag, which stores a string containing matching numbers of nucleotides and non-matching nucleotides


NM:i:count Number of differences (mismatches plus inserted and deleted bases) between the sequence and
reference, counting only (case-insensitive) A, C, G and T bases in sequence and reference as potential
matches, with everything else being a mismatch. Note this means that ambiguity codes in both
sequence and reference that match each other, such as ‘N’ in both, or compatible codes such as ‘A’ and
‘R’, are still counted as mismatches. The special sequence base ‘=’ will always be considered to be a
match, even if the reference is ambiguous at that point. Alignment reference skips, padding, soft and
hard clipping (‘N’, ‘P’, ‘S’ and ‘H’ CIGAR operations) do not count as mismatches, but insertions and
deletions count as one mismatch per base.
Note that historically this has been ill-defined and both data and tools exist that disagree with this
definition.

This is also a good blog post by (David Tang)[https://davetang.org/wiki/tiki-index.php?page=SAM]


```{r}
library(readr)
library(ggplot2)

mappingStat <- read_delim("K1_ONT_rawIllumina_bwamem_sorted_only100_statsforR.txt", "\t", escape_double = FALSE, col_names = TRUE,trim_ws = TRUE)



sum(mappingStat$identity<0.96)
sum(mappingStat$identity<0.97)
sum(mappingStat$identity<0.98)
sum(mappingStat$identity<0.99)
dim(mappingStat)
sum(mappingStat$identity<45)
sum(mappingStat$Mapped_Length<45)
sum(mappingStat$Mapped_Length>45)
sum(mappingStat$EditDistance>5)
sum(mappingStat$EditDistance<5)

summary(mappingStat$Mapped_Length)
summary(mappingStat$EditDistance)
Id_dens <- ggplot(mappingStat,aes(x=identity))+geom_density()
NM_dens <- ggplot(mappingStat,aes(x=EditDistance))+geom_density()
ML_dens <- ggplot(mappingStat,aes(x=Mapped_Length))+geom_density()


   svg("plots/Id_dens.svg",width=5,height=5)
Id_dens
 dev.off()
 
 
   svg("plots/NM_dens.svg",width=5,height=5)
NM_dens
 dev.off()
 
 
   svg("plots/ML_dens.svg",width=5,height=5)
ML_dens
 dev.off()
 
 

```

move local

```{bash}



scp -r vincent@130.223.51.151:/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/plots/ ~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/1AB/assembly/


```

#### Rename contigs

The naming was manually choosen by blasting the small contigs online

```{bash}
###-----------rename contigs
sed 's/contig_1$/S_thermophilus_mag_rmk202/g'   /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/assembly.fasta | \
    sed 's/contig_10$/S_thermophilus_repA/g'| \
    sed 's/contig_11$/Bos_taurus_genome_01/g'| \
    sed 's/contig_12$/L_del_phage_01/g'| \
    sed 's/contig_2$/Bos_taurus_mito/g'| \
    sed 's/contig_3$/L_delbrueckii_mag_rmk202/g'| \
    sed 's/contig_4$/Bos_taurus_genome_02/g'| \
    sed 's/contig_5$/Unknown_01/g'| \
    sed 's/contig_6$/Bos_taurus_genome_03/g'| \
    sed 's/contig_7$/Bos_taurus_genome_04/g'| \
    sed 's/contig_8$/L_del_plasmid_01/g'| \
    sed 's/contig_9$/Bos_taurus_genome_05/g' > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/di_rmk202_MAG_reference_renamed.fasta



```


#### circlator
as circlator did not work. I will do a prokka annotation and annotate manually at the DnaA

```{bash}

###-----------devide all contigs
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs
for contigs in $(grep ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/di_rmk202_MAG_reference_renamed.fasta |sed 's/>//g' )
  do
  echo ${contigs}
samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/di_rmk202_MAG_reference_renamed.fasta ${contigs} > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/${contigs}.fasta
  done
###-----------PRokka
##------------------------------------Streptococcus thermophilus
contigs=S_thermophilus_mag_rmk202
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/unpolished/${contigs}
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/unpolished/${contigs}
    
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/unpolished/${contigs}/ --force --usegenus --genus Streptococcus --locustag Sterm_202 --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/${contigs}.fasta
##------------------------------------Lactoabillus delbrueckii
contigs=L_delbrueckii_mag_rmk202
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/unpolished/${contigs}
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/unpolished/${contigs}
    
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/unpolished/${contigs}/ --force --usegenus --genus Lactobacillus --locustag Ldel_202 --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/${contigs}.fasta
  
  ##=================  
###-----------rearrang contigs
  ##=================  
  
  ##===================================================Lactobacillus delbrueckii
  grep "DNAa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/unpolished/L_delbrueckii_mag_rmk202/PROKKA_11142019.gff
  
  #37825
  contigs=L_delbrueckii_mag_rmk202


###---------------extract fragemtent 1-------------------
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/logs/
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/logs/

echo -e ${contigs}"\t1\t37000" > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/circlaring_ldel/logs/${contigs}_fragment1.bed

samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/${contigs}.fasta

bedtools getfasta -fi /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/${contigs}.fasta \
  -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/circlaring_ldel/logs/${contigs}_fragment1.bed \
  -fo /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/circlaring_ldel/logs/${contigs}_fragment1.fasta

grep -v ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/circlaring_ldel/logs/${contigs}_fragment1.fasta > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/circlaring_ldel/logs/${contigs}_fragment1_megerable.fasta


###---------------extract fragemtent 3-------------------


#seq_length.py /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/${contigs}.fasta

echo -e ${contigs}"\t37001\t2193739" > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/logs//${contigs}_fragment2.bed

#samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring_ldel/input/contig_1.fasta

bedtools getfasta -fi /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/${contigs}.fasta \
  -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/logs//${contigs}_fragment2.bed \
  -fo /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/logs//${contigs}_fragment2.fasta

grep -v ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/logs//${contigs}_fragment2.fasta > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/logs//${contigs}_fragment2_mergeable.fasta


##-------merging fragments-------
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/final

echo ">"${contigs} > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/final/${contigs}_unformated.fasta

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/circlaring_ldel/logs/${contigs}_fragment2_mergeable.fasta \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/logs//${contigs}_fragment1_megerable.fasta | sed 's/\n//g' >>  \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/final/${contigs}_unformated.fasta
  

fasta_formatter -w 50 -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/final/${contigs}_unformated.fasta -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/final/${contigs}.fasta

###------------------checking with annotation
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/final/annoted/${contigs}
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/final/annoted/${contigs}
    
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/final/annoted/${contigs} --force --usegenus --genus Lactobacillus --locustag Ldel_202 --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/final/${contigs}.fasta
  
  grep "DNAa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/final/annoted/${contigs}/PROKKA_*.gff
  ##===================================================Streptococcus_thermophilus


  ##===================================================Lactobacillus delbrueckii
  contigs=S_thermophilus_mag_rmk202
  grep "DNAa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/unpolished/${contigs}/PROKKA_*.gff
  #345399
  
###---------------extract fragemtent 1-------------------
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/logs/
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/logs/

echo -e ${contigs}"\t1\t344900" > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/circlaring_sterm/logs/${contigs}_fragment1.bed

samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/${contigs}.fasta

bedtools getfasta -fi /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/${contigs}.fasta \
  -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/circlaring_sterm/logs/${contigs}_fragment1.bed \
  -fo /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/circlaring_sterm/logs/${contigs}_fragment1.fasta

grep -v ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/circlaring_sterm/logs/${contigs}_fragment1.fasta > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/circlaring_sterm/logs/${contigs}_fragment1_megerable.fasta


###---------------extract fragemtent 3-------------------


#seq_length.py /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/${contigs}.fasta

echo -e ${contigs}"\t344900\t1895563" > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/logs//${contigs}_fragment2.bed

#samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring_sterm/input/contig_1.fasta

bedtools getfasta -fi /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/${contigs}.fasta \
  -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/logs//${contigs}_fragment2.bed \
  -fo /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/logs//${contigs}_fragment2.fasta

grep -v ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/logs//${contigs}_fragment2.fasta > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/logs//${contigs}_fragment2_mergeable.fasta


##-------merging fragments-------
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final

echo ">"${contigs} > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/${contigs}_unformated.fasta

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/circlaring_sterm/logs/${contigs}_fragment2_mergeable.fasta \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/logs//${contigs}_fragment1_megerable.fasta | sed 's/\n//g' >>  \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/${contigs}_unformated.fasta
  

fasta_formatter -w 50 -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/${contigs}_unformated.fasta -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/${contigs}.fasta

###------------------checking with annotation
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/${contigs}
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/${contigs}
    
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/${contigs} --force --usegenus --genus Streptococcus --locustag Sterm_202 --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/${contigs}.fasta
  
  grep "DNAa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/${contigs}/PROKKA_*.gff
  


##--------------------------------merge all again
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_ldel/final/L_delbrueckii_mag_rmk202.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/S_thermophilus_mag_rmk202.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/L_del_plasmid_01.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/S_thermophilus_repA.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/L_del_phage_01.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/Bos_taurus_mito.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/Bos_taurus_genome_01.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/Bos_taurus_genome_02.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/Bos_taurus_genome_03.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/Bos_taurus_genome_04.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/Bos_taurus_genome_05.fasta > \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned_unformated.fasta


fasta_formatter -w 50 -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned_unformated.fasta -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta



#/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/Unknown_01.fasta \

###------------------checking with annotation
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all
    
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all --force --usegenus --genus Streptococcus --locustag Sterm_202 --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta
  
  grep "DNAa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all/PROKKA_*.gff
  


```



#### ONT polishing Racon
After cirlarication we do several rounds fo polishing
This is then check by continious Quality control steps with the postAssemblyAnylsis tool. 
For polishing I consider the following tools:
- [Racon](https://github.com/isovic/racon)
- 

```{bash}

###===========================
###-----------------
##Racon polishing with graphmap
###-----------------
###===========================    
     
##------------------define variables------------------
#for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt |tail -2)
#do


name=di_K2_6h
sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta 
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/di_K2_6h_nanopore_filtered.fastq.gz
#reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta
echo "========================================="
echo $sample
echo "========================================="

echo $reference_fasta
echo "---------------------------"
     
###===========================
###-----------------
##graphmap mapping
###-----------------
###===========================      

name=first
num=1

for name in first second third fourth
do 
rm -r $Overall_output_directory/0${num}_${name}Racon_graphmap/
mkdir -p $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/bamqc

#name=first

graphmap align -t ${threads} \
  -r $reference_fasta \
  -d $ont_reads \
  -o $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam 

#  -d $ont_reads | samtools sort -@${threads} -O BAM -o $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam - 

  
###===========================
###-----------------
##racon polishing
###-----------------
###===========================   

#rm -r$Overall_output_directory/0${num}_${name}Racon_graphmap/assembly

mkdir -p $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly

samtools view -bS $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam | samtools sort -@${threads} -O BAM -o $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam -

#samtools sort -@${threads} -O SAM $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam -

#samtools view -S -b $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam > $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam 

/home/vincent/apps/racon/new/racon/build/bin/racon \
 -t ${threads} \
  $ont_reads \
  $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam \
  $reference_fasta > \
  $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly/${name}Racon_NWC_2Assembly.fasta
  


samtools index $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam 

qualimap bamqc -bam $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam \
  -outdir $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/bamqc --java-mem-size=40G 

#rm $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam

  reference_fasta=$(echo $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly/${name}Racon_NWC_2Assembly.fasta)
echo $reference_fasta
num=$((num+1)) 

done


##--------------------cleanup


name=di_K2_6h
sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta 
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/di_K2_6h_nanopore_filtered.fastq.gz
#reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta
name=first
num=1

for name in first second third fourth
do 


rm -r $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/

  reference_fasta=$(echo $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly/${name}Racon_NWC_2Assembly.fasta)
echo $reference_fasta
num=$((num+1)) 

done
#done # loop for 1AB and 2AB samples


##--------------------test PROKKA
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
name=fourth
num=4

rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all_0${num}_Racon
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all_0${num}_Racon
    
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all_0${num}_Racon --force --usegenus --genus Streptococcus --locustag Sterm_202 --proteins proteins.faa --evalue 0.001 --addgenes   $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly/${name}Racon_NWC_2Assembly.fasta
  
  grep "DNAa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all_0${num}_Racon/PROKKA_*.gff
  
  echo "=======================================================ALT==============================================================="
  grep "DNAa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all/PROKKA_*.gff


  


###===========================
###-----------------
##quast
###-----------------
###===========================  

#for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
#do


name=di_K2_6h
sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta 
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/di_K2_6h_nanopore_filtered.fastq.gz


echo "========================================="
echo $sample
echo "========================================="

echo $reference_fasta
echo "---------------------------"
     
    rm -r $Overall_output_directory/quastwithRaconONT/
    mkdir -p $Overall_output_directory/quastwithRaconONT
    quast.py \
      -t ${threads} -l "FlyeAssembly,firstONTracon,secondONTracon,thirdONTracon" \
      -R $Overall_output_directory/04_fourthRacon_graphmap/assembly/fourthRacon_NWC_2Assembly.fasta \
      $reference_fasta \
      $Overall_output_directory/01_firstRacon_graphmap/assembly/firstRacon_NWC_2Assembly.fasta \
      $Overall_output_directory/02_secondRacon_graphmap/assembly/secondRacon_NWC_2Assembly.fasta \
      $Overall_output_directory/03_thirdRacon_graphmap/assembly/thirdRacon_NWC_2Assembly.fasta \
      -o $Overall_output_directory/quastwithRaconONT
 
 
 ##-------------------making MultiQC output
 rm $Overall_output_directory/logs_MultiQC_raconONT.txt
 
 echo $Overall_output_directory/quastwithRaconONT/report.tsv >> $Overall_output_directory/logs_MultiQC_raconONT.txt
 
 num=1

for name in first second third fourth
do 

echo $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/bamqc >> $Overall_output_directory/logs_MultiQC_raconONT.txt
num=$((num+1)) 

done ##loop for name
 
#done #loop for sample
###===========================
###-----------------
##MultiQC
###-----------------
###===========================  


#for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
#do
name=di_K2_6h
sample=di_K2_6h
threads=37
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${sample}/
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${sample}/assembly/assembly.fasta 
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/${sample}_nanopore_filtered.fastq.gz

echo "========================================="
echo $sample
echo "========================================="



#done #loop for sample
 
rm -r $Overall_output_directory/MultiQCRaconONT/
mkdir -p $Overall_output_directory/MultiQCRaconONT/
  /home/vincent/.local/bin/multiqc --config /archiv/logs/multiQC_config.txt --file-list   $Overall_output_directory/logs_MultiQC_raconONT.txt \
    -o $Overall_output_directory/MultiQCRaconONT/



###--------------on local

name=di_K2_6h
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/quastwithRaconONT/report.html ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC/${name}_quast_report.html

scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/MultiQCRaconONT/multiqc_report.html ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC/${name}_multiqc_report.html



```


#### Illumina polishing freebayes 

Now we polish with the high quality Illumina reads

```{bash}


###-----------------
##Illuminamapping rawReads to circlator genome
###-----------------

   
##------------------define variables------------------
#for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
#do

name=di_K2_6h
sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/04_fourthRacon_graphmap/assembly/fourthRacon_NWC_2Assembly.fasta
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/di_K2_6h_nanopore_filtered.fastq.gz

num=5


for run in first second third fourth
   do 

name=run
##--------------create directories----------------
rm -r $Overall_output_directory/0${num}_${run}Freebayes/
mkdir -p $Overall_output_directory/0${num}_${run}Freebayes/{rawIlluminaMapped,freebayesOuput}
mkdir -p $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/bamqc

#mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fq

# gunzip -c /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R1_val_1.fq.gz > \
#  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fq/${sample}_R1_val_1.fq
# gunzip -c /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R2_val_2.fq.gz > \
#  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fq/${sample}_R2_val_2.fq


bwa index $reference_fasta

bwa mem -t ${threads}  $reference_fasta /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R1_val_1.fq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R2_val_2.fq |samtools sort -@8 -O BAM \
-o $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam - 
     

qualimap bamqc -bam $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam \
        -outdir $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/bamqc --java-mem-size=40G
    
samtools index $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam

###-----------------
##freebayesAfterCirclator
###-----------------
echo freebayes

    SECONDS=0

    freebayes -F 0.5 --min-coverage 4 -p 1 -f \
      $reference_fasta \
      $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam > \
      $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf
      
      
    echo "RUNTIME (h:min:s)= $(($SECONDS / 3600)):$((($SECONDS / 60) % 60)):$(($SECONDS % 60))"
    echo END = `date +"%Y-%m-%d %H:%M:%S"`

##-------   
##bcftools variant integration of Illumina data into Pacbio polished genome
##-------

    SECONDS=0
    echo "bgzip"
bgzip -c  $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf > \
       $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf.gz
      echo "tabix"
tabix -p vcf  $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf.gz
    echo "bcftools"
bcftools consensus -f $reference_fasta \
      -o $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${num}_${run}Freebayes.fasta \
       $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf.gz
      
    echo "RUNTIME (h:min:s)= $(($SECONDS / 3600)):$((($SECONDS / 60) % 60)):$(($SECONDS % 60))"
    echo END = `date +"%Y-%m-%d %H:%M:%S"`
  
  bwa index $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${num}_${run}Freebayes.fasta 
  
   ##-------------------new variable name for next polishing steps-----------
  reference_fasta=$Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${num}_${run}Freebayes.fasta  
  num=$((num+1)) 
done
    
 ##------------------------------------------clean-------------------------  


name=di_K2_6h
sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/04_fourthRacon_graphmap/assembly/fourthRacon_NWC_2Assembly.fasta
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/di_K2_6h_nanopore_filtered.fastq.gz

num=5


for run in first second third fourth
   do 

name=run

rm -r $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped

 num=$((num+1)) 

done

##--------------indexing


bwa index $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${num}_${run}Freebayes.fasta  




##--------------------test PROKKA
#Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
name=second
num=6

rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all_0${num}
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all_0${num}
    
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all_0${num} --force --usegenus --genus Streptococcus --locustag Sterm_202 --proteins proteins.faa --evalue 0.001 --addgenes  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/0${num}_${name}Freebayes/freebayesOuput/${num}_${name}Freebayes.fasta
  
  grep "DNAa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all_0${num}/PROKKA_*.gff
  
  echo "=======================================================ALT==============================================================="
  grep "DNAa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all/PROKKA_*.gff



#done #samples
###===========================
###-----------------
##quast
###-----------------
###===========================  

#for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
#do
#sample=1
name=di_K2_6h
sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta 
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/di_K2_6h_nanopore_filtered.fastq.gz

rm $Overall_output_directory/logs_MultiQC_raconFreebayesONT.txt

echo "========================================="
echo $sample
echo "========================================="

echo $reference_fasta
echo "---------------------------"
     
    rm -r $Overall_output_directory/quastwithRacon_FreebayesONT/
    mkdir -p $Overall_output_directory/quastwithRacon_FreebayesONT
    quast.py \
      -t ${threads} -l "FlyeAssembly,firstONTracon,secondONTracon,thirdONTracon,fourthONTracon,firstFreebayes,secondFreebayes,thirdFreebayes" \
      -R $Overall_output_directory/08_fourthFreebayes/freebayesOuput/8_fourthFreebayes.fasta \
      $reference_fasta \
      $Overall_output_directory/01_firstRacon_graphmap/assembly/firstRacon_NWC_2Assembly.fasta \
      $Overall_output_directory/02_secondRacon_graphmap/assembly/secondRacon_NWC_2Assembly.fasta \
      $Overall_output_directory/03_thirdRacon_graphmap/assembly/thirdRacon_NWC_2Assembly.fasta \
      $Overall_output_directory/04_fourthRacon_graphmap/assembly/fourthRacon_NWC_2Assembly.fasta \
      $Overall_output_directory/05_firstFreebayes/freebayesOuput/5_firstFreebayes.fasta \
      $Overall_output_directory/06_secondFreebayes/freebayesOuput/6_secondFreebayes.fasta  \
      $Overall_output_directory/07_thirdFreebayes/freebayesOuput/7_thirdFreebayes.fasta  \
            -o $Overall_output_directory/quastwithRacon_FreebayesONT
 
      
     
    
      
###===========================
###-----------------
##MultiQC
###-----------------
###===========================  


name=di_K2_6h
sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta 
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/di_K2_6h_nanopore_filtered.fastq.gz

echo "========================================="
echo $sample
echo "========================================="


  rm -r $Overall_output_directory/MultiQCRaconFreebayesONT
    
  mkdir -p $Overall_output_directory/MultiQCRaconFreebayesONT
  
  /home/vincent/.local/bin/multiqc --config /archiv/logs/multiQC_config.txt --file-list  $Overall_output_directory/logs_MultiQC_raconFreebayesONT.txt \
    -o $Overall_output_directory/MultiQCRaconFreebayesONT/





###--------------on local



 name=di_K2_6h
 scp -r vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/quastwithRacon_FreebayesONT/ ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC/


scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/MultiQCRaconFreebayesONT/multiqc_report.html
~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC/${name}_multiqc_report_final.html



```



#### after circlator
as circlator did not work. I will do a prokka annotation and annotate manually at the DnaA

```{bash}
name=di_K2_6h
sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta 
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/di_K2_6h_nanopore_filtered.fastq.gz

###-------------------------------------------------------rename contigs

sed 's/S_thermophilus_mag_rmk202.*$/S_thermophilus_mag_rmk202/g' $Overall_output_directory/08_fourthFreebayes/freebayesOuput/8_fourthFreebayes.fasta | \
    sed 's/S_thermophilus_repA.*$/S_thermophilus_repA/g'| \
    sed 's/Bos_taurus_genome_01.*$/Bos_taurus_genome_01/g'| \
    sed 's/L_del_phage_01.*$/L_del_phage_01/g'| \
    sed 's/Bos_taurus_mito.*$/Bos_taurus_mito/g'| \
    sed 's/L_delbrueckii_mag_rmk202.*$/L_delbrueckii_mag_rmk202/g'| \
    sed 's/Bos_taurus_genome_02.*$/Bos_taurus_genome_02/g'| \
    sed 's/Bos_taurus_genome_03.*$/Bos_taurus_genome_03/g'| \
    sed 's/Bos_taurus_genome_04.*$/Bos_taurus_genome_04/g'| \
    sed 's/L_del_plasmid_01.*$/L_del_plasmid_01/g'| \
    sed 's/Bos_taurus_genome_05.*$/Bos_taurus_genome_05/g' > $Overall_output_directory/08_fourthFreebayes/freebayesOuput/di_rmk202_MAG_reference_polished_renamed.fasta 



###-----------devide all contigs
rm -r $Overall_output_directory/circlatorAfter/splitContigs
mkdir -p $Overall_output_directory/circlatorAfter/splitContigs
for contigs in $(grep ">" $Overall_output_directory/08_fourthFreebayes/freebayesOuput/di_rmk202_MAG_reference_polished_renamed.fasta  |sed 's/>//g' )
  do
  echo ${contigs}
samtools faidx $Overall_output_directory/08_fourthFreebayes/freebayesOuput/di_rmk202_MAG_reference_polished_renamed.fasta  ${contigs} > $Overall_output_directory/circlatorAfter/splitContigs/${contigs}.fasta
  done
###-----------PRokka
##------------------------------------Streptococcus thermophilus
contigs=S_thermophilus_mag_rmk202
rm -r $Overall_output_directory/circlatorAfter/splitContigs/unpolished/${contigs}
mkdir -p $Overall_output_directory/circlatorAfter/splitContigs/unpolished/${contigs}
    
prokka --outdir $Overall_output_directory/circlatorAfter/splitContigs/unpolished/${contigs}/ --force --usegenus --genus Streptococcus --locustag Sterm_202 --proteins proteins.faa --evalue 0.001 --addgenes $Overall_output_directory/circlatorAfter/splitContigs/${contigs}.fasta
  grep -i "dnaa" $Overall_output_directory/circlatorAfter/splitContigs/unpolished/${contigs}/*.gff
##------------------------------------Lactoabillus delbrueckii
contigs=L_delbrueckii_mag_rmk202
rm -r  $Overall_output_directory/circlatorAfter/splitContigs/unpolished/${contigs}
mkdir -p  $Overall_output_directory/circlatorAfter/splitContigs/unpolished/${contigs}
    
prokka --outdir  $Overall_output_directory/circlatorAfter/splitContigs/unpolished/${contigs}/ --force --usegenus --genus Lactobacillus --locustag Ldel_202 --proteins proteins.faa --evalue 0.001 --addgenes  $Overall_output_directory/circlatorAfter/splitContigs/${contigs}.fasta
  
  
  grep -i "dnaa" $Overall_output_directory/circlatorAfter/splitContigs/unpolished/${contigs}/*.gff
  ##=================  
###-----------rearrang contigs
  ##=================  
  
  ##===================================================Lactobacillus delbrueckii
  contigs=L_delbrueckii_mag_rmk202
  grep "DNAa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/unpolished/L_delbrueckii_mag_rmk202/PROKKA_11142019.gff
  seq_length.py $Overall_output_directory/circlatorAfter/splitContigs/${contigs}.fasta
  grep "gene" $Overall_output_directory/circlatorAfter/splitContigs/unpolished/${contigs}/*.gff|tail #2166497
  #37825
  #2166766


###---------------extract fragemtent 1-------------------
rm -r $Overall_output_directory/circlatorAfter///circlaring_ldel/logs/
mkdir -p $Overall_output_directory/circlatorAfter///circlaring_ldel/logs/

echo -e ${contigs}"\t1\t2166500" > $Overall_output_directory/circlatorAfter//circlaring_ldel/logs/${contigs}_fragment1.bed

samtools faidx $Overall_output_directory/circlatorAfter//splitContigs/${contigs}.fasta

bedtools getfasta -fi $Overall_output_directory/circlatorAfter//splitContigs/${contigs}.fasta \
  -bed $Overall_output_directory/circlatorAfter//circlaring_ldel/logs/${contigs}_fragment1.bed \
  -fo $Overall_output_directory/circlatorAfter//circlaring_ldel/logs/${contigs}_fragment1.fasta

grep -v ">" $Overall_output_directory/circlatorAfter//circlaring_ldel/logs/${contigs}_fragment1.fasta > $Overall_output_directory/circlatorAfter//circlaring_ldel/logs/${contigs}_fragment1_megerable.fasta


###---------------extract fragemtent 3-------------------


#seq_length.py $Overall_output_directory/circlatorAfter//splitContigs/${contigs}.fasta

echo -e ${contigs}"\t2166500\t2166766" > $Overall_output_directory/circlatorAfter///circlaring_ldel/logs//${contigs}_fragment2.bed

#samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring_ldel/input/contig_1.fasta

bedtools getfasta -fi $Overall_output_directory/circlatorAfter//splitContigs/${contigs}.fasta \
  -bed $Overall_output_directory/circlatorAfter///circlaring_ldel/logs//${contigs}_fragment2.bed \
  -fo $Overall_output_directory/circlatorAfter///circlaring_ldel/logs//${contigs}_fragment2.fasta

grep -v ">" $Overall_output_directory/circlatorAfter///circlaring_ldel/logs//${contigs}_fragment2.fasta > $Overall_output_directory/circlatorAfter///circlaring_ldel/logs//${contigs}_fragment2_mergeable.fasta


##-------merging fragments-------
mkdir -p $Overall_output_directory/circlatorAfter///circlaring_ldel/final

echo ">"${contigs} > $Overall_output_directory/circlatorAfter///circlaring_ldel/final/${contigs}_unformated.fasta

cat $Overall_output_directory/circlatorAfter//circlaring_ldel/logs/${contigs}_fragment2_mergeable.fasta \
  $Overall_output_directory/circlatorAfter///circlaring_ldel/logs//${contigs}_fragment1_megerable.fasta | sed 's/\n//g' >>  \
  $Overall_output_directory/circlatorAfter///circlaring_ldel/final/${contigs}_unformated.fasta
  

fasta_formatter -w 50 -i $Overall_output_directory/circlatorAfter///circlaring_ldel/final/${contigs}_unformated.fasta -o $Overall_output_directory/circlatorAfter///circlaring_ldel/final/${contigs}.fasta

###------------------checking with annotation
rm -r $Overall_output_directory/circlatorAfter///circlaring_ldel/final/annoted/${contigs}
mkdir -p $Overall_output_directory/circlatorAfter///circlaring_ldel/final/annoted/${contigs}
    
prokka --outdir $Overall_output_directory/circlatorAfter///circlaring_ldel/final/annoted/${contigs} --force --usegenus --genus Lactobacillus --locustag Ldel_202 --proteins proteins.faa --evalue 0.001 --addgenes $Overall_output_directory/circlatorAfter///circlaring_ldel/final/${contigs}.fasta
  
  grep "DNAa" -i $Overall_output_directory/circlatorAfter///circlaring_ldel/final/annoted/${contigs}/PROKKA_*.gff
  ##===================================================Streptococcus_thermophilus


  contigs=S_thermophilus_mag_rmk202
  grep "DNAa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/unpolished/${contigs}/PROKKA_*.gff
  seq_length.py $Overall_output_directory/circlatorAfter//splitContigs/${contigs}.fasta
  grep "gene" $Overall_output_directory/circlatorAfter//splitContigs/unpolished/${contigs}/*.gff|tail #1865282
##1865440
  #345399
  
###---------------extract fragemtent 1-------------------
rm -r $Overall_output_directory/circlatorAfter//circlaring_sterm/logs/
mkdir -p $Overall_output_directory/circlatorAfter//circlaring_sterm/logs/

echo -e ${contigs}"\t1\t1865290" > $Overall_output_directory/circlatorAfter/circlaring_sterm/logs/${contigs}_fragment1.bed

samtools faidx $Overall_output_directory/circlatorAfter/splitContigs/${contigs}.fasta

bedtools getfasta -fi $Overall_output_directory/circlatorAfter/splitContigs/${contigs}.fasta \
  -bed $Overall_output_directory/circlatorAfter/circlaring_sterm/logs/${contigs}_fragment1.bed \
  -fo $Overall_output_directory/circlatorAfter/circlaring_sterm/logs/${contigs}_fragment1.fasta

grep -v ">" $Overall_output_directory/circlatorAfter/circlaring_sterm/logs/${contigs}_fragment1.fasta > $Overall_output_directory/circlatorAfter/circlaring_sterm/logs/${contigs}_fragment1_megerable.fasta


###---------------extract fragemtent 3-------------------


#seq_length.py $Overall_output_directory/circlatorAfter/splitContigs/${contigs}.fasta

echo -e ${contigs}"\t1865290\t1865440" > $Overall_output_directory/circlatorAfter//circlaring_sterm/logs//${contigs}_fragment2.bed

#samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring_sterm/input/contig_1.fasta

bedtools getfasta -fi $Overall_output_directory/circlatorAfter/splitContigs/${contigs}.fasta \
  -bed $Overall_output_directory/circlatorAfter//circlaring_sterm/logs//${contigs}_fragment2.bed \
  -fo $Overall_output_directory/circlatorAfter//circlaring_sterm/logs//${contigs}_fragment2.fasta

grep -v ">" $Overall_output_directory/circlatorAfter//circlaring_sterm/logs//${contigs}_fragment2.fasta > $Overall_output_directory/circlatorAfter//circlaring_sterm/logs//${contigs}_fragment2_mergeable.fasta


##-------merging fragments-------
mkdir -p $Overall_output_directory/circlatorAfter//circlaring_sterm/final

echo ">"${contigs} > $Overall_output_directory/circlatorAfter//circlaring_sterm/final/${contigs}_unformated.fasta

cat $Overall_output_directory/circlatorAfter/circlaring_sterm/logs/${contigs}_fragment2_mergeable.fasta \
  $Overall_output_directory/circlatorAfter//circlaring_sterm/logs//${contigs}_fragment1_megerable.fasta | sed 's/\n//g' >>  \
  $Overall_output_directory/circlatorAfter//circlaring_sterm/final/${contigs}_unformated.fasta
  

fasta_formatter -w 50 -i $Overall_output_directory/circlatorAfter//circlaring_sterm/final/${contigs}_unformated.fasta -o $Overall_output_directory/circlatorAfter//circlaring_sterm/final/${contigs}.fasta

###------------------checking with annotation
rm -r $Overall_output_directory/circlatorAfter//circlaring_sterm/final/annoted/${contigs}
mkdir -p $Overall_output_directory/circlatorAfter//circlaring_sterm/final/annoted/${contigs}
    
prokka --outdir $Overall_output_directory/circlatorAfter//circlaring_sterm/final/annoted/${contigs} --force --usegenus --genus Streptococcus --locustag Sterm_202 --proteins proteins.faa --evalue 0.001 --addgenes $Overall_output_directory/circlatorAfter//circlaring_sterm/final/${contigs}.fasta
  
  grep "DNAa" -i $Overall_output_directory/circlatorAfter//circlaring_sterm/final/annoted/${contigs}/PROKKA_*.gff
  


##--------------------------------merge all again
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/startAligned/

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_ldel/final/L_delbrueckii_mag_rmk202.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_sterm/final/S_thermophilus_mag_rmk202.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//splitContigs/L_del_plasmid_01.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//splitContigs/S_thermophilus_repA.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//splitContigs/L_del_phage_01.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//Bos_taurus_mito.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//Bos_taurus_genome_01.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//Bos_taurus_genome_02.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//Bos_taurus_genome_03.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//Bos_taurus_genome_04.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//splitContigs/Bos_taurus_genome_05.fasta > \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//startAligned/di_rmk202_MAG_reference_startaligned_unformated.fasta


fasta_formatter -w 50 -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//startAligned/di_rmk202_MAG_reference_startaligned_unformated.fasta -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//startAligned/di_rmk202_MAG_reference_startaligned.fasta



#/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/splitContigs/Unknown_01.fasta \

###------------------checking with annotation
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//startAligned/annoted/all
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//startAligned/annoted/all
    
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly//circlaring_sterm/final/annoted/all --force --usegenus --genus Streptococcus --locustag Sterm_202 --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//startAligned/di_rmk202_MAG_reference_startaligned.fasta
  
  grep "DNAa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//startAligned/annoted/all/PROKKA_*.gff
  


```

Finally, these start aligned and polished contigs will be merged with the contigs from the plasmdi assembly further down in the script

####remove cow

```{bash}

###----------------------------------devide all contigs
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/08_fourthFreebayes/splitContigs/
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/08_fourthFreebayes/splitContigs/
for contigs in $(grep ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/08_fourthFreebayes/freebayesOuput/di_rmk202_MAG_reference_polished_renamed.fasta |sed 's/>//g' )
  do
  echo ${contigs}
samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/08_fourthFreebayes/freebayesOuput/di_rmk202_MAG_reference_polished_renamed.fasta ${contigs} > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/08_fourthFreebayes/splitContigs//${contigs}.fasta
  done

##---------------------------------------------only keep non cow

mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/Final_Polished_withoutCow/

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/08_fourthFreebayes/splitContigs//L_delbrueckii_mag_rmk202.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/08_fourthFreebayes/splitContigs//S_thermophilus_mag_rmk202.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/08_fourthFreebayes/splitContigs//L_del_plasmid_01.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/08_fourthFreebayes/splitContigs//S_thermophilus_repA.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/08_fourthFreebayes/splitContigs//L_del_phage_01.fasta > \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/Final_Polished_withoutCow/di_rmk202_MAG_reference_polished_unformated.fasta


fasta_formatter -w 50 -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/Final_Polished_withoutCow/di_rmk202_MAG_reference_polished_unformated.fasta -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/Final_Polished_withoutCow/di_rmk202_MAG_reference_polished.fasta


##---------------------------------------------BRING LOCAL

mkdir -p ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_FINALgENOMEassembly/
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/08_fourthFreebayes/splitContigs/S_thermophilus_mag_rmk202.fasta ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_FINALgENOMEassembly/

scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/08_fourthFreebayes/splitContigs/L_delbrueckii_mag_rmk202.fasta ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_FINALgENOMEassembly/


##-------------phage for RAST
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/08_fourthFreebayes/splitContigs/L_del_phage_01.fasta ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_FINALgENOMEassembly/


```



##QC
###### Quast 
Here, I do the analysis of the Quast output

```{r}

library(readr)
library(ggplot2)
library(gridExtra)

quast_K1 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC/quastwithRacon_FreebayesONT/report.tsv","\t", escape_double = FALSE, trim_ws = TRUE)

rownames(quast_K1) <- quast_K1$Assembly
# quast_K1 <- quast_K1[,-1]

quast_K1_analysis <- t(quast_K1)
View(quast_K1_analysis)
quast_K1_analysis <- as.data.frame(quast_K1_analysis[-1,])
quast_K1_analysis$nameSample <- rownames(quast_K1_analysis)

quast_K1_analysis$`# misassemblies` <- as.double(as.character(quast_K1_analysis$`# misassemblies`))
quast_K1_analysis$`Duplication ratio` <- as.double(as.character(quast_K1_analysis$`Duplication ratio`))
quast_K1_analysis$`# mismatches per 100 kbp` <- as.double(as.character(quast_K1_analysis$`# mismatches per 100 kbp`))
quast_K1_analysis$`# indels per 100 kbp` <- as.double(as.character(quast_K1_analysis$`# indels per 100 kbp`))

#table(quast_K1_analysis$nameSample)
#quast_K1_analysis$nameSample <- revalue(quast_K1_analysis$nameSample, c("S50"="mst1", "S72"="mst2", "Streptococcus_thermophilus_RMK202"="RMK202"))
quast_K1_analysis$nameSample_f = factor(quast_K1_analysis$nameSample, levels=c("FlyeAssembly","firstONTracon","secondONTracon","thirdONTracon","fourthONTracon","firstFreebayes","secondFreebayes","thirdFreebayes"))


##Misassemblies
p1 <- ggplot(quast_K1_analysis,aes(x=nameSample_f,y=`# misassemblies`,group=1))+geom_point(size=2)+geom_line()+
  theme_classic()+
  labs(x="",
       y="# Misassemblies")+
  theme(axis.text.x = element_blank())


##Misassemblies
p2 <- ggplot(quast_K1_analysis,aes(x=nameSample_f,y=`Duplication ratio`,group=1))+geom_point(size=2)+geom_line()+
  theme_classic()+
  labs(x="",
       y="Duplication ratio")+
  theme(axis.text.x =element_blank())


##Mismatches
p3 <- ggplot(quast_K1_analysis,aes(x=nameSample_f,y=`# mismatches per 100 kbp`,group=1))+geom_point(size=2)+geom_line()+
  theme_classic()+
  labs(x="",
       y="# Mismatches per 100 kbp")+
  theme(axis.text.x = element_text(angle = 75, hjust = 1))

##INdels
p4 <- ggplot(quast_K1_analysis,aes(x=nameSample_f,y=`# indels per 100 kbp`,group=1))+geom_point(size=2)+geom_line()+
  theme_classic()+
  labs(x="",
       y="# INDELS per 100 kbp")+
  theme(axis.text.x = element_text(angle = 75, hjust = 1))


grid.arrange(p1,p2,p3,p4, ncol=2, heights=c(1,2))

  svg("~/Desktop/presenations/my_presentations/20190708_groupmeeting/figures/polishing_K1.svg",width=6,height=5)
grid.arrange(p1,p2,p3,p4, ncol=2, heights=c(2,3))
 dev.off()


```


###### Ideel

In the following we run [ideel](https://github.com/phiweger/ideel).
Ideel should be run after all polishing steps in order to see the changes after the single steps. 

1. In order to run Ideel we first need to download the (tremble protein database)[https://www.uniprot.org/downloads]. 
...and after clone from github I had to change the location of the tremble.database in the SnakeFile before getting it to work. 

2. change tremble.database location in script

3. change .fa to .fasta in script

Runs incredibly long (days!)
```{bash}

##download Database
mkdir -p /archiv/TREMBL_database/20190710
cd /archiv/TREMBL_database/20190710 
wget ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_trembl.fasta.gz

##make database
cd /archiv/TREMBL_database/20190710 

pigz -d /archiv/TREMBL_database/20190710/uniprot_trembl.fasta.gz

diamond makedb --in uniprot_trembl.fasta.gz  -d uniprot_trembl

##create ideal
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/assembly/startAligned/di_rmk202_MAG_reference_startaligned.fasta 

mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/IDEEL
cd /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/IDEEL

git clone https://github.com/mw55309/ideel

cd ideel 
mkdir -p genomes
cd genomes



cp $Overall_output_directory/08_fourthFreebayes/freebayesOuput/8_fourthFreebayes.fasta .
cp $reference_fasta .
cp $Overall_output_directory/01_firstRacon_graphmap/assembly/firstRacon_NWC_2Assembly.fasta .
cp $Overall_output_directory/02_secondRacon_graphmap/assembly/secondRacon_NWC_2Assembly.fasta .
cp $Overall_output_directory/03_thirdRacon_graphmap/assembly/thirdRacon_NWC_2Assembly.fasta .
cp $Overall_output_directory/04_fourthRacon_graphmap/assembly/fourthRacon_NWC_2Assembly.fasta .
cp $Overall_output_directory/05_firstFreebayes/freebayesOuput/5_firstFreebayes.fasta .
cp $Overall_output_directory/06_secondFreebayes/freebayesOuput/6_secondFreebayes.fasta .
cp $Overall_output_directory/07_thirdFreebayes/freebayesOuput/7_thirdFreebayes.fasta .
cd ..

snakemake

###________________________move local

scp -r vincent@130.223.51.116:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/IDEEL/ideel/hists/ /home/vincent/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/

scp -r vincent@130.223.51.116:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/IDEEL/ideel/lengths/ /home/vincent/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/


```

```{r}

library(readr)
library(ggplot2)

assembly_01 <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/lengths/K1_ONT_assembly_circlor.data",  "\t", escape_double = FALSE, col_names = c("length","expected"), trim_ws = TRUE)
assembly_02 <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/lengths/firstRacon_NWC_2Assembly.data",  "\t", escape_double = FALSE, col_names = c("length","expected"), trim_ws = TRUE)
assembly_03 <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/lengths/secondRacon_NWC_2Assembly.data",  "\t", escape_double = FALSE, col_names = c("length","expected"), trim_ws = TRUE)
assembly_04 <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/lengths/thirdRacon_NWC_2Assembly.data",  "\t", escape_double = FALSE, col_names = c("length","expected"), trim_ws = TRUE)
assembly_05 <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/lengths/fourthRacon_NWC_2Assembly.data",  "\t", escape_double = FALSE, col_names = c("length","expected"), trim_ws = TRUE)
assembly_06 <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/lengths/_fifthFreebayes.data",  "\t", escape_double = FALSE, col_names = c("length","expected"), trim_ws = TRUE)
assembly_07 <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/lengths/_sixthFreebayes.data",  "\t", escape_double = FALSE, col_names = c("length","expected"), trim_ws = TRUE)
assembly_08 <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/lengths/_seventhFreebayes.data",  "\t", escape_double = FALSE, col_names = c("length","expected"), trim_ws = TRUE)
assembly_09 <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/lengths/_eightFreebayes.data",  "\t", escape_double = FALSE, col_names = c("length","expected"), trim_ws = TRUE)


pseudoPlotData <- data.frame(pseudo=c(100-(sum((assembly_01$length/assembly_01$expected)>0.9)/nrow(assembly_01)*100),
100-(sum((assembly_02$length/assembly_02$expected)>0.9)/nrow(assembly_02)*100),
100-(sum((assembly_03$length/assembly_03$expected)>0.9)/nrow(assembly_03)*100),
100-(sum((assembly_04$length/assembly_04$expected)>0.9)/nrow(assembly_04)*100),
100-(sum((assembly_05$length/assembly_05$expected)>0.9)/nrow(assembly_05)*100),
100-(sum((assembly_06$length/assembly_06$expected)>0.9)/nrow(assembly_06)*100),
100-(sum((assembly_07$length/assembly_07$expected)>0.9)/nrow(assembly_07)*100),
100-(sum((assembly_08$length/assembly_08$expected)>0.9)/nrow(assembly_08)*100),
100-(sum((assembly_09$length/assembly_09$expected)>0.9)/nrow(assembly_09)*100)),names=c("FlyeAssembly","firstONTracon","secondONTracon","thirdONTracon","fourthONTracon","firstFreebayes","secondFreebayes","thirdFreebayes","fourthFreebayes"))

pseudoPlotData$nameSample_f = factor(pseudoPlotData$names, levels=c("FlyeAssembly","firstONTracon","secondONTracon","thirdONTracon","fourthONTracon","firstFreebayes","secondFreebayes","thirdFreebayes","fourthFreebayes"))


p5 <- ggplot(pseudoPlotData,aes(x=nameSample_f,y=pseudo,group=1))+geom_point(size=2)+geom_line()+
  theme_classic()+
  ylim(10,100)+
  scale_y_continuous(breaks = c(12,25,50,75,96),labels=c(12,25,50,75,96))+
  labs(x="",
       y="Pseudogenes [%]")+
  theme(axis.text.x = element_text(angle = 75, hjust = 1))
p5


  svg("~/Desktop/presenations/my_presentations/20190708_groupmeeting/figures/polishing_K1_pseudo.svg",width=6,height=3)
p5
 dev.off()


```
######mapping

```{bash}

for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt)
do
threads=38

#names=di_K2_6h

  echo ${num}"/37  :" ${names}
  num=$((num+1))
  rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/

  mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/

  # bwa index /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/Final_Polished_withoutCow/di_rmk202_MAG_reference_polished.fasta
  
  bwa mem -t ${threads} /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/Final_Polished_withoutCow/di_rmk202_MAG_reference_polished.fasta /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_1.fastq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*kneaddata_paired_2.fastq | samtools sort -@${threads} -O BAM -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 

#cd /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/
#samstat /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam

samtools view -b -f 4 /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

bedtools bamtofastq -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam -fq /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq

rm /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

##-----Qualimap

mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/bamqc

qualimap bamqc -bam /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/bamqc --java-mem-size=80G



 mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/log
echo "/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/bamqc" >> /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/log/multiqc_mapping_logfile.txt
   

samtools index /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam &


done

###--------------------------------multiQC

rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL//MultiQC/
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL//MultiQC/


  multiqc --config /archiv/logs/multiQC_config.txt --file-list /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/log/multiqc_mapping_logfile.txt \
  -o archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL//MultiQC/

###--------------------------------move local


names=di_K2_6h
mkdir -p ~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_demupliplex/mapping/${names}
scp -r vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/bamqc ~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_demupliplex/mapping/${names}


```


## plasmid spades of non-mapping

```{bash}


##===============
##merge all non-mapping reads
##===============
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/nonMappingReads_merged.fq
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/
for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt)
do


cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq >> /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/nonMappingReads_merged.fq

done


##===============
##spades plasmid assembly
##===============

mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly

/home/vincent/miniconda2/pkgs/spades-3.13.0-0/bin/spades.py --plasmid -t 37 -m 100 \
  -s /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/nonMappingReads_merged.fq \
  -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly
  
##--------------meta  

mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly_meta

/home/vincent/miniconda2/pkgs/spades-3.13.0-0/bin/spades.py --meta -t 37 -m 100 \
  -s /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/nonMappingReads_merged.fq \
  -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly_meta
  
###--------------------------------move local


names=di_K2_6h
mkdir -p ~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_demupliplex/assembly/
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly/assembly_graph_with_scaffolds.gfa ~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_demupliplex/assembly/${names}_graph.gfa


###======================================single sample unmapped assembly

for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt|grep "_K")
do
echo ${names}
##--------------plasmid spades
rm - r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly_everySample_indiviually_plasmid/${names}/
mkdir -p  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly_everySample_indiviually_plasmid/${names}/
/home/vincent/miniconda2/pkgs/spades-3.13.0-0/bin/spades.py --plasmid -t 37 -m 100 \
  -s /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq \
  -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly_everySample_indiviually_plasmid/${names}/

##--------------meta  
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly_everySample_indiviually_meta/${names}/
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly_everySample_indiviually_meta/${names}/
/home/vincent/miniconda2/pkgs/spades-3.13.0-0/bin/spades.py --plasmid -t 37 -m 100 \
  -s /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq \
  -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly_everySample_indiviually_meta/${names}/


done


###--------------------------------move local


names=di_K2_6h
mkdir -p ~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_demupliplex/assembly/
scp vincent@130.223.51.116:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly_everySample_indiviually_plasmid/di_K2_6h/contigs.fasta ~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_demupliplex/assembly/${names}_constig.fasta

```


#####merge ONT and spades assembly



```{bash}


###-----------rename contigs
sed 's/NODE_4_length_7994_cov_44.859921_component_2$/L_del_plasmid_02/g'   /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly/contigs.fasta > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly/di_rmk202_MAG_plasmidAssembly_renamed.fasta

###-----------rename contigs 02
num=1
for contigs in $(grep ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly/di_rmk202_MAG_plasmidAssembly_renamed.fasta |grep "component_0"|sed 's/>//g')
  do
  echo ${contigs}
  sed -i "s/$contigs/S_term_phage_01_contig$num/g" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly/di_rmk202_MAG_plasmidAssembly_renamed.fasta
  
  #sed "s/$contigs/S_term_phage_01_contig$num/g" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly/di_rmk202_MAG_plasmidAssembly_renamed.fasta |grep ">"
  
  num=$((num+1))
  done
  
  grep ">"  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly/di_rmk202_MAG_plasmidAssembly_renamed.fasta



###-----------devide all contigs
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/splitContigs/
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/splitContigs/
for contigs in $(grep ">"  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly/di_rmk202_MAG_plasmidAssembly_renamed.fasta|grep "NODE" -v |sed 's/>//g' )
  do
  echo ${contigs}
samtools faidx  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly/di_rmk202_MAG_plasmidAssembly_renamed.fasta ${contigs} > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/splitContigs/${contigs}.fasta
  done

###-----------get PGAP .fsa

mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_ldel/final/PGAP
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_sterm/final/PGAP
  
scp /home/vincent/Downloads/L_delbrueckii_RMK202.fasta vincent@130.223.51.116:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_ldel/final/PGAP/
scp /home/vincent/Downloads/S_thermophilus_RMK202.fasta vincent@130.223.51.116:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_sterm/final/PGAP/


###-----------merge PGAP contigs
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/S_thermophilus_RMK202.current.fna \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/L_delbrueckii_RMK202.current.fna \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//splitContigs/L_del_phage_01.fasta \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/splitContigs/*.fasta > \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished_unformated.fasta



fasta_formatter -w 50 -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished_unformated.fasta -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished.fasta





###-----------merge contigs



mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids/

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_ldel/final/L_delbrueckii_mag_rmk202.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_sterm/final/S_thermophilus_mag_rmk202.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//splitContigs/L_del_plasmid_01.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//splitContigs/S_thermophilus_repA.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//splitContigs/L_del_phage_01.fasta \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/splitContigs/*.fasta > \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids/di_rmk202_MAG_reference_polished_unformated.fasta


fasta_formatter -w 50 -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids/di_rmk202_MAG_reference_polished_unformated.fasta -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids/di_rmk202_MAG_reference_polished.fasta

##--------------------PROKKA test
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids/PROKKA_TEST
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids/PROKKA_TEST
    
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids/PROKKA_TEST --force --usegenus --genus Streptococcus --locustag Sterm_202 --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids/di_rmk202_MAG_reference_polished.fasta
  grep -i "dnaa" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids/PROKKA_TEST/*.gff


###-----------merge all phages
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/splitContigs/Sterm_phage
cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/splitContigs/S_t*.fasta > \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/splitContigs/Sterm_phage/S_term_phage_01.fasta

scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/splitContigs/Sterm_phage/S_term_phage_01.fasta ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_FINALgENOMEassembly/
```


###remove remaining cow contigs

```{bash}
##===================
##prepare remaining cow contigs
##===================

###-----------devide all contigs
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/splitContigs_cow/
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/splitContigs_cow/
for contigs in $(grep ">"  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly/di_rmk202_MAG_plasmidAssembly_renamed.fasta|grep "NODE" |sed 's/>//g' )
  do
  echo ${contigs}
samtools faidx  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/asssembly/di_rmk202_MAG_plasmidAssembly_renamed.fasta ${contigs} > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/splitContigs_cow/${contigs}.fasta
  done
##===================
##build DB
##===================
#mkdir -p /archiv/cowDB/mixed/20190520/

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/splitContigs_cow/*.fasta > \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/splitContigs_cow/cow_genomeANDmito_DB_remain.fna


bowtie2-build /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/splitContigs_cow/cow_genomeANDmito_DB_remain.fna /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/splitContigs_cow/cow_genomeANDmito_DB_remain

##===================
##remove reads
##===================
threads=37

num=1
for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt)
#for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/log/20191031_logDownload_changeName.txt)
  do
  

  echo ${num}"/37  :" ${names}
  num=$((num+1))
rm -r /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata_Remaining/${names}
mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata_Remaining/${names}

#gunzip /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R1_val_1.fq.gz
#gunzip /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R2_val_2.fq.gz


kneaddata --max-memory 90000m --no-discordant -t 37 -p 37 --input /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_1.fastq --input /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*kneaddata_paired_2.fastq  \
  -db /archiv/cowDB/mixed/20190520/cow_genomeANDmito_DB  \
  --trimmomatic /home/vincent/Joanito/brain/Software/Trimmomatic-0.39/ \
  --output /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata_Remaining/${names}/${names}

done

```


##Final annoation

The final annoation will be done with:
- PGAP for Bacteria, Plasmids and the cow mito
- phanotate for the phages
- preliminary prokka

##### PGAP

make PGAP annotation by submitting in the [submission portal](https://submit.ncbi.nlm.nih.gov/).

```{bash}


##-------------------------
##prepare files for PGAP
##-------------------------
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/PGAP_prep/
##----------------------Sterm

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_sterm/final/S_thermophilus_mag_rmk202.fasta \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//splitContigs/S_thermophilus_repA.fasta > \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/PGAP_prep/S_thermophilus_RMK202.fasta
  
##----------------------Ldel

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_ldel/final/L_delbrueckii_mag_rmk202.fasta \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//splitContigs/L_del_plasmid_01.fasta \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/splitContigs/L_del_plasmid_02.fasta > \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/PGAP_prep/L_delbrueckii_RMK202.fasta
  
##------------------------------------BRING LOCAL


mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/assembly/  

scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/PGAP_prep/L_delbrueckii_RMK202.fasta /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/assembly/


mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/assembly/  

scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/PGAP_prep/S_thermophilus_RMK202.fasta /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/assembly/


```

getting files from PGAG NCBI submission portal and moving to bigmachine for further analysis. 





##### Prokka

First we however do PROKKA for all the genomes indiviually.

```{bash}


###================
##description file
###================

Outputlocation=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/03_PROKKA_annotation_polishedGenome
Assembly_sterm=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_sterm/final/S_thermophilus_mag_rmk202.fasta
Assembly_ldel=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_ldel/final/L_delbrueckii_mag_rmk202.fasta
BaseFILNAME_ldel=L_delbrueckii_mag_rmk202
BaseFILNAME_sterm=S_thermophilus_mag_rmk202
locusTagSterm=Sterm_202
locusTagLdel=Ldel_202
Genus_Sterm=Streptococcus
Genus_Ldel=Lactobacillus
threads=37


###================
##script
###================



##===============
##annotaion
##===============


##====================================================================================-Streptococcus_thermophilus================

rm -r ${Outputlocation}/${BaseFILNAME_sterm}
    mkdir -p ${Outputlocation}/${BaseFILNAME_sterm}
    
prokka --outdir ${Outputlocation}/${BaseFILNAME_sterm}/ --force --usegenus --genus ${Genus_Sterm} --locustag ${locusTagSterm} --proteins proteins.faa --evalue 0.001 --addgenes ${Assembly_sterm}


#grep "dnaa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/03_PROKKA_annotation_polishedGenome/${names}/*.gff


##====================================================================================-Lactobacillus delbrueckii================

rm -r ${Outputlocation}/${BaseFILNAME_ldel}
    mkdir -p ${Outputlocation}/${BaseFILNAME_ldel}
    
prokka --outdir ${Outputlocation}/${BaseFILNAME_ldel}/ --force --usegenus --genus ${Genus_Ldel} --locustag ${locusTagLdel} --proteins proteins.faa --evalue 0.001 --addgenes ${Assembly_ldel}


#grep "dnaa" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/03_PROKKA_annotation_polishedGenome/${names}/*.gff

###-----------scp-------------------------------------------------------------------------------

###-----------------local
  
mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_RMK202/annotated/{CDS,eggnog}  

names=S_thermophilus_mag_rmk202
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/03_PROKKA_annotation_polishedGenome/${names}/*.faa /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_RMK202/annotated/CDS/${names}.faa


###-----------------local
  
mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_RMK202/annotated/{CDS,eggnog}  

names=L_delbrueckii_mag_rmk202
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/03_PROKKA_annotation_polishedGenome/${names}/*.faa /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_RMK202/annotated/CDS/${names}.faa
    
```

looking into Prokka annoation

```{bash}

ll /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/

##Lactobacillus_delbrueckii_RMK202
gff_location=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_delbrueckii_RMK202/*gff

grep "dnaa" -i ${gff_location}
grep "pseudogene" -i ${gff_location}
grep "transposase" -i ${gff_location}
grep "16S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "23S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "5S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep " ribosomal RNA" -i ${gff_location} 


##Lactobacillus_delbrueckii_RMK202
gff_location=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Streptococcus_thermophilus_RMK202/*gff

grep "dnaa" -i ${gff_location}
grep "pseudogene" -i ${gff_location}
grep "transposase" -i ${gff_location}
grep "16S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "23S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "5S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep " ribosomal RNA" -i ${gff_location} 


##S72
gff_location=/archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/*gff

grep "dnaa" -i ${gff_location}
grep "pseudogene" -i ${gff_location}
grep "transposase" -i ${gff_location}
grep "16S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "23S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "5S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep " ribosomal RNA" -i ${gff_location} 


##S50
gff_location=/archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/*gff

grep "dnaa" -i ${gff_location}
grep "pseudogene" -i ${gff_location}
grep "transposon" -i ${gff_location}
grep "16S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "23S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "5S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep " ribosomal RNA" -i ${gff_location} 



#L104
gff_location=/archiv/Projects/2018_Culturomics/06_Spades/L104/Prokka/20190203/*gff

grep "dnaa" -i ${gff_location}
grep "pseudogene" -i ${gff_location}
grep "transposon" -i ${gff_location}
grep "16S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "23S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "5S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep " ribosomal RNA" -i ${gff_location} 


```



####eggnog annoation

All assemblies were annotated on the [eggnog homepage](http://eggnog-mapper.embl.de/)
After the annoation the files are downloaded locally and analysed in R.

genes of interest

Transporter: GO:0140104 (molecular carrier activity), GO:0005215 (transporter activity)

The different annotation options:
1. [KEGG](https://www.genome.jp/kegg/kegg1a.html) has many different annotations schems:
  - Probably the most interesting one is [KO (KEGG ORTHOLOGY) Database](https://www.genome.jp/kegg/ko.html).
  - [KEGG modules](https://www.genome.jp/kegg/module.html) less grained the Brite
  - Brite
  
  
Do the mapping of the annotations [here](https://www.genome.jp/kegg/tool/map_pathway.html)
  


```{bash}

scp /home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/phage_ldel/L_delbrueckii_RMK202.annotations vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel//eggnog/

scp /home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/phage_sterm/S_thermophilus_RMK202.annotations vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm//eggnog/
##=================
##create mappable files
##=================

##Ldel  
awk -F "\t" 'BEGIN{OFS="\t"} {print $1, $9}' /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Lactobacillus_delbrueckii_RMK202_pgap_query_seqs.fa.emapper.annotations |sed 's/ko://g' |grep "^#" -v > /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog_Kegg/Lactobacillus_delbrueckii_RMK202_pgap_query_seqs.fa.emapper.annotations

grep -v "^#" /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Lactobacillus_delbrueckii_RMK202_pgap_query_seqs.fa.emapper.annotations | awk -F "\t" 'BEGIN{OFS="\t"} {print $9}'  |sed 's/ko://g' > /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog_Kegg/Lactobacillus_delbrueckii_RMK202_pgap_query_seqs.fa.emapper.annotations

```


I noticed that i first have to seperate the Brite columns with multiple annoations in order to get their annoation

```{r}
library(thacklr)
library(readr)
library(dplyr)
library(tidyverse)
library(tidyr)

S50 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/S50_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3)) %>% add_column(.,sample="S50")

Lactobacillus_delbrueckii_RMK202 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Lactobacillus_delbrueckii_RMK202_pgap_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3)) %>% add_column(.,sample="Lactobacillus_delbrueckii_RMK202")
colnames(S50) <- c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description","genome")


colnames(Lactobacillus_delbrueckii_RMK202) <- c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description","genome")


## The ";\\s+" means that the separator is a ";" followed by one or more spaces


Lactobacillus_delbrueckii_RMK202_subset4Brite <- Lactobacillus_delbrueckii_RMK202[,c("query_name","KEGG_ko")]
# Lactobacillus_delbrueckii_RMK202_subset4Brite$KEGG_ko <- gsub("ko:","",Lactobacillus_delbrueckii_RMK202_subset4Brite$KEGG_ko)
Lactobacillus_delbrueckii_RMK202_subset4Brite <- separate_rows(Lactobacillus_delbrueckii_RMK202_subset4Brite,KEGG_ko,sep=",")

write.table(Lactobacillus_delbrueckii_RMK202_subset4Brite, "~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog_Kegg/Lactobacillus_delbrueckii_RMK202_subset4Brite_pgap_query_seqs.fa.emapper.annotations",na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names = FALSE)

```


#### Circos bacteria

Here, I creat the circos plots for the bacteria. 
I will include the following information:
1. genome (circul)
2. forward genes  
3. reverse genes
2.2. tRNA and rRNA colored
3.2. tRNA and rRNA colored
4. Pseudogenes
5. Prophage location
6. GC-skew (https://dbsloan.github.io/TS2019/exercises/circos.html#add-gc-skew-data-to-the-plot)
```{bash}

##==================
##Sterm
##==================
home=/home/vincent/bin/apps/circos-0.69-6/
cd $home

##organisational 
#cp /home/vincent/bin/apps/circos-0.69-6/etc/ticks_nwc_1_ldel_both.conf /home/vincent/bin/apps/circos-0.69-6/etc/ticks_mag_both.conf

#cp /home/vincent/bin/apps/circos-0.69-6/etc/ideogram.conf /home/vincent/bin/apps/circos-0.69-6/etc/ideogram_mag.conf
mkdir -p $home/data/rmk202/MAG_rmk202_sterm/
##---------------------karyotyp
#seqlength.py "/home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202 (2).fasta" |head -1 > ${home}/data/karyotype/sterm_magg_rmk202.txt
##have to change the format a bit

##---------------------genes
cd /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm
/usr/bin/perl /home/vincent/miniconda3/bin/bp_genbank2gff3.pl S_thermophilus_RMK202.current.gb
cd $home
grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |awk -F "\t" '{OFS="\t"}{if($1=="CP046134" && $3=="CDS" && $7=="+") print "S_thermophilus_mag_rmk202",$4,$5}' > $home/data/rmk202/MAG_rmk202_sterm/genes_forward.txt
grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |awk -F "\t" '{OFS="\t"}{if($1=="CP046134" && $3=="CDS" && $7=="-") print "S_thermophilus_mag_rmk202",$4,$5}' > $home/data/rmk202/MAG_rmk202_sterm/genes_reverse.txt

##---------------------rRNA

grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |awk -F "\t" '{OFS="\t"}{if($1=="CP046134" && $3=="rRNA" && $7=="+") print "S_thermophilus_mag_rmk202",$4,$5}' > $home/data/rmk202/MAG_rmk202_sterm/rRNA_forward.txt
grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |awk -F "\t" '{OFS="\t"}{if($1=="CP046134" && $3=="rRNA" && $7=="-") print "S_thermophilus_mag_rmk202",$4,$5}' > $home/data/rmk202/MAG_rmk202_sterm/rRNA_reverse.txt

##---------------------tRNA

grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |awk -F "\t" '{OFS="\t"}{if($1=="CP046134" && $3=="tRNA" && $7=="+") print "S_thermophilus_mag_rmk202",$4,$5}' > $home/data/rmk202/MAG_rmk202_sterm/tRNA_forward.txt
grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |awk -F "\t" '{OFS="\t"}{if($1=="CP046134" && $3=="tRNA" && $7=="-") print "S_thermophilus_mag_rmk202",$4,$5}' > $home/data/rmk202/MAG_rmk202_sterm/tRNA_reverse.txt

##---------------------transposase from PGAP

grep "transposase" -i /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |awk -F "\t" '{OFS="\t"}{if($1=="CP046134") print "S_thermophilus_mag_rmk202",$4,$5}' > $home/data/rmk202/MAG_rmk202_sterm/transposase.txt

##---------------------pseudogenes

grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |awk -F "\t" '{OFS="\t"}{if($1=="CP046134" && $3=="pseudogene" ) print "S_thermophilus_mag_rmk202",$4,$5}' > $home/data/rmk202/MAG_rmk202_sterm/pseudogenes.txt
#grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |awk -F "\t" '{OFS="\t"}{if($1=="CP046134" && $3=="tRNA" && $7=="-") print "S_thermophilus_mag_rmk202",$4,$5}' > $home/data/rmk202/MAG_rmk202_sterm/tRNA_reverse.txt



##---------------------prophages
awk -F "\t" '{OFS="\t"}{if($1=="CP046134") print "S_thermophilus_mag_rmk202",$4,$5}'  /home/vincent/Desktop/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/prophage_summary_onlyGenome.gff > $home/data/rmk202/MAG_rmk202_sterm/prophages.txt

##---------------------protease

#only in l. delbrueckii 
#I checked online on NCBI if the genomes contain PrtS or PrtB

#/home/vincent/Downloads/prtS_Sterm.fasta

grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |grep "S8 family serine peptidase" |awk -F "\t" '{OFS="\t"}{if($1=="CP046134") print "S_thermophilus_mag_rmk202",$4,$5}'> $home/data/rmk202/MAG_rmk202_sterm/protease.txt
##---------------------transporter


grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |grep "transporter" |awk -F "\t" '{OFS="\t"}{if($1=="CP046134") print "S_thermophilus_mag_rmk202",$4,$5}' > $home/data/rmk202/MAG_rmk202_sterm/transporter.txt


##---------------------CRISPR arrays

scp vincent@130.223.51.116:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/S_thermophilus_RMK202.pilarCR_out $home/data/rmk202/MAG_rmk202_sterm/
#grep "SUMMARY BY POSITION" -A 50 $home/data/rmk202/MAG_rmk202_sterm/S_thermophilus_RMK202.pilarCR_out|grep "^====" -A 50 | sed '1d' | sed 's/^ *//g'| sed 's/ \{1,\}/\t/g'|awk -F "\t" '{OFS="\t"}{print "S_thermophilus_mag_rmk202",$3,$3+$4,"fill_color=blue"}' >  $home/data/rmk202/MAG_rmk202_sterm/CRISPR.txt
##--cas genes

#grep "Cas" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff
grep "CRISPR" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff  |awk -F "\t" '{OFS="\t"}{if($1=="CP046134" && $3=="CDS" ) print "S_thermophilus_mag_rmk202",$4,$5,"color=chr3"}' >  $home/data/rmk202/MAG_rmk202_sterm/CRISPR.txt

grep "CRISPR" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff  |awk -F "\t" '{OFS="\t"}{if($1=="CP046134" && $3=="repeat_region" ) print "S_thermophilus_mag_rmk202",$4,$5,"color=chr2"}' >>  $home/data/rmk202/MAG_rmk202_sterm/CRISPR.txt

##---------------------gc skew

GCcalc.py -f "/home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202 (2).fasta" > $home/data/rmk202/MAG_rmk202_sterm/gc_skew_sterm.txt
awk -F "\t" '{OFS="\t"}{if($5>0) print $1,$2,$2,$5,"fill_color=blue" ; else print $1,$2,$2,$5,"fill_color=orange"}' $home/data/rmk202/MAG_rmk202_sterm/gc_skew_sterm.txt > 
$home/data/rmk202/MAG_rmk202_sterm/gc_skew_sterm_cleaned.txt

#scp -r vincent@130.223.51.116:/usr/bin/seq_length.py ~

##example
cp etc/repeat_nwc1_sterm_01.conf etc/sterm_rmk202_mag_circos.conf

#example run
bin/circos -conf etc/sterm_rmk202_mag_circos.conf -outputfile ./Sterm_mag_rmk202.svg; firefox ./Sterm_mag_rmk202.svg &


##==================
##Ldel
##==================
home=/home/vincent/bin/apps/circos-0.69-6/
cd $home

##organisational 
#cp /home/vincent/bin/apps/circos-0.69-6/etc/ticks_nwc_1_ldel_both.conf /home/vincent/bin/apps/circos-0.69-6/etc/ticks_mag_both.conf ##already done
#cp /home/vincent/bin/apps/circos-0.69-6/etc/ideogram.conf /home/vincent/bin/apps/circos-0.69-6/etc/ideogram_mag.conf ##already done
#cp /home/vincent/bin/apps/circos-0.69-6/etc/sterm_rmk202_mag_circos.conf /home/vincent/bin/apps/circos-0.69-6/etc/ldel_rmk202_mag_circos.conf
mkdir -p $home/data/rmk202/MAG_rmk202_ldel/
##---------------------karyotyp

#seqlength.py /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.fasta |head -1 > ${home}/data/karyotype/ldel_magg_rmk202.txt


##have to change the format a bit

##---------------------genes
cd /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel
/usr/bin/perl /home/vincent/miniconda3/bin/bp_genbank2gff3.pl L_delbrueckii_RMK202.current.gb
cd $home
grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.current.gb.gff |awk -F "\t" '{OFS="\t"}{if($1=="CP046131" && $3=="CDS" && $7=="+") print "L_delbrueckii_mag_rmk202",$4,$5}' > $home/data/rmk202/MAG_rmk202_ldel/genes_forward.txt
grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.current.gb.gff |awk -F "\t" '{OFS="\t"}{if($1=="CP046131" && $3=="CDS" && $7=="-") print "L_delbrueckii_mag_rmk202",$4,$5}' > $home/data/rmk202/MAG_rmk202_ldel/genes_reverse.txt

##---------------------rRNA

grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.current.gb.gff |awk -F "\t" '{OFS="\t"}{if($1=="CP046131" && $3=="rRNA" && $7=="+") print "L_delbrueckii_mag_rmk202",$4,$5}' > $home/data/rmk202/MAG_rmk202_ldel/rRNA_forward.txt
grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.current.gb.gff |awk -F "\t" '{OFS="\t"}{if($1=="CP046131" && $3=="rRNA" && $7=="-") print "L_delbrueckii_mag_rmk202",$4,$5}' > $home/data/rmk202/MAG_rmk202_ldel/rRNA_reverse.txt


##---------------------transposase from PGAP

grep "transposase" -i /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.current.gb.gff |awk -F "\t" '{OFS="\t"}{if($1=="CP046131") print "L_delbrueckii_mag_rmk202",$4,$5}' > $home/data/rmk202/MAG_rmk202_ldel/transposase.txt

##---------------------pseudogenes

grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.current.gb.gff |awk -F "\t" '{OFS="\t"}{if($1=="CP046131" && $3=="pseudogene" ) print "L_delbrueckii_mag_rmk202",$4,$5}' > $home/data/rmk202/MAG_rmk202_ldel/pseudogenes.txt
#grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.current.gb.gff |awk -F "\t" '{OFS="\t"}{if($1=="CP046134" && $3=="tRNA" && $7=="-") print "L_delbrueckii_mag_rmk202",$4,$5}' > $home/data/rmk202/MAG_rmk202_ldel/tRNA_reverse.txt



##---------------------prophages
awk -F "\t" '{OFS="\t"}{if($1=="CP046134") print "L_delbrueckii_mag_rmk202",$4,$5}'  /home/vincent/Desktop/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/prophage_summary_onlyGenome.gff > $home/data/rmk202/MAG_rmk202_ldel/prophages.txt
##---------------------CRISPR arrays


#grep "Cas" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.current.gb.gff
grep "CRISPR" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.current.gb.gff  |awk -F "\t" '{OFS="\t"}{if($1=="CP046131" && $3=="CDS" ) print "L_delbrueckii_mag_rmk202",$4,$5,"color=chr3"}' >  $home/data/rmk202/MAG_rmk202_ldel/CRISPR.txt

grep "CRISPR" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.current.gb.gff  |awk -F "\t" '{OFS="\t"}{if($1=="CP046131" && $3=="repeat_region" ) print "L_delbrueckii_mag_rmk202",$4,$5,"color=chr2"}' >>  $home/data/rmk202/MAG_rmk202_ldel/CRISPR.txt


##---------------------protease

#only in l. delbrueckii 
#I checked online on NCBI if the genomes contain PrtS or PrtB

#/home/vincent/Downloads/prtS_Sterm.fasta

grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.current.gb.gff |grep "S8 family serine peptidase" |awk -F "\t" '{OFS="\t"}{if($1=="CP046131") print "L_delbrueckii_mag_rmk202",$4,$5}'> $home/data/rmk202/MAG_rmk202_ldel/protease.txt
##---------------------transporter


grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.current.gb.gff |grep "transporter" |awk -F "\t" '{OFS="\t"}{if($1=="CP046131") print "L_delbrueckii_mag_rmk202",$4,$5}' > $home/data/rmk202/MAG_rmk202_ldel/transporter.txt



##---------------------gc skew

GCcalc.py -f /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.fasta > $home/data/rmk202/MAG_rmk202_ldel/gc_skew_ldel.txt
awk -F "\t" '{OFS="\t"}{if($5>0) print $1,$2,$2,$5,"fill_color=blue" ; else print $1,$2,$2,$5,"fill_color=orange"}' $home/data/rmk202/MAG_rmk202_ldel/gc_skew_ldel.txt > $home/data/rmk202/MAG_rmk202_ldel/gc_skew_ldel_cleaned.txt


#scp -r vincent@130.223.51.116:/usr/bin/seq_length.py ~

##example
#cp etc/repeat_nwc1_sterm_01.conf etc/ldel_rmk202_mag_circos.conf

#example run
bin/circos -conf etc/ldel_rmk202_mag_circos.conf -outputfile ./ldel_mag_rmk202.svg; firefox ./ldel_mag_rmk202.svg &

```

#####Galactose genes

Here, I look at what galactose genes are annoted. 

By blasting Streptococcus thermophilus GalR (Accession U61402.1 ) I found a region in our mag Sterm assembly I want to look at the annotated genes therein.

The lactose permease is in the same operon as the galactose enzymes

```{bash}


echo -e "CP046134\t572000\t579000" >/home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/tmp.txt
head -7389 /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff|grep "^#" -v > /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/tmp2.txt
bedtools intersect -b /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/tmp.txt -a  /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/tmp2.txt


grep "gal" -i /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |grep "^CP0"

grep "lactose permease" -A 20 -B 40 -i /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |grep "^CP0"|grep "CDS"

grep "phosphotransferase" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff

grep "pts"  /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff
  
grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.annotations |cut -f 9|sed 's/,/\n/g' | sed '/^$/d' > /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.annotations_KEGG

####------------------------
##transporter genes
####------------------------

grep "^CP0" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |grep "gal" -i 


####------------------------
## Leloir pathway enzymes
##is one operon
####------------------------
##transporter
grep "^CP0" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |grep "Galt" -i 

grep "^CP0" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |grep "GalK" -i 

grep "^CP0" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |grep "GalM" -i 


grep "^CP0" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |grep "GalT" -i 


grep "^CP0" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |grep "GalE" -i 

##!!!!!!!!!!!!!!!!!!!!!!!!
##dieser Stamm scheint eine hexose transferase zu haben
grep "^CP0" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |grep "Gal" -i |grep "transf" -i


####------------------------
## lactose to glucose+galactose
####------------------------

#tagatose
grep "^CP0" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |grep "lacg" -i 

# leloir

grep "^CP0" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |grep "lacZ" -i 

grep "^CP0" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |grep "laclm" -i 


####------------------------
## tagatose pathway
####------------------------

grep "^CP0" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |grep "isomerase" -i |grep -i "galactose"


grep "^CP0" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |grep "histidine-containin"

##!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
##pseudogenic
grep "^CP0" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |grep "phosphotransferase" |grep -i "phosphoenolpyruvate"

grep "^CP0" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |grep "phosphoenolpyruvate--protein phosphotransferase" -A 40 -B 40 |grep "CDS" |awk -F "[\t]" '{OFS="\t"}{print $9}'


```
one of the two phosphoenolpyruvate-dependent sugar phosphotransferase system is mutated and pseudogenised. Therefore this might not be usuable anymore and could explain the non-functioning of galactose uptake?

####Proteolytic system

```{bash}


grep "^CP0" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff|grep "prts"  -i

grep "^CP0" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff|grep "prt" -i 

grep "^CP0" -i  /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff|grep "ami" -i

grep "^CP0" -i  /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff|grep "peptidase" -i

grep "^CP0" -i  /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff|grep "proteinase" -i

##SrtA sortase (anchor of prtS)
grep "^CP0" -i  /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff|grep "SrtA" i-
grep "^CP0" -i  /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff|grep "sortase" -i


###---------------serine protease
mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/protease/
grep "^CP0" -i  /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff|grep "serine protease" -i |awk -F "[\t;]" '{OFS="\t"}{print "S_thermophilus_mag_rmk202",$4,$5,$15}' > /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/protease/SerineProtease.bed


bedtools getfasta -s -name  -fi  /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/'S_thermophilus_RMK202 (2).fasta' \
  -bed /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/protease/SerineProtease.bed > /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/protease/SerineProtease.fasta
  
transeq  -sequence /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/protease/SerineProtease.fasta > \
  -outseq  /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/protease/SerineProtease.faa

##----------------------------------------------------------

grep "^CP0" -i  /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/*gb.gff|grep "prtb" -i

grep "^CP0" -i  /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/*gb.gff|grep "proteinase" -i



```


###peptidases

On the other hand maybe peptidases and trasport is interesting


```{bash}

12 cytoplasmic peptidases (pepA, pepC, pepF, pepM, pepN, pepO, pepP, pepQ, pepS, pepT, pepV, pepX

##=============================================================
##Ldel
##=============================================================
####--------------------------
#peptidasese
####--------------------------

speiceisss=$(echo /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/*gb.gff)

grep "^CP0" -i  /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/*gb.gff|grep "pep" -i |grep -i -c "peptidase"
grep "^CP0" -i  /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/*gb.gff|grep "pep" -i |grep -i "peptidase"
grep "^CP0" -i  ${speiceisss} |grep "pepX" -i
grep "^CP0" -i  ${speiceisss} |grep "pepN" -i
grep "^CP0" -i  ${speiceisss} |grep "pepL" -i
grep "^CP0" -i  ${speiceisss} |grep "pepc" -i
grep "^CP0" -i  ${speiceisss} |grep "pep" -i

####--------------------------
#transporter
####--------------------------
speiceisss=$(echo /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/*gb.gff)

grep "^CP0" -i  /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/*gb.gff|grep "pep" -i |grep -i -c "transporter"
grep "^CP0" -i  /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/*gb.gff|grep "pep" -i |grep -i "transporter"



##=============================================================
##Sterm
##=============================================================
####--------------------------
#peptidasese
####--------------------------
speiceisss=$(echo /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff)

grep "^CP0" -i  /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff|grep "pep" -i |grep -i "peptidase"
grep "^CP0" -i  /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff|grep "pep" -i |grep -i -c "peptidase"

####--------------------------
#transporter
####--------------------------
grep "^CP0" -i  /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff|grep "pep" -i |grep -i "transporter"
grep "^CP0" -i  /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff|grep "pep" -i |grep -i -c "transporter"


/home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff
```


##Phage annotation

###phage annotation graph

prepare for aligment blast

```{bash}
cd /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/Prodigal

sed 's/ # /-/g' Streptococcus_phage_1_startAligned_Final_cleaned.faa | sed 's/-1-.*$//g' > Streptococcus_phage_1_startAligned_Final_cleaned_readyBLAST.faa
sed 's/ # /-/g' Streptococcus_phage_2_startAligned_Final_cleaned.faa | sed 's/-1-.*$//g' > Streptococcus_phage_2_startAligned_Final_cleaned_readyBLAST.faa


sed 's/ \[locus.*location=/-/g'  s_term_sw30.faa |sed 's/\].*$//g'|sed 's/\.\./-/g' > s_term_sw30_readyBLAST.faa
sed 's/ \[locus.*location=/-/g'  s_term_9874.faa |sed 's/\].*$//g'|sed 's/\.\./-/g' > s_term_9874_readyBLAST.faa



##-------------after alignment
cd /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/alignments
grep "hits found" -A 1 Sterm_1_vs_2.txt |grep "^#" -v |grep "^-" -v |cut -f 1,2,3 > Sterm_1_vs_2_cleaned.txt

grep "hits found" -A 1 Sterm_2_vs_sw30.txt |grep "^#" -v |grep "^-" -v |cut -f 1,2,3 > Sterm_2_vs_sw30_cleaned.txt

grep "hits found" -A 1 Sterm_9874_vs_1.txt |grep "^#" -v |grep "^-" -v |cut -f 1,2,3 > Sterm_9874_vs_1_cleaned.txt

```
download hit table (txt)


```{r}


data(three_genes)
comparisons[[1]]$col <- apply_color_scheme(c(0.6, 0.4, 0.5), "grey")
comparisons
##-------------------------
##Streptococcus_phage_Sterm_1_vs_2
##-------------------------
library(readr)
Sterm_1_vs_2_cleaned <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/alignments/Sterm_1_vs_2_cleaned.txt",  "\t", escape_double = FALSE, col_names = c("query","subject","score"),  trim_ws = TRUE)

Sterm_1_vs_2_cleaned$start1 <- str_split_fixed(Sterm_1_vs_2_cleaned$query, fixed("-"), 4)[,2]%>% as.numeric()
Sterm_1_vs_2_cleaned$end1 <- str_split_fixed(Sterm_1_vs_2_cleaned$query, fixed("-"), 4)[,3]%>% as.numeric()

Sterm_1_vs_2_cleaned$start2 <- str_split_fixed(Sterm_1_vs_2_cleaned$subject, fixed("-"), 4)[,2]%>% as.numeric()
Sterm_1_vs_2_cleaned$end2 <- str_split_fixed(Sterm_1_vs_2_cleaned$subject, fixed("-"), 4)[,3]%>% as.numeric()

Sterm_1_vs_2_cleaned$col <- ifelse(Sterm_1_vs_2_cleaned$score>95,"black","grey")

Sterm_1_vs_2_cleaned_final <- Sterm_1_vs_2_cleaned %>%  add_column(.,"direction"="1") %>% filter(score>80) %>% select(start1,end1,start2,end2,direction,col) %>% as.comparison()

##-------------------------
##Streptococcus_phage_Sterm_2_vs_sw30
##-------------------------
library(readr)
Sterm_2_vs_sw30_cleaned <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/alignments/Sterm_2_vs_sw30_cleaned.txt",  "\t", escape_double = FALSE, col_names = c("query","subject","score"),  trim_ws = TRUE)

Sterm_2_vs_sw30_cleaned$start1 <- str_split_fixed(Sterm_2_vs_sw30_cleaned$query, fixed("-"), 4)[,2]%>% as.numeric()
Sterm_2_vs_sw30_cleaned$end1 <- str_split_fixed(Sterm_2_vs_sw30_cleaned$query, fixed("-"), 4)[,3]%>% as.numeric()

Sterm_2_vs_sw30_cleaned$start2 <- str_split_fixed(Sterm_2_vs_sw30_cleaned$subject, fixed("-"), 4)[,2]%>% as.numeric()
Sterm_2_vs_sw30_cleaned$end2 <- str_split_fixed(Sterm_2_vs_sw30_cleaned$subject, fixed("-"), 4)[,3]%>% as.numeric()

Sterm_2_vs_sw30_cleaned$col <- ifelse(Sterm_2_vs_sw30_cleaned$score>95,"black","grey")

Sterm_2_vs_sw30_cleaned_final <- Sterm_2_vs_sw30_cleaned %>%  add_column(.,"direction"="1") %>% filter(score>80) %>% select(start1,end1,start2,end2,direction,col) %>% as.comparison()


##-------------------------
##Streptococcus_phage_Sterm_9874_vs_1
##-------------------------
library(readr)
Sterm_9874_vs_1_cleaned <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/alignments/Sterm_9874_vs_1_cleaned.txt",  "\t", escape_double = FALSE, col_names = c("query","subject","score"),  trim_ws = TRUE)

Sterm_9874_vs_1_cleaned$start1 <- str_split_fixed(Sterm_9874_vs_1_cleaned$query, fixed("-"), 4)[,2] %>% as.numeric()
Sterm_9874_vs_1_cleaned$end1 <- str_split_fixed(Sterm_9874_vs_1_cleaned$query, fixed("-"), 4)[,3]%>% as.numeric()

Sterm_9874_vs_1_cleaned$start2 <- str_split_fixed(Sterm_9874_vs_1_cleaned$subject, fixed("-"), 4)[,2]%>% as.numeric()
Sterm_9874_vs_1_cleaned$end2 <- str_split_fixed(Sterm_9874_vs_1_cleaned$subject, fixed("-"), 4)[,3]%>% as.numeric()

Sterm_9874_vs_1_cleaned$col <- ifelse(Sterm_9874_vs_1_cleaned$score>95,"black","grey")

Sterm_9874_vs_1_cleaned_final <- Sterm_9874_vs_1_cleaned %>%  add_column(.,"direction"="1") %>% filter(score>80) %>% select(start1,end1,start2,end2,direction,col) %>% as.comparison()

##-----------------
##bring togethere
##-----------------
comparisons_mine <- list(Sterm_9874_vs_1_cleaned_final,Sterm_1_vs_2_cleaned_final,Sterm_2_vs_sw30_cleaned_final)


```


I annotated the phages by blasting all the proteins to the blast DB. Then I manually transfered the information. 
Now I will create a fill for genoplotR. 

```{bash}
cd /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/annotation/
echo -e "name\tstart\tend\tstrand\tcol\tfill\tgene_type" > Streptococcus_virus_9874_for_genoplotR.gff
awk -F " " '{OFS="\t"}{print $9 $10 $11 $12 $13 $14 $15,$4,$5,"1","black","arrows"}' Streptococcus_virus_9874_forR.gff | sed 's/product=.*;group=//g' |awk -F "\t" '{OFS="\t"}{print $1,$2,$3,$4,$5,$1,$6}' >> Streptococcus_virus_9874_for_genoplotR.gff

###--------------------

echo -e "name\tstart\tend\tstrand\tcol\tfill\tgene_type" > Streptococcus_phage_1_startAligned_Final_for_genoplotR.gff

awk -F " " '{OFS="\t"}{print $9 $10 $11 $12 $13 $14 $15,$4,$5,"1","black","arrows"}' Streptococcus_phage_1_startAligned_Final_forR.gff | sed 's/ID=.*;group=//g' |awk -F "\t" '{OFS="\t"}{print $1,$2,$3,$4,$5,$1,$6}' >> Streptococcus_phage_1_startAligned_Final_for_genoplotR.gff

###--------------------

echo -e "name\tstart\tend\tstrand\tcol\tfill\tgene_type" > Streptococcus_phage_2_startAligned_Final_for_genoplotR.gff

awk -F " " '{OFS="\t"}{print $9 $10 $11 $12 $13 $14 $15,$4,$5,"1","black","arrows"}' Streptococcus_phage_2_startAligned_Final_forR.gff | sed 's/ID=.*;group=//g' |awk -F "\t" '{OFS="\t"}{print $1,$2,$3,$4,$5,$1,$6}' >> Streptococcus_phage_2_startAligned_Final_for_genoplotR.gff
cat Streptococcus_phage_2_startAligned_Final_for_genoplotR.gff

###--------------------

echo -e "name\tstart\tend\tstrand\tcol\tfill\tgene_type" > Streptococcus_phage_SW30_for_genoplotR.gff

awk -F " " '{OFS="\t"}{print $9 $10 $11 $12 $13 $14 $15,$4,$5,"1","black","arrows"}' Streptococcus_phage_SW30_forR.gff | sed 's/product=.*;group=//g' |awk -F "\t" '{OFS="\t"}{print $1,$2,$3,$4,$5,$1,$6}' >> Streptococcus_phage_SW30_for_genoplotR.gff

```


```{r}
library(genoPlotR)
library(plyr)
library(tidyverse)


##-------------------------
##Streptococcus_phage_SW30
##-------------------------
library(readr)
SW30 <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/annotation/Streptococcus_phage_SW30_for_genoplotR.gff", "\t", escape_double = FALSE, trim_ws = TRUE)  %>% as.data.frame() #%>% filter(name!="spacer") #%>% select(-fill)

table(SW30$fill)

SW30$fill <- revalue(SW30$fill, c("portalprotein"="red","largeterminasesubunit"="red","smallterminasesubunit"="red","majorcapsidprotein"="green","head-tailconnectorprotein"="blue","baseplatecomponentprotein"="yellow","tailprotein"="pink","majortailprotein"="pink","tape-measureprotein"="green","tailchaperoneprotein"="green","distaltailprotein"="yellow","hypotheticalprotein"="grey","holin"="green","lysin"="green","replication"="brown","antireceptorprotein"="blue","other"="grey88","cro-likeprotein"="violet","HNHendonuclease"="grey88","endonuclease"="grey88","Protease"="grey88"))

table(SW30$fill)
table(SW30$col)

dna_seg1_SW30 <- dna_seg(SW30)
dna_segs_SW30 <- list(dna_seg1_SW30)
plot_gene_map(dna_segs=dna_segs_SW30)
##-------------------------
##Streptococcus_phage_2
##-------------------------
library(readr)
rmk_2 <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/annotation/Streptococcus_phage_2_startAligned_Final_for_genoplotR.gff", "\t", escape_double = FALSE, trim_ws = TRUE)  %>% as.data.frame() #%>% filter(name!="spacer") #%>% select(-fill)

table(rmk_2$fill)

# rmk_2$fill <- revalue(rmk_2$fill, c("portalprotein"="red","largeterminasesubunit"="red","smallterminasesubunit"="red","majorcapsidprotein"="green","head-tailconnectorprotein"="blue","baseplatecomponentprotein"="yellow","tailprotein"="pink","majortailprotein"="pink","tape-measureprotein"="green","tailchaperoneprotein"="green","distaltailprotein"="yellow","hypotheticalprotein"="grey","holin"="green","lysin"="green","replication"="brown","antireceptorprotein"="blue","other"="grey88","cro-likeprotein"="violet","HNHendonuclease"="grey88","endonuclease"="grey88","Protease"="grey88","Acr-like"="gold","baseplateprotein"="violet","Head-closureprotein"="purple","Integrase"="cyan"))

rmk_2$fill <- revalue(rmk_2$fill, c("portalprotein"="red","largeterminasesubunit"="red","smallterminasesubunit"="red","majorcapsidprotein"="green","minorcapsidprotein"="green","head-tailconnectorprotein"="blue","baseplatecomponentprotein"="yellow","scaffoldingprotein"="pink","tailprotein"="pink","majortailprotein"="pink","tape-measureprotein"="green","tailchaperoneprotein"="green","distaltailprotein"="yellow","hypotheticalprotein"="grey","holin"="green","lysin"="green","tail-associatedlysin"="blue","replication"="brown","antireceptorprotein"="blue","other"="grey88","cro-likeprotein"="violet","HNHendonuclease"="grey88","endonuclease"="grey88","Protease"="grey88","tailcompletionprotein"="blue","Integrase"="cyan","Acr-like"="gold","baseplateprotein"="violet","Head-closureprotein"="blue"))

table(rmk_2$fill)
table(rmk_2$col)

dna_seg1_rmk_2 <- dna_seg(rmk_2)
dna_segs_rmk_2 <- list(dna_seg1_rmk_2)
plot_gene_map(dna_segs=dna_segs_rmk_2)

##-------------------------
##Streptococcus_phage_987
##-------------------------
library(readr)
phage_9874 <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/annotation/Streptococcus_virus_9874_for_genoplotR.gff", "\t", escape_double = FALSE, trim_ws = TRUE)  %>% as.data.frame() #%>% filter(name!="spacer") #%>% select(-fill)

table(phage_9874$fill)

phage_9874$fill <- revalue(phage_9874$fill, c("portalprotein"="red","largeterminasesubunit"="red","smallterminasesubunit"="red","majorcapsidprotein"="green","minorcapsidprotein"="green","head-tailconnectorprotein"="blue","baseplatecomponentprotein"="yellow","scaffoldingprotein"="pink","tailprotein"="pink","majortailprotein"="pink","tape-measureprotein"="green","tailchaperoneprotein"="green","distaltailprotein"="yellow","hypotheticalprotein"="grey","holin"="green","lysin"="green","tail-associatedlysin"="blue","replication"="brown","antireceptorprotein"="blue","other"="grey88","cro-likeprotein"="violet","HNHendonuclease"="grey88","endonuclease"="grey88","Protease"="grey88","tailcompletionprotein"="blue"))

table(phage_9874$fill)
table(phage_9874$col)

dna_seg1_9874 <- dna_seg(phage_9874)
dna_segs_9874 <- list(dna_seg1_9874)
plot_gene_map(dna_segs=dna_segs_9874)


##-------------------------
##Streptococcus_phage_2
##-------------------------
library(readr)
rmk_1 <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/annotation/Streptococcus_phage_1_startAligned_Final_for_genoplotR.gff", "\t", escape_double = FALSE, trim_ws = TRUE)  %>% as.data.frame() #%>% filter(name!="spacer") #%>% select(-fill)

table(rmk_1$fill)

# rmk_2$fill <- revalue(rmk_2$fill, c("portalprotein"="red","largeterminasesubunit"="red","smallterminasesubunit"="red","majorcapsidprotein"="green","head-tailconnectorprotein"="blue","baseplatecomponentprotein"="yellow","tailprotein"="pink","majortailprotein"="pink","tape-measureprotein"="green","tailchaperoneprotein"="green","distaltailprotein"="yellow","hypotheticalprotein"="grey","holin"="green","lysin"="green","replication"="brown","antireceptorprotein"="blue","other"="grey88","cro-likeprotein"="violet","HNHendonuclease"="grey88","endonuclease"="grey88","Protease"="grey88","Acr-like"="gold","baseplateprotein"="violet","Head-closureprotein"="purple","Integrase"="cyan"))

rmk_1$fill <- revalue(rmk_1$fill, c("portalprotein"="red","largeterminasesubunit"="red","smallterminasesubunit"="red","majorcapsidprotein"="green","minorcapsidprotein"="green","head-tailconnectorprotein"="blue","baseplatecomponentprotein"="yellow","scaffoldingprotein"="pink","tailprotein"="pink","majortailprotein"="pink","tape-measureprotein"="green","tailchaperoneprotein"="green","distaltailprotein"="yellow","hypotheticalprotein"="grey","holin"="green","lysin"="green","tail-associatedlysin"="blue","replication"="brown","antireceptorprotein"="blue","other"="grey88","cro-likeprotein"="violet","HNHendonuclease"="grey88","endonuclease"="grey88","Protease"="grey88","tailcompletionprotein"="blue","Integrase"="cyan","Acr-like"="gold","baseplateprotein"="violet","Head-closureprotein"="blue"))

table(rmk_1$fill)
table(rmk_1$col)

dna_seg1_rmk_1 <- dna_seg(rmk_1)
dna_segs_rmk_1 <- list(dna_seg1_rmk_1)
plot_gene_map(dna_segs=dna_segs_rmk_1)


###--------------------
##merge
###--------------------

all_toegther_annotation <- list(dna_seg1_9874,dna_seg1_rmk_1,dna_seg1_rmk_2,dna_seg1_SW30)
plot_gene_map(dna_segs=all_toegther_annotation)

```


with comparision

```{r}

svg("~/Desktop/Projects/2019_RMK202_analysis/plot/phage_annotation_plot.svg",width=7,height=3)
plot_gene_map(dna_segs=all_toegther_annotation,comparisons = comparisons_mine)
 dev.off()


```




####Rast annotation of phages
also merge phages and prepare for Rast and eggnog assembly

the phages are called:
1. S_term_phage_01
2. L_del_phage_01

the id/locustag should be:
1. stph01
2. ldph01

```{bash}

##--------------prepare the phage contigs

#cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/splitContigs/*.fasta > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/splitContigs/S_term_phage_01.fsa
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/Phages_for_RAST

mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/Phages_for_RAST

for phagesss in $(grep ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta | grep "phage"|sed 's/>//g')
do

samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta ${phagesss} > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/Phages_for_RAST/${phagesss}.fasta

done


##----------------------
##move Phages local for Rast annoation and Eggnog
##----------------------


mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/phage_sterm/
mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/phages/

scp -r vincent@130.223.51.116:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/Phages_for_RAST/ /home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/phages/

scp vincent@130.223.51.116:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//splitContigs/L_del_phage_01.fasta /home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/phage_ldel/

##----------------------
##rename faa after RAST
##----------------------

 sed -i 's/fig|6666666.498386.peg./L_del_phage_01_/g' L_del_phage_01.faa 


 sed -i 's/fig|6666666.498387.peg./S_term_phage_01_/g' S_term_phage_01.faa 


##----------------------
##rename gff after RAST
##----------------------

 sed -i 's/fig|6666666.498387.peg./S_term_phage_01_/g' /home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/phage_sterm/S_term_phage_01.gff

 sed 's/fig|6666666.498386.peg./L_del_phage_01_/g' /home/vincent/Downloads/6666666.498386.gff > /home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/phage_ldel/L_del_phage_01.gff


##----------------------
##merge eggnog output
##----------------------
mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/merged/
cat /home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/sterm/*.annotations \
/home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/ldel/*.annotations \
/home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/phage_ldel/*.annotations \
/home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/phage_sterm/*.annotations |grep "^#" -v > \
/home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/merged/merged_all_eggnog.annotations
      

##----------------------
##move Phages back to big machine for Rast annoation and Eggnog
##----------------------

/home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/


mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/annotation_eggnog_rast/

scp -r  /home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/ vincent@130.223.51.116:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/annotation_eggnog_rast/



##----------------------
##merge all gff for snpEFF
##----------------------
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/annotation_eggnog_rast/assembly_20191129/merged


cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/S_thermophilus_RMK202.current.gff | grep -v "^CP046}" |awk -F "\t" '{OFS"\t"}{if($3=="gene") print $0}' > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/annotation_eggnog_rast/assembly_20191129/merged/all_genes_rmk202.gff

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/L_delbrueckii_RMK202.current.gff | grep -v "^CP046}" |awk -F "\t" '{OFS"\t"}{if($3=="gene") print $0}' >> /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/annotation_eggnog_rast/assembly_20191129/merged/all_genes_rmk202.gff

cat \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/annotation_eggnog_rast/assembly_20191129/phage_ldel/*.gff \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/annotation_eggnog_rast/assembly_20191129/phage_sterm/*.gff >> \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/annotation_eggnog_rast/assembly_20191129/merged/all_genes_rmk202.gff
      
```
####Phaster
Here, I extract the annotation of the entire metagenome file from phaster. 
I am doing this on the local machine.
I do this for the following two things:
1. prophage annotation of bacterial strains
2. Phage genome annotations

currently I am not looking at the inseration sites (attL and attR)
```{bash}

Baselocation=/home/vincent/Desktop/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER
output=/home/vincent/Desktop/Projects/2019_RMK202_analysis/phage_annotation/phaster/

mkdir -p ${output}/cleaned
grep "COMPLETENESS(score)" ${Baselocation}/summary.txt -A 20|head -1  |sed 's/^ *//g' | sed 's/ \{1,\}/\t/g' > ${Baselocation}/header_prophage_summary.txt

grep "COMPLETENESS(score)" ${Baselocation}/summary.txt -A 20|sed -e '1,2d' |sed 's/^ *//g' | sed 's/ \{1,\}/\t/g' > ${Baselocation}/prophage_summary.txt

awk -F "[\t:-]" '{OFS="\t"}{print $5,"Phaster","Prophage",$6,$7,".",".","+","compleness="$3";tRNA="$8";nProt="$9";nProtPhage="$10";Hypothecial="$11";BacProt="$13";GC="$19";commonPhagePercent="$18";ATT_SITEs="$14}' ${Baselocation}/prophage_summary.txt > ${Baselocation}/prophage_summary_all.gff

grep "^CP" ${Baselocation}/prophage_summary_all.gff > ${Baselocation}/prophage_summary_onlyGenome.gff


###-----------------------genes detail gff

grep "^####" ${Baselocation}/detail.txt > ${Baselocation}/tmp.txt
rm ${Baselocation}/gene.gff
#for genenomesss in $(grep "^####" ${Baselocation}/detail.txt |awk -F "," '{print $2}' |sed 's/ //g' |sed 's/####//g')

grep "^####" ${Baselocation}/detail.txt| while read genenomes 
do
genenomesss=$(echo ${genenomes} |awk -F "," '{print $2}' |sed 's/ //g' |sed 's/####//g')
startExtract=$(grep "${genenomes}" ${Baselocation}/tmp.txt)
stopExtract=$(grep -A 1 "${genenomes}" ${Baselocation}/tmp.txt |tail -1)

sed -n "/${startExtract}/,/${stopExtract}/p" ${Baselocation}/detail.txt > ${Baselocation}/tmp2.txt

##forward orientation  genes
genesss=$(echo "614..742")
for genesss in $(grep '^[0-9]' ${Baselocation}/tmp2.txt| grep "attR" -v | grep "attL" -v|cut -d ' ' -f 1)
do
# | sed 's/ \{1,\}/\t/g'
start=$(echo "${genesss}" |awk -F "." '{print $1}')
end=$(echo "${genesss}" |awk -F "." '{print $3}')
namelong=$(grep "${genesss}" ${Baselocation}/tmp2.txt |sed 's/: /:/g' | sed 's/ \{2,\}/\t/g'|awk -F '[\t;]' '{print $2}')
namelong2=$(grep "${genesss}" ${Baselocation}/tmp2.txt |sed 's/: /:/g' | sed 's/ \{2,\}/\t/g'|awk -F '[\t;]' '{print $3}')


#name=$(grep "${genesss}" ${Baselocation}/tmp2.txt |sed 's/: /:/g' | sed 's/ \{2,\}/\t/g'|sed 's/; phage/-phage/g' |awk -F '[\t;]' '{print "quality="$4";name="$3}'|sed 's/name= /name=/g'|sed 's/-/;/g')
#description=$(grep "${genesss}" ${Baselocation}/tmp2.txt |sed 's/: /:/g' | sed 's/ \{2,\}/\t/g'|sed 's/; phage/-phage/g' |awk -F '[\t;]' '{print $2}'|sed 's/.*://')
#besthit=$(grep "${genesss}" ${Baselocation}/tmp2.txt |sed 's/: /:/g' | sed 's/ \{2,\}/\t/g'|sed 's/; phage/-phage/g' |awk -F '[\t;]' '{print $2}'|awk -F '[:]' '{print $1}')

#echo -e ${genenomesss}"\t"${start}"\t"${end}"\t"${PHASTER}"\tgene\t.\t.\t+\t"${name}";description="${description}";besthit="${besthit} >> ${Baselocation}/gene.gff
echo -e ${genenomesss}"\t"${start}"\t"${end}"\t"${PHASTER}"\tgene\t.\t.\t+\t"${namelong}";locusTag="${namelong2} >> ${Baselocation}/gene.gff

done # genesss
##reverse orientation genes
genesss=$(echo "complement(25..516)")
for genesss in $(grep "^complement" ${Baselocation}/tmp2.txt| grep "attR" -v | grep "attL" -v |cut -d ' ' -f 1)
do


start=$(echo "${genesss}" |sed 's/complement(//g' |sed 's/)//g'|awk -F "." '{print $1}')
end=$(echo "${genesss}" |sed 's/complement(//g' |sed 's/)//g'|awk -F "." '{print $3}')
namelong=$(grep "${genesss}" ${Baselocation}/tmp2.txt |sed 's/: /:/g' | sed 's/ \{2,\}/\t/g'|awk -F '[\t;]' '{print $2}')
namelong2=$(grep "${genesss}" ${Baselocation}/tmp2.txt |sed 's/: /:/g' | sed 's/ \{2,\}/\t/g'|awk -F '[\t;]' '{print $3}')

#name=$(grep "${genesss}" ${Baselocation}/tmp2.txt |sed 's/: /:/g' | sed 's/ \{2,\}/\t/g'|sed 's/; phage/-phage/g' |awk -F '[\t;]' '{print "quality="$4";name="$3}'|sed 's/name= /name=/g'|sed 's/-/;/g')
#description=$(grep "${genesss}" ${Baselocation}/tmp2.txt |sed 's/: /:/g' | sed 's/ \{2,\}/\t/g'|sed 's/; phage/-phage/g' |awk -F '[\t;]' '{print $2}'|sed 's/.*://')
#besthit=$(grep "${genesss}" ${Baselocation}/tmp2.txt |sed 's/: /:/g' | sed 's/ \{2,\}/\t/g'|sed 's/; phage/-phage/g' |awk -F '[\t;]' '{print $2}'|awk -F '[:]' '{print $1}')

#qualtiy=$(grep "${genesss}" ${Baselocation}/tmp2.txt | sed 's/ \{1,\}/\t/g'|awk -F "\t" '{print $3}')
#namelong=$(grep "${genesss}" ${Baselocation}/tmp2.txt |sed 's/: /:/g' | sed 's/ \{2,\}/\t/g'|awk -F '[\t;]' '{print $2}')

#echo -e ${genenomesss}"\t"${start}"\t"${end}"\t"${PHASTER}"\tgene\t.\t.\t-\t"${name}";description="${description}";besthit="${besthit} >> ${Baselocation}/gene.gff
echo -e ${genenomesss}"\t"${start}"\t"${end}"\t"${PHASTER}"\tgene\t.\t.\t-\t"${namelong}";locusTag="${namelong2} >> ${Baselocation}/gene.gff


done # genesss

 echo "---------------------------------------------------"
done #genomes


###-------------define phage like proteins as proteins with phage homologos but hypothetical
sed 's/PROPHAGE.*:hypothetical protein/Phage-like protein/g' ${Baselocation}/gene.gff |sed 's/PHAGE.*:hypothetical protein/Phage-like protein/g' |sed 's/PROPHAGE.*://g' |sed 's/PHAGE.*://g' > ${Baselocation}/gene_CLEANED.gff


#sed 's/PROPHAGE.*://g' ${Baselocation}/gene.gff |sed 's/PHAGE.*://g'|sed 's/locusTag= /locusTag=/g' > ${Baselocation}/gene_CLEANED.gff

##======================================
##integration sites
##======================================
grep "attL" ${Baselocation}/detail.txt
grep "attR" ${Baselocation}/detail.txt

##======================================
##get faa
##======================================
rm -r ${Baselocation}/faa_for_virfam
mkdir -p ${Baselocation}/faa_for_virfam
#rm ${Baselocation}/gene.faa

for genesss in $(awk -F "[\t;]" '{print $10}' ${Baselocation}/gene_CLEANED.gff|sed 's/locusTag=//g')
do
seq=$(grep "${genesss}" ${Baselocation}/detail.txt |  sed 's/ \{2,\}/\t/g'|awk -F "\t" '{print $4}')
genomess=$(grep "${genesss}" ${Baselocation}/gene_CLEANED.gff |cut -f 1)

echo -e ">"${genesss}"\n"${seq} >> ${Baselocation}/faa_for_virfam/${genomess}.faa

done

```
then I manually change the names of the annotations of phaster according to the findings of virfam.
now I look which are hypothetical and which not and can further be distinguished into:
- morphogenes
- bacterial genes
- protease
- (see phaster distinguishing)
```{bash}

grep -i "hypoth" ${Baselocation}/gene_CLEANED.gff |awk -F "\t" '{OFS="\t"}{print $0, "hypothetical protein"}' > ${Baselocation}/gene_CLEANED_forCircos_final.gff
grep -i "transpos" ${Baselocation}/gene_CLEANED.gff |awk -F "\t" '{OFS="\t"}{print $0, "Transposase"}' >> ${Baselocation}/gene_CLEANED_forCircos_final.gff
grep -i "lysin" ${Baselocation}/gene_CLEANED.gff |awk -F "\t" '{OFS="\t"}{print $0, "Lysis"}' >> ${Baselocation}/gene_CLEANED_forCircos_final.gff
grep -i "holin" ${Baselocation}/gene_CLEANED.gff |awk -F "\t" '{OFS="\t"}{print $0, "Lysis"}' >> ${Baselocation}/gene_CLEANED_forCircos_final.gff
grep -i "protease" ${Baselocation}/gene_CLEANED.gff |awk -F "\t" '{OFS="\t"}{print $0, "Protease"}' >> ${Baselocation}/gene_CLEANED_forCircos_final.gff
grep -i "terminase" ${Baselocation}/gene_CLEANED.gff |awk -F "\t" '{OFS="\t"}{print $0, "Terminase"}' >> ${Baselocation}/gene_CLEANED_forCircos_final.gff
grep -i "integrase" ${Baselocation}/gene_CLEANED.gff |awk -F "\t" '{OFS="\t"}{print $0, "Integrase"}' >>${Baselocation}/gene_CLEANED_forCircos_final.gff
grep -i "phage-like" ${Baselocation}/gene_CLEANED.gff |awk -F "\t" '{OFS="\t"}{print $0, "Phage-like protein"}' >> ${Baselocation}/gene_CLEANED_forCircos_final.gff
grep -i "head" ${Baselocation}/gene_CLEANED.gff |awk -F "\t" '{OFS="\t"}{print $0, "Morphogenes"}' >> ${Baselocation}/gene_CLEANED_forCircos_final.gff

grep -i  -v "hypoth" ${Baselocation}/gene_CLEANED.gff |grep  -i -v "transpos" |grep  -i -v "holin" |grep -i  -v "lysin" |grep -i  -v "protease" |grep -i  -v "terminase" |grep -i  -v "integrase" |grep -i  -v "phage-like protein"|grep -i -v "head" > ${Baselocation}/tmp.txt
##--------------Morphogenes

grep -i "tail" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Morphogenes"}' >> ${Baselocation}/gene_CLEANED_forCircos_final.gff

  grep -i  -v "hypoth" ${Baselocation}/gene_CLEANED.gff |grep  -i -v "transpos" |grep  -i -v "holin" |grep -i  -v "lysin" |grep -i  -v "protease" |grep -i  -v "terminase" |grep -i  -v "integrase" |grep -i  -v "phage-like protein"|grep -i -v "head" |grep -i -v "tail"> ${Baselocation}/tmp.txt
  grep -i "Capsid" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Morphogenes"}' >> ${Baselocation}/gene_CLEANED_forCircos_final.gff

grep -i "portal" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Morphogenes"}' >> ${Baselocation}/gene_CLEANED_forCircos_final.gff
  grep -i  -v "hypoth" ${Baselocation}/gene_CLEANED.gff |grep  -i -v "transpos" |grep  -i -v "holin" |grep -i  -v "lysin" |grep -i  -v "protease" |grep -i  -v "terminase" |grep -i  -v "integrase" |grep -i  -v "phage-like protein"|grep -i -v "head" |grep -i -v "tail"|grep -i -v "portal"> ${Baselocation}/tmp.txt
  
  grep -i "tape" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Morphogenes"}' >> ${Baselocation}/gene_CLEANED_forCircos_final.gff
    grep -i "structur" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Morphogenes"}' >> ${Baselocation}/gene_CLEANED_forCircos_final.gff
        grep -i "adsorption protein" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Morphogenes"}' >> ${Baselocation}/gene_CLEANED_forCircos_final.gff

 
  grep -i  -v "hypoth" ${Baselocation}/gene_CLEANED.gff |grep  -i -v "transpos" |grep  -i -v "holin" |grep -i  -v "lysin" |grep -i  -v "protease" |grep -i  -v "terminase" |grep -i  -v "integrase" |grep -i  -v "phage-like protein"|grep -i -v "head" |grep -i -v "tail"|grep -i -v "portal"|grep -i -v "tape"|grep -v -i "structur"> ${Baselocation}/tmp.txt
##--------------replication
grep -i "replic" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Replication"}' >> ${Baselocation}/gene_CLEANED_forCircos_final.gff
grep -i "transcrip" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Replication"}' >> ${Baselocation}/gene_CLEANED_forCircos_final.gff
grep -i "binding" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Replication"}' >> ${Baselocation}/gene_CLEANED_forCircos_final.gff
grep -i "primase" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Replication"}' >> ${Baselocation}/gene_CLEANED_forCircos_final.gff
grep -i "polymerase" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Replication"}' >> ${Baselocation}/gene_CLEANED_forCircos_final.gff
grep -i "clamp" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Replication"}' >> ${Baselocation}/gene_CLEANED_forCircos_final.gff
grep -i "topoisomerase" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Replication"}' >> ${Baselocation}/gene_CLEANED_forCircos_final.gff
grep -i "helicase" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Replication"}' >> ${Baselocation}/gene_CLEANED_forCircos_final.gff
grep -i "holliday" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Replication"}' >> ${Baselocation}/gene_CLEANED_forCircos_final.gff


##----------------nuclease
grep -i "nuclease" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Nuclease"}' >> ${Baselocation}/gene_CLEANED_forCircos_final.gff


  grep -i  -v "hypoth" ${Baselocation}/gene_CLEANED.gff |grep  -i -v "transpos" |grep  -i -v "holin" |grep -i  -v "lysin" |grep -i  -v "protease" |grep -i  -v "terminase" |grep -i  -v "integrase" |grep -i  -v "phage-like protein"|grep -i -v "head" |grep -i -v "tail"|grep -i -v "portal" |grep -i -v "polymerase" |grep -i -v "primase"|grep -i -v "binding" |grep -i -v "transcrip" |grep -i -v "replic"|grep -i -v "Nuclease" |grep -i -v "tape"|grep -v -i "structur" |grep -i -v "clamp"|grep -i -v "topoisomerase"|grep -v -i "helicase"|grep -i -v "adsorption protein" |grep -v -i "holliday" | grep -i -v "Capsid" > ${Baselocation}/tmp.txt



  grep -i  -v "hypoth" ${Baselocation}/gene_CLEANED.gff |grep  -i -v "transpos" |grep  -i -v "holin" |grep -i  -v "lysin" |grep -i  -v "protease" |grep -i  -v "terminase" |grep -i  -v "integrase" |grep -i  -v "phage-like protein"|grep -i -v "head" |grep -i -v "tail"|grep -i -v "portal" |grep -i -v "polymerase" |grep -i -v "primase"|grep -i -v "binding" |grep -i -v "transcrip" |grep -i -v "replic"|grep -i -v "Nuclease" |grep -i -v "tape"|grep -v -i "structur" |grep -i -v "clamp"|grep -i -v "topoisomerase"|grep -v -i "helicase"|grep -i -v "adsorption protein" |grep -v -i "holliday" | grep -i -v "Capsid"  |awk -F "\t" '{OFS="\t"}{print $0, "others"}'>> ${Baselocation}/gene_CLEANED_forCircos_final.gff

awk -F "\t" '{OFS="\t"}{print $0, "none"}' ${Baselocation}/gene_CLEANED_forCircos_final.gff > ${Baselocation}/gene_CLEANED_forCircos_final_withVirfam.gff


```
now I annotate the virFam genes
```{bash}

awk -F "[\t;]" '{OFS="\t"}{if($12=="none")print $1,$2,$3,$4,$5,$6,$7,$8,$9";"$10; else print $1,$2,$3,$4,$5,$6,$7,$8,$12";"$10}' ${Baselocation}/gene_CLEANED_forCircos_final_withVirfam.gff > ${Baselocation}/gene_CLEANED_forCircos_final_prepFinal.gff

###----------------Assign names

grep -i "hypoth" ${Baselocation}/gene_CLEANED_forCircos_final_prepFinal.gff |awk -F "\t" '{OFS="\t"}{print $0, "hypothetical protein"}' > ${Baselocation}/gene_CLEANED_FINAL_ALL.gff
grep -i "transpos" ${Baselocation}/gene_CLEANED_forCircos_final_prepFinal.gff |awk -F "\t" '{OFS="\t"}{print $0, "Transposase"}' >> ${Baselocation}/gene_CLEANED_FINAL_ALL.gff
grep -i "lysin" ${Baselocation}/gene_CLEANED_forCircos_final_prepFinal.gff |awk -F "\t" '{OFS="\t"}{print $0, "Lysis"}' >> ${Baselocation}/gene_CLEANED_FINAL_ALL.gff
grep -i "holin" ${Baselocation}/gene_CLEANED_forCircos_final_prepFinal.gff |awk -F "\t" '{OFS="\t"}{print $0, "Lysis"}' >> ${Baselocation}/gene_CLEANED_FINAL_ALL.gff
grep -i "protease" ${Baselocation}/gene_CLEANED_forCircos_final_prepFinal.gff |awk -F "\t" '{OFS="\t"}{print $0, "Protease"}' >> ${Baselocation}/gene_CLEANED_FINAL_ALL.gff
grep -i "terminase" ${Baselocation}/gene_CLEANED_forCircos_final_prepFinal.gff |awk -F "\t" '{OFS="\t"}{print $0, "Terminase"}' >> ${Baselocation}/gene_CLEANED_FINAL_ALL.gff
grep -i "integrase" ${Baselocation}/gene_CLEANED_forCircos_final_prepFinal.gff |awk -F "\t" '{OFS="\t"}{print $0, "Integrase"}' >>${Baselocation}/gene_CLEANED_FINAL_ALL.gff
grep -i "phage-like" ${Baselocation}/gene_CLEANED_forCircos_final_prepFinal.gff |awk -F "\t" '{OFS="\t"}{print $0, "Phage-like protein"}' >> ${Baselocation}/gene_CLEANED_FINAL_ALL.gff
grep -i "head" ${Baselocation}/gene_CLEANED_forCircos_final_prepFinal.gff |awk -F "\t" '{OFS="\t"}{print $0, "Morphogenes"}' >> ${Baselocation}/gene_CLEANED_FINAL_ALL.gff

grep -i  -v "hypoth" ${Baselocation}/gene_CLEANED_forCircos_final_prepFinal.gff |grep  -i -v "transpos" |grep  -i -v "holin" |grep -i  -v "lysin" |grep -i  -v "protease" |grep -i  -v "terminase" |grep -i  -v "integrase" |grep -i  -v "phage-like protein"|grep -i -v "head" > ${Baselocation}/tmp.txt
##--------------Morphogenes

grep -i "tail" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Morphogenes"}' >> ${Baselocation}/gene_CLEANED_FINAL_ALL.gff

  grep -i  -v "hypoth" ${Baselocation}/gene_CLEANED_forCircos_final_prepFinal.gff |grep  -i -v "transpos" |grep  -i -v "holin" |grep -i  -v "lysin" |grep -i  -v "protease" |grep -i  -v "terminase" |grep -i  -v "integrase" |grep -i  -v "phage-like protein"|grep -i -v "head" |grep -i -v "tail"> ${Baselocation}/tmp.txt
  grep -i "Capsid" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Morphogenes"}' >> ${Baselocation}/gene_CLEANED_FINAL_ALL.gff

grep -i "portal" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Morphogenes"}' >> ${Baselocation}/gene_CLEANED_FINAL_ALL.gff
  grep -i  -v "hypoth" ${Baselocation}/gene_CLEANED_forCircos_final_prepFinal.gff |grep  -i -v "transpos" |grep  -i -v "holin" |grep -i  -v "lysin" |grep -i  -v "protease" |grep -i  -v "terminase" |grep -i  -v "integrase" |grep -i  -v "phage-like protein"|grep -i -v "head" |grep -i -v "tail"|grep -i -v "portal"> ${Baselocation}/tmp.txt
  
  grep -i "tape" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Morphogenes"}' >> ${Baselocation}/gene_CLEANED_FINAL_ALL.gff
    grep -i "structur" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Morphogenes"}' >> ${Baselocation}/gene_CLEANED_FINAL_ALL.gff
        grep -i "adsorption protein" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Morphogenes"}' >> ${Baselocation}/gene_CLEANED_FINAL_ALL.gff

 
  grep -i  -v "hypoth" ${Baselocation}/gene_CLEANED_forCircos_final_prepFinal.gff |grep  -i -v "transpos" |grep  -i -v "holin" |grep -i  -v "lysin" |grep -i  -v "protease" |grep -i  -v "terminase" |grep -i  -v "integrase" |grep -i  -v "phage-like protein"|grep -i -v "head" |grep -i -v "tail"|grep -i -v "portal"|grep -i -v "tape"|grep -v -i "structur"> ${Baselocation}/tmp.txt
##--------------replication
grep -i "replic" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Replication"}' >> ${Baselocation}/gene_CLEANED_FINAL_ALL.gff
grep -i "transcrip" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Replication"}' >> ${Baselocation}/gene_CLEANED_FINAL_ALL.gff
grep -i "binding" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Replication"}' >> ${Baselocation}/gene_CLEANED_FINAL_ALL.gff
grep -i "primase" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Replication"}' >> ${Baselocation}/gene_CLEANED_FINAL_ALL.gff
grep -i "polymerase" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Replication"}' >> ${Baselocation}/gene_CLEANED_FINAL_ALL.gff
grep -i "clamp" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Replication"}' >> ${Baselocation}/gene_CLEANED_FINAL_ALL.gff
grep -i "topoisomerase" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Replication"}' >> ${Baselocation}/gene_CLEANED_FINAL_ALL.gff
grep -i "helicase" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Replication"}' >> ${Baselocation}/gene_CLEANED_FINAL_ALL.gff
grep -i "holliday" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Replication"}' >> ${Baselocation}/gene_CLEANED_FINAL_ALL.gff


##----------------nuclease
grep -i "nuclease" ${Baselocation}/tmp.txt |awk -F "\t" '{OFS="\t"}{print $0, "Nuclease"}' >> ${Baselocation}/gene_CLEANED_FINAL_ALL.gff


  grep -i  -v "hypoth" ${Baselocation}/gene_CLEANED_forCircos_final_prepFinal.gff |grep  -i -v "transpos" |grep  -i -v "holin" |grep -i  -v "lysin" |grep -i  -v "protease" |grep -i  -v "terminase" |grep -i  -v "integrase" |grep -i  -v "phage-like protein"|grep -i -v "head" |grep -i -v "tail"|grep -i -v "portal" |grep -i -v "polymerase" |grep -i -v "primase"|grep -i -v "binding" |grep -i -v "transcrip" |grep -i -v "replic"|grep -i -v "Nuclease" |grep -i -v "tape"|grep -v -i "structur" |grep -i -v "clamp"|grep -i -v "topoisomerase"|grep -v -i "helicase"|grep -i -v "adsorption protein" |grep -v -i "holliday" | grep -i -v "Capsid" > ${Baselocation}/tmp.txt



  grep -i  -v "hypoth" ${Baselocation}/gene_CLEANED_forCircos_final_prepFinal.gff |grep  -i -v "transpos" |grep  -i -v "holin" |grep -i  -v "lysin" |grep -i  -v "protease" |grep -i  -v "terminase" |grep -i  -v "integrase" |grep -i  -v "phage-like protein"|grep -i -v "head" |grep -i -v "tail"|grep -i -v "portal" |grep -i -v "polymerase" |grep -i -v "primase"|grep -i -v "binding" |grep -i -v "transcrip" |grep -i -v "replic"|grep -i -v "Nuclease" |grep -i -v "tape"|grep -v -i "structur" |grep -i -v "clamp"|grep -i -v "topoisomerase"|grep -v -i "helicase"|grep -i -v "adsorption protein" |grep -v -i "holliday" | grep -i -v "Capsid"  |awk -F "\t" '{OFS="\t"}{print $0, "others"}'>> ${Baselocation}/gene_CLEANED_FINAL_ALL.gff
 cut -f 10 ${Baselocation}/gene_CLEANED_FINAL_ALL.gff |sort|uniq -c
 cut -f 10 ${Baselocation}/gene_CLEANED_FINAL_ALL.gff |sort|uniq 
```

now I can make the circos of the following:

hypothetical protein
others
Phage-like protein

Transposase
Integrase
Nuclease
Protease
Lysis
Terminase

Morphogenes

Replication


#### Circos Phages

Here, I creat the circos plots for the bacteria. 
I will include the following information:
1. genome (circular)
2. forward genes  
3. reverse genes
4. hypthecial genes

```{bash}

##==================
##Sterm phage 1
##==================
home=/home/vincent/bin/apps/circos-0.69-6/
cd $home

mkdir -p $home/data/rmk202/MAG_rmk202_sterm_phage_1/


##organisational 
#cp /home/vincent/bin/apps/circos-0.69-6/etc/ticks_mag_both_plasmids.conf /home/vincent/bin/apps/circos-0.69-6/etc/ticks_mag_both_phages.conf

#cp /home/vincent/bin/apps/circos-0.69-6/etc/ideogram_mag_plasmids.conf /home/vincent/bin/apps/circos-0.69-6/etc/ideogram_mag_phages.conf
#cp /home/vincent/bin/apps/circos-0.69-6/etc/sterm_rmk202_mag_circos.conf /home/vincent/bin/apps/circos-0.69-6/etc/sterm_phage_1_rmk202_mag_circos.conf

#/home/vincent/Desktop/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/gene_CLEANED_FINAL_ALL.gff
##---------------------karyotyp

seqlength.py /home/vincent/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/RMK202_MAG_assembly.fasta |grep "Streptococcus_phage_1" > ${home}/data/karyotype/MAG_rmk202_sterm_pphage_1.txt
##have to change the format a bit

##---------------------general genes
#includes:hypothetical protein, others, Phage-like protein
namesss=Streptococcus_phage_1

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="hypothetical protein" && $1==namesssss) print $1,$2,$3,"color=chr2"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff |uniq > $home/data/rmk202/MAG_rmk202_sterm_phage_1/genes.txt

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="others" && $1==namesssss) print $1,$2,$3,"color=chr1"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff |uniq >> $home/data/rmk202/MAG_rmk202_sterm_phage_1/genes.txt

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Phage-like protein" && $1==namesssss) print $1,$2,$3,"color=chr3"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff|uniq  >> $home/data/rmk202/MAG_rmk202_sterm_phage_1/genes.txt


##---------------------enzyme genes 
##includes: Transposase, Integrase, Nuclease, Protease, Lysis, Terminase


awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Transposase" && $1==namesssss) print $1,$2,$3,"color=chr6"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff|uniq  > $home/data/rmk202/MAG_rmk202_sterm_phage_1/Enyme_genes.txt

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Integrase" && $1==namesssss) print $1,$2,$3,"color=chr7"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff |uniq >> $home/data/rmk202/MAG_rmk202_sterm_phage_1/Enyme_genes.txt


awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Nuclease" && $1==namesssss) print $1,$2,$3,"color=chr8"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff |uniq >> $home/data/rmk202/MAG_rmk202_sterm_phage_1/Enyme_genes.txt

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Protease" && $1==namesssss) print $1,$2,$3,"color=chr9"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff |uniq >> $home/data/rmk202/MAG_rmk202_sterm_phage_1/Enyme_genes.txt

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Lysis" && $1==namesssss) print $1,$2,$3,"color=chr10"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff |uniq >> $home/data/rmk202/MAG_rmk202_sterm_phage_1/Enyme_genes.txt

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Terminase" && $1==namesssss) print $1,$2,$3,"color=chr11"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff |uniq >> $home/data/rmk202/MAG_rmk202_sterm_phage_1/Enyme_genes.txt



##---------------------Morphogenes
##includes: Morphogenes


awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Morphogenes" && $1==namesssss) print $1,$2,$3,"color=chr12"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff |uniq > $home/data/rmk202/MAG_rmk202_sterm_phage_1/Morphogenes.txt


##---------------------Replication
##includes: Morphogenes


awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Replication" && $1==namesssss) print $1,$2,$3,"color=chr13"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff |uniq > $home/data/rmk202/MAG_rmk202_sterm_phage_1/Replication.txt

##example
#cp etc/repeat_nwc1_sterm_01.conf etc/sterm_rmk202_mag_circos.conf

##---------------------Protospacer
#the protospacer location where created in the blast chunk
cat ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/all_differentBlastHits_bothGenomes_circos_protospacers.txt > $home/data/rmk202/MAG_rmk202_sterm_phage_1/all_differentBlastHits_bothGenomes_circos_protospacers.txt

#example run
bin/circos -conf etc/sterm_phage_1_rmk202_mag_circos.conf -outputfile ./sterm_phage_1_mag_rmk202.svg; firefox ./sterm_phage_1_mag_rmk202.svg &




##==================
##Sterm phage 2
##==================
home=/home/vincent/bin/apps/circos-0.69-6/
cd $home

mkdir -p $home/data/rmk202/MAG_rmk202_sterm_phage_2/


##organisational 
#cp /home/vincent/bin/apps/circos-0.69-6/etc/ticks_mag_both_plasmids.conf /home/vincent/bin/apps/circos-0.69-6/etc/ticks_mag_both_phages.conf

#cp /home/vincent/bin/apps/circos-0.69-6/etc/ideogram_mag_plasmids.conf /home/vincent/bin/apps/circos-0.69-6/etc/ideogram_mag_phages.conf
cp /home/vincent/bin/apps/circos-0.69-6/etc/sterm_phage_1_rmk202_mag_circos.conf /home/vincent/bin/apps/circos-0.69-6/etc/sterm_phage_2_rmk202_mag_circos.conf


##---------------------karyotyp

#seqlength.py /home/vincent/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/RMK202_MAG_assembly.fasta |grep "Streptococcus_phage_2" > ${home}/data/karyotype/MAG_rmk202_sterm_pphage_2.txt
##have to change the format a bit

##---------------------general genes
#includes:hypothetical protein, others, Phage-like protein
namesss=Streptococcus_phage_2

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="hypothetical protein" && $1==namesssss) print $1,$2,$3,"color=chr2"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff > $home/data/rmk202/MAG_rmk202_sterm_phage_2/genes.txt

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="others" && $1==namesssss) print $1,$2,$3,"color=chr1"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff >> $home/data/rmk202/MAG_rmk202_sterm_phage_2/genes.txt

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Phage-like protein" && $1==namesssss) print $1,$2,$3,"color=chr3"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff >> $home/data/rmk202/MAG_rmk202_sterm_phage_2/genes.txt


##---------------------enzyme genes 
##includes: Transposase, Integrase, Nuclease, Protease, Lysis, Terminase


awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Transposase" && $1==namesssss) print $1,$2,$3,"color=chr6"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff > $home/data/rmk202/MAG_rmk202_sterm_phage_2/Enyme_genes.txt

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Integrase" && $1==namesssss) print $1,$2,$3,"color=chr7"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff >> $home/data/rmk202/MAG_rmk202_sterm_phage_2/Enyme_genes.txt

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Nuclease" && $1==namesssss) print $1,$2,$3,"color=chr8"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff >> $home/data/rmk202/MAG_rmk202_sterm_phage_2/Enyme_genes.txt

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Protease" && $1==namesssss) print $1,$2,$3,"color=chr9"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff >> $home/data/rmk202/MAG_rmk202_sterm_phage_2/Enyme_genes.txt

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Lysis" && $1==namesssss) print $1,$2,$3,"color=chr10"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff >> $home/data/rmk202/MAG_rmk202_sterm_phage_2/Enyme_genes.txt

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Terminase" && $1==namesssss) print $1,$2,$3,"color=chr11"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff >> $home/data/rmk202/MAG_rmk202_sterm_phage_2/Enyme_genes.txt



##---------------------Morphogenes
##includes: Morphogenes


awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Morphogenes" && $1==namesssss) print $1,$2,$3,"color=chr12"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff > $home/data/rmk202/MAG_rmk202_sterm_phage_2/Morphogenes.txt


##---------------------Replication
##includes: Morphogenes


awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Replication" && $1==namesssss) print $1,$2,$3,"color=chr13"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff > $home/data/rmk202/MAG_rmk202_sterm_phage_2/Replication.txt

##---------------------Protospacer
#the protospacer location where created in the blast chunk
#cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/GenotplotR/GenotplotR/CRISPRblast_rmk202phages_Streptococcus_phage_2_Circos.txt > $home/data/rmk202/MAG_rmk202_sterm_phage_2/CRISPRblast_rmk202phages_Streptococcus_phage_2_Circos.txt
cat ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/all_differentBlastHits_bothGenomes_circos_protospacers.txt > $home/data/rmk202/MAG_rmk202_sterm_phage_1/all_differentBlastHits_bothGenomes_circos_protospacers.txt
sed -i 's/color=chr5_a4/color=chr5_a3/g' $home/data/rmk202/MAG_rmk202_sterm_phage_1/all_differentBlastHits_bothGenomes_circos_protospacers.txt
sed -i 's/black_a1/color=black_a2/g' $home/data/rmk202/MAG_rmk202_sterm_phage_1/all_differentBlastHits_bothGenomes_circos_protospacers.txt

##example
#cp etc/repeat_nwc1_sterm_01.conf etc/sterm_rmk202_mag_circos.conf

#example run
bin/circos -conf etc/sterm_phage_2_rmk202_mag_circos.conf -outputfile ./sterm_phage_2_mag_rmk202.svg; firefox ./sterm_phage_2_mag_rmk202.svg &


##==================
##Ldel phage 1
##==================
home=/home/vincent/bin/apps/circos-0.69-6/
cd $home

mkdir -p $home/data/rmk202/MAG_rmk202_ldel_phage_1/


##organisational 
#cp /home/vincent/bin/apps/circos-0.69-6/etc/ticks_mag_both_plasmids.conf /home/vincent/bin/apps/circos-0.69-6/etc/ticks_mag_both_phages.conf

#cp /home/vincent/bin/apps/circos-0.69-6/etc/ideogram_mag_plasmids.conf /home/vincent/bin/apps/circos-0.69-6/etc/ideogram_mag_phages.conf
cp /home/vincent/bin/apps/circos-0.69-6/etc/sterm_phage_1_rmk202_mag_circos.conf /home/vincent/bin/apps/circos-0.69-6/etc/ldel_phage_1_rmk202_mag_circos.conf


##---------------------karyotyp

#seqlength.py /home/vincent/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/RMK202_MAG_assembly.fasta |grep "Lactobacillus_phage_1" > ${home}/data/karyotype/MAG_rmk202_ldel_pphage_1.txt
##have to change the format a bit

##---------------------general genes
#includes:hypothetical protein, others, Phage-like protein
namesss=Lactobacillus_phage_1

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="hypothetical protein" && $1==namesssss) print $1,$2,$3,"color=chr2"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff > $home/data/rmk202/MAG_rmk202_ldel_phage_1/genes.txt

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="others" && $1==namesssss) print $1,$2,$3,"color=chr1"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff >> $home/data/rmk202/MAG_rmk202_ldel_phage_1/genes.txt

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Phage-like protein" && $1==namesssss) print $1,$2,$3,"color=chr3"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff >> $home/data/rmk202/MAG_rmk202_ldel_phage_1/genes.txt


##---------------------enzyme genes 
##includes: Transposase, Integrase, Nuclease, Protease, Lysis, Terminase


awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Transposase" && $1==namesssss) print $1,$2,$3,"color=chr6"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff > $home/data/rmk202/MAG_rmk202_ldel_phage_1/Enyme_genes.txt

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Integrase" && $1==namesssss) print $1,$2,$3,"color=chr7"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff >> $home/data/rmk202/MAG_rmk202_ldel_phage_1/Enyme_genes.txt

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Nuclease" && $1==namesssss) print $1,$2,$3,"color=chr8"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff >> $home/data/rmk202/MAG_rmk202_ldel_phage_1/Enyme_genes.txt

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Protease" && $1==namesssss) print $1,$2,$3,"color=chr9"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff >> $home/data/rmk202/MAG_rmk202_ldel_phage_1/Enyme_genes.txt

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Lysis" && $1==namesssss) print $1,$2,$3,"color=chr10"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff >> $home/data/rmk202/MAG_rmk202_ldel_phage_1/Enyme_genes.txt

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Terminase" && $1==namesssss) print $1,$2,$3,"color=chr11"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff >> $home/data/rmk202/MAG_rmk202_ldel_phage_1/Enyme_genes.txt



##---------------------Morphogenes
##includes: Morphogenes


awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Morphogenes" && $1==namesssss) print $1,$2,$3,"color=chr12"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff > $home/data/rmk202/MAG_rmk202_ldel_phage_1/Morphogenes.txt


##---------------------Replication
##includes: Morphogenes


awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Replication" && $1==namesssss) print $1,$2,$3,"color=chr13"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff > $home/data/rmk202/MAG_rmk202_ldel_phage_1/Replication.txt

##example
#cp etc/repeat_nwc1_sterm_01.conf etc/sterm_rmk202_mag_circos.conf

#example run
bin/circos -conf etc/ldel_phage_1_rmk202_mag_circos.conf -outputfile ./ldel_phage_1_mag_rmk202.svg; firefox ./ldel_phage_1_mag_rmk202.svg &



##==================
##Ldel phage 2
##==================
home=/home/vincent/bin/apps/circos-0.69-6/
cd $home

mkdir -p $home/data/rmk202/MAG_rmk202_ldel_phage_2/


##organisational 
#cp /home/vincent/bin/apps/circos-0.69-6/etc/ticks_mag_both_plasmids.conf /home/vincent/bin/apps/circos-0.69-6/etc/ticks_mag_both_phages.conf

#cp /home/vincent/bin/apps/circos-0.69-6/etc/ideogram_mag_plasmids.conf /home/vincent/bin/apps/circos-0.69-6/etc/ideogram_mag_phages.conf
cp /home/vincent/bin/apps/circos-0.69-6/etc/sterm_phage_1_rmk202_mag_circos.conf /home/vincent/bin/apps/circos-0.69-6/etc/ldel_phage_2_rmk202_mag_circos.conf


##---------------------karyotyp

#seqlength.py /home/vincent/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/RMK202_MAG_assembly.fasta |grep "Lactobacillus_phage_2" > ${home}/data/karyotype/MAG_rmk202_ldel_pphage_2.txt
##have to change the format a bit

##---------------------general genes
#includes:hypothetical protein, others, Phage-like protein
namesss=Lactobacillus_phage_2

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="hypothetical protein" && $1==namesssss) print $1,$2,$3,"color=chr2"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff > $home/data/rmk202/MAG_rmk202_ldel_phage_2/genes.txt

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="others" && $1==namesssss) print $1,$2,$3,"color=chr1"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff >> $home/data/rmk202/MAG_rmk202_ldel_phage_2/genes.txt

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Phage-like protein" && $1==namesssss) print $1,$2,$3,"color=chr3"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff >> $home/data/rmk202/MAG_rmk202_ldel_phage_2/genes.txt


##---------------------enzyme genes 
##includes: Transposase, Integrase, Nuclease, Protease, Lysis, Terminase


awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Transposase" && $1==namesssss) print $1,$2,$3,"color=chr6"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff > $home/data/rmk202/MAG_rmk202_ldel_phage_2/Enyme_genes.txt

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Integrase" && $1==namesssss) print $1,$2,$3,"color=chr7"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff >> $home/data/rmk202/MAG_rmk202_ldel_phage_2/Enyme_genes.txt

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Nuclease" && $1==namesssss) print $1,$2,$3,"color=chr8"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff >> $home/data/rmk202/MAG_rmk202_ldel_phage_2/Enyme_genes.txt

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Protease" && $1==namesssss) print $1,$2,$3,"color=chr9"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff >> $home/data/rmk202/MAG_rmk202_ldel_phage_2/Enyme_genes.txt

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Lysis" && $1==namesssss) print $1,$2,$3,"color=chr10"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff >> $home/data/rmk202/MAG_rmk202_ldel_phage_2/Enyme_genes.txt

awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Terminase" && $1==namesssss) print $1,$2,$3,"color=chr11"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff >> $home/data/rmk202/MAG_rmk202_ldel_phage_2/Enyme_genes.txt



##---------------------Morphogenes
##includes: Morphogenes


awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Morphogenes" && $1==namesssss) print $1,$2,$3,"color=chr12"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff > $home/data/rmk202/MAG_rmk202_ldel_phage_2/Morphogenes.txt


##---------------------Replication
##includes: Morphogenes


awk -F "\t" -v namesssss="$namesss" '{OFS="\t"}{if($10=="Replication" && $1==namesssss) print $1,$2,$3,"color=chr13"}' ${Baselocation}/gene_CLEANED_FINAL_ALL.gff > $home/data/rmk202/MAG_rmk202_ldel_phage_2/Replication.txt

##example
#cp etc/repeat_nwc1_sterm_01.conf etc/sterm_rmk202_mag_circos.conf

#example run
bin/circos -conf etc/ldel_phage_2_rmk202_mag_circos.conf -outputfile ./ldel_phage_2_mag_rmk202.svg; firefox ./ldel_phage_2_mag_rmk202.svg &



```
#### Attachment site

Here, I extract the attachments site of the phages and blast them against the Streptococcus genomes. 

```{bash}

Baselocation=/home/vincent/Desktop/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER
output=/home/vincent/Desktop/Projects/2019_RMK202_analysis/phage_annotation/phaster/

##======================================
##integration sites
##======================================
grep "attL" ${Baselocation}/detail.txt
grep "attR" ${Baselocation}/detail.txt


###-----------------------genes detail gff

grep "^####" ${Baselocation}/detail.txt > ${Baselocation}/tmp.txt
rm -r  ${Baselocation}/attachmentSite/
mkdir -p ${Baselocation}/attachmentSite/
#for genenomesss in $(grep "^####" ${Baselocation}/detail.txt |awk -F "," '{print $2}' |sed 's/ //g' |sed 's/####//g')

grep "^####" ${Baselocation}/detail.txt | while read genenomes 
do
genenomesss=$(echo ${genenomes} |awk -F "," '{print $2}' |sed 's/ //g' |sed 's/####//g')
startExtract=$(grep "${genenomes}" ${Baselocation}/tmp.txt)
stopExtract=$(grep -A 1 "${genenomes}" ${Baselocation}/tmp.txt |tail -1)

sed -n "/${startExtract}/,/${stopExtract}/p" ${Baselocation}/detail.txt > ${Baselocation}/tmp2.txt

##forward orientation  genes
genesss=$(echo "614..742")
for genesss in $(grep '^[0-9]' ${Baselocation}/tmp2.txt| grep "attL" -i |cut -d ' ' -f 1)
do
# | sed 's/ \{1,\}/\t/g'
start=$(echo "${genesss}" |awk -F "." '{print $1}')
end=$(echo "${genesss}" |awk -F "." '{print $3}')
namelong=$(grep "${genesss}" ${Baselocation}/tmp2.txt |sed 's/: /:/g' | sed 's/ \{2,\}/\t/g'|awk -F '[\t;]' '{print $2}')
namelong2=$(grep "${genesss}" ${Baselocation}/tmp2.txt |sed 's/: /:/g' | sed 's/ \{2,\}/\t/g'|awk -F '[\t;]' '{print $3}')

echo -e ${genenomesss}"\t"${start}"\t"${end}"\t"${PHASTER}"\tgene\t.\t.\t+\t"${namelong}";locusTag="${namelong2} >> ${Baselocation}/attachmentSite/${genenomesss}_attL.gff

fastasss=$(grep "${genesss}" ${Baselocation}/tmp2.txt |sed 's/: /:/g' | sed 's/ \{2,\}/\t/g'|awk -F '[\t;]' '{print $4}')
echo -e ">"${genenomesss}"_"${namelong}"\n"${fastasss} >> ${Baselocation}/attachmentSite/attL.fasta

done # genesss
for genesss in $(grep '^[0-9]' ${Baselocation}/tmp2.txt| grep "attR" -i |cut -d ' ' -f 1)
do
# | sed 's/ \{1,\}/\t/g'
start=$(echo "${genesss}" |awk -F "." '{print $1}')
end=$(echo "${genesss}" |awk -F "." '{print $3}')
namelong=$(grep "${genesss}" ${Baselocation}/tmp2.txt |sed 's/: /:/g' | sed 's/ \{2,\}/\t/g'|awk -F '[\t;]' '{print $2}')
namelong2=$(grep "${genesss}" ${Baselocation}/tmp2.txt |sed 's/: /:/g' | sed 's/ \{2,\}/\t/g'|awk -F '[\t;]' '{print $3}')


echo -e ${genenomesss}"\t"${start}"\t"${end}"\t"${PHASTER}"\tgene\t.\t.\t+\t"${namelong}";locusTag="${namelong2} >> ${Baselocation}/attachmentSite/attR.gff

fastasss=$(grep "${genesss}" ${Baselocation}/tmp2.txt |sed 's/: /:/g' | sed 's/ \{2,\}/\t/g'|awk -F '[\t;]' '{print $4}')
echo -e ">"${genenomesss}"_"${namelong}"\n"${fastasss} >> ${Baselocation}/attachmentSite/${genenomesss}_attR.fasta

done # genesss

done # genesss


###move big machine

scp -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/attachmentSite/ vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/attachmentSite/

```


blast against all genomes
```{bash}


rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/attachmentSite/blasting
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/attachmentSite/blasting

for genenomesss in $(echo "Streptococcus_phage_1 Streptococcus_phage_2")
do

for species in $(echo " Sterm" )
do
echo ==========================
     echo "working on: " ${species}
     echo ==========================
for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/|grep ".fna$" |sed 's/.fna//g')
do
      echo ==========================
     echo -e "blasting to strain: " ${genomes}
     echo ==========================
     
     

makeblastdb -max_file_sz 1GB -in  /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/${genomes}.fna \
 -out /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/${genomes} -parse_seqids -dbtype nucl

  
    blastn -num_threads 35 -max_hsps 1000 -max_target_seqs 1000 -task blastn-short -show_gis \
    -query /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/attachmentSite/attachmentSite/${genenomesss}_attR.fasta \
    -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen sgi sacc sblastnames staxids sscinames" \
    -db /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/${genomes} \
    -out /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/attachmentSite/blasting/${genomes}_${genenomesss}.txt \
    -evalue 100 -word_size 4

awk -F "\t" '{OFS="\t"}{if($3=="100.000" && $4>9) print $0}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/attachmentSite/blasting/${genomes}_${genenomesss}.txt |awk -F "\t" '{OFS="\t"}{if($9>$10) print $2,$10,$9,$1,$4;else print $2,$9,$10,$1,$4 }' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/attachmentSite/blasting/All_cleaned.txt


  done #genomes
 
done #species
done #genenomesss


##-------------------------ldel


for genenomesss in $(echo "Lactobacillus_phage_1")
do

for species in $(echo " Ldel" )
do
echo ==========================
     echo "working on: " ${species}
     echo ==========================
for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/|grep ".fna$" |sed 's/.fna//g')
do
      echo ==========================
     echo -e "blasting to strain: " ${genomes}
     echo ==========================
     
     

makeblastdb -max_file_sz 1GB -in  /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/${genomes}.fna \
 -out /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/${genomes} -parse_seqids -dbtype nucl

  
    blastn -num_threads 35 -max_hsps 1000 -max_target_seqs 1000 -task blastn-short -show_gis \
    -query /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/attachmentSite/attachmentSite/${genenomesss}_attR.fasta \
    -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen sgi sacc sblastnames staxids sscinames" \
    -db /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/${genomes} \
    -out /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/attachmentSite/blasting/${genomes}_${genenomesss}.txt \
    -evalue 100 -word_size 4

awk -F "\t" '{OFS="\t"}{if($3=="100.000" && $4>9) print $0}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/attachmentSite/blasting/${genomes}_${genenomesss}.txt |awk -F "\t" '{OFS="\t"}{if($9>$10) print $2,$10,$9,$1,$4;else print $2,$9,$10,$1,$4 }' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/attachmentSite/blasting/All_cleaned.txt


  done #genomes
 
done #species
done #genenomesss



#----------------------final
cut -f 1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/attachmentSite/blasting/All_cleaned.txt |sort|uniq -c


awk -F "\t" '{OFS="\t"}{if($5>11) print $0}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/attachmentSite/blasting/All_cleaned.txt 


awk -F "\t" '{OFS="\t"}{if($5>11) print $1,$2,$3,$4}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/attachmentSite/blasting/All_cleaned.txt > /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/attachmentSite/blasting/All_cleaned_highmatches.txt

cut -f 1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/attachmentSite/blasting/All_cleaned_highmatches.txt|cut -d '-' -f 1,2 |sort|uniq -c
sort /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/attachmentSite/blasting/All_cleaned_highmatches.txt|uniq


```

MOVE LOCAL
```{bash}


scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/attachmentSite/blasting/All_cleaned.txt /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/attachmentSite/blasting/All_cleaned.txt

```



```{r}
library(readr)
library(ggplot2)
library(plyr)
library(tidyverse)
library(patchwork)
All_cleaned <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/attachmentSite/blasting/All_cleaned.txt",  "\t", escape_double = FALSE, col_names = c("contig","start","end","phageName","matchingLength"),  trim_ws = TRUE) %>% filter(phageName!="Lactobacillus_phage_1_attR")
tmp1 <- data.frame(str_split_fixed(All_cleaned$contig, fixed("-"), 3))[,1:2]

All_cleaned$genome <- paste(tmp1$X1,tmp1$X2,sep="-")
table(All_cleaned$genome)
All_cleaned$genome = factor(All_cleaned$genome, levels=c("202-11141","202-11142","202-11143","202-12104","202-12105","202-12107","202-12109","202-L104","202-L108","202-L35","202-L39","202-L44","202-L70","202-L71" , "202-L80","202-L99","202-LMAG","202-13494","202-S72","202-13498","202-13493","202-13500","202-SMAG","202-13491","202-13492","202-13495","202-13496","202-13497","202-13499c1","202-13499c2","202-S50"))
table(All_cleaned$genome)

All_cleaned$phageName <- revalue(All_cleaned$phageName, c("Streptococcus_phage_1_attR"="Streptococcus phage 1 attP", "Streptococcus_phage_2_attR"="Streptococcus phage 2 attP","Lactobacillus_phage_1_attR"="Lactobacillus phage 1 attP"))


All_cleaned$grouping <- ifelse(All_cleaned$matchingLength>11,"complete","incomplete")
All_cleaned$grouping = factor(All_cleaned$grouping, levels=c("incomplete","complete"))
colorsss <- c("grey89","blue")
attachmentSites_sterm <- ggplot(All_cleaned,aes(x=genome,color=grouping,fill=grouping))+geom_bar()+theme_classic()+theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9),
          # # axis.text.y.left = element_text(color = "blue"),
          # axis.text.y.right = element_text(color="red"),
          # axis.title.y.left = element_text(color="blue"),
          # axis.title.y.right = element_text(color="red"),
          axis.title.x = element_blank(),
          # rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.title = element_blank(),
          legend.position="right")+
  scale_color_manual(values = colorsss)+
  scale_fill_manual(values = colorsss)+
  labs(y="number of attachment sites (attB)")+
  facet_wrap(~phageName,nrow = 2)
attachmentSites_sterm
# 
# svg("~/Desktop/Projects/2019_RMK202_analysis/plot/attachmentSite.svg",width=7,height=4)
# attachmentSites 
#  dev.off()
 
 #------------------------------------------
 ###Ldel
  #------------------------------------------
 
 All_cleaned <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/attachmentSite/blasting/All_cleaned.txt",  "\t", escape_double = FALSE, col_names = c("contig","start","end","phageName","matchingLength"),  trim_ws = TRUE) %>% filter(phageName=="Lactobacillus_phage_1_attR")
tmp1 <- data.frame(str_split_fixed(All_cleaned$contig, fixed("-"), 3))[,1:2]

All_cleaned$genome <- paste(tmp1$X1,tmp1$X2,sep="-")
table(All_cleaned$genome)
All_cleaned$genome = factor(All_cleaned$genome, levels=c("202-11141","202-11142","202-11143","202-12104","202-12105","202-12107","202-12109","202-L104","202-L108","202-L35","202-L39","202-L44","202-L70","202-L71" , "202-L80","202-L99","202-LMAG","202-13494","202-S72","202-13498","202-13493","202-13500","202-SMAG","202-13491","202-13492","202-13495","202-13496","202-13497","202-13499c1","202-13499c2","202-S50"))
table(All_cleaned$genome)

All_cleaned$phageName <- revalue(All_cleaned$phageName, c("Streptococcus_phage_1_attR"="Streptococcus phage 1 attP", "Streptococcus_phage_2_attR"="Streptococcus phage 2 attP","Lactobacillus_phage_1_attR"="Lactobacillus phage 1 attP"))


All_cleaned$grouping <- ifelse(All_cleaned$matchingLength>11,"complete","incomplete")
All_cleaned$grouping = factor(All_cleaned$grouping, levels=c("incomplete","complete"))
colorsss <- c("grey89","blue")
attachmentSites_ldel <- ggplot(All_cleaned,aes(x=genome,color=grouping,fill=grouping))+geom_bar()+theme_classic()+theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9),
          # # axis.text.y.left = element_text(color = "blue"),
          # axis.text.y.right = element_text(color="red"),
          # axis.title.y.left = element_text(color="blue"),
          # axis.title.y.right = element_text(color="red"),
          axis.title.x = element_blank(),
          # rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.title = element_blank(),
          legend.position="none")+
  scale_color_manual(values = colorsss)+
  scale_fill_manual(values = colorsss)+
  labs(y="number of attachment sites (attB)")+
  facet_wrap(~phageName,nrow = 2)
attachmentSites_ldel

svg("~/Desktop/Projects/2019_RMK202_analysis/plot/attachmentSite.svg",width=7,height=4)
attachmentSites_ldel+attachmentSites_sterm 
 dev.off()
 
```

look at location
```{bash}



rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/attachmentSite/genes
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/attachmentSite/genes
for species in $(echo "Sterm Ldel")
do

for genomess in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/ |grep ".fna$"|cut -d '_' -f 1|sed 's/.fna//g')
do
#rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA_genes.gff

grep "^202-" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.gff > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA_genes.txt

bedtools intersect -wa -a /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA_genes.txt  \
  -b /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/attachmentSite/blasting/All_cleaned_highmatches.txt |awk -F "\t" '{OFS="\t"}{if($3=="CDS")print $0}' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/attachmentSite/genes/${genomess}_genes.txt

liness=$(wc -l /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/attachmentSite/genes/${genomess}_genes.txt |cut -d ' ' -f 1)

if [ $liness == "0" ]; then
         
echo -e $liness" contains no genes"
 else
          searchsss=$(cut -f 9 /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/attachmentSite/genes/${genomess}_genes.txt)
   
grep -A 4 -B 4 "${searchsss}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA_genes.txt |awk -F "\t" '{OFS="\t"}{if($3=="CDS")print $0}' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/attachmentSite/genes/${genomess}_genes_extended.txt

echo -e $liness" contains genes"
             fi


done #genomes
done # species

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/attachmentSite/genes/*_genes.txt
cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/attachmentSite/genes/*_genes_extended.txt
```
#### mauve

```{bash}
progressiveMauve \ 
    --output=/home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/phages/Phages_for_RAST/mauve_align \ 
    --output-guide-tree=/home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/phages/Phages_for_RAST/mauve_align.guide_tree \ 
    --backbone-output=/home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/phages/Phages_for_RAST/mauve_align.backbone \ 
    /home/vincent/Downloads/S_phage_sw16_987.fasta \ 
    /home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/phages/Phages_for_RAST/Streptococcus_phage_1.fasta \
    /home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/phages/Phages_for_RAST/Streptococcus_phage_2.fasta
    
    cd /home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/phages/Phages_for_RAST/
    /home/vincent/miniconda3/bin/progressiveMauve /home/vincent/Downloads/S_phage_sw16_987.fasta  /home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/phages/Phages_for_RAST/Streptococcus_phage_1.fasta  /home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/phages/Phages_for_RAST/Streptococcus_phage_2.fasta
    
```
#### artemis

```{bash}

./act /home/vincent/Downloads/S_phage_sw16_987.fasta /home/vincent/Downloads/TJU8MV2R114-Alignment.txt /home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/phages/Phages_for_RAST/Streptococcus_phage_1.fasta

```


#### start align phage
Here, I start align the S. thermophilus phages to ther terminase

```{bash}

Baselocation=/home/vincent/Desktop/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER
output=/home/vincent/Desktop/Projects/2019_RMK202_analysis/phage_annotation/phaster/

grep "ter" -i ${Baselocation}/gene_CLEANED_FINAL_ALL.gff |grep "^Stre" -i

grep "^Stre" -i ${Baselocation}/gene_CLEANED_FINAL_ALL.gff 
##-------------------------
##Sterm phage 1
##-------------------------
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned
#/home/vincent/Desktop/Projects/2020_circleries/02_dependencies/fasta_shift -i #terminas end /home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/phages/Phages_for_RAST/Streptococcus_phage_1.fasta -p 28973 >  /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/Streptococcus_phage_1_startAligned_wrongOrientation.fasta

/home/vincent/Desktop/Projects/2020_circleries/02_dependencies/fasta_shift -i /home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/phages/Phages_for_RAST/Streptococcus_phage_1.fasta -p 29760 >  /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/Streptococcus_phage_1_startAligned_wrongOrientation.fasta


revseq -sequence /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/Streptococcus_phage_1_startAligned_wrongOrientation.fasta -outseq /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/Streptococcus_phage_1_startAligned_Final.fasta -notag 

##-------------------------
##Sterm phage 2
##-------------------------

mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned
#/home/vincent/Desktop/Projects/2020_circleries/02_dependencies/fasta_shift -i #terminas end /home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/phages/Phages_for_RAST/Streptococcus_phage_2.fasta -p 33595 >  /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/Streptococcus_phage_2_startAligned_wrongOrientation.fasta

/home/vincent/Desktop/Projects/2020_circleries/02_dependencies/fasta_shift -i /home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/phages/Phages_for_RAST/Streptococcus_phage_2.fasta -p 34785 >  /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/Streptococcus_phage_2_startAligned_wrongOrientation.fasta


revseq -sequence /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/Streptococcus_phage_2_startAligned_wrongOrientation.fasta -outseq /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/Streptococcus_phage_2_startAligned_Final.fasta -notag 


```

#### annotation
1. prodigal

```{bash}

##-------------------------
##Sterm phage 1
##-------------------------
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/Prodigal
prodigal -i /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/Streptococcus_phage_2_startAligned_Final.fasta -o /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/Prodigal/Streptococcus_phage_2_startAligned_Final.gff -f gff -a /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/Prodigal/Streptococcus_phage_2_startAligned_Final.faa -d /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/Prodigal/Streptococcus_phage_2_startAligned_Final.ffn

sed 's/*//g' /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/Prodigal/Streptococcus_phage_2_startAligned_Final.faa > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/Prodigal/Streptococcus_phage_2_startAligned_Final_cleaned.faa

##-------------------------
##Sterm phage 2
##-------------------------

mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/Prodigal
prodigal -i /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/Streptococcus_phage_1_startAligned_Final.fasta -o /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/Prodigal/Streptococcus_phage_1_startAligned_Final.gff -f gff -a /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/Prodigal/Streptococcus_phage_1_startAligned_Final.faa -d /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/Prodigal/Streptococcus_phage_1_startAligned_Final.ffn

sed 's/*//g' /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/Prodigal/Streptococcus_phage_1_startAligned_Final.faa > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/Prodigal/Streptococcus_phage_1_startAligned_Final_cleaned.faa


###============================================
##do blastP of the faa
###============================================

###============================================
##duplicate prodigal annotation and name the genes according to blast hit. 
###============================================

mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/annotation/

grep "^#" -v /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/Prodigal/Streptococcus_phage_1_startAligned_Final.gff |awk -F ";" '{print $1";name="}' >  /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/annotation/Streptococcus_phage_1_startAligned_Final_02.gff

grep "^#" -v /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/Prodigal/Streptococcus_phage_2_startAligned_Final.gff |awk -F ";" '{print $1";name="}' >  /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/annotation/Streptococcus_phage_2_startAligned_Final_02.gff

###============================================
##transfer blast annotation manually
###============================================


###============================================
##download closely related genome and annotate according to mahony paper all four genomes into categories. 
###============================================

cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/annotation/Streptococcus_phage_1_startAligned_Final_02.gff |awk '{print $0";group="}' > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/annotation/Streptococcus_phage_1_startAligned_Final_forR.gff

cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/annotation/Streptococcus_phage_2_startAligned_Final_02.gff|awk '{print $0";group="}' > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/annotation/Streptococcus_phage_2_startAligned_Final_forR.gff

###============================================
##download closely related genome 
###============================================

#grep "^NC_031023.1" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/annotation/Streptococcus_virus_9874_annotation.gff3 |awk -F "\t" '{if($3=="CDS") print $0}'|awk -F ";" '{print $1";group="}' > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/annotation/Streptococcus_virus_9874_forR.gff


grep "^NC_031023.1" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/annotation/Streptococcus_virus_9874_annotation.gff3 |awk -F "[;\t]" '{if($3=="CDS") print $1,$2,$3,$4,$5,$6,$7,$8,$15";group="}' > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/annotation/Streptococcus_virus_9874_forR.gff


grep "^MH892375.1" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/annotation/Streptococcus_phage_SW30_annotation.gff3 |awk -F "[;\t]" '{if($3=="CDS") print $1,$2,$3,$4,$5,$6,$7,$8,$15";group="}' > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_phage/startaligned/annotation/Streptococcus_phage_SW30_forR.gff


```



##KEGG
When mapping to KEGG homepage I have to manually take the information into and editor and change the columns

[KeggRest](http://bioconductor.org/packages/release/bioc/html/KEGGREST.html)
maybe usfule [link](https://rdrr.io/bioc/keggorthology/src/inst/keggHTML/parseKeg.R)

Potentially interesting R packages:
KEGGgraph
pathView
BioCor

```{r}
library(jsonlite)
library(rjson)
library(dplyr)
library(tibble) 

winners <- rjson::fromJSON(file="~/Downloads/ko00001.json")
# glimpse(winners, max.level = 3, list.len = 4)
# glimpse(winners, max.level = 4, list.len = 4)
# glimpse(winners, max.level = 5, list.len = 4)
# glimpse(winners, max.level = 6, list.len = 4)
# glimpse(winners, max.level = 7, list.len = 4)
# glimpse(winners, max.level = 8, list.len = 4)
# glimpse(winners, max.level = 9, list.len = 4)
# glimpse(winners, max.level = 10, list.len = 4)

new <- glimpse(winners, max.level = 9, list.len = 4)
new$name

new_new <- new$children[[1]]
new_new$name


new
new[[2]][[1]][[1]]

data_raw <- enframe(unlist(winners))

# winners <- jsonlite::fromJSON(file="~/Downloads/ko00001.json")
# jsonlite::fromJSON(winners)
# head(winners)

 # # cat ko00001.json |sed 's/"name"://g' |sed 's/://g'  |sed 's/,//g'|sed 's/"children"//g' > test.json
 #  cat ko00001.json |sed 's/"name"://g'  |sed 's/",//g'|sed 's/"children"://g' |sed 's/[[:blank:]]://g' > test.json
 #  # cat ko00001.json |sed 's/"name"://g' |sed 's/"children"://g' |sed 's/[[:blank:]]://g' > test.json
 #  cat ko00001.json |sed 's/"name"://g'  |sed 's/",/"/g'|sed 's/"children"//g' |sed 's/[[:blank:]]:/:/g' |sed 's/\t://g' > test.json
 #  cat ko00001.json |sed 's/"name"://g'  |sed 's/",/"/g'|sed 's/"children"//g' |sed -r 's/\t+:\[/:\[/g'  > test.json
 #  cat ko00001.json |sed 's/"name"://g'  |sed 's/",/"/g'|sed 's/"children"//g' |sed -r 's/\t+:\[/:\[/g'|sed -z 's/\n:\[/:\[/g'  > test.json

# cat ko00001.json | sed -r 's/^\t"name":/\t"class":/g' | sed -r 's/^\t\t"name":/\t\t"group":/g'| sed -r 's/^\t\t\t"name":/\t\t\t"species":/g' | sed -r 's/^\t\t\t\t"name":/\t\t\t\t"pathway":/g' | sed -r 's/^\t\t\t\t\t"name":/\t\t\t\t\t"gene":/g'   > test.json

# cat test.json | sed -r 's/^\t"children":/\t"class":/g' | sed -r 's/^\t\t"children":/\t\t"group":/g'| sed -r 's/^\t\t\t"children":/\t\t\t"species":/g' | sed -r 's/^\t\t\t\t"children":/\t\t\t\t"pathway":/g' | sed -r 's/^\t\t\t\t\t"children":/\t\t\t\t\t"gene":/g'  > test_02.json

winners <- rjson::fromJSON(file="~/Downloads/test_02.json")
data_raw <- enframe(unlist(winners))
data_raw

data_raw %>% separate(name, into = c(paste0("x", 1:5)))


```


####Go-Annotation

good [intro](https://www.bioconductor.org/packages/release/bioc/vignettes/annotate/inst/doc/GOusage.pdf)
#####topgo

interesting [question](https://www.biostars.org/p/389438/) about read mapping
this might be a [solution](http://avrilomics.blogspot.com/2015/07/using-topgo-to-test-for-go-term.html)
--> first bring data into right format

```{bash}

mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog_GO
grep -v "^#" /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Lactobacillus_delbrueckii_RMK202_pgap_query_seqs.fa.emapper.annotations | awk -F "\t" 'BEGIN{OFS="\t"} {print $1, $7}' |sed 's/gnl|extdb|//g' > /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog_GO/Lactobacillus_delbrueckii_RMK202_pgap_query_seqs.fa.emapper.annotations


grep -v "^#" /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Streptococcus_thermophilus_RMK202_pgap_query_seqs.fa.emapper.annotations | awk -F "\t" 'BEGIN{OFS="\t"} {print $1, $7}' |sed 's/gnl|extdb|//g' > /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog_GO/Streptococcus_thermophilus_RMK202_pgap_query_seqs.fa.emapper.annotations
```

...then go into R

```{r}
library(readr)
library(Rgraphviz)
library(topGO)
geneID2GO <- readMappings(file = "~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog_GO/Lactobacillus_delbrueckii_RMK202_pgap_query_seqs.fa.emapper.annotations") 
geneID2GO <- readMappings(file = "~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog_GO/Streptococcus_thermophilus_RMK202_pgap_query_seqs.fa.emapper.annotations") 


geneUniverse <- names(geneID2GO) 

# genesOfInterest <- read.table("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/Eggnog/variantCallsfreebayes_all_Lactobacillus_delbrueckii_RMK202_highImpactGeneIds_forTopGO.txt",header=FALSE)
# genesOfInterest <- read.table("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/Eggnog/variantCallsfreebayes_all_Streptococcus_thermophilus_RMK202_highImpactGeneIds_forTopGO.txt",header=FALSE)
# genesOfInterest <- read.table("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/Eggnog/variantCallsfreebayes_all_Streptococcus_thermophilus_RMK202_MODERATEImpactGeneIds_forTopGO.txt",header=FALSE)
genesOfInterest <- read.table("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/Eggnog/variantCallsfreebayes_all_Lactobacillus_delbrueckii_RMK202_MODERATEImpactGeneIds_forTopGO.txt",header=FALSE)
# genesOfInterest <- read.table("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/Eggnog/variantCallsfreebayes_all_Streptococcus_thermophilus_RMK202_highImpactGeneIds_forTopGO.txt",header=FALSE)

genesOfInterest <- as.character(genesOfInterest$V1)

table(genesOfInterest)

genesOfInterest <- genesOfInterest[genesOfInterest %in% geneUniverse]

 geneList <- factor(as.integer(geneUniverse %in% genesOfInterest))
 names(geneList) <- geneUniverse
 
 
 myGOdata <- new("topGOdata", description="My project", ontology="BP", allGenes=geneList,  annot = annFUN.gene2GO, gene2GO = geneID2GO)
 
 sg <- sigGenes(myGOdata)
 str(sg)
numSigGenes(myGOdata)
resultFisher <- runTest(myGOdata, algorithm="classic", statistic="fisher")
resultFisher

allRes <- GenTable(myGOdata, classicFisher = resultFisher, orderBy = "resultFisher", ranksOf = "classicFisher", topNodes = 200)
allRes

 showSigOfNodes(myGOdata, score(resultFisher), firstSigNodes = 5, useInfo ='all')
 printGraph(myGOdata, resultFisher, firstSigNodes = 5, fn.prefix = "tGO", useInfo = "all", pdfSW = TRUE)
 
 
 length(usedGO(myGOdata))
 
 # myterms = c("GO:0006207", "GO:0019856", "GO:0046112")
 # mygenes <- genesInTerm(myGOdata, myterms)
 #  for (i in 1:length(myterms))
 #   {
 #       myterm <- myterms[i]
 #       mygenesforterm <- mygenes[myterm][[1]]
 #       mygenesforterm <- paste(mygenesforterm, collapse=',')
 #       print(paste("Term",myterm,"genes:",mygenesforterm))
 #     }
 
 
```


```{r}

library(thacklr)
library(readr)
library(dplyr)
library(tidyverse)

Bos_taurus_mito <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Bos_taurus_mito_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3))

L104 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L104_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3))
L108 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L108_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3))
L35 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L35_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3))
L39 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L39_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3))
L44 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L44_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3))
L70 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L70_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3))
L71 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L71_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3))
L80 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L80_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3))
L99 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L99_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3))


Lactobacillus_delbrueckii_plasmid <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Lactobacillus_delbrueckii_plasmid_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3))


S50 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/S50_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3)) %>% add_column(.,sample="S50")
S72 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/S72_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3)) %>% add_column(.,sample="S72")
Streptococcus_thermophilus_RMK202 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Streptococcus_thermophilus_RMK202_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3)) %>% add_column(.,sample="Streptococcus_thermophilus_RMK202")


ALL_sterm <- rbind(S50,S72,Streptococcus_thermophilus_RMK202)
colnames(ALL_sterm) <- c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description","genome")


S50[grep("GO:0005215",S50$GOs),]
nrow(S50[grep("GO:0005215",S50$GOs),])

##----split multi-column into multiple rows

library(tidyr)
## The ";\\s+" means that the separator is a ";" followed by one or more spaces
separate_rows(df,abilities,sep=";\\s+")

```

####COG analysis

```{r}
library(tidyverse)
library(thacklr)
library(readr)
library(dplyr)
library(ggplot2)

COG_functional_categories <- read_delim("~/Desktop/ScriptRepository/COG_functional_categories.csv",  ",", escape_double = FALSE, col_names = c("short","long","divided"),trim_ws = TRUE, na = "NA")

S50 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/S50_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE, na = "NA") %>% slice( 1:(n()-3)) %>% add_column(.,sample="S50")
S72 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/S72_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE, na = "NA") %>% slice( 1:(n()-3)) %>% add_column(.,sample="S72")
Streptococcus_thermophilus_RMK202 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Streptococcus_thermophilus_RMK202_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE, na = "NA") %>% slice( 1:(n()-3)) %>% add_column(.,sample="Streptococcus_thermophilus_RMK202")


ALL_sterm <- rbind(S50,S72,Streptococcus_thermophilus_RMK202)
colnames(ALL_sterm) <- c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description","genome")

nrow(ALL_sterm)

#unique(sterm_all_unique$COG_Functional_Category)[match(COG_functional_categories$short , unique(sterm_all_unique$COG_Functional_Category))]
unique(ALL_sterm$COG_Functional_Category)[which(!unique(ALL_sterm$COG_Functional_Category) %in% COG_functional_categories$short )]

# table(ALL_sterm$COG_Functional_Category)

ALL_sterm$longCOG <- NA
for (i in 1:nrow(ALL_sterm)) {
  ALL_sterm[i,"longCOG"] <- COG_functional_categories[which(COG_functional_categories$short==as.character(ALL_sterm[i,"COG_Functional_Category"])),"long"]
  
}



CogPlotSterm <- ggplot(ALL_sterm,aes(x=longCOG,fill=genome))+geom_bar(position = "dodge2")+
  theme_classic()+
  labs(x="")+
  theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9),
          # # axis.text.y.left = element_text(color = "blue"),
          # axis.text.y.right = element_text(color="red"),
          # axis.title.y.left = element_text(color="blue"),
          # axis.title.y.right = element_text(color="red"),
          # rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          #legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank(),
          plot.margin=unit(c(0,0,0,2), "cm"),
          axis.text.x.bottom = element_text(size=10)
          #legend.position = c(0.02, 0.98),
          #legend.justification = c("left", "top")
          )
CogPlotSterm
   svg("~/Desktop/presenations/my_presentations/20190807_committeeMeeting/figures/cogPlot_allSterm.svg",width=7,height=7)
CogPlotSterm 
 dev.off()
 
##--------------
 ##Ldel
##--------------
COG_functional_categories <- read_delim("~/Desktop/ScriptRepository/COG_functional_categories.csv",  ",", escape_double = FALSE, col_names = c("short","long","divided"),trim_ws = TRUE, na = "NA")
 

L104 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L104_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE, na = "NA") %>% slice( 1:(n()-3)) %>% add_column(.,sample="L104") 
L108 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L108_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE, na = "NA") %>% slice( 1:(n()-3)) %>% add_column(.,sample="L108")
L35 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L35_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE, na = "NA") %>% slice( 1:(n()-3)) %>% add_column(.,sample="L35")
L39 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L39_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE, na = "NA") %>% slice( 1:(n()-3)) %>% add_column(.,sample="L39")
L44 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L44_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE, na = "NA") %>% slice( 1:(n()-3)) %>% add_column(.,sample="L44")
L70 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L70_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE, na = "NA") %>% slice( 1:(n()-3)) %>% add_column(.,sample="L70")
L71 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L71_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE, na = "NA") %>% slice( 1:(n()-3)) %>% add_column(.,sample="L71")
L80 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L80_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE, na = "NA") %>% slice( 1:(n()-3)) %>% add_column(.,sample="L80")
L99 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L99_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE, na = "NA") %>% slice( 1:(n()-3)) %>% add_column(.,sample="L99")

Lactobacillus_delbrueckii_RMK202 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Lactobacillus_delbrueckii_RMK202_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE, na = "NA") %>% slice( 1:(n()-3)) %>% add_column(.,sample="Lactobacillus_delbrueckii_RMK202")


ALL_sterm <- rbind(L104,L108,L35,L39,L44,L70,L71,L80,L99,Lactobacillus_delbrueckii_RMK202)
colnames(ALL_sterm) <- c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description","genome")

nrow(ALL_sterm)

#unique(sterm_all_unique$COG_Functional_Category)[match(COG_functional_categories$short , unique(sterm_all_unique$COG_Functional_Category))]
unique(ALL_sterm$COG_Functional_Category)[which(!unique(ALL_sterm$COG_Functional_Category) %in% COG_functional_categories$short )]

# table(ALL_sterm$COG_Functional_Category)

ALL_sterm$longCOG <- NA
for (i in 1:nrow(ALL_sterm)) {
  ALL_sterm[i,"longCOG"] <- COG_functional_categories[which(COG_functional_categories$short==as.character(ALL_sterm[i,"COG_Functional_Category"])),"long"]
  
}



CogPlotSterm <- ggplot(ALL_sterm,aes(x=longCOG,fill=genome))+geom_bar(position = "dodge2")+
  theme_classic()+
  labs(x="")+
  theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9),
          # # axis.text.y.left = element_text(color = "blue"),
          # axis.text.y.right = element_text(color="red"),
          # axis.title.y.left = element_text(color="blue"),
          # axis.title.y.right = element_text(color="red"),
          # rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          #legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank(),
          plot.margin=unit(c(0,0,0,2), "cm"),
          axis.text.x.bottom = element_text(size=10)
          #legend.position = c(0.02, 0.98),
          #legend.justification = c("left", "top")
          )
CogPlotSterm
   svg("~/Desktop/presenations/my_presentations/20190807_committeeMeeting/figures/cogPlot_allLdel.svg",width=7,height=7)
CogPlotSterm 
 dev.off()
 

```




#### ANI between the strains

```{bash FASTANI}
date=20190123
source /archiv/Projects/2018_Culturomics/02_script/fileShortcuts.sh
threads=35
id=RMK202
/archiv/Projects/2018_Culturomics

###-------------------
##Lactobacillus delbrueckii
###-------------------

##make list of all files
species=Lactobacillus_delbrueckii

mainPath=/archiv/Projects/2018_Culturomics
rm /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_GenomePath.txt
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/
for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/names_changed.txt | grep "^L")
do
  cp ${mainPath}/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/contigs.fasta ${mainPath}/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/${id}.fasta
  echo ${mainPath}/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/${id}.fasta >> /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_GenomePath.txt

done
##with MAG



echo "/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_ldel/final/L_delbrueckii_mag_rmk202.fasta" >> /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_GenomePath.txt



fastANI -t 5 --ql /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_GenomePath.txt \
  --rl /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_GenomePath.txt \
  -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_fastani.txt

sed 's/\/archiv\/Projects\/2018_Culturomics\/06_Spades\/L[0-9]\+\/assembly\/assemblyMetaSpades_20190121.fasta\///g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_fastani.txt |sed 's/\/archiv\/Projects\/2019_Nano_Meta\/02_flye_Assembly\/02_raw_demultiplexed\/di_K2_6h\/Polishing_FINAL\/circlatorAfter\/\/circlaring_ldel\/final\///g' > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_fastani_forR.txt


#scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/Lactobacillus_delbrueckii/Lactobacillus_delbrueckii_fastani_forR.txt ~/Desktop/Projects/2018_culturomics/fastani/


###-------------------
##Streptococcus thermophilus
###-------------------

##make list of all files
species=Streptococcus_thermophilus

mainPath=/archiv/Projects/2018_Culturomics
rm /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_GenomePath.txt
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/

for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/names_changed.txt | grep "^S")
do

  cp ${mainPath}/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/contigs.fasta ${mainPath}/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/${id}.fasta
  echo ${mainPath}/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/${id}.fasta >> /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_GenomePath.txt

done


echo "/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_sterm/final/S_thermophilus_mag_rmk202.fasta" >> /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_GenomePath.txt


##FASTANI

fastANI -t 37 --ql /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_GenomePath.txt \
  --rl /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_GenomePath.txt \
  -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_fastani.txt
  

sed 's/\/archiv\/Projects\/2019_Nano_Meta\/02_flye_Assembly\/02_raw_demultiplexed\/di_K2_6h\/Polishing_FINAL\/circlatorAfter\/\/circlaring_sterm\/final\///g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_fastani.txt |sed 's/\/archiv\/Projects\/2018_Culturomics\/06_Spades\/S[0-9]\+\/assembly\/assemblyMetaSpades_20190121.fasta\///g' > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_fastani_forR.txt

#cp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/Streptococcus_thermophilus/Streptococcus_thermophilus_fastani_forR.txt ~/Desktop/Projects/2018_culturomics/fastani/


```
make [correlation heatmap](http://www.sthda.com/english/wiki/ggplot2-quick-correlation-matrix-heatmap-r-software-and-data-visualization)
```{r fastaniInR}
library(stringr)
library(readr)
library(ggplot2)
library(reshape2)


##----------------
##sterm
##----------------


Sterm_fastANI_nwc <- read_delim("~/Desktop/Projects/2018_culturomics/fastani/Streptococcus_thermophilus_fastani_forR.txt", "\t", escape_double = FALSE, col_names = FALSE,  trim_ws = TRUE) 
Sterm_fastANI_nwc$AF <- (100*(Sterm_fastANI_nwc$X4/Sterm_fastANI_nwc$X5))


##shorten names by removing path to genomes
Sterm_fastANI_nwc$new_X1 <- gsub("/archiv/Projects/2018_Culturomics/06_Spades/assembly/","",Sterm_fastANI_nwc$X1) 
Sterm_fastANI_nwc$new_X2 <- gsub("/archiv/Projects/2018_Culturomics/06_Spades/assembly/","",Sterm_fastANI_nwc$X2) 


##shorten names by ".fasta"
Sterm_fastANI_nwc$new_X1 <- gsub(".fasta","",Sterm_fastANI_nwc$X1) 
Sterm_fastANI_nwc$new_X2 <- gsub(".fasta","",Sterm_fastANI_nwc$X2) 

##rename genome names
library(plyr)
Sterm_fastANI_nwc$new_X1 <- revalue(Sterm_fastANI_nwc$new_X1, c("S50"="mst1", "S72"="mst2", "Streptococcus_thermophilus_RMK202"="RMK202"))
Sterm_fastANI_nwc$new_X2 <- revalue(Sterm_fastANI_nwc$new_X2, c("S50"="mst1", "S72"="mst2", "Streptococcus_thermophilus_RMK202"="RMK202"))



##remove upper triangle and keep only lower ANI scores
names <- levels(as.factor(Sterm_fastANI_nwc$new_X2))


##creat ANI matrix 
count <- 1
Sterm_fastANI_nwc$combined <- paste(Sterm_fastANI_nwc$new_X1,Sterm_fastANI_nwc$new_X2,sep="_")
sterm_curated <- Sterm_fastANI_nwc[1,]

for (i in 1:length(names)) {
   rowName <- names[i]
    print(rowName)
  for(a in 1:count){
    
    colNames <- names[a]
    print(colNames)
    search_01 <- paste(rowName,colNames,sep="_")
    search_02 <- paste(colNames,rowName,sep="_")
    
     ifelse(
      Sterm_fastANI_nwc[which(Sterm_fastANI_nwc$combined==search_01),"X3"]>=Sterm_fastANI_nwc[which(Sterm_fastANI_nwc$combined==search_02),"X3"],
      temp <- Sterm_fastANI_nwc[which(Sterm_fastANI_nwc$combined==search_02),],temp <-Sterm_fastANI_nwc[which(Sterm_fastANI_nwc$combined==search_01),]
        
    )
    print(temp)
    temp$new_X2 <- colNames
    temp$new_X1 <- rowName

    sterm_curated <- rbind(sterm_curated,temp)
    
    
  }
        
count <- count +1 
  
}
sterm_curated[-1,]



# creat Heatmap
sterm_curated$X3 <- round(sterm_curated$X3,digits = 2)

ggheatmap <- ggplot(sterm_curated, aes(new_X1, new_X2, fill = X3))+
 geom_tile(color = "white",size=1.1)+
 scale_fill_gradient2(low = "blue", high = "red", mid = "white", 
   midpoint = 99.5, limit = c(98.9,100), space = "Lab", 
   name="ANI") +
  theme_minimal()+ # minimal theme
 theme(axis.text.y = element_text(size = 12),
      axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1))+
 coord_fixed()
# Print the heatmap
print(ggheatmap)

plotFinal_sterm <- ggheatmap + 
geom_text(aes(new_X1, new_X2, label = X3), color = "black", size = 4) +
  scale_y_discrete(position = 'right')+
theme(
  axis.title.x = element_blank(),
  axis.title.y = element_blank(),
  panel.grid.major = element_blank(),
  panel.border = element_blank(),
  panel.background = element_blank(),
  axis.ticks = element_blank(),
  legend.justification = c(1, 0),
  legend.position = c(0.4, 0.7),
  legend.text = element_text(angle = 325),
  legend.direction = "horizontal")+
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1,
                title.position = "top", title.hjust = 0.5))

##save heatmap
svg("~/Desktop/Projects/2018_culturomics/fastani/Streptococcus_thermophilus_fastANI_with_MAG.svg",width=2.5,height=2.5)
plotFinal_sterm+theme(legend.position = "none")
dev.off()
# 
# png("~/Desktop/Projects/2018_culturomics/fastani/Streptococcus_thermophilus_fastANI_with_nwcS.png",width=400,height=400)
# plotFinal_sterm
# dev.off()

plotFinal_sterm

##----------------
##ldel
##----------------


  Ldel_fastANI_nwc <- read_delim("~/Desktop/Projects/2018_culturomics/fastani/Lactobacillus_delbrueckii_fastani_forR.txt", "\t", escape_double = FALSE, col_names = FALSE,  trim_ws = TRUE)

Ldel_fastANI_nwc$AF <- (100*(Ldel_fastANI_nwc$X4/Ldel_fastANI_nwc$X5))
##change names
Ldel_fastANI_nwc$new_X1 <- gsub(".fasta","",Ldel_fastANI_nwc$X1) 
Ldel_fastANI_nwc$new_X2 <- gsub(".fasta","",Ldel_fastANI_nwc$X2) 

##remove L39

Ldel_fastANI_nwc <- Ldel_fastANI_nwc[Ldel_fastANI_nwc$new_X1!="L39",]
Ldel_fastANI_nwc <- Ldel_fastANI_nwc[Ldel_fastANI_nwc$new_X2!="L39",]
##rename

Ldel_fastANI_nwc$new_X1 <- revalue(Ldel_fastANI_nwc$new_X1, c("L70"="mst3", "L44"="mst10", "L108"="mst7", "L99"="mst8", "L104"="mst6", "L80"="mst5", "L35"="mst9", "L71"="mst4", "L104"="mst6","Lactobacillus_delbrueckii_RMK202"="RMK202"))
Ldel_fastANI_nwc$new_X2 <- revalue(Ldel_fastANI_nwc$new_X2, c("L70"="mst3", "L44"="mst10", "L108"="mst7", "L99"="mst8", "L104"="mst6", "L80"="mst5", "L35"="mst9", "L71"="mst4", "L104"="mst6","Lactobacillus_delbrueckii_RMK202"="RMK202"))

not_explained_sterm_plot$sample_renamed = factor(not_explained_sterm_plot$sample_renamed, levels=c("Lyoampulle\n2012","Lyoampulle\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018"))

##remove upper triangle and keep only lower ANI score
names <- levels(as.factor(Ldel_fastANI_nwc$new_X2))

count <- 1

Ldel_fastANI_nwc$combined <- paste(Ldel_fastANI_nwc$new_X1,Ldel_fastANI_nwc$new_X2,sep="_")
Ldel_curated <- Ldel_fastANI_nwc[1,]

for (i in 1:length(names)) {
   rowName <- names[i]
    print(rowName)
  for(a in 1:count){
    
    colNames <- names[a]
    print(colNames)
    search_01 <- paste(rowName,colNames,sep="_")
    search_02 <- paste(colNames,rowName,sep="_")
    
     ifelse(
      Ldel_fastANI_nwc[which(Ldel_fastANI_nwc$combined==search_01),"X3"]>=Ldel_fastANI_nwc[which(Ldel_fastANI_nwc$combined==search_02),"X3"],
      temp <- Ldel_fastANI_nwc[which(Ldel_fastANI_nwc$combined==search_02),],temp <-Ldel_fastANI_nwc[which(Ldel_fastANI_nwc$combined==search_01),]
        
    )
    #print(temp)
    temp$new_X1 <- colNames
    temp$new_X2 <- rowName
    
    Ldel_curated <- rbind(Ldel_curated,temp)
    
    
  }
        
count <- count +1 
  
}
Ldel_curated <- Ldel_curated[-1,]



# new <- Ldel_curated[
#   with(Ldel_curated, order(match(new_X1,rev(names)))),
# ]





# Heatmap
Ldel_curated$X3 <- round(Ldel_curated$X3,digits = 2)

ggheatmap <- ggplot(Ldel_curated,aes(x=new_X1, y=new_X2, fill = X3))+
 geom_tile(color = "white",size=1.1)+
 scale_fill_gradient2(low = "blue", high = "red", mid = "white", 
   midpoint = 99.5, limit = c(98.9,100), space = "Lab", 
   name="ANI") +
  theme_minimal()+ # minimal theme
 theme(axis.text.x = element_text(angle = 45, vjust = 1,size = 12, hjust = 0,),
  axis.text.y = element_text(size = 12))+
 coord_fixed()
# Print the heatmap
print(ggheatmap)


plotFinal_ldel <- ggheatmap + 
geom_text(aes(new_X1, new_X2, label = X3), color = "black", size = 3) +
  scale_x_discrete(position = 'top')+
theme(
  axis.title.x = element_blank(),
  axis.title.y = element_blank(),
  panel.grid.major = element_blank(),
  panel.border = element_blank(),
  panel.background = element_blank(),
  axis.ticks = element_blank(),
  legend.justification = c(1, 0),
  legend.position = c(1, 0.1),
  legend.text = element_text(angle = 325),
  legend.direction = "horizontal")+
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1,
                title.position = "top", title.hjust = 0.5))


  svg("~/Desktop/Projects/2018_culturomics/fastani/Lactobacillus_delbrueckii_fastANI_with_MAG.svg",width=5,height=5)
plotFinal_ldel+theme(legend.position = "none")
dev.off()
# 
# png("~/Desktop/Projects/2018_culturomics/fastani/Lactobacillus_delbrueckii_fastANI_with_nwcS.png",width=400,height=400)
# plotFinal_ldel
# dev.off()



  svg("~/Desktop/Projects/2018_culturomics/fastani/Lactobacillus_delbrueckii_fastANI_with_MAG_legend.svg",width=5.8,height=5.8)
plotFinal_ldel
dev.off()

```
##Plasmid scan
I want to look if the plasmids harbour anything relevant
```{bash}


##-------------
##extract all genes faa for eggnog
##-------------
##===========================
##Streptococcus
##===========================
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/plasmids
rm /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/plasmids/CP046135.faa
samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/S_thermophilus_RMK202_genes.faa
for idss in $(grep "CP046135" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/S_thermophilus_RMK202.current.gff |awk -F "[\t;]" '{if($3=="CDS") print $9}'|sed 's/ID=//g'|sed 's/.p01//g')
do

samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/S_thermophilus_RMK202_genes.faa ${idss} >> /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/plasmids/CP046135.faa

done
##===========================
##Lactoacillus
##===========================

grep ">CP" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/L_delbrueckii_RMK202.current.gff|sort|uniq -c

#----------CP046132
rm /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/plasmids/CP046132.faa
samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/S_thermophilus_RMK202_genes.faa
for idss in $(grep "CP046132" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/L_delbrueckii_RMK202.current.gff |awk -F "[\t;]" '{if($3=="CDS") print $9}'|sed 's/ID=//g'|sed 's/.p01//g')
do

samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/L_delbrueckii_RMK202_genes.faa ${idss} >> /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/plasmids/CP046132.faa

done

#----------CP046133
rm /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/plasmids/CP046133.faa
samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/S_thermophilus_RMK202_genes.faa
for idss in $(grep "CP046133" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/L_delbrueckii_RMK202.current.gff |awk -F "[\t;]" '{if($3=="CDS") print $9}'|sed 's/ID=//g'|sed 's/.p01//g')
do

samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/L_delbrueckii_RMK202_genes.faa ${idss} >> /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/plasmids/CP046133.faa

done

```
move plasmid local for eggnog

```{bash}

mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/11_plasmidsMAG
scp -r vincent@130.223.51.116:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/plasmids/ /home/vincent/Desktop/Projects/2019_Pilotplan/11_plasmidsMAG
```
I annote one single gene with eggnog!?! Can this be?
Alternative PROKKA, RAST

Here, I scan the predicted products of the plasmid annotations

```{bash}

##===========================
##Streptococcus
##===========================
grep "CP046135" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/S_thermophilus_RMK202.current.gff| sed 's/;eC_number//g' |awk -F "[\t;]" '{OFS="\t"}{if($3=="CDS") print $9,$15}'


##===========================
##Lactoacillus
##===========================

grep "CP046132" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/L_delbrueckii_RMK202.current.gff| sed 's/;eC_number//g' |awk -F "[\t;]" '{OFS="\t"}{if($3=="CDS") print $9,$15}'


grep "CP046133" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/L_delbrueckii_RMK202.current.gff| sed 's/;eC_number//g' |awk -F "[\t;]" '{OFS="\t"}{if($3=="CDS") print $9,$15}'


```
PROKKA
```{bash}


##===========================
##Streptococcus
##===========================
id=CP046135
rm -r $/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PROKKA_plasmids/$id
    mkdir -p $/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PROKKA_plasmids/$id
    
 samtools faidx  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/S_thermophilus_RMK202.current.fna $id > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/${id}.fna
    
/usr/bin/perl /usr/bin/prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PROKKA_plasmids/$id --force --usegenus --genus Streptococcus --locustag $id --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/${id}.fna
grep "${id}"  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PROKKA_plasmids/$id/PROKKA*.gff |sed 's/Parent.*product=//g' |awk -F "[\t;]" '{OFS="\t"}{if($3=="CDS") print $9,$10}'


##===========================
##Lactoacillus
##===========================

id=CP046132
rm -r $/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PROKKA_plasmids/$id
    mkdir -p $/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PROKKA_plasmids/$id
    
 samtools faidx  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/L_delbrueckii_RMK202.current.fna $id > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/${id}.fna
    
/usr/bin/perl /usr/bin/prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PROKKA_plasmids/$id --force --usegenus --genus Lactobacillus --locustag $id --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/${id}.fna
grep "${id}"  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PROKKA_plasmids/$id/PROKKA*.gff |sed 's/Parent.*product=//g' |awk -F "[\t;]" '{OFS="\t"}{if($3=="CDS") print $9,$10}'



##===========================
##Lactoacillus
##===========================

id=CP046133
rm -r $/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PROKKA_plasmids/$id
    mkdir -p $/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PROKKA_plasmids/$id
    
 samtools faidx  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/L_delbrueckii_RMK202.current.fna $id > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/${id}.fna
    
/usr/bin/perl /usr/bin/prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PROKKA_plasmids/$id --force --usegenus --genus Lactobacillus --locustag $id --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/${id}.fna
grep "${id}"  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PROKKA_plasmids/$id/PROKKA*.gff |sed 's/Parent.*product=//g' |awk -F "[\t;]" '{OFS="\t"}{if($3=="CDS") print $9,$10}'



##---------------------final overview
grep "CP046135"  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PROKKA_plasmids/CP046135/PROKKA*.gff |sed 's/Parent.*product=//g' |awk -F "[\t;]" '{OFS="\t"}{if($3=="CDS") print $9,$10}'


grep "$CP046132"  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PROKKA_plasmids/CP046132/PROKKA*.gff |sed 's/Parent.*product=//g' |awk -F "[\t;]" '{OFS="\t"}{if($3=="CDS") print $9,$10}'
grep "$CP046133"  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PROKKA_plasmids/CP046133/PROKKA*.gff |sed 's/Parent.*product=//g' |awk -F "[\t;]" '{OFS="\t"}{if($3=="CDS") print $9,$10}'
```

Tyrosine recombinase XerC is in the second PLASMID of Lactobacillus
Chromosome partitioning protein ParA in the second PLASMID of Lactobacillus

Nothing to excited....
Therefore blast the faa

```{bash}
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PROKKA_plasmids/forBlast/
cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PROKKA_plasmids/CP046135/PROKKA*.faa >  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PROKKA_plasmids/forBlast/CP046135.faa

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PROKKA_plasmids/CP046132/PROKKA*.faa >  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PROKKA_plasmids/forBlast/CP046132.faa

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PROKKA_plasmids/CP046133/PROKKA*.faa >  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PROKKA_plasmids/forBlast/CP046133.faa
```
bring local for blast
```{bash}
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PROKKA_plasmids/forBlast/

mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/11_plasmidsMAG
scp -r vincent@130.223.51.116:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PROKKA_plasmids/forBlast/ /home/vincent/Desktop/Projects/2019_Pilotplan/11_plasmidsMAG
```
#### Circos Plasmid

Here, I creat the circos plots for the bacteria. 
I will include the following information:
1. genome (circul)
2. forward genes  
3. reverse genes
4. hypthecial genes

```{bash}

##==================
##Sterm
##==================
home=/home/vincent/bin/apps/circos-0.69-6/
cd $home

mkdir -p $home/data/rmk202/MAG_rmk202_sterm_plasmid_1/


##organisational 
#cp /home/vincent/bin/apps/circos-0.69-6/etc/ticks_mag_both.conf /home/vincent/bin/apps/circos-0.69-6/etc/ticks_mag_both_plasmids.conf

#cp /home/vincent/bin/apps/circos-0.69-6/etc/ideogram_mag.conf /home/vincent/bin/apps/circos-0.69-6/etc/ideogram_mag_plasmids.conf
#cp /home/vincent/bin/apps/circos-0.69-6/etc/sterm_rmk202_mag_circos.conf /home/vincent/bin/apps/circos-0.69-6/etc/sterm_plasmid1_rmk202_mag_circos.conf


mkdir -p $home/data/rmk202/MAG_rmk202_sterm/
##---------------------karyotyp
#seqlength.py "/home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202 (2).fasta" |tail -1 > ${home}/data/karyotype/MAG_rmk202_sterm_plasmid_1.txt
##have to change the format a bit

##---------------------genes
#cd /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm
#/usr/bin/perl /home/vincent/miniconda3/bin/bp_genbank2gff3.pl S_thermophilus_RMK202.current.gb
cd $home
grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |awk -F "\t" '{OFS="\t"}{if($1=="CP046135" && $3=="CDS" && $7=="+") print "S_thermophilus_repA",$4,$5}' > $home/data/rmk202/MAG_rmk202_sterm_plasmid_1/genes_forward.txt
grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |awk -F "\t" '{OFS="\t"}{if($1=="CP046135" && $3=="CDS" && $7=="-") print "S_thermophilus_repA",$4,$5}' > $home/data/rmk202/MAG_rmk202_sterm_plasmid_1/genes_reverse.txt

#grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |awk -F "\t" '{OFS="\t"}{if($1=="CP046135" && $3=="CDS") print $0}' |grep "hypo" -v
##---------------------hypothetical 

grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff|grep "hypo" |awk -F "\t" '{OFS="\t"}{if($1=="CP046135" && $3=="CDS") print "S_thermophilus_repA",$4,$5}' > $home/data/rmk202/MAG_rmk202_sterm_plasmid_1/hyothical.txt

##example
#cp etc/repeat_nwc1_sterm_01.conf etc/sterm_rmk202_mag_circos.conf

#example run
bin/circos -conf etc/sterm_plasmid1_rmk202_mag_circos.conf -outputfile ./sterm_plasmid1_mag_rmk202.svg; firefox ./sterm_plasmid1_mag_rmk202.svg &
  

##==================
##Ldel plasmid 1
##==================
home=/home/vincent/bin/apps/circos-0.69-6/
cd $home

##organisational 

#cp /home/vincent/bin/apps/circos-0.69-6/etc/sterm_plasmid1_rmk202_mag_circos.conf /home/vincent/bin/apps/circos-0.69-6/etc/ldel_plasmid1_rmk202_mag_circos.conf
mkdir -p $home/data/rmk202/MAG_rmk202_ldel_plasmid_1/

##---------------------karyotyp

#seqlength.py /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.fasta |head -2|tail -1 > ${home}/data/karyotype/MAG_rmk202_ldel_plasmid_1.txt


##have to change the format a bit

grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.current.gb.gff  |grep "^CP046132"
##---------------------genes
#cd /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm
#/usr/bin/perl /home/vincent/miniconda3/bin/bp_genbank2gff3.pl S_thermophilus_RMK202.current.gb
cd $home
grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.current.gb.gff |awk -F "\t" '{OFS="\t"}{if($1=="CP046132" && $3=="CDS" && $7=="+") print "L_del_plasmid_01",$4,$5}' > $home/data/rmk202/MAG_rmk202_ldel_plasmid_1/genes_forward.txt
grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.current.gb.gff |awk -F "\t" '{OFS="\t"}{if($1=="CP046132" && $3=="CDS" && $7=="-") print "L_del_plasmid_01",$4,$5}' > $home/data/rmk202/MAG_rmk202_ldel_plasmid_1/genes_reverse.txt

#grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.current.gb.gff |awk -F "\t" '{OFS="\t"}{if($1=="CP046132" && $3=="CDS") print $0}' 
##---------------------hypothetical 

grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.current.gb.gff|grep "hypo" |awk -F "\t" '{OFS="\t"}{if($1=="CP046132" && $3=="CDS") print "L_del_plasmid_01",$4,$5}' > $home/data/rmk202/MAG_rmk202_ldel_plasmid_1/hyothical.txt


#example run
bin/circos -conf etc/ldel_plasmid1_rmk202_mag_circos.conf -outputfile ./ldel_plasmid1_mag_rmk202.svg; firefox ./ldel_plasmid1_mag_rmk202.svg &


##==================
##Ldel plasmid 2
##==================
home=/home/vincent/bin/apps/circos-0.69-6/
cd $home

##organisational 

#cp /home/vincent/bin/apps/circos-0.69-6/etc/sterm_plasmid1_rmk202_mag_circos.conf /home/vincent/bin/apps/circos-0.69-6/etc/ldel_plasmid2_rmk202_mag_circos.conf
mkdir -p $home/data/rmk202/MAG_rmk202_ldel_plasmid_2/

##---------------------karyotyp

#seqlength.py /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.fasta |tail -1 > ${home}/data/karyotype/MAG_rmk202_ldel_plasmid_2.txt


##have to change the format a bit


##---------------------genes
#cd /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm
#/usr/bin/perl /home/vincent/miniconda3/bin/bp_genbank2gff3.pl S_thermophilus_RMK202.current.gb
cd $home
grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.current.gb.gff |awk -F "\t" '{OFS="\t"}{if($1=="CP046133" && $3=="CDS" && $7=="+") print "L_del_plasmid_02",$4,$5}' > $home/data/rmk202/MAG_rmk202_ldel_plasmid_2/genes_forward.txt
grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.current.gb.gff |awk -F "\t" '{OFS="\t"}{if($1=="CP046133" && $3=="CDS" && $7=="-") print "L_del_plasmid_02",$4,$5}' > $home/data/rmk202/MAG_rmk202_ldel_plasmid_2/genes_reverse.txt

#grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.current.gb.gff |awk -F "\t" '{OFS="\t"}{if($1=="CP046133" && $3=="CDS") print $0}' 
##---------------------hypothetical 

grep "^#" -v /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.current.gb.gff|grep "hypo" |awk -F "\t" '{OFS="\t"}{if($1=="CP046133" && $3=="CDS") print "L_del_plasmid_02",$4,$5}' > $home/data/rmk202/MAG_rmk202_ldel_plasmid_2/hyothical.txt


#example run
bin/circos -conf etc/ldel_plasmid2_rmk202_mag_circos.conf -outputfile ./ldel_plasmid2_mag_rmk202.svg; firefox ./ldel_plasmid2_mag_rmk202.svg &

```



## Phage scan
after rast we scan the phage annotations.
I do this be downloading the "genomic directory" and by looking at the proposed functions file therin.



## SNP matrix

Here, I try to creat a SNP matrix based on progressive mauve and looking only at sites that are covered by all genomes

```{bash}
##progressive mauve


##----------
##remove SNPs that are not complete
##---------

cut -f 1 Strepto_rmk202_snps.txt | grep "-" -v  > Strepto_rmk202_snps_complete.txt

sed '1d' Strepto_rmk202_snps_complete.txt |tr [a-z] [A-Z]   |sed 's/./&\t/g' > Strepto_rmk202_snps_complete_split.txt

```

```{r}
library(readr)
Strepto_core_snps <- Strepto_core_snps[,-4]

Sterm_matrix <- setNames(data.frame(matrix(ncol = ncol(Strepto_core_snps), nrow = ncol(Strepto_core_snps))), colnames(Strepto_core_snps))
row.names(Sterm_matrix) <-  colnames(Strepto_core_snps)

##testcases
cols=as.numeric("1")
alternativcols=as.numeric("2")

##loop
for (cols in 1:ncol(Strepto_core_snps)) {
  colnamess <- colnames(Strepto_core_snps)[cols]
  
     Sterm_matrix[cols,cols] <- sum(Strepto_core_snps[,cols]!=Strepto_core_snps[,cols])

  
  for (alternativcols in grep(colnamess,colnames(Strepto_core_snps),invert = TRUE)) {
    
   Sterm_matrix[alternativcols,cols] <- sum(Strepto_core_snps[,cols]!=Strepto_core_snps[,alternativcols])

  }

}


Sterm_matrix_rel <- 100*(Sterm_matrix/nrow(Strepto_core_snps))
Sterm_matrix_tot <- 100-100*(Sterm_matrix/1600000) ##this could be ANI


```



####Nucmer:repeat analysis

Here, I use [nucmer](http://mummer.sourceforge.net/) to do the analysis of the largest repeats

```{bash}

###==========================
##sterm
###==========================

mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/{fasta,nucmer,deltaFiltered}

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Streptococcus_thermophilus_RMK202.fasta \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Lactobacillus_delbrueckii_RMK202.fasta > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/fasta/reads.fa

###==========================
##nucmer
###==========================
cd /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer
nucmer -maxmatch -nosimplify /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/fasta/reads.fa /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/fasta/reads.fa


###==========================
##delta filter
###==========================
#mkdir -p $DATA_ROOT/11_nucmer_repeat/cottens_abruijn_delbruecki_streptococcus/{fasta,nucmer,deltaFiltered}
delta-filter -i 85 -l 3000  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/out.delta > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/filtered.txt
##split files
cd /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/
split -t ">" -l 1 /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/filtered.txt

##--------------
##extract Ldel repeats
##--------------
grep "[[:space:]]" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/xab | awk 'BEGIN {OFS ="\t"} {print "L_delbrueckii_RMK202" , $1 , $2 , "L_delbrueckii_RMK202" , $3 , $4}' > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/links_ldel.txt
cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/links_ldel.txt > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/links_ldel_curated.txt
##remove first row and reduant repeats

awk '{print $3-$2}' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/links_ldel_curated.txt |sort


##--------------
##extract sterm repeats
##--------------

grep "[[:space:]]" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/xad | awk 'BEGIN {OFS ="\t"} {print "S_thermophilus_RMK202" , $1 , $2 , "S_thermophilus_RMK202" , $3 , $4}' > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/links_sterm.txt
cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/links_sterm.txt > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/links_sterm_curated.txt
##remove first row and reduant repeats

awk '{print $3-$2}' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/links_sterm_curated.txt |sort


```

##SCOG

##### roary

```{bash}

###================
##description file
###================

Outputlocation=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/03_PROKKA_annotation_polishedGenome
Assembly_sterm=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_sterm/final/S_thermophilus_mag_rmk202.fasta
Assembly_ldel=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_ldel/final/L_delbrueckii_mag_rmk202.fasta
BaseFILNAME_ldel=L_delbrueckii_mag_rmk202
BaseFILNAME_sterm=S_thermophilus_mag_rmk202
locusTagSterm=Sterm_202
locusTagLdel=Ldel_202
Genus_Sterm=Streptococcus
Genus_Ldel=Lactobacillus
threads=37


###================
##script
###================


mkdir -p /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Prokka/{Sterm,Ldel}
##-----------------
###01_annotate Sterm rmk202 with prokka
##-----------------

##Already done in PROKKA chunk

##-----------------
###02_roary
##-----------------

##-----sterm
rm -r /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm
mkdir -p /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/strains

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/03_PROKKA_annotation_polishedGenome/S_thermophilus_mag_rmk202/*.gff > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/strains/Streptococcus_thermophilus_RMK202.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/strains/S72.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/strains/S50.gff

roary -f /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/ -p 5 -e -n -v /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/strains/*.gff

dateName=$(ls /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/|grep "^_")

awk -F '","' '{print $4}' /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/${dateName}/gene_presence_absence.csv |sort |uniq -c

#awk -F '","' '{print $17}' /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/_1571727203/gene_presence_absence.csv |head
mkdir -p /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes
#sudo chmod 777 /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/_1571727203/gene_presence_absence.csv
awk -F '","' '{if ($4=="3") print $17}' /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/${dateName}/gene_presence_absence.csv |sed 's/"//g' |sed 's/^ *//'|sed 's/* $//'  > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202.txt


###--------
##extract genes
###--------
rm /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202.vcf
for ids in $(cat /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202.txt |sed 's/\r//g')
  do
  
  echo "${ids}---"
  grep "Parent=${ids}_gene" /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/strains/Streptococcus_thermophilus_RMK202.gff >> /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202.vcf
echo "------------"

  done
  
  
  
###--------
##change contig name
###--------  

  
    sed 's/^S_thermophilus_mag_rmk202/CP046134/g'   /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202.vcf > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202_renamed.vcf
  
  
##=====================
##-----Ldel
##=====================


rm -r /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/
mkdir -p /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/strains/


cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/03_PROKKA_annotation_polishedGenome/L_delbrueckii_mag_rmk202/*.gff > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/strains/Lactobacillus_delbrueckii_RMK202.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L104//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/strains/L104.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L108//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/strains/L108.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L35//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/strains/L35.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L39//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/strains/L39.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L44//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/strains/L44.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L70//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/strains/L70.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L71//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/strains/L71.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L99//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/strains/L99.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L80//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/strains/L80.gff


sudo roary -f /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/ -p 25 -e -n -v /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/strains/*.gff




dateName=$(ls /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/|grep "^_")

awk -F '","' '{print $4}' /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/${dateName}/gene_presence_absence.csv |sort |uniq -c

#awk -F '","' '{print $17}' /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/_1571727203/gene_presence_absence.csv |head
mkdir -p /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/CoreGenes
#sudo chmod 777 /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/_1571727203/gene_presence_absence.csv
awk -F '","' '{if ($4=="10") print $24}' /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/${dateName}/gene_presence_absence.csv |sed 's/"//g' |sed 's/^ *//'|sed 's/* $//'  > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/CoreGenes/coreGeneNames_rmk202.txt


###--------
##extract genes
###--------
rm /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/CoreGenes/coreGeneNames_rmk202.vcf
for ids in $(cat /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/CoreGenes/coreGeneNames_rmk202.txt |sed 's/\r//g')
  do
  
  echo "${ids}---"
  grep "Parent=${ids}_gene" /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/strains/Lactobacillus_delbrueckii_RMK202.gff >> /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/CoreGenes/coreGeneNames_rmk202.vcf
echo "------------"

  done

###--------
##change contig name
###--------  

  
    sed 's/^L_delbrueckii_mag_rmk202/CP046131/g'  /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/CoreGenes/coreGeneNames_rmk202.vcf > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/CoreGenes/coreGeneNames_rmk202_renamed.vcf
  
##=====================
##-----merge species
##=====================
  

cat /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202_renamed.vcf  \
  /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/CoreGenes/coreGeneNames_rmk202_renamed.vcf |awk -F "\t" '{OFS="\t"}{print $1,$2,$3,$4,$5}'> \
  /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/coreGeneNames_rmk202_renamed_all.vcf

##=====================
##-----bring local
##=====================

mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Sterm
scp vincent@130.223.51.116:/archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/coreGeneNames_rmk202_renamed_all.vcf /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/

#sed 's/"//g' /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/gene_presence_absence.csv > /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/gene_presence_absence_woQuotes.csv

```

##Biolog
Here, I analyse the Biolog data of the different strains.

```{r}
library(readr)
library(ggplot2)
library(plyr)
library(tidyverse)
library(opm)
library(patchwork)
Biolog_all_plates <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/Biolog/strains/Export_All_Plates_Single_File.csv", ";", escape_double = FALSE, trim_ws = TRUE)
Biolog_all_plates$`Field 2`


ggplot(Biolog_all_plates,aes(x=Hour,y=A06,group=`Field 2`,color=`Field 2`))+geom_line()

       
Biolog_long <- gather(Biolog_all_plates, well, intensity, A01:H12, factor_key=TRUE,na.rm = TRUE) 
Biolog_long$intensity <- as.double(Biolog_long$intensity)
Biolog_long$Hour <- as.double(Biolog_long$Hour)

str(Biolog_long)
p1 <- ggplot(Biolog_long,aes(x=Hour,y=intensity,group=well,color=well))+geom_line()+facet_wrap(`Field 2`~.)+theme_classic()
p1
png("~/Desktop/Projects/2019_RMK202_analysis/biolog/plots/Biolog_sterm_rmk202_strains_01.png", width = 4000, height = 4000,res=300)
 # svg("~/Desktop/Projects/2019_RMK202_analysis/10_assembly/heatmap_all_Sterm_strains.svg",width=14,height=14)
p1
dev.off()

##---------------------------------------------
##growth (binary)
##calculate different of highest and lowest intensity value
##---------------------------------------------
Biolog_diff <- aggregate(. ~`Field 2`+well, data=Biolog_long[,c("Field 2","well","intensity")], FUN = function(i)max(i) - min(i))
hist(Biolog_diff$intensity,breaks = 20)
pDensity <- ggplot(Biolog_diff,aes(x=intensity,group=`Field 2`,color=`Field 2`,fill=`Field 2`))+geom_density()+theme_classic()+facet_wrap(`Field 2`~.)+geom_vline(xintercept =90)+labs(x="Intensity difference between maximal and minimal value")+theme(legend.position = "none")


Biolog_diff_high <- Biolog_diff %>% filter(intensity>90) 
Biolog_diff_high$well <- droplevels(Biolog_diff_high$well)
Biolog_diff_high$`Field 2` <- droplevels(Biolog_diff_high$`Field 2`)
table(Biolog_diff_high$well,Biolog_diff_high$`Field 2`)

Biolog_diff %>%  filter(well=="C09")

Biolog_diff_high_ext <- Biolog_diff %>% filter(well %in% c(as.character(unique(Biolog_diff_high$well)),"A06"))

##---------------------------------------------
##make heatmap of strain growth 
##only use the carbohydrates that have at least one strain that grows
##---------------------------------------------


Biolog_diff_wide <- spread(Biolog_diff, `Field 2`, intensity) %>% column_to_rownames(var="well")
Biolog_diff_high_wide <- spread(Biolog_diff_high, `Field 2`, intensity) %>% column_to_rownames(var="well")
Biolog_diff_high_wide_ext <- spread(Biolog_diff_high_ext, `Field 2`, intensity) %>% column_to_rownames(var="well")

rownames(Biolog_diff_high_wide_ext) <- revalue(rownames(Biolog_diff_high_wide_ext), c( "A02"="L-Arabinose", "B08"="D-Xylose", "B11"="D-Mannitol", "C04"="D-Ribose" ,"C07"="D-Fructose", "C09"="alpha-D-Glucose", "D09"="alpha-D-Lactose" ,"D10"="Lactulose", "D11"="Sucrose", "G08"="N-Acetyl-Beta-D-Mannosamine" ,"H06"="L-Lyxose","A06"="D-Galactose"))

# rownames(Biolog_diff_high_wide_ext)

if (!require("gplots")) {
   install.packages("gplots", dependencies = TRUE)
   library(gplots)
   }
if (!require("RColorBrewer")) {
   install.packages("RColorBrewer", dependencies = TRUE)
   library(RColorBrewer)
   }


# mat_data <- as.matrix(Biolog_diff_wide) # transform column 2-5 into a matrix
# mat_data <- as.matrix(Biolog_diff_high_wide) # transform column 2-5 into a matrix
mat_data <- as.matrix(Biolog_diff_high_wide_ext) # transform column 2-5 into a matrix

my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 119)
# col_breaks = c(seq(0,max(mat_data),length=300))
# length(col_breaks)
col_breaks = c(seq(min(mat_data),60,length=40),  # for red
  seq(61,90,length=40),           # for yellow
  seq(91,max(mat_data),length=40))

str(mat_data)
dev.off()
    # png("~/Desktop/Projects/2019_RMK202<_analysis/10_assembly/heatmap_all_Sterm_strains.png", width = 4000, height = 4000,res=300)
 # svg("~/Desktop/Projects/2019_RMK202_analysis/10_assembly/heatmap_all_Sterm_strains.svg",width=14,height=14)
pHeatmap <- heatmap.2(mat_data,
  # cellnote = mat_data,  # same data set for cell labels
  main = "", # heat map title
  notecol="black",      # change font color of cell labels to black
  density.info="none",
  trace="none",         # turns off trace lines inside the heat map
 margins = c(16, 16),     # widens margins around plot
  col=my_palette,   
  breaks=col_breaks,    # enable color transition at specified limits
  sepcolor="black",
  sepwidth=c(0.5,0.5),
  dendrogram="none",
 Colv=FALSE)

svg("~/Desktop/Projects/2019_RMK202_analysis/biolog/plots/DenstiyPlot_biolog_02.svg",width=8,height=8)
heatmap.2(mat_data,
  # cellnote = mat_data,  # same data set for cell labels
  main = "", # heat map title
  notecol="black",      # change font color of cell labels to black
  density.info="none",
  trace="none",         # turns off trace lines inside the heat map
 margins = c(16, 16),     # widens margins around plot
  col=my_palette,   
  breaks=col_breaks,    # enable color transition at specified limits
  sepcolor="black",
  sepwidth=c(0.5,0.5),
  dendrogram="none",
 Colv=FALSE)
dev.off()
# dev.off()

  
  svg("~/Desktop/Projects/2019_RMK202_analysis/biolog/plots/DenstiyPlot_biolog_01.svg",width=5,height=5)
   pDensity
  dev.off()

```


## Genome assembly stats

####PGAP annotation

creat all files from the PGAP annoation

```{bash}

scp /home/vincent/Downloads/L_delbrueckii_RMK202.current.gb vincent@130.223.51.116:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel
scp /home/vincent/Downloads/S_thermophilus_RMK202.current.gb vincent@130.223.51.116:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm


###================
##description file
###================

BaseLocation_Ldel=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/
gb_file_ldel=L_delbrueckii_RMK202.current.gb
BaseFILNAME=L_delbrueckii_RMK202


BaseLocation_Sterm=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/
gb_file_sterm=S_thermophilus_RMK202.current.gb
BaseFILNAME_sterm=S_thermophilus_RMK202




!!!!!!!!!!!!!!!check genome name

##----------------------------------------
##---------Ldel
##----------------------------------------

mkdir -p ${BaseLocation_Ldel}
##---------fna
any2fasta ${BaseLocation_Ldel}/${gb_file_ldel} > ${BaseLocation_Ldel}/${BaseFILNAME}.fna

##---------gff complete
/usr/bin/bp_genbank2gff3 ${BaseLocation_Ldel}/${gb_file_ldel} --outdir - > ${BaseLocation_Ldel}/${BaseFILNAME}.gff


##---------gff complete
cat ${BaseLocation_Ldel}/${BaseFILNAME}.gff | grep -v "^CP046}" |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' > ${BaseLocation_Ldel}/${BaseFILNAME}_gene.gff

##---------gff gene
cat /${BaseLocation_Ldel}/${BaseFILNAME}.gff | grep -v "^CP046}" |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |sed 's/ID=//g' |sed 's/;Name.*//g' |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $1,$4,$5,$9,$8,$7}' >  ${BaseLocation_Ldel}/${BaseFILNAME}_gene.bed


##---------gene ffn
bedtools getfasta -s -name  -fi ${BaseLocation_Ldel}/${BaseFILNAME}.fna \
  -bed ${BaseLocation_Ldel}/${BaseFILNAME}_gene.bed  | sed 's/::.*//g' > ${BaseLocation_Ldel}/${BaseFILNAME}_genes.ffn
  
  
##---------gene faa
transeq  -sequence ${BaseLocation_Ldel}/${BaseFILNAME}_genes.ffn > \
  -outseq  ${BaseLocation_Ldel}/${BaseFILNAME}_genes.faa
sed -i 's/_1$//g'  ${BaseLocation_Ldel}/${BaseFILNAME}_genes.faa

grep ">" ${BaseLocation_Ldel}/${BaseFILNAME}_genes.faa |sed 's/>//g' |cut -d '_' -f 1|sort|uniq -c

##----------------------------------------
##---------Sterm
##----------------------------------------

mkdir -p ${BaseLocation_Sterm}
##---------fna
any2fasta ${BaseLocation_Sterm}/${gb_file_sterm} > ${BaseLocation_Sterm}/${BaseFILNAME_sterm}.fna

##---------gff complete
/usr/bin/bp_genbank2gff3 ${BaseLocation_Sterm}/${gb_file_sterm} --outdir - > ${BaseLocation_Sterm}/${BaseFILNAME_sterm}.gff


##---------gff complete
cat ${BaseLocation_Sterm}/${BaseFILNAME_sterm}.gff | grep -v "^CP046}" |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' > ${BaseLocation_Sterm}/${BaseFILNAME_sterm}_gene.gff

##---------gff gene
cat /${BaseLocation_Sterm}/${BaseFILNAME_sterm}.gff | grep -v "^CP046}" |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $0}' |sed 's/ID=//g' |sed 's/;Name.*//g' |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $1,$4,$5,$9,$8,$7}' >  ${BaseLocation_Sterm}/${BaseFILNAME_sterm}_gene.bed


##---------gene ffn
bedtools getfasta -s -name  -fi ${BaseLocation_Sterm}/${BaseFILNAME_sterm}.fna \
  -bed ${BaseLocation_Sterm}/${BaseFILNAME_sterm}_gene.bed  | sed 's/::.*//g' > ${BaseLocation_Sterm}/${BaseFILNAME_sterm}_genes.ffn
  
  
##---------gene faa
transeq  -sequence ${BaseLocation_Sterm}/${BaseFILNAME_sterm}_genes.ffn > \
  -outseq  ${BaseLocation_Sterm}/${BaseFILNAME_sterm}_genes.faa
sed -i 's/_1$//g'  ${BaseLocation_Sterm}/${BaseFILNAME_sterm}_genes.faa

grep ">" ${BaseLocation_Sterm}/${BaseFILNAME_sterm}_genes.faa |sed 's/>//g' |cut -d '_' -f 1|sort|uniq -c

###================
##bring local
###================
mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/sterm/
mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/ldel/

scp vincent@130.223.51.116:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm//S_thermophilus_RMK202_genes.faa /home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/sterm/

scp vincent@130.223.51.116:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel//L_delbrueckii_RMK202_genes.faa /home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/ldel/



```


#### BUSCO

```{bash}


##-----------------------------------
##move faa files to the directory
##-----------------------------------

cp /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/vincent2/output/L_delbrueckii_RMK202/L_delbrueckii_RMK202_mag.faa /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/


cp /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/vincent2/output/S_thermophilus_RMK202/S_thermophilus_RMK202_mag.faa /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/faa/


##-----------------------------------
##script
##-----------------------------------
rm /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/all_analysis.txt

species=Sterm
species=Ldel
speciesss=Ldel
for speciesss in $(echo "Sterm Ldel")
do
echo "##------------------------"
echo ${speciesss}
echo "##------------------------"

for genemess in $(ls /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${speciesss}/faa/ )
  do
  #genemess=L_delbrueckii_RMK202_mag.faa
  genomesssss=$(echo ${genemess}|cut -d '.' -f 1)
  
  rm -r /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/${genemess}
  rm -r /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/${genomesssss}

  mkdir -p /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/${genomesssss}
  cd /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/${genomesssss}
  
  python /home/vincent/apps/busco/scripts/run_BUSCO.py -c 32 -i /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${speciesss}/faa/${genemess} -o ${genomesssss} -l /home/vincent/apps/busco/lactobacillales_odb9 -m prot

  #grep -v "^#" run_${genomesssss}/short_summary_${genomesssss}.txt| grep "C:" -A 7 
  all_INFO=$(grep -v "^#" run_${genomesssss}/short_summary_${genomesssss}.txt| grep "C:" | sed "s/^[ \t]*//"  )
  complet=$(echo ${all_INFO} | awk -F "[,[]" '{print $1}'|sed 's/%//g'|sed 's/C://g' )
  single=$(echo ${all_INFO} | awk -F "[,[]" '{print $2}'|sed 's/%//g'|sed 's/S://g')
  dupli=$(echo ${all_INFO} | awk -F "[,[]" '{print $3}'|sed 's/%]//g'|sed 's/D://g')
  fragm=$(echo ${all_INFO} | awk -F "[,[]" '{print $4}'|sed 's/%//g'|sed 's/F://g')
  Missi=$(echo ${all_INFO} | awk -F "[,[]" '{print $5}'|sed 's/%//g'|sed 's/M://g')
  numbi=$(echo ${all_INFO} | awk -F "[,[]" '{print $6}'|sed 's/n://g')
  
 echo -e ${speciesss}"\t"${genomesssss}"\t"${complet}"\t"${single}"\t"${dupli}"\t"${fragm}"\t"${Missi}"\t"${numbi}  \
 >> /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/all_analysis.txt

  done
done ## species

sed -i  's/Streptococcus_thermophilus_RMK202_pgap/S_thermophilus_mag_rmk202/g' /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/all_analysis.txt
sed -i  's/Lactobacillus_delbrueckii_RMK202_pgap/L_delbrueckii_mag_rmk202/g' /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/all_analysis.txt

```

#### remove small contigs

```{bash}




for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/isolates_names_changed.txt )
  do
  echo ==========================
   echo ${id}
   echo ==========================
 

  seqkit seq -m 300 /archiv/Projects/2018_Culturomics/06_Spades/${id}/Prokka/*/*19.fna -o /archiv/Projects/2018_Culturomics/06_Spades/${id}/Prokka/20190203/${id}_wo_300bp.fna
 
    done

```



#### genome stats

```{bash}

##---------------
##genome
##---------------

rm /archiv/Projects/2018_Culturomics/06_Spades//all_assembly_analysis.txt
for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/isolates_names_changed.txt )
  do
  echo ==========================
   echo ${id}
   echo ==========================
 
# /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/
 
   contiNumber=$(grep ">" -c /archiv/Projects/2018_Culturomics/06_Spades/${id}/Prokka/*/${id}_wo_300bp.fna) 
   GenomeLength=$(grep -v "^>" /archiv/Projects/2018_Culturomics/06_Spades/${id}/Prokka/*/${id}_wo_300bp.fna  |wc -c)
   
   # contiNumber=$(grep ">" -c /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/contigs.fasta) 
   #GenomeLength=$(grep -v "^>" /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/contigs.fasta  |wc -c)
    
 echo -e ${id}"\t"${contiNumber}"\t"${GenomeLength}  >> /archiv/Projects/2018_Culturomics/06_Spades//all_assembly_analysis.txt

#"\t"${single}"\t"${dupli}"\t"${fragm}"\t"${Missi}"\t"${numbi}  \
    done



##---------------
##MAG
##---------------
##---------Sterm

id=S_thermophilus_mag_rmk202

contiNumber=$(grep ">" -c /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/vincent2/output/S_thermophilus_RMK202//S_thermophilus_RMK202.fasta)
GenomeLength=$(grep -v "^>" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/vincent2/output/S_thermophilus_RMK202//S_thermophilus_RMK202.fasta  |wc -c)

 echo -e ${id}"\t"${contiNumber}"\t"${GenomeLength}  >> /archiv/Projects/2018_Culturomics/06_Spades//all_assembly_analysis.txt
 
##---------Ldel
id=L_delbrueckii_mag_rmk202

contiNumber=$(grep ">" -c /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/vincent2/output/L_delbrueckii_RMK202//L_delbrueckii_RMK202.fasta)
GenomeLength=$(grep -v "^>" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/vincent2/output/L_delbrueckii_RMK202//L_delbrueckii_RMK202.fasta  |wc -c)

 echo -e ${id}"\t"${contiNumber}"\t"${GenomeLength}  >> /archiv/Projects/2018_Culturomics/06_Spades//all_assembly_analysis.txt
 
 
```

#### annotation stats

```{bash}


------

rm /archiv/Projects/2018_Culturomics/06_Spades//all_annotation_analysis.txt
for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/isolates_names_changed.txt )
  do
  echo ==========================
   echo ${id}
   echo ==========================
 
#   contiNumber=$(grep ">" -c /archiv/Projects/2018_Culturomics/06_Spades/${id}/Prokka/2*/*19.gff) 
#   GenomeLength=$(grep -v "^>" /archiv/Projects/2018_Culturomics/06_Spades/${id}/Prokka/2*/*19.gff  |wc -c)
  
  gensss=$(grep "gene: " /archiv/Projects/2018_Culturomics/06_Spades/${id}/Prokka/201*/*19.txt |sed 's/gene: //g')
  tRNAsss=$(grep "tRNA: " /archiv/Projects/2018_Culturomics/06_Spades/${id}/Prokka/201*/*19.txt |sed 's/tRNA: //g')
  rRNAsss=$(grep "rRNA: " /archiv/Projects/2018_Culturomics/06_Spades/${id}/Prokka/201*/*19.txt |sed 's/rRNA: //g')
  
  
 echo -e ${id}"\t"${gensss}"\t"${tRNAsss}"\t"${rRNAsss}  >> /archiv/Projects/2018_Culturomics/06_Spades//all_annotation_analysis.txt

#"\t"${single}"\t"${dupli}"\t"${fragm}"\t"${Missi}"\t"${numbi}  \
    done

###-----------Sterm RMK202 MAG
gbkFile=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/vincent2/output/S_thermophilus_RMK202/annot.gbk
id=S_thermophilus_mag_rmk202
  gensss=$(grep "Genes (total)                     :: " ${gbkFile}|sed 's/Genes (total)                     :: //g'|sed "s/^[ \t]*//"|sed 's/,//g')
  tRNAsss=$(grep "tRNAs                             :: " ${gbkFile} |sed 's/tRNAs                             :: //g'|sed "s/^[ \t]*//"|sed 's/,//g')
  rRNAsss=$(grep "Genes (RNA)                       :: " ${gbkFile} |sed 's/Genes (RNA)                       :: //g'|sed "s/^[ \t]*//"|sed 's/,//g')
  
  
 echo -e ${id}"\t"${gensss}"\t"${tRNAsss}"\t"${rRNAsss}  >> /archiv/Projects/2018_Culturomics/06_Spades//all_annotation_analysis.txt

    
  
###-----------Sterm RMK202 MAG
gbkFile=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/vincent2/output/L_delbrueckii_RMK202//annot.gbk
id=L_delbrueckii_mag_rmk202

  gensss=$(grep "Genes (total)                     :: " ${gbkFile}|sed 's/Genes (total)                     :: //g'|head -1 |sed "s/^[ \t]*//" |sed 's/,//g' )
  tRNAsss=$(grep "tRNAs                             :: " ${gbkFile} |sed 's/tRNAs                             :: //g'|head -1 |sed "s/^[ \t]*//" |sed 's/,//g' )
  rRNAsss=$(grep "Genes (RNA)                       :: " ${gbkFile} |sed 's/Genes (RNA)                       :: //g'|head -1 |sed "s/^[ \t]*//" |sed 's/,//g' )
  
 echo -e ${id}"\t"${gensss}"\t"${tRNAsss}"\t"${rRNAsss}  >> /archiv/Projects/2018_Culturomics/06_Spades//all_annotation_analysis.txt

```

####mapping stats

```{bash}


for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/isolates_names_changed.txt)
  do
  echo ==========================
   echo ${id}
   echo ==========================
    

 rm -r /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/PostaAssembly
mkdir -p /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/PostaAssembly
  
  
  bwa index /archiv/Projects/2018_Culturomics/06_Spades/${id}/Prokka/20190203/${id}_wo_300bp.fna
  
  bwa mem -t 32 /archiv/Projects/2018_Culturomics/06_Spades/${id}/Prokka/20190203/${id}_wo_300bp.fna \
  /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${id}/${id}_R1_val_1.fq /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${id}/${id}_R2_val_2.fq | samtools sort -@$32 -O BAM -o /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/PostaAssembly/${id}_rawIllumina_bwamem_sorted.bam - 

##-----Qualimap

mkdir -p /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/PostaAssembly//bamqc

qualimap bamqc -bam /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/PostaAssembly/${id}_rawIllumina_bwamem_sorted.bam -outdir /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/PostaAssembly//bamqc --java-mem-size=80G

    
    done


##---------------------
##get data
##---------------------

rm /archiv/Projects/2018_Culturomics/06_Spades//all_genome_analysis.txt
for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/isolates_names_changed.txt)
  do

GCss=$(grep "GC percentage" /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/PostaAssembly//bamqc/genome_results.txt| sed 's/GC percentage = //g'| sed "s/^[ \t]*//"|sed 's/%//g'  )

coveragesss=$(grep "mean coverageData = " /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/PostaAssembly//bamqc/genome_results.txt| sed 's/mean coverageData = //g'| sed "s/^[ \t]*//"|sed 's/X//g'  )



#grep -A 10000 ">>>>>>> Coverage per contig" /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/PostaAssembly//bamqc/genome_results.txt| grep -v ">>>>>>> Coverage per contig"| sed "s/^[ \t]*//" |sed -r '/^\s*$/d' 


echo -e ${id}"\t"${GCss}"\t"${coveragesss}"\t""NA" >> /archiv/Projects/2018_Culturomics/06_Spades//all_genome_analysis.txt

done

##---------------------
##get data ONT coverage
##---------------------
## GC
infoseq /archiv/Projects/2019_RMK202_analysis/02_flye_Assembly/merged/RMK202_merged.fasta -noheading -nodatabase -nolength -nousa -nohead -noacc -notype -nodesc -delimiter '_' -outfile /archiv/Projects/2019_RMK202_analysis/02_flye_Assembly/merged/RMK202_mergeg_infoseq.txt


##--------------Sterm
oldName=contig_1
NewName=S_thermophilus_mag_rmk202
GCss=38
## illumina both
IlluminaCoverage=$(grep ">>>>>>> Coverage per contig" -A 20 /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/rawIlluminaMapped/bamqc/genome_results.txt | grep "$oldName" |sed "s/^[ \t]*//"|cut -f 4) 

## ONT both
ONTCoverage=$(grep ">>>>>>> Coverage per contig" -A 20 /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/04_fourthRacon_graphmap/Graphmap_ONT/bamqc/genome_results.txt | grep "$oldName" |sed "s/^[ \t]*//"|cut -f 4)

## GC
GCss=$(grep "S_thermophilus_RMK202" /archiv/Projects/2019_RMK202_analysis/02_flye_Assembly/merged/RMK202_mergeg_infoseq.txt | awk -F " " '{print $2}')

## alltogether
echo -e ${NewName}"\t"${GCss}"\t"${IlluminaCoverage}"\t"${ONTCoverage}  >> /archiv/Projects/2018_Culturomics/06_Spades//all_genome_analysis.txt



##--------------Ldel

oldName=contig_2
NewName=L_delbrueckii_mag_rmk202
GCss=38
## illumina both
IlluminaCoverage=$(grep ">>>>>>> Coverage per contig" -A 20 /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/rawIlluminaMapped/bamqc/genome_results.txt | grep "$oldName" |sed "s/^[ \t]*//"|cut -f 4) 

## ONT both
ONTCoverage=$(grep ">>>>>>> Coverage per contig" -A 20 /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/04_fourthRacon_graphmap/Graphmap_ONT/bamqc/genome_results.txt | grep "$oldName" |sed "s/^[ \t]*//"|cut -f 4)

## GC
GCss=$(grep "L_delbrueckii_RMK202" /archiv/Projects/2019_RMK202_analysis/02_flye_Assembly/merged/RMK202_mergeg_infoseq.txt | awk -F " " '{print $2}')

## alltogether
echo -e ${NewName}"\t"${GCss}"\t"${IlluminaCoverage}"\t"${ONTCoverage}  >> /archiv/Projects/2018_Culturomics/06_Spades//all_genome_analysis.txt


```

#### bring together
```{bash}

cat /archiv/Projects/2018_Culturomics/06_Spades//all_assembly_analysis.txt
cat /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/all_analysis.txt
cat /archiv/Projects/2018_Culturomics/06_Spades//all_genome_analysis.txt
cat /archiv/Projects/2018_Culturomics/06_Spades//all_annotation_analysis.txt

mkdir -p /archiv/Projects/2018_Culturomics/06_Spades/genomeAnalysi/
cp /archiv/Projects/2018_Culturomics/06_Spades//all_assembly_analysis.txt /archiv/Projects/2018_Culturomics/06_Spades/genomeAnalysi/
cp /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/all_analysis.txt /archiv/Projects/2018_Culturomics/06_Spades/genomeAnalysi/
cp /archiv/Projects/2018_Culturomics/06_Spades//all_genome_analysis.txt /archiv/Projects/2018_Culturomics/06_Spades/genomeAnalysi/
cp /archiv/Projects/2018_Culturomics/06_Spades//all_annotation_analysis.txt /archiv/Projects/2018_Culturomics/06_Spades/genomeAnalysi/

scp -r vincent@130.223.51.151:/archiv/Projects/2018_Culturomics/06_Spades/genomeAnalysi/ ~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/

```

```{r}
library(readr)
genome_stats <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/genomeAnalysi/all_assembly_analysis.txt", "\t", escape_double = FALSE, col_names = c("strain","contigNumber","genomeSize"),  trim_ws = TRUE)
Genes_stats <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/genomeAnalysi/all_annotation_analysis.txt", "\t", escape_double = FALSE, col_names = c("strain","genes","tRNA","rRNA"),  trim_ws = TRUE)
BUSCO_stats <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/genomeAnalysi/all_analysis.txt", "\t", escape_double = FALSE, col_names = c("species","strain","completness","singleCopy","duplicated","fragmentation","Missing","numberGenesTested"),  trim_ws = TRUE)
mapping_stats <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/genomeAnalysi/all_genome_analysis.txt", "\t", escape_double = FALSE, col_names =c("strain","GC","IlluminaCoverage","ONTcoverage"),  trim_ws = TRUE)

tmp <- merge(BUSCO_stats,genome_stats,by="strain")
tmp <- merge(tmp,Genes_stats,by="strain")
GenomeAssemblyStats <- merge(tmp,mapping_stats,by="strain")

##----------------------cleaning--------------------------------
table(GenomeAssemblyStats$strain)
GenomeAssemblyStats$species <- revalue(GenomeAssemblyStats$species, c("Sterm"="S.thermophilus","Ldel"="L.delbrueckii"))

GenomeAssemblyStats$strain <- revalue(GenomeAssemblyStats$strain, c("S50"="mst1","S72"="mst2","L70"="mst3", "L44"="mst10", "L108"="mst7", "L99"="mst8", "L104"="mst6", "L80"="mst5", "L35"="mst9", "L71"="mst4","L_delbrueckii_mag_rmk202"="Ldel_mag_202","S_thermophilus_mag_rmk202"="Sterm_mag_202"))

GenomeAssemblyStats <- GenomeAssemblyStats[-which(GenomeAssemblyStats$strain=="L39"),]
colnames(GenomeAssemblyStats)

GenomeAssemblyStats_final <- GenomeAssemblyStats %>% select(species,strain,contigNumber,genomeSize,GC,IlluminaCoverage,ONTcoverage,genes,tRNA,rRNA,completness)

##----------------------write table--------------------------------
  
write.csv(GenomeAssemblyStats_final,file="~/Desktop/presenations/my_presentations/20191007_groupmeeting/figures/genomeAssemblyStats.csv")
# 
# write.xlsx(GenomeAssemblyStats_final,file="Users//Desktop/presenations/my_presentations/20191007_groupmeeting/figures/genomeAssemblyStats.xls", sheetName = "Sheet1",
#   col.names = TRUE, row.names = FALSE, append = FALSE)

```

## Annotation analysis

######Competence system

Here, we look if the Streptococcus thermophilus has the componenents for the competence system [ComR](https://www.ncbi.nlm.nih.gov/protein/116100479) as described in [2016](https://journals.plos.org/plospathogens/article?id=10.1371/journal.ppat.1005980)
```{bash}

##------------transfer to bigmachine

scp /home/vincent/Downloads/ComR_Sterm.faa vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/05_ComR_competence/blast/

##------------make blastDB 

makeblastdb -max_file_sz 1GB -in /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/assemblyFinalwithIsolates.fasta \
 -out /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/assemblyFinalwithIsolates -parse_seqids -dbtype pr

##------------blast


  tblastn -num_threads 37 -max_hsps 3 -max_target_seqs 3 -show_gis \
  -query /archiv/Projects/2019_RMK202_analysis/05_ComR_competence/blast/ComR_Sterm.faa \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen" \
  -db /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/assemblyFinalwithIsolates  \
  -out /archiv/Projects/2019_RMK202_analysis/05_ComR_competence/blast/blastCompetence.txt \
  -evalue 0.01 -word_size 4

##-----------------------------------------------Protein blast
##--------make blastDB 

makeblastdb -max_file_sz 1GB -in /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/S_thermophilus_RMK202_genes.faa \
 -out /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/S_thermophilus_RMK202_genes -parse_seqids -dbtype prot

##------------blast


  blastp -num_threads 37 -max_hsps 3 -max_target_seqs 3 -show_gis \
  -query /archiv/Projects/2019_RMK202_analysis/05_ComR_competence/blast/ComR_Sterm.faa \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen" \
  -db /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/S_thermophilus_RMK202_genes  \
  -out /archiv/Projects/2019_RMK202_analysis/05_ComR_competence/blast/blastCompetence_proteins.txt \
  -evalue 0.01 -word_size 4
  
  grep "GKC13_08270" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/S_thermophilus_RMK202.gff


```
There seems to be a compentence regulator


##compare to old strains

downloaded and now moving to bigmachine

```{bash}

scp -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/12_oldStrains vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/
```


```{bash}

###---------------------------
##make a list of locations of all Sterm and ldel genomes to comapre
###---------------------------

##-------------------------------Ldel
rm /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/filenames_Ldel.txt
mkdir -p /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/

for i in $(ls /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Dialact/Ldel/ |grep ".fna")
do

echo "/archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Dialact/Ldel/"${i} >> /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/filenames_Ldel.txt

done
echo "/archiv/Projects/2018_Culturomics/genomes/L80.fna" >> /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/filenames_Ldel.txt
echo "/archiv/Projects/2018_Culturomics/genomes/L99.fna" >> /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/filenames_Ldel.txt
echo "/archiv/Projects/2018_Culturomics/genomes/L71.fna" >> /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/filenames_Ldel.txt
echo "/archiv/Projects/2018_Culturomics/genomes/L70.fna" >> /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/filenames_Ldel.txt
echo "/archiv/Projects/2018_Culturomics/genomes/L44.fna" >> /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/filenames_Ldel.txt
echo "/archiv/Projects/2018_Culturomics/genomes/L39.fna" >> /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/filenames_Ldel.txt
echo "/archiv/Projects/2018_Culturomics/genomes/L35.fna" >> /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/filenames_Ldel.txt
echo "/archiv/Projects/2018_Culturomics/genomes/L108.fna" >> /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/filenames_Ldel.txt
echo "/archiv/Projects/2018_Culturomics/genomes/L104.fna" >> /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/filenames_Ldel.txt


samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished.fasta CP046131 > /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/ldel_MAG.fasta
echo "/archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/ldel_MAG.fasta" >> /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/filenames_Ldel.txt

##-------------------------------Sterm

rm /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/filenames_Sterm.txt
mkdir -p /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/

for i in $(ls /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Dialact/Sterm/ |grep ".fna")
do

echo "/archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Dialact/Sterm/"${i} >> /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/filenames_Sterm.txt

done

echo "/archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/S50.fna" >> /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/filenames_Sterm.txt 
echo "/archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/S72.fna" >> /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/filenames_Sterm.txt

samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished.fasta CP046134 > /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/Sterm_MAG.fasta
echo "/archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/Sterm_MAG.fasta" >> /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/filenames_Sterm.txt


###---------------------------
##fastaANI
###---------------------------
###----------------------Ldel
mkdir -p /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/fastANI/Ldel
fastANI --fragLen 1000 --minFrag 5 -t 37 --ql /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/filenames_Ldel.txt \
  --rl /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/filenames_Ldel.txt \
  -o /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/fastANI/Ldel/ANI_output.txt

###----------------------Sterm

mkdir -p /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/fastANI/Sterm
fastANI --fragLen 1000 --minFrag 5 -t 37 --ql /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/filenames_Sterm.txt \
  --rl /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/filenames_Sterm.txt \
  -o /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/fastANI/Sterm/ANI_output.txt

```

```{bash}

scp -r  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/fastANI/ /home/vincent/Desktop/Projects/2019_RMK202_analysis/12_oldStrains

```

#####roary
run roary on old strains
```{bash}

##----------------------PROKKA annotation
rm -r /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Roary/Sterm/genomes

mkdir -p /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Roary/Sterm/genomes
for genomes in $(sed 's/.*\///' /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/filenames_Sterm.txt |cut -d '.' -f 1|cut -d '_' -f 1)
do
pathGenome=$(grep "$genomes" /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/filenames_Sterm.txt)
rm -r  /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Prokka/Sterm/${genomes}/
mkdir -p /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Prokka/Sterm/${genomes}/
/usr/bin/perl /usr/bin/prokka --outdir /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Prokka/Sterm/${genomes}/--force --usegenus --genus streptococcus --locustag ${genomes} --proteins proteins.faa --evalue 0.001 --addgenes ${pathGenome}

tmp1=$(ls /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Prokka/Sterm/${genomes}/)
cat /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Prokka/Sterm/${genomes}/${tmp1}/*.gff > /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Roary/Sterm/genomes/${genomes}.gff 

done


cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Streptococcus_thermophilus_RMK202/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/strains/Streptococcus_thermophilus_RMK202.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/strains/S72.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/strains/S50.gff

rm -r /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Roary/Sterm/roary
mkdir -p /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Roary/Sterm/roary
/usr/bin/perl /usr/local/bin/roary -f /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Roary/Sterm/roary -p 35 -e -n -v /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Roary/Sterm/genomes/*.gff


###---------------------
## roary plots
###---------------------
cd /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Roary/Sterm/roary_1585055952/

 sudo fasttreeMP -nt -gtr core_gene_alignment.aln > CdiffTree.newick
 
/home/vincent/apps/Roary/contrib/roary_plots/roary_plots.py  CdiffTree.newick gene_presence_absence.csv
```


```{bash}
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/12_oldStrains/roary/
scp -r  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Roary/Sterm/roary_1585055952/*.png /home/vincent/Desktop/Projects/2019_RMK202_analysis/12_oldStrains/roary/


```



```{r}

###===============================================================
##Streptococcus thermophilus
###===============================================================

library(readr)
library(plyr)
library(tidyverse)
final_ANI <- read_delim("/home/vincent/Desktop/Projects/2019_RMK202_analysis/12_oldStrains/fastANI/Sterm/ANI_output.txt",  "\t", escape_double = FALSE, col_names = c("GenomeA","GenomeB","ANI","mappedFragemnts","totFragemnts"),  trim_ws = TRUE)
final_ANI$coverage_ANI <- 100*(final_ANI$mappedFragemnts/final_ANI$totFragemnts)



final_ANI$GenomeA <- str_remove(final_ANI$GenomeA,"/archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Dialact/Sterm/") %>% str_remove(.,"/archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/") %>% str_remove(.,"/archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/") %>% str_remove(.,"/archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/") %>% str_remove(.,".fna") %>% str_remove(.,".fasta")

final_ANI$GenomeB <- str_remove(final_ANI$GenomeB,"/archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Dialact/Sterm/") %>% str_remove(.,"/archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/") %>% str_remove(.,"/archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/") %>% str_remove(.,"/archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/") %>% str_remove(.,".fna") %>% str_remove(.,".fasta")


final_ANI$GenomeB <- revalue(final_ANI$GenomeB, c("S50"="mst1","S72"="mst2"))
final_ANI$GenomeA <- revalue(final_ANI$GenomeA, c("S50"="mst1","S72"="mst2"))

final_ANI$ANI <- round(final_ANI$ANI,digits = 2)

write.table(final_ANI,"/home/vincent/Desktop/Projects/2019_RMK202_analysis/12_oldStrains/fastANI/Sterm/ANI_output_cleaned.txt",sep = "\t",quote = FALSE,row.names = FALSE,col.names = TRUE)
##------------
##for SNP legend
##------------

ggheatmap <- ggplot(final_ANI, aes(GenomeA, GenomeB, fill = ANI))+
 geom_tile(color = "white",size=1.1)+
 scale_fill_gradient2(low = "blue", high = "red", mid = "white",
  midpoint = 99.5, limit = c(98.9,100), space = "Lab",
  name="ANI") +
  # scale_fill_distiller(name = "SNPs", palette = "Blues", direction = -1)+
  # scale_fill_distiller(name = "ANI", palette = "Reds", direction = 1)+
  theme_minimal()+ # minimal theme
 theme(axis.text.y = element_text(size = 12),
      axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1))+
 coord_fixed()
# Print the heatmap
print(ggheatmap)


##------------
##make matrix
##------------

names<- unique(final_ANI$GenomeA)

heatmap_prep_matrix <- matrix(NA,nrow=length(names),ncol = length(names))
rownames(heatmap_prep_matrix) <- names
colnames(heatmap_prep_matrix) <- names


for (rowsss in names){
  for (colsss in names){
  
heatmap_prep_matrix[rowsss,colsss]  <-  final_ANI[which(final_ANI$GenomeA==rowsss & final_ANI$GenomeB==colsss),"ANI"] %>% as.numeric
  
  
  }  
}


##------------
##make dendro of all snps sites
##------------
library(ggdendro)

otter.dendro <- as.dendrogram(hclust(d = dist(x = heatmap_prep_matrix)))

dev.off()
# Create dendro
dendro.plot <- ggdendrogram(data = otter.dendro, rotate = FALSE)
# Preview the plot
print(dendro.plot)

##===============================================
##heatmap2
##===============================================

if (!require("gplots")) {
   install.packages("gplots", dependencies = TRUE)
   library(gplots)
   }
if (!require("RColorBrewer")) {
   install.packages("RColorBrewer", dependencies = TRUE)
   library(RColorBrewer)
   }


mat_data <- heatmap_prep_matrix # transform column 2-5 into a matrix
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 299)
col_breaks = c(seq(99.3,100,length=300))

dev.off()
    # png("~/Desktop/Projects/2019_RMK202_analysis/10_assembly/heatmap_all_Sterm_strains.png", width = 4000, height = 4000,res=300)
 svg("~/Desktop/Projects/2019_RMK202_analysis/10_assembly/heatmap_all_Sterm_strains.svg",width=14,height=14)
heatmap.2(mat_data,
  cellnote = mat_data,  # same data set for cell labels
  main = "Streptococcus thermophilus ANI", # heat map title
  notecol="black",      # change font color of cell labels to black
  density.info="none",
  trace="none",         # turns off trace lines inside the heat map
 margins = c(16, 16),     # widens margins around plot
  col=my_palette,   
  breaks=col_breaks,    # enable color transition at specified limits
  sepcolor="black",
  sepwidth=c(0.5,0.5),
  dendrogram="both")
dev.off()


###===============================================================
##Lactobacillus delbrueckii
###===============================================================


library(readr)
library(plyr)
library(tidyverse)
final_ANI <- read_delim("/home/vincent/Desktop/Projects/2019_RMK202_analysis/12_oldStrains/fastANI/Ldel/ANI_output.txt",  "\t", escape_double = FALSE, col_names = c("GenomeA","GenomeB","ANI","mappedFragemnts","totFragemnts"),  trim_ws = TRUE)
final_ANI$coverage_ANI <- 100*(final_ANI$mappedFragemnts/final_ANI$totFragemnts)



final_ANI$GenomeA <- str_remove(final_ANI$GenomeA,"/archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Dialact/Ldel/") %>% str_remove(.,"/archiv/Projects/2018_Culturomics/genomes/") %>% str_remove(.,"/archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/") %>% str_remove(.,".fna") %>% str_remove(.,".fasta")

final_ANI$GenomeB <- str_remove(final_ANI$GenomeB,"/archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Dialact/Ldel/") %>% str_remove(.,"/archiv/Projects/2018_Culturomics/genomes/") %>% str_remove(.,"/archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/log/") %>% str_remove(.,".fna") %>% str_remove(.,".fasta")

final_ANI$ANI <- round(final_ANI$ANI,digits = 2)
# final_ANI$GenomeB <- revalue(final_ANI$GenomeB, c("S50"="mst1","S72"="mst2"))
# final_ANI$GenomeA <- revalue(final_ANI$GenomeA, c("S50"="mst1","S72"="mst2"))
##------------
##for SNP legend
##------------
dev.off()

ggheatmap <- ggplot(final_ANI, aes(GenomeA, GenomeB, fill = ANI))+
 geom_tile(color = "white",size=1.1)+
 scale_fill_gradient2(low = "blue", high = "red", mid = "white",
  midpoint = 99.5, limit = c(98.9,100), space = "Lab",
  name="ANI") +
  # scale_fill_distiller(name = "SNPs", palette = "Blues", direction = -1)+
  # scale_fill_distiller(name = "ANI", palette = "Reds", direction = 1)+
  theme_minimal()+ # minimal theme
 theme(axis.text.y = element_text(size = 12),
      axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1))+
 coord_fixed()
# Print the heatmap
print(ggheatmap)


##------------
##make matrix
##------------

names<- unique(final_ANI$GenomeA)

heatmap_prep_matrix <- matrix(NA,nrow=length(names),ncol = length(names))
rownames(heatmap_prep_matrix) <- names
colnames(heatmap_prep_matrix) <- names


for (rowsss in names){
  for (colsss in names){
  
heatmap_prep_matrix[rowsss,colsss]  <-  final_ANI[which(final_ANI$GenomeA==rowsss & final_ANI$GenomeB==colsss),"ANI"] %>% as.numeric
  
  
  }  
}


##------------
##make dendro of all snps sites
##------------
library(ggdendro)

otter.dendro <- as.dendrogram(hclust(d = dist(x = heatmap_prep_matrix)))


# Create dendro
dendro.plot <- ggdendrogram(data = otter.dendro, rotate = FALSE)
# Preview the plot
print(dendro.plot)

##===============================================
##heatmap2
##===============================================

if (!require("gplots")) {
   install.packages("gplots", dependencies = TRUE)
   library(gplots)
   }
if (!require("RColorBrewer")) {
   install.packages("RColorBrewer", dependencies = TRUE)
   library(RColorBrewer)
   }



mat_data <- heatmap_prep_matrix # transform column 2-5 into a matrix
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 299)
col_breaks = c(seq(99.3,100,length=300))

dev.off()
    # png("~/Desktop/Projects/2019_RMK202_analysis/10_assembly/heatmap_all_Ldel_strains.png", width = 4000, height = 4000,res=300)
 svg("~/Desktop/Projects/2019_RMK202_analysis/10_assembly/heatmap_all_Ldel_strains.svg",width=14,height=14)
heatmap.2(mat_data,
  cellnote = mat_data,  # same data set for cell labels
  main = "Lactobacillus delbreuckii ANI", # heat map title
  notecol="black",      # change font color of cell labels to black
  density.info="none",
  trace="none",         # turns off trace lines inside the heat map
 margins = c(16, 16),     # widens margins around plot
  col=my_palette,   
  breaks=col_breaks,    # enable color transition at specified limits
  sepcolor="black",
  sepwidth=c(0.5,0.5),
 # RowSideColors = c(    # grouping row-variables into different
 #     rep("gray", 4),   # categories, Measurement 1-3: green
 #     rep("blue", 6),    # Measurement 4-6: blue
 #     rep("black", 7)),
  dendrogram="both")
dev.off()




```

## Mapping Data

#####mapping to only Bacteria

```{bash}

###================
##extract only bacteria
###================


samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished.fasta CP046134 > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished_onlyBAC.fasta

samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished.fasta CP046131 >> /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished_onlyBAC.fasta

###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt  ##the file with all sample names
logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_onlyBAC
Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished_onlyBAC.fasta
threads=37

##!!!!!!!!!!!!!change names of Forward and Reverse reads

###================
##script
###================
##============
##mapping to reference
##============
bwa index $Assembly
# name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 

mkdir -p ${logFilelocation}

num=1

for names in $(cat ${samplesss})
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))
  rm -r ${BaseLocation}/${names}

  mkdir -p ${BaseLocation}/${names}/bwaMapping2DB/
  
 # bwa mem -t ${threads} /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids/di_rmk202_MAG_reference_polished.fasta
  bwa mem -t ${threads} ${Assembly} \
    /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_1.fastq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_2.fastq | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 

samtools view -b -f 4 ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

bedtools bamtofastq -i ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam -fq ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq

rm ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam &

##-----Qualimap

mkdir -p ${BaseLocation}/${names}/bwaMapping2DB//bamqc

qualimap bamqc -bam ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir ${BaseLocation}/${names}/bwaMapping2DB//bamqc --java-mem-size=80G


 mkdir -p ${BaseLocation}/log
echo ${BaseLocation}"/"${names}"/bwaMapping2DB/bamqc" >> ${BaseLocation}/log/multiqc_logfile_finalGenome.txt
   done 
   
   
#    rm -r ${BaseLocation}/MultiQC
 
#  mkdir -p ${BaseLocation}/MultiQC
  
#  multiqc --config /archiv/logs/multiQC_config.txt --file-list  ${BaseLocation}/log/multiqc_logfile_finalGenome.txt \
#    -o ${BaseLocation}/MultiQC


##-------------------------------------------------  
##move local
##-------------------------------------------------  

##bring together 

cat /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/log/mappings.txt \
/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_allRMKs202/log/mappings.txt > \
/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/log/mappings_final.txt


scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/log/mappings_final.txt /Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/
```

###### unmapped assembly

```{bash}
##-------------------------------------------------  
##-----description
##-------------------------------------------------  
names=Konserve_202
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt  ##the file with all sample names

##-------------------------------------------------  
##-----normal spades on unmaped
##-------------------------------------------------  
#rm -r /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1/
num=1
threads=32
  for names in $(cat ${samplesss})
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))
rm -r ${BaseLocation}/${names}/spadesAssembly_meta_unmapped/assembly
mkdir -p ${BaseLocation}/${names}/spadesAssembly_meta_unmapped/assembly

/home/vincent/miniconda2/pkgs/spades-3.13.0-0/bin/spades.py -t ${threads} -m 100 \
  -s ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq \
  -o ${BaseLocation}/${names}/spadesAssembly_meta_unmapped/assembly
  done
  
  num=1
threads=32
  for names in $(cat ä{samplesss})
  do
  echo"------------------------"
  echo  ${names}
  
  grep ">" ${BaseLocation}/${names}/spadesAssembly_meta_unmapped/assembly/contigs.fasta
  done
  
  
  ##------------------------
##filter short and low coverage contigs
##------------------------
for names in $(cat {samplesss})
 cd ${BaseLocation}/${names}/spadesAssembly_meta_unmapped/assembly/
  for file in contigs.fasta;do
grep -F ">" $file | sed -e 's/_/ /g' |sort -nrk 6 |awk '$6>=2.0 && $4>=500 {print $0}'| sed -e 's/ /_/g'|sed -e 's/>//g'>$file.txt
echo sequences to keep
wc -l $file.txt
echo running fastagrep.pl
/home/vincent/apps/fastagrep.pl -f $file.txt $file > HCov.$file
echo sequences kept
grep -c ">" HCov.$file

done
  
  
  done
  
 
  
  ##--------------------------------------blastn
#rm -r /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_blast_nonBacterialReadsAssembled/metaSpades/
  for names in $(cat ${samplesss})
  do
  echo"------------------------"
  echo  ${names}
  
mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_blast_nonBacterialReadsAssembled/metaSpades/blast/
rm  /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_blast_nonBacterialReadsAssembled/metaSpades/blast/${names}_nonBacterialReadsAssembled_blast.txt
  blastn -num_threads ${threads} -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query ${BaseLocation}/${names}/spadesAssembly_meta_unmapped/assembly/contigs.fasta \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen" \
  -db /archiv/BLAST/blast_db/20190204/viral_nucleotide/viral \
  -out /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_blast_nonBacterialReadsAssembled/metaSpades/blast/${names}_nonBacterialReadsAssembled_blast.txt \
  -evalue 0.01 -word_size 64
  
  done 
  


##------------------------------extract assembly graph

mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_blast_nonBacterialReadsAssembled/metaSpades/AssemblyGraphs/
  for names in $(cat ${samplesss})
do


echo "##====================="
echo ${names}

cp ${BaseLocation}/${names}/spadesAssembly_meta_unmapped/assembly//assembly_graph_with_scaffolds.gfa /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_blast_nonBacterialReadsAssembled/metaSpades/AssemblyGraphs/${names}.gfa

done


##-------------------------------------------------  
##-----plasmid spades
##-------------------------------------------------  

#rm -r /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1/
num=1
threads=32
  for names in $(cat ${samplesss})
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))
rm -r ${BaseLocation}/${names}/PlasmidspadesAssembly_unmapped/assembly
mkdir -p ${BaseLocation}/${names}//PlasmidspadesAssembly_unmapped/assembly

/home/vincent/miniconda2/pkgs/spades-3.13.0-0/bin/spades.py --plasmid -t ${threads} -m 100 \
  -s ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq \
  -o ${BaseLocation}/${names}/PlasmidspadesAssembly_unmapped/assembly
  done
  
  names=lyo202_96
  
  ##------------------------
##filter short and low coverage contigs
##------------------------
for names in $(cat ${samplesss})
 cd ${BaseLocation}/${names}/PlasmidspadesAssembly_unmapped/assembly/
  for file in contigs.fasta;do
grep -F ">" $file | sed -e 's/_/ /g' |sort -nrk 6 |awk '$6>=2.0 && $4>=500 {print $0}'| sed -e 's/ /_/g'|sed -e 's/>//g'>$file.txt
echo sequences to keep
wc -l $file.txt
echo running fastagrep.pl
/home/vincent/apps/fastagrep.pl -f $file.txt $file > HCov.$file
echo sequences kept
grep -c ">" HCov.$file

done
  
  
  done
  
 
  
  ##--------------------------------------blastn

  for names in $(cat ${samplesss})
  do
  echo"------------------------"
  echo  ${names}
  
mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_blast_nonBacterialReadsAssembled/PlasmidSpades/blast/
rm  /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_blast_nonBacterialReadsAssembled/PlasmidSpades/blast/${names}_nonBacterialReadsAssembled_blast.txt
  blastn -num_threads ${threads} -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query ${BaseLocation}/${names}/PlasmidspadesAssembly_unmapped/assembly/contigs.fasta \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen" \
  -db /archiv/BLAST/blast_db/20190204/viral_nucleotide/viral \
  -out /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_blast_nonBacterialReadsAssembled/PlasmidSpades/blast/${names}_nonBacterialReadsAssembled_blast.txt \
  -evalue 0.01 -word_size 64
  
  done 
  


##------------------------------extract assembly graph

mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_blast_nonBacterialReadsAssembled/PlasmidSpades/AssemblyGraphs/
  for names in $(cat ${samplesss})
do


echo "##====================="
echo ${names}

cp ${BaseLocation}/${names}/PlasmidspadesAssembly_unmapped/assembly//assembly_graph_with_scaffolds.gfa /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_blast_nonBacterialReadsAssembled/PlasmidSpades/AssemblyGraphs/${names}.gfa

done

  
  
  
  
  
  
##----remove short contigs (<500bp)


#awk '/^>/ {printf("\n%s\n",$0);next; } { printf("%s",$0);}  END {printf("\n");}' < /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/spadesAssembly_unmapped/assembly/scaffolds.fasta |sed '/^[[:space:]]*$/d' | awk '/^>/ { getline seq } length(seq) >500 { print $0 "\n" seq }' > /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/spadesAssembly_unmapped/assembly/scaffolds_min500bp.fasta



scp -r vincent@130.223.51.116:/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_blast_nonBacterialReadsAssembled/ /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/


##------------------------------move local

#scp -r vincent@130.223.51.151:/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/Versand_202/plasmidSpades_unassembled/assembly_graph_with_scaffolds.gfa /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/Versand_202.gfa


#scp -r vincent@130.223.51.151:/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/Konserve_202/plasmidSpades_unassembled/assembly_graph_with_scaffolds.gfa /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/Konserve_202.gfa


###----------------assembly graph

#spades_contig_graph.py -l  /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/di_K1_0h/spadesAssembly_unmapped/assembly/assembly_graph.fastg  /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/di_K1_0h/spadesAssembly_unmapped/assembly/contigs.fasta  /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/di_K1_0h/spadesAssembly_unmapped/assembly/contigs.paths -o  /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/di_K1_0h/spadesAssembly_unmapped/assembly/contigs.fastg


```

####check phage contig 
Here, I check the phage contigs (non-bacterial) for redundancy. 

```{bash}


###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt  ##the file with all sample names
logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP
Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished.fasta

##---------------------------------------
##extract only Phage contigs
##---------------------------------------

rm /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished_onlyPhage.fasta
for contig in $(grep ">" ${Assembly}|sed 's/>//g'|grep "CP" -v)
do

samtools faidx ${Assembly} ${contig} >> /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished_onlyPhage.fasta

done

##---------------------------------------
##shorten contig name
##---------------------------------------

sed -i 's/_phage//g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished_onlyPhage.fasta
sed -i 's/_contig/_c/g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished_onlyPhage.fasta

####---------------------------------------
##cd -hit to see redundancy
##---------------------------------------
mkdir -p ${BaseLocation}/PhageContigsRedundancy/

cd-hit-est -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished_onlyPhage.fasta \
    -o ${BaseLocation}/PhageContigsRedundancy/both__cdsClustering.txt -c 0.80 -M 30000 -T 37 -sc 1 -sf 1



```
Thereby I have demultiplexed the phage genomes some contigs are redundant and potentially mask SNVs over time.
I now change the metagenome REFERNECE to contain only the nessary phage contigs.

The old mapping, containg all contigs is locate here:
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP


I have done This again by checking which contigs are present in the samples. 
Now I want to see if they are divergent

```{bash}

####---------------------------------------
##script
##---------------------------------------

mkdir -p /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_onlyBAC/PhageAssemblies/{log,Phages}

rm /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_onlyBAC/PhageAssemblies/log/Sterm_phage_contigs_description.txt
rm /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_onlyBAC/PhageAssemblies/Phages/all_Sterm_phageContigs.fasta

for sPhage in $(grep "_component_0" /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_onlyBAC/lyo202_96/PlasmidspadesAssembly_unmapped/assembly/scaffolds.fasta |sed 's/>//g')
do

length=$(echo ${sPhage} | cut -d '_' -f 4)
Coverage=$(echo ${sPhage} | cut -d '_' -f 6)

componentName=$(grep "${sPhage}" /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_onlyBAC/lyo202_96/PlasmidspadesAssembly_unmapped/assembly/assembly_graph_with_scaffolds.gfa)

echo -e ${sPhage}"\t"${length}"\t"${Coverage}"\t"${componentName} >> /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_onlyBAC/PhageAssemblies/log/Sterm_phage_contigs_description.txt


samtools faidx /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_onlyBAC/lyo202_96/PlasmidspadesAssembly_unmapped/assembly/contigs.fasta ${sPhage} >> /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_onlyBAC/PhageAssemblies/Phages/scaffolds.fasta

done


####---------------------------------------
##cd -hit to see redundancy
##---------------------------------------
mkdir -p /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_onlyBAC/PhageAssemblies/cd_hit/
cd-hit-est -i /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_onlyBAC/PhageAssemblies/Phages/all_Sterm_phageContigs.fasta \
    -o /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_onlyBAC/PhageAssemblies/cd_hit/all_Sterm_phages.txt -c 0.80 -M 30000 -T 37 -sc 1 -sf 1

####---------------------------------------
##bring local
##---------------------------------------

mkdir -p Desktop/Projects/2019_Pilotplan/12_mapping2phages/

scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_onlyBAC/PhageAssemblies/log/Sterm_phage_contigs_description.txt Desktop/Projects/2019_Pilotplan/12_mapping2phages/

```

```{r}
library(readr)
library(ggplot2)
Sterm_phage_contigs_description <- read_delim("~/Desktop/Projects/2019_Pilotplan/12_mapping2phages/Sterm_phage_contigs_description.txt",  "\t", escape_double = FALSE, col_names = c("name","length","coverage"),   trim_ws = TRUE)


ggplot(Sterm_phage_contigs_description,aes(x=coverage,y=length))+geom_point()+theme_classic()+geom_line()
Sterm_phage_contigs_description$name


Sterm_phage_contigs_description$group <- Sterm_phage_contigs_description$name


Sterm_phage_contigs_description$group <- Sterm_phage_contigs_description$name
Sterm_phage_contigs_description$group  <- revalue(Sterm_phage_contigs_description$group , c(
  "NODE_2_length_24091_cov_87.467019_component_0"="1B","NODE_4_length_19116_cov_2026.732129_component_0"="1A", "NODE_6_length_4770_cov_1797.099084_component_0" ="6",
  "NODE_7_length_3001_cov_1416.239056_component_0"=
))



```



#####mapping to single polished genome

```{bash}

###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt  ##the file with all sample names
logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_woPhageRedundancy
Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished_cleaned.fasta

##!!!!!!!!!!!!!change names of Forward and Reverse reads

###================
##merge genome
###================

##---------------------------------------
##extract Bacteria contigs
##---------------------------------------

rm /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished_onlyBACTERIA_PLASMID.fasta
for contig in $(grep ">" ${Assembly}|sed 's/>//g'|grep "CP" )
do

samtools faidx ${Assembly} ${contig} >> /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished_onlyBACTERIA_PLASMID.fasta

done

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished_onlyBACTERIA_PLASMID.fasta \
${BaseLocation}/PhageContigsRedundancy/both__cdsClustering.txt > \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished_cleaned.fasta


###================
##script
###================
##============
##mapping to reference
##============
bwa index $Assembly
# name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 

mkdir -p ${logFilelocation}

num=1

for names in $(cat ${samplesss})
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))
  rm -r ${BaseLocation}/${names}

  mkdir -p ${BaseLocation}/${names}/bwaMapping2DB/
  
 # bwa mem -t ${threads} /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids/di_rmk202_MAG_reference_polished.fasta
  bwa mem -t ${threads} ${Assembly} \
    /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_1.fastq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_2.fastq | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 

samtools view -b -f 4 ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

bedtools bamtofastq -i ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam -fq ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq

rm ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam &

##-----Qualimap

mkdir -p ${BaseLocation}/${names}/bwaMapping2DB//bamqc

qualimap bamqc -bam ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir ${BaseLocation}/${names}/bwaMapping2DB//bamqc --java-mem-size=80G


 mkdir -p ${BaseLocation}/log
echo ${BaseLocation}"/"${names}"/bwaMapping2DB/bamqc" >> ${BaseLocation}/log/multiqc_logfile_finalGenome.txt
   done 
   
   
#    rm -r ${BaseLocation}/MultiQC
 
#  mkdir -p ${BaseLocation}/MultiQC
  
#  multiqc --config /archiv/logs/multiQC_config.txt --file-list  ${BaseLocation}/log/multiqc_logfile_finalGenome.txt \
#    -o ${BaseLocation}/MultiQC


##-------------------------------------------------  
##-----check read coverage
##-------------------------------------------------  
name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 
threads=37
names=Konserve_202


num=1
rm -r ${BaseLocation}/log/mappings.txt

#for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt)
for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt)
  do
mapped=$(grep "     number of mapped reads = " ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt  |cut -d '(' -f 2 | sed 's/%)//g')

mkdir -p ${BaseLocation}/log
echo -e ${names}"\tonlyMeta\t"${mapped} >> ${BaseLocation}/log/mappings.txt

done

##-------------------------------------------------  
##move local
##-------------------------------------------------  

##bring together 

cat /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/log/mappings.txt \
/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_allRMKs202/log/mappings.txt > \
/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/log/mappings_final.txt


scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/log/mappings_final.txt /Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/
```

#####check mapping length

```{bash}


###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt  ##the file with all sample names
logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP
Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished.fasta
names=Konserve_202

----------------------
##script
###-----------------------

#seq_length.py ${Assembly} |grep "CP" -v > /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/phageMapping/ContigLength.txt 


rm /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/phageMapping//MappingQualityLength.txt
for names in $(cat ${samplesss})
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))

mkdir -p /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/phageMapping



mkdir -p /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/phageMapping/mappingOnlyPhages/${names}/
samtools view -b -L /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/phageMapping/ContigLength.txt  ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/phageMapping/mappingOnlyPhages/${names}/mapping.bam


# samtools view -L /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/phageMapping/ContigLength.txt  ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/phageMapping/mappingOnlyPhages/${names}/mapping.sam


mkdir -p /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/phageMapping

samtools view /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/phageMapping/mappingOnlyPhages/${names}/mapping.bam |cut -f 13 |sed 's/MD:Z://g'  | awk -F "[CTAG]" '{print $1+$2+$3+$4+$5+$6+$7+$8+$9+$10+$11+$12+$13}' |sort -rn |uniq -c | sed -e 's/^[[:space:]]*//' |awk -F " " -v var="$names" '{OFS=" "} {print $0,var}' >> /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/phageMapping//MappingQualityLength.txt

done

##-------------------------local

mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/05_phageMapping/

scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/phageMapping//MappingQualityLength.txt ~/Desktop/Projects/2019_RMK202_analysis/05_phageMapping/MappingQualityLength.txt

```


```{r}

###-------------------------look at mapping length
library(readr)
library(patchwork)
  
MappingQualityLength <- read_table2("~/Desktop/Projects/2019_RMK202_analysis/05_phageMapping/MappingQualityLength.txt",  col_names = c("count","length","sample"))
  
range(MappingQualityLength$length)
mappinglengthPlot <- ggplot(MappingQualityLength,aes(x=length,y=count,color=sample,fill=sample))+geom_bar(stat="identity")+theme_classic()
  mappinglengthPlot
MappingQualityLength_long <- MappingQualityLength[which(MappingQualityLength$length>42),]
ggplot(MappingQualityLength_long,aes(x=length,y=count,color=sample,fill=sample))+geom_bar(stat="identity")+theme_classic()

table(MappingQualityLength_long$sample)

MappingQualityLength$keeping <- ifelse(MappingQualityLength$length>42,"keep","remove")
MappingQualityLength_keeping <- aggregate(. ~sample+keeping, data=MappingQualityLength[,c("sample","count","keeping")], sum, na.rm=TRUE)

MappingQualityLength_keeping$keeping = factor(MappingQualityLength_keeping$keeping, levels=c("remove","keep"))
# colorsss <- (c("darkcyan","darkturquoise","lightblue","lightgray"))


keepingPlot <- ggplot(MappingQualityLength_keeping,aes(x=sample,y=count,color=keeping,fill=keeping))+geom_bar(stat="identity")+theme_classic()+labs(title = "42bp mapping thresehold",x="")+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9))
keepingPlot


 
  svg("~/Desktop/Projects/2019_RMK202_analysis/plot/mappinglengthplot_phages.svg",width=6,height=3)

(mappinglengthPlot|keepingPlot)
# (p1|p2|p3)+plot_layout(widths =c(2,4,2))

 dev.off()

nrow(CRISPR_spacer_coverage_extended_wide)

```


#####filter mapping

```{bash}

###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt  ##the file with all sample names
logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP
Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished.fasta
names=Konserve_202

###-----------------------
##script
###-----------------------



seq_length.py ${Assembly} |grep "CP" -v > /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/phageMapping/ContigLength.txt 

mkdir -p /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/phageMapping



mkdir -p /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/phageMapping/mappingOnlyPhages/${names}/
#samtools view -b -L /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/phageMapping/ContigLength.txt  ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/phageMapping/mappingOnlyPhages/${names}/mapping.bam


 samtools view -L /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/phageMapping/ContigLength.txt  ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/phageMapping/mappingOnlyPhages/${names}/mapping.sam


cd /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/phageMapping/mappingOnlyPhages/${names}/


/home/vincent/scripts/mappingLengthfiltering_phageReads.R -i mapping.sam -o mapping_filtered.sam
#rm ${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub.sam

samtools flagstat mapping_filtered.sam

done


```
###### unmapped assembly

```{bash}
##-------------------------------------------------  
##-----normal spades on unmaped
##-------------------------------------------------  

#rm -r /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1/
num=1
threads=32
  for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt)
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))
rm -r /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/spadesAssembly_meta_unmapped/assembly
mkdir -p /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/spadesAssembly_meta_unmapped/assembly

/home/vincent/miniconda2/pkgs/spades-3.13.0-0/bin/spades.py -t ${threads} -m 100 \
  -s /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq \
  -o /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/spadesAssembly_meta_unmapped/assembly
  done
  
  num=1
threads=32
  for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt)
  do
  echo"------------------------"
  echo  ${names}
  

  grep ">" /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/spadesAssembly_meta_unmapped/assembly/contigs.fasta
  done
  for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt)
  do
  echo"------------------------"
  echo  ${names}
  

  cd /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/spadesAssembly_meta_unmapped/assembly/
  for file in contigs.fasta;do
grep -F ">" $file | sed -e 's/_/ /g' |sort -nrk 6 |awk '$6>=2.0 && $4>=500 {print $0}'| sed -e 's/ /_/g'|sed -e 's/>//g'>$file.txt
echo sequences to keep
wc -l $file.txt
echo running fastagrep.pl
../fastagrep.pl -f $file.txt $file > HCov.$file
echo sequences kept
grep -c ">" HCov.$file

done
  
  
  done
  
  
    for file in contigs.fasta;do
grep -F ">" $file


done


##-------------------------------------------------  
##-----plasmid spades
##-------------------------------------------------  

#rm -r /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1/
num=1
threads=32
  for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt)
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))
rm -r /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/spadesAssembly_unmapped/assembly
mkdir -p /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/spadesAssembly_unmapped/assembly

/home/vincent/miniconda2/pkgs/spades-3.13.0-0/bin/spades.py --plasmid -t ${threads} -m 100 \
  -s /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq \
  -o /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/spadesAssembly_unmapped/assembly
  done
  
  
  
##----remove short contigs (<500bp)


awk '/^>/ {printf("\n%s\n",$0);next; } { printf("%s",$0);}  END {printf("\n");}' < /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/spadesAssembly_unmapped/assembly/scaffolds.fasta |sed '/^[[:space:]]*$/d' | awk '/^>/ { getline seq } length(seq) >500 { print $0 "\n" seq }' > /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/spadesAssembly_unmapped/assembly/scaffolds_min500bp.fasta


##----blastn


mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/spadesAssembly_unmapped/blastn
  
  blastn -num_threads ${threads} -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/spadesAssembly_unmapped/assembly/scaffolds_min500bp.fasta \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen" \
  -db /archiv/BLAST/blast_db/20190204/phage_nucleotide/phage_nucleotide_cow \
  -out /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/spadesAssembly_unmapped/blastn/blastAssemblyaginstScaffolds.txt \
  -evalue 0.01 -word_size 64
  
  done 
  

#scp vincent@130.223.51.151:/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/di_K1_0h/spadesAssembly_unmapped/assembly/assembly_graph_with_scaffolds.gfa 


##------------------------------extract assembly graph

mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/AssemblyGraphs/
  for names in $(cat /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name_allOfExperiment.txt)
do


echo "##====================="
echo ${names}

cp /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/spadesAssembly_unmapped/assembly/assembly_graph_with_scaffolds.gfa /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/AssemblyGraphs/${names}.gfa

done


scp -r vincent@130.223.51.151:/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/AssemblyGraphs/ /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/


##------------------------------move local

scp -r vincent@130.223.51.151:/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/Versand_202/plasmidSpades_unassembled/assembly_graph_with_scaffolds.gfa /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/Versand_202.gfa


scp -r vincent@130.223.51.151:/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/Konserve_202/plasmidSpades_unassembled/assembly_graph_with_scaffolds.gfa /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/Konserve_202.gfa


###----------------assembly graph

spades_contig_graph.py -l  /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/di_K1_0h/spadesAssembly_unmapped/assembly/assembly_graph.fastg  /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/di_K1_0h/spadesAssembly_unmapped/assembly/contigs.fasta  /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/di_K1_0h/spadesAssembly_unmapped/assembly/contigs.paths -o  /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/di_K1_0h/spadesAssembly_unmapped/assembly/contigs.fastg


```




```{r}
library(ggplot2)
library(readr)

mappings <- read_delim("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/mappings_final.txt",  "\t", escape_double = FALSE, col_names = c("sample","type","mapping"),  trim_ws = TRUE)

median(mappings$mapping)

##------------------------treamtents
mappings$sample <- revalue(mappings$sample, c("L71"="mst4","lyo202_96"="Lyo\n1996","Lyo_202_2012"="Lyo\n2012", "Lyo_202_2014"="Lyo\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018","di_K2_6h"="cheesemaking\nday1","th_K2_8h"="cheesemaking\nday2"))

mappings$sample <- factor(mappings$sample, levels=(c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2")))
mappings$type <- factor(mappings$type, levels=(c("onlyMeta","allisolates")))


##------------------------plotting

 # coloursplotting <-  c(rep("cadetblue3",5),rep("azure4",16),rep("#EB4D4D",2),rep("#10B552",9))
coloursplotting <-  c("cadetblue3","azure4")

 rawREad <-  ggplot(mappings,aes(x=sample,y=mapping,fill=type))+geom_bar( stat="identity", position="dodge")+
    labs(y="Percent\nraw reads mapping",
        x="")+
         coord_cartesian(ylim=c(95,100))+
   scale_fill_manual(values=coloursplotting)+
       theme_classic()+
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
          # # axis.text.y.left = element_text(color = "blue"),
          # axis.text.y.right = element_text(color="red"),
          # axis.title.y.left = element_text(color="blue"),
          # axis.title.y.right = element_text(color="red"),
          # rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="right",
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank()
          )
 rawREad
  # svg("~/Desktop/presenations/my_presentations/20190807_committeeMeeting/figures/04_rawReadMapping.svg",width=5,height=3)
  svg("~/Desktop/Manuscripts/2019_RMK202/Figures/S05_rawReadMapping.svg",width=6,height=4)
    # png("~/Desktop/Manuscripts/2019_RMK202/Figures/20191114_draft_02/PNGs_for_manuscript/S05_rawReadMapping.png", width = 1000, height = 800,res=300)


rawREad
 dev.off()
 


p1 <- ggplot(mappings,aes(x=type,y=mapping))+geom_boxplot()+theme_classic()+scale_y_continuous(breaks =c(97.5,98,98.5,99,99.5,100),labels=c(97.5,98,98.5,99,99.5,100),limits = c(97.5,100))+labs(x="",y="mapping [%]") #+lims(y=c(90,100))
p1

svg("~/Desktop/Manuscripts/2019_RMK202/Figures/20191114_draft_02/S03_mapping.svg",width=2,height=4)
#    # png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 1000, height = 800,res=300)
p1
 dev.off()

```

###phage disentangle

```{bash}
rm - r /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/log/
mkidr -p /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/log/

 num=1
  for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt )
    do
    
    echo ${num}"/37  :" ${names} 
    num=$((num+1))

##extract from qualimap

grep -A 100 ">>>> Coverage per contig" /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/${names}/bwaMapping2DB/bamqc/genome_results.txt | sed '1,2d' | sed '/^$/d' | sed -e 's/^[ \t]*//'  |grep -v "CP" |awk -F "\t" -v namess="$names"  'BEGIN{OFS="\t"}{print $0, namess}' >> /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/log/mappingcoverage.txt

done


scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/log/mappingcoverage.txt ~/Desktop/Projects/2019_Pilotplan/PhagemappingCoverage.txt

```


```{r}

library(readr)
library(plyr)
library(tidyverse)


PhagemappingCoverage <- read_delim("Desktop/Projects/2019_Pilotplan/PhagemappingCoverage.txt",  "\t", escape_double = FALSE, col_names = c("contig","size","reads","coverage","stdv","sample"),  trim_ws = TRUE)

tmp1 <- data.frame(str_split_fixed(PhagemappingCoverage$contig, fixed("_"), 7))[,1:4]
PhagemappingCoverage$genome <- paste(tmp1$X1,tmp1$X2,tmp1$X3,tmp1$X4,sep="_")

PhagemappingCoverage$coverage <- PhagemappingCoverage$coverage+0.1

ggplot(PhagemappingCoverage,aes(x=coverage,y=size,color=genome,fill=genome))+geom_point()+theme_classic()


PhagemappingCoverage$sample <- revalue(PhagemappingCoverage$sample, c("lyo202_96"="Lyo\n1996","Lyo_202_2012"="Lyo\n2012", "Lyo_202_2014"="Lyo\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018","di_K2_6h"="cheesemaking\nday1","th_K2_8h"="cheesemaking\nday2"))
PhagemappingCoverage$sample = factor(PhagemappingCoverage$sample, levels=c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2"))
levels(PhagemappingCoverage$sample)
# PhagemappingCoverage$sample <- as.factor(PhagemappingCoverage$sample)

Phage_mapping_coverage <- ggplot(PhagemappingCoverage,aes(x=sample,y=coverage,group=contig,color=genome,fill=genome))+ geom_line(size=0.5, alpha=.8)+ 
       facet_wrap(genome~.,scales = "free")+
   labs("",
         x="",
         # y="",
          y="all")+
       # scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  coord_trans( y="log2")+
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
Phage_mapping_coverage


```

I had to manually disentangle the two genomes manually. 
By looking at the coverages of the shared and divergent regions. 
In order to plot this I had to manually get the coverage of the different components out of bandage. 
The components are not the contigs. After disentengling the genomes I merge and extracted the path

```{r}
library(readr)
library(ggplot2)
library(plyr)
library(dplyr)
library(tidyverse)
library(patchwork)

phages_sterm_contigs_mapping <- read_delim("~/Desktop/Projects/2019_Pilotplan/11_phages/phages_sterm_contigs_mapping/phages_sterm_contigs_mapping.csv", "\t", escape_double = FALSE, trim_ws = TRUE)
  phages_sterm_contigs_mapping$contigFraction <- factor(phages_sterm_contigs_mapping$contigFraction, levels=c("A","B","C"))

  
phages_sterm_contigs_mapping$contigFraction <- revalue(phages_sterm_contigs_mapping$contigFraction, c("A"="High\ncoverage\ncontigs","B"="Low\ncoverage\ncontigs","C"="Shared\ncontigs"))



# phages_sterm_contigs_mapping$contigNumber

plotPoints <- ggplot(phages_sterm_contigs_mapping,aes(x=depth,y=length,group=contigNumber,color=contigFraction,fill=contigFraction))+geom_point(size=2)+geom_line()+labs(x="Contig coverage",y="Contig length")+theme_classic()+coord_trans( y="log2")+scale_y_continuous(breaks =c(0,100,250,500,1000,5000,10000,20000,25000),labels=c(0,100,250,500,1000,5000,10000,20000,25000),limits = c(100,25000))+theme(legend.position = "none")

plot_box <- ggplot(phages_sterm_contigs_mapping,aes(x=contigFraction,y=depth,color=contigFraction,fill=contigFraction))+geom_boxplot(outlier.size =2)+scale_y_continuous(limits=c(0,2100))+coord_flip()+theme_classic()+theme(legend.position = "none",axis.title = element_blank(),axis.text.x=element_blank())



  svg("~/Desktop/Projects/2019_Pilotplan/11_phages/phages_sterm_contigs_mapping/phage_contigsPlot.svg",width=6,height=7)
 # png("~/Desktop/Projects/2019_Pilotplan/11_phages/phages_sterm_contigs_mapping/phage_contigsPlot.png", width = 1400, height = 1600,res=300)
(plot_box/plotPoints)+plot_layout(heights =c(1,2))
 dev.off()

```




###polish disentangled genomes

1. take the phages out of the plasmid assemblies 
  - demultiplex two sterm phages (lyo202_96)
  - Lactobacillus phage (RMK202)
  - Sterm phage from cheesemaking (di_K2_6h)
2. rename the genomes
2. merge the demultiplexed phages 
3. cd-hit (check)
4. polish (on bigmachine)
5. map

```{bash}
###------------------------
#rename
###------------------------
cd /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/02_blast_nonBacterialReadsAssembled/PlasmidSpades/AssemblyGraphs

sed 's/>.*$/>Streptococcus_phage_1/g' lyo96_phage_sterm_high_path > Phages_rmk202.fasta
sed 's/>.*$/>Streptococcus_phage_2/g' Sterm_low_coverage_lyo20296.fasta >> Phages_rmk202.fasta
sed 's/>.*$/>Lactobacillus_phage_2/g' Sterm_3_lyo20296.fasta >> Phages_rmk202.fasta
sed  's/>.*$/>Lactobacillus_phage_1/g' Lactobacillus_phage_rmk202.fasta >> Phages_rmk202.fasta


##==================
##cd-hit
##==================
mkdir -p cd_hit_phages
cd-hit-est -i Phages_rmk202.fasta\
    -o cd_hit_phages/phages.txt -c 0.90 -M 30000 -T 37 -sc 1 -sf 1


###------------------------
#bring  bigmachine
###------------------------


scp /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/02_blast_nonBacterialReadsAssembled/PlasmidSpades/AssemblyGraphs/Phages_rmk202.fasta vincent@130.223.51.116:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/


###------------------------
#merge with bacterial genomes
###------------------------
cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished_onlyBACTERIA_PLASMID.fasta /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/Phages_rmk202.fasta > \
 /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/rmk202_bacteria_phage_mag_ref.fasta


###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt  ##the file with all sample names
logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_withPhageAssembly
Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/rmk202_bacteria_phage_mag_ref.fasta

##!!!!!!!!!!!!!change names of Forward and Reverse reads

###================
##script
###================
##============
##mapping to reference
##============
bwa index $Assembly
# name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 

mkdir -p ${logFilelocation}

num=1

for names in $(cat ${samplesss})
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))
  rm -r ${BaseLocation}/${names}

  mkdir -p ${BaseLocation}/${names}/bwaMapping2DB/
  
 # bwa mem -t ${threads} /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids/di_rmk202_MAG_reference_polished.fasta
  bwa mem -t ${threads} ${Assembly} \
    /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_1.fastq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_2.fastq | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 

samtools view -b -f 4 ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

bedtools bamtofastq -i ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam -fq ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq

rm ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam &

##-----Qualimap

mkdir -p ${BaseLocation}/${names}/bwaMapping2DB//bamqc

qualimap bamqc -bam ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir ${BaseLocation}/${names}/bwaMapping2DB//bamqc --java-mem-size=80G


 mkdir -p ${BaseLocation}/log
echo ${BaseLocation}"/"${names}"/bwaMapping2DB/bamqc" >> ${BaseLocation}/log/multiqc_logfile_finalGenome.txt
   done 
   
   ##----------------------------------
   ##bamqc
      ##----------------------------------
for names in $(cat ${samplesss})
  do
  
  cp -r ${BaseLocation}/${names}/bwaMapping2DB//bamqc ${BaseLocation}/log/bamQC_${names}
  
  done
  mkdir -p ~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/bamQCs
```


```{bash}
scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_withPhageAssembly/log  ~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/bamQCs
```


```{bash}
###-----------------
##Illuminamapping rawReads to circlator genome
###-----------------

   
##------------------define variables------------------
#for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
#do

name=di_K2_6h
sample=di_K2_6h
#sample=1
threads=37
Overall_output_directory=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_withPhageAssembly/polishing
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/rmk202_bacteria_phage_mag_ref.fasta
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/di_K2_6h_nanopore_filtered.fastq.gz


for name in $(echo "RMK202 lyo202_96 di_K2_6h")
do

Overall_output_directory=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_withPhageAssembly/polishing/$name
sample=$name
num=1



for run in first second third fourth
   do 

name=run
##--------------create directories----------------
rm -r $Overall_output_directory/0${num}_${run}Freebayes/
mkdir -p $Overall_output_directory/0${num}_${run}Freebayes/{rawIlluminaMapped,freebayesOuput}
mkdir -p $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/bamqc

#mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fq

# gunzip -c /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R1_val_1.fq.gz > \
#  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fq/${sample}_R1_val_1.fq
# gunzip -c /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R2_val_2.fq.gz > \
#  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fq/${sample}_R2_val_2.fq


bwa index $reference_fasta

bwa mem -t ${threads}  $reference_fasta /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_1.fastq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_2.fastq |samtools sort -@8 -O BAM \
-o $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam - 
     

qualimap bamqc -bam $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam \
        -outdir $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/bamqc --java-mem-size=40G
    
samtools index $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam

###-----------------
##freebayesAfterCirclator
###-----------------
echo freebayes

    SECONDS=0

    freebayes -F 0.5 --min-coverage 4 -p 1 -f \
      $reference_fasta \
      $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam > \
      $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf
      
      
    echo "RUNTIME (h:min:s)= $(($SECONDS / 3600)):$((($SECONDS / 60) % 60)):$(($SECONDS % 60))"
    echo END = `date +"%Y-%m-%d %H:%M:%S"`

##-------   
##bcftools variant integration of Illumina data into Pacbio polished genome
##-------

    SECONDS=0
    echo "bgzip"
bgzip -c  $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf > \
       $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf.gz
      echo "tabix"
tabix -p vcf  $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf.gz
    echo "bcftools"
bcftools consensus -f $reference_fasta \
      -o $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${num}_${run}Freebayes.fasta \
       $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf.gz
      
    echo "RUNTIME (h:min:s)= $(($SECONDS / 3600)):$((($SECONDS / 60) % 60)):$(($SECONDS % 60))"
    echo END = `date +"%Y-%m-%d %H:%M:%S"`
  
  bwa index $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${num}_${run}Freebayes.fasta 
  
   ##-------------------new variable name for next polishing steps-----------
  reference_fasta=$Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${num}_${run}Freebayes.fasta  
  num=$((num+1)) 
done
   
done # samples


###===========================
###-----------------
##quast
###-----------------
###===========================  

#for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
#do


for name in $(echo "RMK202 lyo202_96 di_K2_6h")
do

Overall_output_directory=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_withPhageAssembly/polishing/$name
sample=$name
num=1



#sample=1
threads=37
Overall_output_directory=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_withPhageAssembly/polishing/$name
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/rmk202_bacteria_phage_mag_ref.fasta
#ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/di_K2_6h_nanopore_filtered.fastq.gz
OUPUT=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_withPhageAssembly/polishing/Quast/$name


echo "========================================="
echo $sample
echo "========================================="

echo $reference_fasta
echo "---------------------------"
     
    rm -r $Overall_output_directory/quastwithRaconONT/
    mkdir -p $Overall_output_directory/quastwithRaconONT
    quast.py \
      -t ${threads} -l "FlyeAssembly,firstONTracon,secondONTracon" \
      -R $Overall_output_directory/03_thirdFreebayes/freebayesOuput/3_thirdFreebayes.fasta \
      $reference_fasta \
      $Overall_output_directory/01_firstFreebayes/freebayesOuput/1_firstFreebayes.fasta \
  $Overall_output_directory/02_secondFreebayes/freebayesOuput/2_secondFreebayes.fasta \
      -o $OUPUT
 
 done
```
look at Quast 

```{bash}

###------------------------
##local
###------------------------
mkdir -p ~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/Quast
scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_withPhageAssembly/polishing/Quast/  ~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/Quast


```
extract polished contigs and merge with Bacterial contigs for Final Reference file
for Streptococcus_phage_1 and Streptococcus_phage_2 take lyo202_96
for Streptococcus_phage_3 and di_K2_6h
for Lactococcus_phage_1 take RMK202

```{bash}

###------------------------
##extract contigs
###------------------------
##------------------------- bacteria

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished_onlyBACTERIA_PLASMID.fasta > \
 /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta
##------------------------- Streptococcus_phage_1
name=lyo202_96
i=Streptococcus_phage_1

samtools faidx /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_withPhageAssembly/polishing/$name/03_thirdFreebayes/freebayesOuput/3_thirdFreebayes.fasta ${i} >>  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta

##------------------------- Streptococcus_phage_2

name=lyo202_96
i=Streptococcus_phage_2

samtools faidx /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_withPhageAssembly/polishing/$name/03_thirdFreebayes/freebayesOuput/3_thirdFreebayes.fasta ${i} >>  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta

##------------------------- Lactobacillus_phage_1

name=RMK202
i=Lactobacillus_phage_1

samtools faidx /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_withPhageAssembly/polishing/$name/03_thirdFreebayes/freebayesOuput/3_thirdFreebayes.fasta ${i} >>  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta

##------------------------- Lactobacillus_phage_2

name=di_K2_6h
i=Streptococcus_phage_3

samtools faidx /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_withPhageAssembly/polishing/$name/03_thirdFreebayes/freebayesOuput/3_thirdFreebayes.fasta ${i} |sed 's/Streptococcus_phage_3/Lactobacillus_phage_2/g'  >>  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta

```

## mapping to genome

```{bash}
scp vincent@130.223.51.116:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta /Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/
```

first I need to gather the different raw files

```{bash}
rm /archiv/Projects/2019_RMK202_analysis/01_log/locationRawData_allSamples.txt
for genomess in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt)
do

echo -e ${genomess}"\t/archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/"${genomess}"/"${genomess}"/"${genomess}"_*_kneaddata_paired_1.fastq\t/archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/"${genomess}"/"${genomess}"/"${genomess}"_*_kneaddata_paired_2.fastq" >> /archiv/Projects/2019_RMK202_analysis/01_log/locationRawData_allSamples.txt

done

##-----------------------------
for genomess in $(awk -F "\t" '{OFS="\t"}{if($4=="202") print $1}' /data/Project/2020_StarterCultureDiversity/99_log/allStrains_collection.tsv|grep -v "11141" | grep -v  "11142"|grep  -v "11143"| grep -v  "12104"|grep  -v "12105"| grep -v  "12109")
do
echo -e ${genomess}"\t/data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM"${genomess}"/FAM"${genomess}"_*_1.fq.gz\t/data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM"${genomess}"/FAM"${genomess}"_*_2.fq.gz" >> /archiv/Projects/2019_RMK202_analysis/01_log/locationRawData_allSamples.txt


#zcat /data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM"${genomess}"/FAM"${genomess}"_*_1.fq.gz |head -1
done
#/data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz

cut -f 1 /archiv/Projects/2019_RMK202_analysis/01_log/locationRawData_allSamples.txt |wc -l
cut -f 1 /archiv/Projects/2019_RMK202_analysis/01_log/locationRawData_allSamples.txt |sort|uniq -c



###------------------------------------------
##adding evolution experiment
###------------------------------------------


for genomess in $(ls /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/ |grep "^G")
do
echo $genomess

echo -e ${genomess}"\t/archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/"${genomess}"/"${genomess}"/"${genomess}"-*_kneaddata_paired_1.fastq\t/archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/"${genomess}"/"${genomess}"/"${genomess}"-*_kneaddata_paired_2.fastq" >> /archiv/Projects/2019_RMK202_analysis/01_log/locationRawData_allSamples.txt

done

sed -i 's/18_/18-/g' /archiv/Projects/2019_RMK202_analysis/01_log/locationRawData_allSamples.txt

/archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/G1_6_18/G1_6_18/G1_6_18-*_kneaddata_paired_1.fastq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/G1_6_18/G1_6_18/G1_6_18-*_kneaddata_paired_2.fastq
```


```{bash}
###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt  ##the file with all sample names
logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final
Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta

##!!!!!!!!!!!!!change names of Forward and Reverse reads
names=G4_6_18
r1=/home/vincent/Projects/2020_StarterEvolution/01_data/20200929_Novogene/X204SC20090774-Z01-F001/trimm_Galore/gz/${names}/${names}-R1_val_1.fq.gz
r2=/home/vincent/Projects/2020_StarterEvolution/01_data/20200929_Novogene/X204SC20090774-Z01-F001/trimm_Galore/gz/${names}/${names}-R2_val_2.fq.gz

###================
##merge genome
###================



###================
##script
###================
##============
##mapping to reference
##============
bwa index $Assembly
# name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 

mkdir -p ${logFilelocation}

num=1

for names in $(cut -f 1 /archiv/Projects/2019_RMK202_analysis/01_log/locationRawData_allSamples.txt)
#for names in $(ls /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/ |grep "^G"|head -2)

  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))
  rm -r ${BaseLocation}/${names}/bwaMapping2DB/

  mkdir -p ${BaseLocation}/${names}/bwaMapping2DB/
  
  r1=$(grep "^${names}" /archiv/Projects/2019_RMK202_analysis/01_log/locationRawData_allSamples.txt |cut -f 2)
   r2=$(grep "^${names}" /archiv/Projects/2019_RMK202_analysis/01_log/locationRawData_allSamples.txt |cut -f 3)
   
 # bwa mem -t ${threads} /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids/di_rmk202_MAG_reference_polished.fasta
  bwa mem -t ${threads} ${Assembly} \
    ${r1} ${r2} | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 

samtools view -b -f 4 ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

bedtools bamtofastq -i ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam -fq ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq

rm ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam &

##-----Qualimap

mkdir -p ${BaseLocation}/${names}/bwaMapping2DB//bamqc

qualimap bamqc -bam ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir ${BaseLocation}/${names}/bwaMapping2DB//bamqc --java-mem-size=80G


 mkdir -p ${BaseLocation}/log
echo ${BaseLocation}"/"${names}"/bwaMapping2DB/bamqc" >> ${BaseLocation}/log/multiqc_logfile_finalGenome.txt
   done 
   
   
#    rm -r ${BaseLocation}/MultiQC
 
#  mkdir -p ${BaseLocation}/MultiQC
  
#  multiqc --config /archiv/logs/multiQC_config.txt --file-list  ${BaseLocation}/log/multiqc_logfile_finalGenome.txt \
#    -o ${BaseLocation}/MultiQC


##-------------------------------------------------  
##-----check read coverage
##-------------------------------------------------  
name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final 
threads=37
names=Konserve_202


num=1
rm -r ${BaseLocation}/log/mappings.txt

#for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt)
for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt)
  do
mapped=$(grep "     number of mapped reads = " ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt  |cut -d '(' -f 2 | sed 's/%)//g')

mkdir -p ${BaseLocation}/log
echo -e ${names}"\tonlyMeta\t"${mapped} >> ${BaseLocation}/log/mappings.txt

done


##bring together 

cat /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/log/mappings.txt \
/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_allRMKs202/log/mappings.txt > \
/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/log/mappings_final.txt
```


```{bash}
##-------------------------------------------------  
##move local
##-------------------------------------------------  

scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/log/mappings_final.txt /Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/
```

```{bash}

##-----------for phaster
mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all
scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all/* ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all

sed -i 's/contig_1/13496/g' /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all/S_O_202_13496.fna
sed -i 's/contig/13494/g' /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all/S_O_202_13494.fna

rm /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all/*.fna.*
rm /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all/S_O_202_SMAG.fna

cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all/*.fna > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all/all_contigs.fasta



#wget --post-file=/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all/S_O_202_24739.fna "http://phaster.ca/phaster_api?contigs=1" -O /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all/S_O_202_24739_phaster_output

#wget "http://phaster.ca/phaster_api?acc=ZZ_d240a070d2" -O home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all/S_O_202_24739_phaster_output_02
#ZZ_d240a070d2 

##----------------------------
#ldel

mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Ldel/FNA_all
scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Ldel/FNA_all/* ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Ldel/FNA_all

grep ">" ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Ldel/FNA_all
```

##### calculate mapping

Here, I calculate the number of reads only mapping to genome and plasmids (w/o phages)

```{bash}
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final
rm ${BaseLocation}/mapping_wo_phages.txt
for names in $(grep "K2" -v /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples_withEvolution.txt)
do
#echo "=================================="
#echo ${names}

totBass=$(grep "     number of sequenced bases = " ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt  | sed 's/     number of sequenced bases = //g' |sed 's/ bp//g' |sed 's/,//g')

#echo $totBass


grep "CP046134" -A 4 ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt  |awk -F "\t" -v bassss="$totBass" -v sampss="$names" '{OFS="\t"}{sum+=$4} END {print sampss,sum/bassss}' >> ${BaseLocation}/mapping_wo_phages.txt


done
cat ${BaseLocation}/mapping_wo_phages.txt

```


```{bash}

scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/mapping_wo_phages.txt ~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/
```

```{r}

library(readr)
mappings_cleaned_soPhages <- read_delim("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/mapping_wo_phages.txt",  "\t", escape_double = FALSE, col_names = c("sample","mapped"), trim_ws = TRUE)


  mappings_cleaned_soPhages$sample <- factor(mappings_cleaned_soPhages$sample, levels=c("lyo202_96","Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202","G1_6_18","G2_6_18","G3_6_18","G4_6_18","G5_6_18"))
mappings_cleaned_soPhages$mapped_percent <- mappings_cleaned_soPhages$mapped*100

mappingPercetn <- ggplot(mappings_cleaned_soPhages,aes(x=sample,y=mapped_percent))+geom_bar(stat="identity")+theme_classic()+ coord_cartesian(ylim=c(50,100))+theme(axis.text.x = element_text(angle = 45, vjust = 1, size = 8, hjust = 1))+labs(x="",y="Mapped raw reads\n[%]")
  

mappingPercetn
  svg("~/Desktop/mid_thesis/report/figures/20200430/chapter1/20201006/supplement/MappingPercent2meta_wophages.svg",width=4,height=3)

mappingPercetn
 dev.off()

 mean(mappings_cleaned_soPhages$mapped)
  sd(mappings_cleaned_soPhages$mapped)

```


#####mapping to all genomes
```{bash}
cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all/*.fna /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Ldel/FNA_all/*.fna > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly_allStrains.fasta


bwa index /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly_allStrains.fasta

threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples_withEvolution.txt  ##the file with all sample names
logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final_withALLSTRAINS
Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly_allStrains.fasta


##============
##mapping to reference
##============
rm /archiv/Projects/2019_RMK202_analysis/01_log/locationRawData_allSamples_onlyMETA.txt
for namess in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples_withEvolution.txt)
do
grep "${namess}" /archiv/Projects/2019_RMK202_analysis/01_log/locationRawData_allSamples.txt >> /archiv/Projects/2019_RMK202_analysis/01_log/locationRawData_allSamples_onlyMETA.txt 

done
 name_folder=02_againstpolished_allRMKs202

###================
##description file
###================

##!!!!!!!!!!!!!change names of Forward and Reverse reads

###================
##script
###================
##============
##mapping to reference
##============

# name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 

mkdir -p ${logFilelocation}

num=1
rm -r ${BaseLocation}/log/mappings.txt

for names in $(cut -f 1 /archiv/Projects/2019_RMK202_analysis/01_log/locationRawData_allSamples_onlyMETA.txt)
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))
  rm -r ${BaseLocation}/${names}

  mkdir -p ${BaseLocation}/${names}/bwaMapping2DB/
  
  
  r1=$(grep "^${names}" /archiv/Projects/2019_RMK202_analysis/01_log/locationRawData_allSamples.txt |cut -f 2)
   r2=$(grep "^${names}" /archiv/Projects/2019_RMK202_analysis/01_log/locationRawData_allSamples.txt |cut -f 3)
   
  #   names=G4_6_18
#r1=/home/vincent/Projects/2020_StarterEvolution/01_data/20200929_Novogene/X204SC20090774-Z01-F001/trimm_Galore/gz/${names}/${names}-R1_val_1.fq
#r2=/home/vincent/Projects/2020_StarterEvolution/01_data/20200929_Novogene/X204SC20090774-Z01-F001/trimm_Galore/gz/${names}/${names}-R2_val_2.fq

   
 # bwa mem -t ${threads} /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids/di_rmk202_MAG_reference_polished.fasta
  bwa mem -t ${threads} ${Assembly} \
    ${r1} ${r2} | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 

  
 # bwa mem -t ${threads} /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids/di_rmk202_MAG_reference_polished.fasta
  #bwa mem -t ${threads} ${Assembly} \
  #/archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_1.fastq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_2.fastq | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 

samtools view -b -f 4 ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

bedtools bamtofastq -i ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam -fq ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq

rm ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam &

##-----Qualimap

mkdir -p ${BaseLocation}/${names}/bwaMapping2DB//bamqc

qualimap bamqc -bam ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir ${BaseLocation}/${names}/bwaMapping2DB//bamqc --java-mem-size=80G


 mkdir -p ${BaseLocation}/log
echo ${BaseLocation}"/"${names}"/bwaMapping2DB/bamqc" >> ${BaseLocation}/log/multiqc_logfile_finalGenome.txt



#mapped=$(grep "     number of mapped reads = " ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt  |cut -d '(' -f 2 | sed 's/%)//g')
mapped=$(grep "     number of mapped reads = " ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt  )

mkdir -p ${BaseLocation}/log/
echo -e ${names}"\tallisolates\t"${mapped} #>> ${BaseLocation}/log/mappings.txt


   done 
   
   
#    rm -r ${BaseLocation}/MultiQC
 
#  mkdir -p ${BaseLocation}/MultiQC
  
#  multiqc --config /archiv/logs/multiQC_config.txt --file-list  ${BaseLocation}/log/multiqc_logfile_finalGenome.txt \
#    -o ${BaseLocation}/MultiQC


##-------------------------------------------------  
##-----check read coverage
##-------------------------------------------------  

#name_folder=02_againstpolished_allRMKs202 
#threads=37
#names=Konserve_202

  
  num=1
  rm -r ${BaseLocation}/log/mappings.txt
  #for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt)
  for names in $(cut -f 1 /archiv/Projects/2019_RMK202_analysis/01_log/locationRawData_allSamples_onlyMETA.txt )
    do
  #qualimap bamqc -bam ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir ${BaseLocation}/${names}/bwaMapping2DB//bamqc --java-mem-size=80G
  
  samtools flagstat ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > ${BaseLocation}/${names}/bwaMapping2DB//bamqc/flagstat.out
  
  #mapped=$(grep "     number of mapped reads = " ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt  |cut -d '(' -f 2 | sed 's/%)//g')
  mapped=$(grep "mapped (" ${BaseLocation}/${names}/bwaMapping2DB//bamqc/flagstat.out |awk -F "[(%)]" '{print $2}'  )
  
  mkdir -p ${BaseLocation}/log/
  echo -e ${names}"\tallisolates\t"${mapped} >> ${BaseLocation}/log/mappings.txt
  
  done
```


```{bash}
##-------------------------------------------------  
##move local
##-------------------------------------------------  

#scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains/log/mappings.txt Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/

scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final_withALLSTRAINS/log/mappings_cleaned.txt ~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/

```

```{r}

library(readr)
mappings_cleaned <- read_delim("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/mappings_cleaned.txt",  "\t", escape_double = FALSE, col_names = c("sample","isolates","mapped"), trim_ws = TRUE)


  mappings_cleaned$sample <- factor(mappings_cleaned$sample, levels=c("lyo202_96","Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202","G1_6_18","G2_6_18","G3_6_18","G4_6_18","G5_6_18"))


mappingPercetn <- ggplot(mappings_cleaned,aes(x=sample,y=mapped))+geom_bar(stat="identity")+theme_classic()+ coord_cartesian(ylim=c(99.5,100))+theme(axis.text.x = element_text(angle = 45, vjust = 1, size = 8, hjust = 1))+labs(x="",y="Mapped raw reads\n[%]")
  


  svg("~/Desktop/mid_thesis/report/figures/20200430/chapter1/20201006/supplement/MappingPercent2meta.svg",width=4,height=3)

mappingPercetn
 dev.off()

 mean(mappings_cleaned$mapped)
  sd(mappings_cleaned$mapped)

```

##prophage 24739

```{bash}
threads=37
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping4PROPHAGE/
long=S_O_202_24739

#names=$(echo $long |cut -d '_' -f 4)
names=24739

Assembly=/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all/${long}.fna
 rm -r ${BaseLocation}/${names}/bwaMapping2DB/

  mkdir -p ${BaseLocation}/${names}/bwaMapping2DB/
  
  r1=$(grep "^${names}" /archiv/Projects/2019_RMK202_analysis/01_log/locationRawData_allSamples.txt |cut -f 2)
   r2=$(grep "^${names}" /archiv/Projects/2019_RMK202_analysis/01_log/locationRawData_allSamples.txt |cut -f 3)
   
 # bwa mem -t ${threads} /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids/di_rmk202_MAG_reference_polished.fasta
 bwa index $Assembly
  bwa mem -t ${threads} ${Assembly} \
    ${r1} ${r2} | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 

samtools depth -a ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > ${BaseLocation}/${names}/bwaMapping2DB/${names}_depth.txt

###-------------------------------------
#Map phage against genome
###-------------------------------------
threads=37
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping4PROPHAGE/phage2_genome_24739
long=S_O_202_24739
names=24739

Assembly=/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/FNA_all/${long}.fna

mkdir -p ${BaseLocation}

  bwa mem -x intractg -t ${threads} ${Assembly} \
    /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_phage/activePhagemapping/repeating_regions/genomes/Sterm_phage2.fasta | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}_rawIllumina_bwamem_sorted.bam - 


```

```{bash}
mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/03_mapping4PROPHAGE/
scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/03_mapping4PROPHAGE//24739/bwaMapping2DB/24739_depth.txt ~/Desktop/Projects/2019_RMK202_analysis/03_mapping4PROPHAGE/

```


```{r}
library(ggplot2)
my_cov <- read_delim("Desktop/Projects/2019_RMK202_analysis/03_mapping4PROPHAGE/24739_depth.txt",   "\t", escape_double = FALSE, col_names = c("Genome", "Locus", "Depth"), trim_ws = TRUE)

n_bins <- ceiling(nrow(my_cov) / 100)
bin_assign <- cut(my_cov$Locus, breaks = n_bins, labels = FALSE)
my_cov$bin <- bin_assign
# compute mean depth per bin
Depth_bin <- tapply(X = my_cov$Depth, INDEX = my_cov$bin, FUN = mean)
# compute corresponding mean locus per bin
Locus_bin <- tapply(X = my_cov$Locus, INDEX = my_cov$bin, FUN = mean)
# put together in data frame
my_cov_bin <- data.frame("Locus" = Locus_bin, 
                         "Depth" = Depth_bin)
my_cov_bin$LocusKB <- my_cov_bin$Locus / 1000



ggplot(my_cov_bin, aes(x = LocusKB, y = Depth)) + 
  geom_line() + 
  labs(x = "Locus (kb)") + 
  theme_bw(base_size = 16)+lims(x=c(1070,1130))


```



## check plasmid cov
here I check if all ldel strains have the plasmid

```{bash}


rm ${BaseLocation}/LdelPlasmid_coverage.txt
for names in $(cat ${samplesss}|grep "^L"|grep "Lyo" -v)
  do
  echo "==================="
  echo ${names}
CovGenome=$(grep ">>>>>>> Coverage per contig" -A 20 ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt |grep "CP046131"  |cut -f 5)
Covplasmid=$(grep ">>>>>>> Coverage per contig" -A 20 ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt |grep "CP046132" |cut -f 5)
 
echo -e ${names}"\t"${CovGenome}"\t"${Covplasmid} >> ${BaseLocation}/LdelPlasmid_coverage.txt
 done
```

```{bash}
/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/LdelPlasmid_coverage.txt


scp vincent@130.223.49.145:/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/LdelPlasmid_coverage.txt /home/vincent/Desktop/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final

```

```{r}
library(readr)
library(ggplot2)
LdelPlasmid_coverage <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/LdelPlasmid_coverage.txt",  "\t", escape_double = FALSE, col_names = c("sample","genomeCoverage","plasmidCoverage"),  trim_ws = TRUE) 

LdelPlasmid_coverage <- LdelPlasmid_coverage[1:9,]
LdelPlasmid_coverage$copyNumber <- LdelPlasmid_coverage$plasmidCoverage/LdelPlasmid_coverage$genomeCoverage

range(LdelPlasmid_coverage$copyNumber)
ggplot(LdelPlasmid_coverage,aes(x=sample,y=copyNumber))+geom_bar( stat="identity")+theme_classic()
```


## SNP calling freebayes (K1)

This is a good (blog)[https://bioinformatics-core-shared-training.github.io/cruk-summer-school-2017/Day2/vcf-intro.nb.html] on how to use freebayes and how to parse the output. 
Finally I will do freebayes on all samples simultaniously. 

Interesting output from column 11 on:

GT	Genotype v :1
DP	Read Depth  :1040
DPR	Number of observation for each allele  : 1,1037
RO	Reference allele observation count :1
QR	Sum of quality of the reference observations :10
AO	Alternate allele observation count :1037
QA	Sum of quality of the alternate observations :36917
GL	Genotype Likelihood, log10-scaled likelihoods of the data given the called genotype for each possible genotype generated from the reference and alternate alleles given the sample ploidy

Very nice [blog](http://www.ddocent.com/filtering/) illustrating good filtering practices.

```{bash,eval=FALSE,echo=FALSE}
#/data/Project/2020_StarterCultureDiversity/01_Data/20200515_genomeSeq/01_rawdata/trimmed_Galore/gz/FAM${sample}/FAM${sample}_R1_val_1.fq.gz

###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt  ##the file with all sample names
logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final
Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta

###--------special
FREEBAYES_out=freebayesOuput_WithONT_Parallel_default

###================
##script
###================

##--------------------------------------------------------------------------------------------------------------------------------------
##-----------change read group
##--------------------------------------------------------------------------------------------------------------------------------------

#name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 
names=G4_6_18

###------------------adding readgroup to bam
#rm ${BaseLocation}/log/multiqc_logfile_bams.txt
mkdir -p  ${BaseLocation}/log/
for names in $(cut -f 1 /archiv/Projects/2019_RMK202_analysis/01_log/locationRawData_allSamples.txt)
#for names in $(cut -f 1 /archiv/Projects/2019_RMK202_analysis/01_log/locationRawData_allSamples.txt |grep "^G")
do
echo ${names}
#echo ${BaseLocation}"/"${names}"/bwaMapping2DB/"${names}"_mapping2ref.bam " >> ${BaseLocation}/log/multiqc_logfile_bams.txt

#done
java -jar /home/vincent/apps/picard/picard.jar AddOrReplaceReadGroups \
       I=${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam \
       O=${BaseLocation}/${names}/bwaMapping2DB/${names}_mapping2ref.bam \
       RGID=${names} \
       RGLB=lib1 \
       RGPL=illumina \
       RGPU=unit1 \
       RGSM=${names}

rm ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam

echo ${BaseLocation}"/"${names}"/bwaMapping2DB/"${names}"_mapping2ref.bam " >> ${BaseLocation}/log/multiqc_logfile_bams.txt

done




##remove trailing whitespaces

 sed -i 's/[ \t]*$//' ${BaseLocation}/log/multiqc_logfile_bams.txt

##--------------------------------------------------------------------------------------------------------------------------------------
##-----------check bam mapping
##--------------------------------------------------------------------------------------------------------------------------------------

# for map in $(cat ${BaseLocation}/log/multiqc_logfile_bams.txt)
 for names in $(cut -f 1 /archiv/Projects/2019_RMK202_analysis/01_log/locationRawData_allSamples.txt |grep "^G")
    do
    echo "----------------------------------------------------"
    echo $map
    
    samtools flagstat $map |grep "mapped ("
    
    done

##--------------------------------------------------------------------------------------------------------------------------------------
##-----------indexing file
##--------------------------------------------------------------------------------------------------------------------------------------
##need before freebayes parallel

#  for names in $(cat ${BaseLocation}/log/multiqc_logfile_bams.txt)
  for names in $(cut -f 1 /archiv/Projects/2019_RMK202_analysis/01_log/locationRawData_allSamples.txt |grep -v "^G")
do
echo $names

samtools index ${BaseLocation}/${names}/bwaMapping2DB/${names}_mapping2ref.bam

done

##--------------------------------------------------------------------------------------------------------------------------------------
##-----------SNP calling
##--------------------------------------------------------------------------------------------------------------------------------------

#rm -r ${BaseLocation}/freebayesOuput_WithONT_notParallel_default
#mkdir -p ${BaseLocation}/freebayesOuput_WithONT_notParallel_default


  #  freebayes -F 0.05 -C 3 --min-coverage 30 --pooled-discrete \
  #    --bam-list ${BaseLocation}/log/multiqc_logfile_bams.txt -f \
  #    ${Assembly} > \
  #   ${BaseLocation}/freebayesOuput_WithONT_notParallel_default/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf
  
##--------------------------------------------------------------------------------------------------------------------------------------
##-----------SNP calling parallel
##--------------------------------------------------------------------------------------------------------------------------------------


  rm -r ${BaseLocation}/${FREEBAYES_out}
  mkdir -p ${BaseLocation}/${FREEBAYES_out}
  
  samtools faidx $Assembly
 
    #sort  ${BaseLocation}/log/multiqc_logfile_bams.txt | uniq  |grep -v "12107" |grep "13491" -v | grep "13492"-v| grep "13493" -v > ${BaseLocation}/log/multiqc_logfile_bams_uniq.txt

  sort  ${BaseLocation}/log/multiqc_logfile_bams.txt | uniq  > ${BaseLocation}/log/multiqc_logfile_bams_uniq.txt
    freebayes-parallel <(/home/vincent/apps/freebayes/scripts/fasta_generate_regions.py $Assembly 100000) 36 -f ${Assembly} -C 5 --pooled-continuous --min-alternate-fraction 0.05 --min-coverage 10 --bam-list ${BaseLocation}/log/multiqc_logfile_bams_uniq.txt > ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf
  
  
  grep "^#" -v ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf |cut -f 1 |sort|uniq -c
  
##-------------zipping

#bgzip -c /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains/freebayesOuput_WithONT_02/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf > /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains/freebayesOuput_WithONT_02/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf.gz
#tabix -p  vcf /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final/freebayesOuput_WithONT/variantCallsfreebayes_all.vcf.gz


##--------------------------------------------------------------------------------------------------------------------------------------
##------------vcftools filtering
##--------------------------------------------------------------------------------------------------------------------------------------


###==============================1. Qualty filtering QC>30

vcftools --vcf ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf --minQ 30 --remove-indels --recode --recode-INFO-all --out ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL  

  grep "^#" -v ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf |cut -f 1 |sort|uniq -c

#vcftools --vcf ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x.vcf  --counts2 --out ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC  

###==============================4. hist-indel-len


#vcftools --vcf ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC.recode.vcf --hist-indel-len --out ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC

###==============================2. remove indel

#vcftools --vcf ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC.recode.vcf --mac 1 --remove-indels --minQ 30 --recode --recode-INFO-all --out ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL

###==============================3. INFO

#vcftools --vcf ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf --missing-indv --out ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC  
###==============================4. hist-indel-len


#vcftools --vcf ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf --hist-indel-len --out ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL

#grep "^#" -v ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf |cut -f 1 |sort|uniq -c

##--------------------------------------------------------------------------------------------------------------------------------------
##------------prep for R
##--------------------------------------------------------------------------------------------------------------------------------------



grep "^##" -v ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf >  ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_prepR.vcf


##--------------------------------------------------------------------------------------------------------------------------------------
##------------bring local
##--------------------------------------------------------------------------------------------------------------------------------------




mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/06_snpCalling2ONT_new

scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/freebayesOuput_WithONT_Parallel_default/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_prepR.vcf /home/vincent/Desktop/Projects/2019_RMK202_analysis/06_snpCalling2ONT_new/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_prepR_new.vcf


```

## POGENOM
This is the analysis done with [POGENOM](https://pogenom.readthedocs.io/en/latest/POGENOM_Usage.html)
We do this to get population genomic insights such as dN/dS ratios.
Important to note is that the gff file should not contain any fasta entries. 

```{bash}


scp /home/vincent/Downloads/*eggnog.annotations vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/ 

scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/S_I_202_SMAG/PROKKA_08282020.faa /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/


scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Ldel/L_I_202_LMAG/*.faa /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/


/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA_new/all_interaction_functioning_Genes_cleaned.txt

```


```{bash}
VCFFILE=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/freebayesOuput_WithONT_Parallel_default/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf
Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta


mkdir -p  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PGAP/ForPOGENOM/

#grep "^202-LMAG-1" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Ldel/202-LMAG/PROKKA_04012020.gff |sed 's/^202-LMAG-1/CP046131/g' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/bothSpecies_prepped.gff

#grep "^202-SMAG-1" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA_04012020.gff |sed 's/^202-SMAG-1/CP046134/g' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/bothSpecies_prepped.gff

grep "^#" -v /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Ldel/202-LMAG/PROKKA_04012020.gff |sed 's/^202-LMAG-1/CP046131/g' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/bothSpecies_prepped.gff

grep "^#" -v /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA_04012020.gff |sed 's/^202-SMAG-1/CP046134/g' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/bothSpecies_prepped.gff


awk -F "\t" '{OFS="\t"}{if($3=="CDS")print $0}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/bothSpecies_prepped.gff > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/bothSpecies_prepped_ready.gff

#GFF_file=/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/MAG_annotation_onlyBAC_mod.gff 


###--------------------------------
#RMK202
#Konserve_202
#Versand_202
#Lyo_202_2012
#lyo202_96
#Lyo_202_2014




#vcftools --vcf /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/freebayesOuput_WithONT_Parallel_default/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf --minQ 30 --remove-indels --recode --recode-INFO-all \
#  --keep /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/SamplesMeta.txt --out \
#  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_onlyMeta

sed -e 's/\r/\n/g' /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/MAG_annotation_onlyBAC_mod.gff  >  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/MAG_annotation_onlyBAC_mod_cleaned.gff

#GFF_file=/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/MAG_annotation_onlyBAC_mod_cleaned.gff
GFF_file=/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/bothSpecies_prepped_ready.gff


Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta

VCFFILE=/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_onlyMeta.recode.vcf

grep "^#" -v ${GFF_file} |cut -f 1|sort|uniq -c
#tail ${GFF_file}
grep ">" ${Assembly}
grep "^#" -v ${VCFFILE} |cut -f 1|sort|uniq -c
##========================================
##Run POGENOME
##========================================


###-----------------
##Sterm
###-----------------
sed 's/^202-SMAG-1/CP046134/g' /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/S_I_202_SMAG/PROKKA_08282020.gff|sed 's/>202-SMAG-1/CP046134/g' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/bothSpecies_prepped_sterm_cleaned.gff
GFF_file_03=/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/bothSpecies_prepped_sterm_cleaned.gff

#genomes=S_M_202_SMAG
#echo "##gff-version 3" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gff/${genomes}_forPOGENOM.gff
#echo "##sequence-region 202-SMAG-1 1 1865439" >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gff/${genomes}_forPOGENOM.gff

#grep "^CP046134" /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gff/${genomes}.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene")print $0}' |sed 's/ID=.*locus_tag=/ID=/g' |sed 's/gene/CDS/g'|sed 's/;.*$//g'| awk '{print $0"_gene"}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gff/${genomes}_forPOGENOM.gff
#GFF_file_03=/archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gff/${genomes}_forPOGENOM.gff
#echo "##FASTA" >> ${GFF_file_03}
#cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/S_I_202_SMAG/PROKKA_08282020.fna |sed 's/>202-SMAG-1/CP046134/g' >> ${GFF_file_03}
#samtools faidx /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/fasta/${genomes}.fasta CP046134 > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/fasta/${genomes}_forPOGENOM.fasta


mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PGAP/ForPOGENOM/

vcftools --vcf /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/freebayesOuput_WithONT_Parallel_default/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf --chr CP046134 --minQ 30 --remove-indels --recode --recode-INFO-all \
  --keep /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples_withEvolution.txt --out \
  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PGAP/ForPOGENOM/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_onlyMeta_onlySterm

VCFFILE=/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PGAP/ForPOGENOM/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_onlyMeta_onlySterm.recode.vcf


rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/POGENOM_PGAP/bothGEnomes_allSamples/sterm/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/POGENOM_PGAP/bothGEnomes_allSamples/sterm/


perl /home/vincent/apps/POGENOM/POGENOM-0.8.1/pogenom.pl --vcf_file ${VCFFILE} --out /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/POGENOM_PGAP/bothGEnomes_allSamples/sterm/RunPogenome_all_rmk202_new --gff_file ${GFF_file_03}   --genetic_code_file /home/vincent/apps/POGENOM/POGENOM-0.8.1/bacterial_genetic_code_table11_ncbi.txt

#--fasta_file /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/fasta/${genomes}_forPOGENOM.fasta 
#--genome_size 1865459
###-----------------
##Ldel
###-----------------
sed 's/^202-LMAG-1/CP046131/g' /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Ldel/L_I_202_LMAG/PROKKA_10022020.gff|sed 's/>202-LMAG-1/CP046131/g' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/bothSpecies_prepped_ldel_cleaned.gff
GFF_file_03=/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/bothSpecies_prepped_ldel_cleaned.gff


#genomes=L_M_202_LMAG



#grep "^CP046131" /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gff/${genomes}.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene")print $0}' |sed 's/ID=.*locus_tag=/ID=/g' |sed 's/gene/CDS/g'|sed 's/;.*$//g'| awk '{print $0"_gene"}' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gff/${genomes}_forPOGENOM.gff
#GFF_file_03=/archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gff/${genomes}_forPOGENOM.gff
#echo "##FASTA" >> ${GFF_file_03}
#samtools faidx /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/fasta/${genomes}.fasta CP046131 |sed 's/>//g' >> ${GFF_file_03}


mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PGAP/ForPOGENOM/


vcftools --vcf /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/freebayesOuput_WithONT_Parallel_default/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf --chr CP046131 --minQ 30 --remove-indels --recode --recode-INFO-all \
  --keep /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples_withEvolution.txt --out \
  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PGAP/ForPOGENOM/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_onlyMeta_onlyLdel

VCFFILE=/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PGAP/ForPOGENOM/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_onlyMeta_onlyLdel.recode.vcf


rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/POGENOM_PGAP/bothGEnomes_allSamples/Ldel/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/POGENOM_PGAP/bothGEnomes_allSamples/Ldel/


perl /home/vincent/apps/POGENOM/POGENOM-0.8.1/pogenom.pl --vcf_file ${VCFFILE} --out /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/POGENOM_PGAP/bothGEnomes_allSamples/Ldel/RunPogenome_all_rmk202_new --gff_file ${GFF_file_03}   --genetic_code_file /home/vincent/apps/POGENOM/POGENOM-0.8.1/bacterial_genetic_code_table11_ncbi.txt

#--genome_size 2166765
```




move local for Anders
```{bash}

mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/POGENOM_test/

scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_onlyMeta.recode.vcf ~/Desktop/Projects/2019_RMK202_analysis/POGENOM_test/longitudinalSamples_snv.vcf

scp vincent@130.223.51.116:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta ~/Desktop/Projects/2019_RMK202_analysis/POGENOM_test/MAG_assembly.fasta

scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/ForPOGENOM/bothSpecies_prepped.gff ~/Desktop/Projects/2019_RMK202_analysis/POGENOM_test/MAG_annotation_onlyBAC.gff

##-------------------------final

scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/POGENOM_PGAP/bothGEnomes_allSamples/Ldel/RunPogenome_all_rmk202_new.pNpS-per-gene.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/RunPogenome_Ldel.pNpS-per-gene.txt

scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/POGENOM_PGAP/bothGEnomes_allSamples/sterm/RunPogenome_all_rmk202_new.pNpS-per-gene.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/RunPogenome_Sterm.pNpS-per-gene.txt

sed -i 's/ID=//g' ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/RunPogenome_Ldel.pNpS-per-gene.txt
sed -i 's/ID=//g' ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/RunPogenome_Sterm.pNpS-per-gene.txt

scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA_new/all_interaction_functioning_Genes_cleaned.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/




```

```{r}
library(readr)
library(ggplot2)
library(lubridate)
library(tidyverse)
Genes_of_Interest <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/all_interaction_functioning_Genes.txt",  "\t", escape_double = FALSE, col_names = c("species","gene","OG","numberGenomes"),  trim_ws = TRUE) %>% select(-numberGenomes)


OGs_Ldel <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/OGs_Ldel.txt", "\t", escape_double = FALSE, col_names = c("OG","Gene"), trim_ws = TRUE)
OGs_Sterm <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/OGs_Sterm.txt", "\t", escape_double = FALSE, col_names = c("OG","Gene"), trim_ws = TRUE)
Genes_of_Interest_Ldel <- Genes_of_Interest %>% filter(species=="Ldel")
Genes_of_Interest_Sterm <- Genes_of_Interest %>% filter(species=="Sterm")

GENES_Ldel <- merge(OGs_Ldel,Genes_of_Interest_Ldel,by="OG",all = TRUE)
# table(GENES_Ldel$gene)
GENES_Sterm <- merge(OGs_Sterm,Genes_of_Interest_Sterm,by="OG",all = TRUE)
# table(GENES_Sterm$gene)

table(GENES_Ldel_extended$gene)

RunPogenome_Ldel_pNpS_per_gene <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/RunPogenome_Ldel.pNpS-per-gene.txt",   "\t", escape_double = FALSE, trim_ws = TRUE) %>% select(-c(`RMK202 pNpS`,`Konserve_202 pNpS`,`Versand_202 pNpS`,`Lyo_202_2012 pNpS`,`lyo202_96 pNpS`,`Lyo_202_2014 pNpS`))
RunPogenome_Sterm_pNpS_per_gene <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/RunPogenome_Sterm.pNpS-per-gene.txt",   "\t", escape_double = FALSE, trim_ws = TRUE) %>% select(-c(`RMK202 pNpS`,`Konserve_202 pNpS`,`Versand_202 pNpS`,`Lyo_202_2012 pNpS`,`lyo202_96 pNpS`,`Lyo_202_2014 pNpS`))

ggplot(RunPogenome_Sterm_pNpS_per_gene,aes(x=`All_samples_combined pNpS`))+geom_density()
ggplot(RunPogenome_Ldel_pNpS_per_gene,aes(x=`All_samples_combined pNpS`))+geom_density()

all_interaction_functioning_Genes_cleaned <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/all_interaction_functioning_Genes_cleaned.txt",   "\t", escape_double = FALSE, col_names = c("geness","Name"),  trim_ws = TRUE)





all_interaction_functioning_Genes_cleaned$gene <-  str_split_fixed(all_interaction_functioning_Genes_cleaned$geness, fixed("_"), 2)[,2]
all_interaction_functioning_Genes_cleaned$species <-  str_split_fixed(all_interaction_functioning_Genes_cleaned$geness, fixed("_"), 2)[,1]

all_interaction_functioning_Genes_cleaned <- all_interaction_functioning_Genes_cleaned %>% select(-geness)


GENES_Sterm_extended <- merge(RunPogenome_Sterm_pNpS_per_gene,all_interaction_functioning_Genes_cleaned,by.x="Gene",by.y="Name",all.x = TRUE)
GENES_Ldel_extended <- merge(RunPogenome_Ldel_pNpS_per_gene,all_interaction_functioning_Genes_cleaned,by.x="Gene",by.y="Name",all.x = TRUE)


# GENES_Ldel_extended <- merge(GENES_Ldel,RunPogenome_Ldel_pNpS_per_gene,by="Gene",all = TRUE)
# table(GENES_Ldel$gene)
# GENES_Sterm_extended  <- merge(GENES_Sterm,RunPogenome_Sterm_pNpS_per_gene,by="Gene",all = TRUE)
# table(GENES_Sterm$gene)


GENES_Sterm_reduced <- GENES_Sterm_extended %>% filter(!is.na(gene)) %>% filter(!is.na(Num_loci)) 
GENES_Ldel_reduced <- GENES_Ldel_extended %>% filter(!is.na(gene)) %>% filter(!is.na(Num_loci))
GENES_Ldel_reduced$`All_samples_combined pNpS`[is.na(GENES_Ldel_reduced$`All_samples_combined pNpS`)] <- 0
GENES_Sterm_reduced$`All_samples_combined pNpS`[is.na(GENES_Sterm_reduced$`All_samples_combined pNpS`)] <- 0

GENES_Sterm_reduced_prep <- GENES_Sterm_reduced %>% select(c(gene,`All_samples_combined pNpS`,species,Num_loci))
GENES_ldel_reduced_prep <- GENES_Ldel_reduced %>% select(c(gene,`All_samples_combined pNpS`,species,Num_loci))




GENES_long <- rbind(GENES_Sterm_reduced_prep,GENES_ldel_reduced_prep) #%>% add_column(functioning="protocooperation\nrelated")

GENES_long$functioning <- ifelse(grepl("pep",GENES_long$gene),"peptidase","protocooperation\nrelated")


GENES_long <- GENES_long[
  order( GENES_long[,5] ),
]



ggheatmap_CRISPR <- ggplot(GENES_long, aes(species, gene,color = `All_samples_combined pNpS`))+
 # geom_tile(color = "white",size=1.1,shape="circle")+
 geom_point(size=8,shape="circle")+
  geom_point(shape = 1,size = 8,colour = "black")+
  # theme_classic()+
 # scale_fill_gradient2(low = "grey", high = "red",
  # midpoint = 90, limit = c(80,100), space = "Lab",
  # name="protein id",na.value = 'white',colour="black",pch=21) +
   scale_color_gradient2(low = "red", high = "grey",
  midpoint = 1, limit = c(0,2), space = "Lab",
  name="pN/pS",na.value = 'white') +
  # scale_fill_distiller(name = "SNPs", palette = "Blues", direction = -1)+
  # scale_fill_viridis(alpha=0.8)+
  labs(legend="CRISPR ID",x="",y="")+
  # scale_fill_distiller(name = "SNPs", palette = "Viridis", direction = -1)+
  # scale_fill_distiller(name = "ANI", palette = "Reds", direction = 1)+
  theme_minimal()+ # minimal theme
 theme(axis.text.y = element_text(size = 12),
      axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1))+
 coord_fixed()
# Print the heatmap
print(ggheatmap_CRISPR)


##===================================================
##plot
##=================================================== 

  svg("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder//HEATMAP_interactionGenes_pN_pS.svg",width=3,height=9)

print(ggheatmap_CRISPR)

 dev.off()
 
##===================================================
##distribution
##=================================================== 

 
GENES_Sterm_reverse <- GENES_Sterm_extended %>% filter(is.na(gene)) %>% filter(!is.na(Num_loci)) 
GENES_Ldel_reverse <- GENES_Ldel_extended %>% filter(is.na(gene)) %>% filter(!is.na(Num_loci))
GENES_Ldel_reverse$`All_samples_combined pNpS`[is.na(GENES_Ldel_reverse$`All_samples_combined pNpS`)] <- 0
GENES_Sterm_reverse$`All_samples_combined pNpS`[is.na(GENES_Sterm_reverse$`All_samples_combined pNpS`)] <- 0

GENES_Sterm_reverse_prep <- GENES_Sterm_reverse %>% select(c(gene,`All_samples_combined pNpS`,species,Num_loci))
GENES_ldel_reverse_prep <- GENES_Ldel_reverse %>% select(c(gene,`All_samples_combined pNpS`,species,Num_loci))


GENES_long_reverse <- rbind(GENES_Sterm_reverse_prep,GENES_ldel_reverse_prep) %>% add_column(functioning="other function")

GENES_complete <- rbind(GENES_long_reverse,GENES_long)

ggplot(GENES_complete,aes(x=`All_samples_combined pNpS`))+geom_density()+facet_wrap(~functioning)
 
ggplot(GENES_complete,aes(x=`All_samples_combined pNpS`,y=Num_loci))+geom_point()+facet_wrap(~functioning)


  GENES_complete$functioning <- factor(GENES_complete$functioning, levels=c("protocooperation\nrelated","peptidase","other function"))


colorssss <- c("red","orange","grey88")
dnDSplot <- ggplot(GENES_complete,aes(x=`All_samples_combined pNpS`,y=Num_loci,fill=functioning,color=functioning))+geom_point(size=2)+theme_classic()+labs(x="pN/pS",y="Number of mutations")+scale_fill_manual(values = colorssss)+scale_color_manual(values = colorssss)+theme(legend.title = element_blank())
dnDSplot


  svg("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder//dot_plot_pN_pS.svg",width=4.5,height=4)

dnDSplot

 dev.off()
 

```


#####Prophage detection

Here, I try to identify reads that map both to the Streptococcus genome and the phage

```{bash}



name_folder=02_againstpolished_single_K1_final_kneaddata

threads=37
#rm -r /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/

  for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/names_onlymeta.txt)
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))

  rm -r /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/03_mapping_Prophage/${names}/
  mkdir -p /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/03_mapping_Prophage/${names}/
  
samtools view -H /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/03_mapping_Prophage/${names}/header.bam

samtools view /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam |grep  "S_phage_RMK202" |grep "S_thermophilus_RMK202" >  /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/03_mapping_Prophage/${names}/data_tmp.bam

cat /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/03_mapping_Prophage/${names}/header.bam /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/03_mapping_Prophage/${names}/data_tmp.bam | samtools view -S -b  >  /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/03_mapping_Prophage/${names}/${names}_mappingProphage.bam

done
```


getting coverage information

```{bash}

date=20190513

#rm /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/log/mappingcoverage.txt
#mkdir -p /home/vincent/Projects/2019_pilotplant/06_MetaSNV_ONT/{99_logs,01_analysis_01}

  threads=37

###------------make a file file
  num=1
  for names in $(cat /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name_allOfExperiment.txt)
    do
    
   # sampleShort=$(echo ${names}| awk -F "_" 'BEGIN{OFS="_"}{print $1,$2}')
  
   #NanoSample=$(grep "$sampleShort" /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt) 
    
    echo ${num}"/37  :" ${names} 
    num=$((num+1))


#samtools depth  -m 80000 /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam  |  awk '{sum+=$3} END { print "Average = ",sum/NR}' |awk -F "\t" -v namess="$names"  'BEGIN{OFS="\t"}{print $0, namess}' #>> /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/${names}/bwaMapping2DB/${names}_coverage.txt

##extract from qualimap

grep -A 100 ">>>> Coverage per contig" /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata/${names}/bwaMapping2DB/bamqc/genome_results.txt | sed '1,2d' | sed '/^$/d' | sed -e 's/^[ \t]*//'  |awk -F "\t" -v namess="$names"  'BEGIN{OFS="\t"}{print $0, namess}' >> /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata/log/mappingcoverage.txt

done

##transfer coverage local

scp vincent@130.223.51.151:/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata/log/mappingcoverage.txt ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/mappingcoverage_2ONT.txt

```

```{r}

library(readr)
library("stringr")
library("ggplot2")
library(tidyverse)


mappingcoverage <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/mappingcoverage_2ONT.txt",  "\t", escape_double = FALSE, col_names = c("contig","size","readsMapped","coverage","std","sample"),  trim_ws = TRUE)

mappingcoverage$genome <- mappingcoverage$contig
mappingcoverage[grep("S_phage_RMK202",mappingcoverage$genome),"genome"] <- "S_phage_RMK202"
mappingcoverage[grep("L_plasmid_RMK202_",mappingcoverage$genome),"genome"] <- "L_plasmid_RMK202"
unique(mappingcoverage$genome)

mappingcoverage_reduced <- mappingcoverage[,-c(4,5)]
mappingcoverage_reduced$sample <- as.factor(mappingcoverage_reduced$sample)
mappingcoverage_reduced$genome <- as.factor(mappingcoverage_reduced$genome)


  ##=============coverage    

   mappingcoverage_reduced_summed <-aggregate(.~sample+genome, data=mappingcoverage_reduced[,c("sample","genome","readsMapped")], sum, na.rm=TRUE)
      mappingcoverage_reduced_size <-aggregate(.~sample+genome, data=mappingcoverage_reduced[,c("sample","genome","size")], sum, na.rm=TRUE)
      mappingcoverage_reduced_final <- cbind(mappingcoverage_reduced_summed,size=mappingcoverage_reduced_size$size)
      mappingcoverage_reduced_final$coverage <- (mappingcoverage_reduced_final$readsMapped*500)/mappingcoverage_reduced_final$size
      
      
  ##=============percent    


  mappingcoverage_reduced_final_summed <-aggregate(. ~sample , data=mappingcoverage_reduced_final[,c("sample","coverage")], sum, na.rm=TRUE)
  mappingcoverage_reduced_final$total_coverage <- mappingcoverage_reduced_final_summed[match(mappingcoverage_reduced_final$sample,mappingcoverage_reduced_final_summed$sample),"coverage"]
  mappingcoverage_reduced_final$percent_coverage <- 100*(mappingcoverage_reduced_final$coverage/mappingcoverage_reduced_final$total_coverage)

  mappingcoverage_reduced_final <- mappingcoverage_reduced_final[which(mappingcoverage_reduced_final$genome!="Bos_taurus_mitochondrion"),]
  
 ##=============sample order    
  levels(factor(mappingcoverage_reduced_final$genome))
  mappingcoverage_reduced_final$genome_f <- factor(mappingcoverage_reduced_final$genome, levels=c("S_thermophilus_RMK202","S_ther_potential_Prophage","S_phage_RMK202","L_delbrueckii_RMK202","L_delbrueckii_plasmid_RMK202_01","L_plasmid_RMK202","L_phage_RMK202_01","L_phage_RMK202_02"))

  mappingcoverage_reduced_final$genome_f <- factor(mappingcoverage_reduced_final$genome, levels=rev(c("S_thermophilus_RMK202","S_ther_potential_Prophage","S_phage_RMK202","L_delbrueckii_RMK202","L_delbrueckii_plasmid_RMK202_01","L_plasmid_RMK202","L_phage_RMK202_01","L_phage_RMK202_02")))
  

  mappingcoverage_reduced_final$sample_f = factor(mappingcoverage_reduced_final$sample, levels=c("Lyo_202_2012","Lyo_202_2014","RMK202","Konserve_202","Versand_202","di_K1_0h","th_K1_0h","di_K1_2h","th_K1_2h","di_K1_4h","th_K1_4h","di_K1_6h","th_K1_6h","di_K1_8h","th_K1_8h","di_K1_10h","th_K1_10h","di_K1_12h","th_K1_12h","di_K1_24h","th_K1_24h","S72","S50","L104","L108","L35","L39","L44","L70","L71","L80","L99"))

   ##=============colours    


Sterm_colours <- rev(c("#EB4D4D","#FAA0A0","indianred1","#10B552","#36E37B","#6CF5A3","cyan3","cyan"))

levels(mappingcoverage_reduced_final$sample)

mappingcoverage_reduced_final <- mappingcoverage_reduced_final[mappingcoverage_reduced_final$sample %in% c("Konserve_202","Lyo_202_2012","Lyo_202_2014","RMK202","Versand_202"),]


p2 <-  ggplot( data = mappingcoverage_reduced_final,aes(y = percent_coverage, x = sample_f, group=genome_f,fill = genome_f))+ geom_bar( stat="identity")+
    labs("",
         x="",
         y="relative Abundance")+
    theme_classic()+
   # scale_color_viridis(discrete=TRUE)+
  scale_fill_manual(values=Sterm_colours)+
  # scale_fill_viridis(discrete=TRUE)+
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
      #axis.text.x = element_blank(),
          legend.position="right",
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank()
          )

  p2
#   svg("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.svg",width=7,height=4)
#    # png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 1000, height = 800,res=300)
# p2
#  dev.off()
#  
 

```



###CIGAR parsing


```{bash}

#rm /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/CIGAR_parsing
 
  num=1
  for names in $(cat /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name_allOfExperiment.txt)
    do
echo ${num}"/33  :" ${names}
    num=$((num+1))
    
    #names=th_K1_12h
    #rm -r /home/vincent/Projects/2019_pilotplant/05_mapping2DB/${names}/BacteriaDBmapping
  
    mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/CIGAR_parsing/
    
    samtools index /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam
    
perl /home/vincent/scripts/mapped_read_Length_extended_reduced.pl /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/CIGAR_parsing/${names}_rawIllumina_bwamem_sorted_statsforR.txt

done



##-----------------move local
#names=th_K1_12h
scp -r vincent@130.223.51.151:/home/vincent/Projects/2019_pilotplant/05_mapping2DB/logs_mapping/ ~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/QC_ofDBmapping


```
## Species Coverage

####all together



```{bash}

scp -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/gene_CLEANED_FINAL_ALL.gff vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/



scp /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff vincent@130.223.51.116:/archiv/Projects/2019_Pilotplan/11_PGAP/final/sterm/



scp /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.current.gb.gff vincent@130.223.51.116:/archiv/Projects/2019_Pilotplan/11_PGAP/final/ldel/


```

```{bash}


###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt  ##the file with all sample names
logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final
Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples_withEvolution.txt  ##the file with all sample names

###================
##bring all CDS gffs together
###================
mkdir -p  /archiv/Projects/2019_RMK202_analysis/00_FINAL/07_abundances/prep/

grep "^CP" -v /archiv/Projects/2019_RMK202_analysis/phage_annotation/phaster/ZZ_be76e5e924.PHASTER/gene_CLEANED_FINAL_ALL.gff |awk -F "\t" '{OFS="\t"}{print $1,$2,$3}' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/07_abundances/prep/genes_for_Abundances.bed


grep "^#" -v /archiv/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |grep "^CP" |awk -F "\t" '{OFS="\t"}{if($3=="CDS")print $1,$4,$5}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/07_abundances/prep/genes_for_Abundances.bed


grep "^#" -v /archiv/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.current.gb.gff |grep "^CP" |awk -F "\t" '{OFS="\t"}{if($3=="CDS")print $1,$4,$5}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/07_abundances/prep/genes_for_Abundances.bed



###================
##mapping
###================

##============
##mapping to reference
##============
names=RMK202
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/07_abundances/mapping/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/07_abundances/mapping/

for names in $(cat ${samplesss} )
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))



bedtools coverage -bed -a /archiv/Projects/2019_RMK202_analysis/00_FINAL/07_abundances/prep/genes_for_Abundances.bed -b ${BaseLocation}/${names}/bwaMapping2DB/${names}_mapping2ref.bam  > \
  /archiv/Projects/2019_RMK202_analysis/00_FINAL/07_abundances/mapping/${names}2bacteriaDB.bed 
  
  #${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam
    cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/07_abundances/mapping/${names}2bacteriaDB.bed |awk -F "[\t]" -v namess="$names"  'BEGIN{OFS="\t"}{print $0,namess}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/07_abundances/mapping/all_2bacteria_and_phages_from_MAG.bed 
    
   done 
   



```

```{bash}

mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/07_abundances/mapping/

scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/07_abundances/mapping/all_2bacteria_and_phages_from_MAG.bed /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/07_abundances/mapping/all_2bacteria_and_phages_from_MAG.bed


```


```{r}


source("/home/vincent/Desktop/ScriptRepository/R_functions-master/R_functions/g_legend.R") ##for plotting only legends
library(readr)
library(ggplot2)
library(dplyr)
library(tidyverse)
library(grid)
library(gridExtra)
library(cowplot) ## for arranging plots
library(stringr) ##for split string
library(rlist) ##rbind all dataframe of list
library(viridis) ##colour palette
##===================================
#-------------file import


  read_count <- read_delim("/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/07_abundances/mapping/all_2bacteria_and_phages_from_MAG.bed","\t", escape_double = FALSE, col_names = c("chr","start","end","count","length_mapped","geneLength","unknown","sample"),trim_ws = TRUE)

table(read_count$chr)

# read_count$species <- ifelse(read_count$chr=="L_del_phage_01","L_del_phage_01","S_term_phage_01")
  
  table(read_count$chr)
  
  read_count$geneCoverage  <- (read_count$count*600)/read_count$geneLength
  
  # ggplot(read_count,aes(y=geneCoverage,group=sort,color=sort,fill=sort))+geom_boxplot()+facet_grid(sample~species, scales="free")
  
  library(dplyr)
  
  
  # all_final <- read_count %>% 
  #   group_by(sample,chr) %>% 
  #   dplyr::summarize(median = median(geneCoverage)) 
  # 
   all_final <- read_count %>% 
    group_by(sample,chr) %>% 
    dplyr::summarize(median = median(geneCoverage)) %>% filter(chr %in% c("CP046131","CP046134","Lactobacillus_phage_1","Streptococcus_phage_1","Streptococcus_phage_2"))
  
  
    total_samples_sumTreatment <- aggregate(. ~sample, data=all_final[,c("sample","median")], sum, na.rm=TRUE)
  
  all_final$total_coverage <- total_samples_sumTreatment[match(all_final$sample,total_samples_sumTreatment$sample),"median"]
  all_final$percent_coverage <- 100*(all_final$median/all_final$total_coverage)
  # 
  # all_final$species <- as.factor(all_final$species)
  # 
  # levels(all_final$species)
  
  
  # all_final$species <- factor(all_final$species, levels=rev(c("S_thermophilus_RMK202","S_term_plasmid_01","S_term_phage_01","L_delbrueckii_RMK202","L_del_plasmid_01","L_plasmid_RMK202","L_del_plasmid_02","L_del_phage_01")))
  # all_final$species <- factor(all_final$species, levels=rev(c("S_thermophilus_RMK202","S_phage_RMK202","L_delbrueckii_RMK202","L_delbrueckii_plasmid_RMK202_01","L_plasmid_RMK202","L_phage_RMK202_01","L_phage_RMK202_02")))
  
  # all_final$sample <- as.factor(all_final$sample)
  
  table(all_final$sample)
  all_final <- all_final %>% filter(! sample %in% c("th_K2_8h","di_K2_6h"))
  all_final$sample <- factor(all_final$sample, levels=(c("lyo202_96","Lyo_202_2012","Konserve_202","Lyo_202_2014","RMK202","Versand_202","G1_6_18","G2_6_18","G3_6_18","G4_6_18","G5_6_18")))
  # bac_final_02$sample <- revalue(bac_final_02$sample, c("lyo202_96"="Lyo\n1996","Lyo_202_2012"="Lyo\n2012", "Lyo_202_2014"="Lyo\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018","di_K2_6h"="cheesemaking\nday1","th_K2_8h"="cheesemaking\nday2"))
  # bac_final_02$sample <- factor(bac_final_02$sample, levels=(c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2")))
  



# table(total_samples$phage) %>% length()
write.table(all_final,"/home/vincent/Desktop/Projects/2019_Pilotplan/coverage_rmk202.tsv",sep = "\t",quote = FALSE,col.names = FALSE)

write.table(all_final,"/home/vincent/Desktop/Projects/2019_Pilotplan/coverage_rmk202_n32.tsv",sep = "\t",quote = FALSE,col.names = FALSE)

 # all_colours <-c("#C5F6FA" ,"#6CF5A3" ,"#36E37B" ,"#10B552", "darkorange",  "#FAA0A0","#EB4D4D")
 all_colours <-c("#C5F6FA" ,"#6CF5A3" ,"#36E37B" ,"#10B552", "darkorange",  "#FAA0A0","#EB4D4D")


# mito_colours <- c("#AFBACC","#BE99AB") # sequential_hcl(5,palette="Grays")[c(4,3)]
# Sterm_colours <- c("#FCC2C2","#EB4D4D")# c("#FAA0A0","#EB4D4D") #c("#FF7D87","#E71D32")    
# Ldel_colours <- c("#6CF5A3","#10B552")  # c("#8CD211","#4C8400")   # c("#5AA700","#2D660A") #brewer.pal(9,name="YlGnBu")[c(4,6)] #sequential_hcl(5,palette="Purples 3")[c(3,2)]
# lactococcus_colours <- c("#5AAAFA","#4178BE")  #sequential_hcl(5,palette="Terrain 2")[c(3,2)]
# rest_colours <-  c("#C5F6FA","#99E9F2","#66D9E8","#3BC9DB") #brewer.pal(9,name="YlOrBr")[c(5,7,8,9)] #sequential_hcl(5,palette="BluGrn")[2:5]
# colours_phages_02 <- c(rest_colours,lactococcus_colours,Ldel_colours,Sterm_colours,mito_colours)


##----------------change name
library(plyr)
library(dplyr)
all_final$sample <- revalue(all_final$sample, c("lyo202_96"="Lyo 1996","Lyo_202_2012"="Lyo\n2012", "Lyo_202_2014"="Lyo\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018"))
##----------------plot
# all_final <- all_final %>% filter(!sample %in% c("cheesemaking\nday1","cheesemaking\nday2"))

levels(all_final$chr)
# all_final$species <- revalue(all_final$species, c("lyo202_96"="Lyo\n1996","Lyo_202_2012"="Lyo\n2012", "Lyo_202_2014"="Lyo\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018","di_K2_6h"="cheesemaking\nday1","th_K2_8h"="cheesemaking\nday2"))
#all_final$chr <- factor(all_final$chr, levels=c("L_del_phage_01" , "L_del_plasmid_02" ,"L_plasmid_RMK202", "L_del_plasmid_01","S_term_phage_01","S_term_plasmid_01","L_delbrueckii_RMK202","S_thermophilus_RMK202"))

all_final$chr <- factor(all_final$chr, levels=c("Lactobacillus_phage_2" , "Lactobacillus_phage_1" ,"CP046133", "CP046132","Streptococcus_phage_2","Streptococcus_phage_1","CP046135","CP046131","CP046134"))



# all_colours
all_colours_new <-  c("#36E37A","#C5F6FA","#6CF5A3","#36E37B","orange","darkorange","#FAA0A0", "#10B552","#EB4D4D")
  
all_colours_new <-  c("#dbece1","#a0cbd2","#6bf5a2","#66c264","#ffa300","#ff8a00","#ff5200", "#10B552","#EB4D4D")
  
all_colours_new <-  c("#a0cbd2","#ffa300","#ff8a00", "#10B552","#EB4D4D")

# c("#C5F6FA","#6CF5A3","#36E37B", "#10B552","darkorange","#FAA0A0","#EB4D4D")
# c("L_del_phage_01" , "L_del_plasmid_02" ,"L_plasmid_RMK202", "L_del_plasmid_01","S_term_phage_01","S_term_plasmid_01","L_delbrueckii_RMK202","S_thermophilus_RMK202")

PrelAbundance <-  ggplot( data = all_final,aes(y = percent_coverage, x = sample, group=interaction(chr),fill = chr))+ geom_bar( stat="identity")+
    labs("",
         x="",
         y="relative abundance")+
    theme_classic()+
   # scale_color_viridis(discrete=TRUE)+
   scale_fill_manual(values=all_colours_new)+
  # scale_fill_viridis(discrete=TRUE)+
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
        # axis.text.x = element_blank(),
          legend.position="right",
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank()
          )

  PrelAbundance
  svg("~/Desktop/mid_thesis/report/figures/20200430/chapter1/relative_abundance.svg",width=4.5,height=3)
#    # png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 1000, height = 800,res=300)
# 
PrelAbundance
# 
dev.off()
# 
 # svg("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.svg",width=7,height=4)
# png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 2000, height = 1000,res=300)
# 
# PrelAbundance
# 
# dev.off()

##----------------amount phages


# all_final$genus <- ifelse(grepl("S_",all_final$species),"CP046131","Lactobacillus")

all_final_phage <- all_final %>% filter(chr %in% c("Lactobacillus_phage_1","Lactobacillus_phage_2","Streptococcus_phage_1","Streptococcus_phage_2"))

all_final_phage %>% 
  group_by(sample) %>% 
  dplyr::summarize(sum = sum(percent_coverage))  


all_final_phage$median_01 <- all_final_phage$median
all_final_phage %>% filter(chr=="Lactobacillus_phage_1") %>% group_by(chr) %>% 
   dplyr::summarize(mean = mean(median_01),sd=sd(median_01))


all_final_phage %>% 
  group_by(sample) %>% 
  dplyr::summarize(sum = sum(median)) %>%   dplyr::summarize(mean = mean(sum),sd=sd(sum))

all_final_phage %>% 
  group_by(sample) %>% 
  dplyr::summarize(sum = sum(percent_coverage)) %>%   dplyr::summarize(mean = mean(sum),sd=sd(sum))
# %>% 
#    group_by(chr) %>% 
#   dplyr::summarize(min = min(sum),
#             max = max(sum))  



  total_samples_phages <- aggregate(. ~sample, data=all_final_phage[,c("sample","median")], sum, na.rm=TRUE)

all_final_phage$total_coverage_phages <- total_samples_phages[match(all_final_phage$sample,total_samples_phages$sample),"median"]
all_final_phage$percent_coverage_phages <- 100*(all_final_phage$median/all_final_phage$total_coverage_phages)
# 

all_final_phage$genus <- ifelse(grepl("S",all_final_phage$chr),"Streptococcus_phages","Lactobacillus_phages")


all_final_phage %>% 
  group_by(sample,chr) %>% 
  dplyr::summarize(sum = sum(percent_coverage_phages))  

all_final_phage %>% 
  group_by(sample,chr) %>% 
  dplyr::summarize(sum = sum(percent_coverage_phages))  %>% filter(chr=="Lactobacillus_phage_1")

table(all_final_phage$chr)

# %>% 
#    group_by(chr) %>% 
#   dplyr::summarize(min = min(sum),
#             max = max(sum))  
all_final_phage %>% 
  group_by(sample,genus) %>% 
  dplyr::summarize(sum = sum(percent_coverage_phages))  

##----------------amount of Streptococci and Lactobacilli


all_final$genus <- ifelse(grepl("S_",all_final$species),"CP046131","Lactobacillus")

all_final_bacteria <- all_final %>% filter(chr %in% c("CP046131","CP046134"))

all_final_bacteria %>% 
  group_by(sample,chr) %>% 
  dplyr::summarize(sum = sum(percent_coverage))  %>% 
   group_by(chr) %>% 
  dplyr::summarize(min = min(sum),
            max = max(sum),median=median(sum))  

##----------------amount Ldelplasmid


# all_final$genus <- ifelse(grepl("S_",all_final$species),"CP046131","Lactobacillus")

all_final_plasmid <- all_final %>% filter(chr %in% c("CP046132"))

all_final_plasmid %>% 
  group_by(sample,chr) %>% 
  dplyr::summarize(sum = sum(percent_coverage))  %>% 
   group_by(chr) %>% 
  dplyr::summarize(min = min(sum),
            max = max(sum))  

##----------------amount of Lactobacillus delbrueckii RMK202


all_final_bacteria %>% 
  group_by(chr) %>% 
  dplyr::summarize(min = min(percent_coverage),
            max = max(percent_coverage))  


##----------------coverage of Lactobacillus delbrueckii RMK202
table(all_final_bacteria$species)

all_final_bacteria[which(all_final_bacteria$species=="CP046131"),]


##-------------------------------plasmid copy number

all_final_copyNUMBER <- all_final %>% filter(chr %in% c("CP046132","CP046131")) %>% select(sample,chr, median) %>% spread(., chr, median)
all_final_copyNUMBER$copyNUMBER <- 100*(all_final_copyNUMBER$CP046132/all_final_copyNUMBER$CP046131)

plasmidCopyNUMber <- ggplot(all_final_copyNUMBER,aes(x=sample,y=copyNUMBER))+geom_point()+theme_classic()+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9))+coord_trans( y="log2")+labs(x="",y="plasmid copy number\n[%]")

plasmidCopyNUMber

svg("~/Desktop/mid_thesis/report/figures/20200430/chapter1/20201006/supplement/plasmidCopyNUMBER.svg",width=4.5,height=3)
   # png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 1000, height = 800,res=300)

plasmidCopyNUMber

dev.off()

all_final_copyNUMBER %>% 
  dplyr::summarize(sum = mean(copyNUMBER),sd=sd(copyNUMBER)) 


mean(all_final_copyNUMBER$copyNUMBER)
sd(all_final_copyNUMBER$copyNUMBER)




##----------------------normalise with actual mapping percent--------------------------------

# library(readr)
# 
# mappings <- read_delim("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/mappings_final.txt",  "\t", escape_double = FALSE, col_names = c("sample","type","mapping"),  trim_ws = TRUE) %>% filter(type=="onlyMeta") %>% select(add=-type)
# mappings$sample <- revalue(mappings$sample, c("L71"="mst4","lyo202_96"="Lyo 1996","Lyo_202_2012"="Lyo\n2012", "Lyo_202_2014"="Lyo\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018","di_K2_6h"="cheesemaking\nday1","th_K2_8h"="cheesemaking\nday2"))
# table(all_final$sample)
# table(mappings$sample)
# 
# all_final_normalised <-  merge(all_final,mappings,by.x ="sample",by.y="sample" )
# table(all_final_normalised$sample)
# 
# revalue(all_final_normalised$chr,c()
# 
# table(all_final_normalised$sample)
# all_final_normalised$percent_coverage_normalised <- all_final_normalised$percent_coverage*((all_final_normalised$mapping/100))
# PrelAbundance <-  ggplot( data = all_final_normalised,aes(y = percent_coverage_normalised, x = sample, group=interaction(chr),fill = chr))+ geom_bar( stat="identity")+
#     labs("",
#          x="",
#          y="relative abundance")+
#     theme_classic()+
#   geom_hline(yintercept=100, linetype="dashed", color = "grey")+
#    # scale_color_viridis(discrete=TRUE)+
#    scale_fill_manual(values=all_colours_new)+
#   # scale_fill_viridis(discrete=TRUE)+
#     theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
#         # axis.text.x = element_blank(),
#           legend.position="right",
#           #legend.justification=c(1,1), legend.position=c(1,1),
#           legend.title = element_blank()
#           )
# 
#   PrelAbundance
#   svg("~/Desktop/mid_thesis/report/figures/20200430/chapter1/relative_abundance.svg",width=4.5,height=3)
#    # png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 1000, height = 800,res=300)
# 
# PrelAbundance
# 
# dev.off()
# # 

##===============================
##only Bacteria
##===============================

all_final_bacteria

  total_samples_sumTreatment <- aggregate(. ~sample, data=all_final_bacteria[,c("sample","median")], sum, na.rm=TRUE)

all_final_bacteria$total_coverage <- total_samples_sumTreatment[match(all_final_bacteria$sample,total_samples_sumTreatment$sample),"median"]
all_final_bacteria$percent_coverage <- 100*(all_final_bacteria$median/all_final_bacteria$total_coverage)



# all_final$chr <- factor(all_final$chr, levels=c("Lactobacillus_phage_2" , "Lactobacillus_phage_1" ,"CP046133", "CP046132","Streptococcus_phage_2","Streptococcus_phage_1","CP046135","CP046131","CP046134"))



# all_colours
# all_colours_new <-  c("#36E37A","#C5F6FA","#6CF5A3","#36E37B","orange","darkorange","#FAA0A0", "#10B552","#EB4D4D")
  
all_colours_new <-  c( "#10B552","#EB4D4D")
  
# c("#C5F6FA","#6CF5A3","#36E37B", "#10B552","darkorange","#FAA0A0","#EB4D4D")
# c("L_del_phage_01" , "L_del_plasmid_02" ,"L_plasmid_RMK202", "L_del_plasmid_01","S_term_phage_01","S_term_plasmid_01","L_delbrueckii_RMK202","S_thermophilus_RMK202")

PrelAbundance_bacteria <-  ggplot( data = all_final_bacteria,aes(y = percent_coverage, x = sample, group=interaction(chr),fill = chr))+ geom_bar( stat="identity")+
    labs("",
         x="",
         y="relative abundance")+
    theme_classic()+
   # scale_color_viridis(discrete=TRUE)+
   scale_fill_manual(values=all_colours_new)+
  # scale_fill_viridis(discrete=TRUE)+
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
        # axis.text.x = element_blank(),
          legend.position="right",
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank()
          )

library(patchwork)

  PrelAbundance_bacteria
  svg("~/Desktop/mid_thesis/report/figures/20200430/chapter1/relative_abundance_all.svg",width=10,height=4.5)
#    # png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 1000, height = 800,res=300)
# 
(PrelAbundance+theme(legend.position = "none"))+(PrelAbundance_bacteria+theme(legend.position = "none"))+PrelAbundance
# 
dev.off()
# 
```

##### phage mapping

```{bash}



###================
##old
###================



###================
##description file
###================
threads=37
BaseAnnotation=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/annotation_eggnog_rast/assembly_20191129/
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples_withEvolution.txt  ##the file with all sample names
#BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP
BaseLocation=/archiv/Projects/2019_R1MK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final

###================
##Script
###================

/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm

##----------------------------------------------
##convert RAST annotation to bed
##----------------------------------------------
mkdir -p ${BaseAnnotation}/merged_all_ForCoverage/
rm ${BaseAnnotation}/merged_all_ForCoverage/all_phages.bed
for phageNAMES in $(echo "phage_ldel phage_sterm")
do
convert2bed --input=gff < ${BaseAnnotation}/${phageNAMES}/*.gff > ${BaseAnnotation}/${phageNAMES}/${phageNAMES}.bed
#convert2bed --input=gff < /archiv/Projects/2019_pilotplant/02_annotation/rastServer/Streptococcus_phage.fasta.gbk.gff > /archiv/Projects/2019_pilotplant/02_annotation/rastServer/Streptococcus_phage.bed
#convert2bed --input=gff < /archiv/Projects/2019_pilotplant/02_annotation/rastServer/Lactobacillus_phage_RMK202_02.fasta.gbk.gff > /archiv/Projects/2019_pilotplant/02_annotation/rastServer/Lactobacillus_phage_RMK202_02.bed

cat \
${BaseAnnotation}/${phageNAMES}/${phageNAMES}.bed  >> \
${BaseAnnotation}/merged_all_ForCoverage/all_phages.bed
done #phageNAMES


##----------------------------------------------
##Phage coverage
##----------------------------------------------

rm -r ${BaseAnnotation}/merged_all_ForCoverage/phageDBmapping_rmk202/

 # rm  /home/vincent/Projects/2019_pilotplant/05_mapping2DB//log/multiqc_logfile.txt
  num=1
  for names in $(cat ${samplesss})
    do
echo ${num}"/8  :" ${names}
    num=$((num+1))
    rm -r ${BaseAnnotation}/merged_all_ForCoverage/phageDBmapping_rmk202//${names}
  
    mkdir -p ${BaseAnnotation}/merged_all_ForCoverage/phageDBmapping_rmk202//${names}
    

bedtools coverage -bed -a ${BaseAnnotation}/merged_all_ForCoverage/all_phages.bed -b ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam  > \
  ${BaseAnnotation}/merged_all_ForCoverage/phageDBmapping_rmk202//${names}/${names}2phageDB.bed 
  
done


##----------------------------------------------
###----------------change name of bed file
##----------------------------------------------
names=Konserve_202
 
 rm -r ${BaseAnnotation}/merged_all_ForCoverage/phageDBmapping_rmk202/finalmapping/
 mkdir -p ${BaseAnnotation}/merged_all_ForCoverage/phageDBmapping_rmk202/finalmapping/
   num=1
for names in $(cat ${samplesss})
    do
    
echo ${num}"/8  :" ${names}
    num=$((num+1))
  
    cat ${BaseAnnotation}/merged_all_ForCoverage/phageDBmapping_rmk202//${names}/${names}2phageDB.bed  | sed -e 's/ID=.*;OG=/OG=/g'|awk -F "[\t]" -v namess="$names"  'BEGIN{OFS="\t"}{print $0,namess}'  > ${BaseAnnotation}/merged_all_ForCoverage/phageDBmapping_rmk202/finalmapping/${names}_finalPhages_forR.bed
    

    done
cat ${BaseAnnotation}/merged_all_ForCoverage/phageDBmapping_rmk202/finalmapping//*.bed > ${BaseAnnotation}/merged_all_ForCoverage/phageDBmapping_rmk202/finalmapping//all_Phages_allSamples.txt



##----------------------------------------------
###-------------on local machine
##----------------------------------------------


echo "transfer local with the following command:"
echo " "
echo -e "\nscp vincent@130.223.51.116:"${BaseAnnotation}"/merged_all_ForCoverage/phageDBmapping_rmk202/finalmapping//all_Phages_allSamples.txt /home/vincent/Desktop/Projects/2019_Pilotplan/09_annoation/all_Phages_allSamples_rmk202.txt"



```

```{r}


source("/home/vincent/Desktop/ScriptRepository/R_functions-master/R_functions/g_legend.R") ##for plotting only legends
library(readr)
library(ggplot2)
library(dplyr)
library(tidyverse)
library(grid)
library(gridExtra)
library(cowplot) ## for arranging plots
library(stringr) ##for split string
library(rlist) ##rbind all dataframe of list
library(viridis) ##colour palette
##===================================
#-------------file import


  read_count <- read_delim("/home/vincent/Desktop/Projects/2019_Pilotplan/09_annoation/all_Phages_allSamples_rmk202.txt","\t", escape_double = FALSE, col_names = c("chr","start","end","name","unknown2","orientation","origin","sort","unknown","description","count","length_mapped","geneLength","sample2","sample"),trim_ws = TRUE)

table(read_count$chr)

read_count$species <- ifelse(read_count$chr=="L_del_phage_01","L_del_phage_01","S_term_phage_01")

table(read_count$species)

read_count$geneCoverage  <- (read_count$count*600)/read_count$geneLength

ggplot(read_count,aes(y=geneCoverage,group=sort,color=sort,fill=sort))+geom_boxplot()+facet_grid(sample~species, scales="free")

library(dplyr)


phage_final <- read_count %>% 
  group_by(sample,species) %>% 
  dplyr::summarize(median = median(geneCoverage))


```
##### bacteria reference genome

Currently, I am leaving out the bam file filtering (CIGAR.filtering) and going straight to the gene coverage

```{bash}


###================
##description file
###================

BaseLocation_Ldel=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/
gb_file_ldel=L_delbrueckii_RMK202.current.gb
BaseFILNAME_ldel=L_delbrueckii_RMK202

BaseLocation_Sterm=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/
gb_file_sterm=S_thermophilus_RMK202.current.gb
BaseFILNAME_sterm=S_thermophilus_RMK202

threads=37
BaseAnnotation=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/annotation_eggnog_rast/assembly_20191129/
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt  ##the file with all sample names
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP


###================
##script
###================

##-------------------------------------
###----------------prep gffs
##-------------------------------------

cat ${BaseLocation_Sterm}/${BaseFILNAME_sterm}_gene.gff \
${BaseLocation_Ldel}/${BaseFILNAME_ldel}_gene.gff > \
${BaseAnnotation}/merged_all_ForCoverage/all_bacteria.gff


cut -f 1 ${BaseAnnotation}/merged_all_ForCoverage/all_bacteria.gff|sort|uniq -c

##-------------------------------------
###----------------extract coverage
##-------------------------------------



rm -r ${BaseAnnotation}/merged_all_ForCoverage/bacteriaDBmapping_rmk202/

 # rm  /home/vincent/Projects/2019_pilotplant/05_mapping2DB//log/multiqc_logfile.txt
  num=1
  for names in $(cat ${samplesss})
    do
echo ${num}"/8  :" ${names}
    num=$((num+1))
    rm -r ${BaseAnnotation}/merged_all_ForCoverage/bacteriaDBmapping_rmk202//${names}
  
    mkdir -p ${BaseAnnotation}/merged_all_ForCoverage/bacteriaDBmapping_rmk202//${names}
    

bedtools coverage -bed -a ${BaseAnnotation}/merged_all_ForCoverage/all_bacteria.gff -b ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam  > \
  ${BaseAnnotation}/merged_all_ForCoverage/bacteriaDBmapping_rmk202//${names}/${names}2bacteriaDB.bed 
  
done




##-------------------------------------
###----------------change name of bed file
##-------------------------------------

names=Konserve_202
 
 rm -r ${BaseAnnotation}/merged_all_ForCoverage/bacteriaDBmapping_rmk202/finalmapping/
 mkdir -p ${BaseAnnotation}/merged_all_ForCoverage/bacteriaDBmapping_rmk202/finalmapping/
   num=1
for names in $(cat ${samplesss})
    do
    
echo ${num}"/8  :" ${names}
    num=$((num+1))
  
    cat ${BaseAnnotation}/merged_all_ForCoverage/bacteriaDBmapping_rmk202//${names}/${names}2bacteriaDB.bed  | sed -e 's/ID=.*;OG=/OG=/g'|awk -F "[\t]" -v namess="$names"  'BEGIN{OFS="\t"}{print $0,namess}'  > ${BaseAnnotation}/merged_all_ForCoverage/bacteriaDBmapping_rmk202/finalmapping/${names}_finalBacteria_forR.bed
    

    done
cat ${BaseAnnotation}/merged_all_ForCoverage/bacteriaDBmapping_rmk202/finalmapping/*.bed > ${BaseAnnotation}/merged_all_ForCoverage/bacteriaDBmapping_rmk202/finalmapping//all_Bacteria_allSamples.txt

 cut -f 1 ${BaseAnnotation}/merged_all_ForCoverage/bacteriaDBmapping_rmk202/finalmapping//all_Bacteria_allSamples.txt | sort|uniq -c
 

##----------------------------------------------
###-------------on local machine
##----------------------------------------------


echo "transfer local with the following command:"
echo " "
echo -e "\nscp vincent@130.223.51.116:"${BaseAnnotation}"/merged_all_ForCoverage/bacteriaDBmapping_rmk202/finalmapping//all_Bacteria_allSamples.txt /home/vincent/Desktop/Projects/2019_Pilotplan/09_annoation/all_Bacteria_allSamples_rmk202.txt"


```



```{r}


source("/home/vincent/Desktop/ScriptRepository/R_functions-master/R_functions/g_legend.R") ##for plotting only legends
library(readr)
library(ggplot2)
library(dplyr)
library(tidyverse)
library(grid)
library(gridExtra)
library(cowplot) ## for arranging plots
library(stringr) ##for split string
library(rlist) ##rbind all dataframe of list
library(viridis) ##colour palette
library(plyr)
library(dplyr)
##===================================
#-------------file import


  read_count_bac <- read_delim("/home/vincent/Desktop/Projects/2019_Pilotplan/09_annoation/all_Bacteria_allSamples_rmk202.txt","\t", escape_double = FALSE, col_names = c("chr","type","sort","start","end","unknown2","orientation","unknown","description","count","length_mapped","geneLength","covered","sample"),trim_ws = TRUE)

table(read_count_bac$chr)

read_count_bac$species <- revalue(read_count_bac$chr, c("CP046131"="L_delbrueckii_RMK202","CP046132"="L_del_plasmid_01","CP046133"="L_del_plasmid_02","CP046134"="S_thermophilus_RMK202","CP046135"="S_term_plasmid_01"))



table(read_count_bac$species)

read_count_bac$geneCoverage  <- (read_count_bac$count*600)/read_count_bac$geneLength

ggplot(read_count_bac,aes(y=geneCoverage))+geom_boxplot()+facet_grid(sample~species, scales="free")

library(dplyr)

bac_final <- read_count_bac %>% 
  dplyr::group_by(sample,species) %>% 
  dplyr::summarize(median = median(geneCoverage))


##=======================only bac plot

bac_final_02 <- bac_final %>% filter(species %in% c("L_delbrueckii_RMK202","S_thermophilus_RMK202")) 
 bac_final_03 <- aggregate(. ~sample, data=bac_final_02[,c("sample","median")], sum, na.rm=TRUE)

bac_final_02$total_coverage <- bac_final_03[match(bac_final_02$sample,bac_final_03$sample),"median"]
bac_final_02$percent_coverage <- 100*(bac_final_02$median/bac_final_02$total_coverage)
bac_final_02$species <- as.factor(bac_final_02$species)
bac_final_02$species <- factor(bac_final_02$species, levels=rev(c("S_thermophilus_RMK202","L_delbrueckii_RMK202")))
bac_final_02$sample <- as.factor(bac_final_02$sample)
table(bac_final_02$sample)
bac_final_02$sample <- revalue(bac_final_02$sample, c("lyo202_96"="Lyo\n1996","Lyo_202_2012"="Lyo\n2012", "Lyo_202_2014"="Lyo\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018","di_K2_6h"="cheesemaking\nday1","th_K2_8h"="cheesemaking\nday2"))
bac_final_02$sample <- factor(bac_final_02$sample, levels=(c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2")))



##----------------change name

#bac_final_02$sample <- revalue(bac_final_02$sample, c("lyo202_96"="Lyo\n1996","Lyo_202_2012"="Lyo\n2012", "Lyo_202_2014"="Lyo\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018"))
# all_final$sample <- sample_relative_wide %>% select(site,"Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018")

##----------------plot
all_colours <- rev(c("#EB4D4D","#10B552") ) 

bac_final_02 <- bac_final_02 %>% filter(!sample %in% c("cheesemaking\nday1","cheesemaking\nday2"))

PrelAbundance <-  ggplot( data = bac_final_02,aes(y = percent_coverage, x = sample, group=interaction(species),fill = species))+ geom_bar( stat="identity")+
    labs("",
         x="",
         y="relative abundance")+
    theme_classic()+
   # scale_color_viridis(discrete=TRUE)+
   scale_fill_manual(values=all_colours)+
  # scale_fill_viridis(discrete=TRUE)+
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
        # axis.text.x = element_blank(),
          legend.position="right",
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank()
          )

  PrelAbundance
  svg("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance_bac.svg",width=4.5,height=3)
   # png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 1000, height = 800,res=300)

PrelAbundance

dev.off()

```


#####bring together
can only run with previous two chapters.

```{r}
library(dplyr)
all_final <- rbind(phage_final,bac_final)
# all_final <- rbind(phage_final,bac_final)


  total_samples_sumTreatment <- aggregate(. ~sample, data=all_final[,c("sample","median")], sum, na.rm=TRUE)

all_final$total_coverage <- total_samples_sumTreatment[match(all_final$sample,total_samples_sumTreatment$sample),"median"]
all_final$percent_coverage <- 100*(all_final$median/all_final$total_coverage)

all_final$species <- as.factor(all_final$species)

levels(all_final$species)


all_final$species <- factor(all_final$species, levels=rev(c("S_thermophilus_RMK202","S_term_plasmid_01","S_term_phage_01","L_delbrueckii_RMK202","L_del_plasmid_01","L_plasmid_RMK202","L_del_plasmid_02","L_del_phage_01")))
# all_final$species <- factor(all_final$species, levels=rev(c("S_thermophilus_RMK202","S_phage_RMK202","L_delbrueckii_RMK202","L_delbrueckii_plasmid_RMK202_01","L_plasmid_RMK202","L_phage_RMK202_01","L_phage_RMK202_02")))

all_final$sample <- as.factor(all_final$sample)

levels(all_final$sample)

all_final$sample <- factor(all_final$sample, levels=(c("lyo202_96","Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202","di_K2_6h","th_K2_8h")))
# bac_final_02$sample <- revalue(bac_final_02$sample, c("lyo202_96"="Lyo\n1996","Lyo_202_2012"="Lyo\n2012", "Lyo_202_2014"="Lyo\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018","di_K2_6h"="cheesemaking\nday1","th_K2_8h"="cheesemaking\nday2"))
# bac_final_02$sample <- factor(bac_final_02$sample, levels=(c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2")))




# table(total_samples$phage) %>% length()

write.table(all_final,"/home/vincent/Desktop/Projects/2019_Pilotplan/coverage_rmk202.tsv",sep = "\t",quote = FALSE,col.names = FALSE)

 all_colours <-c("#C5F6FA" ,"#6CF5A3" ,"#36E37B" ,"#10B552", "darkorange",  "#FAA0A0","#EB4D4D")


# mito_colours <- c("#AFBACC","#BE99AB") # sequential_hcl(5,palette="Grays")[c(4,3)]
# Sterm_colours <- c("#FCC2C2","#EB4D4D")# c("#FAA0A0","#EB4D4D") #c("#FF7D87","#E71D32")    
# Ldel_colours <- c("#6CF5A3","#10B552")  # c("#8CD211","#4C8400")   # c("#5AA700","#2D660A") #brewer.pal(9,name="YlGnBu")[c(4,6)] #sequential_hcl(5,palette="Purples 3")[c(3,2)]
# lactococcus_colours <- c("#5AAAFA","#4178BE")  #sequential_hcl(5,palette="Terrain 2")[c(3,2)]
# rest_colours <-  c("#C5F6FA","#99E9F2","#66D9E8","#3BC9DB") #brewer.pal(9,name="YlOrBr")[c(5,7,8,9)] #sequential_hcl(5,palette="BluGrn")[2:5]
# colours_phages_02 <- c(rest_colours,lactococcus_colours,Ldel_colours,Sterm_colours,mito_colours)


##----------------change name
library(plyr)
library(dplyr)
all_final$sample <- revalue(all_final$sample, c("lyo202_96"="Lyo\n1996","Lyo_202_2012"="Lyo\n2012", "Lyo_202_2014"="Lyo\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018","di_K2_6h"="cheesemaking\nday1","th_K2_8h"="cheesemaking\nday2"))
##----------------plot
all_final <- all_final %>% filter(!sample %in% c("cheesemaking\nday1","cheesemaking\nday2"))

levels(all_final$species)
# all_final$species <- revalue(all_final$species, c("lyo202_96"="Lyo\n1996","Lyo_202_2012"="Lyo\n2012", "Lyo_202_2014"="Lyo\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018","di_K2_6h"="cheesemaking\nday1","th_K2_8h"="cheesemaking\nday2"))
all_final$species <- factor(all_final$species, levels=c("L_del_phage_01" , "L_del_plasmid_02" ,"L_plasmid_RMK202", "L_del_plasmid_01","S_term_phage_01","S_term_plasmid_01","L_delbrueckii_RMK202","S_thermophilus_RMK202"))

all_colours
all_colours_new <-  c("#C5F6FA","#6CF5A3","#36E37B","darkorange","#FAA0A0", "#10B552","#EB4D4D")
  
  
c("#C5F6FA","#6CF5A3","#36E37B", "#10B552","darkorange","#FAA0A0","#EB4D4D")
# c("L_del_phage_01" , "L_del_plasmid_02" ,"L_plasmid_RMK202", "L_del_plasmid_01","S_term_phage_01","S_term_plasmid_01","L_delbrueckii_RMK202","S_thermophilus_RMK202")

PrelAbundance <-  ggplot( data = all_final,aes(y = percent_coverage, x = sample, group=interaction(species),fill = species))+ geom_bar( stat="identity")+
    labs("",
         x="",
         y="relative abundance")+
    theme_classic()+
   # scale_color_viridis(discrete=TRUE)+
   scale_fill_manual(values=all_colours_new)+
  # scale_fill_viridis(discrete=TRUE)+
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
        # axis.text.x = element_blank(),
          legend.position="right",
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank()
          )

  PrelAbundance
  svg("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.svg",width=4.5,height=3)
   # png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 1000, height = 800,res=300)

PrelAbundance

dev.off()
# 
 # svg("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.svg",width=7,height=4)
# png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 2000, height = 1000,res=300)
# 
# PrelAbundance
# 
# dev.off()


##----------------amount of Streptococci and Lactobacilli


all_final$genus <- ifelse(grepl("S_",all_final$species),"Streptococcus","Lactobacillus")

all_final %>% 
  group_by(sample,genus) %>% 
  dplyr::summarize(sum = sum(percent_coverage))  %>% 
   group_by(genus) %>% 
  dplyr::summarize(min = min(sum),
            max = max(sum))  

##----------------amount of Lactobacillus delbrueckii RMK202


all_final %>% 
  group_by(species) %>% 
  dplyr::summarize(min = min(percent_coverage),
            max = max(percent_coverage))  


##----------------coverage of Lactobacillus delbrueckii RMK202
table(all_final$species)

all_final[which(all_final$species=="L_delbrueckii_RMK202"),]

##----------------------normalise with actual mapping percent--------------------------------

library(readr)

mappings <- read_delim("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/mappings_final.txt",  "\t", escape_double = FALSE, col_names = c("sample","type","mapping"),  trim_ws = TRUE) %>% filter(type=="onlyMeta") %>% select(add=-type)
mappings$sample <- revalue(mappings$sample, c("L71"="mst4","lyo202_96"="Lyo\n1996","Lyo_202_2012"="Lyo\n2012", "Lyo_202_2014"="Lyo\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018","di_K2_6h"="cheesemaking\nday1","th_K2_8h"="cheesemaking\nday2"))

all_final_normalised <-  merge(all_final,mappings,by.x ="sample",by.y="sample" )
all_final_normalised$percent_coverage_normalised <- all_final_normalised$percent_coverage*((all_final_normalised$mapping/100))
PrelAbundance <-  ggplot( data = all_final_normalised,aes(y = percent_coverage_normalised, x = sample, group=interaction(species),fill = species))+ geom_bar( stat="identity")+
    labs("",
         x="",
         y="relative abundance")+
    theme_classic()+
  geom_hline(yintercept=100, linetype="dashed", color = "grey")+
   # scale_color_viridis(discrete=TRUE)+
   scale_fill_manual(values=all_colours_new)+
  # scale_fill_viridis(discrete=TRUE)+
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
        # axis.text.x = element_blank(),
          legend.position="right",
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank()
          )

  PrelAbundance
  svg("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance_normalised.svg",width=4.5,height=3)
   # png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 1000, height = 800,res=300)

PrelAbundance

dev.off()
# 

```

#####phage mapping again

Here I quickly look how large the ldel phage coverage is
```{bash}


##---------------------------------checking for plasmid mapping


threads=37
#samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt  ##the file with all sample names
#logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final
#Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta
what=plasmid2
short=RMK202
#genus=Lactobacillus
#mkdir -p /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}

#rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_*.txt
echo -e "Genome\tStermphage1\tStermphage2\tLdelphage1\tLdelphage2" > /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/Ldel_phage_mapping.txt
for genomes in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt)
do
echo ${genomes}
  #shortGenome=$(echo ${genomes}|cut -d '_' -f 4)
  bamqccsss=$(echo -e "/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/"${genomes}"/bwaMapping2DB/bamqc/genome_results.txt") 
  
  StermPhage1=$(grep ">>>>>>> Coverage per contig" -A 14 ${bamqccsss} |grep "Streptococcus_phage_1" |  sed "s/^[ \t]*//" |sed -r '/^\s*$/d' |awk -F "\t" '{print $4}')
  StermPhage2=$(grep ">>>>>>> Coverage per contig" -A 14 ${bamqccsss} |grep "Streptococcus_phage_2" |  sed "s/^[ \t]*//" |sed -r '/^\s*$/d' |awk -F "\t" '{print $4}')
  
  LdelPhage1=$(grep ">>>>>>> Coverage per contig" -A 14 ${bamqccsss} |grep "Lactobacillus_phage_1" |  sed "s/^[ \t]*//" |sed -r '/^\s*$/d' |awk -F "\t" '{print $4}')
  LdelPhage2=$(grep ">>>>>>> Coverage per contig" -A 14 ${bamqccsss} |grep "Lactobacillus_phage_2" |  sed "s/^[ \t]*//" |sed -r '/^\s*$/d' |awk -F "\t" '{print $4}')

echo -e ${genomes}"\t"${StermPhage1}"\t"${StermPhage2}"\t"${LdelPhage1}"\t"${LdelPhage2} >> /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/Ldel_phage_mapping.txt

  done
  

```

move local
```{bash}



scp -r  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/Ldel_phage_mapping.txt /home/vincent/Desktop/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/

```

analysis of phage mapping

```{r}
phage_mapping <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/Ldel_phage_mapping.txt", "\t", escape_double = FALSE, trim_ws = TRUE) %>% select(-Ldelphage2) %>% filter(!Genome %in% c("di_K2_6h","th_K2_8h"))


# "lyo202_96"="Lyo\n1996","Lyo_202_2012"="Lyo\n2012", "Lyo_202_2014"="Lyo\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018","di_K2_6h"="cheesemaking\nday1","th_K2_8h"="cheesemaking\nday2"
phage_mapping$Genome <- revalue(phage_mapping$Genome, c("lyo202_96"="Lyo\n1996","Lyo_202_2012"="Lyo\n2012", "Lyo_202_2014"="Lyo\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018","di_K2_6h"="cheesemaking\nday1","th_K2_8h"="cheesemaking\nday2"))


```


##Add Information

#####snpEff

Here we do [snpEff](http://snpeff.sourceforge.net/SnpEff_manual.html#buildAddConfig).


```{bash}


###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt  ##the file with all sample names
logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP
Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished.fasta
FREEBAYES_out=freebayesOuput_WithONT_Parallel_default
sampleShort=PROKKA_meta_all ##For st_thermophilus



threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt  ##the file with all sample names
logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final
Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta

##!!!!!!!!!!!!!change names of Forward and Reverse reads


##----------------
##bring PGAP LOCAL
##----------------

mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP

scp -r /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/LD_singu vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/

scp -r /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/

##=======================================================================================================================
##-----------------------------------------------all together----------------------------------------------
##=======================================================================================================================
##==================
##Building Database
##==================
#/archiv/Projects/2018_Culturomics/genomes/Lactobacillus_delbrueckii_RMK202.fna
#/archiv/Projects/2018_Culturomics/genomes/Streptococcus_thermophilus_RMK202.fna

##-------------
##01_add genome name to
##-------------

vim /home/vincent/miniconda2/share/snpeff-4.3.1t-2/snpEff.config
#S_thermophilus_mag_rmk202.genome : S_thermophilus_mag_rmk202
#L_delbrueckii_mag_rmk202.genome : L_delbrueckii_mag_rmk202
#Streptococcus_phage_rmk202.genome : Streptococcus_phage_rmk202
#PGAP_meta_all.genome : PGAP_meta_all
PROKKA_meta_all.genome : PROKKA_meta_all

##-------------
#02_put genome fasta and gff into right directory:
##-------------

#sampleShort=S_thermophilus_mag_rmk202
#sampleShort=CP046134 ##For st_thermophilus
#sampleShort=PROKKA_meta_all/
sampleShort=PGAP_meta_all

#/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_ldel/final/L_delbrueckii_mag_rmk202.fasta 

##-------------
#03_put genome fasta 
##-------------
rm -r /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}
mkdir -p /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}

cat ${Assembly} > /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}/sequences.fa



##-------------
#04 genes gff 
##-------------

#grep "^#" -v /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/S_thermophilus_RMK202.current.gff|grep -e "${sampleShort}" |cut -f 1|sort|uniq -c

#cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/S_thermophilus_RMK202.current.gff | grep -e "^${sampleShort}" |awk -F "\t" '{OFS"\t"}{if($3=="CDS") print $0}' > /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}//genes.gff


##-------------
#04 all genes gff 
##-------------

##already preped in RAST chunk
 #cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/annotation_eggnog_rast/assembly_20191129/merged/all_genes_rmk202.gff |sed 's/gene/CDS/g' > /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}//genes.gff

 awk -F "\t" '{OFS="\t"}{if( $3=="CDS")print $0}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA_04012020.gff|sed 's/^202-SMAG-1/CP046134/g'  > /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}//genes.gff

 awk -F "\t" '{OFS="\t"}{if( $3=="CDS")print $0}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Ldel//202-LMAG/PROKKA_04012020.gff|sed 's/^202-LMAG-1/CP046131/g'  >> /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}//genes.gff

##-------------
##05_build database
##-------------

cd /home/vincent/miniconda2/share/snpeff-4.3.1t-2/
java -jar snpEff.jar build -gff3 -v ${sampleShort}


##==================
##running snpEff
##==================

##-------------
##06_preperation of SNV file
##-------------

#grep -e "${sampleShort}" ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf > ${BaseLocation}/${FREEBAYES_out}/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_all.vcf

##-------------
##07 SNveff calling
##-------------

cd /home/vincent/miniconda2/share/snpeff-4.3.1t-2/
#rm -r ${sampleShort} ${BaseLocation}/${FREEBAYES_out}/SnpEff/${sampleShort}/
mkdir -p ${sampleShort} ${BaseLocation}/${FREEBAYES_out}/SnpEff/${sampleShort}/


#java -Xmx64g -jar /home/vincent/miniconda2/share/snpeff-4.3.1t-2/snpEff.jar ${sampleShort} /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/freebayesOuput_WithONT_Parallel_default/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf > ${BaseLocation}/${FREEBAYES_out}/SnpEff/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_snpeff.vcf

grep "phage" -v /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/freebayesOuput_WithONT_Parallel_default/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode.vcf > /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/freebayesOuput_WithONT_Parallel_default/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode_wophage.vcf


head -1 /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/freebayesOuput_WithONT_Parallel_default/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_prepR.vcf|sed 's/\t/\n/g'|sed '1,9d' > /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/freebayesOuput_WithONT_Parallel_default/samples_names.txt


java -Xmx64g -jar /home/vincent/miniconda2/share/snpeff-4.3.1t-2/snpEff.jar ${sampleShort} \
/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/freebayesOuput_WithONT_Parallel_default/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL.recode_wophage.vcf > /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/freebayesOuput_WithONT_Parallel_default//variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_snpeff.vcf


##-------------
##08_prep for R
##-------------

grep "^#" -v /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/freebayesOuput_WithONT_Parallel_default//variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_snpeff.vcf  | \
  awk -F "\t" 'BEGIN{OFS="\t"} {print $1,$2,$3,$4,$5,$6,$7,$8}' > /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/freebayesOuput_WithONT_Parallel_default//variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_snpeff_prepforR.vcf


##==================
##transfer local 
##==================

threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt  ##the file with all sample names
logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP
Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished.fasta
FREEBAYES_out=freebayesOuput_WithONT_Parallel_default



##-------------
###snpEff
##-------------

scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/freebayesOuput_WithONT_Parallel_default//variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_snpeff_prepforR.vcf ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new/

##-------------
###eggnog
##-------------

#scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/eggnog/${sampleShort}_pgap_query_seqs.fa.emapper.annotations  ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new/variantCallsfreebayes_all_${sampleShort}_eggnog.vcf 



```

here we merge all annotation knowledge we have of the SNPs:

1. SNPeffect
2. eggnog annotation
3. repeat annotation
4. core genes

Thereafter we have a long list of information for every SNP. 

#####repeat genes

We could potentially have a problem of alignement in genes highly similiar. Therefore I look if which genes are cluster together. 

```{bash}



###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt  ##the file with all sample names
logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP
Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/di_rmk202_MAG_reference_polished.fasta



BaseLocation_Ldel=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/ldel/
BaseFILNAME=L_delbrueckii_RMK202


BaseLocation_Sterm=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/
BaseFILNAME_sterm=S_thermophilus_RMK202


###================
##script
###================


##==================
##merge all fna files
##==================


rm -r ${BaseLocation}/Repeats/
mkdir -p ${BaseLocation}/Repeats/

cat ${BaseLocation_Ldel}/${BaseFILNAME}_genes.ffn \
  ${BaseLocation_Sterm}/${BaseFILNAME_sterm}_genes.ffn > \
 ${BaseLocation}/Repeats/both_genes.ffn

cat ${BaseLocation_Ldel}/${BaseFILNAME}.current.gff \
  ${BaseLocation_Sterm}/${BaseFILNAME_sterm}.current.gff > \
 ${BaseLocation}/Repeats/both_genes.ffn
 
##==================
##cd-hit
##==================

cd-hit-est -i ${BaseLocation}/Repeats/both_genes.ffn \
    -o ${BaseLocation}/Repeats/both__cdsClustering.txt -c 0.90 -M 30000 -T 37 -sc 1 -sf 1

  # cd-hit -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pgap_cds.fna \
  #  -o /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/sterm_cdsClustering.txt -c 0.85 -M 30000 -T 37 -sc 1 -sf 1
  






  ###multifasta2fasta
  mkdir -p ${BaseLocation}/Repeats/splitClusters
  cd ${BaseLocation}/Repeats/splitClusters
  awk 'NR%1==0{ print > "Cluster_"++i".txt" }' RS='>Cl' ${BaseLocation}/Repeats/both__cdsClustering.txt.clstr

  ##prepare log file
  #cluster_2.txt till cluster_24.txt
  
  ls ${BaseLocation}/Repeats/splitClusters |sort -V |head -84 |tail -83 > ${BaseLocation}/Repeats/splitClusters/files_of_INterest.txt
     
   ##get gene names
   num=2
   rm ${BaseLocation}/Repeats/bigCluster/all_repeatingGenes.txt
   mkdir -p ${BaseLocation}/Repeats/bigCluster/
   i=Cluster_2.txt
   for i in $(cat ${BaseLocation}/Repeats/splitClusters/files_of_INterest.txt)
    do
   grep -v "^uster" ${BaseLocation}/Repeats/splitClusters/${i} |awk -F ">" '{print $2}' |awk -F "... at " 'BEGIN{OFS="\t"}{print $1,$2}' |sed 's/... \*//g' |sed '$ d' > \
    ${BaseLocation}/Repeats/bigCluster/names_${i}
    
   grep -v "^uster" ${BaseLocation}/Repeats/splitClusters/${i} |awk -F ">" '{print $2}'  |awk -F "... at " -v count="$num" 'BEGIN{OFS="\t"}{print $1,$2,count}' |sed 's/... \*//g' |sed '$ d' >> \
   ${BaseLocation}/Repeats/bigCluster/all_repeatingGenes.txt
      
    num=$((num+1))
    
    
    done
 

    ##get gff
    
  # genes=pgaptmp_001292
    mkdir -p  ${BaseLocation}/Repeats//bigCluster_genes/
    for i in $(cat ${BaseLocation}/Repeats/splitClusters/files_of_INterest.txt)
    do
    rm  ${BaseLocation}/Repeats//bigCluster_genes//genes_${i}
    for genes in $(cut -f 1 ${BaseLocation}/Repeats/bigCluster//names_${i})
      do
        
        grep "${genes}.p" ${BaseLocation}/Repeats/both_genes.ffn >> \
          ${BaseLocation}/Repeats//bigCluster_genes/genes_${i}
       
        
      done # $genes   
    done   # $i
    
    
    
  mkdir -p ~/Desktop/Projects/2019_Pilotplan/09_annoation/repeatingGenes


scp vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/Repeats/bigCluster/all_repeatingGenes.txt ~/Desktop/Projects/2019_Pilotplan/09_annoation/repeatingGenes/

```


#####Core genes prep

```{bash}

rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coreGenes_rmk202_both.txt

##=======================================================================================================================
##-----------------------------------------------Streptococcus thermophilus----------------------------------------------
##=======================================================================================================================
species=Sterm

##-------extract OG names
#rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/Orthogroups/test.txt
rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01//Orthogroups/OG_names_${species}.txt
for OG in $(cat /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01//Orthogroups/Orthogroups_SingleCopyOrthologues.txt)
do

grep "$OG" /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01//Orthogroups/Orthogroups.txt | sed 's/gnl|extdb|//g'| awk -F " " 'BEGIN{OFS="\t"}{print $1,"S_thermophilus_RMK202",$2}'  >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01/Orthogroups/OG_names_${species}.txt
#grep "$OG" /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01//Orthogroups/Orthogroups.txt | sed 's/gnl|extdb|//g'| awk -F " " 'BEGIN{OFS="\t"}{print $1,$2}'  >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01/Orthogroups/OG_names_${species}.txt


done #OG

cat /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01//Orthogroups/OG_names_${species}.txt >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coreGenes_rmk202_both.txt


##=======================================================================================================================
##-----------------------------------------------Lactobacillus delbrueckii----------------------------------------------
##=======================================================================================================================
species=Ldel

##-------extract OG names
#rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/Orthogroups/test.txt
rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01//Orthogroups/OG_names_${species}.txt
for OG in $(cat /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01//Orthogroups/Orthogroups_SingleCopyOrthologues.txt)
do

grep "$OG" /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01//Orthogroups/Orthogroups.txt| sed 's/gnl|extdb|//g' |awk -F " " 'BEGIN{OFS="\t"}{print $1,"	L_delbrueckii_RMK202",$11}'  >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01//Orthogroups/OG_names_${species}.txt
#grep "$OG" /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01//Orthogroups/Orthogroups.txt| sed 's/gnl|extdb|//g' |awk -F " " 'BEGIN{OFS="\t"}{print $1,$2}'  >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01//Orthogroups/OG_names_${species}.txt


done #OG

cat /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Dec01//Orthogroups/OG_names_${species}.txt >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coreGenes_rmk202_both.txt



##=======================================================================================================================
##-----------------------------------------------Lactobacillus delbrueckii----------------------------------------------
##=======================================================================================================================

mkdir -p ~/Desktop/Projects/2019_Pilotplan/09_annoation/coreGenes

scp vincent@130.223.51.116:/archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coreGenes_rmk202_both.txt  ~/Desktop/Projects/2019_Pilotplan/09_annoation/coreGenes/coreGenes_rmk202_both.txt


```

#####merge all infomration

```{r}
library(readr)
library(stringr)

##==================
##01-SNPeff
##==================
RMK202_both_snpEff <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_snpeff_prepforR.vcf",  "\t", escape_double = FALSE, col_names = c("X.CHROM","POS","unknown","REF","ALT","quality","uknown2","snpeff"), trim_ws = TRUE)

RMK202_both_snpEff_02 <- subset(RMK202_both_snpEff, select=-c(unknown,uknown2))
RMK202_both_snpEff_02$site <- paste(RMK202_both_snpEff_02$X.CHROM,RMK202_both_snpEff_02$POS,RMK202_both_snpEff_02$REF,RMK202_both_snpEff_02$ALT,sep="_")

tmp_snpeff <-  data.frame(str_split_fixed(RMK202_both_snpEff_02$snpeff, fixed("|"), 136)[,2:8])
colnames(tmp_snpeff) <- c("effect","significance","geneName","PGAP_name","type","typeName","typeGene")

RMK202_both_snpEff_02 <- subset(RMK202_both_snpEff_02, select=-snpeff)
RMK202_both_SNPeff_final <-  cbind(RMK202_both_snpEff_02,tmp_snpeff)
# table(S_thermophilus_snpEff_03$typeGene)
# 
# S_thermophilus_snpEff_03[which(S_thermophilus_snpEff_03$typeGene=="pseudogene"),]
# table(S_thermophilus_snpEff_03[which(S_thermophilus_snpEff_03$typeGene=="protein_coding"),]$type)
# S_thermophilus_snpEff_03[which(S_thermophilus_snpEff_03$typeGene=="rRNA"),]

RMK202_both_SNPeff_final$finalName <- RMK202_both_SNPeff_final$typeName %>% gsub("rna-","",.) %>%  gsub("TRANSCRIPT_gene-","",.) %>%  gsub("gene-","",.) 
RMK202_both_SNPeff_final$finalName <- paste(RMK202_both_SNPeff_final$X.CHROM,RMK202_both_SNPeff_final$finalName,sep = "_")
# length(grep("pgaptmp_",S_thermophilus_snpEff_03$finalName,invert = TRUE))
table(RMK202_both_SNPeff_final$X.CHROM)
#table(RMK202_both_SNPeff_final$significance)
##==================
##02 eggnog
##==================
RMK202_both_eggnog <- read_delim("/home/vincent/Desktop/Projects/2019_Pilotplan/assembly_20191129/merged/merged_all_eggnog.annotations",  "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description"))
# RMK202_both_eggnog[RMK202_both_eggnog$query_name %in% SNPeff_final$finalName,]

tmp_pep <- RMK202_both_eggnog[grep("pep",RMK202_both_eggnog$Preferred_name),]

##==================
##03 repeats
##==================

RMK202_both_repeats <- read_delim("~/Desktop/Projects/2019_Pilotplan/09_annoation/repeatingGenes/all_repeatingGenes.txt",  "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("geneNAME","Repeat_identity","Repeat_cluster")) #%>% slice( 1:(n()-3))
# nrow(RMK202_both_repeats)

##==================
##04_core genes
##==================
library(tidyverse)
RMK202_both_coreGenes <- read_delim("~/Desktop/Projects/2019_Pilotplan/09_annoation/coreGenes/coreGenes_rmk202_both.txt",  "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("orthogroup","species","geneName")) %>% add_column(.,core="core")

##==================
##05_merge
##==================
nrow(RMK202_both_SNPeff_final)

RMK202_snpeff_eggnog <- merge(RMK202_both_SNPeff_final, RMK202_both_eggnog, by.x = "geneName", by.y = "query_name",all.x = TRUE)
RMK202_snpeff_eggnog_repeats <- merge(RMK202_snpeff_eggnog, RMK202_both_repeats, by.x = "geneName", by.y = "geneNAME",all.x = TRUE)
RMK202_snpeff_eggnog_repeats_core <- merge(RMK202_snpeff_eggnog_repeats, RMK202_both_coreGenes, by.x = "geneName", by.y = "geneName",all.x = TRUE)
nrow(RMK202_snpeff_eggnog_repeats_core)

table(RMK202_snpeff_eggnog_repeats_core$core)
table(RMK202_snpeff_eggnog_repeats_core$X.CHROM)
write.csv(RMK202_snpeff_eggnog_repeats_core,"~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new/RMK202_snpeff_eggnog_repeats_core.txt")

RMK202_snpeff_eggnog_repeats_core[grep("S_thermophilus",RMK202_snpeff_eggnog_repeats_core$finalName),]

```

## analysis snps

The coverage of the alternative allel must be at least 3 or zero. 

We have to be careful when removing low quality SNP calls. As they seem to be not not low quality in the sense of low coverage but rather suported by only a fraction of samples. This makes sense for many snps which are only valied for Sterm and have NAs in the Ldel genomes. Therefore we rather select the SNP by selecting for a minimum of coverage supporting the SNP per sample. 

!!IMPORTANT!!
Before starting we have to add the following to the SNP.vcf:

1. SNPeff
2. if core or not with roary and interesect
3. (if repeat within repeat)

###### first Analysis

```{r}
library(tidyverse)
library(vcfR)
library(readr)
library(plyr)
library(dplyr)
library(ggplot2)
library(xlsx)

##==================
##01_import data
##==================
#snps_freebayes <- read_delim("/home/vincent/Desktop/Projects/2019_RMK202_analysis/06_snpCalling2ONT/variantCallsfreebayes_all_f_005_min_20x_prepforR.vcf", "\t", escape_double = FALSE, trim_ws = TRUE)
snps_freebayes <- read_delim("/home/vincent/Desktop/Projects/2019_RMK202_analysis/06_snpCalling2ONT_new/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_prepR_new.vcf", "\t", escape_double = FALSE, trim_ws = TRUE)


table(snps_freebayes$`#CHROM`)

snps_freebayes$MQM <- str_split_fixed(snps_freebayes$INFO, ";", 130)[,16] %>% gsub("MQM=","",.) 
snps_freebayes$MQM <- str_split_fixed(snps_freebayes$MQM, ",", 3)[,1] %>%  as.numeric()


# colnames(snps_freebayes)
# snps_freebayes[snps_freebayes$QUAL<=20,]
# snps_freebayes <- snps_freebayes[snps_freebayes$QUAL>20,]
# dim(snps_freebayes_new)
# dim(snps_freebayes)
# snps_freebayes <- subset(snps_freebayes, select=-c(di_K1_10h,th_K1_8h))


##==================
##01_filter SNVs MQM larger than 30 
##INFO=<ID=MQM,Number=A,Type=Float,Description="Mean mapping quality of observed alternate alleles">
##==================
tmp_remove <- snps_freebayes[which(snps_freebayes$MQM<30),]
table(tmp_remove$`#CHROM`)
snps_freebayes <- snps_freebayes[which(snps_freebayes$MQM>30),] %>% select(-MQM)
table(snps_freebayes$`#CHROM`)


##==================
##01_filter SNVs MQM larger than 30 
##INFO=<ID=MQM,Number=A,Type=Float,Description="Mean mapping quality of observed alternate alleles">
##==================

##==================
##02_long list and frequency calc
##==================

table(snps_freebayes$`#CHROM`)
##wide2long

 # snps_freebayes[,10]
snps_freebayes_long <- gather(snps_freebayes, sample, Snps, colnames(snps_freebayes)[10]:colnames(snps_freebayes)[ncol(snps_freebayes)], factor_key=TRUE,na.rm = TRUE) 
nrow(snps_freebayes_long)
table(snps_freebayes_long$sample)
###------------------------remove non-called snps
snps_freebayes_long_cleaned <- snps_freebayes_long[snps_freebayes_long$Snps!=".",]
nrow(snps_freebayes_long_cleaned)
# snps_freebayes_long[c(7438,7437,7439, 7440, 7441, 7442, 7443, 7444, 7445),]

###------------------------calculate allel frequency   

sample_relative_temp <-snps_freebayes_long_cleaned %>% separate("Snps", into=c("GT","DP","AD","RO","QR","AO","QA","GL"), sep = ":") %>% transform(.,allel_frequency=as.numeric(AO)/as.numeric(DP),QualityScore=as.numeric(QA)/as.numeric(DP))



##==================
##03_filter QA/DP > 30
##INFO=<ID=QA,Number=A,Type=Integer,Description="Alternate allele quality sum in phred">
##INFO=<ID=DP,Number=1,Type=Integer,Description="Total read depth at the locus"
##==================
sample_relative_temp_filter <- sample_relative_temp[which(!is.na(sample_relative_temp$QualityScore)),]

# tmp <- sample_relative_temp_filter[which(sample_relative_temp_filter$sample=="di_K2_6h"),]
# nrow(tmp)
# table(tmp$sample)
# tmp$QualityScore <- as.numeric(tmp$QualityScore)
# sum(tmp$QualityScore>5)
# hist(tmp$QualityScore)
# hist(sample_relative_temp_filter$QualityScore)

sample_relative_temp_withONT <-sample_relative_temp_filter[which(sample_relative_temp_filter$QualityScore>5),]
table(sample_relative_temp_withONT$sample)
# sample_relative_temp_withONT <- sample_relative_temp
sample_relative_temp_withONT$site <- paste(sample_relative_temp_withONT$X.CHROM,sample_relative_temp_withONT$POS,sample_relative_temp_withONT$REF,sample_relative_temp_withONT$ALT,sep="_")

sample_relative_temp <- sample_relative_temp_withONT
# sample_relative_temp <- sample_relative_temp_filter ### without QS>30

##==================
##03_filter
##==================

##remove SNPs that have more than two allel frequencies
# nrow(sample_relative_temp)
# nas <- sample_relative_temp[which(is.na(sample_relative_temp$allel_frequency)),]
# nrow(nas)
sample_relative_temp_cleaned <- sample_relative_temp[which(!is.na(sample_relative_temp$allel_frequency)),]
# nrow(sample_relative_temp_cleaned)
##remove low coverage (>30x)
cutoff <- 30
sample_relative_temp_cleaned$DP <- as.numeric(sample_relative_temp_cleaned$DP)
nrow(sample_relative_temp_cleaned)
sample_relative_safe <- sample_relative_temp_cleaned[which(sample_relative_temp_cleaned$DP>cutoff),]
nrow(sample_relative_safe)

##make all allele frequencies smaller than 0.05 to NA
# sample_relative_safe[sample_relative_safe$allel_frequency<0.05,"allel_frequency"] <- NA

#is.na(sample_relative_safe$allel_frequency)

##remove snps with less than 2x alternative allel frequency or zero
sample_relative_safe$AO <- as.numeric(sample_relative_safe$AO)
# nrow(sample_relative_safe[sample_relative_safe$AO<=2,])
# nrow(sample_relative_safe[sample_relative_safe$AO==0 | sample_relative_safe$AO>2,])
nrow(sample_relative_safe[sample_relative_safe$AO>0 & sample_relative_safe$AO<=2,])
sample_relative_safe <- (sample_relative_safe[sample_relative_safe$AO==0 | sample_relative_safe$AO>2,])
nrow(sample_relative_safe)


##==================
##04_preperation and make wide
##==================

sample_relative_temp_cleaned <- sample_relative_safe
#ggplot(sample_relative_temp_cleaned, aes(y=DP))+geom_boxplot()+ylim(c(0,1000))


##make site name
sample_relative_temp_cleaned$site <- paste(sample_relative_temp_cleaned$X.CHROM,sample_relative_temp_cleaned$POS,sample_relative_temp_cleaned$REF,sample_relative_temp_cleaned$ALT,sep="_")


###spread the dataframe again

nrow(sample_relative_temp_cleaned)

sample_relative_safe_final_prep <- sample_relative_temp_cleaned[,c("X.CHROM","POS","site","allel_frequency","sample")]
#nrow(sample_relative_safe_final_prep[grep("S_thermophilus_RMK202",sample_relative_safe_final_prep$site),])
sample_relative_safe_final_tsne <- spread(sample_relative_safe_final_prep, sample, allel_frequency)  %>%replace(is.na(.), 0)
nrow(sample_relative_safe_final_tsne)
table(sample_relative_safe_final_prep$sample)
##==================
##05_filter rowsums!=0
##==================

##remove row sums ==0
nrow(sample_relative_safe_final_tsne)
sum(rowSums(sample_relative_safe_final_tsne[,4:ncol(sample_relative_safe_final_tsne)],na.rm = TRUE)==0)


sample_relative_wide <- sample_relative_safe_final_tsne[rowSums(sample_relative_safe_final_tsne[,4:ncol(sample_relative_safe_final_tsne)],na.rm = TRUE)>0,]
table(sample_relative_wide$X.CHROM)
sample_relative_wide_phagesONly <- sample_relative_wide[grep("phage",sample_relative_wide$X.CHROM),]
# table(sample_relative_wide_phagesONly$X.CHROM)
# nrow(sample_relative_wide)
sample_relative_wide <- subset(sample_relative_wide, select=-POS)
sample_relative_wide$species <- sample_relative_wide$X.CHROM
sample_relative_wide$species <- revalue(sample_relative_wide$species, c("CP046131"="L. delbrueckii","CP046134"="S. thermophilus"))
sample_relative_wide$culture <- "RMK202"
nrow(sample_relative_wide)
##==================
##06_add gene information
##==================

RMK202_snpeff_eggnog_repeats_core <- read_csv("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new/RMK202_snpeff_eggnog_repeats_core.txt")
# nrow(RMK202_snpeff_eggnog_repeats_core)
# nrow(sample_relative_wide)
#RMK202_snpeff_eggnog_repeats_core_subset <- subset(RMK202_snpeff_eggnog_repeats_core, select=c(X.CHROM,site,finalName,effect,significance,geneName,GOs,EC,KEGG_ko,COG_Functional_Category,Repeat_identity,core))

sample_relative_wide_description <- merge(sample_relative_wide,RMK202_snpeff_eggnog_repeats_core, by = "site",all.x = TRUE)
# nrow(sample_relative_wide_description)

#---------------------------------subset for interesting columns
colnames(sample_relative_wide_description)
sample_relative_wide_description_interest <- subset(sample_relative_wide_description, select=c("X.CHROM.x","species.x","site","culture","finalName","Preferred_name","effect","significance","geneName","COG_Functional_Category","Repeat_cluster","core","L104","L108","L35","L44","L70","L71","L80","L99","S50","S72","Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202","lyo202_96","di_K2_6h","th_K2_8h","G1_6_18","G2_6_18","G3_6_18","G4_6_18","G5_6_18","12107","24776",
"24778",
"24779",
"24780",
"24781",
"24782",
"24783",
"24798",
"24777",
"13491",
"13492",
"13493",
"13494",
"13495",
"13496",
"13497",
"13498",
"13500",
"24737",
"24738",
"24739",
"24740",
"13499",
"24853",
"24854",
"24855"))

#"G4_6_18"

colnames(sample_relative_wide_description_interest)
nrow(sample_relative_wide_description_interest)
##==================
##07 rename samples
##==================

# colnames(sample_relative_wide_description_interest) <- revalue(colnames(sample_relative_wide_description_interest), c("species.x"="species","S50"="mst1","S72"="mst2","L70"="mst3", "L44"="mst10", "L108"="mst7", "L99"="mst8", "L104"="mst6", "L80"="mst5", "L35"="mst9", "L71"="mst4","lyo202_96"="Lyo\n1996","Lyo_202_2012"="Lyo\n2012", "Lyo_202_2014"="Lyo\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018","di_K2_6h"="cheesemaking\nday1","th_K2_8h"="cheesemaking\nday2"))

colnames(sample_relative_wide_description_interest) <- revalue(colnames(sample_relative_wide_description_interest), c("species.x"="species","L70"="mst3", "L44"="mst10", "L108"="mst7", "L99"="mst8", "L104"="mst6", "L80"="mst5", "L35"="mst9", "L71"="mst4","lyo202_96"="Lyo\n1996","Lyo_202_2012"="Lyo\n2012", "Lyo_202_2014"="Lyo\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018","di_K2_6h"="cheesemaking\nday1","th_K2_8h"="cheesemaking\nday2","G1_6_18"="experiment_A","G2_6_18"="experiment_B","G3_6_18"="experiment_C","G4_6_18"="experiment_D","G5_6_18"="experiment_E"))


##==================
##08_remove non-bacterial SNPs
##==================
table(sample_relative_wide_description_interest$X.CHROM.x)
nrow(sample_relative_wide_description_interest)
# sample_relative_wide_description_interest_subset <- sample_relative_wide_description_interest[which(sample_relative_wide_description_interest$X.CHROM.x %in% c("L_delbrueckii_RMK202","S_thermophilus_RMK202")),]
sample_relative_wide_description_interest_subset <- sample_relative_wide_description_interest[which(sample_relative_wide_description_interest$X.CHROM.x %in% c("CP046131","CP046134")),]
nrow(sample_relative_wide_description_interest_subset)
table(sample_relative_wide_description_interest_subset$X.CHROM.x)
sample_relative_wide <- sample_relative_wide_description_interest_subset


##==================
##09_order variables properly
##==================


sample_relative_wide$significance <- factor(sample_relative_wide$significance, levels=c("MODIFIER","LOW","MODERATE","HIGH"))


##==================
##10_dN_dS ratio
##==================
###-----------prep for 
prep_for_dN_dS_wide <- sample_relative_wide[which(sample_relative_wide$significance!="MODIFIER"),] 

###-----------calculate dS/dN ratio

dN_dS_ratios <- data.frame()
for (gene in unique(prep_for_dN_dS_wide$finalName)) {
  
  dS <- prep_for_dN_dS_wide %>% filter(finalName==gene) %>% filter(significance=="LOW")
   dN <-  prep_for_dN_dS_wide %>% filter(finalName==gene) %>% filter(significance %in% c("MODERATE","HIGH"))
tmp <- data.frame(gene, "dS"=nrow(dS),"dN"=nrow(dN),"dN_dN+dS"=(nrow(dN))/(nrow(dS)+nrow(dN)),"dN_dS"=(nrow(dN))/(nrow(dS)))
dN_dS_ratios <- rbind(dN_dS_ratios,tmp)
}

plotDens <- ggplot(dN_dS_ratios,aes(x=dN_dS))+geom_histogram()+
  geom_vline(xintercept=mean(dN_dS_ratios$dN_dS))+
  theme_classic()+
  labs(x="dN/(dN+dS)",
       y="gene count")

plotDens
dN_dS_ratios <- arrange(dN_dS_ratios,desc(dN_dS)) 

##-------------------------------prep snpeff
annotation_prep <- RMK202_snpeff_eggnog_repeats_core %>% select(-c(X1,X.CHROM,species,POS,REF,ALT,quality,site,effect,significance)) %>% distinct()
dN_dS_ratios_final <- merge(dN_dS_ratios,annotation_prep,by.x="gene",by.y="finalName",all.x = TRUE)
dN_dS_ratios_final_02 <- arrange(dN_dS_ratios_final,desc(dN_dS)) 

##-------------------------------wirte to file

# write.xlsx(dN_dS_ratios_final_02,file="Users//Desktop/presenations/my_presentations/20191007_groupmeeting/figures/dN_dS_ratios_genes_rmk202.xls", sheetName = "Sheet1", 
  # col.names = TRUE, row.names = FALSE, append = FALSE)

##-------------------------------merge with sample_relative_wide
dN_dS_ratios$mutations <- dN_dS_ratios$dS+dN_dS_ratios$dN
dN_dS_ratios_prep <- dN_dS_ratios %>% select(gene,dN_dN.dS,mutations)
sample_relative_wide <- merge(sample_relative_wide,dN_dS_ratios_prep,by.y="gene",by.x="finalName",all.x = TRUE)
# colnames(sample_relative_wide)
# RMK202_snpeff_eggnog_repeats_core_uniuqes <- sample_relative_wide %>%select(-c("X.CHROM.x","POS","REF","ALT","quality","site","effect","significance","best_tax_level")) %>% distinct()
##==================
##11_make long
##==================
colnames(sample_relative_wide)
sample_relative_long <- gather(sample_relative_wide, sample, Snps, "mst6":"24855", factor_key=TRUE,na.rm = TRUE) 
nrow(sample_relative_long)
table(sample_relative_long$X.CHROM)


##==================
##12_make a unique gene infomoration
##==================

RMK202_snpeff_eggnog_repeats_core <- read_csv("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new/RMK202_snpeff_eggnog_repeats_core.txt")
RMK202_snpeff_eggnog_repeats_core_uniuqes <- RMK202_snpeff_eggnog_repeats_core %>% select(-c("X1","X.CHROM","POS","REF","ALT","quality","site","effect","significance","best_tax_level")) %>% distinct()
RMK202_snpeff_eggnog_repeats_core_uniuqes_final <- merge(RMK202_snpeff_eggnog_repeats_core_uniuqes,dN_dS_ratios_prep,by.y="gene",by.x="finalName",all.x = TRUE)  

write.table(RMK202_snpeff_eggnog_repeats_core_uniuqes_final,quote=FALSE,sep="\t",file="~/Desktop/presenations/my_presentations/20191007_groupmeeting/figures/RMK202_snpeff_eggnog_repeats_core_dNdS_uniqueGenes.txt")

colnames(RMK202_snpeff_eggnog_repeats_core_uniuqes_final)
RMK202_snpeff_eggnog_repeats_core_uniuqes_final_reduce <- RMK202_snpeff_eggnog_repeats_core %>% select(-c("finalName","Preferred_name","EC","KEGG_ko","KEGG_Reaction","CAZy","BiGG_Reaction","COG_Functional_Category","eggNOG free text description")) %>% distinct()


###------------------------------------------
#wide only metagenomes
###------------------------------------------

table(sample_relative_long$sample)

sample_relative_long_meta <- sample_relative_long %>% filter(sample %in% c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2","experiment_A","experiment_B","experiment_C","experiment_D","experiment_E"))


sample_relative_wide_meta <- spread(sample_relative_long_meta, sample, Snps) %>%  filter(core=="core")#%>%replace(is.na(.), 0) 

```

######rarefaction

```{r}
library(vegan)


##-----------------------------------------------------------------------own samples------------------------
library(tidyverse) 
library(lubridate)

sample_relative_safe_final_prep_rarefy <- sample_relative_temp_cleaned[,c("X.CHROM","POS","site","RO","sample")]
#nrow(sample_relative_safe_final_prep[grep("S_thermophilus_RMK202",sample_relative_safe_final_prep$site),])
sample_relative_safe_final_prep_rarefy_wide <- spread(sample_relative_safe_final_prep_rarefy, sample, RO) %>%replace(is.na(.), 0) %>% select(site,Konserve_202,RMK202,Versand_202,lyo202_96,di_K2_6h,th_K2_8h)

row.names(sample_relative_safe_final_prep_rarefy_wide) <- sample_relative_safe_final_prep_rarefy_wide$site
sample_relative_safe_final_prep_rarefy_wide <- sample_relative_safe_final_prep_rarefy_wide %>% select(-site) #%>% as.matrix() %>% t()
View(sample_relative_safe_final_prep_rarefy_wide_new)

sample_relative_safe_final_prep_rarefy_wide_new <- sample_relative_safe_final_prep_rarefy_wide %>%
    mutate_all(type.convert)  %>% as.matrix() %>% t()
    # mutate_if(as.character,as.numeric)

S <- specnumber(sample_relative_safe_final_prep_rarefy_wide_new) # observed number of species
(raremax <- min(rowSums(sample_relative_safe_final_prep_rarefy_wide_new)))

Srare <- rarefy(sample_relative_safe_final_prep_rarefy_wide_new,961678)
plot(S, Srare, xlab = "Observed No. of Species", ylab = "Rarefied No. of Species")
abline(0, 1)

 rarecurve(sample_relative_safe_final_prep_rarefy_wide_new, step = 100000, sample = raremax, col = "blue", cex = 0.6,xlab="Sampling size","number of SNPs")

# svg("~/Desktop/Manuscripts/2019_RMK202/Figures/rarefaction.svg",width=6,height=5)
# # png("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_sterm.png",width=1800,height=1600,res=300)
# 
# p1 <- rarecurve(sample_relative_safe_final_prep_rarefy_wide_new, step = 1000, sample = raremax, col = "blue", cex = 0.6,xlab="Sampling size","number of SNPs")
# 
# dev.off()




```


##### dN/dS ratio plots

```{r}

all_interaction_functioning_Genes_cleaned <- read_table2("~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/For_dN_dS/all_interaction_functioning_Genes_cleaned.txt",  col_names = c("name","gene"))

dN_dS_ratios <- merge(dN_dS_ratios,all_interaction_functioning_Genes_cleaned,by="gene",all.x=TRUE)

dN_dS_ratios$functionalll <- ifelse(is.na(dN_dS_ratios$name),"non-functional","functional")
table(dN_dS_ratios$name)

dN_dS_ratios %>% filter(functional=="functional")
##-------------------------------plotting

png("~/Desktop/presenations/my_presentations/20191007_groupmeeting/figures/dN_dS_ratios_genes_density_rmk202.png",width=1300,height=1000,res=300)
plotDens
dev.off()
# prep_for_dN_dS_wide %>% filter(geneName=="pgaptmp_000898")
##------------
##prepare 
##------------
threshold_mutation <- 1

dN_dS_ratios$category <- ifelse((dN_dS_ratios$dN_dN.dS>mean(dN_dS_ratios$dN_dN.dS) &dN_dS_ratios$dN>threshold_mutation),"used","not used") 
table(dN_dS_ratios$category)
dN_dS_ratios$category
dN_dS_ratios$category = factor(dN_dS_ratios$category, levels=c("used","not used"))
colors <- c("red","grey")

##------------
##plot dN/dS to mutation number
##------------
plotpoint <- ggplot(dN_dS_ratios,aes(x=dN_dN.dS,y=dN,group=category,color=category,fill=category))+geom_point()+
  geom_vline(xintercept=mean(dN_dS_ratios$dN_dN.dS))+
  geom_hline(yintercept=threshold_mutation)+
  scale_fill_manual(values=colors)+
  scale_color_manual(values=colors)+
  theme_classic()+
  annotate(geom="text", x=0.9, y=0.9*(max(dN_dS_ratios$dN)), label=paste0("used:",sum(dN_dS_ratios$category=="used")), color="red")+
  annotate(geom="text", x=0.1, y=0.9*(max(dN_dS_ratios$dN)), label=paste0("not used:",sum(dN_dS_ratios$category=="not used")), color="grey")+
  theme(legend.title = element_blank())+
  # coord_trans( y="log2")+
  labs(x="dN/(dN+dS)",
       y="dN")
plotpoint


plotDens_x <- ggplot(dN_dS_ratios,aes(x=dN_dN.dS,group=category,fill=category))+geom_histogram()+
  geom_vline(xintercept=mean(dN_dS_ratios$dN_dN.dS))+
  scale_fill_manual(values=colors)+
  # scale_color_manual(values=colors)+
  theme_classic()+
  theme(legend.position="none")+
  labs(x="",
       y="gene count")
plotDens_x

plotDens_y <- ggplot(dN_dS_ratios,aes(x=dN,group=category,fill=category))+geom_histogram()+
  geom_vline(xintercept=threshold_mutation)+
  scale_fill_manual(values=colors)+
  coord_flip()+
  # scale_color_manual(values=colors)+
  theme_classic()+
  theme(legend.position="none")+
  labs(x="",
       y="gene count")
plotDens_y



##-------put together
source("/home/vincent/Desktop/ScriptRepository/R_functions-master/R_functions/g_legend.R")

 legend <- g_legend(plotpoint)
# blank <- grid.rect(gp=gpar(col="white"))


grid.arrange(plotDens_x,legend,plotpoint+theme(legend.position="none"),plotDens_y,
        ncol=2, nrow=2, widths=c(4, 2), heights=c(2,4))

svg("~/Desktop/Manuscripts/2019_RMK202/Figures/dN_dS_ratio.svg",width=6,height=5)
# png("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_sterm.png",width=1800,height=1600,res=300)


grid.arrange(plotDens_x,legend,plotpoint+theme(legend.position="none"),plotDens_y,
        ncol=2, nrow=2, widths=c(4, 2), heights=c(2,4))

dev.off()

###-------------------------
##protocooperation plot
###-------------------------
RMK202_snpeff_eggnog_repeats_core_uniuqes_final

```

#### protocooperation
```{r}

table(sample_relative_wide$significance)


```


```{bash}

cd /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/
species=Sterm
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt
for species in $(echo "Ldel Sterm")
do
####================
##OGs
####================
filleeezzzz=~/Desktop/presenations/my_presentations/20191007_groupmeeting/figures/RMK202_snpeff_eggnog_repeats_core_dNdS_uniqueGenes.txt
##-----------prt
prtSsss=$(grep "prtS" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/eggnog_orthifinder_${species}_annotations.tsv | cut -f 1)
##-----------pep
#prtSsss=$(grep "pep" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/eggnog_orthifinder_${species}_annotations.tsv | cut -f 1)
pepA=$(awk -F "\t" '{OFS="\t"}{if($6=="pepA")print $0}' ${filleeezzzz} | cut -f 1|head -1)
pepC=$(awk -F "\t" '{OFS="\t"}{if($6=="pepC")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepD=$(awk -F "\t" '{OFS="\t"}{if($6=="pepD")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepF=$(awk -F "\t" '{OFS="\t"}{if($6=="pepF")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepI=$(awk -F "\t" '{OFS="\t"}{if($6=="pepI")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepL=$(awk -F "\t" '{OFS="\t"}{if($6=="pepL")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepN=$(awk -F "\t" '{OFS="\t"}{if($6=="pepN")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepP=$(awk -F "\t" '{OFS="\t"}{if($6=="pepP")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepQ=$(awk -F "\t" '{OFS="\t"}{if($6=="pepQ")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepS=$(awk -F "\t" '{OFS="\t"}{if($6=="pepS")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepT=$(awk -F "\t" '{OFS="\t"}{if($6=="pepT")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepT2=$(awk -F "\t" '{OFS="\t"}{if($6=="pepT2")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepV=$(awk -F "\t" '{OFS="\t"}{if($6=="pepV")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)
pepX=$(awk -F "\t" '{OFS="\t"}{if($6=="pepX")print $0}' eggnog_orthifinder_${species}_annotations.tsv | cut -f 1|head -1)


##-----------Urea
UreCCC=$(grep "3\.5\.1\.5" ${filleeezzzz} | grep "ureC" | cut -f 1)
UreBBB=$(grep "3\.5\.1\.5" ${filleeezzzz} | grep "ureB" | cut -f 1)
UreAAA=$(grep "3\.5\.1\.5" ${filleeezzzz} | grep "ureA" | cut -f 1)
##-----------formate
formmsss=$(grep "2\.3\.1\.54" ${filleeezzzz} | grep "pflB" | cut -f 1)
##-----------folicAcid
FolBBB=$(grep "4\.1\.2\.25" ${filleeezzzz} | grep "folB" | cut -f 1)
Folkkk=$(grep "2\.7\.6\.3" ${filleeezzzz} | grep "folK" | cut -f 1)
Folppp=$(grep "2\.5\.1\.15" ${filleeezzzz}| grep "folP" | cut -f 1)
FolCCC=$(grep "6\.3\.2\.12" ${filleeezzzz}| grep "folC"|head -1 | cut -f 1)
FolAAA=$(grep "1\.5\.1\.3" ${filleeezzzz}| grep "folA"|head -1 | cut -f 1)

####================
##number of Genes
####================
num_UreCCC=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Apr02/Orthogroup_Sequences/${UreCCC}.fa)
num_UreBBB=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Apr02/Orthogroup_Sequences/${UreBBB}.fa)
num_UreAAA=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Apr02/Orthogroup_Sequences/${UreAAA}.fa)
num_formmsss=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Apr02/Orthogroup_Sequences/${formmsss}.fa)
num_FolBBB=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Apr02/Orthogroup_Sequences/${FolBBB}.fa)
num_Folkkk=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Apr02/Orthogroup_Sequences/${Folkkk}.fa)
num_Folppp=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Apr02/Orthogroup_Sequences/${Folppp}.fa)
num_FolCCC=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Apr02/Orthogroup_Sequences/${FolCCC}.fa)
num_FolAAA=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Apr02/Orthogroup_Sequences/${FolAAA}.fa)
num_prtSsss=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Apr02/Orthogroup_Sequences/${prtSsss}.fa)


num_pepA=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Apr02/Orthogroup_Sequences/${pepA}.fa)
num_pepC=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Apr02/Orthogroup_Sequences/${pepC}.fa)
num_pepD=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Apr02/Orthogroup_Sequences/${pepD}.fa)
num_pepF=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Apr02/Orthogroup_Sequences/${pepF}.fa)
num_pepI=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Apr02/Orthogroup_Sequences/${pepI}.fa)
num_pepL=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Apr02/Orthogroup_Sequences/${pepL}.fa)
num_pepN=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Apr02/Orthogroup_Sequences/${pepN}.fa)
num_pepP=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Apr02/Orthogroup_Sequences/${pepP}.fa)
num_pepQ=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Apr02/Orthogroup_Sequences/${pepQ}.fa)
num_pepS=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Apr02/Orthogroup_Sequences/${pepS}.fa)
num_pepT=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Apr02/Orthogroup_Sequences/${pepT}.fa)
num_pepT2=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Apr02/Orthogroup_Sequences/${pepT2}.fa)
num_pepV=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Apr02/Orthogroup_Sequences/${pepV}.fa)
num_pepX=$(grep -c ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Apr02/Orthogroup_Sequences/${pepX}.fa)


####================
#output
####================
echo -e ${species}"\tUreC\t"${UreCCC}"\t"${num_UreCCC} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt
echo -e ${species}"\tUreB\t"${UreBBB}"\t"${num_UreBBB} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt
echo -e ${species}"\tUreA\t"${UreAAA}"\t"${num_UreAAA} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt
echo -e ${species}"\tformate\t"${formmsss}"\t"${num_formmsss} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt
echo -e ${species}"\tfolB\t"${FolBBB}"\t"${num_FolBBB} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt
echo -e ${species}"\tfolK\t"${Folkkk}"\t"${num_Folkkk} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt
echo -e ${species}"\tfolP\t"${Folppp}"\t"${num_Folppp} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt
echo -e ${species}"\tfolC\t"${FolCCC}"\t"${num_FolCCC} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt
echo -e ${species}"\tfolA\t"${FolAAA}"\t"${num_FolAAA} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt
echo -e ${species}"\tprotease\t"${prtSsss}"\t"${num_prtSsss} >> all_interaction_functioning_Genes.txt


echo -e ${species}"\tpepA\t"${pepA}"\t"${num_pepA} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepC\t"${pepC}"\t"${num_pepC} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepD\t"${pepD}"\t"${num_pepD} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepF\t"${pepF}"\t"${num_pepF} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepI\t"${pepI}"\t"${num_pepI} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepL\t"${pepL}"\t"${num_pepL} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepN\t"${pepN}"\t"${num_pepN} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepP\t"${pepP}"\t"${num_pepP} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepQ\t"${pepQ}"\t"${num_pepQ} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepS\t"${pepS}"\t"${num_pepS} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepT\t"${pepT}"\t"${num_pepT} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepT2\t"${pepT2}"\t"${num_pepT2} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepV\t"${pepV}"\t"${num_pepV} >> all_interaction_functioning_Genes.txt
echo -e ${species}"\tpepX\t"${pepX}"\t"${num_pepX} >> all_interaction_functioning_Genes.txt


done # species


####================
#msa of OGs
####================
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/all_clustalwAlignement_final.txt
for species in $(echo "Ldel Sterm")
do
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/InteractionGenesAlignment/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/InteractionGenesAlignment/
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}_clustalwAlignement_final.txt
for OGsss in $(grep "${species}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt |cut -f 3)
do

namesss=$(grep "${species}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt |grep "${OGsss}"|cut -f 2)
numbersss=$(grep "${species}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt |grep "${OGsss}"|cut -f 4)



   echo ==========================
     echo -e "extracting geness: " ${OGsss}
     echo ==========================

clustalw -TYPE=PROTEIN -INFILE=/archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/Results_Apr02/Orthogroup_Sequences/${OGsss}.fa -ALIGN -OUTFILE=/archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/InteractionGenesAlignment/${OGsss}_clustalwAlignement | grep "Aligned. Score:" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/InteractionGenesAlignment/${OGsss}_clustalwAlignement.txt


sed 's/) Aligned. Score:  /\t/g' /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}/InteractionGenesAlignment/${OGsss}_clustalwAlignement.txt | sed 's/Sequences (//g' |awk -F "\t" -v namesss="$species" -v arraysss="$namesss" -v genezzz="$numbersss" -v OGsssssss="$OGsss" '{OFS="\t"}{sum+=$2} END {print namesss,arraysss,OGsssssss,genezzz,sum/NR}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}_clustalwAlignement_final.txt

done #OGsss
cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/${species}_clustalwAlignement_final.txt >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/all_clustalwAlignement_final.txt
done # species
awk -F "\t" '{OFS="\t"}{if($3=="")print $0,"NA","NA","NA"}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/all_interaction_functioning_Genes.txt >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/all_clustalwAlignement_final.txt
```


###### eps/rgp search
here, I want to find the different eps and rgp genes because they seem to play a crucial role in first line phage response. 

This is a nice illustration of [eps](https://www.nature.com/articles/srep04974/figures/4)
```{bash}

mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/
grep "exopolysaccharide" -A 500 -B 500 -i /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff  > /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp.txt

grep "purine" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp.txt > /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp2.txt
grep "regulati" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp.txt >> /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp2.txt
grep "chitin" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp.txt >> /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp2.txt
grep "glycosyltransferases" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp.txt >> /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp2.txt
grep "acetyl" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp.txt >> /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp2.txt
grep "pyruv" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp.txt >> /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp2.txt
grep "transferas" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp.txt >> /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp2.txt
grep "rhamn" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp.txt >> /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp2.txt
grep "transpos" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp.txt >> /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp2.txt
grep "polymer" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp.txt >> /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp2.txt
grep "translocat" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp.txt >> /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp2.txt
grep "phosph" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp.txt >> /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp2.txt
grep "permease" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp.txt >> /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp2.txt
grep "mutase" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp.txt >> /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp2.txt

sort -k 4 -n /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp2.txt |uniq 

grep "deod" -i -A 300 /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/EPs/tmp.txt |awk -F "\t" '{if($3=="CDS")print $0}'


###----------------------prokka
grep "deod_1" -i -A 100 /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA_genes.txt |awk -F "\t" '{if($3=="CDS")print $0}'

/home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff

grep "deod_1" -i -A 100 /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA_genes.txt |awk -F "\t" '{if($3=="CDS")print $0}' |grep "glycosyltransferase"


grep "eps" -i /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff 

grep "rgp" -i /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff 
grep "rhamnos" -i /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff 
grep "glycos" -i /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff 
grep "rhamnos" -i /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff |grep "glycos"

grep "rhamnosyltransferase" -i /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff

grep "Gkc13_02695" -i -A 100 /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/sterm/S_thermophilus_RMK202.current.gb.gff|awk -F "\t" '{if($3=="CDS")print $0}' |grep "glycosyltransferase"


mkdir -p /home/vincent/Desktop/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Ldel/202-LMAG

scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Ldel/202-LMAG/PROKKA_04012020.faa /home/vincent/Desktop/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Ldel/202-LMAG


mkdir -p /home/vincent/Desktop/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG//
scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA_04012020.faa /home/vincent/Desktop/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG//

/archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Sterm/202-SMAG/PROKKA_04012020.faa

```


######snpeff of snps

for this I have to first run the snpEff chunk. 

```{r}

##----------------
##significance
##----------------
ggplot(sample_relative_wide,aes(x=significance,fill=significance))+geom_bar()+
  facet_grid(species~.,scales = "free")+
  theme_classic()+
  xlab("") + 
  ylab("variant sites") +
  scale_fill_manual(values=c("grey","gold1","orange","red"))+   #wes_palette(n=3, name="Moonrise3")[1:3]))+
  theme(legend.position = "none",
        plot.margin=unit(c(0,0,0,2.5),"cm"),
        axis.text.x=element_text(angle=45, hjust=1,size=9))

##----------------
##effect of high significant mutations
##----------------
annotation_snps_high <- sample_relative_wide[sample_relative_wide$significance=="HIGH",]
table(annotation_snps_high$typeGene)
ggplot(annotation_snps_high,aes(x=effect))+geom_bar()+
  facet_grid(species~.,scales = "free")+
  theme_classic()+
theme(legend.position = "none",
        plot.margin=unit(c(0,0,0,2.5),"cm"),
        axis.text.x=element_text(angle=45, hjust=1,size=9))
##----------------
##--------non-modifier mutations
##----------------
annotation_snps_nonModifier <- sample_relative_wide[sample_relative_wide$significance!="MODIFIER",]
# table(annotation_sterm_snps_nonModifier$significance)
##----------------
##--------COG effect of non-modifier mutations
##----------------
# annotation_snps_nonModifier$COG_Functional_Category
ggplot(annotation_snps_nonModifier,aes(x=COG_Functional_Category))+geom_bar()+
  facet_grid(species~.,scales = "free")+
  theme_classic()+
theme(legend.position = "none",
        plot.margin=unit(c(0,0,0,2.5),"cm"),
        axis.text.x=element_text(angle=45, hjust=1,size=9))



```
######only  isolates
Here, I want to see how many SNPs we have on the core genome
```{r}
table(sample_relative_long$sample)
# nrow(sample_relative_long)
sample_relative_safe_onlySTRAINS_sub01 <- sample_relative_long
# sample_relative_safe_woSTRAINS_sub01 <- sample_relative_safe_woSTRAINS[which(sample_relative_safe_woSTRAINS$significance!="MODIFIER"),]
nrow(sample_relative_safe_onlySTRAINS_sub01)
sample_relative_safe_onlySTRAINS_sub02 <- sample_relative_safe_onlySTRAINS_sub01
sample_relative_safe_onlySTRAINS_sub02 <- sample_relative_safe_onlySTRAINS_sub01[which(sample_relative_safe_onlySTRAINS_sub01$core=="core"),]
table(sample_relative_safe_onlySTRAINS_sub02$sample)
sample_relative_safe_onlySTRAINS_sub03 <- sample_relative_safe_onlySTRAINS_sub02[which(sample_relative_safe_onlySTRAINS_sub02$Snps>0.8),]
# sample_relative_safe_woSTRAINS_sub05 <- sample_relative_safe_woSTRAINS_sub03
table(sample_relative_safe_onlySTRAINS_sub03$sample)

sample_relative_safe_onlySTRAINS_sub04 <- sample_relative_safe_onlySTRAINS_sub03 %>% filter(!sample %in% c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2","experiment_A","experiment_B","experiment_C","experiment_D","experiment_E"))

table(sample_relative_safe_onlySTRAINS_sub04$sample)

table(sample_relative_safe_onlySTRAINS_sub04$site) %>% length()
table(sample_relative_safe_onlySTRAINS_sub04$site) %>% table()
```


######remove isolates

```{r}

##----------------
##stats
##----------------

table((sample_relative_wide_meta$core))
length((sample_relative_wide_meta$core))
table(sample_relative_wide_meta$species)
table(sample_relative_wide_meta$significance)

sample_relative_wide_meta$synonomous <- revalue(sample_relative_wide_meta$significance,c("MODIFIER"="synonymous","LOW"="synonymous","MODERATE"="non-synonymous","HIGH"="non-synonymous"))##add certain clusters to others
table(interaction(sample_relative_wide_meta$species,sample_relative_wide_meta$synonomous))

plotSNPs <- ggplot(sample_relative_wide_meta,aes(x=synonomous,fill=significance))+geom_bar()+theme_classic()+facet_wrap(~species,scales="free")+labs(x="",y="count",fill="SNP effect")+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9))
plotSNPs


  # svg("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance_bac_snps.svg",width=5,height=4.5)
   png("~/Desktop/Manuscripts/2019_RMK202/Figures/supplement_SNP_effect.png", width = 1600, height = 1800,res=300)

plotSNPs

dev.off()

##----------------
##01_remove strains
##----------------

# sample_relative_safe_woSTRAINS <- sample_relative_long[grep("^mst",sample_relative_long$sample,invert = TRUE),] 

# sample_relative_safe_woSTRAINS$sample <- droplevels(sample_relative_safe_woSTRAINS$sample)
# sample_relative_safe_final==sample_relative_safe_woSTRAINS
# nrow(sample_relative_safe_woSTRAINS)
# sample_relative_safe_woSTRAINS <- sample_relative_safe_woSTRAINS[which(sample_relative_safe_woSTRAINS$core=="core"),]
# nrow(sample_relative_safe_woSTRAINS)

##reorder samples
# 
# sample_relative_safe_woSTRAINS$sample = factor(sample_relative_safe_woSTRAINS$sample, levels=c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2"))
# levels(sample_relative_safe_woSTRAINS$sample)



  ##============
### plot 
   ##============

all_colours <- rev(c("#EB4D4D","#10B552") ) 

colnames(sample_relative_safe_woSTRAINS)

table(sample_relative_safe_woSTRAINS$sample)
# 
# ##plot
#   p2 <- ggplot(sample_relative_safe_woSTRAINS,aes(x=sample,y=Snps,group=site,color=species,fill=species, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nRepeat_identity:",Repeat_cluster,"\ncore:",core,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.2, alpha=.1)+ 
#     # facet_grid(day~kessel~species)+
#     facet_grid(species~.)+
#     # facet_grid(treatment~species)+
#     # facet_grid(treatment~species)+
#     labs("",
#          x=" ",
#          y="Alternative allele frequency")+
#     #scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
#     #scale_y_continuous(limits = c(0,0.3))+
#     theme_classic()+
#    scale_fill_manual(values=all_colours)+
#    scale_color_manual(values=all_colours)+
#     scale_x_discrete( expand = c(0, 0)) +
#     theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
#           rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
#       #axis.text.x = element_blank(),
#           legend.position="none"
#           #legend.justification=c(1,1), legend.position=c(1,1),
#           #legend.title = element_blank()
#           )
#   
#   p2
  
#   png("~/Desktop/presenations/my_presentations/20190919_VUA/abundance_snps.png", width = 1900, height = 1200,res=300)
# 
# p2
# 
# dev.off()

#  library(plotly)
# # ggp <- ggplotly(p2)
# # htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_bacteria_all.html")
# 
#   svg("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance_bac_snps.svg",width=5,height=4.5)
#    # png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 1000, height = 800,res=300)
# 
# p2
# 
# dev.off()

  ##============
### subset :
##----01==only genic mutations
##----02==only core genes mutations
##----03==only snps above 0.03
##----04==only moderate or high significance SNVs
   ##============
table(sample_relative_long$sample)
nrow(sample_relative_long)
sample_relative_safe_woSTRAINS_sub01 <- sample_relative_long
# sample_relative_safe_woSTRAINS_sub01 <- sample_relative_safe_woSTRAINS[which(sample_relative_safe_woSTRAINS$significance!="MODIFIER"),]
nrow(sample_relative_safe_woSTRAINS_sub01)
sample_relative_safe_woSTRAINS_sub02 <- sample_relative_safe_woSTRAINS_sub01
sample_relative_safe_woSTRAINS_sub02 <- sample_relative_safe_woSTRAINS_sub01[which(sample_relative_safe_woSTRAINS_sub01$core=="core"),]
table(sample_relative_safe_woSTRAINS_sub02$sample)
sample_relative_safe_woSTRAINS_sub03 <- sample_relative_safe_woSTRAINS_sub02[which(sample_relative_safe_woSTRAINS_sub02$Snps>0.03),]
# sample_relative_safe_woSTRAINS_sub05 <- sample_relative_safe_woSTRAINS_sub03
table(sample_relative_safe_woSTRAINS_sub03$sample)

sample_relative_safe_woSTRAINS_sub04 <- sample_relative_safe_woSTRAINS_sub02[which(sample_relative_safe_woSTRAINS_sub02$Snps<=0.03),]
sample_relative_safe_woSTRAINS_sub04$Snps <- 0
nrow(sample_relative_safe_woSTRAINS_sub04)
sample_relative_safe_woSTRAINS_sub05 <- rbind(sample_relative_safe_woSTRAINS_sub03,sample_relative_safe_woSTRAINS_sub04)
sample_relative_safe_woSTRAINS_sub06 <- sample_relative_safe_woSTRAINS_sub05[which(sample_relative_safe_woSTRAINS_sub05$significance %in% c("MODERATE","HIGH")),]

nrow(sample_relative_safe_woSTRAINS_sub05)
table(sample_relative_safe_woSTRAINS_sub05$sample)

sample_relative_safe_Sterm_final <- sample_relative_safe_woSTRAINS_sub06[which(sample_relative_safe_woSTRAINS_sub06$species=="S. thermophilus"),]
table(sample_relative_safe_Sterm_final$sample)
sample_relative_safe_Sterm_final_wide <- spread(sample_relative_safe_Sterm_final, sample, Snps)  %>%replace(is.na(.), 0)

table(sample_relative_safe_Sterm_final_wide$sample)
# length(unique(sample_relative_safe_Sterm_final$site))
  ##============
### plot Streptococcus thermophius
   ##============



colnames(sample_relative_safe_Sterm_final)
table(sample_relative_safe_Sterm_final$sample)

# sample_relative_safe_woSTRAINS <- sample_relative_safe_Sterm_final[grep("^mst",sample_relative_safe_Sterm_final$sample,invert = TRUE),] 

sample_relative_safe_woSTRAINS <- sample_relative_safe_Sterm_final %>% filter(sample %in% c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2","experiment_A","experiment_B","experiment_C","experiment_D","experiment_E"))


sample_relative_safe_woSTRAINS$sample <- droplevels(sample_relative_safe_woSTRAINS$sample)
sample_relative_safe_woSTRAINS$sample = factor(sample_relative_safe_woSTRAINS$sample, levels=c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2","experiment_A","experiment_B","experiment_C","experiment_D","experiment_E"))

levels(sample_relative_safe_woSTRAINS$sample)
table(sample_relative_safe_woSTRAINS$sample)

  sample_relative_safe_woSTRAINS <- sample_relative_safe_woSTRAINS %>% filter(sample!="cheesemaking\nday1")  %>% filter(sample!="cheesemaking\nday2")


  p3 <- ggplot(sample_relative_safe_woSTRAINS,aes(x=sample,y=Snps,group=site,color=species,fill=species, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nRepeat_identity:",Repeat_cluster,"\ncore:",core,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.2, alpha=.1)+
    # facet_grid(day~kessel~species)+
    # facet_grid(species~.)+
    # facet_grid(treatment~species)+
    # facet_grid(treatment~species)+
    labs("",
         x="",
         y="Alternative allele frequency")+
    #scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    #scale_y_continuous(limits = c(0,0.3))+
    theme_classic()+
   scale_fill_manual(values=all_colours[2])+
   scale_color_manual(values=all_colours[2])+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
      #axis.text.x = element_blank(),
          legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          )

#   svg("~/Desktop/presenations/my_presentations/20190919_VUA/abundance_snps.png", width = 1900, height = 1200,res=300)
# #
p3
#
# dev.off()
  svg("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance_bac_snps_sterm.svg",width=6,height=5)
   # png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 1800, height = 1200,res=300)
# 
# p3+theme(axis.title = element_blank(),axis.text.x = element_blank())
p3

dev.off()




  ##============
### plot Lactobacillus delbrueckii
   ##============


nrow(sample_relative_long)
sample_relative_safe_woSTRAINS_sub01 <- sample_relative_long
# sample_relative_safe_woSTRAINS_sub01 <- sample_relative_safe_woSTRAINS[which(sample_relative_safe_woSTRAINS$significance!="MODIFIER"),]
nrow(sample_relative_safe_woSTRAINS_sub01)
table(sample_relative_safe_woSTRAINS_sub01$species)

sample_relative_safe_woSTRAINS_sub02 <- sample_relative_safe_woSTRAINS_sub01
sample_relative_safe_woSTRAINS_sub02 <- sample_relative_safe_woSTRAINS_sub01[which(sample_relative_safe_woSTRAINS_sub01$core=="core"),]
nrow(sample_relative_safe_woSTRAINS_sub02)
table(sample_relative_safe_woSTRAINS_sub02$species)
sample_relative_safe_woSTRAINS_sub03 <- sample_relative_safe_woSTRAINS_sub01[which(sample_relative_safe_woSTRAINS_sub01$Snps>0.03),]
# sample_relative_safe_woSTRAINS_sub05 <- sample_relative_safe_woSTRAINS_sub03
table(sample_relative_safe_woSTRAINS_sub03$species)
sample_relative_safe_woSTRAINS_sub04 <- sample_relative_safe_woSTRAINS_sub03 %>% filter(species=="L. delbrueckii") 
unique(sample_relative_safe_woSTRAINS_sub04$site) %>% length()
sample_relative_safe_woSTRAINS_sub03 <- sample_relative_safe_woSTRAINS_sub02[which(sample_relative_safe_woSTRAINS_sub02$Snps<=0.03),]
sample_relative_safe_woSTRAINS_sub04$Snps <- 0
table(sample_relative_safe_woSTRAINS_sub04$species)

nrow(sample_relative_safe_woSTRAINS_sub04)
sample_relative_safe_woSTRAINS_sub05 <- rbind(sample_relative_safe_woSTRAINS_sub03,sample_relative_safe_woSTRAINS_sub04)
sample_relative_safe_woSTRAINS_sub06 <- sample_relative_safe_woSTRAINS_sub05[which(sample_relative_safe_woSTRAINS_sub05$significance %in% c("MODERATE","HIGH")),]




sample_relative_safe_Sterm_final_ldel <- sample_relative_safe_woSTRAINS_sub06[which(sample_relative_safe_woSTRAINS_sub06$species=="S. thermophilus"),]

sample_relative_safe_Sterm_final_wide <- spread(sample_relative_safe_Sterm_final_ldel, sample, Snps)  %>%replace(is.na(.), 0)

sample_relative_safe_woSTRAINS

table(sample_relative_safe_woSTRAINS_sub06$species)

sample_relative_safe_woSTRAINS <- sample_relative_safe_Sterm_final[grep("^mst",sample_relative_safe_Sterm_final$sample,invert = TRUE),] 
sample_relative_safe_woSTRAINS$sample <- droplevels(sample_relative_safe_woSTRAINS$sample)
sample_relative_safe_woSTRAINS$sample = factor(sample_relative_safe_woSTRAINS$sample, levels=c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2"))
levels(sample_relative_safe_woSTRAINS$sample)

  sample_relative_safe_woSTRAINS <- sample_relative_safe_woSTRAINS %>% filter(sample!="cheesemaking\nday1")  %>% filter(sample!="cheesemaking\nday2")




table(sample_relative_safe_woSTRAINS$X.CHROM.x)

sample_relative_safe_woSTRAINS_sub_ldel <- sample_relative_safe_woSTRAINS %>% filter(X.CHROM.x=="CP046131") %>% filter(Snps>0.05)


#   p4 <- ggplot(sample_relative_safe_woSTRAINS_sub_ldel,aes(x=sample,y=Snps,group=site,color=species,fill=species, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nRepeat_identity:",Repeat_cluster,"\ncore:",core,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.2, alpha=0.8)+ 
#     # facet_grid(day~kessel~species)+
#     # facet_grid(species~.)+
#     # facet_grid(treatment~species)+
#     # facet_grid(treatment~species)+
#     labs("",
#          x="time [h]",
#          y="Alternative allele frequency")+
#     #scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
#     #scale_y_continuous(limits = c(0,0.3))+
#     theme_classic()+
#    scale_fill_manual(values=all_colours[1])+
#    scale_color_manual(values=all_colours[1])+
#     scale_x_discrete( expand = c(0, 0)) +
#     theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
#           rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
#       # axis.text.x = element_blank(),
#           legend.position="none",
#           #legend.justification=c(1,1), legend.position=c(1,1),
#           # axis.title = element_blank()
#           )
#   
# #   svg("~/Desktop/presenations/my_presentations/20190919_VUA/abundance_snps.png", width = 1900, height = 1200,res=300)
# # # 
#   p4
#  svg("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance_bac_snps_ldel.svg",width=4,height=1.8)
#    # png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 1000, height = 800,res=300)
# 
# p4+theme(axis.title = element_blank(),axis.text.x = element_blank())
# 
# dev.off()
#   svg("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance_bac_snps_title.svg",width=4,height=3)
#    # png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 1000, height = 800,res=300)
# 
# p4
# 
# dev.off()

# 
#  library(plotly)
# ggp <- ggplotly(p3)
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_bacteria_onlyModifier.html")

```

#####phase streptococcus


```{r}
library(forcats) 
library(Rtsne)
library(fpc)
library(ggrepel)
###--------------------------------------------------------------------
##colors
###--------------------------------------------------------------------

# colorsTSNE <- c("#DD8D29","#E2D200","#46ACC8","black","#B40F20")

colorsTSNE <- c("#E69F00", "#56B4E9", "#009E73",
          "#F0E442", "#0072B2", "#D55E00", "#CC79A7","grey","#000000")

colorsTSNE <- rev(c("#FF9E79","#FB6D4C","#C23B22","#8A0000","#580000","#D55E00"))
###--------------------------------------------------------------------
##subset
###--------------------------------------------------------------------

# mst1_metagenomicMutations <- sample_relative_wide %>% filter(.,species=="S. thermophilus") %>% filter(.,significance %in% c("MODERATE","HIGH")) %>% filter(.,mst1>0.9 & mst2 <=0.9) %>% add_column(group="mst1") %>%replace(is.na(.), 0)
mst1_metagenomicMutations <- sample_relative_safe_Sterm_final_wide %>% filter(.,mst1>0.9 & mst2 <=0.9)%>% add_column(group="mst1") 


# mst1_metagenomicMutations_long <-  mst1_metagenomicMutations %>% gather(., sample, Snps, "Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2", factor_key=TRUE,na.rm = TRUE)# %>% filter(.,Snps>0.005)
mst1_metagenomicMutations_long <-  sample_relative_safe_woSTRAINS

mst1_metagenomicMutations_tsne <- spread(mst1_metagenomicMutations_long, sample, Snps)%>%replace(is.na(.), 0)


###--------------------------------------------------------------------
##prep for tSNE
###--------------------------------------------------------------------



fortsne <- mst1_metagenomicMutations_tsne %>% select(.,"site","Lyo\n1996","Lyo\n2012","working\nstock","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018")
fortsne <- fortsne[!duplicated(fortsne[,-1]), ]
# fortsne[duplicated(fortsne), ] %>% nrow()
dim(fortsne)
set.seed(1234567)
# set.seed(12314567)

tsne <- Rtsne(fortsne[,-1], dims = 2, perplexity=10, verbose=TRUE, max_iter = 500)
# tsne <- Rtsne(fortsne[,41:45], dims = 2, perplexity=10, verbose=TRUE, max_iter = 500)
tsne_plot <- as.data.frame(tsne$Y)
tsne_plot$site <- fortsne[,1]
# tsne_plot$species <- paste(str_split_fixed(tsne_plot$site, "_", 5)[,1],str_split_fixed(tsne_plot$site, "_", 5)[,2],sep = "_") %>%  gsub("_",". ",.)



##------------plot blank
# ggplot(tsne_plot,aes(x=V1,y=V2))+geom_point()

##------------clusters

# set.seed(123456)

set.seed(1234567)
ds.real = dbscan(tsne_plot[,c(1,2)], 4)
clusterNames <- (names(sort(table(ds.real$cluster),decreasing = TRUE)))
# change_name <- setNames(1:length(decreasing_clusternames),decreasing_clusternames)
# clusterNames <- revalue(as.factor(ds.real$cluster),change_name)
# clusterNames<- revalue(clusterNames,c("4"="1","6"="1","7"="1","10"="1","11"="1","5"="4","8"="5","9"="6"))##add certain clusters to others
tsne_plot$density = ds.real$cluster
tsne_plot <- tsne_plot %>% filter(density %in% c("1","3","4","5","6"))
table(tsne_plot$density)
# tsne_plot$density <- revalue(as.character(tsne_plot$density),c("5"="2","6"="2","3"="4","4"="3","2"="1")) %>% as.numeric()
# tsne_plot$density <- revalue(as.character(tsne_plot$density),c("3"="4","4"="3","2"="1")) %>% as.numeric()
tsne_plot$density <- revalue(as.character(tsne_plot$density),c("5"="2")) %>% as.numeric()

table(tsne_plot$density)
# levels(tsne_plot$density) <- c(levels(tsne_plot$density), "5")

# special_cases <- c("CP046134_494890_A_G","CP046134_482906_A_C")
# tsne_plot[which(tsne_plot$site %in% special_cases),"density"] <- "5"
ds.cent = tsne_plot %>% group_by(density) %>% select(V1,
    V2) %>% summarize_all(mean)

# colors_substrains <- c(colorsTSNE[1:length(unique(tsne_plot$density))])

# tsne_plot$density <- as.character(tsne_plot$density)

tsnePlot1 <- ggplot(tsne_plot, aes(x = V1, y = V2, color = density))+labs(x="tsne1",y="tsne2") +
    geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label = density),
    data = ds.cent) + guides(colour = FALSE)




svg("~/Desktop/mid_thesis/report/figures/20200430/chapter1/Tsne_clustering_snps_sterm.svg",width=5,height=4.5)
tsnePlot1
dev.off()

##-----------make clustering manual
# tsne_plot$final_clust <- tsne_plot$density
# tsne_plot_final_clust <- tsne_plot  %>% select(site,  final_clust)
# table(tsne_plot_final_clust$final_clust)

# tsne_plot_final_clust <- tsne_plot %>% mutate(final_clust = ifelse(density==3,2,
#                                                                     ifelse(density==2,3,1)))  %>% select(site,  final_clust)
#  tsne_plot <- tsne_plot %>% mutate(final_clust = ifelse(density==3,2,
#                                                                     ifelse(density==2,3,1)))
# table(tsne_plot_final_clust$final_clust)
# tsne_plot$density <- revalue(tsne_plot$density, c("4"="1"))

tsne_plot <- tsne_plot %>% mutate(final_clust=density)
tsne_plot_final_clust <- tsne_plot %>% select(site,  final_clust)

###------------------------------------
##-------sumplementary plot tsne
###------------------------------------
tsne_plot$final_clust = factor(tsne_plot$final_clust, levels=1:length(unique(tsne_plot_final_clust$final_clust)))
ds.cent = tsne_plot %>% group_by(final_clust) %>% select(V1, 
    V2) %>% summarize_all(mean)
colors_substrains <- c(colorsTSNE[1:length(unique(tsne_plot_final_clust$final_clust))])

# Sterm_mst1_suplementaryTsne <- ggplot(tsne_plot, aes(x = V1, y = V2, colour = final_clust)) + 
#     geom_point(alpha = 0.3) + theme_classic() + geom_label_repel(aes(label =final_clust), 
#     data = ds.cent) + guides(colour = FALSE)+scale_color_manual(values = colors_substrains)+theme(axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank())
# 
# Sterm_mst1_suplementaryTsne

###------------------------------------
##-------sumplementary histogramm
###------------------------------------

# Sterm_mst1_suplementarybarPlot <- ggplot(tsne_plot,aes(x=final_clust,color=final_clust,fill=final_clust))+geom_bar()+theme_classic()+
#   scale_fill_manual(values = colors_substrains)+
#   scale_color_manual(values = colors_substrains)+
#   labs(x="",y="count")+theme(legend.position = "none")
# Sterm_mst1_suplementarybarPlot
###------------------------------------
##---------clustering
###------------------------------------

explained_mst1_plot_nonModi_CLUSTERED <- merge(mst1_metagenomicMutations_long,tsne_plot_final_clust,by = "site",all.x = TRUE)
# table(explained_mst1_plot_nonModi_CLUSTERED$final_clust)
# unique(explained_mst1_plot_nonModi_CLUSTERED$final_clust)
# explained_mst1_plot_nonModi_CLUSTERED[which(is.na(explained_mst1_plot_nonModi_CLUSTERED$final_clust)),]
explained_mst1_plot_nonModi_CLUSTERED <- explained_mst1_plot_nonModi_CLUSTERED[which(!is.na(explained_mst1_plot_nonModi_CLUSTERED$final_clust)),]
explained_mst1_plot_nonModi_CLUSTERED$final_clust <- fct_explicit_na(as.character(explained_mst1_plot_nonModi_CLUSTERED$final_clust), na_level = "not clustered")
# unique(explained_mst1_plot_nonModi_CLUSTERED$final_clust)
explained_mst1_plot_nonModi_CLUSTERED$final_clust <- droplevels(explained_mst1_plot_nonModi_CLUSTERED$final_clust)

###------------------------------------
##plotting
###------------------------------------
# explained_mst1_plot_nonModi_CLUSTERED
explained_mst1_plot_nonModi_CLUSTERED$final_clust = factor(explained_mst1_plot_nonModi_CLUSTERED$final_clust, levels=(sort(unique(explained_mst1_plot_nonModi_CLUSTERED$final_clust),decreasing = FALSE)))

# levels(explained_mst1_plot_nonModi_CLUSTERED$final_clust)
plotColors <- colorsTSNE[as.numeric(levels(explained_mst1_plot_nonModi_CLUSTERED$final_clust))]

##assign colorsi
explained_mst1_plot_nonModi_CLUSTERED$final_clust <- droplevels(explained_mst1_plot_nonModi_CLUSTERED$final_clust)
colors_mst1 <- data.frame(cluster=paste("mst1",as.character(levels(explained_mst1_plot_nonModi_CLUSTERED$final_clust)),sep="_"),color=plotColors[1:length(levels(explained_mst1_plot_nonModi_CLUSTERED$final_clust))])


  p_mst1_sterm_woModi_clustered <- ggplot(explained_mst1_plot_nonModi_CLUSTERED,aes(x=sample,y=Snps,group=site,color=final_clust,fill=final_clust, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.1, alpha=.5)+ 
       # facet_grid(final_clust~.)+
   # labs("",
         # x="",
         # y="",
          # y="all")+
       scale_fill_manual(values=(plotColors))+
      scale_color_manual(values=(plotColors))+
       scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          ylim(c(0,1))+
 theme(#axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="bottom",
       legend.title = element_blank(),
       axis.text.x = element_blank(),
       axis.text.y = element_text(size=8),
       # axis.title = element_text(size=9),
       axis.title = element_blank(),
       
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )

p_mst1_sterm_woModi_clustered
# 
# library(plotly)
# ggp <- ggplotly(p_mst1_sterm_woModi_clustered)
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_mst1_clustered_woModi.html")
# 

##===========================================
##-------------------------------------plotting withou clustering
##===========================================

    p_mst1_sterm_woModi_NOT_clustered <- ggplot(explained_mst1_plot_nonModi_CLUSTERED,aes(x=sample,y=Snps,group=site, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.1, alpha=.5)+ 
       # facet_grid(final_clust~.)+
   # labs("",
         # x="",
         # y="",
          # y="all")+
       # scale_fill_manual(values=(plotColors))+
      # scale_color_manual(values=(plotColors))+
       scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="bottom",
       legend.title = element_blank(),
       # axis.text.x = element_blank(),
       axis.text.y = element_text(size=8),
       # axis.title = element_text(size=9),
       axis.title = element_blank(),
       
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
p_mst1_sterm_woModi_NOT_clustered

# library(plotly)
# ggp <- ggplotly(p_mst1_sterm_woModi_clustered)
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_mst1_clustered_woModi.html")

tsne_plot_mst1 <- tsne_plot
##===========================================
##-------------------------------------plotting without cheesemaking samples
##===========================================

###------------------------------------
##density plot
###------------------------------------
explained_mst1_plot_nonModi_CLUSTERED$location <- as.numeric(str_split_fixed(explained_mst1_plot_nonModi_CLUSTERED$site, "_", 5)[,2])

forDensityPlot <- explained_mst1_plot_nonModi_CLUSTERED %>% select(.,site,location,final_clust) %>% distinct()

forDensityPlot$final_clust <- droplevels(forDensityPlot$final_clust)
forDensityPlot$final_clust = factor(forDensityPlot$final_clust, levels=(sort(unique(forDensityPlot$final_clust),decreasing = FALSE)))
# plotColors <- colorsTSNE[as.numeric(levels(forDensityPlot$final_clust))]
unique(forDensityPlot$final_clust)

Sterm_mst1_density <- ggplot(forDensityPlot,aes(x=location,group=final_clust,color=final_clust,fill=final_clust))+geom_density(alpha=0.2)+theme_classic()+
  facet_wrap(~final_clust,scale = "free_y")+
  scale_fill_manual(values = c(colors_substrains,"grey"))+
  scale_color_manual(values = c(colors_substrains,"grey"))+
  labs(x="genomic location",y="density")+theme(strip.text.x = element_blank(),legend.position = "none",legend.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank())
Sterm_mst1_density

Sterm_mst1_frequency <- ggplot(forDensityPlot,aes(x=location,group=final_clust,color=final_clust,fill=final_clust))+geom_freqpoly()+theme_classic()+
  # facet_wrap(~final_clust,scale = "free_y")+
  scale_fill_manual(values = c(colors_substrains,"grey"))+
  scale_color_manual(values = c(colors_substrains,"grey"))+
  labs(x="genomic location",y="density")+theme(legend.position = "none",legend.title = element_blank())
Sterm_mst1_frequency


Sterm_mst1_histo <- ggplot(forDensityPlot,aes(x=location,group=final_clust,color=final_clust,fill=final_clust))+geom_histogram(position="dodge2")+theme_classic()+
  # facet_wrap(~final_clust,scale = "free_y")+
  scale_fill_manual(values = c(colors_substrains,"grey"))+
  scale_color_manual(values = c(colors_substrains,"grey"))+
  labs(x="genomic location",y="density")+theme(legend.position = "none",legend.title = element_blank())
Sterm_mst1_histo


Sterm_mst1_point <- ggplot(forDensityPlot,aes(x=location,y=final_clust,group=final_clust,color=final_clust,fill=final_clust))+geom_jitter(width = 0.0, height = 0.4,alpha=0.3 ,size = 0.35)+theme_classic()+
  # facet_wrap(~final_clust,scale = "free_y")+
  scale_fill_manual(values = c(colors_substrains,"grey"))+
  scale_color_manual(values = c(colors_substrains,"grey"))+
  labs(x="genomic location",y="cluster")+theme(legend.position = "none",legend.title = element_blank(),axis.text.x = element_blank(),axis.ticks.x = element_blank())
Sterm_mst1_point

# theme(strip.text.x = element_blank())



##====================================================
###------RIBBON of interquartile range
##====================================================

##=====================here I apply a hack!!
##in order to recreat eh labels from ggplot_build I color the samples and final_clust according to a known color scale

colorSamples <- c("gray1","gray2","gray3","gray4","gray5","gray6","gray7","gray8","gray9","gray10","gray11")
 # explained_mst1_plot_nonModi_CLUSTERED$final_clust = factor(explained_mst1_plot_nonModi_CLUSTERED$final_clust, levels=c(6:1))
explained_mst1_plot_nonModi_CLUSTERED$final_clust = factor(explained_mst1_plot_nonModi_CLUSTERED$final_clust, levels=(sort(unique(explained_mst1_plot_nonModi_CLUSTERED$final_clust),decreasing = FALSE)))

 explained_mst1_plot_nonModi_CLUSTERED$sample = factor(explained_mst1_plot_nonModi_CLUSTERED$sample, levels=(c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2")))

##----------------------prepare table

p <- ggplot(explained_mst1_plot_nonModi_CLUSTERED, aes(x=sample, y=Snps,color=sample,fill=final_clust)) + geom_boxplot() +scale_fill_manual(values = colorSamples)+scale_color_manual(values = colorSamples)
p
mst1_phase_qunatiles <- ggplot_build(p)$data[[1]] %>% as.tibble() %>%  rename(sample=colour)  %>%  rename(final_clust=fill) %>% mutate(final_clust = revalue(final_clust, c("gray1"="1", "gray2"="2", "gray3"="3", "gray4"="4", "gray5"="5", "gray6"="6", "gray7"="7", "gray8"="8")))%>% mutate(sample = revalue(sample, c("gray1"="Lyo\n1996", "gray2"="Lyo\n2012", "gray3"="Lyo\n2014", "gray4"="working\nstock", "gray5"="starter\nculture\n2012", "gray6"="starter\nculture\n2018", "gray7"="cheesemaking\nday1", "gray8"="cheesemaking\nday2")))
 table( mst1_phase_qunatiles$final_clust)
str(mst1_phase_qunatiles$final_clust)

mst1_phase_qunatiles <- ggplot_build(p)$data[[1]] %>% as.tibble() %>%  rename(sample=colour)  %>%  rename(final_clust=fill) %>% mutate(final_clust = revalue(final_clust, c("gray1"="1", "gray2"="2", "gray3"="3", "gray4"="4", "gray5"="5", "gray6"="6", "gray7"="7", "gray8"="8")))%>% mutate(sample = revalue(sample, c("gray1"="Lyo\n1996", "gray2"="Lyo\n2012", "gray3"="Lyo\n2014", "gray4"="working\nstock", "gray5"="starter\nculture\n2012", "gray6"="starter\nculture\n2018", "gray7"="cheesemaking\nday1", "gray8"="cheesemaking\nday2"))) %>% select(-"y")



# mst1_phase_qunatiles$final_clust = factor(mst1_phase_qunatiles$final_clust, levels=c(6:1))
# plotColors <- colorsTSNE[as.numeric(levels(mst1_phase_qunatiles$final_clust))]

mst1_phase_qunatiles$final_clust = factor(mst1_phase_qunatiles$final_clust, levels=(sort(unique(mst1_phase_qunatiles$final_clust),decreasing = FALSE)))
plotColors <- colorsTSNE[as.numeric(levels(mst1_phase_qunatiles$final_clust))]


 mst1_phase_qunatiles$sample = factor(mst1_phase_qunatiles$sample, levels=(c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2")))


mst1_phased_ribbon <- ggplot()+
    geom_ribbon(aes(x=sample,ymin = ymin, ymax = ymax,group=final_clust,fill=final_clust),data = mst1_phase_qunatiles, alpha=.2)+
    # geom_ribbon(aes(x=sample,ymin = X1st.Qu., ymax = X3rd.Qu.,group=final_clust,fill=final_clust),data = both_phase_qunatiles, alpha=.5)+
    # geom_ribbon(aes(x=sample,ymin = Min., ymax = Max.,group=final_clust,fill=final_clust),data = mst1_phase_qunatiles, alpha=.5)+
    geom_line(data=mst1_metagenomicMutations_long,aes(x=sample,y=Snps,group=site),color="red",size=0.1, alpha=.15)+
    # geom_line(data=explained_mst1_plot_nonModi_CLUSTERED,aes(x=sample,y=Snps,group=site,color=final_clust,fill=final_clust),size=0.1, alpha=.1)+
    # geom_line(data = mappingcoverage,aes(y = coverage, x = time_f_02,group=contig))+
      # facet_grid(contig~treatment )+
      # scale_x_continuous(breaks = c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
       # scale_x_discrete(breaks = c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    scale_fill_manual(values=(plotColors))+
      scale_color_manual(values=(plotColors))+
       scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
  labs(y="alternative allele frequency")+
          ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="bottom",
       legend.title = element_blank(),
       # axis.text.x = element_blank(),
       axis.text.y = element_text(size=9),
       # axis.title = element_text(size=9),
       axis.title.x = element_blank(),
       
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )


  
  mst1_phased_ribbon

svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/STERM_line_with_ribbon_snp.svg",width=5,height=4.5)
mst1_phased_ribbon
dev.off()

  
```

##### Phasing new
```{r}

# colnames(sample_relative_wide)
# 
# sample_relative_wide_snpsUsed <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% 
#   filter("Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 )
# 
# total_tmp <- sample_relative_wide_snpsUsed  %>%
#       filter(mst1>0.9 | mst2>0.9) 
#   
  
  
sample_relative_wide_snpsUsed <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")    %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 )

sample_relative_wide_snpsUsed_lin1 <- sample_relative_wide_snpsUsed %>% filter(`24853`>0.8| `24798`>0.8| `13493`>0.8| `13500`>0.8 | `24737`>0.8 | `24854`>0.8| `13491`>0.8 | `13492`>0.8 ) %>% add_column(explained="lin1")
sample_relative_wide_snpsUsed_lin2 <- sample_relative_wide_snpsUsed %>% filter(`S72`>0.8| `24855`>0.8| `13494`>0.8)%>% add_column(explained="lin2")
sample_relative_wide_snpsUsed_lin3 <- sample_relative_wide_snpsUsed %>% filter(`S50`>0.8| `24740`>0.8| `24738`>0.8| `13499`>0.8 )%>% add_column(explained="lin3")
sample_relative_wide_snpsUsed_lin4 <- sample_relative_wide_snpsUsed %>% filter(`24739`>0.8| `13497`>0.8| `13496`>0.8| `13495`>0.8 )%>% add_column(explained="lin4")



allsites <- c(sample_relative_wide_snpsUsed_lin1$site, sample_relative_wide_snpsUsed_lin2$site, sample_relative_wide_snpsUsed_lin3$site,sample_relative_wide_snpsUsed_lin4$site)


sample_relative_wide_snpsUsed_lin1only <- sample_relative_wide_snpsUsed %>% 
  filter(`24853`>0.8| `24798`>0.8| `13493`>0.8| `13500`>0.8 | `24737`>0.8 | `24854`>0.8| `13491`>0.8 | `13492`>0.8 ) %>% add_column(explained="lin1") %>% 
  filter(`S72`<=0.8| `24855`<=0.8| `13494`<=0.8) %>% 
  filter(`S50`<=0.8| `24740`<=0.8| `24738`<=0.8| `13499`<=0.8 ) %>% 
  filter(`24739`<=0.8| `13497`<=0.8| `13496`<=0.8| `13495`<=0.8 )

sample_relative_wide_snpsUsed_lin2only <- sample_relative_wide_snpsUsed %>% 
  filter(`24853`<=0.8& `24798`<=0.8& `13493`<=0.8& `13500`<=0.8 & `24737`<=0.8 & `24854`<=0.8& `13491`<=0.8 & `13492`<=0.8 ) %>% add_column(explained="lin2") %>% 
  filter(`S72`>0.8| `24855`>0.8| `13494`>0.8) %>% 
  filter(`S50`<=0.8& `24740`<=0.8& `24738`<=0.8& `13499`<=0.8 ) %>% 
  filter(`24739`<=0.8& `13497`<=0.8& `13496`<=0.8& `13495`<=0.8 )

sample_relative_wide_snpsUsed_lin2 <- sample_relative_wide_snpsUsed %>% filter(`S72`>0.8| `24855`>0.8| `13494`>0.8)%>% add_column(explained="lin2")
sample_relative_wide_snpsUsed_lin3 <- sample_relative_wide_snpsUsed %>% filter(`S50`>0.8| `24740`>0.8| `24738`>0.8| `13499`>0.8 )%>% add_column(explained="lin3")
sample_relative_wide_snpsUsed_lin4 <- sample_relative_wide_snpsUsed %>% filter(`24739`>0.8| `13497`>0.8| `13496`>0.8| `13495`>0.8 )%>% add_column(explained="lin4")
allsites <- c(sample_relative_wide_snpsUsed_lin1$site, sample_relative_wide_snpsUsed_lin2$site, sample_relative_wide_snpsUsed_lin3$site,sample_relative_wide_snpsUsed_lin4$site)


```


#####Venn-Diagramm

```{r}




library(gplots)
library(gridExtra)
library(grid)
library(VennDiagram)

# colnames(sample_relative_wide)
# nrow(sample_relative_wide)

 # sample_relative_safe_final_no_snps <- sample_relative_wide[,c("site","RMK202","Lyo_202_2014","S72","Lyo_202_2012","L99","L80","Versand_202","L71","L70","Konserve_202","L108","L35","S50","L39","L104","L44")]

 ##-------------------------------
 ##meta has coverage
##-------------------------------
sample_relative_safe_final_snps_meta <- sample_relative_wide %>% select(site,"Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018")
sum(rowSums(sample_relative_safe_final_snps_meta[,-1],na.rm = TRUE)==0)
sum(rowSums(sample_relative_safe_final_snps_meta[,-1],na.rm = TRUE)>0)
sample_relative_safe_final_meta_cleaned <- sample_relative_wide[(rowSums(sample_relative_safe_final_snps_meta[,-1],na.rm = TRUE)>0),]
# sample_relative_safe_final_meta_cleaned_02 <- sample_relative_safe_final_snps_meta[(rowSums(sample_relative_safe_final_snps_meta[,-1],na.rm = TRUE)>0),]


 ##-------------------------------
 ##isolates has coverage
##-------------------------------

dim(sample_relative_wide)
in_isolates <- sample_relative_wide[which((sample_relative_wide$mst1>0.9 | sample_relative_wide$mst2>0.9| sample_relative_wide$mst3>0.9| sample_relative_wide$mst4>0.9| sample_relative_wide$mst5>0.9| sample_relative_wide$mst6>0.9 | sample_relative_wide$mst7>0.9 | sample_relative_wide$mst8>0.9 |sample_relative_wide$mst9>0.9 |  sample_relative_wide$mst10>0.9) %in% TRUE),]
dim(in_isolates)


##----------------
##plotting Venn with Isolates
##----------------
##==================================================================ldel

#----------------------total snps in analysis
 # %>% filter(core=="core")
total_tmp <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(species=="L. delbrueckii")  %>% filter(mst3>0.9| mst7>0.9| mst6>0.9 |  mst9>0.9|mst4>0.9) %>% filter( mst5<=0.9) %>% filter( mst8<=0.9 & mst10<=0.9) %>% filter("Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05| "starter\nculture\n2018">0.05 ) %>% nrow()
#----------------------in isolate
isolates_tmp <- sample_relative_wide %>%replace(is.na(.), 0)%>% filter(species=="L. delbrueckii")  %>% filter(mst3>0.9 | mst4>0.9| mst5>0.9| mst6>0.9| mst7>0.9| mst8>0.9| mst9>0.9| mst10>0.9)%>% nrow()
#----------------------in meta
meta_tmp <- sample_relative_wide %>%replace(is.na(.), 0)%>% filter(species=="L. delbrueckii")  %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05| "starter\nculture\n2018">0.05 ) %>% nrow()
##---------------------in meta or isolate
shared_tmp <- sample_relative_wide  %>%replace(is.na(.), 0)  %>%replace(is.na(.), 0)%>% filter(species=="L. delbrueckii")  %>% filter(mst3>0.9 | mst4>0.9| mst5>0.9| mst6>0.9| mst7>0.9| mst8>0.9| mst9>0.9| mst10>0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05| "starter\nculture\n2018">0.05 ) %>% nrow()

##----------------
##plotting Venn in meta or not
##----------------
plot.new()
venn.plot <- draw.pairwise.venn(isolates_tmp,meta_tmp, shared_tmp, c("in isolates", "only in metagenome"), lty = rep("blank",
    2), fill = c("light blue", "grey"), alpha = rep(0.5, 2), cat.pos = c(0,
    0), cat.dist = rep(0.025, 2))
grid.draw(venn.plot)

svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/04_venn_diagramm_ldel.svg",width=6,height=4)
grid.draw(venn.plot)
dev.off()

##==============================
##-------------group1
group1_count <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter(mst3>0.9| mst7>0.9| mst6>0.9 |  mst9>0.9|mst4>0.9)%>% nrow()
##-------------group2
group2_count <-  sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter( mst8>0.9) %>% nrow()
##-------------group3
group3_count <-  sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter( mst5>0.9|mst10>0.9) %>% nrow()
#----------------------in meta
meta_tmp <-  sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% nrow()
##==============================
group1_group2_match <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter(mst3>0.9| mst7>0.9| mst6>0.9 |  mst9>0.9|mst4>0.9) %>% filter( mst8>0.9) %>% nrow() 
group1_group3_match <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter(mst3>0.9| mst7>0.9| mst6>0.9 |  mst9>0.9|mst4>0.9) %>% filter( mst5>0.9|mst10>0.9) %>% nrow() 
group1_meta_match <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter(mst3>0.9| mst7>0.9| mst6>0.9 |  mst9>0.9|mst4>0.9) %>%filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% nrow() 
group2_group3_match <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>%filter( mst8>0.9) %>% filter( mst5>0.9|mst10>0.9) %>% nrow() 
group2_meta_match <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>%filter( mst8>0.9) %>%filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% nrow() 
group3_meta_match <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>%filter( mst5>0.9|mst10>0.9) %>%filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% nrow() 
##==============================
group1_group2_group3_match <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter(mst3>0.9| mst7>0.9| mst6>0.9 |  mst9>0.9|mst4>0.9) %>% filter( mst8>0.9) %>% filter( mst5>0.9|mst10>0.9) %>%nrow() 
group1_group2_meta_match <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter(mst3>0.9| mst7>0.9| mst6>0.9 |  mst9>0.9|mst4>0.9) %>% filter( mst8>0.9) %>%  filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% nrow() 
group1_group3_meta_match <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter(mst3>0.9| mst7>0.9| mst6>0.9 |  mst9>0.9|mst4>0.9) %>%  filter( mst5>0.9|mst10>0.9) %>%filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>%nrow() 
group2_group3_meta_match <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter( mst8>0.9) %>%  filter( mst5>0.9|mst10>0.9) %>%filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>%nrow() 
##==============================
group1_group2_group3_match <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter(mst3>0.9| mst7>0.9| mst6>0.9 |  mst9>0.9|mst4>0.9) %>% filter( mst8>0.9) %>% filter( mst5>0.9|mst10>0.9)%>%filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>%nrow() 


##----------------
##plotting Venn triple
##----------------

plot.new()
venn.plot <- draw.triple.venn(group1_count, group2_count, group3_count, group1_group2_match,  group2_group3_match, group1_group3_match,
    group1_group2_group3_match, category = c("L.delbrueckii\ngroup1", "L.delbrueckii\ngroup2", "L.delbrueckii\ngroup3"), lty = "blank", 
    fill = c("skyblue", "pink1", "mediumorchid"))
grid.draw(venn.plot)


svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/04_venn_diagramm_ldel_tri.svg",width=6,height=4)
grid.draw(venn.plot)
dev.off()


##==================================================================Sterm

#----------------------total snps in analysis
total_tmp <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% 
  filter(species=="S. thermophilus")  %>% filter(mst1>0.9 | mst2>0.9) %>%
  filter("Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% nrow()

#----------------------in isolate
isolates_tmp <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter(mst1>0.9 | mst2>0.9) %>% nrow()
#----------------------in meta
meta_tmp <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% nrow()
##---------------------in meta or isolate
shared_tmp <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter((mst1>0.9 | mst2>0.9) , ("Lyo\n1996">0.9| "Lyo\n2012">0.9| "Lyo\n2014">0.9| "working\nstock">0.9 | "starter\nculture\n2012">0.9) | "starter\nculture\n2018">0.9 ) %>% nrow()
##----------------
##plotting Venn in meta or not
##----------------
plot.new()
venn.plot <- draw.pairwise.venn(isolates_tmp,meta_tmp, shared_tmp, c("in isolates", "only in metagenome"), lty = rep("blank", 
    2), fill = c("light blue", "grey"), alpha = rep(0.5, 2), cat.pos = c(0, 
    0), cat.dist = rep(0.025, 2))
grid.draw(venn.plot)


svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/04_venn_diagramm_sterm.svg",width=6,height=4)
grid.draw(venn.plot)
dev.off()

##==============================
##-------------mst1
mst1_count <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter(mst1>0.9 ) %>% nrow()
##-------------mst2
mst2_count <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter( mst2>0.9) %>% nrow()
#----------------------in meta
meta_tmp <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% nrow()
##==============================
mst1_mst2_match <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter(mst1>0.9 ) %>% filter( mst2>0.9) %>% nrow() 
mst1_meta_match <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter(mst1>0.9 ) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% nrow() 
mst2_meta_match <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter(mst2>0.9 ) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% nrow() 
##==============================
mst1_mst2_meta_match <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter( mst1>0.9) %>% filter(mst2>0.9 ) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% nrow() 


##----------------
##plotting Venn di
##----------------


plot.new()
venn.plot <- draw.pairwise.venn(mst1_count,mst2_count, mst1_mst2_match, c("S.thermophilus\nmst1", "S.thermophilus\nmst2"), lty = rep("blank", 
    2), fill = c("light blue","pink"), alpha = rep(0.5, 2), cat.pos = c(0, 
    0), cat.dist = rep(0.025, 2))
grid.draw(venn.plot)

svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/04_venn_diagramm_sterm_di_strains.svg",width=6,height=4)
grid.draw(venn.plot)
dev.off()

##=====================================
##frequency plot
##=====================================


##==================================================Streptococcus thermophilus===================================================================


##-------------mst1
mst1_only_frequency <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter(mst1>0.9 )  %>% filter( mst2<=0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% gather(.,sample,Frequency,"Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018") %>% filter(Frequency>0.05) %>% add_column(.,group="S.thermophilus\nmst1")
##-------------mst2
mst2_only_frequency <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter(mst1<=0.9 )  %>% filter( mst2>0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% gather(.,sample,Frequency,"Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018")  %>% filter(Frequency>0.05) %>% add_column(.,group="S.thermophilus\nmst2")
##-------------not explained
not_explained_frequency <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter(mst1<=0.9 )  %>% filter( mst2<=0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% gather(.,sample,Frequency,"Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018")  %>% filter(Frequency>0.05) %>% add_column(.,group="not\nexplained")

all_sterm_frequency <- rbind(mst1_only_frequency,mst2_only_frequency,not_explained_frequency)

MAG_frequency <- 1-(median(mst1_only_frequency$Frequency)+median(mst2_only_frequency$Frequency))
##----------------
##plotting frequency
##----------------

all_sterm_frequency$group = factor(all_sterm_frequency$group, levels=c("S.thermophilus\nmst1","S.thermophilus\nmst2","not\nexplained"))

sterm_freqeunyPlot <- ggplot(all_sterm_frequency,aes(x=group,y=Frequency))+ geom_violin(aes(color=group,fill=group))+geom_boxplot(aes(alpha = 0.2),outlier.shape=NA)+theme_classic()+
      scale_fill_manual(values=c("light blue","pink","grey"))+
      scale_color_manual(values=c("light blue","pink","grey"))+
  labs(y="alternative allele\nfrequeny")+
    theme(legend.position="none",
       # axis.text.x = element_text(size=8),
       axis.title.x = element_blank())

svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/04_frequeny_sterm.svg",width=6,height=4)
sterm_freqeunyPlot
dev.off()

##==================================================Lactobacillus delbrueckii===================================================================
##-------------group1
group1_count_snps <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(species=="L. delbrueckii")  %>% filter(mst6>0.9) %>% filter(mst7<0.9& mst4<0.9&mst3<0.9 & mst9<0.9&mst8<0.9& mst5<0.9&mst10<0.9)
group1_count <- group1_count_snps %>% nrow()

##-------------group2
group2_count_snps <-  sample_relative_wide %>%replace(is.na(.), 0) %>% filter(species=="L. delbrueckii")  %>% filter(mst7>0.9| mst4>0.9|mst3>0.9 | mst9>0.9) %>%filter(mst6<0.9&mst8<0.9& mst5<0.9&mst10<0.9)
group2_count <- group2_count_snps %>% nrow()

##-------------group3
group3_count_snps <-  sample_relative_wide %>%replace(is.na(.), 0)%>% filter(species=="L. delbrueckii")  %>%  filter(mst7<0.9& mst4<0.9&mst3<0.9 & mst9<0.9&mst6<0.9 ) %>%filter(mst8>0.9|mst5>0.9|mst10>0.9) 
group3_count <- group3_count_snps %>% nrow()

##-------------explained by multiple
allGroups_count_snps <-  sample_relative_wide %>%replace(is.na(.), 0)%>% filter(species=="L. delbrueckii")  %>%  filter(((mst6>0.9)+(mst7>0.9| mst4>0.9|mst3>0.9 | mst9>0.9)+(mst8>0.9|mst5>0.9|mst10>0.9))>1)
allGroups_count_snps <-  sample_relative_wide %>%replace(is.na(.), 0)%>% filter(species=="L. delbrueckii")  %>%  filter((mst6>0.9)&(mst7>0.9| mst4>0.9|mst3>0.9|mst9>0.9))
allGroups_count_snps <-  sample_relative_wide %>%replace(is.na(.), 0)%>% filter(species=="L. delbrueckii")  %>%  filter((mst6>0.9)&(mst8>0.9|mst5>0.9|mst10>0.9))
allGroups_count_snps <-  sample_relative_wide %>%replace(is.na(.), 0)%>% filter(species=="L. delbrueckii")  %>%  filter((mst7>0.9| mst4>0.9|mst3>0.9|mst9>0.9)&(mst8>0.9|mst5>0.9|mst10>0.9))
allGroups_count_snps <-  sample_relative_wide %>%replace(is.na(.), 0)%>% filter(species=="L. delbrueckii")  %>%  filter((mst6>0.9)&(mst7>0.9| mst4>0.9|mst3>0.9|mst9>0.9)&(mst8>0.9|mst5>0.9|mst10>0.9))
###non at all


##-------------explained by none
none_count_snps <-  sample_relative_wide %>%replace(is.na(.), 0)%>% filter(species=="L. delbrueckii")  %>%  filter((mst6<0.9)&(mst7<0.9&mst4<0.9&mst3<0.9&mst9<0.9)&(mst8<0.9&mst5<0.9&mst10<0.9))
 
allGroups_count_snps <-  sample_relative_wide %>%replace(is.na(.), 0)%>% filter(species=="L. delbrueckii")  %>%  filter((mst6>0.9)&(mst7>0.9| mst4>0.9|mst3>0.9))
allGroups_count_snps <-  sample_relative_wide %>%replace(is.na(.), 0)%>% filter(species=="L. delbrueckii")  %>%  filter((mst6>0.9)&(mst8>0.9|mst5>0.9|mst10>0.9))


#----------------------in meta
meta_tmp <-  sample_relative_wide %>%replace(is.na(.), 0) %>% filter(species=="L. delbrueckii")  %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% nrow()

##-------------group1

 # %>% filter(core=="core")
group1_only_frequency <- sample_relative_wide  %>%replace(is.na(.), 0)   %>% filter(species=="L. delbrueckii")  %>% filter(species=="L. delbrueckii")  %>% filter(mst3>0.9| mst7>0.9| mst6>0.9 |  mst9>0.9|mst4>0.9)%>% filter( mst8<=0.9) %>% filter(mst5<=0.9&mst10<=0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% gather(.,sample,Frequency,"Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018") %>% filter(Frequency>0.0) %>% add_column(.,group="L.delbrueckii\ngroup1")
##-------------group2
group2_only_frequency <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(species=="L. delbrueckii")  %>% filter(species=="L. delbrueckii")  %>% filter(mst3<=0.9& mst7<=0.9&mst6<=0.9 &  mst9<=0.9&mst4<=0.9)%>% filter( mst8>0.9) %>% filter( mst5<=0.9&mst10<=0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% gather(.,sample,Frequency,"Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018")  %>% add_column(.,group="L.delbrueckii\ngroup2")
##-------------group3
group3_only_frequency <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(species=="L. delbrueckii")  %>% filter(species=="L. delbrueckii")  %>% filter(mst3<=0.9& mst7<=0.9&mst6<=0.9 &  mst9<=0.9&mst4<=0.9)%>% filter( mst8<=0.9) %>% filter( mst5>0.9|mst10>0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% gather(.,sample,Frequency,"Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018")  %>% filter(Frequency>0.0)%>% add_column(.,group="L.delbrueckii\ngroup3")

##-------------not explained
not_explained_frequency_ldel <- sample_relative_wide  %>%replace(is.na(.), 0) %>% filter(species=="L. delbrueckii")  %>% filter(species=="L. delbrueckii")  %>% filter(mst3<=0.9& mst7<=0.9&mst6<=0.9 &  mst9<=0.9&mst4<=0.9)%>% filter( mst8<=0.9) %>% filter( mst5<=0.9&mst10<=0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% gather(.,sample,Frequency,"Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018") %>% filter(Frequency>0.0) %>% add_column(.,group="not\nexplained")


all_ldel_frequency <- rbind(group1_only_frequency,group2_only_frequency,not_explained_frequency_ldel,group3_only_frequency)

# MAG_frequency <- 1-(median(mst1_only_frequency$Frequency)+median(mst2_only_frequency$Frequency))
##----------------
##plotting frequency
##----------------

all_ldel_frequency$group = factor(all_ldel_frequency$group, levels=c("L.delbrueckii\ngroup1","L.delbrueckii\ngroup2","L.delbrueckii\ngroup3","not\nexplained"))
table(all_ldel_frequency$group)

ldel_freqeunyPlot <- ggplot(all_ldel_frequency,aes(x=group,y=Frequency))+ geom_violin(aes(color=group,fill=group))+geom_boxplot(aes(alpha = 0.2),outlier.shape=NA)+theme_classic()+
      scale_fill_manual(values=c("skyblue", "pink1", "mediumorchid","grey"))+
      scale_color_manual(values=c("skyblue", "pink1", "mediumorchid","grey"))+
  lims(y=c(0,0.2))+
  labs(y="alternative allele\nfrequeny")+
    theme(legend.position="none",
       # axis.text.x = element_text(size=8),
       axis.title.x = element_blank())
ldel_freqeunyPlot
svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/04_frequeny_ldel.svg",width=6,height=4)
ldel_freqeunyPlot
# dev.off()

```

##### strain count
needs previous chunk to run
```{r}
library(robustbase)
library(VennDiagram)
##==================================================Streptococcus thermophilus===================================================================
colnames(sample_relative_wide)
sample_relative_wide_snpsUsed <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")    %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 | "experiment_A">0.05| "experiment_B">0.05| "experiment_C">0.05 | "experiment_D">0.05 | "experiment_E">0.05)

colnames(sample_relative_wide_snpsUsed)
sample_relative_wide_snpsUsed_lin1 <- sample_relative_wide_snpsUsed %>% filter(`24853`>0.8| `24798`>0.8| `13493`>0.8| `13500`>0.8 | `24737`>0.8 | `24854`>0.8| `13491`>0.8 | `13492`>0.8 ) %>% add_column(explained="lin1")
sample_relative_wide_snpsUsed_lin2 <- sample_relative_wide_snpsUsed %>% filter(`S72`>0.8| `24855`>0.8| `13494`>0.8)%>% add_column(explained="lin2")
sample_relative_wide_snpsUsed_lin3 <- sample_relative_wide_snpsUsed %>% filter(`S50`>0.8| `24740`>0.8| `24738`>0.8| `13499`>0.8 )%>% add_column(explained="lin3")
sample_relative_wide_snpsUsed_lin4 <- sample_relative_wide_snpsUsed %>% filter(`24739`>0.8| `13497`>0.8| `13496`>0.8| `13495`>0.8 )%>% add_column(explained="lin4")
allsites <- c(sample_relative_wide_snpsUsed_lin1$site, sample_relative_wide_snpsUsed_lin2$site, sample_relative_wide_snpsUsed_lin3$site,sample_relative_wide_snpsUsed_lin4$site)

##----------------------
# Chart venn diagramm
##----------------------
temp <-venn.diagram(
  x = list(sample_relative_wide_snpsUsed_lin1$site, sample_relative_wide_snpsUsed_lin2$site, sample_relative_wide_snpsUsed_lin3$site,sample_relative_wide_snpsUsed_lin4$site),
  category.names = c("lin 1" , "lin 2 " , "lin 3", "lin 4"),
  filename = NULL
)

plot.new() 

grid.draw(temp)
# 
# pdf("testpdf", width = 14, height = 7)
# 
# grid.draw(temp)
# 
# dev.off()

grid.draw(temp)

allsites[duplicated(allsites)]
sum(duplicated(allsites))
sum(!duplicated(allsites))

sum(table(allsites)>1)
sum(table(allsites)==1)

duplictednames <- names(table(allsites)[(table(allsites)>1)])

##----------------------
# Chart venn diagramm
##----------------------

sample_relative_wide_snpsUsed_new_01 <- rbind(sample_relative_wide_snpsUsed_lin1,sample_relative_wide_snpsUsed_lin2,sample_relative_wide_snpsUsed_lin3,sample_relative_wide_snpsUsed_lin4)
sample_relative_wide_snpsUsed_new_02 <- sample_relative_wide_snpsUsed_new_01 %>% filter(!site %in%duplictednames) #lineage specific snps
table(sample_relative_wide_snpsUsed_new_02$explained)
sample_relative_wide_snpsUsed_new_temp <- sample_relative_wide_snpsUsed %>% filter(site %in%duplictednames)%>% add_column(explained="multiple") ##duplicated sites
sample_relative_wide_snpsUsed_new_03 <- rbind(sample_relative_wide_snpsUsed_new_02,sample_relative_wide_snpsUsed_new_temp) ##all explained sites
sample_relative_wide_snpsUsed_tmp <-  sample_relative_wide_snpsUsed %>% filter(!site %in%sample_relative_wide_snpsUsed_new_03$site) %>% add_column(explained="not explained")
sample_relative_wide_snpsUsed_final <- rbind(sample_relative_wide_snpsUsed_new_03,sample_relative_wide_snpsUsed_tmp) ##all explained sites


##EXKURSION LINEAGE SPECIFIC DUPLICATIONS
TMP <- sample_relative_wide_snpsUsed %>% filter(site %in%duplictednames)
TMP$e

short_explained <- sample_relative_wide_snpsUsed_new_01 %>%filter(site %in%duplictednames) %>%  select(c("site","species","explained")) %>% mutate(abundance = explained) %>% spread(., explained, abundance)
short_explained$explained <- paste(short_explained$lin1,short_explained$lin2,short_explained$lin3,short_explained$lin4,sep="_")
short_explained_multiple <- short_explained%>% select("site","explained")
sample_relative_wide_snpsUsed_new_temp <- sample_relative_wide_snpsUsed %>% filter(site %in%duplictednames) ##duplicated sites
sample_relative_wide_snpsUsed_new_temp_02 <- merge(sample_relative_wide_snpsUsed_new_temp,short_explained_multiple,by="site")
sample_relative_wide_snpsUsed_new_03 <- rbind(sample_relative_wide_snpsUsed_new_02,sample_relative_wide_snpsUsed_new_temp_02) ##all explained sites
sample_relative_wide_snpsUsed_tmp <-  sample_relative_wide_snpsUsed %>% filter(!site %in%sample_relative_wide_snpsUsed_new_03$site) %>% add_column(explained="not explained")
sample_relative_wide_snpsUsed_final <- rbind(sample_relative_wide_snpsUsed_new_03,sample_relative_wide_snpsUsed_tmp) ##all explained sites


totalSNPS <- nrow(sample_relative_wide_snpsUsed_final)
100*(table(sample_relative_wide_snpsUsed_final$explained)/totalSNPS)
table(sample_relative_wide_snpsUsed_final$explained)
##----------------------
# boxplot
##----------------------
sample_relative_wide_snpsUsed_final_long <- sample_relative_wide_snpsUsed_final %>% select( "site","Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","experiment_A","experiment_B","experiment_C","experiment_D","experiment_E","explained")%>% gather(.,sample,Median,"Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","experiment_A","experiment_B","experiment_C","experiment_D","experiment_E") 


ggplot(sample_relative_wide_snpsUsed_final_long,aes(x=sample,y=Median))+geom_boxplot()+theme_classic()+facet_wrap(~explained)

#remove zeroes
sample_relative_wide_snpsUsed_final_long_woZeros <- sample_relative_wide_snpsUsed_final_long %>% filter(Median>0.1)
ggplot(sample_relative_wide_snpsUsed_final_long_woZeros,aes(x=sample,y=Median))+geom_boxplot()+theme_classic()+facet_wrap(~explained)


##----------------------
# lineplot
##----------------------
sample_relative_wide_snpsUsed_final_long_woZeros$sample = factor(sample_relative_wide_snpsUsed_final_long_woZeros$sample, levels=c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2","experiment_A","experiment_B","experiment_C","experiment_D","experiment_E"))

sample_relative_wide_snpsUsed_final_long_woZeros$explained = factor(sample_relative_wide_snpsUsed_final_long_woZeros$explained, levels=c("lin1","lin2","lin3","lin4","not explained","lin1_lin2_NA_NA","NA_NA_lin3_lin4","lin1_lin2_lin3_lin4","lin1_lin2_NA_lin4","lin1_NA_lin3_lin4","lin1_NA_lin3_NA","lin1_NA_NA_lin4","NA_lin2_lin3_NA","NA_lin2_NA_lin4","NA_lin2_lin3_lin4"))


# ggplot(sample_relative_wide_snpsUsed_final_long_woZeros,aes(x=sample,y=Median,group=site))+geom_line(color="red",size=0.2, alpha=.1)+theme_classic()


pExplainted <- ggplot(sample_relative_wide_snpsUsed_final_long_woZeros,aes(x=sample,y=Median,group=site,color=explained,fill=explained))+geom_line(alpha=.5,size=0.5)+theme_classic()+facet_wrap(~explained)+theme(legend.position = "none",axis.text.x = element_text(angle = 75, hjust = 1,size=9))

table(sample_relative_wide_snpsUsed_final_long_woZeros$explained)
pExplainted


  # svg("~/Desktop/mid_thesis/report/figures/20200430/chapter1/Supplement_SNVs_explained_by_isolates..svg",width=7,height=4.5)
png("~/Desktop/supp_altNucFre_lineages.png", width = 3400, height = 2600,res=300)
# 
# p3+theme(axis.title = element_blank(),axis.text.x = element_blank())
pExplainted
dev.off()


sample_relative_wide_snpsUsed_final_long$explained_02 <- revalue(sample_relative_wide_snpsUsed_final_long$explained, c("lin1"="explained by isolates", "lin2"="explained by isolates","lin3"="explained by isolates","lin4"="explained by isolates","multiple"="explained by isolates","not explained"="not explained by isolates"))

pExplainted <- ggplot(sample_relative_wide_snpsUsed_final_long,aes(x=sample,y=Median,group=site,color=explained_02,fill=explained_02))+geom_line(size=0.5,alpha=0.5)+theme_classic()+facet_wrap(~explained_02)+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),legend.position = "none")+
    labs("",
         x="",
         y="Alternative allele frequency")
pExplainted

pExplainted <- ggplot(sample_relative_wide_snpsUsed_final_long,aes(x=sample,y=Median,group=site,color=explained_02,fill=explained_02))+geom_line(size=0.5,alpha=0.5)+theme_classic()+facet_wrap(~explained_02)+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),legend.position = "none")+
    labs("",
         x="",
         y="Alternative allele frequency")
pExplainted


  svg("~/Desktop/mid_thesis/report/figures/20200430/chapter1/Supplement_SNVs_explained_by_isolates..svg",width=7,height=4.5)
#    # png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 1000, height = 800,res=300)
# 
# p3+theme(axis.title = element_blank(),axis.text.x = element_blank())
pExplainted

dev.off()

##----------------------
# where do these funky linkes (~recombinatnts) locate on the genome
##----------------------

sample_relative_wide_snpsUsed_final_long_recombinants <- sample_relative_wide_snpsUsed_final_long %>% filter(explained %in% c("lin1_lin2_NA_lin4","lin1_NA_lin3_lin4","lin1_NA_lin3_NA","lin1_NA_NA_lin4","NA_lin2_lin3_NA","NA_lin2_NA_lin4","NA_lin2_lin3_lin4")) %>% select("site","explained") %>% unique() 

sample_relative_wide_snpsUsed_final_long_recombinants <- sample_relative_wide_snpsUsed_final_long %>% select("site","explained") %>% unique() 


sample_relative_wide_snpsUsed_final_long_recombinants$location <- as.numeric(str_split_fixed(sample_relative_wide_snpsUsed_final_long_recombinants$site, "_", 4)[,2])
sample_relative_wide_snpsUsed_final_long_recombinants$explained = factor(sample_relative_wide_snpsUsed_final_long_recombinants$explained, levels=c("lin1","lin2","lin3","lin4","not explained","lin1_lin2_NA_NA","NA_NA_lin3_lin4","lin1_lin2_lin3_lin4","NA_lin2_lin3_lin4","lin1_lin2_NA_lin4","lin1_NA_lin3_lin4","lin1_NA_lin3_NA","lin1_NA_NA_lin4","NA_lin2_lin3_NA","NA_lin2_NA_lin4"))


sample_relative_wide_snpsUsed_final_long_recombinants$explained_2 <- revalue(sample_relative_wide_snpsUsed_final_long_recombinants$explained, c("lin1_lin2_NA_NA"="lin1_lin2_NA_NA=~evolved_after_split","NA_NA_lin3_lin4"="NA_NA_lin3_lin4=~evolved_after_split","lin1_lin2_lin3_lin4"="lin1_lin2_lin3_lin4=polishingERRORs","NA_lin2_lin3_lin4"="NA_lin2_lin3_lin4=lost_in_lin1","lin1_NA_lin3_lin4"="lin1_NA_lin3_lin4=lost_in_lin2","lin1_lin2_NA_lin4"="lin1_lin2_NA_lin4=lost_in_lin3","lin1_NA_lin3_NA"="lin1_NA_lin3_NA=recombination","lin1_NA_NA_lin4"="lin1_NA_NA_lin4=recombination","NA_lin2_lin3_NA"="NA_lin2_lin3_NA=recombination","NA_lin2_NA_lin4"="NA_lin2_NA_lin4=recombination"))

# ggplot(sample_relative_wide_snpsUsed_final_long_recombinants,aes(x=location,color=explained,fill=explained))+geom_density()+facet_wrap(~explained,scales = "free_y")+theme_classic()
plocation <- ggplot(sample_relative_wide_snpsUsed_final_long_recombinants,aes(x=location,color=explained,fill=explained))+geom_histogram()+facet_wrap(~explained_2,scales = "free_y")+theme_classic()+labs(title="location of SNVs coming from different ",x="genomic location")+theme(legend.position = "none")

plocation


  # svg("~/Desktop/mid_thesis/report/figures/20200430/chapter1/Supplement_SNVs_explained_by_isolates..svg",width=7,height=4.5)
png("~/Desktop/supp_altNucFre_lineages_location.png", width = 3400, height = 2600,res=300)
# 
# p3+theme(axis.title = element_blank(),axis.text.x = element_blank())
plocation
dev.off()

##----------------------
# not explained
##----------------------
table(sample_relative_wide_snpsUsed_final_long$explained)
sample_relative_wide_snpsUsed_final_long_notExplained <- sample_relative_wide_snpsUsed_final_long %>% filter(explained=="not explained")

ggplot(sample_relative_wide_snpsUsed_final_long_notExplained,aes(x=Median))+geom_density()+facet_wrap(~sample)+theme_classic()


###nomany multiallelic sites
sitesss <- str_split_fixed(sample_relative_wide_snpsUsed_final_long_notExplained$site, "_", 4)[,2]
length(sitesss)/6
unique(sitesss) %>% length()

sitesss <- str_split_fixed(sample_relative_wide_snpsUsed_final_long$site, "_", 4)[,2]
length(sitesss) /6
unique(sitesss) %>% length()

##----------------------
# meansss and medianss
##----------------------
library(patchwork)
# mediannssss <-  aggregate(.~sample+explained, data=sample_relative_wide_snpsUsed_final_long[,c("explained","sample","Median")], median, na.rm=TRUE)%>% filter(explained!="multiple")
# meanssss <-  aggregate(.~sample+explained, data=sample_relative_wide_snpsUsed_final_long[,c("explained","sample","Median")], median, na.rm=TRUE) %>% filter(explained!="multiple")
# 
# mediansplot <- ggplot(mediannssss,aes(x=sample,y=Median,color=explained,fill=explained))+geom_bar(stat="identity")+theme_classic()+theme(legend.position = "none")
# meansssplot <- ggplot(meanssss,aes(x=sample,y=Median,color=explained,fill=explained))+geom_bar(stat="identity")+theme_classic()
# 
# mediansplot+meansssplot

##---without zeros

mediannssss <-  aggregate(.~sample+explained, data=sample_relative_wide_snpsUsed_final_long_woZeros[,c("explained","sample","Median")], median, na.rm=TRUE)%>% filter(explained!="multiple")
meanssss <-  aggregate(.~sample+explained, data=sample_relative_wide_snpsUsed_final_long_woZeros[,c("explained","sample","Median")], median, na.rm=TRUE) %>% filter(explained!="multiple")

mediansplot <- ggplot(mediannssss,aes(x=sample,y=Median,color=explained,fill=explained))+geom_bar(stat="identity")+theme_classic()+theme(legend.position = "none")
meansssplot <- ggplot(meanssss,aes(x=sample,y=Median,color=explained,fill=explained))+geom_bar(stat="identity")+theme_classic()

mediansplot+meansssplot

##---------------------------------
##lineage abundance
##---------------------------------
ratio_larger <- mediannssss %>% filter(explained=="NA_NA_lin3_lin4") %>% select("sample","Median")  %>% dplyr::rename(Overall_ratio = Median) 
ratio_larger_2 <- mediannssss %>% filter(explained=="lin2") %>% select("sample","Median")  %>% dplyr::rename(rel_lin2 = Median) 
ratio_larger_3 <- mediannssss %>% filter(explained=="lin3") %>% select("sample","Median")  %>% dplyr::rename(rel_lin3 = Median) 
# ratio_larger_4 <- mediannssss %>% filter(explained=="lin4") %>% select("sample","Median")  %>% dplyr::rename(rel_lin4 = Median) 

ratio_together_1 <- merge(ratio_larger,ratio_larger_2,by="sample",all.x = TRUE)
ratio_together_2 <- merge(ratio_together_1,ratio_larger_3,by="sample",all.x = TRUE)
ratio_together_2$lin1 <- (1-ratio_together_2$Overall_ratio)*(1-ratio_together_2$rel_lin2)
ratio_together_2$lin2 <- (1-ratio_together_2$Overall_ratio)*(ratio_together_2$rel_lin2)
ratio_together_2$lin3 <- (ratio_together_2$Overall_ratio)*(ratio_together_2$rel_lin3)
ratio_together_2$lin4 <- (ratio_together_2$Overall_ratio)*(1-ratio_together_2$rel_lin3)
# ratio_together_2$Unknown <- 1-(ratio_together_2$lin1+ratio_together_2$lin2+ratio_together_2$lin3+ratio_together_2$lin4)

# ratio_together_final <- ratio_together_2 %>% select(sample,lin1,lin2,lin3,lin4) %>% gather(.,lineage,"Relative abundance","lin1","lin2","lin3","lin4")
ratio_together_final <- ratio_together_2 %>% select(sample,lin4,lin3,lin2,lin1) %>% gather(.,lineage,"Relative abundance","lin4","lin3","lin2","lin1")


ratio_together_final$lineage = factor(ratio_together_final$lineage, levels=(c("lin4","lin3","lin2","lin1")))


# colorsss <- c("#0000FF","#6699FF","#99CCFF","#00FFFF")
colorsss <- c("#99CCFF","#00FFFF","#0000FF","#6699FF")


plot_strain_abundance <- ggplot(ratio_together_final,aes(x=sample,y=`Relative abundance`,color=lineage,fill=lineage))+geom_bar(stat="identity")+theme_classic()+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9))+
    labs(x="")+scale_color_manual(values=colorsss)+scale_fill_manual(values=colorsss)+theme(legend.position = "none")
plot_strain_abundance
# png("~/Desktop/supp_strain_Abundance.png", width = 2800, height = 2200,res=300)
svg("~/Desktop/Projects/2019_RMK202_analysis/plot/strain_rel_abundance.svg",width=6,height=3.5)

# p3+theme(axis.title = element_blank(),axis.text.x = element_blank())
plot_strain_abundance
dev.off()

##---------------------------------
##Sample medians calulation per group
##---------------------------------
##-------------mst1
mst1_Median <-  sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter(mst1>0.9 )  %>% filter( mst2<=0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% select("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018") %>% as.matrix %>%colMedians()
##-------------mst2
mst2_Median <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter(mst1<=0.9 )  %>% filter( mst2>0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% select("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018") %>% as.matrix %>%colMedians()
##-------------not explained
not_explained_Median <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter(mst1<=0.9 )  %>% filter( mst2<=0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% select("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018") %>% as.matrix %>%colMedians()

##---------------------------------
##Gather data
##---------------------------------
relAbundance_isolates_sterm <- rbind(mst1_Median,mst2_Median,not_explained_Median)
S_thermophilusMAG_rmk202 <- 1-(relAbundance_isolates_sterm[1,]+relAbundance_isolates_sterm[2,])
relAbundance_isolates_sterm <- rbind(mst1_Median,mst2_Median,S_thermophilusMAG_rmk202)
relAbundance_isolates_sterm_long <-  rbind(mst1_Median,mst2_Median,S_thermophilusMAG_rmk202)%>% as.data.frame()%>% add_column(group=rownames(relAbundance_isolates_sterm))%>% gather(.,sample,Median,"Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018") 


##---------------------------------
##stacked barplot
##---------------------------------
table(relAbundance_isolates_sterm_long$group)
relAbundance_isolates_sterm_long$group <- revalue(relAbundance_isolates_sterm_long$group, c("mst1_Median"="S.thermophilus\nmst1", "mst2_Median"="S.thermophilus\nmst2", "S_thermophilusMAG_rmk202"="S.thermophilus\nMAG RMK202"))
relAbundance_isolates_sterm_long$group = factor(relAbundance_isolates_sterm_long$group, levels=rev(c("S.thermophilus\nmst1","S.thermophilus\nmst2","S.thermophilus\nMAG RMK202")))

ggplot(relAbundance_isolates_sterm_long,aes(x=sample,y=Median,group=group,color=group,fill=group))+geom_bar(stat="identity")+theme_classic()+
  scale_color_manual(values=rev(c("light blue","pink","grey")))+
 scale_fill_manual(values=rev(c("light blue","pink","grey")))


##==================================================Lactobacillus delbrueckii===================================================================
library(naniar)

##---------------------------------
##Sample medians calulation per group
##---------------------------------
# %>% filter(core=="core")
group1_Median <- sample_relative_wide  %>%replace(is.na(.), 0)   %>% filter(species=="L. delbrueckii")  %>% filter(species=="L. delbrueckii")  %>% filter(mst3>0.9| mst7>0.9| mst6>0.9 |  mst9>0.9|mst4>0.9)%>% filter( mst8<=0.9) %>% filter(mst5<=0.9&mst10<=0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 )%>% select("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018") %>%  replace_with_na_at(.vars = c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018"),condition = ~.x == 0)%>% as.matrix()%>% colMedians(na.rm = TRUE)   %>%replace(is.na(.), 0) 


##-------------group2
group2_Median <- sample_relative_wide  %>%replace(is.na(.), 0)   %>% filter(species=="L. delbrueckii")  %>% filter(species=="L. delbrueckii")  %>% filter(mst3<=0.9& mst7<=0.9&mst6<=0.9 &  mst9<=0.9&mst4<=0.9)%>% filter( mst8>0.9) %>% filter( mst5<=0.9&mst10<=0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% select("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018") %>%  replace_with_na_at(.vars = c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018"),condition = ~.x == 0)%>% as.matrix()%>% colMedians(na.rm = TRUE)   %>%replace(is.na(.), 0) 
##-------------group3
group3_Median <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(species=="L. delbrueckii")  %>% filter(species=="L. delbrueckii")  %>% filter(mst3<=0.9& mst7<=0.9&mst6<=0.9 &  mst9<=0.9&mst4<=0.9)%>% filter( mst8<=0.9) %>% filter( mst5>0.9|mst10>0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% select("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018") %>%  replace_with_na_at(.vars = c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018"),condition = ~.x == 0)%>% as.matrix()%>% colMedians(na.rm = TRUE)   %>%replace(is.na(.), 0) 
##-------------not explained
not_explained_Median_ldel <- sample_relative_wide  %>%replace(is.na(.), 0)   %>% filter(species=="L. delbrueckii")  %>% filter(species=="L. delbrueckii")  %>% filter(mst3<=0.9& mst7<=0.9&mst6<=0.9 &  mst9<=0.9&mst4<=0.9)%>% filter( mst8<=0.9) %>% filter( mst5<=0.9&mst10<=0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% select("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018") %>%  replace_with_na_at(.vars = c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018"),condition = ~.x == 0)%>% as.matrix()%>% colMedians(na.rm = TRUE)   %>%replace(is.na(.), 0) 


##---------------------------------
##Gather data
##---------------------------------

relAbundance_isolates_ldel <- rbind(group1_Median,group2_Median,group3_Median,not_explained_Median_ldel)
Ldel_MAG_rmk202 <- 1-(colSums(relAbundance_isolates_ldel))
group1_Median_new <- Ldel_MAG_rmk202+group1_Median
relAbundance_isolates_ldel <- rbind(group1_Median_new,group2_Median,group3_Median,not_explained_Median_ldel)
relAbundance_isolates_ldel_long <- rbind(group1_Median_new,group2_Median,group3_Median,not_explained_Median_ldel) %>% as.data.frame()%>% add_column(group=rownames(relAbundance_isolates_ldel))%>% gather(.,sample,Median,"Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018") 

# colSums(relAbundance_isolates_ldel)

##---------------------------------
##stacked barplot
##---------------------------------

relAbundance_isolates_ldel_long$group <- revalue(relAbundance_isolates_ldel_long$group, c("group1_Median_new"="L.delbrueckii\ngroup1", "group2_Median"="L.delbrueckii\ngroup2", "group3_Median"="L.delbrueckii\ngroup3", "not_explained_Median_ldel"="not\nexplained"))
relAbundance_isolates_ldel_long$group = factor(relAbundance_isolates_ldel_long$group, levels=rev(c("L.delbrueckii\ngroup1","L.delbrueckii\ngroup2","L.delbrueckii\ngroup3","not\nexplained")))

ggplot(relAbundance_isolates_ldel_long,aes(x=sample,y=Median,group=group,color=group,fill=group))+geom_bar(stat="identity")+theme_classic()+
      scale_fill_manual(values=rev(c("skyblue", "pink1", "mediumorchid","grey")))+
      scale_color_manual(values=rev(c("skyblue", "pink1", "mediumorchid","grey")))


```

#####Muller plot strains

good [blog](https://cran.r-project.org/web/packages/ggmuller/vignettes/ggmuller.html)
or [this blog](https://cran.r-project.org/web/packages/MullerPlot/vignettes/MullerPlot.html)


```{r}
library(ggmuller)
library(MullerPlot)
library(viridis)

##==================================================Streptococcus thermophilus===================================================================
library(MullerPlot)

parent_identity_file <- as.matrix(data.frame(names=rownames(relAbundance_isolates_sterm),parents=NA,colors=c("light blue","pink","grey")))
Final_Abundance_Sterm_matrix <- relAbundance_isolates_sterm 
colnames(Final_Abundance_Sterm_matrix) <- 1:6
layout(matrix(c(1,2), nrow = 1), widths = c(0.8, 0.2))
par(mar=c(3, 3, 1, 0))
m.plot <- Muller.plot(attributes = parent_identity_file, population.data = Final_Abundance_Sterm_matrix,
                      data.method = "table", time.interval.method = "equal",mgp=c(2,0.5,0))

##==================================================Lactobacillus delbrueckii===================================================================

library(MullerPlot)

parent_identity_file <- as.matrix(data.frame(names=rownames(relAbundance_isolates_ldel),parents=NA,colors=c("skyblue", "pink1", "mediumorchid","grey")))
Final_Abundance_Ldel_matrix <- relAbundance_isolates_ldel
colnames(Final_Abundance_Ldel_matrix) <- 1:6
layout(matrix(c(1,2), nrow = 1), widths = c(0.8, 0.2))
par(mar=c(3, 3, 1, 0))
m.plot <- Muller.plot(attributes = parent_identity_file, population.data = Final_Abundance_Ldel_matrix,
                      data.method = "table", time.interval.method = "equal",mgp=c(2,0.5,0))


###------------------------------------------------------------
##all Bacteria together
###------------------------------------------------------------


library(readr)
coverage_rmk202 <- read_delim("~/Desktop/Projects/2019_Pilotplan/coverage_rmk202.tsv","\t", escape_double = FALSE, col_names = c("rowNumber","sample","species","Median_Coverage","total_coverage","percent_coverage"),trim_ws = TRUE)

coverage_rmk202$sample <- revalue(coverage_rmk202$sample, c("lyo202_96"="Lyo\n1996","Lyo_202_2012"="Lyo\n2012", "Lyo_202_2014"="Lyo\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018"))

##----------------------percent calculation
coverage_rmk202_percent <- coverage_rmk202 %>% filter(species %in% c("S_thermophilus_RMK202","L_delbrueckii_RMK202")) %>% group_by(., sample) %>% mutate(percent_bacteria = Median_Coverage/sum(Median_Coverage))

##---------------------plot bacterial diversity

coverage_rmk202_percent$sample = factor(coverage_rmk202_percent$sample, levels=c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018"))
all_colours <- rev(c("#EB4D4D","#10B552") )


PrelAbundance <-  ggplot( data = coverage_rmk202_percent,aes(y = percent_bacteria, x = sample, group=interaction(species),fill = species))+ geom_bar( stat="identity")+
    labs("",
         x="",
         y="relative abundance")+
    theme_classic()+
   scale_color_viridis(discrete=TRUE)+
   scale_fill_manual(values=all_colours)+
  # scale_fill_viridis(discrete=TRUE)+
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
        # axis.text.x = element_blank(),
          legend.position="right",
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank()
          )

  PrelAbundance
  # svg("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.svg",width=7,height=4)
   # png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 1000, height = 800,res=300)
svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/F1_relativeAbundance_speciesLevel.svg",width=7,height=4)

PrelAbundance

dev.off()


##---------------------calculate subspecies divesity

##------------------------------------------------------------------Streptococcus thermophilus
relAbundance_isolates_sterm
coverage_rmk202_percent
relAbundance_isolates_sterm_normalized <- relAbundance_isolates_sterm
for (sampleName in colnames(relAbundance_isolates_sterm)) {
  print(sampleName)
  coverage_rmk202_percent_tmp <- coverage_rmk202_percent %>% filter(sample==sampleName)  %>% filter(species=="S_thermophilus_RMK202") %>% select(percent_bacteria) %>% as.double()
  print(coverage_rmk202_percent_tmp[2])
relAbundance_isolates_sterm_normalized[,sampleName] <- relAbundance_isolates_sterm[,sampleName]*coverage_rmk202_percent_tmp[2]
  
}

##------------------------------------------------------------------Lactobacillus delbrueckii
relAbundance_isolates_ldel
coverage_rmk202_percent
relAbundance_isolates_ldel_normalized <- relAbundance_isolates_ldel
for (sampleName in colnames(relAbundance_isolates_ldel)) {
  print(sampleName)
    coverage_rmk202_percent_tmp <- coverage_rmk202_percent %>% filter(sample==sampleName)  %>% filter(species=="L_delbrueckii_RMK202") %>% select(percent_bacteria) %>% as.double()
  print(coverage_rmk202_percent_tmp[2])
relAbundance_isolates_ldel_normalized[,sampleName] <- relAbundance_isolates_ldel[,sampleName]*coverage_rmk202_percent_tmp[2]
  
}
##------------------------------------------------------------------together

relAbundance_isolates_togther <- rbind(relAbundance_isolates_sterm_normalized,relAbundance_isolates_ldel_normalized) %>% as.data.frame()

# relAbundance_isolates_togther$group <- revalue(rownames(relAbundance_isolates_togther),c("mst1_Median"="S.thermophilus\nmst1", "mst2_Median"="S.thermophilus\nmst2", "S_thermophilusMAG_rmk202"="S.thermophilus\nMAG RMK202","group1_Median_new"="L.delbrueckii\ngroup1", "group2_Median"="L.delbrueckii\ngroup2", "group3_Median"="L.delbrueckii\ngroup3", "not_explained_Median_ldel"="not\nexplained"))
# 
# 
# relAbundance_isolates_togther$group = factor(relAbundance_isolates_sterm_long$group, levels=(c("S.thermophilus\nmst1","S.thermophilus\nmst2","S.thermophilus\nMAG RMK202","L.delbrueckii\ngroup1","L.delbrueckii\ngroup2","L.delbrueckii\ngroup3","not\nexplained")))



parent_identity_file <- as.matrix(data.frame(names=rownames(relAbundance_isolates_togther),parents=NA,colors=c("light blue","pink","grey","skyblue", "pink1", "mediumorchid","grey")))
Final_Abundance_together_matrix <- relAbundance_isolates_togther
colnames(Final_Abundance_together_matrix) <- 1:6
layout(matrix(c(1,2), nrow = 1), widths = c(0.8, 0.2))
par(mar=c(3, 3, 1, 0))
m.plot <- Muller.plot(attributes = parent_identity_file, population.data = Final_Abundance_together_matrix,
                      data.method = "table", time.interval.method = "equal",mgp=c(2,0.5,0))


svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/F1_relativeAbundance_subspeciesLevel.svg",width=7,height=4)

m.plot <- Muller.plot(attributes = parent_identity_file, population.data = Final_Abundance_together_matrix,
                      data.method = "table", time.interval.method = "equal",mgp=c(2,0.5,0))

dev.off()
```


######nucleotide diversity over time

Here, we calculate nucleotide diversity within genes
```{r}

###------------------------------------------------------------------------new try
##--------------
##get length of core genome
##-------------
coreGeneNames_sterm_ldel <- read_delim("~/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/coreGeneNames_rmk202_renamed_all.vcf", "\t", escape_double = FALSE, col_names = c("contig","method","annotation","start","end"), trim_ws = TRUE)
table(coreGeneNames_sterm_ldel$contig)
coreGeneNames_sterm <- coreGeneNames_sterm_ldel %>% filter(contig=="CP046134") %>% summarize(GeneLength = end-start) %>% colSums()
coreGeneNames_ldel <- coreGeneNames_sterm_ldel %>% filter(contig=="CP046131") %>% summarize(GeneLength = end-start) %>% colSums() 



genome_stats <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/genomeAnalysi/all_assembly_analysis.txt", "\t", escape_double = FALSE, col_names = c("strain","contigNumber","genomeSize"),  trim_ws = TRUE)
coreGenome_sterm <- genome_stats %>% filter(strain=="S_thermophilus_mag_rmk202") %>% select(.,genomeSize) %>% as.double()
coreGenome_ldel <- genome_stats %>% filter(strain=="L_delbrueckii_mag_rmk202") %>% select(.,genomeSize) %>% as.double()



##--------------
##Snp count per sample and species
##-------------
colnames(sample_relative_long)
table(sample_relative_long$sample)

sample_relative_long_Core <- sample_relative_long %>% filter(core=="core" & Snps>0.05) 
table(sample_relative_long_Core$sample)


sample_relative_long_Core <-  gather(sample_relative_wide, sample, Snps, "Lyo\n1996","Lyo\n2012","Lyo\n2014","starter\nculture\n2012","working\nstock","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2","experiment_A","experiment_B","experiment_C","experiment_D","experiment_E", factor_key=TRUE,na.rm = TRUE) %>% filter(Snps>0.05)   %>% group_by(sample,species) %>% tally()# %>% filter(core=="core" & Snps>0.05) 
# sample_relative_long_Core$coreGenomeSize <- if_else(sample_relative_long_Core$species=="S. thermophilus",coreGeneNames_sterm,coreGeneNames_ldel) 
sample_relative_long_Core$coreGenomeSize <- if_else(sample_relative_long_Core$species=="S. thermophilus",coreGenome_sterm,coreGenome_ldel) 
CoreGenome_nucleotide_div <- sample_relative_long_Core %>% mutate(.,nucleotide_diversity=100*(n/coreGenomeSize))

table(sample_relative_long_Core$sample)
##--------------
##add metagenome coverage to plot
##-------------

library(readr)
coverage_rmk202_setForMerge <- read_delim("~/Desktop/Projects/2019_Pilotplan/coverage_rmk202.tsv","\t", escape_double = FALSE, col_names = c("rowNumber","sample","species","Median_Coverage","total_coverage","percent_coverage"),trim_ws = TRUE)%>% select(sample,species,Median_Coverage) %>% filter(species %in% c("CP046131","CP046134")) %>% mutate(species=recode(species, CP046134="S. thermophilus",CP046131="L. delbrueckii"))%>% mutate(sample=recode(sample, lyo202_96="Lyo\n1996",Lyo_202_2012="Lyo\n2012", Lyo_202_2014="Lyo\n2014", Konserve_202="working\nstock", RMK202="starter\nculture\n2012", Versand_202="starter\nculture\n2018","di_K2_6h"="cheesemaking\nday1","th_K2_8h"="cheesemaking\nday2","G1_6_18"="experiment_A","G2_6_18"="experiment_B","G3_6_18"="experiment_C","G4_6_18"="experiment_D","G5_6_18"="experiment_E")) 



table(coverage_rmk202_setForMerge$species)

CoreGenome_nucleotide_div <-  CoreGenome_nucleotide_div %>% filter(!sample %in% c("cheesemaking\nday1","cheesemaking\nday2"))

CoreGenome_nucleotide_div <- merge(CoreGenome_nucleotide_div,coverage_rmk202_setForMerge,by=c("sample","species"),all.x = TRUE)
range(CoreGenome_nucleotide_div$Median_Coverage)

median(CoreGenome_nucleotide_div$Median_Coverage)

##--------------
##plot
##-------------

  nucleotid_div_sterm <- CoreGenome_nucleotide_div %>% filter(species=="S. thermophilus")
  nucleotid_div_ldel<- CoreGenome_nucleotide_div %>% filter(species!="S. thermophilus")

  sd(nucleotid_div_sterm$nucleotide_diversity)
    sd(nucleotid_div_ldel$nucleotide_diversity)

  ##-------------nucleotide diversity in one plot
  
   ###---prepare axis 
  ylim.pH <- c(min(nucleotid_div_sterm$nucleotide_diversity),max(nucleotid_div_sterm$nucleotide_diversity)) # in this example, precipitation
  ylim.temp <- c(min(nucleotid_div_ldel$nucleotide_diversity),max(nucleotid_div_ldel$nucleotide_diversity))  # in this example, temperature

  b <- diff(ylim.pH)/diff(ylim.temp)-650
  a <- 0.50
  
  nucleotid_div_sterm <- nucleotid_div_sterm %>% filter(sample!="cheesemaking\nday1")  %>% filter(sample!="cheesemaking\nday2")
  nucleotid_div_ldel <- nucleotid_div_ldel %>% filter(sample!="cheesemaking\nday1")  %>% filter(sample!="cheesemaking\nday2")


   pall <- ggplot()+ #stat_summary(fun.y=sum, geom="line")+
     geom_point(data = nucleotid_div_sterm,aes(y = nucleotide_diversity, x = sample,size=Median_Coverage),fill="#EB4D4D",color="#EB4D4D")+
     geom_line(data = nucleotid_div_sterm,aes(y = nucleotide_diversity, x = sample,group=species),fill="#EB4D4D",color="#EB4D4D")+
     geom_point(data = nucleotid_div_ldel,aes(y = (a+nucleotide_diversity*b), x = sample,size=Median_Coverage),fill="#10B552",color="#10B552")+
     geom_line(data = nucleotid_div_ldel,aes(y = (a+nucleotide_diversity*b), x = sample,group=species),fill="#10B552",color="#10B552")+
   labs("",
         x="",
         y="nucleotide diversity [%]",
          title="")+
  theme_classic()+
     scale_x_discrete( expand = c(0, 0)) +
 scale_y_continuous(name = "S. thermophilus\n nucleotide diversity [%]", sec.axis = sec_axis(~ (. - a)/b, name = "L. delbreuckii\n nucleotide diversity [%]"))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="top",
       axis.text.y.left = element_text(color = "#EB4D4D"),
          axis.text.y.right = element_text(color="#10B552"),
          axis.title.y.left = element_text(color="#EB4D4D"),
          axis.title.y.right = element_text(color="#10B552"),
       )
  pall
# png("~/Desktop/presenations/my_presentations/2020_groupmeeting/figures/nucleotid_diversity.png", width = 2000, height = 1000,res=300)

  svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/05_sterm_nucleotideDiv.svg",width=6,height=4.5)
pall
dev.off()

###-----------------RUN ONLY WHEN run phage diversity chunk

library(patchwork)

  svg("~/Desktop/presenations/my_presentations/20200318_comittee/figures/fig_1_nucleotide_div.svg",width=12,height=4.5)
pall+p12

dev.off()

###AVERAGE
mean(nucleotid_div_sterm$nucleotide_diversity)
mean(nucleotid_div_ldel$nucleotide_diversity)

###----------------------------------------
#einzelne plots
###----------------------------------------

 CoreGenome_nucleotide_div$species = factor(CoreGenome_nucleotide_div$species, levels=c("S. thermophilus","L. delbrueckii" ))

 pSterm <- ggplot()+ #stat_summary(fun.y=sum, geom="line")+
     geom_point(data = CoreGenome_nucleotide_div,aes(y = nucleotide_diversity, x = sample,size=Median_Coverage,fill=species,color=species))+
     geom_line(data = CoreGenome_nucleotide_div,aes(y = nucleotide_diversity, x = sample,group=species,fill=species,color=species))+
   facet_wrap(~species,nrow = 2,scales = "free_y")+
   labs("",
         x="",
         y="nucleotide diversity [%]",
          title="")+
   scale_fill_manual(values = c("#EB4D4D","#10B552"))+
   scale_color_manual(values = c("#EB4D4D","#10B552"))+
  theme_classic()+
     scale_x_discrete( expand = c(0, 0)) +
    scale_y_continuous(labels = scales::comma)+
  # lims
 # scale_y_continuous(name = "nucleotide diversity [%]",breaks = c(0.3,0.35,0.4),labels =c(0.3,0.35,0.4) )+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="top"
       # axis.text.y.left = element_text(color = "#EB4D4D"),
        # axis.title.y.left = element_text(color="#EB4D4D"),
       )
  pSterm
  
 svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/05_nucleotideDiv_both.svg",width=6.2,height=5)
pSterm
dev.off()
  

```


Finally I have the conclusion that the large majority of the ldel community is exlained with the metagenome assembled genome. The remaining variation is in average very small. 

##### new SNPs

```{r}


##-----------------------------------------------------------
##cases where Lyo96==0 New SNPs
##-----------------------------------------------------------
sample_relative_long_newSNPS <- sample_relative_wide %>% filter(.,`Lyo\n1996`==0.00) %>%  filter(.,`starter\nculture\n2018`>0.05) %>% gather(., sample, Snps, "Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018", factor_key=TRUE,na.rm = TRUE) %>% filter(.,significance!="MODIFIER")  %>% filter(.,significance!="LOW")  %>% filter(.,is.na(Repeat_cluster))

  ##============
### plot 
   ##============


 sample_relative_long_newSNPS$LABELS <- ifelse(is.na(sample_relative_long_newSNPS$Preferred_name),sample_relative_long_newSNPS$geneName,sample_relative_long_newSNPS$Preferred_name)

table(sample_relative_long_newSNPS$LABELS)
sample_relative_long_newSNPS$LABELS = factor(sample_relative_long_newSNPS$LABELS, levels=c("GKC13_08910",'GKC13_05765','GKC13_04170','GKC13_06155','amiA'))


colors <- c("grey","grey","grey","grey","red")

  p3 <- ggplot(sample_relative_long_newSNPS,aes(x=sample,y=Snps,group=site,color=LABELS,fill=LABELS, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nRepeat_identity:",Repeat_cluster,"\ncore:",core,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.8, alpha=.5)+ 

    labs("",
         x="",
         y="Alternative allele frequency")+
    #scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    scale_y_continuous(limits = c(0,0.68))+
    theme_classic()+
   scale_fill_manual(values=colors)+
   scale_color_manual(values=colors)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
      #axis.text.x = element_blank(),
          legend.position="right",
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank()
          )
  
   svg("~/Desktop/Manuscripts/2019_RMK202/Figures//F1_new_mutations.svg",width=7,height=3)

#   png("~/Desktop/presenations/my_presentations/20190919_VUA/abundance_snps.png", width = 1900, height = 1200,res=300)
# 
p3+theme(axis.text.x = element_blank())# 
dev.off()


 
   svg("~/Desktop/Manuscripts/2019_RMK202/Figures//F1_new_mutations_names.svg",width=7,height=3)

#   png("~/Desktop/presenations/my_presentations/20190919_VUA/abundance_snps.png", width = 1900, height = 1200,res=300)
# 
p3
# 
dev.off()

  ##============
### genes of interest 
   ##============
RMK202_snpeff_eggnog_repeats_core <- read_csv("~/Desktop/presenations/my_presentations/20191007_groupmeeting/figures/RMK202_snpeff_eggnog_repeats_core_dNdS_uniqueGenes.txt")  %>% select(-X1)
table(sample_relative_long_newSNPS$finalName)

RMK202_snpeff_eggnog_repeats_core[which(RMK202_snpeff_eggnog_repeats_core$finalName %in% unique(sample_relative_long_newSNPS$finalName)),] 
RMK202_snpeff_eggnog_repeats_core[which(RMK202_snpeff_eggnog_repeats_core$finalName %in% sample_relative_long_newSNPS$finalName),] %>%  select(Preferred_name) %>% table()
# RMK202_snpeff_eggnog_repeats_core$
  
  # sum(RMK202_snpeff_eggnog_repeats_core$finalName %in% sample_relative_long_newSNPS$finalName)

```



##### lost SNPs

```{r}
##--------------------------------------------------------------------------------------------
##--------------------------------------------------------lost SNPs
##--------------------------------------------------------------------------------------------

  ##============
### Threshold 
   ##============
threshold_01 <- 0.1
threshold_02 <- 0.01

  ##============
### prepare file
   ##============
sample_relative_long_lostSNPS <- sample_relative_wide %>% filter(.,`Lyo\n1996`>threshold_01)  %>% filter(.,`starter\nculture\n2018`<=threshold_02) %>%   gather(., sample, Snps, "Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018", factor_key=TRUE,na.rm = TRUE) %>% filter(.,significance!="MODIFIER")  %>% filter(.,significance!="LOW")  %>% filter(.,is.na(Repeat_cluster))

nrow(sample_relative_long_lostSNPS)
sample_relative_long_lostSNPS$geneName
  ##============
### plot 
   ##============

# p3 <- ggplot(sample_relative_long_lostSNPS,aes(x=sample,y=Snps,group=site,color=Preferred_name,fill=Preferred_name, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nRepeat_identity:",Repeat_cluster,"\ncore:",core,"\nCOG:",COG_Functional_Category)))+geom_line(size=0.8, alpha=.5)+ 
  p3 <- ggplot(sample_relative_long_lostSNPS,aes(x=sample,y=Snps,group=site,color=geneName,fill=geneName, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nRepeat_identity:",Repeat_cluster,"\ncore:",core,"\nCOG:",COG_Functional_Category)))+geom_line(size=0.8, alpha=.5)+ 
    labs("",
         x="",
         y="Alternative \nallele\n frequency")+
    #scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    scale_y_continuous(limits = c(0,1))+
    theme_classic()+
   # scale_fill_manual(values=all_colours)+
   # scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
      #axis.text.x = element_blank(),
          legend.position="right"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          )
  
#   png("~/Desktop/presenations/my_presentations/20190919_VUA/abundance_snps.png", width = 1900, height = 1200,res=300)
# 
p3
# 
# dev.off()

  ##============
### genes of interest 
   ##============
RMK202_snpeff_eggnog_repeats_core <- read_csv("~/Desktop/presenations/my_presentations/20191007_groupmeeting/figures/RMK202_snpeff_eggnog_repeats_core_dNdS_uniqueGenes.txt")  %>% select(-X1)
RMK202_snpeff_eggnog_repeats_core[which(RMK202_snpeff_eggnog_repeats_core$finalName %in% unique(sample_relative_long_lostSNPS$finalName)),]


```

##### fixed SNPs


```{r}
##--------------------------------------------------------------------------------------------
##--------------------------------------------------------fixed SNPs
##--------------------------------------------------------------------------------------------

  ##============
### Threshold 
   ##============
threshold_03 <- 0.95
threshold_04 <- 0.98

  ##============
### prepare file
   ##============
sample_relative_long_fixedSNPS <- sample_relative_wide %>% filter(.,`Lyo\n2012`<=threshold_03)  %>% filter(.,`starter\nculture\n2018`>threshold_04) %>%   gather(., sample, Snps, "Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018", factor_key=TRUE,na.rm = TRUE) %>% filter(.,significance!="MODIFIER")  %>% filter(.,significance!="LOW")  %>% filter(.,is.na(Repeat_cluster))

nrow(sample_relative_long_fixedSNPS)

  ##============
### plot 
   ##============
sample_relative_long_fixedSNPS$LABELS <- ifelse(is.na(sample_relative_long_fixedSNPS$Preferred_name),sample_relative_long_fixedSNPS$geneName,sample_relative_long_fixedSNPS$Preferred_name)

colors <- c("grey","grey","grey","gold")

  p3 <- ggplot(sample_relative_long_fixedSNPS,aes(x=sample,y=Snps,group=site,color=LABELS,fill=LABELS, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nRepeat_identity:",Repeat_cluster,"\ncore:",core,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.8, alpha=.5)+ 

    labs("",
         x="",
         y="Alternative allele frequency")+
    #scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    scale_y_continuous(limits = c(0.68,1))+
    theme_classic()+
   scale_fill_manual(values=colors)+
   scale_color_manual(values=colors)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(#axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
      axis.text.x = element_blank(),
          legend.position="right",
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank()
          )
  
  p3
 svg("~/Desktop/Manuscripts/2019_RMK202/Figures//F1_fixed_mutations.svg",width=7,height=3)
 
#   png("~/Desktop/presenations/my_presentations/20190919_VUA/abundance_snps.png", width = 1900, height = 1200,res=300)
# 
p3
# 
dev.off()


  ##============
### genes of interest 
   ##============
RMK202_snpeff_eggnog_repeats_core <- read_csv("~/Desktop/presenations/my_presentations/20191007_groupmeeting/figures/RMK202_snpeff_eggnog_repeats_core_dNdS_uniqueGenes.txt")  %>% select(-X1)
RMK202_snpeff_eggnog_repeats_core[which(RMK202_snpeff_eggnog_repeats_core$finalName %in% unique(sample_relative_long_fixedSNPS$finalName)),]
table(sample_relative_long_fixedSNPS$finalName)
```
#####merge mutatations
```{r}


 ##============
### prepare file
   ##============
# sample_relative_alltogether_newSNPS <- rbind(sample_relative_long_newSNPS,sample_relative_long_lostSNPS,sample_relative_long_fixedSNPS)
sample_relative_long_fixedSNPS$type <- "fixed"
sample_relative_long_newSNPS$type<- "new"
sample_relative_alltogether_newSNPS <- rbind(sample_relative_long_newSNPS,sample_relative_long_fixedSNPS)


  ##============
### plot 
   ##============


  p3 <- ggplot(sample_relative_alltogether_newSNPS,aes(x=sample,y=Snps,group=site,color=Preferred_name,fill=Preferred_name, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nRepeat_identity:",Repeat_cluster,"\ncore:",core,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.8, alpha=.5)+ 
  facet_grid(type~.,scales="free_y")+
    labs("",
         x="",
         y="Alternative \nallele\n frequency")+
    #scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    scale_y_continuous(limits = c(0,1))+
    theme_classic()+
   # scale_fill_manual(values=all_colours)+
   # scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
      #axis.text.x = element_blank(),
          legend.position="right"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          )
  
#   png("~/Desktop/presenations/my_presentations/20190919_VUA/abundance_snps.png", width = 1900, height = 1200,res=300)
# 
p3
# 
# dev.off()



table(sample_relative_alltogether_newSNPS$significance)

svg("~/Desktop/Manuscripts/2019_RMK202/Figures//F1_fixedandNewMutations.svg",width=7,height=4)

p3
dev.off()

```




(glcU)[https://www.uniprot.org/uniprot/Q5HDV2]
Involved in the uptake of glucose.
(mntH)[https://www.uniprot.org/uniprot/P0A769]
H+-stimulated, divalent metal cation uptake system. Involved in manganese and iron uptake. Can also transport cadmium, cobalt, zinc and to a lesser extent nickel and copper. Involved in response to reactive oxygen
(ylbM)[https://www.uniprot.org/uniprot/L8AHH4]
Catalyzes the formation of N4-acetylcytidine (ac4C) at the wobble position of elongator tRNA(Met), using acetate and ATP as substrates. First activates an acetate ion to form acetyladenylate (Ac-AMP) and then transfers the acetyl group to tRNA to form ac4C34.



####explain metagenome snps with isolates

good [cluster](https://jmonlong.github.io/Hippocamplus/2018/02/13/tsne-and-clustering/) from tsne blog


###### subsetting and effect hisotgramm

Finally, we decided to look at following SNPs:
1. location with higher than 20x coverage 
2. minimum 3x support for SNP
3. min 30x coverage
4. MQM (average alternative allele quality) > 30
5. AQ/DP > 5
6. minimum 0.05 allele frequency
7. only modifications within genes are used
8. only core genes
9. only non-synomous SNVs (moderate and high impact mutations) are looked at
10. SNVs that are not present in all samples are not used for phasing


```{r}
###-------------
##--------merge all
###-------------

# sample_relative_safe_Sterm_final

not_explained_sterm_plot_nonModi <- merge(not_explained_sterm_plot,annotation_all,by="site",all.x = TRUE)
explained_sterm_both_plot_nonModi <- merge(explained_sterm_both_plot,annotation_all,by="site",all.x = TRUE)
explained_s50_plot_nonModi <- merge(explained_s50_plot,annotation_all,by="site",all.x = TRUE)
explained_s72_plot_nonModi <- merge(explained_s72_plot,annotation_all,by="site",all.x = TRUE)

all_sterm_plot_hist <- rbind(explained_sterm_both_plot_nonModi ,explained_s50_plot_nonModi,explained_s72_plot_nonModi,not_explained_sterm_plot_nonModi)
nrow(all_sterm_plot_hist)


###-------------
##------creat histogramm
###-------------
# cbind(all_sterm_plot$site,all_sterm_plot$group)
table(all_sterm_plot_hist$significance)  
table(all_sterm_plot_hist$group)  

all_sterm_plot_hist$detection <- ifelse(is.na(all_sterm_plot_hist$allele_frequency),"bellow detection\nlimit",ifelse(all_sterm_plot_hist$allele_frequency>0.05,"used","bellow detection\nlimit"))
table(all_sterm_plot_hist$detection)
# readsExplained <- [,c("site","group")]

all_sterm_plot_hist$significance = factor(all_sterm_plot_hist$significance, levels=c('MODIFIER','LOW','MODERATE','HIGH'))
all_sterm_plot_hist$group = factor(all_sterm_plot_hist$group, levels=c('both','S50','S72','not_explained'))
all_sterm_plot_hist$group <- revalue(all_sterm_plot_hist$group, c("S50"="mst1", "S72"="mst2"))



histogramm_SNVs <- ggplot(all_sterm_plot_hist,aes(x=group))+geom_bar(aes(fill = interaction(significance,detection)))+theme_classic()+
  facet_grid(~significance)+
  xlab("") + 
  ylab("variant sites") +
  scale_fill_manual(values=c("grey89","grey89","grey89","grey89","grey","gold1","orange","red"))+   #wes_palette(n=3, name="Moonrise3")[1:3]))+
  theme(legend.position = "right",
        plot.margin=unit(c(0,0,0,2.5),"cm"),
        axis.text.x=element_text(angle=45, hjust=1,size=9))

histogramm_SNVs
 svg("~/Desktop/Manuscripts/2019_RMK202/Figures/S09_SNVs_histogramm",width=8,height=4)
histogramm_SNVs

 dev.off()
 
#  
#   library("tidyverse")
# library("officer")
# library("rvg")

# read_pptx() %>%
#     add_slide(layout = "Title and Content", master = "Office Theme") %>%
#     ph_with_vg(code = print(grid.arrange(p1,p2, p3,p0,ncol=1, nrow=4,heights=c(1.15,1,1,1.55))), type = "body",height = 7, width = 8) %>% 
#     print(target = "~/Desktop/presenations/my_presentations/20190919_VUA/slides_03.pptx")

 
##================
###-------------
##----------subsett only dominant non-synomous mutations
###-------------
##================


###---------------------
# ---------------------subset data1: low abundance reads
###---------------------
 # table(not_explained_sterm_plot_nonModi$group)
not_explained_sterm_plot_nonModi <- not_explained_sterm_plot_nonModi[which(not_explained_sterm_plot_nonModi$allele_frequency>0.05),]
explained_sterm_both_plot_nonModi <- explained_sterm_both_plot_nonModi[which(explained_sterm_both_plot_nonModi$allele_frequency>0.05),]
explained_s50_plot_nonModi$group <- revalue(explained_s50_plot_nonModi$group, c("S50"="mst1"))
explained_s50_plot_nonModi <- explained_s50_plot_nonModi[which(explained_s50_plot_nonModi$allele_frequency>0.05),]
explained_s72_plot_nonModi$group <- revalue(explained_s72_plot_nonModi$group, c( "S72"="mst2"))
explained_s72_plot_nonModi <- explained_s72_plot_nonModi[which(explained_s72_plot_nonModi$allele_frequency>0.05),]


all_sterm_plot <- rbind(explained_sterm_both_plot_nonModi ,explained_s50_plot_nonModi,explained_s72_plot_nonModi,not_explained_sterm_plot_nonModi)
nrow(all_sterm_plot)
all_sterm_plot$group = factor(all_sterm_plot$group, levels=c("both","mst1","mst2","not_explained"))


 
 ##----plooting without modification

  pall_sterm_withModi <- ggplot(all_sterm_plot,aes(x=sample,y=allele_frequency,group=site,color=group,fill=group))+ geom_line(size=0.1, alpha=.1)+ 
       facet_grid(group~.)+
   labs("",
         x="",
         # y="",
          y="all")+
       # scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )

pall_sterm_withModi
#  library(plotly)
# ggp <- ggplotly(pall_sterm_withModi)
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_bacteria_all_split.html")



###---------------------
# ---------------------subset data2: non-complete snps
###---------------------

all_sterm_plot_wide <- spread(all_sterm_plot, sample, allele_frequency)
dim(all_sterm_plot_wide)
all_sterm_plot_wide_woNA <- all_sterm_plot_wide[!is.na(rowSums(all_sterm_plot_wide[,41:45])),]
dim(all_sterm_plot_wide_woNA)
all_sterm_plot_wide_onlyNA <- all_sterm_plot_wide[is.na(rowSums(all_sterm_plot_wide[,41:45])),]
dim(all_sterm_plot_wide_onlyNA)
all_sterm_plot_wide_onlyNA_long <- gather(all_sterm_plot_wide_onlyNA, sample, allele_frequency, colnames(all_sterm_plot_wide_onlyNA)[41:45], factor_key=TRUE,na.rm = TRUE) 

all_sterm_plot_wide_onlyNA_long_nonMOD <- all_sterm_plot_wide_onlyNA_long[all_sterm_plot_wide_onlyNA_long$significance!="MODIFIER",]
all_sterm_plot_wide_onlyNA_long_nonMOD_LOW <- all_sterm_plot_wide_onlyNA_long_nonMOD[all_sterm_plot_wide_onlyNA_long_nonMOD$significance!="LOW",]

 ##----plooting without modification

  pall_sterm_withModi_onlyNA <- ggplot(all_sterm_plot_wide_onlyNA_long_nonMOD_LOW,aes(x=sample,y=allele_frequency,group=site,color=group,fill=group, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nEC:",EC,"\nKEGG_ko:",KEGG_ko,"\nCAZy:",CAZy,"\nCOG:",COG_Functional_Category))) + geom_line(size=0.1, alpha=.1)+ 
       facet_grid(group~.)+
   labs("",
         x="",
         # y="",
          y="all")+
       # scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )

pall_sterm_withModi_onlyNA


 library(plotly)
ggp <- ggplotly(pall_sterm_withModi_onlyNA)
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_bacteria_all_split_woModi_snps that go away.html")



###-------------
##remove modifier muations
###-------------
table(all_sterm_plot$significance)
all_sterm_plot_woModi <- all_sterm_plot[all_sterm_plot$significance!="MODIFIER",]
(table(all_sterm_plot_woModi$group)/5)/(nrow(all_sterm_plot_woModi)/5)



  pall_sterm_woModi <- ggplot(all_sterm_plot_woModi,aes(x=sample,y=allele_frequency,group=site,color=group,fill=group, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nEC:",EC,"\nKEGG_ko:",KEGG_ko,"\nCAZy:",CAZy,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.1, alpha=.5)+ 
       facet_grid(group~.)+
   labs("",
         x="",
         # y="",
          y="all")+
       # scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
pall_sterm_woModi
#  library(plotly)
# ggp <- ggplotly(pall_sterm_woModi)
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_bacteria_all_split_woModi.html")


###-------------
##remove Low effect muations
###-------------
table(all_sterm_plot$significance)
all_sterm_plot_woModi_LOW <- all_sterm_plot_woModi[all_sterm_plot_woModi$significance!="LOW",]
(table(all_sterm_plot_woModi_LOW$group)/5)/(nrow(all_sterm_plot_woModi_LOW)/5)



  pall_sterm_woModi_LOW <- ggplot(all_sterm_plot_woModi_LOW,aes(x=sample,y=allele_frequency,group=site,color=group,fill=group, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nEC:",EC,"\nKEGG_ko:",KEGG_ko,"\nCAZy:",CAZy,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.1, alpha=.5)+ 
       facet_grid(group~.)+
   labs("",
         x="",
         # y="",
          y="all")+
       # scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
pall_sterm_woModi_LOW
#  library(plotly)
# ggp <- ggplotly(pall_sterm_woModi)
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_bacteria_all_split_woModi.html")


###-------------
##only copre
###-------------

coreGeneNames_rmk202 <- read_delim("~/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/coreGeneNames_rmk202_FINAL.txt", "\t", escape_double = FALSE, col_names = c("site","core"), trim_ws = TRUE)
dim(all_sterm_plot)
all_sterm_plot_core <- merge(all_sterm_plot,coreGeneNames_rmk202,by="site",all.x = TRUE,all.y = FALSE)
dim(all_sterm_plot_core)

table(all_sterm_plot_core$core)
all_sterm_plot_core <- all_sterm_plot_core[all_sterm_plot_core$core=="core",]
table(all_sterm_plot_core$core)
(table(all_sterm_plot_core$group)/5)/(nrow(all_sterm_plot_core)/5)



  pall_sterm_CORE <- ggplot(all_sterm_plot_core,aes(x=sample,y=allele_frequency,group=site,color=group,fill=group, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nEC:",EC,"\nKEGG_ko:",KEGG_ko,"\nCAZy:",CAZy,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.1, alpha=.5)+ 
       facet_grid(group~.)+
   labs("",
         x="",
         # y="",
          y="all")+
       # scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
pall_sterm_CORE
#  library(plotly)
# ggp <- ggplotly(pall_sterm_woModi)
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_bacteria_all_split_woModi.html")

###---------
##boxplot
###---------

  pall_sterm_CORE_box <- ggplot(all_sterm_plot_core,aes(x=sample,y=allele_frequency,color=group,fill=group))+ geom_boxplot()+
       # facet_grid(group~.)+
   labs("",
         x="",
         # y="",
          y="all")+
       # scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="right",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
pall_sterm_CORE_box


  pall_sterm_CORE_box <- ggplot(all_sterm_plot_core,aes(x=group,y=allele_frequency,color=sample,fill=sample))+ geom_boxplot()+
       # facet_grid(group~.)+
   labs("",
         x="",
         # y="",
          y="all")+
       # scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="right",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
pall_sterm_CORE_box

pall_sterm_CORE_box_median <- aggregate(.~sample+group, data=all_sterm_plot_core[,c("allele_frequency","sample","group")], median, na.rm=TRUE)
pall_sterm_CORE_box_median

pall_sterm_CORE_box_median_wide <- spread(pall_sterm_CORE_box_median, group, allele_frequency)
pall_sterm_CORE_box_median_wide

Final_Abundance_Sterm_wide <- pall_sterm_CORE_box_median_wide %>% select(sample,mst1,mst2,not_explained)

Final_Abundance_Sterm_wide$MAG_rmk202 <- 1-pall_sterm_CORE_box_median_wide$both
Final_Abundance_Sterm_wide$mst1 <- pall_sterm_CORE_box_median_wide$both*pall_sterm_CORE_box_median_wide$mst1
Final_Abundance_Sterm_wide$mst2 <- pall_sterm_CORE_box_median_wide$both*pall_sterm_CORE_box_median_wide$mst2
Final_Abundance_Sterm_wide$not_explained <- pall_sterm_CORE_box_median_wide$both*pall_sterm_CORE_box_median_wide$not_explained
Final_Abundance_Sterm_wide
Final_Abundance_Sterm_long <- gather(Final_Abundance_Sterm_wide,Identity, Population, "mst1":"MAG_rmk202", factor_key=TRUE,na.rm = TRUE) 
colnames(Final_Abundance_Sterm_long)[1] <- "Generation"
Final_Abundance_Sterm_long$Generation <- revalue(Final_Abundance_Sterm_long$Generation, c("Lyoampulle\n2012"=as.double(1) ,"Lyoampulle\n2014"=as.double(2),"working\nstock"=as.double(3), "starter\nculture\n2012"=as.double(4), "starter\nculture\n2018"=as.double(5)))

Final_Abundance_Sterm_long

```

###### plot for presentation

Here, I make a plot of the two species snp profile ready for the group meeting presentation

```{r}

SNP_profile_presentation_long <- sample_relative_safe_Sterm_final
# SNP_profile_presentation_wide <- sample_relative_wide %>%  filter(.,significance %in% c("MODERATE","HIGH"))  %>%replace(is.na(.), 0)
# SNP_profile_presentation_long <-  SNP_profile_presentation_wide %>% gather(., sample, Snps, "Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2", factor_key=TRUE,na.rm = TRUE) %>% filter(.,Snps>0.005)

##---------------plot all samples

PLOT_allSNPs_presentation <- ggplot(mst1_metagenomicMutations_long,aes(x=sample,y=Snps,group=site,color=species,fill=species, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nRepeat_identity:",Repeat_cluster,"\ncore:",core,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.2, alpha=.1)+ 
    # facet_grid(day~kessel~species)+
    facet_grid(species~.)+
    # facet_grid(treatment~species)+
    # facet_grid(treatment~species)+
    labs("",
         x="time [h]",
         y="Alternative allele frequency")+
    #scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    #scale_y_continuous(limits = c(0,0.3))+
    theme_classic()+
   scale_fill_manual(values=all_colours)+
   scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
      #axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          )
  
PLOT_allSNPs_presentation



##---------------plot without cheesemaking

SNP_profile_presentation_long_subset <- SNP_profile_presentation_long %>% filter(sample!="cheesemaking\nday1")  %>% filter(sample!="cheesemaking\nday2")

PLOT_allSNPs_presentation <- ggplot(SNP_profile_presentation_long_subset,aes(x=sample,y=Snps,group=site,color=species,fill=species, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nRepeat_identity:",Repeat_cluster,"\ncore:",core,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.2, alpha=.1)+ 
    # facet_grid(day~kessel~species)+
    facet_grid(species~.)+
    # facet_grid(treatment~species)+
    # facet_grid(treatment~species)+
    labs("",
         x="time [h]",
         y="Alternative allele frequency")+
    #scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    #scale_y_continuous(limits = c(0,0.3))+
    theme_classic()+
   scale_fill_manual(values=all_colours)+
   scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
      #axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          )
  
PLOT_allSNPs_presentation

 png("~/Desktop/presenations/my_presentations/2020_groupmeeting/figures/snpProfile_withoutCheesmaking.png",width=1800,height=1600,res=300)


PLOT_allSNPs_presentation

dev.off()

##---------------plot SNP profile with disentagling

# ---split into mst's
p2 <- p_both_sterm_woModi_NOT_clustered_withougCheese / p_mst1_sterm_woModi_NOT_clustered_withougCheese / p_mst2_sterm_woModi_NOT_clustered_withougCheese / p_none_sterm_woModi_NOT_clustered_withougCheese

##--------snp profile sterm

all_snps_sterm_long_sub <- all_snps_sterm_long %>% filter(sample!="cheesemaking\nday1")  %>% filter(sample!="cheesemaking\nday2")

PLOT_allSNPs <- ggplot(all_snps_sterm_long_sub,aes(x=sample,y=Snps,group=site,color=species,fill=species, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nRepeat_identity:",Repeat_cluster,"\ncore:",core,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.2, alpha=.1)+ 
    labs("",
         x="time [h]",
         y="Alternative allele frequency")+
  theme_classic()+
   scale_fill_manual(values=all_colours[2])+
   scale_color_manual(values=all_colours[2])+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          legend.position="none")
  

p1 <- plot_spacer() +  PLOT_allSNPs +  plot_spacer() +   plot_layout(nrow=3,heights = c(1,2, 1))
p1
#===========================
 ##-----==combine all
#===========================

# png("~/Desktop/presenations/my_presentations/2020_groupmeeting/figures/snps_msts_withoutCheesemaking.png",width=2400,height=1600,res=300)

 svg("~/Desktop/presenations/my_presentations/2020_groupmeeting/figures/snps_msts_withoutCheesemaking.svg",width=12,height=8)

(p1|p2)+plot_layout(widths =c(2,2))
 
 dev.off()

```




######mst1 substrains

good [color schemes](https://www.schemecolor.com/pastel-red-monochromatic.php)

```{r}
library(forcats) 
library(Rtsne)
library(fpc)
library(ggrepel)
###--------------------------------------------------------------------
##colors
###--------------------------------------------------------------------

# colorsTSNE <- c("#DD8D29","#E2D200","#46ACC8","black","#B40F20")

colorsTSNE <- c("#E69F00", "#56B4E9", "#009E73",
          "#F0E442", "#0072B2", "#D55E00", "#CC79A7","grey","#000000")

colorsTSNE <- rev(c("#FF9E79","#FB6D4C","#C23B22","#8A0000","#580000","#D55E00"))
###--------------------------------------------------------------------
##subset
###--------------------------------------------------------------------

# mst1_metagenomicMutations <- sample_relative_wide %>% filter(.,species=="S. thermophilus") %>% filter(.,significance %in% c("MODERATE","HIGH")) %>% filter(.,mst1>0.9 & mst2 <=0.9) %>% add_column(group="mst1") %>%replace(is.na(.), 0)
mst1_metagenomicMutations <- sample_relative_safe_Sterm_final_wide %>% filter(.,mst1>0.9 & mst2 <=0.9)%>% add_column(group="mst1") 


mst1_metagenomicMutations_long <-  mst1_metagenomicMutations %>% gather(., sample, Snps, "Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2", factor_key=TRUE,na.rm = TRUE)# %>% filter(.,Snps>0.005)

mst1_metagenomicMutations_tsne <- spread(mst1_metagenomicMutations_long, sample, Snps)%>%replace(is.na(.), 0)


###--------------------------------------------------------------------
##prep for tSNE
###--------------------------------------------------------------------



fortsne <- mst1_metagenomicMutations %>% select(.,"site","Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2")
fortsne <- fortsne[!duplicated(fortsne[,-1]), ]
# fortsne[duplicated(fortsne), ] %>% nrow()

set.seed(1234567)
# set.seed(12314567)

tsne <- Rtsne(fortsne[,-1], dims = 2, perplexity=10, verbose=TRUE, max_iter = 500)
# tsne <- Rtsne(fortsne[,41:45], dims = 2, perplexity=10, verbose=TRUE, max_iter = 500)
tsne_plot <- as.data.frame(tsne$Y)
tsne_plot$site <- fortsne[,1]
tsne_plot$species <- paste(str_split_fixed(tsne_plot$site, "_", 5)[,1],str_split_fixed(tsne_plot$site, "_", 5)[,2],sep = "_") %>%  gsub("_",". ",.)



##------------plot blank
# ggplot(tsne_plot,aes(x=V1,y=V2))+geom_point()

##------------clusters

# set.seed(123456)

set.seed(1234567)
ds.real = dbscan(tsne_plot[,c(1,2)], 3)
decreasing_clusternames <- (names(sort(table(ds.real$cluster),decreasing = TRUE)))
change_name <- setNames(1:length(decreasing_clusternames),decreasing_clusternames)
clusterNames <- revalue(as.factor(ds.real$cluster),change_name)
clusterNames<- revalue(clusterNames,c("4"="1","6"="1","7"="1","10"="1","11"="1","5"="4","8"="5","9"="6"))##add certain clusters to others
tsne_plot$density = factor(clusterNames)
levels(tsne_plot$density) <- c(levels(tsne_plot$density), "5")

# special_cases <- c("CP046134_494890_A_G","CP046134_482906_A_C")
# tsne_plot[which(tsne_plot$site %in% special_cases),"density"] <- "5"
ds.cent = tsne_plot %>% group_by(density) %>% select(V1,
    V2) %>% summarize_all(mean)
ggplot(tsne_plot, aes(x = V1, y = V2, colour = density)) +
    geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label = density),
    data = ds.cent) + guides(colour = FALSE)



##-----------make clustering manual
# tsne_plot$final_clust <- tsne_plot$density
# tsne_plot_final_clust <- tsne_plot  %>% select(site,  final_clust)
# table(tsne_plot_final_clust$final_clust)

# tsne_plot_final_clust <- tsne_plot %>% mutate(final_clust = ifelse(density==3,2,
#                                                                     ifelse(density==2,3,1)))  %>% select(site,  final_clust)
#  tsne_plot <- tsne_plot %>% mutate(final_clust = ifelse(density==3,2,
#                                                                     ifelse(density==2,3,1)))
# table(tsne_plot_final_clust$final_clust)
# tsne_plot$density <- revalue(tsne_plot$density, c("4"="1"))

tsne_plot <- tsne_plot %>% mutate(final_clust=density)
tsne_plot_final_clust <- tsne_plot %>% select(site,  final_clust)

###------------------------------------
##-------sumplementary plot tsne
###------------------------------------
tsne_plot$final_clust = factor(tsne_plot$final_clust, levels=1:length(unique(tsne_plot_final_clust$final_clust)))
ds.cent = tsne_plot %>% group_by(final_clust) %>% select(V1, 
    V2) %>% summarize_all(mean)
colors_substrains <- c(colorsTSNE[1:length(unique(tsne_plot_final_clust$final_clust))])

Sterm_mst1_suplementaryTsne <- ggplot(tsne_plot, aes(x = V1, y = V2, colour = final_clust)) + 
    geom_point(alpha = 0.3) + theme_classic() + geom_label_repel(aes(label =final_clust), 
    data = ds.cent) + guides(colour = FALSE)+scale_color_manual(values = colors_substrains)+theme(axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank())

Sterm_mst1_suplementaryTsne

###------------------------------------
##-------sumplementary histogramm
###------------------------------------

Sterm_mst1_suplementarybarPlot <- ggplot(tsne_plot,aes(x=final_clust,color=final_clust,fill=final_clust))+geom_bar()+theme_classic()+
  scale_fill_manual(values = colors_substrains)+
  scale_color_manual(values = colors_substrains)+
  labs(x="",y="count")+theme(legend.position = "none")
Sterm_mst1_suplementarybarPlot
###------------------------------------
##---------clustering
###------------------------------------

explained_mst1_plot_nonModi_CLUSTERED <- merge(mst1_metagenomicMutations_long,tsne_plot_final_clust,by = "site",all.x = TRUE)
# table(explained_mst1_plot_nonModi_CLUSTERED$final_clust)
# unique(explained_mst1_plot_nonModi_CLUSTERED$final_clust)
# explained_mst1_plot_nonModi_CLUSTERED[which(is.na(explained_mst1_plot_nonModi_CLUSTERED$final_clust)),]
explained_mst1_plot_nonModi_CLUSTERED <- explained_mst1_plot_nonModi_CLUSTERED[which(!is.na(explained_mst1_plot_nonModi_CLUSTERED$final_clust)),]
explained_mst1_plot_nonModi_CLUSTERED$final_clust <- fct_explicit_na(as.character(explained_mst1_plot_nonModi_CLUSTERED$final_clust), na_level = "not clustered")
# unique(explained_mst1_plot_nonModi_CLUSTERED$final_clust)
explained_mst1_plot_nonModi_CLUSTERED$final_clust <- droplevels(explained_mst1_plot_nonModi_CLUSTERED$final_clust)

###------------------------------------
##plotting
###------------------------------------
# explained_mst1_plot_nonModi_CLUSTERED
explained_mst1_plot_nonModi_CLUSTERED$final_clust = factor(explained_mst1_plot_nonModi_CLUSTERED$final_clust, levels=(sort(unique(explained_mst1_plot_nonModi_CLUSTERED$final_clust),decreasing = FALSE)))

# levels(explained_mst1_plot_nonModi_CLUSTERED$final_clust)
plotColors <- colorsTSNE[as.numeric(levels(explained_mst1_plot_nonModi_CLUSTERED$final_clust))]

##assign colorsi
explained_mst1_plot_nonModi_CLUSTERED$final_clust <- droplevels(explained_mst1_plot_nonModi_CLUSTERED$final_clust)
colors_mst1 <- data.frame(cluster=paste("mst1",as.character(levels(explained_mst1_plot_nonModi_CLUSTERED$final_clust)),sep="_"),color=plotColors[1:length(levels(explained_mst1_plot_nonModi_CLUSTERED$final_clust))])


  p_mst1_sterm_woModi_clustered <- ggplot(explained_mst1_plot_nonModi_CLUSTERED,aes(x=sample,y=Snps,group=site,color=final_clust,fill=final_clust, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.1, alpha=.5)+ 
       # facet_grid(final_clust~.)+
   # labs("",
         # x="",
         # y="",
          # y="all")+
       scale_fill_manual(values=(plotColors))+
      scale_color_manual(values=(plotColors))+
       scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          ylim(c(0,1))+
 theme(#axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="bottom",
       legend.title = element_blank(),
       axis.text.x = element_blank(),
       axis.text.y = element_text(size=8),
       # axis.title = element_text(size=9),
       axis.title = element_blank(),
       
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )

p_mst1_sterm_woModi_clustered
# 
# ggp <- ggplotly(p_mst1_sterm_woModi_clustered)
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_mst1_clustered_woModi.html")
# 

##===========================================
##-------------------------------------plotting withou clustering
##===========================================

    p_mst1_sterm_woModi_NOT_clustered <- ggplot(explained_mst1_plot_nonModi_CLUSTERED,aes(x=sample,y=Snps,group=site, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.1, alpha=.5)+ 
       # facet_grid(final_clust~.)+
   # labs("",
         # x="",
         # y="",
          # y="all")+
       # scale_fill_manual(values=(plotColors))+
      # scale_color_manual(values=(plotColors))+
       scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="bottom",
       legend.title = element_blank(),
       # axis.text.x = element_blank(),
       axis.text.y = element_text(size=8),
       # axis.title = element_text(size=9),
       axis.title = element_blank(),
       
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
p_mst1_sterm_woModi_NOT_clustered

# library(plotly)
# ggp <- ggplotly(p_mst1_sterm_woModi_clustered)
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_mst1_clustered_woModi.html")

tsne_plot_mst1 <- tsne_plot
##===========================================
##-------------------------------------plotting without cheesemaking samples
##===========================================

explained_mst1_plot_nonModi_CLUSTERED_withoutcheese <- explained_mst1_plot_nonModi_CLUSTERED %>% filter(sample!="cheesemaking\nday1")  %>% filter(sample!="cheesemaking\nday2")

    p_mst1_sterm_woModi_NOT_clustered_withougCheese <- ggplot(explained_mst1_plot_nonModi_CLUSTERED_withoutcheese,aes(x=sample,y=Snps,group=site, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.1, alpha=.5)+ 
       scale_x_discrete( expand = c(0, 0)) +
  theme_classic()+
          ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="bottom",
       legend.title = element_blank(),
       axis.text.y = element_text(size=8),
       axis.title = element_blank(),
       
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
p_mst1_sterm_woModi_NOT_clustered_withougCheese
###------------------------------------
##density plot
###------------------------------------
explained_mst1_plot_nonModi_CLUSTERED$location <- as.numeric(str_split_fixed(explained_mst1_plot_nonModi_CLUSTERED$site, "_", 5)[,2])

forDensityPlot <- explained_mst1_plot_nonModi_CLUSTERED %>% select(.,site,location,final_clust) %>% distinct()

forDensityPlot$final_clust <- droplevels(forDensityPlot$final_clust)
forDensityPlot$final_clust = factor(forDensityPlot$final_clust, levels=(sort(unique(forDensityPlot$final_clust),decreasing = FALSE)))
# plotColors <- colorsTSNE[as.numeric(levels(forDensityPlot$final_clust))]
unique(forDensityPlot$final_clust)

Sterm_mst1_density <- ggplot(forDensityPlot,aes(x=location,group=final_clust,color=final_clust,fill=final_clust))+geom_density(alpha=0.2)+theme_classic()+
  facet_wrap(~final_clust,scale = "free_y")+
  scale_fill_manual(values = c(colors_substrains,"grey"))+
  scale_color_manual(values = c(colors_substrains,"grey"))+
  labs(x="genomic location",y="density")+theme(strip.text.x = element_blank(),legend.position = "none",legend.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank())
Sterm_mst1_density

Sterm_mst1_frequency <- ggplot(forDensityPlot,aes(x=location,group=final_clust,color=final_clust,fill=final_clust))+geom_freqpoly()+theme_classic()+
  # facet_wrap(~final_clust,scale = "free_y")+
  scale_fill_manual(values = c(colors_substrains,"grey"))+
  scale_color_manual(values = c(colors_substrains,"grey"))+
  labs(x="genomic location",y="density")+theme(legend.position = "none",legend.title = element_blank())
Sterm_mst1_frequency


Sterm_mst1_histo <- ggplot(forDensityPlot,aes(x=location,group=final_clust,color=final_clust,fill=final_clust))+geom_histogram(position="dodge2")+theme_classic()+
  # facet_wrap(~final_clust,scale = "free_y")+
  scale_fill_manual(values = c(colors_substrains,"grey"))+
  scale_color_manual(values = c(colors_substrains,"grey"))+
  labs(x="genomic location",y="density")+theme(legend.position = "none",legend.title = element_blank())
Sterm_mst1_histo


Sterm_mst1_point <- ggplot(forDensityPlot,aes(x=location,y=final_clust,group=final_clust,color=final_clust,fill=final_clust))+geom_jitter(width = 0.0, height = 0.4,alpha=0.3 ,size = 0.35)+theme_classic()+
  # facet_wrap(~final_clust,scale = "free_y")+
  scale_fill_manual(values = c(colors_substrains,"grey"))+
  scale_color_manual(values = c(colors_substrains,"grey"))+
  labs(x="genomic location",y="cluster")+theme(legend.position = "none",legend.title = element_blank(),axis.text.x = element_blank(),axis.ticks.x = element_blank())
Sterm_mst1_point

# theme(strip.text.x = element_blank())



##====================================================
###------RIBBON of interquartile range
##====================================================

##=====================here I apply a hack!!
##in order to recreat eh labels from ggplot_build I color the samples and final_clust according to a known color scale

colorSamples <- c("gray1","gray2","gray3","gray4","gray5","gray6","gray7","gray8","gray9","gray10","gray11")
 # explained_mst1_plot_nonModi_CLUSTERED$final_clust = factor(explained_mst1_plot_nonModi_CLUSTERED$final_clust, levels=c(6:1))
explained_mst1_plot_nonModi_CLUSTERED$final_clust = factor(explained_mst1_plot_nonModi_CLUSTERED$final_clust, levels=(sort(unique(explained_mst1_plot_nonModi_CLUSTERED$final_clust),decreasing = FALSE)))

 explained_mst1_plot_nonModi_CLUSTERED$sample = factor(explained_mst1_plot_nonModi_CLUSTERED$sample, levels=(c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2")))

##----------------------prepare table

p <- ggplot(explained_mst1_plot_nonModi_CLUSTERED, aes(x=sample, y=Snps,color=sample,fill=final_clust)) + geom_boxplot() +scale_fill_manual(values = colorSamples)+scale_color_manual(values = colorSamples)
p
mst1_phase_qunatiles <- ggplot_build(p)$data[[1]] %>% as.tibble() %>%  rename(sample=colour)  %>%  rename(final_clust=fill) %>% mutate(final_clust = revalue(final_clust, c("gray1"="1", "gray2"="2", "gray3"="3", "gray4"="4", "gray5"="5", "gray6"="6", "gray7"="7", "gray8"="8")))%>% mutate(sample = revalue(sample, c("gray1"="Lyo\n1996", "gray2"="Lyo\n2012", "gray3"="Lyo\n2014", "gray4"="working\nstock", "gray5"="starter\nculture\n2012", "gray6"="starter\nculture\n2018", "gray7"="cheesemaking\nday1", "gray8"="cheesemaking\nday2")))
 table( mst1_phase_qunatiles$final_clust)
str(mst1_phase_qunatiles$final_clust)

mst1_phase_qunatiles <- ggplot_build(p)$data[[1]] %>% as.tibble() %>%  rename(sample=colour)  %>%  rename(final_clust=fill) %>% mutate(final_clust = revalue(final_clust, c("gray1"="1", "gray2"="2", "gray3"="3", "gray4"="4", "gray5"="5", "gray6"="6", "gray7"="7", "gray8"="8")))%>% mutate(sample = revalue(sample, c("gray1"="Lyo\n1996", "gray2"="Lyo\n2012", "gray3"="Lyo\n2014", "gray4"="working\nstock", "gray5"="starter\nculture\n2012", "gray6"="starter\nculture\n2018", "gray7"="cheesemaking\nday1", "gray8"="cheesemaking\nday2"))) %>% select(-"y")



# mst1_phase_qunatiles$final_clust = factor(mst1_phase_qunatiles$final_clust, levels=c(6:1))
# plotColors <- colorsTSNE[as.numeric(levels(mst1_phase_qunatiles$final_clust))]

mst1_phase_qunatiles$final_clust = factor(mst1_phase_qunatiles$final_clust, levels=(sort(unique(mst1_phase_qunatiles$final_clust),decreasing = FALSE)))
plotColors <- colorsTSNE[as.numeric(levels(mst1_phase_qunatiles$final_clust))]


 mst1_phase_qunatiles$sample = factor(mst1_phase_qunatiles$sample, levels=(c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2")))


mst1_phased_ribbon <- ggplot()+
    geom_ribbon(aes(x=sample,ymin = ymin, ymax = ymax,group=final_clust,fill=final_clust),data = mst1_phase_qunatiles, alpha=.4)+
    # geom_ribbon(aes(x=sample,ymin = X1st.Qu., ymax = X3rd.Qu.,group=final_clust,fill=final_clust),data = both_phase_qunatiles, alpha=.5)+
    # geom_ribbon(aes(x=sample,ymin = Min., ymax = Max.,group=final_clust,fill=final_clust),data = mst1_phase_qunatiles, alpha=.5)+
    # geom_line(data=explained_mst1_plot_nonModi_CLUSTERED,aes(x=sample,y=Snps,group=site,color=final_clust,fill=final_clust),size=0.1, alpha=.1)+ 
    # geom_line(data = mappingcoverage,aes(y = coverage, x = time_f_02,group=contig))+
      # facet_grid(contig~treatment )+
      # scale_x_continuous(breaks = c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
       # scale_x_discrete(breaks = c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    scale_fill_manual(values=(plotColors))+
      scale_color_manual(values=(plotColors))+
       scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="bottom",
       legend.title = element_blank(),
       # axis.text.x = element_blank(),
       axis.text.y = element_text(size=8),
       # axis.title = element_text(size=9),
       axis.title = element_blank(),
       
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )


  mst1_phased_ribbon
  
##====================================================
###------mutated genes
##====================================================
colnames(explained_mst1_plot_nonModi_CLUSTERED)
  
mst1_Explained_sterm_wide <- explained_mst1_plot_nonModi_CLUSTERED %>% select(c("site","finalName","X.CHROM.x","species","culture", "Preferred_name","effect","significance","geneName", "COG_Functional_Category","Repeat_cluster","core" , "mst6", "mst7","mst9", "mst10","mst3","mst4","mst5" ,"mst8" , "mst1", "mst2", "dN_dN.dS"  , "mutations" , "group",  "sample" , "Snps","final_clust" , "location")) %>% spread(., sample, Snps)
  
mst1_Explained_sterm_wide$LABELS <- ifelse(mst1_Explained_sterm_wide$Preferred_name=="0",mst1_Explained_sterm_wide$geneName,mst1_Explained_sterm_wide$Preferred_name)

mst1_genes_Mutated <- with(mst1_Explained_sterm_wide, table(LABELS,final_clust)) %>% as.data.frame() %>% filter(Freq>0) %>% arrange(desc(Freq))

mst1_genes_Mutated$LABELS <-  factor(mst1_genes_Mutated$LABELS, levels=unique(mst1_genes_Mutated$LABELS))
mst1_genes_Mutated$final_clust <-  factor(mst1_genes_Mutated$final_clust, levels=c("4","3","2","1"))


genes_mutated_mst1_ <- ggplot(mst1_genes_Mutated,aes(x=LABELS,y=Freq,color=final_clust,fill=final_clust))+geom_point()+theme_classic()+facet_wrap(final_clust~., scales="free")+
      scale_fill_manual(values=(plotColors))+
      scale_color_manual(values=(plotColors))+
   theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7),
         axis.title.x = element_blank(),
         strip.background = element_blank(),
          strip.text.x = element_blank(),
         legend.position = "none")

genes_mutated_mst1_

  
```


######mst2 substrain

```{r}



colorsTSNE <- rev(c("#0B97B9","#6FD8E5","#C4EFED","#8DDBCF","#30C0C0"))

###--------------------------------------------------------------------
##subset
###--------------------------------------------------------------------
mst2_metagenomicMutations <- sample_relative_safe_Sterm_final_wide %>% filter(.,mst1<=0.9 & mst2>0.9) %>% add_column(group="mst2")


# mst1_metagenomicMutations <- sample_relative_safe_Sterm_final_wide %>% filter(.,mst1>0.9 & mst2 <=0.9)%>% add_column(group="mst1") 
# 
# 
# mst1_metagenomicMutations_long <-  mst1_metagenomicMutations %>% gather(., sample, Snps, "Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2", factor_key=TRUE,na.rm = TRUE)# %>% filter(.,Snps>0.005)
# 
# mst1_metagenomicMutations_tsne <- spread(mst1_metagenomicMutations_long, sample, Snps)%>%replace(is.na(.), 0)


# mst2_metagenomicMutations <- sample_relative_wide %>% filter(.,species=="S. thermophilus") %>% filter(.,significance %in% c("MODERATE","HIGH")) %>% filter(.,mst1<=0.9 & mst2>0.9) %>% add_column(group="mst2")%>%replace(is.na(.), 0)
mst2_metagenomicMutations_long <-  mst2_metagenomicMutations %>% gather(., sample, Snps, "Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2", factor_key=TRUE,na.rm = TRUE) #%>% filter(.,Snps>0.005)

mst2_metagenomicMutations_tsne <- spread(mst2_metagenomicMutations_long, sample, Snps)%>%replace(is.na(.), 0)


###--------------------------------------------------------------------
##prep for tSNE
###--------------------------------------------------------------------


# 
fortsne <- mst2_metagenomicMutations_tsne %>% select(.,"site","Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2")
fortsne <- fortsne[!duplicated(fortsne[,-1]), ]
# fortsne[duplicated(fortsne), ] %>% nrow()
# 
set.seed(1234567)
tsne <- Rtsne(fortsne[,-1], dims = 2, perplexity=10, verbose=TRUE, max_iter = 500)
# tsne <- Rtsne(fortsne[,41:45], dims = 2, perplexity=10, verbose=TRUE, max_iter = 500)
tsne_plot <- as.data.frame(tsne$Y)
tsne_plot$site <- fortsne[,1]
tsne_plot$species <- paste(str_split_fixed(tsne_plot$site, "_", 5)[,1],str_split_fixed(tsne_plot$site, "_", 5)[,2],sep = "_") %>%  gsub("_",". ",.)



##------------plot blank
ggplot(tsne_plot,aes(x=V1,y=V2))+geom_point()

##------------clusters


set.seed(123456)
ds.real = dbscan(tsne_plot[,c(1,2)], 6)
# decreasing_clusternames <- (names(sort(table(ds.real$cluster),decreasing = TRUE)))
# change_name <- setNames(1:length(decreasing_clusternames),decreasing_clusternames)
# clusterNames <- revalue(as.factor(ds.real$cluster),change_name)
clusterNames <-as.factor(ds.real$cluster)
clusterNames<- revalue(clusterNames,c("1"="3","2"="1","3"="2"))##add certain clusters to others
tsne_plot$density = factor(clusterNames)
ds.cent = tsne_plot %>% group_by(density) %>% select(V1, 
    V2) %>% summarize_all(mean)
ggplot(tsne_plot, aes(x = V1, y = V2, colour = density)) +
    geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label = density),
    data = ds.cent) + guides(colour = FALSE)


##-----------make clustering manual
# tsne_plot$final_clust <- tsne_plot$density
# tsne_plot_final_clust <- tsne_plot  %>% select(site,  final_clust)
# table(tsne_plot_final_clust$final_clust)

# tsne_plot_final_clust <- tsne_plot %>% mutate(final_clust = ifelse(density==3,2,
#                                                                     ifelse(density==2,3,1)))  %>% select(site,  final_clust)
#  tsne_plot <- tsne_plot %>% mutate(final_clust = ifelse(density==3,2,
#                                                                     ifelse(density==2,3,1)))
# table(tsne_plot_final_clust$final_clust)
tsne_plot <- tsne_plot %>% mutate(final_clust=density)
tsne_plot_final_clust <- tsne_plot %>% select(site,  final_clust)

###------------------------------------
##-------sumplementary plot tsne
###------------------------------------
# tsne_plot$final_clust = factor(tsne_plot$final_clust, levels=1:length(unique(tsne_plot_final_clust$final_clust)))
# ds.cent = tsne_plot %>% group_by(final_clust) %>% select(V1, 
#     V2) %>% summarize_all(mean)
# colors_substrains <- c(colorsTSNE[1:length(unique(tsne_plot_final_clust$final_clust))])


tsne_plot$final_clust = factor(tsne_plot$final_clust, levels=1:length(unique(tsne_plot_final_clust$final_clust)))
ds.cent = tsne_plot %>% group_by(final_clust) %>% select(V1, 
    V2) %>% summarize_all(mean)
colors_substrains <- c(colorsTSNE[1:length(unique(tsne_plot_final_clust$final_clust))])


# Sterm_mst2_suplementaryTsne <- ggplot(tsne_plot, aes(x = V1, y = V2, colour = final_clust)) + 
#     geom_point(alpha = 0.3) + theme_classic() + geom_label_repel(aes(label =final_clust), 
#     data = ds.cent) + guides(colour = FALSE)+scale_color_manual(values = colors_substrains)


Sterm_mst2_suplementaryTsne <- ggplot(tsne_plot, aes(x = V1, y = V2, colour = final_clust)) + 
    geom_point(alpha = 0.3) + theme_classic() + geom_label_repel(aes(label =final_clust), 
    data = ds.cent) + guides(colour = FALSE)+scale_color_manual(values = colors_substrains)+theme(axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank())


Sterm_mst2_suplementaryTsne





###------------------------------------
##-------sumplementary histogramm
###------------------------------------

Sterm_mst2_suplementarybarPlot <- ggplot(tsne_plot,aes(x=final_clust,color=final_clust,fill=final_clust))+geom_bar()+theme_classic()+
  scale_fill_manual(values = colors_substrains)+
  scale_color_manual(values = colors_substrains)+
  labs(x="",y="count")+theme(legend.position = "none")
Sterm_mst2_suplementarybarPlot
###------------------------------------
##---------clustering
###------------------------------------

explained_mst2_plot_nonModi_CLUSTERED <- merge(mst2_metagenomicMutations_long,tsne_plot_final_clust,by = "site",all.x = TRUE)
# table(explained_mst2_plot_nonModi_CLUSTERED$final_clust)
# unique(explained_mst2_plot_nonModi_CLUSTERED$final_clust)
explained_mst2_plot_nonModi_CLUSTERED <- explained_mst2_plot_nonModi_CLUSTERED[which(!is.na(explained_mst2_plot_nonModi_CLUSTERED$final_clust)),]
explained_mst2_plot_nonModi_CLUSTERED$final_clust <- fct_explicit_na(as.character(explained_mst2_plot_nonModi_CLUSTERED$final_clust), na_level = "not clustered")
# unique(explained_mst2_plot_nonModi_CLUSTERED$final_clust)
explained_mst2_plot_nonModi_CLUSTERED$final_clust <- droplevels(explained_mst2_plot_nonModi_CLUSTERED$final_clust)

###------------------------------------
##plotting
###------------------------------------
# explained_mst2_plot_nonModi_CLUSTERED
explained_mst2_plot_nonModi_CLUSTERED$final_clust = factor(explained_mst2_plot_nonModi_CLUSTERED$final_clust, levels=(sort(unique(explained_mst2_plot_nonModi_CLUSTERED$final_clust),decreasing = FALSE)))

# levels(explained_mst2_plot_nonModi_CLUSTERED$final_clust)
plotColors <- colorsTSNE[as.numeric(levels(explained_mst2_plot_nonModi_CLUSTERED$final_clust))]

##assign colors
colors_mst2 <- data.frame(cluster=paste("mst2",as.character(levels(explained_mst2_plot_nonModi_CLUSTERED$final_clust)),sep="_"),color=plotColors)
explained_mst2_plot_nonModi_CLUSTERED$final_clust <- droplevels(explained_mst2_plot_nonModi_CLUSTERED$final_clust)
colors_mst2 <- data.frame(cluster=paste("mst2",as.character(levels(explained_mst2_plot_nonModi_CLUSTERED$final_clust)),sep="_"),color=plotColors[1:length(levels(explained_mst2_plot_nonModi_CLUSTERED$final_clust))])

  
    p_mst2_sterm_woModi_clustered <- ggplot(explained_mst2_plot_nonModi_CLUSTERED,aes(x=sample,y=Snps,group=site,color=final_clust,fill=final_clust, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.1, alpha=.5)+ 
       # facet_grid(final_clust~.)+
   # labs("",
         # x="",
         # y="",
          # y="all")+
       scale_fill_manual(values=(plotColors))+
      scale_color_manual(values=(plotColors))+
       scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          ylim(c(0,1))+
 theme(#axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="bottom",
       legend.title = element_blank(),
       axis.text.x = element_blank(),
       axis.text.y = element_text(size=8),
       # axis.title = element_text(size=9),
       axis.title = element_blank(),
       
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )

p_mst2_sterm_woModi_clustered


##===========================================
##-------------------------------------plotting withou clustering
##===========================================

    p_mst2_sterm_woModi_NOT_clustered <- ggplot(explained_mst2_plot_nonModi_CLUSTERED,aes(x=sample,y=Snps,group=site, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.1, alpha=.5)+ 
       # facet_grid(final_clust~.)+
   # labs("",
         # x="",
         # y="",
          # y="all")+
       # scale_fill_manual(values=(plotColors))+
      # scale_color_manual(values=(plotColors))+
       scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="bottom",
       legend.title = element_blank(),
       # axis.text.x = element_blank(),
       axis.text.y = element_text(size=8),
       # axis.title = element_text(size=9),
       axis.title = element_blank(),
       
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
p_mst2_sterm_woModi_NOT_clustered

# library(plotly)
# ggp <- ggplotly(p_mst2_sterm_woModi_clustered)
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_mst2_clustered_woModi.html")

tsne_plot_mst2 <- tsne_plot

##===========================================
##-------------------------------------plotting without cheesemaking samples
##===========================================

explained_mst2_plot_nonModi_CLUSTERED_withoutcheese <- explained_mst2_plot_nonModi_CLUSTERED %>% filter(sample!="cheesemaking\nday1")  %>% filter(sample!="cheesemaking\nday2")

    p_mst2_sterm_woModi_NOT_clustered_withougCheese <- ggplot(explained_mst2_plot_nonModi_CLUSTERED_withoutcheese,aes(x=sample,y=Snps,group=site, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.1, alpha=.5)+ 
       scale_x_discrete( expand = c(0, 0)) +
  theme_classic()+
          ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="bottom",
       legend.title = element_blank(),
       axis.text.y = element_text(size=8),
       axis.title = element_blank(),
       
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
p_mst2_sterm_woModi_NOT_clustered_withougCheese
###------------------------------------
##density plot
###------------------------------------
explained_mst2_plot_nonModi_CLUSTERED$location <- as.numeric(str_split_fixed(explained_mst2_plot_nonModi_CLUSTERED$site, "_", 5)[,2])

# explained_mst2_plot_nonModi_CLUSTERED$location <- str_split_fixed(explained_mst2_plot_nonModi_CLUSTERED$site, "_", 5)[,4]
forDensityPlot <- explained_mst2_plot_nonModi_CLUSTERED %>% select(.,site,location,final_clust) %>% distinct()

forDensityPlot$final_clust <- droplevels(forDensityPlot$final_clust)
forDensityPlot$final_clust = factor(forDensityPlot$final_clust, levels=(sort(unique(forDensityPlot$final_clust),decreasing = FALSE)))
plotColors <- colorsTSNE[as.numeric(levels(forDensityPlot$final_clust))]



Sterm_mst2_density <- ggplot(forDensityPlot,aes(x=location,group=final_clust,color=final_clust,fill=final_clust))+geom_density(alpha=0.2)+theme_classic()+
  facet_wrap(~final_clust,scale = "free_y")+
  scale_fill_manual(values = colors_substrains)+
  scale_color_manual(values = colors_substrains)+
  labs(x="genomic location",y="density")+theme(strip.text.x = element_blank(),legend.position = "none",legend.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank())

Sterm_mst2_density


Sterm_mst2_point <- ggplot(forDensityPlot,aes(x=location,y=final_clust,group=final_clust,color=final_clust,fill=final_clust))+geom_jitter(width = 0.0, height = 0.4,alpha=0.7 ,size = 1)+theme_classic()+
  # facet_wrap(~final_clust,scale = "free_y")+
  scale_fill_manual(values = colors_substrains)+
  scale_color_manual(values = colors_substrains)+
  labs(x="genomic location",y="cluster")+theme(legend.position = "none",legend.title = element_blank(),axis.text.x = element_blank(),axis.ticks.x = element_blank())
Sterm_mst2_point
##====================================================
###------RIBBON of interquartile range
##====================================================

##=====================here I apply a hack!!
##in order to recreat eh labels from ggplot_build I color the samples and final_clust according to a known color scale

colorSamples <- c("gray1","gray2","gray3","gray4","gray5","gray6","gray7","gray8","gray9","gray10","gray11")
 # explained_mst2_plot_nonModi_CLUSTERED$final_clust = factor(explained_mst2_plot_nonModi_CLUSTERED$final_clust, levels=c(6:1))
explained_mst2_plot_nonModi_CLUSTERED$final_clust = factor(explained_mst2_plot_nonModi_CLUSTERED$final_clust, levels=(sort(unique(explained_mst2_plot_nonModi_CLUSTERED$final_clust),decreasing = FALSE)))
 explained_mst2_plot_nonModi_CLUSTERED$sample = factor(explained_mst2_plot_nonModi_CLUSTERED$sample, levels=(c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2")))

##----------------------prepare table

p <- ggplot(explained_mst2_plot_nonModi_CLUSTERED, aes(x=sample, y=Snps,color=sample,fill=final_clust)) + geom_boxplot() +scale_fill_manual(values = colorSamples)+scale_color_manual(values = colorSamples)
p
mst2_phase_qunatiles <- ggplot_build(p)$data[[1]] %>% as.tibble() %>%  rename(sample=colour)  %>%  rename(final_clust=fill) %>% mutate(final_clust = revalue(final_clust, c("gray1"="1", "gray2"="2", "gray3"="3", "gray4"="4", "gray5"="5", "gray6"="6", "gray7"="7", "gray8"="8")))%>% mutate(sample = revalue(sample, c("gray1"="Lyo\n1996", "gray2"="Lyo\n2012", "gray3"="Lyo\n2014", "gray4"="working\nstock", "gray5"="starter\nculture\n2012", "gray6"="starter\nculture\n2018", "gray7"="cheesemaking\nday1", "gray8"="cheesemaking\nday2")))
 table( mst2_phase_qunatiles$final_clust)
str(mst2_phase_qunatiles$final_clust)
colnames(mst2_phase_qunatiles)
# mst2_phase_qunatiles$final_clust = factor(mst2_phase_qunatiles$final_clust, levels=c(6:1))
mst2_phase_qunatiles$final_clust = factor(mst2_phase_qunatiles$final_clust, levels=(sort(unique(mst2_phase_qunatiles$final_clust),decreasing = FALSE)))

plotColors <- colorsTSNE[as.numeric(levels(mst2_phase_qunatiles$final_clust))]


 mst2_phase_qunatiles$sample = factor(mst2_phase_qunatiles$sample, levels=(c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2")))


mst2_phased_ribbon <- ggplot()+
    geom_ribbon(aes(x=sample,ymin = ymin, ymax = ymax,group=final_clust,fill=final_clust),data = mst2_phase_qunatiles, alpha=.6)+
    # geom_ribbon(aes(x=sample,ymin = X1st.Qu., ymax = X3rd.Qu.,group=final_clust,fill=final_clust),data = both_phase_qunatiles, alpha=.5)+
    # geom_ribbon(aes(x=sample,ymin = Min., ymax = Max.,group=final_clust,fill=final_clust),data = mst2_phase_qunatiles, alpha=.5)+
    # geom_line(data=explained_mst2_plot_nonModi_CLUSTERED,aes(x=sample,y=Snps,group=site,color=final_clust,fill=final_clust),size=0.1, alpha=.1)+ 
    # geom_line(data = mappingcoverage,aes(y = coverage, x = time_f_02,group=contig))+
      # facet_grid(contig~treatment )+
      # scale_x_continuous(breaks = c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
       # scale_x_discrete(breaks = c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    scale_fill_manual(values=(plotColors))+
      scale_color_manual(values=(plotColors))+
       scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="bottom",
       legend.title = element_blank(),
       # axis.text.x = element_blank(),
       axis.text.y = element_text(size=8),
       # axis.title = element_text(size=9),
       axis.title = element_blank(),
       
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )


  mst2_phased_ribbon
  
  
##====================================================
###------mutated genes
##====================================================
colnames(explained_mst1_plot_nonModi_CLUSTERED)
  
mst1_Explained_sterm_wide <- explained_mst1_plot_nonModi_CLUSTERED %>% select(c("site","finalName","X.CHROM.x","species","culture", "Preferred_name","effect","significance","geneName", "COG_Functional_Category","Repeat_cluster","core" , "mst6", "mst7","mst9", "mst10","mst3","mst4","mst5" ,"mst8" , "mst1", "mst2", "dN_dN.dS"  , "mutations" , "group",  "sample" , "Snps","final_clust" , "location")) %>% spread(., sample, Snps)
  
mst1_Explained_sterm_wide$LABELS <- ifelse(mst1_Explained_sterm_wide$Preferred_name=="0",mst1_Explained_sterm_wide$geneName,mst1_Explained_sterm_wide$Preferred_name)

mst1_genes_Mutated <- with(mst1_Explained_sterm_wide, table(LABELS,final_clust)) %>% as.data.frame() %>% filter(Freq>0) %>% arrange(desc(Freq))

mst1_genes_Mutated$LABELS <-  factor(mst1_genes_Mutated$LABELS, levels=unique(mst1_genes_Mutated$LABELS))
mst1_genes_Mutated$final_clust <-  factor(mst1_genes_Mutated$final_clust, levels=c("4","3","2","1"))


genes_mutated_mst1_ <- ggplot(mst1_genes_Mutated,aes(x=LABELS,y=Freq,color=final_clust,fill=final_clust))+geom_point()+theme_classic()+facet_wrap(final_clust~., scales="free")+
      scale_fill_manual(values=(plotColors))+
      scale_color_manual(values=(plotColors))+
   theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7),
         axis.title.x = element_blank(),
         strip.background = element_blank(),
          strip.text.x = element_blank(),
         legend.position = "none")

genes_mutated_mst1_
```



######both substrain

```{r}


colorsTSNE <- (c("#88654E","#C39953","#DECC6F"))

###--------------------------------------------------------------------
##subset
###--------------------------------------------------------------------


mst_both_metagenomicMutations <- sample_relative_safe_Sterm_final_wide  %>% filter(.,mst1>0.9 & mst2>0.9) %>% add_column(group="both")


# mst_both_metagenomicMutations <- sample_relative_wide %>% filter(.,species=="S. thermophilus") %>% filter(.,significance %in% c("MODERATE","HIGH")) %>% filter(.,mst1>0.9 & mst2>0.9) %>% add_column(group="both")%>%replace(is.na(.), 0)
mst_both_metagenomicMutations_long <-  mst_both_metagenomicMutations %>% gather(., sample, Snps, "Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2", factor_key=TRUE,na.rm = TRUE) #%>% filter(.,Snps>0.005)

mst_both_metagenomicMutations_tsne <- spread(mst_both_metagenomicMutations_long, sample, Snps)%>%replace(is.na(.), 0)


###--------------------------------------------------------------------
##prep for tSNE
###--------------------------------------------------------------------

fortsne <- mst_both_metagenomicMutations_tsne %>% select(.,"site","Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2")
fortsne <- fortsne[!duplicated(fortsne[,-1]), ]
# fortsne[duplicated(fortsne), ] %>% nrow()

set.seed(1234567)
# set.seed(2)

tsne <- Rtsne(fortsne[,-1], dims = 2, perplexity=10, verbose=TRUE, max_iter = 500)
# tsne <- Rtsne(fortsne[,41:45], dims = 2, perplexity=10, verbose=TRUE, max_iter = 500)
tsne_plot <- as.data.frame(tsne$Y)
tsne_plot$site <- fortsne[,1]
tsne_plot$species <- paste(str_split_fixed(tsne_plot$site, "_", 5)[,1],str_split_fixed(tsne_plot$site, "_", 5)[,2],sep = "_") %>%  gsub("_",". ",.)



##------------plot blank
ggplot(tsne_plot,aes(x=V1,y=V2))+geom_point()

##------------clusters


set.seed(123456)
# set.seed(2)

ds.real = dbscan(tsne_plot[,c(1,2)], 8)
# decreasing_clusternames <- (names(sort(table(ds.real$cluster),decreasing = TRUE)))
# change_name <- setNames(1:length(decreasing_clusternames),decreasing_clusternames)
clusterNames <- as.factor(ds.real$cluster)

# clusterNames <- revalue(as.factor(ds.real$cluster),change_name)
clusterNames<- revalue(clusterNames,c("2"="1","1"="2"))##add certain clusters to others
tsne_plot$density = factor(clusterNames)
ds.cent = tsne_plot %>% group_by(density) %>% select(V1, 
    V2) %>% summarize_all(mean)
ggplot(tsne_plot, aes(x = V1, y = V2, colour = density)) +
    geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label = density),
    data = ds.cent) + guides(colour = FALSE)


##-----------make clustering manual
# tsne_plot$final_clust <- tsne_plot$density
# tsne_plot_final_clust <- tsne_plot  %>% select(site,  final_clust)
# table(tsne_plot_final_clust$final_clust)

# tsne_plot_final_clust <- tsne_plot %>% mutate(final_clust = ifelse(density==3,2,
#                                                                     ifelse(density==2,3,1)))  %>% select(site,  final_clust)
#  tsne_plot <- tsne_plot %>% mutate(final_clust = ifelse(density==3,2,
#                                                                     ifelse(density==2,3,1)))
# table(tsne_plot_final_clust$final_clust)
tsne_plot <- tsne_plot %>% mutate(final_clust=density)
tsne_plot_final_clust <- tsne_plot %>% select(site,  final_clust)

###------------------------------------
##-------sumplementary plot tsne
###------------------------------------
tsne_plot$final_clust = factor(tsne_plot$final_clust, levels=1:length(unique(tsne_plot_final_clust$final_clust)))
ds.cent = tsne_plot %>% group_by(final_clust) %>% select(V1, 
    V2) %>% summarize_all(mean)
colors_substrains <- c(colorsTSNE[1:length(unique(tsne_plot_final_clust$final_clust))])

# Sterm_both_suplementaryTsne <- ggplot(tsne_plot, aes(x = V1, y = V2, colour = final_clust)) + 
#     geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label =final_clust), 
#     data = ds.cent) + guides(colour = FALSE)+scale_color_manual(values = colors_substrains)


Sterm_both_suplementaryTsne <- ggplot(tsne_plot, aes(x = V1, y = V2, colour = final_clust)) + 
    geom_point(alpha = 0.3) + theme_classic() + geom_label_repel(aes(label =final_clust), 
    data = ds.cent) + guides(colour = FALSE)+scale_color_manual(values = colors_substrains)+theme(axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank())


Sterm_both_suplementaryTsne

###------------------------------------
##-------sumplementary histogramm
###------------------------------------

Sterm_both_suplementarybarPlot <- ggplot(tsne_plot,aes(x=final_clust,color=final_clust,fill=final_clust))+geom_bar()+theme_classic()+
  scale_fill_manual(values = colors_substrains)+
  scale_color_manual(values = colors_substrains)+
  labs(x="",y="count")+theme(legend.position = "none")
Sterm_both_suplementarybarPlot
###------------------------------------
##---------clustering
###------------------------------------

explained_both_plot_nonModi_CLUSTERED <- merge(mst_both_metagenomicMutations_long,tsne_plot_final_clust,by = "site",all.x = TRUE)
# table(explained_both_plot_nonModi_CLUSTERED$final_clust)
# unique(explained_both_plot_nonModi_CLUSTERED$final_clust)
explained_both_plot_nonModi_CLUSTERED <- explained_both_plot_nonModi_CLUSTERED[which(!is.na(explained_both_plot_nonModi_CLUSTERED$final_clust)),]

explained_both_plot_nonModi_CLUSTERED$final_clust <- fct_explicit_na(as.character(explained_both_plot_nonModi_CLUSTERED$final_clust), na_level = "not clustered")
# unique(explained_both_plot_nonModi_CLUSTERED$final_clust)
explained_both_plot_nonModi_CLUSTERED$final_clust <- droplevels(explained_both_plot_nonModi_CLUSTERED$final_clust)

###------------------------------------
##plotting
###------------------------------------
# explained_both_plot_nonModi_CLUSTERED
# explained_both_plot_nonModi_CLUSTERED$final_clust = factor(explained_both_plot_nonModi_CLUSTERED$final_clust, levels=c("not clustered",2:1))
explained_both_plot_nonModi_CLUSTERED$final_clust = factor(explained_both_plot_nonModi_CLUSTERED$final_clust, levels=(sort(unique(explained_both_plot_nonModi_CLUSTERED$final_clust),decreasing = FALSE)))

# levels(explained_both_plot_nonModi_CLUSTERED$final_clust)
# plotColors <- c(colorsTSNE[as.numeric(levels(explained_both_plot_nonModi_CLUSTERED$final_clust))],"grey")
# plotColors <- c("grey",colorsTSNE[as.numeric(levels(explained_both_plot_nonModi_CLUSTERED$final_clust))[2:5]])
plotColors <- c(colorsTSNE[as.numeric(levels(explained_both_plot_nonModi_CLUSTERED$final_clust))[1:5]])

# explained_both_plot_nonModi_CLUSTERED <- explained_both_plot_nonModi_CLUSTERED %>% filter(.,final_clust=="2")
##assign colors
explained_both_plot_nonModi_CLUSTERED$final_clust <- droplevels(explained_both_plot_nonModi_CLUSTERED$final_clust)
colors_both <- data.frame(cluster=paste("both",as.character(levels(explained_both_plot_nonModi_CLUSTERED$final_clust)),sep="_"),color=plotColors[1:length(levels(explained_both_plot_nonModi_CLUSTERED$final_clust))])


    p_both_sterm_woModi_clustered <- ggplot(explained_both_plot_nonModi_CLUSTERED,aes(x=sample,y=Snps,group=site,color=final_clust,fill=final_clust, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.1, alpha=.5)+ 
       # facet_grid(final_clust~.)+
   # labs("",
         # x="",
         # y="",
          # y="all")+
       scale_fill_manual(values=(plotColors))+
      scale_color_manual(values=(plotColors))+
       scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          ylim(c(0,1))+
 theme(#axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="bottom",
       legend.title = element_blank(),
       axis.text.x = element_blank(),
       axis.text.y = element_text(size=8),
       # axis.title = element_text(size=9),
       axis.title = element_blank(),
       
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
p_both_sterm_woModi_clustered


##===========================================
##-------------------------------------plotting withou clustering
##===========================================

    p_both_sterm_woModi_NOT_clustered <- ggplot(explained_both_plot_nonModi_CLUSTERED,aes(x=sample,y=Snps,group=site, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.1, alpha=.5)+ 
       scale_x_discrete( expand = c(0, 0)) +
  theme_classic()+
          ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="bottom",
       legend.title = element_blank(),
       axis.text.y = element_text(size=8),
       axis.title = element_blank(),
       
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
p_both_sterm_woModi_NOT_clustered

# library(plotly)
# ggp <- ggplotly(p_both_sterm_woModi_clustered)
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_both_clustered_woModi.html")

tsne_plot_both <- tsne_plot

##===========================================
##-------------------------------------plotting without cheesemaking samples
##===========================================

explained_both_plot_nonModi_CLUSTERED_withoutcheese <- explained_both_plot_nonModi_CLUSTERED %>% filter(sample!="cheesemaking\nday1")  %>% filter(sample!="cheesemaking\nday2")

    p_both_sterm_woModi_NOT_clustered_withougCheese <- ggplot(explained_both_plot_nonModi_CLUSTERED_withoutcheese,aes(x=sample,y=Snps,group=site, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.1, alpha=.5)+ 
       scale_x_discrete( expand = c(0, 0)) +
  theme_classic()+
          ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="bottom",
       legend.title = element_blank(),
       axis.text.y = element_text(size=8),
       axis.title = element_blank(),
       
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
p_both_sterm_woModi_NOT_clustered_withougCheese
###------------------------------------
##density plot
###------------------------------------
explained_both_plot_nonModi_CLUSTERED$location <- as.numeric(str_split_fixed(explained_both_plot_nonModi_CLUSTERED$site, "_", 5)[,2])

# explained_both_plot_nonModi_CLUSTERED$location <- str_split_fixed(explained_both_plot_nonModi_CLUSTERED$site, "_", 5)[,4]
forDensityPlot <- explained_both_plot_nonModi_CLUSTERED %>% select(.,site,location,final_clust) %>% distinct()

# forDensityPlot$final_clust <- droplevels(forDensityPlot$final_clust)
# forDensityPlot$final_clust = factor(forDensityPlot$final_clust, levels=(sort(unique(forDensityPlot$final_clust),decreasing = TRUE)))
# plotColors <- colorsTSNE[as.numeric(levels(forDensityPlot$final_clust))]

forDensityPlot$final_clust <- droplevels(forDensityPlot$final_clust)
# forDensityPlot$final_clust = factor(forDensityPlot$final_clust, levels=(sort(unique(forDensityPlot$final_clust),decreasing = TRUE)))

# forDensityPlot$final_clust = factor(forDensityPlot$final_clust, levels=c("not clustered",4:1),decreasing = TRUE)))
# plotColors <-  c(colorsTSNE[as.numeric(levels(explained_both_plot_nonModi_CLUSTERED$final_clust))[2:5]])

forDensityPlot$final_clust = factor(forDensityPlot$final_clust, levels=(sort(unique(forDensityPlot$final_clust),decreasing = FALSE)))
# colors_substrains <- c(colorsTSNE[1:length(unique(tsne_plot_final_clust$final_clust))])


Sterm_both_density <- ggplot(forDensityPlot,aes(x=location,group=final_clust,color=final_clust,fill=final_clust))+geom_density(alpha=0.2)+theme_classic()+
  facet_wrap(~final_clust,scale = "free_y")+
  scale_fill_manual(values = colors_substrains)+
  scale_color_manual(values = colors_substrains)+
  # labs(x="genomic location",y="density")+theme(legend.position = "none",legend.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank())
labs(x="genomic location",y="density")+theme(strip.text.x = element_blank(),legend.position = "none",legend.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank())
Sterm_both_density


Sterm_both_point <- ggplot(forDensityPlot,aes(x=location,y=final_clust,group=final_clust,color=final_clust,fill=final_clust))+geom_jitter(width = 0.0, height = 0.4,alpha=0.7 ,size = 1)+theme_classic()+
  # facet_wrap(~final_clust,scale = "free_y")+
  scale_fill_manual(values = colors_substrains)+
  scale_color_manual(values = colors_substrains)+
  labs(x="genomic location",y="cluster")+theme(legend.position = "none",legend.title = element_blank(),axis.text.x = element_blank(),axis.ticks.x = element_blank())
Sterm_both_point

##====================================================
###------RIBBON of interquartile range
##====================================================

##=====================here I apply a hack!!
##in order to recreat eh labels from ggplot_build I color the samples and final_clust according to a known color scale

colorSamples <- c("gray1","gray2","gray3","gray4","gray5","gray6","gray7","gray8","gray9","gray10","gray11")
# explained_both_plot_nonModi_CLUSTERED$final_clust = factor(explained_both_plot_nonModi_CLUSTERED$final_clust, levels=c("not clustered",2:1))
 explained_both_plot_nonModi_CLUSTERED$sample = factor(explained_both_plot_nonModi_CLUSTERED$sample, levels=(c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2")))

##----------------------prepare table

p <- ggplot(explained_both_plot_nonModi_CLUSTERED, aes(x=sample, y=Snps,color=sample,fill=final_clust)) + geom_boxplot() +scale_fill_manual(values = colorSamples)+scale_color_manual(values = colorSamples)
p
both_phase_qunatiles <- ggplot_build(p)$data[[1]] %>% as.tibble() %>%  rename(sample=colour)  %>%  rename(final_clust=fill) %>% mutate(final_clust = revalue(final_clust, c("gray1"="1", "gray2"="2", "gray3"="3", "gray4"="4", "gray5"="5", "gray6"="6", "gray7"="7", "gray8"="8")))%>% mutate(sample = revalue(sample, c("gray1"="Lyo\n1996", "gray2"="Lyo\n2012", "gray3"="Lyo\n2014", "gray4"="working\nstock", "gray5"="starter\nculture\n2012", "gray6"="starter\nculture\n2018", "gray7"="cheesemaking\nday1", "gray8"="cheesemaking\nday2")))
 table( both_phase_qunatiles$final_clust)
str(both_phase_qunatiles$final_clust)


# both_phase_qunatiles$final_clust = factor(both_phase_qunatiles$final_clust, levels=c(2:1))
# plotColors <- colorsTSNE[as.numeric(levels(both_phase_qunatiles$final_clust))]


 both_phase_qunatiles$sample = factor(both_phase_qunatiles$sample, levels=(c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2")))


both_phased_ribbon <- ggplot()+
    geom_ribbon(aes(x=sample,ymin = ymin, ymax = ymax,group=final_clust,fill=final_clust),data = both_phase_qunatiles, alpha=.4)+
    # geom_ribbon(aes(x=sample,ymin = X1st.Qu., ymax = X3rd.Qu.,group=final_clust,fill=final_clust),data = both_phase_qunatiles, alpha=.5)+
    # geom_ribbon(aes(x=sample,ymin = Min., ymax = Max.,group=final_clust,fill=final_clust),data = both_phase_qunatiles, alpha=.5)+
    # geom_line(data=explained_both_plot_nonModi_CLUSTERED,aes(x=sample,y=Snps,group=site,color=final_clust,fill=final_clust),size=0.1, alpha=.1)+ 
    # geom_line(data = mappingcoverage,aes(y = coverage, x = time_f_02,group=contig))+
      # facet_grid(contig~treatment )+
      # scale_x_continuous(breaks = c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
       # scale_x_discrete(breaks = c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    scale_fill_manual(values=(plotColors))+
      scale_color_manual(values=(plotColors))+
       scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="bottom",
       legend.title = element_blank(),
       # axis.text.x = element_blank(),
       axis.text.y = element_text(size=8),
       # axis.title = element_text(size=9),
       axis.title = element_blank(),
       
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )


  both_phased_ribbon

##====================================================
###------mutated genes
##====================================================
colnames(explained_both_plot_nonModi_CLUSTERED)
  
both_Explained_sterm_wide <- explained_both_plot_nonModi_CLUSTERED %>% select(c("site","finalName","X.CHROM.x","species","culture", "Preferred_name","effect","significance","geneName", "COG_Functional_Category","Repeat_cluster","core" , "mst6", "mst7","mst9", "mst10","mst3","mst4","mst5" ,"mst8" , "mst1", "mst2", "dN_dN.dS"  , "mutations" , "group",  "sample" , "Snps","final_clust" , "location")) %>% spread(., sample, Snps)
  
both_Explained_sterm_wide$LABELS <- ifelse(both_Explained_sterm_wide$Preferred_name=="0",both_Explained_sterm_wide$geneName,both_Explained_sterm_wide$Preferred_name)

both_genes_Mutated <- with(both_Explained_sterm_wide, table(LABELS,final_clust)) %>% as.data.frame() %>% filter(Freq>0) %>% arrange(desc(Freq))

both_genes_Mutated$LABELS <-  factor(both_genes_Mutated$LABELS, levels=unique(both_genes_Mutated$LABELS))
both_genes_Mutated$final_clust <-  factor(both_genes_Mutated$final_clust, levels=c("2","1"))


genes_mutated_both_ <- ggplot(both_genes_Mutated,aes(x=LABELS,y=Freq,color=final_clust,fill=final_clust))+geom_point()+theme_classic()+facet_wrap(final_clust~., scales="free")+
      scale_fill_manual(values=(plotColors))+
      scale_color_manual(values=(plotColors))+
   theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7),
         axis.title.x = element_blank(),
         strip.background = element_blank(),
          strip.text.x = element_blank(),
         legend.position = "none")

genes_mutated_both_

```



######not explained

```{r}


colorsTSNE <- rev(c("gray7","#676867","#777877","#909090","#AAAAAA","#BCBCBC","grey"))

###--------------------------------------------------------------------
##subset
###--------------------------------------------------------------------


mst_none_metagenomicMutations <- sample_relative_safe_Sterm_final_wide  %>% filter(.,mst1<=0.9 & mst2<=0.9) %>% add_column(group="not explained")

# mst_none_metagenomicMutations <- sample_relative_wide %>% filter(.,species=="S. thermophilus") %>% filter(.,significance %in% c("MODERATE","HIGH")) %>% filter(.,mst1<=0.9 & mst2<=0.9) %>% add_column(group="not explained")%>%replace(is.na(.), 0)
mst_none_metagenomicMutations_long <-  mst_none_metagenomicMutations %>% gather(., sample, Snps, "Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2", factor_key=TRUE,na.rm = TRUE)# %>% filter(.,Snps>0.005)


mst_none_metagenomicMutations_tsne <- spread(mst_none_metagenomicMutations_long, sample, Snps)%>%replace(is.na(.), 0)




###--------------------------------------------------------------------
##prep for tSNE
###--------------------------------------------------------------------




fortsne <- mst_none_metagenomicMutations_tsne %>% select(.,"site","Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2")
fortsne <- fortsne[!duplicated(fortsne[,-1]), ]
# fortsne[duplicated(fortsne), ] %>% nrow()

set.seed(1234567)
# set.seed(2)

tsne <- Rtsne(fortsne[,-1], dims = 2, perplexity=10, verbose=TRUE, max_iter = 500)
# tsne <- Rtsne(fortsne[,41:45], dims = 2, perplexity=10, verbose=TRUE, max_iter = 500)
tsne_plot <- as.data.frame(tsne$Y)
tsne_plot$site <- fortsne[,1]
tsne_plot$species <- paste(str_split_fixed(tsne_plot$site, "_", 5)[,1],str_split_fixed(tsne_plot$site, "_", 5)[,2],sep = "_") %>%  gsub("_",". ",.)



##------------plot blank
# ggplot(tsne_plot,aes(x=V1,y=V2))+geom_point()

##------------clusters


set.seed(123456)
# set.seed(2)

ds.real = dbscan(tsne_plot[,c(1,2)], 5)
# decreasing_clusternames <- (names(sort(table(ds.real$cluster),decreasing = TRUE)))
# change_name <- setNames(1:length(decreasing_clusternames),decreasing_clusternames)
# clusterNames <- revalue(as.fdecreasing_clusternamesactor(ds.real$cluster),change_name)
# clusterNames<- revalue(clusterNames,c("7"="3"))##add certain clusters to others
tsne_plot$density = as.factor(ds.real$cluster)
 
##-----------make clustering manual
# tsne_plot$final_clust <- tsne_plot$density
# tsne_plot_final_clust <- tsne_plot  %>% select(site,  final_clust)
# table(tsne_plot_final_clust$final_clust)

# tsne_plot_final_clust <- tsne_plot %>% mutate(final_clust = ifelse(density==3,2,
#                                                                     ifelse(density==2,3,1)))  %>% select(site,  final_clust)
#  tsne_plot <- tsne_plot %>% mutate(final_clust = ifelse(density==3,2,
#                                                                     ifelse(density==2,3,1)))
# table(tsne_plot_final_clust$final_clust)
tsne_plot <- tsne_plot %>% mutate(final_clust=density)
tsne_plot_final_clust <- tsne_plot %>% select(site,  final_clust)

###------------------------------------
##-------sumplementary plot tsne
###------------------------------------
tsne_plot$final_clust = factor(tsne_plot$final_clust, levels=1:length(unique(tsne_plot_final_clust$final_clust)))
ds.cent = tsne_plot %>% group_by(final_clust) %>% select(V1, 
    V2) %>% summarize_all(mean)
colors_substrains <- c(colorsTSNE[1:length(unique(tsne_plot_final_clust$final_clust))])

# Sterm_none_suplementaryTsne <- ggplot(tsne_plot, aes(x = V1, y = V2, colour = final_clust)) +
#   labs(x="",y="")+
#     geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label =final_clust), 
#     data = ds.cent) + guides(colour = FALSE)+scale_color_manual(values = colors_substrains)

Sterm_none_suplementaryTsne <- ggplot(tsne_plot, aes(x = V1, y = V2, colour = final_clust)) + 
    geom_point(alpha = 0.3) + theme_classic() + geom_label_repel(aes(label =final_clust), 
    data = ds.cent) + guides(colour = FALSE)+scale_color_manual(values = colors_substrains)+theme(axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank())

Sterm_none_suplementaryTsne

###------------------------------------
##-------sumplementary histogramm
###------------------------------------

Sterm_none_suplementarybarPlot <- ggplot(tsne_plot,aes(x=final_clust,color=final_clust,fill=final_clust))+geom_bar()+theme_classic()+
  scale_fill_manual(values = colors_substrains)+
  scale_color_manual(values = colors_substrains)+
  labs(x="",y="count")+theme(legend.position = "none")
Sterm_none_suplementarybarPlot

###------------------------------------
##---------clustering
###------------------------------------

explained_none_plot_nonModi_CLUSTERED <- merge(mst_none_metagenomicMutations_long,tsne_plot_final_clust,by = "site",all.x = TRUE)
# table(explained_none_plot_nonModi_CLUSTERED$final_clust)
# unique(explained_none_plot_nonModi_CLUSTERED$final_clust)
explained_none_plot_nonModi_CLUSTERED <- explained_none_plot_nonModi_CLUSTERED[which(!is.na(explained_none_plot_nonModi_CLUSTERED$final_clust)),]

# explained_none_plot_nonModi_CLUSTERED$final_clust <- fct_explicit_na(as.character(explained_none_plot_nonModi_CLUSTERED$final_clust), na_level = "not clustered")
# unique(explained_none_plot_nonModi_CLUSTERED$final_clust)
explained_none_plot_nonModi_CLUSTERED$final_clust <- droplevels(explained_none_plot_nonModi_CLUSTERED$final_clust)

###------------------------------------
##plotting
###------------------------------------
table(explained_none_plot_nonModi_CLUSTERED$final_clust)
# explained_none_plot_nonModi_CLUSTERED$final_clust = factor(explained_none_plot_nonModi_CLUSTERED$final_clust, levels=c(6:1))

# levels(explained_none_plot_nonModi_CLUSTERED$final_clust)
# plotColors <- c(colorsTSNE[as.numeric(levels(explained_none_plot_nonModi_CLUSTERED$final_clust))],"grey")
# plotColors <- c("grey",colorsTSNE[as.numeric(levels(explained_none_plot_nonModi_CLUSTERED$final_clust))[2:6]])
# plotColors <- c(colorsTSNE[as.numeric(levels(explained_none_plot_nonModi_CLUSTERED$final_clust))[1:6]])
explained_none_plot_nonModi_CLUSTERED$final_clust = factor(explained_none_plot_nonModi_CLUSTERED$final_clust, levels=(sort(as.numeric(unique(explained_none_plot_nonModi_CLUSTERED$final_clust)),decreasing = FALSE)))

# explained_none_plot_nonModi_CLUSTERED$final_clust = factor(explained_none_plot_nonModi_CLUSTERED$final_clust, levels=(sort(as.numeric(unique(explained_none_plot_nonModi_CLUSTERED$final_clust)),decreasing = TRUE)))
plotColors <- rev(colorsTSNE[as.numeric(levels(explained_none_plot_nonModi_CLUSTERED$final_clust))])
# table(explained_none_plot_nonModi_CLUSTERED$final_clust)

##assign colors
# colors_none <- data.frame(cluster=paste("none",as.character(levels(explained_none_plot_nonModi_CLUSTERED$final_clust)),sep="_"),color=plotColors)

explained_none_plot_nonModi_CLUSTERED$final_clust <- droplevels(explained_none_plot_nonModi_CLUSTERED$final_clust)
colors_none <- data.frame(cluster=paste("none",as.character(levels(explained_none_plot_nonModi_CLUSTERED$final_clust)),sep="_"),color=plotColors[1:length(levels(explained_none_plot_nonModi_CLUSTERED$final_clust))])


    p_none_sterm_woModi_clustered <- ggplot(explained_none_plot_nonModi_CLUSTERED,aes(x=sample,y=Snps,group=site,color=final_clust,fill=final_clust, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.1, alpha=.5)+ 
       # facet_grid(final_clust~.)+
   # labs("",
         # x="",
         # y="",
          # y="all")+
       scale_fill_manual(values=(plotColors))+
      scale_color_manual(values=(plotColors))+
       scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="bottom",
       legend.title = element_blank(),
       # axis.text.x = element_blank(),
       axis.text.y = element_text(size=8),
       # axis.title = element_text(size=9),
       axis.title = element_blank(),
       
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
p_none_sterm_woModi_clustered

# 
# ggp <- ggplotly(p_none_sterm_woModi_clustered)
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_NONE_clustered_woModi.html")


##===========================================
##-------------------------------------plotting withou clustering
##===========================================

    p_none_sterm_woModi_NOT_clustered <- ggplot(explained_none_plot_nonModi_CLUSTERED,aes(x=sample,y=Snps,group=site, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.1, alpha=.5)+ 
       # facet_grid(final_clust~.)+
   # labs("",
         # x="",
         # y="",
          # y="all")+
       # scale_fill_manual(values=(plotColors))+
      # scale_color_manual(values=(plotColors))+
       scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="bottom",
       legend.title = element_blank(),
       # axis.text.x = element_blank(),
       axis.text.y = element_text(size=8),
       # axis.title = element_text(size=9),
       axis.title = element_blank(),
       
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
p_none_sterm_woModi_NOT_clustered

# library(plotly)
# ggp <- ggplotly(p_none_sterm_woModi_clustered)
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_none_clustered_woModi.html")

tsne_plot_none <- tsne_plot
 
 ##===========================================
##-------------------------------------plotting without cheesemaking samples
##===========================================

explained_none_plot_nonModi_CLUSTERED_withoutcheese <- explained_none_plot_nonModi_CLUSTERED %>% filter(sample!="cheesemaking\nday1")  %>% filter(sample!="cheesemaking\nday2")

    p_none_sterm_woModi_NOT_clustered_withougCheese <- ggplot(explained_none_plot_nonModi_CLUSTERED_withoutcheese,aes(x=sample,y=Snps,group=site, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.1, alpha=.5)+ 
       scale_x_discrete( expand = c(0, 0)) +
  theme_classic()+
          ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="bottom",
       legend.title = element_blank(),
       axis.text.y = element_text(size=8),
       axis.title = element_blank(),
       
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
p_none_sterm_woModi_NOT_clustered_withougCheese
###------------------------------------
##density plot
###------------------------------------

explained_none_plot_nonModi_CLUSTERED$location <- as.numeric(str_split_fixed(explained_none_plot_nonModi_CLUSTERED$site, "_", 5)[,2])
forDensityPlot <- explained_none_plot_nonModi_CLUSTERED %>% select(.,site,location,final_clust) %>% distinct()

# forDensityPlot$final_clust <- droplevels(forDensityPlot$final_clust)
# forDensityPlot$final_clust = factor(forDensityPlot$final_clust, levels=(sort(unique(forDensityPlot$final_clust),decreasing = TRUE)))
# plotColors <- colorsTSNE[as.numeric(levels(forDensityPlot$final_clust))]

forDensityPlot$final_clust <- droplevels(forDensityPlot$final_clust)
# forDensityPlot$final_clust = factor(forDensityPlot$final_clust, levels=c("not clustered",6:1))
forDensityPlot$final_clust = factor(forDensityPlot$final_clust, levels=(sort(unique(forDensityPlot$final_clust),decreasing = TRUE)))


Sterm_none_density <- ggplot(forDensityPlot,aes(x=location,group=final_clust,color=final_clust,fill=final_clust))+geom_density(alpha=0.2)+theme_classic()+
    facet_wrap(~final_clust,scale = "free_y")+
  scale_fill_manual(values = colors_substrains)+
  scale_color_manual(values = colors_substrains)+
  labs(x="genomic location",y="density")+theme(strip.text.x = element_blank(),legend.position = "none",legend.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank())
  # labs(x="genomic location",y="density")+theme(legend.position = "none",legend.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank())

Sterm_none_density

Sterm_none_point <- ggplot(forDensityPlot,aes(x=location,y=final_clust,group=final_clust,color=final_clust,fill=final_clust))+geom_jitter(width = 0.0, height = 0.4,alpha=0.7 ,size = 1)+theme_classic()+
  # facet_wrap(~final_clust,scale = "free_y")+
  scale_fill_manual(values = colors_substrains)+
  scale_color_manual(values = colors_substrains)+
  labs(x="genomic location",y="cluster")+theme(legend.position = "none",legend.title = element_blank(),axis.text.x = element_blank(),axis.ticks.x = element_blank())
Sterm_none_point

##====================================================
###------RIBBON of interquartile range
##====================================================

##=====================here I apply a hack!!
##in order to recreat eh labels from ggplot_build I color the samples and final_clust according to a known color scale

colorSamples <- c("gray1","gray2","gray3","gray4","gray5","gray6","gray7","gray8","gray9","gray10","gray11")
 # explained_none_plot_nonModi_CLUSTERED$final_clust = factor(explained_none_plot_nonModi_CLUSTERED$final_clust, levels=c(6:1))
  # non_phase_qunatiles$final_clust = factor(non_phase_qunatiles$final_clust, levels=c("not clustered",6:1))

# explained_none_plot_nonModi_CLUSTERED$final_clust = factor(explained_none_plot_nonModi_CLUSTERED$final_clust, levels=c(6:1))
 explained_none_plot_nonModi_CLUSTERED$sample = factor(explained_none_plot_nonModi_CLUSTERED$sample, levels=(c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2")))

 
 levels(explained_none_plot_nonModi_CLUSTERED$final_clust)
##----------------------prepare table

p <- ggplot(explained_none_plot_nonModi_CLUSTERED, aes(x=sample, y=Snps,color=sample,fill=final_clust)) + geom_boxplot() +scale_fill_manual(values = colorSamples)+scale_color_manual(values = colorSamples)
p
non_phase_qunatiles <- ggplot_build(p)$data[[1]] %>% as.tibble() %>%  rename(sample=colour)  %>%  rename(final_clust=fill) %>% mutate(final_clust = revalue(final_clust, c("gray1"="1", "gray2"="2", "gray3"="3", "gray4"="4", "gray5"="5", "gray6"="6", "gray7"="7", "gray8"="8")))%>% mutate(sample = revalue(sample, c("gray1"="Lyo\n1996", "gray2"="Lyo\n2012", "gray3"="Lyo\n2014", "gray4"="working\nstock", "gray5"="starter\nculture\n2012", "gray6"="starter\nculture\n2018", "gray7"="cheesemaking\nday1", "gray8"="cheesemaking\nday2"))) %>% select(-"y")
 table( non_phase_qunatiles$final_clust)
unique(non_phase_qunatiles$final_clust)


# non_phase_qunatiles$final_clust = factor(non_phase_qunatiles$final_clust, levels=c(6:1))
# plotColors <- colorsTSNE[as.numeric(levels(non_phase_qunatiles$final_clust))]
# non_phase_qunatiles$final_clust = factor(non_phase_qunatiles$final_clust, levels=c("not clustered",6:1))
# plotColors <- c(colorsTSNE[as.numeric(levels(non_phase_qunatiles$final_clust))[2:7]])

# plotColors <- c(colorsTSNE[as.numeric(levels(non_phase_qunatiles$final_clust))[2:7]])


 non_phase_qunatiles$sample = factor(non_phase_qunatiles$sample, levels=(c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2")))


none_phased_ribbon <- ggplot()+
    geom_ribbon(aes(x=sample,ymin = ymin, ymax = ymax,group=final_clust,fill=final_clust),data = non_phase_qunatiles, alpha=.4)+
      # geom_ribbon(aes(x=sample,ymin = lower, ymax = upper,group=final_clust,fill=final_clust),data = non_phase_qunatiles, alpha=.4)+
    # geom_ribbon(aes(x=sample,ymin = X1st.Qu., ymax = X3rd.Qu.,group=final_clust,fill=final_clust),data = none_phase_qunatiles, alpha=.5)+
    # geom_ribbon(aes(x=sample,ymin = Min., ymax = Max.,group=final_clust,fill=final_clust),data = non_phase_qunatiles, alpha=.5)+
    # geom_line(data=explained_none_plot_nonModi_CLUSTERED,aes(x=sample,y=Snps,group=site,color=final_clust,fill=final_clust),size=0.1, alpha=.1)+ 
    # geom_line(data = mappingcoverage,aes(y = coverage, x = time_f_02,group=contig))+
      # facet_grid(contig~treatment )+
      # scale_x_continuous(breaks = c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
       # scale_x_discrete(breaks = c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    scale_fill_manual(values=(plotColors))+
      scale_color_manual(values=(plotColors))+
       scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="bottom",
       legend.title = element_blank(),
       # axis.text.x = element_blank(),
       axis.text.y = element_text(size=8),
       # axis.title = element_text(size=9),
       axis.title = element_blank(),
       
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )


  none_phased_ribbon
  
  
##====================================================
###------mutated genes
##====================================================
colnames(explained_none_plot_nonModi_CLUSTERED)
  table(explained_none_plot_nonModi_CLUSTERED$final_clust)
none_Explained_sterm_wide <- explained_none_plot_nonModi_CLUSTERED %>% select(c("site","finalName","X.CHROM.x","species","culture", "Preferred_name","effect","significance","geneName", "COG_Functional_Category","Repeat_cluster","core" , "mst6", "mst7","mst9", "mst10","mst3","mst4","mst5" ,"mst8" , "mst1", "mst2", "dN_dN.dS"  , "mutations" , "group",  "sample" , "Snps","final_clust" , "location")) %>% spread(., sample, Snps)
  
none_Explained_sterm_wide$LABELS <- ifelse(none_Explained_sterm_wide$Preferred_name=="0",none_Explained_sterm_wide$geneName,none_Explained_sterm_wide$Preferred_name)

none_genes_Mutated <- with(none_Explained_sterm_wide, table(LABELS,final_clust)) %>% as.data.frame() %>% filter(Freq>0) %>% arrange(desc(Freq))

none_genes_Mutated$LABELS <-  factor(none_genes_Mutated$LABELS, levels=unique(none_genes_Mutated$LABELS))
none_genes_Mutated$final_clust <-  factor(none_genes_Mutated$final_clust, levels=c("5","4","3","2","1"))


genes_mutated_none_ <- ggplot(none_genes_Mutated,aes(x=LABELS,y=Freq,color=final_clust,fill=final_clust))+geom_point()+theme_classic()+facet_wrap(final_clust~., scales="free")+
      scale_fill_manual(values=(plotColors))+
      scale_color_manual(values=(plotColors))+
   theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7),
         axis.title.x = element_blank(),
         strip.background = element_blank(),
          strip.text.x = element_blank(),
         legend.position = "none")

genes_mutated_none_
```

#####Replication

```{r}

non_phase_qunatiles$grouping <- paste("none",non_phase_qunatiles$final_clust,sep = "_")
both_phase_qunatiles$grouping <- paste("both",both_phase_qunatiles$final_clust,sep = "_")
mst1_phase_qunatiles$grouping <- paste("mst1",mst1_phase_qunatiles$final_clust,sep = "_")
mst2_phase_qunatiles$grouping <- paste("mst2",mst2_phase_qunatiles$final_clust,sep = "_")

table(non_phase_qunatiles$grouping)
colnames(non_phase_qunatiles)
colnames(both_phase_qunatiles)
colnames(mst1_phase_qunatiles)
colnames(mst2_phase_qunatiles)

colnames(non_phase_qunatiles)
colnames(mst1_phase_qunatiles)

all_SNPs_phased <- rbind(non_phase_qunatiles,both_phase_qunatiles,mst1_phase_qunatiles,mst2_phase_qunatiles)
colors_all_phases <- rbind(colors_none,colors_both,colors_mst1,colors_mst2)
 
high_type <- c("both_2","mst1_5","mst2_2","mst2_4","none_6")
low_type <- c("both_1","mst1_3","mst2_1","mst2_3","none_2","none_4","none_5")
lost_type <- c("mst1_1","mst1_2","mst1_4","none_1","none_3")


##--------high---------
high_genotype_snps <- all_SNPs_phased %>% filter(grouping %in% high_type) 
colors_high <- as.character(colors_all_phases[which(colors_all_phases$cluster %in% high_type),"color"])
high_type_order <- as.character(colors_all_phases[which(colors_all_phases$cluster %in% high_type),"cluster"])
high_genotype_snps$grouping = factor(high_genotype_snps$grouping, levels=high_type_order)
levels(high_genotype_snps$grouping)
droplevels(high_genotype_snps$grouping)


high_phased_ribbon <- ggplot()+
    geom_ribbon(aes(x=sample,ymin = ymin, ymax = ymax,group=grouping,fill=grouping),data = high_genotype_snps, alpha=.4)+
      # geom_ribbon(aes(x=sample,ymin = lower, ymax = upper,group=final_clust,fill=final_clust),data = non_phase_qunatiles, alpha=.4)+
    # geom_ribbon(aes(x=sample,ymin = X1st.Qu., ymax = X3rd.Qu.,group=final_clust,fill=final_clust),data = none_phase_qunatiles, alpha=.5)+
    # geom_ribbon(aes(x=sample,ymin = Min., ymax = Max.,group=final_clust,fill=final_clust),data = non_phase_qunatiles, alpha=.5)+
    # geom_line(data=explained_none_plot_nonModi_CLUSTERED,aes(x=sample,y=Snps,group=site,color=final_clust,fill=final_clust),size=0.1, alpha=.1)+ 
    # geom_line(data = mappingcoverage,aes(y = coverage, x = time_f_02,group=contig))+
      # facet_grid(contig~treatment )+
      # scale_x_continuous(breaks = c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
       # scale_x_discrete(breaks = c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    scale_fill_manual(values=(colors_high))+
      scale_color_manual(values=(colors_high))+
       scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="bottom",
       legend.title = element_blank(),
       # axis.text.x = element_blank(),
       axis.text.y = element_text(size=8),
       # axis.title = element_text(size=9),
       axis.title = element_blank(),
       
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )


  high_phased_ribbon

Sterm_high_point <- ggplot(high_genotype_snps,aes(x=location,y=final_clust,group=final_clust,color=final_clust,fill=final_clust))+geom_jitter(width = 0.0, height = 0.4,alpha=0.7 ,size = 1)+theme_classic()+
  # facet_wrap(~final_clust,scale = "free_y")+
  scale_fill_manual(values = colors_substrains)+
  scale_color_manual(values = colors_substrains)+
  labs(x="genomic location",y="cluster")+theme(legend.position = "none",legend.title = element_blank(),axis.text.x = element_blank(),axis.ticks.x = element_blank())
Sterm_mst2_point
  

##--------LOW---------
low_genotype_snps <- all_SNPs_phased %>% filter(grouping %in% low_type) 
colors_low <- as.character(colors_all_phases[which(colors_all_phases$cluster %in% low_type),"color"]) 
low_type_order <- as.character(colors_all_phases[which(colors_all_phases$cluster %in% low_type),"cluster"])    
low_genotype_snps$grouping = factor(low_genotype_snps$grouping, levels=low_type_order)
droplevels(low_genotype_snps$grouping)
levels(low_genotype_snps$grouping)


low_phased_ribbon <- ggplot()+
    geom_ribbon(aes(x=sample,ymin = ymin, ymax = ymax,group=grouping,fill=grouping),data = low_genotype_snps, alpha=.4)+
      # geom_ribbon(aes(x=sample,ymin = lower, ymax = upper,group=final_clust,fill=final_clust),data = non_phase_qunatiles, alpha=.4)+
    # geom_ribbon(aes(x=sample,ymin = X1st.Qu., ymax = X3rd.Qu.,group=final_clust,fill=final_clust),data = none_phase_qunatiles, alpha=.5)+
    # geom_ribbon(aes(x=sample,ymin = Min., ymax = Max.,group=final_clust,fill=final_clust),data = non_phase_qunatiles, alpha=.5)+
    # geom_line(data=explained_none_plot_nonModi_CLUSTERED,aes(x=sample,y=Snps,group=site,color=final_clust,fill=final_clust),size=0.1, alpha=.1)+ 
    # geom_line(data = mappingcoverage,aes(y = coverage, x = time_f_02,group=contig))+
      # facet_grid(contig~treatment )+
      # scale_x_continuous(breaks = c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
       # scale_x_discrete(breaks = c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    scale_fill_manual(values=(colors_low))+
      scale_color_manual(values=(colors_low))+
       scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="bottom",
       legend.title = element_blank(),
       # axis.text.x = element_blank(),
       axis.text.y = element_text(size=8),
       # axis.title = element_text(size=9),
       axis.title = element_blank(),
       
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )


  low_phased_ribbon


##--------LOST---------
lost_genotype_snps <- all_SNPs_phased %>% filter(grouping %in% lost_type) 
colors_lost <- as.character(colors_all_phases[which(colors_all_phases$cluster %in% lost_type),"color"])
# droplevels(colors_lost)
lost_type_order <- as.character(colors_all_phases[which(colors_all_phases$cluster %in% lost_type),"cluster"])    
lost_genotype_snps$grouping = factor(lost_genotype_snps$grouping, levels=lost_type_order)
droplevels(lost_genotype_snps$grouping)
 levels(lost_genotype_snps$grouping)


lost_phased_ribbon <- ggplot()+
    geom_ribbon(aes(x=sample,ymin = ymin, ymax = ymax,group=grouping,fill=grouping),data = lost_genotype_snps, alpha=.4)+
      # geom_ribbon(aes(x=sample,ymin = lower, ymax = upper,group=final_clust,fill=final_clust),data = non_phase_qunatiles, alpha=.4)+
    # geom_ribbon(aes(x=sample,ymin = X1st.Qu., ymax = X3rd.Qu.,group=final_clust,fill=final_clust),data = none_phase_qunatiles, alpha=.5)+
    # geom_ribbon(aes(x=sample,ymin = Min., ymax = Max.,group=final_clust,fill=final_clust),data = non_phase_qunatiles, alpha=.5)+
    # geom_line(data=explained_none_plot_nonModi_CLUSTERED,aes(x=sample,y=Snps,group=site,color=final_clust,fill=final_clust),size=0.1, alpha=.1)+ 
    # geom_line(data = mappingcoverage,aes(y = coverage, x = time_f_02,group=contig))+
      # facet_grid(contig~treatment )+
      # scale_x_continuous(breaks = c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
       # scale_x_discrete(breaks = c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    scale_fill_manual(values=(colors_lost))+
      scale_color_manual(values=(colors_lost))+
       scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="bottom",
       legend.title = element_blank(),
       # axis.text.x = element_blank(),
       axis.text.y = element_text(size=8),
       # axis.title = element_text(size=9),
       axis.title = element_blank(),
       
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )


  lost_phased_ribbon

  
  ###------------------------------
  ###merge all plots togther
  ###------------------------------
  all_genotype_snps <- rbind(high_genotype_snps,low_genotype_snps,lost_genotype_snps)
  all_genotype_snps$grouping = factor(all_genotype_snps$grouping, levels=c(high_type_order,low_type_order,lost_type_order))
colors_all <- c(colors_high,colors_low,colors_lost)
  

all_phased_ribbon <- ggplot()+
    geom_ribbon(aes(x=sample,ymin = ymin, ymax = ymax,group=grouping,fill=grouping),data = all_genotype_snps, alpha=.4)+
      # geom_ribbon(aes(x=sample,ymin = lower, ymax = upper,group=final_clust,fill=final_clust),data = non_phase_qunatiles, alpha=.4)+
    # geom_ribbon(aes(x=sample,ymin = X1st.Qu., ymax = X3rd.Qu.,group=final_clust,fill=final_clust),data = none_phase_qunatiles, alpha=.5)+
    # geom_ribbon(aes(x=sample,ymin = Min., ymax = Max.,group=final_clust,fill=final_clust),data = non_phase_qunatiles, alpha=.5)+
    # geom_line(data=explained_none_plot_nonModi_CLUSTERED,aes(x=sample,y=Snps,group=site,color=final_clust,fill=final_clust),size=0.1, alpha=.1)+ 
    # geom_line(data = mappingcoverage,aes(y = coverage, x = time_f_02,group=contig))+
      # facet_grid(contig~treatment )+
      # scale_x_continuous(breaks = c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
       # scale_x_discrete(breaks = c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    scale_fill_manual(values=(colors_all))+
      scale_color_manual(values=(colors_all))+
       scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=7),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="bottom",
       legend.title = element_blank(),
       # axis.text.x = element_blank(),
       axis.text.y = element_text(size=8),
       # axis.title = element_text(size=9),
       axis.title = element_blank(),
       
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )


  all_phased_ribbon
```





among the most mutate genes are:
[addA](https://www.uniprot.org/uniprot/P23478) a double strand break repair gene
[mvk]
[dnaG](https://www.uniprot.org/uniprot/P0ABS5) RNA polymerase that catalyzes the synthesis of short RNA molecules used as primers for DNA polymerase during DNA replication
[murA](https://www.uniprot.org/uniprot/P0A749) Cell wall formation (PubMed:1512209). Adds enolpyruvyl to UDP-N-acetylglucosamine (PubMed:1512209, PubMed:20392080). Target for the antibiotic fosfomycin.
[glmS](https://www.uniprot.org/uniprot/P17169) Catalyzes the first step in hexosamine metabolism, converting fructose-6P into glucosamine-6P using glutamine as a nitrogen source
[purH]


#####bring all together and plot with tsne cluster


```{r}
library(grid)
library(gridExtra)
library(patchwork)


source("/home/vincent/Desktop/ScriptRepository/R_functions-master/R_functions/g_legend.R") ##for plotting only legends
# legend_1 <- g_legend(p_mst1_sterm_woModi_clustered)
# legend_2 <- g_legend(p_mst2_sterm_woModi_clustered)
# legend_both <- g_legend(p_both_sterm_woModi_clustered)
# blank <- grid.rect(gp=gpar(col="white"))
# 
# grid.arrange(p_both_sterm_woModi_clustered+theme(legend.position = "none"),Sterm_both_suplementaryTsne,Sterm_both_suplementarybarPlot,Sterm_both_density,
#              p_mst1_sterm_woModi_clustered+theme(legend.position = "none"),Sterm_mst1_suplementaryTsne,Sterm_mst1_suplementarybarPlot,Sterm_mst1_density, 
#              p_mst2_sterm_woModi_clustered+theme(legend.position = "none"),Sterm_mst2_suplementaryTsne,Sterm_mst2_suplementarybarPlot,Sterm_mst2_density, 
#              p_none_sterm_woModi_clustered+theme(legend.position = "none"),Sterm_none_suplementaryTsne,Sterm_none_suplementarybarPlot,Sterm_none_density, 
#              legend_1,legend_2,legend_both,blank,
#              ncol=4,heights=c(1,1,1,1.3,1),widths=c(4,1,1,1))
# 
# 
# 
# 
# 
# 
#  svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_tsne_all.svg",width=12,height=8)
# 
# grid.arrange(p_both_sterm_woModi_clustered+theme(legend.position = "none"),Sterm_both_suplementaryTsne,Sterm_both_suplementarybarPlot,Sterm_both_density,
#              p_mst1_sterm_woModi_clustered+theme(legend.position = "none"),Sterm_mst1_suplementaryTsne,Sterm_mst1_suplementarybarPlot,Sterm_mst1_density,
#              p_mst2_sterm_woModi_clustered+theme(legend.position = "none"),Sterm_mst2_suplementaryTsne,Sterm_mst2_suplementarybarPlot,Sterm_mst2_density,
#              p_none_sterm_woModi_clustered+theme(legend.position = "none"),Sterm_none_suplementaryTsne,Sterm_none_suplementarybarPlot,Sterm_none_density,
#              legend_1,legend_2,legend_both,blank,
#              ncol=4,heights=c(1,1,1,1,1),widths=c(4,1,1,1))
# 
# 
# dev.off()



# 
#  svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_tsne_all_02.svg",width=12,height=8)
# 
# grid.arrange(p_both_sterm_woModi_clustered+theme(legend.position = "none")+annotate("text", label = "both strains", x = 5, y = 0.25, size = 7),Sterm_both_suplementarybarPlot,Sterm_both_density,Sterm_both_point,
#              p_mst1_sterm_woModi_clustered+theme(legend.position = "none")+annotate("text", label = "mst1", x = 5, y = 0.25, size = 7),Sterm_mst1_suplementarybarPlot,Sterm_mst1_density,Sterm_mst1_point,
#              p_mst2_sterm_woModi_clustered+theme(legend.position = "none")+annotate("text", label = "mst2", x = 5, y = 0.75, size = 7),Sterm_mst2_suplementarybarPlot,Sterm_mst2_density,Sterm_mst2_point,
#              p_none_sterm_woModi_clustered+theme(legend.position = "none")+annotate("text", label = "not explained", x = 5, y = 0.85, size = 7),Sterm_none_suplementarybarPlot,Sterm_none_density,Sterm_none_point,
#              
#          
#              ncol=4,heights=c(1,1,1,1.2),widths=c(4,1,1,1))
# 
# 
# dev.off()


###--------------------------------Patchwork

# 
#  svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_tsne_all_03.svg",width=12,height=8)
# 
#  (p_both_sterm_woModi_clustered+theme(legend.position = "none")+annotate("text", label = "both strains", x = 5, y = 0.25, size = 7)|Sterm_both_suplementarybarPlot|Sterm_both_density+theme(axis.title.x = element_blank())|Sterm_both_point+theme(axis.title.x = element_blank())|  plot_layout(widths = c(3, 1,1,1))) /
#  (p_mst1_sterm_woModi_clustered+theme(legend.position = "none")+annotate("text", label = "mst1", x = 5, y = 0.25, size = 7)|Sterm_mst1_suplementarybarPlot|Sterm_mst1_density+theme(axis.title.x = element_blank())|Sterm_mst1_point+theme(axis.title.x = element_blank())| plot_layout(widths = c(3, 1,1,1))) /
#  (p_mst2_sterm_woModi_clustered+theme(legend.position = "none")+annotate("text", label = "mst2", x = 5, y = 0.75, size = 7)|Sterm_mst2_suplementarybarPlot|Sterm_mst2_density+theme(axis.title.x = element_blank())|Sterm_mst2_point+theme(axis.title.x = element_blank())|  plot_layout(widths = c(3, 1,1,1))) /
#  (p_none_sterm_woModi_clustered+theme(legend.position = "none")+annotate("text", label = "not explained", x = 5, y = 0.85, size = 7)|Sterm_none_suplementarybarPlot|Sterm_none_density|Sterm_none_point|plot_layout(widths = c(3, 1,1,1)))
# 
#  
#  dev.off()
##-----===Ribbon
 
  p2 <- (p_both_sterm_woModi_NOT_clustered+theme(axis.text.x = element_blank())+annotate("text", label = "both strains", x = 5, y = 0.25, size = 6)|both_phased_ribbon+theme(axis.text.x = element_blank(),legend.position = "none")|Sterm_both_suplementarybarPlot|Sterm_both_point+theme(axis.title.x = element_blank())|  plot_layout(widths = c(2,1,2, 1,1))) /
 (p_mst1_sterm_woModi_NOT_clustered+theme(axis.text.x = element_blank())+annotate("text", label = "mst1", x = 5, y = 0.25, size = 6)|mst1_phased_ribbon+theme(axis.text.x = element_blank(),legend.position = "none")|Sterm_mst1_suplementarybarPlot|Sterm_mst1_point+theme(axis.title.x = element_blank())| plot_layout(widths = c(2,1,2, 1,1))) /
 (p_mst2_sterm_woModi_NOT_clustered+theme(axis.text.x = element_blank())+annotate("text", label = "mst2", x = 5, y = 0.75, size = 6)|mst2_phased_ribbon+theme(axis.text.x = element_blank(),legend.position = "none")|Sterm_mst2_suplementarybarPlot|Sterm_mst2_point+theme(axis.title.x = element_blank())|  plot_layout(widths = c(2,1,2, 1,1))) /
 (p_none_sterm_woModi_NOT_clustered+annotate("text", label = "not explained", x = 5, y = 0.85, size = 6)|none_phased_ribbon+theme(legend.position = "none")|Sterm_none_suplementarybarPlot|Sterm_none_point|plot_layout(widths = c(2,1,2, 1,1)))

#  
#  svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_tsne_all_03.svg",width=15,height=8)
# 
p2
#  
#  dev.off()
 
#===========================
 ##-----===with original Streptococcus thermophilus profile
#===========================
 
##----------------
##prep
##----------------
all_snps_sterm_long <-  rbind(explained_mst1_plot_nonModi_CLUSTERED,explained_mst2_plot_nonModi_CLUSTERED,explained_both_plot_nonModi_CLUSTERED,explained_none_plot_nonModi_CLUSTERED)
 
total <- sum(table(all_snps_sterm_long$group))
table(all_snps_sterm_long$group)/total
##----------------
##plot
##----------------
PLOT_allSNPs <- ggplot(all_snps_sterm_long,aes(x=sample,y=Snps,group=site,color=species,fill=species, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nRepeat_identity:",Repeat_cluster,"\ncore:",core,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.2, alpha=.1)+ 
    # facet_grid(day~kessel~species)+
    # facet_grid(species~.)+
    # facet_grid(treatment~species)+
    # facet_grid(treatment~species)+
    labs("",
         x="time [h]",
         y="Alternative allele frequency")+
    #scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    #scale_y_continuous(limits = c(0,0.3))+
    theme_classic()+
   scale_fill_manual(values=all_colours[2])+
   scale_color_manual(values=all_colours[2])+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
      #axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          )
  
p1 <- plot_spacer() +  PLOT_allSNPs +  plot_spacer() +   plot_layout(nrow=3,heights = c(1,2, 1))
p1

#===========================
 ##-----==gganimate
#===========================

all_snps_sterm_long$sample
all_snps_sterm_long$sample2 <- revalue(all_snps_sterm_long$sample, c("Lyo\n1996"="starter culture 1","Lyo\n2012"="starter culture 2", "Lyo\n2014"="starter culture 3", "working\nstock"="starter culture 4", "starter\nculture\n2012"="starter culture 5", "starter\nculture\n2018"="starter culture 6","cheesemaking\nday1"="starter culture 7", "cheesemaking\nday2"="starter culture 8"))


PLOT_allSNPs2 <- ggplot(all_snps_sterm_long,aes(x=sample2,y=Snps,group=site,color=species,fill=species, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nRepeat_identity:",Repeat_cluster,"\ncore:",core,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.2, alpha=.1)+ 
    # facet_grid(day~kessel~species)+
    # facet_grid(species~.)+
    # facet_grid(treatment~species)+
    # facet_grid(treatment~species)+
    labs("",
         x="time [h]",
         y="Alternative allele frequency")+
    #scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    #scale_y_continuous(limits = c(0,0.3))+
    theme_classic()+
   scale_fill_manual(values=all_colours[2])+
   scale_color_manual(values=all_colours[2])+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
      #axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          )
PLOT_allSNPs2

 png("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_tsne_all_06.png",width=1800,height=1400,res=300)
 # svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_tsne_all_06.svg",width=4,height=2)

PLOT_allSNPs2
 
 dev.off()
#===========================
 ##-----==combine all
#===========================


 svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_tsne_all_04.svg",width=18,height=8)

(p1|p2)+plot_layout(widths =c(2,6))
 
 dev.off()
 
 #===========================
 ##-----==with replicons and nucleotide diversity
#===========================
# plot_nucleotide_div <- pall
p3 <-   plot_spacer()+(high_phased_ribbon+theme(axis.text.x = element_blank(),legend.position = "right")) + (low_phased_ribbon+theme(axis.text.x = element_blank(),legend.position = "right"))+ (lost_phased_ribbon+theme(axis.text.x = element_blank(),legend.position = "right")) +plot_spacer()+  plot_layout(nrow=5)
p3

# 
# plot_nucleotide_div <- pall
# p3 <-   (high_phased_ribbon+theme(axis.text.x = element_blank(),legend.position = "right")) + (low_phased_ribbon+theme(axis.text.x = element_blank(),legend.position = "right"))+ (lost_phased_ribbon+theme(axis.text.x = element_blank(),legend.position = "right")) +(plot_nucleotide_div+theme(legend.position = "none"))+  plot_layout(nrow=4)
# p3
# 
#  svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_tsne_all_05.svg",width=20,height=8)
# 
# (p1|p2|p3)+plot_layout(widths =c(2,6,2))
#  
#  dev.off()
 
 ##--------------------
 ##reduced figure for paper
 ##--------------------
 
  p2 <- (p_both_sterm_woModi_NOT_clustered+theme(axis.text.x = element_blank())+annotate("text", label = "both strains", x = 5, y = 0.2, size = 6)|both_phased_ribbon+theme(axis.text.x = element_blank(),legend.position = "none")|  plot_layout(widths = c(2,2))) /
 (p_mst1_sterm_woModi_NOT_clustered+theme(axis.text.x = element_blank())+annotate("text", label = "mst1", x = 5, y = 0.25, size = 6)|mst1_phased_ribbon+theme(axis.text.x = element_blank(),legend.position = "none")|plot_layout(widths = c(2,2))) /
 (p_mst2_sterm_woModi_NOT_clustered+theme(axis.text.x = element_blank())+annotate("text", label = "mst2", x = 5, y = 0.9, size = 6)|mst2_phased_ribbon+theme(axis.text.x = element_blank(),legend.position = "none")|plot_layout(widths = c(2,2))) /
 (p_none_sterm_woModi_NOT_clustered+annotate("text", label = "not explained", x = 5, y = 0.9, size = 6)|none_phased_ribbon+theme(legend.position = "none")|plot_layout(widths = c(2,2)))

 p2
 
  svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_tsne_all_06.svg",width=15,height=8)

(p1|p2|p3)+plot_layout(widths =c(2,4,2))

 dev.off()

##-------------------------------------
###SNPs explained in one figure
 ##-------------------------------------
 explained_all_plot_nonModi_CLUSTERED <- rbind(explained_both_plot_nonModi_CLUSTERED,explained_mst2_plot_nonModi_CLUSTERED,explained_mst1_plot_nonModi_CLUSTERED,explained_none_plot_nonModi_CLUSTERED)


 
 
# explained_all_plot_nonModi_CLUSTERED <- rbind(explained_S72_plot_nonModi_CLUSTERED,
# explained_S50_plot_nonModi_CLUSTERED,
# explained_both_plot_nonModi_CLUSTERED,
# explained_not_plot_nonModi_CLUSTERED)
nrow(explained_all_plot_nonModi_CLUSTERED)
##----------plot line
colors <- c("#C39953","#580000","#8DDBCF","#909090")
# none_3 #909090
# both_2 #C39953
# mst1_1 #580000
# mst2_2 #8DDBCF

(table(explained_all_plot_nonModi_CLUSTERED$group)/5)/(nrow(explained_all_plot_nonModi_CLUSTERED)/5)*100

unique(explained_all_plot_nonModi_CLUSTERED$group)
unique(explained_all_plot_nonModi_CLUSTERED$final_clust)
explained_all_plot_nonModi_CLUSTERED[which(is.na(explained_all_plot_nonModi_CLUSTERED$group)),"group"] <- "not_explained"
explained_all_plot_nonModi_CLUSTERED$group = factor(explained_all_plot_nonModi_CLUSTERED$group, levels=c("both","mst1","not_explained","mst2"))
explained_all_plot_nonModi_CLUSTERED$final_clust = factor(explained_all_plot_nonModi_CLUSTERED$final_clust, levels=rev(c("1","2","3","4","5","not clustered")))

explained_all_plot_nonModi_CLUSTERED$sample2 <- revalue(explained_all_plot_nonModi_CLUSTERED$sample, c("Lyo\n1996"="starter culture 1","Lyo\n2012"="starter culture 2", "Lyo\n2014"="starter culture 3", "working\nstock"="starter culture 4", "starter\nculture\n2012"="starter culture 5", "starter\nculture\n2018"="starter culture 6","cheesemaking\nday1"="starter culture 7", "cheesemaking\nday2"="starter culture 8"))



  p_S72_sterm_woModi_clustered <- ggplot(explained_all_plot_nonModi_CLUSTERED,aes(x=sample2,y=Snps,group=site,color=group,fill=group))+ geom_line(size=0.2, alpha=.1)+ 
    # facet_grid(day~kessel~species)+
    # facet_grid(species~.)+
    # facet_grid(treatment~species)+
    # facet_grid(treatment~species)+
    labs("",
         x="time [h]",
         y="Alternative allele frequency")+
    #scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    #scale_y_continuous(limits = c(0,0.3))+
    theme_classic()+
   scale_fill_manual(values=colors)+
   scale_color_manual(values=colors)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
      #axis.text.x = element_blank(),
          axis.title.x = element_blank(),
          legend.position="right"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          )

p_S72_sterm_woModi_clustered

 # png("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_tsne_all_07.png",width=1800,height=1400,res=300)
 svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_tsne_all_07.svg",width=4,height=2)

p_S72_sterm_woModi_clustered+theme(  legend.position="left")
 
 dev.off()

```

###Venn Diagramm

```{r}

all_snps_sterm_long <-  rbind(explained_mst1_plot_nonModi_CLUSTERED,explained_mst2_plot_nonModi_CLUSTERED,explained_both_plot_nonModi_CLUSTERED,explained_none_plot_nonModi_CLUSTERED)
 
total <- sum(table(all_snps_sterm_long$group))
table(all_snps_sterm_long$group)/total

```



```{r}

##--------colors

library("RColorBrewer")
display.brewer.all()
display.brewer.pal(n = 5, name = 'RdYlBu')
# colors_substrains <- c(brewer.pal(n = 5, name = 'RdYlBu'),"grey60")
colors_substrains <- c(colorsTSNE[1:5],"grey")
library(wesanderson)
# colors_substrains <- c(wes_palette(n=5, name="FantasticFox1"),"grey60")
##----------merge
explained_all_plot_nonModi_CLUSTERED <- rbind(explained_S72_plot_nonModi_CLUSTERED,
explained_S50_plot_nonModi_CLUSTERED,
explained_both_plot_nonModi_CLUSTERED,
explained_not_plot_nonModi_CLUSTERED)
nrow(explained_all_plot_nonModi_CLUSTERED)
##----------plot line

(table(explained_all_plot_nonModi_CLUSTERED$group)/5)/(nrow(explained_all_plot_nonModi_CLUSTERED)/5)*100

unique(explained_all_plot_nonModi_CLUSTERED$group)
unique(explained_all_plot_nonModi_CLUSTERED$final_clust)

explained_all_plot_nonModi_CLUSTERED$group = factor(explained_all_plot_nonModi_CLUSTERED$group, levels=c("both","mst1","mst2","not_explained"))
explained_all_plot_nonModi_CLUSTERED$final_clust = factor(explained_all_plot_nonModi_CLUSTERED$final_clust, levels=rev(c("1","2","3","4","5","not clustered")))

  p_S72_sterm_woModi_clustered <- ggplot(explained_all_plot_nonModi_CLUSTERED,aes(x=sample,y=allele_frequency,group=site,color=final_clust,fill=final_clust, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nEC:",EC,"\nKEGG_ko:",KEGG_ko,"\nCAZy:",CAZy,"\nCOG:",COG_Functional_Category,"\nCluster:",final_clust)))+ geom_line(size=0.1, alpha=.5)+ 
       facet_grid(group~.)+
   labs("",
         x="",
         # y="",
          y="alternative allele frequency")+
        scale_fill_manual(values=rev(colors_substrains))+
        scale_color_manual(values=rev(colors_substrains))+
       scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.25,0.5,0.75,1),labels=c(0,0.25,0.5,0.75,1), expand = c(0, 0)) +
  theme_classic()+
          ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=6),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
           legend.position="top",
       legend.title = element_blank(),
       axis.text.y = element_text(size=6),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm"),legend.direction = "horizontal"
       )

p_S72_sterm_woModi_clustered
svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_CLUSTERED_02.svg",width=3,height=4.2)
# png("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_sterm.png",width=1800,height=1600,res=300)


p_S72_sterm_woModi_clustered

dev.off()

ggp <- ggplotly(p_S72_sterm_woModi_clustered)
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_all_clustered_woModi.html")




##----------plot box

# explained_s72_plot_nonModi_CLUSTERED$final_clust = factor(explained_s72_plot_nonModi_CLUSTERED$final_clust, levels=c("1","2","3","NA"))

  p_S72_sterm_woModi_clustered_box <- ggplot(explained_all_plot_nonModi_CLUSTERED,aes(x=final_clust,y=allele_frequency,color=final_clust,fill=final_clust))+ geom_boxplot(alpha=.5)+ 
       facet_grid(group~.)+
   labs("",
         x="",
         # y="",
          y="allele frequency")+
       # scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,1))+
 theme(#axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            # plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )

p_S72_sterm_woModi_clustered_box


ggp <- ggplotly(p_S72_sterm_woModi_clustered_box)
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_all_clustered_woModi.html")


###-----------------
##Tsne plots
###-----------------

p1 <- Sterm_both_suplementaryTsne+labs(title = "explained by both strains")
p2 <- Sterm_S50_suplementaryTsne+labs(title = "explained by mst1")
p3 <- Sterm_S72_suplementaryTsne+labs(title = "explained by mst2")
p4 <- Sterm_notExplained_suplementaryTsne+labs(title = "not-explained")

grid.arrange(p1,p2,p3,p4, ncol=1)

library(grid)
library(gridExtra)
 svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_tsne_all.svg",width=2.5,height=10)

   grid.arrange(p1,p2,p3,p4, ncol=1)

dev.off()
```


```{r}
##plot

  library("tidyverse")
library("officer")
library("rvg")

# read_pptx() %>%
#     add_slide(layout = "Title and Content", master = "Office Theme") %>%
#     ph_with_vg(code = print(grid.arrange(p1,p2, p3,p0,ncol=1, nrow=4,heights=c(1.15,1,1,1.55))), type = "body",height = 7, width = 8) %>% 
#     print(target = "~/Desktop/presenations/my_presentations/20190919_VUA/slides_03.pptx")

  
     # grid.arrange(p0,p1, p2,p3,ncol=1, nrow=4,heights=c(1,1,1,2))
  # pdf("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored.pdf",width=9.8,height=20,paper='special')
# svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored.svg",width=5,height=4)
# png("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_sterm.png",width=1800,height=1600,res=300)
#  svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_sterm.svg",width=4,height=7)
# 
# grid.arrange(p1,p2, p3,p0,ncol=1, nrow=4,heights=c(1.15,1,1,1.55))
# 
# dev.off()

  
  ##============
  #look at special things
  ##============
  ##-------
  ##decreasing snp
  ##-------
library(plotly)
ggp <- ggplotly(p1)
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_sterm_both_decreasing.html")
sample_relative_temp_cleaned[grep("RMK202_1049174_ACAA",sample_relative_temp_cleaned$site),]
sample_relative_temp_cleaned[grep("RMK202_1049189_",sample_relative_temp_cleaned$site),]

  ##-------
  ##increasing snp
  ##-------
library(plotly)
ggp <- ggplotly(p2)
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_sterm_both_increase.html")
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_sterm_both_increase.html")


sample_relative_temp_cleaned[grep("RMK202_1624846",sample_relative_temp_cleaned$site),]
sample_relative_temp_cleaned[grep("RMK202_1049189_",sample_relative_temp_cleaned$site),]

  ##-------
  ##S72
  ##-------
library(plotly)
ggp <- ggplotly(p3)
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_sterm_S72.html")

  ##-------
  ##not explained
  ##-------
library(plotly)
ggp <- ggplotly(p0)
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_sterm_S72.html")
sample_relative_temp_cleaned[grep("RMK202_1353308_",sample_relative_temp_cleaned$site),]


##------------------------
##ldel
##------------------------


##----------START

# both_metaIso_ldel <- meta_ldel[onlysecond_sterm %in% onlyfirst_sterm,]
# nrow(both_metaIso_ldel)
# only_meta_ldel <- meta_ldel[!onlysecond_sterm %in% onlyfirst_sterm,]
# nrow(only_meta_ldel)
# # sum(duplicated(only_meta_sterm))
# only_isolates_ldel <- sample_relative_safe_final_iso_cleaned_ldel[!onlyfirst_sterm %in% onlysecond_sterm,]
# nrow(only_isolates_ldel)

###-----not explained by isolates
  not_explained_ldel_plot <- gather(only_meta_ldel, sample, allele_frequency, "Lyo_202_2012":"Versand_202", factor_key=TRUE)
  # not_explained_sterm_plot$site <- paste(not_explained_sterm_plot$X.CHROM,not_explained_sterm_plot$POS,not_explained_sterm_plot$REF,not_explained_sterm_plot$ALT,sep="_")

  not_explained_ldel_plot$group <- "not_explained"
  
  boxplot_not_explained_ldel <- data.frame(name=(only_meta_ldel[,1]),rowSums=rowSums(only_meta_ldel[,-1]),name="not explained\nldel")

###-----explained by isolates

   explained_isolates_ldel_plot <- gather(both_metaIso_ldel, sample, allele_frequency, "Lyo_202_2012":"Versand_202", factor_key=TRUE)
  # not_explained_sterm_plot$site <- paste(not_explained_sterm_plot$X.CHROM,not_explained_sterm_plot$POS,not_explained_sterm_plot$REF,not_explained_sterm_plot$ALT,sep="_")

  explained_isolates_ldel_plot$group <- "explained_by_isolates"

  boxplot_both_metaIso_ldel <- data.frame(name=(both_metaIso_ldel[,1]),rowSums=rowSums(both_metaIso_ldel[,-1]),name="explained\nldel")

##---------------plot

  p1 <- ggplot(explained_isolates_ldel_plot,aes(x=sample,y=allele_frequency,group=site,color=group,fill=group))+ geom_line(size=0.2, alpha=.2)+ 
      # facet_grid(species~.)+
   labs("",
         x="",
         # y="",
          y="explained by\nisolates")+
       scale_fill_manual(values="black")+
      scale_color_manual(values="black")+
           scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0),limits = c(0,0.2)) +
      scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
    # ylim(0,0.2)+
  theme_classic()+
 theme(#axis.text.x = element_text(angle = 75, hjust = 1,size=9),
       axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
         plot.margin=unit(c(t = 0.5, r = 0.5, b = 0, l = 0.1),"cm")
       )
  
  # p1
  
  
  p0 <- ggplot(not_explained_ldel_plot,aes(x=sample,y=allele_frequency,group=site,color=group,fill=group))+ geom_line(size=0.2, alpha=.2)+ 
      # facet_grid(species~.)+
   labs("",
         x="",
         # y="",
          y="not\nexplained")+
       scale_fill_manual(values="gray41")+
      scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0),limits = c(0,0.2)) +
     scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
    # ylim(0,0.2)+
  theme_classic()+
 theme(#axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
  
  # p0

  
  # pfinal_ldel <-  grid.arrange(p1, p0, ncol=1, nrow=2,heights=c(1,1.5))
   
#    
# library(plotly)
# ggp <- ggplotly(p0)
# 
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_ldel.html")
#   

  # pdf("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored.pdf",width=9.8,height=20,paper='special')
# svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored.svg",width=5,height=4)
# png("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_ldel.png",width=1800,height=1600,res=300)
# 
#    grid.arrange(p1, p0, ncol=1, nrow=2,heights=c(1,1.5))
# 
# dev.off()

##---------------Nucleotide diversity

nucleotid_div_all <- rbind(nucleotid_div_sterm,nucleotid_div)

all_colours <- rev(c("#EB4D4D","#FAA0A0","#10B552","#36E37B","#6CF5A3","#C5F6FA","#99E9F2") ) 

nucleotid_div_all$samples = factor(nucleotid_div_all$samples, levels=c("Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202"))
library(plyr)
nucleotid_div_all$samples_renamed <- revalue(nucleotid_div_all$samples, c("Lyo_202_2012"="Lyoampulle\n2012", "Lyo_202_2014"="Lyoampulle\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018"))
nucleotid_div_all$samples_renamed = factor(nucleotid_div_all$samples_renamed, levels=c("Lyoampulle\n2012","Lyoampulle\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018"))



 pnucdiv <- ggplot(nucleotid_div_all,aes(x=samples_renamed,y=div,group=group,fill=group,color=group))+ geom_point(size=3)+ geom_line()+ #stat_summary(fun.y=sum, geom="line")+
   labs("",
         x="",
         y="nucleotide\ndiversity [%]")+
   scale_color_manual(values=c("#10B552","#EB4D4D"))+
          scale_x_discrete( expand = c(0, 0)) +
   scale_y_continuous(breaks =c(0,0.4,0.8),labels=c(0,0.4,0.8),expand = c(0, 0),limits = c(-0.1,0.9)) +
  theme_classic()+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,vjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
        plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       # legend.justification=c(1,1), legend.position=c(0.7,0.7),
       # legend.title = element_blank()
       )
  pnucdiv
  
#   
#   png("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_ldel.png",width=1800,height=1600,res=300)
# 
#    grid.arrange(p1, p0,pnucdiv, ncol=1, nrow=3,heights=c(1,1,1.5))
# 
# dev.off()

##---------boxplot rowsums of alternative allele frequencies

boxplot_data_rowsums <- rbind(boxplot_explained_sterm_both,boxplot_explained_s50,boxplot_explained_s72,boxplot_not_explained_sterm,boxplot_both_metaIso_ldel, boxplot_not_explained_ldel)



PSummedFrequency <- ggplot(boxplot_data_rowsums,aes(y=rowSums,x=name.1,color=name.1))+geom_boxplot()+
   labs(x="",
         y="summed\nfrequency")+
   #scale_color_manual(values=c("#10B552","#EB4D4D"))+
          #scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.4,0.8),labels=c(0,0.4,0.8),expand = c(0, 0),limits = c(-0.1,0.9)) +
  theme_classic()+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
        plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.38),"cm")
       # legend.justification=c(1,1), legend.position=c(0.7,0.7),
       # legend.title = element_blank()
       )

 PSummedFrequency
  # png("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_ldel.png",width=1800,height=1600,res=300)
 svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_ldel.svg",width=4,height=7)

   grid.arrange(p1, p0,pnucdiv,PSummedFrequency, ncol=1, nrow=4,heights=c(1.3,1,1.8,2))

dev.off()

###t-test


 rbind(boxplot_explained_sterm_both,boxplot_explained_s50,boxplot_explained_s72,boxplot_not_explained_sterm,boxplot_both_metaIso_ldel, boxplot_not_explained_ldel)
#=============
##-ldel
#=============
 # summary(boxplot_both_metaIso_ldel)
 #  summary(boxplot_not_explained_ldel)

ldel_Ttest <- t.test(boxplot_both_metaIso_ldel$rowSums,boxplot_not_explained_ldel$rowSums)
ldel_Ttest$p.value
#=============
#====-sterm
#=============
 # summary(boxplot_both_metaIso_ldel)
 #  summary(boxplot_not_explained_ldel)

lsterm_Ttest <- t.test(boxplot_explained_sterm_both$rowSums,boxplot_explained_s72$rowSums)
lsterm_Ttest$p.value

lsterm_Ttest <- t.test(boxplot_explained_sterm_both$rowSums,boxplot_explained_s72$rowSums)
lsterm_Ttest$p.value

```

######Go-Analysis

```{r}

table(explained_all_plot_nonModi_CLUSTERED$significance)
table(str_split_fixed(explained_all_plot_nonModi_CLUSTERED$PGAP_name, "-", 2)[,2])
explained_all_plot_nonModi_CLUSTERED$geneNameprep <- str_split_fixed(explained_all_plot_nonModi_CLUSTERED$PGAP_name, "-", 2)[,2]

##================
###prepare for topGO analysis
##================

library(readr)
library(Rgraphviz)
library(topGO)

##------allgenes
geneID2GO_ldel <- readMappings(file = "~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog_GO/Lactobacillus_delbrueckii_RMK202_pgap_query_seqs.fa.emapper.annotations") 
geneID2GO_sterm <- readMappings(file = "~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog_GO/Streptococcus_thermophilus_RMK202_pgap_query_seqs.fa.emapper.annotations") 
length(geneID2GO_sterm)

##============
##topGO 
##============
##------without modifier snp genes


geneUniverse <- names(geneID2GO_sterm) 
genesOfInterest <- as.character(explained_all_plot_nonModi_CLUSTERED$geneNameprep)
length(genesOfInterest)


table(genesOfInterest)

genesOfInterest <- genesOfInterest[genesOfInterest %in% geneUniverse]

 geneList <- factor(as.integer(geneUniverse %in% genesOfInterest))
 names(geneList) <- geneUniverse
 
 
 myGOdata <- new("topGOdata", description="My project", ontology="BP", allGenes=geneList,  annot = annFUN.gene2GO, gene2GO = geneID2GO_sterm)
 
 sg <- sigGenes(myGOdata)
 str(sg)
numSigGenes(myGOdata)
resultFisher <- runTest(myGOdata, algorithm="classic", statistic="fisher")
resultFisher

allRes <- GenTable(myGOdata, classicFisher = resultFisher, orderBy = "resultFisher", ranksOf = "classicFisher", topNodes = 200)
allRes

 showSigOfNodes(myGOdata, score(resultFisher), firstSigNodes = 5, useInfo ='all')
 printGraph(myGOdata, resultFisher, firstSigNodes = 5, fn.prefix = "tGO", useInfo = "all", pdfSW = TRUE)
 
 
 length(usedGO(myGOdata))
 
  write.csv(allRes,"~/Desktop/presenations/my_presentations/20191007_groupmeeting/figures/Goterms.csv")
  
  
##------high significance mutations
 
annotation_sterm_snps_HIGH <- explained_all_plot_nonModi_CLUSTERED[explained_all_plot_nonModi_CLUSTERED$significance=="HIGH",]
table(annotation_sterm_snps_HIGH$significance)
# - as.character(explained_all_plot_nonModi_CLUSTERED$)
genesOfInterest <- as.character(annotation_sterm_snps_HIGH$geneNameprep)
genesOfInterest <- genesOfInterest[genesOfInterest %in% geneUniverse]

geneList <- factor(as.integer(geneUniverse %in% genesOfInterest))
names(geneList) <- geneUniverse
 
myGOdata <- new("topGOdata", description="My project", ontology="BP", allGenes=geneList,  annot = annFUN.gene2GO, gene2GO = geneID2GO_sterm)
 
sg <- sigGenes(myGOdata)
numSigGenes(myGOdata)

resultFisher <- runTest(myGOdata, algorithm="classic", statistic="fisher")
resultFisher

allRes <- GenTable(myGOdata, classicFisher = resultFisher, orderBy = "resultFisher", ranksOf = "classicFisher", topNodes = 200)
allRes

  write.csv(allRes,"~/Desktop/presenations/my_presentations/20191007_groupmeeting/figures/Goterms_loseOFfunction.csv")


##------moderate significance mutations
 
annotation_sterm_snps_HIGH <- explained_all_plot_nonModi_CLUSTERED[explained_all_plot_nonModi_CLUSTERED$significance=="MODERATE",]
table(annotation_sterm_snps_HIGH$significance)
# - as.character(explained_all_plot_nonModi_CLUSTERED$)
genesOfInterest <- as.character(annotation_sterm_snps_HIGH$geneNameprep)
genesOfInterest <- genesOfInterest[genesOfInterest %in% geneUniverse]

geneList <- factor(as.integer(geneUniverse %in% genesOfInterest))
names(geneList) <- geneUniverse
 
myGOdata <- new("topGOdata", description="My project", ontology="BP", allGenes=geneList,  annot = annFUN.gene2GO, gene2GO = geneID2GO_sterm)
 
sg <- sigGenes(myGOdata)
numSigGenes(myGOdata)

resultFisher <- runTest(myGOdata, algorithm="classic", statistic="fisher")
resultFisher

allRes <- GenTable(myGOdata, classicFisher = resultFisher, orderBy = "resultFisher", ranksOf = "classicFisher", topNodes = 200)
allRes

  write.csv(allRes,"~/Desktop/presenations/my_presentations/20191007_groupmeeting/figures/Goterms_MODERATE.csv")

```



#####look at interesting genes

```{bash}

##sterm decreasing

grep "1049189" /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata/SnpEff/Streptococcus_thermophilus_RMK202_pgap/variantCallsfreebayes_all_Streptococcus_thermophilus_RMK202_snpEff.vcf 

grep "1049174" /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata/SnpEff/Streptococcus_thermophilus_RMK202_pgap/variantCallsfreebayes_all_Streptococcus_thermophilus_RMK202_snpEff.vcf

grep "pgaptmp_001104" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/*_pgap.gff


##sterm not explained strain all same location 

grep "347968" /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata/SnpEff/Streptococcus_thermophilus_RMK202_pgap/variantCallsfreebayes_all_Streptococcus_thermophilus_RMK202_snpEff.vcf 

grep "pgaptmp_000374" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/*_pgap.gff

```


#####tsne test

```{r}

library(ggplot2)
library(dplyr)
library(magrittr)
library(ggrepel)

set.seed(123456)
N = 5000
D = 100
data.norm = matrix(rnorm(N * D, 2), N)
groups.probs = runif(10)
groups = sample(1:10, N, TRUE, groups.probs/sum(groups.probs))
for (gp in unique(groups)) {
    dev = rep(1, D)
    dev[sample.int(D, 3)] = runif(3, -10, 10)
    data.norm[which(groups == gp), ] = data.norm[which(groups == 
        gp), ] %*% diag(dev)
}
info.norm = tibble(truth = factor(groups))


data.norm <- as.matrix(sample_relative_safe_final_tsne_safe_sterm_woDuplicates[,c(-1,-2)])

pca.norm = prcomp(data.norm)
info.norm %<>% cbind(pca.norm$x[, 1:4])
ggplot(info.norm, aes(x = PC1, y = PC2, colour = truth)) + 
    geom_point(alpha = 0.3) + theme_bw()


library(Rtsne)
tsne.norm = Rtsne(pca.norm$x, pca = FALSE)
tsne_plot <- as.data.frame(tsne.norm$Y)
tsne_plot$site <- sample_relative_safe_final_tsne_safe_sterm_woDuplicates[,1]
tsne_plot$contig <- paste(str_split_fixed(tsne_plot$site, "_", 5)[,1],str_split_fixed(tsne_plot$site, "_", 5)[,2],sep = "_")


tsne_plot$species <- tsne_plot$contig

##plot blank
ggplot(tsne_plot,aes(x=V1,y=V2))+geom_point()



# info.norm %<>% mutate(tsne1 = tsne.norm$Y[, 1], tsne2 = tsne.norm$Y[,2])
# ggplot(info.norm, aes(x = tsne.norm$Y[, 1], y = tsne.norm$Y[,2])) + 
#     geom_point(alpha = 0.3) + theme_bw()

##ward
hc.norm = hclust(dist(tsne_plot), method = "ward.D")
tsne_plot$hclust =factor(cutree(hc.norm, 7))
hc.norm.cent = tsne_plot %>% group_by(hclust) %>% select(V1, 
    V2) %>% summarize_all(mean)
plotTsne <- ggplot(tsne_plot, aes(x = V1, y = V2, colour = hclust)) + 
  facet_grid(species~.)+
    geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label = hclust), 
    data = hc.norm.cent) + guides(colour = FALSE) + 
    ggtitle("Linkage criterion: Ward")
plotTsne

##kmeans
hc.norm = hclust(dist(tsne_plot))
tsne_plot$hclust =factor(cutree(hc.norm, 6))
hc.norm.cent = tsne_plot %>% group_by(hclust) %>% select(V1, 
    V2) %>% summarize_all(mean)
plotTsne <- ggplot(tsne_plot, aes(x = V1, y = V2, colour = hclust)) + 
  facet_grid(species~.)+
    geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label = hclust), 
    data = hc.norm.cent) + guides(colour = FALSE) + 
    ggtitle("Linkage criterion: kmeans")
plotTsne


km.norm = kmeans(tsne_plot, 9, nstart = 100)
tsne_plot$kmeans = factor(km.norm$cluster)

km.cent = info.norm %>% group_by(kmeans) %>% select(tsne1, 
    tsne2) %>% summarize_all(mean)
ggplot(info.norm, aes(x = tsne1, y = tsne2, colour = kmeans)) + 
    geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label = kmeans), 
    data = km.cent) + guides(colour = FALSE) + ggtitle("9 clusters")


```


#####t-sne

Here, I try to cluster the previous snp profiles with t-sne. explained [here](https://www.analyticsvidhya.com/blog/2017/01/t-sne-implementation-r-python/)


```{r}
library(Rtsne)
library(ggrepel)

##---------prep not explained
not_explained_sterm_plot_prep <- not_explained_sterm_plot[,c(-2,-4)]
# not_explained_sterm_plot_prep$sample_renamed <- factor(not_explained_sterm_plot_prep$sample_renamed)
notExplained_sterm_wide <- spread(not_explained_sterm_plot_prep, sample_renamed, allele_frequency)
nrow(notExplained_sterm_wide)



##-------------------sterm

sample_relative_safe_final_tsne_safe_sterm <- notExplained_sterm_wide
# sample_relative_safe_final_tsne_safe_sterm <- sample_relative_safe_final_meta_cleaned_02[grep("S_thermophilus_RMK202",sample_relative_safe_final_meta_cleaned_02$site),]
#    nrow(sample_relative_safe_final_tsne_safe_sterm)

   # sample_relative_safe_final_tsne_safe_sterm[which(rowSums(is.na(sample_relative_safe_final_tsne_safe_sterm[,c("Lyo_202_2012","Lyo_202_2014","RMK202","Konserve_202","Versand_202")]))!=0),]

##duplicate rows
nrow(sample_relative_safe_final_tsne_safe_sterm)
sample_relative_safe_final_tsne_safe_sterm_woDuplicates <- sample_relative_safe_final_tsne_safe_sterm[!duplicated(sample_relative_safe_final_tsne_safe_sterm[,-1]), ]
nrow(sample_relative_safe_final_tsne_safe_sterm_woDuplicates)   
sample_relative_safe_final_tsne_safe_sterm_woDuplicates <- sample_relative_safe_final_tsne_safe_sterm_woDuplicates[!rowSums(is.na(sample_relative_safe_final_tsne_safe_sterm_woDuplicates))>0,]

library(philentropy)
dist_sterm <- distance(as.matrix(sample_relative_safe_final_tsne_safe_sterm_woDuplicates[,c(-1,-2)]), method = "jaccard")

tsne <- Rtsne(dist_sterm, dims = 2, perplexity=30, verbose=TRUE, max_iter = 500)
 

# tsne <- Rtsne(sample_relative_safe_final_tsne_safe_sterm_woDuplicates[,c("Lyoampulle\n2012","Lyoampulle\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018")], dims = 2, perplexity=30, verbose=TRUE, max_iter = 500)
 # tsne <- Rtsne(sample_relative_safe_final_tsne_safe_sterm_woDuplicates[,c("Lyo_202_2012","Lyo_202_2014","RMK202","Konserve_202","Versand_202")], dims = 2, perplexity=30, verbose=TRUE, max_iter = 500)
tsne_plot <- as.data.frame(tsne$Y)
tsne_plot$site <- sample_relative_safe_final_tsne_safe_sterm_woDuplicates[,1]
tsne_plot$contig <- paste(str_split_fixed(tsne_plot$site, "_", 5)[,1],str_split_fixed(tsne_plot$site, "_", 5)[,2],sep = "_")




unique(tsne_plot$contig)
tsne_plot$species <- tsne_plot$contig
# tsne_plot$species <- ifelse(tsne_plot$contig=="contig_1","S.thermophilus",
#                                              ifelse(tsne_plot$contig=="contig_2","L.delbrueckii",
#                                                     ifelse(tsne_plot$contig=="contig_3","Lactococcus phage",
#                                                            ifelse(tsne_plot$contig=="contig_4","Cow",
#                                                                   ifelse(tsne_plot$contig=="contig_5","L.delbrueckii plasmid",
#                                                                         ifelse(tsne_plot$contig=="plasmidSpades_CowMito","Cow mitogenome",NA))))))

unique(tsne_plot$species)

##plot blank
ggplot(tsne_plot,aes(x=V1,y=V2))+geom_point()

##clusters
hc.norm = hclust(dist(tsne_plot[,c(1,2)]))
tsne_plot$hclust = factor(cutree(hc.norm, 7))
hc.norm.cent = tsne_plot %>% group_by(hclust) %>% select(V1, 
    V2) %>% summarize_all(mean)
plotTsne <- ggplot(tsne_plot, aes(x = V1, y = V2, colour = hclust)) + 
  facet_grid(species~.)+
    geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label = hclust), 
    data = hc.norm.cent) + guides(colour = FALSE) + 
    ggtitle("Linkage criterion: Complete")

  # pdf("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_tsnePlot_sterm.pdf",width=5,height=5,paper='special')
  # svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_tsnePlot_sterm.svg",width=4,height=4)
  # png("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_tsnePlot_sterm.png",width=1600,height=1600,res=300)
plotTsne

  
library(plotly)
ggp <- ggplotly(plotTsne)
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_notexplained_sterm_colored_tsne.html")



# dev.off()
tsne_plot_sterm <- tsne_plot
##------------------Ldel
# 
# sample_relative_safe_final_tsne_safe_ldel_plasmid <- sample_relative_safe_final_meta_cleaned_02[grep("L_delbrueckii_RMK202_",sample_relative_safe_final_meta_cleaned_02$site),]
#   
# ##duplicate rows
# nrow(sample_relative_safe_final_tsne_safe_ldel_plasmid)
# sample_relative_safe_final_tsne_safe_ldel_woDuplicates <- sample_relative_safe_final_tsne_safe_ldel_plasmid[!duplicated(sample_relative_safe_final_tsne_safe_ldel_plasmid[,-1]), ]
# nrow(sample_relative_safe_final_tsne_safe_ldel_woDuplicates)   
# sample_relative_safe_final_tsne_safe_ldel_woDuplicates <- sample_relative_safe_final_tsne_safe_ldel_woDuplicates[!rowSums(is.na(sample_relative_safe_final_tsne_safe_ldel_woDuplicates))>0,]
# library(philentropy)
# distance(sample_relative_safe_final_tsne_safe_ldel_woDuplicates[,-1], method = "jaccard")
# 
# 
# tsne <- Rtsne(sample_relative_safe_final_tsne_safe_ldel_woDuplicates[,-1], dims = 2, perplexity=30, verbose=TRUE, max_iter = 500,)
# tsne_plot <- as.data.frame(tsne$Y)
# tsne_plot$site <- sample_relative_safe_final_tsne_safe_ldel_woDuplicates[,1]
# tsne_plot$contig <- paste(str_split_fixed(tsne_plot$site, "_", 5)[,1],str_split_fixed(tsne_plot$site, "_", 5)[,2],sep = "_")
# 
# 
# 
# 
# unique(tsne_plot$contig)
# tsne_plot$species <- tsne_plot$contig
# # tsne_plot$species <- ifelse(tsne_plot$contig=="contig_1","S.thermophilus",
# #                                              ifelse(tsne_plot$contig=="contig_2","L.delbrueckii",
# #                                                     ifelse(tsne_plot$contig=="contig_3","Lactococcus phage",
# #                                                            ifelse(tsne_plot$contig=="contig_4","Cow",
# #                                                                   ifelse(tsne_plot$contig=="contig_5","L.delbrueckii plasmid",
# #                                                                         ifelse(tsne_plot$contig=="plasmidSpades_CowMito","Cow mitogenome",NA))))))
# 
# unique(tsne_plot$species)
# 
# ##plot blank
# ggplot(tsne_plot,aes(x=V1,y=V2))+geom_point()
# 
# ##clusters
# hc.norm = hclust(dist(tsne_plot[,c(1,2)]))
# tsne_plot$hclust = factor(cutree(hc.norm, 20))
# hc.norm.cent = tsne_plot %>% group_by(hclust) %>% select(V1, 
#     V2) %>% summarize_all(mean)
# plotTsne <- ggplot(tsne_plot, aes(x = V1, y = V2, colour = hclust)) + 
#   facet_grid(species~.)+
#     geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label = hclust), 
#     data = hc.norm.cent) + guides(colour = FALSE) + 
#     ggtitle("Linkage criterion: Complete")
#   plotTsne
# 
#   # pdf("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_tsnePlot_ldel.pdf",width=5,height=5,paper='special')
# # svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_tsnePlot_ldel.svg",width=4,height=4)
# png("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_tsnePlot_ldel.png",width=1600,height=1600,res=300)
#   plotTsne
# 
# dev.off()
# tsne_plot_ldel <- tsne_plot
# tsne_plot_all <- rbind(tsne_plot_sterm,tsne_plot_ldel)


```

######add tsne annoation to plot

```{r}

not_explained_sterm_plot$grouping <- "none"
not_explained_sterm_plot$grouping <- tsne_plot_sterm[match(not_explained_sterm_plot$site,tsne_plot_sterm$site),"hclust"]
table(not_explained_sterm_plot$grouping)

# unique(sample_relative_safe_final$sample_02)
# sample_relative_safe_final_tsne_safe_ldel<- sample_relative_safe_final[grep("L_delbrueckii_RMK202_",sample_relative_safe_final$site),]
# sample_relative_safe_final_tsne_safe_sterm <- sample_relative_safe_final[grep("S_thermophilus_RMK202",sample_relative_safe_final$site),]
# 
# nrow(sample_relative_safe_final_tsne_safe_sterm)
# sample_relative_safe_final_reduce <- rbind(sample_relative_safe_final_tsne_safe_ldel,sample_relative_safe_final_tsne_safe_sterm)


  
  pcoloredtsne <- ggplot(not_explained_sterm_plot,aes(x=sample_renamed,y=allele_frequency,group=site,color=grouping,fill=grouping))+ geom_line(size=0.1, alpha=.1)+ 
      # facet_grid(species~.)+
   labs("",
         x="",
         # y="",
          y="not\nexplained")+
       # scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="right",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
  
  pcoloredtsne
  
library(plotly)
ggp <- ggplotly(pcoloredtsne)

htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_notexplained_sterm_colored.html")

  
  
  # pdf("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored.pdf",width=9.8,height=20,paper='special')
svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored.svg",width=5,height=4)
# png("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored.png",width=2000,height=1500,res=300)
  pcoloredtsne

dev.off()


```
######look at tsne grouping locations

```{r}

nrow(sample_relative_safe_final_tsne_safe_ldel)
nrow(sample_relative_safe_final_tsne_safe_sterm)
sample_relative_safe_final_reduce <- rbind(sample_relative_safe_final_tsne_safe_ldel,sample_relative_safe_final_tsne_safe_sterm)

##sterm

ggplot(sample_relative_safe_final_tsne_safe_sterm,aes(x=POS,group=grouping,color=grouping,fill=grouping))+geom_density(alpha=0.4)

##ldel

ldel_group1 <- sample_relative_safe_final_tsne_safe_ldel[sample_relative_safe_final_tsne_safe_ldel$grouping=="1",]
ggplot(ldel_group1,aes(x=POS))+geom_histogram()

ldel_group2 <- sample_relative_safe_final_tsne_safe_ldel[sample_relative_safe_final_tsne_safe_ldel$grouping=="2",]
ggplot(ldel_group2,aes(x=POS))+geom_histogram()

ldel_group3 <- sample_relative_safe_final_tsne_safe_ldel[sample_relative_safe_final_tsne_safe_ldel$grouping=="3",]
ggplot(ldel_group3,aes(x=POS))+geom_histogram()

ggplot(sample_relative_safe_final_tsne_safe_ldel,aes(x=POS,group=grouping,color=grouping,fill=grouping))+geom_density(alpha=0.2)

```





The two snps that decrease and are explained by both strains are modifier mutations of a zinc ABC transporter substrate-binding protein AdcA.


The snps that cannot be explained by the two strain seem to be upstream of a ABC transporter ATP-binding protein. 

#####annote the SCOG with 
```{bash}
##=============
###make a query file
##=============

subgroups=Lactobacillus_delbrueckii_group_subgroup_2

for subgroups in $(echo "Lactobacillus_delbrueckii_group_subgroup_2 Streptococcus_thermophilus_group_subgroup_4")
do
#date_fix=Results_May17
date_fix=$(ls /archiv/Dialact/ReferenceDatabase/20190513/Orthofinder/01_normalgrouping/${subgroups}/ )

echo "##=============================="
echo ${subgroups}
echo "##=============================="


rm /archiv/Dialact/ReferenceDatabase/20190513/Orthofinder/01_normalgrouping/${subgroups}/${date_fix}/blastOGs.fasta

for OGs in $(cat /archiv/Dialact/ReferenceDatabase/20190513/Orthofinder/01_normalgrouping/${subgroups}/${date_fix}/Orthogroups/Orthogroups_SingleCopyOrthologues.txt)
do
#OGs=$(echo ${OG} | cut -d '.' -f 1)

## extract first fasta in multiFasta
awk '/^>/{if(N)exit;++N;} {print;}' /archiv/Dialact/ReferenceDatabase/20190513/Orthofinder/01_normalgrouping/${subgroups}/${date_fix}/Orthogroup_Sequences/${OGs}.fa | sed "s/>/>$OGs /g" >> /archiv/Dialact/ReferenceDatabase/20190513/Orthofinder/01_normalgrouping/${subgroups}/${date_fix}/blastOGs.fasta


done # subgroups
done #OG

##=============
###make blast DB
##=============


threads=37

num=1
for names in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt )
  do
  echo ${num}"/4  :" ${names}
  num=$((num+1))
echo "---------------"
${names}
makeblastdb -max_file_sz 10GB -in /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/03_PROKKA_annotation_polishedGenome/${names}/PROKKA_*.ffn \
 -out /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/03_PROKKA_annotation_polishedGenome/${names}/BlastDB -parse_seqids -dbtype nucl

done

##=============
###tblastn query to DB
##=============
threads=37

num=1
for names in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt )
  do
  echo ${num}"/4  :" ${names}
  num=$((num+1))
echo "---------------"
echo ${names}

for subgroups in $(echo "Lactobacillus_delbrueckii_group_subgroup_2 Streptococcus_thermophilus_group_subgroup_4")
do

rm -r /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/BLASTannotation_OG/${names}/${subgroups}
mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/BLASTannotation_OG/${names}/${subgroups}

date_fix=$(ls /archiv/Dialact/ReferenceDatabase/20190513/Orthofinder/01_normalgrouping/${subgroups}/ )

  tblastn -num_threads ${threads} -max_hsps 1 -max_target_seqs 1 -show_gis \
  -query /archiv/Dialact/ReferenceDatabase/20190513/Orthofinder/01_normalgrouping/${subgroups}/${date_fix}/blastOGs.fasta \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen" \
  -db /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/03_PROKKA_annotation_polishedGenome/${names}/BlastDB \
  -out /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/BLASTannotation_OG/${names}/${subgroups}/${subgroups}_${names}_annotation_blast.txt \
  -evalue 0.01 -word_size 4


done #subgroups
done #name


##=============
###create gff file of OGs
##=============

threads=37

num=1
for names in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt )
  do
  echo ${num}"/4  :" ${names}
  num=$((num+1))
echo "---------------"
echo ${names}

for subgroups in $(echo "Lactobacillus_delbrueckii_group_subgroup_2 Streptococcus_thermophilus_group_subgroup_4")
do

rm /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/BLASTannotation_OG/${names}/${subgroups}/${subgroups}_${names}_annotation_blast.gff

echo "============="
echo ${subgroups}

n=$(wc -l  /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/BLASTannotation_OG/${names}/${subgroups}/${subgroups}_${names}_annotation_blast.txt |awk '{print $1}')

for OG in $(seq 1 $n)
do

OGs=$(sed "${OG}q;d" /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/BLASTannotation_OG/${names}/${subgroups}/${subgroups}_${names}_annotation_blast.txt |awk -F "\t" '{print $1}')
gene=$(sed "${OG}q;d" /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/BLASTannotation_OG/${names}/${subgroups}/${subgroups}_${names}_annotation_blast.txt |awk -F "\t" '{print $2}')

#gene=$(echo "th_K2_8h_00003")

#/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/03_PROKKA_annotation_polishedGenome/${names}/

#grep "IfD=$gene;" /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/03_PROKKA_annotation_polishedGenome/${names}/PROKKA_*.gff | awk -F "\t"

grep "ID=${gene};" /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/03_PROKKA_annotation_polishedGenome/${names}/PROKKA_*.gff |awk -F "--" -v og="$OGs" -v spe="$subgroups" -v gens="$gene" '{print $0";OG="og";grouping="spe";gene="gens}' >> \
                /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/BLASTannotation_OG/${names}/${subgroups}/${subgroups}_${names}_annotation_blast.gff


done #OG
done #subgroups
done #name


##=============
##extracting coverage
##=============




num=1
for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name.txt)
  do
  echo ${num}"/37  :" ${names}
  num=$((num+1))
#  rm -r /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/${names}

#  mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/${names}/bwaMapping2DB/
  
  sampleShort=$(echo ${names}| awk -F "_" 'BEGIN{OFS="_"}{print $1,$2}')
  
   NanoSample=$(grep "$sampleShort" /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt) 
  


  threads=30
mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/raw/
rm /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/raw/${names}_mapping_2_OG_${subgroups}.bed

for subgroups in $(echo "Lactobacillus_delbrueckii_group_subgroup_2 Streptococcus_thermophilus_group_subgroup_4")
do
    #rm -r /home/vincent/Projects/2019_pilotplant/05_mapping2DB/${names}/BacteriaDBmapping
  
    #mkdir -p /home/vincent/Projects/2019_pilotplant/05_mapping2DB/${names}/BacteriaDBmapping
    

echo "============="
echo ${subgroups}



bedtools coverage -bed -a /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/BLASTannotation_OG/${NanoSample}/${subgroups}/${subgroups}_${NanoSample}_annotation_blast.gff -b /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam  > \
  /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/raw/${names}_mapping_2_OG_${subgroups}.bed
  
  
done #subgroup
done #names





###----------------change name of bed file
 
 rm -r /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/tmp/
 rm -r /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/final/
 mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/tmp/
  mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/final/

for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name.txt )
    do
    
       # name=th_K2_4h


Day=$(echo $names |cut -d '_' -f 1)
Kessel=$(echo $names |cut -d '_' -f 2)
time=$(echo $names |cut -d '_' -f 3|sed 's/h//g')

 echo ${num}"/37  :" ${names}
  num=$((num+1))
#  rm -r /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/${names}

#  mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/${names}/bwaMapping2DB/
  
  sampleShort=$(echo ${names}| awk -F "_" 'BEGIN{OFS="_"}{print $1,$2}')
  
   NanoSample=$(grep "$sampleShort" /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt) 
  
 # rm /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/tmp/
  mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/tmp/
  mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/final/


for subgroups in $(echo "Lactobacillus_delbrueckii_group_subgroup_2 Streptococcus_thermophilus_group_subgroup_4")
do

#home/vincent/Projects/2019_pilotplant/05_mapping2DB/${name}/BacteriaDBmapping/${name}2BacteriaDB.bed
   
    cat /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/raw/${names}_mapping_2_OG_${subgroups}.bed | sed -e 's/ID=.*;OG=/OG=/g'|awk -F "[\t;]" -v namess="$name" -v days="$Day" -v Kessels="$Kessel" -v times="$time" 'BEGIN{OFS="\t"}{print $1,$4,$5,$7,$9,$10,$11,$12,$13,$14,$15,days,Kessels,times}' |sed 's/OG=//g' |sed 's/grouping=//g'|sed 's/gene=//g' |sed 's/genome=//g' > /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/tmp/${names}_mapping_2_OG_${subgroups}.tmp
  
#cut -d '_' -f 1,2 /home/vincent/Projects/2019_pilotplant/05_mapping2DB/BacteriaDBmapping/tmp.bed > /home/vincent/Projects/2019_pilotplant/05_mapping2DB/BacteriaDBmapping/tmp.species

#awk -F "[-_]" 'BEGIN{OFS="_"}{print $1,$2}' /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/tmp/${names}_mapping_2_OG_${subgroups}.tmp > /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/tmp/${names}_mapping_2_OG_${subgroups}.species


#paste /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/tmp/${names}_mapping_2_OG_${subgroups}.species /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/tmp/${names}_mapping_2_OG_${subgroups}.tmp > /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/final/${names}_mapping_2_OG_${subgroups}_final.bed

#rm /home/vincent/Projects/2019_pilotplant/05_mapping2DB/BacteriaDBmapping/tmp.bed
#rm /home/vincent/Projects/2019_pilotplant/05_mapping2DB/BacteriaDBmapping/tmp.species

#cut -f 1 /home/vincent/Projects/2019_pilotplant/05_mapping2DB/BacteriaDBmapping/${name}2BacteriaDB_final.bed|sort|uniq -c
#awk '{if($1=="_")print $0}' /home/vincent/Projects/2019_pilotplant/05_mapping2DB/BacteriaDBmapping/${name}2BacteriaDB_final.bed

    done ##subgroups
    done ##names
    
    
##local


scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRcasFinder/test_meta/202-SMAG_cas.tar.gz /home/vincent/Desktop/

scp -r vincent@130.223.51.151:/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/tmp/ /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/OG_DB
```


```{r}
source("/home/vincent/Desktop/ScriptRepository/R_functions-master/R_functions/g_legend.R") ##for plotting only legends
library(readr)
library(ggplot2)
library(dplyr)
library(tidyverse)
library(grid)
library(gridExtra)
library(cowplot) ## for arranging plots
library(stringr) ##for split string
library(rlist) ##rbind all dataframe of list
library(viridis) ##colour palette
##===================================
#-------------file import
# startingGff <- read_delim("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/phageDBmapping/di_K2_6h2phageDB_final.bed","\t", escape_double = FALSE, col_names = c("species","chr","start","end","orientation","OG","group","gene","genome","genomeSize","count","length_unmapped","geneLength","percent","name","day","kessel","time"),trim_ws = TRUE)

samples_names <- read_csv("~/Desktop/Projects/2019_Pilotplan/samples_ordered.txt", col_names = FALSE)
listcsv_ldel <- as.vector(paste0("/home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/OG_DB/tmp/",samples_names$X1,"_mapping_2_OG_Lactobacillus_delbrueckii_group_subgroup_2.tmp"))
listcsv_sterm <- as.vector(paste0("/home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/OG_DB/tmp/",samples_names$X1,"_mapping_2_OG_Streptococcus_thermophilus_group_subgroup_4.tmp"))

#read_count <- read_delim("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/04_againstAllWithCow/Mulitqc/table_count_final.txt", 
 #   "\t", escape_double = FALSE, col_names = c("name","MioReads"), 
    # trim_ws = TRUE)
# k <- 1
ldf_ldel <- list()
ldf_sterm <- list()

for (k in 1:length(listcsv_ldel)){
 ldf_ldel[[k]] <- read_delim(listcsv_ldel[[k]],"\t", escape_double = FALSE, col_names = c("chr","start","end","orientation","OG","group","gene","count","length_unmapped","geneLength","percent","day","kessel","time"),trim_ws = TRUE)
 ldf_sterm[[k]] <- read_delim(listcsv_sterm[[k]],"\t", escape_double = FALSE, col_names = c("chr","start","end","orientation","OG","group","gene","count","length_unmapped","geneLength","percent","day","kessel","time"),trim_ws = TRUE)

  
  # ldf_ldel[[k]] <- ldf_ldel[[k]]
}
#ldf[[1]]

# head(ldf[[2]])
# 
# dim(startingGff)
# table(startingGff$count)
# startingGff$geneCoverage <- (startingGff$count*600)/startingGff$geneLength

# ggplot(startingGff,aes(x=group,y=geneCoverage,fill=species,colour=species) )+geom_boxplot()+
#     theme(axis.text.x = element_text(angle = 75, hjust = 1),
#           #legend.position="top",
#           legend.justification=c(1,1), legend.position=c(1,1),
#           legend.title = element_blank()
#           )
##===================================
###---TPKM calculations


# startingGff$RawReads <- unlist(read_count[match(startingGff$name,read_count$name),2])
# # aggregate(startingGff$RawReads, list(startingGff$name), mean)
# startingGff$TPKM <- ((startingGff$count*600)/startingGff$RawReads)/startingGff$geneLength
# summary(startingGff$TPKM)


p <- list()
for (i in  1:length(listcsv_sterm)){
  ldf_ldel[[i]]$geneCoverage  <- (ldf_ldel[[i]]$count*600)/ldf_ldel[[i]]$geneLength
  ldf_ldel[[i]]$name  <- paste0(ldf_ldel[[i]]$day,"_",ldf_ldel[[i]]$kessel,"_",ldf_ldel[[i]]$time,"h")
  # ldf_ldel[[i]]$RawReads <- unlist(read_count[match(ldf_ldel[[i]]$name,read_count$name),"MioReads"])
  # ldf_ldel[[i]]$TPKM <- ((ldf_ldel[[i]]$count)/ldf_ldel[[i]]$RawReads)/ldf_ldel[[i]]$geneLength
  ldf_sterm[[i]]$geneCoverage  <- (ldf_sterm[[i]]$count*600)/ldf_sterm[[i]]$geneLength
  ldf_sterm[[i]]$name  <- paste0(ldf_sterm[[i]]$day,"_",ldf_sterm[[i]]$kessel,"_",ldf_sterm[[i]]$time,"h")

}


##===================================
####-----------------plot only grouping=Streptococcus_thermophilus_group_subgroup_4


sterm_s4_list <- list()
p1 <- list()
p2 <- list()
sterm_s4Plot <- list()
OG_sterm_s4_sum_list <- list()
# OG_Ldel_s2_sum_list <- list()
OG_sterm_s4_quantile_list <- as.data.frame(matrix(0, ncol = 7, nrow = 32))
colnames(OG_sterm_s4_quantile_list) <- c("sample","0%","25%","50%","75%","100%","PTR")

for (i in  1:length(ldf_sterm)){

  sterm_s4_list[[i]] <- ldf_sterm[[i]][ldf_sterm[[i]]$group=="Streptococcus_thermophilus_group_subgroup_4",]

  # p1[[i]] <- ggplot() + geom_bar(aes(y = TPKM, x = reorder(OG, -TPKM), fill = genome), data = sterm_s4_list[[i]], stat="identity")+
  #   labs(title = paste0(unique(sterm_s4_list[[i]]$group), " from ",unique(sterm_s4_list[[i]]$name)),
  #        x="Orthologous groups",
  #        y="")+
  #   theme_classic()+
  #   theme(#axis.text.x = element_text(angle = 75, hjust = 1),
  #         axis.text.x = element_blank(),
  #         plot.title = element_text(size=9,hjust = 0.5),
  #         legend.position="none",
  #         #legend.justification=c(1,1), legend.position=c(1,1),
  #         #legend.title = element_blank()
  #         )

  # OG_sterm_s4_sum_list[[i]] <- aggregate(sterm_s4_list[[i]]$TPKM, list(sterm_s4_list[[i]]$OG), sum)
  OG_sterm_s4_sum_list[[i]] <- aggregate(sterm_s4_list[[i]]$geneCoverage, list(sterm_s4_list[[i]]$OG), sum)

   # OG_Ldel_s2_sum_list[[i]] <- aggregate(Ldel_s2_list[[i]]$TPKM, list(Ldel_s2_list[[i]]$OG), sum)
    OG_sterm_s4_quantile_list[i,1:6] <- c(unique(sterm_s4_list[[i]]$name),quantile(OG_sterm_s4_sum_list[[i]]$x))
OG_sterm_s4_quantile_list[i,7] <- as.numeric(OG_sterm_s4_quantile_list[i,5])/as.numeric(OG_sterm_s4_quantile_list[i,3])
  
  
  p1[[i]] <- ggplot(sterm_s4_list[[i]],aes(x=start,y=geneCoverage,fill=name,colour=name) )+geom_point(size = 0.1)+
    facet_grid(~name)+
    # labs(title = paste0(unique(ldf[[i]]$name)),
    #      x="",
    #      y="")+
    theme(axis.text.x = element_text(angle = 75, hjust = 1),
          #axis.text.x =element_blank(),
          plot.title = element_text(size=20),
          legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          )
  
}

pdf("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/figures/coverageOG_sterm.pdf",width=8,height=16,paper='special') 

# do.call(grid.arrange, c(p, ncol=4))

do.call(plot_grid, c(p1, labels = "TPKM", align = 'h', ncol=4))


dev.off()



  OG_sterm_s4_quantile_list$day <- str_split_fixed(OG_sterm_s4_quantile_list$sample, "_", 3)[,1]
  OG_sterm_s4_quantile_list$kessel <- str_split_fixed(OG_sterm_s4_quantile_list$sample, "_", 3)[,2]
  OG_sterm_s4_quantile_list$time <- str_split_fixed(OG_sterm_s4_quantile_list$sample, "_", 3)[,3]
  OG_sterm_s4_quantile_list$treatment <- paste0(OG_sterm_s4_quantile_list$day,"_",OG_sterm_s4_quantile_list$kessel)
 
  OG_sterm_s4_quantile_list$treatment_f = factor(OG_sterm_s4_quantile_list$treatment, levels=c('di_K1','th_K1','di_K2','th_K2'))
  OG_sterm_s4_quantile_list$time_f = factor(OG_sterm_s4_quantile_list$time, levels=c("0h","2h","4h","6h","8h","10h","12h","24h"))

OG_sterm_s4_quantile_list$species <- "S. thermophilus"

ggplot(OG_sterm_s4_quantile_list,aes(x=time_f,y=PTR,group=treatment_f,color=treatment_f,fill=treatment_f))+geom_line()+geom_point()+facet_grid(~kessel)

##viridis

#   pdf("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/Bacteria/Streptococcus_thermophilus_group_subgroup_4_OGs.pdf",width=20,height=32,paper='special')
# do.call(plot_grid, c(sterm_s4Plot, labels = "S_the_subgroup_4", align = 'h', ncol=4))
# 
# 
# dev.off()

# sum_ldels9 <- quantile(OG_ldels9_sum$x)
# sum_ldels9 <- c(unique(ldel_s9$group),unique(ldel_s9$name),unique(ldel_s9$day),unique(ldel_s9$kessel),unique(ldel_s9$time),sum_ldels9)
# names(sum_ldels9) <- c("subgroup","name","day","kessel","time","min","1st","median","3rd","max")
# sum_ldels9

##===================================
####-----------------plot only grouping=Lactobacillus_delbrueckii_group_subgroup_2


Ldel_s2_list <- list()
p1 <- list()
p2 <- list()
Ldel_s2Plot <- list()
OG_Ldel_s2_sum_list <- list()
OG_Ldel_s2_quantile_list <- as.data.frame(matrix(0, ncol = 7, nrow = 32))
colnames(OG_Ldel_s2_quantile_list) <- c("sample","0%","25%","50%","75%","100%","PTR")
# OG_Ldel_s2_quantile_list <- data.frame()

for (i in  1:length(ldf_ldel)){

  Ldel_s2_list[[i]] <- ldf_ldel[[i]][ldf_ldel[[i]]$group=="Lactobacillus_delbrueckii_group_subgroup_2",]

  # p1[[i]] <- ggplot() + geom_bar(aes(y = TPKM, x = reorder(OG, -TPKM), fill = genome), data = Ldel_s2_list[[i]], stat="identity")+
  #   labs(title = paste0(unique(Ldel_s2_list[[i]]$group), " from ",unique(Ldel_s2_list[[i]]$name)),
  #        x="Orthologous groups",
  #        y="")+
  #   theme_classic()+
  #   theme(#axis.text.x = element_text(angle = 75, hjust = 1),
  #         axis.text.x = element_blank(),
  #         plot.title = element_text(size=9,hjust = 0.5),
  #         legend.position="none"
  #         #legend.justification=c(1,1), legend.position=c(1,1),
  #         #legend.title = element_blank()
  #         )

  # OG_Ldel_s2_sum_list[[i]] <- aggregate(Ldel_s2_list[[i]]$TPKM, list(Ldel_s2_list[[i]]$OG), sum)
  OG_Ldel_s2_sum_list[[i]] <- aggregate(Ldel_s2_list[[i]]$geneCoverage, list(Ldel_s2_list[[i]]$OG), sum)
    OG_Ldel_s2_quantile_list[i,1:6] <- c(unique(Ldel_s2_list[[i]]$name),quantile(OG_Ldel_s2_sum_list[[i]]$x))
OG_Ldel_s2_quantile_list[i,7] <- as.numeric(OG_Ldel_s2_quantile_list[i,5])/as.numeric(OG_Ldel_s2_quantile_list[i,3])
  
  
  p1[[i]] <- ggplot(Ldel_s2_list[[i]],aes(x=start,y=geneCoverage,fill=name,colour=name) )+geom_point()+
    facet_grid(~name)+
    # labs(title = paste0(unique(ldf[[i]]$name)),
    #      x="",
    #      y="")+
    theme(#axis.text.x = element_text(angle = 75, hjust = 1),
          axis.text.x =element_blank(),
          plot.title = element_text(size=20),
          legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          )
  
}


# do.call(plot_grid, c(p1, labels = "TPKM", align = 'h', ncol=4))



pdf("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/figures/coverageOG.pdf",width=8,height=16,paper='special') 

# do.call(grid.arrange, c(p, ncol=4))

do.call(plot_grid, c(p1, labels = "TPKM", align = 'h', ncol=4))


dev.off()




  OG_Ldel_s2_quantile_list$day <- str_split_fixed(OG_Ldel_s2_quantile_list$sample, "_", 3)[,1]
  OG_Ldel_s2_quantile_list$kessel <- str_split_fixed(OG_Ldel_s2_quantile_list$sample, "_", 3)[,2]
  OG_Ldel_s2_quantile_list$time <- str_split_fixed(OG_Ldel_s2_quantile_list$sample, "_", 3)[,3]
  OG_Ldel_s2_quantile_list$treatment <- paste0(OG_Ldel_s2_quantile_list$day,"_",OG_Ldel_s2_quantile_list$kessel)
 
  OG_Ldel_s2_quantile_list$treatment_f = factor(OG_Ldel_s2_quantile_list$treatment, levels=c('di_K1','th_K1','di_K2','th_K2'))
  OG_Ldel_s2_quantile_list$time_f = factor(OG_Ldel_s2_quantile_list$time, levels=c("0h","2h","4h","6h","8h","10h","12h","24h"))

OG_Ldel_s2_quantile_list$species <- "L. delbrueckii"

all_OG <- rbind(OG_Ldel_s2_quantile_list,OG_sterm_s4_quantile_list)


ggplot(OG_Ldel_s2_quantile_list,aes(x=time_f,y=PTR,group=treatment_f,color=treatment_f,fill=treatment_f))+geom_line()+geom_point()

ggplot(all_OG,aes(x=time_f,y=PTR,group=treatment_f,color=treatment_f,fill=treatment_f))+geom_line()+geom_point(aes_string(size=as.numeric(all_OG$`50%`)))+facet_grid(~species~kessel)
# 
# 
# all_OG_ONT <- all_OG
# 
# all_OG_ONT$Method <- "ONT_assembled"

# png("~/Desktop/Projects/2019_Pilotplan/PTR_referenceDB.png",width=1500,height=1200,res=300)
# 
# ggplot(all_OG,aes(x=time_f,y=PTR,group=treatment_f,color=treatment_f,fill=treatment_f))+geom_line()+geom_point(aes_string(size=as.numeric(all_OG$`50%`)))+facet_grid(~species~kessel)+
#       labs("",
#          x="time",
#          y="Peak-to-trough ratio")+
#     theme_classic()+
#    # scale_color_viridis(discrete=TRUE)+
#   # scale_fill_viridis(discrete=TRUE)+
#     theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
#       #axis.text.x = element_blank(),
#           legend.position="right"
#           #legend.justification=c(1,1), legend.position=c(1,1),
#           #legend.title = element_blank()
#           )
# dev.off()


##======================
##only K1_di
##======================


samples_names <- read_csv("~/Desktop/Projects/2019_Pilotplan/samples_ordered.txt", col_names = FALSE)
samples_names_di_K1 <- samples_names[grep("di_K1",samples_names$X1),]

listcsv_ldel_di_K1 <- as.vector(paste0("/home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/OG_DB/tmp/",samples_names_di_K1$X1,"_mapping_2_OG_Lactobacillus_delbrueckii_group_subgroup_2.tmp"))
listcsv_sterm_di_K1 <- as.vector(paste0("/home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/OG_DB/tmp/",samples_names_di_K1$X1,"_mapping_2_OG_Streptococcus_thermophilus_group_subgroup_4.tmp"))



ldf_ldel_di_K1 <- list()
ldf_sterm_di_K1 <- list()

for (k in 1:length(listcsv_ldel_di_K1)){
 ldf_ldel_di_K1[[k]] <- read_delim(listcsv_ldel_di_K1[[k]],"\t", escape_double = FALSE, col_names = c("chr","start","end","orientation","OG","group","gene","count","length_unmapped","geneLength","percent","day","kessel","time"),trim_ws = TRUE)
 ldf_sterm_di_K1[[k]] <- read_delim(listcsv_sterm_di_K1[[k]],"\t", escape_double = FALSE, col_names = c("chr","start","end","orientation","OG","group","gene","count","length_unmapped","geneLength","percent","day","kessel","time"),trim_ws = TRUE)

  
}
##===================================
###---TPKM calculations



p <- list()
for (i in  1:length(ldf_sterm_di_K1)){
  ldf_ldel_di_K1[[i]]$geneCoverage  <- (ldf_ldel_di_K1[[i]]$count*600)/ldf_ldel_di_K1[[i]]$geneLength
  ldf_ldel_di_K1[[i]]$name  <- paste0(ldf_ldel_di_K1[[i]]$day,"_",ldf_ldel_di_K1[[i]]$kessel,"_",ldf_ldel_di_K1[[i]]$time,"h")

  
  ldf_sterm_di_K1[[i]]$geneCoverage  <- (ldf_sterm_di_K1[[i]]$count*600)/ldf_sterm_di_K1[[i]]$geneLength
  ldf_sterm_di_K1[[i]]$name  <- paste0(ldf_sterm_di_K1[[i]]$day,"_",ldf_sterm_di_K1[[i]]$kessel,"_",ldf_sterm_di_K1[[i]]$time,"h")

}


##===================================
####-----------------plot only grouping=Streptococcus_thermophilus_group_subgroup_4


sterm_s4_list <- list()
p1 <- list()
p2 <- list()
sterm_s4Plot <- list()
OG_sterm_s4_sum_list <- list()
# OG_Ldel_s2_sum_list <- list()
OG_sterm_s4_quantile_list <- as.data.frame(matrix(0, ncol = 7, nrow = 32))
colnames(OG_sterm_s4_quantile_list) <- c("sample","0%","25%","50%","75%","100%","PTR")

sterm_s4_list_01 <- list()
sterm_s4_list_02 <- list()

for (i in  1:length(ldf_sterm_di_K1)){

  sterm_s4_list[[i]] <- ldf_sterm_di_K1[[i]][ldf_sterm_di_K1[[i]]$group=="Streptococcus_thermophilus_group_subgroup_4",]

  # OG_sterm_s4_sum_list[[i]] <- aggregate(sterm_s4_list[[i]]$TPKM, list(sterm_s4_list[[i]]$OG), sum)
  OG_sterm_s4_sum_list[[i]] <- aggregate(sterm_s4_list[[i]]$geneCoverage, list(sterm_s4_list[[i]]$OG), sum)

   # OG_Ldel_s2_sum_list[[i]] <- aggregate(Ldel_s2_list[[i]]$TPKM, list(Ldel_s2_list[[i]]$OG), sum)
    OG_sterm_s4_quantile_list[i,1:6] <- c(unique(sterm_s4_list[[i]]$name),quantile(OG_sterm_s4_sum_list[[i]]$x))
OG_sterm_s4_quantile_list[i,7] <- as.numeric(OG_sterm_s4_quantile_list[i,5])/as.numeric(OG_sterm_s4_quantile_list[i,3])
  
  
  p1[[i]] <- ggplot(sterm_s4_list[[i]],aes(x=start,y=geneCoverage,fill=name,colour=name) )+geom_point(size = 0.5)+
    facet_grid(~name)+
    # labs(title = paste0(unique(ldf[[i]]$name)),
    #      x="",
    #      y="")+
    theme(axis.text.x = element_text(angle = 75, hjust = 1),
          #axis.text.x =element_blank(),
          plot.title = element_text(size=20),
          legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          )
  
sterm_s4_list_01[[i]] <- sterm_s4_list[[i]][which(sterm_s4_list[[i]]$start<922302),]
sterm_s4_list_02[[i]] <- sterm_s4_list[[i]][which(sterm_s4_list[[i]]$start>922302),]

}

pdf("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/figures/coverageOG_sterm_di_K1.pdf",width=3,height=24,paper='special') 

# do.call(grid.arrange, c(p, ncol=4))

do.call(plot_grid, c(p1, labels = "TPKM", align = 'h', ncol=1))


dev.off()

##linear regression

# summary(sterm_s4_list[[1]]$start)

sterm_s4_list_01[[i]] <- sterm_s4_list[[i]][which(sterm_s4_list[[i]]$start<922302),]
sterm_s4_list_02[[i]] <- sterm_s4_list[[i]][which(sterm_s4_list[[i]]$start>922302),]

linearMod <- lm(sterm_s4_list_01[[1]]$geneCoverage ~ sterm_s4_list_01[[1]]$start)
print(linearMod)

linearMod <- lm(sterm_s4_list_02[[1]]$geneCoverage ~ sterm_s4_list_02[[1]]$start)
print(linearMod)


linearMod <- lm(dist ~ speed, data=sterm_s4_list)  # build linear regression model on full data
print(linearMod)

```
#### Phage stability

Here, I test if the phages are also stabil. If we have a stabil bacterila community and phage community we could have a carrier state life cycle. 

```{r}
# colnames(sample_relative_wide_phagesONly) <- revalue(colnames(sample_relative_wide_phagesONly), c("S50"="mst1","S72"="mst2","L70"="mst3", "L44"="mst10", "L108"="mst7", "L99"="mst8", "L104"="mst6", "L80"="mst5", "L35"="mst9", "L71"="mst4","lyo202_96"="Lyo\n1996","Lyo_202_2012"="Lyo\n2012", "Lyo_202_2014"="Lyo\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018","di_K2_6h"="cheesemaking\nday1","th_K2_8h"="cheesemaking\nday2"))
# colnames(sample_relative_wide_phagesONly) <- revalue(colnames(sample_relative_wide_phagesONly), c("species.x"="species","S50"="mst1","S72"="mst2","L70"="mst3", "L44"="mst10", "L108"="mst7", "L99"="mst8", "L104"="mst6", "L80"="mst5", "L35"="mst9", "L71"="mst4","lyo202_96"="Lyo\n1996","Lyo_202_2012"="Lyo\n2012", "Lyo_202_2014"="Lyo\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018","di_K2_6h"="cheesemaking\nday1","th_K2_8h"="cheesemaking\nday2"))
colnames(sample_relative_wide_phagesONly)
colnames(sample_relative_wide_phagesONly) <- revalue(colnames(sample_relative_wide_phagesONly), c("Streptococcus_phage_3"="Lactobacillus phage 2","Streptococcus_phage_2"="Streptococcus phage 2","Streptococcus_phage_1"="Streptococcus phage 1"))

table(sample_relative_wide_phagesONly$X.CHROM)
sample_relative_wide_phagesONly_prep <- sample_relative_wide_phagesONly%>%replace(is.na(.), 0)

# sample_relative_long_phage <- gather(sample_relative_wide_phagesONly, sample, Snps, "cheesemaking\nday2":"mst2", factor_key=TRUE,na.rm = TRUE) 
sample_relative_long_phage <- gather(sample_relative_wide_phagesONly_prep, sample, Snps, "Versand_202":"G2_6_18", factor_key=TRUE,na.rm = TRUE) 

sample_relative_long_phage$sample <- revalue(sample_relative_long_phage$sample, c("lyo202_96"="Lyo 1996","Lyo_202_2012"="Lyo 2012", "Lyo_202_2014"="Lyo 2014", "Konserve_202"="working stock", "RMK202"="starter culture 2012", "Versand_202"="starter culture 2018","di_K2_6h"="cheesemaking day1","th_K2_8h"="cheesemaking day2","G1_6_18"="experiment_A","G2_6_18"="experiment_B","G3_6_18"="experiment_C","G4_6_18"="experiment_D","G5_6_18"="experiment_E"))

sample_relative_long_phage_woIsolates <- sample_relative_long_phage %>% filter(sample %in% c("Lyo\ 1996","Lyo 2012","Lyo 2014","working stock","starter culture 2012","starter culture 2018","experiment_A","experiment_B","experiment_C","experiment_D","experiment_E"))

sample_relative_long_phage_woIsolates$sample <- droplevels(sample_relative_long_phage_woIsolates$sample)
sample_relative_long_phage_woIsolates$sample = factor(sample_relative_long_phage_woIsolates$sample, levels=c("Lyo 1996","Lyo 2012","Lyo 2014","working stock","starter culture 2012","starter culture 2018","experiment_A","experiment_B","experiment_C","experiment_D","experiment_E"))


##reorder samples









sample_relative_long_phage_woIsolates <- sample_relative_long_phage_woIsolates %>%replace(is.na(.), 0)
# nrow()


  ##============
### plot 
   ##============

# all_colours <- rev(c("#EB4D4D","#10B552") ) 

colnames(sample_relative_long_phage_woIsolates)
# sample_relative_long_phage_woIsolates$X.CHROM
##plot
  p2 <- ggplot(sample_relative_long_phage_woIsolates,aes(x=sample,y=Snps,group=site,color=X.CHROM,fill=X.CHROM))+ geom_line(size=1, alpha=0.5)+ 
    # facet_grid(day~kessel~species)+
    # facet_grid(species~.)+
    # facet_grid(treatment~species)+
    # facet_grid(treatment~species)+
    labs("",
         x=" ",
         y="Alternative allele frequency")+
    #scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    scale_y_continuous(limits = c(0,1))+
    theme_classic()+
   # scale_fill_manual(values=all_colours)+
   # scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
      #axis.text.x = element_blank(),
          legend.position="right",
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank()
          )
  
  p2

   svg("~/Desktop/mid_thesis/report/figures/20200430/chapter1/20201006/supplement/SNVs_phages_OverTime.svg",width=6,height=3.5)
#
p2
#
 dev.off()
 
 ###---------------------------------------
 ###for manuscript
 ###this means without the cheesemaking samples
 ###---------------------------------------
 sample_relative_long_phage_woIsolates_woCheesemaking <- sample_relative_long_phage_woIsolates %>% filter(sample %in% c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018"))
 sample_relative_long_phage_woIsolates_woCheesemaking <- sample_relative_long_phage_woIsolates_woCheesemaking %>% filter(X.CHROM %in% c("Streptococcus_phage_1" ,"Streptococcus_phage_2"))

 sample_relative_long_phage_woIsolates_woCheesemaking$X.CHROM <- revalue(sample_relative_long_phage_woIsolates_woCheesemaking$X.CHROM, c("Streptococcus_phage_1"="S. phage 1","Streptococcus_phage_2"="S. phage 2"))

 table(sample_relative_long_phage_woIsolates_woCheesemaking$X.CHROM)
# colnames(sample_relative_long_phage_woIsolates)
# sample_relative_long_phage_woIsolates$X.CHROM
##plot
  p12 <- ggplot(sample_relative_long_phage_woIsolates_woCheesemaking,aes(x=sample,y=Snps,group=site,color=X.CHROM,fill=X.CHROM))+ geom_line(size=1, alpha=0.5)+ 
    # facet_grid(day~kessel~species)+
    # facet_grid(species~.)+
    # facet_grid(treatment~species)+
    # facet_grid(treatment~species)+
    labs("",
         x=" ",
         y="alternative allele frequency")+
    #scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    scale_y_continuous(limits = c(0,1))+
    theme_classic()+
   # scale_fill_manual(values=all_colours)+
   # scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
      #axis.text.x = element_blank(),
          legend.position="bottom",
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank()
          )
  
  p12
library(patchwork)
   svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/Phage_rmk202_SNVs_new_manuscript.svg",width=7,height=4.5)
#
mst1_phased_ribbon+p12+plot_layout(widths = c(2,1))
#
 dev.off()
 
 
 ###-----------------------------
 ##export phage mutation
###-----------------------------
 
 tmp <- unique(sample_relative_long_phage_woIsolates$site)[grep("Streptococcus_phage_3",unique(sample_relative_long_phage_woIsolates$site),invert = TRUE)]
 genomomesss <- paste0("Streptococcus_phage_",as.character(str_split_fixed(tmp, "_", 5)[,3]))
 start <- as.numeric(str_split_fixed(tmp, "_", 5)[,4])
 end <- as.numeric(str_split_fixed(tmp, "_", 5)[,4])+10
 
 
 snps_freebayes <- read_delim("/home/vincent/Desktop/Projects/2019_RMK202_analysis/06_snpCalling2ONT_new/variantCallsfreebayes_minAlt3x_0.05_minCov10x_QC_noINDEL_prepR.vcf", "\t", escape_double = FALSE, trim_ws = TRUE)

 
 final_locations <- data.frame(genome=genomomesss,starts=start,endss=end)
 write.table(final_locations,file="/home/vincent/Desktop/Projects/2019_RMK202_analysis/06_snpCalling2ONT_new/snpSPhages.txt",sep = "\t",col.names = FALSE,row.names =FALSE,quote = FALSE )

```

move local

```{bash}

mkdir -p /archiv/Projects/2019_RMK202_analysis/06_snpCalling2ONT_new/
scp /home/vincent/Desktop/Projects/2019_RMK202_analysis/06_snpCalling2ONT_new/snpSPhages.txt vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/06_snpCalling2ONT_new/snpSPhages.txt

```

```{bash}

cat //archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_Streptococcus_phage_1.txt /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_Streptococcus_phage_2.txt > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages_Streptococcus_phage_all.txt

```



## ANI Figure

add hierachial clustering and order accordingly. This is trivial in the heatmap2 function but non-trivial for ggplot heatmpa. [here](https://jcoliver.github.io/learn-r/008-ggplot-dendrograms-and-heatmaps.html) is a blog post how to do it. 

```{r}
###=================
##Sterm
###=================

##--------------
##snps
##-------------
annotation_sterm_snps_01 <- sample_relative_wide ##I had to add this to run
colnames(annotation_sterm_snps_01)
sterm_snps_core <- annotation_sterm_snps_01
table(sterm_snps_core$X.CHROM.x)
sterm_snps_core <- sterm_snps_core[sterm_snps_core$X.CHROM.x=="CP046134",]
table(sterm_snps_core$X.CHROM.x)

sterm_snps_core$Sterm_rmk202 <- TRUE
sterm_snps_core$mst1 <- sterm_snps_core$mst1<0.9
sterm_snps_core$mst2 <- sterm_snps_core$mst2<0.9

sterm_snps_core <- sterm_snps_core %>% dplyr::select(site,core,Sterm_rmk202,mst1,mst2)
sterm_snps_core[c("mst1", "mst2")][is.na(sterm_snps_core[c("mst1", "mst2")])] <- TRUE

##--------------
##get length
##-------------
coreGeneNames_sterm_ldel <- read_delim("~/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/coreGeneNames_rmk202_bothLdelSterm.vcf", "\t", escape_double = FALSE, col_names = c("contig","method","annotation","start","end"), trim_ws = TRUE)
# table(coreGeneNames_sterm_ldel$contig)
coreGeneNames_sterm <- coreGeneNames_sterm_ldel[coreGeneNames_sterm_ldel$contig=="S_thermophilus_RMK202",]
# table(coreGeneNames_sterm$contig)
CoreGeneSize_sterm <- sum(coreGeneNames_sterm$end-coreGeneNames_sterm$start)
# 
##--------------
##plot
##-------------
library(stringr)
library(readr)
library(ggplot2)
library(reshape2)

table(sterm_snps_core$core)
sterm_snps_CORE <- sterm_snps_core[which(sterm_snps_core$core=="core"),]
table(sterm_snps_CORE$core)



##------------
##prep
##------------


colnames(sterm_snps_core)[3:5]
names_normal <- colnames(sterm_snps_core)[3:5]
count <- 1 ## triangle above
count <- 3 ; names <- rev(names_normal) ## triangle bellow


sterm_curated_rev_names <- data.frame()
# for (i in 1:length(names)) {  ## triangle above
for (i in rev(1:length(names))) { ## triangle bellow

   rowName <- names[i]
    print(rowName)
    
  for(a in rev(1:length(names))){
    
    colNames <- names[a]
    print(colNames)
    
    


    countSNPs <- sum(sterm_snps_CORE[,rowName]!=sterm_snps_CORE[,colNames])
    ANI_core <- 100*((CoreGeneSize_sterm-countSNPs)/CoreGeneSize_sterm)
    tmp <- data.frame("genome1"=rowName,"genome2"=colNames,"SNPs"=countSNPs,"ANI"=ANI_core)
    sterm_curated_rev_names <- rbind(sterm_curated_rev_names,tmp)
    

  }

# count <- count +1  ## triangle above
count <- count -1  ## triangle bellow

}
sterm_curated_rev_names


##------------
##make dendro of all snps sites
##------------
library(ggdendro)

sterm_matrix <- as.matrix(t(sterm_snps_CORE[, -c(1:2)]))
rownames(sterm_matrix) <- names_normal
otter.dendro <- as.dendrogram(hclust(d = dist(x = sterm_matrix)))


# Create dendro
dendro.plot <- ggdendrogram(data = otter.dendro, rotate = FALSE)
# Preview the plot
print(dendro.plot)
# 
# labels(otter.dendro)
# 
svg("~/Desktop/Projects/2018_culturomics/fastani/STERM_SNPs_DENDROGRAM.svg",width=2.5,height=3)
print(dendro.plot)
dev.off()

sterm_curated_rev_names
##reorder according to dendro
levels(sterm_curated_rev_names$genome1)
sterm_curated_rev_names$genome1 = factor(sterm_curated_rev_names$genome1, levels=labels(otter.dendro))
levels(sterm_curated_rev_names$genome1)
levels(sterm_curated_rev_names$genome2)
sterm_curated_rev_names$genome2 = factor(sterm_curated_rev_names$genome2, levels=labels(otter.dendro))
levels(sterm_curated_rev_names$genome2)

# creat Heatmap
# sterm_curated$X3 <- round(sterm_curated$X3,digits = 2)

##-----------
##change names
##-----------
sterm_curated_rev_names$genome1 <- revalue(sterm_curated_rev_names$genome1, c("Sterm_rmk202"="MAG RMK202"))
sterm_curated_rev_names$genome2 <- revalue(sterm_curated_rev_names$genome2, c("Sterm_rmk202"="MAG RMK202"))


##------------
##for SNP legend
##------------

ggheatmap <- ggplot(sterm_curated_rev_names, aes(genome1, genome2, fill = SNPs))+
 geom_tile(color = "white",size=1.1)+
 #scale_fill_gradient2(low = "blue", high = "red", mid = "white", 
  # midpoint = 99.5, limit = c(98.9,100), space = "Lab", 
  # name="ANI") +
  scale_fill_distiller(name = "SNPs", palette = "Blues", direction = -1)+
  # scale_fill_distiller(name = "ANI", palette = "Reds", direction = 1)+
  theme_minimal()+ # minimal theme
 theme(axis.text.y = element_text(size = 12),
      axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1))+
 coord_fixed()
# Print the heatmap
print(ggheatmap)

##------------
##SNPs
##------------


plotFinal_sterm <- ggheatmap + 
geom_text(aes(genome1, genome2, label = SNPs), color = "black", size = 2.5) +
    labs(title=paste0("# core genes :",nrow(coreGeneNames_sterm)," (",CoreGeneSize_sterm,"bp)"))+
  scale_y_discrete(position = 'right')+
theme(
  axis.title.x = element_blank(),
  axis.title.y = element_blank(),
  # axis.text.y.left = element_text(angle = 75, hjust = 1,size=9),
  panel.grid.major = element_blank(),
  # scale_y_date(position = "right"),
  panel.border = element_blank(),
  panel.background = element_blank(),
  axis.ticks = element_blank(),
  # legend.justification = c(1, 0),
  legend.position = "right",
  legend.text = element_text(angle = 0),
  legend.direction = "vertical")+
  guides(fill = guide_colorbar(barwidth = 1, barheight = 7,
                title.position = "top", title.hjust = 0.5))
plotFinal_sterm
##save heatmap
svg("~/Desktop/Projects/2018_culturomics/fastani/Sterm_SNPs_core_absolute_final.svg",width=5.5,height=4)
plotFinal_sterm#+theme(legend.position = "none")
dev.off()


##------------
##ANI 2
##------------
library(viridis)
# creat Heatmap
sterm_curated_rev_names$ANI_2 <- round(sterm_curated_rev_names$ANI,digits = 2)
min(sterm_curated_rev_names$ANI)
ggheatmap <- ggplot(sterm_curated_rev_names, aes(genome1, genome2, fill = ANI))+
 geom_tile(color = "white",size=1.1)+
 # scale_fill_gradient2(low = "red", high = "red4",
 # midpoint = 99.7, limit = c(99.2,100), space = "Lab",
 # name="ANI") +
  # scale_fill_viridis(discrete = FALSE, option = "B",direction = -1)+
  scale_fill_distiller(name = "ANI", palette = "Reds", direction = 1)+
  theme_minimal()+ # minimal theme
 theme(axis.text.y = element_text(size = 12),
      axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1))+
 coord_fixed()
# Print the heatmap
print(ggheatmap)

plotFinal_sterm <- ggheatmap + 
geom_text(aes(genome1, genome2, label = ANI_2), color = "black", size = 2.5) +
  scale_y_discrete(position = 'right')+
  labs(title=paste0("# core genes :",nrow(coreGeneNames_sterm)," (",CoreGeneSize_sterm,"bp)"))+
theme(
  axis.title.x = element_blank(),
  axis.title.y = element_blank(),
  # axis.text = element_blank(),
  panel.grid.major = element_blank(),
  panel.border = element_blank(),
  panel.background = element_blank(),
  axis.ticks = element_blank(),
  # legend.justification = c(1, 0),
  legend.position = "right",
  legend.text = element_text(angle = 0),
  legend.direction = "vertical")+
  guides(fill = guide_colorbar(barwidth = 1, barheight = 7,
                title.position = "top", title.hjust = 0.5))
plotFinal_sterm
##save heatmap
svg("~/Desktop/Projects/2018_culturomics/fastani/Sterm_SNPs_core_ANI.svg",width=5.5,height=4)
plotFinal_sterm#+theme(legend.position = "none")
dev.off()

##--------------------------
##snp location
##-------------------------

##----SNPs between trains
isolates_different <- sterm_snps_core[which(sterm_snps_core$mst1!=sterm_snps_core$mst2),]


  isolates_different$location <- as.numeric(str_split_fixed(isolates_different$site, "_", 3)[,2])

  
  
  
isolate_differnt_density <- ggplot(isolates_different,aes(x=location))+geom_density(alpha=0.4)+theme_classic()
  # facet_wrap(~final_clust,scale = "free_y")+
  # scale_fill_manual(values = c(colors_substrains,"grey"))+
  # scale_color_manual(values = c(colors_substrains,"grey"))+
  # labs(x="genomic location",y="density")+theme(strip.text.x = element_blank(),legend.position = "none",legend.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank())
isolate_differnt_density

isolate_differnt_point <- ggplot(isolates_different,aes(x=location,y=1))+geom_jitter(width = 0.0, height = 0.4,alpha=0.3 ,size = 0.35)+theme_classic()
  # facet_wrap(~final_clust,scale = "free_y")+
  # scale_fill_manual(values = c(colors_substrains,"grey"))+
  # scale_color_manual(values = c(colors_substrains,"grey"))+
  # labs(x="genomic location",y="cluster")+theme(legend.position = "none",legend.title = element_blank(),axis.text.x = element_blank(),axis.ticks.x = element_blank())
isolate_differnt_point


##----shared SNPs
isolates_SAME<- sterm_snps_core[which(sterm_snps_core$mst1==FALSE & sterm_snps_core$mst2==FALSE),]
dim(isolates_SAME)

  isolates_SAME$location <- as.numeric(str_split_fixed(isolates_SAME$site, "_", 3)[,2])

  
  
  
isolate_differnt_density <- ggplot(isolates_SAME,aes(x=location))+geom_density(alpha=0.4)+theme_classic()
  # facet_wrap(~final_clust,scale = "free_y")+
  # scale_fill_manual(values = c(colors_substrains,"grey"))+
  # scale_color_manual(values = c(colors_substrains,"grey"))+
  # labs(x="genomic location",y="density")+theme(strip.text.x = element_blank(),legend.position = "none",legend.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank())
isolate_differnt_density

isolate_differnt_point <- ggplot(isolates_SAME,aes(x=location,y=1))+geom_jitter(width = 0.0, height = 0.4,alpha=0.3 ,size = 0.35)+theme_classic()
  # facet_wrap(~final_clust,scale = "free_y")+
  # scale_fill_manual(values = c(colors_substrains,"grey"))+
  # scale_color_manual(values = c(colors_substrains,"grey"))+
  # labs(x="genomic location",y="cluster")+theme(legend.position = "none",legend.title = element_blank(),axis.text.x = element_blank(),axis.ticks.x = element_blank())
isolate_differnt_point
```


```{r}
###=================
##Ldel
###=================
library(plyr)
##--------------
##snps
##-------------

colnames(annotation_sterm_snps_01)
sterm_snps_core <- annotation_sterm_snps_01
table(sterm_snps_core$X.CHROM.x)
sterm_snps_core <- sterm_snps_core[sterm_snps_core$X.CHROM.x=="CP046131",]
table(sterm_snps_core$X.CHROM.x)

sterm_snps_core$Ldel_rmk202 <- TRUE
sterm_snps_core$mst3 <- sterm_snps_core$mst3<0.9
sterm_snps_core$mst10 <- sterm_snps_core$mst10<0.9
sterm_snps_core$mst7 <- sterm_snps_core$mst7<0.9
sterm_snps_core$mst8 <- sterm_snps_core$mst8<0.9
sterm_snps_core$mst6 <- sterm_snps_core$mst6<0.9
sterm_snps_core$mst5 <- sterm_snps_core$mst5<0.9
sterm_snps_core$mst9 <- sterm_snps_core$mst9<0.9
sterm_snps_core$mst4 <- sterm_snps_core$mst4<0.9
sterm_snps_core$mst6 <- sterm_snps_core$mst6<0.9

# sterm_snps_core$mst3 <- sterm_snps_core$L70<0.9
# sterm_snps_core$mst10 <- sterm_snps_core$L44<0.9
# sterm_snps_core$mst7 <- sterm_snps_core$L108<0.9
# sterm_snps_core$mst8 <- sterm_snps_core$L99<0.9
# sterm_snps_core$mst6 <- sterm_snps_core$L104<0.9
# sterm_snps_core$mst5 <- sterm_snps_core$L80<0.9
# sterm_snps_core$mst9 <- sterm_snps_core$L35<0.9
# sterm_snps_core$mst4 <- sterm_snps_core$L71<0.9
# sterm_snps_core$mst6 <- sterm_snps_core$L104<0.9
sterm_snps_core <- sterm_snps_core %>% select(site,core,Ldel_rmk202,mst3, mst4,mst5, mst6,mst7, mst8,mst9, mst10)
sterm_snps_core[c("mst3", "mst4","mst5", "mst6","mst7", "mst8","mst9", "mst10")][is.na(sterm_snps_core[c("mst3", "mst4","mst5", "mst6","mst7", "mst8","mst9", "mst10")])] <- TRUE


##--------------
##get length
##-------------
coreGeneNames_sterm_ldel <- read_delim("~/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/coreGeneNames_rmk202_bothLdelSterm.vcf", "\t", escape_double = FALSE, col_names = c("contig","method","annotation","start","end"), trim_ws = TRUE)
# table(coreGeneNames_sterm_ldel$contig)
coreGeneNames_sterm <- coreGeneNames_sterm_ldel[coreGeneNames_sterm_ldel$contig=="L_delbrueckii_RMK202",]
# table(coreGeneNames_sterm$contig)
CoreGeneSize_sterm <- sum(coreGeneNames_sterm$end-coreGeneNames_sterm$start)



##--------------
##plot
##-------------
library(stringr)
library(readr)
library(ggplot2)
library(reshape2)

table(sterm_snps_core$core)
sterm_snps_CORE <- sterm_snps_core
# sterm_snps_CORE <- sterm_snps_core[which(sterm_snps_core$core=="core"),]
table(sterm_snps_CORE$core)



##------------
##prep
##------------
colnames(sterm_snps_core)
colnames(sterm_snps_core)[3:11]
names_normal <- colnames(sterm_snps_core)[3:11]
# count <- 1 ## triangle above
count <- 9 ; names <- rev(names_normal) ## triangle bellow

sterm_curated_rev_names <- data.frame()
# for (i in 1:length(names)) {  ## triangle above
for (i in rev(1:length(names))) { ## triangle bellow

   rowName <- names[i]
    print(rowName)
    
  for(a in rev(1:length(names))){
    
    colNames <- names[a]
    print(colNames)
    
    


    countSNPs <- sum(sterm_snps_CORE[,rowName]!=sterm_snps_CORE[,colNames])
    ANI_core <- 100*((CoreGeneSize_sterm-countSNPs)/CoreGeneSize_sterm)
    tmp <- data.frame("genome1"=rowName,"genome2"=colNames,"SNPs"=countSNPs,"ANI"=ANI_core)
    sterm_curated_rev_names <- rbind(sterm_curated_rev_names,tmp)
    

  }

# count <- count +1  ## triangle above
count <- count -1  ## triangle bellow

}
sterm_curated_rev_names

##------------
##make dendro of all snps sites
##------------##------------
##make dendro of all snps sites


library(ggdendro)
# 
sterm_matrix <- as.matrix(t(sterm_snps_CORE[, -c(1:2)]))
rownames(sterm_matrix) <- names_normal
otter.dendro <- as.dendrogram(hclust(d = dist(x = sterm_matrix)))

# 
# # Create dendro
dendro.plot <- ggdendrogram(data = otter.dendro, rotate = FALSE)
# # Preview the plot
print(dendro.plot)
# 
# labels(otter.dendro)
# 
svg("~/Desktop/Projects/2018_culturomics/fastani/Ldel_SNPs_DENDROGRAM.svg",width=2.5,height=3)
print(dendro.plot)
dev.off()

sterm_curated_rev_names
##reorder according to dendro
levels(sterm_curated_rev_names$genome1)
sterm_curated_rev_names$genome1 = factor(sterm_curated_rev_names$genome1, levels=labels(otter.dendro))
levels(sterm_curated_rev_names$genome1)
levels(sterm_curated_rev_names$genome2)
sterm_curated_rev_names$genome2 = factor(sterm_curated_rev_names$genome2, levels=labels(otter.dendro))
levels(sterm_curated_rev_names$genome2)

# creat Heatmap
# sterm_curated$X3 <- round(sterm_curated$X3,digits = 2)
##-----------
##change names
##-----------
sterm_curated_rev_names$genome1 <- revalue(sterm_curated_rev_names$genome1, c("Ldel_rmk202"="MAG RMK202"))
sterm_curated_rev_names$genome2 <- revalue(sterm_curated_rev_names$genome2, c("Ldel_rmk202"="MAG RMK202"))

# ##------------
# ##for SNP legend
# ##------------

ggheatmap <- ggplot(sterm_curated_rev_names, aes(genome1, genome2, fill = SNPs))+
 geom_tile(color = "white",size=1.1)+
 #scale_fill_gradient2(low = "blue", high = "red", mid = "white",
  # midpoint = 99.5, limit = c(98.9,100), space = "Lab",
  # name="ANI") +
  scale_fill_distiller(name = "SNPs", palette = "Blues", direction = -1)+
  # scale_fill_distiller(name = "ANI", palette = "Reds", direction = 1)+
  theme_minimal()+ # minimal theme
 theme(axis.text.y = element_text(size = 12),
      axis.text.x = element_text(angle = 45, vjust = 1,
    size = 12, hjust = 1))+
 coord_fixed()
# Print the heatmap
print(ggheatmap)

##------------
##SNPs
##------------


plotFinal_sterm <- ggheatmap + 
geom_text(aes(genome1, genome2, label = SNPs), color = "black", size = 2.5) +
    labs(title=paste0("# core genes :",nrow(coreGeneNames_sterm)," (",CoreGeneSize_sterm,"bp)"))+
  scale_y_discrete(position = 'right')+
theme(
  axis.title.x = element_blank(),
  axis.title.y = element_blank(),
  # axis.text.y.left = element_text(angle = 75, hjust = 1,size=9),
  panel.grid.major = element_blank(),
  # scale_y_date(position = "right"),
  panel.border = element_blank(),
  panel.background = element_blank(),
  axis.ticks = element_blank(),
  # legend.justification = c(1, 0),
  legend.position = "right",
  legend.text = element_text(angle = 0),
  legend.direction = "vertical")+
  guides(fill = guide_colorbar(barwidth = 1, barheight = 7,
                title.position = "top", title.hjust = 0.5))
plotFinal_sterm
##save heatmap
svg("~/Desktop/Projects/2018_culturomics/fastani/Ldel_SNPs_core_absolute_final.svg",width=5.5,height=4)
plotFinal_sterm#+theme(legend.position = "none")
dev.off()


##------------
##ANI 2
##------------


library(viridis)
# creat Heatmap
sterm_curated_rev_names$ANI_2 <- round(sterm_curated_rev_names$ANI,digits = 2)
min(sterm_curated_rev_names$ANI)
ggheatmap <- ggplot(sterm_curated_rev_names, aes(genome1, genome2, fill = ANI))+
 geom_tile(color = "white",size=1.1)+
 # scale_fill_gradient2(low = "red", high = "red4",
 # midpoint = 99.995, limit = c(99.99,100), space = "Lab",
 # name="ANI") +
  # scale_fill_viridis(discrete = FALSE, option = "B",direction = -1)+
  scale_fill_distiller(name = "ANI", palette = "Reds", direction = 1)+
  theme_minimal()+ # minimal theme
 theme(axis.text.y = element_text(size = 12),
      axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1))+
 coord_fixed()
# Print the heatmap
print(ggheatmap)


plotFinal_sterm <- ggheatmap + 
# geom_text(aes(genome1, genome2, label = ANI_2), color = "black", size = 2.5) + ##this version is with the ANI value in the heatmap
geom_text(aes(genome1, genome2, label = SNPs), color = "black", size = 2.5) + #this version is with the SNP counts in the heatmap
  scale_y_discrete(position = 'right')+
  labs(title=paste0("# core genes :",nrow(coreGeneNames_sterm)," (",CoreGeneSize_sterm,"bp)"))+
theme(
  axis.title.x = element_blank(),
  axis.title.y = element_blank(),
  # axis.text = element_blank(),
  panel.grid.major = element_blank(),
  panel.border = element_blank(),
  panel.background = element_blank(),
  axis.ticks = element_blank(),
  # legend.justification = c(1, 0),
  legend.position = "right",
  legend.text = element_text(angle = 0),
  legend.direction = "vertical")+
  guides(fill = guide_colorbar(barwidth = 1, barheight = 7,
                title.position = "top", title.hjust = 0.5))
plotFinal_sterm
##save heatmap
# svg("~/Desktop/Projects/2018_culturomics/fastani/Ldel_SNPs_core_ANI.svg",width=5.5,height=4)
svg("~/Desktop/presenations/my_presentations/2020_groupmeeting/figures/Ldel_SNPs_core_ANI.svg",width=5.5,height=4)
plotFinal_sterm#+theme(legend.position = "none")
dev.off()

```




##SNPeff analysis supplementary


rAnalysis

```{r}


##==========snpEff !!!!do after snpEff chunk!!!
#Ldel
snpEff_Ldel <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/effects/variantCallsfreebayes_all_Lactobacillus_delbrueckii_RMK202-withOld_merged_forR_unmodified.csv", "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("count","effect","impact"));snpEff_Ldel$species <- "L.delbrueckii"

#Sterm
snpEff_Sterm <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/effects/variantCallsfreebayes_all_Streptococcus_thermophilus_RMK202-withOld_merged_forR_unmodified.csv", "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("count","effect","impact"));snpEff_Sterm$species <- "S. thermophilus"

snpEff <- rbind(snpEff_Ldel,snpEff_Sterm)

snpEff$location <- ifelse(snpEff$impact=="MODIFIER","intragenic","genic")

# ##seperate
# snpEff_Sterm <- snpEff[which(snpEff$species=="Sterm"),]
# snpEff_Ldel <- snpEff[which(snpEff$species=="Ldel"),]

##==========plot abundance 



#--- genomic or intergenomic

table(snpEff$sample)

# aggregate(startingGff$RawReads, list(startingGff$name), mean)

interPlot <- ggplot(data=snpEff, aes(x=location, weight=count,fill=location)) +
  facet_grid(species~., scales="free")+
  geom_bar()+
  # geom_bar(stat="identity", position = "dodge")+
labs(y="count",
     x="genomic location")+
scale_fill_manual(values=c("black","grey"))+ 
  theme_classic()+
theme(#axis.text.y=element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "none")


interPlot
 svg("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/snpEff/genomicLocation_snps.svg",width=3,height=5)
interPlot
 dev.off()
 

source("/home/vincent/Desktop/ScriptRepository/R_functions-master/R_functions/g_legend.R")
library(sparklyr)
##==========change order

##-Sterm

snpEff_new <- snpEff


factor(snpEff_new$impact)
snpEff_new$impact <- factor(snpEff_new$impact, levels=(c("MODIFIER","LOW","MODERATE","HIGH")))
factor(snpEff_new$impact)



snpEff_new[match(levels(factor(snpEff_new$effect)),snpEff_new$effect),c(2,3)][match(c("MODIFIER","LOW","MODERATE","HIGH"),snpEff_new[match(levels(factor(snpEff_new$effect)),snpEff_new$effect),c(2,3)]$impact),]

orderEffect <- snpEff_new[match(levels(factor(snpEff_new$effect)),snpEff_new$effect),c(2,3)][( snpEff_new[match(levels(factor(snpEff_new$effect)),snpEff_new$effect),c(2,3)]$impact == "MODIFIER"),]
orderEffect <- rbind(orderEffect,snpEff_new[match(levels(factor(snpEff_new$effect)),snpEff_new$effect),c(2,3)][( snpEff_new[match(levels(factor(snpEff_new$effect)),snpEff_new$effect),c(2,3)]$impact == "LOW"),])
orderEffect <- rbind(orderEffect,snpEff_new[match(levels(factor(snpEff_new$effect)),snpEff_new$effect),c(2,3)][( snpEff_new[match(levels(factor(snpEff_new$effect)),snpEff_new$effect),c(2,3)]$impact == "MODERATE"),])
orderEffect <- rbind(orderEffect,snpEff_new[match(levels(factor(snpEff_new$effect)),snpEff_new$effect),c(2,3)][( snpEff_new[match(levels(factor(snpEff_new$effect)),snpEff_new$effect),c(2,3)]$impact == "HIGH"),])

orderEffect_true <- droplevels(factor(orderEffect$effect))

factor(snpEff_new$effect)
snpEff_new$effect <- factor(snpEff_new$effect, levels=(orderEffect_true))
factor(snpEff_new$effect)


##==========plot insertion sequence per mutation type
##---Sterm


all_impact <- ggplot(data=snpEff_new, aes(x=effect, y=count,fill=impact)) +
  facet_grid(species~., scales="free")+
  geom_bar(stat="identity", position = "dodge")+
# ggtitle(paste0("SNV sites in Sterm n= " ,(sum(snpEff_Sterm$count))," ( " , round(100*sum(snpEff_Sterm$count)/StermContigSize,digits=2), " % )")) +
xlab("effect") + 
ylab("variant sites") +
scale_fill_manual(values=c("grey","gold1","orange","red"))+   #wes_palette(n=3, name="Moonrise3")[1:3]))+
# scale_fill_manual(values=c("red","gold1","orange","grey"))+   #wes_palette(n=3, name="Moonrise3")[1:3]))+
  theme_classic()+
  theme(legend.position = "none",
        plot.margin=unit(c(0,0,0,2.5),"cm"),
        axis.text.x=element_text(angle=45, hjust=1,size=9))

all_impact

 svg("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/snpEff/inter_genomic_snp_impact.svg",width=5,height=5.5)
all_impact
 dev.off()
 

all_impact <- ggplot(data=snpEff_new, aes(x=impact,weight=count,fill=impact)) +
  facet_grid(species~., scales="free")+
  geom_bar()+
# ggtitle(paste0("SNV sites in Sterm n= " ,(sum(snpEff_Sterm$count))," ( " , round(100*sum(snpEff_Sterm$count)/StermContigSize,digits=2), " % )")) +
xlab("impact") + 
ylab("variant sites") +
scale_fill_manual(values=c("grey","gold1","orange","red"))+   #wes_palette(n=3, name="Moonrise3")[1:3]))+
# scale_fill_manual(values=c("red","gold1","orange","grey"))+   #wes_palette(n=3, name="Moonrise3")[1:3]))+
  theme_classic()+
  theme(legend.position = "none",
        plot.margin=unit(c(0,0,0,2.5),"cm"),
        axis.text.x=element_text(angle=45, hjust=1,size=9))

all_impact

 svg("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/snpEff/inter_genomic_snp_impact_02.svg",width=5,height=5.5)
all_impact
 dev.off()
 
###---------------------------high impact
 
snpEff_new_high <- snpEff_new[which(snpEff_new$impact=="HIGH"),]
 

high_impact <- ggplot(data=snpEff_new_high, aes(x=effect, y=count,fill=impact)) +
  facet_grid(species~., scales="free")+
  geom_bar(stat="identity", position = "dodge")+
# ggtitle(paste0("SNV sites in Sterm n= " ,(sum(snpEff_Sterm$count))," ( " , round(100*sum(snpEff_Sterm$count)/StermContigSize,digits=2), " % )")) +
xlab("effect") + 
ylab("variant sites") +
scale_fill_manual(values=c("red"))+   #wes_palette(n=3, name="Moonrise3")[1:3]))+
    theme_classic()+
  theme(legend.position = "none",
        plot.margin=unit(c(0,0,0,2.5),"cm"),
        axis.text.x=element_text(angle=45, hjust=1,size=10))


high_impact

 svg("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/snpEff/inter_genomic_snp_impact_high.svg",width=5,height=5.5)
high_impact
 dev.off()

```


eggnog analysis with (COG)[http://clovr.org/docs/clusters-of-orthologous-groups-cogs/]categories
```{r}
library(readr)

##==========data import data prep
COG_functional_categories <- read_delim("~/Desktop/ScriptRepository/COG_functional_categories.csv",  ",", escape_double = FALSE, col_names = c("short","long","divided"),trim_ws = TRUE)

cogFunctionsHighImapactSNVs <- read_delim("~/Desktop/Projects/2018_culturomics/MetaSNV/cogFunctionsHighImapactSNVs.csv", "\t", escape_double = FALSE, trim_ws = TRUE)

cogFunctionsHighImapactSNVs$long <- NA

for (i in 1:nrow(cogFunctionsHighImapactSNVs)) {
  cogFunctionsHighImapactSNVs[i,"long"] <- COG_functional_categories[which(COG_functional_categories$short==as.character(cogFunctionsHighImapactSNVs[i,"function"])),"long"]
  
}

library(ggplot2)


source("/home/vincent/Desktop/ScriptRepository/R_functions-master/R_functions/g_legend.R")

##==========change order

cogFunctionsHighImapactSNVs_Sterm <- cogFunctionsHighImapactSNVs[which(cogFunctionsHighImapactSNVs$species=="Sterm"),]
cogFunctionsHighImapactSNVs_Ldel <- cogFunctionsHighImapactSNVs[which(cogFunctionsHighImapactSNVs$species=="Ldel"),]

colnames(cogFunctionsHighImapactSNVs_Sterm)[2] <- "functionss"
colnames(cogFunctionsHighImapactSNVs_Ldel)[2] <- "functionss"


##-Sterm

cogFunctionsHighImapactSNVs_Sterm_new <- cogFunctionsHighImapactSNVs_Sterm

factor(cogFunctionsHighImapactSNVs_Sterm_new$'functionss')
cogFunctionsHighImapactSNVs_Sterm_new$'functionss' <- factor(cogFunctionsHighImapactSNVs_Sterm_new$'functionss', levels=(c("L","S","M","V","T","K","F","E","G","P","O","C","H","unknown")))
factor(cogFunctionsHighImapactSNVs_Sterm_new$'functionss')

cogFunctionsHighImapactSNVs_Sterm_new$'long' <- gsub(" and ","\n",cogFunctionsHighImapactSNVs_Sterm_new$'long') %>% gsub(", ","\n",.) %>% gsub("/e","\ne",.)

factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')

cogFunctionsHighImapactSNVs_Sterm_new$'long' <- factor(cogFunctionsHighImapactSNVs_Sterm_new$'long', levels=(c(as.character(factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')[8]),as.character(factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')[12]),as.character(factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')[9]),as.character(factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')[14]),as.character(factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')[13]),as.character(factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')[7]),as.character(factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')[4]),as.character(factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')[3]),as.character(factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')[5]),as.character(factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')[11]),as.character(factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')[10]),as.character(factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')[2]),as.character(factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')[6]),as.character(factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')[1]))))

factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')

##-Ldel

cogFunctionsHighImapactSNVs_Ldel_new <- cogFunctionsHighImapactSNVs_Ldel

factor(cogFunctionsHighImapactSNVs_Ldel_new$'functionss')
cogFunctionsHighImapactSNVs_Ldel_new$'functionss' <- factor(cogFunctionsHighImapactSNVs_Ldel_new$'functionss', levels=(c("L","E","K","I")))
factor(cogFunctionsHighImapactSNVs_Ldel_new$'functionss')

cogFunctionsHighImapactSNVs_Ldel_new$'long' <- gsub(" and ","\n",cogFunctionsHighImapactSNVs_Ldel_new$'long')

factor(cogFunctionsHighImapactSNVs_Ldel_new$'long')

cogFunctionsHighImapactSNVs_Ldel_new$'long' <- factor(cogFunctionsHighImapactSNVs_Ldel_new$'long', levels=(c(as.character(factor(cogFunctionsHighImapactSNVs_Ldel_new$'long')[4]),as.character(factor(cogFunctionsHighImapactSNVs_Ldel_new$'long')[1]),as.character(factor(cogFunctionsHighImapactSNVs_Ldel_new$'long')[3]),as.character(factor(cogFunctionsHighImapactSNVs_Ldel_new$'long')[2]))))

factor(cogFunctionsHighImapactSNVs_Ldel_new$'long')



##==========plot insertion sequence per mutation type
##---Sterm

Sterm_COG <- ggplot(data=cogFunctionsHighImapactSNVs_Sterm_new, aes(x=long, y=count,fill=long)) +
  #facet_grid(~species)+
  geom_bar(stat="identity")+
ggtitle(paste0("COG categories of high impact SNV in Sterm n= " ,(sum(cogFunctionsHighImapactSNVs_Sterm_new$count)))) +
xlab("genomic location") + 
ylab("variant sites") +
#scale_fill_manual(values=c("red","gold1","orange","grey"))+   #wes_palette(n=3, name="Moonrise3")[1:3]))+
theme(#axis.text.y=element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank(),
        #axis.text.x=element_blank(),
        axis.title.x = element_blank(),
        #axis.text.x=element_blank(),
        legend.position = "none",
        axis.line = element_line(colour = "black"),
        panel.background = element_blank(),
        panel.grid.major.x = element_blank() ,
        axis.text.x=element_text(angle=45, hjust=1),
        plot.margin=unit(c(0,0,0,0), "cm"),
        panel.border = element_rect(colour = "black", fill=NA, size=1),
        #panel.background = element_rect(fill = "white", colour = "grey50"),
        panel.grid.major =element_line(colour = "grey90"),panel.grid.major.y = element_blank())

#---Ldel


Ldel_COG <- ggplot(data=cogFunctionsHighImapactSNVs_Ldel_new, aes(x=long, y=count,fill=long)) +
  #facet_grid(~species)+
  geom_bar(stat="identity")+
ggtitle(paste0("Ldel n= " ,(sum(cogFunctionsHighImapactSNVs_Ldel_new$count)))) +
xlab("genomic location") + 
ylab("variant sites") +
scale_x_discrete(labels= as.character(cogFunctionsHighImapactSNVs_Ldel_new$long))+
#scale_fill_manual(values=c("red","gold1","orange","grey"))+   #wes_palette(n=3, name="Moonrise3")[1:3]))+
theme(#axis.text.y=element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank(),
        #axis.text.x=element_blank(),
        axis.title.x = element_blank(),
        #axis.text.x=element_blank(),
        legend.position = "none",
        axis.line = element_line(colour = "black"),
        panel.background = element_blank(),
        panel.grid.major.x = element_blank() ,
        axis.text.x=element_text(angle=45, hjust=1),
        plot.margin=unit(c(0,0,0,0), "cm"),
        panel.border = element_rect(colour = "black", fill=NA, size=1),
        #panel.background = element_rect(fill = "white", colour = "grey50"),
        panel.grid.major =element_line(colour = "grey90"),panel.grid.major.y = element_blank())

Sterm_COG
Ldel_COG 

 grid.arrange(Sterm_COG, Ldel_COG, 
         ncol=2, nrow=1, widths=c(4,1.5))

svg("~/Desktop/Projects/2018_culturomics/MetaSNV/genomicLocation_short_Highimpact_COG.svg",width=10,height=4.5)

 grid.arrange(Sterm_COG, Ldel_COG, 
         ncol=2, nrow=1, widths=c(4,1.5))
 dev.off()
 



```



####PTR calculation



own pipeline
```{bash}


##=============
##bowtie2 mapping
##=============

num=1
for NanoSample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt )
  do
  echo ${num}"/37  :" ${names}
  num=$((num+1))
  echo "======================================"
bowtie2-build /archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${NanoSample}/08_eightFreebayes/freebayesOuput/_eightFreebayes.fasta /archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${NanoSample}/08_eightFreebayes/freebayesOuput/_eightFreebayes


done
threads=37

num=1
for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name.txt )
  do
  echo ${num}"/37  :" ${names}
  num=$((num+1))
  #rm -r /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/${names}

  mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_bowtie2/${names}/bwaMapping2DB/
  
  #gunzip /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R1_val_1.fq.gz
  #gunzip /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R2_val_2.fq.gz
  
  sampleShort=$(echo ${names}| awk -F "_" 'BEGIN{OFS="_"}{print $1,$2}')
  
   NanoSample=$(grep "$sampleShort" /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt) 
  
  #/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${NanoSample}/assembly/assembly.fasta
  

  
bowtie2 -p ${threads} -x /archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${NanoSample}/08_eightFreebayes/freebayesOuput/_eightFreebayes  -1 /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R1_val_1.fq -2 /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R2_val_2.fq | samtools sort -@${threads} -O SAM -o /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_bowtie2/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.sam - 

#samtools index /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam

 # gzip /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R1_val_1.fq
#  gzip /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R2_val_2.fq
  
  
  done 

##================
#irep
##================



threads=37

num=1
for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name.txt | grep "th_K1_")
  do
  echo ${num}"/37  :" ${names}
  num=$((num+1))

  sampleShort=$(echo ${names}| awk -F "_" 'BEGIN{OFS="_"}{print $1,$2}')
  
   NanoSample=$(grep "$sampleShort" /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt) 
  
  mkdir -p /archiv/Projects/2019_Nano_Meta/04_polishing/02_PTR_irep/all_th_K1/
  

iRep -f /archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/th_K1_8h/08_eightFreebayes/freebayesOuput/genomes/contig_16.fasta  -s /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_bowtie2/${names}/bwaMapping2DB/*.sam -o /archiv/Projects/2019_Nano_Meta/04_polishing/02_PTR_irep/all_th_K1/${names}_PTR.tsv 

done

```



(irep)[https://github.com/christophertbrown/iRep] of all indiviual and then put into excel manually

```{r}

library(readr)
library(tidyverse)
library(dplyr)
library(tidyr)

irep_K1_PTR <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/PTR_irep/irep_K1_PTR.csv", "\t", escape_double = FALSE, trim_ws = TRUE)
irep_K1_PTR <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/PTR_irep/irep_K1_PTR.csv", ",", escape_double = FALSE, trim_ws = TRUE)



  irep_K1_PTR$day <- str_split_fixed(irep_K1_PTR$sample, "_", 3)[,1]
  irep_K1_PTR$kessel <- str_split_fixed(irep_K1_PTR$sample, "_", 3)[,2]
  irep_K1_PTR$time <- str_split_fixed(irep_K1_PTR$sample, "_", 3)[,3]
  irep_K1_PTR$treatment <- paste0(irep_K1_PTR$day,"_",irep_K1_PTR$kessel)
 
#  di_K1_long$treatment_f = factor(di_K1_long$treatment, levels=c('di_K1','th_K1','di_K2','th_K2'))
  irep_K1_PTR$time_f = factor(irep_K1_PTR$time, levels=c("0h","2h","4h","6h","8h","10h","12h","24h"))

  colours <-  c("#10B552","#EB4D4D") #c("#FAA0A0","#FCC2C2","#EB4D4D","#D42828") #c("#FAA0A0","#FAA0A0","#FAA0A0","#FAA0A0")
# Ldel_colours <- c("#6CF5A3","#B3FCD1","#10B552","#088A3C") 
  irep_K1_PTR <- irep_K1_PTR[which(irep_K1_PTR$day=="di"),]
  
  
    irep_K1_PTR$time_f_02 <- as.character(gsub("h","",irep_K1_PTR_poster$time_f) )
  irep_K1_PTR$time_f_02 = as.numeric(irep_K1_PTR_poster$time_f_02, levels=c("0","2","4","6","8","10","12","24"))

  
 ptr_01 <-  ggplot(irep_K1_PTR,aes(x=time_f,y=PTR,size=coverage,group=genome,color=genome,fill=genome))+geom_point()+geom_line()+facet_grid(~treatment)+
    labs("",
         x="time",
         y="Peak-to-trough")+
    theme_classic()+
    #scale_color_viridis(discrete=TRUE)+
    scale_fill_manual(values=colours)+
    scale_color_manual(values=colours)

  # scale_fill_viridis(discrete=TRUE)+
    # theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
      #axis.text.x = element_blank(),
          # legend.position="right"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          # )
  ptr_01
svg("~/Desktop/presenations/my_presentations/20190708_groupmeeting/figures/PTR.svg",width=6,height=3.2)
ptr_01
 dev.off()

 
 ##absolute count model
 
 total_samples
 
unique(total_samples$group)

 
 
 Ldel_abs <- total_samples[which(total_samples$group=="Lactobacillus_delbrueckii_group_subgroup_2"),] %>% filter(., kessel=="K1") 
 sterm_abs <- total_samples[which(total_samples$group=="Streptococcus_thermophilus_group_subgroup_4"),] %>% filter(., kessel=="K1") 
 
 Ldel_abs$genome <- "L. delbrueckii"
 sterm_abs$genome <- "S. thermophilus"

   sterm_abs$time_f_02 <- as.character(gsub("h","",sterm_abs$time_f) )
   Ldel_abs$time_f_02 <- as.character(gsub("h","",Ldel_abs$time_f) )

 

   sterm_abs_new <- subset(sterm_abs, select=c("name","genome","day","kessel","time_f","time_f_02","treatment","absCount"))
  Ldel_abs_new <- subset(Ldel_abs, select=c("name","genome","day","kessel","time_f","time_f_02","treatment","absCount"))

  all_abs_new <- rbind(Ldel_abs_new,sterm_abs_new)
  write.csv(all_abs_new,"~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/PTR_irep/PTR_fromR.csv")

  ##change in R
 
 
 # irep_K1_PTR_new <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/PTR_irep/PTR_fromR.csv", ",", escape_double = FALSE, trim_ws = TRUE)

 # ggplot(irep_K1_PTR_new,aes(x=absCount,y=expecCount))+geom_point()
 
 
  ##-------plot for poster
  irep_K1_PTR_poster <- irep_K1_PTR[which(irep_K1_PTR$day=="di"),]
  
    irep_K1_PTR_poster$time_f_02 <- as.character(gsub("h","",irep_K1_PTR_poster$time_f) )
  irep_K1_PTR_poster$time_f_02 = as.numeric(irep_K1_PTR_poster$time_f_02, levels=c("0","2","4","6","8","10","12","24"))

 ptr_01 <-  ggplot(irep_K1_PTR_poster,aes(x=time_f_02,y=PTR,size=coverage,group=genome,color=genome,fill=genome))+geom_point()+geom_line()+
    labs("",
         x="time [h]",
         y="Peak-to-trough")+
   scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    theme_classic()+
    #scale_color_viridis(discrete=TRUE)+
    scale_fill_manual(values=colours)+
    scale_color_manual(values=colours)

  # scale_fill_viridis(discrete=TRUE)+
    # theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
      #axis.text.x = element_blank(),
          # legend.position="right"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          # )
  ptr_01
svg("~/Desktop/posters/2019_ISCB/figures/PTR_poster.svg",width=9,height=4.7)
ptr_01
 dev.off()
  
```


##PTR with segmented



```{r}
library(readr)
library(segmented)
library(plotly)
library(ggplot2)
library(gridExtra)
library(cowplot)

coverage <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/1AB/mapping2SCOG_reference/coverage_2_referenceGenome_allSamples.bed","\t", escape_double = FALSE, col_names = c("contig","annotation","gene","start","end","unknown1","orientation","unknown2","id","count","length_unmapped","geneLength","percent","name"), trim_ws = TRUE)


  coverage <- read_delim("/home/vincent/Desktop/Projects/2019_Pilotplan/09_annoation/all_Bacteria_allSamples.txt","\t", escape_double = FALSE, col_names = c("chr","type","sort","start","end","unknown2","orientation","unknown","description","count","length_mapped","geneLength","covered","sample"),trim_ws = TRUE)


coverage$coverage <- (coverage$count*600)/coverage$geneLength

#coverage$species <- ifelse(coverage$contig=="contig_1","S. thermophilus","L. delbrueckii")
coverage <- coverage[which(coverage$end-coverage$start>600),]


##-------------Ldel


psterm  <- list()
pldel  <- list()
namesss = NULL
PTRss= NULL
#coverage_sample <- list
# p <- list()
for (i in  1:length(unique(coverage$sample))){
  
  
  sample <- unique(coverage$sample)[i]
  coverage_sample  <- coverage[which(coverage$sample==sample),]
  coverage_sample_Ldel <- coverage_sample[which(coverage_sample$chr=="L_delbrueckii_RMK202"),]
  mean_ldel <- max(coverage_sample_Ldel$end)/2

  y <- as.double(coverage_sample_Ldel$coverage)
  x <- as.double(coverage_sample_Ldel$start)

  lin.mod <- lm(y~x)
summary( lin.mod)
  segmented.mod <- segmented(lin.mod, seg.Z = ~x, psi=mean_ldel)
  midpoint <- segmented.mod$psi[2]
  maxpoint <- max(coverage_sample_Ldel$end)
  summary(segmented.mod)
  slope_01 <- segmented.mod$coefficients[2]
  slope_02 <- segmented.mod$coefficients[3]+segmented.mod$coefficients[2]
  intercept_01 <- segmented.mod$coefficients[1]
  break1 <- segmented.mod$psi[[3]]
  intercept_02 <-  intercept_01 + (slope_01* midpoint) - (slope_02* midpoint) 
  
  minValue_01 <- intercept_01 + (slope_01* midpoint)
  minValue_02 <- intercept_02 + (slope_02* midpoint)
  
  maxValue_01 <- intercept_01
  maxValue_02 <- intercept_02 + (slope_02* maxpoint)
  
  maxValue_total <- max(maxValue_01,maxValue_02)
  minValue_total <- min(minValue_01,minValue_02)
  PTR <- maxValue_total/ minValue_total
  
    namesss = append(namesss, as.character(sample))
  PTRss = append(PTRss, PTR)
  
  fit <- numeric(length(x)) * NA
  fit[complete.cases(rowSums(cbind(y, x)))] <- broken.line(segmented.mod)$fit
  data1 <- data.frame(x = x, y = y, fit = fit,geneID=coverage_sample_Ldel$description,geneStart=coverage_sample_Ldel$start,geneSize=coverage_sample_Ldel$end-coverage_sample_Ldel$start,Species=coverage_sample_Ldel$chr,Name=coverage_sample_Ldel$sample)

pldel[[i]] <- ggplot(data1, aes(x=x,y=y,label=geneID,label2=geneStart,label3=Name))+
  geom_point(aes(size=geneSize,color=Species))+
  annotate(geom="text", x=500000, y=max(data1$y)-(0.2*max(data1$y)), label=paste0(sample," (PTR= ",round(PTR,digits = 2),")"),color="black")+
  geom_line(aes(x = x, y = fit),size=2, color = 'grey')+
  labs(x="genome location",y="coverage")+
  scale_color_manual(values="#10B552")+
  theme_bw()+
  theme(legend.position="none")


}
dldel = data.frame(namesss,PTRss,species="Lactobacillus delbrueckii")


# htmlwidgets::saveWidget(pldel, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/PTR_ldel.html")

# pdf("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/1AB/mapping2SCOG_reference/mapping2ONTreference_ldel.pdf",width=16,height=32,paper='special') 
svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/PTR_ldel.svg",width=20,height=4)
do.call(plot_grid, c(pldel, align = 'h', ncol=5))
dev.off()


##-------------Sterm
# table(coverage_sample$chr)
psterm  <- list()
pldel  <- list()
namesss = NULL
PTRss= NULL
# PTR_total <- data.frame(c("name","PTR"))
#coverage_sample <- list
# p <- list()
for (i in  1:length(unique(coverage$sample))){
  
  sample <- unique(coverage$sample)[i]
  coverage_sample  <- coverage[which(coverage$sample==sample),]
  coverage_sample_Ldel <- coverage_sample[which(coverage_sample$chr=="S_thermophilus_RMK202"),]
  mean_ldel <- max(coverage_sample_Ldel$end)/2

  y <- as.double(coverage_sample_Ldel$coverage)
  x <- as.double(coverage_sample_Ldel$start)

  lin.mod <- lm(y~x)
summary( lin.mod)
  segmented.mod <- segmented(lin.mod, seg.Z = ~x, psi=mean_ldel)
  midpoint <- segmented.mod$psi[2]
  maxpoint <- max(coverage_sample_Ldel$end)
  summary(segmented.mod)
  slope_01 <- segmented.mod$coefficients[2]
  slope_02 <- segmented.mod$coefficients[3]+segmented.mod$coefficients[2]
  intercept_01 <- segmented.mod$coefficients[1]
  break1 <- segmented.mod$psi[[3]]
  intercept_02 <-  intercept_01 + (slope_01* midpoint) - (slope_02* midpoint) 
  
  minValue_01 <- intercept_01 + (slope_01* midpoint)
  minValue_02 <- intercept_02 + (slope_02* midpoint)
  
  maxValue_01 <- intercept_01
  maxValue_02 <- intercept_02 + (slope_02* maxpoint)
  
  maxValue_total <- max(maxValue_01,maxValue_02)
  minValue_total <- min(minValue_01,minValue_02)
  PTR <- maxValue_total/ minValue_total
  
  # PTR[[i,]] <- c(as.character(sample),PTR)
  namesss = append(namesss, as.character(sample))
  PTRss = append(PTRss, PTR)

  fit <- numeric(length(x)) * NA
  fit[complete.cases(rowSums(cbind(y, x)))] <- broken.line(segmented.mod)$fit
  data1 <- data.frame(x = x, y = y, fit = fit,geneID=coverage_sample_Ldel$description,geneStart=coverage_sample_Ldel$start,geneSize=coverage_sample_Ldel$end-coverage_sample_Ldel$start,Species=coverage_sample_Ldel$chr,Name=coverage_sample_Ldel$sample)

pldel[[i]] <- ggplot(data1, aes(x=x,y=y,label=geneID,label2=geneStart,label3=Name))+
  geom_point(aes(size=geneSize,color=Species))+
  annotate(geom="text", x=500000, y=max(data1$y)-(0.2*max(data1$y)), label=paste0(sample," (PTR= ",round(PTR,digits = 2),")"),color="black")+
  geom_line(aes(x = x, y = fit),size=2, color = 'grey')+
  labs(x="genome location",y="coverage")+
  scale_color_manual(values="#10B552")+
  theme_bw()+
  theme(legend.position="none")


}

d = data.frame(namesss,PTRss,species="Streptococcus thermophilus")
PTR_all <- rbind(dldel,d)

svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/PTR_sterm.svg",width=20,height=4)
do.call(plot_grid, c(pldel, align = 'h', ncol=5))
dev.off()


##--------------PTR plot

PTR_all$namesss = factor(PTR_all$namesss, levels=c("Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202"))


ggplot(PTR_all,aes(x=namesss,y=PTRss,))


all_colours <- rev(c("#EB4D4D","#FAA0A0","#10B552","#36E37B","#6CF5A3","#C5F6FA","#99E9F2") ) 


 pPTR <- ggplot(PTR_all,aes(x=namesss,y=PTRss,group=species,fill=species,color=species))+ geom_point(size=3)+ geom_line()+ #stat_summary(fun.y=sum, geom="line")+
   labs("",
         x="",
         y="Peak-to-trough ratio",
          title="")+
   scale_color_manual(values=c("#10B552","#EB4D4D"))+
  theme_classic()+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
           legend.position="none"
       #egend.justification=c(1,1), legend.position=c(0.9,0.9),
       #legend.title = element_blank()
       )
  pPTR
  
svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/06_PTR_relAbundance.svg",width=3.5,height=8)
grid.arrange(PrelAbundance,PrelAbundance+theme(legend.position = "none"),pPTR,ncol=1, heights=c(3,3,3))

dev.off()



```


##Mapping colony picking strains to ONT-Ref

In the following we want to map colony picked strains to the ONT-REF.

```{bash}

#rm /archiv/Projects/2019_Nano_Meta/colonyPickingIsolates/bwaMapping2ONT_K1/log/multiqc_logfile.txt

threads=3

for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/isolates_names_changed.txt)
  do
  echo ==========================
   echo ${id}
   echo ==========================
   
   mkdir -p /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${id}/

    
#gunzip -c /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/gz/${id}/${id}_R1_val_1.fq.gz  > \
#  /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${id}/${id}_R1_val_1.fq
# gunzip -c /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/gz/${id}/${id}_R2_val_2.fq.gz > \
#  /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${id}/${id}_R2_val_2.fq

mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${id}/bwaMapping2DB/

#bwa mem -t ${threads}  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/_eightFreebayes.fasta \
#/archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/gz/${id}/${id}_R1_val_1.fq.gz /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${id}/${id}_R2_val_2.fq |\
#samtools sort -@8 -O BAM \
#-o /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${id}/bwaMapping2DB/${id}_rawIllumina_bwamem_sorted.bam - 

#qualimap bamqc -bam /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${id}/bwaMapping2DB/${id}_rawIllumina_bwamem_sorted.bam \
#        -outdir /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${id}/bwaMapping2DB/bamqc --java-mem-size=40G

#mkdir -p /archiv/Projects/2019_Nano_Meta/colonyPickingIsolates/bwaMapping2ONT_K1/log

echo "/archiv/Projects/2019_Nano_Meta/colonyPickingIsolates/bwaMapping2ONT_K1/"${id}"/bamqc/raw_data_qualimapReport/" >> /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/log/multiqc_logfile.txt
   
#samtools index /archiv/Projects/2019_Nano_Meta/colonyPickingIsolates/bwaMapping2ONT_K1/${id}/rawIllumina_${id}_bwamem_Assembly_ONTK1_sorted.bam


##===========SNV calling
mkdir -p /archiv/Projects/2019_Nano_Meta/colonyPickingIsolates/SNV_freebayes/${id}

#freebayes -F 0.5 --min-coverage 4 -p 1 -f \
#      /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/_eightFreebayes.fasta  \
#      /archiv/Projects/2019_Nano_Meta/colonyPickingIsolates/bwaMapping2ONT_K1/${id}/rawIllumina_${id}_bwamem_Assembly_ONTK1_sorted.bam > \
#      /archiv/Projects/2019_Nano_Meta/colonyPickingIsolates/SNV_freebayes/${id}//${id}_variantCallsfreebayes.vcf

##===========assembly of non-assembled


#samtools view -b -f 4 /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${id}/bwaMapping2DB/${id}_rawIllumina_bwamem_sorted.bam > /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${id}/bwaMapping2DB/${id}_rawIllumina_bwamem_sorted_unmapped.bam

#bedtools bamtofastq -i /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${id}/bwaMapping2DB/${id}_rawIllumina_bwamem_sorted_unmapped.bam -fq /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${id}/bwaMapping2DB/${id}_rawIllumina_bwamem_sorted_unmapped.fq
     
rm -r /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${id}/bwaMapping2DB/${id}_rawIllumina_bwamem_sorted_unmapped.bam
mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${id}/spadesAssembly_unmapped/assembly


/home/vincent/miniconda2/pkgs/spades-3.13.0-0/bin/spades.py --plasmid -t 35 -m 100 \
  -s /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${id}/bwaMapping2DB/${id}_rawIllumina_bwamem_sorted_unmapped.fq \
   -o /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${id}/spadesAssembly_unmapped/assembly


##----remove short contigs (<500bp)


awk '/^>/ {printf("\n%s\n",$0);next; } { printf("%s",$0);}  END {printf("\n");}' < /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${id}/spadesAssembly_unmapped/assembly/scaffolds.fasta |sed '/^[[:space:]]*$/d' | awk '/^>/ { getline seq } length(seq) >500 { print $0 "\n" seq }' > /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${id}/spadesAssembly_unmapped/assembly/scaffolds_min500bp.fasta



    done




##MULTIQC
rm -r /home/vincent/Projects/2019_pilotplant/02_mapping2referenceDB/02_againstAllCowMito/MultiQC/
mkdir -p /home/vincent/Projects/2019_pilotplant/02_mapping2referenceDB/02_againstAllCowMito/MultiQC/

multiqc --cl_config "qualimap_config: { total_reads: TRUE }" /archiv/logs/multiQC_config.txt --file-list /home/vincent/Projects/2019_pilotplant/02_mapping2referenceDB/02_againstAllCowMito/log/multiqc_logfile.txt \
  -o /home/vincent/Projects/2019_pilotplant/02_mapping2referenceDB/02_againstAllCowMito/MultiQC/










```


##Mapping old RMK samples

In the following we want to map colony picked strains to the ONT-REF.

```{bash}

##----------
##add old RMK202 samples
##----------


threads=3

for names in $(echo "Konserve_202 Versand_202 RMK202")
  do
  echo ==========================
   echo ${names}
   echo ==========================
   
#   mkdir -p /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${id}/

    
# gunzip -c /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/gz/${id}/${id}_R1_val_1.fq.gz  > \
#  /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${id}/${id}_R1_val_1.fq
# gunzip -c /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/gz/${id}/${id}_R2_val_2.fq.gz > \
#  /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${id}/${id}_R2_val_2.fq
 # rm -r /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}

 mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/
  
 # mkdir -p /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${names}/
  
 # gunzip -c /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/gz/${names}/${names}_R1*.fq.gz > /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${names}/${names}_R1.fq
#  gunzip -c /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/gz/${names}/${names}_R2*.fq.gz > /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${names}/${names}_R2.fq
  
  #/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${NanoSample}/assembly/assembly.fasta
  
#  bwa mem -t ${threads} /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/_eightFreebayes.fasta /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${names}/${names}_R1.fq /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${names}/${names}_R2.fq | samtools sort -@${threads} -O BAM -o /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 

#rm /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

##-----Qualimap

mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB//bamqc

#qualimap bamqc -bam /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB//bamqc --java-mem-size=80G


mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/log
#echo "/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/"${names}"/bwaMapping2DB//bamqc" >> /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/log/multiqc_logfile.txt
   

#samtools index /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam

##===========SNV calling
#mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/SNV_freebayes/

#freebayes -F 0.5 --min-coverage 4 -p 1 -f \
#      /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/_eightFreebayes.fasta  \
#      /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > \
#     /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/SNV_freebayes/${names}_variantCallsfreebayes.vcf

##===========assembly of non-assembled


#samtools view -b -f 4 /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

#bedtools bamtofastq -i /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam -fq /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq

rm /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/spadesAssembly_unmapped/assembly/

/home/vincent/miniconda2/pkgs/spades-3.13.0-0/bin/spades.py --plasmid -t 35 -m 100 \
  -s /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq \
  -o /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/spadesAssembly_unmapped/assembly


##----remove short contigs (<500bp)


awk '/^>/ {printf("\n%s\n",$0);next; } { printf("%s",$0);}  END {printf("\n");}' < /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/spadesAssembly_unmapped/assembly/scaffolds.fasta |sed '/^[[:space:]]*$/d' | awk '/^>/ { getline seq } length(seq) >500 { print $0 "\n" seq }' > /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/spadesAssembly_unmapped/assembly/scaffolds_min500bp.fasta




    done

##----blastn


mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/spadesAssembly_unmapped/blastn
  
  blastn -num_threads ${threads} -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/spadesAssembly_unmapped/assembly/scaffolds_min500bp.fasta \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen" \
  -db /archiv/BLAST/blast_db/20190204/phage_nucleotide/phage_nucleotide_cow \
  -out /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/spadesAssembly_unmapped/blastn/blastAssemblyaginstScaffolds.txt \
  -evalue 0.01 -word_size 64
  



##MULTIQC
rm -r /home/vincent/Projects/2019_pilotplant/02_mapping2referenceDB/02_againstAllCowMito/MultiQC/
mkdir -p /home/vincent/Projects/2019_pilotplant/02_mapping2referenceDB/02_againstAllCowMito/MultiQC/

multiqc --cl_config "qualimap_config: { total_reads: TRUE }" /archiv/logs/multiQC_config.txt --file-list /home/vincent/Projects/2019_pilotplant/02_mapping2referenceDB/02_againstAllCowMito/log/multiqc_logfile.txt \
  -o /home/vincent/Projects/2019_pilotplant/02_mapping2referenceDB/02_againstAllCowMito/MultiQC/










```

####Map to Clostridium botulinum strain MAP 5
I have now seen mulitple times that I mapped to  Clostridium botulinum. 
Here, I want to see what is going on. Therefore I map to the whole genome of  Clostridium botulinum strain MAP 5.

```{bash}

 Clostridium botulinum strain MAP 5


  date=20181226
  #species=Streptococcus_thermophilus

mkdir -p /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/genome/

  curl 'ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt' >  /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/genome/all_refseq.txt

 grep "Clostridium botulinum" -i /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/genome/all_refseq.txt | grep "MAP 5" 

 grep "Clostridium botulinum" -i /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/genome/all_refseq.txt | grep "MAP 5"


grep "Clostridium botulinum" -i /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/genome/all_refseq.txt | grep "MAP 5" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' >    \
    /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/genome/download.txt
    cd /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/
    wget -i /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/genome/download.txt



/home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/mapping
    
##-------------------mapping

threads=37
names=th_K1_12h

num=1
for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name_K1.txt |grep -v "Lyo")
  do
  echo ${num}"/37  :" ${names}
  num=$((num+1))

  mkdir -p /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/mapping/${names}/
  
  bwa mem -t ${threads} /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/GCF_003017265.1_ASM301726v1_genomic.fna /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R1_val_1.fq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R2_val_2.fq | samtools sort -@${threads} -O BAM -o /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/mapping/${names}/${names}_rawIllumina_bwamem_sorted.bam - 

##-----Qualimap

mkdir -p /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/mapping/${names}/bamqc 

qualimap bamqc -bam /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/mapping/${names}/${names}_rawIllumina_bwamem_sorted.bam -outdir /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/mapping/${names}/bamqc --java-mem-size=80G


 mkdir -p /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/log/
echo "/home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/mapping/"${names}"/"${names}"_rawIllumina_bwamem_sorted.bam" >> /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/log/multiqc_logfile.txt
       
done

mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/filesTest
names=th_K1_12h

scp -r vincent@130.223.51.151:/home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/mapping/${names}/bamqc/ /home/vincent/Desktop/Projects/2019_Pilotplan/filesTest



```

#Structural variation detection 
In order to detect structural variation over time and in the isolates see if the single copy orthologous genes have a weird coverage ordering. 

#####Prokka
We first annotate the CDS for all the genomes

```{bash}

##===============
###split MAG into genome
##===============

###get PGAP annotations


scp /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/LD_singu/Lactobacillus_delbrueckii_RMK202_pgap.faa vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/

scp /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/Streptococcus_thermophilus_RMK202_pgap.faa vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/faa/



###multifasta2fasta

mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta
for i in $(grep ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/_eightFreebayes.fasta |sed 's/>//g' |cut -d ' ' -f 1)
do
samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/_eightFreebayes.fasta ${i} > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/${i}.fasta
done

##===============
###annotate the genomes
##===============

mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/PROKKA_singleGenomes/{contig_1,contig_2}

##sterm
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/PROKKA_singleGenomes/contig_1/ --force --usegenus --genus streptococcus --locustag Sterm_K1 --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/contig_1.fasta

##Ldel 
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/PROKKA_singleGenomes/contig_2/ --force --usegenus --genus lactobacillus --locustag Ldel_K1 --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/contig_2.fasta


##===============
###gather all faa files
##===============

##Sterm

mkdir -p /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/{Sterm,Ldel}
mkdir -p /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/faa


#cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/PROKKA_singleGenomes/contig_1/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/faa/contig_1_meta.faa
cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/annotation_eggnog_rast/assembly_20191129/sterm/S_thermophilus_RMK202_genes.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/faa/S_thermophilus_RMK202_mag.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/faa/S72.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/faa/S50.faa


scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/faa/* /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/CDS/

##Ldel

mkdir -p /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa


#cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/PROKKA_singleGenomes/contig_2/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/contig_2_meta.faa
cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/annotation_eggnog_rast/assembly_20191129/ldel/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L_delbrueckii_RMK202_mag.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L104//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L104.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L108//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L108.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L35//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L35.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L39//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L39.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L44//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L44.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L70//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L70.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L71//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L71.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L99//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L99.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L80//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L80.faa


scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/* /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/CDS/


scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/contig_5.fasta /home/vincent/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/1AB/mapping2SCOG_reference/

scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/contig_1.fasta /home/vincent/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/1AB/mapping2SCOG_reference/


```

#####Contigs

Here, I describe the assembled contigs in K1:

contig_1: Sterm
contig_2: Ldel
contig_3: Lactococcus phage
contig_4: Cow
contig_5: ldel plasmid



#####orthofinder
identify SCOG with orthofinder of all picked isolates and the metagenome assembled genome.

```{bash}


##sterm

rm -r /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder

    orthofinder -f /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/faa/ \
      -t 37 \
      -o /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/ \
      -a 37



##Ldel

rm -r /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder

    orthofinder -f /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/ \
      -t 37 \
      -o /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/ \
      -a 37


```

####extract single copy locations

```{bash}
##===============
##sterm
##===============
mkdir -p /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/extraction_of_OGs/
##-------extract OG names
#rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/Orthogroups/test.txt
rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/extraction_of_OGs/OG_names_sterm.txt
for OG in $(cat /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/Orthogroups/Orthogroups_SingleCopyOrthologues.txt)
do

grep "$OG" /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/Orthogroups/Orthogroups.txt |cut -d ' ' -f 4 >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/extraction_of_OGs/OG_names_sterm.txt


done #OG
wc -l /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/extraction_of_OGs/OG_names_sterm.txt

##-------getting gff of OGs 
#rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/Orthogroups/test.txt
rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/extraction_of_OGs/OG_genes_sterm.gff
for OG in $(cat /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/extraction_of_OGs/OG_names_sterm.txt)
do

grep "ID=${OG}_gene" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/PROKKA_singleGenomes/contig_1/*.gff >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/extraction_of_OGs/OG_genes_sterm.gff


done #OG
wc -l /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/extraction_of_OGs/OG_genes_sterm.gff
#wc -l /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/Orthogroups/test.txt

sort -k4 -n /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/extraction_of_OGs/OG_genes_sterm.gff > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/extraction_of_OGs/OG_genes_sterm_sorted.gff

##===============
##Ldel
##===============
mkdir -p /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/extraction_of_OGs/
##-------extract OG names
#rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/Orthogroups/test.txt
rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/extraction_of_OGs/OG_names_ldel.txt
for OG in $(cat /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/Orthogroups/Orthogroups_SingleCopyOrthologues.txt)
do

grep "$OG" /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/Orthogroups/Orthogroups.txt |cut -d ' ' -f 11 >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/extraction_of_OGs/OG_names_ldel.txt


done #OG
wc -l /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/extraction_of_OGs/OG_names_ldel.txt

##-------getting gff of OGs 
#rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/Orthogroups/test.txt
rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/extraction_of_OGs/OG_genes_ldel.gff
for OG in $(cat /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/extraction_of_OGs/OG_names_ldel.txt)
do

grep "ID=${OG}_gene" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/PROKKA_singleGenomes/contig_2/*.gff >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/extraction_of_OGs/OG_genes_ldel.gff


done #OG
wc -l /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/extraction_of_OGs/OG_genes_ldel.gff
#wc -l /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/Orthogroups/test.txt

sort -k4 -n /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/extraction_of_OGs/OG_genes_ldel.gff > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/extraction_of_OGs/OG_genes_ldel_sorted.gff


```

#### Coverage at all SCOG

```{bash}

mkdir -p /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/merging_gffs

cat /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/extraction_of_OGs/OG_genes_sterm_sorted.gff /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/extraction_of_OGs/OG_genes_ldel_sorted.gff > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/merging_gffs/sterm_ldel_SCOG.gff

##===============
##extracting coverage
##===============


  num=1
  
  for names in $(cat /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name_allOfExperiment.txt)
do

echo ${num}"/37  :" ${names}
    num=$((num+1))
    
    #names=th_K1_12h
    #rm -r /home/vincent/Projects/2019_pilotplant/05_mapping2DB/${names}/BacteriaDBmapping
  
    mkdir -p /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coverage/
 

bedtools coverage -bed -a /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/merging_gffs/sterm_ldel_SCOG.gff -b /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam  > \
  /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coverage/${names}2Bacteria.bed 
  

done

##============
## Add sample name to file
##============

  num=1
  
  for names in $(cat /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name_allOfExperiment.txt)
do

echo ${num}"/37  :" ${names}
    num=$((num+1))
    
    #names=th_K1_12h
    #rm -r /home/vincent/Projects/2019_pilotplant/05_mapping2DB/${names}/BacteriaDBmapping
  
    mkdir -p /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coverage_final/
 
awk -F "\t" -v name="$names"  '{OFS="\t"} {print $0, name}' /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coverage/${names}2Bacteria.bed  > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coverage_final/${names}2Bacteria.bed
  

done

mkdir -p /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coverage_allToghther

cat /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coverage_final/*.bed > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coverage_allToghther/coverage_2_referenceGenome_allSamples.bed


###Bring local

scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coverage_allToghther/coverage_2_referenceGenome_allSamples.bed /home/vincent/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/1AB/mapping2SCOG_reference/

```

analysing the SCOG mapping in R

with a piecewise regression: Piecewise regression comes about when you have ‘breakpoints’, where there are clearly two different linear relationships in the data with a sudden, sharp change in directionality

```{r}
library(readr)
library(segmented)
library(plotly)
library(ggplot2)
library(gridExtra)
library(cowplot)

coverage <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/1AB/mapping2SCOG_reference/coverage_2_referenceGenome_allSamples.bed","\t", escape_double = FALSE, col_names = c("contig","annotation","gene","start","end","unknown1","orientation","unknown2","id","count","length_unmapped","geneLength","percent","name"), trim_ws = TRUE)

coverage$coverage <- (coverage$count*600)/coverage$geneLength

coverage$species <- ifelse(coverage$contig=="contig_1","S. thermophilus","L. delbrueckii")
coverage <- coverage[which(coverage$end-coverage$start>600),]




coverage_th_K1_6h <- coverage[which(coverage$name=="di_K1_24h"),]


##-----Ldel
coverage_th_K1_6h_Ldel <- coverage_th_K1_6h[which(coverage_th_K1_6h$species=="L. delbrueckii"),]


mean_ldel <- max(coverage_th_K1_6h$end)/2

y <- as.double(coverage_th_K1_6h_Ldel$coverage)
x <- as.double(coverage_th_K1_6h_Ldel$start)

lin.mod <- lm(y~x)
# summary(lin.mod)

segmented.mod <- segmented(lin.mod, seg.Z = ~x, psi=1000000)
# summary(segmented.mod)

fit <- numeric(length(x)) * NA
fit[complete.cases(rowSums(cbind(y, x)))] <- broken.line(segmented.mod)$fit
data1 <- data.frame(x = x, y = y, fit = fit,geneID=coverage_th_K1_6h_Ldel$id,geneStart=coverage_th_K1_6h_Ldel$start,geneSize=coverage_th_K1_6h_Ldel$end-coverage_th_K1_6h_Ldel$start,Species=coverage_th_K1_6h_Ldel$species)

pldel <- ggplot(data1, aes(x=x,y=y,label=geneID,label2=geneStart))+
  geom_point(aes(size=geneSize,color=Species))+
  geom_line(aes(x = x, y = fit),size=2, color = 'grey')+
  labs(x=" ",y="coverage",title =unique(coverage_th_K1_6h_Ldel$name) )+
  scale_color_manual(values="#10B552")+
  theme(legend.position="none",axis.title.x=element_blank())


##-----sterm
coverage_th_K1_6h_Ldel <- coverage_th_K1_6h[which(coverage_th_K1_6h$species=="S. thermophilus"),]


mean_ldel <- max(coverage_th_K1_6h$end)/2

y <- as.double(coverage_th_K1_6h_Ldel$coverage)
x <- as.double(coverage_th_K1_6h_Ldel$start)

lin.mod <- lm(y~x)
# summary(lin.mod)

segmented.mod <- segmented(lin.mod, seg.Z = ~x, psi=1000000)
# summary(segmented.mod)

fit <- numeric(length(x)) * NA
fit[complete.cases(rowSums(cbind(y, x)))] <- broken.line(segmented.mod)$fit
data1 <- data.frame(x = x, y = y, fit = fit,geneID=coverage_th_K1_6h_Ldel$id,geneStart=coverage_th_K1_6h_Ldel$start,geneSize=coverage_th_K1_6h_Ldel$end-coverage_th_K1_6h_Ldel$start,Species=coverage_th_K1_6h_Ldel$species)

psterm <- ggplot(data1, aes(x=x,y=y,label=geneID,label2=geneStart))+
  geom_point(aes(size=geneSize,color=Species))+
  geom_line(aes(x = x, y = fit),size=2, color = 'grey')+
  labs(x="genome location",y="coverage")+
  scale_color_manual(values="#EB4D4D")+
  theme(legend.position="none")


# p

# pboth <- grid.arrange(pldel,psterm)

# pboth <- grid.arrange(pboth,pboth)

# ggp <- ggplotly(pboth)
# plot(x,y, pch=16)
# plot(segmented.mod,col="grey",lwd=3, add=T)
# 
# ggp

  # colours <-  c("#10B552","#EB4D4D")
  
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/1AB/mapping2SCOG_reference/L44_coveragePlot.html")

psterm  <- list()
pldel  <- list()
#coverage_sample <- list
# p <- list()
for (i in  1:length(unique(coverage$name))){
  
  
  sample <- unique(coverage$name)[i]
  coverage_sample  <- coverage[which(coverage$name==sample),]
  coverage_sample_Ldel <- coverage_sample[which(coverage_sample$species=="L. delbrueckii"),]
  mean_ldel <- max(coverage_sample_Ldel$end)/2

  y <- as.double(coverage_sample_Ldel$coverage)
  x <- as.double(coverage_sample_Ldel$start)

  lin.mod <- lm(y~x)
summary( lin.mod)
  segmented.mod <- segmented(lin.mod, seg.Z = ~x, psi=mean_ldel)
  midpoint <- segmented.mod$psi[2]
  maxpoint <- max(coverage_sample_Ldel$end)
  summary(segmented.mod)
  slope_01 <- segmented.mod$coefficients[2]
  slope_02 <- segmented.mod$coefficients[3]+segmented.mod$coefficients[2]
  intercept_01 <- segmented.mod$coefficients[1]
  break1 <- segmented.mod$psi[[3]]
  intercept_02 <-  intercept_01 + (slope_01* midpoint) - (slope_02* midpoint) 
  
  minValue_01 <- intercept_01 + (slope_01* midpoint)
  minValue_02 <- intercept_02 + (slope_02* midpoint)
  
  maxValue_01 <- intercept_01
  maxValue_02 <- intercept_02 + (slope_02* maxpoint)
  
  maxValue_total <- max(maxValue_01,maxValue_02)
  minValue_total <- min(minValue_01,minValue_02)
  PTR <- maxValue_total/ minValue_total
  
  
  
  fit <- numeric(length(x)) * NA
  fit[complete.cases(rowSums(cbind(y, x)))] <- broken.line(segmented.mod)$fit
  data1 <- data.frame(x = x, y = y, fit = fit,geneID=coverage_sample_Ldel$id,geneStart=coverage_sample_Ldel$start,geneSize=coverage_sample_Ldel$end-coverage_sample_Ldel$start,Species=coverage_sample_Ldel$species,Name=coverage_sample_Ldel$name)

pldel[[i]] <- ggplot(data1, aes(x=x,y=y,label=geneID,label2=geneStart,label3=Name))+
  geom_point(aes(size=geneSize,color=Species))+
  annotate(geom="text", x=500000, y=max(data1$y)-(0.2*max(data1$y)), label=paste0(sample," (PTR= ",round(PTR,digits = 2),")"),color="black")+
  geom_line(aes(x = x, y = fit),size=2, color = 'grey')+
  labs(x="genome location",y="coverage")+
  scale_color_manual(values="#10B552")+
  theme_bw()+
  theme(legend.position="none")






# ggplot(data1, aes(x=x,y=y,label=geneID,label2=geneStart))+
#   geom_point(aes(size=geneSize,color=Species))+ geom_abline(intercept =  intercept_01, slope =  slope_01, 
#                      aes(colour = "first part"), show_guide = TRUE)+ geom_abline(intercept =  intercept_02, slope =  slope_02, 
#                      aes(colour = "first part"), show_guide = TRUE)


}



 
#subplot(pldel, nrows = length(pldel)/4)
#subplot(pldel, nrows = length(pldel)/4, shareX = TRUE, titleX = FALSE)

htmlwidgets::saveWidget(subplot(pldel, nrows = length(pldel)/4), "~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/1AB/mapping2SCOG_reference/L44_coveragePlot.html")

pdf("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/1AB/mapping2SCOG_reference/mapping2ONTreference_ldel.pdf",width=16,height=32,paper='special') 

# do.call(grid.arrange, c(p, ncol=4))

# do.call(plot_grid, c(pldel, labels = "TPKM", align = 'h', ncol=4))
#do.call(plot_grid, c(psterm, labels = "TPKM", align = 'h', ncol=4))
do.call(grid.arrange, c(psterm, ncol=4))


dev.off()

do.call(grid.arrange, c(pldel, ncol=4))

do.call(grid.arrange, c(pldel, ncol=4))



psterm  <- list()
pldel  <- list()
#coverage_sample <- list
# p <- list()
for (i in  1:length(unique(coverage$name))){
  
  
  sample <- unique(coverage$name)[i]
  coverage_sample  <- coverage[which(coverage$name==sample),]
  coverage_sample_Ldel <- coverage_sample[which(coverage_sample$species=="S. thermophilus"),]
  mean_ldel <- max(coverage_sample_Ldel$end)/2

  y <- as.double(coverage_sample_Ldel$coverage)
  x <- as.double(coverage_sample_Ldel$start)

  lin.mod <- lm(y~x)
summary( lin.mod)
  segmented.mod <- segmented(lin.mod, seg.Z = ~x, psi=mean_ldel)
  midpoint <- segmented.mod$psi[2]
  maxpoint <- max(coverage_sample_Ldel$end)
  summary(segmented.mod)
  slope_01 <- segmented.mod$coefficients[2]
  slope_02 <- segmented.mod$coefficients[3]+segmented.mod$coefficients[2]
  intercept_01 <- segmented.mod$coefficients[1]
  break1 <- segmented.mod$psi[[3]]
  intercept_02 <-  intercept_01 + (slope_01* midpoint) - (slope_02* midpoint) 
  
  minValue_01 <- intercept_01 + (slope_01* midpoint)
  minValue_02 <- intercept_02 + (slope_02* midpoint)
  
  maxValue_01 <- intercept_01
  maxValue_02 <- intercept_02 + (slope_02* maxpoint)
  
  maxValue_total <- max(maxValue_01,maxValue_02)
  minValue_total <- min(minValue_01,minValue_02)
  PTR <- maxValue_total/ minValue_total
  
  
  
  fit <- numeric(length(x)) * NA
  fit[complete.cases(rowSums(cbind(y, x)))] <- broken.line(segmented.mod)$fit
  data1 <- data.frame(x = x, y = y, fit = fit,geneID=coverage_sample_Ldel$id,geneStart=coverage_sample_Ldel$start,geneSize=coverage_sample_Ldel$end-coverage_sample_Ldel$start,Species=coverage_sample_Ldel$species,Name=coverage_sample_Ldel$name)



psterm[[i]] <- ggplot(data1, aes(x=x,y=y,label=geneID,label2=geneStart,label3=Name))+
  geom_point(aes(size=geneSize,color=Species))+
  annotate(geom="text", x=500000, y=max(data1$y)-(0.2*max(data1$y)), label=paste0(sample," (PTR= ",round(PTR,digits = 2),")"),color="black")+
  geom_line(aes(x = x, y = fit),size=2, color = 'grey')+
  labs(x="genome location",y="coverage")+
  scale_color_manual(values="#EB4D4D")+
  theme_bw()+
  theme(legend.position="none")




# ggplot(data1, aes(x=x,y=y,label=geneID,label2=geneStart))+
#   geom_point(aes(size=geneSize,color=Species))+ geom_abline(intercept =  intercept_01, slope =  slope_01, 
#                      aes(colour = "first part"), show_guide = TRUE)+ geom_abline(intercept =  intercept_02, slope =  slope_02, 
#                      aes(colour = "first part"), show_guide = TRUE)


}

pdf("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/1AB/mapping2SCOG_reference/mapping2ONTreference_sterm.pdf",width=16,height=32,paper='special') 

# do.call(grid.arrange, c(p, ncol=4))

# do.call(plot_grid, c(pldel, labels = "TPKM", align = 'h', ncol=4))
do.call(plot_grid, c(psterm, labels = "TPKM", align = 'h', ncol=4))
#do.call(grid.arrange, c(psterm, ncol=4))


dev.off()

```

plottly for multiple plots [subplots](https://plotly-r.com/arranging-views.html)


## Pandora for Strains



#####Roary

First, run roary to identify all orthologous genes. 
A good [tutorial](https://github.com/microgenomics/tutorials/blob/master/pangenome.md)


```{bash}

##-----sterm
sudo rm -r /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/
mkdir -p /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/strains

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Streptococcus_thermophilus_RMK202/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/strains/Streptococcus_thermophilus_RMK202.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/strains/S72.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/strains/S50.gff

sudo roary -f /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/ -p 5 -e -n -v /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/strains/*.gff

cd /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/_1564994606/

sudo mkdir -p /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/_1564994606//ffn/

sudo cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Streptococcus_thermophilus_RMK202/*.ffn > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/_1564994606/ffn/Streptococcus_thermophilus_RMK202.ffn
sudo cat /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/*.ffn > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/_1564994606/ffn/S72.ffn
sudo cat /archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/*.ffn > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/_1564994606/ffn/S50.ffn

python3 /home/vincent/apps/DBGenerator.py ffn

mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Sterm
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/_1564994606//pangenoma_locus.csv /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/

##=====================
##-----Ldel
##=====================
sudo rm -r /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/
mkdir -p /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/


cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_delbrueckii_RMK202/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/Lactobacillus_delbrueckii_RMK202.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L104//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L104.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L108//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L108.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L35//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L35.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L39//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L39.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L44//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L44.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L70//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L70.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L71//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L71.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L99//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L99.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L80//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L80.gff


sudo roary -f /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/ -e -n -v /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/*.gff



cd /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/_1564994796/

sudo mkdir -p /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/_1564994796/ffn/


cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_delbrueckii_RMK202/*.ffn > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/Lactobacillus_delbrueckii_RMK202.ffn
cat /archiv/Projects/2018_Culturomics/06_Spades/L104//Prokka/20190203/*.ffn > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L104.ffn
cat /archiv/Projects/2018_Culturomics/06_Spades/L108//Prokka/20190203/*.ffn > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L108.ffn
cat /archiv/Projects/2018_Culturomics/06_Spades/L35//Prokka/20190203/*.ffn > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L35.ffn
cat /archiv/Projects/2018_Culturomics/06_Spades/L39//Prokka/20190203/*.ffn > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L39.ffn
cat /archiv/Projects/2018_Culturomics/06_Spades/L44//Prokka/20190203/*.ffn > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L44.ffn
cat /archiv/Projects/2018_Culturomics/06_Spades/L70//Prokka/20190203/*.ffn > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L70.ffn
cat /archiv/Projects/2018_Culturomics/06_Spades/L71//Prokka/20190203/*.ffn > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L71.ffn
cat /archiv/Projects/2018_Culturomics/06_Spades/L99//Prokka/20190203/*.ffn > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L99.ffn
cat /archiv/Projects/2018_Culturomics/06_Spades/L80//Prokka/20190203/*.ffn > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L80.ffn


python3 /home/vincent/apps/DBGenerator.py ffn



mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Ldel
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/_1564994796/pangenoma_locus.csv /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Ldel


```


create gene family files from roary output.
The following files are promising:
gene_presence_absence.csv : contains total genes in rows and column $15,$16,$17 different genomes


```{bash}

##======
##sterm
##======

##----------------
##merge all ffn files
##----------------
mkdir -p /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/ffn

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/PROKKA_singleGenomes/contig_1/*.ffn \
  /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/*.ffn \
  /archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/*.ffn > \
  /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/ffn/merged_sterm.ffn

##----------------
##prepare orthologous gene file
##----------------

mkdir -p /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned

sed '1d' /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_1564466937/gene_presence_absence.csv >  /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/gene_presence_absence_woHeade.csv
rowwss=$(wc -l /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/gene_presence_absence_woHeade.csv | cut -d  ' ' -f 1)

row=2
genomes=15

rm -r /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/geneOrthologous/
mkdir -p /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/geneOrthologous/

for row in $(seq 1 $rowwss)
do

    rowsss=$(sed -n "${row}p" /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/gene_presence_absence_woHeade.csv) 
    geneName=$(echo ${rowsss} |cut -d ',' -f 1 |sed 's/"//g')

    rm /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/geneOrthologous/${geneName}.ffn

for genomes in $(echo "15 16 17")
do

  gene=$(echo ${rowsss} |cut -d ',' -f ${genomes} |sed 's/"//g' |sed 's/[[:space:]]*$//')
  gene=$(echo ${rowsss} |awk -F '","' -v name="$genomes"  '{print $name}' |sed 's/"//g' |sed 's/[[:space:]]*$//')

  
  
  samtools faidx /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/ffn/merged_sterm.ffn ${gene} >> /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/geneOrthologous/${geneName}.ffn


done #genomes


done ##row


```



#####Unique gene analysis
analysis of roary output
```{r}
library(readr)
library(ggplot2)


##=========
##Sterm
##=========

  uniquees <-read.delim("/home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/pangenoma_locus.csv",sep="|",header = FALSE)
plotData <- table(table(uniquees$V1))
data <- data.frame(genomes=c(1,2,3),genes=as.vector(plotData))

plotsterm <- ggplot(data,aes(x=genomes,y=genes))+geom_bar(stat="identity")+
  theme_classic()

svg("/home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/pangenoma_locus_plot.svg",width=3,height=2)
plotsterm 
 dev.off()
 
 
uniquegenes <- names(table(uniquees$V1)[which((table(uniquees$V1)==1))])
 
     # uniques_who <- uniquees[which(uniquegenes,uniquees$V1),]
      uniques_who <- uniquees[which(uniquees$V1  %in% uniquegenes),]
     uniques_who$sample <- str_split_fixed(uniques_who$V2, "_", 3)[,1]
     uniques_who[uniques_who$sample=="st","sample"] <- "Streptococcus_thermophilus_RMK202"
     
plotData <- table(uniques_who$sample)
plotData
sum(plotData)

write.table(uniques_who,"/home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/pangenoma_locus_plot.csv",sep = "\t",quote = FALSE,col.names = FALSE)

data <- data.frame(genomes=c("s50","S72","MAG"),genes=as.vector(plotData))


plotsterm <- ggplot(data,aes(x=genomes,y=genes))+geom_bar(stat="identity")+
  theme_classic()+
  labs(x="",
       y="unique genes")

plotsterm 
svg("/home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/pangenoma_locus_plot_uniques.svg",width=3,height=2)
plotsterm 
 dev.off()


 
##=========
##Ldel
##=========

  uniquees <-read.delim("/home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Ldel//pangenoma_locus.csv",sep="|",header = FALSE)
plotData <- table(table(uniquees$V1))
data <- data.frame(genomes=1:10,genes=as.vector(plotData))

plotsterm <- ggplot(data,aes(x=genomes,y=genes))+geom_bar(stat="identity")+
  theme_classic()+
  scale_x_continuous(breaks = 1:10,labels=1:10)
plotsterm 
svg("/home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Ldel/pangenoma_locus_plot.svg",width=3,height=2)
plotsterm 
 dev.off()
 
 
uniquegenes <- names(table(uniquees$V1)[which((table(uniquees$V1)==1))])
 
     # uniques_who <- uniquees[which(uniquegenes,uniquees$V1),]
      uniques_who <- uniquees[which(uniquees$V1  %in% uniquegenes),]
     uniques_who$sample <- str_split_fixed(uniques_who$V2, "_", 3)[,1]
     uniques_who[uniques_who$sample=="ld","sample"] <- "Lactobacillus_delbrueckii_RMK202"
     
plotData <- table(uniques_who$sample)
plotData
sum(plotData)
write.table(uniques_who,"/home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Ldel/pangenoma_locus_plot.csv",sep = "\t",quote = FALSE,col.names = FALSE)

data <- data.frame(genomes=c("L104","L108","L35","L39","L44","L70","L71","L80","L99","MAG"),genes=as.vector(plotData))


plotsterm <- ggplot(data,aes(x=genomes,y=genes))+geom_bar(stat="identity")+
  theme_classic()+
  labs(x="",
       y="unique genes")+
theme(axis.text.x=element_text(angle=45, hjust=1))

plotsterm 
svg("/home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Ldel/pangenoma_locus_plot_uniques.svg",width=3,height=2)
plotsterm 
 dev.off()
 
```

make a unique genes file for eggnog
```{bash}

##=========
##transfer to bigmachine
##=========
##-------------
##Sterm
##-------------

scp /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/pangenoma_locus_plot.csv vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/
scp -r /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/ vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/


##-------------
##Ldel
##-------------

scp /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Ldel/pangenoma_locus_plot.csv vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/

##=========
##getting annotations
##=========
##-------------
##Sterm
##-------------
row=2
genome=S50
gene=group_156

rowwss=$(wc -l /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/pangenoma_locus_plot.csv | cut -d  ' ' -f 1)
rm -r /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/uniqueGenes/
mkdir -p /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/uniqueGenes/

for row in $(seq 1 $rowwss)
do

   genome=$(sed -n "${row}p" /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/pangenoma_locus_plot.csv |awk '{print $4}') 
   gene=$(sed -n "${row}p" /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/pangenoma_locus_plot.csv |awk '{print $2}') 
   
   
   geneName=$(grep "${gene}:"  /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/_1564994606/clustered_proteins | sed "s/${gene}: //g" |sed -e's/[[:space:]]*$//')

   grep "ID=${geneName};" /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/strains/${genome}.gff | awk -F "\t" -v genomeee="$genome"  'BEGIN{OFS="\t"}{print $0 , genomeee}' >> /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/uniqueGenes/${genome}_uniques_Prokka.gff
  
 grep "${geneName}" /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/eggnog/${genome}_query_seqs.fa.emapper.annotations | awk -F "\t" -v genomeee="$genome"  'BEGIN{OFS="\t"}{print $0 , genomeee}' >> /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/uniqueGenes/${genome}_uniques_eggnog.gff
  
done

cat /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/uniqueGenes/*eggnog.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/uniqueGenes/sterm_merged.eggnog.txt
cat /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/uniqueGenes/*Prokka.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/uniqueGenes/sterm_merged.Prokka.txt
##-------------
##Ldel
##-------------

row=2
genome=S50
gene=group_156

rowwss=$(wc -l /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/pangenoma_locus_plot.csv | cut -d  ' ' -f 1)
rm -r /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/uniqueGenes/
mkdir -p /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/uniqueGenes/

for row in $(seq 1 $rowwss)
do

   genome=$(sed -n "${row}p" /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/pangenoma_locus_plot.csv |awk '{print $4}') 
   gene=$(sed -n "${row}p" /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/pangenoma_locus_plot.csv |awk '{print $2}') 
   
   
   geneName=$(grep "${gene}:"  /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/_1564994796/clustered_proteins | sed "s/${gene}: //g" |sed -e's/[[:space:]]*$//')

   grep "ID=${geneName};" /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/${genome}.gff | awk -F "\t" -v genomeee="$genome"  'BEGIN{OFS="\t"}{print $0 , genomeee}' >> /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/uniqueGenes/${genome}_uniques_Prokka.gff
  
 grep "${geneName}" /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/eggnog/${genome}_query_seqs.fa.emapper.annotations | awk -F "\t" -v genomeee="$genome"  'BEGIN{OFS="\t"}{print $0 , genomeee}' >> /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/uniqueGenes/${genome}_uniques_eggnog.gff
  
done

cat /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/uniqueGenes/*eggnog.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/uniqueGenes/Ldel_merged.eggnog.txt
cat /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/uniqueGenes/*Prokka.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/uniqueGenes/Ldel_merged.Prokka.txt


##=========
##getting local
##=========
mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/unique_sterm
scp -r  vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/uniqueGenes/ /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/unique_sterm

mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/unique_ldel
scp -r  vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/uniqueGenes/ /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/unique_ldel

```

[eggnog documentation](https://github.com/eggnogdb/eggnog-mapper/wiki/eggNOG-mapper-v2)

```{r}

library(thacklr)
library(readr)
library(dplyr)
library(ggplot2)

COG_functional_categories <- read_delim("~/Desktop/ScriptRepository/COG_functional_categories.csv",  ",", escape_double = FALSE, col_names = c("short","long","divided"),trim_ws = TRUE, na = "NA")

sterm_all_unique <- read_delim("/home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/unique_sterm/uniqueGenes/sterm_merged.eggnog.txt", "\t", escape_double = FALSE, trim_ws = TRUE, na = "NA",col_names = c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description","genome")) 


#unique(sterm_all_unique$COG_Functional_Category)[match(COG_functional_categories$short , unique(sterm_all_unique$COG_Functional_Category))]
unique(sterm_all_unique$COG_Functional_Category)[which(!unique(sterm_all_unique$COG_Functional_Category) %in% COG_functional_categories$short )]

# table(sterm_all_unique$COG_Functional_Category)

sterm_all_unique$longCOG <- NA
for (i in 1:nrow(sterm_all_unique)) {
  sterm_all_unique[i,"longCOG"] <- COG_functional_categories[which(COG_functional_categories$short==as.character(sterm_all_unique[i,"COG_Functional_Category"])),"long"]
  
}



CogPlotSterm <- ggplot(sterm_all_unique,aes(x=longCOG,fill=genome))+geom_bar(position = "dodge2")+
  theme_classic()+
  labs(x="")+
  theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9),
          # # axis.text.y.left = element_text(color = "blue"),
          # axis.text.y.right = element_text(color="red"),
          # axis.title.y.left = element_text(color="blue"),
          # axis.title.y.right = element_text(color="red"),
          # rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          #legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank(),
          plot.margin=unit(c(0,0,0,2), "cm"),
          axis.text.x.bottom = element_text(size=10),
          legend.position = c(0.02, 0.98),
          legend.justification = c("left", "top")
          )
CogPlotSterm
   svg("~/Desktop/presenations/my_presentations/20190807_committeeMeeting/figures/cogPlot.svg",width=7,height=7)
CogPlotSterm 
 dev.off()

 sterm_all_unique$species ="S. thermophilus"
 sterm_uniques <- sterm_all_unique
##-------------
##Ldel
##-------------

 
sterm_all_unique <- read_delim("/home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/unique_ldel/uniqueGenes/Ldel_merged.eggnog.txt", "\t", escape_double = FALSE, trim_ws = TRUE, na = "NA",col_names = c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description","genome")) 


#unique(sterm_all_unique$COG_Functional_Category)[match(COG_functional_categories$short , unique(sterm_all_unique$COG_Functional_Category))]
unique(sterm_all_unique$COG_Functional_Category)[which(!unique(sterm_all_unique$COG_Functional_Category) %in% COG_functional_categories$short )]

# table(sterm_all_unique$COG_Functional_Category)

sterm_all_unique$longCOG <- NA
for (i in 1:nrow(sterm_all_unique)) {
  sterm_all_unique[i,"longCOG"] <- COG_functional_categories[which(COG_functional_categories$short==as.character(sterm_all_unique[i,"COG_Functional_Category"])),"long"]
  
}



CogPlotSterm <- ggplot(sterm_all_unique,aes(x=longCOG,fill=genome))+geom_bar(position = "dodge2")+
  theme_classic()+
  labs(x="")+
  theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9),
          # # axis.text.y.left = element_text(color = "blue"),
          # axis.text.y.right = element_text(color="red"),
          # axis.title.y.left = element_text(color="blue"),
          # axis.title.y.right = element_text(color="red"),
          # rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          #legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank(),
          plot.margin=unit(c(0,0,0,2), "cm"),
          axis.text.x.bottom = element_text(size=10),
          legend.position = c(0.02, 0.99),
          legend.justification = c("left", "top")
          )
CogPlotSterm
   svg("~/Desktop/presenations/my_presentations/20190807_committeeMeeting/figures/cogPlot_ldel.svg",width=7,height=5)
CogPlotSterm 
 dev.off()

 
 sterm_all_unique$species ="L. delbrueckii"
 Ldel_uniques <- sterm_all_unique
 
 
##-------------
##Merge and plot
##-------------
```




#####Multiseqalignment for all strains

first I need to creat a Multi-Sequence-Alignment of all isolated strains. 
I do this with (Mugsy)[http://mugsy.sourceforge.net/]
or mafft

```{bash}

geneName=yfkN_3

mkdir -p /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/MSA/

for gene in $(ls /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/geneOrthologous/)
do

geneName=$(echo ${gene}| cut -d '.' -f 1)

rm /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/MSA/${geneName}.msa


mafft --auto /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/geneOrthologous/${geneName}.ffn > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/MSA/${geneName}.msa


done

##----------
##remove empty files
##----------

find /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/MSA/ -size 0 -delete

##----------
##make index file
##----------

mkdir -p /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/log

awk 'BEGIN{OFS="\t"}{print "sample_id","infile"}' > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/log/index.file

for gene in $(ls /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/MSA/)
do

geneName=$(echo ${gene}| cut -d '.' -f 1)


echo "/archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/MSA/" | awk -v name="$geneName"  'BEGIN{OFS="\t"}{print name,$2name".msa"}' > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/log/index.file


done

```


```{bash}

#wget https://sourceforge.net/projects/mugsy/files/latest/download/mugsy_x86-64-v1r2.2.tgz

###------------Sterm

mkdir -p /archiv/Projects/2019_Nano_Meta/06_MSA_Mugsy/Sterm

#export MUGSY_INSTALL=/home/vincent/apps/mugsy/mugsy_x86-64-v1r2.2/

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/contig_1.fasta > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/MAG_contig_1.fasta
cat /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/*.fna > /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/S72.fna
cat /archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/*.fna > /archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/S50.fna


/home/vincent/apps/mugsy/mugsy_x86-64-v1r2.2/mugsy --directory /archiv/Projects/2019_Nano_Meta/06_MSA_Mugsy/Sterm --prefix mugsy_sterm_msa /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/MAG_contig_1.fasta \
/archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/S72.fna \
/archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/S50.fna

mkdir -p /archiv/Projects/2018_Culturomics/genomes/
cp /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/S72.fna /archiv/Projects/2018_Culturomics/genomes/
cp /archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/S50.fna /archiv/Projects/2018_Culturomics/genomes/

###------------Ldel


cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/PROKKA_singleGenomes/contig_1/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/faa/contig_1_meta.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/faa/S72.faa \
cat /archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/faa/S50.faa


##Ldel

mkdir -p /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa


cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/PROKKA_singleGenomes/contig_2/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/contig_2_meta.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L104//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L104.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L108//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L108.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L35//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L35.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L39//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L39.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L44//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L44.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L70//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L70.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L71//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L71.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L99//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L99.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L80//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L80.faa



cp /archiv/Projects/2018_Culturomics/06_Spades/L104//Prokka/20190203/*.fna /archiv/Projects/2018_Culturomics/genomes/L104.fna
cp /archiv/Projects/2018_Culturomics/06_Spades/L108//Prokka/20190203/*.fna /archiv/Projects/2018_Culturomics/genomes/L108.fna
cp /archiv/Projects/2018_Culturomics/06_Spades/L35//Prokka/20190203/*.fna /archiv/Projects/2018_Culturomics/genomes/L35.fna
cp /archiv/Projects/2018_Culturomics/06_Spades/L39//Prokka/20190203/*.fna /archiv/Projects/2018_Culturomics/genomes/L39.fna
cp /archiv/Projects/2018_Culturomics/06_Spades/L44//Prokka/20190203/*.fna /archiv/Projects/2018_Culturomics/genomes/L44.fna
cp /archiv/Projects/2018_Culturomics/06_Spades/L70//Prokka/20190203/*.fna /archiv/Projects/2018_Culturomics/genomes/L70.fna
cp /archiv/Projects/2018_Culturomics/06_Spades/L71//Prokka/20190203/*.fna /archiv/Projects/2018_Culturomics/genomes/L71.fna
cp /archiv/Projects/2018_Culturomics/06_Spades/L99//Prokka/20190203/*.fna /archiv/Projects/2018_Culturomics/genomes/L99.fna
cp /archiv/Projects/2018_Culturomics/06_Spades/L80//Prokka/20190203/*.fna /archiv/Projects/2018_Culturomics/genomes/L80.fna

cp /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_delbrueckii_RMK202/*.fna /archiv/Projects/2018_Culturomics/genomes/Lactobacillus_delbrueckii_RMK202.fna

cp /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Streptococcus_thermophilus_RMK202//*.fna /archiv/Projects/2018_Culturomics/genomes/Streptococcus_thermophilus_RMK202.fna


mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/{afterAssembly,startaligned}
scp -r vincent@130.223.51.151:/archiv/Projects/2018_Culturomics/genomes/ /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/afterAssembly
```



#####make_prg

Before running pandora we need to create a pangenome graph with (make prg)[https://github.com/rmcolq/make_prg]

```{bash}
nextflow run /home/vincent/apps/make_prg/make_prg_nexflow.nf /archiv/Projects/2019_Nano_Meta/06_MSA_Mugsy/Sterm/mugsy_sterm_msa.maf

scp vincent@130.223.51.151:/home/vincent/apps/make_prg/pytest.error /

```


##Circos plot

```{bash}
home=/home/vincent/bin/apps/circos-0.69-6/

##karyotyp
${home}/data/karyotype/Ldel_rmk202.txt

##example
cp etc/repeat_nwc1_sterm_01.conf etc/ldel_rmk202_circos.conf

#example run
bin/circos -conf etc/ldel_rmk202_circos.conf ; firefox ./circos.svg &


##links
scp -r vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/links_ldel_curated.txt /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/links_ldel.txt

data/nwc_rmk/links_ldel.txt

##CDS
scp -r vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_delbrueckii_RMK202/PROKKA_08052019.gff /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/

grep "^#" -v /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/PROKKA_08052019.gff |awk ' { if ($3=="CDS") print $0 } ' |awk 'BEGIN{OFS="\t"} { if ($7=="+") print $1,$4,$5,"color=green";else print $1,$4,$5,"color=black" } ' > /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/cds_ldel.txt

##rRNA

  grep "^#" -v /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/PROKKA_08052019.gff |awk ' { if ($3=="rRNA") print $0 } ' |awk 'BEGIN{OFS="\t"} { if ($7=="+") print $1,$4,$5,"color=green";else print $1,$4,$5,"color=black" } ' > /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/rRNA_ldel.txt

##eggnog

awk -F "\t" ' { if ($21=="L") print $1 } ' ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Lactobacillus_delbrueckii_RMK202_query_seqs.fa.emapper.annotations > /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/Lgenes_names_ldel.txt

rm -r /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/Lgenes_names_ldel_final.txt
for genes in $(cat /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/Lgenes_names_ldel.txt)
do
grep "ID=${genes};" /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/PROKKA_08052019.gff | awk 'BEGIN{OFS="\t"}{print $1,$4,$5}' >> /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/Lgenes_names_ldel_final.txt

done

###transposase from PGAP

grep "transposase" -i ~/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/LD_singu/annot.gff | awk -F "\t" 'BEGIN{OFS="\t"}{print $1,$4,$5}' > /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/transposase_ldel.txt


##======================
##Sterm
##======================

home=/home/vincent/bin/apps/circos-0.69-6/

##karyotyp
${home}/data/karyotype/sterm_rmk202.txt

##example
cp etc/ldel_rmk202_circos.conf etc/sterm_rmk202_circos.conf

#example run
bin/circos -conf etc/sterm_rmk202_circos.conf ; firefox ./circos.svg &


##links
scp -r vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/links_sterm_curated.txt /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/links_sterm.txt

data/nwc_rmk/links_ldel.txt

##CDS
scp -r vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Streptococcus_thermophilus_RMK202/PROKKA_08052019.gff /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/

grep "^#" -v /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/PROKKA_08052019.gff |awk ' { if ($3=="CDS") print $0 } ' |awk 'BEGIN{OFS="\t"} { if ($7=="+") print $1,$4,$5,"color=green";else print $1,$4,$5,"color=black" } ' > /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/cds_sterm.txt

##rRNA

  grep "^#" -v /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/PROKKA_08052019.gff |awk ' { if ($3=="rRNA") print $0 } ' |awk 'BEGIN{OFS="\t"} { if ($7=="+") print $1,$4,$5,"color=green";else print $1,$4,$5,"color=black" } ' > /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/rRNA_sterm.txt

##eggnog

awk -F "\t" ' { if ($21=="L") print $1 } ' ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Streptococcus_thermophilus_RMK202_query_seqs.fa.emapper.annotations > /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/Lgenes_names_sterm.txt

rm -r /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/Lgenes_names_sterm_final.txt
for genes in $(cat /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/Lgenes_names_sterm.txt)
do
grep "ID=${genes};" /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/PROKKA_08052019.gff | awk 'BEGIN{OFS="\t"}{print $1,$4,$5}' >> /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/Lgenes_names_sterm_final.txt

done

###transposase from PGAP

grep "transposase" -i ~/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/annot.gff | awk -F "\t" 'BEGIN{OFS="\t"}{print $1,$4,$5}' > /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/transposase_sterm.txt


```

##PGAP analysis

moving genomes from local (previously downloaded from NCBI submission portal) to big machine

```{bash}

scp /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/PGAP_genomes/* vincent@130.223.51.116:/data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/NCBI_PGAP_FINAL/allRMKs/

```

change names of files

```{bash}

##--------------names of strains
ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Ldel/ |grep "_202" |grep "fasta$" |sed 's/.fasta//g' > /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/names_RMK202_strains.txt
ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/Sterm/ |grep "_202" |grep "fasta$" |sed 's/.fasta//g' >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/names_RMK202_strains.txt

##--------------change name
starterCulturesss=202
IFS=$'\n'
for files in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/NCBI_PGAP_FINAL/allRMKs/ |grep -v ".gb" )
do

echo ${files}
genomeID=$(head -2 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/NCBI_PGAP_FINAL/allRMKs/${files} |tail -1 |awk -F "[,]" '{OFS="\t"}{print $1}'| sed 's/ chromosome//g' |awk -F "strain " '{OFS="\t"}{print $2}')
Species=$(head -2 /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/NCBI_PGAP_FINAL/allRMKs/${files} |tail -1 |awk -F "[ ,]" '{OFS="\t"}{print $3}'|head -c1)
Techhh=$(grep "Sequencing Technology  ::" /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/NCBI_PGAP_FINAL/allRMKs/${files} | sed 's/            Sequencing Technology  :: //g'  |head -1 | sed 's/.*; //g' |head -c1)

echo -e ${genomeID} ${Species} ${Techhh}

cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/NCBI_PGAP_FINAL/allRMKs/${files} /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/NCBI_PGAP_FINAL/allRMKs/${Species}_${Techhh}_${starterCulturesss}_${genomeID}.gb

rm /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/NCBI_PGAP_FINAL/allRMKs/${files}
 echo "---------------------------"
done

###metagenom assembled genomes change name

mv /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/NCBI_PGAP_FINAL/allRMKs//202_LMAG.gb /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/NCBI_PGAP_FINAL/allRMKs//L_M_202_LMAG.gb 

mv /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/NCBI_PGAP_FINAL/allRMKs//202_SMAG.gb /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/NCBI_PGAP_FINAL/allRMKs//S_M_202_SMAG.gb 

```

download the two reference strains (type strains)


```{bash}

mkdir -p /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/TypeStrainsNCBI/{log,genomes}

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/download.file |grep "lactobacillus" -i|grep "DSM"|grep "20072"|tail -1|awk 'BEGIN{FS="\t";OFS="\t"}  {print $20}'| sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >  /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/TypeStrainsNCBI/log/download_gb.txt

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/coreGenomeSNPtree/download.file |grep "streptococcus" -i|grep "19258"|grep "representative genome"|awk 'BEGIN{FS="\t";OFS="\t"}  {print $20}'| sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >> /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/TypeStrainsNCBI/log/download_gb.txt

cd /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/TypeStrainsNCBI/genomes

wget -i /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/TypeStrainsNCBI/log/download_gb.txt
gunzip /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/TypeStrainsNCBI/genomes/*.gz

mv /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/TypeStrainsNCBI/genomes/GCF_010120595.1_ASM1012059v1_genomic.gbff /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/TypeStrainsNCBI/genomes/S_T_typeStrain_19258.gb

mv /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/TypeStrainsNCBI/genomes/GCF_002278095.1_ASM227809v1_genomic.gbff /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/TypeStrainsNCBI/genomes/L_T_typeStrain_20072.gb

```

move all genbank files together

```{bash}
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb
cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/TypeStrainsNCBI/genomes/* /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/
cp /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/NCBI_PGAP_FINAL/allRMKs/* /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/


```

convert to genbank to fasta

```{bash}
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/fasta
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/fasta


for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/ |sed 's/.gb//g')
do
seqret /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/${genomes}.gb /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/fasta/${genomes}.fasta
done

```

convert to genbank to gff

```{bash}
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gff
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gff


for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/ |sed 's/.gb//g')
do
/home/vincent/perl5/bin/bp_genbank2gff.pl -stdout -file /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/${genomes}.gb > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gff/${genomes}.gff
echo -e ${genomes}
done

```

convert genbank to faa

```{bash}
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/faa
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/faa

for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/ |sed 's/.gb//g')
do
python2.7 /home/vincent/apps/genbank2faa.py /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/${genomes}.gb /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/faa/${genomes}.faa
echo -e ${genomes}
done

```


#####pseudogene
Here, I want to create a pseudo gene comparison

```{bash}
short=RMK202
#genus=Lactobacillus
what=pseudogene

##============
##pseudogenes
##============
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/
##--------------------------
##Ldel
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_*.txt
echo -e "Genome\tSpecies\tTechnology\t"${what} > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_${short}.txt

for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/)
do
echo ${genomes}
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 4)
  #short=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 3)
  shortSpeciesss=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 1)
  shortTechnology=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 2)
  
  cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/${genomes} |head -1000 | grep "Pseudo Genes (total)              ::" |head -1| sed 's/Pseudo Genes (total)              :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk -v genomesss="$shortGenome" -v shortSpeciesss="$shortSpeciesss" -v techsss="$shortTechnology" 'BEGIN{OFS="\t"}{print genomesss , shortSpeciesss , techsss,$0}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_${short}.txt

done


```


#####rRNA

```{bash}

##============
##rRNA
##============
what=rRNA
short=RMK202
#genus=Lactobacillus
mkdir -p /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}

rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_*.txt
echo -e "Genome\t"${what} >  /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_${short}.txt

for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/)
do
echo ${genomes}
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 4)
  #short=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 3)
  #shortSpeciesss=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 1)
  #shortTechnology=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 2)
  
  cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/${genomes} |head -1000 | grep "Genes (RNA)                       :: " |head -1| sed 's/Genes (RNA)                       :: //g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk -v genomesss="$shortGenome"  'BEGIN{OFS="\t"}{print genomesss,$0}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_${short}.txt

done
```

#####genes

```{bash}

##============
##genes
##============
what=genes
short=RMK202
#genus=Lactobacillus




rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_*.txt
echo -e "Genome\t"${what} > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_${short}.txt
for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/)
do
echo ${genomes}
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 4)
  #short=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 3)
  #shortSpeciesss=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 1)
  #shortTechnology=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 2)
  
  cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/${genomes} |head -1000 | grep "Genes (total)                     :: " |head -1| sed 's/Genes (total)                     :: //g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |sed 's/,//g'|awk -v genomesss="$shortGenome"  'BEGIN{OFS="\t"}{print genomesss,$0}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_${short}.txt

done


```


#####CRISPR

```{bash}

##============
##CRISPR
##============
what=CRISPR
short=RMK202
#genus=Lactobacillus




rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_*.txt
echo -e "Genome\t"${what} > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_${short}.txt
for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/)
do
echo ${genomes}
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 4)
  #short=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 3)
  #shortSpeciesss=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 1)
  #shortTechnology=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 2)
  
  cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/${genomes} |head -1000 | grep "CRISPR Arrays                     ::" |head -1| sed 's/CRISPR Arrays                     :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |sed 's/,//g'|awk -v genomesss="$shortGenome"  'BEGIN{OFS="\t"}{print genomesss,$0}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_${short}.txt

done


```

#####tRNAs

```{bash}

##============
##tRNAs
##============
what=tRNAs
short=RMK202
#genus=Lactobacillus



rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_*.txt
echo -e "Genome\t"${what} > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_${short}.txt
for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/)
do
echo ${genomes}
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 4)
  #short=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 3)
  #shortSpeciesss=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 1)
  #shortTechnology=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 2)
  
  cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/${genomes} |head -1000 | grep "tRNAs                             ::" |head -1| sed 's/tRNAs                             :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |sed 's/,//g'|awk -v genomesss="$shortGenome"  'BEGIN{OFS="\t"}{print genomesss,$0}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_${short}.txt

done


```

#####ncRNAs

```{bash}

##============
##ncRNAs
##============
what=ncRNAs
short=RMK202
#genus=Lactobacillus



rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_*.txt
echo -e "Genome\t"${what} > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_${short}.txt
for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/)
do
echo ${genomes}
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 4)
  #short=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 3)
  #shortSpeciesss=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 1)
  #shortTechnology=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 2)
  
  cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/${genomes} |head -1000 | grep "ncRNAs                            ::" |head -1| sed 's/ncRNAs                            :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |sed 's/,//g'|awk -v genomesss="$shortGenome"  'BEGIN{OFS="\t"}{print genomesss,$0}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_${short}.txt

done




```

#####transposase

```{bash}

##============
##transposase
##============
what=transposase
short=RMK202
#genus=Lactobacillus
#mkdir -p /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}


rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_*.txt
echo -e "Genome\t"${what} > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_${short}.txt
for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/ )
do
echo ${genomes}
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 4)
  #short=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 3)
  #shortSpeciesss=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 1)
  #shortTechnology=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 2)
  
  #cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/${genomes} |grep "/product=\"5S" 
  #cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/${genomes} |grep "/product=\"16S" 
  #cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/${genomes} |grep "/product=\"23S"
  cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/${genomes} |grep "transposase" -c |awk -v genomesss="$shortGenome"  'BEGIN{OFS="\t"}{print genomesss,$0}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_${short}.txt

done


```



####rRNAs
```{bash}
what=rRNAs
short=RMK202
#genus=Lactobacillus
#mkdir -p /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}


rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_*.txt
echo -e "Genome\t"${what} > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_${short}.txt
for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/ |head -5)
do
echo ${genomes}
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 4)
  #short=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 3)
  #shortSpeciesss=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 1)
  #shortTechnology=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 2)
  
  #cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/${genomes} |grep "/product=\"5S" 
  #cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/${genomes} |grep "/product=\"16S" 
  #cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/${genomes} |grep "/product=\"23S"
  cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/${genomes} |grep "transposase" -c |awk -v genomesss="$shortGenome"  'BEGIN{OFS="\t"}{print genomesss,$0}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_${short}.txt

done
```


#### contig number

```{bash}

##============
##contigNumber
##============
what=contigs
short=RMK202
#genus=Lactobacillus
#mkdir -p /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}


rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_*.txt
echo -e "Genome\t"${what} > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_${short}.txt
for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/ )
do
echo ${genomes}
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 4)
  #short=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 3)
  #shortSpeciesss=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 1)
  #shortTechnology=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 2)
  
  #cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/${genomes} |grep "/product=\"5S" 
  #cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/${genomes} |grep "/product=\"16S" 
  #cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/${genomes} |grep "/product=\"23S"
  cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/${genomes} |grep "^LOCUS" -c |awk -v genomesss="$shortGenome"  'BEGIN{OFS="\t"}{print genomesss,$0}'  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_${short}.txt

done
```

####genome size 

```{bash}

what=genome
short=RMK202
#genus=Lactobacillus
#mkdir -p /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}


rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_*.txt
echo -e "Genome\tSize\tGC" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_${short}.txt
for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/ |sed 's/.gb//g')
do
echo ${genomes}
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 4)

sed '2,${/^>/d}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/fasta/${genomes}.fasta >  /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/fasta/tmp.fasta
#seq_length.py /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/fasta/tmp.fasta
geecee /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/fasta/tmp.fasta /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/fasta/tmp2.txt

Gcsss=$(tail -1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/fasta/tmp2.txt|awk '{print $2}'|sed 's/0.//g')

genomesSizzz=$(seq_length.py /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/fasta/${genomes}.fasta |awk -F '\t' '{sum += $3} END {print sum}')

echo -e ${shortGenome}"\t"${genomesSizzz}"\t"${Gcsss} >>/archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_${short}.txt
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/fasta/tmp*.fasta
done

```


#### plasmid

```{bash}

source activate staramr
#staramr --help

rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/plasmid
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/plasmid
#staramr search -o /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/plasmid/out /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/fasta/*.fasta
cd /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/plasmid
#plasmidfinder.py -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/fasta/L_I_202_11141.fasta

for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/ |sed 's/.gb//g')
do
echo $genomes
#rm -r ${genomes}
mkdir -p ${genomes}
#plasmidfinder.py -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/fasta/${genomes}.fasta -o ${genomes}
done




for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/ |sed 's/.gb//g')
do
#/home/vincent/perl5/bin/bp_genbank2gff.pl -stdout -file /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/${genomes}.gb > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gff/${genomes}.gff
echo -e ${genomes}
plasmidCount=$(grep "plasmid" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gff/${genomes}.gff |cut -f 1|sort|uniq -c |wc -l)
done
##============
##contigNumber
##============
what=plasmid
short=RMK202
#genus=Lactobacillus
#mkdir -p /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}

#rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_*.txt
echo -e "Genome\t"${what} > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_${short}.txt
for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/ |cut -d '.' -f 1 )
do
echo ${genomes}
  shortGenome=$(echo ${genomes}|cut -d '_' -f 4)
  #short=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 3)
  #shortSpeciesss=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 1)
  #shortTechnology=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 2)
  
  #cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/${genomes} |grep "/product=\"5S" 
  #cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/${genomes} |grep "/product=\"16S" 
  #cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/${genomes} |grep "/product=\"23S"
plasmidCount=$(grep "plasmid" -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gff/${genomes}.gff |cut -f 1|sort|uniq -c |wc -l)
echo -e ${shortGenome}"\t"${plasmidCount} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_${short}.txt
done

##---------------------------------checking for plasmid mapping


threads=37
#samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt  ##the file with all sample names
#logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final
#Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta
what=plasmid2
short=RMK202
#genus=Lactobacillus
#mkdir -p /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}

#rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_*.txt
echo -e "Genome\tStermphage\tLdelphage1\tLdelphage2" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_${short}.txt
#for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/ |cut -d '.' -f 1 )
for genomes in $(echo "11141 11142 11143 12104 12105 12107 12109 24776 24777 24778 24779 24780 24781 24782 24783 24798")
do
echo "========================"
echo ${genomes}
  shortGenome=$(echo ${genomes}|cut -d '_' -f 4)
  
  #shortGenome=24780
  bamqccsss=$(echo -e "/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/"${shortGenome}"/bwaMapping2DB/bamqc/genome_results.txt") 
  STermPhage=$(grep ">>>>>>> Coverage per contig" -A 14 ${bamqccsss} |grep "CP046131" |  sed "s/^[ \t]*//" |sed -r '/^\s*$/d' |awk -F "\t" '{print $4-$5}')
  LdelPhage1=$(grep ">>>>>>> Coverage per contig" -A 14 ${bamqccsss} |grep "CP046132" |  sed "s/^[ \t]*//" |sed -r '/^\s*$/d' |awk -F "\t" '{print $4-$5}')
  LdelPhage2=$(grep ">>>>>>> Coverage per contig" -A 14 ${bamqccsss} |grep "CP046133" |  sed "s/^[ \t]*//" |sed -r '/^\s*$/d' |awk -F "\t" '{print $4-$5}')

echo -e ${shortGenome}"\t"${STermPhage}"\t"${LdelPhage1}"\t"${LdelPhage2} #>> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_${short}.txt

  done
  

echo "11141 11142 11143 12104 12105 12107 12109 24776 24777 24778 24779 24780 24781 24782 24783 24798" 

##---------------------------------aligne Ldel genome assemblies to MAG


threads=37
#samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt  ##the file with all sample names
#logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final
Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta
what=plasmid3
short=RMK202
#genus=Lactobacillus
#mkdir -p /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}

#rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_*.txt
echo -e "Genome\tLdelCov\tLdelplasmid1\tLdelplasmid2" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_${short}.txt
for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/ |cut -d '.' -f 1 |grep "^L")
#for genomes in $(echo "11141 11142 11143 12104 12105 12107 12109 24776 24777 24778 24779 24780 24781 24782 24783 24798")
do
echo "========================"
echo ${genomes}
  shortGenome=$(echo ${genomes}|cut -d '_' -f 4)
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/plasmidAlignment/BamQC/${genomes}


  
#bwa mem -x intractg ${Assembly} /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/fasta/${genomes}.fasta  | samtools sort -@${threads} -O BAM -o /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/plasmidAlignment/${genomes}.bam - 



#qualimap bamqc -bam /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/plasmidAlignment/${genomes}.bam -outdir /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/plasmidAlignment/BamQC/${genomes} --java-mem-size=80G

  
  #shortGenome=24780
  bamqccsss=$(echo -e "/archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/plasmidAlignment/BamQC/"${genomes}"/genome_results.txt") 
  STermPhage=$(grep ">>>>>>> Coverage per contig" -A 14 ${bamqccsss} |grep "CP046131" |  sed "s/^[ \t]*//" |sed -r '/^\s*$/d' |awk -F "\t" '{print $4}')
  LdelPhage1=$(grep ">>>>>>> Coverage per contig" -A 14 ${bamqccsss} |grep "CP046132" |  sed "s/^[ \t]*//" |sed -r '/^\s*$/d' |awk -F "\t" '{print $4}')
  LdelPhage2=$(grep ">>>>>>> Coverage per contig" -A 14 ${bamqccsss} |grep "CP046133" |  sed "s/^[ \t]*//" |sed -r '/^\s*$/d' |awk -F "\t" '{print $4}')

echo -e ${shortGenome}"\t"${STermPhage}"\t"${LdelPhage1}"\t"${LdelPhage2} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_${short}.txt

  done


```

####Busco

```{bash}



what=Busco
short=RMK202
#genus=Lactobacillus
#mkdir -p /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}


rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_*.txt
echo -e "Genome\tCompleteness" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_${short}.txt
for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/gb/ |sed 's/.gb//g')
do
echo ${genomes}
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1|cut -d '_' -f 4)

#rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/busco/${shortGenome}
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/busco/${shortGenome}

#busco -m MODE -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/fasta/${genomes}.fasta -o /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/busco/${shortGenome} --auto-lineage-prok

cd /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/busco/${shortGenome}
#
 # python3 /home/vincent/apps/busco/scripts/run_BUSCO.py -c 32 -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes//FINAL_mix_typestrains_RMK202/faa/${genomes}.faa -o ${shortGenome} -l /home/vincent/apps/busco/lactobacillales_odb9 -m prot


  #grep -v "^#" run_${genomesssss}/short_summary_${genomesssss}.txt| grep "C:" -A 7 
  all_INFO=$(grep -v "^#" run_*/short_summary_*.txt| grep "C:" | sed "s/^[ \t]*//"  )
  complet=$(echo ${all_INFO} | awk -F "[,[]" '{print $1}'|sed 's/%//g'|sed 's/C://g' )
  single=$(echo ${all_INFO} | awk -F "[,[]" '{print $2}'|sed 's/%//g'|sed 's/S://g')
  dupli=$(echo ${all_INFO} | awk -F "[,[]" '{print $3}'|sed 's/%]//g'|sed 's/D://g')
  fragm=$(echo ${all_INFO} | awk -F "[,[]" '{print $4}'|sed 's/%//g'|sed 's/F://g')
  Missi=$(echo ${all_INFO} | awk -F "[,[]" '{print $5}'|sed 's/%//g'|sed 's/M://g')
  numbi=$(echo ${all_INFO} | awk -F "[,[]" '{print $6}'|sed 's/n://g')
  
 echo -e ${speciesss}"\t"${genomesssss}"\t"${complet}"\t"${single}"\t"${dupli}"\t"${fragm}"\t"${Missi}"\t"${numbi}  
# >> /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/all_analysis.txt

 echo -e ${shortGenome}"\t"${complet} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/${what}_${short}.txt

done

```


move all analysis local

```{bash}

scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/plasmid3_RMK202.txt /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/

mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/
scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/* /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/

```

####rAnalysis
```{r}
library(readr)
library(plyr)
library(tidyverse)
#source("/home/vincent/Desktop/ScriptRepository/R_functions-master/R_functions/g_legend.R") ##for plotting only legends

##=========================================
##---Pseudo
RMK_pseudogene <- read_delim("/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/pseudogene_RMK202.txt",  "\t", escape_double = FALSE,  trim_ws = TRUE)
RMK_contigs <- read_delim("/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/contigs_RMK202.txt",  "\t", escape_double = FALSE,  trim_ws = TRUE)
RMK_genome <- read_delim("/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/genome_RMK202.txt",  "\t", escape_double = FALSE,  trim_ws = TRUE)
RMK_genes <- read_delim("/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/genes_RMK202.txt",  "\t", escape_double = FALSE,  trim_ws = TRUE)
RMK_CRISPR <- read_delim("/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/CRISPR_RMK202.txt",  "\t", escape_double = FALSE,  trim_ws = TRUE)
RMK_ncRNAs <- read_delim("/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/ncRNAs_RMK202.txt",  "\t", escape_double = FALSE,  trim_ws = TRUE)
RMK_plasmid <- read_delim("/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/plasmid_RMK202.txt",  "\t", escape_double = FALSE,  trim_ws = TRUE)
RMK_plasmid2 <- read_delim("/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/plasmid3_RMK202.txt",  "\t", escape_double = FALSE,  trim_ws = TRUE) %>% dplyr::select(-LdelCov) %>% dplyr::select(-Ldelplasmid2) 

RMK_plasmid2$plasmid <- ifelse(RMK_plasmid2$Ldelplasmid1>0.5,1,0)
RMK_plasmid3 <- RMK_plasmid2 %>% dplyr::select(-Ldelplasmid1)
RMK_plasmid <- RMK_plasmid %>% filter(!Genome %in% RMK_plasmid3$Genome)

RMK_plasmid <- rbind(RMK_plasmid3,RMK_plasmid)

RMK_rRNA <- read_delim("/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/rRNA_RMK202.txt",  "\t", escape_double = FALSE,  trim_ws = TRUE)
RMK_transposase <- read_delim("/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/transposase_RMK202.txt",  "\t", escape_double = FALSE,  trim_ws = TRUE)
RMK_tRNAs <- read_delim("/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/tRNAs_RMK202.txt",  "\t", escape_double = FALSE,  trim_ws = TRUE)

RMK_Busco <- read_delim("/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/Busco_RMK202.txt",  "\t", escape_double = FALSE,  trim_ws = TRUE)

##=========================================
##merge all samples
##=========================================
library(plyr)
library(tidyverse)
library(xtable)
tmp1 <- as.tibble(merge(RMK_pseudogene,RMK_contigs,by="Genome"))
tmp2 <- as.tibble(merge(tmp1,RMK_genome,by="Genome"))
tmp3 <- as.tibble(merge(tmp2,RMK_genes,by="Genome"))
tmp4 <- as.tibble(merge(tmp3,RMK_CRISPR,by="Genome"))
tmp5 <- as.tibble(merge(tmp4,RMK_ncRNAs,by="Genome"))
tmp6 <- as.tibble(merge(tmp5,RMK_plasmid,by="Genome"))
tmp7 <- as.tibble(merge(tmp6,RMK_rRNA,by="Genome"))
tmp8 <- as.tibble(merge(tmp7,RMK_transposase,by="Genome"))
tmp9 <- as.tibble(merge(tmp8,RMK_tRNAs,by="Genome"))
tmp10 <- as.tibble(merge(tmp9,RMK_Busco,by="Genome"))

  RMK_final <- tmp10

  
  RMK_final$Species <- revalue(RMK_final$Species, c("L"="L. delbrueckii", "S"="S. thermophilus"))
  RMK_final$Technology <- revalue(RMK_final$Technology, c("I"="Illumina", "O"="Nanopore&I.", "M"="MAG","T"="Typestrain"))
RMK_final_again <- RMK_final %>% dplyr::select(Species,Genome,Technology,contigs,Size,GC,plasmid,genes,pseudogene,rRNA,tRNAs,transposase,CRISPR,Completeness)
 colnames(RMK_final_again) <- revalue(colnames(RMK_final_again), c("Size"="Genome size","Completeness"="BUSCO completeness","pseudogene"="Pseudogenes","Genome"="Strain", "contigs"="Contig number", "tRNAs"="tRNA"))

 target <- c("13494","13496","13498","13499","24737","24738","24739","24740","24853","24854","24855","13491","13492","13493","13495","13497","13499c1","13499c2","13500","S50","S72","SMAG","19258","11141","11142","11143","12104","12105","12107","12109","24776","24777","24778","24779","24780","24781","24782","24783","24798","LMAG","20072")

 RMK_final_again$`Contig number` <- as.integer(RMK_final_again$`Contig number`)
  RMK_final_again$`Genome size` <- as.integer(RMK_final_again$`Genome size`)
  RMK_final_again$GC <- as.integer(RMK_final_again$GC)
RMK_final_again$plasmid <- as.integer(RMK_final_again$plasmid)
 RMK_final_again$genes <- as.integer(RMK_final_again$genes)
 RMK_final_again$Pseudogenes <- as.integer(RMK_final_again$Pseudogenes)
 RMK_final_again$rRNA <- as.integer(RMK_final_again$rRNA)
 RMK_final_again$tRNA <- as.integer(RMK_final_again$tRNA)
 RMK_final_again$CRISPR <- as.integer(RMK_final_again$CRISPR)
 RMK_final_again$transposase <- as.integer(RMK_final_again$transposase)
 RMK_final_again$`BUSCO completeness` <- as.integer(RMK_final_again$`BUSCO completeness`)
 
 RMK_final_again_final <- RMK_final_again %>% arrange(factor(Strain, levels = target))
xtable(RMK_final_again_final,include.rownames = FALSE)

RMK_final_again %>%  filter(Technology!="Typestrain") %>% group_by(Species) %>% summarize_all(mean)
RMK_final_again %>%  filter(Technology!="Typestrain") %>% group_by(Species) %>% summarize_all(sd)


tsne_plot %>% group_by(density) %>% select(V1, 
    V2) %>% summarize_all(mean)
##--------------------go with this into overleaf--------------------------

library(kableExtra)
knitr::kable(RMK_final_again, "latex", booktabs = TRUE, linesep = "") %>%
    kable_styling(full_width = TRUE, font_size = 9) %>%  as_image()
    # column_spec(1, width = "4cm") %>%
    # save_kable(file = "/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/comparativeGenomics/table.png")

##=========================================
pseudoPlot <- ggplot(Lactobacillus_pseudogene,aes(x=species,y=count))+geom_boxplot()+geom_dotplot(aes(color=location,fill=location),binaxis='y', stackdir='center', dotsize=1)+
  labs(x="",
       y="pseudogene")+
  theme_classic()+
  theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
        #legend.position = c(0.02, 0.98),
          # legend.justification = c("left", "top"),
          # legend.title = element_blank()
          legend.position="none"        
          )
pseudoPlot

##=========================================
##---rRNA

Lactobacillus_rRNA <- read_delim("~/Projects/2019_pilotplant/comparativeGenomics/rRNA/all_rRNA.txt",  "\t", escape_double = FALSE, col_names =c("genome","species","location","count"),  trim_ws = TRUE)

rRNAPlot <- ggplot(Lactobacillus_rRNA,aes(x=species,y=count))+geom_boxplot()+geom_dotplot(aes(color=location,fill=location),binaxis='y', stackdir='center', dotsize=1)+
  labs(x="",
       y="rRNA")+
  theme_classic()+
  theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
        #legend.position = c(0.02, 0.98),
          # legend.justification = c("left", "top"),
          # legend.title = element_blank()
        legend.position="none"
          )
rRNAPlot

##=========================================
##---ncRNAs

Lactobacillus_ncRNAs <- read_delim("~/Projects/2019_pilotplant/comparativeGenomics/ncRNAs/all_ncRNAs.txt",  "\t", escape_double = FALSE, col_names =c("genome","species","location","count"),  trim_ws = TRUE)

ncRNAsPlot <- ggplot(Lactobacillus_ncRNAs,aes(x=species,y=count))+geom_boxplot()+geom_dotplot(aes(color=location,fill=location),binaxis='y', stackdir='center', dotsize=1)+
  labs(x="",
       y="ncRNAs")+
  theme_classic()+
  theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
        legend.position = c(0.02, 0.98),
          legend.justification = c("left", "top"),
          legend.title = element_blank()
          )
ncRNAsPlot

##=========================================
##---genes

Lactobacillus_genes <- read_delim("~/Projects/2019_pilotplant/comparativeGenomics/genes/all_genes.txt",  "\t", escape_double = FALSE, col_names =c("genome","species","location","count"),  trim_ws = TRUE)

genesPlot <- ggplot(Lactobacillus_genes,aes(x=species,y=count))+geom_boxplot()+geom_dotplot(aes(color=location,fill=location),binaxis='y', stackdir='center', dotsize=1)+
  labs(x="",
       y="genes")+
  theme_classic()+
  theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
        legend.position = c(0.02, 0.98),
          legend.justification = c("left", "top"),
          legend.title = element_blank()
          )
genesPlot

##=========================================
##---CRISPR

Lactobacillus_CRISPR <- read_delim("~/Projects/2019_pilotplant/comparativeGenomics/CRISPR/all_CRISPR.txt",  "\t", escape_double = FALSE, col_names =c("genome","species","location","count"),  trim_ws = TRUE)

CRISPRPlot <- ggplot(Lactobacillus_CRISPR,aes(x=species,y=count))+geom_boxplot()+geom_dotplot(aes(color=location,fill=location),binaxis='y', stackdir='center', dotsize=1)+
  labs(x="",
       y="CRISPR")+
  theme_classic()+
  theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
        #legend.position = c(0.02, 0.98),
          # legend.justification = c("left", "top"),
          # legend.title = element_blank()
        legend.position="none"
          )
CRISPRPlot


##=========================================
##---transposase

Lactobacillus_transposase <- read_delim("~/Projects/2019_pilotplant/comparativeGenomics/transposase/all_transposase.txt",  "\t", escape_double = FALSE, col_names =c("genome","species","location","count"),  trim_ws = TRUE)

transposasePlot <- ggplot(Lactobacillus_transposase,aes(x=species,y=count))+geom_boxplot()+geom_dotplot(aes(color=location,fill=location),binaxis='y', stackdir='center', dotsize=1)+
  labs(x="",
       y="transposase")+
  theme_classic()+
  theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
        #legend.position = c(0.02, 0.98),
          # legend.justification = c("left", "top"),
          # legend.title = element_blank()
        legend.position="none"
          )
transposasePlot

##=========================================
##---tRNAs

Lactobacillus_tRNAs <- read_delim("~/Projects/2019_pilotplant/comparativeGenomics/tRNAs/all_tRNAs.txt",  "\t", escape_double = FALSE, col_names =c("genome","species","location","count"),  trim_ws = TRUE)

tRNAsPlot <- ggplot(Lactobacillus_transposase,aes(x=species,y=count))+geom_boxplot()+geom_dotplot(aes(color=location,fill=location),binaxis='y', stackdir='center', dotsize=1)+
  labs(x="",
       y="tRNAs")+
  theme_classic()+
  theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
        #legend.position = c(0.02, 0.98),
          # legend.justification = c("left", "top"),
          # legend.title = element_blank()
        legend.position="none"
          )
tRNAsPlot

##=========================================
##---arrange

legend <- g_legend(genesPlot)
grid.arrange(genesPlot+theme(legend.position="none"),rRNAPlot,tRNAsPlot,legend,pseudoPlot,transposasePlot,CRISPRPlot , ncol=4)


svg("~/Desktop/presenations/my_presentations/20190807_committeeMeeting/figures/ldelGenomes.svg",width=10,height=5)
grid.arrange(genesPlot+theme(legend.position="none"),rRNAPlot,tRNAsPlot,legend,pseudoPlot,transposasePlot,CRISPRPlot , ncol=4)

dev.off()
```



##Appendix


##CG Viewer
changed at 1.10.202

Here, I want to create the CG view files of the assembled genomes
```{bash}

mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/13_CG_view/{Sterm,Ldel}
##===================preperation pseudogenes file

cd /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/

/usr/bin/perl /home/vincent/miniconda3/bin/bp_genbank2gff3.pl /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/L_delbrueckii_RMK202.current.gb
echo -e 'seqname\tsource\tfeature\tstart\tend\tscore\tstrand\tframe' > /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/L_delbrueckii_pseudogenes_vor_cgview.gff
grep "pseudogene"  L_delbrueckii_RMK202.current.gb.gff >> /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/L_delbrueckii_pseudogenes_vor_cgview.gff

##===================preperation prophages file

cd /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/

/usr/bin/perl /home/vincent/miniconda3/bin/bp_genbank2gff3.pl /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/L_delbrueckii_RMK202.current.gb
echo -e 'seqname\tsource\tfeature\tstart\tend\tscore\tstrand\tframe' > /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/L_delbrueckii_pseudogenes_vor_cgview.gff
grep "pseudogene"  L_delbrueckii_RMK202.current.gb.gff >> /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/L_delbrueckii_pseudogenes_vor_cgview.gff


##===================creat file

/home/vincent/Downloads
##-----------------Ldel
mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/13_CG_view/Ldel/

/usr/bin/perl  /home/vincent/apps/cgview/cgview_xml_builder/cgview_xml_builder.pl -sequence /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/L_delbrueckii_RMK202.current.gb -output  /home/vincent/Desktop/Projects/2019_Pilotplan/13_CG_view/Ldel/ldel.file.xml -tick_density 0.5 -gc_content F -genes /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/L_delbrueckii_pseudogenes_vor_cgview.gff 

java -jar -Xmx1500m /home/vincent/apps/cgview/cgview.jar -i /home/vincent/Desktop/Projects/2019_Pilotplan/13_CG_view/Ldel/ldel.file.xml -o /home/vincent/Desktop/Projects/2019_Pilotplan/13_CG_view/Ldel/default_ldel.svg -f svg

##-----------------Ldel
mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/13_CG_view/Sterm/

/usr/bin/perl  cgview_xml_builder.pl -sequence /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/S_thermophilus_RMK202.current.gb  -output /home/vincent/Desktop/Projects/2019_Pilotplan/13_CG_view/Sterm/Sterm.file.xml -tick_density 0.7 -genes /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/L_delbrueckii_pseudogenes_vor_cgview.gff


java -jar -Xmx1500m /home/vincent/apps/cgview/cgview.jar -i /home/vincent/Desktop/Projects/2019_Pilotplan/13_CG_view/Sterm/Sterm.file.xml -o /home/vincent/Desktop/Projects/2019_Pilotplan/13_CG_view/Sterm/default_Sterm.svg -f svg


##===================cgviewer create

##-----------------Ldel


java -jar -Xmx1500m /home/vincent/apps/cgview/cgview.jar -i /home/vincent/Desktop/Projects/2019_Pilotplan/13_CG_view/Ldel/ldel.file.xml -o /home/vincent/Desktop/Projects/2019_Pilotplan/13_CG_view/Ldel/default_ldel.svg -f svg


```

##Circos plot for CRISPR

changed at 1.10.202
Previously we had the PGap annotation analysis before here but I removed it 

Here I do three CIRCOS plots to show the CRISPRs and finally put them together with the phage to show where they actually match onto the phage

Now we go step by step what we actually need:

#####Prokka annoation and pilerCR of start aligned fasta

```{bash}
###-----------------
###---------move files on big machine
###-----------------
mkdir -p /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/{S50,S72}

#/home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/S50/alignment3/S50.fna.fas
scp /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/S50/alignment3/S50.fna.fas vincent@130.223.51.151:/archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/

scp /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/s72_03/alignment2/S72.fna.fas vincent@130.223.51.151:/archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/


###-----------------
###--------PROKKA annotation
###-----------------

rm -r mkdir -p /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/PROKKA
    mkdir -p /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/PROKKA
    
prokka --outdir /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/PROKKA --force --usegenus --genus Streptococcus --locustag mst1 --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/S50.fna.fas
  
rm -r mkdir -p /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/PROKKA
    mkdir -p /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/PROKKA
    
prokka --outdir /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/PROKKA --force --usegenus --genus Streptococcus --locustag mst2 --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/S72.fna.fas


###-----------------
###--------pilerCR
###-----------------

rm -r mkdir -p /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR
    mkdir -p /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR

~/apps/PilerCR/pilercr1.06/pilercr -in /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/S50.fna.fas \
    -out /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out -noinfo\
    -seq /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out.fasta

rm -r mkdir -p /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR
    mkdir -p /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR

~/apps/PilerCR/pilercr1.06/pilercr -in /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/S72.fna.fas \
    -out /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out -noinfo\
    -seq /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out.fasta

###-----------------
###---------move files back local
###-----------------
mkdir -p /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/{S50,S72}

#/home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/S50/alignment3/S50.fna.fas
scp  -r vincent@130.223.51.151:/archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/ /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/S50/alignment3/

scp -r vincent@130.223.51.151:/archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/ /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/s72_03/alignment2/


```


#####2. genome file

```{bash}

home=/home/vincent/bin/apps/circos-0.69-6/

##karyotyp
${home}/data/karyotype/sterm_rmk202.txt


mkdir -p /home/vincent/bin/apps/circos-0.69-6/data/rmk202/


###------------
##meta
###------------
grep "CRISPR" -i ~/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/annot.gff | awk -F "\t" 'BEGIN{OFS="\t"}{print $1,$4,$5}' > /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/transposase_sterm.txt

less  ~/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/*.gbk

grep "CRISPR Arrays" -i ~/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/*.gbk
grep "CRISPR" -i ~/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/*.gbk


##repeat region --> look in pilerCR how many repeats
grep "CRISPR;rpt_type" -i ~/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/annot.gff | awk -F "\t" 'BEGIN{OFS="\t"}{print $1,$4-2000,$5+2000}' > /home/vincent/bin/apps/circos-0.69-6/data/rmk202/metaRMK202_CRISPRarray.txt

grep "CRISPR;rpt_type" -i ~/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/annot.gff | awk -F "\t" 'BEGIN{OFS="\t"}{print $1,$4-20000,$5}' > /home/vincent/bin/apps/circos-0.69-6/data/rmk202/metaRMK202_CRISPRarray_large.txt

##cas-gene endonuclease
grep "endonuclease Cas" -i ~/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/annot.gff | awk -F "[\t;]" 'BEGIN{OFS="\t"}{print $1,$4,$5+20000,$13}' | sed 's/gene_//g' > /home/vincent/bin/apps/circos-0.69-6/data/rmk202/metaRMK202_Cas.txt
#grep "endonuclease Cas" -i ~/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/annot.gff | awk -F "[\t;]" 'BEGIN{OFS="\t"}{print $1,$4,$5,$13}' | sed 's/gene_//g' > /home/vincent/bin/apps/circos-0.69-6/data/rmk202/metaRMK202_Cas.txt

##crispr-associated gene (csm)
grep "protein Csm" -i ~/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/annot.gff | awk -F "[\t;]" 'BEGIN{OFS="\t"}{print $1,$4,$5+20000,$13}' | sed 's/gene_//g' >> /home/vincent/bin/apps/circos-0.69-6/data/rmk202/metaRMK202_Csm.txt

cat /home/vincent/bin/apps/circos-0.69-6/data/rmk202/metaRMK202_Cas.txt /home/vincent/bin/apps/circos-0.69-6/data/rmk202/metaRMK202_Csm.txt

#example run
cd /home/vincent/bin/apps/circos-0.69-6/
cp etc/sterm_rmk202_circos.conf etc/sterm_rmk202_metagenome_circos.conf
bin/circos -conf etc/sterm_rmk202_metagenome_circos.conf ; firefox ./circos.svg &
bin/circos -conf etc/sterm_rmk202_metagenome_circos.conf ; cp ./circos.svg ./meta_rmk202_circos.svg; firefox ./meta_rmk202_circos.svg &


##--------------CRISPR repeats

grep "SUMMARY BY SIMILARITY" -A 15 /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm//S_thermophilus_RMK202.pilarCR_out |grep "S_thermophilus_R"| sed -e 's/^[ \t]*//' |awk -F " " 'BEGIN{OFS="\t"}{print "S_thermophilus_RMK202",$3-2000,$3+$4+20000,$5,$6,$7,$8,$9,"meta_rmk202"}' > /home/vincent/bin/apps/circos-0.69-6/data/rmk202/metaRMK202_CRISPR_Repeat_long.txt

###------------
##mst 1
###------------


##--------------genome mapping

less /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/S50/alignment3/S50.fna_contigs.tab
grep "^contig" /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/S50/alignment3/S50.fna_contigs.tab | awk -F "[\t]" 'BEGIN{OFS="\t"}{print "S_thermophilus_RMK202",$5,$6}' >  /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst1_genome.txt

##--------------CRISPR repeats

grep "SUMMARY BY SIMILARITY" -A 15 /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/S50/alignment3/S50/pilerCR/pilarCR_out |grep "NODE" > /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst1_infoRepeat.txt
id=NODE_20_length_4

rm /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst1_CRISPR_Repeat.txt
rm /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst1_CRISPR_Repeat_long.txt
for id in $(grep "SUMMARY BY SIMILARITY" -A 15 /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/S50/alignment3/S50/pilerCR/pilarCR_out |grep "NODE"| sed -e 's/^[ \t]*//'|awk -F " " 'BEGIN{OFS="\t"}{print $2}')
  do
  ##get location of rmk202 meta
  baselocation=$(grep "${id}" /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/S50/alignment3/S50.fna_contigs.tab |awk -F "\t" 'BEGIN{OFS="\t"}{print $5}')
  ###get new locations
   grep "${id}" /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst1_infoRepeat.txt | sed -e 's/^[ \t]*//' |awk -F " " -v loc="$baselocation"  'BEGIN{OFS="\t"}{print "S_thermophilus_RMK202",loc+$3-2000,loc+$3+$4+2000}' >> /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst1_CRISPR_Repeat.txt
    grep "${id}" /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst1_infoRepeat.txt | sed -e 's/^[ \t]*//' |awk -F " " -v loc="$baselocation"  'BEGIN{OFS="\t"}{print "S_thermophilus_RMK202",loc+$3-2000,loc+$3+$4+20000,$5,$6,$7,$8,$9,"mst1"}' >> /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst1_CRISPR_Repeat_long.txt
  done


##--------------cas
rm /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst1_Cas.txt
for id in $(grep "SUMMARY BY SIMILARITY" -A 15 /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/S50/alignment3/S50/pilerCR/pilarCR_out |grep "NODE"| sed -e 's/^[ \t]*//'|awk -F " " 'BEGIN{OFS="\t"}{print $2}')
  do
  ##get location of rmk202 meta
  baselocation=$(grep "${id}" /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/S50/alignment3/S50.fna_contigs.tab |awk -F "\t" 'BEGIN{OFS="\t"}{print $5}')
  ###get new locations

grep "CRISPR" -i /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/S50/alignment3/S50/PROKKA/PROKKA_09272019.gff | sed 's/;eC//g'|grep "${id}"| awk -F "[\t;]" -v loc="$baselocation"  'BEGIN{OFS="\t"}{print "S_thermophilus_RMK202",loc+$4,loc+$5+20000,$12}' | grep -v "locus_tag=mst1_01056"| sed 's/gene=//g' >> \
/home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst1_Cas.txt
done

#example run
cd /home/vincent/bin/apps/circos-0.69-6/
cp etc/sterm_rmk202_metagenome_circos.conf etc/sterm_rmk202_mst1_circos.conf
bin/circos -conf etc/sterm_rmk202_mst1_circos.conf ; cp ./circos.svg ./mst1_circos.svg; firefox ./mst1_circos.svg &



###------------
##mst 2
###------------


##--------------genome mapping

less  /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/s72_03/alignment2/S72.fna_contigs.tab
grep "^contig" /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/s72_03/alignment2/S72.fna_contigs.tab | awk -F "[\t]" 'BEGIN{OFS="\t"}{print "S_thermophilus_RMK202",$5,$6}'|sort|uniq >  /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst2_genome.txt

##--------------CRISPR repeats

grep "SUMMARY BY SIMILARITY" -A 15 /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/s72_03/alignment2/S72/pilerCR/pilarCR_out |grep "NODE" > /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst2_infoRepeat.txt
id=NODE_20_length_4

rm /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst2_CRISPR_Repeat.txt
rm /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst2_CRISPR_Repeat_long.txt
for id in $(grep "SUMMARY BY SIMILARITY" -A 15 /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/s72_03/alignment2/S72/pilerCR/pilarCR_out |grep "NODE"| sed -e 's/^[ \t]*//'|awk -F " " 'BEGIN{OFS="\t"}{print $2}')
  do
  ##get location of rmk202 meta
  baselocation=$(grep "${id}" /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/s72_03/alignment2/S72.fna_contigs.tab |awk -F "\t" 'BEGIN{OFS="\t"}{print $5}')
  ###get new locations
   grep "${id}" /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst2_infoRepeat.txt | sed -e 's/^[ \t]*//' |awk -F " " -v loc="$baselocation"  'BEGIN{OFS="\t"}{print "S_thermophilus_RMK202",loc+$3-2000,loc+$3+$4+2000}' >> /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst2_CRISPR_Repeat.txt
    grep "${id}" /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst2_infoRepeat.txt | sed -e 's/^[ \t]*//' |awk -F " " -v loc="$baselocation"  'BEGIN{OFS="\t"}{print "S_thermophilus_RMK202",loc+$3-2000,loc+$3+$4+20000,$5,$6,$7,$8,$9,"mst2"}' >> /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst2_CRISPR_Repeat_long.txt
  done


##--------------cas
rm /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst2_Cas.txt
for id in $(grep "SUMMARY BY SIMILARITY" -A 15 /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/s72_03/alignment2/S72/pilerCR/pilarCR_out |grep "NODE"| sed -e 's/^[ \t]*//'|awk -F " " 'BEGIN{OFS="\t"}{print $2}')
  do
  ##get location of rmk202 meta
  baselocation=$(grep "${id}" /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/s72_03/alignment2/S72.fna_contigs.tab |awk -F "\t" 'BEGIN{OFS="\t"}{print $5}')
  ###get new locations

grep "CRISPR" -i /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/s72_03/alignment2/S72/PROKKA/PROKKA_09272019.gff | sed 's/;eC//g'|grep "${id}"| awk -F "[\t;]" -v loc="$baselocation"  'BEGIN{OFS="\t"}{print "S_thermophilus_RMK202",loc+$4,loc+$5+20000,$12}' | grep -v "locus_tag=mst2_01056"| sed 's/gene=//g' >> \
/home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst2_Cas.txt
done

#example run
cd /home/vincent/bin/apps/circos-0.69-6/
cp etc/sterm_rmk202_mst1_circos.conf etc/sterm_rmk202_mst2_circos.conf
bin/circos -conf etc/sterm_rmk202_mst2_circos.conf ; cp ./circos.svg ./mst2_circos.svg; firefox ./mst2_circos.svg  &


grep "CRISPR" -i /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/PROKKA_08052019.gff
grep "CRISPR" -i /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/PROKKA_08052019.gbk




##===============
###=======meta analysis
##===============


##----------CRISPR_array

cat /home/vincent/bin/apps/circos-0.69-6/data/rmk202/metaRMK202_CRISPR_Repeat_long.txt \
  /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst1_CRISPR_Repeat_long.txt \
  /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst2_CRISPR_Repeat_long.txt >   /home/vincent/bin/apps/circos-0.69-6/data/rmk202/all_CRISPR_Repeat_long.txt


##----------CRISPR_array CD-HIT
mkdir -p /home/vincent/bin/apps/circos-0.69-6/data/rmk202/CRISPRspacers
cut -f 7,8 /home/vincent/bin/apps/circos-0.69-6/data/rmk202/all_CRISPR_Repeat_long.txt | sed 's/+\t/>CRISPR_spacer\n/g' > /home/vincent/bin/apps/circos-0.69-6/data/rmk202/CRISPRspacers/spacers.fasta


cd-hit -i /home/vincent/bin/apps/circos-0.69-6/data/rmk202/CRISPRspacers/spacers.fasta \
    -o q -c 0.9 -M 30000 -T 37
  
##----------mafft

cp /home/vincent/bin/apps/circos-0.69-6/data/rmk202/CRISPRspacers/spacers.fasta /home/vincent/bin/apps/circos-0.69-6/data/rmk202/CRISPRspacers/spacers_array1.fasta
cp /home/vincent/bin/apps/circos-0.69-6/data/rmk202/CRISPRspacers/spacers.fasta /home/vincent/bin/apps/circos-0.69-6/data/rmk202/CRISPRspacers/spacers_array2.fasta
cp /home/vincent/bin/apps/circos-0.69-6/data/rmk202/CRISPRspacers/spacers.fasta /home/vincent/bin/apps/circos-0.69-6/data/rmk202/CRISPRspacers/spacers_array3.fasta

mafft --clustalout --auto /home/vincent/bin/apps/circos-0.69-6/data/rmk202/CRISPRspacers/spacers_array1.fasta > /home/vincent/bin/apps/circos-0.69-6/data/rmk202/CRISPRspacers/spacers_array1.msa

##----------Cas


```

```{r}
library(knitr)
library(readr)
spacers_array1 <- read_table2("/home/vincent/bin/apps/circos-0.69-6/data/rmk202/CRISPRspacers/spacers_array1.msa",  col_names = FALSE, skip = 1)
kable(spacers_array1)
```

##Phage uniprot annotation

```{bash}

mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/phage_uniprot

##---bring to bigmachine
scp /home/vincent/Desktop/Projects/2019_Pilotplan/09_annoation/rastServer/Streptococcus_thermophilus_ViSo-RMK202_b.faa vincent@130.223.51.151:/archiv/Projects/2019_pilotplant/02_annotation/phage_uniprot/

##---bring to bigmachine

diamond blastp --threads 32 --max-target-seqs 1 --db /archiv/TREMBL_database/20190710/uniprot_trembl.dmnd --query /archiv/Projects/2019_pilotplant/02_annotation/phage_uniprot/Streptococcus_thermophilus_ViSo-RMK202_b.faa --out /archiv/Projects/2019_pilotplant/02_annotation/phage_uniprot/Streptococcus_thermophilus_ViSo-RMK202_b.txt


##---bring local
scp -r vincent@130.223.51.151:/archiv/Projects/2018_Culturomics/12_CRISPR_isolates/cdhit/20190312/Streptococcus_thermophilus/clustering_results.txt ~/Desktop/Projects/2018_culturomics/CRISPR/

```

#### Snippy between isolates

Here, I test the SNP calling between different assembled genomes on the core genome. 
this should be feasable with [snippy](https://github.com/tseemann/snippy).
See the following section in the snippy github:
Finding SNPs between contigs

If you supply snippy with a .gbk reference instead of the .fna than it also does a snpEff analysis


```{bash}

##===============
##Sterm rmk202
##===============
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/snippy/{log,sterm,ldel}

echo -e "S72"'\t'"/archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/S72.fna" > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/snippy/log/log_sterm.txt
echo -e "S50"'\t'"/archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/S50.fna" >> /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/snippy/log/log_sterm.txt


#/archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/S72.fna
#/archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/S50.fna
#/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Streptococcus_thermophilus_RMK202.fasta

<<<<<<<<<<<<<<
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/snippy/sterm
cd /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/snippy/sterm

snippy-multi  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/snippy/log/log_sterm.txt \
  --cpus 32 --ram 100 \
  --ref /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Streptococcus_thermophilus_RMK202.fasta \
  --tmpdir /Fast/tmp/


snippy --outdir 'S72' --ctgs '/archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/S72.fna' --cpus 32 --ram 100 --ref /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Streptococcus_thermophilus_RMK202.fasta --tmpdir /Fast/tmp/
snippy --outdir 'S50' --ctgs '/archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/S50.fna' --cpus 32 --ram 100 --ref /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Streptococcus_thermophilus_RMK202.fasta --tmpdir /Fast/tmp/

snippy-core --ref 'S50/ref.fa' S50 S72

 snippy-clean_full_aln core.full.aln > clean.full.aln
 run_gubbins.py -p gubbins clean.full.aln
 snp-sites -c gubbins.filtered_polymorphic_sites.fasta > clean.core.aln
 FastTree -gtr -nt clean.full.aln > clean.core.tree


##===============
##Ldel
##===============

```


####Prophage detection (Virsorter)

For phage and prophage detection I use Virsorter from the (Sullivan lab)[http://u.osu.edu/viruslab/]

```{bash}


for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
do
#sample=1
threads=35
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${sample}/


echo "========================================="
echo $sample
echo "========================================="



  rm -r /home/vincent/Projects/2019_pilotplant/07_Virsorter/${sample}
  mkdir -p /home/vincent/Projects/2019_pilotplant/07_Virsorter/${sample}
 
    source activate virsorter
    wrapper_phage_contigs_sorter_iPlant.pl \
    -f $Overall_output_directory/08_eightFreebayes/freebayesOuput/_eightFreebayes.fasta \
    --db 1 \
    --no_c \
    --wdir /home/vincent/Projects/2019_pilotplant/07_Virsorter/${sample} --ncpu ${threads} --data-dir /archiv/VirSorter/virsorter-data

grep -v "^#" /home/vincent/Projects/2019_pilotplant/07_Virsorter/${sample}/VIRSorter_global-phage-signal.csv 
 
 
 done #samples

##look at files


for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
do
echo "========================================="
echo $sample
echo "========================================="


cat /home/vincent/Projects/2019_pilotplant/07_Virsorter/${sample}/VIRSorter_global-phage-signal.csv 
 
 
 done #samples

##extract single fasta from multifasta (multifasta2fasta)

samtools faidx th_K1_8h_metagenome.fasta contig_6 > contig_6.fasta
```

####PHASTER
Here, I also used (PHASTER)[http://phaster.ca/] for prophage/phage detection.

```{bash}

##make Bed out of PHASTER output
mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/08_phaster/phasterBED


for name in $(echo "di_K1_10h" "di_K2_6h" "th_K1_8h" "th_K2_8h")
do

 echo "------------"
 echo ${name}
  echo "------------"

grep ">" /home/vincent/Desktop/Projects/2019_Pilotplan/08_phaster/${name}/phage_regions.fna  |awk -F "\t" '{print $2}'| sed 's/ //g' |awk -F[,:-] 'BEGIN{OFS="\t"}{print $1,$11,$12}' > /home/vincent/Desktop/Projects/2019_Pilotplan/08_phaster/phasterBED/${name}.bed

cat /home/vincent/Desktop/Projects/2019_Pilotplan/08_phaster/phasterBED/${name}.bed
done

##move to big machine

###--------------on local
name=di_K1_10h
scp /home/vincent/Desktop/Projects/2019_Pilotplan/08_phaster/phasterBED/${name}.bed vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${name}/

name=th_K1_8h
scp /home/vincent/Desktop/Projects/2019_Pilotplan/08_phaster/phasterBED/${name}.bed vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${name}/

name=di_K2_6h
scp /home/vincent/Desktop/Projects/2019_Pilotplan/08_phaster/phasterBED/${name}.bed vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${name}/

name=th_K2_8h
scp /home/vincent/Desktop/Projects/2019_Pilotplan/08_phaster/phasterBED/${name}.bed vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${name}/


```



######KEGG annotation
Currently I use the following annotation forms in inform myself:
- CazY for carbohydrate metabolic enzymes (e.g. lipolyse)
- KEGG_enzymes(EC) for enymes (e.g. Hydrolase(very broad--> no specific pathway))
- KEGG_pathway for most genes (e.g. metabolic, transporters,...) [help](https://www.genome.jp/kegg-bin/get_htext#A1)
- KEGG_ko for Transposons

I will the annotation in the order listed above. 


```{r}
library(KEGGREST)

annotation_sterm_snps_high <- annotation_sterm_snps_01[annotation_sterm_snps_01$significance=="HIGH",]
annotation_sterm_snps_high


##number of unananoted genes
table(!is.na(annotation_sterm_snps_high$KEGG_ko))

##===========
###KEGG ko
##===========
annotation_sterm_snps_high_withko <- annotation_sterm_snps_high[!is.na(annotation_sterm_snps_high$KEGG_ko),]
annotation_sterm_snps_high_woko <- annotation_sterm_snps_high[is.na(annotation_sterm_snps_high$KEGG_ko),]

annotation_sterm_snps_high_kegg_02 <- separate_rows(annotation_sterm_snps_high_withko,KEGG_ko,sep = ",",)

# count_keggs <- (sort(table(annotation_sterm_snps_high_kegg_02$KEGG_ko),decreasing = TRUE)) %>% gsub("ko:","",.)
# keggss <- names(sort(table(annotation_sterm_snps_high_kegg_02$KEGG_ko),decreasing = TRUE)) %>% gsub("ko:","",.)
# count_keggs <- (annotation_sterm_snps_high_kegg_02$KEGG_ko) %>% gsub("ko:","",.)

keggss <- (annotation_sterm_snps_high_kegg_02$KEGG_ko) %>% gsub("ko:","",.)


keggnames <- data.frame()
i<- 10
for (i in 1:length(keggss)) {
  keggsss <- keggGet(keggss[i])[[1]]
def_i <- keggsss$DEFINITION  
pathway_contains <- is.null(keggsss$PATHWAY)
  print(paste0(def_i," AND ",keggss[i]))
 keggnames[i,1:3] <- c(keggss[i],def_i,pathway_contains) 
  
}

colnames(keggnames) <- c("KEGG","DEFINITION","pathway_contains")
# keggnames$count <- count_keggs
keggnames$KEGG_original <- gsub("K","ko:K",keggnames$KEGG)
keggnames$GeneName <- separate_rows(annotation_sterm_snps_high_withko,KEGG_ko,sep = ",",)$finalName

###transposase associated KEGG
keggnames[grep("transposase",keggnames$DEFINITION),]


sort(table(keggnames$DEFINITION),decreasing = TRUE)
sort(table(keggnames$pathway_contains),decreasing = TRUE)
sort(table(keggnames$GeneName),decreasing = TRUE)



##===========
###ENYME (EC-NUMBER)
##===========
annotation_sterm_snps_high_withEC <- annotation_sterm_snps_high[!is.na(annotation_sterm_snps_high$EC),]
annotation_sterm_snps_high_woEC <- annotation_sterm_snps_high[is.na(annotation_sterm_snps_high$EC),]

annotation_sterm_snps_high_withEC_02 <- separate_rows(annotation_sterm_snps_high_withEC,EC,sep = ",",)$EC

enzymenames <- data.frame()
 i<- 1
for (i in 1:length(annotation_sterm_snps_high_withEC_02)) {
  keggsss <- keggGet(annotation_sterm_snps_high_withEC_02[i])[[1]]
NAME_i <- keggsss$NAME[1]
def_i <- keggsss$CLASS[1]
def_i_lowest <- keggsss$CLASS[length(keggsss$CLASS)]
enzymenames[i,1:4] <- c(annotation_sterm_snps_high_withEC_02[i],NAME_i,def_i,def_i_lowest) 
  
}

colnames(enzymenames) <- c("EC","NAME","CLASS","lowestCLASS")
enzymenames$GeneName <- separate_rows(annotation_sterm_snps_high_withEC,EC,sep = ",",)$finalName

sort(table(enzymenames$CLASS),decreasing = TRUE)
sort(table(enzymenames$EC),decreasing = TRUE)
sort(table(enzymenames$NAME),decreasing = TRUE)
sort(table(enzymenames$lowestCLASS),decreasing = TRUE)
##===========
###Pathway
##===========
listDatabases()
table(!is.na(annotation_sterm_snps_high$KEGG_Pathway))
annotation_sterm_snps_high_withPathWay <- annotation_sterm_snps_high[!is.na(annotation_sterm_snps_high$KEGG_Pathway),]
annotation_sterm_snps_high_woPathWay <- annotation_sterm_snps_high[is.na(annotation_sterm_snps_high$KEGG_Pathway),]

annotation_sterm_snps_high_kegg_02 <- separate_rows(annotation_sterm_snps_high_withPathWay,KEGG_Pathway,sep = ",",)

nrow(annotation_sterm_snps_high_withPathWay)
nrow(annotation_sterm_snps_high_kegg_02)

keggss <- annotation_sterm_snps_high_kegg_02$KEGG_Pathway

Pathwaynames <- data.frame()
i<- 1
for (i in 1:length(keggss)) {
  keggsss <- keggGet(keggss[i])[[1]]
  name_i <- keggsss$NAME
def_i <- ifelse(is.null(keggsss$CLASS),NA,keggsss$CLASS)
pathway_contains <- ifelse(is.null(keggsss$PATHWAY_MAP),NA,as.vector(keggsss$PATHWAY_MAP))
  print(paste0(name_i ," ||| ",def_i," ||| ",pathway_contains," ||| ",keggss[i]))
 Pathwaynames[i,1:4] <- c(keggss[i],keggsss$NAME,def_i,pathway_contains) 
  
}

colnames(Pathwaynames) <- c("KEGG","NAME","CLASS","pathway")
Pathwaynames$GeneName <- separate_rows(annotation_sterm_snps_high_withPathWay,KEGG_Pathway,sep = ",",)$finalName

table(Pathwaynames[grep("Metabolism",Pathwaynames$CLASS),]$CLASS)
Pathwaynames[(Pathwaynames$CLASS=="Metabolism; Carbohydrate metabolism")==TRUE,]
sort(table(Pathwaynames[grep("Metabolism; Carbohydrate metabolism",Pathwaynames$CLASS),"NAME"]),decreasing = TRUE)
sort(table(Pathwaynames[grep("Environmental Information Processing; Membrane transport",Pathwaynames$CLASS),"NAME"]),decreasing = TRUE)
sort(table(Pathwaynames[grep("Metabolism; Xenobiotics biodegradation and metabolism",Pathwaynames$CLASS),"NAME"]),decreasing = TRUE) ##not needed?
sort(table(Pathwaynames[grep("Metabolism; Nucleotide metabolism",Pathwaynames$CLASS),"NAME"]),decreasing = TRUE)


namess_extract <- Pathwaynames[grep("Pyrimidine metabolism",Pathwaynames$NAME),"GeneName"]

Pathwaynames[Pathwaynames$GeneName %in% namess_extract,]
keggnames[keggnames$GeneName %in% namess_extract,]
 

# sort(table(Pathwaynames$pathway),decreasing = TRUE)
sort(table(Pathwaynames$CLASS),decreasing = TRUE)
sort(table(Pathwaynames$NAME),decreasing = TRUE)

##===========
###CazY
##===========

annotation_sterm_snps_high[!is.na(annotation_sterm_snps_high$CAZy),]

```



######Go-Analysis

```{r}

##================
###prepare for topGO analysis
##================

library(readr)
library(Rgraphviz)
library(topGO)

##------allgenes
geneID2GO_ldel <- readMappings(file = "~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog_GO/Lactobacillus_delbrueckii_RMK202_pgap_query_seqs.fa.emapper.annotations") 
geneID2GO_sterm <- readMappings(file = "~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog_GO/Streptococcus_thermophilus_RMK202_pgap_query_seqs.fa.emapper.annotations") 
length(geneID2GO_sterm)

##============
##topGO 
##============
##------without modifier snp genes


geneUniverse <- names(geneID2GO_sterm) 
genesOfInterest <- as.character(annotation_sterm_snps_nonModifier$finalName)
length(genesOfInterest)


table(genesOfInterest)

genesOfInterest <- genesOfInterest[genesOfInterest %in% geneUniverse]

 geneList <- factor(as.integer(geneUniverse %in% genesOfInterest))
 names(geneList) <- geneUniverse
 
 
 myGOdata <- new("topGOdata", description="My project", ontology="BP", allGenes=geneList,  annot = annFUN.gene2GO, gene2GO = geneID2GO)
 
 sg <- sigGenes(myGOdata)
 str(sg)
numSigGenes(myGOdata)
resultFisher <- runTest(myGOdata, algorithm="classic", statistic="fisher")
resultFisher

allRes <- GenTable(myGOdata, classicFisher = resultFisher, orderBy = "resultFisher", ranksOf = "classicFisher", topNodes = 200)
allRes

 showSigOfNodes(myGOdata, score(resultFisher), firstSigNodes = 5, useInfo ='all')
 printGraph(myGOdata, resultFisher, firstSigNodes = 5, fn.prefix = "tGO", useInfo = "all", pdfSW = TRUE)
 
 
 length(usedGO(myGOdata))
 
 ##------high and modereate significance mutations
 
annotation_sterm_snps_MODERATE_HIGH <- annotation_sterm_snps_01[annotation_sterm_snps_01$significance=="MODERATE" | annotation_sterm_snps_01$significance=="HIGH",]
table(annotation_sterm_snps_MODERATE_HIGH$significance)
genesOfInterest <- as.character(annotation_sterm_snps_MODERATE_HIGH$finalName)
genesOfInterest <- genesOfInterest[genesOfInterest %in% geneUniverse]

geneList <- factor(as.integer(geneUniverse %in% genesOfInterest))
names(geneList) <- geneUniverse
 
myGOdata <- new("topGOdata", description="My project", ontology="BP", allGenes=geneList,  annot = annFUN.gene2GO, gene2GO = geneID2GO)
 
sg <- sigGenes(myGOdata)
numSigGenes(myGOdata)

resultFisher <- runTest(myGOdata, algorithm="classic", statistic="fisher")
resultFisher

allRes <- GenTable(myGOdata, classicFisher = resultFisher, orderBy = "resultFisher", ranksOf = "classicFisher", topNodes = 200)
allRes

##------high significance mutations
 
annotation_sterm_snps_HIGH <- annotation_sterm_snps_01[annotation_sterm_snps_01$significance=="HIGH",]
table(annotation_sterm_snps_HIGH$significance)
genesOfInterest <- as.character(annotation_sterm_snps_HIGH$finalName)
genesOfInterest <- genesOfInterest[genesOfInterest %in% geneUniverse]

geneList <- factor(as.integer(geneUniverse %in% genesOfInterest))
names(geneList) <- geneUniverse
 
myGOdata <- new("topGOdata", description="My project", ontology="BP", allGenes=geneList,  annot = annFUN.gene2GO, gene2GO = geneID2GO)
 
sg <- sigGenes(myGOdata)
numSigGenes(myGOdata)

resultFisher <- runTest(myGOdata, algorithm="classic", statistic="fisher")
resultFisher

allRes <- GenTable(myGOdata, classicFisher = resultFisher, orderBy = "resultFisher", ranksOf = "classicFisher", topNodes = 200)
allRes


```

##### disentengle Ldel

```{r}



###-----------
##plot all SNPs that occur in one sample
###-----------

###---------change name
table(sample_relative_wide$X.CHROM)
sample_relative_wide_renamed <- sample_relative_wide
# colnames(sample_relative_wide)
colnames(sample_relative_wide_renamed)


###----------not explained by isolates or reference
colMeans(only_meta_ldel[2:6],na.rm = TRUE) 
# meta_ldel
meta_ldel_notExplained <- gather(only_meta_ldel, sample, Snps, "Lyo_202_2012":"Versand_202", factor_key=TRUE,na.rm = TRUE) 
meta_ldel_notExplained$grouping <- "not Explained\nbyGenomes"
colMeans(only_meta_ldel[2:6],na.rm = TRUE) ##EXPLAINED by unknown

###----------explained by isolates

Ldel_L99_snps <- sample_relative_wide_renamed[which((sample_relative_wide_renamed$mst8>0.9) %in% TRUE),c("site","Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202")]
colMeans(Ldel_L99_snps[2:6],na.rm = TRUE)
Ldel_mst8_snps_Explained <- gather(Ldel_L99_snps, sample, Snps, "Lyo_202_2012":"Versand_202", factor_key=TRUE,na.rm = TRUE) 
Ldel_mst8_snps_Explained$grouping <- "mst8"

Ldel_L80_snps <- sample_relative_wide_renamed[which((sample_relative_wide_renamed$mst5>0.9) %in% TRUE),c("site","Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202")];Ldel_L80_snps
colMeans(Ldel_L80_snps[2:6],na.rm = TRUE)
Ldel_mst5_snps_Explained <- gather(Ldel_L80_snps, sample, Snps, "Lyo_202_2012":"Versand_202", factor_key=TRUE,na.rm = TRUE) 
Ldel_mst5_snps_Explained$grouping <- "mst5";Ldel_mst5_snps_Explained

Ldel_L71_snps <- sample_relative_wide_renamed[which((sample_relative_wide_renamed$mst4>0.9) %in% TRUE),c("site","Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202")];Ldel_L71_snps
colMeans(Ldel_L71_snps[2:6],na.rm = TRUE)
Ldel_mst4_snps_Explained <- gather(Ldel_L71_snps, sample, Snps, "Lyo_202_2012":"Versand_202", factor_key=TRUE,na.rm = TRUE) 
Ldel_mst4_snps_Explained$grouping <- "mst4";Ldel_mst4_snps_Explained

Ldel_L70_snps <- sample_relative_wide_renamed[which((sample_relative_wide_renamed$mst3>0.9) %in% TRUE),c("site","Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202")];Ldel_L70_snps
colMeans(Ldel_L70_snps[2:6],na.rm = TRUE)
Ldel_mst3_snps_Explained <- gather(Ldel_L70_snps, sample, Snps, "Lyo_202_2012":"Versand_202", factor_key=TRUE,na.rm = TRUE) 
Ldel_mst3_snps_Explained$grouping <- "mst3";Ldel_mst3_snps_Explained

Ldel_L108_snps <- sample_relative_wide_renamed[which((sample_relative_wide_renamed$mst7>0.9) %in% TRUE),c("site","Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202")];Ldel_L108_snps
colMeans(Ldel_L108_snps[2:6],na.rm = TRUE)
Ldel_mst7_snps_Explained <- gather(Ldel_L108_snps, sample, Snps, "Lyo_202_2012":"Versand_202", factor_key=TRUE,na.rm = TRUE) 
Ldel_mst7_snps_Explained$grouping <- "mst7";Ldel_mst7_snps_Explained

Ldel_L35_snps <- sample_relative_wide_renamed[which((sample_relative_wide_renamed$mst9>0.9) %in% TRUE),c("site","Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202")];Ldel_L35_snps
colMeans(Ldel_L35_snps[2:6],na.rm = TRUE)
Ldel_mst9_snps_Explained <- gather(Ldel_L35_snps, sample, Snps, "Lyo_202_2012":"Versand_202", factor_key=TRUE,na.rm = TRUE) 
Ldel_mst9_snps_Explained$grouping <- "mst9";Ldel_mst9_snps_Explained

Ldel_L104_snps <- sample_relative_wide_renamed[which((sample_relative_wide_renamed$mst6>0.9) %in% TRUE),c("site","Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202")];Ldel_L104_snps
colMeans(Ldel_L104_snps[2:6],na.rm = TRUE)
Ldel_mst6_snps_Explained <- gather(Ldel_L104_snps, sample, Snps, "Lyo_202_2012":"Versand_202", factor_key=TRUE,na.rm = TRUE) 
Ldel_mst6_snps_Explained$grouping <- "mst6";Ldel_mst6_snps_Explained

Ldel_L44_snps <- sample_relative_wide_renamed[which((sample_relative_wide_renamed$mst10>0.9) %in% TRUE),c("site","Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202")];Ldel_L44_snps
colMeans(Ldel_L44_snps[2:6],na.rm = TRUE)
Ldel_mst10_snps_Explained <- gather(Ldel_L44_snps, sample, Snps, "Lyo_202_2012":"Versand_202", factor_key=TRUE,na.rm = TRUE) 
Ldel_mst10_snps_Explained$grouping <- "mst10";Ldel_mst10_snps_Explained


###----------merge different samples

alltogether_SNPs <- rbind(meta_ldel_notExplained,Ldel_mst3_snps_Explained,Ldel_mst4_snps_Explained,Ldel_mst5_snps_Explained,Ldel_mst6_snps_Explained,Ldel_mst7_snps_Explained,Ldel_mst8_snps_Explained,Ldel_mst9_snps_Explained,Ldel_mst10_snps_Explained)

table(alltogether_SNPs$grouping)
alltogether_SNPs$grouping = factor(alltogether_SNPs$grouping, levels=c("not Explained\nbyGenomes","mst3","mst7","mst6","mst9","mst4","mst8","mst5","mst10"))

###----------boxplot


  pall_sterm_CORE_box <- ggplot(alltogether_SNPs,aes(x=grouping,y=Snps,color=sample,fill=sample))+ geom_boxplot()+
       # facet_grid(group~.)+
   labs("",
         x="",
         # y="",
          y="all")+
       # scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="right",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
pall_sterm_CORE_box

###----------look at SNP frequency 
# table(sample_relative_wide_renamed$X.CHROM)
sample_relative_wide_renamed_Ldel <- sample_relative_wide_renamed[which(sample_relative_wide_renamed$X.CHROM=="L_delbrueckii_RMK202"),]
sample_relative_wide_frequency <- sample_relative_wide_renamed_Ldel %>% select(site,mst3,mst7,mst6,mst9,mst4,mst8,mst5,mst10) %>% gather(., sample, Snps, "mst3","mst7","mst6","mst9","mst4","mst8","mst5","mst10", factor_key=TRUE,na.rm = TRUE) 

ggplot(sample_relative_wide_frequency,aes(x=Snps))+geom_histogram()+facet_grid(sample~.,scales = "free")+
  lims(x=c(0.1,1))+
  theme_classic()

###----------frequeny in Isolate vs. frequency in meta
meta_vs_iso <- data.frame()

for (meta in c("Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202")) {
  
  
  for (iso in c("mst3","mst7","mst6","mst9","mst4","mst8","mst5","mst10")) {
    
    grouping <- ifelse(iso %in% c("mst3","mst7","mst6","mst9","mst4"),"group1","group2")
    
    tmp <- data.frame(site=sample_relative_wide_renamed_Ldel$site,metaName=meta,isoName=iso,groupingName=grouping,metaSNPs=sample_relative_wide_renamed_Ldel[,meta],isoSNPs=sample_relative_wide_renamed_Ldel[,iso])
    meta_vs_iso <- rbind(meta_vs_iso,tmp)
    
  }

}##meta

ggplot(meta_vs_iso,aes(x=isoSNPs,y=metaSNPs,color=isoName))+geom_point()+theme_classic()
ggplot(meta_vs_iso,aes(x=isoSNPs,y=metaSNPs,color=groupingName))+geom_point()+theme_classic()
ggplot(meta_vs_iso,aes(x=isoSNPs,y=metaSNPs,color=metaName))+geom_point()+theme_classic()


##make NA to zero
meta_vs_iso_cleaned <- meta_vs_iso
# meta_vs_iso_cleaned <- meta_vs_iso[!is.na(meta_vs_iso$metaSNPs),]
# meta_vs_iso_cleaned <- meta_vs_iso_cleaned[!is.na(meta_vs_iso_cleaned$isoSNPs),]

meta_vs_iso_cleaned[c("metaSNPs", "isoSNPs")][is.na(meta_vs_iso_cleaned[c("metaSNPs", "isoSNPs")])] <- 0


meta_vs_iso_cleaned <- meta_vs_iso_cleaned[which(meta_vs_iso_cleaned$metaSNPs>"0" |meta_vs_iso_cleaned$isoSNPs>"0"),]

# ggplot(meta_vs_iso_cleaned)+stat_density2d(aes(x=isoSNPs,y=metaSNPs))+theme_classic()
# ggplot(meta_vs_iso_cleaned,aes(x=isoSNPs,y=metaSNPs))+geom_hex(binwidth = c(0.02, 0.02),aes(fill=log(..count..)))+theme_classic()
# ggplot(meta_vs_iso_cleaned,aes(x=isoSNPs,y=metaSNPs))+geom_hex(binwidth = c(0.02, 0.02))+theme_classic()
ggplot(meta_vs_iso_cleaned,aes(x=isoSNPs,y=metaSNPs))+geom_hex(binwidth = c(0.03, 0.03),aes(fill=log(..count..)))+theme_classic()

###----------summed frequeny in Isolate vs. frequency in meta

sample_relative_wide_renamed_Ldel_rowsumsIsolates <- sample_relative_wide_renamed_Ldel %>% select(site,Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202)
  sample_relative_wide_renamed_Ldel_rowsumsIsolates$rowSumsIsolates <- rowSums(sample_relative_wide_renamed_Ldel[,c("mst3","mst7","mst6","mst9","mst4","mst8","mst5","mst10")],na.rm = TRUE)
  sample_relative_wide_renamed_Ldel_rowsumsIsolates_long <- gather(sample_relative_wide_renamed_Ldel_rowsumsIsolates, sample, Snps, "Lyo_202_2012":"Versand_202", factor_key=TRUE,na.rm = TRUE) 

 
sample_relative_wide_renamed_Ldel_rowsumsIsolates_long[c("rowSumsIsolates", "Snps")][is.na(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long[c("rowSumsIsolates", "Snps")])] <- 0


sample_relative_wide_renamed_Ldel_rowsumsIsolates_long <- sample_relative_wide_renamed_Ldel_rowsumsIsolates_long[which(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long$rowSumsIsolates>"0" |sample_relative_wide_renamed_Ldel_rowsumsIsolates_long$Snps>"0"),]

ggplot(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long,aes(x=rowSumsIsolates,y=Snps))+geom_hex(binwidth = c(0.3, 0.02),aes(fill=log(..count..)))+theme_classic()
ggplot(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long,aes(x=rowSumsIsolates,y=Snps))+geom_point()+theme_classic()


###----------tsne

library(Rtsne)
library(ggrepel)
library(philentropy)

sample_relative_wide_renamed_Ldel_onlyIsolates <- sample_relative_wide_renamed_Ldel %>% select(mst3,mst7,mst6,mst9,mst4,mst8,mst5,mst10)
sample_relative_wide_renamed_Ldel_onlyIsolates[c("mst3","mst7","mst6","mst9","mst4","mst8","mst5","mst10")][is.na(sample_relative_wide_renamed_Ldel_onlyIsolates[c("mst3","mst7","mst6","mst9","mst4","mst8","mst5","mst10")])] <- 0
annotation_prep <- sample_relative_wide_renamed_Ldel_onlyIsolates[!duplicated(sample_relative_wide_renamed_Ldel_onlyIsolates), ]

dist_sterm <- distance(as.matrix(annotation_prep), method = "jaccard")

tsne <- Rtsne(dist_sterm, dims = 2, perplexity=30, verbose=TRUE, max_iter = 500)
 

# tsne <- Rtsne(sample_relative_safe_final_tsne_safe_sterm_woDuplicates[,c("Lyoampulle\n2012","Lyoampulle\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018")], dims = 2, perplexity=30, verbose=TRUE, max_iter = 500)
 # tsne <- Rtsne(sample_relative_safe_final_tsne_safe_sterm_woDuplicates[,c("Lyo_202_2012","Lyo_202_2014","RMK202","Konserve_202","Versand_202")], dims = 2, perplexity=30, verbose=TRUE, max_iter = 500)
tsne_plot <- as.data.frame(tsne$Y)
tsne_plot$site <- sample_relative_wide_renamed_Ldel[!duplicated(sample_relative_wide_renamed_Ldel_onlyIsolates), 3]

sample_relative_wide_renamed_Ldel_onlyIsolates_02 <- sample_relative_wide_renamed_Ldel %>% select(site,mst3,mst7,mst6,mst9,mst4,mst8,mst5,mst10)

tsne_plot <- merge(tsne_plot,sample_relative_wide_renamed_Ldel_onlyIsolates_02,by="site",all.x = TRUE)


##plot blank
ggplot(tsne_plot,aes(x=V1,y=V2))+geom_point()


##------------clusters
# set.seed(123456)
# ds.real = dbscan(tsne_plot[,c(1,2)], 1.5)
# tsne_plot$density = factor(ds.real$cluster)
# ds.cent = tsne_plot %>% group_by(density) %>% select(V1, 
#     V2) %>% summarize_all(mean)
# ggplot(tsne_plot, aes(x = V1, y = V2, colour = density)) + 
#     geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label = density), 
#     data = ds.cent) + guides(colour = FALSE)


##clusters
hc.norm = hclust(dist(tsne_plot[,c(1,2)]))
tsne_plot$hclust = factor(cutree(hc.norm, 7))
hc.norm.cent = tsne_plot %>% group_by(hclust) %>% select(V1, 
    V2) %>% summarize_all(mean)
plotTsne <- ggplot(tsne_plot, aes(x = V1, y = V2, color = hclust, text =paste("effect:", site,"\nmst3:",mst3,"\nmst7:",mst7,"\nmst6:",mst6,"\nmst9:",mst9,"\nmst4:",mst4,"\nmst8:",mst8,"\nmst5:",mst5,"\nmst10:",mst10))) + 
  # facet_grid(species~.)+
    geom_point(alpha = 0.3) + theme_bw() +# geom_label_repel(aes(label = hclust), data = hc.norm.cent) +
  guides(colour = FALSE) + 
    ggtitle("Linkage criterion: Complete")

  # pdf("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_tsnePlot_sterm.pdf",width=5,height=5,paper='special')
  # svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_tsnePlot_sterm.svg",width=4,height=4)
  # png("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_tsnePlot_sterm.png",width=1600,height=1600,res=300)
plotTsne

library(plotly)
ggp <- ggplotly(plotTsne)
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_notexplained_ldel_colored_tsne_isolates.html")

##--------------
##make a line graph
##--------------

  pall_sterm_CORE_box <- ggplot(alltogether_SNPs,aes(x=grouping,y=Snps,color=sample,fill=sample))+ geom_boxplot()+
       # facet_grid(group~.)+
   labs("",
         x="",
         # y="",
          y="all")+
       # scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="right",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
pall_sterm_CORE_box

nrow(alltogether_SNPs)
alltogether_SNPs_reduced <- alltogether_SNPs[which(alltogether_SNPs$Snps!=0),]
nrow(alltogether_SNPs_reduced)

  ggplot(alltogether_SNPs_reduced,aes(x=grouping,y=Snps,group=site))+ geom_line(size=0.2, alpha=.2)+ 
       facet_grid(sample~.)+
   labs("",
         x="",
         # y="",
          y="not\nexplained")+
      #  scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       # scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0),limits = c(0,0.2)) +
     scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
    # ylim(0,0.2)+
  theme_classic()+
 theme(#axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       # axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
  
  
  ##-------------
  ##plot uniques
  ##-------------
  length(alltogether_SNPs$site)
  length(unique(alltogether_SNPs$site))
  
  
  sum(sample_relative_wide$site %in% unique(alltogether_SNPs$site))
  sample_relative_wide_subset_wide <- sample_relative_wide[which(sample_relative_wide$site %in% unique(alltogether_SNPs$site)),]

sample_relative_wide_subset_long <- gather(sample_relative_wide_subset_wide, sample, Snps, c("L104","L108","L35","L39","L44","L70","L71","L80","L99"), factor_key=TRUE,na.rm = TRUE) 
nrow(sample_relative_wide_subset_long)

# sample_relative_wide_subset_long$sample"L70"="mst3", "L44"="mst10", "L108"="mst7", "L99"="mst8", "L104"="mst6", "L80"="mst5", "L35"="mst9", "L71"="mst4", "L104"="mst6","Lactobacillus_delbrueckii_RMK202"="RMK202"

sample_relative_wide_subset_long$sample <- revalue(sample_relative_wide_subset_long$sample, c("L70"="mst3", "L44"="mst10", "L108"="mst7", "L99"="mst8", "L104"="mst6", "L80"="mst5", "L35"="mst9", "L71"="mst4", "L104"="mst6","Lactobacillus_delbrueckii_RMK202"="RMK202"))

sample_relative_wide_subset_long[sample_relative_wide_subset_long$Snps<0.8,"Snps"] <- 0

sample_relative_wide_subset_long$sample = factor(sample_relative_wide_subset_long$sample, levels=c("mst3","mst7","mst6","mst9","mst4","mst8","mst5","mst10"))

  ggplot(sample_relative_wide_subset_long,aes(x=sample,y=Snps,group=site))+ geom_line(size=0.2, alpha=.2)+ 
       # facet_grid(sample~.)+
   labs("",
         x="",
         # y="",
          y="not\nexplained")+
      #  scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       # scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0),limits = c(0,0.2)) +
     scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
    # ylim(0,0.2)+
  theme_classic()+
 theme(#axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       # axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
  
  ###----------tsne this

library(Rtsne)
library(ggrepel)
library(philentropy)

  

  
sample_relative_wide_subset_wide_set <- sample_relative_wide_subset_wide
sample_relative_wide_subset_wide_set[is.na(sample_relative_wide_subset_wide_set)] <- 0

colnames(sample_relative_wide_subset_wide_set) <- revalue(colnames(sample_relative_wide_subset_wide_set), c("L70"="mst3", "L44"="mst10", "L108"="mst7", "L99"="mst8", "L80"="mst5", "L35"="mst9", "L71"="mst4", "L104"="mst6"))  


sample_relative_wide_subset_wide_preped <- sample_relative_wide_subset_wide_set[,c("mst3","mst7","mst6","mst9","mst4","mst8","mst5","mst10")]
sample_relative_wide_subset_wide_preped[duplicated(sample_relative_wide_subset_wide_preped), ]

nrow(sample_relative_wide_subset_wide_preped)
annotation_prep <- sample_relative_wide_subset_wide_preped[!duplicated(sample_relative_wide_subset_wide_preped), ]
nrow(annotation_prep)

annotation_prep[is.na(annotation_prep)] <- 0

dist_sterm <- distance(as.matrix(annotation_prep), method = "jaccard")

tsne <- Rtsne(dist_sterm, dims = 2, perplexity=30, verbose=TRUE, max_iter = 500)
 

# tsne <- Rtsne(sample_relative_safe_final_tsne_safe_sterm_woDuplicates[,c("Lyoampulle\n2012","Lyoampulle\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018")], dims = 2, perplexity=30, verbose=TRUE, max_iter = 500)
 # tsne <- Rtsne(sample_relative_safe_final_tsne_safe_sterm_woDuplicates[,c("Lyo_202_2012","Lyo_202_2014","RMK202","Konserve_202","Versand_202")], dims = 2, perplexity=30, verbose=TRUE, max_iter = 500)
tsne_plot <- as.data.frame(tsne$Y)
tsne_plot$site <- sample_relative_wide_subset_wide_set[!duplicated(sample_relative_wide_subset_wide_preped), "site"]

sample_relative_wide_renamed_Ldel_onlyIsolates_02 <- sample_relative_wide_subset_wide_set %>% select(site,mst3,mst7,mst6,mst9,mst4,mst8,mst5,mst10)
# sample_relative_wide_renamed_Ldel_onlyIsolates_02 <- sample_relative_wide_subset_wide_set %>% select(site,L104,L108,L35,L39,L44,L70,L71,L80,L99)

tsne_plot <- merge(tsne_plot,sample_relative_wide_renamed_Ldel_onlyIsolates_02,by="site",all.x = TRUE)


##plot blank
ggplot(tsne_plot,aes(x=V1,y=V2))+geom_point()


##------------clusters
# set.seed(123456)
# ds.real = dbscan(tsne_plot[,c(1,2)], 1.5)
# tsne_plot$density = factor(ds.real$cluster)
# ds.cent = tsne_plot %>% group_by(density) %>% select(V1, 
#     V2) %>% summarize_all(mean)
# ggplot(tsne_plot, aes(x = V1, y = V2, colour = density)) + 
#     geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label = density), 
#     data = ds.cent) + guides(colour = FALSE)


##clusters
hc.norm = hclust(dist(tsne_plot[,c(1,2)]))
tsne_plot$hclust = factor(cutree(hc.norm, 7))
hc.norm.cent = tsne_plot %>% group_by(hclust) %>% select(V1, 
    V2) %>% summarize_all(mean)
plotTsne <- ggplot(tsne_plot, aes(x = V1, y = V2, color = hclust, text =paste("effect:", site,"\nmst3:",mst3,"\nmst7:",mst7,"\nmst6:",mst6,"\nmst9:",mst9,"\nmst4:",mst4,"\nmst8:",mst8,"\nmst5:",mst5,"\nmst10:",mst10))) + 
  # facet_grid(species~.)+
    geom_point(alpha = 0.3) + theme_bw() +# geom_label_repel(aes(label = hclust), data = hc.norm.cent) +
  guides(colour = FALSE) + 
    ggtitle("Linkage criterion: Complete")

  # pdf("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_tsnePlot_sterm.pdf",width=5,height=5,paper='special')
  # svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_tsnePlot_sterm.svg",width=4,height=4)
  # png("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_tsnePlot_sterm.png",width=1600,height=1600,res=300)
plotTsne


library(plotly)
ggp <- ggplotly(plotTsne)
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_notexplained_ldel_colored_tsne_isolates_02.html")


##-------------
##Unique for Strains
##-------------

sample_relative_wide_strains <- sample_relative_wide
sample_relative_wide_strains[is.na(sample_relative_wide_strains)] <- 0

colnames(sample_relative_wide_strains) <- revalue(colnames(sample_relative_wide_strains), c("L70"="mst3", "L44"="mst10", "L108"="mst7", "L99"="mst8", "L80"="mst5", "L35"="mst9", "L71"="mst4", "L104"="mst6"))  


colMeans(sample_relative_wide_strains[which(sample_relative_wide_strains$site %in% inAll_Ldel$site),c("Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202")])

Ldel_mst8_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8>0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_mst8_snps

Ldel_mst5_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5>0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_mst5_snps

Ldel_mst4_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4>0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_mst4_snps

Ldel_mst3_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3>0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_mst3_snps

Ldel_mst7_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7>0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_mst7_snps

Ldel_mst9_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9>0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_mst9_snps

Ldel_L39_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39>0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_L39_snps

Ldel_mst6_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6>0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_mst6_snps

Ldel_mst10_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10>0.9) %in% TRUE),];Ldel_mst10_snps



##-------------
##Unique for all strains in one of three subgroups
##-------------
library(robustbase)
sample_relative_wide_strains <- sample_relative_wide
sample_relative_wide_strains[is.na(sample_relative_wide_strains)] <- 0

colnames(sample_relative_wide_strains) <- revalue(colnames(sample_relative_wide_strains), c("L70"="mst3", "L44"="mst10", "L108"="mst7", "L99"="mst8", "L80"="mst5", "L35"="mst9", "L71"="mst4", "L104"="mst6"))  

colMeans(sample_relative_wide_strains[which(sample_relative_wide_strains$site %in% inAll_Ldel$site),c("Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202")])

Ldel_subcluster_01 <- sample_relative_wide_strains[which(((sample_relative_wide_strains$mst8<0.1 & sample_relative_wide_strains$mst5<0.1&sample_relative_wide_strains$mst10<0.1) &(sample_relative_wide_strains$mst4>0.9& sample_relative_wide_strains$mst3>0.9& sample_relative_wide_strains$mst7>0.9& sample_relative_wide_strains$mst9>0.9 & sample_relative_wide_strains$mst6>0.9)) %in% TRUE),];Ldel_subcluster_01;Ldel_subcluster_01 %>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% as.matrix() %>% colMedians()

Ldel_subcluster_02 <- sample_relative_wide_strains[which(((sample_relative_wide_strains$mst8>0.9) & (sample_relative_wide_strains$mst5<0.1&sample_relative_wide_strains$mst10<0.1 & sample_relative_wide_strains$mst4<0.1& sample_relative_wide_strains$mst3<0.1& sample_relative_wide_strains$mst7<0.1& sample_relative_wide_strains$mst9<0.1 & sample_relative_wide_strains$mst6<0.1)) %in% TRUE),];Ldel_subcluster_02;Ldel_subcluster_02 %>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% as.matrix() %>% colMedians()

Ldel_subcluster_03 <- sample_relative_wide_strains[which(((sample_relative_wide_strains$mst8<0.1) & (sample_relative_wide_strains$mst5>0.9&sample_relative_wide_strains$mst10>0.9) & (sample_relative_wide_strains$mst4<0.1& sample_relative_wide_strains$mst3<0.1& sample_relative_wide_strains$mst7<0.1& sample_relative_wide_strains$mst9<0.1 & sample_relative_wide_strains$mst6<0.1)) %in% TRUE),];Ldel_subcluster_03;Ldel_subcluster_03 %>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% as.matrix() %>% colMedians()

Ldel_subcluster_largeCulster_02_03 <- sample_relative_wide_strains[which(((sample_relative_wide_strains$mst8>0.9 & sample_relative_wide_strains$mst5>0.9&sample_relative_wide_strains$mst10>0.9) & (sample_relative_wide_strains$mst4<0.1& sample_relative_wide_strains$mst3<0.1& sample_relative_wide_strains$mst7<0.1& sample_relative_wide_strains$mst9<0.1 & sample_relative_wide_strains$mst6<0.1)) %in% TRUE),];Ldel_subcluster_largeCulster_02_03;Ldel_subcluster_largeCulster_02_03 %>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% as.matrix() %>% colMedians();Ldel_subcluster_largeCulster_02_03 %>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% gather(.) %>% ggplot(.,aes(x=key,y=value))+geom_boxplot()


Ldel_mst5_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5>0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_mst5_snps

Ldel_mst4_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4>0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_mst4_snps

Ldel_mst3_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3>0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_mst3_snps

Ldel_mst7_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7>0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_mst7_snps

Ldel_mst9_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9>0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_mst9_snps

Ldel_L39_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39>0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_L39_snps

Ldel_mst6_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6>0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_mst6_snps

Ldel_mst10_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10>0.9) %in% TRUE),];Ldel_mst10_snps

###----------summed frequeny in Isolate vs. frequency in meta

sample_relative_wide_renamed_Ldel_rowsumsIsolates <- sample_relative_wide_renamed_Ldel %>% select(site,Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202)
  sample_relative_wide_renamed_Ldel_rowsumsIsolates$rowSumsIsolates <- rowSums(sample_relative_wide_renamed_Ldel[,c("mst3","mst7","mst6","mst9","mst4","mst8","mst5","mst10")],na.rm = TRUE)
  sample_relative_wide_renamed_Ldel_rowsumsIsolates_long <- gather(sample_relative_wide_renamed_Ldel_rowsumsIsolates, sample, Snps, "Lyo_202_2012":"Versand_202", factor_key=TRUE,na.rm = TRUE) 

 
sample_relative_wide_renamed_Ldel_rowsumsIsolates_long[c("rowSumsIsolates", "Snps")][is.na(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long[c("rowSumsIsolates", "Snps")])] <- 0


sample_relative_wide_renamed_Ldel_rowsumsIsolates_long <- sample_relative_wide_renamed_Ldel_rowsumsIsolates_long[which(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long$rowSumsIsolates>"0" |sample_relative_wide_renamed_Ldel_rowsumsIsolates_long$Snps>"0"),]

ggplot(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long,aes(x=rowSumsIsolates,y=Snps))+geom_hex(binwidth = c(0.3, 0.02),aes(fill=log(..count..)))+theme_classic()
ggplot(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long,aes(x=rowSumsIsolates,y=Snps))+geom_point()+theme_classic()


sample_relative_wide_renamed_Ldel_rowsumsIsolates_long_subset <- sample_relative_wide_renamed_Ldel_rowsumsIsolates_long[which(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long$rowSumsIsolates>1),]
ggplot(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long_subset,aes(x=rowSumsIsolates,y=Snps))+geom_point()+theme_classic()


sample_relative_wide_renamed_Ldel_rowsumsIsolates_long_subset <- sample_relative_wide_renamed_Ldel_rowsumsIsolates_long_subset[which(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long_subset$Snps>0.2),]
ggplot(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long_subset,aes(x=rowSumsIsolates,y=Snps))+geom_point()+theme_classic()

length(unique(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long_subset$site))

sample_relative_wide_strains_ldel_interesting <- sample_relative_wide_strains_ldel[which(sample_relative_wide_strains_ldel$site %in% unique(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long_subset$site)),]


sample_relative_wide_renamed_Ldel_rowsumsIsolates_long_subset_PLOT <- merge(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long_subset,sample_relative_wide_strains_ldel,by="site",all.x = TRUE)
PLOT1 <- ggplot(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long_subset_PLOT,aes(x=rowSumsIsolates,y=Snps, text =paste("effect:", site,"\nmst3:",mst3,"\nmst7:",mst7,"\nmst6:",mst6,"\nmst9:",mst9,"\nmst4:",mst4,"\nmst8:",mst8,"\nmst5:",mst5,"\nmst10:",mst10)))+geom_point()+theme_classic()

PLOT1

library(plotly)
ggp <- ggplotly(PLOT1)
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_notexplained_ldel_colored_tsne_isolates_03.html")


##---------------
##subset snps to the ones that only occure in the metagenome
##---------------

keep <- sample_relative_wide_strains_ldel%>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% rowSums() >0 

sample_relative_wide_strains_ldel%>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% as.matrix() %>% colMedians()
sample_relative_wide_strains_ldel%>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% gather(.) %>% ggplot(.,aes(x=key,y=value))+geom_boxplot()

# sample_relative_wide_strains_ldel_kept <- sample_relative_wide_strains_ldel[which(keep),]
# sample_relative_wide_strains_ldel_kept%>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% as.matrix() %>% colMedians()
sample_relative_wide_strains_ldel_kept%>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% gather(.) %>% ggplot(.,aes(x=key,y=value))+geom_boxplot()
sample_relative_wide_strains_ldel_kept_long <- sample_relative_wide_strains_ldel_kept %>% gather(.,sample, snps,c("Lyo_202_2012","Lyo_202_2014","Konserve_202",RMK202,"Versand_202")) 
sample_relative_wide_renamed_Ldel_rowsumsIsolates_long_subset_PLOT <- merge(sample_relative_wide_strains_ldel_kept_long,sample_relative_wide_strains_ldel,by="site",all.x = TRUE)

nrow(sample_relative_wide_strains_ldel_kept)
library(plyr)
# Renamed_gained_lost_SNPs_long_02$sample <- revalue(Renamed_gained_lost_SNPs_long_02$sample, c("Lyo_202_2012"="Lyoampulle\n2012", "Lyo_202_2014"="Lyoampulle\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018"))
Renamed_gained_lost_SNPs_long_02$sample = factor(Renamed_gained_lost_SNPs_long_02$sample, levels=c("Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202"))




# plot1 <- ggplot(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long_subset_PLOT,aes(x=sample,y=snps, text =paste("effect:", site,"\nmst3:",mst3,"\nmst7:",mst7,"\nmst6:",mst6,"\nmst9:",mst9,"\nmst4:",mst4,"\nmst8:",mst8,"\nmst5:",mst5,"\nmst10:",mst10)))+geom_boxplot()
# plot1
# library(plotly)
# ggp <- ggplotly(plot1)
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_notexplained_ldel_colored_tsne_isolates_04.html")

##-------------
##Unique for strains in one of three subgroups
##-------------

Ldel_subcluster_cluster_01_parts <- sample_relative_wide_strains_ldel_kept[which(((sample_relative_wide_strains_ldel_kept$mst8<0.1 & sample_relative_wide_strains_ldel_kept$mst5<0.1 & sample_relative_wide_strains_ldel_kept$mst10<0.1) & (sample_relative_wide_strains_ldel_kept$mst4>0.9 | sample_relative_wide_strains_ldel_kept$mst3>0.9 | sample_relative_wide_strains_ldel_kept$mst7>0.9 | sample_relative_wide_strains_ldel_kept$mst9>0.9 | sample_relative_wide_strains_ldel_kept$mst6>0.9)) %in% TRUE),];Ldel_subcluster_cluster_01_parts;Ldel_subcluster_cluster_01_parts %>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% as.matrix() %>% colMedians();Ldel_subcluster_cluster_01_parts %>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% gather(.) %>% ggplot(.,aes(x=key,y=value))+geom_boxplot()
cluster1_wo_mag <- Ldel_subcluster_cluster_01_parts %>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% as.matrix() %>% colMedians()%>% t()  %>% data.frame() 
cluster1_wo_mag$group <- "cluster1_woMAG"

Ldel_subcluster_cluster_023_parts <- sample_relative_wide_strains_ldel_kept[which(((sample_relative_wide_strains_ldel_kept$mst8>0.9 | sample_relative_wide_strains_ldel_kept$mst5>0.9|sample_relative_wide_strains_ldel_kept$mst10>0.9) & (sample_relative_wide_strains_ldel_kept$mst4<0.1& sample_relative_wide_strains_ldel_kept$mst3<0.1& sample_relative_wide_strains_ldel_kept$mst7<0.1& sample_relative_wide_strains_ldel_kept$mst9<0.1 & sample_relative_wide_strains_ldel_kept$mst6<0.1)) %in% TRUE),];Ldel_subcluster_cluster_023_parts;Ldel_subcluster_cluster_023_parts %>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% as.matrix() %>% colMedians();Ldel_subcluster_cluster_023_parts %>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% gather(.) %>% ggplot(.,aes(x=key,y=value))+geom_boxplot()
cluster2_wo_mag <- Ldel_subcluster_cluster_023_parts %>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% as.matrix() %>% colMedians()%>% t()  %>% data.frame() 
cluster2_wo_mag$group <- "cluster2_all"


##not explained
table(sample_relative_wide_strains$X.CHROM)
sample_relative_wide_strains_ldel <- sample_relative_wide_strains[which(sample_relative_wide_strains$X.CHROM=="L_delbrueckii_RMK202"),]

Ldel_subcluster_cluster_non <- sample_relative_wide_strains_ldel_kept[which(((sample_relative_wide_strains_ldel_kept$mst8<0.1& sample_relative_wide_strains_ldel_kept$mst5<=0.9&sample_relative_wide_strains_ldel_kept$mst10<=0.9& sample_relative_wide_strains_ldel_kept$mst4<=0.9& sample_relative_wide_strains_ldel_kept$mst3<=0.9& sample_relative_wide_strains_ldel_kept$mst7<=0.9& sample_relative_wide_strains_ldel_kept$mst9<=0.9 & sample_relative_wide_strains_ldel_kept$mst6<=0.9)) %in% TRUE),];Ldel_subcluster_cluster_non;Ldel_subcluster_cluster_non %>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% as.matrix() %>% colMedians();Ldel_subcluster_cluster_non %>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% gather(.) %>% ggplot(.,aes(x=key,y=value))+geom_boxplot()

unexplained <- Ldel_subcluster_cluster_non %>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% as.matrix() %>% colMedians()%>% t()  %>% data.frame() 
unexplained$group <- "unexplainsed"

##cluster one with mag

merged_medians <- rbind(cluster1_wo_mag,cluster2_wo_mag,unexplained)
MAG_COUNT <- 1-colSums(merged_medians[1:5]) %>% data.frame() %>% t()  %>% data.frame() 
cluster_1_wMag <- MAG_COUNT+cluster1_wo_mag[1:5]

cluster_1_wMag$group <- "cluster1 with MAG"


merged_medians <- rbind(cluster_1_wMag,cluster2_wo_mag,unexplained)

```


######rarefaction

```{r}
library(vegan)








##test old


library(diveRsity)
install.packages("hierfstat")
library(hierfstat)




data(gtrunchier)
result<- allelic.richness(gtrunchier[,-1])
gtrunchier
dim(gtrunchier)
plot(result$Ar)
dim(result$Ar)


rarefy(x, sample, se = FALSE, MARGIN = 1)
sample_relative_temp_withONT

install.packages("adiv")
library("adiv")

dim(sample_relative_safe_final_meta_cleaned_02)
forRarefaction <- t(samples)
dim(forRarefaction)
# rare_Rao(forRarefaction, dis, sim = TRUE, resampling = 999, formula = c("QE", "EDI"))
rare_Rao(forRarefaction,dis=NULL)
is.na(forRarefaction)


data(aviurba, package="ade4")
distances<-dist.ktab(ktab.list.df(list(aviurba$traits)), type = "N")
abundances<- aviurba$fau
dim(aviurba$fau)
rare_Rao(abundances, distances, sim = TRUE, resampling = 100)
rare_Rao(abundances)
```

##old phage Blast

```{bash}

  ####------------exkursion
  ##------check if the matches are not undetected CRISPR regions
  
  
  grep "NODE_" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages.txt |awk -F "\t" '{OFS="\t"} {if ($9<$10) print $2,$9-50,$10+50,$1 ; else print $2,$10-50,$9+50,$1 }' > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.txt

  
bedtools getfasta -fi /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/Streptococcus_CRISPR.masked_rmk202Genome.fa \
  -bed /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.txt \
  -fo /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.fasta

grep -B 1 --color "GAGCTGTGTTGTTTCGAATGGT" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.fasta |grep ">" > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_filtered.fasta
grep -B 1 --color "GTCCCCTCTCGAGGTAATTAGG" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.fasta|grep ">" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_filtered.fasta
grep -B 1  --color "ACAGTTACTTAAATCTTGAGAGTA" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.fasta |grep ">"  >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_filtered.fasta

sed -i 's/>//g' /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_filtered.fasta

#remove the CRISPR repeats

rm /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_spacers.txt
for removess in $(cat /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_filtered.fasta)
do
contigss=$(echo $removess | cut -d ':' -f 1)
start=$(echo $removess | cut -d ':' -f 2|cut -d '-' -f 1)
end=$(echo $removess | cut -d ':' -f 2|cut -d '-' -f 2)

grep "${contigss}" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.txt | grep "${start}" |grep "${end}" |awk -F "\t" '{print $4}' >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_spacers.txt
done
cat /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages.txt > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_new.txt
for spazerss in $(cat /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_spacers.txt)
do
echo ${spazerss}
grep "${spazerss}" -v /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_new.txt > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_tmp.txt 
cat /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_tmp.txt > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_new.txt

done

wc -l /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages.txt
wc -l /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_new.txt

 ##-------------continue

  
  
  
  cut -f 1 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_new.txt  > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/mapped2local_spacersNames_onlyNames.txt
  

  diff /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/mapped2local_spacersNames_onlyNames.txt \
  /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/all_spacersNames_onlyNames.txt | grep ">" | sed 's/> //g' >\
  /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/unmapped2local_spacersNames_onlyNames.txt


```


###PilerCR BAckup


```{bash pilercr_dialact extraction}
  
##=============
##pilerCR
##=============
#cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Streptococcus_thermophilus_RMK202.fasta \
 
##---------------
species=Streptococcus_thermophilus
##Streptococcus thermophilus
##---------------


id=S_thermophilus_RMK202
      echo ==========================
     echo ${id}
     echo ==========================
     # cd /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/
      
      mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR

  
  ~/apps/PilerCR/pilercr1.06/pilercr -in /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_sterm/final/S_thermophilus_mag_rmk202.fasta \
    -out /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/${id}.pilarCR_out -noinfo\
    -seq /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/${id}.pilarCR_out.fasta

  
  perl ~/apps/PilerCR/PilerCR_parser_new /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//annotated/CRISPR/${id}.pilarCR_out > \
    /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/${id}.pilarCR_out_final
  
  
  perl ~/apps/PilerCR/PilerCR_parser_new_vincent /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//annotated/CRISPR/${id}.pilarCR_out > \
    /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/${id}.pilarCR_out_final_new
  sed -i 's/>/>RMK202:/g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/${id}.pilarCR_out_final_new
  grep ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/${id}.pilarCR_out_final_new | sed 's/>//g' |awk -F "__" '{OFS="\t"} {print $4,$2,$3,$1}' > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/${id}.pilarCR_out_spacer.bed


##---------------
species=Lactobacillus_delbrueckii
##Lactobacillus delbrueckii
##---------------

id=L_delbrueckii_RMK202
      echo ==========================
     echo ${id}
     echo ==========================
     # cd /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/
      
      mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//annotated/CRISPR/

  
  ~/apps/PilerCR/pilercr1.06/pilercr -in /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_ldel/final/L_delbrueckii_mag_rmk202.fasta  \
    -out /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/${id}.pilarCR_out -noinfo\
    -seq /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/${id}.pilarCR_out.fasta

  
  perl ~/apps/PilerCR/PilerCR_parser_new /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/${id}.pilarCR_out > \
    /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/${id}.pilarCR_out_final
  
  
```

In these section I aim to identify and extract all CRISPR spacers in all genomes sequenced from RMK202. 
I do this with [PilerCR](https://www.drive5.com/pilercr/).

```{bash pilercr_dialact extraction}
  
  date=20190312


##=============
##pilerCR
##=============

##---------------
species=Streptococcus_thermophilus
##Streptococcus thermophilus
##---------------

##prep
/archiv/Projects/2018_Culturomics/06_Spades//S50/metabat2/20190203/binning_contigs

/archiv/Projects/2018_Culturomics/06_Spades//S72/metabat2/20190203/binning_contigs

mkdir -p /archiv/Projects/2018_Culturomics/06_Spades//{S72,S50}/metabat2/20190203/binning_contigs/blast
id=S72
threads=30
blastn -num_threads ${threads} -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query /archiv/Projects/2018_Culturomics/06_Spades/${id}/metabat2/20190203//binning_contigs.1.fa \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen" \
  -db /archiv/BLAST/blast_db/20190204/nucleotide/prokaryotes \
  -out /archiv/Projects/2018_Culturomics/06_Spades/${id}/metabat2/20190203/binning_contigs/blast/blast_binning_contig_1.txt \
  -evalue 0.01 -word_size 64

blastn -num_threads ${threads} -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query /archiv/Projects/2018_Culturomics/06_Spades/${id}/metabat2/20190203/binning_contigs.2.fa \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen" \
  -db /archiv/BLAST/blast_db/20190204/nucleotide/prokaryotes \
  -out /archiv/Projects/2018_Culturomics/06_Spades/${id}/metabat2/20190203/binning_contigs/blast/blast_binning_contig_2.txt \
  -evalue 0.01 -word_size 64

  
  for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/names_changed_isolates.txt |grep "^S")
    do
      echo ==========================
     echo ${id}
     echo ==========================
      cd /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/
      
      mkdir -p /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}
  cd /archiv/Projects/2018_Culturomics/06_Spades/${id}/metabat2/20190203/
  
  ~/apps/PilerCR/pilercr1.06/pilercr -in binning_contigs.2.fa \
    -out /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}/${id}.pilarCR_out -noinfo\
    -seq /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}/${id}.pilarCR_out.fasta
  
  #id=S50
  #date=20190214
  
  perl ~/apps/PilerCR/PilerCR_parser_new /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}/${id}.pilarCR_out > \
    /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}/${id}.pilarCR_out_final
  
  
    done > /archiv/Projects/2018_Culturomics/06_Spades/Pilar_${date}_log.out 2> /archiv/Projects/2018_Culturomics/06_Spades/Pilar_${date}_log_error.out
  
  
  ##------------------new parser--------
  #50
  perl ~/apps/PilerCR/PilerCR_parser_new_vincent /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out > \
    /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out_final_new
  sed -i 's/>/>mst1:/g' /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out_final_new
  grep ">" /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out_final_new | sed 's/>//g' |awk -F "__" '{OFS="\t"} {print $4,$2,$3,$1}' > /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out_spacer.bed

 #72
  perl ~/apps/PilerCR/PilerCR_parser_new_vincent /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out > \
    /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out_final_new
  sed -i 's/>/>mst2:/g' /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out_final_new
  grep ">" /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out_final_new | sed 's/>//g' |awk -F "__" '{OFS="\t"} {print $4,$2,$3,$1}' > /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out_spacer.bed


  perl ~/apps/PilerCR/PilerCR_parser_new /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out > \
    /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out_final
    
  perl ~/apps/PilerCR/PilerCR_parser_new /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out > \
    /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out_final

##---------------
species=Lactobacillus_delbrueckii
##Lactobacillus delbrueckii
##---------------

for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/names_changed_isolates.txt |grep "^L")
  do
    echo ==========================
   echo ${id}
   echo ==========================
    cd /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/
    
    mkdir -p /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}
  date=20190312
~/apps/PilerCR/pilercr1.06/pilercr -in scaffolds.fasta \
  -out /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}/${id}.pilarCR_out -noinfo\
  -seq /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}/${id}.pilarCR_out.fasta

perl ~/apps/PilerCR/PilerCR_parser_new /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}/${id}.pilarCR_out > \
  /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}/${id}.pilarCR_out_final


  done > /archiv/Projects/2018_Culturomics/06_Spades/Pilar_${date}_log.out 2> /archiv/Projects/2018_Culturomics/06_Spades/Pilar_${date}_log_error.out


```

Now I do it with all Dialact strains that were originally extracted out of the RMK202.

```{bash}


  date=20190312


##=============
##pilerCR
##=============

##---------------
species=Streptococcus_thermophilus
##Streptococcus thermophilus
##---------------

  
  for i in $(ls /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Dialact/Sterm/ |grep ".fna")
    do
    id=$(echo $i |sed 's/.fna//g')
      echo ==========================
     echo ${id}
     echo ==========================
      cd /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Dialact/Sterm/
      rm -r /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/oldSample
     mkdir -p /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/oldSample
  date=20190312
~/apps/PilerCR/pilercr1.06/pilercr -in ${i} \
  -out /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/oldSample/${id}.pilarCR_out -noinfo\
  -seq /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/oldSample/${id}.pilarCR_out.fasta

  #id=S50
  #date=20190214
  
  perl ~/apps/PilerCR/PilerCR_parser_new /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/oldSample/${id}.pilarCR_out > \
    /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/oldSample/${id}.pilarCR_out_final
  
  
    done # > /archiv/Projects/2018_Culturomics/06_Spades/Pilar_${date}_log.out 2> /archiv/Projects/2018_Culturomics/06_Spades/Pilar_${date}_log_error.out
  
  
##---------------
species=Lactobacillus_delbrueckii
##Lactobacillus delbrueckii
##---------------




for i in $(ls /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Dialact/Ldel/ |grep ".fna")
  do
  
  id=$(echo $i |sed 's/.fna//g')
    echo ==========================
   echo ${id}
   echo ==========================
    cd /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Dialact/Ldel/
    
    mkdir -p /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/oldSample
  date=20190312
~/apps/PilerCR/pilercr1.06/pilercr -in ${i} \
  -out /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/oldSample/${id}.pilarCR_out -noinfo\
  -seq /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/oldSample/${id}.pilarCR_out.fasta

perl ~/apps/PilerCR/PilerCR_parser_new /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/oldSample/${id}.pilarCR_out > \
  /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/oldSample/${id}.pilarCR_out_final


  done # > /archiv/Projects/2018_Culturomics/06_Spades/Pilar_${date}_log.out 2> /archiv/Projects/2018_Culturomics/06_Spades/Pilar_${date}_log_error.out





```
#####merge all CRISPR spacers
Here, I want to bring all annotated CRISPR spacers into one file.

```{bash}

for species in $(echo "Ldel Sterm")
do
echo ==========================
     echo "working on: " ${species}
     echo ==========================
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/CRISPR_spacer_merged.fasta
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/
for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/ |sed 's/.fna//g')
do
      echo ==========================
     echo -e "extracting strain: " ${genomes}
     echo ==========================

 cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${genomes}/${genomes}.pilarCR_out_final >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/CRISPR_spacer_merged.fasta


done #genomes

grep ">" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}//CRISPR_spacer_merged.fasta
grep ">"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}//CRISPR_spacer_merged.fasta| awk -F "[-__]" '{OFS="-"}{print $7,$8}'|sort|uniq -c
done #species

##====================================================
##Streptococcus thermophilus$
##====================================================



cat /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out_final_named  \
   /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out_finall_named   \
   /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/S_thermophilus_RMK202.pilarCR_out_final_named > \
   /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_new_withOLDsamples
   

  for i in $(ls /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Dialact/Sterm/ |grep ".fna")
    do
    id=$(echo $i |sed 's/.fna//g')
     sed "s/>/>${id}:/g" /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/oldSample/${id}.pilarCR_out_final > /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/oldSample/${id}.pilarCR_out_final_named
     
     cat /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/oldSample/${id}.pilarCR_out_final_named >> /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_new_withOLDsamples
     
     echo "======================================="
     echo $id
     grep ">" -c /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/oldSample/${id}.pilarCR_out_final
     
    done 
  
  grep ">" -c /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out_final
  grep ">" -c /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out_final
  grep ">" -c /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/S_thermophilus_RMK202.pilarCR_out_final
  
grep ">" -c /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_new_withOLDsamples
tail /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/oldSample/${id}.pilarCR_out_final_named
##====================================================
##Lactobacillus delbrueckii
##====================================================


rm /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_new_withOLDsamples_ldel.txt

for id in $(ls /archiv/Projects/2018_Culturomics/06_Spades/ |grep "^L" |grep -v "L39")
do

sed "s/>/>${id}:/g" /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/20190203/${id}.pilarCR_out_final > /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/20190203/${id}.pilarCR_out_final_named
     

cat /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/20190203/${id}.pilarCR_out_final_named > /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_new_withOLDsamples_ldel.txt

echo "======================================="
     echo $id
     grep ">" -c /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/20190203/${id}.pilarCR_out_final
done

sed 's/>/>MAG_rmk/g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/L_delbrueckii_RMK202.pilarCR_out_final > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/L_delbrueckii_RMK202.pilarCR_out_final_named

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/L_delbrueckii_RMK202.pilarCR_out_final_named >> /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_new_withOLDsamples_ldel.txt


  for i in $(ls /archiv/Projects/2019_RMK202_analysis/12_oldStrains_comparison/Dialact/Ldel/ |grep ".fna")
    do
    id=$(echo $i |sed 's/.fna//g')
     
     
sed "s/>/>${id}:/g" /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/oldSample/${id}.pilarCR_out_final > /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/oldSample/${id}.pilarCR_out_final_named
     
     
     cat /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/oldSample/${id}.pilarCR_out_final_named >> /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_new_withOLDsamples_ldel.txt
     
     echo "======================================="
     echo $id
     grep ">" -c /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/oldSample/${id}.pilarCR_out_final
     
    done 
echo "==================="
echo "total"
grep ">" -c //archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_new_withOLDsamples_ldel.txt
tail //archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_new_withOLDsamples_ldel.txt
```

##### pairwise CRISPR 
want to check for pairwise CRISPR diversity
```{bash}
##====================================
##Streptococci
##====================================
###split  every CRIPSR into indiviual files

rm -r /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/CRISPRspacersRaw_new
mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/CRISPRspacersRaw_new

sed 's/__.*$//g' /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_new_withOLDsamples |sed 's/:/__/g' > /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_new_withOLDsamples_newNAMED

samtools faidx /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_new_withOLDsamples_newNAMED

for spacer in $(grep "^>" /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_new_withOLDsamples_newNAMED|sed 's/>//g')
  do

#spacer1_final=$(echo ${spacer} |awk -F "__" '{OFS="_"}{print $1,2}' |sed 's/:/_/g')
samtools faidx /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_new_withOLDsamples_newNAMED ${spacer} > /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/CRISPRspacersRaw_new/${spacer}.fasta

  done
  
##-------------
##pairwise needle-wunsch alignment
##-------------

rm -r /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle_new
mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle_new

for spacer1 in $(grep "^>" /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_new_withOLDsamples_newNAMED|sed 's/>//g')
  do
  
  spacer1_final=$(echo ${spacer1})

  for spacer2 in $(grep "^>" /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_new_withOLDsamples_newNAMED|sed 's/>//g')
    do
    
    spacer2_final=$(echo ${spacer2} )

    needle -asequence /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/CRISPRspacersRaw_new/${spacer1_final}.fasta \
      -bsequence /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/CRISPRspacersRaw_new/${spacer2_final}.fasta -gapopen 20 -gapextend 0 -outfile /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle_new/${spacer1_final}__${spacer2_final}.msa

      length=$(grep "Length" /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle_new/${spacer1_final}__${spacer2_final}.msa | cut -d ':' -f 2 | sed -e 's/^[ \t]*//' )
      Identity=$(grep "Identity" /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle_new/${spacer1_final}__${spacer2_final}.msa | cut -d ':' -f 2 | cut -d '/' -f 1| sed -e 's/^[ \t]*//' )
      Gaps=$(grep "Gaps" /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle_new/${spacer1_final}__${spacer2_final}.msa | cut -d ':' -f 2 | cut -d '/' -f 1| sed -e 's/^[ \t]*//' )
      Score=$(grep "Score" /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle_new/${spacer1_final}__${spacer2_final}.msa | cut -d ':' -f 2 | sed -e 's/^[ \t]*//' )
      
     echo -e ${spacer1_final}"\t"${spacer2_final}"\t"${length}"\t"${Identity}"\t"${Gaps}"\t"${Score} >> /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle_new/final_pairwise_alignment.txt
     
    rm /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle_new/${spacer1_final}__${spacer2_final}.msa
    done
    
  done
  

##-------------do parallel
###------define TASK



task(){ 
    spacer2_final=$(echo ${spacer2} )
    needle -asequence /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/CRISPRspacersRaw_new/${spacer1_final}.fasta \
      -bsequence /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/CRISPRspacersRaw_new/${spacer2_final}.fasta -gapopen 20 -gapextend 0 -outfile /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle_new/${spacer1_final}__${spacer2_final}.msa

      length=$(grep "Length" /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle_new/${spacer1_final}__${spacer2_final}.msa | cut -d ':' -f 2 | sed -e 's/^[ \t]*//' )
      Identity=$(grep "Identity" /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle_new/${spacer1_final}__${spacer2_final}.msa | cut -d ':' -f 2 | cut -d '/' -f 1| sed -e 's/^[ \t]*//' )
      Gaps=$(grep "Gaps" /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle_new/${spacer1_final}__${spacer2_final}.msa | cut -d ':' -f 2 | cut -d '/' -f 1| sed -e 's/^[ \t]*//' )
      Score=$(grep "Score" /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle_new/${spacer1_final}__${spacer2_final}.msa | cut -d ':' -f 2 | sed -e 's/^[ \t]*//' )
      
     echo -e ${spacer1_final}"\t"${spacer2_final}"\t"${length}"\t"${Identity}"\t"${Gaps}"\t"${Score} >> /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle_new/final_pairwise_alignment.txt
     
    rm /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle_new/${spacer1_final}__${spacer2_final}.msa

}



###------Run parallel in batches



rm -r /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle_new
mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle_new
counts=$(wc -l /data/Project/CRISPRspacers/02_PilerCR/20200217/all/pilerCR_out_repeats_short|cut -d ' ' -f 1)
N=300
for spacer1 in $(grep "^>" /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_new_withOLDsamples_newNAMED|sed 's/>//g')
  do
  
  spacer1_final=$(echo ${spacer1})

  for spacer2 in $(grep "^>" /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_new_withOLDsamples_newNAMED|sed 's/>//g')
    do
    
    ((i=i%N)); ((i++==0)) && wait
    ##--------------run task
    task &
    
    done
    
  done
```
move local

```{bash}
scp -r vincent@130.223.51.116:/archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle_new/final_pairwise_alignment.txt /home/vincent/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/final_pairwise_alignment_new.txt

```

```{r}
library(readr)
library(ggplot2)
spacer_pairs <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/final_pairwise_alignment_new.txt",  "\t", escape_double = FALSE, col_names = c("spacer1","spacer2","length","matching","gaps","score"),trim_ws = TRUE)

spacer_pairs$id <- 100*(spacer_pairs$matching/spacer_pairs$length)



##----------subset
100*(sum(spacer_pairs$id>90)/nrow(spacer_pairs))
100*(sum(spacer_pairs$id>90)/nrow(spacer_pairs))

##----------histogramm

hist(spacer_pairs$id)

spacer_pairs_similiar <- spacer_pairs[(spacer_pairs$id>80),]
hist(spacer_pairs_similiar$id)


 ##---------------export the shared spacers
 

unique_spacers <- unique(c(spacer_pairs_similiar$spacer1,spacer_pairs_similiar$spacer2))

write.table(unique_spacers, "~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/unique_Sterm_spacers_new.txt",na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names = FALSE)

##----------create matrix

names <- levels(as.factor(spacer_pairs$spacer1))

pairwise_spacer_Matrix <- data.frame(matrix(NA, nrow = length(names), ncol = length(names)))
colnames(pairwise_spacer_Matrix) <- names
rownames(pairwise_spacer_Matrix) <- names

for (i in 1:length(names)) {
   rowName <- names[i]
    print(rowName)
  for (a in 1:length(names)) {
   rowName <- names[i]
    print(rowName)
    
    colNames <- names[a]
    print(colNames)
    
    pairwise_spacer_Matrix[rowName,colNames] <- spacer_pairs[spacer_pairs$spacer1==rowName & spacer_pairs$spacer2==colNames,"id"]
    
    }
        
# count <- count +1 
  
}


##-----------heatmap 2
library(stringr)
library(rafalib)
library(RColorBrewer)
names_genome <- str_split_fixed(names, "_", 2)[,1]
cols <- palette(brewer.pal(3, "Dark2"))[as.fumeric(names_genome)]



library(gplots) ##Available from CRAN
pairwise_spacer_Matrix_ready <- as.matrix(pairwise_spacer_Matrix)
image <- heatmap.2(pairwise_spacer_Matrix_ready,
          trace="none",
          # Rowv = "none",
          Colv=NA,
          ColSideColors=cols,
          # density.info = "none") ##historgram in col scale
          tracecol="black",
          )


svg("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/Streptococcus_thermophilus_SPACERS.svg",width=5,height=5)
 heatmap.2(pairwise_spacer_Matrix_ready,
          trace="none",
          # Rowv = "none",
          Colv=NA,
          cexRow=0.1,
          ColSideColors=cols,
          # density.info = "none") ##historgram in col scale
          tracecol="black",
          )
 dev.off()

 
 
library(plotly)

library(orca)
p <- heatmaply(pairwise_spacer_Matrix_ready, k_row = NA, dendrogram = "row",ColSideColors=cols)
plotly::orca(p, "Streptococcus_thermophilus_SPACERS_plotly.svg")

##-----------plot histogramm

ggplot(spacer_pairs,aes(x=id))+geom_histogram(binwidth = 3)+theme_classic()+
  labs(x="pairwise spacer identity [%]")


##-----------subset for only unique

names <- unique(spacer_pairs$spacer1)


##creat ANI matrix 
count <- 2
spacer_pairs$combined <- paste(spacer_pairs$spacer1,spacer_pairs$spacer2,sep="_")
sterm_curated <- spacer_pairs[1,]

for (i in 1:length(names)) {
   rowName <- names[i]
    print(rowName)
  for(a in count:length(names)){
    
    colNames <- names[a]
    print(colNames)
 
      temp <- spacer_pairs[spacer_pairs$spacer1==rowName &spacer_pairs$spacer2==colNames,]
        
   
    sterm_curated <- rbind(sterm_curated,temp)
    
    
  }
        
count <- count +1 
  
}
sterm_curated <- sterm_curated[-1,]
sterm_curated <- sterm_curated[which(!is.na(sterm_curated$id>80)),]

densityPlot <- ggplot(sterm_curated,aes(x=id))+geom_histogram(binwidth = 3)+theme_classic()+
  labs(x="pairwise spacer identity [%]")

100*(sum(sterm_curated$id>80)/nrow(sterm_curated))
100*(15/nrow(sterm_curated))

svg("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/Streptococcus_thermophilus_SPACERS_histo.svg",width=5,height=3)

densityPlot
          
 dev.off()
 
 ##---------------export the shared spacers
 
sharedspacers <- sterm_curated[which(sterm_curated$id>80),]

unique_spacers <- unique(c(sharedspacers$spacer1,sharedspacers$spacer2))

write.table(unique_spacers, "~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/unique_Sterm_spacers.txt",na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names = FALSE)
```
Based on this analysis (especially the histogramm), we see that we have a very distinct biomodial distribution. By setting a clustering cutoff of 80% we should be able to distinguish common CRISPR spacers and newly aquired

#####make CRISPR clusters plot

now we try to see if there are differences in spacers between the CRISPR spacers of all genomes. 
This is the old way, I used cd-hit to see how many clusters I have within a certain similiartity threshold


```{bash}
date=20190312
source /archiv/Projects/2018_Culturomics/02_script/fileShortcuts.sh

##==================
##cd-hit
##==================
  
  #--------------
  ##Streptococcus thermophilus
  #--------------
  
  
  
  ###-------------
  ##assign representaive Spacers to Cluster
   ###-------------
   cat /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/sterm/sterm_results.txt > /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/sterm/sterm_results_ClusterName.txt
   for spacersss in $(grep ">" /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/sterm/sterm_results.txt)
   do
   echo "--------------------------"
   grep "${spacersss}" -A 1 /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/sterm/sterm_results.txt  > /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/sterm/tmp_spacer.txt
   
   newNAME=$(grep -B 20 "${spacersss}" /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/sterm/sterm_results.txt.clstr |grep "^>Cluster" |tail -1 |sed 's/ /_/g')
   
   sed -i "s/${spacersss}/${newNAME}/g" /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/sterm/sterm_results_ClusterName.txt
   
   done
   rm /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/sterm/tmp_spacer.txt
   
   
   
  
  rm -r /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit_new
  mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit_new/{sterm,ldel}

sed 's/FAM//g' /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_new_withOLDsamples_newNAMED |sed 's/_0.*__/_/g'  |sed 's/__/_/g' > /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_new_withOLDsamples_newNAMED_short

  
   cd-hit -i /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_new_withOLDsamples_newNAMED_short \
    -o /archiv/Projects/2019_pilotplant/02_annotation/CRISPR//CD-hit_new/sterm/sterm_results.txt -c 0.9 -M 30000 -T 37
  
  
  grep -c ">" -c /archiv/Projects/2018_Culturomics/06_Spades/S72/PilerCR/${date}/S72.pilarCR_out_final
  grep -c ">" -c /archiv/Projects/2018_Culturomics/06_Spades/S50/PilerCR/${date}/S50.pilarCR_out_final
  grep -c ">" -c /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/S_thermophilus_RMK202.pilarCR_out_final

  
  ##count clusters and how many genomes per cluster
  date=20191002
  i=0
  start=0
  end=10
  mkdir -p /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/cdhit/20191002/Streptococcus_thermophilus/
  rm /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CD-hit_new/20191002/Streptococcus_thermophilus/clustering_results.txt
  
  for ((i=start; i<=end; i++))
  do
     echo "number: $i"
     
      grep "^${i}" -c /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/sterm/sterm_results.txt.clstr >> \
      /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/cdhit/20191002/Streptococcus_thermophilus/clustering_results.txt
      
  done

#scp -r vincent@130.223.51.116:/archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/sterm/sterm_results.txt.clstr ~/Desktop/Projects/2018_culturomics/CRISPR/
#scp -r vincent@130.223.51.116:/archiv/Projects/2018_Culturomics/12_CRISPR_isolates/cdhit/20191002/Streptococcus_thermophilus/clustering_results.txt ~/Desktop/Projects/2018_culturomics/CRISPR/

#--------------
#Lactobacillus delbrueckii
#--------------

mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/{sterm,ldel}
rm /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/ldel/GenomePath.txt

for id in $(ls /archiv/Projects/2018_Culturomics/06_Spades/ |grep "^L" |grep -v "L39")
do

cat /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/20190203/${id}.pilarCR_out_final | sed -e "s/>/>${id}_/g"  >> \
/archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/ldel/GenomePath.txt

done

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/L_delbrueckii_RMK202.pilarCR_out_final >> \
/archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/ldel/GenomePath.txt

grep "^>" -c /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/ldel/GenomePath.txt



 cd-hit -i /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/ldel/GenomePath.txt \
  -o /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/ldel/ldel_results.txt -c 0.7 -M 30000 -T 37


##count clusters and how many genomes per cluster

start=0
end=12
for ((i=start; i<=end; i++))
do
   echo "number: $i"
   
    grep "^${i}" -c /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/ldel/ldel_results.txt.clstr >> \
    /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/ldel/ldel_clustering_results.txt
    
done

#scp -r vincent@130.223.51.116:/archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/ldel/ldel_clustering_results.txt ~/Desktop/Projects/2018_culturomics/CRISPR/


```




```{r}

library(readr)
library(ggplot2)

##----------Ldel-------------
# ldel_CRISPR <- read_csv("~/Desktop/Projects/2018_culturomics/CRISPR/ldel_clustering.txt",col_names = FALSE)
ldel_CRISPR <- read_csv("~/Desktop/Projects/2018_culturomics/CRISPR/ldel_clustering_results.txt",col_names = FALSE)
#as.vector(ldel_CRISPR)
ldel_CRISPR_new <- cbind(1:13,ldel_CRISPR)
ldel_CRISPR_new[10,]
ldel_CRISPR_new$correctedCount <- NA
for (i in 10:2) {
  ldel_CRISPR_new[(i-1),3] <- ldel_CRISPR_new[i-1,2] - ldel_CRISPR_new[i,2]
}
# ldel_CRISPR_new <- ldel_CRISPR_new[c(-10,-11),]
colnames(ldel_CRISPR_new)[1] <- "cluster"
ldel_CRISPR_new$cluster <- as.factor(ldel_CRISPR_new$cluster)

ldel_CRISPR_new <- ldel_CRISPR_new[1:9,]
# ldel_CRISPR_new <- ldel_CRISPR_new[(ldel_CRISPR_new$correctedCount!=0 & !is.na(ldel_CRISPR_new$correctedCount)),]

plotldel <- ggplot(ldel_CRISPR_new,aes(x=cluster,y=correctedCount))+
    geom_bar(stat = "identity", colour = 'grey')+
          labs(y="count",x="cluster size") + 
    theme(legend.title = element_blank(),
      legend.position="none",
  axis.line = element_line(colour = "black"),
 # axis.text.x =element_blank(),
  panel.background = element_blank(),
  plot.margin=unit(c(0,0,0,0), "cm"),
  text = element_text(size=18),
  panel.border = element_rect(colour = "black", fill=NA, size=1))
plotldel
  svg("~/Desktop/Projects/2018_culturomics/CRISPR/ldel_clustering.svg",width=4,height=2)
plotldel
dev.off()

##----------Sterm-------------
# ldel_CRISPR <- read_csv("~/Desktop/Projects/2018_culturomics/CRISPR/sterm_clustering.txt",col_names = FALSE)
ldel_CRISPR <- read_csv("~/Desktop/Projects/2018_culturomics/CRISPR/clustering_results.txt",col_names = FALSE)
#as.vector(ldel_CRISPR)
ldel_CRISPR_new <- cbind(1:11,ldel_CRISPR)
ldel_CRISPR_new[10,]
ldel_CRISPR_new$correctedCount <- NA
for (i in 10:2) {
  ldel_CRISPR_new[(i-1),3] <- ldel_CRISPR_new[i-1,2] - ldel_CRISPR_new[i,2]
}
# ldel_CRISPR_new <- ldel_CRISPR_new[-(3:11),]
colnames(ldel_CRISPR_new)[1] <- "cluster"
ldel_CRISPR_new$cluster <- as.factor(ldel_CRISPR_new$cluster)

ldel_CRISPR_new <- ldel_CRISPR_new[(ldel_CRISPR_new$correctedCount!=0 & !is.na(ldel_CRISPR_new$correctedCount)),]

plotldel <- ggplot(ldel_CRISPR_new,aes(x=cluster,y=correctedCount))+
    geom_bar(stat = "identity")+
          labs(y="count",x="cluster size") + 
    theme(legend.title = element_blank(),
      legend.position="none",
  axis.line = element_line(colour = "black"),
 # axis.text.x =element_blank(),
  panel.background = element_blank(),
  plot.margin=unit(c(0,0,0,0), "cm"),
  text = element_text(size=18),
  panel.border = element_rect(colour = "black", fill=NA, size=1))
plotldel
  svg("~/Desktop/Projects/2018_culturomics/CRISPR/sterm_clustering.svg",width=2,height=2)
plotldel
dev.off()


```

##### extract CRISPR REGION

```{bash}
##--------------------------
##Sterm
##--------------------------
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations


####==================================
##---getlocation
####==================================

###---------------meta
grep "SUMMARY BY SIMILARITY" -A 15 /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/S_thermophilus_RMK202.pilarCR_out |grep "S_thermophilus_m"| sed -e 's/^[ \t]*//' |awk -F " " 'BEGIN{OFS="\t"}{print $2,$3-3000,$3+$4+3000}' > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/S_thermophilus_RMK202.CRISPRlocation.bed

###---------------S72
grep "SUMMARY BY SIMILARITY" -A 15 /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out |grep "NODE_"| sed -e 's/^[ \t]*//' |awk -F " " 'BEGIN{OFS="\t"}{print $2,$3-3000,$3+$4+3000}' > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/S_thermophilus_S72.CRISPRlocation.bed

###---------------S50
grep "SUMMARY BY SIMILARITY" -A 15 /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out  |grep "NODE_"| sed -e 's/^[ \t]*//' |awk -F " " 'BEGIN{OFS="\t"}{print $2,$3-3000,$3+$4+3000}' > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/S_thermophilus_S50.CRISPRlocation.bed



####==================================
##---change name
##because pilar truncates names the names have to be chnaged afterwards
####==================================
rm /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/genomes_contigs.txt
grep ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_sterm/final/S_thermophilus_mag_rmk202.fasta > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/genomes_contigs.txt
grep ">" /archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/S50.fna >> /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/genomes_contigs.txt
grep ">" /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/S72.fna >> /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/genomes_contigs.txt

grep ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/genomes_contigs.txt
for filesss in $(echo "/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/S_thermophilus_RMK202.CRISPRlocation.bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/S_thermophilus_S72.CRISPRlocation.bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/S_thermophilus_S50.CRISPRlocation.bed")
  do
    for arraysss in $(cut -f 1 $filesss |sort|uniq)
      do
      echo "-----------------------"
      rightName=$(grep "$arraysss" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/genomes_contigs.txt|sed 's/>//g')
      echo $rightName
      
      sed -i -e "s/${arraysss}/$rightName/g" $filesss
      
      done #arraysss
    done #files

####==================================
##---getFASTA
####==================================
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/FASTA/
###---------------meta
bedtools getfasta -fi /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_sterm/final/S_thermophilus_mag_rmk202.fasta \
  -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/S_thermophilus_RMK202.CRISPRlocation.bed \
  -fo //archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/FASTA/S_thermophilus_RMK202.CRISPRlocation.fasta 

###---------------S72
bwa index /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/S72.fna
bedtools getfasta -fi /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/S72.fna \
  -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/S_thermophilus_S72.CRISPRlocation.bed \
  -fo //archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/FASTA/S_thermophilus_S72.CRISPRlocation.fasta 
sed -i 's/>/>mst2-/g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/FASTA/S_thermophilus_S72.CRISPRlocation.fasta 
###---------------S50
bedtools getfasta -fi /archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/S50.fna \
  -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/S_thermophilus_S50.CRISPRlocation.bed \
  -fo //archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/FASTA/S_thermophilus_S50.CRISPRlocation.fasta
  sed -i 's/>/>mst1-/g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/FASTA/S_thermophilus_S50.CRISPRlocation.fasta 

  
cat //archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/FASTA/S_thermophilus_RMK202.CRISPRlocation.fasta \
  //archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/FASTA/S_thermophilus_S72.CRISPRlocation.fasta  \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/FASTA/S_thermophilus_S50.CRISPRlocation.fasta > \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/FASTA/S_thermophilus_alltogether.CRISPRlocation.fasta

```
#####map to CRISPR region

```{bash}
####==================================
##---mapping reads and filtering
####==================================


###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt  ##the file with all sample names
logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion
Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/FASTA/S_thermophilus_alltogether.CRISPRlocation.fasta




##!!!!!!!!!!!!!change names of Forward and Reverse reads

###================
##script
###================
##============
##mapping to reference
##============
bwa index $Assembly
# name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 

mkdir -p ${logFilelocation}

num=1

for names in $(cat ${samplesss})
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))
  rm -r ${BaseLocation}/${names}

  mkdir -p ${BaseLocation}/${names}/bwaMapping2DB/
  
 # bwa mem -t ${threads} /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids/di_rmk202_MAG_reference_polished.fasta
  bwa mem -t ${threads} ${Assembly} \
  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_1.fastq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_2.fastq | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 



samtools view -b -F 4 ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped.bam

bedtools bamtofastq -i ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped.bam -fq ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped.fq


##-----Qualimap

mkdir -p ${BaseLocation}/${names}/bwaMapping2DB//bamqc

qualimap bamqc -bam ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir ${BaseLocation}/${names}/bwaMapping2DB//bamqc --java-mem-size=80G

rm ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam &
 mkdir -p ${BaseLocation}/log
echo ${BaseLocation}"/"${names}"/bwaMapping2DB/bamqc" >> ${BaseLocation}/log/multiqc_logfile_finalGenome.txt

done
```


#####assemble CRISPR regions

```{bash}



###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt  ##the file with all sample names
logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion
Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/FASTA/S_thermophilus_alltogether.CRISPRlocation.fasta
/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledSpacers/allSpacersAssembled_short.fasta
##!!!!!!!!!!!!!change names of Forward and Reverse reads

##--------------meta  assembly


for names in $(cat ${samplesss})
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))
 # rm -r ${BaseLocation}/${names}/assemblyMeta_of_CRISPR-02/
mkdir -p ${BaseLocation}/${names}/assemblyMeta_of_CRISPR-02/

#/home/vincent/miniconda2/pkgs/spades-3.13.0-0/bin/spades.py -k 45 -t 37 -m 100 \
#  -s ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped.fq \
#  -o ${BaseLocation}/${names}/assemblyMeta_of_CRISPR-02/
  
 #reformat.sh ${BaseLocation}/${names}/assemblyMeta_of_CRISPR-02/contigs.fasta
 
 reformat.sh in=${BaseLocation}/${names}/assemblyMeta_of_CRISPR-02/contigs.fasta out=${BaseLocation}/${names}/assemblyMeta_of_CRISPR-02/contigs_min500.fasta minlength=500
  done

##=============
##pilerCR
##=============

 
 names=RMK202
num=1
rm -r ${BaseLocation}/assembledSpacers/
mkdir -p ${BaseLocation}/assembledSpacers/
for names in $(cat ${samplesss})
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))


# ${BaseLocation}/${names}/assemblyMeta_of_CRISPR/contigs.fasta


      mkdir -p ${BaseLocation}/${names}/assembled_and_CRISPR_pilerCR/

  
  ~/apps/PilerCR/pilercr1.06/pilercr -in  ${BaseLocation}/${names}/assemblyMeta_of_CRISPR-02/contigs_min500.fasta \
    -out ${BaseLocation}/${names}/assembled_and_CRISPR_pilerCR//${id}.pilarCR_out -noinfo\
    -seq ${BaseLocation}/${names}/assembled_and_CRISPR_pilerCR//${id}.pilarCR_out.fasta

  
  perl ~/apps/PilerCR/PilerCR_parser_new_vincent ${BaseLocation}/${names}/assembled_and_CRISPR_pilerCR//${id}.pilarCR_out > \
    ${BaseLocation}/${names}/assembled_and_CRISPR_pilerCR//${id}.pilarCR_out_final
  
  sed "s/>/>$names:/g" ${BaseLocation}/${names}/assembled_and_CRISPR_pilerCR//${id}.pilarCR_out_final > ${BaseLocation}/${names}/assembled_and_CRISPR_pilerCR//${id}.pilarCR_out_final_named 
  cat ${BaseLocation}/${names}/assembled_and_CRISPR_pilerCR//${id}.pilarCR_out_final_named >> ${BaseLocation}/assembledSpacers/allSpacersAssembled.fasta
done

sed 's/>/>REF_/g' /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_named  >> ${BaseLocation}/assembledSpacers/allSpacersAssembled.fasta

sed 's/ary/a/g' ${BaseLocation}/assembledSpacers/allSpacersAssembled.fasta  |sed 's/spc0/spc/g'|sed 's/spc/s/g' |sed 's/Lyo/L/g' |sed 's/Konserve/Kon/g' |sed 's/Versand/Ver/g' >> ${BaseLocation}/assembledSpacers/allSpacersAssembled_short.fasta

rm -r ${BaseLocation}/assembledSpacers/cdHIT
  mkdir -p ${BaseLocation}/assembledSpacers/cdHIT
/home/vincent/apps/cd-hit-v4.8.1-2019-0228/cd-hit-est -i ${BaseLocation}/assembledSpacers/allSpacersAssembled_short.fasta \
    -o  ${BaseLocation}/assembledSpacers/cdHIT/CRISPR_Clustering.txt -c 0.95
    grep ">" ${BaseLocation}/assembledSpacers/allSpacersAssembled_short.fasta|cut -d ':' -f 1|sort|uniq -c
##=============
##rename and cluster contigs
##=============

num=1
rm -r ${BaseLocation}/AllSamplesCRISPRassembled/allcontigs.fasta
for names in $(cat ${samplesss})
  do
  sed -i "s/>NODE/>${names}-NODE/" ${BaseLocation}/${names}/assemblyMeta_of_CRISPR/contigs.fasta  


mkdir -p ${BaseLocation}/AllSamplesCRISPRassembled/

cat ${BaseLocation}/${names}/assemblyMeta_of_CRISPR/contigs.fasta  >> ${BaseLocation}/AllSamplesCRISPRassembled/allcontigs.fasta

  done
  
  mkdir -p ${BaseLocation}/AllSamplesCRISPRassembled/cd_hitClustered/
/home/vincent/apps/cd-hit-v4.8.1-2019-0228/cd-hit-est -i ${BaseLocation}/AllSamplesCRISPRassembled/allcontigs.fasta \
    -o  ${BaseLocation}/AllSamplesCRISPRassembled/cd_hitClustered/CRISPR_Clustering.txt -c 0.85
    


##=============
##mapping contigs to the original
##=============

bwa index ${Assembly} 
rm ${BaseLocation}/AllSamplesCRISPRassembled/allcontigsMapped2CRISPRs.bam

bwa mem -x intractg ${Assembly} ${BaseLocation}/AllSamplesCRISPRassembled/allcontigs.fasta  | samtools sort -@${threads} -O BAM -o ${BaseLocation}/AllSamplesCRISPRassembled/allcontigsMapped2CRISPRs.bam - 



num=1
mkdir -p ${BaseLocation}/AllSamplesCRISPRassembled/CRISPRsArrays/
for names in $(samtools view ${BaseLocation}/AllSamplesCRISPRassembled/allcontigsMapped2CRISPRs.bam |cut -f 3 |sort|uniq|grep "*" -v)
  do 
  echo ${num}"/16  :" ${names}
  num=$((num+1))

sampless=$(echo $names | awk -F "[:-]" -v num="$num" '{OFS=""} {print $1,"_",num}')
echo $sampless

samtools view ${BaseLocation}/AllSamplesCRISPRassembled/allcontigsMapped2CRISPRs.bam |awk -v names="$names" '{OFS="\n"} {if($3==names) print ">"$1,$10}' > ${BaseLocation}/AllSamplesCRISPRassembled/CRISPRsArrays/${sampless}.fasta

  done
  
#none mapping
samtools view -b -f 4 ${BaseLocation}/AllSamplesCRISPRassembled/allcontigsMapped2CRISPRs.bam |samtools view - |awk '{OFS="\n"} {print ">"$1,$10}'> ${BaseLocation}/AllSamplesCRISPRassembled/CRISPRsArrays/unmapped.fasta


```




old ...I do not know what this chunk does
```{r}
library(readr)
clusterCRISPRs <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/clusterCRISPRs_sterm_new.txt", "\t", escape_double = FALSE, trim_ws = TRUE)


colnames(clusterCRISPRs)

clusterCRISPRs_new <- clusterCRISPRs %>% filter(.,REF_RMK202==0 & REF_mst1==0& REF_mst2==0)
clusterCRISPRs_old <- clusterCRISPRs %>% filter(.,REF_RMK202!=0 |REF_mst1!=0|REF_mst2!=0)
# clusterCRISPRs %>% filter(.,lyo202_96==0 )
# clusterCRISPRs %>% filter(.,Lyo_202_2014==0 )

write.table(clusterCRISPRs_new, "~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/clusterCRISPRs_new.txt",na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names = FALSE)
write.table(clusterCRISPRs_old, "~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/clusterCRISPRs_old.txt",na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names = FALSE)

clusterCRISPRs_new %>% filter(.,Konserve_202!=0)
clusterCRISPRs_new %>% filter(.,Versand_202!=0)
```

this section was used to check if we have additional spacers in the genome assemblies that were not detected with PilerCR (orphin spacer).
I decided not to use this analysis because it is not of great interest. 


```{bash}


scp  vincent@130.223.51.116:/home/vincent/scripts/

##===================================================
###-----------1. mask Genome from current CRISPR calls
##===================================================
#cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/S_thermophilus_RMK202.CRISPRlocation.bed \
#  //archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/S_thermophilus_S72.CRISPRlocation.bed \
#  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/S_thermophilus_S50.CRISPRlocation.bed |awk -F "\t" '{OFS="\t"} {print $1,$2,$3}' >  \
#  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/allStreptococcus_CRISPRlocation.bed


bedtools maskfasta -fi /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/assemblyFinalwithIsolates.fasta -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/allStreptococcus_CRISPRlocation.bed \
-fo /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/Streptococcus_CRISPR.masked_rmk202Genome.fa


grep " " /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/Streptococcus_CRISPR.masked_rmk202Genome.fa

##---------rmk202
bedtools maskfasta -fi /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids/di_rmk202_MAG_reference_polished.fasta -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/allStreptococcus_CRISPRlocation.bed \
-fo /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_meta_rmk202_Genome.fa

#grep "NN" -i  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids/di_rmk202_MAG_reference_polished.fasta 
#grep "NN" -i  /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_meta_rmk202Genome.fa

##---------S50
bedtools maskfasta -fi /archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/S50.fna -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/allStreptococcus_CRISPRlocation.bed \
-fo /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_S50_Genome.fa


##---------rmk202
bedtools maskfasta -fi /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/S72.fna -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/allStreptococcus_CRISPRlocation.bed \
-fo /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_S72_Genome.fa

##===================================================
###-----------2. blast repeats against each Genome
##===================================================

for genome in $(echo "S72" "S50" "meta_rmk202")
do

makeblastdb -max_file_sz 1GB -in /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_${genome}_Genome.fa \
 -out /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_${genome}_Genome.fa -parse_seqids -dbtype nucl

done


### find adaps

genome=S72
for genome in $(echo "S72" "S50" "meta_rmk202")
do

echo "------------------------------------------------"
echo $adap
echo $adapSeq
echo $genome
echo "------------------------------------------------"
mkdir -p /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/

  blastn -num_threads 35 -max_hsps 100 -max_target_seqs 100 -task megablast -show_gis \
  -query /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt  \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen sgi sacc sblastnames staxids sscinames" \
  -db /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_${genome}_Genome.fa \
  -out /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/CRISPRrepeats_to_${genome}_blast.txt \
  -evalue 0.01 -word_size 16
  
done #genome


##===================================================
###-----------3. extract fasta of regions with blast hit
##===================================================

#/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut_reversed.txt

for genome in $(echo "S72" "S50" "meta_rmk202")
do
#increasing
awk -F "\t" '{OFS="\t"} {print $2,$9,$10,$1}' /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/CRISPRrepeats_to_${genome}_blast.txt | sort -k2 -n  > /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/CRISPRrepeats_to_${genome}_blast_increasing.txt

#decreasing
awk -F "\t" '{OFS="\t"} {print $2,$9,$10,$1}' /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/CRISPRrepeats_to_${genome}_blast.txt | sort -k2 -n -r   > /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/CRISPRrepeats_to_${genome}_blast_decreasing.txt

done #genome
##-----manually make a list of regions to extract spacers from
genome=S72

cat /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/CRISPRrepeats_to_${genome}_blast_increasing.txt
#cat /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/CRISPRrepeats_to_${genome}_blast_decreasing.txt

#vim /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/new_CRISPRrepeats_region_${genome}.txt

#grep "GTTTTAGAGCTGTGTTGTTTCGAATGGTTCCAAAAC" /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_${genome}_Genome.fa
#grep "GTTTTGGAACCATTCGAAACAACACAGCTCTAAAAC" /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_${genome}_Genome.fa
awk -F "\t" '{OFS="\t"} {print $1 ,$2-100,$3+100,$4}' /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/new_CRISPRrepeats_region_${genome}.txt > /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/new_CRISPRrepeats_region_${genome}_extended.txt

samtools faidx /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_${genome}_Genome.fa

 
bedtools getfasta -fi /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_${genome}_Genome.fa \
  -bed /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/new_CRISPRrepeats_region_${genome}_extended.txt \
  -fo /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/new_CRISPRrepeats_region_${genome}.fasta


genome=S50

cat /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/CRISPRrepeats_to_${genome}_blast_increasing.txt
#cat /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/CRISPRrepeats_to_${genome}_blast_decreasing.txt

#vim /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/new_CRISPRrepeats_region_${genome}.txt

#awk -F "\t" '{OFS="\t"} {print $1 ,$2-100,$3+100,$4}' /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/new_CRISPRrepeats_region_${genome}.txt > /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/new_CRISPRrepeats_region_${genome}_extended.txt

samtools faidx /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_${genome}_Genome.fa

 
bedtools getfasta -fi /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_${genome}_Genome.fa \
  -bed /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/new_CRISPRrepeats_region_${genome}_extended.txt \
  -fo /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/new_CRISPRrepeats_region_${genome}.fasta



genome=meta_rmk202
awk -F "\t" '{OFS="\t"} {print $1 ,$2-100,$3+100,$4}' /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/CRISPRrepeats_to_meta_rmk202_blast_decreasing.txt > /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/new_CRISPRrepeats_region_${genome}_extended.txt

samtools faidx /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_${genome}_Genome.fa

 
bedtools getfasta -fi /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_${genome}_Genome.fa \
  -bed /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/new_CRISPRrepeats_region_${genome}_extended.txt \
  -fo /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/new_CRISPRrepeats_region_${genome}.fasta



less /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/CRISPRrepeats_to_${genome}_blast_increasing.txt
cat /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/CRISPRrepeats_to_${genome}_blast.txt >
/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/new_CRISPRrepeats_region_${genome}.txt

/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/new_CRISPRrepeats_region_${genome}.txt

##===================================================
###-----------4. make fastq out of fasta
##===================================================

/home/vincent/scripts/fasta_to_fastq.pl
genome=S50
for genome in $(echo "S72" "S50" "meta_rmk202")
do

perl /home/vincent/scripts/fasta_to_fastq.pl /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/new_CRISPRrepeats_region_${genome}.fasta > /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/new_CRISPRrepeats_region_${genome}.fastq

done #genome

##===================================================
###-----------5. cutadapt fastqs to extract more spacers (orphin spacers)
##===================================================

###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt  ##the file with all sample names
#logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Genomes/
#Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/FASTA/S_thermophilus_alltogether.CRISPRlocation.fasta
species=Sterm
BaseLocation_bam=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP


rm -r /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Genomes/
mkdir -p /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Genomes/

for names in $(echo "S72" "S50" "meta_rmk202")
  do
 

read=/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/repeats2GenomeBlast/new_CRISPRrepeats_region_${names}.fastq


for adapName in $(grep ">" /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt )
  do
adapSeq=$(grep "${adapName}" -A 1 /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt|tail -1)
adap=$(echo ${adapName} | sed 's/>//g'|cut -d '_' -f 1)
cleanRound=1

echo "------------------------------------------------"
echo $adap
echo $adapSeq
echo $names
echo $read
echo $species
echo "------------------------------------------------"

mkdir -p ${BaseLocation}/${names}/ReadPair_${pair}//tmp/

cutadapt ${read} -b ${adapSeq}  -e  0.1 -O 15 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}.txt
rm ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}.txt &

## reads with match
awk '{if($2!="-1") print $0}' ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt

rm ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt & 

##make a list of current (potentially still containing 5' adapters)
awk -F "\t" '{OFS="\n"} {print "@"$1, $5,"+", $9}' ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt  > \
${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp.fastq


##trim 5' ends of these (with relaxed matching errors)
cutadapt ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp.fastq -g ${adapSeq}  -e  0.1 -O 5 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_.fastq.txt

rm ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_.fastq.txt &

## finalise first spacers (we will not be able to keep reads that don't have a small (>5bp) of repeat left)
##removing potential 5' adapters of first trim
mkdir -p ${BaseLocation}/${names}/final/
awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($7)>"25" && length($7)<"46") print ">"$1, $7}' ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_info.txt > \
${BaseLocation}/${names}/ReadPair_${pair}/${names}_${adap}_final_round1_R${pair}.fasta

cat ${BaseLocation}/${names}/ReadPair_${pair}/${names}_${adap}_final_round1_R${pair}.fasta > ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R${pair}.fasta


##take remaining reads and remove 3' adapters again --> do this 5 times
awk -F "\t" '{OFS="\n"} {print "@"$1, $7,"+", $11}' ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt  > \
${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round2_matchestmp_timmed5_readyForRound2.fastq

#grep --color "GTTTTAGAGCTGTGTTGTTTCGAATGGTTCCAAAAC" /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/readswithrepeats_${adap}_intraReads_info_trimmed_matches_readyForRound2.fastq  |head
cleanRound=2
for cleanRound in $(echo "2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20")
  do
  echo ${cleanRound}
  echo "------------------------------------"
cleanRound_next=$((cleanRound+1))
  
cutadapt ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_readyForRound${cleanRound}.fastq -b ${adapSeq}  -e  0.1 -O 5 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.fasta
rm ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.fasta &
##finalised spacers
awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($5)>"25"  && length($5)<"46") print ">"$1, $5}' ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt >> ${BaseLocation}/${names}/ReadPair_${pair}/${names}_${adap}_final_round${cleanRound}_R${pair}.fasta

cat ${BaseLocation}/${names}/ReadPair_${pair}/${names}_${adap}_final_round${cleanRound}_R${pair}.fasta >> ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R${pair}.fasta


##finalised prepered
awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($7)>"25" && length($7)<"46") print "@"$1, $7,"+", $11}' ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound_next}_matchestmp_timmed5_readyForRound${cleanRound_next}.fastq

  done #cleanRound
  cat ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R1.fasta \
  ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R2.fasta > \
  ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads.fasta
  done #adap (spacer)
  
  ###-------------====
  ##reverse
    ###-------------====

for adapName in $(grep ">" /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut_reversed.txt| cut -d ' ' -f 1)
  do
adapSeq=$(grep "${adapName}" -A 1 /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut_reversed.txt|tail -1)
adap=$(echo ${adapName} | sed 's/>//g'|cut -d '_' -f 1)
cleanRound=1

echo "------------------------------------------------"
echo $adap
echo $adapSeq
echo $names
echo $read
echo $species
echo "------------------------------------------------"

mkdir -p ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/

cutadapt ${read} -b ${adapSeq}  -e  0.1 -O 15 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}.txt
rm ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}.txt &

## reads with match
awk '{if($2!="-1") print $0}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt

rm ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt & 

##make a list of current (potentially still containing 5' adapters)
awk -F "\t" '{OFS="\n"} {print "@"$1, $5,"+", $9}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt  > \
${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp.fastq


##trim 5' ends of these (with relaxed matching errors)
cutadapt ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp.fastq -g ${adapSeq}  -e  0.1 -O 5 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_.fastq.txt

rm ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_.fastq.txt &

## finalise first spacers (we will not be able to keep reads that don't have a small (>5bp) of repeat left)
##removing potential 5' adapters of first trim
mkdir -p ${BaseLocation}/${names}/final/
awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($7)>"25" && length($7)<"46") print ">"$1, $7}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_info.txt > \
${BaseLocation}/${names}/ReadPair_${pair}_reversed/${names}_${adap}_final_round1_R${pair}_reversed.fasta

cat ${BaseLocation}/${names}/ReadPair_${pair}_reversed/${names}_${adap}_final_round1_R${pair}_reversed.fasta > ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R${pair}_reversed.fasta


##take remaining reads and remove 3' adapters again --> do this 5 times
awk -F "\t" '{OFS="\n"} {print "@"$1, $7,"+", $11}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt  > \
${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round2_matchestmp_timmed5_readyForRound2.fastq

#grep --color "GTTTTAGAGCTGTGTTGTTTCGAATGGTTCCAAAAC" /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/readswithrepeats_${adap}_intraReads_info_trimmed_matches_readyForRound2.fastq  |head
cleanRound=2
for cleanRound in $(echo "2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20")
  do
  echo ${cleanRound}
  echo "------------------------------------"
cleanRound_next=$((cleanRound+1))
  
cutadapt ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_readyForRound${cleanRound}.fastq -b ${adapSeq}  -e  0.1 -O 5 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.fasta
rm ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.fasta &
##finalised spacers
awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($5)>"25" && length($5)<"46") print ">"$1, $5}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt >> ${BaseLocation}/${names}/ReadPair_${pair}_reversed/${names}_${adap}_final_round${cleanRound}_R${pair}_reversed.fasta

cat ${BaseLocation}/${names}/ReadPair_${pair}_reversed/${names}_${adap}_final_round${cleanRound}_R${pair}_reversed.fasta >> ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R${pair}_reversed.fasta


##finalised prepered
awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($7)>"25" && length($7)<"46") print "@"$1, $7,"+", $11}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound_next}_matchestmp_timmed5_readyForRound${cleanRound_next}.fastq

  done #cleanRound
  cat ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R1_reversed.fasta \
  ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R2_reversed.fasta > \
  ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed.fasta
  revseq ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed.fasta ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular.fasta
  done #adap (spacer)
  

  done #names (sampleName)


##===================================================
###-----------
##===================================================
```
#####map CRISPR spacers onto Sterm phage genome. 

```{bash}

threads=37

 bwa index /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Streptococcus_phage.fasta
 
 mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/
 
bwa mem -t 37 /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Streptococcus_phage.fasta /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_named | samtools sort -@${threads} -O BAM -o /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/mapping2StermPhageGenome_sort_02.bam - 

##intersect CRISPR spacers maps and SNPs --> to see if snps are in crispr regions
bedtools intersect -a /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/mapping2StermPhageGenome_sort.bam  \
  -b /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata/freebayesOuput_WithONT/variantCallsfreebayes_all_f_005_min_20x.vcf |samtools view


 ## from which genome do they originate from
  
  date=20190312

samtools view -b -F 4 /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/mapping2StermPhageGenome_sort_02.bam |samtools view| cut -f 1 > /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_matched_names.txt

sed 's/>//g' /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_matched_names.txt | cut -d ':' -f 1


samtools view -b -F 4 /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/mapping2StermPhageGenome_sort_02.bam |samtools view
##are they in repeat CRISPRs

cut -d '_' -f 1 /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_matched_names.txt > /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_matched_names_short.txt

for crispii in $(cat /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_matched_names_short.txt)
  do
  echo "====================="
  echo $crispii
  grep "$crispii" -A 1 /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/sterm/sterm_results.txt.clstr
  done

##check if assembly is aquiring new crispr spacers---> find indels in crispr array

```
What we detect:

- There are 5 CRISPR spacers that match to the phage in all strains perfectly to the phage
- ,,,
- all CRISPR spacers are different and seem to have evolved seperately
- There are no snps within the CRISPR spacers

##### map unique spacers
```{bash}
scp ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/unique_Sterm_spacers.txt vincent@130.223.51.151:/archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/

###-----------------
##-get unique spacers sequence
###-----------------


rm -r /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/CRISPRspacersUnique
mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/CRISPRspacersUnique

for spacer in $(cat /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/unique_Sterm_spacers.txt)
  do

cat /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/CRISPRspacersRaw/${spacer}.fasta > /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/CRISPRspacersUnique/${spacer}.fasta

cat /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/CRISPRspacersRaw/${spacer}.fasta >> /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/CRISPRspacersUnique/Unique_Sterm_CRISPR.fasta
  done
  
###-----------------
##-blast against phage database 
###-----------------


#/archiv/BLAST/blast_db/20190204/phage_nucleotide/phage_nucleotide 
rm -r /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique
mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique


  blastn -num_threads 35 -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/CRISPRspacersUnique/Unique_Sterm_CRISPR.fasta \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen sgi sacc sblastnames staxids sscinames" \
  -db /archiv/BLAST/blast_db/20190204/viral_nucleotide/viral \
  -out /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF.txt \
  -evalue 0.01 -word_size 16
 head /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF.txt
 
awk -F "\t" '{OFS="\t"} {print $2,$9,$10 }' /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF.txt | sed 's/ref|//g' |sed 's/|//g' > /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF.bed

awk -F "\t" '{OFS="\t"} {if ($9<$10) print $2,$9,$10 ; else print $2,$10,$9 } ' /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF.txt | sed 's/ref|//g' |sed 's/|//g' > /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF.bed


###-----------------
##-get genome annotation of these phages
###-----------------


rm /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF_refseq.txt
 
for phage in $(cut -f 13 /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF.txt |awk -F "[ ,]" '{print $3}' |sed 's/\.//g')
  do
  echo "${phage}"
  grep -P "${phage}\t" /archiv/BLAST/blast_db/20190204/viral_genomic_file.txt |grep "Streptococc" >> /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF_refseq.txt
  
  grep -P "${phage}\t" /archiv/BLAST/blast_db/20190204/viral_genomic_file.txt |grep "Streptococc"  |cut -f 8 |cut -d ' ' -f 3
  echo "--------"
  done
  
  wc -l /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF.txt 
  wc -l /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF_refseq.txt
 
 sort /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF_refseq.txt| uniq > /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF_refseq_cleaned.txt

###-----------------
##-download gff file
###-----------------
rm -r /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/
mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/

 awk 'BEGIN{FS="\t";OFS="\t"}  {print $8 , $20}' /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF_refseq_cleaned.txt | cut -f 2 | \
    sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gff.gz|' >    \
    /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/downloadFile.txt
    

mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/GFF
cd /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/GFF
wget -i /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/downloadFile.txt 
gunzip *.gff.gz

cat /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/GFF/*.gff > /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/GFF/mergedAllMatching.gff

###-----------------
##-download fna file
###-----------------


 awk 'BEGIN{FS="\t";OFS="\t"}  {print $8 , $20}' /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF_refseq_cleaned.txt | cut -f 2 | \
    sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' >    \
    /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/downloadFile.txt
   # ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/841/145/GCF_000841145.1_ViralProj14226/GCF_000841145.1_ViralProj14226_genomic.fna.gz

mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FNA
cd /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FNA
wget -i /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/downloadFile.txt 
gunzip *.fna.gz

cat /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FNA/*fna > /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FNA/mergedAllMatching.fna

#ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/841/145/GCF_000841145.1_ViralProj14226/GCF_000841145.1_ViralProj14226_protein.faa.gz


###-----------------
##-download faa file
###-----------------


 awk 'BEGIN{FS="\t";OFS="\t"}  {print $8 , $20}' /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF_refseq_cleaned.txt | cut -f 2 | \
    sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_protein.faa.gz|' >    \
    /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/downloadFile.txt
   # ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/841/145/GCF_000841145.1_ViralProj14226/GCF_000841145.1_ViralProj14226_genomic.fna.gz

mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FAA
cd /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FAA
wget -i /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/downloadFile.txt 
gunzip *.faa.gz

cat /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FAA/*faa > /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FAA/mergedAllMatching.faa

###-----------------
##orthofinder
###-----------------


cd /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FAA/
threads=35
##--------------------adding one species
orthofinder -t ${threads} -a ${threads} -o /archiv/Projects/2019_pilotplant/03_phagePhylogeny/ -f  /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FAA/

##--------------------move local

mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/03_phagePhylogeny/
scp vincent@130.223.51.151:/archiv/Projects/2019_pilotplant/03_phagePhylogeny/Results_Nov13/Species_Tree/SpeciesTree_rooted.txt  /home/vincent/Desktop/Projects/2019_Pilotplan/03_phagePhylogeny/phage_SpeciesTree_rooted.txt
```

include cos,pac, 987 and 5093 reference genomes
```{bash}
###-----------------
##orthofinder
###-----------------

mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FAA_include_04

cp /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FAA/*faa /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FAA_include_04/

#cp /archiv/Phage_DB/BLAST/20190519/downloads/Sterm_proteins/Renamed/Str-SW10-cos_GCA_003925185.faa /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FAA_include/
cp /archiv/Phage_DB/BLAST/20190519/downloads/Sterm_proteins/Renamed/Str-SW11-cos_GCA_003925205.faa /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FAA_include_04/ #not needed

#cp /archiv/Phage_DB/BLAST/20190519/downloads/Sterm_proteins/Renamed/Str-SW1151-pac_GCA_003925525.faa /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FAA_include/
cp /archiv/Phage_DB/BLAST/20190519/downloads/Sterm_proteins/Renamed/Str-SW13-pac_GCA_003925245.faa /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FAA_include_04/ ##too diverse
cp /archiv/Phage_DB/BLAST/20190519/downloads/Sterm_proteins/Renamed/Str-SW18-pac_GCA_003925325.faa /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FAA_include_04/

#cp /archiv/Phage_DB/BLAST/20190519/downloads/Sterm_proteins/Renamed/Str-SW16-987_GCA_003925005.faa /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FAA_include/
cp /archiv/Phage_DB/BLAST/20190519/downloads/Sterm_proteins/Renamed/Str-SW22-987_GCA_003925385.faa /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FAA_include_04/ ##identical to first
cp /archiv/Phage_DB/BLAST/20190519/downloads/Sterm_proteins/Renamed/Str-SW28-987_GCA_003925485.faa /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FAA_include_04/

cp /archiv/Phage_DB/BLAST/20190519/downloads/Sterm_proteins/Renamed/Str-SW19-5093_GCA_003925345.faa /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FAA_include_04/
#cp /archiv/Phage_DB/BLAST/20190519/downloads/Sterm_proteins/Renamed/Str-SW4-5093_GCA_003925105.faa /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FAA_include_02/ ##not needed


cd /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FAA_include_04/
threads=35
##--------------------adding one species
orthofinder -t ${threads} -a ${threads} -o /archiv/Projects/2019_pilotplant/03_phagePhylogeny/Results_Nov13_new/ -f  /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FAA_include_04/


cat /archiv/Projects/2019_pilotplant/03_phagePhylogeny/Results_Nov13_3/Species_Tree/SpeciesTree_rooted.txt
cat /archiv/Projects/2019_pilotplant/03_phagePhylogeny/Results_Nov13_3/Orthogroups/Orthogroups_SingleCopyOrthologues.txt
##--------------------move local

mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/03_phagePhylogeny/
scp vincent@130.223.51.151:/archiv/Projects/2019_pilotplant/03_phagePhylogeny/Results_Nov13_new/Results_Nov13/Species_Tree/SpeciesTree_rooted.txt /home/vincent/Desktop/Projects/2019_Pilotplan/03_phagePhylogeny/phage_SpeciesTree_rooted_withRef_04.txt
###-----------------
##look for intersecting genes
###-----------------


bedtools intersect -wa -a /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/GFF/mergedAllMatching.gff  \
  -b /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF.bed | awk '{if ($3=="CDS") print $0}' > /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF_genes.gff
#bedtools intersect -a /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/GFF/mergedAllMatching.gff  \
#  -b /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF.bed | awk '{if ($3=="gene") print $0}'


###-----------------
##create faa for the matching GENES
###-----------------


bedtools getfasta -fi /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FNA/mergedAllMatching.fna \
  -bed /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF_genes.gff \
  -fo /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/GENES_phages.fna

scp  vincent@130.223.51.151:/archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/GENES_phages.fna ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/

  
  transeq -sequence /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/GENES_phages.fna \
  -outseq /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/GENES_phages.faa


###-----------------
##blast to UNIPROT db
###-----------------
rm /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/UNIPROT_DIAMOND_matching_CRISPRgenes.txt
diamond blastp --threads 32 --max-target-seqs 1 --db /archiv/TREMBL_database/20190710/uniprot_trembl.dmnd --query /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/GENES_phages.faa --out /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/UNIPROT_DIAMOND_matching_CRISPRgenes.txt


```


```{r}

library(readr)
library(ggplot2)

##----------Ldel-------------
# ldel_CRISPR <- read_csv("~/Desktop/Projects/2018_culturomics/CRISPR/ldel_clustering.txt",col_names = FALSE)
ldel_CRISPR <- read_csv("~/Desktop/Projects/2018_culturomics/CRISPR/ldel_clustering_results.txt",col_names = FALSE)
#as.vector(ldel_CRISPR)
ldel_CRISPR_new <- cbind(1:13,ldel_CRISPR)
ldel_CRISPR_new[10,]
ldel_CRISPR_new$correctedCount <- NA
for (i in 10:2) {
  ldel_CRISPR_new[(i-1),3] <- ldel_CRISPR_new[i-1,2] - ldel_CRISPR_new[i,2]
}
# ldel_CRISPR_new <- ldel_CRISPR_new[c(-10,-11),]
colnames(ldel_CRISPR_new)[1] <- "cluster"
ldel_CRISPR_new$cluster <- as.factor(ldel_CRISPR_new$cluster)

ldel_CRISPR_new <- ldel_CRISPR_new[1:9,]
# ldel_CRISPR_new <- ldel_CRISPR_new[(ldel_CRISPR_new$correctedCount!=0 & !is.na(ldel_CRISPR_new$correctedCount)),]

plotldel <- ggplot(ldel_CRISPR_new,aes(x=cluster,y=correctedCount))+
    geom_bar(stat = "identity", colour = 'grey')+
          labs(y="count",x="cluster size") + 
    theme(legend.title = element_blank(),
      legend.position="none",
  axis.line = element_line(colour = "black"),
 # axis.text.x =element_blank(),
  panel.background = element_blank(),
  plot.margin=unit(c(0,0,0,0), "cm"),
  text = element_text(size=18),
  panel.border = element_rect(colour = "black", fill=NA, size=1))
plotldel
  svg("~/Desktop/Projects/2018_culturomics/CRISPR/ldel_clustering.svg",width=4,height=2)
plotldel
dev.off()

##----------Sterm-------------
# ldel_CRISPR <- read_csv("~/Desktop/Projects/2018_culturomics/CRISPR/sterm_clustering.txt",col_names = FALSE)
ldel_CRISPR <- read_csv("~/Desktop/Projects/2018_culturomics/CRISPR/clustering_results.txt",col_names = FALSE)
#as.vector(ldel_CRISPR)
ldel_CRISPR_new <- cbind(1:11,ldel_CRISPR)
ldel_CRISPR_new[10,]
ldel_CRISPR_new$correctedCount <- NA
for (i in 10:2) {
  ldel_CRISPR_new[(i-1),3] <- ldel_CRISPR_new[i-1,2] - ldel_CRISPR_new[i,2]
}
# ldel_CRISPR_new <- ldel_CRISPR_new[-(3:11),]
colnames(ldel_CRISPR_new)[1] <- "cluster"
ldel_CRISPR_new$cluster <- as.factor(ldel_CRISPR_new$cluster)

ldel_CRISPR_new <- ldel_CRISPR_new[(ldel_CRISPR_new$correctedCount!=0 & !is.na(ldel_CRISPR_new$correctedCount)),]

plotldel <- ggplot(ldel_CRISPR_new,aes(x=cluster,y=correctedCount))+
    geom_bar(stat = "identity")+
          labs(y="count",x="cluster size") + 
    theme(legend.title = element_blank(),
      legend.position="none",
  axis.line = element_line(colour = "black"),
 # axis.text.x =element_blank(),
  panel.background = element_blank(),
  plot.margin=unit(c(0,0,0,0), "cm"),
  text = element_text(size=18),
  panel.border = element_rect(colour = "black", fill=NA, size=1))
plotldel
  svg("~/Desktop/Projects/2018_culturomics/CRISPR/sterm_clustering.svg",width=2,height=2)
plotldel
dev.off()


```



move back to bigchmaine
```{bash}
##----------------
scp  ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/clusterCRISPRs_new.txt vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledSpacers/cdHIT/count/
scp  ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/clusterCRISPRs_old.txt vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledSpacers/cdHIT/count/
```


```{bash}
##----------------bigmachine
rm /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledSpacers/cdHIT/count/newCRISPRS.fasta
for i in $(cut -f 1 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledSpacers/cdHIT/count/clusterCRISPRs_new.txt|cut -d '_' -f 2)
  do 
  end_num=$(($i+1))
  echo $i

 sed -n "/>Cluster $i$/,/>Cluster $end_num$/p" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledSpacers/cdHIT/CRISPR_Clustering.txt.clstr >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledSpacers/cdHIT/count/newCRISPRS.fasta

done

##----------------new labelling
rm /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledSpacers/cdHIT/count/longCrispr.txt
for i in $(cut -f 1 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledSpacers/cdHIT/count/clusterCRISPRs_new.txt|cut -d '_' -f 2)
  do 
  end_num=$(($i+1))
  echo $i

 sed -n "/>Cluster $i$/,/>Cluster $end_num$/p" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledSpacers/cdHIT/CRISPR_Clustering.txt.clstr |grep -v ">Cluster"|awk -F "\t" '{print $2}'|awk -F " " '{print $2}' |sed 's/\.\.\.//g' | awk -F "__" -v clust="$i" '{OFS="\t"} {print "Cluster"clust,$1,"notExplained"}'  >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledSpacers/cdHIT/count/longCrispr.txt

done

for i in $(cut -f 1 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledSpacers/cdHIT/count/clusterCRISPRs_old.txt|cut -d '_' -f 2)
  do 
  end_num=$(($i+1))
  echo $i

 sed -n "/>Cluster $i$/,/>Cluster $end_num$/p" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledSpacers/cdHIT/CRISPR_Clustering.txt.clstr |grep -v ">Cluster"|awk -F "\t" '{print $2}'|awk -F " " '{print $2}' |sed 's/\.\.\.//g' | awk -F "__" -v clust="$i" '{OFS="\t"} {print "Cluster"clust,$1,"Explained"}'  >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledSpacers/cdHIT/count/longCrispr.txt

done

###put coverage in the table
rm /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledSpacers/cdHIT/count/longCrispr_new.txt
for namessss in $(cut -f 2 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledSpacers/cdHIT/count/longCrispr.txt)
  do
  echo $namessss
  coverage=$(grep -e "${namessss}" ${BaseLocation}/assembledSpacers/allSpacersAssembled_short.fasta | sed 's/$/_cov_NA/g' |awk -F "cov_" '{print $2}'|awk -F "." '{print $1}')
  echo $coverage
  grep -e "${namessss}" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledSpacers/cdHIT/count/longCrispr.txt |awk -v cov="$coverage" '{OFS="\t"} {print $0, cov}' |sed 's/>//g' >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledSpacers/cdHIT/count/longCrispr_new.txt 
  done
  
  
scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledSpacers/cdHIT/count/longCrispr_new.txt ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/
```


old stuff
```{bash}
###................................
##------assigning arrays to the clusters
##this is still in parts manuel work
##cluster one is an undefined array

##we start with the undefined array

rm ${BaseLocation}/assembledRepeats/cdHIT/CRISPR_Clustering_assigned_tmp.txt

arrayNum=1
ENDarray=$((arrayNum+1))

Assignedtooo=$(echo "NEW")
  
  sed -n "/>Cluster $arrayNum$/,/>Cluster $ENDarray$/p" ${BaseLocation}/assembledRepeats/cdHIT/CRISPR_Clustering.txt.clstr|grep -v ">Cluster"|awk -F "\t" '{print $2}'|awk -F " " '{print $2}' |sed 's/\.\.\.//g'| awk -F "__" -v clust="$Assignedtooo" '{OFS="\t"} {print $1,clust}' >> ${BaseLocation}/assembledRepeats/cdHIT/CRISPR_Clustering_assigned_tmp.txt
  

##--assigned arrays


##---------start analysis 
for arrayNum in $(echo "0 2 3")
  do 
  echo $i

ENDarray=$((arrayNum+1))

Assignedtooo=$(sed -n "/>Cluster $arrayNum$/,/>Cluster $ENDarray$/p" ${BaseLocation}/assembledRepeats/cdHIT/CRISPR_Clustering.txt.clstr |grep "mst1" |awk -F "\t" '{print $2}'|awk -F " " '{print $2}' |sed 's/\.\.\.//g' |sed 's/>REF_mst1\://g')

  sed -n "/>Cluster $arrayNum$/,/>Cluster $ENDarray$/p" ${BaseLocation}/assembledRepeats/cdHIT/CRISPR_Clustering.txt.clstr|grep -v ">Cluster"|awk -F "\t" '{print $2}'|awk -F " " '{print $2}' |sed 's/\.\.\.//g'| awk -F "__" -v clust="$Assignedtooo" '{OFS="\t"} {print $1,clust}' >> ${BaseLocation}/assembledRepeats/cdHIT/CRISPR_Clustering_assigned_tmp.txt

done

echo ${BaseLocation}/assembledRepeats/cdHIT/CRISPR_Clustering_assigned_tmp.txt

##bring local
scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledRepeats/cdHIT/CRISPR_Clustering_assigned_tmp.txt ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/

```


##### CRISPR-spacer stats

```{r}
library(readr)

longCrispr_new <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/longCrispr_new.txt",  "\t", escape_double = FALSE, col_names = c("clusterName","spacerName","grouping","coverage"),  trim_ws = TRUE)

# longCrispr_new<- longCrispr_new[which(!is.na(longCrispr_new$coverage)),]

longCrispr_new$sample <- str_split_fixed(longCrispr_new$spacerName, ":", 2)[,1]
longCrispr_new$crisprName <- str_split_fixed(longCrispr_new$spacerName, ":", 2)[,2]
longCrispr_new$array <- str_split_fixed(longCrispr_new$crisprName, "s", 2)[,1]
longCrispr_new$spacer <- str_split_fixed(longCrispr_new$crisprName, "s", 2)[,2]
longCrispr_new$arrayName <- paste(longCrispr_new$sample,longCrispr_new$array,sep=":")

longCrispr_new$grouping = factor(longCrispr_new$grouping, levels=c( "notExplained","Explained"))
unique(longCrispr_new$sample)
longCrispr_new$sample = factor(longCrispr_new$sample, levels=c( "REF_RMK202","REF_mst1","REF_mst2" ,"lyo202_96","L_202_2012","L_202_2014","Kon_202","RMK202","Ver_202","di_K2_6h","th_K2_8h" ))
                              

##----------------------------
##plotting
##----------------------------

##---------------------------------Histogramm per sample

      

explainedplot <- ggplot(longCrispr_new,aes(x=sample,group=grouping,color=grouping,fill=grouping))+geom_bar()+
  theme_classic()+
  theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
      #axis.text.x = element_blank(),
          legend.position="right"
          # legend.justification=c(50,50), legend.position=c(50,50)
          #legend.title = element_blank()
          )


    png("~/Desktop/Projects/2019_RMK202_analysis/plot/explainedSPACERS.png", width = 1800, height = 1200,res=300)

explainedplot

dev.off()


##---------------------------------Histogramm per array

ggplot(longCrispr_new,aes(x=arrayName,group=grouping,color=grouping,fill=grouping))+geom_bar()+
  theme_classic()+
  theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
      #axis.text.x = element_blank(),
          legend.position="right"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          )


##---------------------------------Histogramm percent explained per array

explained_arrayes <- data.frame(table(longCrispr_new$arrayName,longCrispr_new$grouping))
explained_arrayes <- explained_arrayes[grep("REF",explained_arrayes$Var1,invert = TRUE),]
explained_arrayes_wide <- spread(explained_arrayes, Var2, Freq)
explained_arrayes_wide$percentExplained <- 100*(explained_arrayes_wide$Explained/(explained_arrayes_wide$Explained+explained_arrayes_wide$notExplained))



ggplot(explained_arrayes_wide,aes(x=percentExplained))+geom_histogram()+
  theme_classic()
  # theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
  #         rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
  #     #axis.text.x = element_blank(),
  #         legend.position="right"
  #         #legend.justification=c(1,1), legend.position=c(1,1),
  #         #legend.title = element_blank()
  #         )

##---------------------------------point plot with size of array

explained_arrayes_wide$totalsize <- explained_arrayes_wide$Explained+explained_arrayes_wide$notExplained
explained_arrayes_wide$sample <- str_split_fixed(explained_arrayes_wide$Var1, ":", 2)[,1]
explained_arrayes_wide$array <- str_split_fixed(explained_arrayes_wide$Var1, ":", 2)[,2]

ggplot(explained_arrayes_wide,aes(x=percentExplained,y=totalsize,color=sample))+geom_point()+
  theme_classic()

###--------------------------
###now I take the arrays which have some explained spacers and see if they are explained by the same reference array or if they are chimeric


explained_arrayes_wide_explained <- explained_arrayes_wide[which(explained_arrayes_wide$percentExplained>0),]


##nested loop 
#1. clusters
#2. through 

###---------------see explained clusters and how many refs are in there
longCrispr_new_explained_test <- longCrispr_new[which(longCrispr_new$grouping=="Explained"),] 
longCrispr_new_explained <- longCrispr_new[which(longCrispr_new$grouping=="Explained"),]%>%filter(., !grepl("REF",sample))
REFSS="Cluster192"
clusterFrame <- data.frame(table(longCrispr_new_explained$clusterName))
clusterFrame$NumberREFS <- NA
 longCrispr_new_explained$arrayNameassigned <- NA
for (REFSS in unique(longCrispr_new_explained$clusterName)) {
  tmp_master <-  longCrispr_new_explained[which(longCrispr_new_explained$clusterName==REFSS),]
  tmp <- longCrispr_new_explained_test[which(longCrispr_new_explained_test$clusterName==REFSS),] %>% filter(., grepl("REF",sample)) %>%unique() %>% nrow()
  clusterFrame[which(clusterFrame$Var1==REFSS),"NumberREFS"] <- tmp
  
  arrayNameassigned <- longCrispr_new_explained_test[which(longCrispr_new_explained_test$clusterName==REFSS),] %>% filter(., grepl("REF",sample))%>%unique() %>% slice(1) %>% select(spacerName) %>% as.character()
  
  longCrispr_new_explained[which(longCrispr_new_explained$clusterName==REFSS),"arrayNameassigned"] <-  arrayNameassigned
  # longCrispr_new_explained[which(longCrispr_new_explained$clusterName==REFSS),]
  # longCrispr_new_explained_test[which(longCrispr_new_explained_test$clusterName==REFSS),]
  if (tmp>="2") {arrayNameassigned <- longCrispr_new_explained_test[which(longCrispr_new_explained_test$clusterName==REFSS),] %>% filter(., grepl("REF",sample)) %>%unique()%>% slice(2) %>% select(spacerName) %>% as.character()
    tmp_02 <- tmp_master
    tmp_02[which(tmp_02==REFSS),"arrayNameassigned"] <-  arrayNameassigned
     longCrispr_new_explained <- rbind(longCrispr_new_explained,tmp_02)
  }
  if (tmp>="3") {arrayNameassigned <- longCrispr_new_explained_test[which(longCrispr_new_explained_test$clusterName==REFSS),] %>% filter(., grepl("REF",sample))%>%unique() %>% slice(3) %>% select(spacerName) %>% as.character()
    tmp_02 <- tmp_master
    tmp_02[which(tmp_02==REFSS),"arrayNameassigned"] <-  arrayNameassigned
     longCrispr_new_explained <- rbind(longCrispr_new_explained,tmp_02)
  }
  if (tmp>="4") {arrayNameassigned <- longCrispr_new_explained_test[which(longCrispr_new_explained_test$clusterName==REFSS),] %>% filter(., grepl("REF",sample))%>%unique() %>% slice(4) %>% select(spacerName) %>% as.character()
    
    tmp_02 <- tmp_master
    tmp_02[which(tmp_02==REFSS),"arrayNameassigned"] <-  arrayNameassigned
    longCrispr_new_explained <- rbind(longCrispr_new_explained,tmp_02)
  }
  
}
table(clusterFrame$NumberREFS)

###---------make names
# longCrispr_new_explained$onlyArray <- str_split_fixed(longCrispr_new_explained$arrayNameassigned,"s",2)[1]
longCrispr_new_explained$sampleAssigned <- str_split_fixed(longCrispr_new_explained$arrayNameassigned, ":", 2)[,1]
longCrispr_new_explained$crisprNameAssigned <- str_split_fixed(longCrispr_new_explained$arrayNameassigned, ":", 2)[,2]
longCrispr_new_explained$arrayAssigned <- str_split_fixed(longCrispr_new_explained$crisprNameAssigned, "s", 2)[,1]
longCrispr_new_explained$spacerAssigned <- str_split_fixed(longCrispr_new_explained$crisprNameAssigned, "s", 2)[,2]
longCrispr_new_explained$arrayNameAssignedFinal <- paste(longCrispr_new_explained$sampleAssigned,longCrispr_new_explained$arrayAssigned,sep=":")



##----------------how many different refclusters those one sampleArray map to 

sampleArray_frame <- data.frame(table(longCrispr_new_explained$arrayName))

sampleArray_frame$numberspacersss <- NA
sampleArray_frame$numMatchingArrays <- NA
sampleArray_frame$explainedByONE <- NA
sampleArray_frame$dominantArrayAssigned <- NA


for (samplearraysss in unique(longCrispr_new_explained$arrayName)) {
 tmp_01 <-  longCrispr_new_explained[which(longCrispr_new_explained$arrayName==samplearraysss),"arrayNameAssignedFinal"]
 tmp_02 <-  longCrispr_new_explained[which(longCrispr_new_explained$arrayName==samplearraysss),]
 table(tmp_02$spacerName)
 
 numberspacers <- unique(tmp_02$spacerName) %>% length()

  numUniques <- tmp_01 %>% unique() %>% nrow()
  tmp_03 <- data.frame(spacerName="Kon_202:a12s02",arrayNameAssignedFinal="REF_mst2:a01")
  for (uniquzz in unique(tmp_02$spacerName)) {
    tmp_04 <- tmp_02[tmp_02$spacerName==uniquzz,] %>% select(c(spacerName,arrayNameAssignedFinal)) %>% unique() 
    tmp_03 <- rbind(tmp_03,tmp_04)
  }
  tmp_03 <- tmp_03[-1,]
   dominantCount <-
    tmp_03$arrayNameAssignedFinal %>% table() %>% sort(decreasing = TRUE) %>% first()
    dominantArrayAssigned_tmp <- tmp_03$arrayNameAssignedFinal %>% table() %>% sort(decreasing = TRUE) %>% names() %>% first() %>%as.character()
  percentExplained_02 <- 100*(dominantCount/numberspacers)
  
  sampleArray_frame[which(sampleArray_frame$Var1==samplearraysss),"numberspacersss"] <- numberspacers
  sampleArray_frame[which(sampleArray_frame$Var1==samplearraysss),"numMatchingArrays"] <- numUniques
  sampleArray_frame[which(sampleArray_frame$Var1==samplearraysss),"explainedByONE"] <- percentExplained_02
  sampleArray_frame[which(sampleArray_frame$Var1==samplearraysss),"dominantArrayAssigned"] <- dominantArrayAssigned_tmp 
  # sampleArray_frame$dominantArrayAssigned 
  
}

###-------------plot with percent explained

ggplot(sampleArray_frame,aes(x=explainedByONE))+geom_histogram()+theme_classic()

arrayesFinal <- merge(explained_arrayes_wide,sampleArray_frame,by="Var1",all = TRUE)


ggplot(arrayesFinal,aes(x=percentExplained,y=totalsize,color=explainedByONE))+geom_point()+
  theme_classic()


###-------------plot with percent explained
arrayesFinal_explained <- arrayesFinal[which(arrayesFinal$percentExplained>0),]
# arrayesFinal_explained$dominantArrayAssigned
ggplot(arrayesFinal_explained,aes(x=percentExplained,y=totalsize,color=dominantArrayAssigned))+geom_point(size=3)+
  theme_classic()


table(arrayesFinal_explained$dominantArrayAssigned,arrayesFinal_explained$sample)

###----add array information

CRISPR_assigned <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/CRISPR_Clustering_assigned_tmp.txt", "\t", escape_double = FALSE, col_names = c("arrayNamesss","assignedArray"), trim_ws = TRUE)
CRISPR_assigned$arrayNamesss <- gsub(">","",CRISPR_assigned$arrayNamesss)
CRISPR_assigned$sampleArray <- str_split_fixed(CRISPR_assigned$arrayNamesss, ":", 2)[,1]
CRISPR_assigned$sampleArrayName <- str_split_fixed(CRISPR_assigned$arrayNamesss, ":", 2)[,2]
CRISPR_assigned$arrayNamesss <- paste0(CRISPR_assigned$sampleArray,":a",str_sub(CRISPR_assigned$sampleArrayName, start= -2))

arrayesFinal$arrayNamesss <- paste0(arrayesFinal$sample,":",arrayesFinal$array)

arrayesFinal_new <- merge(CRISPR_assigned,arrayesFinal,by="arrayNamesss",all.y = TRUE)
arrayesFinal_new <- arrayesFinal_new[which(arrayesFinal_new$assignedArray!="NEW"),]


explainedArrays <- ggplot(arrayesFinal_new,aes(x=percentExplained,y=totalsize,color=assignedArray))+geom_point()+labs(x="percent of spacers explained by strains in contig",y="number of spacers on contig")+
  theme_classic()

explainedArrays

    # png("~/Desktop/Projects/2019_RMK202_analysis/plot/explainedarrays.png", width = 1800, height = 1200,res=300)

explainedArrays

# dev.off()

##---------------------------------------------------------------------------------------------------------------
##----------assign array Information to all single spacers

##spacer list
longCrispr_new$spacerName

##extended information for explained spacers
longCrispr_new_explained
grep("ssigne",colnames(longCrispr_new_explained))
tmp_2 <- longCrispr_new_explained[,c(2,grep("ssigne",colnames(longCrispr_new_explained)))]

longCrispr_new_final <- merge(longCrispr_new,tmp_2,by="spacerName",all = TRUE)
##array assignment
CRISPR_assigned$arrayNamesss
##---------------------------------------------------------------------------------------------------------------
###-----------check if all spacers from one cluster come from the same array
longCrispr_new_final_01 <- merge(longCrispr_new,CRISPR_assigned,by.x = "arrayName",by.y = "arrayNamesss",all = TRUE)

###here we remove the few spacer which have a new repeat because we assume they are spurious repeats
longCrispr_new_final_01 <- longCrispr_new_final_01[which(longCrispr_new_final_01$assignedArray!="NEW"),]
tmp1 <- table(longCrispr_new_final_01$clusterName,longCrispr_new_final_01$assignedArray) %>% data.frame() %>% spread(.,Var2,Freq) %>% remove_rownames %>% column_to_rownames(var="Var1") 

##!!!!none spacers are assigned to the same array in every cluster
which(rowSums(as.data.frame(tmp1)>0)>1)
##---------------------------------------------------------------------------------------------------------------
##----------------assign array to the cluster
# 
# tmp1$ClusterName <- rownames(tmp1)
# clusterAssigned <- gather(tmp1, array, count, 1:3, factor_key=TRUE,na.rm = TRUE)
# clusterAssigned <- clusterAssigned[which(clusterAssigned$count>0),-3]

tmp2 <- longCrispr_new_final_01[,c("clusterName","grouping","assignedArray")]

ClusterAssigned_final <- tmp2[!duplicated(tmp2),]
ClusterAssigned_final_plot <- table(ClusterAssigned_final$grouping,ClusterAssigned_final$assignedArray) %>%as.data.frame()

numArrayExplained <- ggplot(ClusterAssigned_final_plot,aes(x=Var2,y=Freq,group=Var1,color=Var1,fill=Var1))+geom_bar(stat="identity")+theme_classic()+labs(x="Array",y="number of CRISPR spacers in <Pan-CRISPR-array>")

    # png("~/Desktop/Projects/2019_RMK202_analysis/plot/explainedarrays_02.png", width = 1800, height = 1200,res=300)

numArrayExplained

# dev.off()

##---------------------------------------------------------------------------------------------------------------
##explained by one or multiple strains


tmp3 <- table(longCrispr_new_explained$clusterName,longCrispr_new_explained$sampleAssigned)  %>% data.frame() %>% spread(.,Var2,Freq)  %>% remove_rownames %>% column_to_rownames(var="Var1") 

ClusterAssigned_final$groupingExtended <- ClusterAssigned_final$grouping
ClusterAssigned_final$groupingExtended <- revalue(ClusterAssigned_final$groupingExtended, c("Explained"="Explained by one strain","notExplained"="not Explained"))
ClusterAssigned_final$groupingExtended <- as.character(ClusterAssigned_final$groupingExtended)

for (Clustersss in (which(rowSums(as.data.frame(tmp3)>0)>1) %>% names())) {
  print(Clustersss)
  ClusterAssigned_final[which(ClusterAssigned_final$clusterName==Clustersss),"groupingExtended"] <- "Explained by multiple strains"
} 

ClusterAssigned_final_plot <- table(ClusterAssigned_final$groupingExtended,ClusterAssigned_final$assignedArray) %>%as.data.frame()
ClusterAssigned_final_plot$Var1 = factor(ClusterAssigned_final_plot$Var1, levels=c("not Explained","Explained by one strain","Explained by multiple strains"))
colorsss <- rev(c("darkcyan","darkturquoise","lightgray"))

numArrayExplained_02 <- ggplot(ClusterAssigned_final_plot,aes(x=Var2,y=Freq,group=Var1,color=Var1,fill=Var1))+geom_bar(stat="identity")+theme_classic()+labs(x="Array",y="number of CRISPR spacers",title = "Pan-CRISPR-arrays in S. thermophilus")+
   scale_fill_manual(values=colorsss)+
   scale_color_manual(values=colorsss)+
  theme(legend.title = element_blank())

 # png("~/Desktop/Projects/2019_RMK202_analysis/plot/explainedarrays_03.png", width = 1800, height = 1200,res=300)

numArrayExplained_02

# dev.off()
##---------------------------------------------------------------------------------------------------------------
##Pan-CRISPR-array-Graph (which spacers are unique and which are shared)

longCrispr_new_REF <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/longCrispr_new.txt",  "\t", escape_double = FALSE, col_names = c("clusterName","spacerName","grouping","coverage"),  trim_ws = TRUE)


##-------------------
##multiple CRISPRs

longCrispr_new_REF <- longCrispr_new_REF[grep("REF_",longCrispr_new_REF$spacerName),]
CRISPRPLOT_data <- table(table(longCrispr_new_REF$clusterName)) %>% as.data.frame()

ggplot(CRISPRPLOT_data,aes(x=Var1,y=Freq))+geom_bar(stat="identity")+theme_classic()+labs(x="Number of SPACER occurences",y="Frequency",title = "Pan-CRISPR-arrays-graphs in S. thermophilus")

##-------------------
##1. identify and remove intra strain spacer duplication


longCrispr_new_REF$genome <- str_split_fixed(longCrispr_new_REF$spacerName, ":", 2)[,1]
longCrispr_new_REF_subset <- longCrispr_new_REF[,c(1,5)] 
longCrispr_new_REF_subset <- longCrispr_new_REF_subset[!duplicated(longCrispr_new_REF_subset),]
CRISPRPLOT_data <- table(table(longCrispr_new_REF_subset$clusterName)) %>% as.data.frame()

ggplot(CRISPRPLOT_data,aes(x=Var1,y=Freq))+geom_bar(stat="identity")+theme_classic()+labs(x="Number of Genomes in which the spacer occurs",y="Frequency",title = "Pan-CRISPR-arrays-graphs in S. thermophilus")

longCrispr_new_REF_subset <- longCrispr_new_REF[,c(1,5)] 
longCrispr_new_REF_subset <- longCrispr_new_REF_subset[duplicated(longCrispr_new_REF_subset),1] %>% as.vector() %>% unique()%>% pull(clusterName)
repeatedCRISPRspacer <- longCrispr_new_REF[which(longCrispr_new_REF$clusterName %in% longCrispr_new_REF_subset),]


##---------------------------------------------------------------------------------------------------------------
##Heatmap Pan-CRISPR-array-Graph (which spacers are unique and which are pairwise-shared)

##-------------------
##2. identify shared CRISPRs

REFSSSS <- unique(longCrispr_new_REF$genome)
i<- 1

pairwiseSharedSpacers <- data.frame()
for (i in 1:length(REFSSSS)) {
  REF1 <- REFSSSS[-i][1]
  REF2 <- REFSSSS[-i][2]
  
  tmp_4 <- longCrispr_new_REF[longCrispr_new_REF$genome %in% c(REFSSSS[-i]),c(1,5)] %>%unique()
  total_spacers <-  tmp_4$clusterName %>% unique() %>%length()
  shared_spacers <-  tmp_4$clusterName %>% table() %>% table()
  clusterSize_tmp <- tmp_4$clusterName %>% table() 
  names_tmp <- which(clusterSize_tmp>1) %>% names()
  
  shared_spacers_size <- tmp_4[which(tmp_4$clusterName %in% names_tmp ),"clusterName"] %>% unique() %>% nrow()
  
  uniqueSpacers_tmp <- tmp_4[which(!tmp_4$clusterName %in% names_tmp ),"genome"] %>% table() %>% as.data.frame() 
  
  Ref1_spacersize <- uniqueSpacers_tmp[which(uniqueSpacers_tmp$.==REF1),2]
  Ref2_spacersize <- uniqueSpacers_tmp[which(uniqueSpacers_tmp$.==REF2),2]
  total_shared_Percent <- 100*(shared_spacers_size/total_spacers)
  tmp_05 <- data.frame(Ref1=REF1,Ref2=REF2,totalSpacer=total_spacers,sharedSpacers=shared_spacers_size,UniqueREF1=Ref1_spacersize,UniqueREF2=Ref2_spacersize,PercentShared=round(total_shared_Percent,digits=1))
  
  pairwiseSharedSpacers <- rbind(pairwiseSharedSpacers,tmp_05)
}

##--------------
##plot
##-------------
library(stringr)
library(readr)
library(ggplot2)
library(reshape2)

# sterm_curated_rev_names
# ##reorder according to dendro
# levels(sterm_curated_rev_names$genome1)
# sterm_curated_rev_names$genome1 = factor(sterm_curated_rev_names$genome1, levels=labels(otter.dendro))
# levels(sterm_curated_rev_names$genome1)
# levels(sterm_curated_rev_names$genome2)
# sterm_curated_rev_names$genome2 = factor(sterm_curated_rev_names$genome2, levels=labels(otter.dendro))
# levels(sterm_curated_rev_names$genome2)

# creat Heatmap
# sterm_curated$X3 <- round(sterm_curated$X3,digits = 2)

##-----------
##change names
##-----------
# sterm_curated_rev_names$genome1 <- revalue(sterm_curated_rev_names$genome1, c("Sterm_rmk202"="MAG RMK202"))
# sterm_curated_rev_names$genome2 <- revalue(sterm_curated_rev_names$genome2, c("Sterm_rmk202"="MAG RMK202"))


##------------
##for SNP legend
##------------
pairwiseSharedSpacers
# pairwiseSharedSpacers_tmp <- pairwiseSharedSpacers[,c("Ref2","Ref1","PercentShared")]
# pairwiseSharedSpacers_tmp_02 <- c(Ref2=unique(pairwiseSharedSpacers_final$Ref2)))

pairwiseSharedSpacers_tmp_02 <-  data.frame(Ref1=c(as.character(pairwiseSharedSpacers$Ref2),as.character(pairwiseSharedSpacers$Ref1)),Ref2=c(as.character(pairwiseSharedSpacers$Ref1),as.character(pairwiseSharedSpacers$Ref2)),PercentShared=c(as.numeric(pairwiseSharedSpacers$PercentShared),as.numeric(pairwiseSharedSpacers$PercentShared)))

pairwiseSharedSpacers_tmp_02$Ref1 = factor(pairwiseSharedSpacers_tmp_02$Ref1, levels=c("REF_mst1","REF_RMK202","REF_mst2"))
pairwiseSharedSpacers_tmp_02$Ref2 = factor(pairwiseSharedSpacers_tmp_02$Ref2, levels=c("REF_mst1","REF_RMK202","REF_mst2"))

# levels(CRISPR_spacer_coverage$sample)


# pairwiseSharedSpacers_final <- rbind(pairwiseSharedSpacers_tmp_02[,c("Ref2","Ref1","PercentShared")],pairwiseSharedSpacers[,c("Ref1","Ref2","PercentShared")])
ggheatmap <- ggplot(pairwiseSharedSpacers_tmp_02, aes(Ref2, Ref1, fill = PercentShared))+
 geom_tile(color = "white",size=1.1)+
 #scale_fill_gradient2(low = "blue", high = "red", mid = "white", 
  # midpoint = 99.5, limit = c(98.9,100), space = "Lab", 
  # name="ANI") +
  scale_fill_distiller(name = "SNPs", palette = "Blues", direction = -1)+
  # scale_fill_distiller(name = "ANI", palette = "Reds", direction = 1)+
  theme_minimal()+ # minimal theme
 theme(axis.text.y = element_text(size = 12),
      axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1))+
 coord_fixed()
# Print the heatmap
print(ggheatmap)

##------------
##CRISPR
##------------

plotFinal_sterm <- ggheatmap + 
geom_text(aes(Ref2, Ref1, label =PercentShared), color = "black", size = 2.5) +
    # labs(title=paste0("# core genes :",nrow(coreGeneNames_sterm)," (",CoreGeneSize_sterm,"bp)"))+
  scale_y_discrete(position = 'right')+
theme(
  axis.title.x = element_blank(),
  axis.title.y = element_blank(),
  # axis.text.y.left = element_text(angle = 75, hjust = 1,size=9),
  panel.grid.major = element_blank(),
  # scale_y_date(position = "right"),
  panel.border = element_blank(),
  panel.background = element_blank(),
  axis.ticks = element_blank(),
  # legend.justification = c(1, 0),
  legend.position = "right",
  legend.text = element_text(angle = 0),
  legend.direction = "vertical")+
  guides(fill = guide_colorbar(barwidth = 1, barheight = 7,
                title.position = "top", title.hjust = 0.5))
plotFinal_sterm
##-----------save heatmap
# svg("~/Desktop/Projects/2018_culturomics/fastani/Sterm_CRISPR_core_absolute_final.svg",width=5.5,height=4)
plotFinal_sterm#+theme(legend.position = "none")
# dev.off()



```
It seems like there is a huge CRISPR spacer repetoire apart from the assembled contigs


##old:Metagenome CRISPR spacers
this is the old version of the METAGENOMIC CRISPR spacer extraction pipeline (removed on 8.4. 2020)

```{bash}
#---test
adap=GTTTTAGAGCTGTGTTGTTTCGAATGGTTCCAAAAC
reverse=GTTTTGGAACCATTCGAAACAACACAGCTCTAAAAC

cat /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt

head -1000000 /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/RMK202/RMK202/RMK202_R1_kneaddata_paired_1.fastq |grep -c ${adap} 
grep -c ${reverse} /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/RMK202/RMK202/RMK202_R1_kneaddata_paired_1.fastq

echo $adap
echo $reverse


#manually extract the CRISPR repeats

#cat /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt

###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt  ##the file with all sample names
#BaseLocation=/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/
BaseLocation=/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract/
species=Sterm
#BaseLocation_bam=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP
#BaseLocation_bam=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP
repeatsFile=/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt

#revseq /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut_reversed.txt

###================
##script
###================
##test
pair=1
names=Konserve_202
adapName=$(grep ">" ${repeatsFile} |head -1)
###----end

for names in $(cat ${samplesss})
  do
 
for pair in $(echo "1 2")
  do
#pair=1

read=/archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_${pair}.fastq


for adapName in $(grep ">" ${repeatsFile} )
  do
adapSeq=$(grep "${adapName}" -A 1 ${repeatsFile}|tail -1)
adap=$(echo ${adapName} | sed 's/>//g'|cut -d '_' -f 1)
cleanRound=1

echo "------------------------------------------------"
echo $adap
echo $adapSeq
echo $names
echo $read
echo $species
echo "------------------------------------------------"

mkdir -p ${BaseLocation}/${names}/ReadPair_${pair}//tmp/

cutadapt ${read} -b ${adapSeq}  -e  0.1 -O 15 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}.txt
rm ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}.txt &

## reads with match
awk '{if($2!="-1") print $0}' ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt

rm ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt & 

##make a list of current (potentially still containing 5' adapters)
awk -F "\t" '{OFS="\n"} {print "@"$1, $5,"+", $9}' ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt  > \
${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp.fastq


##trim 5' ends of these (with relaxed matching errors)
cutadapt ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp.fastq -g ${adapSeq}  -e  0.1 -O 5 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_.fastq.txt

rm ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_.fastq.txt &

## finalise first spacers (we will not be able to keep reads that don't have a small (>5bp) of repeat left)
##removing potential 5' adapters of first trim
mkdir -p ${BaseLocation}/${names}/final/
#awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($7)>"25" && length($7)<"46") print ">"$1, $7}' #${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_info.txt > \
#${BaseLocation}/${names}/ReadPair_${pair}/${names}_${adap}_final_round1_R${pair}.fasta


awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($7)>"25"  && length($7)<"46" && length($7)==length($11)) print "@"$1, $7 ,"+", $11}' ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}/${names}_${adap}_final_round1_R${pair}.fastq

cat ${BaseLocation}/${names}/ReadPair_${pair}/${names}_${adap}_final_round1_R${pair}.fastq > ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R${pair}.fastq


##take remaining reads and remove 3' adapters again --> do this 5 times
awk -F "\t" '{OFS="\n"} {print "@"$1, $7,"+", $11}' ${BaseLocation}/${names}/ReadPair_${pair}//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt  > \
${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round2_matchestmp_timmed5_readyForRound2.fastq

#grep --color "GTTTTAGAGCTGTGTTGTTTCGAATGGTTCCAAAAC" /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/readswithrepeats_${adap}_intraReads_info_trimmed_matches_readyForRound2.fastq  |head
cleanRound=2
for cleanRound in $(echo "2 3 4 5 6")
  do
  echo ${cleanRound}
  echo "------------------------------------"
cleanRound_next=$((cleanRound+1))
  
cutadapt ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_readyForRound${cleanRound}.fastq -b ${adapSeq}  -e  0.1 -O 5 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.fasta
rm ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.fasta &
##finalised spacers
awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($5)>"25"  && length($5)<"46"&& length($5)==length($9)) print "@"$1, $5 ,"+", $9}' ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}/${names}_${adap}_final_round${cleanRound}_R${pair}.fastq


cat ${BaseLocation}/${names}/ReadPair_${pair}/${names}_${adap}_final_round${cleanRound}_R${pair}.fastq >> ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R${pair}.fastq


##finalised prepered
awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($7)>"25" && length($7)<"46" && length($7)==length($11)) print "@"$1, $7,"+", $11}' ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}/tmp/readswithrepeats_${adap}_${species}_round${cleanRound_next}_matchestmp_timmed5_readyForRound${cleanRound_next}.fastq

  done #cleanRound
  cat ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R1.fastq \
  ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R2.fastq > \
  ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads.fastq
  done #adap (spacer)
  
  ###-------------====
  ##reverse
    ###-------------====
#adapName=$(grep ">" /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut_reversed.txt| cut -d ' ' -f 1 |head -1)
for adapName in $(grep ">" ${repeatsFile}| cut -d ' ' -f 1)
  do
adapSeq=$(grep "${adapName}" -A 1 ${repeatsFile}|tail -1)
adap=$(echo ${adapName} | sed 's/>//g'|cut -d '_' -f 1)
cleanRound=1

echo "------------------------------------------------"
echo $adap
echo $adapSeq
echo $names
echo $read
echo $species
echo "------------------------------------------------"

mkdir -p ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/

cutadapt ${read} -b ${adapSeq}  -e  0.1 -O 15 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}.txt
rm ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}.txt &

## reads with match
awk '{if($2!="-1") print $0}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt

rm ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt & 

##make a list of current (potentially still containing 5' adapters)
awk -F "\t" '{OFS="\n"} {print "@"$1, $5,"+", $9}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt  > \
${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp.fastq


##trim 5' ends of these (with relaxed matching errors)
cutadapt ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp.fastq -g ${adapSeq}  -e  0.1 -O 5 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_.fastq.txt

rm ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_.fastq.txt &

## finalise first spacers (we will not be able to keep reads that don't have a small (>5bp) of repeat left)
##removing potential 5' adapters of first trim
mkdir -p ${BaseLocation}/${names}/final/
awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($7)>"25" && length($7)<"46" && length($7)==length($11)) print "@"$1, $7 ,"+", $11}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_info.txt > \
${BaseLocation}/${names}/ReadPair_${pair}_reversed/${names}_${adap}_final_round1_R${pair}_reversed.fastq

cat ${BaseLocation}/${names}/ReadPair_${pair}_reversed/${names}_${adap}_final_round1_R${pair}_reversed.fastq > ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R${pair}_reversed.fastq


##take remaining reads and remove 3' adapters again --> do this 5 times
awk -F "\t" '{OFS="\n"} {print "@"$1, $7,"+", $11}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed//tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info_trimmed.txt  > \
${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round2_matchestmp_timmed5_readyForRound2.fastq

#grep --color "GTTTTAGAGCTGTGTTGTTTCGAATGGTTCCAAAAC" /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/readswithrepeats_${adap}_intraReads_info_trimmed_matches_readyForRound2.fastq  |head
cleanRound=2
for cleanRound in $(echo "2 3 4 5 6")
  do
  echo ${cleanRound}
  echo "------------------------------------"
cleanRound_next=$((cleanRound+1))
  
cutadapt ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_matchestmp_timmed5_readyForRound${cleanRound}.fastq -b ${adapSeq}  -e  0.1 -O 5 --action=none --trimmed-only --info-file=${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt -o ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.fasta
rm ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.fasta &
##finalised spacers
awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($5)>"25" && length($5)<"46" && length($5)==length($9)) print "@"$1, $5 ,"+", $9}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}_reversed/${names}_${adap}_final_round${cleanRound}_R${pair}_reversed.fastq

cat ${BaseLocation}/${names}/ReadPair_${pair}_reversed/${names}_${adap}_final_round${cleanRound}_R${pair}_reversed.fastq >> ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R${pair}_reversed.fastq


##finalised prepered
awk -F "\t" '{OFS="\n"} {if($2!="-1" && length($7)>"25" && length($7)<"46" && length($7)==length($11)) print "@"$1, $7,"+", $11}' ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound}_info.txt > ${BaseLocation}/${names}/ReadPair_${pair}_reversed/tmp/readswithrepeats_${adap}_${species}_round${cleanRound_next}_matchestmp_timmed5_readyForRound${cleanRound_next}.fastq



  done #cleanRound
  cat ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R1_reversed.fastq \
  ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_R2_reversed.fastq > \
  ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed.fastq
  #revseq ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed.fastq ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular.fastq
  seqtk seq -r  ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed.fastq > ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular.fastq

  done #adap (spacer)
  
  done #pair (readPair)
  done #names (sampleName)

#seqtk seq -r /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//RMK202/final/RMK202_Array3_final_allRounds_allReads.fastq > /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//RMK202/final/RMK202_Array3_final_allRounds_allReads_tmp.fastq

##==============================================
##-----------------Spacers of Reference
##add fastq of the Reference 
##==============================================

##RMK202 metgenome
awk -F '__' '{print $1}' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/S_thermophilus_RMK202.pilarCR_out_final_new > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/S_thermophilus_RMK202.pilarCR_out_final_newshort

reformat.sh in=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/S_thermophilus_RMK202.pilarCR_out_final_new_short out=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/S_thermophilus_RMK202.pilarCR_out_final_newshort.fastq qfake=70

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/S_thermophilus_RMK202.pilarCR_out_final_newshort.fastq |sed 's/ary0/a/g' |sed 's/spc00/:s/g' | sed 's/spc0/:s/g' > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/S_thermophilus_RMK202.pilarCR_out_final_newshort_cleaned.fastq


    name=meta_rmk202
    grep ":a1:" -A 3 /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/S_thermophilus_RMK202.pilarCR_out_final_newshort_cleaned.fastq > /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//FinalSpacers/ForDADA2/${name}_allSpacermerged_Array1.fastq
    
    grep ":a2:" -A 3 /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/S_thermophilus_RMK202.pilarCR_out_final_newshort_cleaned.fastq > /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//FinalSpacers/ForDADA2/${name}_allSpacermerged_Array2.fastq
    
    grep ":a3:" -A 3 /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/S_thermophilus_RMK202.pilarCR_out_final_newshort_cleaned.fastq > /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//FinalSpacers/ForDADA2/${name}_allSpacermerged_Array3.fastq

##--------------------mst1
awk -F '__' '{print $1}' /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out_final_new > /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out_final_newshort

reformat.sh in=/archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out_final_newshort out=/archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out_final_newshort.fastq qfake=70

cat /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out_final_newshort.fastq |sed 's/ary0/a/g' |sed 's/spc00/:s/g' | sed 's/spc0/:s/g' > /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out_final_newshort_cleaned.fastq

  
    name=mst1
    grep ":a1:" -A 3 /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out_final_newshort_cleaned.fastq > /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//FinalSpacers/ForDADA2/${name}_allSpacermerged_Array1.fastq
    
    grep ":a2:" -A 3 /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out_final_newshort_cleaned.fastq > /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//FinalSpacers/ForDADA2/${name}_allSpacermerged_Array2.fastq
    
    grep ":a3:" -A 3 /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out_final_newshort_cleaned.fastq > /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//FinalSpacers/ForDADA2/${name}_allSpacermerged_Array3.fastq


##-------------------mst2
awk -F '__' '{print $1}' /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out_final_new > /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out_final_newshort

reformat.sh in=/archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out_final_newshort out=/archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out_final_newshort.fastq qfake=70

cat /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out_final_newshort.fastq |sed 's/ary0/a/g' |sed 's/spc00/:s/g' | sed 's/spc0/:s/g' > /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out_final_newshort_cleaned.fastq

  
  name=mst2
  grep ":a1:" -A 3 /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out_final_newshort_cleaned.fastq > /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//FinalSpacers/ForDADA2/${name}_allSpacermerged_Array1.fastq
  
  grep ":a2:" -A 3 /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out_final_newshort_cleaned.fastq > /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//FinalSpacers/ForDADA2/${name}_allSpacermerged_Array2.fastq
  
  grep ":a3:" -A 3 /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out_final_newshort_cleaned.fastq > /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//FinalSpacers/ForDADA2/${name}_allSpacermerged_Array3.fastq

##remove spacers again from the fastq collection
name=meta_rmk202
rm /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//FinalSpacers/ForDADA2/${name}*
name=mst1
rm /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//FinalSpacers/ForDADA2/${name}*
name=mst2
rm /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//FinalSpacers/ForDADA2/${name}*
##==============================================
##-----------------merge all samples spacers
##==============================================
rm ${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_${adap}.fastq
rm -r ${BaseLocation}/FinalSpacers/ForDADA2/
mkdir -p ${BaseLocation}/FinalSpacers/ForDADA2/
for names in $(cat ${samplesss} )
  do
for adapName in $(grep ">" /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt  )
  do
adapSeq=$(grep "${adapName}" -A 1 /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt)
adap=$(echo ${adapName} | sed 's/>//g'|cut -d '_' -f 1)
cleanRound=1

seqtk seq -r  ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed.fastq > ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular.fastq


#-----------change name



#awk -v sampless="$names" -v arrayss="$adapShor"  '{if(NR%4==1) $0=sprintf("@"sampless":"arrayss":s"++k,(1+i++)); print;}' ${BaseLocation}/FinalSpacers/ForDADA2/di_K2_6h_allSpacermerged_Array1.fastq


adapShor=$(echo $adap |sed 's/Array/a/g')
awk -v sampless="$names" -v arrayss="$adapShor"  '{if(NR%4==1) $0=sprintf("@"sampless":"arrayss":s"++k,(1+i++)); print;}' ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads.fastq > ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_renamed.fastq

adapShor=$(echo $adap |sed 's/Array/a/g')
awk -v sampless="$names" -v arrayss="$adapShor"  '{if(NR%4==1) $0=sprintf("@"sampless":R"arrayss":s"++k,(1+i++)); print;}' ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular.fastq > ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular_renamed.fastq



#-----------merge samples
mkdir -p  ${BaseLocation}/FinalSpacers/mergeSpacersFile
cat ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_renamed.fastq >> ${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_${adap}.fastq
cat ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular_renamed.fastq >> ${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_${adap}.fastq


##------ make a folder for dada2
cat ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_renamed.fastq >> ${BaseLocation}/FinalSpacers/ForDADA2/${names}_allSpacermerged_${adap}.fastq
cat ${BaseLocation}/${names}/final/${names}_${adap}_final_allRounds_allReads_reversed_regular_renamed.fastq >> ${BaseLocation}/FinalSpacers/ForDADA2/${names}_allSpacermerged_${adap}.fastq


done #adap (spacer)
  done #names (sampleName)

##----------------move local for Dada2

scp -r vincent@130.223.51.116:/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//FinalSpacers/ForDADA2/ ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/



##----Qualtiy control
grep ">" ${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_Array1.fasta | cut -d ':' -f 1  |sort|uniq -c
grep ">" ${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_Array2.fasta | cut -d ':' -f 1  |sort|uniq -c
grep ">" ${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_Array3.fasta | cut -d ':' -f 1  |sort|uniq -c

grep ">" ${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_Array1.fasta | cut -d ':' -f 2  |sort|uniq -c
grep ">" ${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_Array2.fasta | cut -d ':' -f 2  |sort|uniq -c
grep ">" ${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_Array3.fasta | cut -d ':' -f 2  |sort|uniq -c


##add REF
grep ":a01" -A 1 ${BaseLocation}/FinalSpacers/cdHit_spacers/allREF_CRISPRspacers.fasta |grep -v "\-\-"  > ${BaseLocation}/FinalSpacers/cdHit_spacers/allREF_CRISPRspacers_Array1.fasta
grep ":a02" -A 1 ${BaseLocation}/FinalSpacers/cdHit_spacers/allREF_CRISPRspacers.fasta |grep -v "\-\-"  > ${BaseLocation}/FinalSpacers/cdHit_spacers/allREF_CRISPRspacers_Array2.fasta
grep ":a03" -A 1 ${BaseLocation}/FinalSpacers/cdHit_spacers/allREF_CRISPRspacers.fasta |grep -v "\-\-"  > ${BaseLocation}/FinalSpacers/cdHit_spacers/allREF_CRISPRspacers_Array3.fasta


cat ${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_Array1.fasta ${BaseLocation}/FinalSpacers/cdHit_spacers/allREF_CRISPRspacers_Array1.fasta | sed 's/di_K2/d2/g' | sed 's/th_K2/t2/g' | sed 's/Lyo_202/L/g' | sed 's/lyo202/L/g' | sed 's/Konserve_202/K_/g'| sed 's/Versand/V_/g' | sed 's/RMK202/R202/g' > \
${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_Array1_REF.fasta

cat ${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_Array2.fasta ${BaseLocation}/FinalSpacers/cdHit_spacers/allREF_CRISPRspacers_Array2.fasta | sed 's/di_K2/d2/g' | sed 's/th_K2/t2/g' | sed 's/Lyo_202/L/g' | sed 's/lyo202/L/g' | sed 's/Konserve_202/K_/g'| sed 's/Versand/V_/g' | sed 's/RMK202/R202/g' > \
${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_Array2_REF.fasta

cat ${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_Array3.fasta ${BaseLocation}/FinalSpacers/cdHit_spacers/allREF_CRISPRspacers_Array3.fasta | sed 's/di_K2/d2/g' | sed 's/th_K2/t2/g' | sed 's/Lyo_202/L/g' | sed 's/lyo202/L/g' | sed 's/Konserve_202/K_/g'| sed 's/Versand/V_/g' | sed 's/RMK202/R202/g' > \
${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_Array3_REF.fasta



##--------------------------------------------------
##-----------cd -hit
##--------------------------------------------------

rm ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays.txt
for adapName in $(grep ">" /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt  )
  do
adapSeq=$(grep "${adapName}" -A 1 /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt)
adap=$(echo ${adapName} | sed 's/>//g'|cut -d '_' -f 1)
cleanRound=1
mkdir -p ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples.txt

/home/vincent/apps/cd-hit-v4.8.1-2019-0228/cd-hit-est -i ${BaseLocation}/FinalSpacers/mergeSpacersFile/allSamplesMergedspacers_${adap}_REF.fasta \
    -o  ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_${adap}.txt -c 0.88

echo "number of unique cluster for" $adap ":"
grep "^>Cluster" -c /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_${adap}.txt.clstr

###merge spacers
cat ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_${adap}.txt >> ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays.txt
    
done #adap (spacer)


##95= 585|314|603
##90= 342|181|430
##88= 295|179|402
##85= 289|177|394
##80= 285|173|387




##--Qualtiy control

grep ">" -c ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays.txt

##check for singletons
#/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_${adap}.txt.clstr

##--------------------------------------------------
##-----------check for singletons 
#we do this by looking at the size of clusters
##--------------------------------------------------
rm ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays.fasta
echo -e "Cluster\tadap\tREF_RMK202\tREF_mst1\tREF_mst2\tlyo202_96\tL_202_2012\tL_202_2014\tRMK202\tKonserve_202\tVersand_202\tdi_K2_6h\tth_K2_6h\tall\tall_REF\tREFspacer" > ${BaseLocation}/FinalSpacers/cdHit_spacers/counts/clusterCRISPRs_allArrays.txt

for adapName in $(grep ">" /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt)
  do
#adapSeq=$(grep "${adapName}" -A 1 /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt)
adap=$(echo ${adapName} | sed 's/>//g'|cut -d '_' -f 1)
rm ${BaseLocation}/FinalSpacers/cdHit_spacers/counts/clusterCRISPRs_${adap}.txt


#grep ">" ${BaseLocation}/assembledSpacers/allSpacersAssembled_short.fasta|cut -d ':' -f 1|sort|uniq -c
count=$(grep ">Cluster" -c ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_${adap}.txt.clstr)

#echo ">Cluster "${count} >> ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_${adap}.txt.clstr


##----------prime
start_num=0
i=1
 mkdir -p ${BaseLocation}/FinalSpacers/cdHit_spacers/counts
 
  #echo -e  "ClusterName\tREF_RMK202\tREF_mst1\tREF_mst2\tlyo202_96\tLyo_202_2012\tLyo_202_2014\tRMK202\tKonserve_202\tVersand_202\tdi_K2_6h\tth_K2_8h" > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledSpacers/cdHIT/count/clusterCRISPRs.txt
  
##---------start analysis 
for i in $(seq 1 $count)
  do 
  echo $i
  

  sed -n "/>Cluster $start_num$/,/>Cluster $i$/p" ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_${adap}.txt.clstr > ${BaseLocation}/FinalSpacers/cdHit_spacers/counts/tmpCluster.txt
  
  

  
di_K2_6h=$(grep ">d2_6h" -c ${BaseLocation}/FinalSpacers/cdHit_spacers/counts/tmpCluster.txt)  
Konserve_202=$(grep ">K_" -c ${BaseLocation}/FinalSpacers/cdHit_spacers/counts/tmpCluster.txt)  
L_202_2012=$(grep ">L_2012" -c ${BaseLocation}/FinalSpacers/cdHit_spacers/counts/tmpCluster.txt)  
L_202_2014=$(grep ">L_2014" -c ${BaseLocation}/FinalSpacers/cdHit_spacers/counts/tmpCluster.txt)  
lyo202_96=$(grep ">L_96" -c ${BaseLocation}/FinalSpacers/cdHit_spacers/counts/tmpCluster.txt)  
REF_mst1=$(grep ">REF_mst1" -c ${BaseLocation}/FinalSpacers/cdHit_spacers/counts/tmpCluster.txt)  
REF_mst2=$(grep ">REF_mst2" -c ${BaseLocation}/FinalSpacers/cdHit_spacers/counts/tmpCluster.txt)  
REF_RMK202=$(grep ">REF_R202" -c ${BaseLocation}/FinalSpacers/cdHit_spacers/counts/tmpCluster.txt)  
RMK202=$(grep ">R202" -c ${BaseLocation}/FinalSpacers/cdHit_spacers/counts/tmpCluster.txt)  
th_K2_6h=$(grep ">t2_8h" -c ${BaseLocation}/FinalSpacers/cdHit_spacers/counts/tmpCluster.txt)  
Versand_202=$(grep ">V__202" -c ${BaseLocation}/FinalSpacers/cdHit_spacers/counts/tmpCluster.txt)  
all=$(grep -v ">Cluster" ${BaseLocation}/FinalSpacers/cdHit_spacers/counts/tmpCluster.txt |wc -l)  
all_REF=$(grep ">REF" -c ${BaseLocation}/FinalSpacers/cdHit_spacers/counts/tmpCluster.txt)  
REFspacer=$(cat ${BaseLocation}/FinalSpacers/cdHit_spacers/counts/tmpCluster.txt| grep "*$"| awk '{print $3}' |sed 's/\.\.\.//g')

 echo -e "Cluster_"${start_num}"\t"${adap}"\t"${REF_RMK202}"\t"${REF_mst1}"\t"${REF_mst2}"\t"${lyo202_96}"\t"${L_202_2012}"\t"${L_202_2014}"\t"${RMK202}"\t"${Konserve_202}"\t"${Versand_202}"\t"${di_K2_6h}"\t"${th_K2_6h}"\t"${all}"\t"${all_REF}"\t"${REFspacer} >> ${BaseLocation}/FinalSpacers/cdHit_spacers/counts/clusterCRISPRs_${adap}.txt

  
  start_num=$((start_num+1))
  done #Clusters
  
    cat ${BaseLocation}/FinalSpacers/cdHit_spacers/counts/clusterCRISPRs_${adap}.txt >> ${BaseLocation}/FinalSpacers/cdHit_spacers/counts/clusterCRISPRs_allArrays.txt
    cat ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_${adap}.txt >> ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays.fasta
  
done #adap (spacer)



##--------bring local


scp  vincent@130.223.51.116:/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//FinalSpacers/cdHit_spacers/counts/clusterCRISPRs_allArrays.txt ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/


```

```{r}
library(ggplot2)
library(readr)
clusterCRISPRs_allArrays <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/clusterCRISPRs_allArrays.txt", "\t", escape_double = FALSE, trim_ws = TRUE)
clusterCRISPRs_allArrays[(is.na(clusterCRISPRs_allArrays$all)),]
###-------------------
##Verteilung der spacers
###-------------------


ggplot(clusterCRISPRs_allArrays,aes(x=all))+geom_density()+theme_classic()+lims(x=c(0,100))
ggplot(clusterCRISPRs_allArrays,aes(x=all))+geom_bar(width=20)+theme_classic()
ggplot(clusterCRISPRs_allArrays,aes(x=all))+geom_histogram(binwidth = 20)+theme_classic()+lims(x=c(0,100))

##only keep reads with 4 or more spacers

clusterCRISPRs_allArrays$keep <- ifelse(clusterCRISPRs_allArrays$all>4|clusterCRISPRs_allArrays$all_REF>0,"keep","singletons")
nrow(clusterCRISPRs_allArrays)
clusterCRISPRs_allArrays[which(clusterCRISPRs_allArrays$all<4&clusterCRISPRs_allArrays$all_REF>0),]
clusterCRISPRs_allArrays$adap
ggplot(clusterCRISPRs_allArrays,aes(x=keep,color=adap, fill=adap))+geom_bar(stat="count")
table(clusterCRISPRs_allArrays$keep)
clusterCRISPRs_keep <- clusterCRISPRs_allArrays[which(clusterCRISPRs_allArrays$keep=="keep"),]

write.table(clusterCRISPRs_keep, "~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/clusterCRISPRs_allArrays_keep.txt",na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names = TRUE)

```





```{bash}

##--------bring bigmachine

scp ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/clusterCRISPRs_allArrays_keep.txt  vincent@130.223.51.116:/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//FinalSpacers/cdHit_spacers/counts/clusterCRISPRs_allArrays_keep.txt

##--------------------------------------------------
##-----------subset and remove singlentons (everything that is present more than twice)
##--------------------------------------------------

sed -i 's/a01s16__48/a1:s16/g' /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//FinalSpacers/cdHit_spacers/counts/clusterCRISPRs_allArrays_keep.txt


sed -i 's/a01s16__481781__481813__S_thermophilus_mag_rmk202/a1:s16/g' ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays.fasta


mkdir -p ${BaseLocation}/FinalSpacers/cdHit_spacers/final/
rm ${BaseLocation}/FinalSpacers/cdHit_spacers/final/clusterCRISPRs_allArrays_keeps.txt
for keeps in $(cut -f 16 /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//FinalSpacers/cdHit_spacers/counts/clusterCRISPRs_allArrays_keep.txt);do

grep -A 1 "${keeps}$" ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays.fasta >> ${BaseLocation}/FinalSpacers/cdHit_spacers/final/clusterCRISPRs_allArrays_keeps.txt


done
#grep -A 1 ">REF_R202:a01s16__48" ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays.fasta >> ${BaseLocation}/FinalSpacers/cdHit_spacers/final/clusterCRISPRs_allArrays_keeps.txt

wc -l /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats//FinalSpacers/cdHit_spacers/counts/clusterCRISPRs_allArrays_keep.txt
grep -c "^>" ${BaseLocation}/FinalSpacers/cdHit_spacers/final/clusterCRISPRs_allArrays_keeps.txt


##--------------------------------------------------
##-----------create mappable and blastable file
##--------------------------------------------------
rm ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable.fasta
rm ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable.bed
rm ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable_final.fasta
#rm ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_blastable_final.fasta


for spacerss in $(grep ">" ${BaseLocation}/FinalSpacers/cdHit_spacers/final/clusterCRISPRs_allArrays_keeps.txt)
do

#grep ">" ${BaseLocation}/FinalSpacers/cdHit_spacers/final/clusterCRISPRs_allArrays_keeps.txt|cut -d ':' -f 2|cut -d 's' -f 1|sed 's/0//g' |sed 's/R//g' | sed 's/a/Array/g'
adap=$(echo $spacerss |cut -d ':' -f 2|cut -d 's' -f 1|sed 's/0//g' |sed 's/R//g' | sed 's/a/Array/g')
adapSeq=$(grep "${adap}" -A 1 /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/manual_spacer_cut.txt|tail -1)
startSpacer=$(echo $adapSeq|wc -c)

echo "=========================="
echo $adap
echo $adapSeq
echo "length:" $startSpacer
echo "=========================="



echo $spacerss

seq=$(grep "$spacerss" -A 1 ${BaseLocation}/FinalSpacers/cdHit_spacers/final/clusterCRISPRs_allArrays_keeps.txt |tail -1)

echo -e $spacerss"\n"$adapSeq"\n"$seq"\n"$adapSeq >> ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable.fasta


LENGTHpacer=$(echo $seq|wc -c)
endspacer=$((startSpacer+LENGTHpacer))
spacershort=$(echo $spacerss |sed 's/>//g')


echo -e $spacershort"\t"$startSpacer"\t"$endspacer"\t"$spacershort >> ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable.bed



done #spacerss (spacer)


fasta_formatter -w 50 -i ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable.fasta -o ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable_final.fasta

## Qualtiy control

grep ">" -c ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable_final.fasta
grep ">" ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable_final.fasta | cut -d ':' -f 1  |sort|uniq -c
grep ">" ${BaseLocation}/FinalSpacers/cdHit_spacers/uniqueSpacers_rmk202_sterm_allsamples_allArrays_mappabable_final.fasta | cut -d ':' -f 2  |sort|uniq -c


##--------------------------------------------------
##-----------coverage of Cas genes
#to compare with coverage of spcaers
##--------------------------------------------------
mkdir -p ${BaseLocation}/Cas_gene/

grep "cas" -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/S_thermophilus_RMK202_gene.gff > ${BaseLocation}/Cas_gene/Sterm_cas_genes.gff

BaseLocation_bam=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP
#rm ${BaseLocation}/Cas_gene/all_Sterm_cas_genes_coverage.gff
mkdir -p  ${BaseLocation}/Cas_gene/bedtools_cov/
for names in $(cat ${samplesss})
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))


#bedtools coverage -bed -a ${BaseLocation}/Cas_gene/Sterm_cas_genes.gff -b ${BaseLocation_bam}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam  >  ${BaseLocation}/Cas_gene/${names}_Sterm_cas_genes_coverage.gff
  
 # awk -v samplezz="$names" '{print samplezz,$9,$10,$11,$10/$11}' ${BaseLocation}/Cas_gene/${names}_Sterm_cas_genes_coverage.gff >> ${BaseLocation}/Cas_gene/all_Sterm_cas_genes_coverage.gff

bedtools bamtobed -i ${BaseLocation_bam}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam | \
bedtools coverage -a - -b ${BaseLocation}/Cas_gene/${names}_Sterm_cas_genes_coverage.gff \
> ${BaseLocation}/Cas_gene/bedtools_cov/${names}_Sterm_cas_genes_coverage_02.gff

   done
   

   
   ##-------------------------blast
   
   
   ##-----------------------map spacers
   
  
```



#####blast Spacers

```{bash}

##--------------------------blast to local phages 
  
#rm -r /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast
mkdir -p /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast

  mkdir -p /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/starterPhagesPlasmid
  
#  cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//splitContigs/L_del_plasmid_01.fasta \
#/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//splitContigs/S_thermophilus_repA.fasta \
#/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//splitContigs/L_del_phage_01.fasta \
#  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/splitContigs/*.fasta \
#  /archiv/Pro
#  jects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/splitContigs/L_del_plasmid_02.fsa > \
#/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/starterPhagesPlasmid/rmk202_phages_plasmids.fasta


#makeblastdb -max_file_sz 1GB -in /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/starterPhagesPlasmid/rmk202_phages_plasmids.fasta \
# -out /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/starterPhagesPlasmid/rmk202_phages_plasmids -parse_seqids -dbtype nucl

makeblastdb -max_file_sz 1GB -in /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/starterPhagesPlasmid/rmk202_phages_plasmids.fasta \
 -out /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/starterPhagesPlasmid/rmk202_phages_plasmids -parse_seqids -dbtype nucl


  blastn -num_threads 35 -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query ${BaseLocation}/FinalSpacers/cdHit_spacers/final/clusterCRISPRs_allArrays_keeps.txt \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen sgi sacc sblastnames staxids sscinames" \
  -db /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/starterPhagesPlasmid/rmk202_phages_plasmids \
  -out /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages.txt \
  -evalue 0.01 -word_size 16
  
##--------------------------blast to local bacteria
  
  ##create bacteria DB
  ##---------rmk202
bedtools maskfasta -fi /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_sterm/final/S_thermophilus_mag_rmk202.fasta -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/allStreptococcus_CRISPRlocation.bed \
-fo /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/Streptococcus_CRISPR.masked_rmk202Genome.fa

##---------S50
bedtools maskfasta -fi /archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/S50.fna -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/allStreptococcus_CRISPRlocation.bed \
-fo /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_S50_Genome.fa

##---------rmk202
bedtools maskfasta -fi /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/S72.fna -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/allStreptococcus_CRISPRlocation.bed \
-fo /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_S72_Genome.fa

##_rename contigs for easy search

sed -i 's/>/>REF_R202:/g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/Streptococcus_CRISPR.masked_rmk202Genome.fa
sed -i 's/>/>REF_mst1:/g' /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_S50_Genome.fa
sed -i 's/>/>REF_mst2:/g' /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_S72_Genome.fa

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/Streptococcus_CRISPR.masked_rmk202Genome.fa \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//circlaring_ldel/final/L_delbrueckii_mag_rmk202.fasta \
/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_S50_Genome.fa \
/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_S72_Genome.fa > \
/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_allStrepGenomesrmk202_Genome.fa


#makeblastdb -max_file_sz 1GB -in /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/starterPhagesPlasmid/rmk202_phages_plasmids.fasta \
# -out /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/starterPhagesPlasmid/rmk202_phages_plasmids -parse_seqids -dbtype nucl

makeblastdb -max_file_sz 1GB -in /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_allStrepGenomesrmk202_Genome.fa \
 -out /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_allStrepGenomesrmk202_Genome -parse_seqids -dbtype nucl


  blastn -num_threads 35 -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query ${BaseLocation}/FinalSpacers/cdHit_spacers/final/clusterCRISPRs_allArrays_keeps.txt \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen sgi sacc sblastnames staxids sscinames" \
  -db /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_allStrepGenomesrmk202_Genome \
  -out /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Bacteria.txt \
  -evalue 0.01 -word_size 16
  




  ####------------exkursion
  ##------check if the matches are not undetected CRISPR regions
  
  
  grep "NODE_" //archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Bacteria.txt |awk -F "\t" '{OFS="\t"} {if ($9<$10) print $2,$9-50,$10+50,$1 ; else print $2,$10-50,$9+50,$1 }' > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.txt

  
samtools faidx /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_allStrepGenomesrmk202_Genome.fa
  
bedtools getfasta -fi /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_allStrepGenomesrmk202_Genome.fa\
  -bed /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.txt \
  -fo /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.fasta

grep -B 1 --color "GAGCTGTGTTGTTTCGAATGGT" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.fasta |grep ">" > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_filtered.fasta
grep -B 1 --color "GTCCCCTCTCGAGGTAATTAGG" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.fasta|grep ">" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_filtered.fasta
grep -B 1  --color "ACAGTTACTTAAATCTTGAGAGTA" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.fasta |grep ">"  >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_filtered.fasta



grep -B 1 --color "GAACCATTCGAAACAACACAGCT" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.fasta |grep ">" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_filtered.fasta
grep -B 1 --color "CCTAATTACCTCGAGAGGGGACG" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.fasta|grep ">" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_filtered.fasta
grep -B 1  --color "GTACTCTCAAGATTTAAGTAACTG" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.fasta |grep ">"  >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_filtered.fasta

sed -i 's/>//g' /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_filtered.fasta

#remove the CRISPR repeats
rm /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/all_differentBlastHits.txt

for removess in $(cat /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_filtered.fasta)
do

contigss=$(echo $removess | cut -d ':' -f 1,2)
start=$(echo $removess | cut -d ':' -f 3|cut -d '-' -f 1)
end=$(echo $removess | cut -d ':' -f 3|cut -d '-' -f 2)
hit=$(echo $removess | cut -d ':' -f 1)
#echo $removess

explained=$(grep "${contigss}" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.txt | grep "${start}" |grep "${end}" |awk -F "\t" '{print $4}')

echo -e "OrphanCRISPR""\t"${explained}"\t"${hit} >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/all_differentBlastHits.txt

done




rm /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_spacers.txt
for removess in $(cat /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_filtered.fasta)
do
contigss=$(echo $removess | cut -d ':' -f 1)
start=$(echo $removess | cut -d ':' -f 2|cut -d '-' -f 1)
end=$(echo $removess | cut -d ':' -f 2|cut -d '-' -f 2)

grep "${contigss}" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes.txt | grep "${start}" |grep "${end}" |awk -F "\t" '{print $4}' >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_spacers.txt
done
cat /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages.txt > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_new.txt
for spazerss in $(cat /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_Genomes_spacers.txt)
do
echo ${spazerss}
grep "${spazerss}" -v /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_new.txt > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_tmp.txt 
cat /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_tmp.txt > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_new.txt

done

wc -l /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages.txt
wc -l /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages_new.txt

 ##-------------continue

  
# cut -f 1 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_viralREF.txt > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/mapped_spacersNames_onlyNames.txt
  

 # diff /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/mapped_spacersNames_onlyNames.txt \
#  /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/unmapped2local_spacersNames_onlyNames.txt | grep ">" | sed 's/> //g' >\
#  /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/unmapped_spacersNames_onlyNames.txt

##--------------------------blast to viral DB
#this is the vContact database

cat /archiv/Phage_DB/BLAST/20190519/downloads/Sterm/Renamed/RenamedContigs/*.fna \
/archiv/Phage_DB/BLAST/20190519/downloads/Ldel/Renamed/RenamedContigs/*.fna \
/archiv/Phage_DB/BLAST/20190519/downloads/Lactococcus/Renamed/RenamedContigs/*.fna > \
/archiv/Phage_DB/BLAST/20190519/Vcontact/Database/alltogether/data/merged_all_phages.fna



makeblastdb -max_file_sz 1GB -in /archiv/Phage_DB/BLAST/20190519/Vcontact/Database/alltogether/data/merged_all_phages.fna \
 -out /archiv/Phage_DB/BLAST/20190519/Vcontact/Database/alltogether/data/merged_all_phages -parse_seqids -dbtype nucl


  blastn -num_threads 35 -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query ${BaseLocation}/FinalSpacers/cdHit_spacers/final/clusterCRISPRs_allArrays_keeps.txt \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen sgi sacc sblastnames staxids sscinames" \
  -db /archiv/Phage_DB/BLAST/20190519/Vcontact/Database/alltogether/data/merged_all_phages \
  -out /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_ViralDB.txt \
  -evalue 0.01 -word_size 16
  


##--------------------------blast to old viral DB
#this is the RefSeq database

  blastn -num_threads 35 -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query ${BaseLocation}/FinalSpacers/cdHit_spacers/final/clusterCRISPRs_allArrays_keeps.txt \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen sgi sacc sblastnames staxids sscinames" \
  -db /archiv/BLAST/blast_db/20190204/viral_nucleotide/viral \
  -out /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_ViralDB.txt \
  -evalue 0.01 -word_size 16
  
  
  
##--------------------------blast unmapped to nucleotide DB

#rm /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/spacers/unmapped_spacers4blast.fasta
#for i in $(cat /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/unmapped_spacersNames_onlyNames.txt)
#do
#grep -A 1 "${i}" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/spacers/all_spacers4blast.fasta >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/spacers/unmapped_spacers4blast.fasta
#done


blastn -num_threads 37 -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query ${BaseLocation}/FinalSpacers/cdHit_spacers/final/clusterCRISPRs_allArrays_keeps.txt \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen" \
  -db /archiv/BLAST/blast_db/20190204/nucleotide/prokaryotes \
  -out  /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_nucleotideDB.txt \
  -evalue 0.01 -word_size 16




##-----------------------merge all blasts


  cut -f 1,2 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202phages.txt |sed 's/_contig[[:digit:]]\+//g'  | awk -F "[\t]" '{OFS="\t"} {print "localPHAGE",$1,$2}' >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/all_differentBlastHits.txt

  awk -F "\t" '{OFS="\t"} {print $1 ,$2 }' /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Bacteria.txt | awk -F "[\t:]" '{OFS="\t"} {print "localBAC",$1":"$2":"$3,$4}' >>  /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/all_differentBlastHits.txt


  cut -f 1,13 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_ViralDB.txt |cut -d ',' -f 1 | sed 's/ complete prophage genome//g' |sed 's/ complete genome//g'  | awk -F "[\t]" '{OFS="\t"} {print "phageDB",$1,$2}' >>  /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/all_differentBlastHits.txt

cut -f 1,13 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_nucleotideDB.txt |cut -d ',' -f 1 | awk -F "[\t]" '{OFS="\t"} {print "BactDB",$1,$2}' >>  /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/all_differentBlastHits.txt

cut -f 1 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/all_differentBlastHits.txt | sort|uniq -c

  ##-------------------------transfer local
  #scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_viralREF.txt ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/
 # scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/all_spacersNames.txt ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/
#  scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_unmapped2prokaryoteREF.txt ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/
  
 # scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages.txt ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/ 
  
   scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/all_differentBlastHits.txt ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/ 
  
```


#####BacBlast
Here, I want to disentangle to what the CRISPRs are mapping to on the bacteria. 
I hyptophesised that they might against parasitic elements, like:
-transposase
-prophages
-mobile elements
##### Transposase

Annotated with new prokka to get IS-finder calling.
This step is already done in the annotation chunk. 
I do not know if it is at all necessary anymore.

```{bash}

##========================
##PROKKA annotation
##already done in annotation chunk
##========================
num=0
species=Sterm

##---------------------S50
genomes=202-S50
head -5 /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA_*.gff


num_tmp=$(grep -c "ISfinder" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA_*.gff)
num=$((num+num_tmp))
echo $num_tmp " in " $genome
echo $num
##---------------------S50
genome=202-S72
#rm -r /archiv/Projects/2019_RMK202_analysis/Prokka_1_14_withIS/${genome}
#mkdir -p /archiv/Projects/2019_RMK202_analysis/Prokka_1_14_withIS/${genome}
    
#prokka --outdir /archiv/Projects/2019_RMK202_analysis/Prokka_1_14_withIS/${genome} --force --usegenus --genus Streptococcus --locustag ${genome} --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2018_Culturomics/06_Spades/${genome}/Prokka/*/${genome}.fna

grep -c "ISfinder" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA_*.gff

num_tmp=$(grep -c "ISfinder" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA_*.gff)
num=$((num+num_tmp))
echo $num_tmp " in " $genome
echo $num
##---------------------S50
genome=202-SMAG
#rm -r /archiv/Projects/2019_RMK202_analysis/Prokka_1_14_withIS/${genome}
#mkdir -p /archiv/Projects/2019_RMK202_analysis/Prokka_1_14_withIS/${genome}
    
#prokka --outdir /archiv/Projects/2019_RMK202_analysis/Prokka_1_14_withIS/${genome} --force --usegenus --genus Streptococcus --locustag ${genome} --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/PGAP_assembly/sterm/S_thermophilus_RMK202.current.fna

grep -c "ISfinder" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA_*.gff


num_tmp=$(grep -c "ISfinder" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomes}/PROKKA_*.gff)
num=$((num+num_tmp))
echo $num_tmp " in " $genome
echo $num

mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/

echo -e "transposase\ttotal\t"${num}"\tall" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest.txt

echo -e "transposase\tmatched\t0\tall" >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest.txt

##---------------------alltogether
/data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_allStrepGenomesrmk202_Genome.fa

#rm -r /archiv/Projects/2019_RMK202_analysis/Prokka_1_14_withIS/${genome}
mkdir -p /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/PROKKA
   
sed 's/_length.*$//g' /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_allStrepGenomesrmk202_Genome.fa > /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_allStrepGenomesrmk202_Genome_short.fa
    
/usr/bin/perl /usr/bin/prokka --outdir //data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/PROKKA --force --usegenus --genus Streptococcus --locustag alltogether --proteins proteins.faa --evalue 0.001 --addgenes /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/Streptococcus_CRISPR.masked_allStrepGenomesrmk202_Genome_short.fa

grep -c "ISfinder" /data/Project/CRISPRspacers/02_PilerCR/20190112_Streptococcus_thermophilus_withRMK202/ReadsWithRepeats/PROKKA/**.gff


num_tmp=$(grep -c "ISfinder" /archiv/Projects/2019_RMK202_analysis/Prokka_1_14_withIS/${genome}/**.gff)
num=$((num+num_tmp))
echo $num_tmp " in " $genome
echo $num

echo -e "transposase\ttotal\t"${num}"\tall" > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest.txt

echo -e "transposase\tmatched\t0\tall" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest.txt

```


######BacBlast
Where do they blast to on bacterial genome

```{bash}

###--------------------------
##prepare bed of protospacers
###--------------------------
#/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/CRISPRblast_rmk202Bacteria.txt

awk -F "\t" '{OFS="\t"}{if($9>$10) print $2,$10,$9,$1;else print $2,$9,$10,$1 }' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/CRISPRblast_rmk202Bacteria_02.txt |sed 's/REF_mst2://g' |sed 's/REF_mst1://g' |sed 's/REF_R202://g'| sed s'/S_thermophilus_mag_rmk202/CP046134/g' | sed s'/L_delbrueckii_mag_rmk202/CP046131/g' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/locations_bacteria.bed


###--------------------------
##prepare large gff of all bacteria used for protospacer mapping
###--------------------------
species=Sterm
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/all_gffs_toExtractfrom.gff
for genomess in $(echo "202-SMAG 202-S50 202-S72")
do

grep "$(printf '^')${genomess}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA_*.gff  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/all_gffs_toExtractfrom.gff

done

grep "^202-LMAG" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/Ldel/202-LMAG/PROKKA_*.gff  >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/all_gffs_toExtractfrom.gff

###===============================0
##also do this for the uniq bacteria DB matches
#n=5
##look this up in chunk 
###===============================
for gensss in $(echo "META_Cluster_106 META_Cluster_132 META_Cluster_51")
do
grep "${gensss}" ${BaseLocation}/CRISPRspacerBLAST/blast/CRISPRblast_nucleotideDB.txt
done

###--------------------------
##intersect protspacers
##with bacteria Prokka genes
###--------------------------

bedtools intersect -a /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/all_gffs_toExtractfrom.gff  \
  -b /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast/locations_bacteria.bed -wb > \
   /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Spacers_from_Metagenome/SpacerExtract//CRISPRspacerBLAST/blast//intersect_genes.gff

num=0
genesCount=$(awk '{if($3=="gene") print $0}' /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/intersect_genes.gff |cut -f 13|sort|uniq |wc -l)
 num=$((num+genesCount))
 echo ${num}
 
 num_tot=0
 genesCount=$(grep "^CP046131" -v /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/all_gffs_toExtractfrom.gff|awk '{if($3=="gene") print $0}'|sort|uniq |wc -l)
 
 num_tot=$((num_tot+genesCount))
 echo ${num_tot}

echo -e "genes\ttotal\t"${num_tot}"\tall" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest.txt

echo -e "genes\tmatched" > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest_tmp.txt
echo -e "genes\tmatched" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest_tmp.txt
echo -e "genes\tmatched" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest_tmp.txt

awk '{if($3=="gene") print $0}' /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/intersect_genes.gff |cut -f 13|sort|uniq| sed 's/:R/:/g' |cut -d ':' -f 2 |sort|uniq -c| sed -e 's/^[[:space:]]*//' |sed 's/ /\t/g' > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest_tmp2.txt

paste -d ',' /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest_tmp.txt /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest_tmp2.txt |sed 's/,/\t/g' >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest.txt


###------------------------------
##extract genes
###------------------------------
awk '{if($3=="gene") print $0}' /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/intersect_genes.gff | sed 's/_gene;/\t/g' | sed 's/;Name=/\t/g' > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/intersect_genes_prep.gff
 
 
rm -r /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/genesMapped/
mkdir -p /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/genesMapped/
 
genesss=S72_00155
for genesss in $(cut -f 9 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/intersect_genes_prep.gff | sed 's/ID=//g')
do

genome=$(echo ${genesss} | awk -F "_0" '{print $1}')
 
  echo ${genesss}
samtools faidx /archiv/Projects/2019_RMK202_analysis/Prokka_1_14_withIS/${genome}/*.faa ${genesss} >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/genesMapped/mappedGenes_forKEGG.faa

done

cut -f 9,14 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/intersect_genes_prep.gff | sed 's/ID=//g' > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/intersect_genes_prep_prepped.txt

##-----------------

#is elements


###-----------------transfer local for eggnog


scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/genesMapped/mappedGenes_forKEGG.faa ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/


scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/intersect_genes_prep_prepped.txt ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/


```

#####Prophage

Detection of prophage with phaster.
The genomic location has to be manually extracted out of the output file. 
The output files are save.

```{bash}

#saved on local machine at
ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/06_Phaster


###------------------on bigmachine
mkdir -p /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/Prophages/

##rmk202_sterm (complete)
echo -e "CP046134\t850670\t877821" > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/Prophages/phasterAnnotatedProphage.bed

##S50 (incomplete)
echo -e "NODE_37_length_31935_cov_62.349252\t10121\t20946" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/Prophages/phasterAnnotatedProphage.bed

##S72 (incomplete)
echo -e "NODE_23_length_38834_cov_28.794430\t17059\t27885" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/Prophages/phasterAnnotatedProphage.bed

##ldel rmk202 (two incomplete)
echo -e "CP046131\t212504\t220802" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/Prophages/phasterAnnotatedProphage.bed
echo -e "CP046131\t1990687\t1997908" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/Prophages/phasterAnnotatedProphage.bed

sed 's/CP046134/S_thermophilus_mag_rmk202/g'  /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/Prophages/phasterAnnotatedProphage.bed | sed s'/CP046131/L_delbrueckii_mag_rmk202/g' > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/Prophages/phasterAnnotatedProphage_changedNames.bed

##--------------intersect

bedtools intersect -a /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/Prophages/phasterAnnotatedProphage_changedNames.bed  \
  -b /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/locations_bacteria.bed -wb > \
   /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/intersect_prophage.gff


num=0
num_tmp=$(wc -l /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/intersect_prophage.gff)
num=$((num+num_tmp))
echo $num_tmp " in " $genome
echo $num

num_total=0
num_tmp=$(wc -l //archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/Prophages/phasterAnnotatedProphage_changedNames.bed|cut -d ' ' -f 1)
num_total=$((num_total+num_tmp))
echo $num_tmp " in " $genome
echo $num_total

echo -e "prophage\ttotal\t"${num_total}"\tall" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest.txt

echo -e "prophage\tmatched\t"${num}"\tall" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest.txt


```

The spacers do not seem to be matching to the prophages!

#####mobile elements

prediction with SRID from the [ALM paper](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0223680#sec002)

```{bash}


 
###================
##description file
###================
threads=35
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt  ##the file with all sample names
logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_allRMKs202
#Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/assemblyFinalwithIsolates.fasta
#name_folder=02_againstpolished_allRMKs202

##------------------
#prep
##------------------


###================
##script
###================
##============
##mapping to reference
##============

mkdir -p ${logFilelocation}

num=1

for names in $(cat ${samplesss})
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))
  
  
#mean insertSize

meanInsert=$(grep "mean insert size" ${BaseLocation}/${names}/bwaMapping2DB/bamqc//genome_results.txt |awk -F "= " '{print $2}'|sed 's/,//g' |cut -d '.' -f 1)
stdInsert=$(grep "std insert size" ${BaseLocation}/${names}/bwaMapping2DB/bamqc//genome_results.txt |awk -F "= " '{print $2}'|sed 's/,//g'|cut -d '.' -f 1)


##average read length
ReadLength=$(samtools view ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam  |awk '{print length($10)}' | awk '{ total += $1; count++ } END { print total/count }'|cut -d '.' -f 1 )


echo "========================================"
echo "sample: "$names
echo "mean read length: " ${ReadLength}
echo "mean insert size: " ${meanInsert}
echo "std insert size: " ${stdInsert}
echo "========================================"

  rm -r ${BaseLocation}/02_mobile_elements/${names}

  mkdir -p ${BaseLocation}/02_mobile_elements/${names}
  
/home/vincent/scripts/SRID/SRID.py \
  --bam ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam \
  --out ${BaseLocation}/02_mobile_elements/${names} \
  --readlen ${ReadLength} \
  --mean ${meanInsert} \
  --sd ${stdInsert} \
  --threads ${threads} \
  

done


##----------extract sequences
rm ${BaseLocation}/02_mobile_elements/all_samples_mobileElements.bed
for names in $(cat ${samplesss})
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))
  
  
  awk -F "\t" -v samplezzz="$names" '{OFS="\t"} {print $1,$2,$3,samplezzz}' ${BaseLocation}/02_mobile_elements/${names}/*.tab | grep -v "^Scaffold"  >>  ${BaseLocation}/02_mobile_elements/all_samples_mobileElements.bed
  
done


##---------------------------
##see if spacers map to putative mobile elements
##---------------------------
cut -f 1,2,3 ${BaseLocation}/02_mobile_elements/all_samples_mobileElements.bed |sort|uniq -c

cut -f 1,2,3 ${BaseLocation}/02_mobile_elements/all_samples_mobileElements.bed |sort|uniq  > ${BaseLocation}/02_mobile_elements/all_samples_mobileElements_mapping.bed

bedtools intersect -a ${BaseLocation}/02_mobile_elements/all_samples_mobileElements_mapping.bed  \
  -b /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/locations_bacteria.bed -wb > \
   ${BaseLocation}/02_mobile_elements/mappedCRISPRs_mobileElements_mapping.bed


##------------------------count
#cut -f 7 ${BaseLocation}/02_mobile_elements/mappedCRISPRs_mobileElements_mapping.bed|sort|uniq|sed 's/:R/:/g' |cut -d ':' -f 2|sort|uniq -c | sed -e 's/^[[:space:]]*//' |sed 's/ /\t/g'


 
 num_tot=0
 genesCount=$(wc -l ${BaseLocation}/02_mobile_elements/all_samples_mobileElements_mapping.bed |cut -d ' ' -f 1)
 
 num_tot=$((num_tot+genesCount))
 echo ${num_tot}

echo -e "mobile elements\ttotal\t"${num_tot}"\tall" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest.txt

echo -e "mobile elements\tmatched" > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest_tmp.txt

cut -f 7 /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_allRMKs202/02_mobile_elements/mappedCRISPRs_mobileElements_mapping.bed|sort|uniq|sed 's/:R/:/g' |cut -d ':' -f 2|sort|uniq -c | sed -e 's/^[[:space:]]*//' |sed 's/ /\t/g' > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest_tmp2.txt

paste -d ',' /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest_tmp.txt /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest_tmp2.txt |sed 's/,/\t/g' >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest.txt





```

The spacers do not seem to be matching to the mobile elements!
only spacers that map to the CRISPR arrays

###remaining spacers

```{bash}

awk '{if($3=="gene") print $0}' /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/intersect_genes.gff |cut -f 13|sort|uniq >  /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/matchspacers.txt

cut -f 7 ${BaseLocation}/02_mobile_elements/mappedCRISPRs_mobileElements_mapping.bed|sort|uniq >>  /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/matchspacers.txt


sort /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/matchspacers.txt |uniq|sort > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/matchspacers_uniq.txt

##all spacers
cut -f 4 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/locations_bacteria.bed |sort> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/matchspacers_all.txt

diff /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/matchspacers_uniq.txt /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/matchspacers_all.txt |grep ">" |sed 's/> //g' |sed 's/:R/:/g' |cut -d ':' -f 2 |sort|uniq -c | sed -e 's/^[[:space:]]*//' |sed 's/ /\t/g' >  /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/matchspacers_all_remaining_tmp1.txt


echo -e "remaining\tmatched" > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/matchspacers_all_remaining_tmp2.txt
echo -e "remaining\tmatched" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/matchspacers_all_remaining_tmp2.txt
echo -e "remaining\tmatched" >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/matchspacers_all_remaining_tmp2.txt

paste -d ',' /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/matchspacers_all_remaining_tmp2.txt /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/matchspacers_all_remaining_tmp1.txt |sed 's/,/\t/g' >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest.txt


##-----------------------------------------



scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/Number_genes_of_interest.txt ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/


```


```{r}
library(ggplot2)
library(readr)
library(RColorBrewer)
library(plyr)
library(tidyverse)


Number_genes_of_interest <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/Number_genes_of_interest.txt",  "\t", escape_double = FALSE, col_names = c("grouping","fraction","spacerCount","array"),  trim_ws = TRUE)

Number_genes_of_interest_01 <- Number_genes_of_interest[which(Number_genes_of_interest$fraction!="total"),]

Number_genes_of_interest_02 <- rbind(Number_genes_of_interest_01,c("mobile elements","mapped","0","a2"))
Number_genes_of_interest_03 <- rbind(Number_genes_of_interest_02,c("mobile elements","mapped","0","a3"))

Number_genes_of_interest_03$spacerCount <- as.numeric(Number_genes_of_interest_03$spacerCount)
 Number_genes_of_interest_03$array  <- revalue( Number_genes_of_interest_03$array, c("a1"="Array 1", "a2"="Array 2",   "a3"="Array 3"))

Number_genes_of_interest_03$grouping = factor(Number_genes_of_interest_03$grouping, levels=(c("transposase" ,"prophage" , "mobile elements" , "genes" , "remaining")))
Number_genes_of_interest_03$array = factor(Number_genes_of_interest_03$array, levels=(c("Array 1", "Array 2",   "Array 3","all")))

spacersmapped_genes_plot <-  ggplot(Number_genes_of_interest_03,aes(x=grouping,y=spacerCount,group=array,color=array,fill=array))+geom_bar(width=0.7,stat="identity",position = position_dodge(width=0.8))+theme_classic()+scale_y_discrete( expand = c(0, 0)) +lims(y=c(-0.5,50))+ theme(legend.title = element_blank(),axis.text.x = element_text(angle = 75, hjust = 1,size=9))+labs(x="",y="spacers mapped")+
    scale_fill_manual(values=three_reds)+
   scale_color_manual(values=three_reds)

spacersmapped_genes_plot
png("~/Desktop/Projects/2019_RMK202_analysis/plot/MappingDB_mappingtoBac.png", width = 1000, height = 1000,res=300)
 # svg("~/Desktop/Projects/2019_RMK202_analysis/plot/MappingDB_perArray.svg",width=11,height=8)

spacersmapped_genes_plot


dev.off()


```

######eggnog

```{r}
library(readr)
genes_soacer <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE,  skip = 3)

intersect_genes_prep_prepped <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/intersect_genes_prep_prepped.txt", "\t", escape_double = FALSE, col_names = c("query_name","spacerName"),   trim_ws = TRUE)


S50 <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3)) 

genes_soacer <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3)) 
colnames(genes_soacer) <- c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description")

COG_functional_categories <- read_delim("~/Desktop/ScriptRepository/COG_functional_categories.csv",  ",", escape_double = FALSE, col_names = c("short","long","divided"),trim_ws = TRUE, na = "NA")
##----------------COG

ALL_sterm <- genes_soacer
nrow(ALL_sterm)

ALL_sterm <- ALL_sterm %>% 
    mutate(COG_Functional_Category = strsplit(as.character(COG_Functional_Category), "")) %>% 
    unnest(COG_Functional_Category)

nrow(ALL_sterm)
# ALL_sterm$co
#unique(sterm_all_unique$COG_Functional_Category)[match(COG_functional_categories$short , unique(sterm_all_unique$COG_Functional_Category))]
unique(ALL_sterm$COG_Functional_Category)[which(!unique(ALL_sterm$COG_Functional_Category) %in% COG_functional_categories$short )]
# ALL_sterm$
# table(ALL_sterm$COG_Functional_Category)

i=4
ALL_sterm$longCOG <- NA
for (i in 1:nrow(ALL_sterm)) {
  ALL_sterm[i,"longCOG"] <- COG_functional_categories[which(COG_functional_categories$short==as.character(ALL_sterm[i,"COG_Functional_Category"])),"long"] %>% as.character()
  
}

# ALL_sterm[which(is.na(ALL_sterm$longCOG)),"COG_Functional_Category"] %>% table()


# ALL_sterm_unnested_not <- ALL_sterm %>% select(COG_Functional_Category)

# ALL_sterm <- ALL_sterm[which(!is.na(ALL_sterm$longCOG)),]
# unique(ALL_sterm$COG_Functional_Category)[which(!unique(ALL_sterm$COG_Functional_Category) %in% COG_functional_categories$short )]


   # svg("~/Desktop/presenations/my_presentations/20190807_committeeMeeting/figures/cogPlot_allSterm.svg",width=7,height=7)
# CogPlotSterm 
 # dev.off()

##-----------------------
##add array information
##-----------------------
ALL_sterm$query_name



ALL_sterm_sub <- ALL_sterm %>% drop_na(COG_Functional_Category) %>% select(query_name,longCOG)
# ALL_sterm[which(ALL_sterm$longCOG=="character(0)"),]%>% select(-1,-2,-3)
ALL_sterm_extended <- merge(ALL_sterm_sub,intersect_genes_prep_prepped,by="query_name") 

ALL_sterm_extended$spacerName <- gsub(":R",":",ALL_sterm_extended$spacerName) 

ALL_sterm_extended$array <- str_split_fixed(ALL_sterm_extended$spacerName, ":", 3)[,2]

 ALL_sterm_extended$array  <- revalue( ALL_sterm_extended$array, c("a1"="Array 1", "a2"="Array 2",   "a3"="Array 3"))

ALL_sterm_extended$array = factor(ALL_sterm_extended$array, levels=(c("Array 1", "Array 2",   "Array 3","all")))


CogPlotSterm <- ggplot(ALL_sterm_extended,aes(x=forcats::fct_infreq(longCOG),group=array,color=array,fill=array))+geom_bar()+
  theme_classic()+
  # facet_wrap(array ~ .)+
  labs(x="",y="spacers mapped")+
    scale_fill_manual(values=three_reds)+
   scale_color_manual(values=three_reds)+
  theme(axis.text.x = element_text(angle = 60, hjust = 1,size=6),
          # # axis.text.y.left = element_text(color = "blue"),
          # axis.text.y.right = element_text(color="red"),
          # axis.title.y.left = element_text(color="blue"),
          # axis.title.y.right = element_text(color="red"),
          # rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          #legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank(),
          plot.margin=unit(c(0,0,0,2), "cm"),
          # axis.text.x.bottom = element_text(size=10)
          #legend.position = c(0.02, 0.98),
          #legend.justification = c("left", "top")
          )

library(patchwork)

spacersmapped_genes_plot/CogPlotSterm

png("~/Desktop/Projects/2019_RMK202_analysis/plot/MappingDB_mappingtoBac_eggnog.png", width = 1000, height = 2000,res=300)

CogPlotSterm

dev.off()


```


###Cas-genes alignment
added 28.8.2020

In order to annotate the Cas-genes we need to annote the genomes with Prokka

```{bash}

grep -i "isfinder"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/*.gff
grep -i "=cas"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/*.gff
grep -i "=csn"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/*.gff
grep -i "=Csy"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/*.gff
grep -i "=Csm"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/*.gff



```


##### 16s rRNA analysis
added 28.8.2020

Here, I look if the 16s regions are divergent.

```{bash}

for species in $(echo "Ldel Sterm")
do

rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/barnap/
mkdir -p //archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/barnap

#for genomess in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/ |grep ".fna$"|cut -d '_' -f 1|sed 's/.fna//g')
for genomess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/${species}/ |grep "_202_"|grep ".fasta$" |sed 's/.fasta//g')
do


 barrnap -q -k bac /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.fna --outseq //archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/barnap/${genomess}_rRNA.fasta > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/barnap/${genomess}_rRNA.gff
 ###-------------------gather 16s genes
 for genessss in $(grep "16S_" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/barnap/${genomess}_rRNA.gff|grep "partial" -v -i)
do
samtools faidx /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/barnap/${genomess}_rRNA.fasta ${genessss} >>  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/barnap/${genomess}_16s_rRNA.fasta

done #genessss
#----------------cd-hit
/home/vincent/apps/cd-hit-v4.8.1-2019-0228/cd-hit-est -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/barnap/${genomess}_16s_rRNA.fasta \
    -o  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/barnap/${genomess}_16s_rRNA_clustering.txt -c 0.80
    
  sed -i 's/\*/at +\/100.00%/g' /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/barnap/${genomess}_16s_rRNA_clustering.txt.clstr  
  
average16s=$(grep -v ">Cl" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/barnap/${genomess}_16s_rRNA_clustering.txt.clstr|sed 's/%//g' |awk -F "+/"  '{sum+=$2} END { print sum/NR}')
number16s=$(grep "^>" -c /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/barnap/${genomess}_16s_rRNA.fasta)

echo -e ${genomess}"\t"${number16s}"\t"${average16s} >>  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/barnap/${genomess}_16s_rRNA_intraspecies_div.txt


done #genomes
done # species

```

##### Codon usage
added 28.8.2020
Here, I look if the proteins of the different strains or species show a bias for certain codons or proteins. 
I do this with a perl package callled fascodon. 
I have the Codon (n=60) usage for all 20 nucleotides for every gene. 
Maybe I have to subset for only the core genes or something like that.

```{bash}

###------------------------------------
##make header
###------------------------------------

species=Ldel
genomess=$(echo "202-11142")

sed 's/ .*//g' /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.ffn  > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/tmp.ffn

grep "^>" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.ffn |  awk -F " " '{OFS=" "}{print $2,$3,$4,$5,$6,$7}' |sed 's/[[:space:]]*$//'  > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/geneNamesfromFFN.txt

rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/05_CodonUsage/fascodon/${species}/${genomess}
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/05_CodonUsage/fascodon/${species}/${genomess}

cd /archiv/Projects/2019_RMK202_analysis/00_FINAL/05_CodonUsage/fascodon/${species}/${genomess}
fascodon  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/tmp.ffn |grep ">"|head -1 |sed 's/>//g' |sed 's/:[0-9]\+//g' |sed 's/202-11142_00001 /geneName/g'| sed 's/ /\t/g' |awk -F "\t"  '{OFS="\t"}{print "geneDescription","SPECIES","GENOMESS",$0}' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/05_CodonUsage/fascodon/codonUsageAllstrains.txt

###------------------------------------
##calculate codon usage
###------------------------------------

for species in $(echo "Ldel Sterm")
do

#for genomess in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/ |grep ".fna$"|cut -d '_' -f 1|sed 's/.fna//g')
for genomess in $(ls /data/Project/2020_StarterCultureDiversity/10_REFERENCE_GEN0MES/genomes/RMKsonly/FINAL/${species}/ |grep "_202_"|grep ".fasta$" |sed 's/.fasta//g')
do

sed 's/ .*//g' /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.ffn |sed 's/ //g' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/tmp.ffn

grep "^>" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.ffn|  awk -F " " '{OFS=" "}{print $2,$3,$4,$5,$6,$7}' |sed 's/[[:space:]]*$//'  > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/geneNamesfromFFN.txt

rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/05_CodonUsage/fascodon/${species}/${genomess}
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/05_CodonUsage/fascodon/${species}/${genomess}

cd /archiv/Projects/2019_RMK202_analysis/00_FINAL/05_CodonUsage/fascodon/${species}/${genomess}
fascodon  /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/tmp.ffn |grep ">" |  sed 's/>//g' |sed 's/  / /g' |sed 's/ /\t/g'|sed 's/freq_.\{1\}_.\{3\}://g' |awk -F "\t" -v SPECIES="$species" -v GENOMESS="$genomess" '{OFS="\t"}{print SPECIES,GENOMESS,$0}' > ${genomess}_codonUsage.txt

paste /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/geneNamesfromFFN.txt ${genomess}_codonUsage.txt >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/05_CodonUsage/fascodon/codonUsageAllstrains.txt

rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/tmp.ffn &
done #genomes
done # species


```

## check plasmid cov
here I check if all ldel strains have the plasmid
added 28.8.2020

```{bash}
###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2018_Culturomics/02_script/names_changed.txt ##the file with all sample names
logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/PlasmidCoverage
Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids_PGAP/RMK202_MAG_assembly.fasta

##!!!!!!!!!!!!!change names of Forward and Reverse reads

for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/names_changed.txt)
  do
  echo ==========================
   echo ${id}
   echo ==========================
 
 
       mkdir -p $mainPath/06_Spades/${id}/assembly/
 
 
     /home/vincent/miniconda2/pkgs/spades-3.13.0-0/bin/spades.py --pe1-1 $mainPath/01_rawData/trimmed_Galore/gz/${id}/${id}_R1_val_1.fq.gz \
       --pe1-2 $mainPath/01_rawData/trimmed_Galore/gz/${id}/${id}_R2_val_2.fq.gz \
       -o $mainPath/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta -t ${threads} -m 100

   
    done

###================
##merge genome
###================



###================
##script
###================
##============
##mapping to reference
##============
bwa index $Assembly
# name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 

mkdir -p ${logFilelocation}

num=1

for names in $(cut -f 2 ${samplesss}| )
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))
  rm -r ${BaseLocation}/${names}

  mkdir -p ${BaseLocation}/${names}/bwaMapping2DB/
  
 # bwa mem -t ${threads} /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids/di_rmk202_MAG_reference_polished.fasta
  bwa mem -t ${threads} ${Assembly} \
    /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_1.fastq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_2.fastq | samtools sort -@${threads} -O BAM -o ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 

samtools view -b -f 4 ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

bedtools bamtofastq -i ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam -fq ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq

rm ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam &

##-----Qualimap

mkdir -p ${BaseLocation}/${names}/bwaMapping2DB//bamqc

qualimap bamqc -bam ${BaseLocation}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir ${BaseLocation}/${names}/bwaMapping2DB//bamqc --java-mem-size=80G


 mkdir -p ${BaseLocation}/log
echo ${BaseLocation}"/"${names}"/bwaMapping2DB/bamqc" >> ${BaseLocation}/log/multiqc_logfile_finalGenome.txt
   done 
   
   
#    rm -r ${BaseLocation}/MultiQC
 
#  mkdir -p ${BaseLocation}/MultiQC
  
#  multiqc --config /archiv/logs/multiQC_config.txt --file-list  ${BaseLocation}/log/multiqc_logfile_finalGenome.txt \
#    -o ${BaseLocation}/MultiQC


##-------------------------------------------------  
##-----check read coverage
##-------------------------------------------------  
name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final 
threads=37
names=Konserve_202


num=1
rm -r ${BaseLocation}/log/mappings.txt

#for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt)
for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt)
  do
mapped=$(grep "     number of mapped reads = " ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt  |cut -d '(' -f 2 | sed 's/%)//g')

mkdir -p ${BaseLocation}/log
echo -e ${names}"\tonlyMeta\t"${mapped} >> ${BaseLocation}/log/mappings.txt

done


##bring together 

cat /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/log/mappings.txt \
/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_allRMKs202/log/mappings.txt > \
/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/log/mappings_final.txt
```

```{bash}


rm ${BaseLocation}/LdelPlasmid_coverage.txt
for names in $(cat ${samplesss}|grep "^L")
  do
  echo "==================="
  echo ${names}
CovGenome=$(grep ">>>>>>> Coverage per contig" -A 20 ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt |grep "CP046131"  |cut -f 5)
Covplasmid=$(grep ">>>>>>> Coverage per contig" -A 20 ${BaseLocation}/${names}/bwaMapping2DB//bamqc/genome_results.txt |grep "CP046132" |cut -f 5)
 
echo -e ${names}"\t"${CovGenome}"\t"${Covplasmid} >> ${BaseLocation}/LdelPlasmid_coverage.txt
 done
```

```{bash}
/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/LdelPlasmid_coverage.txt


scp vincent@130.223.49.145:/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/LdelPlasmid_coverage.txt /home/vincent/Desktop/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final

```

```{r}
library(readr)
library(ggplot2)
LdelPlasmid_coverage <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP_Final/LdelPlasmid_coverage.txt",  "\t", escape_double = FALSE, col_names = c("sample","genomeCoverage","plasmidCoverage"),  trim_ws = TRUE) 

LdelPlasmid_coverage <- LdelPlasmid_coverage[1:9,]
LdelPlasmid_coverage$copyNumber <- LdelPlasmid_coverage$plasmidCoverage/LdelPlasmid_coverage$genomeCoverage

range(LdelPlasmid_coverage$copyNumber)
ggplot(LdelPlasmid_coverage,aes(x=sample,y=copyNumber))+geom_bar( stat="identity")+theme_classic()
```

####Decipher
added 28.8.2020
Trying out [Decipher](http://www2.decipher.codes/) to find ampliconable regions.



```{bash}
if (!requireNamespace("BiocManager", quietly=TRUE))
    install.packages("BiocManager")
BiocManager::install("DECIPHER")
```



###CRISPR location Circos
added 28.8.2020

```{bash}

##------------------------
###bwa mem
##------------------------
threads=37
for species in $(echo "Ldel Sterm")
do
rm -r  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/forCRISPRs/${species}/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/forCRISPRs/${species}/
rm -r  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/forCircos_contigs/${species}/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/forCircos_contigs/${species}/
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/startaligned/Aligned/${species}/

REF=$(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/ |grep "MAG" |grep ".fna$")
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/${REF}.*
bwa index /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/${REF}
#for genomess in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/ |grep ".fna$"|cut -d '_' -f 1|grep "MAG" -v |sed 's/.fna//g')
for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/ |grep "_202_"|grep ".fasta$" |sed 's/.fasta//g'|grep "MAG" -v)

do

#/archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs//202-SMAG.fna

mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/startaligned/Aligned/${species}/${genomess}

bwa mem -x intractg /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned//StartAligned/${REF} /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned//${genomess}.fna | samtools sort -@${threads} -O BAM -o /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/startaligned/Aligned/${species}/${genomess}/allcontigsMapped2CRISPRs.bam - 

samtools view -F 4 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/startaligned/Aligned/${species}/${genomess}/allcontigsMapped2CRISPRs.bam |awk -F "\t" '{OFS="\t"}{print $3,$4,$4+length($10)}' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/forCircos_contigs/${species}/${genomess}_contigs.txt
samtools view -F 4 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/startaligned/Aligned/${species}/${genomess}/allcontigsMapped2CRISPRs.bam |awk -F "\t" '{OFS="\t"}{print $3,$4,$4+length($10),$1}' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/forCRISPRs/${species}/${genomess}_contigs.txt
done #genomes
done # species

###why they still map reverse complement
#samtools view /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/startaligned/Aligned/${species}/202-13495/allcontigsMapped2CRISPRs.bam | awk -F "\t"  '{OFS="\t"}{if($2=="16")print $1,$2,$6}'

#samtools view /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/startaligned/Aligned/${species}/202-13495/allcontigsMapped2CRISPRs.bam | grep "534M" |cut -f 1,2,6
#samtools view /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/startaligned/Aligned/${species}/202-13495/allcontigsMapped2CRISPRs.bam | grep "667" |cut -f 1,2,6



#grep "534M"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/tmp/202-13495_allcontigsMapped2CRISPRs.cleaned |cut -f 1,2,6
##--------------------
##check alignment 
##--------------------
for species in $(echo "Ldel Sterm")
do
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/startaligned/Aligned/${species}/tmp.alignment
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/startaligned/Aligned/${species}/tmp.alignment_02
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/startaligned/Aligned/${species}/
for genomess in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/ |grep ".fna$"|cut -d '_' -f 1|grep "MAG" -v |sed 's/.fna//g')
do
samtools view /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/startaligned/Aligned/${species}/${genomess}/allcontigsMapped2CRISPRs.bam |cut -f 1,2 >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/startaligned/Aligned/${species}/tmp.alignment

samtools view /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/startaligned/Aligned/${species}/${genomess}/allcontigsMapped2CRISPRs.bam | awk -F "\t"  '{OFS="\t"}{if($2=="16")print $1,$2,$6}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/startaligned/Aligned/${species}/tmp.alignment_02

done #genomes
cut -f 2 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/startaligned/Aligned/${species}/tmp.alignment |sort|uniq -c
cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/startaligned/Aligned/${species}/tmp.alignment_02
echo "==================="
done # species

wc -l /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/startaligned/Aligned/${species}/tmp.alignment
cut -f 1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/startaligned/Aligned/${species}/tmp.alignment |sort|uniq |wc -l

##------------------------
###CRISPRs
##------------------------
for species in $(echo "Ldel Sterm")
do
echo "##=========================="
echo $species
echo "##=========================="
rm -r  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/CRISPRlocation/${species}
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/forCRISPRs/${species}_allCRISPR.txt
for genomess in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned//StartAligned/ |grep ".fna$" |grep "MAG" -v|sed 's/.fna//g')
do
echo "#----------------------"
$genomess
echo "#----------------------"

mkdir -p  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/CRISPRlocation/${species}/${genomess}

  sed -n "/SUMMARY BY SIMILARITY$/,/SUMMARY BY POSITION$/p" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${genomess}/${genomess}.pilarCR_out | grep "Array" -v |grep "^====" -v |grep "SUMM" -v|sed '/^$/d' | sed 's/^ *//g'|grep "^\*" -v |sed -E 's/\s+/\t/g' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/CRISPRlocation/${species}/${genomess}/tmp.txt
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/forCircos_contigs/${species}/${genomess}_CRISPR.txt
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/forCRISPRs/${species}/${genomess}_CRISPR.txt

for CRISPRsss in $(cut -f 1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/CRISPRlocation/${species}/${genomess}/tmp.txt)
do
echo $CRISPRsss

contigName=$(grep "^${CRISPRsss}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/CRISPRlocation/${species}/${genomess}/tmp.txt| cut -f 2)
contigpsoitionOrig=$(grep "^${CRISPRsss}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/CRISPRlocation/${species}/${genomess}/tmp.txt| cut -f 3)
contigLengthOrig=$(grep "^${CRISPRsss}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/CRISPRlocation/${species}/${genomess}/tmp.txt| cut -f 4)

RepeatSeq=$(grep "^${CRISPRsss}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/CRISPRlocation/${species}/${genomess}/tmp.txt| cut -f 9|head -c 20|tail -c 10)
RepeatArray=$(grep "${RepeatSeq}" -B 1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/${species}_CRISPR_repeat.txt |grep "^>" |head -1|sed 's/>//g')

REFname=$(grep "${contigName}$" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/forCRISPRs/${species}/${genomess}_contigs.txt |head -1|cut -f 1)
REFStart=$(grep "${contigName}$" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/forCRISPRs/${species}/${genomess}_contigs.txt |head -1|cut -f 2)
#echo -e ${REFname}"\t"${REFStart}

CRISPRstart=$((REFStart+contigpsoitionOrig))
CRISPREND=$((REFStart+contigpsoitionOrig+contigLengthOrig))
echo -e ${REFname}"\t"${CRISPRstart}"\t"${CRISPREND}"\t"${RepeatArray} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/forCircos_contigs/${species}/${genomess}_CRISPR.txt
echo -e ${REFname}"\t"${CRISPRstart}"\t"${CRISPREND}"\t"${contigName} >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/forCRISPRs/${species}_allCRISPR.txt

#echo -e ${contigName}"\t"${contigpsoitionOrig}"\t"${contigLengthOrig}
done #CRISPRsss

##change name to color
sed -i 's/a1/stroke_color=orange/g' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/forCircos_contigs/${species}/${genomess}_CRISPR.txt
sed -i 's/a2/stroke_color=dpurple/g' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/forCircos_contigs/${species}/${genomess}_CRISPR.txt
sed -i 's/a3/stroke_color=dred/g' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/forCircos_contigs/${species}/${genomess}_CRISPR.txt

done #genomes

done # species

cut -f 2,4 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/forCircos_contigs/${species}_allCRISPR.txt |sort -n
##--location
for species in $(echo "Ldel Sterm")
do
echo "##=========================="
echo $species
echo "##=========================="
cut -f 2,4 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/forCircos_contigs/${species}/*_CRISPR.txt |sort -n
done # species
```

make circos with the following data that I created in the last two chunks
1. Start alignment
2. CRISPR locations

make the plot according to the pyhlogeny

move all necessary files local
```{bash}

mkdir -p /home/vincent/bin/apps/circos-0.69-6/data/CRISPR
scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Location_Circos/forCircos_contigs/ /home/vincent/bin/apps/circos-0.69-6/data/CRISPR
```

some manual work still needs to be done:
1. larger CRISPR array bands (data/CRISPR/forCircos_contigs/Sterm/arrays.txt)
2. links file for jumping arrays (data/CRISPR/forCircos_contigs/Sterm/links.txt)

```{bash}

##==================
##Sterm
##==================
home=/home/vincent/bin/apps/circos-0.69-6/
cd $home

##organisational 
#cp /home/vincent/bin/apps/circos-0.69-6/etc/ticks_mag_both.conf /home/vincent/bin/apps/circos-0.69-6/etc/ticks_mag_CRISPR.conf

#cp /home/vincent/bin/apps/circos-0.69-6/etc/ideogram_mag.conf /home/vincent/bin/apps/circos-0.69-6/etc/ideogram_mag_CRISPR.conf
<!-- mkdir -p $home/data/rmk202/MAG_rmk202_sterm_CRISPR/ -->
#cp /home/vincent/bin/apps/circos-0.69-6/etc/sterm_rmk202_mag_circos.conf /home/vincent/bin/apps/circos-0.69-6/etc/sterm_rmk202_mag_CRISPR.conf
##---------------------karyotyp
#seqlength.py /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/Sterm/renamedContigs/StartAligned//StartAligned/ |head -1 > ${home}/data/karyotype/sterm_magg_rmk202.txt

sed 's/S_thermophilus_mag_rmk202/202-SMAG-1/g' ${home}/data/karyotype/sterm_magg_rmk202.txt > ${home}/data/karyotype/sterm_magg_rmk202_CRISPR.txt
##have to change the format a bit

##---------------------number genomes

ls /home/vincent/bin/apps/circos-0.69-6/data/CRISPR/forCircos_contigs/Sterm/ |cut -d '_' -f 1|sort|uniq
202-13494
202-S72
202-13498
202-13493
202-13500
202-13491
202-13492

202-13495
202-13496
202-13497

202-13499c1
202-13499c2
202-S50


#202-13494


##example
#cp etc/repeat_nwc1_sterm_01.conf etc/sterm_rmk202_mag_circos.conf

#example run
bin/circos -conf etc/sterm_rmk202_mag_CRISPR.conf -outputfile ./Sterm_mag_rmk202_CRISPR.svg; firefox ./Sterm_mag_rmk202_CRISPR.svg &


##==================
##Ldel
##==================
home=/home/vincent/bin/apps/circos-0.69-6/
cd $home

##organisational 
#cp /home/vincent/bin/apps/circos-0.69-6/etc/ticks_nwc_1_ldel_both.conf /home/vincent/bin/apps/circos-0.69-6/etc/ticks_mag_both.conf ##already done
#cp /home/vincent/bin/apps/circos-0.69-6/etc/ideogram.conf /home/vincent/bin/apps/circos-0.69-6/etc/ideogram_mag.conf ##already done
#cp /home/vincent/bin/apps/circos-0.69-6/etc/sterm_rmk202_mag_circos.conf /home/vincent/bin/apps/circos-0.69-6/etc/ldel_rmk202_mag_circos.conf
mkdir -p $home/data/rmk202/MAG_rmk202_ldel/
##---------------------karyotyp

#seqlength.py /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/final/ldel/L_delbrueckii_RMK202.fasta |head -1 > ${home}/data/karyotype/ldel_magg_rmk202.txt


##have to change the format a bit

```

#### mismatch repair genes
added 28.8.2020
```{bash}


cd /archiv/Projects/2019_RMK202_analysis/00_FINAL/06_Orthofinder/eggnog_of_orthofinder/
species=Sterm
for species in $(echo "Ldel Sterm")
do
####================
##OGs
####================
echo "==================================================="
echo ${species}
echo "==================================================="

 awk -F "\t" '{OFS="\t"}{if($6~"mut")print $0}' eggnog_orthifinder_${species}_annotations.tsv |cut -f 6

done
```


###CRISPR region tree
added 28.9.2020
Here, I want to make a CRISPR tree array (with Cas genes)
Is not used for the paper. 
```{bash}

###--------------------------------
##CRISPR repeat location
###--------------------------------
species=Sterm
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISP§R_region_tree/${species}/array/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/Tree_location/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/Tree_location/
for genomess in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/ |grep ".fna$"|cut -d '_' -f 1 |sed 's/.fna//g') #|grep "202-13497"
do
echo "===================================="
echo ${genomess}
grep "repeat" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.gff > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${genomess}_repeats.txt
cut -f 1,4,5 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${genomess}_repeats.txt > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${genomess}_repeats.bed
for arraysss in $(grep ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Sterm_CRISPR_repeat_PROKKA.txt|sed 's/>//g')
do
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${arraysss}/
arraySeq=$(grep -A 1 "${arraysss}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Sterm_CRISPR_repeat_PROKKA.txt |tail -1|head -c 25|tail -c 20)

grep "${arraySeq}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${genomess}_repeats.txt > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${arraysss}/${genomess}_${arraysss}_repeats.txt
cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${arraysss}/${genomess}_${arraysss}_repeats.txt >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${arraysss}_repeats.txt

tmp=$(cut -f 9 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${arraysss}/${genomess}_${arraysss}_repeats.txt)
cut -f 1,4,5 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${arraysss}/${genomess}_${arraysss}_repeats.txt
#grep -A 15 -B 15 "${tmp}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.gff |grep "cas9" -i


if [[ ${arraysss} == "a1"  && [$(echo ${tmp})>0 ]]; then
            echo -e "ary1"
            grep -B 15 "${tmp}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.gff |grep "cas9" -i |head -1 >            /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${arraysss}/${genomess}_${arraysss}_cas9.txt
            cut -f 1,4,5 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${arraysss}/${genomess}_${arraysss}_cas9.txt
            
            contig=$(cut -f 1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${arraysss}/${genomess}_${arraysss}_cas9.txt)
            start=$(cut -f 4 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${arraysss}/${genomess}_${arraysss}_cas9.txt|awk '{print $1}')
            end=$(cut -f 5 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${arraysss}/${genomess}_${arraysss}_repeats.txt|awk '{print $1}')
            echo -e ${contig}"\t"${start}"\t"${end}"\t"${genomess}";"${arraysss} > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/Tree_location/${genomess}_${arraysss}_CRISPR_location.bed
            echo -e ${contig}"\tVINCENT\tCRISPRLOCATION\t"${start}"\t"${end}"\t.\t+\t0\t"${genomess}";"${arraysss} > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/Tree_location/${genomess}_${arraysss}_CRISPR_location.gff
            
            elif [[ ${arraysss} == "a2"  && [$(echo ${tmp})>0 ]] 
            then
            echo -e "ary2"
            grep -B 20 "${tmp}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.gff |grep "csm6" -i |head -1 >            /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${arraysss}/${genomess}_${arraysss}_csm6.txt
            cut -f 1,4,5 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${arraysss}/${genomess}_${arraysss}_csm6.txt
            ##---
            grep -A 20 "${tmp}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.gff |grep "cas1" -i |head -1 >            /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${arraysss}/${genomess}_${arraysss}_cas1.txt
            cut -f 1,4,5 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${arraysss}/${genomess}_${arraysss}_cas1.txt
            
            contig=$(cut -f 1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${arraysss}/${genomess}_${arraysss}_cas1.txt)
            start=$(cut -f 4 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${arraysss}/${genomess}_${arraysss}_csm6.txt)
            end=$(cut -f 5 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${arraysss}/${genomess}_${arraysss}_cas1.txt)
             echo -e ${contig}"\t"${start}"\t"${end}"\t"${genomess}";"${arraysss} > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/Tree_location/${genomess}_${arraysss}_CRISPR_location.bed
            echo -e ${contig}"\tVINCENT\tCRISPRLOCATION\t"${start}"\t"${end}"\t.\t-\t0\t"${genomess}";"${arraysss} > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/Tree_location/${genomess}_${arraysss}_CRISPR_location.gff
            
            elif [[ ${arraysss} == "a3"  && [$(echo ${tmp})>0 ]]
            then
            echo -e "ary3"
            grep -A 15 "${tmp}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.gff |grep "cas9" -i |head -1 >             /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${arraysss}/${genomess}_${arraysss}_cas9.txt 
            cut -f 1,4,5 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${arraysss}/${genomess}_${arraysss}_cas9.txt
            
            contig=$(cut -f 1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${arraysss}/${genomess}_${arraysss}_cas9.txt)
            start=$(cut -f 4 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${arraysss}/${genomess}_${arraysss}_repeats.txt)
            end=$(cut -f 5 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${arraysss}/${genomess}_${arraysss}_cas9.txt)
             echo -e ${contig}"\t"${start}"\t"${end}"\t"${genomess}";"${arraysss} > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/Tree_location/${genomess}_${arraysss}_CRISPR_location.bed
             echo -e ${contig}"\tVINCENT\tCRISPRLOCATION\t"${start}"\t"${end}"\t.\t-\t0\t"${genomess}";"${arraysss} > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/Tree_location/${genomess}_${arraysss}_CRISPR_location.gff
            
            
               
            fi
echo "--------------"
done #arraysss
done #genomes


cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/Tree_location/*.gff |awk -F "\t" '{OFS="\t"}{print $0,$5-$4}'




###--------------------------------
##mask CRISPR repeat region
###--------------------------------

rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/maskedGenomes/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/maskedGenomes/
for genomess in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/ |grep ".fna$"|cut -d '_' -f 1 |sed 's/.fna//g') #|grep "202-13497"
do
echo "===================================="
echo ${genomess}

awk -F "\t" '{OFS="\t"}{print $1,$2-500,$3+500}' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${genomess}_repeats.bed > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${genomess}_repeats_extended.bed

for arraysss in $(grep ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Sterm_CRISPR_repeat_PROKKA.txt|sed 's/>//g')
do

bedtools maskfasta -fi /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/Sterm/renamedContigs/StartAligned/StartAligned/${genomess}.fna -bed /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${genomess}_repeats.bed -fo /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/maskedGenomes/${genomess}_CRISPRmasked.fna


bedtools maskfasta -fi /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/Sterm/renamedContigs/StartAligned/StartAligned/${genomess}.fna -bed /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/array/${genomess}_repeats_extended.bed -fo /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/maskedGenomes/${genomess}_CRISPRmasked_extended.fna

done #arraysss
done #genomes




###--------------------------------
##extract CRISPR region out of genome
###--------------------------------


rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/extractedCRISPR/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/extractedCRISPR/
for genomess in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/ |grep ".fna$"|cut -d '_' -f 1 |sed 's/.fna//g') #|grep "202-13497"
do
echo "===================================="
echo ${genomess}

for arraysss in $(grep ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Sterm_CRISPR_repeat_PROKKA.txt|sed 's/>//g')
do

bedtools getfasta -fi /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/maskedGenomes/${genomess}_CRISPRmasked.fna \
  -bed /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/Tree_location/${genomess}_${arraysss}_CRISPR_location.gff \
  -fo /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/extractedCRISPR/${genomess}_${arraysss}_CRISPR_location.fasta



sed "s/>.*$/>${genomess}-${arraysss}/g" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/extractedCRISPR/${genomess}_${arraysss}_CRISPR_location.fasta >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/extractedCRISPR/${arraysss}_CRISPR_locations.fasta
done #arraysss
done #genomes



```
now make a multiseq alignment file

I go information for the [alignment](https://www.protocols.io/view/week-5-aligning-with-muscle-and-making-trees-with-g2sbyee?step=4) and for [RaxML](http://evomics.org/learning/phylogenetics/raxml/)

```{bash}
###--------------------------------
##Multiseq alignment
###--------------------------------

rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/extractedCRISPR/all_CRISPR_locations.fasta
species=Sterm
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/MultiseqAlignment/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/MultiseqAlignment/
for arraysss in $(grep ">" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/Sterm_CRISPR_repeat_PROKKA.txt|sed 's/>//g')
do
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/MultiseqAlignment/${arraysss}/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/MultiseqAlignment/${arraysss}/

muscle -in /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/extractedCRISPR/${arraysss}_CRISPR_locations.fasta -out /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/MultiseqAlignment/${arraysss}_MultiseqAlignment.fasta 

/usr/bin/perl /home/vincent/apps/Fasta2Phylip.pl /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/MultiseqAlignment/${arraysss}_MultiseqAlignment.fasta /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/MultiseqAlignment/${arraysss}/${arraysss}_MultiseqAlignment.phy


cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/extractedCRISPR/${arraysss}_CRISPR_locations.fasta >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/extractedCRISPR/all_CRISPR_locations.fasta
done #arraysss

###---------------all
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/MultiseqAlignment/all_together/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/MultiseqAlignment/all_together/

muscle  -in /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/extractedCRISPR/all_CRISPR_locations.fasta -out /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/MultiseqAlignment/all_MultiseqAlignment.fasta

/usr/bin/perl /home/vincent/apps/Fasta2Phylip.pl /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/MultiseqAlignment/all_MultiseqAlignment.fasta /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/MultiseqAlignment/all_together/all_MultiseqAlignment.phy


###---------------only array1 and array3

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/extractedCRISPR/a1_CRISPR_locations.fasta /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/extractedCRISPR/a3_CRISPR_locations.fasta > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/extractedCRISPR/a1_a3_CRISPR_locations.fasta

rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/MultiseqAlignment/a1_a3_together/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/MultiseqAlignment/a1_a3_together/

muscle  -in /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/extractedCRISPR/a1_a3_CRISPR_locations.fasta -out /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/MultiseqAlignment/a1_a3_MultiseqAlignment.fasta

/usr/bin/perl /home/vincent/apps/Fasta2Phylip.pl /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/MultiseqAlignment/a1_a3_MultiseqAlignment.fasta /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/MultiseqAlignment/a1_a3_together/a1_a3_MultiseqAlignment.phy


###--------------------------------
##make tree
###--------------------------------
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/RaxML_tree/
#mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/RaxML_tree/
for arraysss in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/MultiseqAlignment/ |grep ".fasta" -v)
do
cd /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/MultiseqAlignment/${arraysss}
raxmlHPC -s *_MultiseqAlignment.phy -n ${arraysss}_RaxML_tree_out -m GTRCAT -f a -x 123 -N autoMRE -p 456

done #arraysss


```

move local 

```{bash}
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/Sterm/MultiseqAlignment
scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/Sterm/MultiseqAlignment /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/Sterm/MultiseqAlignment

```

#####special:DGCC 7710
added 28.9.2020
Here, I want to look into the S. thermophilus strain DGCC 7710 because this streptococcus has 4 CRISPR arrays and could be used for the referencing of my 3 S. thermophilus CRISPR arrays. 

```{bash}
##=================================
##download genome
##=================================
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/genome/

curl 'ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt' |grep "DGCC 7710" |tail -1 |awk 'BEGIN{FS="\t";OFS="\t"}  {print $20}'| sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/genome/download.file



cd /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/genome/
wget -i download.file
gunzip *.fna.gz

##=================================
##PilarCR
##=================================


mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/pilerCR/

  ~/apps/PilerCR/pilercr1.06/pilercr -in /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/genome/GCF_002843115.1_ASM284311v1_genomic.fna \
    -out /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/pilerCR/strain_DGCC_7710.pilarCR_out -noinfo\
    -seq /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/pilerCR/strain_DGCC_7710.pilarCR_out.fasta

  
  perl ~/apps/PilerCR/PilerCR_parser_new_vincent /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/pilerCR/strain_DGCC_7710.pilarCR_out > \
    /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/pilerCR/strain_DGCC_7710.pilarCR_out_final


##=================================
##Prokka
##=================================

mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/PROKKA
/usr/bin/perl /usr/bin/prokka --outdir /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/PROKKA --force --usegenus --genus Streptococcus --locustag DGCC_7710 --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/genome/GCF_002843115.1_ASM284311v1_genomic.fna 

##=================================
##Cas genes
##=================================
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/PROKKA/CRISPRarray.txt
#grep -i "isfinder"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/PROKKA/*.gff
grep -i "=cas" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/PROKKA/*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $1,$4,$5,$7";"$9}'   >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/PROKKA/CRISPRarray.txt
grep -i "=csn"  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/PROKKA/*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $1,$4,$5,$7";"$9}'   >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/PROKKA/CRISPRarray.txt
grep -i "=Csy" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/PROKKA/*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $1,$4,$5,$7";"$9}'   >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/PROKKA/CRISPRarray.txt
grep -i "=Csm" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/PROKKA/*.gff |awk -F "\t" '{OFS="\t"}{if($3=="gene") print $1,$4,$5,$7";"$9}'   >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/PROKKA/CRISPRarray.txt

grep "SUMMARY BY POSITION" -A 15 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/pilerCR/strain_DGCC_7710.pilarCR_out |grep "===========" -A 6 |sed '1d'  |sed 's/^ *//' |sed 's/  */\t/g' |awk -F "\t" '{OFS="\t"}{print $2,$4,$4+$5,"CRISPRspacers"}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/PROKKA/CRISPRarray.txt

sort -k2 -n /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/PROKKA/CRISPRarray.txt

mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation

sort -k2 -n /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/PROKKA/CRISPRarray.txt | awk -F "\t" '{if($2~"^6")print $0}' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/CRISPRarray_a1.txt
sort -k2 -n /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/PROKKA/CRISPRarray.txt | awk -F "\t" '{if($2~"^8")print $0}' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/CRISPRarray_a2.txt
sort -k2 -n /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/PROKKA/CRISPRarray.txt | awk -F "\t" '{if($2~"^9")print $0}' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/CRISPRarray_a3.txt
sort -k2 -n /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/PROKKA/CRISPRarray.txt | awk -F "\t" '{if($2~"^1")print $0}' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/CRISPRarray_a4.txt

##=================================
##mask CRISPR spacers
##=================================
grep "repeat" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/PROKKA/*.gff > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/PROKKA/strain_DGCC_7710_repeats.txt

cut -f 1,4,5 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/PROKKA/strain_DGCC_7710_repeats.txt > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/PROKKA/strain_DGCC_7710_repeats.bed

bedtools maskfasta -fi /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/genome/GCF_002843115.1_ASM284311v1_genomic.fna  -bed /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/PROKKA/strain_DGCC_7710_repeats.bed -fo /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/genome/GCF_002843115.1_ASM284311v1_genomic_CRISPRmasked.fna 


##=================================
##CRISPR extract
##=================================
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/CRISPRarray_all_locations.fasta
for arrayzzz in $(echo "a1 a2 a3 a4")
do
echo $arrayzzz
contig=$(cut -f 1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/CRISPRarray_${arrayzzz}.txt|head -1)
start=$(cut -f 2 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/CRISPRarray_${arrayzzz}.txt|head -1)
end=$(cut -f 3 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/CRISPRarray_${arrayzzz}.txt|tail -1)

echo -e ${contig}"\t"${start}"\t"${end}"\tstrain_DGCC_7710;"${arrayzzz} > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/CRISPRarray_${arrayzzz}.bed


bedtools getfasta -fi /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/genome/GCF_002843115.1_ASM284311v1_genomic_CRISPRmasked.fna  \
  -bed /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/CRISPRarray_${arrayzzz}.bed \
  -fo /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/CRISPRarray_${arrayzzz}.fasta


sed "s/>.*$/>strain_DGCC_7710-${arrayzzz}/g" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/CRISPRarray_${arrayzzz}.fasta > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/CRISPRarray_${arrayzzz}_locations.fasta

cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/CRISPRarray_${arrayzzz}_locations.fasta >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/CRISPRarray_all_locations.fasta
done #arrayzzz

##=================================
##merge and MSA with other Arrays
##=================================
cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/Sterm/extractedCRISPR/all_CRISPR_locations.fasta >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/CRISPRarray_all_locations.fasta

rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/MultiseqAlignment/all_together/
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/MultiseqAlignment/all_together/

muscle  -in /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/CRISPRarray_all_locations.fasta -out /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/MultiseqAlignment/all_together/all_MultiseqAlignment.fasta -maxiters 2

/usr/bin/perl /home/vincent/apps/Fasta2Phylip.pl /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/MultiseqAlignment/all_together/all_MultiseqAlignment.fasta /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/MultiseqAlignment/all_together/all_MultiseqAlignment.phy

###for alignment score
cd /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/MultiseqAlignment/all_together
/home/vincent/apps/MstatX/mstatx -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/MultiseqAlignment/all_together/all_MultiseqAlignment.fasta -s trident

##=================================
##make tree
##=================================
#rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation//RaxML_tree/
#mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPR_region_tree/${species}/RaxML_tree/

cd /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/MultiseqAlignment/all_together/
raxmlHPC -s all_MultiseqAlignment.phy -n withDGCC_RaxML_tree_out -m GTRCAT -f a -x 123 -N autoMRE -p 456



```

```{bash}
/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/MultiseqAlignment/all_together
```

move local 

```{bash}
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/MultiseqAlignment/all_together
scp -r vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/MultiseqAlignment/all_together /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/strain_DGCC_7710/ARRAYannotation/MultiseqAlignment/all_together

```



##fastani for identification
added 29.8.2020
```{bash}
!!!!!!!!!!!!THIS
###-----------------------------------------
##get references
###-----------------------------------------
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log
curl 'ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt' > /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200529_NCBI_log.txt

##ldel
species=$(echo "Lactobacillus delbrueckii")
species_short=Ldel
typeStrain=$(echo "equicursoris")

#rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}


cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200529_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' >    \
    /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download.txt


cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200529_NCBI_log.txt |grep -e "${typeStrain}" -i |grep "type strain"|head -1 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' >>    \
    /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download.txt
    
    rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/FNA
mkdir -p //home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/FNA
    cd //home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/FNA
    wget -i /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download.txt
    gunzip *.fna.gz

#-------------------------------Sterm

species=$(echo "Streptococcus thermophilus")
species_short=Sterm
typeStrain=$(echo "S735")
typeStrain=$(echo "vestibularis")

#rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}

cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200529_NCBI_log.txt |grep -e "${species}"|cut -f 12|sort|uniq -c
#cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200529_NCBI_log.txt |grep -e "${species}"|awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'

cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200529_NCBI_log.txt |grep -e "${species}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' >    \
    /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download.txt
    

cat /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200529_NCBI_log.txt |grep -e "${typeStrain}" |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome" ||$12=="Chromosome" ) print $0}'|head -1|cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' >>    \
    /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download.txt
    
rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/FNA
mkdir -p //home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/FNA
    cd //archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FNA_all
    wget -i /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species_short}/download.txt
gunzip *.fna.gz



###-----------------------------------------
##move all fna
###-----------------------------------------

for species in $(echo "Ldel Sterm")
do

rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FNA_all
mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FNA_all

for genomess in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned/ |grep ".fna$"|cut -d '_' -f 1|sed 's/.fna//g')
do
cut -d ' ' -f 1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/${genomess}/PROKKA*.fna > /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FNA_all/${genomess}.fna

done #genomes
rm -r /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/FNA_all/
mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/FNA_all/
cp /archiv/Projects/2019_RMK202_analysis/00_FINAL/03_annotation/PROKKA/${species}/FNA_all/*.fna /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/FNA_all/
cp /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/FNA/*.fna /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/FNA_all/

done # species


###-----------------------------------------
##make list
###-----------------------------------------

for species in $(echo "Ldel Sterm")
do
rm /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/fastani.txt

for genomssss in $(ls /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/FNA_all/ |grep ".fna$")
do
echo "/home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/"${species}"/FNA_all/"${genomssss} >> /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/fastani.txt


done #genomes
done # species


###-----------------------------------------
##fastani
###-----------------------------------------



for species in $(echo "Ldel Sterm")
do



fastANI --fragLen 1000  -t 37 --ql /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/fastani.txt \
  --rl /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/fastani.txt \
  -o /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/fastaANI_comparision.txt

done # species


###-----------------------------------------
##closest genome
###-----------------------------------------
for species in $(echo "Ldel Sterm")
do


simss=$(grep "MAG" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/fastaANI_comparision.txt|grep "GCF"|sort -k3  |tail -1 |cut -f 3)
covesss=$(grep "MAG" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/fastaANI_comparision.txt |grep "GCF"|sort -k3  |tail -1 |awk -F "\t" '{OFS="\t"}{print ($4/$5)*100}')
genomesss=$(grep "MAG" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/fastaANI_comparision.txt |grep "GCF"|sort -k3  |tail -1|cut -f 1|sed 's/\/home\/vincent\/Desktop\/Projects\/2019_RMK202_analysis\/00_FINAL\/09_referenceTree\/Sterm\/FNA_all\///g'|sed 's/_genomic.fna//g')


speciessss=$(grep "${genomesss}" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200529_NCBI_log.txt |cut -f 8)
strain=$(grep "${genomesss}" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200529_NCBI_log.txt |cut -f 9|sed 's/strain=//g')

echo -e "closest strain for the Metagenome of "${species} " is : "${speciessss} ${strain} " with a similarity " ${simss} "% and coverage "${covesss}"%"

done # species


species=Ldel
simss=$(grep "MAG" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/fastaANI_comparision.txt|grep "GCF"|sort -k3  |tail -1 |cut -f 3)
covesss=$(grep "MAG" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/fastaANI_comparision.txt |grep "GCF"|sort -k3  |tail -1 |awk -F "\t" '{OFS="\t"}{print ($4/$5)*100}')
genomesss=$(grep "MAG" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/${species}/fastaANI_comparision.txt |grep "GCF"|sort -k3  |tail -1|cut -f 1|sed 's/\/home\/vincent\/Desktop\/Projects\/2019_RMK202_analysis\/00_FINAL\/09_referenceTree\/Ldel\/FNA_all\///g'|sed 's/_genomic.fna//g')


speciessss=$(grep "${genomesss}" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200529_NCBI_log.txt |cut -f 8)
strain=$(grep "${genomesss}" /home/vincent/Desktop/Projects/2019_RMK202_analysis/00_FINAL/09_referenceTree/log/20200529_NCBI_log.txt |cut -f 9|sed 's/strain=//g')

echo -e "closest strain for the Metagenome of "${species} " is : "${speciessss} ${strain} " with a similarity " ${simss} "% and coverage "${covesss}"%"


```
###find leader sequence
here I try to identify the leader sequence of the CRISPR arrays
Finally, I did it online with the [CRISPRdetect](http://crispr.otago.ac.nz/CRISPRDetect/predict_crispr_array.html)

### CRISPR-repeats
The CRISPR repeats are conserved. Therefore in the assemblies we should have the same repeats occuring again and again. 
working list:
-add sample name
-add REFs
-cd-hit

```{bash}
###================
##description file
###================
threads=37
#samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt  ##the file with all sample names
#logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/04_withOlDsamples/repeats
#Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/FASTA/S_thermophilus_alltogether.CRISPRlocation.fasta



#===============================
##merge the repeats all together
#===============================

for species in $(echo "Ldel Sterm")
do
echo ==========================
     echo "working on: " ${species}
     echo ==========================
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/allRepeatsAssembled.fasta
#mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/
#for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/strains/${species}/renamedContigs/StartAligned/StartAligned| grep ".fna$" |sed 's/.fna//g')
for genomes in $(ls /archiv/Projects/2019_RMK202_analysis/00_FINAL/01_genomes/Final/${species}/ |grep "_202_"|grep ".fasta$" |sed 's/.fasta//g')
do
      echo ==========================
     echo -e "extracting repeats of strain: " ${genomes}
     echo ==========================

genomes_short=$(echo ${genomes}|sed 's/S_O_202_//g'|sed 's/S_I_202_//g'|sed 's/L_I_202_//g')

sed "s/.*\[Array/>_a/g" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${genomes}/${genomes}.pilarCR_out.fasta|sed 's/;.*$//g'|sed "s/>_/>${genomes_short}_/g" >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/allRepeatsAssembled.fasta



#sed 's/__.*$//g' /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${genomes}/${genomes}.pilarCR_out.fasta | sed "s/>/>${genomes_short}_/g"| sed 's/ary0/a/g' |sed 's/spc0/s/g' |sed 's/202-//g' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/all/${species}/CRISPR_spacer_merged_short.fasta

done #genomes

done #species


#sed 's/\:.*\[/:/' ${BaseLocation}/assembledRepeats/allRepeatsAssembled.fasta |sed 's/Array/a0/' |sed 's/\;.*\]//' |sed 's/_0.*\:/:/' > ${BaseLocation}/assembledRepeats/allRepeatsAssembled_short.fasta

#===============================
##cd-hit the repeats
#===============================

for species in $(echo "Ldel Sterm")
do
echo ==========================
     echo "working on: " ${species}
     echo ==========================
rm -r /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/cdHIT/
  mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/cdHIT/
/home/vincent/apps/cd-hit-v4.8.1-2019-0228/cd-hit-est -i /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/allRepeatsAssembled.fasta \
    -o  /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/cdHIT/CRISPR_Clustering.txt -c 0.80
    
    #grep ">" ${BaseLocation}/assembledSpacers/allSpacersAssembled_short.fasta|cut -d ':' -f 1|sort|uniq -c

#cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/cdHIT/CRISPR_Clustering.txt

###------------
##change array names
###every strain has a different array labelling according to the CRISPR locations 
##here I want to make them all the same (because the CRISPRs are within one of three arrays)
###------------
##-------------------check!!
##this only runs if all strains have zero, one or two annotated CRISPR arrays for all of the three arrays
##all other scenario it cannot cope with
done #species

##!!!!!!!!!!!!!!!!!!!!!this step has to be done manually
##by looking in which Cluster the Reference (circular assembled) arrays lie
species=Sterm
genomes=202-SMAG
cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/cdHIT/CRISPR_Clustering.txt.clstr

mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/logs/
echo -e ">Cluster_0\ta3\n>Cluster_1\ta1\n>Cluster_2\ta2" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/logs/Arrays.log

species=Ldel
genomes=202-LMAG
cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/cdHIT/CRISPR_Clustering.txt.clstr

mkdir -p /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/logs/
echo -e ">Cluster_0\ta1\n>Cluster_1\ta2" > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/logs/Arrays.log

#===============================
##assign the arrays of all strain to the reference array
#===============================

for species in $(echo "Ldel Sterm")
do
echo ==========================
     echo "working on: " ${species}
     echo ==========================
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final.txt
rm /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final_WithouAccumulation.txt

for clustersss in $(cut -f 1 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/logs/Arrays.log)
do
aryName=$(grep "${clustersss}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/logs/Arrays.log |cut -f 2)
ClusterNumber=$(echo $clustersss|awk -F "_" '{print $2}')
EndCluster=$((ClusterNumber+1))

sed -n "/>Cluster $ClusterNumber$/,/>Cluster $EndCluster$/p" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}/cdHIT/CRISPR_Clustering.txt.clstr | grep -v "^>Cl" |awk -F "[>...]" '{print $2}' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt

#for samplesss in $(cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt)
num=1
for samplesss in $(cut -f 3 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt|sort|uniq|grep -v "sample")
do

echo -e ${num}
NumberCRISPRarrays=$(grep -c "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt)


if [ ${NumberCRISPRarrays} == "1" ]; then
              WhichCRISPRarrays=$(grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt|cut -d '_' -f 2|sed 's/a//g')
              grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,$5}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final.txt
              grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,$5}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final_WithouAccumulation.txt
            elif [[ ${NumberCRISPRarrays} == "0"  ]]
            then
            echo -e "No arrays in this Cluster "${aryName} "for "${samplesss} 
            elif [[ ${NumberCRISPRarrays} == "2"  ]]
            then
            echo -e "Two Arrays matcht for Cluster" ${aryName} "for "${samplesss}
            ##C----Array_ONe
            WhichCRISPRarrays=$(grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt|head -1|cut -d '_' -f 2|sed 's/a//g')
            grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,$5}' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats_Spacers.txt
            spacerCount=$(cut -f 5 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats_Spacers.txt |sort|tail -1)
            cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats_Spacers.txt >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final.txt
            cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats_Spacers.txt >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final_WithouAccumulation.txt
            ##C----Array_TWO
            WhichCRISPRarrays=$(grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt|tail -1|cut -d '_' -f 2|sed 's/a//g')
            grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Countss="${spacerCount}" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,Countss+$5}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final.txt
            grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Countss="${spacerCount}" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,$5}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final_WithouAccumulation.txt
            elif [[ ${NumberCRISPRarrays} == "3"  ]]
            then
            echo -e "Three Arrays matcht for Cluster" ${aryName} "for "${samplesss}
            ##C----Array_ONe
            WhichCRISPRarrays=$(grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt|head -1|cut -d '_' -f 2|sed 's/a//g')
            grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,$5}' > /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats_Spacers.txt
            spacerCount=$(cut -f 5 /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats_Spacers.txt |sort|tail -1)
            cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats_Spacers.txt >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final.txt
            cat /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats_Spacers.txt >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final_WithouAccumulation.txt
            ##C----Array_TWO
            WhichCRISPRarrays=$(grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt|head -2|tail -1|cut -d '_' -f 2|sed 's/a//g')
            grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Countss="${spacerCount}" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,Countss+$5}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final.txt
            grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Countss="${spacerCount}" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,$5}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final_WithouAccumulation.txt
            ##C----Array_THREE
            WhichCRISPRarrays=$(grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRrepeatAnalysis/${species}//cdHIT//tmpCRISPRS_repeats.txt|head -3|tail -1|cut -d '_' -f 2|sed 's/a//g')
            grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Countss="${spacerCount}" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,Countss+$5}' >> /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final.txt
            grep "$samplesss" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}.txt |awk -F "\t" -v Countss="${spacerCount}" -v Name="${aryName}" -v arraysss="$WhichCRISPRarrays" '{OFS="\t"}{if($4==arraysss)print $0,Name,$5}' >> /archiv/Projects/2019_RMK202_an
            else
               echo "more than three arrays-----Currently nothing is done with these"
               
               
            fi
num=$((num+1))
done #samplesss

done #clusterssss

done #species


```
Here, we can created the list of in which one of the the three Streptococci CRISPR arrays the spacer lies and at which postion.

```{bash}
##bring local
species=Sterm
mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/
scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/

scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final_WithouAccumulation.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final_WithouAccumulation.txt

##bring local
species=Ldel
mkdir -p ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/
scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final_WithouAccumulation.txt

scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final_WithouAccumulation.txt ~/Desktop/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CRISPRspacerAnalysis/${species}/spacer_Infos_${species}_final_WithouAccumulation.txt

```


#####blast Spacers
old
```{bash}
###----------------------------- Get all Spacers
mkdir -p /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/spacers/

samtools faidx /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerExtractall/CRISPR_spacers_demultiplexed_final.fasta
  
bedtools getfasta -fi /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerExtractall/CRISPR_spacers_demultiplexed_final.fasta \
  -bed /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerExtractall/CRISPR_spacers_only.bed \
  -fo /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/spacers/all_spacers4blast.fasta

grep ">" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/spacers/all_spacers4blast.fasta |sed 's/>//g' |sed 's/:/\t/g' > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/all_spacersNames.txt

grep ">" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/spacers/all_spacers4blast.fasta |sed 's/>//g'  > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/all_spacersNames_onlyNames.txt


##--------------------------blast to local phages  
  
rm -r /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast
mkdir -p /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast

  mkdir -p /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/starterPhagesPlasmid
  
  cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//splitContigs/L_del_plasmid_01.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//splitContigs/S_thermophilus_repA.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter//splitContigs/L_del_phage_01.fasta \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/splitContigs/*.fasta \
  /archiv/Pro
  jects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/spades_nonMapping_assembly/splitContigs/L_del_plasmid_02.fsa > \
/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/starterPhagesPlasmid/rmk202_phages_plasmids.fasta


makeblastdb -max_file_sz 1GB -in /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/starterPhagesPlasmid/rmk202_phages_plasmids.fasta \
 -out /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/starterPhagesPlasmid/rmk202_phages_plasmids -parse_seqids -dbtype nucl


  blastn -num_threads 35 -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/spacers/all_spacers4blast.fasta \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen sgi sacc sblastnames staxids sscinames" \
  -db /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/starterPhagesPlasmid/rmk202_phages_plasmids \
  -out /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages.txt \
  -evalue 0.01 -word_size 16
  
  
  cut -f 1 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages.txt  > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/mapped2local_spacersNames_onlyNames.txt
  

  diff /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/mapped2local_spacersNames_onlyNames.txt \
  /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/all_spacersNames_onlyNames.txt | grep ">" | sed 's/> //g' >\
  /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/unmapped2local_spacersNames_onlyNames.txt



##--------------------------blast to old viral DB

rm /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/spacers/unmapped2local_spacers4blast.fasta
for i in $(cat /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/unmapped2local_spacersNames_onlyNames.txt)
do
grep -A 1 "${i}" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/spacers/all_spacers4blast.fasta >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/spacers/unmapped2local_spacers4blast.fasta
done

  blastn -num_threads 35 -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/spacers/unmapped2local_spacers4blast.fasta \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen sgi sacc sblastnames staxids sscinames" \
  -db /archiv/BLAST/blast_db/20190204/viral_nucleotide/viral \
  -out /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_viralREF.txt \
  -evalue 0.01 -word_size 16
  
  
  cut -f 1 /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_viralREF.txt > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/mapped_spacersNames_onlyNames.txt
  

  diff /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/mapped_spacersNames_onlyNames.txt \
  /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/unmapped2local_spacersNames_onlyNames.txt | grep ">" | sed 's/> //g' >\
  /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/unmapped_spacersNames_onlyNames.txt

##--------------------------blast unmapped to nucleotide DB

rm /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/spacers/unmapped_spacers4blast.fasta
for i in $(cat /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/unmapped_spacersNames_onlyNames.txt)
do
grep -A 1 "${i}" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/spacers/all_spacers4blast.fasta >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/spacers/unmapped_spacers4blast.fasta
done


blastn -num_threads 37 -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/spacers/unmapped_spacers4blast.fasta \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen" \
  -db /archiv/BLAST/blast_db/20190204/nucleotide/prokaryotes \
  -out  /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_unmapped2prokaryoteREF.txt \
  -evalue 0.01 -word_size 16


  ##-------------------------transfer local
  scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_viralREF.txt ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/
  scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/all_spacersNames.txt ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/
  scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_unmapped2prokaryoteREF.txt ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/
  
  scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerBLAST/blast/CRISPRblast_rmk202Phages.txt ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/ 
  
```


```{r}

library(readr)
all_spacersNames <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/all_spacersNames.txt", "\t", escape_double = FALSE, col_names = c("sample","spacerName","location"), trim_ws = TRUE); all_spacersNames$Name <- paste(all_spacersNames$sample,all_spacersNames$spacerName,sep=":")


CRISPRblast_rmk202Phages <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/CRISPRblast_rmk202Phages.txt", "\t", escape_double = FALSE, col_names = c("Name","mappingContig","MappingQual","spacerLength","mismatch","gapopen","Querystart","QueryEND","Sequencestart","SequenceEND","evalue","bitscore","tax","lengthMappingSeq","donotknow_02","mappingName"),  trim_ws = TRUE) ; CRISPRblast_rmk202Phages$Name <- paste(str_split_fixed(CRISPRblast_rmk202Phages$Name, ":", 3)[,1],str_split_fixed(CRISPRblast_rmk202Phages$Name, ":", 3)[,2],sep=":")
CRISPRblast_PhageDB <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/CRISPRblast_viralREF.txt", "\t", escape_double = FALSE, col_names = c("Name","mappingContig","MappingQual","spacerLength","mismatch","gapopen","Querystart","QueryEND","Sequencestart","SequenceEND","evalue","bitscore","tax","lengthMappingSeq","donotknow_02","mappingName"),  trim_ws = TRUE); CRISPRblast_PhageDB$Name <- paste(str_split_fixed(CRISPRblast_PhageDB$Name, ":", 3)[,1],str_split_fixed(CRISPRblast_PhageDB$Name, ":", 3)[,2],sep=":")
CRISPRblast_Prokaryote <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/CRISPRblast_unmapped2prokaryoteREF.txt", "\t", escape_double = FALSE, col_names = c("Name","mappingContig","MappingQual","spacerLength","mismatch","gapopen","Querystart","QueryEND","Sequencestart","SequenceEND","evalue","bitscore","tax","lengthMappingSeq","donotknow_02","mappingName"),  trim_ws = TRUE);CRISPRblast_Prokaryote$Name <- paste(str_split_fixed(CRISPRblast_Prokaryote$Name, ":", 3)[,1],str_split_fixed(CRISPRblast_Prokaryote$Name, ":", 3)[,2],sep=":")

###-------------------preperation

CRISPRblast_rmk202Phages_prepped <- CRISPRblast_rmk202Phages %>% select(Name,mappingContig) %>% add_column(.,mapping2what="rmk202Phages") %>% add_column(.,mapping="mapping")
CRISPRblast_rmk202Phages_prepped$mappingContig <- gsub("_contig([0-9])","",CRISPRblast_rmk202Phages_prepped$mappingContig) %>% gsub("016","01",.)
colnames(CRISPRblast_rmk202Phages_prepped) <- c("Name","mappedGenome","DB","mapping")

CRISPRblast_PhageDB_prepped <- CRISPRblast_PhageDB %>% select(Name,tax) %>% add_column(.,mapping2what="PhageDB") %>% add_column(.,mapping="mapping")
CRISPRblast_PhageDB_prepped$tax<- gsub(" complete ([a-zA-Z])","",CRISPRblast_PhageDB_prepped$tax) %>% gsub("enome","",.)  %>% gsub(",","",.) %>% gsub("rophage g","",.)
colnames(CRISPRblast_PhageDB_prepped) <- c("Name","mappedGenome","DB","mapping")

CRISPRblast_Prokaryote_prepped <- CRISPRblast_Prokaryote %>% select(Name,tax) %>% add_column(.,mapping2what="BakteriaDB") %>% add_column(.,mapping="mapping")
CRISPRblast_Prokaryote_prepped$tax<- gsub(", complete genome","",CRISPRblast_Prokaryote_prepped$tax) %>% gsub(", complete sequence","",.)  %>% gsub(" chromosome","",.) %>% gsub(", chromosome: 1","",.) %>% gsub(" genome assembly,: 1","",.)
colnames(CRISPRblast_Prokaryote_prepped) <- c("Name","mappedGenome","DB","mapping")

CRISPRblast_mapped <- rbind(CRISPRblast_rmk202Phages_prepped,CRISPRblast_PhageDB_prepped,CRISPRblast_Prokaryote_prepped)

unmapped_spacer <- all_spacersNames[which(!all_spacersNames$Name %in% CRISPRblast_mapped$Name),] %>% select(Name) %>% add_column(.,mappedGenome="NA") %>% add_column(.,DB="non-mapping") %>% add_column(.,mapping="non-mapping")

CRISPRblast_all <- rbind(CRISPRblast_rmk202Phages_prepped,CRISPRblast_PhageDB_prepped,CRISPRblast_Prokaryote_prepped,unmapped_spacer)

###-------------------How many SPACERS map to the known viruses
CRISPRblast_all$DB = factor(CRISPRblast_all$DB, levels=c("BakteriaDB","PhageDB","rmk202Phages","non-mapping"))
colorsss <- (c("darkcyan","darkturquoise","lightblue","lightgray"))

ggplot(CRISPRblast_all, aes(x=mapping))+geom_bar()+theme_classic()+labs(x="",y="Frequency",title = "CRISPR spacers mapping")
MAPPINGTO <- ggplot(CRISPRblast_all, aes(x=mapping, fill=DB, color=DB))+geom_bar()+theme_classic()+labs(x="",y="Frequency",title = "CRISPR spacers mapping explained DB")+
   scale_fill_manual(values=colorsss)+
   scale_color_manual(values=colorsss)

MAPPINGTO
    png("~/Desktop/Projects/2019_RMK202_analysis/plot/MAPPINGTO.PNG", width = 1000, height = 1200,res=300)

MAPPINGTO

dev.off()
  
###-------------------To how many different viruses do they map

CRISPRblast_all_mapping <- CRISPRblast_all[which(CRISPRblast_all$mappedGenome!="NA"),]
MAPPINGTO2 <- ggplot(CRISPRblast_all_mapping, aes(x = forcats::fct_infreq(mappedGenome),fill=DB))+geom_bar(stat = "count")+theme_classic()+labs(x="",y="Frequency",title = "CRISPR spacers mapping to which phages")+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9))+scale_fill_manual(values=colorsss)

MAPPINGTO2
    png("~/Desktop/Projects/2019_RMK202_analysis/plot/MAPPINGTO2.PNG", width = 2500, height = 1900,res=300)

MAPPINGTO2

dev.off()
  


```


####map and follow CRISPR spacers over time

Here we map all raw reads against the Pan-CRISPR-array and follow the mapping coverage over time.

What we do:
1. Identify on reference SPACER per cluster. 
2. extract the spacer with the flanking repeat region
3. make a bwa index all all pan-CRISPR-arrays
4. map all timepoints against the pan-CRISPR-array

```{bash}



###================
##description file
###================
threads=37
samplesss=/archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt  ##the file with all sample names
logFilelocation=/archiv/Projects/2019_RMK202_analysis/01_log
BaseLocation=/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion
Assembly=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR_locations/FASTA/S_thermophilus_alltogether.CRISPRlocation.fasta
BaseLocation_mapping=/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/bwaMapping

##---------------------------Prepare
mkdir -p ${BaseLocation}/REF_mst2/assembled_and_CRISPR_pilerCR//
cat /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out > ${BaseLocation}/REF_mst2/assembled_and_CRISPR_pilerCR//.pilarCR_out
mkdir -p ${BaseLocation}/REF_mst1/assembled_and_CRISPR_pilerCR/
cat /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out > ${BaseLocation}/REF_mst1/assembled_and_CRISPR_pilerCR//.pilarCR_out
mkdir -p ${BaseLocation}/REF_RMK202/assembled_and_CRISPR_pilerCR//
cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/circlatorAfter/annotated/CRISPR/S_thermophilus_RMK202.pilarCR_out > ${BaseLocation}/REF_RMK202/assembled_and_CRISPR_pilerCR//.pilarCR_out



###=================================================================================
##olished
###=================================================================================

##---------------------------Prepare
mkdir -p /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerExtractall/tmp
rm /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerExtractall/CRISPR_spacers_demultiplexed.fasta
rm /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerExtractall/CRISPR_spacers_only.bed

spazerss=RMK202:a01s24
spazerss=13494_a3s06

#for spazerss in $(grep "*" /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/assembledSpacers/cdHIT/CRISPR_Clustering.txt.clstr |awk -F "\t" '{print $2}'|awk -F " " '{print $2}' |sed 's/\.\.\.//g')

for spazerss in $(grep "*" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results.txt.clstr |awk -F "\t" '{print $2}'|awk -F " " '{print $2}' |sed 's/\.\.\.//g')
  do
  
  ClusterName=$(grep -B 15 "${spazerss}" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/CD-hit_spacers/${species}/${species}_results.txt.clstr | grep "^>Cluster" |tail -1|sed 's/>//g' |sed 's/ //g')
  sampless=$(echo $spazerss | cut -d '_' -f 1|sed 's/>//g')
  samplessName=$(echo -e "S_*_202_"${sampless})
  array=$(echo $spazerss | cut -d '_' -f 2| cut -d 's' -f 1|sed 's/a//g' |sed 's/^0//g')
  spacer=$(echo $spazerss | cut -d '_' -f 2| cut -d 's' -f 2|cut -d '_' -f 1|sed 's/^0//g')
  arrayquery=$(echo "Array "${array})
  spacerROW=$((spacer+4))
  
  echo ${spacerROW}
  echo ${arrayquery}


#/archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${samplessName}/${samplessName}.pilarCR_out.fasta

  genomeFastaLocation=$(grep ".fas" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${samplessName}/${samplessName}.pilarCR_out |sed 's/^.*pilarCR_out://g' | cut -d ':' -f 1|head -1)
  contigName=$(grep -A 1 "${arrayquery}$" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${samplessName}/${samplessName}.pilarCR_out |tail -1 | sed 's/>//g')
  spacerINfo=$(grep -A ${spacerROW} "${arrayquery}$" /archiv/Projects/2019_RMK202_analysis/00_FINAL/02_CRISPR/pilerCR/${species}/${samplessName}/${samplessName}.pilarCR_out |tail -1 )
  
  start=$(echo $spacerINfo | awk '{print $1}')
  end=$(echo $spacerINfo | awk '{print $1+$2+$4}')
  
  startSpacernew=$(echo $spacerINfo | awk '{print $2}')
  Endspacernew=$(echo $spacerINfo | awk '{print $2+$4-1}')
  
  ##Because the assembler will most likely cut the last repeat and give it to the next sequence we most often have a contig end after a spacer. Therefore we add the repeat from before to the spacer
    startRepeatnew=$(echo $spacerINfo | awk '{print $1}')
  EndRepeatnew=$(echo $spacerINfo | awk '{print $1+$2-1}')
  
  echo -e ${contigName}"\t"${start}"\t"${end}"\t"${sampless}":ary"${array}"spc"${spacer} > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerExtractall/tmp/tmp.bed
  
   
  echo -e ${contigName}"\t"${startRepeatnew}"\t"${EndRepeatnew}"\t"${sampless}":ary"${array}"spc"${spacer} >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerExtractall/tmp/tmp.bed
  
  
  samtools faidx ${genomeFastaLocation}
  
bedtools getfasta -fi ${genomeFastaLocation} \
  -bed /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerExtractall/tmp/tmp.bed \
  -fo /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerExtractall/tmp/tmp.fasta

echo -e ">"${ClusterName} >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerExtractall/CRISPR_spacers_demultiplexed.fasta
#echo -e ">"${sampless}":ary"${array}"spc"${spacer} >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerExtractall/CRISPR_spacers_demultiplexed.fasta


grep ">" -v /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerExtractall/tmp/tmp.fasta >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerExtractall/CRISPR_spacers_demultiplexed.fasta

echo -e ">"${sampless}":ary"${array}"spc"${spacer} 

 echo -e ${sampless}":ary"${array}"spc"${spacer}"\t"${startSpacernew}"\t"${Endspacernew}"\t"${sampless}":ary"${array}"spc"${spacer}:${ClusterName} >> /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerExtractall/CRISPR_spacers_only.bed
  
  done
  
seqkit seq -w 60  /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerExtractall/CRISPR_spacers_demultiplexed.fasta > /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerExtractall/CRISPR_spacers_demultiplexed_final.fasta



##-----------------------------------------map rawReads

##============
##mapping to reference
##============
bwa index /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerExtractall/CRISPR_spacers_demultiplexed_final.fasta
# name_folder=02_againstpolished_single_rmk202_final_kneaddata_withStrains 

mkdir -p ${logFilelocation}

num=1
rm ${BaseLocation_mapping}/log/flagstat_logfile_CRISPRmappingGenome.txt
rm ${BaseLocation_mapping}/log/multiqc_logfile_CRISPRmappingGenome.txt
for names in $(cat ${samplesss})
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))
  rm -r ${BaseLocation_mapping}/${names}/bwaMapping2DB/

  mkdir -p ${BaseLocation_mapping}/${names}/bwaMapping2DB/
  
 # bwa mem -t ${threads} /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/di_K2_6h/Polishing_FINAL/FINAL_assembly_withPlasmids/di_rmk202_MAG_reference_polished.fasta
  bwa mem -t ${threads} /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerExtractall/CRISPR_spacers_demultiplexed_final.fasta \
  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_1.fastq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_2.fastq | samtools sort -@${threads} -O BAM -o ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 

samtools view -b -F 4 ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped.bam

##-------------------coverage


  samtools view -h -q 30 -b ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped.bam > ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30.bam
 
##----take only mappings longer than 42 

samtools view ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30.bam |cut -f 13 |sed 's/MD:Z://g'  | awk -F "[CTAG]" '{print $1+$2+$3+$4+$5+$6+$7+$8}' |sort -rn |uniq -c | sed -e 's/^[[:space:]]*//' |awk -F " " -v var="$names" '{OFS=" "} {print $0,var}' >> ${BaseLocation_mapping}/log/MappingQualityLength.txt


#samtools view ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped.bam |grep "M02102:390:000000000-C7G6T:1:1102:13014:15583:N:0:GGACTCCT+GTAAGGAG#0"

cd ${BaseLocation_mapping}/${names}/bwaMapping2DB/

samtools view ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30.bam > ${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub.sam
/home/vincent/scripts/mappingLengthfiltering.R -i ${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub.sam -o ${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub_out.sam
rm ${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub.sam

#samtools view ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30.bam |cut -f 13 |sed 's/MD:Z://g'  | awk -F "[CTAG]" '{if($1+$2+$3+$4+$5+$6+$7+$8>42) print $0}' |sort -rn |uniq -c | sed -e 's/^[[:space:]]*//' |awk -F " " -v var="$names" '{OFS=" "} {print $0,var}' >> ${BaseLocation_mapping}/log/MappingQualityLength.txt


samtools view -H ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped.bam > ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_header.sam  # extract header only

awk '{OFS="\t"} {print $1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14}' ${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub_out.sam > ${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub_out_02.sam
cat ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_header.sam \
 ${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub_out_02.sam |samtools view -S -b > ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub_out_final.bam


bedtools coverage -bed -a /archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/CRISPRspacerExtractall/CRISPR_spacers_only.bed -b ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub_out_final.bam  >  ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_coverage.bed 
  
  mkdir -p ${BaseLocation_mapping}/final
  
awk -F "\t" -v samplessss="$names" '{OFS="\t"} {print $0,samplessss}' ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_coverage.bed  >> ${BaseLocation_mapping}/final/CRISPR_spacer_coverage.bed

##-----Qualimap

mkdir -p ${BaseLocation_mapping}/${names}/bwaMapping2DB//bamqc

qualimap bamqc -bam ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir $${BaseLocation_mapping}/${names}/bwaMapping2DB//bamqc --java-mem-size=80G

echo ${names} >>  ${BaseLocation_mapping}/log/flagstat_logfile_CRISPRmappingGenome.txt
samtools flagstat {BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam >>  ${BaseLocation_mapping}/log/flagstat_logfile_CRISPRmappingGenome.txt

rm ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam &
 mkdir -p ${BaseLocation_mapping}/log
echo ${BaseLocation_mapping}"/"${names}"/bwaMapping2DB/bamqc" >> ${BaseLocation_mapping}/log/multiqc_logfile_CRISPRmappingGenome.txt

done

   rm -r ${BaseLocation}/log/MultiQC
 
  mkdir -p ${BaseLocation}/log/MultiQC
  
  /home/vincent/.local/bin/multiqc --config /archiv/logs/multiQC_config.txt --file-list  ${BaseLocation_mapping}/log/multiqc_logfile_CRISPRmappingGenome.txt \
    -o ${BaseLocation}/log/MultiQC

##-------------------calculate mapping reads for normalization
rm ${BaseLocation_mapping}/log/reads4normalization.txt
num=1
names=RMK202
for names in $(cat ${samplesss})
  do
  echo ${num}"/16  :" ${names}

reads=$(grep "CP046134" /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_rmk202_final_kneaddata_withStrains_PGAP/${names}/bwaMapping2DB/bamqc/genome_results.txt | sed -e 's/^[[:space:]]*//'|cut -f 3)

echo -e ${names}"\t"${reads}"\tStreptococcus thermophilus"  >> ${BaseLocation_mapping}/log/reads4normalization.txt

done

##-----------------------filtering reads for full length mapping
rm ${BaseLocation_mapping}/log/MappingQualityLength.txt
for names in $(cat ${samplesss})
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))

samtools view ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30.bam |cut -f 13 |sed 's/MD:Z://g'  | awk -F "[CTAG]" '{print $1+$2+$3+$4+$5+$6+$7+$8}' |sort -rn |uniq -c | sed -e 's/^[[:space:]]*//' |awk -F " " -v var="$names" '{OFS=" "} {print $0,var}' >> ${BaseLocation_mapping}/log/MappingQualityLength.txt

done

samtools view ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30.bam |head -1000 >  ${BaseLocation_mapping}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub.sam

cd ${BaseLocation_mapping}/${names}/bwaMapping2DB/

/home/vincent/scripts/mappingLengthfiltering.R -i ${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub.sam -o ${names}_rawIllumina_bwamem_sorted_mapped_mapQ_30_sub_out.sam


###------------------transfer local

scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/bwaMapping/final/CRISPR_spacer_coverage.bed ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/

scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/log/MultiQC/multiqc_report.html ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/

scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/bwaMapping/log/reads4normalization.txt ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/


scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/bwaMapping/log/MappingQualityLength.txt ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/


scp  vincent@130.223.51.116:/archiv/Projects/2019_RMK202_analysis/04_CRISPRmappings/01_mapping2CRISPRregion/bwaMapping/RMK202/bwaMapping2DB/RMK202_rawIllumina_bwamem_sorted_mapped_mapQ_30.bam ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/

scp  /home/vincent/apps/mappingLengthfiltering.R vincent@130.223.51.116:/home/vincent/scripts/
```


```{r}
library(readr)
CRISPR_spacer_coverage <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/CRISPR_spacer_coverage.bed",   "\t", escape_double = FALSE, col_names = c("Name","startSpacer","endSpacer","ClusterName","numReads","basesCoverd","basesTotal","numcov","sample"),  trim_ws = TRUE)

# length(unique(CRISPR_spacer_coverage$Name))

CRISPR_spacer_coverage$ClusterName <- str_split_fixed(CRISPR_spacer_coverage$ClusterName, ":", 3)[,3]

library(readr)
reads4normalization <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/reads4normalization.txt", "\t", escape_double = FALSE, col_names = c("sample","readNumber","species"), trim_ws = TRUE) ;  reads4normalization <- reads4normalization[,-3]

CRISPR_spacer_coverage <- merge(CRISPR_spacer_coverage,reads4normalization,by="sample",all.x = TRUE)

CRISPR_spacer_coverage$CPM <- 1000000*(CRISPR_spacer_coverage$numReads/CRISPR_spacer_coverage$readNumber)
##-----------------------plot over time
CRISPR_spacer_coverage$sample <- revalue(CRISPR_spacer_coverage$sample, c("lyo202_96"="Lyo\n1996","Lyo_202_2012"="Lyo\n2012", "Lyo_202_2014"="Lyo\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018","di_K2_6h"="cheesemaking\nday1","th_K2_8h"="cheesemaking\nday2"))

colnames(CRISPR_spacer_coverage)

CRISPR_spacer_coverage$sample = factor(CRISPR_spacer_coverage$sample, levels=c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018","cheesemaking\nday1","cheesemaking\nday2"))
levels(CRISPR_spacer_coverage$sample)


  ##---------------put infomration of location into plot
  
  CRISPR_spacer_coverage_extended <- merge(CRISPR_spacer_coverage,ClusterAssigned_final,by.y="clusterName",by.x = "ClusterName",all.x = TRUE)
CRISPR_spacer_coverage_extended$grouping
CRISPR_spacer_coverage_extended$grouping

 ##---------------add blast information
CRISPRblast_all_extended[,c(1,3)] %>% duplicate()


longCrispr_new_tmp <- longCrispr_new
longCrispr_new_tmp$array <- longCrispr_new_tmp$array %>% gsub("a","",.) %>% as.numeric() %>% as.character()
longCrispr_new_tmp$Name <- paste0(longCrispr_new_tmp$sample,":a",longCrispr_new_tmp$array,"s",as.character(as.numeric(longCrispr_new_tmp$spacer)))
longCrispr_new_tmp <- longCrispr_new_tmp[,c(1,10)]
# longCrispr_new_tmp$spacerName <- longCrispr_new_tmp$spacerName %>% gsub(":a",":ary",.) %>% gsub("s",":ary",.)
CRISPRblast_all_extended_forPlotting <- CRISPRblast_all_extended[!duplicated(CRISPRblast_all_extended[,c(1,3)]),c(1,3)]
CRISPRblast_all_extended_forPlotting$Name <- CRISPRblast_all_extended_forPlotting$Name %>% gsub(":ary",":a",.) %>% gsub("spc","s",.)  %>% gsub("Versand","Ver",.) %>% gsub("Konserve","Kon",.) %>% gsub("Lyo","L",.)

CRISPRblast_all_extended_forPlotting$Name %in% longCrispr_new_tmp$Name

CRISPRblast_all_extended_forPlotting_02 <- merge(CRISPRblast_all_extended_forPlotting,longCrispr_new_tmp,by.y="Name",by.x = "Name",all.x = TRUE)
CRISPRblast_all_extended_forPlotting_02 <- merge(CRISPRblast_all_extended_forPlotting,longCrispr_new_tmp,by.y="Name",by.x = "Name",all.x = TRUE)
  CRISPR_spacer_coverage_02 <- merge(CRISPR_spacer_coverage_extended,CRISPRblast_all_extended_forPlotting_02,by.y="clusterName",by.x = "ClusterName",all.x = TRUE)


# CRISPRblast_all_extended_forPlotting_02$clusterName %>% unique() %>% length()
  ##---------------plot
  
# range(CRISPR_spacer_coverage$Cpm)

  # p2 <- ggplot(CRISPR_spacer_coverage_02,aes(x=sample,y=CPM,group=ClusterName,color=assignedArray))+ geom_line(size=0.5, alpha=.5)+ 
  p2 <- ggplot(CRISPR_spacer_coverage_02,aes(x=sample,y=CPM,group=ClusterName,color=assignedArray))+ geom_line(size=0.5, alpha=.5)+ 
    # facet_grid(day~kessel~species)+
     # facet_grid(DB~.)+
    facet_wrap(DB~., scales="free")+
    # coord_trans(y="log2")+
    # facet_grid(treatment~species)+
    # facet_grid(treatment~species)+
    labs("",
         x=" ",
         y="Number of reads mapping")+
    #scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    # scale_y_continuous(limits = c(0,10))+
    theme_classic()+
   # scale_fill_manual(values=all_colours)+
   # scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
      #axis.text.x = element_blank(),
          legend.position="right"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          )
  
  p2
    png("~/Desktop/Projects/2019_RMK202_analysis/plot/spacer_following.png", width = 2500, height = 1200,res=300)

p2

dev.off()
  
  
###-------------------------look at mapping length
library(readr)
library(patchwork)
  
MappingQualityLength <- read_table2("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/MappingQualityLength.txt",  col_names = c("count","length","sample"))
  
mappinglengthPlot <- ggplot(MappingQualityLength,aes(x=length,y=count,color=sample,fill=sample))+geom_bar(stat="identity")+theme_classic()
  mappinglengthPlot
MappingQualityLength_long <- MappingQualityLength[which(MappingQualityLength$length>42),]
ggplot(MappingQualityLength_long,aes(x=length,y=count,color=sample,fill=sample))+geom_bar(stat="identity")+theme_classic()

table(MappingQualityLength_long$sample)

MappingQualityLength$keeping <- ifelse(MappingQualityLength$length>42,"keep","remove")
MappingQualityLength_keeping <- aggregate(. ~sample+keeping, data=MappingQualityLength[,c("sample","count","keeping")], sum, na.rm=TRUE)

MappingQualityLength_keeping$keeping = factor(MappingQualityLength_keeping$keeping, levels=c("remove","keep"))
# colorsss <- (c("darkcyan","darkturquoise","lightblue","lightgray"))


keepingPlot <- ggplot(MappingQualityLength_keeping,aes(x=sample,y=count,color=keeping,fill=keeping))+geom_bar(stat="identity")+theme_classic()+labs(title = "42bp mapping thresehold",x="")+theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9))
keepingPlot


 
  svg("~/Desktop/Projects/2019_RMK202_analysis/plot/mappinglengthplot.svg",width=6,height=3)

(mappinglengthPlot|keepingPlot)
# (p1|p2|p3)+plot_layout(widths =c(2,4,2))

 dev.off()


```

all spacers that are represented by the local phages are very highly abundant
- most likely we have also protospacers read mappings
we need to find a method to split protospacer and spacer hits 
--> therefore I only take mappings that are longer than 42 (which is the longest spacer length)
with this i ensure that I do not have protospacer mappings but also some repeat mapping


