---
title: "RMK202_cheese_experiment_20190128"
author: "Vincent Somerville"
date: "4 January 2019"
output: html_document
---
##Setup

```{r setup ,echo=FALSE}
knitr::opts_chunk$set(message=FALSE, echo=FALSE, eval=FALSE)

```

#Tempatur modelling

#isolates

### renaming

```{bash}

#old labels
cat /archiv/Projects/2018_Culturomics/02_script/names_changed.txt
#new labels
181213_3	L70	mst3
181213_4	L44	mst10
181213_5	S50	mst1
181213_6	L108	mst7
181213_7	S72	mst2
181213_8	L99	mst8
181213_10	L104	mst6
181213_11	L80	mst5
181213_12	L35	mst9
181213_13	L71	mst4

```



###FastQC
In a first step we creat [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) reports for all the libraries. 
These are then visualized with [MultiQC](https://multiqc.info/)

```{bash fastqc}
source /archiv/Projects/2018_Culturomics/02_script/fileShortcuts.sh
threads=8

mkdir -p $mainPath/01_rawData/IlluminaFastQC/


for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/names_changed.txt)
  do
  echo ${id}

    
    fastqc $mainPath/01_rawData/fastq/gz/${id}_R1.fastq.gz -t ${threads} -o $mainPath/01_rawData/IlluminaFastQC/
    fastqc $mainPath/01_rawData/fastq/gz/${id}_R2.fastq.gz -t ${threads} -o $mainPath/01_rawData/IlluminaFastQC/
    
  done
  mkdir -p $mainPath/01_rawData/IlluminaFastQC/Multiqc,output_file='test.html')

  
  multiqc $mainPath/01_rawData/IlluminaFastQC/ -o $mainPath/01_rawData/IlluminaFastQC/Multiqc

##Merging both RMK202

zcat $mainPath/01_rawData/fastq/gz/Konserve_202_R1.fastq.gz $mainPath/01_rawData/fastq/gz/Versand_202_R1.fastq.gz | gzip > $mainPath/01_rawData/fastq/gz/RMK202_R1.fastq.gz

zcat $mainPath/01_rawData/fastq/gz/Konserve_202_R2.fastq.gz $mainPath/01_rawData/fastq/gz/Versand_202_R2.fastq.gz | gzip > $mainPath/01_rawData/fastq/gz/RMK202_R2.fastq.gz

##Merging all isolates

zcat $mainPath/01_rawData/fastq/gz/L104_R1.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L108_R1.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L35_R1.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L39_R1.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L44_R1.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L70_R1.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L71_R1.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L80_R1.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L99_R1.fastq.gz \
  $mainPath/01_rawData/fastq/gz/S50_R1.fastq.gz \
  $mainPath/01_rawData/fastq/gz/S72_R1.fastq.gz \
  | gzip > $mainPath/01_rawData/fastq/gz/isolates_R1.fastq.gz

zcat $mainPath/01_rawData/fastq/gz/L104_R2.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L108_R2.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L35_R2.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L39_R2.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L44_R2.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L70_R2.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L71_R2.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L80_R2.fastq.gz \
  $mainPath/01_rawData/fastq/gz/L99_R2.fastq.gz \
  $mainPath/01_rawData/fastq/gz/S50_R2.fastq.gz \
  $mainPath/01_rawData/fastq/gz/S72_R2.fastq.gz \
  | gzip > $mainPath/01_rawData/fastq/gz/isolates_R2.fastq.gz

##Merging both RMK202

base=$mainPath/01_rawData/fastq/gz/

zcat $base/Konserve_202_R1.fastq.gz $base/Versand_202_R1.fastq.gz $base/RMK202_old_R1.fastq.gz  | gzip > $base/RMK202_withOld_R1.fastq.gz

zcat $base/Konserve_202_R2.fastq.gz $base/Versand_202_R2.fastq.gz $base/RMK202_old_R2.fastq.gz $ | gzip > $base/RMK202_withOld_R2.fastq.gz


```

### Trim-galore
In the following we clip adaptors with [trim-galore]()

```{bash trimGalore}
source /archiv/Projects/2018_Culturomics/02_script/fileShortcuts.sh
threads=30
id=RMK202_withOld

for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/names_changed_new.txt)
  do
  echo "##=============="
  echo ${id}
  echo "##=============="
      rm -r $mainPath/01_rawData/trimmed_Galore/gz/${id}
      mkdir -p $mainPath/01_rawData/trimmed_Galore/gz/${id}
      trim_galore -paired -o $mainPath/01_rawData/trimmed_Galore/gz/${id} \
        $mainPath/01_rawData/fastq/gz/${id}_R1.fastq.gz  \
        $mainPath/01_rawData/fastq/gz/${id}_R2.fastq.gz 
        
  echo "##=============="
  echo ${id} "fastQC of trimmed"
  echo "##=============="
      
    mkdir -p $mainPath/01_rawData/trimmed_Galore/fastQC/
      
    fastqc $mainPath/01_rawData/trimmed_Galore/gz/${id}/${id}_R1_val_1.fq.gz -t ${threads} -o $mainPath/01_rawData/trimmed_Galore/fastQC/
    
    fastqc $mainPath/01_rawData/trimmed_Galore/gz/${id}/${id}_R2_val_2.fq.gz -t ${threads} -o $mainPath/01_rawData/trimmed_Galore/fastQC/
    
        
  done 
  

  mkdir -p $mainPath/01_rawData/trimmed_Galore/fastQC/Multiqc,output_file='test.html')

  multiqc $mainPath/01_rawData/trimmed_Galore/fastQC/ -o $mainPath/01_rawData/trimmed_Galore/fastQC//Multiqc




```

###spades

In this step we assemble the raw reads of the isolate with [spades](http://bioinf.spbau.ru/en/spades3.7).

```{bash spades}
source /archiv/Projects/2018_Culturomics/02_script/fileShortcuts.sh

threads=35

for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/names_changed.txt)
  do
  echo ==========================
   echo ${id}
   echo ==========================
 
 
       mkdir -p $mainPath/06_Spades/${id}/assembly/
 
 
     /home/vincent/miniconda2/pkgs/spades-3.13.0-0/bin/spades.py --pe1-1 $mainPath/01_rawData/trimmed_Galore/gz/${id}/${id}_R1_val_1.fq.gz \
       --pe1-2 $mainPath/01_rawData/trimmed_Galore/gz/${id}/${id}_R2_val_2.fq.gz \
       -o $mainPath/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta -t ${threads} -m 100

   
    done

#---------------
##short info
##---------------

for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/isolates_names_changed.txt)
  do
  echo ==========================
   echo ${id}
   echo ==========================
 
   grep ">" -c $mainPath/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/contigs.fasta 
    done
##---------------
##PostAssemblyAnalysis
##---------------

for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/isolates_names_changed.txt)
  do
  echo ==========================
   echo ${id}
   echo ==========================
    
    #tail -3 $mainPath/03_metaphlan2/${id}_trimmed/metaphlan_profile/profiled_metagenome_${id}_trimmed.txt
    PostAssemblyAnalysis.R -a $mainPath/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/scaffolds.fasta \
      -F $mainPath/01_rawData/trimmed_Galore/gz/${id}/${id}_R1_val_1.fq.gz \
      -R $mainPath/01_rawData/trimmed_Galore/gz/${id}/${id}_R2_val_2.fq.gz \
      -o $mainPath/06_Spades/${id}/assembly/PostaAssembly \
      -t 35
    
    done


```



### megabat2

here, we try to remove the Lactobacillus delbrueckii contamination from the Streptococcus thermophilus isolates with [metabat2](https://bitbucket.org/berkeleylab/metabat)

```{bash metabat}
threads=10
id=S50
date=20190203
##-------
##S50
##-------
mkdir -p /archiv/Projects/2018_Culturomics/06_Spades//${id}/metabat2/${date}/

metabat2 -t ${threads} \
  -i /archiv/Projects/2018_Culturomics/06_Spades//${id}/assembly/assemblyMetaSpades_20190121.fasta/scaffolds.fasta \
  /archiv/Projects/2018_Culturomics/06_Spades/S50/assembly/PostaAssembly/Allillumina2assembly/rawIllumina_bwamem_Scaffolds_sorted.bam \
  -o /archiv/Projects/2018_Culturomics/06_Spades//${id}/metabat2/${date}/binning_contigs


id=S72
##-------
##S72
##-------
mkdir -p /archiv/Projects/2018_Culturomics/06_Spades//${id}/metabat2/${date}/

metabat2 -t ${threads} \
  -i /archiv/Projects/2018_Culturomics/06_Spades//${id}/assembly/assemblyMetaSpades_20190121.fasta/scaffolds.fasta \
  /archiv/Projects/2018_Culturomics/06_Spades/S50/assembly/PostaAssembly/Allillumina2assembly/rawIllumina_bwamem_Scaffolds_sorted.bam \
  -o /archiv/Projects/2018_Culturomics/06_Spades//${id}/metabat2/${date}/binning_contigs
  
  
##-------
##BLAST QC
##-------

for genome in $(ls /archiv/Projects/2018_Culturomics/06_Spades//${id}/metabat2/${date}/)
  do
  
  genome_renamed=$(echo $genome |sed 's/.fa//g'  )
  
  #mkdir -p /archiv/Projects/2018_Culturomics/06_Spades//${id}/metabat2/${date}/${genome_renamed}_metaphlan
  
  
   infoseq /archiv/Projects/2018_Culturomics/06_Spades//${id}/metabat2/${date}/$genome 
   
  
 
  done

```


###PilerCR crispr for MAG

I identify the CRISPR spacers with [PilerCR](https://www.drive5.com/pilercr/).

```{bash pilercr_dialact extraction}
  

##=============
##pilerCR
##=============
cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Streptococcus_thermophilus_RMK202.fasta \
  \
##---------------
species=Streptococcus_thermophilus
##Streptococcus thermophilus
##---------------


id=S_thermophilus_RMK202
      echo ==========================
     echo ${id}
     echo ==========================
     # cd /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/
      
      mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/CRISPR/

  
  ~/apps/PilerCR/pilercr1.06/pilercr -in /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Streptococcus_thermophilus_RMK202.fasta \
    -out /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/CRISPR/${id}.pilarCR_out -noinfo\
    -seq /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/CRISPR/${id}.pilarCR_out.fasta

  
  perl ~/apps/PilerCR/PilerCR_parser_new /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/CRISPR/${id}.pilarCR_out > \
    /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/CRISPR/${id}.pilarCR_out_final
  
  



##---------------
species=Lactobacillus_delbrueckii
##Lactobacillus delbrueckii
##---------------

id=L_delbrueckii_RMK202
      echo ==========================
     echo ${id}
     echo ==========================
     # cd /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/
      
      mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/CRISPR/

  
  ~/apps/PilerCR/pilercr1.06/pilercr -in /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Lactobacillus_delbrueckii_RMK202.fasta  \
    -out /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/CRISPR/${id}.pilarCR_out -noinfo\
    -seq /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/CRISPR/${id}.pilarCR_out.fasta

  
  perl ~/apps/PilerCR/PilerCR_parser_new /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/CRISPR/${id}.pilarCR_out > \
    /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/CRISPR/${id}.pilarCR_out_final
  
  
```

In these section I aim to identify and extract all CRISPR spacers in all genomes sequenced from RMK202. 
I do this with [PilerCR](https://www.drive5.com/pilercr/).

```{bash pilercr_dialact extraction}
  
  date=20190312


##=============
##pilerCR
##=============

##---------------
species=Streptococcus_thermophilus
##Streptococcus thermophilus
##---------------

##prep
/archiv/Projects/2018_Culturomics/06_Spades//S50/metabat2/20190203/binning_contigs

/archiv/Projects/2018_Culturomics/06_Spades//S72/metabat2/20190203/binning_contigs

mkdir -p /archiv/Projects/2018_Culturomics/06_Spades//{S72,S50}/metabat2/20190203/binning_contigs/blast
id=S72
threads=30
blastn -num_threads ${threads} -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query /archiv/Projects/2018_Culturomics/06_Spades/${id}/metabat2/20190203//binning_contigs.1.fa \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen" \
  -db /archiv/BLAST/blast_db/20190204/nucleotide/prokaryotes \
  -out /archiv/Projects/2018_Culturomics/06_Spades/${id}/metabat2/20190203/binning_contigs/blast/blast_binning_contig_1.txt \
  -evalue 0.01 -word_size 64

blastn -num_threads ${threads} -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query /archiv/Projects/2018_Culturomics/06_Spades/${id}/metabat2/20190203/binning_contigs.2.fa \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen" \
  -db /archiv/BLAST/blast_db/20190204/nucleotide/prokaryotes \
  -out /archiv/Projects/2018_Culturomics/06_Spades/${id}/metabat2/20190203/binning_contigs/blast/blast_binning_contig_2.txt \
  -evalue 0.01 -word_size 64

  
  for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/names_changed_isolates.txt |grep "^S")
    do
      echo ==========================
     echo ${id}
     echo ==========================
      cd /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/
      
      mkdir -p /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}
  cd /archiv/Projects/2018_Culturomics/06_Spades/${id}/metabat2/20190203/
  
  ~/apps/PilerCR/pilercr1.06/pilercr -in binning_contigs.2.fa \
    -out /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}/${id}.pilarCR_out -noinfo\
    -seq /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}/${id}.pilarCR_out.fasta
  
  #id=S50
  #date=20190214
  
  perl ~/apps/PilerCR/PilerCR_parser_new /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}/${id}.pilarCR_out > \
    /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}/${id}.pilarCR_out_final
  
  
    done > /archiv/Projects/2018_Culturomics/06_Spades/Pilar_${date}_log.out 2> /archiv/Projects/2018_Culturomics/06_Spades/Pilar_${date}_log_error.out
  


##---------------
species=Lactobacillus_delbrueckii
##Lactobacillus delbrueckii
##---------------

for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/names_changed_isolates.txt |grep "^L")
  do
    echo ==========================
   echo ${id}
   echo ==========================
    cd /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/
    
    mkdir -p /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}

~/apps/PilerCR/pilercr1.06/pilercr -in scaffolds.fasta \
  -out /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}/${id}.pilarCR_out -noinfo\
  -seq /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}/${id}.pilarCR_out.fasta

perl ~/apps/PilerCR/PilerCR_parser_new /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}/${id}.pilarCR_out > \
  /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/${date}/${id}.pilarCR_out_final


  done > /archiv/Projects/2018_Culturomics/06_Spades/Pilar_${date}_log.out 2> /archiv/Projects/2018_Culturomics/06_Spades/Pilar_${date}_log_error.out


```

now we try to see if there are differences in spacers between the CRISPR spacers of all genomes. 

```{bash}
date=20190312
source /archiv/Projects/2018_Culturomics/02_script/fileShortcuts.sh


##==================
##cd-hit
##==================
  
  perl ~/apps/PilerCR/PilerCR_parser_new /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out > \
    /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out_final
    
  perl ~/apps/PilerCR/PilerCR_parser_new /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out > \
    /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out_final
  #--------------
  ##Streptococcus thermophilus
  #--------------
  rm -r /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit
  mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/{sterm,ldel}

  cat /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out_final \
      /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out_final \
      /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/CRISPR/S_thermophilus_RMK202.pilarCR_out_final  >> \
  /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/sterm/GenomePath.txt
  
  grep ">" -c /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/sterm/GenomePath.txt
  
  
sed 's/>/>mst2:/g' /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out_final > /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out_final_named 
sed 's/>/>mst1:/g' /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out_final > /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out_finall_named 
sed 's/>/>RMK202:/g' /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/CRISPR/S_thermophilus_RMK202.pilarCR_out_final >  /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/CRISPR/S_thermophilus_RMK202.pilarCR_out_final_named

cat /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out_final_named  \
   /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out_finall_named   \
   /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/CRISPR/S_thermophilus_RMK202.pilarCR_out_final_named > \
   /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_named
   
  grep ">" -c /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_named
  
  
   cd-hit -i /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_named \
    -o /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/sterm/sterm_results.txt -c 0.8 -M 30000 -T 37
  
  
  grep -c ">" -c /archiv/Projects/2018_Culturomics/06_Spades/S72/PilerCR/${date}/S72.pilarCR_out_final
  grep -c ">" -c /archiv/Projects/2018_Culturomics/06_Spades/S50/PilerCR/${date}/S50.pilarCR_out_final
  grep -c ">" -c /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/CRISPR/S_thermophilus_RMK202.pilarCR_out_final

  
  ##count clusters and how many genomes per cluster
  date=20191002
  i=0
  start=0
  end=10
  mkdir -p /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/cdhit/20191002/Streptococcus_thermophilus/
  rm /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/cdhit/20191002/Streptococcus_thermophilus/clustering_results.txt
  
  for ((i=start; i<=end; i++))
  do
     echo "number: $i"
     
      grep "^${i}" -c /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/sterm/sterm_results.txt.clstr >> \
      /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/cdhit/20191002/Streptococcus_thermophilus/clustering_results.txt
      
  done
  
scp -r vincent@130.223.51.151:/archiv/Projects/2018_Culturomics/12_CRISPR_isolates/cdhit/20190312/Streptococcus_thermophilus/clustering_results.txt ~/Desktop/Projects/2018_culturomics/CRISPR/

#--------------
#Lactobacillus delbrueckii
#--------------

mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/{sterm,ldel}
rm /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/ldel/GenomePath.txt

for id in $(ls /archiv/Projects/2018_Culturomics/06_Spades/ |grep "^L" |grep -v "L39")
do

cat /archiv/Projects/2018_Culturomics/06_Spades/${id}/PilerCR/20190203/${id}.pilarCR_out_final | sed -e "s/>/>${id}_/g"  >> \
/archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/ldel/GenomePath.txt

done

cat /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/CRISPR/L_delbrueckii_RMK202.pilarCR_out_final >> \
/archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/ldel/GenomePath.txt

grep "^>" -c /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/ldel/GenomePath.txt



 cd-hit -i /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/ldel/GenomePath.txt \
  -o /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/ldel/ldel_results.txt -c 0.7 -M 30000 -T 37


##count clusters and how many genomes per cluster

start=0
end=12
for ((i=start; i<=end; i++))
do
   echo "number: $i"
   
    grep "^${i}" -c /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/ldel/ldel_results.txt.clstr >> \
    /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/ldel/ldel_clustering_results.txt
    
done

scp -r vincent@130.223.51.151:/archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/ldel/ldel_clustering_results.txt ~/Desktop/Projects/2018_culturomics/CRISPR/


```


#####make CRISPR clusters plot

```{r}

library(readr)
library(ggplot2)

##----------Ldel-------------
# ldel_CRISPR <- read_csv("~/Desktop/Projects/2018_culturomics/CRISPR/ldel_clustering.txt",col_names = FALSE)
ldel_CRISPR <- read_csv("~/Desktop/Projects/2018_culturomics/CRISPR/ldel_clustering_results.txt",col_names = FALSE)
#as.vector(ldel_CRISPR)
ldel_CRISPR_new <- cbind(1:13,ldel_CRISPR)
ldel_CRISPR_new[10,]
ldel_CRISPR_new$correctedCount <- NA
for (i in 10:2) {
  ldel_CRISPR_new[(i-1),3] <- ldel_CRISPR_new[i-1,2] - ldel_CRISPR_new[i,2]
}
# ldel_CRISPR_new <- ldel_CRISPR_new[c(-10,-11),]
colnames(ldel_CRISPR_new)[1] <- "cluster"
ldel_CRISPR_new$cluster <- as.factor(ldel_CRISPR_new$cluster)

ldel_CRISPR_new <- ldel_CRISPR_new[1:9,]
# ldel_CRISPR_new <- ldel_CRISPR_new[(ldel_CRISPR_new$correctedCount!=0 & !is.na(ldel_CRISPR_new$correctedCount)),]

plotldel <- ggplot(ldel_CRISPR_new,aes(x=cluster,y=correctedCount))+
    geom_bar(stat = "identity", colour = 'grey')+
          labs(y="count",x="cluster size") + 
    theme(legend.title = element_blank(),
      legend.position="none",
  axis.line = element_line(colour = "black"),
 # axis.text.x =element_blank(),
  panel.background = element_blank(),
  plot.margin=unit(c(0,0,0,0), "cm"),
  text = element_text(size=18),
  panel.border = element_rect(colour = "black", fill=NA, size=1))
plotldel
  svg("~/Desktop/Projects/2018_culturomics/CRISPR/ldel_clustering.svg",width=4,height=2)
plotldel
dev.off()

##----------Sterm-------------
# ldel_CRISPR <- read_csv("~/Desktop/Projects/2018_culturomics/CRISPR/sterm_clustering.txt",col_names = FALSE)
ldel_CRISPR <- read_csv("~/Desktop/Projects/2018_culturomics/CRISPR/clustering_results.txt",col_names = FALSE)
#as.vector(ldel_CRISPR)
ldel_CRISPR_new <- cbind(1:11,ldel_CRISPR)
ldel_CRISPR_new[10,]
ldel_CRISPR_new$correctedCount <- NA
for (i in 10:2) {
  ldel_CRISPR_new[(i-1),3] <- ldel_CRISPR_new[i-1,2] - ldel_CRISPR_new[i,2]
}
# ldel_CRISPR_new <- ldel_CRISPR_new[-(3:11),]
colnames(ldel_CRISPR_new)[1] <- "cluster"
ldel_CRISPR_new$cluster <- as.factor(ldel_CRISPR_new$cluster)

ldel_CRISPR_new <- ldel_CRISPR_new[(ldel_CRISPR_new$correctedCount!=0 & !is.na(ldel_CRISPR_new$correctedCount)),]

plotldel <- ggplot(ldel_CRISPR_new,aes(x=cluster,y=correctedCount))+
    geom_bar(stat = "identity")+
          labs(y="count",x="cluster size") + 
    theme(legend.title = element_blank(),
      legend.position="none",
  axis.line = element_line(colour = "black"),
 # axis.text.x =element_blank(),
  panel.background = element_blank(),
  plot.margin=unit(c(0,0,0,0), "cm"),
  text = element_text(size=18),
  panel.border = element_rect(colour = "black", fill=NA, size=1))
plotldel
  svg("~/Desktop/Projects/2018_culturomics/CRISPR/sterm_clustering.svg",width=2,height=2)
plotldel
dev.off()


```


#####map CRISPR spacers onto Sterm phage genome. 

```{bash}

threads=37

 bwa index /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Streptococcus_phage.fasta
 
 mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/
 
bwa mem -t 37 /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Streptococcus_phage.fasta /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_named | samtools sort -@${threads} -O BAM -o /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/mapping2StermPhageGenome_sort_02.bam - 

##intersect CRISPR spacers maps and SNPs --> to see if snps are in crispr regions
bedtools intersect -a /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/mapping2StermPhageGenome_sort.bam  \
  -b /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata/freebayesOuput_WithONT/variantCallsfreebayes_all_f_005_min_20x.vcf |samtools view


## from which genome do they originate from
  
  date=20190312

samtools view -b -F 4 /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/mapping2StermPhageGenome_sort_02.bam |samtools view| cut -f 1 > /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_matched_names.txt

sed 's/>//g' /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_matched_names.txt | cut -d ':' -f 1


samtools view -b -F 4 /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/mapping2StermPhageGenome_sort_02.bam |samtools view
##are they in repeat CRISPRs

cut -d '_' -f 1 /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_matched_names.txt > /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_matched_names_short.txt

for crispii in $(cat /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_matched_names_short.txt)
  do
  echo "====================="
  echo $crispii
  grep "$crispii" -A 1 /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/CD-hit/sterm/sterm_results.txt.clstr
  done

##check if assembly is aquiring new crispr spacers---> find indels in crispr array

```
What we detect:

- There are 5 CRISPR spacers that match to the phage in all strains perfectly to the phage
- ,,,
- all CRISPR spacers are different and seem to have evolved seperately
- There are no snps within the CRISPR spacers

##### pairwise CRISPR 
want to check for pairwise CRISPR diversity
```{bash}
###split  every CRIPSR into indiviual files

rm -r /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/CRISPRspacersRaw
mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/CRISPRspacersRaw

for spacer in $(grep "^>" /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_named|sed 's/>//g')
  do

spacer1_final=$(echo ${spacer} | cut -d '_' -f 1| sed 's/:/_/g' )
samtools faidx /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_named ${spacer} > /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/CRISPRspacersRaw/${spacer1_final}.fasta

  done

##-------------
##pairwise needle-wunsch alignment
##-------------

rm -r /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle
mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle

for spacer1 in $(grep "^>" /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_named|sed 's/>//g')
  do
  
  spacer1_final=$(echo ${spacer1} | cut -d '_' -f 1| sed 's/:/_/g' )

  for spacer2 in $(grep "^>" /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/mapping2phageGenome/CRISPR_alltogether_named|sed 's/>//g')
    do
    
    spacer2_final=$(echo ${spacer2} | cut -d '_' -f 1| sed 's/:/_/g' )

    needle -asequence /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/CRISPRspacersRaw/${spacer1_final}.fasta \
      -bsequence /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/CRISPRspacersRaw/${spacer2_final}.fasta -gapopen 20 -gapextend 0 -outfile /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle/${spacer1_final}__${spacer2_final}.msa

      length=$(grep "Length" /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle/${spacer1_final}__${spacer2_final}.msa | cut -d ':' -f 2 | sed -e 's/^[ \t]*//' )
      Identity=$(grep "Identity" /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle/${spacer1_final}__${spacer2_final}.msa | cut -d ':' -f 2 | cut -d '/' -f 1| sed -e 's/^[ \t]*//' )
      Gaps=$(grep "Gaps" /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle/${spacer1_final}__${spacer2_final}.msa | cut -d ':' -f 2 | cut -d '/' -f 1| sed -e 's/^[ \t]*//' )
      Score=$(grep "Score" /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle/${spacer1_final}__${spacer2_final}.msa | cut -d ':' -f 2 | sed -e 's/^[ \t]*//' )
      
     echo -e ${spacer1_final}"\t"${spacer2_final}"\t"${length}"\t"${Identity}"\t"${Gaps}"\t"${Score} >> /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle/final_pairwise_alignment.txt
     
    rm /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle/${spacer1_final}__${spacer2_final}.msa
    done
    
  done
  
  scp -r vincent@130.223.51.151:/archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/needle/final_pairwise_alignment.txt /home/vincent/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/


```

```{r}
library(readr)
library(ggplot2)
spacer_pairs <- read_delim("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/final_pairwise_alignment.txt",  "\t", escape_double = FALSE, col_names = c("spacer1","spacer2","length","matching","gaps","score"),trim_ws = TRUE)

spacer_pairs$id <- 100*(spacer_pairs$matching/spacer_pairs$length)



##----------subset
100*(sum(spacer_pairs$id>90)/nrow(spacer_pairs))
100*(sum(spacer_pairs$id>90)/nrow(spacer_pairs))

##----------create matrix

names <- levels(as.factor(spacer_pairs$spacer1))

pairwise_spacer_Matrix <- data.frame(matrix(NA, nrow = length(names), ncol = length(names)))
colnames(pairwise_spacer_Matrix) <- names
rownames(pairwise_spacer_Matrix) <- names

for (i in 1:length(names)) {
   rowName <- names[i]
    print(rowName)
  for (a in 1:length(names)) {
   rowName <- names[i]
    print(rowName)
    
    colNames <- names[a]
    print(colNames)
    
    pairwise_spacer_Matrix[rowName,colNames] <- spacer_pairs[spacer_pairs$spacer1==rowName & spacer_pairs$spacer2==colNames,"id"]
    
    }
        
# count <- count +1 
  
}


##-----------heatmap 2
library(stringr)
library(rafalib)
library(RColorBrewer)
names_genome <- str_split_fixed(names, "_", 2)[,1]
cols <- palette(brewer.pal(3, "Dark2"))[as.fumeric(names_genome)]



library(gplots) ##Available from CRAN
pairwise_spacer_Matrix_ready <- as.matrix(pairwise_spacer_Matrix)
image <- heatmap.2(pairwise_spacer_Matrix_ready,
          trace="none",
          # Rowv = "none",
          Colv=NA,
          ColSideColors=cols,
          # density.info = "none") ##historgram in col scale
          tracecol="black",
          )


svg("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/Streptococcus_thermophilus_SPACERS.svg",width=5,height=5)
 heatmap.2(pairwise_spacer_Matrix_ready,
          trace="none",
          # Rowv = "none",
          Colv=NA,
          cexRow=0.1,
          ColSideColors=cols,
          # density.info = "none") ##historgram in col scale
          tracecol="black",
          )
 dev.off()

 
 
library(plotly)

library(orca)
p <- heatmaply(pairwise_spacer_Matrix_ready, k_row = NA, dendrogram = "row",ColSideColors=cols)
plotly::orca(p, "Streptococcus_thermophilus_SPACERS_plotly.svg")

##-----------plot histogramm

ggplot(spacer_pairs,aes(x=id))+geom_histogram(binwidth = 3)+theme_classic()+
  labs(x="pairwise spacer identity [%]")


##-----------subset for only unique

names <- unique(spacer_pairs$spacer1)


##creat ANI matrix 
count <- 2
spacer_pairs$combined <- paste(spacer_pairs$spacer1,spacer_pairs$spacer2,sep="_")
sterm_curated <- spacer_pairs[1,]

for (i in 1:length(names)) {
   rowName <- names[i]
    print(rowName)
  for(a in count:length(names)){
    
    colNames <- names[a]
    print(colNames)
 
      temp <- spacer_pairs[spacer_pairs$spacer1==rowName &spacer_pairs$spacer2==colNames,]
        
   
    sterm_curated <- rbind(sterm_curated,temp)
    
    
  }
        
count <- count +1 
  
}
sterm_curated <- sterm_curated[-1,]
sterm_curated <- sterm_curated[which(!is.na(sterm_curated$id>80)),]

densityPlot <- ggplot(sterm_curated,aes(x=id))+geom_histogram(binwidth = 3)+theme_classic()+
  labs(x="pairwise spacer identity [%]")

100*(sum(sterm_curated$id>80)/nrow(sterm_curated))
100*(15/nrow(sterm_curated))

svg("~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/Streptococcus_thermophilus_SPACERS_histo.svg",width=5,height=3)

densityPlot
          
 dev.off()
 
 ##---------------export the shared spacers
 
sharedspacers <- sterm_curated[which(sterm_curated$id>80),]

unique_spacers <- unique(c(sharedspacers$spacer1,sharedspacers$spacer2))

write.table(unique_spacers, "~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/unique_Sterm_spacers.txt",na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names = FALSE)
```

##### map unique spacers
```{bash}

scp ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/unique_Sterm_spacers.txt vincent@130.223.51.151:/archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/

###-----------------
##-get unique spacers sequence
###-----------------


rm -r /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/CRISPRspacersUnique
mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/CRISPRspacersUnique

for spacer in $(cat /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/unique_Sterm_spacers.txt)
  do

cat /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/CRISPRspacersRaw/${spacer}.fasta > /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/CRISPRspacersUnique/${spacer}.fasta

cat /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/CRISPRspacersRaw/${spacer}.fasta >> /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/CRISPRspacersUnique/Unique_Sterm_CRISPR.fasta
  done
  
###-----------------
##-blast against phage database
###-----------------


#/archiv/BLAST/blast_db/20190204/phage_nucleotide/phage_nucleotide 
rm -r /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique
mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique


  blastn -num_threads 35 -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/CRISPRspacersUnique/Unique_Sterm_CRISPR.fasta \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen sgi sacc sblastnames staxids sscinames" \
  -db /archiv/BLAST/blast_db/20190204/viral_nucleotide/viral \
  -out /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF.txt \
  -evalue 0.01 -word_size 16
 head /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF.txt
 
awk -F "\t" '{OFS="\t"} {print $2,$9,$10 }' /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF.txt | sed 's/ref|//g' |sed 's/|//g' > /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF.bed

awk -F "\t" '{OFS="\t"} {if ($9<$10) print $2,$9,$10 ; else print $2,$10,$9 } ' /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF.txt | sed 's/ref|//g' |sed 's/|//g' > /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF.bed


###-----------------
##-get genome annotation of these phages
###-----------------


rm /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF_refseq.txt
 
for phage in $(cut -f 13 /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF.txt |awk -F "[ ,]" '{print $3}' |sed 's/\.//g')
  do
  echo "${phage}"
  grep -P "${phage}\t" /archiv/BLAST/blast_db/20190204/viral_genomic_file.txt |grep "Streptococc" >> /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF_refseq.txt
  
  grep -P "${phage}\t" /archiv/BLAST/blast_db/20190204/viral_genomic_file.txt |grep "Streptococc"  |cut -f 8 |cut -d ' ' -f 3
  echo "--------"
  done
  
  wc -l /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF.txt 
  wc -l /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF_refseq.txt
 
 sort /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF_refseq.txt| uniq > /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF_refseq_cleaned.txt

###-----------------
##-download gff file
###-----------------
rm -r /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/
mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/

 awk 'BEGIN{FS="\t";OFS="\t"}  {print $8 , $20}' /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF_refseq_cleaned.txt | cut -f 2 | \
    sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gff.gz|' >    \
    /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/downloadFile.txt
    

mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/GFF
cd /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/GFF
wget -i /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/downloadFile.txt 
gunzip *.gff.gz

cat /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/GFF/*.gff > /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/GFF/mergedAllMatching.gff

###-----------------
##-download fna file
###-----------------


 awk 'BEGIN{FS="\t";OFS="\t"}  {print $8 , $20}' /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF_refseq_cleaned.txt | cut -f 2 | \
    sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' >    \
    /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/downloadFile.txt
   # ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/841/145/GCF_000841145.1_ViralProj14226/GCF_000841145.1_ViralProj14226_genomic.fna.gz

mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FNA
cd /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FNA
wget -i /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/downloadFile.txt 
gunzip *.fna.gz

cat /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FNA/*fna > /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FNA/mergedAllMatching.fna

#ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/841/145/GCF_000841145.1_ViralProj14226/GCF_000841145.1_ViralProj14226_protein.faa.gz
###-----------------
##look for intersecting genes
###-----------------


bedtools intersect -wa -a /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/GFF/mergedAllMatching.gff  \
  -b /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF.bed | awk '{if ($3=="CDS") print $0}' > /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF_genes.gff
#bedtools intersect -a /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/GFF/mergedAllMatching.gff  \
#  -b /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF.bed | awk '{if ($3=="gene") print $0}'


###-----------------
##create faa for the matching GENES
###-----------------


bedtools getfasta -fi /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/PHAGES_CRISPRspacersUnique/FNA/mergedAllMatching.fna \
  -bed /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/CRISPRblast_viralREF_genes.gff \
  -fo /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/GENES_phages.fna

scp  vincent@130.223.51.151:/archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/GENES_phages.fna ~/Desktop/Projects/2019_RMK202_analysis/04_CRISPR_spacer/

  
  transeq -sequence /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/GENES_phages.fna \
  -outseq /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/GENES_phages.faa


###-----------------
##blast to UNIPROT db
###-----------------
rm /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/UNIPROT_DIAMOND_matching_CRISPRgenes.txt
diamond blastp --threads 32 --max-target-seqs 1 --db /archiv/TREMBL_database/20190710/uniprot_trembl.dmnd --query /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/BLAST_CRISPRspacersUnique/GENES_phages.faa --out /archiv/Projects/2019_pilotplant/02_annotation/CRISPR/pairwiseID/UNIPROT_DIAMOND_matching_CRISPRgenes.txt


```



# Illumina analysis

#### download

1. create a file containing all paths to the files from [LIMS](https://uhts-lgtf.vital-it.ch/user/symlinkdataruns/1453/951)
2. download the file with wget -i [File] in the appropitate folder
```{bash}
http://uhts-lgtf.vital-it.ch/symlink/V1_L6_R1_001_eVimQCCGzvPR.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V1_L6_R2_001_X0NeL654UFhg.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V1_L7_R1_001_jvgJBy864KhI.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V1_L7_R2_001_kzfyKrQ5YJuO.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V1_L8_R1_001_M9EtbZWANTlH.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V1_L8_R2_001_6LCQu5kVhdme.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V2_L6_R1_001_D3J3VNGUO2sb.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V2_L6_R2_001_Uh5JXQhzM0Rt.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V2_L7_R1_001_JJDw9KIxeXvd.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V2_L7_R2_001_6a1Vrn1zZPPQ.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V2_L8_R1_001_lPTEu1ohnTPS.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V2_L8_R2_001_eyxyKJ20KL3i.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V3_L6_R1_001_hf6ISQBSU3vw.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V3_L6_R2_001_eXTnM0N6Wnps.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V3_L7_R1_001_dQNFjCHgsKmE.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V3_L7_R2_001_gfhhG5a76yw4.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V3_L8_R1_001_DrKMggM4S7Eo.fastq.gz
http://uhts-lgtf.vital-it.ch/symlink/V3_L8_R2_001_4fB1hvNMF3Ts.fastq.gz

ls /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz |grep "^V"

```





###name change

create a file with the labels and the names of the file in each a column

```{bash}


for name in $(ls /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz |grep "^V")
do

  name_01=$(echo $name |cut -d '_' -f 1)
  Lane=$(echo $name |cut -d '_' -f 2)
  Read=$(echo $name |cut -d '_' -f 3)  
  
  newName=$(awk -F "\t" -v name="$name_01" '{if($1==name) {print $2}}' /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name.txt)
  
  echo ${newName}"-"${Lane}"-"${Read}".fastq.gz"
    
  mv /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/${name} /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/${newName}-${Lane}-${Read}.fastq.gz
    
    
done

```


concatenenat the fastq.gz files

```{bash}

mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/all_fastq_gz/ 

for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name.txt)
  do
  
  echo ${names}
  
  forward_01=$(ls /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/| grep "$names"  |grep "\-L1\-R1.fastq.gz")
  forward_02=$(ls /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/| grep "$names"  |grep "\-L2\-R1.fastq.gz")
  
  cat /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/${forward_01} \
  /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/${forward_02} > \
  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/all_fastq_gz/${names}_R1.fastq.gz
  
  reverse_01=$(ls /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/| grep "$names"  |grep "\-L1\-R2.fastq.gz")
  reverse_02=$(ls /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/| grep "$names"  |grep "\-L2\-R2.fastq.gz")

  cat /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/${reverse_01} \
  /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/${reverse_02} > \
  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/all_fastq_gz/${names}_R2.fastq.gz
  
  done

##-----------added files

mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/all_fastq_gz/ 
names=V1
for names in $(cut -f 1 /archiv/Projects/2019_pilotplant/01_rawData/log/20191031_logDownload_changeName.txt)
  do
  
  echo ${names}
  
  newName=$(grep "${names}" /archiv/Projects/2019_pilotplant/01_rawData/log/20191031_logDownload_changeName.txt |cut -f 2)
  
  forward_01=$(ls /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/| grep "^${names}_L6_R1")
  forward_02=$(ls /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/| grep "^${names}_L7_R1")
  forward_03=$(ls /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/| grep "^${names}_L8_R1")
  

  cat /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/${forward_01} \
  /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/${forward_02}  \
  /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/${forward_03} > \
  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/all_fastq_gz/${newName}_R1.fastq.gz
  
  
  reverse_01=$(ls /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/| grep "^${names}_L6_R2")
  reverse_02=$(ls /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/| grep "^${names}_L7_R2")
  reverse_03=$(ls /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/| grep "^${names}_L8_R2")
  
  cat /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/${reverse_01} \
  /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/${reverse_02}  \
  /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/fastq_gz/${reverse_03} > \
  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/all_fastq_gz/${newName}_R2.fastq.gz
  
   done



```


####FastQC

In a first step we creat [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) reports for all the libraries. 
These are then visualized with [MultiQC](https://multiqc.info/)

```{bash}
threads=4

rm -r /archiv/Projects/2019_pilotplant/01_rawData/03_rawFastQCs/allSamples_fastqcs

mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/03_rawFastQCs/allSamples_fastqcs

num=1
for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name.txt )
  do
  echo ${num}"/37  :" ${names}
  num=$((num+1))
    
    fastqc /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/all_fastq_gz/${names}_R1.fastq.gz -t ${threads} -o /archiv/Projects/2019_pilotplant/01_rawData/03_rawFastQCs/allSamples_fastqcs
    fastqc /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/all_fastq_gz/${names}_R2.fastq.gz -t ${threads} -o /archiv/Projects/2019_pilotplant/01_rawData/03_rawFastQCs/allSamples_fastqcs
    
  done
  mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/Multiqc,output_file='test.html')

  
  multiqc /archiv/Projects/2019_pilotplant/01_rawData/03_rawFastQCs/allSamples_fastqcs -o /archiv/Projects/2019_pilotplant/01_rawData/Multiqc

```




####Adapter removal


In the following we clip adaptors with [trim-galore](https://github.com/FelixKrueger/TrimGalore)

```{bash trimGalore}
threads=35

num=1
for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/log/20191031_logDownload_changeName.txt)
  do
  echo ${num}"/3  :" ${names}
  num=$((num+1))
  
  
  
  #    rm -r /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq//trimmed_Galore/gz/
      mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq//trimmed_Galore/gz/
      trim_galore --fastqc -paired -o /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/ \
        /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/all_fastq_gz/${names}_R1.fastq.gz   \
        /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/all_fastq_gz/${names}_R2.fastq.gz 
        
  echo "##=============="
  echo ${id} "fastQC of trimmed"
  echo "##=============="
      
    mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC/
      
    fastqc /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R1_val_1.fq.gz -t ${threads} -o /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore//fastQC/
    #names=th_K2_8h
    fastqc /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R2_val_2.fq.gz -t ${threads} -o /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC/
    
        
  done 
  

  mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC//Multiqc,output_file='test.html')

  multiqc /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC/ -o /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC/Multiqc


  for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/log/20191031_logDownload_changeName.txt)
  do
 mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC/log
echo "/archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC/" >>  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC/log/multiqc_logfile.txt
   
     done 
  
  mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC//Multiqc_02
  
  multiqc --config /archiv/logs/multiQC_config.txt --file-list /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/log/multiqc_logfile.txt \
  -o /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fastQC//Multiqc_02


```


x####KneadData

Here, we filter out the cow genome contamination reads

```{bash}

##prepare reference DB
mkdir -p /archiv/cowDB/mixed/20190520/

cat /archiv/cowDB/mito/20190520/cowDB_renamed.fasta \
  /archiv/cowDB/genome/20190520/GCF_002263795.1_ARS-UCD1.2_genomic.fna > \
  /archiv/cowDB/mixed/20190520/cow_genomeANDmito_DB.fna


bowtie2-build /archiv/cowDB/mixed/20190520/cow_genomeANDmito_DB.fna /archiv/cowDB/mixed/20190520/cow_genomeANDmito_DB

##===================
##remove reads
##===================

##--------------
##experiment
##--------------

threads=37

num=1
#for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name_K1.txt )
for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/log/20191031_logDownload_changeName.txt)
  do
  echo ${num}"/37  :" ${names}
  num=$((num+1))
rm -r /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}
mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}

gunzip /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R1_val_1.fq.gz
gunzip /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R2_val_2.fq.gz


kneaddata --input /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R1_val_1.fq \
  --input /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R2_val_2.fq \
  -db /archiv/cowDB/mixed/20190520/cow_genomeANDmito_DB  \
  --trimmomatic /home/vincent/Joanito/brain/Software/Trimmomatic-0.39/ \
  --output /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}

done


##--------------
##strains
##--------------


threads=37

for names in $(cat /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name_allOfExperiment.txt |grep -v "K1"|grep "Lyo" -v)
  do
  echo ==========================
   echo ${id}
   echo ==========================
 
rm -r /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}
mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}



kneaddata --input /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${names}/${names}_R1*fq \
  --input /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${names}/${names}_R2*fq \
  -db /archiv/cowDB/mixed/20190520/cow_genomeANDmito_DB  \
  --trimmomatic /home/vincent/Joanito/brain/Software/Trimmomatic-0.39/ \
  --output /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}

done

```

###mOTUs

```{bash}


num=1
  for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/names.txt)
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))
  rm -r /archiv/Projects/2019_pilotplant/mOTU/${names}/

  mkdir -p /archiv/Projects/2019_pilotplant/mOTU/${names}/
 
motus profile -f /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_1.fastq \
  -r /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_2.fastq -o /archiv/Projects/2019_pilotplant/mOTU/${names}/ -t 37 -n ${names}_mOTU

done

rm /archiv/Projects/2019_pilotplant/mOTU/all_mOTUs_prepforR.txt
  for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/names_onlymeta.txt)
  do
#  echo "-----------------------------------------"
#  echo ${names}

grep "^#" -v /archiv/Projects/2019_pilotplant/mOTU/${names}/* |grep -v "0.0000000000" |grep -v "sp." |awk -F "\t" -v sampless="${names}" 'BEGIN{OFS="\t"} {print $1, $2, sampless}' >> /archiv/Projects/2019_pilotplant/mOTU/all_mOTUs_prepforR.txt

done

scp -r vincent@130.223.51.151:/archiv/Projects/2019_pilotplant/mOTU/all_mOTUs_prepforR.txt /home/vincent/Desktop/Projects/2019_Nano_Meta/Illumina_analysis/


```

```{r}
library(readr)
library(ggplot2)

motus_abundance <- read_delim("/home/vincent/Desktop/Projects/2019_Nano_Meta/Illumina_analysis/all_mOTUs_prepforR.txt", "\t", escape_double = FALSE, col_names = c("species","abundance","sample"),trim_ws = TRUE)

motus_abundance$sample

motus_abundance$sample = factor(motus_abundance$sample, levels=c("Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202"))

Sterm_colours <- rev(c("#EB4D4D","#10B552"))
# Sterm_colours <- rev(c("#EB4D4D","#FAA0A0","indianred1","#10B552","#36E37B","#6CF5A3","cyan3","cyan"))

library(plyr)
motus_abundance$sample <- revalue(motus_abundance$sample, c("Lyo_202_2012"="Lyoampulle\n2012", "Lyo_202_2014"="Lyoampulle\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018"))

motus_abundance_meragable <- motus_abundance[,c("sample","species","abundance")]

motus_abundance_meragable$species <- revalue(motus_abundance_meragable$species, c("Streptococcus thermophilus [ref_mOTU_v25_01348]"="S. thermophilus", "Lactobacillus delbrueckii [ref_mOTU_v25_01298]"="L. delbrueckii\nspp. lactis"))
motus_abundance_meragable$abundance <- 100*motus_abundance_meragable$abundance
motus_abundance_meragable$method <- "mOTUs"

pMOTUs <-  ggplot( data = motus_abundance,aes(y = abundance, x = sample, group=species,fill = species))+ geom_bar( stat="identity")+
    labs("",
         x="",
         y="relative Abundance")+
    theme_classic()+
   # scale_color_viridis(discrete=TRUE)+
  scale_fill_manual(values=Sterm_colours)+
  # scale_fill_viridis(discrete=TRUE)+
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
      #axis.text.x = element_blank(),
          legend.position="right",
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank()
          )

  pMOTUs
  
###===========
##relative Abundance my method

  
 onlyBac <- bac_final[bac_final$species=="L_delbrueckii_RMK202"| bac_final$species=="S_thermophilus_RMK202",]
  
  
  

  onlyBac_sum <- aggregate(. ~sample, data=onlyBac[,c("sample","median")], sum, na.rm=TRUE)

onlyBac$total_coverage <- onlyBac_sum[match(onlyBac$sample,onlyBac_sum$sample),"median"]
onlyBac$percent_coverage <- 100*(onlyBac$median/onlyBac$total_coverage)

onlyBac$species <- as.factor(onlyBac$species)

levels(onlyBac$species)
onlyBac$species <- factor(onlyBac$species, levels=rev(c("S_thermophilus_RMK202","L_delbrueckii_RMK202")))
onlyBac$species <- revalue(onlyBac$species, c("S_thermophilus_RMK202"="S. thermophilus", "L_delbrueckii_RMK202"="L. delbrueckii\nspp. lactis"))

onlyBac$sample <- as.factor(onlyBac$sample)

levels(onlyBac$sample)
onlyBac$sample <- factor(onlyBac$sample, levels=(c("Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202")))

library(plyr)
onlyBac$sample <- revalue(onlyBac$sample, c("Lyo_202_2012"="Lyoampulle\n2012", "Lyo_202_2014"="Lyoampulle\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018"))

onlyBac_mergable <- onlyBac[,c("sample","species","percent_coverage")]
colnames(onlyBac_mergable)[3] <- "abundance"
onlyBac_mergable$method <- "mapping"

all_colours <- rev(c("#EB4D4D","#10B552") ) 



##----------------plot

PrelAbundance <-  ggplot( data = onlyBac,aes(y = percent_coverage, x = sample, group=interaction(species),fill = species))+ geom_bar( stat="identity")+
    labs("",
         x="",
         y="Relative\n  abundance [%]")+
    theme_classic()+
   # scale_color_viridis(discrete=TRUE)+
   scale_fill_manual(values=all_colours)+
  # scale_fill_viridis(discrete=TRUE)+
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
        #axis.text.x = element_blank(),
          legend.position="right",
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank()
          )

  PrelAbundance
  
    png("~/Desktop/presenations/my_presentations/20190919_VUA/abundance_species.png", width = 1900, height = 1200,res=300)

PrelAbundance

dev.off()
  
  
##-------merged methods
  motus_abundance_meragable
  onlyBac_mergable
all_methods <- rbind(as.data.frame(motus_abundance_meragable),as.data.frame(onlyBac_mergable))



PrelAbundance <-  ggplot( data = all_methods,aes(y = abundance, x = method, group=interaction(species),fill = species))+ geom_bar( stat="identity")+
  facet_grid(~sample)+
    labs("",
         x="",
         y="Relative\n  abundance [%]")+
    theme_classic()+
   # scale_color_viridis(discrete=TRUE)+
   scale_fill_manual(values=all_colours)+
  # scale_fill_viridis(discrete=TRUE)+
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
        # axis.text.x = element_blank(),
          legend.position="right",
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank()
          )

  PrelAbundance
  

#   svg("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.svg",width=7,height=4)

# 
# #  svg("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.svg",width=7,height=4)
png("~/Desktop/Manuscripts/2019_RMK202/Figures/sF01_relativeAbundance_motus.png", width = 2000, height = 1000,res=300)

PrelAbundance

dev.off()


```


##metaphlan2


```{bash}


num=1
  for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/names.txt)
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))
  rm -r /archiv/Projects/2019_pilotplant/metaphlan2/${names}/

  mkdir -p /archiv/Projects/2019_pilotplant/metaphlan2/${names}/{bowtie2ouput,metaphlan_profile}
 
    metaphlan2 --input_type fastq <(cat /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_1.fastq  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_2.fastq ) \
   --bowtie2out  /archiv/Projects/2019_pilotplant/metaphlan2/${names}//bowtie2ouput/metagenome_${names}_trimmed.bowtie2.bz2 --nproc 37 > \
   /archiv/Projects/2019_pilotplant/metaphlan2/${names}/metaphlan_profile/profiled_metagenome_${names}_trimmed.txt
   

done



```


## Nanopore analysis

The data was basecalled with ont-guppy-for-minknow/now 2.0.5-1~xenial amd64. 

#### get DATA

download data

```{bash}

wget https://campuscloud.unibe.ch/ssf/a/do?p_name=ss_forum&p_action=1&binderId=2753552&action=view_permalink&entityType=folder&novl_url=1&euet=null#1557129336614/fast5_pass.tar.gz.md5

```



#### Read filtering
In a first step I filter the fastq Nanopore reads for the pooled replicates with [filtlong](https://github.com/rrwick/Filtlong). 
I use a filter for min read length of 8kb and the best quality reads. 

```{bash}

##2AB
/home/vincent/apps/Filtlong/bin/filtlong --min_length 7000 --keep_percent 90 --target_bases 1000000000 --mean_q_weight 10 /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_2A2B/fastq_pass/all_2AB_raw.fq | gzip > /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_2A2B/fastq_pass/all_2AB_qualityFiltered_min7000_Q90.fq.gz

mv /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_2A2B/fastq_pass/all_2AB_qualityFiltered_min7000_Q90.fq /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_2A2B/fastq_pass/all_2AB_qualityFiltered_min7000_Q90.fq.gz

##1AB
/home/vincent/apps/Filtlong/bin/filtlong --min_length 7000 --keep_percent 90 --target_bases 1000000000 --mean_q_weight 10 /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_1A1B/fastq_pass/all_1AB_raw.fq | gzip > /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_1A1B/fastq_pass/all_1AB_qualityFiltered_min7000_Q90.fq.gz

mv /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_1A1B/fastq_pass/all_1AB_qualityFiltered_min7000_Q90.fq /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_1A1B/fastq_pass/all_1AB_qualityFiltered_min7000_Q90.fq.gz


##===============
##Demultiplexed data
##===============

mkdir -p /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered

for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
  do

  echo "=============="
  echo ${sample}
  echo "=============="

    /home/vincent/apps/Filtlong/bin/filtlong --min_length 7000 --keep_percent 90 --target_bases 1000000000 --mean_q_weight 10 /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_renamed/${sample}_nanopore_raw.fastq.gz | gzip > /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/${sample}_nanopore_filtered.fastq.gz


  done


scp -r vincent@130.223.51.151:/archiv/Projects/2019_pilotplant/01_rawData/Multiqc /home/vincent/Desktop/Projects/2019_Nano_Meta/Illumina_analysis


```


## Genome Assembly


#### Flye
Here, we create a [flye](https://github.com/fenderglass/Flye/blob/flye/docs/graph_example.png) assembly of the following:
1. Undemultiplexed samples with Qualtiy filtering max 100000000 and minLength 7000 reads
2. ...

Than I look at the Assembly graph with Bandage. 

The plasmid undemultiplexed seemed to be the best for K1 (1AB). this is based on the coverage the output and the 

```{bash}

mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/{1AB,2AB}/assembly/



##===========
##Assembly plasmid
##===========
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/{1AB,2AB}/assembly/


##2AB
flye --threads 37  --iterations 4 --genome-size 4500000 --plasmids --nano-raw /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_2A2B/fastq_pass/all_2AB_qualityFiltered_min7000_Q90.fq.gz --out-dir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/2AB/assembly/


##1AB
flye --threads 37 --iterations 4 --genome-size 4500000 --plasmids --nano-raw /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_1A1B/fastq_pass/all_1AB_qualityFiltered_min7000_Q90.fq.gz --out-dir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/assembly/

###---Bandage


scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/2AB/assembly/assembly_graph.gfa ~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/2AB/assembly/assembly_Graph_plasmid.gfa

scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/assembly/assembly_graph.gfa ~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/1AB/assembly/assembly_Graph_plasmid.gfa


##===========
##Demultiplexed
##===========

##===========
##Assembly plasmid
##===========
for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
  do
  
    sample=th_K2_8h
    
    rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${sample}/assembly/
    mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${sample}/assembly/
  echo "=============="
  echo ${sample}
  echo "=============="

    flye --threads 15  --iterations 5 --genome-size 4500000 --plasmids --nano-raw \
      /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/${sample}_nanopore_filtered.fastq.gz \
      --out-dir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${sample}/assembly/

  done



###---Bandage

for sample in $(echo "di_K1_10h
th_K1_8h
di_K2_6h
th_K2_8h")
do

mkdir -p ~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/assembly/

scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${sample}/assembly/assembly_graph.gfa ~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/assembly/${sample}_assembly_Graph_plasmid.gfa


done
```

#### PostAssembly analysis



```{bash}


##===========
##Assembly plasmid
##===========

##2AB

PostAssemblyAnalysis.R   -a /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/2AB/assembly/assembly.fasta \
  \ ## -l /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_2A2B/fastq_pass/all_2AB_raw.fq.gz \
  -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/2AB/postAssemblyAnalysis \
  -t 37 \
  --BlastDB /archiv/BLAST/blast_db/20190204/phage_nucleotide/phage_nucleotide 
  

##1AB

PostAssemblyAnalysis.R \
  -a /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/assembly/assembly.fasta \
  \ # -l /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_1A1B/fastq_pass/all_1AB_raw.fq.gz \
  -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/postAssemblyAnalysis \
  -t 37 \
  --BlastDB /archiv/BLAST/blast_db/20190204/phage_nucleotide/phage_nucleotide
  
##===========
##Assembly demultiplexed
##===========


for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
  do
  
    #sample=di_K1_10h
    
    rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${sample}/postAssemblyAnalysis
    mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${sample}/postAssemblyAnalysis
  echo "=============="
  echo ${sample}
  echo "=============="


  PostAssemblyAnalysis.R \
  -i Y \
  -x ${sample} \
  -a /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${sample}/assembly/assembly.fasta \
  -l /archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/${sample}_nanopore_filtered.fastq.gz \
  -F /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R1_val_1.f* \
  -R /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R2_val_2.f* \
  -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${sample}/postAssemblyAnalysis \
  -t 35 \
  -s Y \
  --BlastDB /archiv/BLAST/blast_db/20190204/phage_nucleotide/phage_nucleotide


  done
  
```

#### plasmid spades of non-mapping

```{bash}


##===============
##Assembly plasmid
##===============

###--------merge illumina libraries
mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/forONT_K1/
cat /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/di_K1_10h_R1_val_1.fq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/th_K1_8h_R1_val_1.fq > /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/forONT_K1/K1_bothDay1_2_ONT_R1.fq

cat /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/di_K1_10h_R2_val_2.fq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/th_K1_8h_R2_val_2.fq > /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/forONT_K1/K1_bothDay1_2_ONT_R2.fq

##--------Illumina mapping

#rm -r /home/vincent/Projects/2019_pilotplant/05_mapping2Nanopre/01_againstUnpolished

threads=37

names=K1_ONT

  echo ${num}"/37  :" ${names}
  num=$((num+1))
  rm -r /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}

  mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/

  # bwa index /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/assembly/assembly.fasta
  
  bwa mem -t ${threads} /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/assembly/assembly.fasta  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/forONT_K1/K1_bothDay1_2_ONT_R1.fq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/forONT_K1/K1_bothDay1_2_ONT_R2.fq | samtools sort -@${threads} -O BAM -o /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 

cd /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/
samstat /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam

samtools view -b -f 4 /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

bedtools bamtofastq -i /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam -fq /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq

rm /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

##-----Qualimap

mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB//bamqc

qualimap bamqc -bam /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB//bamqc --java-mem-size=80G


#scp -r vincent@130.223.51.151:/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/K1_ONT/bwaMapping2DB//bamqc ~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/1AB/assembly/


samtools index /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam

  
##-----plasmid spades/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/K1_ONT/spadesAssembly_unmapped/assembly/
mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/spadesAssembly_unmapped/assembly

/home/vincent/miniconda2/pkgs/spades-3.13.0-0/bin/spades.py --plasmid -t ${threads} -m 100 \
  -s /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq \
  -o /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/spadesAssembly_unmapped/assembly
  
##----blastn

mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/spadesAssembly_unmapped/blastn
  
  blastn -num_threads ${threads} -max_hsps 2 -max_target_seqs 2 -task megablast -show_gis \
  -query /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/spadesAssembly_unmapped//assembly/scaffolds.fasta \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen" \
  -db /archiv/BLAST/blast_db/20190204/phage_nucleotide/phage_nucleotide_cow_ \
  -out /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/spadesAssembly_unmapped/blastn/blastAssemblyaginstScaffolds.txt \
  -evalue 0.01 -word_size 64
  
  
  /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/K1_ONT/spadesAssembly_unmapped/assembly/

scp vincent@130.223.51.151:/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/K1_ONT/spadesAssembly_unmapped/assembly/assembly_graph_with_scaffolds.gfa ~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/1AB/assembly/PlasmidSpades_assembly_Graph_plasmid_K1_ONT.gfa

##cigar extraction

perl /home/vincent/scripts/mapped_read_Length_extended_reduced.pl /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_only100_statsforR.txt


###Go into next chunk to analyse data in R on bigmachine

#perl /home/vincent/scripts/mapped_read_Length_extended_reduced.pl /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_only100.bam




#scp vincent@130.223.51.151:/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_only100_statsforR.txt ~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/1AB/assembly/


##----------
##put plasmids and flye assembly together and map again
##----------
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/assembly_and_plasmidSpades/

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/assembly/assembly.fasta \
/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/K1_ONT/spadesAssembly_unmapped/assembly/scaffolds.fasta > \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/assembly_and_plasmidSpades/K1_ONT_assembly.fasta



scp /home/vincent/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/1AB/assembly/PlasmidSpades/selected_from_Bandage.fasta vincent@130.223.51.151:/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/K1_ONT/spadesAssembly_unmapped/assembly/ 

##Here, I first blasted all plasmidspades contigs and selected all non-cow contigs manually. 


cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/assembly/assembly.fasta \
/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/K1_ONT/spadesAssembly_unmapped/assembly/selected_from_Bandage.fasta  > \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/assembly_and_plasmidSpades/K1_ONT_assembly_unpolished.fasta


##===============
##demultiplexed
##===============

#rm -r /home/vincent/Projects/2019_pilotplant/05_mapping2Nanopre/01_againstUnpolished

threads=8

num=1
for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name.txt )
  do
  echo ${num}"/37  :" ${names}
  num=$((num+1))
  rm -r /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}

  mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/
  
  #gunzip /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R1_val_1.fq.gz
  #gunzip /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R2_val_2.fq.gz
  
  sampleShort=$(echo ${names}| awk -F "_" 'BEGIN{OFS="_"}{print $1,$2}')
  
   NanoSample=$(grep "$sampleShort" /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt) 
  
  #/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${NanoSample}/assembly/assembly.fasta
  
  bwa mem -t ${threads} /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${NanoSample}/assembly/assembly.fasta  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R1_val_1.fq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R2_val_2.fq | samtools sort -@${threads} -O BAM -o /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 

samtools view -b -f 4 /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

bedtools bamtofastq -i /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam -fq /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq

rm /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

##-----Qualimap

mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB//bamqc

qualimap bamqc -bam /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB//bamqc --java-mem-size=80G


 mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/log
echo "/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/"${names}"/bwaMapping2DB//bamqc" >> /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/log/multiqc_logfile.txt
   

samtools index /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam

 # gzip /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R1_val_1.fq
#  gzip /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R2_val_2.fq
  
  
##-----plasmid spades
mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/spadesAssembly_unmapped/assembly

/home/vincent/miniconda2/pkgs/spades-3.13.0-0/bin/spades.py --plasmid -t ${threads} -m 100 \
  -s /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq \
  -o /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/spadesAssembly_unmapped/assembly
  
##----blastn

mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/spadesAssembly_unmapped/blastn
  
  blastn -num_threads ${threads} -max_hsps 2 -max_target_seqs 2 -task megablast -show_gis \
  -query /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/spadesAssembly_unmapped//assembly/scaffolds.fasta \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen" \
  -db /archiv/BLAST/blast_db/20190204/phage_nucleotide/phage_nucleotide_cow_01 \
  -out /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/spadesAssembly_unmapped/blastn/blastAssemblyaginstScaffolds.txt \
  -evalue 0.01 -word_size 64
  
  
  done 

rm -r /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/MultiQC/
mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/MultiQC/


  
  multiqc --config /archiv/logs/multiQC_config.txt --file-list /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/log/multiqc_logfile.txt \
  -o /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/MultiQC/


```

###looking into read filtering with the edit distance
Good recources are the (SAM description)[https://samtools.github.io/hts-specs/SAMv1.pdf] and (additional file)[https://samtools.github.io/hts-specs/SAMtags.pdf] and the (blog post of WouterDeCoster)[https://gigabaseorgigabyte.wordpress.com/2017/04/14/getting-the-edit-distance-from-a-bam-alignment-a-journey/].

Interesting BWA output columns
- The edit distance is the NM tag ($12) in the sam file. The edit distance is the number of changes to the query to get the reference
- The CIGAR string is $6
- The read alignment length is the AS tag ($15)
- The MD tag, which stores a string containing matching numbers of nucleotides and non-matching nucleotides


NM:i:count Number of differences (mismatches plus inserted and deleted bases) between the sequence and
reference, counting only (case-insensitive) A, C, G and T bases in sequence and reference as potential
matches, with everything else being a mismatch. Note this means that ambiguity codes in both
sequence and reference that match each other, such as ‘N’ in both, or compatible codes such as ‘A’ and
‘R’, are still counted as mismatches. The special sequence base ‘=’ will always be considered to be a
match, even if the reference is ambiguous at that point. Alignment reference skips, padding, soft and
hard clipping (‘N’, ‘P’, ‘S’ and ‘H’ CIGAR operations) do not count as mismatches, but insertions and
deletions count as one mismatch per base.
Note that historically this has been ill-defined and both data and tools exist that disagree with this
definition.

This is also a good blog post by (David Tang)[https://davetang.org/wiki/tiki-index.php?page=SAM]


```{r}
library(readr)
library(ggplot2)

mappingStat <- read_delim("K1_ONT_rawIllumina_bwamem_sorted_only100_statsforR.txt", "\t", escape_double = FALSE, col_names = TRUE,trim_ws = TRUE)



sum(mappingStat$identity<0.96)
sum(mappingStat$identity<0.97)
sum(mappingStat$identity<0.98)
sum(mappingStat$identity<0.99)
dim(mappingStat)
sum(mappingStat$identity<45)
sum(mappingStat$Mapped_Length<45)
sum(mappingStat$Mapped_Length>45)
sum(mappingStat$EditDistance>5)
sum(mappingStat$EditDistance<5)

summary(mappingStat$Mapped_Length)
summary(mappingStat$EditDistance)
Id_dens <- ggplot(mappingStat,aes(x=identity))+geom_density()
NM_dens <- ggplot(mappingStat,aes(x=EditDistance))+geom_density()
ML_dens <- ggplot(mappingStat,aes(x=Mapped_Length))+geom_density()


   svg("plots/Id_dens.svg",width=5,height=5)
Id_dens
 dev.off()
 
 
   svg("plots/NM_dens.svg",width=5,height=5)
NM_dens
 dev.off()
 
 
   svg("plots/ML_dens.svg",width=5,height=5)
ML_dens
 dev.off()
 
 

```

move local

```{bash}



scp -r vincent@130.223.51.151:/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/${names}/bwaMapping2DB/plots/ ~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/1AB/assembly/


```


as circlator did not work. I will do a prokka annotation and annotate manually at the DnaA

```{bash}
###-----------shorten plasmidSpades contig Names
sed -i 's/>NODE_4215108_component_4+_length_7994_cov_7.50538/>plasmidSpades_ldelPlasmid/g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/assembly_and_plasmidSpades/K1_ONT_assembly_unpolished.fasta
sed -i 's/>NODE_4288578_component_0+_length_16466_cov_24.6103/>plasmidSpades_CowMito/g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/assembly_and_plasmidSpades/K1_ONT_assembly_unpolished.fasta

###-----------PRokka

rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB//03_PROKKA_annotation_polishedGenome/
    mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB//03_PROKKA_annotation_polishedGenome/
    
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB//03_PROKKA_annotation_polishedGenome/ --force --usegenus --genus Meta --locustag k1_ONT --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/assembly_and_plasmidSpades/K1_ONT_assembly_unpolished.fasta
  
  ##=================  
###-----------rearrang contigs
  ##=================  
  
  contig_1 22969
  contig_2 494618
  
###multifasta2fasta

mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/input/
for i in $(grep ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/assembly_and_plasmidSpades/K1_ONT_assembly_unpolished.fasta |sed 's/>//g' )
do
samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/assembly_and_plasmidSpades/K1_ONT_assembly_unpolished.fasta ${i} > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/input/${i}.fasta
done


###---------------extract fragemtent 1-------------------
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/logs/
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/output/

echo -e "contig_1\t1\t22962" > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/logs/contig1_fragment1.bed

samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/input/contig_1.fasta

bedtools getfasta -fi /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/input/contig_1.fasta \
  -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/logs/contig1_fragment1.bed \
  -fo /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/output/contig_1_fragment_1.fasta

grep -v ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/output/contig_1_fragment_1.fasta > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/output/contig_1_fragment_1_mergeable.fasta


###---------------extract fragemtent 3-------------------
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/logs/
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/output/

#seq_length.py /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/input/contig_1.fasta

echo -e "contig_1\t22962\t1894275" > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/logs/contig1_fragment2.bed

samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/input/contig_1.fasta

bedtools getfasta -fi /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/input/contig_1.fasta \
  -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/logs/contig1_fragment2.bed \
  -fo /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/output/contig_1_fragment_2.fasta

grep -v ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/output/contig_1_fragment_2.fasta > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/output/contig_1_fragment_2_mergeable.fasta


##-------merging fragments-------
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/final/

echo '>contig_1' > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/final/contig_1.fasta

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/output/contig_1_fragment_2_mergeable.fasta \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/output/contig_1_fragment_1_mergeable.fasta | sed 's/\n//g' >>  \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/final/contig_1.fasta
  

fasta_formatter -w 50 -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/final/contig_1.fasta -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/final/contig_1_new.fasta





###---------------extract fragemtent 1-------------------
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/logs/
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/output/

echo -e "contig_2\t1\t494612" > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/logs/contig2_fragment1.bed

samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/input/contig_2.fasta

bedtools getfasta -fi /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/input/contig_2.fasta \
  -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/logs/contig2_fragment1.bed \
  -fo /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/output/contig_2_fragment_1.fasta

grep -v ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/output/contig_2_fragment_1.fasta > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/output/contig_2_fragment_1_mergeable.fasta


###---------------extract fragemtent 3-------------------
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/logs/
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/output/

seq_length.py /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/input/contig_2.fasta

echo -e "contig_2\t494612\t2195337" > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/logs/contig2_fragment2.bed

samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/input/contig_2.fasta

bedtools getfasta -fi /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/input/contig_2.fasta \
  -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/logs/contig2_fragment2.bed \
  -fo /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/output/contig_2_fragment_2.fasta

grep -v ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/output/contig_2_fragment_2.fasta > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/output/contig_2_fragment_2_mergeable.fasta


##-------merging fragments-------
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/final/

echo '>contig_2' > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/final/contig_2.fasta

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/output/contig_2_fragment_2_mergeable.fasta \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/output/contig_2_fragment_1_mergeable.fasta | sed 's/\n//g' >>  \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/final/contig_2.fasta
  

fasta_formatter -w 50 -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/final/contig_2.fasta -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/final/contig_2_new.fasta


##merge all again
cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/final/contig_1_new.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/final/contig_2_new.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/input/contig_3.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/input/contig_4.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/input/contig_5.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/input/plasmidSpades_CowMito.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/input/plasmidSpades_ldelPlasmid.fasta > \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/final/K1_ONT_assembly_circlor_unformated.fasta


fasta_formatter -w 50 -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/final/K1_ONT_assembly_circlor_unformated.fasta -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/final/K1_ONT_assembly_circlor.fasta

##======
##check with PROKKa
##========
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB//04_PROKKA_annotation_polishedGenome_cirlazide/
    mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB//04_PROKKA_annotation_polishedGenome_cirlazide/
    
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB//04_PROKKA_annotation_polishedGenome_cirlazide/ --force --usegenus --genus Meta --locustag k1_ONT --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/final/K1_ONT_assembly_circlor.fasta
  
  
grep -i "dnaa" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB//04_PROKKA_annotation_polishedGenome_cirlazide/*gff


```



#### several rounds of polishing
After cirlarication we do several rounds fo polishing
This is then check by continious Quality control steps with the postAssemblyAnylsis tool. 
For polishing I consider the following tools:
- [Racon](https://github.com/isovic/racon)
- 

```{bash}

###===========================
###-----------------
##Racon polishing with graphmap
###-----------------
###===========================    
     
##------------------define variables------------------
for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt |tail -2)
do


name=K1_ONT
sample=K1_ONT
#sample=1
threads=37
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/final/K1_ONT_assembly_circlor.fasta 
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_1A1B/fastq_pass/all_1AB_raw.fq.gz

echo "========================================="
echo $sample
echo "========================================="

echo $reference_fasta
echo "---------------------------"
     
###===========================
###-----------------
##graphmap mapping
###-----------------
###===========================      


num=1

for name in first second third fourth
do 
rm -r $Overall_output_directory/0${num}_${name}Racon_graphmap/
mkdir -p $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/bamqc

#name=first

graphmap align -t ${threads} \
  -r $reference_fasta \
  -d $ont_reads \
  -o $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam 

#  -d $ont_reads | samtools sort -@${threads} -O BAM -o $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam - 

  
###===========================
###-----------------
##racon polishing
###-----------------
###===========================   

#rm -r$Overall_output_directory/0${num}_${name}Racon_graphmap/assembly

mkdir -p $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly

samtools view -bS $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam | samtools sort -@${threads} -O BAM -o $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam -

#samtools sort -@${threads} -O BAM $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam -

#samtools view -S -b $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam > $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam 

racon \
 -t ${threads} \
  $ont_reads \
  $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam \
  $reference_fasta > \
  $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly/${name}Racon_NWC_2Assembly.fasta
  


samtools index $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam 

qualimap bamqc -bam $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.bam \
  -outdir $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/bamqc --java-mem-size=40G 

rm $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/${name}Racon_ONT2FinalAssembly_sorted.sam

  reference_fasta=$(echo $Overall_output_directory/0${num}_${name}Racon_graphmap/assembly/${name}Racon_NWC_2Assembly.fasta)
echo $reference_fasta
num=$((num+1)) 

done

done # loop for 1AB and 2AB samples


###===========================
###-----------------
##quast
###-----------------
###===========================  

for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
do
#sample=1
threads=3
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${sample}/
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${sample}/assembly/assembly.fasta 
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/${sample}_nanopore_filtered.fastq.gz

echo "========================================="
echo $sample
echo "========================================="

echo $reference_fasta
echo "---------------------------"
     
    rm -r $Overall_output_directory/quastwithRaconONT/
    mkdir -p $Overall_output_directory/quastwithRaconONT
    quast.py \
      -t ${threads} -l "FlyeAssembly,firstONTracon,secondONTracon,thirdONTracon" \
      -R $Overall_output_directory/04_fourthRacon_graphmap/assembly/fourthRacon_NWC_2Assembly.fasta \
      $reference_fasta \
      $Overall_output_directory/01_firstRacon_graphmap/assembly/firstRacon_NWC_2Assembly.fasta \
      $Overall_output_directory/02_secondRacon_graphmap/assembly/secondRacon_NWC_2Assembly.fasta \
      $Overall_output_directory/03_thirdRacon_graphmap/assembly/thirdRacon_NWC_2Assembly.fasta \
      -o $Overall_output_directory/quastwithRaconONT
 
 
 ##-------------------making MultiQC output
 rm $Overall_output_directory/logs_MultiQC_raconONT.txt
 
 echo $Overall_output_directory/quastwithRaconONT/report.tsv >> $Overall_output_directory/logs_MultiQC_raconONT.txt
 
 num=1

for name in first second third fourth
do 

echo $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/bamqc >> $Overall_output_directory/logs_MultiQC_raconONT.txt
num=$((num+1)) 

done ##loop for name
 
done #loop for sample
###===========================
###-----------------
##MultiQC
###-----------------
###===========================  


for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
do
#sample=1
threads=3
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${sample}/
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${sample}/assembly/assembly.fasta 
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/Nanopore_demultiplexed_filtered/${sample}_nanopore_filtered.fastq.gz

echo "========================================="
echo $sample
echo "========================================="


  rm -r $Overall_output_directory/MultiQCRaconONT
    
  mkdir -p $Overall_output_directory/MultiQCRaconONT
  
  multiqc --config /archiv/logs/multiQC_config.txt --file-list  $Overall_output_directory/logs_MultiQC_raconONT.txt \
    -o $Overall_output_directory/MultiQCRaconONT/


done #loop for sample


###--------------on local
name=di_K1_10h
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${name}/MultiQCRaconONT/multiqc_report.html Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC/${name}_multiqc_report.html

name=th_K1_8h
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${name}/MultiQCRaconONT/multiqc_report.html Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC/${name}_multiqc_report.html
name=di_K2_6h
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${name}/MultiQCRaconONT/multiqc_report.html Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC/${name}_multiqc_report.html
name=th_K2_8h
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${name}/MultiQCRaconONT/multiqc_report.html Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC/${name}_multiqc_report.html
```


####freebayes polishing

```{bash}


###-----------------
##Illuminamapping rawReads to circlator genome
###-----------------

   
##------------------define variables------------------
for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
do


#sample=1
threads=37
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/04_fourthRacon_graphmap/assembly/fourthRacon_NWC_2Assembly.fasta
ont_reads=/archiv/Projects/2019_Nano_Meta/01_rawData/01_data/files/20190426_NB_1A1B/fastq_pass/all_1AB_raw.fq.gz



num=5


for run in fifth sixth seventh eight
   do 

name=run
##--------------create directories----------------
rm -r $Overall_output_directory/0${num}_${run}Freebayes/
mkdir -p $Overall_output_directory/0${num}_${run}Freebayes/{rawIlluminaMapped,freebayesOuput}
mkdir -p $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/bamqc

mkdir -p /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fq

# gunzip -c /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R1_val_1.fq.gz > \
#  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fq/${sample}_R1_val_1.fq
# gunzip -c /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${sample}_R2_val_2.fq.gz > \
#  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/fq/${sample}_R2_val_2.fq



bwa index $reference_fasta

bwa mem -t ${threads}  $reference_fasta /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/forONT_K1/K1_bothDay1_2_ONT_R1.fq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/forONT_K1/K1_bothDay1_2_ONT_R2.fq |samtools sort -@8 -O BAM \
-o $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam - 
     

qualimap bamqc -bam $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam \
        -outdir $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/bamqc --java-mem-size=40G
    
samtools index $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam

###-----------------
##freebayesAfterCirclator
###-----------------
echo freebayes

    SECONDS=0

    freebayes -F 0.5 --min-coverage 4 -p 1 -f \
      $reference_fasta \
      $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/rawIllumina_${name}_bwamem_Assembly_sorted.bam > \
      $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf
      
      
    echo "RUNTIME (h:min:s)= $(($SECONDS / 3600)):$((($SECONDS / 60) % 60)):$(($SECONDS % 60))"
    echo END = `date +"%Y-%m-%d %H:%M:%S"`

##-------   
##bcftools variant integration of Illumina data into Pacbio polished genome
##-------

    SECONDS=0
    echo "bgzip"
bgzip -c  $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf > \
       $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf.gz
      echo "tabix"
tabix -p vcf  $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf.gz
    echo "bcftools"
bcftools consensus -f $reference_fasta \
      -o $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${species}_${run}Freebayes.fasta \
       $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/variantCallsfreebayes.vcf.gz
      
    echo "RUNTIME (h:min:s)= $(($SECONDS / 3600)):$((($SECONDS / 60) % 60)):$(($SECONDS % 60))"
    echo END = `date +"%Y-%m-%d %H:%M:%S"`
  
  bwa index $Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${species}_${run}Freebayes.fasta 
  
   ##-------------------new variable name for next polishing steps-----------
  reference_fasta=$Overall_output_directory/0${num}_${run}Freebayes/freebayesOuput/${species}_${run}Freebayes.fasta  
  num=$((num+1)) 
done
    
done #samples
##indexing

for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
do

bwa index /archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${sample}/08_eightFreebayes/freebayesOuput/_eightFreebayes.fasta
done

###===========================
###-----------------
##quast
###-----------------
###===========================  

#for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
#do
#sample=1
threads=6
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL
reference_fasta=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/final/K1_ONT_assembly_circlor.fasta  

rm $Overall_output_directory/logs_MultiQC_raconFreebayesONT.txt

echo "========================================="
echo $sample
echo "========================================="

echo $reference_fasta
echo "---------------------------"
     
    rm -r $Overall_output_directory/quastwithRacon_FreebayesONT/
    mkdir -p $Overall_output_directory/quastwithRacon_FreebayesONT
    quast.py \
      -t ${threads} -l "FlyeAssembly,firstONTracon,secondONTracon,thirdONTracon,fourthFreebayes,fiftFreebayes,sixthFreebayes,seventhFreebayes" \
      -R $Overall_output_directory/08_eightFreebayes/freebayesOuput/_eightFreebayes.fasta \
      $reference_fasta \
      $Overall_output_directory/01_firstRacon_graphmap/assembly/firstRacon_NWC_2Assembly.fasta \
      $Overall_output_directory/02_secondRacon_graphmap/assembly/secondRacon_NWC_2Assembly.fasta \
      $Overall_output_directory/03_thirdRacon_graphmap/assembly/thirdRacon_NWC_2Assembly.fasta \
      $Overall_output_directory/04_fourthRacon_graphmap/assembly/fourthRacon_NWC_2Assembly.fasta \
      $Overall_output_directory/05_fifthFreebayes/freebayesOuput/_fifthFreebayes.fasta \
      $Overall_output_directory/06_sixthFreebayes/freebayesOuput/_sixthFreebayes.fasta \
      $Overall_output_directory/07_seventhFreebayes/freebayesOuput/_seventhFreebayes.fasta \
      -o $Overall_output_directory/quastwithRacon_FreebayesONT
 
 
 
 scp -r vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/quastwithRacon_FreebayesONT/ ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC_bothK1/
###---------------analaysis of Quast in following chunk
 
 ##-------------------making MultiQC output
 #rm $Overall_output_directory/logs_MultiQC_raconONT.txt
 
 #echo $Overall_output_directory/quastwithRacon_FreebayesONT/report.tsv >> $Overall_output_directory/logs_MultiQC_racon_freebayesONT.txt
 
 #num=1

#for name in first second third fourth
#do 

#echo $Overall_output_directory/0${num}_${name}Racon_graphmap/Graphmap_ONT/bamqc >> $Overall_output_directory/logs_MultiQC_raconFreebayesONT.txt
#num=$((num+1)) 

#done ##loop for name
 
# for run in fifth sixth seventh eight
#   do 

#echo $Overall_output_directory/0${num}_${run}Freebayes/rawIlluminaMapped/bamqc >> $Overall_output_directory/logs_MultiQC_raconFreebayesONT.txt

#done #run loops
 
#done #loop for sample
###===========================
###-----------------
##MultiQC
###-----------------
###===========================  


for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
do
#sample=1
threads=3
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${sample}/

echo "========================================="
echo $sample
echo "========================================="


  rm -r $Overall_output_directory/MultiQCRaconFreebayesONT
    
  mkdir -p $Overall_output_directory/MultiQCRaconFreebayesONT
  
  multiqc --config /archiv/logs/multiQC_config.txt --file-list  $Overall_output_directory/logs_MultiQC_raconFreebayesONT.txt \
    -o $Overall_output_directory/MultiQCRaconFreebayesONT/


done #loop for sample


###--------------on local
name=di_K1_10h
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${name}/MultiQCRaconFreebayesONT/multiqc_report.html Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC/${name}_multiqc_report.html

name=th_K1_8h
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${name}/MultiQCRaconFreebayesONT/multiqc_report.html Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC/${name}_multiqc_report.html
name=di_K2_6h
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${name}/MultiQCRaconFreebayesONT/multiqc_report.html Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC/${name}_multiqc_report.html
name=th_K2_8h
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${name}/MultiQCRaconFreebayesONT/multiqc_report.html Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC/${name}_multiqc_report.html


###==============
##Blast
###==============


for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
do
#sample=1
threads=3
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${sample}/


echo "========================================="
echo $sample
echo "========================================="


    rm -r $Overall_output_directory/08_eightFreebayes/BLAST/
    mkdir -p $Overall_output_directory/08_eightFreebayes/BLAST/

blastn -num_threads 35 -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis -query $Overall_output_directory/08_eightFreebayes/freebayesOuput/_eightFreebayes.fasta -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen" -db /archiv/BLAST/blast_db/20190204/phage_nucleotide/phage_nucleotide_cow_03 -out $Overall_output_directory/08_eightFreebayes/BLAST/finalBlast.txt -evalue 0.01 -word_size 12

 
 done #samples


```

Here, I do the analysis of the Quast output

```{r}

library(readr)
library(ggplot2)
library(gridExtra)
quast_K1 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/01_polishingQC_bothK1/quastwithRacon_FreebayesONT/report.tsv","\t", escape_double = FALSE, trim_ws = TRUE)

rownames(quast_K1) <- quast_K1$Assembly
# quast_K1 <- quast_K1[,-1]

quast_K1_analysis <- t(quast_K1)
View(quast_K1_analysis)
quast_K1_analysis <- as.data.frame(quast_K1_analysis[-1,])
quast_K1_analysis$nameSample <- rownames(quast_K1_analysis)

quast_K1_analysis$`# misassemblies` <- as.double(as.character(quast_K1_analysis$`# misassemblies`))
quast_K1_analysis$`Duplication ratio` <- as.double(as.character(quast_K1_analysis$`Duplication ratio`))
quast_K1_analysis$`# mismatches per 100 kbp` <- as.double(as.character(quast_K1_analysis$`# mismatches per 100 kbp`))
quast_K1_analysis$`# indels per 100 kbp` <- as.double(as.character(quast_K1_analysis$`# indels per 100 kbp`))

quast_K1_analysis$nameSample_f = factor(quast_K1_analysis$nameSample, levels=c("FlyeAssembly","firstONTracon","secondONTracon","thirdONTracon","fourthFreebayes","fiftFreebayes","sixthFreebayes","seventhFreebayes"))


##Misassemblies
p1 <- ggplot(quast_K1_analysis,aes(x=nameSample_f,y=`# misassemblies`,group=1))+geom_point(size=2)+geom_line()+
  theme_classic()+
  labs(x="",
       y="# Misassemblies")+
  theme(axis.text.x = element_blank())


##Misassemblies
p2 <- ggplot(quast_K1_analysis,aes(x=nameSample_f,y=`Duplication ratio`,group=1))+geom_point(size=2)+geom_line()+
  theme_classic()+
  labs(x="",
       y="Duplication ratio")+
  theme(axis.text.x =element_blank())


##Mismatches
p3 <- ggplot(quast_K1_analysis,aes(x=nameSample_f,y=`# mismatches per 100 kbp`,group=1))+geom_point(size=2)+geom_line()+
  theme_classic()+
  labs(x="",
       y="# Mismatches per 100 kbp")+
  theme(axis.text.x = element_text(angle = 75, hjust = 1))

##INdels
p4 <- ggplot(quast_K1_analysis,aes(x=nameSample_f,y=`# indels per 100 kbp`,group=1))+geom_point(size=2)+geom_line()+
  theme_classic()+
  labs(x="",
       y="# INDELS per 100 kbp")+
  theme(axis.text.x = element_text(angle = 75, hjust = 1))


grid.arrange(p1,p2,p3,p4, ncol=2, heights=c(1,2))

  svg("~/Desktop/presenations/my_presentations/20190708_groupmeeting/figures/polishing_K1.svg",width=6,height=5)
grid.arrange(p1,p2,p3,p4, ncol=2, heights=c(2,3))
 dev.off()


```


#### Ideel

In the following we run [ideel](https://github.com/phiweger/ideel).
Ideel should be run after all polishing steps in order to see the changes after the single steps. 

In order to run Ideel we first need to download the (tremble protein database)[https://www.uniprot.org/downloads]. 
...and after clone from github I had to change the location of the tremble.database in the SnakeFile before getting it to work. 


```{bash}

##download Database
mkdir -p /archiv/TREMBL_database/20190710
cd /archiv/TREMBL_database/20190710 
wget ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_trembl.fasta.gz

##make database
cd /archiv/TREMBL_database/20190710 

pigz -d /archiv/TREMBL_database/20190710/uniprot_trembl.fasta.gz

diamond makedb --in uniprot_trembl.fasta.gz  -d uniprot_trembl

##create ideal


mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/IDEEL
cd /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/IDEEL

git clone https://github.com/mw55309/ideel

cd ideel 
mkdir -p genomes
cd genomes
cp /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/final/K1_ONT_assembly_circlor.fasta K1_ONT_assembly_circlor.fa
cp /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/01_firstRacon_graphmap/assembly/firstRacon_NWC_2Assembly.fasta firstRacon_NWC_2Assembly.fa
cp /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/02_secondRacon_graphmap/assembly/secondRacon_NWC_2Assembly.fasta secondRacon_NWC_2Assembly.fa
cp /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/03_thirdRacon_graphmap/assembly/thirdRacon_NWC_2Assembly.fasta thirdRacon_NWC_2Assembly.fa
cp /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/04_fourthRacon_graphmap/assembly/fourthRacon_NWC_2Assembly.fasta fourthRacon_NWC_2Assembly.fa
cp /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/05_fifthFreebayes/freebayesOuput/_fifthFreebayes.fasta _fifthFreebayes.fa
cp /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/06_sixthFreebayes/freebayesOuput/_sixthFreebayes.fasta _sixthFreebayes.fa
cp /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/07_seventhFreebayes/freebayesOuput/_seventhFreebayes.fasta _seventhFreebayes.fa
cp /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/_eightFreebayes.fasta _eightFreebayes.fa
cd ..

snakemake

###________________________move local

scp -r vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/IDEEL/ideel/hists/ /home/vincent/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/

scp -r vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/IDEEL/ideel/lengths/ /home/vincent/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/


```

```{r}

library(readr)
library(ggplot2)

assembly_01 <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/lengths/K1_ONT_assembly_circlor.data",  "\t", escape_double = FALSE, col_names = c("length","expected"), trim_ws = TRUE)
assembly_02 <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/lengths/firstRacon_NWC_2Assembly.data",  "\t", escape_double = FALSE, col_names = c("length","expected"), trim_ws = TRUE)
assembly_03 <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/lengths/secondRacon_NWC_2Assembly.data",  "\t", escape_double = FALSE, col_names = c("length","expected"), trim_ws = TRUE)
assembly_04 <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/lengths/thirdRacon_NWC_2Assembly.data",  "\t", escape_double = FALSE, col_names = c("length","expected"), trim_ws = TRUE)
assembly_05 <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/lengths/fourthRacon_NWC_2Assembly.data",  "\t", escape_double = FALSE, col_names = c("length","expected"), trim_ws = TRUE)
assembly_06 <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/lengths/_fifthFreebayes.data",  "\t", escape_double = FALSE, col_names = c("length","expected"), trim_ws = TRUE)
assembly_07 <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/lengths/_sixthFreebayes.data",  "\t", escape_double = FALSE, col_names = c("length","expected"), trim_ws = TRUE)
assembly_08 <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/lengths/_seventhFreebayes.data",  "\t", escape_double = FALSE, col_names = c("length","expected"), trim_ws = TRUE)
assembly_09 <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/ideel/lengths/_eightFreebayes.data",  "\t", escape_double = FALSE, col_names = c("length","expected"), trim_ws = TRUE)


pseudoPlotData <- data.frame(pseudo=c(100-(sum((assembly_01$length/assembly_01$expected)>0.9)/nrow(assembly_01)*100),
100-(sum((assembly_02$length/assembly_02$expected)>0.9)/nrow(assembly_02)*100),
100-(sum((assembly_03$length/assembly_03$expected)>0.9)/nrow(assembly_03)*100),
100-(sum((assembly_04$length/assembly_04$expected)>0.9)/nrow(assembly_04)*100),
100-(sum((assembly_05$length/assembly_05$expected)>0.9)/nrow(assembly_05)*100),
100-(sum((assembly_06$length/assembly_06$expected)>0.9)/nrow(assembly_06)*100),
100-(sum((assembly_07$length/assembly_07$expected)>0.9)/nrow(assembly_07)*100),
100-(sum((assembly_08$length/assembly_08$expected)>0.9)/nrow(assembly_08)*100),
100-(sum((assembly_09$length/assembly_09$expected)>0.9)/nrow(assembly_09)*100)),names=c("FlyeAssembly","firstONTracon","secondONTracon","thirdONTracon","fourthFreebayes","fiftFreebayes","sixthFreebayes","seventhFreebayes","eigthFreeebayes"))

pseudoPlotData$nameSample_f = factor(pseudoPlotData$names, levels=c("FlyeAssembly","firstONTracon","secondONTracon","thirdONTracon","fourthFreebayes","fiftFreebayes","sixthFreebayes","seventhFreebayes","eigthFreeebayes"))


p5 <- ggplot(pseudoPlotData,aes(x=nameSample_f,y=pseudo,group=1))+geom_point(size=2)+geom_line()+
  theme_classic()+
  ylim(10,100)+
  scale_y_continuous(breaks = c(12,25,50,75,96),labels=c(12,25,50,75,96))+
  labs(x="",
       y="Pseudogenes [%]")+
  theme(axis.text.x = element_text(angle = 75, hjust = 1))
p5


  svg("~/Desktop/presenations/my_presentations/20190708_groupmeeting/figures/polishing_K1_pseudo.svg",width=6,height=3)
p5
 dev.off()


```



#### Prokka annotation of polished assembly

```{bash}

threads=37

rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/03_PROKKA_annotation_polishedGenome/ 
    mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/03_PROKKA_annotation_polishedGenome/ 
    
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/03_PROKKA_annotation_polishedGenome/ --force --usegenus --genus Meta --locustag K1_ONT --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/_eightFreebayes.fasta 
    


```


####Prophage detection (Virsorter)

For phage and prophage detection I use Virsorter from the (Sullivan lab)[http://u.osu.edu/viruslab/]

```{bash}


for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
do
#sample=1
threads=35
Overall_output_directory=/archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${sample}/


echo "========================================="
echo $sample
echo "========================================="



  rm -r /home/vincent/Projects/2019_pilotplant/07_Virsorter/${sample}
  mkdir -p /home/vincent/Projects/2019_pilotplant/07_Virsorter/${sample}
 
    source activate virsorter
    wrapper_phage_contigs_sorter_iPlant.pl \
    -f $Overall_output_directory/08_eightFreebayes/freebayesOuput/_eightFreebayes.fasta \
    --db 1 \
    --no_c \
    --wdir /home/vincent/Projects/2019_pilotplant/07_Virsorter/${sample} --ncpu ${threads} --data-dir /archiv/VirSorter/virsorter-data

grep -v "^#" /home/vincent/Projects/2019_pilotplant/07_Virsorter/${sample}/VIRSorter_global-phage-signal.csv 
 
 
 done #samples

##look at files


for sample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt)
do
echo "========================================="
echo $sample
echo "========================================="


cat /home/vincent/Projects/2019_pilotplant/07_Virsorter/${sample}/VIRSorter_global-phage-signal.csv 
 
 
 done #samples

##extract single fasta from multifasta (multifasta2fasta)

samtools faidx th_K1_8h_metagenome.fasta contig_6 > contig_6.fasta
```

####PHASTER
Here, I also used (PHASTER)[http://phaster.ca/] for prophage/phage detection.

```{bash}

##make Bed out of PHASTER output
mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/08_phaster/phasterBED


for name in $(echo "di_K1_10h" "di_K2_6h" "th_K1_8h" "th_K2_8h")
do

 echo "------------"
 echo ${name}
  echo "------------"

grep ">" /home/vincent/Desktop/Projects/2019_Pilotplan/08_phaster/${name}/phage_regions.fna  |awk -F "\t" '{print $2}'| sed 's/ //g' |awk -F[,:-] 'BEGIN{OFS="\t"}{print $1,$11,$12}' > /home/vincent/Desktop/Projects/2019_Pilotplan/08_phaster/phasterBED/${name}.bed

cat /home/vincent/Desktop/Projects/2019_Pilotplan/08_phaster/phasterBED/${name}.bed
done

##move to big machine

###--------------on local
name=di_K1_10h
scp /home/vincent/Desktop/Projects/2019_Pilotplan/08_phaster/phasterBED/${name}.bed vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${name}/

name=th_K1_8h
scp /home/vincent/Desktop/Projects/2019_Pilotplan/08_phaster/phasterBED/${name}.bed vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${name}/

name=di_K2_6h
scp /home/vincent/Desktop/Projects/2019_Pilotplan/08_phaster/phasterBED/${name}.bed vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${name}/

name=th_K2_8h
scp /home/vincent/Desktop/Projects/2019_Pilotplan/08_phaster/phasterBED/${name}.bed vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${name}/


```



#### extract from Bandage

I noticed that I did not use the polished genomes for the final assembly...therefore I am correcting it. 

```{bash}
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina


##================
##POLISHED ONT
##================

##Streptococcus_thermophilus_RMK202.fasta

grep "^>" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/contig_1.fasta

sed 's/>contig_1/>S_thermophilus_RMK202/g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/contig_1.fasta > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Streptococcus_thermophilus_RMK202.fasta

grep "^>" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Streptococcus_thermophilus_RMK202.fasta

##Lactobacillus_delbrueckii_RMK202.fasta

grep "^>" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/contig_2.fasta
grep ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Lactobacillus_delbrueckii_RMK202.fasta


sed 's/>contig_2/>L_delbrueckii_RMK202/g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/contig_2.fasta > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Lactobacillus_delbrueckii_RMK202.fasta

grep "^>" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Lactobacillus_delbrueckii_RMK202.fasta

##Lactobacillus_phage_RMK202_01.fasta

grep "^>" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/contig_3.fasta
grep ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Lactobacillus_phage_RMK202_01.fasta


sed 's/>contig_3/>L_phage_RMK202_01/g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/contig_3.fasta > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Lactobacillus_phage_RMK202_01.fasta

grep "^>" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Lactobacillus_phage_RMK202_01.fasta

ll /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/

##Bos_taurus_mito.fasta

grep "^>" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/plasmidSpades_CowMito.fasta
grep ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Bos_taurus_mito.fasta


sed 's/>plasmidSpades_CowMito/>Bos_taurus_mitochondrion/g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/plasmidSpades_CowMito.fasta > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Bos_taurus_mito.fasta

grep "^>" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Bos_taurus_mito.fasta

ll /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/


##Lactobacillus_delbrueckii_plasmid.fasta

grep "^>" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/plasmidSpades_ldelPlasmid.fasta
grep ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Lactobacillus_delbrueckii_plasmid.fasta


sed 's/>plasmidSpades_ldelPlasmid/>L_delbrueckii_plasmid_RMK202_01/g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/plasmidSpades_ldelPlasmid.fasta > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Lactobacillus_delbrueckii_plasmid.fasta

grep "^>" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Lactobacillus_delbrueckii_plasmid.fasta

ll /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/


##================
##remaining Illumina assemblies
##================

##Lactobacillus_plasmid.fasta


/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/

##Streptococcus_phage.fasta

##Streptococcus_thermophilus_potenial_prophage.fasta

## Lactobacillus_phage_RMK202_02.fasta

##================
##merge
##================

mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/merged

ls /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/
ls /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/



mkdir -p  /archiv/Projects/2019_RMK202_analysis/02_flye_Assembly/merged/
cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Streptococcus_thermophilus_RMK202.fasta \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Lactobacillus_delbrueckii_RMK202.fasta \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Bos_taurus_mito.fasta \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Lactobacillus_phage_RMK202_01.fasta \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Lactobacillus_delbrueckii_plasmid.fasta \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Lactobacillus_plasmid.fasta \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Lactobacillus_phage_RMK202_02.fasta \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Streptococcus_phage.fasta > \
  /archiv/Projects/2019_RMK202_analysis/02_flye_Assembly/merged/RMK202_merged.fasta
        

##with all genomes
  cat /archiv/Projects/2019_RMK202_analysis/02_flye_Assembly/merged/RMK202_merged.fasta \
    /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/S72.fna \
    /archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/S50.fna \
    /archiv/Projects/2018_Culturomics/genomes/L*.fna > \
    /archiv/Projects/2019_RMK202_analysis/02_flye_Assembly/merged/RMK202_merged_withStrains.fasta


```



####Final annoation

The final annoation will be done with:
- PGAP for Bacteria, Plasmids and the cow mito
- phanotate for the phages


```{bash}

mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_phage_RMK202_01/

cd /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_phage_RMK202_01

/home/vincent/apps/phanotate/PHANOTATE-master/phanotate.py /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Lactobacillus_phage_RMK202_01.fasta |grep "^#" -v | awk 'BEGIN{OFS="\t"}{print $4,$1,$2,".",".",$3}' | \
awk 'BEGIN{OFS="\t"} {
if ($6 =="+")
	print $1,$2,$3,$4,$5,$6;
else
	print $1,$3,$2,$4,$5,$6;
}' > \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_phage_RMK202_01/Lactobacillus_phage_RMK202_01.bed


##change if on negative strand switch $2 and $3



bedtools getfasta  -s -fi /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Lactobacillus_phage_RMK202_01.fasta \
  -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_phage_RMK202_01/Lactobacillus_phage_RMK202_01.bed > \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_phage_RMK202_01/Lactobacillus_phage_RMK202_01.fasta
  
  
  transeq -sequence /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_phage_RMK202_01/Lactobacillus_phage_RMK202_01.fasta \
  -outseq /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_phage_RMK202_01/Lactobacillus_phage_RMK202_01.faa


##move local
mkdir -p  /home/vincent/Desktop/Projects/2019_Pilotplan/09_annoation/phages
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_phage_RMK202_01/Lactobacillus_phage_RMK202_01.fasta /home/vincent/Desktop/Projects/2019_Pilotplan/09_annoation/phages





```

First we however do PROKKA for all the genomes indiviually.

```{bash}


###get assembly from local

mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta

scp -r /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/single_contigs vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/ 

##===============
##replace bacterial genomes with start aligned
##===============

##sterm
rm /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Streptococcus_thermophilus_RMK202.fasta
cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/final/contig_1_new.fasta > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Streptococcus_thermophilus_RMK202.fasta
sed -i 's/contig_1/S_thermophilus_RMK202/g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Streptococcus_thermophilus_RMK202.fasta
grep ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Streptococcus_thermophilus_RMK202.fasta

##Ldel
rm /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Lactobacillus_delbrueckii_RMK202.fasta
cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/circlaring/final/contig_2_new.fasta > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Lactobacillus_delbrueckii_RMK202.fasta
sed -i 's/contig_2/L_delbrueckii_RMK202/g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Lactobacillus_delbrueckii_RMK202.fasta
grep ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Lactobacillus_delbrueckii_RMK202.fasta


##===============
##shorten contigs name
##===============

for id in $(ls /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/)
do
sed -i 's/NODE_//g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/${id} 
sed -i 's/+.*//g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/${id} 
done
##===============
##annotaion
##===============


        

##-----------
###Bos_taurus_mito.fasta
##-----------
rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Bos_taurus_mito/
    mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Bos_taurus_mito/
    
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Bos_taurus_mito/ --force --usegenus --genus Bos --locustag bt_mito --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Bos_taurus_mito.fasta
    
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Bos_taurus_mito/*.faa /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/CDS/Bos_taurus_mito.faa


##-----------
###Lactobacillus_delbrueckii_plasmid.fasta
##-----------

#sed -i 's/Lactobacillus/L/g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Lactobacillus_delbrueckii_plasmid.fasta

rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_delbrueckii_plasmid/
    mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_delbrueckii_plasmid/
    
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_delbrueckii_plasmid/ --force --usegenus --genus Lactobacillus --locustag ld_plasmid --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Lactobacillus_delbrueckii_plasmid.fasta
    
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_delbrueckii_plasmid/*.faa /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/CDS/Lactobacillus_delbrueckii_plasmid.faa


    
##-----------
###Lactobacillus_delbrueckii_RMK202.fasta
##-----------

#sed -i 's/Lactobacillus/L/g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Lactobacillus_delbrueckii_RMK202.fasta

rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_delbrueckii_RMK202/
    mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_delbrueckii_RMK202/
    
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_delbrueckii_RMK202/ --force --usegenus --genus Lactobacillus --locustag ld_rmk202 --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Lactobacillus_delbrueckii_RMK202.fasta
  
mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/{CDS,eggnog}  

scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_delbrueckii_RMK202/*.faa /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/CDS/Lactobacillus_delbrueckii_RMK202.faa


##-----------
###Streptococcus_thermophilus_RMK202.fasta
##-----------


rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Streptococcus_thermophilus_RMK202/
    mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Streptococcus_thermophilus_RMK202/
    
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Streptococcus_thermophilus_RMK202/ --force --usegenus --genus Streptococcus --locustag st_rmk202 --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Streptococcus_thermophilus_RMK202.fasta
    
    
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Streptococcus_thermophilus_RMK202/*.faa /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/CDS/Streptococcus_thermophilus_RMK202.faa


##-----------
###Lactobacillus_plasmid.fasta
##-----------
sed -i 's/Lactobacillus/L/g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Lactobacillus_plasmid.fasta

rm -r /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_plasmid/
    mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_plasmid/
    
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_plasmid/ --force --usegenus --genus Lactobacillus --locustag ld_plasmid --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Lactobacillus_plasmid.fasta
    

    
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_plasmid/*.faa /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/CDS/Lactobacillus_plasmid.faa


##==========0
##shorten Names
##==========
grep "^>" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Lactobacillus_phage_RMK202_01.fasta
sed -i 's/Lactobacillus/L/g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Lactobacillus_phage_RMK202_01.fasta
grep "^>" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Lactobacillus_phage_RMK202_01.fasta

grep "^>" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Streptococcus_phage.fasta
sed -i 's/Streptococcus/S/g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Streptococcus_phage.fasta
grep "^>" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Streptococcus_phage.fasta

grep "^>" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Streptococcus_thermophilus_potenial_prophage.fasta
sed -i 's/Streptococcus_thermohilus/S_ther/g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Streptococcus_thermophilus_potenial_prophage.fasta
grep "^>" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Streptococcus_thermophilus_potenial_prophage.fasta

grep "^>" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Lactobacillus_phage_RMK202_02.fasta
sed -i 's/Lactobacillus/L/g' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Lactobacillus_phage_RMK202_02.fasta
grep "^>" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Lactobacillus_phage_RMK202_02.fasta
 

 
##=================
##merge Ldel
##=================

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Lactobacillus_delbrueckii_RMK202.fasta  \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Lactobacillus_delbrueckii_plasmid.fasta \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/fasta/single_contigs/Lactobacillus_plasmid.fasta      > \
/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/merged/Ldel_RMK202.fasta

###-----------scp
rm -r/home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/single_contigs_final

mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/single_contigs_final

scp -r vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/merged/Ldel_RMK202.fasta /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/single_contigs_final

scp -r vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Streptococcus_thermophilus_RMK202.fasta /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/single_contigs_final


```

looking into Prokka annoation

```{bash}

ll /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/

##Lactobacillus_delbrueckii_RMK202
gff_location=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_delbrueckii_RMK202/*gff

grep "dnaa" -i ${gff_location}
grep "pseudogene" -i ${gff_location}
grep "transposase" -i ${gff_location}
grep "16S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "23S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "5S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep " ribosomal RNA" -i ${gff_location} 


##Lactobacillus_delbrueckii_RMK202
gff_location=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Streptococcus_thermophilus_RMK202/*gff

grep "dnaa" -i ${gff_location}
grep "pseudogene" -i ${gff_location}
grep "transposase" -i ${gff_location}
grep "16S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "23S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "5S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep " ribosomal RNA" -i ${gff_location} 


##S72
gff_location=/archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/*gff

grep "dnaa" -i ${gff_location}
grep "pseudogene" -i ${gff_location}
grep "transposase" -i ${gff_location}
grep "16S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "23S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "5S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep " ribosomal RNA" -i ${gff_location} 


##S50
gff_location=/archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/*gff

grep "dnaa" -i ${gff_location}
grep "pseudogene" -i ${gff_location}
grep "transposon" -i ${gff_location}
grep "16S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "23S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "5S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep " ribosomal RNA" -i ${gff_location} 



#L104
gff_location=/archiv/Projects/2018_Culturomics/06_Spades/L104/Prokka/20190203/*gff

grep "dnaa" -i ${gff_location}
grep "pseudogene" -i ${gff_location}
grep "transposon" -i ${gff_location}
grep "16S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "23S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep "5S ribosomal RNA" -i ${gff_location} | awk '{print $5-$4}'
grep " ribosomal RNA" -i ${gff_location} 


```



####eggnog annoation

All assemblies were annotated on the [eggnog homepage](http://eggnog-mapper.embl.de/)
After the annoation the files are downloaded locally and analysed in R.

genes of interest

Transporter: GO:0140104 (molecular carrier activity), GO:0005215 (transporter activity)

The different annotation options:
1. [KEGG](https://www.genome.jp/kegg/kegg1a.html) has many different annotations schems:
  - Probably the most interesting one is [KO (KEGG ORTHOLOGY) Database](https://www.genome.jp/kegg/ko.html).
  - [KEGG modules](https://www.genome.jp/kegg/module.html) less grained the Brite
  - Brite
  
  
Do the mapping of the annotations [here](https://www.genome.jp/kegg/tool/map_pathway.html)
  


```{bash}

scp /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/*_pgap* vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/eggnog/

##=================
##create mappable files
##=================

##Ldel  
awk -F "\t" 'BEGIN{OFS="\t"} {print $1, $9}' /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Lactobacillus_delbrueckii_RMK202_pgap_query_seqs.fa.emapper.annotations |sed 's/ko://g' |grep "^#" -v > /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog_Kegg/Lactobacillus_delbrueckii_RMK202_pgap_query_seqs.fa.emapper.annotations

grep -v "^#" /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Lactobacillus_delbrueckii_RMK202_pgap_query_seqs.fa.emapper.annotations | awk -F "\t" 'BEGIN{OFS="\t"} {print $9}'  |sed 's/ko://g' > /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog_Kegg/Lactobacillus_delbrueckii_RMK202_pgap_query_seqs.fa.emapper.annotations

```


I noticed that i first have to seperate the Brite columns with multiple annoations in order to get their annoation

```{r}
library(thacklr)
library(readr)
library(dplyr)
library(tidyverse)
library(tidyr)

S50 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/S50_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3)) %>% add_column(.,sample="S50")

Lactobacillus_delbrueckii_RMK202 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Lactobacillus_delbrueckii_RMK202_pgap_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3)) %>% add_column(.,sample="Lactobacillus_delbrueckii_RMK202")
colnames(S50) <- c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description","genome")


colnames(Lactobacillus_delbrueckii_RMK202) <- c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description","genome")


## The ";\\s+" means that the separator is a ";" followed by one or more spaces


Lactobacillus_delbrueckii_RMK202_subset4Brite <- Lactobacillus_delbrueckii_RMK202[,c("query_name","KEGG_ko")]
# Lactobacillus_delbrueckii_RMK202_subset4Brite$KEGG_ko <- gsub("ko:","",Lactobacillus_delbrueckii_RMK202_subset4Brite$KEGG_ko)
Lactobacillus_delbrueckii_RMK202_subset4Brite <- separate_rows(Lactobacillus_delbrueckii_RMK202_subset4Brite,KEGG_ko,sep=",")

write.table(Lactobacillus_delbrueckii_RMK202_subset4Brite, "~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog_Kegg/Lactobacillus_delbrueckii_RMK202_subset4Brite_pgap_query_seqs.fa.emapper.annotations",na = "", quote = FALSE, sep = "\t",row.names = FALSE, col.names = FALSE)

```

####Rast annotation of phages


####KEGG
When mapping to KEGG homepage I have to manually take the information into and editor and change the columns

[KeggRest](http://bioconductor.org/packages/release/bioc/html/KEGGREST.html)
maybe usfule [link](https://rdrr.io/bioc/keggorthology/src/inst/keggHTML/parseKeg.R)

Potentially interesting R packages:
KEGGgraph
pathView
BioCor

```{r}
library(jsonlite)
library(rjson)
library(dplyr)
library(tibble) 

winners <- rjson::fromJSON(file="~/Downloads/ko00001.json")
# glimpse(winners, max.level = 3, list.len = 4)
# glimpse(winners, max.level = 4, list.len = 4)
# glimpse(winners, max.level = 5, list.len = 4)
# glimpse(winners, max.level = 6, list.len = 4)
# glimpse(winners, max.level = 7, list.len = 4)
# glimpse(winners, max.level = 8, list.len = 4)
# glimpse(winners, max.level = 9, list.len = 4)
# glimpse(winners, max.level = 10, list.len = 4)

new <- glimpse(winners, max.level = 9, list.len = 4)
new$name

new_new <- new$children[[1]]
new_new$name


new
new[[2]][[1]][[1]]

data_raw <- enframe(unlist(winners))

# winners <- jsonlite::fromJSON(file="~/Downloads/ko00001.json")
# jsonlite::fromJSON(winners)
# head(winners)

 # # cat ko00001.json |sed 's/"name"://g' |sed 's/://g'  |sed 's/,//g'|sed 's/"children"//g' > test.json
 #  cat ko00001.json |sed 's/"name"://g'  |sed 's/",//g'|sed 's/"children"://g' |sed 's/[[:blank:]]://g' > test.json
 #  # cat ko00001.json |sed 's/"name"://g' |sed 's/"children"://g' |sed 's/[[:blank:]]://g' > test.json
 #  cat ko00001.json |sed 's/"name"://g'  |sed 's/",/"/g'|sed 's/"children"//g' |sed 's/[[:blank:]]:/:/g' |sed 's/\t://g' > test.json
 #  cat ko00001.json |sed 's/"name"://g'  |sed 's/",/"/g'|sed 's/"children"//g' |sed -r 's/\t+:\[/:\[/g'  > test.json
 #  cat ko00001.json |sed 's/"name"://g'  |sed 's/",/"/g'|sed 's/"children"//g' |sed -r 's/\t+:\[/:\[/g'|sed -z 's/\n:\[/:\[/g'  > test.json

# cat ko00001.json | sed -r 's/^\t"name":/\t"class":/g' | sed -r 's/^\t\t"name":/\t\t"group":/g'| sed -r 's/^\t\t\t"name":/\t\t\t"species":/g' | sed -r 's/^\t\t\t\t"name":/\t\t\t\t"pathway":/g' | sed -r 's/^\t\t\t\t\t"name":/\t\t\t\t\t"gene":/g'   > test.json

# cat test.json | sed -r 's/^\t"children":/\t"class":/g' | sed -r 's/^\t\t"children":/\t\t"group":/g'| sed -r 's/^\t\t\t"children":/\t\t\t"species":/g' | sed -r 's/^\t\t\t\t"children":/\t\t\t\t"pathway":/g' | sed -r 's/^\t\t\t\t\t"children":/\t\t\t\t\t"gene":/g'  > test_02.json

winners <- rjson::fromJSON(file="~/Downloads/test_02.json")
data_raw <- enframe(unlist(winners))
data_raw

data_raw %>% separate(name, into = c(paste0("x", 1:5)))




```


####Go-Annotation

good [intro](https://www.bioconductor.org/packages/release/bioc/vignettes/annotate/inst/doc/GOusage.pdf)
#####topgo

interesting [question](https://www.biostars.org/p/389438/) about read mapping
this might be a [solution](http://avrilomics.blogspot.com/2015/07/using-topgo-to-test-for-go-term.html)
--> first bring data into right format

```{bash}

mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog_GO
grep -v "^#" /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Lactobacillus_delbrueckii_RMK202_pgap_query_seqs.fa.emapper.annotations | awk -F "\t" 'BEGIN{OFS="\t"} {print $1, $7}' |sed 's/gnl|extdb|//g' > /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog_GO/Lactobacillus_delbrueckii_RMK202_pgap_query_seqs.fa.emapper.annotations


grep -v "^#" /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Streptococcus_thermophilus_RMK202_pgap_query_seqs.fa.emapper.annotations | awk -F "\t" 'BEGIN{OFS="\t"} {print $1, $7}' |sed 's/gnl|extdb|//g' > /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog_GO/Streptococcus_thermophilus_RMK202_pgap_query_seqs.fa.emapper.annotations
```

...then go into R

```{r}
library(readr)
library(Rgraphviz)
library(topGO)
geneID2GO <- readMappings(file = "~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog_GO/Lactobacillus_delbrueckii_RMK202_pgap_query_seqs.fa.emapper.annotations") 
geneID2GO <- readMappings(file = "~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog_GO/Streptococcus_thermophilus_RMK202_pgap_query_seqs.fa.emapper.annotations") 


geneUniverse <- names(geneID2GO) 

# genesOfInterest <- read.table("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/Eggnog/variantCallsfreebayes_all_Lactobacillus_delbrueckii_RMK202_highImpactGeneIds_forTopGO.txt",header=FALSE)
# genesOfInterest <- read.table("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/Eggnog/variantCallsfreebayes_all_Streptococcus_thermophilus_RMK202_highImpactGeneIds_forTopGO.txt",header=FALSE)
# genesOfInterest <- read.table("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/Eggnog/variantCallsfreebayes_all_Streptococcus_thermophilus_RMK202_MODERATEImpactGeneIds_forTopGO.txt",header=FALSE)
genesOfInterest <- read.table("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/Eggnog/variantCallsfreebayes_all_Lactobacillus_delbrueckii_RMK202_MODERATEImpactGeneIds_forTopGO.txt",header=FALSE)
# genesOfInterest <- read.table("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/Eggnog/variantCallsfreebayes_all_Streptococcus_thermophilus_RMK202_highImpactGeneIds_forTopGO.txt",header=FALSE)

genesOfInterest <- as.character(genesOfInterest$V1)

table(genesOfInterest)

genesOfInterest <- genesOfInterest[genesOfInterest %in% geneUniverse]

 geneList <- factor(as.integer(geneUniverse %in% genesOfInterest))
 names(geneList) <- geneUniverse
 
 
 myGOdata <- new("topGOdata", description="My project", ontology="BP", allGenes=geneList,  annot = annFUN.gene2GO, gene2GO = geneID2GO)
 
 sg <- sigGenes(myGOdata)
 str(sg)
numSigGenes(myGOdata)
resultFisher <- runTest(myGOdata, algorithm="classic", statistic="fisher")
resultFisher

allRes <- GenTable(myGOdata, classicFisher = resultFisher, orderBy = "resultFisher", ranksOf = "classicFisher", topNodes = 200)
allRes

 showSigOfNodes(myGOdata, score(resultFisher), firstSigNodes = 5, useInfo ='all')
 printGraph(myGOdata, resultFisher, firstSigNodes = 5, fn.prefix = "tGO", useInfo = "all", pdfSW = TRUE)
 
 
 length(usedGO(myGOdata))
 
 # myterms = c("GO:0006207", "GO:0019856", "GO:0046112")
 # mygenes <- genesInTerm(myGOdata, myterms)
 #  for (i in 1:length(myterms))
 #   {
 #       myterm <- myterms[i]
 #       mygenesforterm <- mygenes[myterm][[1]]
 #       mygenesforterm <- paste(mygenesforterm, collapse=',')
 #       print(paste("Term",myterm,"genes:",mygenesforterm))
 #     }
 
 
```


```{r}

library(thacklr)
library(readr)
library(dplyr)
library(tidyverse)

Bos_taurus_mito <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Bos_taurus_mito_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3))

L104 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L104_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3))
L108 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L108_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3))
L35 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L35_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3))
L39 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L39_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3))
L44 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L44_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3))
L70 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L70_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3))
L71 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L71_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3))
L80 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L80_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3))
L99 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L99_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3))


Lactobacillus_delbrueckii_plasmid <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Lactobacillus_delbrueckii_plasmid_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3))


S50 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/S50_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3)) %>% add_column(.,sample="S50")
S72 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/S72_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3)) %>% add_column(.,sample="S72")
Streptococcus_thermophilus_RMK202 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Streptococcus_thermophilus_RMK202_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE) %>% slice( 1:(n()-3)) %>% add_column(.,sample="Streptococcus_thermophilus_RMK202")


ALL_sterm <- rbind(S50,S72,Streptococcus_thermophilus_RMK202)
colnames(ALL_sterm) <- c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description","genome")


S50[grep("GO:0005215",S50$GOs),]
nrow(S50[grep("GO:0005215",S50$GOs),])

##----split multi-column into multiple rows

library(tidyr)
## The ";\\s+" means that the separator is a ";" followed by one or more spaces
separate_rows(df,abilities,sep=";\\s+")

```

####COG analysis

```{r}
library(tidyverse)
library(thacklr)
library(readr)
library(dplyr)
library(ggplot2)

COG_functional_categories <- read_delim("~/Desktop/ScriptRepository/COG_functional_categories.csv",  ",", escape_double = FALSE, col_names = c("short","long","divided"),trim_ws = TRUE, na = "NA")

S50 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/S50_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE, na = "NA") %>% slice( 1:(n()-3)) %>% add_column(.,sample="S50")
S72 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/S72_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE, na = "NA") %>% slice( 1:(n()-3)) %>% add_column(.,sample="S72")
Streptococcus_thermophilus_RMK202 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Streptococcus_thermophilus_RMK202_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE, na = "NA") %>% slice( 1:(n()-3)) %>% add_column(.,sample="Streptococcus_thermophilus_RMK202")


ALL_sterm <- rbind(S50,S72,Streptococcus_thermophilus_RMK202)
colnames(ALL_sterm) <- c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description","genome")

nrow(ALL_sterm)

#unique(sterm_all_unique$COG_Functional_Category)[match(COG_functional_categories$short , unique(sterm_all_unique$COG_Functional_Category))]
unique(ALL_sterm$COG_Functional_Category)[which(!unique(ALL_sterm$COG_Functional_Category) %in% COG_functional_categories$short )]

# table(ALL_sterm$COG_Functional_Category)

ALL_sterm$longCOG <- NA
for (i in 1:nrow(ALL_sterm)) {
  ALL_sterm[i,"longCOG"] <- COG_functional_categories[which(COG_functional_categories$short==as.character(ALL_sterm[i,"COG_Functional_Category"])),"long"]
  
}



CogPlotSterm <- ggplot(ALL_sterm,aes(x=longCOG,fill=genome))+geom_bar(position = "dodge2")+
  theme_classic()+
  labs(x="")+
  theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9),
          # # axis.text.y.left = element_text(color = "blue"),
          # axis.text.y.right = element_text(color="red"),
          # axis.title.y.left = element_text(color="blue"),
          # axis.title.y.right = element_text(color="red"),
          # rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          #legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank(),
          plot.margin=unit(c(0,0,0,2), "cm"),
          axis.text.x.bottom = element_text(size=10)
          #legend.position = c(0.02, 0.98),
          #legend.justification = c("left", "top")
          )
CogPlotSterm
   svg("~/Desktop/presenations/my_presentations/20190807_committeeMeeting/figures/cogPlot_allSterm.svg",width=7,height=7)
CogPlotSterm 
 dev.off()
 
##--------------
 ##Ldel
##--------------
COG_functional_categories <- read_delim("~/Desktop/ScriptRepository/COG_functional_categories.csv",  ",", escape_double = FALSE, col_names = c("short","long","divided"),trim_ws = TRUE, na = "NA")
 

L104 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L104_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE, na = "NA") %>% slice( 1:(n()-3)) %>% add_column(.,sample="L104") 
L108 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L108_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE, na = "NA") %>% slice( 1:(n()-3)) %>% add_column(.,sample="L108")
L35 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L35_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE, na = "NA") %>% slice( 1:(n()-3)) %>% add_column(.,sample="L35")
L39 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L39_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE, na = "NA") %>% slice( 1:(n()-3)) %>% add_column(.,sample="L39")
L44 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L44_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE, na = "NA") %>% slice( 1:(n()-3)) %>% add_column(.,sample="L44")
L70 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L70_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE, na = "NA") %>% slice( 1:(n()-3)) %>% add_column(.,sample="L70")
L71 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L71_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE, na = "NA") %>% slice( 1:(n()-3)) %>% add_column(.,sample="L71")
L80 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L80_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE, na = "NA") %>% slice( 1:(n()-3)) %>% add_column(.,sample="L80")
L99 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/L99_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE, na = "NA") %>% slice( 1:(n()-3)) %>% add_column(.,sample="L99")

Lactobacillus_delbrueckii_RMK202 <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Lactobacillus_delbrueckii_RMK202_query_seqs.fa.emapper.annotations", "\t", escape_double = FALSE, trim_ws = TRUE, skip = 4,col_names = FALSE, na = "NA") %>% slice( 1:(n()-3)) %>% add_column(.,sample="Lactobacillus_delbrueckii_RMK202")


ALL_sterm <- rbind(L104,L108,L35,L39,L44,L70,L71,L80,L99,Lactobacillus_delbrueckii_RMK202)
colnames(ALL_sterm) <- c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description","genome")

nrow(ALL_sterm)

#unique(sterm_all_unique$COG_Functional_Category)[match(COG_functional_categories$short , unique(sterm_all_unique$COG_Functional_Category))]
unique(ALL_sterm$COG_Functional_Category)[which(!unique(ALL_sterm$COG_Functional_Category) %in% COG_functional_categories$short )]

# table(ALL_sterm$COG_Functional_Category)

ALL_sterm$longCOG <- NA
for (i in 1:nrow(ALL_sterm)) {
  ALL_sterm[i,"longCOG"] <- COG_functional_categories[which(COG_functional_categories$short==as.character(ALL_sterm[i,"COG_Functional_Category"])),"long"]
  
}



CogPlotSterm <- ggplot(ALL_sterm,aes(x=longCOG,fill=genome))+geom_bar(position = "dodge2")+
  theme_classic()+
  labs(x="")+
  theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9),
          # # axis.text.y.left = element_text(color = "blue"),
          # axis.text.y.right = element_text(color="red"),
          # axis.title.y.left = element_text(color="blue"),
          # axis.title.y.right = element_text(color="red"),
          # rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          #legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank(),
          plot.margin=unit(c(0,0,0,2), "cm"),
          axis.text.x.bottom = element_text(size=10)
          #legend.position = c(0.02, 0.98),
          #legend.justification = c("left", "top")
          )
CogPlotSterm
   svg("~/Desktop/presenations/my_presentations/20190807_committeeMeeting/figures/cogPlot_allLdel.svg",width=7,height=7)
CogPlotSterm 
 dev.off()
 

```




#### ANI between the strains

```{bash FASTANI}
date=20190123
source /archiv/Projects/2018_Culturomics/02_script/fileShortcuts.sh
threads=35
id=RMK202
/archiv/Projects/2018_Culturomics

###-------------------
##Lactobacillus delbrueckii
###-------------------

##make list of all files
species=Lactobacillus_delbrueckii

mainPath=/archiv/Projects/2018_Culturomics
rm /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_GenomePath.txt
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/
for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/names_changed.txt | grep "^L")
do
  cp ${mainPath}/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/contigs.fasta ${mainPath}/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/${id}.fasta
  echo ${mainPath}/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/${id}.fasta >> /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_GenomePath.txt

done
##with MAG


echo "/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Lactobacillus_delbrueckii_RMK202.fasta" >> /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_GenomePath.txt



fastANI -t 5 --ql /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_GenomePath.txt \
  --rl /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_GenomePath.txt \
  -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_fastani.txt

sed 's/\/archiv\/Projects\/2019_Nano_Meta\/02_flye_Assembly\/02_raw_undeplexed_plasmids\/1AB\/Polishing_FINAL\/08_eightFreebayes\/freebayesOuput\/splitFasta\/finalWithIllumina\///g'  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_fastani.txt |sed 's/\/archiv\/Projects\/2018_Culturomics\/06_Spades\/L[0-9]\+\/assembly\/assemblyMetaSpades_20190121.fasta\///g' > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_fastani_forR.txt


scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/Lactobacillus_delbrueckii/Lactobacillus_delbrueckii_fastani_forR.txt ~/Desktop/Projects/2018_culturomics/fastani/


###-------------------
##Streptococcus thermophilus
###-------------------

##make list of all files
species=Streptococcus_thermophilus

mainPath=/archiv/Projects/2018_Culturomics
rm /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_GenomePath.txt
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/

for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/names_changed.txt | grep "^S")
do

  cp ${mainPath}/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/contigs.fasta ${mainPath}/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/${id}.fasta
  echo ${mainPath}/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/${id}.fasta >> /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_GenomePath.txt

done

##with christians MAG


echo "/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Streptococcus_thermophilus_RMK202.fasta" >> /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_GenomePath.txt


##FASTANI

fastANI -t 5 --ql /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_GenomePath.txt \
  --rl /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_GenomePath.txt \
  -o /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_fastani.txt
  

sed 's/\/archiv\/Projects\/2019_Nano_Meta\/02_flye_Assembly\/02_raw_undeplexed_plasmids\/1AB\/Polishing_FINAL\/08_eightFreebayes\/freebayesOuput\/splitFasta\/finalWithIllumina\///g'  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_fastani.txt |sed 's/\/archiv\/Projects\/2018_Culturomics\/06_Spades\/S[0-9]\+\/assembly\/assemblyMetaSpades_20190121.fasta\///g' > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/${species}/${species}_fastani_forR.txt

scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/fastANI/Streptococcus_thermophilus/Streptococcus_thermophilus_fastani_forR.txt ~/Desktop/Projects/2018_culturomics/fastani/


```
make [correlation heatmap](http://www.sthda.com/english/wiki/ggplot2-quick-correlation-matrix-heatmap-r-software-and-data-visualization)
```{r fastaniInR}
library(stringr)
library(readr)
library(ggplot2)
library(reshape2)


##----------------
##sterm
##----------------


Sterm_fastANI_nwc <- read_delim("~/Desktop/Projects/2018_culturomics/fastani/Streptococcus_thermophilus_fastani_forR.txt", "\t", escape_double = FALSE, col_names = FALSE,  trim_ws = TRUE) 
Sterm_fastANI_nwc$AF <- (100*(Sterm_fastANI_nwc$X4/Sterm_fastANI_nwc$X5))


##shorten names by removing path to genomes
Sterm_fastANI_nwc$new_X1 <- gsub("/archiv/Projects/2018_Culturomics/06_Spades/assembly/","",Sterm_fastANI_nwc$X1) 
Sterm_fastANI_nwc$new_X2 <- gsub("/archiv/Projects/2018_Culturomics/06_Spades/assembly/","",Sterm_fastANI_nwc$X2) 


##shorten names by ".fasta"
Sterm_fastANI_nwc$new_X1 <- gsub(".fasta","",Sterm_fastANI_nwc$X1) 
Sterm_fastANI_nwc$new_X2 <- gsub(".fasta","",Sterm_fastANI_nwc$X2) 

##rename genome names
library(plyr)
Sterm_fastANI_nwc$new_X1 <- revalue(Sterm_fastANI_nwc$new_X1, c("S50"="mst1", "S72"="mst2", "Streptococcus_thermophilus_RMK202"="RMK202"))
Sterm_fastANI_nwc$new_X2 <- revalue(Sterm_fastANI_nwc$new_X2, c("S50"="mst1", "S72"="mst2", "Streptococcus_thermophilus_RMK202"="RMK202"))



##remove upper triangle and keep only lower ANI scores
names <- levels(as.factor(Sterm_fastANI_nwc$new_X2))


##creat ANI matrix 
count <- 1
Sterm_fastANI_nwc$combined <- paste(Sterm_fastANI_nwc$new_X1,Sterm_fastANI_nwc$new_X2,sep="_")
sterm_curated <- Sterm_fastANI_nwc[1,]

for (i in 1:length(names)) {
   rowName <- names[i]
    print(rowName)
  for(a in 1:count){
    
    colNames <- names[a]
    print(colNames)
    search_01 <- paste(rowName,colNames,sep="_")
    search_02 <- paste(colNames,rowName,sep="_")
    
     ifelse(
      Sterm_fastANI_nwc[which(Sterm_fastANI_nwc$combined==search_01),"X3"]>=Sterm_fastANI_nwc[which(Sterm_fastANI_nwc$combined==search_02),"X3"],
      temp <- Sterm_fastANI_nwc[which(Sterm_fastANI_nwc$combined==search_02),],temp <-Sterm_fastANI_nwc[which(Sterm_fastANI_nwc$combined==search_01),]
        
    )
    print(temp)
    temp$new_X2 <- colNames
    temp$new_X1 <- rowName

    sterm_curated <- rbind(sterm_curated,temp)
    
    
  }
        
count <- count +1 
  
}
sterm_curated[-1,]



# creat Heatmap
sterm_curated$X3 <- round(sterm_curated$X3,digits = 2)

ggheatmap <- ggplot(sterm_curated, aes(new_X1, new_X2, fill = X3))+
 geom_tile(color = "white",size=1.1)+
 scale_fill_gradient2(low = "blue", high = "red", mid = "white", 
   midpoint = 99.5, limit = c(98.9,100), space = "Lab", 
   name="ANI") +
  theme_minimal()+ # minimal theme
 theme(axis.text.y = element_text(size = 12),
      axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1))+
 coord_fixed()
# Print the heatmap
print(ggheatmap)

plotFinal_sterm <- ggheatmap + 
geom_text(aes(new_X1, new_X2, label = X3), color = "black", size = 4) +
  scale_y_discrete(position = 'right')+
theme(
  axis.title.x = element_blank(),
  axis.title.y = element_blank(),
  panel.grid.major = element_blank(),
  panel.border = element_blank(),
  panel.background = element_blank(),
  axis.ticks = element_blank(),
  legend.justification = c(1, 0),
  legend.position = c(0.4, 0.7),
  legend.text = element_text(angle = 325),
  legend.direction = "horizontal")+
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1,
                title.position = "top", title.hjust = 0.5))

##save heatmap
svg("~/Desktop/Projects/2018_culturomics/fastani/Streptococcus_thermophilus_fastANI_with_MAG.svg",width=2.5,height=2.5)
plotFinal_sterm+theme(legend.position = "none")
dev.off()
# 
# png("~/Desktop/Projects/2018_culturomics/fastani/Streptococcus_thermophilus_fastANI_with_nwcS.png",width=400,height=400)
# plotFinal_sterm
# dev.off()

plotFinal_sterm

##----------------
##ldel
##----------------


  Ldel_fastANI_nwc <- read_delim("~/Desktop/Projects/2018_culturomics/fastani/Lactobacillus_delbrueckii_fastani_forR.txt", "\t", escape_double = FALSE, col_names = FALSE,  trim_ws = TRUE)

Ldel_fastANI_nwc$AF <- (100*(Ldel_fastANI_nwc$X4/Ldel_fastANI_nwc$X5))
##change names
Ldel_fastANI_nwc$new_X1 <- gsub(".fasta","",Ldel_fastANI_nwc$X1) 
Ldel_fastANI_nwc$new_X2 <- gsub(".fasta","",Ldel_fastANI_nwc$X2) 

##remove L39

Ldel_fastANI_nwc <- Ldel_fastANI_nwc[Ldel_fastANI_nwc$new_X1!="L39",]
Ldel_fastANI_nwc <- Ldel_fastANI_nwc[Ldel_fastANI_nwc$new_X2!="L39",]
##rename

Ldel_fastANI_nwc$new_X1 <- revalue(Ldel_fastANI_nwc$new_X1, c("L70"="mst3", "L44"="mst10", "L108"="mst7", "L99"="mst8", "L104"="mst6", "L80"="mst5", "L35"="mst9", "L71"="mst4", "L104"="mst6","Lactobacillus_delbrueckii_RMK202"="RMK202"))
Ldel_fastANI_nwc$new_X2 <- revalue(Ldel_fastANI_nwc$new_X2, c("L70"="mst3", "L44"="mst10", "L108"="mst7", "L99"="mst8", "L104"="mst6", "L80"="mst5", "L35"="mst9", "L71"="mst4", "L104"="mst6","Lactobacillus_delbrueckii_RMK202"="RMK202"))

# not_explained_sterm_plot$sample_renamed = factor(not_explained_sterm_plot$sample_renamed, levels=c("Lyoampulle\n2012","Lyoampulle\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018"))

##remove upper triangle and keep only lower ANI score
names <- levels(as.factor(Ldel_fastANI_nwc$new_X2))

count <- 1

Ldel_fastANI_nwc$combined <- paste(Ldel_fastANI_nwc$new_X1,Ldel_fastANI_nwc$new_X2,sep="_")
Ldel_curated <- Ldel_fastANI_nwc[1,]

for (i in 1:length(names)) {
   rowName <- names[i]
    print(rowName)
  for(a in 1:count){
    
    colNames <- names[a]
    print(colNames)
    search_01 <- paste(rowName,colNames,sep="_")
    search_02 <- paste(colNames,rowName,sep="_")
    
     ifelse(
      Ldel_fastANI_nwc[which(Ldel_fastANI_nwc$combined==search_01),"X3"]>=Ldel_fastANI_nwc[which(Ldel_fastANI_nwc$combined==search_02),"X3"],
      temp <- Ldel_fastANI_nwc[which(Ldel_fastANI_nwc$combined==search_02),],temp <-Ldel_fastANI_nwc[which(Ldel_fastANI_nwc$combined==search_01),]
        
    )
    #print(temp)
    temp$new_X1 <- colNames
    temp$new_X2 <- rowName
    
    Ldel_curated <- rbind(Ldel_curated,temp)
    
    
  }
        
count <- count +1 
  
}
Ldel_curated <- Ldel_curated[-1,]



# new <- Ldel_curated[
#   with(Ldel_curated, order(match(new_X1,rev(names)))),
# ]





# Heatmap
Ldel_curated$X3 <- round(Ldel_curated$X3,digits = 2)

ggheatmap <- ggplot(Ldel_curated,aes(x=new_X1, y=new_X2, fill = X3))+
 geom_tile(color = "white",size=1.1)+
 scale_fill_gradient2(low = "blue", high = "red", mid = "white", 
   midpoint = 99.5, limit = c(98.9,100), space = "Lab", 
   name="ANI") +
  theme_minimal()+ # minimal theme
 theme(axis.text.x = element_text(angle = 45, vjust = 1,size = 12, hjust = 0,),
  axis.text.y = element_text(size = 12))+
 coord_fixed()
# Print the heatmap
print(ggheatmap)


plotFinal_ldel <- ggheatmap + 
geom_text(aes(new_X1, new_X2, label = X3), color = "black", size = 3) +
  scale_x_discrete(position = 'top')+
theme(
  axis.title.x = element_blank(),
  axis.title.y = element_blank(),
  panel.grid.major = element_blank(),
  panel.border = element_blank(),
  panel.background = element_blank(),
  axis.ticks = element_blank(),
  legend.justification = c(1, 0),
  legend.position = c(1, 0.1),
  legend.text = element_text(angle = 325),
  legend.direction = "horizontal")+
  guides(fill = guide_colorbar(barwidth = 7, barheight = 1,
                title.position = "top", title.hjust = 0.5))


  svg("~/Desktop/Projects/2018_culturomics/fastani/Lactobacillus_delbrueckii_fastANI_with_MAG.svg",width=5,height=5)
plotFinal_ldel+theme(legend.position = "none")
dev.off()
# 
# png("~/Desktop/Projects/2018_culturomics/fastani/Lactobacillus_delbrueckii_fastANI_with_nwcS.png",width=400,height=400)
# plotFinal_ldel
# dev.off()



  svg("~/Desktop/Projects/2018_culturomics/fastani/Lactobacillus_delbrueckii_fastANI_with_MAG_legend.svg",width=5.8,height=5.8)
plotFinal_ldel
dev.off()

```


#### Snippy between isolates

Here, I test the SNP calling between different assembled genomes on the core genome. 
this should be feasable with [snippy](https://github.com/tseemann/snippy).
See the following section in the snippy github:
Finding SNPs between contigs

If you supply snippy with a .gbk reference instead of the .fna than it also does a snpEff analysis


```{bash}

##===============
##Sterm rmk202
##===============
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/snippy/{log,sterm,ldel}

echo -e "S72"'\t'"/archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/S72.fna" > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/snippy/log/log_sterm.txt
echo -e "S50"'\t'"/archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/S50.fna" >> /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/snippy/log/log_sterm.txt


#/archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/S72.fna
#/archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/S50.fna
#/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Streptococcus_thermophilus_RMK202.fasta

<<<<<<<<<<<<<<
mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/snippy/sterm
cd /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/snippy/sterm

snippy-multi  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/snippy/log/log_sterm.txt \
  --cpus 32 --ram 100 \
  --ref /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Streptococcus_thermophilus_RMK202.fasta \
  --tmpdir /Fast/tmp/


snippy --outdir 'S72' --ctgs '/archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/S72.fna' --cpus 32 --ram 100 --ref /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Streptococcus_thermophilus_RMK202.fasta --tmpdir /Fast/tmp/
snippy --outdir 'S50' --ctgs '/archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/S50.fna' --cpus 32 --ram 100 --ref /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Streptococcus_thermophilus_RMK202.fasta --tmpdir /Fast/tmp/

snippy-core --ref 'S50/ref.fa' S50 S72

 snippy-clean_full_aln core.full.aln > clean.full.aln
 run_gubbins.py -p gubbins clean.full.aln
 snp-sites -c gubbins.filtered_polymorphic_sites.fasta > clean.core.aln
 FastTree -gtr -nt clean.full.aln > clean.core.tree


##===============
##Ldel
##===============

```

## SNP matrix

Here, I try to creat a SNP matrix based on progressive mauve and looking only at sites that are covered by all genomes

```{bash}
##progressive mauve


##----------
##remove SNPs that are not complete
##---------

cut -f 1 Strepto_rmk202_snps.txt | grep "-" -v  > Strepto_rmk202_snps_complete.txt

sed '1d' Strepto_rmk202_snps_complete.txt |tr [a-z] [A-Z]   |sed 's/./&\t/g' > Strepto_rmk202_snps_complete_split.txt

```

```{r}
library(readr)
Strepto_core_snps <- read_delim("~/Desktop/Projects/2019_Pilotplan/snps/Strepto_rmk202_snps_complete_split.txt",  "\t", escape_double = FALSE, col_names = c("S72","S50","rmk202_mag","NAs"), trim_ws = TRUE) 
Strepto_core_snps <- Strepto_core_snps[,-4]

Sterm_matrix <- setNames(data.frame(matrix(ncol = ncol(Strepto_core_snps), nrow = ncol(Strepto_core_snps))), colnames(Strepto_core_snps))
row.names(Sterm_matrix) <-  colnames(Strepto_core_snps)

##testcases
cols=as.numeric("1")
alternativcols=as.numeric("2")

##loop
for (cols in 1:ncol(Strepto_core_snps)) {
  colnamess <- colnames(Strepto_core_snps)[cols]
  
     Sterm_matrix[cols,cols] <- sum(Strepto_core_snps[,cols]!=Strepto_core_snps[,cols])

  
  for (alternativcols in grep(colnamess,colnames(Strepto_core_snps),invert = TRUE)) {
    
   Sterm_matrix[alternativcols,cols] <- sum(Strepto_core_snps[,cols]!=Strepto_core_snps[,alternativcols])

  }

}


Sterm_matrix_rel <- 100*(Sterm_matrix/nrow(Strepto_core_snps))
Sterm_matrix_tot <- 100-100*(Sterm_matrix/1600000) ##this could be ANI


```



####Nucmer:repeat analysis

Here, I use [nucmer](http://mummer.sourceforge.net/) to do the analysis of the largest repeats

```{bash}

###==========================
##sterm
###==========================

mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/{fasta,nucmer,deltaFiltered}

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Streptococcus_thermophilus_RMK202.fasta \
  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/finalWithIllumina/Lactobacillus_delbrueckii_RMK202.fasta > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/fasta/reads.fa

###==========================
##nucmer
###==========================
cd /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer
nucmer -maxmatch -nosimplify /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/fasta/reads.fa /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/fasta/reads.fa


###==========================
##delta filter
###==========================
#mkdir -p $DATA_ROOT/11_nucmer_repeat/cottens_abruijn_delbruecki_streptococcus/{fasta,nucmer,deltaFiltered}
delta-filter -i 85 -l 3000  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/out.delta > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/filtered.txt
##split files
cd /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/
split -t ">" -l 1 /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/filtered.txt

##--------------
##extract Ldel repeats
##--------------
grep "[[:space:]]" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/xab | awk 'BEGIN {OFS ="\t"} {print "L_delbrueckii_RMK202" , $1 , $2 , "L_delbrueckii_RMK202" , $3 , $4}' > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/links_ldel.txt
cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/links_ldel.txt > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/links_ldel_curated.txt
##remove first row and reduant repeats

awk '{print $3-$2}' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/links_ldel_curated.txt |sort


##--------------
##extract sterm repeats
##--------------

grep "[[:space:]]" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/xad | awk 'BEGIN {OFS ="\t"} {print "S_thermophilus_RMK202" , $1 , $2 , "S_thermophilus_RMK202" , $3 , $4}' > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/links_sterm.txt
cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/links_sterm.txt > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/links_sterm_curated.txt
##remove first row and reduant repeats

awk '{print $3-$2}' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/links_sterm_curated.txt |sort


```

##SCOG

##### roary

```{bash}

mkdir -p /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Prokka/{Sterm,Ldel}
##-----------------
###01_annotate Sterm rmk202 with prokka
##-----------------


scp /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/Ster_rmk202_mag.fasta  vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Prokka/Sterm/
scp /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/LD_singu/Ldel_RMK202.fasta  vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Prokka/Ldel/


##sterm
/home/vincent/miniconda2/pkgs/prokka-1.12-1/bin/prokka --outdir /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Prokka/Sterm/roary --force --usegenus --genus streptococcus --locustag Sterm_rmk202 --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Prokka/Sterm/Ster_rmk202_mag.fasta


##Ldel 
prokka --outdir /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Prokka/Ldel/roary/ --force --usegenus --genus lactobacillus --locustag Ldel_K1 --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Prokka/Ldel/Ldel_RMK202.fasta

##-----------------
###02_roary
##-----------------

##-----sterm
rm -r /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm
mkdir -p /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/strains

cat /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Prokka/Sterm/roary/PROKKA_10222019.gff > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/strains/Streptococcus_thermophilus_RMK202.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/strains/S72.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/strains/S50.gff

sudo roary -f /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/ -p 5 -e -n -v /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/strains/*.gff

awk -F '","' '{print $4}' /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/_1571727203/gene_presence_absence.csv |sort |uniq -c

#awk -F '","' '{print $17}' /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/_1571727203/gene_presence_absence.csv |head
mkdir -p /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes
sudo chmod 777 /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/_1571727203/gene_presence_absence.csv
awk -F '","' '{if ($4=="3") print $17}' /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/_1571727203/gene_presence_absence.csv |sed 's/"//g' |sed 's/^ *//'|sed 's/* $//'  > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202.txt


###--------
##extract genes
###--------
rm /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202.vcf
for ids in $(cat /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202.txt |sed 's/\r//g')
  do
  
  echo "${ids}---"
  grep "Parent=${ids}_gene" /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Prokka/Sterm/roary/PROKKA_10222019.gff >> /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202.vcf
echo "------------"

  done
  
  
##-----BRING local

mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Sterm
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/_1571727203/gene_presence_absence.csv /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/

sed 's/"//g' /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/gene_presence_absence.csv > /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/gene_presence_absence_woQuotes.csv


##=====================
##-----Ldel
##=====================


sudo rm -r /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/strains/
mkdir -p /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/strains/


cat /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Prokka/Ldel/roary/*.gff > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/strains/Lactobacillus_delbrueckii_RMK202.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L104//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/strains/L104.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L108//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/strains/L108.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L35//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/strains/L35.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L39//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/strains/L39.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L44//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/strains/L44.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L70//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/strains/L70.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L71//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/strains/L71.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L99//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/strains/L99.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L80//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/strains/L80.gff


sudo roary -f /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/ -p 25 -e -n -v /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/strains/*.gff


mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Ldel
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/_1571722318/gene_presence_absence.csv /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Ldel/

  sed 's/"//g' /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Ldel/gene_presence_absence.csv > /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Ldel/gene_presence_absence_woQuotes.csv


awk -F '","' '{print $4}' /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/_1571727325/gene_presence_absence.csv |sort |uniq -c

#awk -F '","' '{print $17}' /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/_1571727203/gene_presence_absence.csv |head
mkdir -p /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/CoreGenes
sudo chmod 777 /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/_1571727325//gene_presence_absence.csv
awk -F '","' '{if ($4=="10") print $24}' /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/_1571727325//gene_presence_absence.csv |sed 's/"//g' |sed 's/^ *//'|sed 's/* $//'  > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/CoreGenes/coreGeneNames_rmk202.txt



###--------
##extract genes
###--------
rm /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/CoreGenes/coreGeneNames_rmk202.vcf
for ids in $(cat /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/CoreGenes/coreGeneNames_rmk202.txt |sed 's/\r//g')
  do
  
  echo "${ids}---"
  grep "Parent=${ids}_gene" /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/strains/Lactobacillus_delbrueckii_RMK202.gff  >> /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/CoreGenes/coreGeneNames_rmk202.vcf
echo "------------"

  done
  
  
  
###--------
##intersect with SNP and find core SNPs
###--------

cut -f 1 /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata/freebayesOuput_WithONT/variantCallsfreebayes_all_f_005_min_20x.vcf |sort|uniq -c
cut -f 1 /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/CoreGenes/coreGeneNames_rmk202.vcf |sort|uniq -c

cat /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202.vcf \
  /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/CoreGenes/coreGeneNames_rmk202.vcf > \
  /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202_bothLdelSterm.vcf

bedtools intersect -a /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata/freebayesOuput_WithONT/variantCallsfreebayes_all_f_005_min_20x.vcf  \
  -b /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202_bothLdelSterm.vcf > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202_snps.vcf

bedtools intersect -v -a /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata/freebayesOuput_WithONT/variantCallsfreebayes_all_f_005_min_20x.vcf  \
  -b /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202_bothLdelSterm.vcf > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202_nonCore_snps.vcf


grep -v "^#" /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata/freebayesOuput_WithONT/variantCallsfreebayes_all_f_005_min_20x.vcf |wc -l
wc -l /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202_snps.vcf
wc -l /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202_nonCore_snps.vcf


cut -f 1 /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202_nonCore_snps.vcf |sort|uniq -c
cut -f 1 /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202_snps.vcf |sort|uniq -c

awk -F "\t" '{OFS="\t"}{print $1"_"$2"_"$4"_"$5,"core"}' /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202_snps.vcf > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202_snps_cleaned.vcf 

awk -F "\t" '{OFS="\t"}{print $1"_"$2"_"$4"_"$5,"not"}' /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202_nonCore_snps.vcf > /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202_nonCore_snps_cleaned.vcf 

cat /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202_snps_cleaned.vcf  \
  /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202_nonCore_snps_cleaned.vcf > \
  /archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202_FINAL.txt

##-----BRING local

mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Sterm
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/Ldel/_1571727325//gene_presence_absence.csv /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/

sed 's/"//g' /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/gene_presence_absence.csv > /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/gene_presence_absence_woQuotes.csv

##core and non-core SNPs
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202_FINAL.txt /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/

##core genes
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/08_StrainAnalysis/01_roaryStrains/sterm/CoreGenes/coreGeneNames_rmk202_bothLdelSterm.vcf /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/

```

## Genome assembly stats

#### BUSCO

```{bash}


rm /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/all_analysis.txt

species=Sterm
species=Ldel
for speciesss in $(echo "Sterm Ldel")
do
echo "##------------------------"
echo ${speciesss}
echo "##------------------------"

for genemess in $(ls /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${speciesss}/faa/ )
  do
  
  genomesssss=$(echo ${genemess}|cut -d '.' -f 1)
  
  rm -r /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/${genemess}
  rm -r /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/${genomesssss}

  mkdir -p /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/${genomesssss}
  cd /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/${genomesssss}
  
  python /home/vincent/apps/busco/scripts/run_BUSCO.py -c 32 -i /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${speciesss}/faa/${genemess} -o ${genomesssss} -l /home/vincent/apps/busco/lactobacillales_odb9 -m prot

  #grep -v "^#" run_${genomesssss}/short_summary_${genomesssss}.txt| grep "C:" -A 7 
  all_INFO=$(grep -v "^#" run_${genomesssss}/short_summary_${genomesssss}.txt| grep "C:" | sed "s/^[ \t]*//"  )
  complet=$(echo ${all_INFO} | awk -F "[,[]" '{print $1}'|sed 's/%//g'|sed 's/C://g' )
  single=$(echo ${all_INFO} | awk -F "[,[]" '{print $2}'|sed 's/%//g'|sed 's/S://g')
  dupli=$(echo ${all_INFO} | awk -F "[,[]" '{print $3}'|sed 's/%]//g'|sed 's/D://g')
  fragm=$(echo ${all_INFO} | awk -F "[,[]" '{print $4}'|sed 's/%//g'|sed 's/F://g')
  Missi=$(echo ${all_INFO} | awk -F "[,[]" '{print $5}'|sed 's/%//g'|sed 's/M://g')
  numbi=$(echo ${all_INFO} | awk -F "[,[]" '{print $6}'|sed 's/n://g')
  
 echo -e ${speciesss}"\t"${genomesssss}"\t"${complet}"\t"${single}"\t"${dupli}"\t"${fragm}"\t"${Missi}"\t"${numbi}  \
 >> /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/all_analysis.txt

  done
done ## species

sed -i  's/Streptococcus_thermophilus_RMK202_pgap/S_thermophilus_mag_rmk202/g' /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/all_analysis.txt
sed -i  's/Lactobacillus_delbrueckii_RMK202_pgap/L_delbrueckii_mag_rmk202/g' /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/all_analysis.txt

```

#### remove small contigs

```{bash}




for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/isolates_names_changed.txt )
  do
  echo ==========================
   echo ${id}
   echo ==========================
 

  seqkit seq -m 300 /archiv/Projects/2018_Culturomics/06_Spades/${id}/Prokka/*/*19.fna -o /archiv/Projects/2018_Culturomics/06_Spades/${id}/Prokka/20190203/${id}_wo_300bp.fna
 
    done

```



#### genome stats

```{bash}

##---------------
##genome
##---------------

rm /archiv/Projects/2018_Culturomics/06_Spades//all_assembly_analysis.txt
for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/isolates_names_changed.txt )
  do
  echo ==========================
   echo ${id}
   echo ==========================
 
# /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/
 
   contiNumber=$(grep ">" -c /archiv/Projects/2018_Culturomics/06_Spades/${id}/Prokka/*/${id}_wo_300bp.fna) 
   GenomeLength=$(grep -v "^>" /archiv/Projects/2018_Culturomics/06_Spades/${id}/Prokka/*/${id}_wo_300bp.fna  |wc -c)
   
   # contiNumber=$(grep ">" -c /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/contigs.fasta) 
   #GenomeLength=$(grep -v "^>" /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/assemblyMetaSpades_20190121.fasta/contigs.fasta  |wc -c)
    
 echo -e ${id}"\t"${contiNumber}"\t"${GenomeLength}  >> /archiv/Projects/2018_Culturomics/06_Spades//all_assembly_analysis.txt

#"\t"${single}"\t"${dupli}"\t"${fragm}"\t"${Missi}"\t"${numbi}  \
    done



##---------------
##MAG
##---------------
##---------Sterm

id=S_thermophilus_mag_rmk202

contiNumber=$(grep ">" -c /archiv/Projects/2018_Culturomics/genomes/Streptococcus_thermophilus_RMK202.fna)
GenomeLength=$(grep -v "^>" /archiv/Projects/2018_Culturomics/genomes/Streptococcus_thermophilus_RMK202.fna  |wc -c)

 echo -e ${id}"\t"${contiNumber}"\t"${GenomeLength}  >> /archiv/Projects/2018_Culturomics/06_Spades//all_assembly_analysis.txt
 
##---------Ldel
id=L_delbrueckii_mag_rmk202

contiNumber=$(grep ">" -c /archiv/Projects/2018_Culturomics/genomes/RMK_Lactobacillus_delbrueckii_RMK202.fna)
GenomeLength=$(grep -v "^>" /archiv/Projects/2018_Culturomics/genomes/RMK_Lactobacillus_delbrueckii_RMK202.fna  |wc -c)

 echo -e ${id}"\t"${contiNumber}"\t"${GenomeLength}  >> /archiv/Projects/2018_Culturomics/06_Spades//all_assembly_analysis.txt
 
 
```

#### annotation stats

```{bash}


------

rm /archiv/Projects/2018_Culturomics/06_Spades//all_annotation_analysis.txt
for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/isolates_names_changed.txt )
  do
  echo ==========================
   echo ${id}
   echo ==========================
 
#   contiNumber=$(grep ">" -c /archiv/Projects/2018_Culturomics/06_Spades/${id}/Prokka/2*/*19.gff) 
#   GenomeLength=$(grep -v "^>" /archiv/Projects/2018_Culturomics/06_Spades/${id}/Prokka/2*/*19.gff  |wc -c)
  
  gensss=$(grep "gene: " /archiv/Projects/2018_Culturomics/06_Spades/${id}/Prokka/201*/*19.txt |sed 's/gene: //g')
  tRNAsss=$(grep "tRNA: " /archiv/Projects/2018_Culturomics/06_Spades/${id}/Prokka/201*/*19.txt |sed 's/tRNA: //g')
  rRNAsss=$(grep "rRNA: " /archiv/Projects/2018_Culturomics/06_Spades/${id}/Prokka/201*/*19.txt |sed 's/rRNA: //g')
  
  
 echo -e ${id}"\t"${gensss}"\t"${tRNAsss}"\t"${rRNAsss}  >> /archiv/Projects/2018_Culturomics/06_Spades//all_annotation_analysis.txt

#"\t"${single}"\t"${dupli}"\t"${fragm}"\t"${Missi}"\t"${numbi}  \
    done

###-----------Sterm RMK202 MAG
gbkFile=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/annot.gbk
id=S_thermophilus_mag_rmk202
  gensss=$(grep "Genes (total)                     :: " ${gbkFile}|sed 's/Genes (total)                     :: //g'|sed "s/^[ \t]*//"|sed 's/,//g')
  tRNAsss=$(grep "tRNAs                             :: " ${gbkFile} |sed 's/tRNAs                             :: //g'|sed "s/^[ \t]*//"|sed 's/,//g')
  rRNAsss=$(grep "Genes (RNA)                       :: " ${gbkFile} |sed 's/Genes (RNA)                       :: //g'|sed "s/^[ \t]*//"|sed 's/,//g')
  
  
 echo -e ${id}"\t"${gensss}"\t"${tRNAsss}"\t"${rRNAsss}  >> /archiv/Projects/2018_Culturomics/06_Spades//all_annotation_analysis.txt

    
  
###-----------Sterm RMK202 MAG
gbkFile=/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/annot.gbk
id=L_delbrueckii_mag_rmk202

  gensss=$(grep "Genes (total)                     :: " ${gbkFile}|sed 's/Genes (total)                     :: //g'|head -1 |sed "s/^[ \t]*//" |sed 's/,//g' )
  tRNAsss=$(grep "tRNAs                             :: " ${gbkFile} |sed 's/tRNAs                             :: //g'|head -1 |sed "s/^[ \t]*//" |sed 's/,//g' )
  rRNAsss=$(grep "Genes (RNA)                       :: " ${gbkFile} |sed 's/Genes (RNA)                       :: //g'|head -1 |sed "s/^[ \t]*//" |sed 's/,//g' )
  
 echo -e ${id}"\t"${gensss}"\t"${tRNAsss}"\t"${rRNAsss}  >> /archiv/Projects/2018_Culturomics/06_Spades//all_annotation_analysis.txt

```

####mapping stats

```{bash}


for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/isolates_names_changed.txt)
  do
  echo ==========================
   echo ${id}
   echo ==========================
    

 rm -r /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/PostaAssembly
mkdir -p /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/PostaAssembly
  
  
  bwa index /archiv/Projects/2018_Culturomics/06_Spades/${id}/Prokka/20190203/${id}_wo_300bp.fna
  
  bwa mem -t 32 /archiv/Projects/2018_Culturomics/06_Spades/${id}/Prokka/20190203/${id}_wo_300bp.fna \
  /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${id}/${id}_R1_val_1.fq /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${id}/${id}_R2_val_2.fq | samtools sort -@$32 -O BAM -o /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/PostaAssembly/${id}_rawIllumina_bwamem_sorted.bam - 

##-----Qualimap

mkdir -p /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/PostaAssembly//bamqc

qualimap bamqc -bam /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/PostaAssembly/${id}_rawIllumina_bwamem_sorted.bam -outdir /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/PostaAssembly//bamqc --java-mem-size=80G

    
    done


##---------------------
##get data
##---------------------

rm /archiv/Projects/2018_Culturomics/06_Spades//all_genome_analysis.txt
for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/isolates_names_changed.txt)
  do

GCss=$(grep "GC percentage" /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/PostaAssembly//bamqc/genome_results.txt| sed 's/GC percentage = //g'| sed "s/^[ \t]*//"|sed 's/%//g'  )

coveragesss=$(grep "mean coverageData = " /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/PostaAssembly//bamqc/genome_results.txt| sed 's/mean coverageData = //g'| sed "s/^[ \t]*//"|sed 's/X//g'  )



#grep -A 10000 ">>>>>>> Coverage per contig" /archiv/Projects/2018_Culturomics/06_Spades/${id}/assembly/PostaAssembly//bamqc/genome_results.txt| grep -v ">>>>>>> Coverage per contig"| sed "s/^[ \t]*//" |sed -r '/^\s*$/d' 


echo -e ${id}"\t"${GCss}"\t"${coveragesss}"\t""NA" >> /archiv/Projects/2018_Culturomics/06_Spades//all_genome_analysis.txt

done

##---------------------
##get data ONT coverage
##---------------------
## GC
infoseq /archiv/Projects/2019_RMK202_analysis/02_flye_Assembly/merged/RMK202_merged.fasta -noheading -nodatabase -nolength -nousa -nohead -noacc -notype -nodesc -delimiter '_' -outfile /archiv/Projects/2019_RMK202_analysis/02_flye_Assembly/merged/RMK202_mergeg_infoseq.txt


##--------------Sterm
oldName=contig_1
NewName=S_thermophilus_mag_rmk202
GCss=38
## illumina both
IlluminaCoverage=$(grep ">>>>>>> Coverage per contig" -A 20 /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/rawIlluminaMapped/bamqc/genome_results.txt | grep "$oldName" |sed "s/^[ \t]*//"|cut -f 4) 

## ONT both
ONTCoverage=$(grep ">>>>>>> Coverage per contig" -A 20 /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/04_fourthRacon_graphmap/Graphmap_ONT/bamqc/genome_results.txt | grep "$oldName" |sed "s/^[ \t]*//"|cut -f 4)

## GC
GCss=$(grep "S_thermophilus_RMK202" /archiv/Projects/2019_RMK202_analysis/02_flye_Assembly/merged/RMK202_mergeg_infoseq.txt | awk -F " " '{print $2}')

## alltogether
echo -e ${NewName}"\t"${GCss}"\t"${IlluminaCoverage}"\t"${ONTCoverage}  >> /archiv/Projects/2018_Culturomics/06_Spades//all_genome_analysis.txt



##--------------Ldel

oldName=contig_2
NewName=L_delbrueckii_mag_rmk202
GCss=38
## illumina both
IlluminaCoverage=$(grep ">>>>>>> Coverage per contig" -A 20 /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/rawIlluminaMapped/bamqc/genome_results.txt | grep "$oldName" |sed "s/^[ \t]*//"|cut -f 4) 

## ONT both
ONTCoverage=$(grep ">>>>>>> Coverage per contig" -A 20 /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/04_fourthRacon_graphmap/Graphmap_ONT/bamqc/genome_results.txt | grep "$oldName" |sed "s/^[ \t]*//"|cut -f 4)

## GC
GCss=$(grep "L_delbrueckii_RMK202" /archiv/Projects/2019_RMK202_analysis/02_flye_Assembly/merged/RMK202_mergeg_infoseq.txt | awk -F " " '{print $2}')

## alltogether
echo -e ${NewName}"\t"${GCss}"\t"${IlluminaCoverage}"\t"${ONTCoverage}  >> /archiv/Projects/2018_Culturomics/06_Spades//all_genome_analysis.txt


```

#### bring together
```{bash}

cat /archiv/Projects/2018_Culturomics/06_Spades//all_assembly_analysis.txt
cat /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/all_analysis.txt
cat /archiv/Projects/2018_Culturomics/06_Spades//all_genome_analysis.txt
cat /archiv/Projects/2018_Culturomics/06_Spades//all_annotation_analysis.txt

mkdir -p /archiv/Projects/2018_Culturomics/06_Spades/genomeAnalysi/
cp /archiv/Projects/2018_Culturomics/06_Spades//all_assembly_analysis.txt /archiv/Projects/2018_Culturomics/06_Spades/genomeAnalysi/
cp /archiv/Projects/2019_Nano_Meta/07_BUSCO_assemblies/all_analysis.txt /archiv/Projects/2018_Culturomics/06_Spades/genomeAnalysi/
cp /archiv/Projects/2018_Culturomics/06_Spades//all_genome_analysis.txt /archiv/Projects/2018_Culturomics/06_Spades/genomeAnalysi/
cp /archiv/Projects/2018_Culturomics/06_Spades//all_annotation_analysis.txt /archiv/Projects/2018_Culturomics/06_Spades/genomeAnalysi/

scp -r vincent@130.223.51.151:/archiv/Projects/2018_Culturomics/06_Spades/genomeAnalysi/ ~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/

```

```{r}
library(readr)
genome_stats <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/genomeAnalysi/all_assembly_analysis.txt", "\t", escape_double = FALSE, col_names = c("strain","contigNumber","genomeSize"),  trim_ws = TRUE)
Genes_stats <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/genomeAnalysi/all_annotation_analysis.txt", "\t", escape_double = FALSE, col_names = c("strain","genes","tRNA","rRNA"),  trim_ws = TRUE)
BUSCO_stats <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/genomeAnalysi/all_analysis.txt", "\t", escape_double = FALSE, col_names = c("species","strain","completness","singleCopy","duplicated","fragmentation","Missing","numberGenesTested"),  trim_ws = TRUE)
mapping_stats <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/genomeAnalysi/all_genome_analysis.txt", "\t", escape_double = FALSE, col_names =c("strain","GC","IlluminaCoverage","ONTcoverage"),  trim_ws = TRUE)

tmp <- merge(BUSCO_stats,genome_stats,by="strain")
tmp <- merge(tmp,Genes_stats,by="strain")
GenomeAssemblyStats <- merge(tmp,mapping_stats,by="strain")

```


## Mapping Data

####mapping to single polished genome

```{bash}


##============
##mapping to reference
##============

 name_folder=02_againstpolished_single_K1_final_kneaddata_withStrains 

threads=37
#rm -r /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/

mkdir -p /archiv/Projects/2019_RMK202_analysis/01_log
#rm /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/log/multiqc_logfile_finalGenome.txt
#cp /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name_allOfExperiment.txt /archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt
#cat /archiv/Projects/2019_RMK202_analysis/01_log/names.txt /archiv/Projects/2019_RMK202_analysis/01_log/names_only_extended.txt > /archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt
num=1
#rm -r /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/

#  for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/names_onlymeta.txt)
#for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/log/20191031_logDownload_changeName.txt|tail -1)
for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt)
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))
  rm -r /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final/${names}

  mkdir -p /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/bwaMapping2DB/
  
 # bwa mem -t ${threads} /archiv/Projects/2019_RMK202_analysis/02_flye_Assembly/merged/RMK202_merged.fasta
  bwa mem -t ${threads} /archiv/Projects/2019_RMK202_analysis/02_flye_Assembly/merged/RMK202_merged.fasta \
  /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_1.fastq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/kneaddata/${names}/${names}/${names}*neaddata_paired_2.fastq | samtools sort -@${threads} -O BAM -o /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 

#samtools view -b -f 4 /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

#bedtools bamtofastq -i /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam -fq /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq

#rm /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

##-----Qualimap

mkdir -p /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/bwaMapping2DB//bamqc

qualimap bamqc -bam /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/bwaMapping2DB//bamqc --java-mem-size=80G


 mkdir -p /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/log
echo "/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/"${name_folder}"/"${names}"/bwaMapping2DB//bamqc" >> /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/log/multiqc_logfile_finalGenome.txt
   done 
   
   
    rm -r /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/MultiQC
    
  mkdir -p /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/MultiQC
  
  multiqc --config /archiv/logs/multiQC_config.txt --file-list  /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/log/multiqc_logfile_finalGenome.txt \
    -o /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/MultiQC


  
#scp vincent@130.223.51.151:/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrains/MultiQC/multiqc_report.html ~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/1AB/
   

  
##-----plasmid spades/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/01_againstUnpolished/K1_ONT/spadesAssembly_unmapped/assembly/
rm -r /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1/
num=1
  for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/names.txt)
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))
rm -r /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata/${names}/spadesAssembly_unmapped/assembly
mkdir -p /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata/${names}/spadesAssembly_unmapped/assembly

/home/vincent/miniconda2/pkgs/spades-3.13.0-0/bin/spades.py --plasmid -t ${threads} -m 100 \
  -s /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq \
  -o /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata/${names}/spadesAssembly_unmapped/assembly
  done
##----remove short contigs (<500bp)


awk '/^>/ {printf("\n%s\n",$0);next; } { printf("%s",$0);}  END {printf("\n");}' < /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/spadesAssembly_unmapped/assembly/scaffolds.fasta |sed '/^[[:space:]]*$/d' | awk '/^>/ { getline seq } length(seq) >500 { print $0 "\n" seq }' > /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/spadesAssembly_unmapped/assembly/scaffolds_min500bp.fasta


##----blastn


mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/spadesAssembly_unmapped/blastn
  
  blastn -num_threads ${threads} -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/spadesAssembly_unmapped/assembly/scaffolds_min500bp.fasta \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen" \
  -db /archiv/BLAST/blast_db/20190204/phage_nucleotide/phage_nucleotide_cow \
  -out /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/spadesAssembly_unmapped/blastn/blastAssemblyaginstScaffolds.txt \
  -evalue 0.01 -word_size 64
  
  done 
  

#scp vincent@130.223.51.151:/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/di_K1_0h/spadesAssembly_unmapped/assembly/assembly_graph_with_scaffolds.gfa 


##------------------------------extract assembly graph

mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/AssemblyGraphs/
  for names in $(cat /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name_allOfExperiment.txt)
do


echo "##====================="
echo ${names}

cp /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/spadesAssembly_unmapped/assembly/assembly_graph_with_scaffolds.gfa /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/AssemblyGraphs/${names}.gfa

done


scp -r vincent@130.223.51.151:/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/AssemblyGraphs/ /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/


##------------------------------move local

scp -r vincent@130.223.51.151:/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/Versand_202/plasmidSpades_unassembled/assembly_graph_with_scaffolds.gfa /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/Versand_202.gfa


scp -r vincent@130.223.51.151:/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/Konserve_202/plasmidSpades_unassembled/assembly_graph_with_scaffolds.gfa /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/Konserve_202.gfa


###----------------assembly graph

spades_contig_graph.py -l  /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/di_K1_0h/spadesAssembly_unmapped/assembly/assembly_graph.fastg  /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/di_K1_0h/spadesAssembly_unmapped/assembly/contigs.fasta  /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/di_K1_0h/spadesAssembly_unmapped/assembly/contigs.paths -o  /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/di_K1_0h/spadesAssembly_unmapped/assembly/contigs.fastg


```

#####Prophage detection

Here, I try to identify reads that map both to the Streptococcus genome and the phage

```{bash}



name_folder=02_againstpolished_single_K1_final_kneaddata

threads=37
#rm -r /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/

  for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/names_onlymeta.txt)
  do
  echo ${num}"/16  :" ${names}
  num=$((num+1))

  rm -r /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/03_mapping_Prophage/${names}/
  mkdir -p /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/03_mapping_Prophage/${names}/
  
samtools view -H /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/03_mapping_Prophage/${names}/header.bam

samtools view /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/${name_folder}/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam |grep  "S_phage_RMK202" |grep "S_thermophilus_RMK202" >  /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/03_mapping_Prophage/${names}/data_tmp.bam

cat /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/03_mapping_Prophage/${names}/header.bam /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/03_mapping_Prophage/${names}/data_tmp.bam | samtools view -S -b  >  /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/03_mapping_Prophage/${names}/${names}_mappingProphage.bam

done
```


getting coverage information

```{bash}

date=20190513

#rm /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/log/mappingcoverage.txt
#mkdir -p /home/vincent/Projects/2019_pilotplant/06_MetaSNV_ONT/{99_logs,01_analysis_01}

  threads=37

###------------make a file file
  num=1
  for names in $(cat /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name_allOfExperiment.txt)
    do
    
   # sampleShort=$(echo ${names}| awk -F "_" 'BEGIN{OFS="_"}{print $1,$2}')
  
   #NanoSample=$(grep "$sampleShort" /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt) 
    
    echo ${num}"/37  :" ${names} 
    num=$((num+1))


#samtools depth  -m 80000 /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam  |  awk '{sum+=$3} END { print "Average = ",sum/NR}' |awk -F "\t" -v namess="$names"  'BEGIN{OFS="\t"}{print $0, namess}' #>> /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/${names}/bwaMapping2DB/${names}_coverage.txt

##extract from qualimap

grep -A 100 ">>>> Coverage per contig" /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata/${names}/bwaMapping2DB/bamqc/genome_results.txt | sed '1,2d' | sed '/^$/d' | sed -e 's/^[ \t]*//'  |awk -F "\t" -v namess="$names"  'BEGIN{OFS="\t"}{print $0, namess}' >> /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata/log/mappingcoverage.txt

done

##transfer coverage local

scp vincent@130.223.51.151:/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata/log/mappingcoverage.txt ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/mappingcoverage_2ONT.txt

```

```{r}

library(readr)
library("stringr")
library("ggplot2")
library(tidyverse)


mappingcoverage <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/mappingcoverage_2ONT.txt",  "\t", escape_double = FALSE, col_names = c("contig","size","readsMapped","coverage","std","sample"),  trim_ws = TRUE)

mappingcoverage$genome <- mappingcoverage$contig
mappingcoverage[grep("S_phage_RMK202",mappingcoverage$genome),"genome"] <- "S_phage_RMK202"
mappingcoverage[grep("L_plasmid_RMK202_",mappingcoverage$genome),"genome"] <- "L_plasmid_RMK202"
unique(mappingcoverage$genome)

mappingcoverage_reduced <- mappingcoverage[,-c(4,5)]
mappingcoverage_reduced$sample <- as.factor(mappingcoverage_reduced$sample)
mappingcoverage_reduced$genome <- as.factor(mappingcoverage_reduced$genome)


  ##=============coverage    

   mappingcoverage_reduced_summed <-aggregate(.~sample+genome, data=mappingcoverage_reduced[,c("sample","genome","readsMapped")], sum, na.rm=TRUE)
      mappingcoverage_reduced_size <-aggregate(.~sample+genome, data=mappingcoverage_reduced[,c("sample","genome","size")], sum, na.rm=TRUE)
      mappingcoverage_reduced_final <- cbind(mappingcoverage_reduced_summed,size=mappingcoverage_reduced_size$size)
      mappingcoverage_reduced_final$coverage <- (mappingcoverage_reduced_final$readsMapped*500)/mappingcoverage_reduced_final$size
      
      
  ##=============percent    


  mappingcoverage_reduced_final_summed <-aggregate(. ~sample , data=mappingcoverage_reduced_final[,c("sample","coverage")], sum, na.rm=TRUE)
  mappingcoverage_reduced_final$total_coverage <- mappingcoverage_reduced_final_summed[match(mappingcoverage_reduced_final$sample,mappingcoverage_reduced_final_summed$sample),"coverage"]
  mappingcoverage_reduced_final$percent_coverage <- 100*(mappingcoverage_reduced_final$coverage/mappingcoverage_reduced_final$total_coverage)

  mappingcoverage_reduced_final <- mappingcoverage_reduced_final[which(mappingcoverage_reduced_final$genome!="Bos_taurus_mitochondrion"),]
  
 ##=============sample order    
  levels(factor(mappingcoverage_reduced_final$genome))
  mappingcoverage_reduced_final$genome_f <- factor(mappingcoverage_reduced_final$genome, levels=c("S_thermophilus_RMK202","S_ther_potential_Prophage","S_phage_RMK202","L_delbrueckii_RMK202","L_delbrueckii_plasmid_RMK202_01","L_plasmid_RMK202","L_phage_RMK202_01","L_phage_RMK202_02"))

  mappingcoverage_reduced_final$genome_f <- factor(mappingcoverage_reduced_final$genome, levels=rev(c("S_thermophilus_RMK202","S_ther_potential_Prophage","S_phage_RMK202","L_delbrueckii_RMK202","L_delbrueckii_plasmid_RMK202_01","L_plasmid_RMK202","L_phage_RMK202_01","L_phage_RMK202_02")))
  

  mappingcoverage_reduced_final$sample_f = factor(mappingcoverage_reduced_final$sample, levels=c("Lyo_202_2012","Lyo_202_2014","RMK202","Konserve_202","Versand_202","di_K1_0h","th_K1_0h","di_K1_2h","th_K1_2h","di_K1_4h","th_K1_4h","di_K1_6h","th_K1_6h","di_K1_8h","th_K1_8h","di_K1_10h","th_K1_10h","di_K1_12h","th_K1_12h","di_K1_24h","th_K1_24h","S72","S50","L104","L108","L35","L39","L44","L70","L71","L80","L99"))

   ##=============colours    


Sterm_colours <- rev(c("#EB4D4D","#FAA0A0","indianred1","#10B552","#36E37B","#6CF5A3","cyan3","cyan"))

levels(mappingcoverage_reduced_final$sample)

mappingcoverage_reduced_final <- mappingcoverage_reduced_final[mappingcoverage_reduced_final$sample %in% c("Konserve_202","Lyo_202_2012","Lyo_202_2014","RMK202","Versand_202"),]


p2 <-  ggplot( data = mappingcoverage_reduced_final,aes(y = percent_coverage, x = sample_f, group=genome_f,fill = genome_f))+ geom_bar( stat="identity")+
    labs("",
         x="",
         y="relative Abundance")+
    theme_classic()+
   # scale_color_viridis(discrete=TRUE)+
  scale_fill_manual(values=Sterm_colours)+
  # scale_fill_viridis(discrete=TRUE)+
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
      #axis.text.x = element_blank(),
          legend.position="right",
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank()
          )

  p2
  svg("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.svg",width=7,height=4)
   # png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 1000, height = 800,res=300)
p2
 dev.off()
 
 

```

mappping stats

```{r}
library(readr)
library(stringr)
library(ggplot2)
mappingStat <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/mappingQC/mappingStat_forR.csv", "\t", escape_double = FALSE, col_names = c("name","empty1","empty2","empty3","percent","Mio.reads"),trim_ws = TRUE)


mappingStat <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/mappingQC/mapping_after_kneaddata_cleaned.csv", "\t", escape_double = FALSE, col_names = c("name","percent","Mio.reads"),trim_ws = TRUE)




  mappingStat$day <- str_split_fixed(mappingStat$name, "_", 3)[,1]
  mappingStat$kessel <- str_split_fixed(mappingStat$name, "_", 3)[,2]
  mappingStat$time <- str_split_fixed(mappingStat$name, "_", 3)[,3]
  mappingStat$treatment <- paste0(mappingStat$day,"_",mappingStat$kessel)
 
  mappingStat$treatment_f = factor(mappingStat$treatment, levels=c('di_K1','th_K1','di_K2','th_K2'))
  mappingStat$time_f = factor(mappingStat$time, levels=c("0h","2h","4h","6h","8h","10h","12h","24h"))

mappingStat$name <- as.factor(mappingStat$name)
  mappingStat$name_f = factor(mappingStat$name, levels=c("Lyo_202_2012","Lyo_202_2014","RMK202","Konserve_202","Versand_202","di_K1_0h","th_K1_0h","di_K1_2h","th_K1_2h","di_K1_4h","th_K1_4h","di_K1_6h","th_K1_6h","di_K1_8h","th_K1_8h","di_K1_10h","th_K1_10h","di_K1_12h","th_K1_12h","di_K1_24h","th_K1_24h","S72","S50","L104","L108","L35","L39","L44","L70","L71","L80","L99"))

  
  mappingStat <- mappingStat[mappingStat$name %in% c("Lyo_202_2012","Lyo_202_2014","RMK202","Konserve_202","Versand_202"),]
  
  levels(mappingStat$name)
  min(mappingStat$percent)
  coloursplotting <-  c(rep("cadetblue3",5),rep("azure4",16),rep("#EB4D4D",2),rep("#10B552",9))
  
 rawREad <-  ggplot(mappingStat,aes(x=name_f,y=percent,fill=name_f))+geom_bar( stat="identity", position="dodge")+
    labs(y="Percent\nraw reads mapping",
        x="")+
         coord_cartesian(ylim=c(90,100))+
   scale_fill_manual(values=coloursplotting)+
       theme_classic()+
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          # # axis.text.y.left = element_text(color = "blue"),
          # axis.text.y.right = element_text(color="red"),
          # axis.title.y.left = element_text(color="blue"),
          # axis.title.y.right = element_text(color="red"),
          # rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          )
 rawREad
 
   # svg("~/Desktop/presenations/my_presentations/20190807_committeeMeeting/figures/04_rawReadMapping.svg",width=5,height=3)
  svg("~/Desktop/Manuscripts/2019_RMK202/Figures/S05_rawReadMapping.svg",width=5,height=3)
    png("~/Desktop/Manuscripts/2019_RMK202/Figures/S05_rawReadMapping.png", width = 1000, height = 800,res=300)


rawREad
 dev.off()
 
 
 mappingStat$newReads <- (mappingStat$percent/100)*mappingStat$Mio.reads
  
 Mio_reads <- ggplot(mappingStat,aes(x=time_f,y=newReads,group=treatment,color=kessel,fill=kessel))+geom_bar( stat="identity", position="dodge")+
    labs(y="Mio. reads mapping",
        y="")
 
    svg("~/Desktop/presenations/my_presentations/20190708_groupmeeting/figures/rawReadMapping_mio.svg",width=5,height=3)
Mio_reads
 dev.off()
 
 
```


###CIGAR parsing


```{bash}

#rm /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/CIGAR_parsing
 
  num=1
  for names in $(cat /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name_allOfExperiment.txt)
    do
echo ${num}"/33  :" ${names}
    num=$((num+1))
    
    #names=th_K1_12h
    #rm -r /home/vincent/Projects/2019_pilotplant/05_mapping2DB/${names}/BacteriaDBmapping
  
    mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/CIGAR_parsing/
    
    samtools index /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam
    
perl /home/vincent/scripts/mapped_read_Length_extended_reduced.pl /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/CIGAR_parsing/${names}_rawIllumina_bwamem_sorted_statsforR.txt

done



##-----------------move local
#names=th_K1_12h
scp -r vincent@130.223.51.151:/home/vincent/Projects/2019_pilotplant/05_mapping2DB/logs_mapping/ ~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/QC_ofDBmapping


```
## Species Coverage



##### phage mapping

```{bash}
##get RAST annotation to bigmachine
mkdir -p /archiv/Projects/2019_pilotplant/02_annotation 
scp -r /home/vincent/Desktop/Projects/2019_Pilotplan/09_annoation/rastServer vincent@130.223.51.151:/archiv/Projects/2019_pilotplant/02_annotation 

sed -i 's/S_phage_RMK202_08/L_phage_RMK202_b/g' /archiv/Projects/2019_pilotplant/02_annotation/rastServer/Lactobacillus_delbrueckii_ViSo-RMK202_b.gbk

##----------------------------------------------
##convert RAST annotation to gff
##----------------------------------------------
cd /archiv/Projects/2019_pilotplant/02_annotation/rastServer/
/home/vincent/miniconda2/bin/bp_genbank2gff3.pl /archiv/Projects/2019_pilotplant/02_annotation/rastServer/Lactobacillus_phage_RMK202_01.fasta.gbk
/home/vincent/miniconda2/bin/bp_genbank2gff3.pl --split /archiv/Projects/2019_pilotplant/02_annotation/rastServer/Streptococcus_phage.fasta.gbk
/home/vincent/miniconda2/bin/bp_genbank2gff3.pl /archiv/Projects/2019_pilotplant/02_annotation/rastServer/Lactobacillus_phage_RMK202_02.fasta.gbk

convert2bed --input=gff < /archiv/Projects/2019_pilotplant/02_annotation/rastServer/Lactobacillus_phage_RMK202_01.fasta.gbk.gff > /archiv/Projects/2019_pilotplant/02_annotation/rastServer/Lactobacillus_phage_RMK202_01.bed
convert2bed --input=gff < /archiv/Projects/2019_pilotplant/02_annotation/rastServer/Streptococcus_phage.fasta.gbk.gff > /archiv/Projects/2019_pilotplant/02_annotation/rastServer/Streptococcus_phage.bed
convert2bed --input=gff < /archiv/Projects/2019_pilotplant/02_annotation/rastServer/Lactobacillus_phage_RMK202_02.fasta.gbk.gff > /archiv/Projects/2019_pilotplant/02_annotation/rastServer/Lactobacillus_phage_RMK202_02.bed

cat \
/archiv/Projects/2019_pilotplant/02_annotation/rastServer/Lactobacillus_phage_RMK202_01.bed \
/archiv/Projects/2019_pilotplant/02_annotation/rastServer/Streptococcus_phage.bed \
/archiv/Projects/2019_pilotplant/02_annotation/rastServer/Lactobacillus_phage_RMK202_02.bed > \
/archiv/Projects/2019_pilotplant/02_annotation/rastServer/all_phages.bed
  
  
##----------------------------------------------
##Phage coverage
##----------------------------------------------

rm -r /home/vincent/Projects/2019_pilotplant/05_mapping2RefAssembly/phageDBmapping_rmk202/

 # rm  /home/vincent/Projects/2019_pilotplant/05_mapping2DB//log/multiqc_logfile.txt
  num=1
  for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt)
    do
echo ${num}"/16  :" ${names}
    num=$((num+1))
    rm -r /home/vincent/Projects/2019_pilotplant/05_mapping2RefAssembly/phageDBmapping_rmk202/${names}
  
    mkdir -p /home/vincent/Projects/2019_pilotplant/05_mapping2RefAssembly/phageDBmapping_rmk202/${names}
    

bedtools coverage -bed -a /archiv/Projects/2019_pilotplant/02_annotation/rastServer/all_phages.bed -b /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrains/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam  > \
  /home/vincent/Projects/2019_pilotplant/05_mapping2RefAssembly/phageDBmapping_rmk202/${names}/${names}2phageDB.bed 
  
done


##----------------------------------------------
###----------------change name of bed file
##----------------------------------------------
names=Konserve_202
 #rm -r /home/vincent/Projects/2019_pilotplant/05_mapping2RefAssembly/phageDBmapping_rmk202
 mkdir -p /home/vincent/Projects/2019_pilotplant/05_mapping2DB/phageDBmapping_rmk202/finalmapping/
for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt)
    do
    
echo $names
  
    cat /home/vincent/Projects/2019_pilotplant/05_mapping2RefAssembly/phageDBmapping_rmk202/${names}/${names}2phageDB.bed | sed -e 's/ID=.*;OG=/OG=/g'|awk -F "[\t]" -v namess="$names"  'BEGIN{OFS="\t"}{print $0,namess}'  > /home/vincent/Projects/2019_pilotplant/05_mapping2DB/phageDBmapping_rmk202/finalmapping/${names}_finalPhages_forR.bed
    

    done
cat /home/vincent/Projects/2019_pilotplant/05_mapping2DB/phageDBmapping_rmk202/finalmapping/*.bed > /home/vincent/Projects/2019_pilotplant/05_mapping2DB/phageDBmapping_rmk202/finalmapping/all_Phages_allSamples.txt

##----------------------------------------------
###-------------on local machine
##----------------------------------------------

scp vincent@130.223.51.151:/home/vincent/Projects/2019_pilotplant/05_mapping2DB/phageDBmapping_rmk202/finalmapping/all_Phages_allSamples.txt /home/vincent/Desktop/Projects/2019_Pilotplan/09_annoation/all_Phages_allSamples_rmk202.txt


#cat /home/vincent/Desktop/Projects/2019_Pilotplan/09_annoation/finalmapping/*.bed > /home/vincent/Desktop/Projects/2019_Pilotplan/09_annoation/finalmapping/all_Phages_allSamples.txt

```

```{r}


source("/home/vincent/Desktop/ScriptRepository/R_functions-master/R_functions/g_legend.R") ##for plotting only legends
library(readr)
library(ggplot2)
library(dplyr)
library(tidyverse)
library(grid)
library(gridExtra)
library(cowplot) ## for arranging plots
library(stringr) ##for split string
library(rlist) ##rbind all dataframe of list
library(viridis) ##colour palette
##===================================
#-------------file import


  read_count <- read_delim("/home/vincent/Desktop/Projects/2019_Pilotplan/09_annoation/all_Phages_allSamples_rmk202.txt","\t", escape_double = FALSE, col_names = c("chr","start","end","name","unknown2","orientation","origin","sort","unknown","description","count","length_mapped","geneLength","sample2","sample"),trim_ws = TRUE)

table(read_count$chr)

read_count$species <- ifelse(read_count$chr=="L_phage_RMK202_01","L_phage_RMK202_01",ifelse(
    read_count$chr=="L_phage_RMK202_02","L_phage_RMK202_02","S_phage_RMK202"
))

table(read_count$species)

read_count$geneCoverage  <- (read_count$count*600)/read_count$geneLength

ggplot(read_count,aes(y=geneCoverage,group=sort,color=sort,fill=sort))+geom_boxplot()+facet_grid(sample~species, scales="free")

library(dplyr)


phage_final <- read_count %>% 
  group_by(sample,species) %>% 
  dplyr::summarize(median = median(geneCoverage))


```
##### bacteria reference genome

Currently, I am leaving out the bam file filtering (CIGAR.filtering) and going straight to the gene coverage

```{bash}
##-------------------------------------
###----------------prep gffs
##-------------------------------------
mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/PGAP_for_Abundance

grep -v "^#" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pgap.gff | awk -F "\t" 'BEGIN{OFS="\t"} {if($3=="CDS") print $0}' > /archiv/Projects/2019_pilotplant/02_annotation/PGAP_for_Abundance/Streptococcus_thermophilus_RMK202_pgap_forAbundance.gff

grep -v "^#" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Lactobacillus_delbrueckii_RMK202_pgap.gff | awk -F "\t" 'BEGIN{OFS="\t"} {if($3=="CDS") print $0}' > /archiv/Projects/2019_pilotplant/02_annotation/PGAP_for_Abundance/Lactobacillus_delbrueckii_RMK202_pgap_forAbundance.gff

cat /archiv/Projects/2019_pilotplant/02_annotation/PGAP_for_Abundance/Streptococcus_thermophilus_RMK202_pgap_forAbundance.gff \
/archiv/Projects/2019_pilotplant/02_annotation/PGAP_for_Abundance/Lactobacillus_delbrueckii_RMK202_pgap_forAbundance.gff > \
/archiv/Projects/2019_pilotplant/02_annotation/PGAP_for_Abundance/Bacteria_pgap_forAbundance.gff

cut -f 1 /archiv/Projects/2019_pilotplant/02_annotation/PGAP_for_Abundance/Bacteria_pgap_forAbundance.gff|sort|uniq -c

##-------------------------------------
###----------------extract coverage
##-------------------------------------
rm -r /home/vincent/Projects/2019_pilotplant/05_mapping2RefAssembly/bacteriaDBmapping_rmk202/

 # rm  /home/vincent/Projects/2019_pilotplant/05_mapping2DB//log/multiqc_logfile.txt
  num=1
  for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt)
    do
echo ${num}"/16  :" ${names}
    num=$((num+1))
  
    rm -r /home/vincent/Projects/2019_pilotplant/05_mapping2RefAssembly/bacteriaDBmapping_rmk202/${names}
    mkdir -p /home/vincent/Projects/2019_pilotplant/05_mapping2RefAssembly/bacteriaDBmapping_rmk202/${names}
    

bedtools coverage -bed -a /archiv/Projects/2019_pilotplant/02_annotation/PGAP_for_Abundance/Bacteria_pgap_forAbundance.gff -b /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrains/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam  > \
  /home/vincent/Projects/2019_pilotplant/05_mapping2RefAssembly/bacteriaDBmapping_rmk202/${names}/${names}2bacteriaDB.bed 
  
done


##-------------------------------------
###----------------change name of bed file
##-------------------------------------
names=Konserve_202
 #rm -r /home/vincent/Projects/2019_pilotplant/05_mapping2RefAssembly/bacteriaDBmapping_rmk202
 mkdir -p /home/vincent/Projects/2019_pilotplant/05_mapping2DB/bacteriaDBmapping_rmk202/finalmapping/
for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/onlyMeta_samples.txt)
    do
    
echo $names
  
    cat /home/vincent/Projects/2019_pilotplant/05_mapping2RefAssembly/bacteriaDBmapping_rmk202/${names}/${names}2bacteriaDB.bed | sed -e 's/ID=.*;OG=/OG=/g'|awk -F "[\t]" -v namess="$names"  'BEGIN{OFS="\t"}{print $0,namess}'  > /home/vincent/Projects/2019_pilotplant/05_mapping2DB/bacteriaDBmapping_rmk202/finalmapping/${names}_finalBacteria_forR.bed
    

    done
cat /home/vincent/Projects/2019_pilotplant/05_mapping2DB/bacteriaDBmapping_rmk202/finalmapping/*.bed > /home/vincent/Projects/2019_pilotplant/05_mapping2DB/bacteriaDBmapping_rmk202/finalmapping/all_Bacteria_allSamples.txt

 cut -f 1 /home/vincent/Projects/2019_pilotplant/05_mapping2DB/bacteriaDBmapping_rmk202/finalmapping/all_Bacteria_allSamples.txt | sort|uniq -c
 
###-------------on local machine

scp vincent@130.223.51.151:/home/vincent/Projects/2019_pilotplant/05_mapping2DB/bacteriaDBmapping_rmk202/finalmapping/all_Bacteria_allSamples.txt /home/vincent/Desktop/Projects/2019_Pilotplan/09_annoation/all_Bacteria_allSamples_rmk202.txt


```



```{r}


source("/home/vincent/Desktop/ScriptRepository/R_functions-master/R_functions/g_legend.R") ##for plotting only legends
library(readr)
library(ggplot2)
library(dplyr)
library(tidyverse)
library(grid)
library(gridExtra)
library(cowplot) ## for arranging plots
library(stringr) ##for split string
library(rlist) ##rbind all dataframe of list
library(viridis) ##colour palette
##===================================
#-------------file import


  read_count_bac <- read_delim("/home/vincent/Desktop/Projects/2019_Pilotplan/09_annoation/all_Bacteria_allSamples_rmk202.txt","\t", escape_double = FALSE, col_names = c("chr","type","sort","start","end","unknown2","orientation","unknown","description","count","length_mapped","geneLength","covered","sample"),trim_ws = TRUE)

table(read_count_bac$chr)

read_count_bac$species <- ifelse(read_count_bac$chr=="L_delbrueckii_plasmid_RMK202_01","L_delbrueckii_plasmid_RMK202_01",ifelse(
    read_count_bac$chr=="L_delbrueckii_RMK202","L_delbrueckii_RMK202",ifelse(
      read_count_bac$chr=="S_thermophilus_RMK202","S_thermophilus_RMK202","L_plasmid_RMK202"
    )
))

table(read_count_bac$species)

read_count_bac$geneCoverage  <- (read_count_bac$count*600)/read_count_bac$geneLength

ggplot(read_count_bac,aes(y=geneCoverage))+geom_boxplot()+facet_grid(sample~species, scales="free")

library(dplyr)


bac_final <- read_count_bac %>% 
  dplyr::group_by(sample,species) %>% 
  dplyr::summarize(median = median(geneCoverage))


```


#####bring together
can only run with previous two chapters.

```{r}
library(dplyr)
all_final <- rbind(phage_final,bac_final)
# all_final <- rbind(phage_final,bac_final)


  total_samples_sumTreatment <- aggregate(. ~sample, data=all_final[,c("sample","median")], sum, na.rm=TRUE)

all_final$total_coverage <- total_samples_sumTreatment[match(all_final$sample,total_samples_sumTreatment$sample),"median"]
all_final$percent_coverage <- 100*(all_final$median/all_final$total_coverage)

all_final$species <- as.factor(all_final$species)

levels(all_final$species)

all_final$species <- factor(all_final$species, levels=rev(c("S_thermophilus_RMK202","S_phage_RMK202","L_delbrueckii_RMK202","L_delbrueckii_plasmid_RMK202_01","L_plasmid_RMK202","L_phage_RMK202_01","L_phage_RMK202_02")))

all_final$sample <- as.factor(all_final$sample)

levels(all_final$sample)

all_final$sample <- factor(all_final$sample, levels=(c("lyo202_96","Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202")))


# table(total_samples$phage) %>% length()

write.table(all_final,"/home/vincent/Desktop/Projects/2019_Pilotplan/coverage_rmk202.tsv",sep = "\t",quote = FALSE,col.names = FALSE)


#Sterm_colours <- c("#FAA0A0","#FAA0A0")
all_colours <- rev(c("#EB4D4D","#FAA0A0","#10B552","#36E37B","#6CF5A3","#C5F6FA","#99E9F2") ) 
# 
# mito_colours <- c("#AFBACC","#BE99AB") # sequential_hcl(5,palette="Grays")[c(4,3)]
# Sterm_colours <- c("#FCC2C2","#EB4D4D")# c("#FAA0A0","#EB4D4D") #c("#FF7D87","#E71D32")    
# Ldel_colours <- c("#6CF5A3","#10B552")  # c("#8CD211","#4C8400")   # c("#5AA700","#2D660A") #brewer.pal(9,name="YlGnBu")[c(4,6)] #sequential_hcl(5,palette="Purples 3")[c(3,2)]
# lactococcus_colours <- c("#5AAAFA","#4178BE")  #sequential_hcl(5,palette="Terrain 2")[c(3,2)]
# rest_colours <-  c("#C5F6FA","#99E9F2","#66D9E8","#3BC9DB") #brewer.pal(9,name="YlOrBr")[c(5,7,8,9)] #sequential_hcl(5,palette="BluGrn")[2:5]
# colours_phages_02 <- c(rest_colours,lactococcus_colours,Ldel_colours,Sterm_colours,mito_colours)


##----------------plot

PrelAbundance <-  ggplot( data = all_final,aes(y = percent_coverage, x = sample, group=interaction(species),fill = species))+ geom_bar( stat="identity")+
    labs("",
         x="",
         y="relative abundance")+
    theme_classic()+
   # scale_color_viridis(discrete=TRUE)+
   scale_fill_manual(values=all_colours)+
  # scale_fill_viridis(discrete=TRUE)+
    theme(#axis.text.x = element_text(angle = 75, hjust = 1,size=9),
        axis.text.x = element_blank(),
          legend.position="right",
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank()
          )

  PrelAbundance
  svg("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.svg",width=7,height=4)
   # png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 1000, height = 800,res=300)

PrelAbundance

dev.off()

#  svg("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.svg",width=7,height=4)
png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 2000, height = 1000,res=300)

PrelAbundance

dev.off()


##----------------amount of Streptococci and Lactobacilli


all_final$genus <- ifelse(grepl("S_",all_final$species),"Streptococcus","Lactobacillus")

all_final %>% 
  group_by(sample,genus) %>% 
  dplyr::summarize(sum = sum(percent_coverage))  %>% 
   group_by(genus) %>% 
  dplyr::summarize(min = min(sum),
            max = max(sum))  

##----------------amount of Lactobacillus delbrueckii RMK202


all_final %>% 
  group_by(species) %>% 
  dplyr::summarize(min = min(percent_coverage),
            max = max(percent_coverage))  


##----------------coverage of Lactobacillus delbrueckii RMK202
table(all_final$species)

all_final[which(all_final$species=="L_delbrueckii_RMK202"),]

```


## SNP calling freebayes (K1)

This is a good (blog)[https://bioinformatics-core-shared-training.github.io/cruk-summer-school-2017/Day2/vcf-intro.nb.html] on how to use freebayes and how to parse the output. 
Finally I will do freebayes on all samples simultaniously. 

Interesting output from column 11 on:

GT	Genotype v :1
DP	Read Depth  :1040
DPR	Number of observation for each allele  : 1,1037
RO	Reference allele observation count :1
QR	Sum of quality of the reference observations :10
AO	Alternate allele observation count :1037
QA	Sum of quality of the alternate observations :36917
GL	Genotype Likelihood, log10-scaled likelihoods of the data given the called genotype for each possible genotype generated from the reference and alternate alleles given the sample ploidy

```{bash}



###------------------adding readgroup to bam
rm /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrains/log/multiqc_logfile_bams.txt
  for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt)
do


java -jar /home/vincent/apps/picard/picard.jar AddOrReplaceReadGroups \
       I=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrains/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam \
       O=/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrains/${names}/bwaMapping2DB/${names}_mapping2ref.bam \
       RGID=${names} \
       RGLB=lib1 \
       RGPL=illumina \
       RGPU=unit1 \
       RGSM=${names}

done
rm /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata/log/multiqc_logfile_bams.txt

  for names in $(cat /archiv/Projects/2019_RMK202_analysis/01_log/names_extended.txt)
do
echo "/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrains/"${names}"/bwaMapping2DB/"${names}"_mapping2ref.bam " >> /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrains/log/multiqc_logfile_bams.txt

done

##remove trailing whitespaces

 sed -i 's/[ \t]*$//' /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrains/log/multiqc_logfile_bams.txt

##-------------calling
rm -r /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrains/freebayesOuput_WithONT
mkdir -p /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrains/freebayesOuput_WithONT


    freebayes -F 0.05 -C 2 --min-coverage 30 -p 1 \
      --bam-list /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrains/log/multiqc_logfile_bams.txt -f \
      /archiv/Projects/2019_RMK202_analysis/02_flye_Assembly/merged/RMK202_merged.fasta > \
      /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrains/freebayesOuput_WithONT/variantCallsfreebayes_all_f_005_min_20x.vcf
  
##-------------zipping

#bgzip -c /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final/freebayesOuput_WithONT/variantCallsfreebayes_all.vcf > /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final/freebayesOuput_WithONT/variantCallsfreebayes_all.vcf.gz
#tabix -p  vcf /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final/freebayesOuput_WithONT/variantCallsfreebayes_all.vcf.gz

##-------------zipping

grep "^##" -v /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrains/freebayesOuput_WithONT/variantCallsfreebayes_all_f_005_min_20x.vcf >  /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrains/freebayesOuput_WithONT/variantCallsfreebayes_all_f_005_min_20x_prepforR.vcf


##------------bring local




mkdir -p /home/vincent/Desktop/Projects/2019_RMK202_analysis/06_snpCalling2ONT

scp vincent@130.223.51.151:/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrains/freebayesOuput_WithONT/variantCallsfreebayes_all_f_005_min_20x_prepforR.vcf /home/vincent/Desktop/Projects/2019_RMK202_analysis/06_snpCalling2ONT/


```


######snpEff

Here we do [snpEff](http://snpeff.sourceforge.net/SnpEff_manual.html#buildAddConfig).


```{bash}

##----------------
##bring PGAP LOCAL
##----------------

mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP

scp -r /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/LD_singu vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/

scp -r /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/

##=======================================================================================================================
##-----------------------------------------------Streptococcus thermophilus----------------------------------------------
##=======================================================================================================================
##==================
##Building Database
##==================
#/archiv/Projects/2018_Culturomics/genomes/Lactobacillus_delbrueckii_RMK202.fna
#/archiv/Projects/2018_Culturomics/genomes/Streptococcus_thermophilus_RMK202.fna

##-------------
##01_add genome name to
##-------------

vim /home/vincent/miniconda2/share/snpeff-4.3.1t-2/snpEff.config
Streptococcus_thermophilus_RMK202_pgap.genome : Streptococcus_thermophilus_RMK202_pgap
Lactobacillus_delbrueckii_RMK202_pgap.genome : Lactobacillus_delbrueckii_RMK202_pgap
Streptococcus_phage_rmk202.genome : Streptococcus_phage_rmk202

##-------------
#02_put genome fasta and gff into right directory:
##-------------

sampleShort=Streptococcus_thermophilus_RMK202

##-------------
#03_put genome fasta 
##-------------
rm -r /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}_pgap
mkdir -p /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}_pgap

cat  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202.fasta  > /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}_pgap/sequences.fa



##-------------
#04 genes gff 
##-------------

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pgap.gff > /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}_pgap//genes.gff


##-------------
##05_build database
##-------------

cd /home/vincent/miniconda2/share/snpeff-4.3.1t-2/
java -jar snpEff.jar build -gff3 -v ${sampleShort}_pgap


##==================
##running snpEff
##==================

##-------------
##06_preperation of SNV file
##-------------

grep "S_thermophilus_RMK202" /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrains/freebayesOuput_WithONT/variantCallsfreebayes_all_f_005_min_20x.vcf > /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrains/freebayesOuput_WithONT/variantCallsfreebayes_all_f_005_min_20x_sterm.vcf

##-------------
##07 SNveff calling
##-------------

cd /home/vincent/miniconda2/share/snpeff-4.3.1t-2/
rm -r /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrain/SnpEff/${sampleShort}_pgap/
mkdir -p /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrain/SnpEff/${sampleShort}_pgap/


java -Xmx64g -jar snpEff.jar ${sampleShort}_pgap  /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrains/freebayesOuput_WithONT/variantCallsfreebayes_all_f_005_min_20x_sterm.vcf > /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrain/SnpEff/${sampleShort}_pgap/variantCallsfreebayes_all_${sampleShort}_snpEff.vcf 


##-------------
##08_prep for R
##-------------

grep "^##" -v /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrain/SnpEff/${sampleShort}_pgap/variantCallsfreebayes_all_${sampleShort}_snpEff.vcf | \
  awk -F "\t" 'BEGIN{OFS="\t"} {print $1,$2,$3,$4,$5,$6,$7,$8}' > /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrain/SnpEff/${sampleShort}_pgap/variantCallsfreebayes_all_${sampleShort}_snpEff_forR.vcf


##==================
##transfer local 
##==================


sampleShort=Streptococcus_thermophilus_RMK202


##-------------
###snpEff
##-------------

scp vincent@130.223.51.151:/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrain/SnpEff/${sampleShort}_pgap/variantCallsfreebayes_all_${sampleShort}_snpEff_forR.vcf ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new/variantCallsfreebayes_all_${sampleShort}_snpEff_02.vcf 

##-------------
###eggnog
##-------------

scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/eggnog/${sampleShort}_pgap_query_seqs.fa.emapper.annotations  ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new/variantCallsfreebayes_all_${sampleShort}_eggnog.vcf 





##=======================================================================================================================
##-----------------------------------------------Streptococcus thermophilus----------------------------------------------
##=======================================================================================================================
##==================
##Building Database
##==================
#/archiv/Projects/2018_Culturomics/genomes/Lactobacillus_delbrueckii_RMK202.fna
#/archiv/Projects/2018_Culturomics/genomes/Streptococcus_thermophilus_RMK202.fna

##-------------
##01_add genome name to
##-------------

vim /home/vincent/miniconda2/share/snpeff-4.3.1t-2/snpEff.config
Streptococcus_thermophilus_RMK202_pgap.genome : Streptococcus_thermophilus_RMK202_pgap
Lactobacillus_delbrueckii_RMK202_pgap.genome : Lactobacillus_delbrueckii_RMK202_pgap

##-------------
#02_put genome fasta and gff into right directory:
##-------------

sampleShort=Lactobacillus_delbrueckii_RMK202

##-------------
#03_put genome fasta 
##-------------
rm -r /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}_pgap
mkdir -p /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}_pgap

cat  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Ldel_RMK202.fasta  > /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}_pgap/sequences.fa



##-------------
#04 genes gff 
##-------------

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Lactobacillus_delbrueckii_RMK202_pgap.gff > /home/vincent/miniconda2/share/snpeff-4.3.1t-2/data/${sampleShort}_pgap//genes.gff


##-------------
##05_build database
##-------------

cd /home/vincent/miniconda2/share/snpeff-4.3.1t-2/
java -jar snpEff.jar build -gff3 -v ${sampleShort}_pgap


##==================
##running snpEff
##==================

##-------------
##06_preperation of SNV file
##-------------

grep "L_delbrueckii_RMK202" /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrains/freebayesOuput_WithONT/variantCallsfreebayes_all_f_005_min_20x.vcf > /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrains/freebayesOuput_WithONT/variantCallsfreebayes_all_f_005_min_20x_ldel.vcf

##-------------
##07 SNveff calling
##-------------

cd /home/vincent/miniconda2/share/snpeff-4.3.1t-2/
rm -r /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrain/SnpEff/${sampleShort}_pgap/
mkdir -p /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrain/SnpEff/${sampleShort}_pgap/


java -Xmx64g -jar snpEff.jar ${sampleShort}_pgap  /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrains/freebayesOuput_WithONT/variantCallsfreebayes_all_f_005_min_20x_ldel.vcf > /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrain/SnpEff/${sampleShort}_pgap/variantCallsfreebayes_all_${sampleShort}_snpEff.vcf 

##-------------
##08_prep for R
##-------------

grep "^##" -v /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrain/SnpEff/${sampleShort}_pgap/variantCallsfreebayes_all_${sampleShort}_snpEff.vcf | \
  awk -F "\t" 'BEGIN{OFS="\t"} {print $1,$2,$3,$4,$5,$6,$7,$8}' > /archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrain/SnpEff/${sampleShort}_pgap/variantCallsfreebayes_all_${sampleShort}_snpEff_forR.vcf


##==================
##transfer local 
##==================


#sampleShort=Streptococcus_thermophilus_RMK202
sampleShort=Lactobacillus_delbrueckii_RMK202

##-------------
###snpEff
##-------------

scp vincent@130.223.51.151:/archiv/Projects/2019_RMK202_analysis/03_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata_withStrain/SnpEff/${sampleShort}_pgap/variantCallsfreebayes_all_${sampleShort}_snpEff_forR.vcf ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new/variantCallsfreebayes_all_${sampleShort}_snpEff_02.vcf 

##-------------
###eggnog
##-------------

scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/eggnog/${sampleShort}_pgap_query_seqs.fa.emapper.annotations  ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new/variantCallsfreebayes_all_${sampleShort}_eggnog.vcf 


##=======================================================================================================================
##-----------------------------------------------merge species----------------------------------------------
##=======================================================================================================================

##-------------
###snpEff
##-------------
cat ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new/variantCallsfreebayes_all_S*_snpEff_02.vcf  \
  ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new/variantCallsfreebayes_all_L*_snpEff_02.vcf > \
  ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new/variantCallsfreebayes_both_snpEff.vcf


##-------------
###eggnog
##-------------

grep -v "^#" ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new/variantCallsfreebayes_all_S*_eggnog.vcf | sed 's/gnl|extdb|//g'|awk -F " " 'BEGIN{OFS="_"}{print "S_thermophilus_RMK202",$0}' >  ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new/variantCallsfreebayes_all_both_eggnog.vcf
grep -v "^#" ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new/variantCallsfreebayes_all_L*_eggnog.vcf | sed 's/gnl|extdb|//g'|awk -F " " 'BEGIN{OFS="_"}{print "L_delbrueckii_RMK202",$0}' >>  ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new/variantCallsfreebayes_all_both_eggnog.vcf


scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/eggnog/${sampleShort}_pgap_query_seqs.fa.emapper.annotations  ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new/variantCallsfreebayes_all_${sampleShort}_eggnog.vcf 


```

here we merge all annotation knowledge we have of the SNPs:

1. SNPeffect
2. eggnog annotation
3. repeat annotation
4. core genes

Thereafter we have a long list of information for every SNP. 

#####repeat genes

We could potentially have a problem of alignement in genes highly similiar. Therefore I look if which genes are cluster together. 

```{bash}

##=======================================================================================================================
##-----------------------------------------------Streptococcus thermophilus----------------------------------------------
##=======================================================================================================================
  
awk -F "[\t;]" 'BEGIN{OFS="\t"} {if($3=="CDS") print $1, $4,$5,$9}' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/annot.gff |sed 's/ID=cds-//g' > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/annot_cds.gff

#bedtools index /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202.fasta

bedtools getfasta -name -fi /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202.fasta -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/annot_cds.gff > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pgap_cds.fna

##==================
##cd-hit
##==================
  


   cd-hit -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pgap_cds.fna \
    -o /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/sterm_cdsClustering.txt -c 0.85 -M 30000 -T 37 -sc 1 -sf 1
  
  
  
  
  ###multifasta2fasta
  cd /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/splitClusters
  awk 'NR%1==0{ print > "Cluster_"++i".txt" }' RS='>Cl' /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/sterm_cdsClustering.txt.clstr

  ##prepare log file
  #cluster_2.txt till cluster_24.txt
  
  ls /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/splitClusters |sort -V |head -24 |tail -23 > /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/splitClusters/files_of_INterest.txt
     
   ##get gene names
   num=2
   rm /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/bigCluster/all_repeatingGenes.txt
   mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/bigCluster/
   i=Cluster_2.txt
   for i in $(cat /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/splitClusters/files_of_INterest.txt)
    do
   grep -v "^uster" /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/splitClusters/${i} |awk -F ">" '{print $2}' |awk -F "... at " 'BEGIN{OFS="\t"}{print $1,$2}' |sed 's/... \*//g' |sed '$ d' > \
    /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/bigCluster/names_${i}
    
   grep -v "^uster" /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/splitClusters/${i} |awk -F ">" '{print $2}'  |awk -F "... at " -v count="$num" 'BEGIN{OFS="\t"}{print $1,$2,count}' |sed 's/... \*//g' |sed '$ d' >> \
    /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/bigCluster/all_repeatingGenes.txt
      
    num=$((num+1))
    
    
    done
 

    ##get gff
    
    genes=pgaptmp_001292
    mkdir -p  /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/bigCluster_genes/
    for i in $(cat /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/splitClusters/files_of_INterest.txt)
    do
    rm  /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/bigCluster_genes/genes_${i}
    for genes in $(cut -f 1 /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/bigCluster/names_${i})
      do
        
        grep "cds-${genes};" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202_pgap.gff >> \
          /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/bigCluster_genes/genes_${i}
       
        
      done # $genes   
    done   # $i
    
    
    
  mkdir -p ~/Desktop/Projects/2019_Pilotplan/09_annoation/repeatingGenes


scp vincent@130.223.51.151:/archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/bigCluster/all_repeatingGenes.txt ~/Desktop/Projects/2019_Pilotplan/09_annoation/repeatingGenes/

##=======================================================================================================================
##-----------------------------------------------Lactobacillus delbrueckii----------------------------------------------
##=======================================================================================================================


  
awk -F "[\t;]" 'BEGIN{OFS="\t"} {if($3=="CDS") print $1, $4,$5,$9}' /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/annot.gff |sed 's/ID=cds-//g' > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/annot_cds.gff

#bedtools index /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/Streptococcus_thermophilus_RMK202.fasta

bedtools getfasta -name -fi /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Ldel_RMK202.fasta -bed /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/annot_cds.gff > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Ldel_RMK202_pgap_cds.fna

##==================
##cd-hit
##==================
  


   cd-hit -i /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/LD_singu/Ldel_RMK202_pgap_cds.fna \
    -o /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/ldel_cdsClustering.txt -c 0.85 -M 30000 -T 37 -sc 1 -sf 1
  
  
  
  
  ###multifasta2fasta
  mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/splitClusters_ldel
  cd /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/splitClusters_ldel
  awk 'NR%1==0{ print > "Cluster_"++i".txt" }' RS='>Cl' /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/ldel_cdsClustering.txt.clstr

  ##prepare log file
  #cluster_2.txt till cluster_24.txt
  
  ls /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/splitClusters |sort -V |head -43  > /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/splitClusters_ldel/files_of_INterest.txt
     
   ##get gene names
   num=2
   rm /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/splitClusters_ldel/bigCluster_ldel/all_repeatingGenes.txt
   mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/splitClusters_ldel/bigCluster_ldel/
   i=Cluster_2.txt
   for i in $(cat /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/splitClusters_ldel/files_of_INterest.txt)
    do
   grep -v "^uster" /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/splitClusters_ldel/${i} |awk -F ">" '{print $2}' |awk -F "... at " 'BEGIN{OFS="\t"}{print $1,$2}' |sed 's/... \*//g' |sed '$ d' > \
    /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/splitClusters_ldel/bigCluster_ldel/names_${i}
    
   grep -v "^uster" /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/splitClusters_ldel/${i} |awk -F ">" '{print $2}'  |awk -F "... at " -v count="$num" 'BEGIN{OFS="\t"}{print $1,$2,count}' |sed 's/... \*//g' |sed '$ d' >> \
    /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/splitClusters_ldel/bigCluster_ldel/all_repeatingGenes.txt
      
    num=$((num+1))
    
    
    done
 
 
##=======================================================================================================================
##-----------------------------------------------merge and transfer----------------------------------------------
##=======================================================================================================================


awk -F " " 'BEGIN{OFS="_"}{print "S_thermophilus_RMK202",$0}' /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/bigCluster/all_repeatingGenes.txt > /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/bigCluster/all_repeatingGenes_both.txt 
    
awk -F " " 'BEGIN{OFS="_"}{print "L_delbrueckii_RMK202",$0}' /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/splitClusters_ldel/bigCluster_ldel/all_repeatingGenes.txt >> /archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/bigCluster/all_repeatingGenes_both.txt 

##-------------------
##local
##-------------------

mkdir -p ~/Desktop/Projects/2019_Pilotplan/09_annoation/repeatingGenes

scp vincent@130.223.51.151:/archiv/Projects/2019_pilotplant/02_annotation/CDS_clustering/bigCluster/all_repeatingGenes_both.txt  ~/Desktop/Projects/2019_Pilotplan/09_annoation/repeatingGenes/all_repeatingGenes_both.txt


```


#####Core genes prep

```{bash}

rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coreGenes_rmk202_both.txt

##=======================================================================================================================
##-----------------------------------------------Streptococcus thermophilus----------------------------------------------
##=======================================================================================================================
species=Sterm

##-------extract OG names
#rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/Orthogroups/test.txt
rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Oct21/Orthogroups/OG_names_${species}.txt
for OG in $(cat /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Oct21/Orthogroups/Orthogroups_SingleCopyOrthologues.txt)
do

grep "$OG" /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Oct21/Orthogroups/Orthogroups.txt | sed 's/gnl|extdb|//g'| awk -F " " 'BEGIN{OFS="\t"}{print $1,"S_thermophilus_RMK202_"$4}'  >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Oct21/Orthogroups/OG_names_${species}.txt


done #OG

cat /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Oct21/Orthogroups/OG_names_${species}.txt >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coreGenes_rmk202_both.txt


##=======================================================================================================================
##-----------------------------------------------Lactobacillus delbrueckii----------------------------------------------
##=======================================================================================================================
species=Ldel

##-------extract OG names
#rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/Orthogroups/test.txt
rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Oct21/Orthogroups/OG_names_${species}.txt
for OG in $(cat /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Oct21/Orthogroups/Orthogroups_SingleCopyOrthologues.txt)
do

grep "$OG" /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Oct21/Orthogroups/Orthogroups.txt| sed 's/gnl|extdb|//g' |awk -F " " 'BEGIN{OFS="\t"}{print $1,"	L_delbrueckii_RMK202_"$11}'  >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Oct21/Orthogroups/OG_names_${species}.txt


done #OG

cat /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/${species}/orthofinder/Results_Oct21/Orthogroups/OG_names_${species}.txt >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coreGenes_rmk202_both.txt



##=======================================================================================================================
##-----------------------------------------------Lactobacillus delbrueckii----------------------------------------------
##=======================================================================================================================

mkdir -p ~/Desktop/Projects/2019_Pilotplan/09_annoation/coreGenes

scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coreGenes_rmk202_both.txt  ~/Desktop/Projects/2019_Pilotplan/09_annoation/coreGenes/coreGenes_rmk202_both.txt


```

#####merge all infomration

```{r}
library(readr)
library(stringr)

##==================
##01-SNPeff
##==================
RMK202_both_snpEff <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new/variantCallsfreebayes_both_snpEff.vcf",  "\t", escape_double = FALSE, col_names = c("X.CHROM","POS","unknown","REF","ALT","quality","uknown2","snpeff"), trim_ws = TRUE)

RMK202_both_snpEff_02 <- subset(RMK202_both_snpEff, select=-c(unknown,uknown2))
RMK202_both_snpEff_02$site <- paste(RMK202_both_snpEff_02$X.CHROM,RMK202_both_snpEff_02$POS,RMK202_both_snpEff_02$REF,RMK202_both_snpEff_02$ALT,sep="_")

tmp_snpeff <-  data.frame(str_split_fixed(RMK202_both_snpEff_02$snpeff, fixed("|"), 136)[,2:8])
colnames(tmp_snpeff) <- c("effect","significance","geneName","PGAP_name","type","typeName","typeGene")

RMK202_both_snpEff_02 <- subset(RMK202_both_snpEff_02, select=-snpeff)
RMK202_both_SNPeff_final <-  cbind(RMK202_both_snpEff_02,tmp_snpeff)
# table(S_thermophilus_snpEff_03$typeGene)
# 
# S_thermophilus_snpEff_03[which(S_thermophilus_snpEff_03$typeGene=="pseudogene"),]
# table(S_thermophilus_snpEff_03[which(S_thermophilus_snpEff_03$typeGene=="protein_coding"),]$type)
# S_thermophilus_snpEff_03[which(S_thermophilus_snpEff_03$typeGene=="rRNA"),]

RMK202_both_SNPeff_final$finalName <- RMK202_both_SNPeff_final$typeName %>% gsub("rna-","",.) %>%  gsub("TRANSCRIPT_gene-","",.) #%>%  gsub("pgaptmp_","pgaptmp ",.) 
RMK202_both_SNPeff_final$finalName <- paste(RMK202_both_SNPeff_final$X.CHROM,RMK202_both_SNPeff_final$finalName,sep = "_")
# length(grep("pgaptmp_",S_thermophilus_snpEff_03$finalName,invert = TRUE))
table(RMK202_both_SNPeff_final$X.CHROM)

##==================
##02 eggnog
##==================
RMK202_both_eggnog <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new/variantCallsfreebayes_all_both_eggnog.vcf",  "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description"))
# RMK202_both_eggnog[RMK202_both_eggnog$query_name %in% SNPeff_final$finalName,]

##==================
##03 repeats
##==================

RMK202_both_repeats <- read_delim("~/Desktop/Projects/2019_Pilotplan/09_annoation/repeatingGenes/all_repeatingGenes_both.txt",  "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("geneNAME","Repeat_identity","Repeat_cluster")) #%>% slice( 1:(n()-3))
# nrow(RMK202_both_repeats)

##==================
##04_core genes
##==================

RMK202_both_coreGenes <- read_delim("~/Desktop/Projects/2019_Pilotplan/09_annoation/coreGenes/coreGenes_rmk202_both.txt",  "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("orthogroup","geneName")) %>% add_column(.,core="core")

##==================
##05_merge
##==================
nrow(RMK202_both_SNPeff_final)
RMK202_snpeff_eggnog <- merge(RMK202_both_SNPeff_final, RMK202_both_eggnog, by.x = "finalName", by.y = "query_name",all.x = TRUE)
RMK202_snpeff_eggnog_repeats <- merge(RMK202_snpeff_eggnog, RMK202_both_repeats, by.x = "finalName", by.y = "geneNAME",all.x = TRUE)
RMK202_snpeff_eggnog_repeats_core <- merge(RMK202_snpeff_eggnog_repeats, RMK202_both_coreGenes, by.x = "finalName", by.y = "geneName",all.x = TRUE)
nrow(RMK202_snpeff_eggnog_repeats_core)

write.csv(RMK202_snpeff_eggnog_repeats_core,"~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new/RMK202_snpeff_eggnog_repeats_core.txt")

```

## analysis snps

The coverage of the alternative allel must be at least 3 or zero. 

We have to be careful when removing low quality SNP calls. As they seem to be not not low quality in the sense of low coverage but rather suported by only a fraction of samples. This makes sense for many snps which are only valied for Sterm and have NAs in the Ldel genomes. Therefore we rather select the SNP by selecting for a minimum of coverage supporting the SNP per sample. 

!!IMPORTANT!!
Before starting we have to add the following to the SNP.vcf:

1. SNPeff
2. if core or not with roary and interesect
3. (if repeat within repeat)

###### first Analysis

```{r}
library(tidyverse)
library(vcfR)
library(readr)
library(plyr)
library(dplyr)
library(ggplot2)
##==================
##01_import data
##==================
snps_freebayes <- read_delim("/home/vincent/Desktop/Projects/2019_RMK202_analysis/06_snpCalling2ONT/variantCallsfreebayes_all_f_005_min_20x_prepforR.vcf", "\t", escape_double = FALSE, trim_ws = TRUE)

# snps_freebayes[snps_freebayes$QUAL<=20,]
# snps_freebayes <- snps_freebayes[snps_freebayes$QUAL>20,]
# dim(snps_freebayes_new)
# dim(snps_freebayes)
snps_freebayes <- subset(snps_freebayes, select=-c(di_K1_10h,th_K1_8h))



##==================
##02_long list and frequency calc
##==================

table(snps_freebayes$`#CHROM`)
##wide2long

 # snps_freebayes[,10]
snps_freebayes_long <- gather(snps_freebayes, sample, Snps, colnames(snps_freebayes)[10]:colnames(snps_freebayes)[ncol(snps_freebayes)], factor_key=TRUE,na.rm = TRUE) 
nrow(snps_freebayes_long)
table(snps_freebayes_long$sample)
###------------------------remove non-called snps
snps_freebayes_long_cleaned <- snps_freebayes_long[snps_freebayes_long$Snps!=".",]
nrow(snps_freebayes_long_cleaned)
# snps_freebayes_long[c(7438,7437,7439, 7440, 7441, 7442, 7443, 7444, 7445),]

###------------------------calculate allel frequency   

sample_relative_temp <-snps_freebayes_long_cleaned %>% separate("Snps", into=c("GT","DP","AD","RO","QR","AO","QA","GL"), sep = ":") %>% transform(.,allel_frequency=as.numeric(AO)/as.numeric(DP))

sample_relative_temp_withONT <- sample_relative_temp
sample_relative_temp_withONT$site <- paste(sample_relative_temp_withONT$X.CHROM,sample_relative_temp_withONT$POS,sample_relative_temp_withONT$REF,sample_relative_temp_withONT$ALT,sep="_")


##==================
##03_filter
##==================

##remove SNPs that have more than two allel frequencies
nrow(sample_relative_temp)
nas <- sample_relative_temp[which(is.na(sample_relative_temp$allel_frequency)),]
nrow(nas)
sample_relative_temp_cleaned <- sample_relative_temp[which(!is.na(sample_relative_temp$allel_frequency)),]

##remove low coverage (>40x)
cutoff <- 30
sample_relative_temp_cleaned$DP <- as.numeric(sample_relative_temp_cleaned$DP)
nrow(sample_relative_temp_cleaned)
sample_relative_safe <- sample_relative_temp_cleaned[which(sample_relative_temp_cleaned$DP>cutoff),]
nrow(sample_relative_safe)

##make all allele frequencies smaller than 0.05 to NA
# sample_relative_safe[sample_relative_safe$allel_frequency<0.05,"allel_frequency"] <- NA

#is.na(sample_relative_safe$allel_frequency)

##remove snps with less than 2x alternative allel frequency or zero
sample_relative_safe$AO <- as.numeric(sample_relative_safe$AO)
# nrow(sample_relative_safe[sample_relative_safe$AO<=2,])
# nrow(sample_relative_safe[sample_relative_safe$AO==0 | sample_relative_safe$AO>2,])
nrow(sample_relative_safe[sample_relative_safe$AO>0 & sample_relative_safe$AO<=2,])
sample_relative_safe <- (sample_relative_safe[sample_relative_safe$AO==0 | sample_relative_safe$AO>2,])
nrow(sample_relative_safe)


##==================
##04_preperation and make wide
##==================

sample_relative_temp_cleaned <- sample_relative_safe
#ggplot(sample_relative_temp_cleaned, aes(y=DP))+geom_boxplot()+ylim(c(0,1000))


##make site name
sample_relative_temp_cleaned$site <- paste(sample_relative_temp_cleaned$X.CHROM,sample_relative_temp_cleaned$POS,sample_relative_temp_cleaned$REF,sample_relative_temp_cleaned$ALT,sep="_")


###spread the dataframe again

nrow(sample_relative_temp_cleaned)

sample_relative_safe_final_prep <- sample_relative_temp_cleaned[,c("X.CHROM","POS","site","allel_frequency","sample")]
#nrow(sample_relative_safe_final_prep[grep("S_thermophilus_RMK202",sample_relative_safe_final_prep$site),])
sample_relative_safe_final_tsne <- spread(sample_relative_safe_final_prep, sample, allel_frequency)
nrow(sample_relative_safe_final_tsne)

##==================
##05_filter rowsums!=0
##==================

##remove row sums ==0
nrow(sample_relative_safe_final_tsne)
sum(rowSums(sample_relative_safe_final_tsne[,4:ncol(sample_relative_safe_final_tsne)],na.rm = TRUE)==0)


sample_relative_wide <- sample_relative_safe_final_tsne[rowSums(sample_relative_safe_final_tsne[,4:ncol(sample_relative_safe_final_tsne)],na.rm = TRUE)>0,]
nrow(sample_relative_wide)
sample_relative_wide <- subset(sample_relative_wide, select=-POS)
sample_relative_wide$species <- sample_relative_wide$X.CHROM
sample_relative_wide$species <- revalue(sample_relative_wide$species, c("L_delbrueckii_RMK202"="L. delbrueckii","S_thermophilus_RMK202"="S. thermophilus"))
sample_relative_wide$culture <- "RMK202"

##==================
##06_add gene information
##==================

RMK202_snpeff_eggnog_repeats_core <- read_csv("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/new/RMK202_snpeff_eggnog_repeats_core.txt")
# nrow(RMK202_snpeff_eggnog_repeats_core)
# nrow(sample_relative_wide)
#RMK202_snpeff_eggnog_repeats_core_subset <- subset(RMK202_snpeff_eggnog_repeats_core, select=c(X.CHROM,site,finalName,effect,significance,geneName,GOs,EC,KEGG_ko,COG_Functional_Category,Repeat_identity,core))

sample_relative_wide_description <- merge(sample_relative_wide,RMK202_snpeff_eggnog_repeats_core, by = "site",all.x = TRUE)
# nrow(sample_relative_wide_description)

##---------------------------------subset for interesting columns
colnames(sample_relative_wide_description)
sample_relative_wide_description_interest <- subset(sample_relative_wide_description, select=c(X.CHROM.x,site,species,culture,finalName,effect,significance,geneName,COG_Functional_Category,Repeat_identity,core,L104,L108,L35,L44,L70,L71,L80,L99,S50,S72,Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202,lyo202_96))

##==================
##07 rename samples
##==================

colnames(sample_relative_wide_description_interest) <- revalue(colnames(sample_relative_wide_description_interest), c("S50"="mst1","S72"="mst2","L70"="mst3", "L44"="mst10", "L108"="mst7", "L99"="mst8", "L104"="mst6", "L80"="mst5", "L35"="mst9", "L71"="mst4","lyo202_96"="Lyo\n1996","Lyo_202_2012"="Lyo\n2012", "Lyo_202_2014"="Lyo\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018"))


##==================
##08_remove non-bacterial SNPs
##==================
table(sample_relative_wide_description_interest$X.CHROM.x)
nrow(sample_relative_wide_description_interest)
sample_relative_wide_description_interest_subset <- sample_relative_wide_description_interest[which(sample_relative_wide_description_interest$X.CHROM.x %in% c("L_delbrueckii_RMK202","S_thermophilus_RMK202")),]
nrow(sample_relative_wide_description_interest_subset)
table(sample_relative_wide_description_interest_subset$X.CHROM.x)
sample_relative_wide <- sample_relative_wide_description_interest_subset




##==================
##09_order variables properly
##==================


sample_relative_wide$significance <- factor(sample_relative_wide$significance, levels=c("MODIFIER","LOW","MODERATE","HIGH"))


##==================
##10_dN_dS ratio
##==================
###-----------prep for 
prep_for_dN_dS_wide <- sample_relative_wide[which(sample_relative_wide$significance!="MODIFIER"),] 

###-----------calculate dS/dN ratio

dN_dS_ratios <- data.frame()
for (gene in unique(prep_for_dN_dS_wide$geneName)) {
  
  dS <- prep_for_dN_dS_wide %>% filter(geneName==gene) %>% filter(significance=="LOW")
   dN <-  prep_for_dN_dS_wide %>% filter(geneName==gene) %>% filter(significance %in% c("MODERATE","HIGH"))
tmp <- data.frame(gene, "dS"=nrow(dS),"dN"=nrow(dN),"dN_dN+dS"=(nrow(dN))/(nrow(dS)+nrow(dN)),"dN_dS"=(nrow(dN))/(nrow(dS)))
dN_dS_ratios <- rbind(dN_dS_ratios,tmp)
}

plotDens <- ggplot(dN_dS_ratios,aes(x=dN_dS))+geom_histogram()+
  geom_vline(xintercept=mean(dN_dS_ratios$dN_dS))+
  theme_classic()+
  labs(x="dN/(dN+dS)",
       y="gene count")

plotDens
dN_dS_ratios <- arrange(dN_dS_ratios,desc(dN_dS)) 

##-------------------------------prep snpeff
annotation_prep <- RMK202_snpeff_eggnog_repeats_core %>% select(-c(X1,finalName,X.CHROM,POS,REF,ALT,quality,site,effect,significance)) %>% distinct()
dN_dS_ratios_final <- merge(dN_dS_ratios,annotation_prep,by.x="gene",by.y="geneName",all.x = TRUE)
dN_dS_ratios_final_02 <- arrange(dN_dS_ratios_final,desc(dN_dS)) 

##-------------------------------wirte to file

library(xlsx)

# write.xlsx(dN_dS_ratios_final_02,file="Users//Desktop/presenations/my_presentations/20191007_groupmeeting/figures/dN_dS_ratios_genes_rmk202.xls", sheetName = "Sheet1", 
  # col.names = TRUE, row.names = FALSE, append = FALSE)

##-------------------------------merge with sample_relative_wide
dN_dS_ratios$mutations <- dN_dS_ratios$dS+dN_dS_ratios$dN
dN_dS_ratios_prep <- dN_dS_ratios %>% select(gene,dN_dN.dS,mutations)
sample_relative_wide <- merge(sample_relative_wide,dN_dS_ratios_prep,by.y="gene",by.x="geneName",all.x = TRUE)

##==================
##11_make long
##==================
colnames(sample_relative_wide)
sample_relative_long <- gather(sample_relative_wide, sample, Snps, "mst6":"Lyo\n1996", factor_key=TRUE,na.rm = TRUE) 
nrow(sample_relative_long)
table(sample_relative_long$X.CHROM)


```


##### dN/dS ratio plots

```{r}

##-------------------------------plotting

png("~/Desktop/presenations/my_presentations/20191007_groupmeeting/figures/dN_dS_ratios_genes_density_rmk202.png",width=1300,height=1000,res=300)
plotDens
dev.off()
# prep_for_dN_dS_wide %>% filter(geneName=="pgaptmp_000898")
##------------
##prepare 
##------------
threshold_mutation <- 1

dN_dS_ratios$category <- ifelse((dN_dS_ratios$dN_dN.dS>mean(dN_dS_ratios$dN_dN.dS) &dN_dS_ratios$dN>threshold_mutation),"used","not used") 
table(dN_dS_ratios$category)
dN_dS_ratios$category
dN_dS_ratios$category = factor(dN_dS_ratios$category, levels=c("used","not used"))
colors <- c("red","grey")

##------------
##plot dN/dS to mutation number
##------------
plotpoint <- ggplot(dN_dS_ratios,aes(x=dN_dN.dS,y=dN,group=category,color=category,fill=category))+geom_point()+
  geom_vline(xintercept=mean(dN_dS_ratios$dN_dN.dS))+
  geom_hline(yintercept=threshold_mutation)+
  scale_fill_manual(values=colors)+
  scale_color_manual(values=colors)+
  theme_classic()+
  annotate(geom="text", x=0.9, y=0.9*(max(dN_dS_ratios$dN)), label=paste0("used:",sum(dN_dS_ratios$category=="used")), color="red")+
  annotate(geom="text", x=0.1, y=0.9*(max(dN_dS_ratios$dN)), label=paste0("not used:",sum(dN_dS_ratios$category=="not used")), color="grey")+
  theme(legend.title = element_blank())+
  # coord_trans( y="log2")+
  labs(x="dN/(dN+dS)",
       y="dN")
plotpoint


plotDens_x <- ggplot(dN_dS_ratios,aes(x=dN_dN.dS,group=category,fill=category))+geom_histogram()+
  geom_vline(xintercept=mean(dN_dS_ratios$dN_dN.dS))+
  scale_fill_manual(values=colors)+
  # scale_color_manual(values=colors)+
  theme_classic()+
  theme(legend.position="none")+
  labs(x="",
       y="gene count")
plotDens_x

plotDens_y <- ggplot(dN_dS_ratios,aes(x=dN,group=category,fill=category))+geom_histogram()+
  geom_vline(xintercept=threshold_mutation)+
  scale_fill_manual(values=colors)+
  coord_flip()+
  # scale_color_manual(values=colors)+
  theme_classic()+
  theme(legend.position="none")+
  labs(x="",
       y="gene count")
plotDens_y



##-------put together
source("/home/vincent/Desktop/ScriptRepository/R_functions-master/R_functions/g_legend.R")

 legend <- g_legend(plotpoint)
# blank <- grid.rect(gp=gpar(col="white"))


grid.arrange(plotDens_x,legend,plotpoint+theme(legend.position="none"),plotDens_y,
        ncol=2, nrow=2, widths=c(4, 2), heights=c(2,4))

svg("~/Desktop/Manuscripts/2019_RMK202/Figures/dN_dS_ratio.svg",width=6,height=5)
# png("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_sterm.png",width=1800,height=1600,res=300)


grid.arrange(plotDens_x,legend,plotpoint+theme(legend.position="none"),plotDens_y,
        ncol=2, nrow=2, widths=c(4, 2), heights=c(2,4))

dev.off()


```

######snpeff of snps

for this I have to first run the snpEff chunk. 

```{r}

##----------------
##significance
##----------------
ggplot(sample_relative_wide,aes(x=significance,fill=significance))+geom_bar()+
  facet_grid(species~.,scales = "free")+
  theme_classic()+
  xlab("") + 
  ylab("variant sites") +
  scale_fill_manual(values=c("grey","gold1","orange","red"))+   #wes_palette(n=3, name="Moonrise3")[1:3]))+
  theme(legend.position = "none",
        plot.margin=unit(c(0,0,0,2.5),"cm"),
        axis.text.x=element_text(angle=45, hjust=1,size=9))

##----------------
##effect of high significant mutations
##----------------
annotation_snps_high <- sample_relative_wide[sample_relative_wide$significance=="HIGH",]
table(annotation_snps_high$typeGene)
ggplot(annotation_snps_high,aes(x=effect))+geom_bar()+
  facet_grid(species~.,scales = "free")+
  theme_classic()+
theme(legend.position = "none",
        plot.margin=unit(c(0,0,0,2.5),"cm"),
        axis.text.x=element_text(angle=45, hjust=1,size=9))
##----------------
##--------non-modifier mutations
##----------------
annotation_snps_nonModifier <- sample_relative_wide[sample_relative_wide$significance!="MODIFIER",]
# table(annotation_sterm_snps_nonModifier$significance)
##----------------
##--------COG effect of non-modifier mutations
##----------------
# annotation_snps_nonModifier$COG_Functional_Category
ggplot(annotation_snps_nonModifier,aes(x=COG_Functional_Category))+geom_bar()+
  facet_grid(species~.,scales = "free")+
  theme_classic()+
theme(legend.position = "none",
        plot.margin=unit(c(0,0,0,2.5),"cm"),
        axis.text.x=element_text(angle=45, hjust=1,size=9))



```

######remove isolates

```{r}


##----------------
##01_remove strains
##----------------
sample_relative_safe_woSTRAINS <- sample_relative_long[grep("^mst",sample_relative_long$sample,invert = TRUE),] 

sample_relative_safe_woSTRAINS$sample <- droplevels(sample_relative_safe_woSTRAINS$sample)
# sample_relative_safe_final==sample_relative_safe_woSTRAINS

##reorder samples

sample_relative_safe_woSTRAINS$sample = factor(sample_relative_safe_woSTRAINS$sample, levels=c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018"))
levels(sample_relative_safe_woSTRAINS$sample)



  ##============
### plot 
   ##============

all_colours <- rev(c("#EB4D4D","#10B552") ) 

colnames(sample_relative_safe_woSTRAINS)

##plot
  p2 <- ggplot(sample_relative_safe_woSTRAINS,aes(x=sample,y=Snps,group=site,color=species,fill=species, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nRepeat_identity:",Repeat_identity,"\ncore:",core,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.2, alpha=.1)+ 
    # facet_grid(day~kessel~species)+
    facet_grid(species~.)+
    # facet_grid(treatment~species)+
    # facet_grid(treatment~species)+
    labs("",
         x="time [h]",
         y="Alternative \nallele\n frequency")+
    #scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    #scale_y_continuous(limits = c(0,0.3))+
    theme_classic()+
   scale_fill_manual(values=all_colours)+
   scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
      #axis.text.x = element_blank(),
          legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          )
  
  p2
  
  png("~/Desktop/presenations/my_presentations/20190919_VUA/abundance_snps.png", width = 1900, height = 1200,res=300)

p2

dev.off()

 library(plotly)
# ggp <- ggplotly(p2)
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_bacteria_all.html")


  ##============
### subset :
##----01==only genic mutations
##----02==only core genes mutations
##----03==only snps above 0.05
   ##============

nrow(sample_relative_safe_woSTRAINS)
sample_relative_safe_woSTRAINS_sub01 <- sample_relative_safe_woSTRAINS[which(sample_relative_safe_woSTRAINS$significance!="MODIFIER"),]
nrow(sample_relative_safe_woSTRAINS_sub01)
sample_relative_safe_woSTRAINS_sub02 <- sample_relative_safe_woSTRAINS_sub01[which(sample_relative_safe_woSTRAINS_sub01$core=="core"),]
nrow(sample_relative_safe_woSTRAINS_sub02)
sample_relative_safe_woSTRAINS_sub03 <- sample_relative_safe_woSTRAINS_sub02[which(sample_relative_safe_woSTRAINS_sub02$Snps>0.05),]
nrow(sample_relative_safe_woSTRAINS_sub03)
table(sample_relative_safe_woSTRAINS_sub03$species)

  ##============
### plot 
   ##============


  p3 <- ggplot(sample_relative_safe_woSTRAINS_sub03,aes(x=sample,y=Snps,group=site,color=species,fill=species, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nRepeat_identity:",Repeat_identity,"\ncore:",core,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.2, alpha=.1)+ 
    # facet_grid(day~kessel~species)+
    facet_grid(species~.)+
    # facet_grid(treatment~species)+
    # facet_grid(treatment~species)+
    labs("",
         x="time [h]",
         y="Alternative \nallele\n frequency")+
    #scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    #scale_y_continuous(limits = c(0,0.3))+
    theme_classic()+
   scale_fill_manual(values=all_colours)+
   scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
      #axis.text.x = element_blank(),
          legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          )
  
#   png("~/Desktop/presenations/my_presentations/20190919_VUA/abundance_snps.png", width = 1900, height = 1200,res=300)
# 
p3
# 
# dev.off()
# 
#  library(plotly)
# ggp <- ggplotly(p3)
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_bacteria_onlyModifier.html")

```

#####Venn-Diagramm

```{r}

library(gplots)
library(gridExtra)
library(grid)
library(VennDiagram)

# colnames(sample_relative_wide)
# nrow(sample_relative_wide)

 # sample_relative_safe_final_no_snps <- sample_relative_wide[,c("site","RMK202","Lyo_202_2014","S72","Lyo_202_2012","L99","L80","Versand_202","L71","L70","Konserve_202","L108","L35","S50","L39","L104","L44")]

 ##-------------------------------
 ##meta has coverage
##-------------------------------
sample_relative_safe_final_snps_meta <- sample_relative_wide %>% select(site,"Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018")
sum(rowSums(sample_relative_safe_final_snps_meta[,-1],na.rm = TRUE)==0)
sum(rowSums(sample_relative_safe_final_snps_meta[,-1],na.rm = TRUE)>0)
sample_relative_safe_final_meta_cleaned <- sample_relative_wide[(rowSums(sample_relative_safe_final_snps_meta[,-1],na.rm = TRUE)>0),]
# sample_relative_safe_final_meta_cleaned_02 <- sample_relative_safe_final_snps_meta[(rowSums(sample_relative_safe_final_snps_meta[,-1],na.rm = TRUE)>0),]


 ##-------------------------------
 ##isolates has coverage
##-------------------------------

dim(sample_relative_wide)
in_isolates <- sample_relative_wide[which((sample_relative_wide$mst1>0.9 | sample_relative_wide$mst2>0.9| sample_relative_wide$mst3>0.9| sample_relative_wide$mst4>0.9| sample_relative_wide$mst5>0.9| sample_relative_wide$mst6>0.9 | sample_relative_wide$mst7>0.9 | sample_relative_wide$mst8>0.9 |sample_relative_wide$mst9>0.9 |  sample_relative_wide$mst10>0.9) %in% TRUE),]
dim(in_isolates)


##----------------
##plotting Venn with Isolates
##----------------
##==================================================================ldel

#----------------------total snps in analysis
total_tmp <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core") %>% filter(species=="L. delbrueckii")  %>% filter(mst3>0.9| mst7>0.9| mst6>0.9 |  mst9>0.9|mst4>0.9) %>% filter( mst5<=0.9) %>% filter( mst8<=0.9 & mst10<=0.9) %>% filter("Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05| "starter\nculture\n2018">0.05 ) %>% nrow()
#----------------------in isolate
isolates_tmp <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter(mst3>0.9 | mst4>0.9| mst5>0.9| mst6>0.9| mst7>0.9| mst8>0.9| mst9>0.9| mst10>0.9)%>% nrow()
#----------------------in meta
meta_tmp <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05| "starter\nculture\n2018">0.05 ) %>% nrow()
##---------------------in meta or isolate
shared_tmp <- sample_relative_wide  %>%replace(is.na(.), 0)  %>%replace(is.na(.), 0)%>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter(mst3>0.9 | mst4>0.9| mst5>0.9| mst6>0.9| mst7>0.9| mst8>0.9| mst9>0.9| mst10>0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05| "starter\nculture\n2018">0.05 ) %>% nrow()

##----------------
##plotting Venn in meta or not
##----------------
plot.new()
venn.plot <- draw.pairwise.venn(isolates_tmp,meta_tmp, shared_tmp, c("in isolates", "only in metagenome"), lty = rep("blank", 
    2), fill = c("light blue", "grey"), alpha = rep(0.5, 2), cat.pos = c(0, 
    0), cat.dist = rep(0.025, 2))
grid.draw(venn.plot)

svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/04_venn_diagramm_ldel.svg",width=6,height=4)
grid.draw(venn.plot)
dev.off()

##==============================
##-------------group1
group1_count <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter(mst3>0.9| mst7>0.9| mst6>0.9 |  mst9>0.9|mst4>0.9)%>% nrow()
##-------------group2
group2_count <-  sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter( mst8>0.9) %>% nrow()
##-------------group3
group3_count <-  sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter( mst5>0.9|mst10>0.9) %>% nrow()
#----------------------in meta
meta_tmp <-  sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% nrow()
##==============================
group1_group2_match <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter(mst3>0.9| mst7>0.9| mst6>0.9 |  mst9>0.9|mst4>0.9) %>% filter( mst8>0.9) %>% nrow() 
group1_group3_match <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter(mst3>0.9| mst7>0.9| mst6>0.9 |  mst9>0.9|mst4>0.9) %>% filter( mst5>0.9|mst10>0.9) %>% nrow() 
group1_meta_match <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter(mst3>0.9| mst7>0.9| mst6>0.9 |  mst9>0.9|mst4>0.9) %>%filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% nrow() 
group2_group3_match <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>%filter( mst8>0.9) %>% filter( mst5>0.9|mst10>0.9) %>% nrow() 
group2_meta_match <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>%filter( mst8>0.9) %>%filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% nrow() 
group3_meta_match <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>%filter( mst5>0.9|mst10>0.9) %>%filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% nrow() 
##==============================
group1_group2_group3_match <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter(mst3>0.9| mst7>0.9| mst6>0.9 |  mst9>0.9|mst4>0.9) %>% filter( mst8>0.9) %>% filter( mst5>0.9|mst10>0.9) %>%nrow() 
group1_group2_meta_match <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter(mst3>0.9| mst7>0.9| mst6>0.9 |  mst9>0.9|mst4>0.9) %>% filter( mst8>0.9) %>%  filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% nrow() 
group1_group3_meta_match <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter(mst3>0.9| mst7>0.9| mst6>0.9 |  mst9>0.9|mst4>0.9) %>%  filter( mst5>0.9|mst10>0.9) %>%filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>%nrow() 
group2_group3_meta_match <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter( mst8>0.9) %>%  filter( mst5>0.9|mst10>0.9) %>%filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>%nrow() 
##==============================
group1_group2_group3_match <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter(mst3>0.9| mst7>0.9| mst6>0.9 |  mst9>0.9|mst4>0.9) %>% filter( mst8>0.9) %>% filter( mst5>0.9|mst10>0.9)%>%filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>%nrow() 


##----------------
##plotting Venn triple
##----------------

plot.new()
venn.plot <- draw.triple.venn(group1_count, group2_count, group3_count, group1_group2_match,  group2_group3_match, group1_group3_match,
    group1_group2_group3_match, category = c("L.delbrueckii\ngroup1", "L.delbrueckii\ngroup2", "L.delbrueckii\ngroup3"), lty = "blank", 
    fill = c("skyblue", "pink1", "mediumorchid"))
grid.draw(venn.plot)


svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/04_venn_diagramm_ldel_tri.svg",width=6,height=4)
grid.draw(venn.plot)
dev.off()


##==================================================================Sterm

#----------------------total snps in analysis
total_tmp <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter(mst1>0.9 | mst2>0.9| "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% nrow()

#----------------------in isolate
isolates_tmp <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter(mst1>0.9 | mst2>0.9) %>% nrow()
#----------------------in meta
meta_tmp <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% nrow()
##---------------------in meta or isolate
shared_tmp <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter((mst1>0.9 | mst2>0.9) , ("Lyo\n1996">0.9| "Lyo\n2012">0.9| "Lyo\n2014">0.9| "working\nstock">0.9 | "starter\nculture\n2012">0.9) | "starter\nculture\n2018">0.9 ) %>% nrow()
##----------------
##plotting Venn in meta or not
##----------------
plot.new()
venn.plot <- draw.pairwise.venn(isolates_tmp,meta_tmp, shared_tmp, c("in isolates", "only in metagenome"), lty = rep("blank", 
    2), fill = c("light blue", "grey"), alpha = rep(0.5, 2), cat.pos = c(0, 
    0), cat.dist = rep(0.025, 2))
grid.draw(venn.plot)


svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/04_venn_diagramm_sterm.svg",width=6,height=4)
grid.draw(venn.plot)
dev.off()

##==============================
##-------------mst1
mst1_count <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter(mst1>0.9 ) %>% nrow()
##-------------mst2
mst2_count <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter( mst2>0.9) %>% nrow()
#----------------------in meta
meta_tmp <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% nrow()
##==============================
mst1_mst2_match <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter(mst1>0.9 ) %>% filter( mst2>0.9) %>% nrow() 
mst1_meta_match <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter(mst1>0.9 ) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% nrow() 
mst2_meta_match <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter(mst2>0.9 ) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% nrow() 
##==============================
mst1_mst2_meta_match <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter( mst1>0.9) %>% filter(mst2>0.9 ) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% nrow() 


##----------------
##plotting Venn di
##----------------


plot.new()
venn.plot <- draw.pairwise.venn(mst1_count,mst2_count, mst1_mst2_match, c("S.thermophilus\nmst1", "S.thermophilus\nmst2"), lty = rep("blank", 
    2), fill = c("light blue","pink"), alpha = rep(0.5, 2), cat.pos = c(0, 
    0), cat.dist = rep(0.025, 2))
grid.draw(venn.plot)

svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/04_venn_diagramm_sterm_di_strains.svg",width=6,height=4)
grid.draw(venn.plot)
dev.off()

##=====================================
##frequency plot
##=====================================


##==================================================Streptococcus thermophilus===================================================================


##-------------mst1
mst1_only_frequency <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter(mst1>0.9 )  %>% filter( mst2<=0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% gather(.,sample,Frequency,"Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018") %>% filter(Frequency>0.05) %>% add_column(.,group="S.thermophilus\nmst1")
##-------------mst2
mst2_only_frequency <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter(mst1<=0.9 )  %>% filter( mst2>0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% gather(.,sample,Frequency,"Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018")  %>% filter(Frequency>0.05) %>% add_column(.,group="S.thermophilus\nmst2")
##-------------not explained
not_explained_frequency <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter(mst1<=0.9 )  %>% filter( mst2<=0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% gather(.,sample,Frequency,"Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018")  %>% filter(Frequency>0.05) %>% add_column(.,group="not\nexplained")

all_sterm_frequency <- rbind(mst1_only_frequency,mst2_only_frequency,not_explained_frequency)

MAG_frequency <- 1-(median(mst1_only_frequency$Frequency)+median(mst2_only_frequency$Frequency))
##----------------
##plotting frequency
##----------------

all_sterm_frequency$group = factor(all_sterm_frequency$group, levels=c("S.thermophilus\nmst1","S.thermophilus\nmst2","not\nexplained"))

sterm_freqeunyPlot <- ggplot(all_sterm_frequency,aes(x=group,y=Frequency))+ geom_violin(aes(color=group,fill=group))+geom_boxplot(aes(alpha = 0.2),outlier.shape=NA)+theme_classic()+
      scale_fill_manual(values=c("light blue","pink","grey"))+
      scale_color_manual(values=c("light blue","pink","grey"))+
  labs(y="alternative allele\nfrequeny")+
    theme(legend.position="none",
       # axis.text.x = element_text(size=8),
       axis.title.x = element_blank())

svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/04_frequeny_sterm.svg",width=6,height=4)
sterm_freqeunyPlot
dev.off()

##==================================================Lactobacillus delbrueckii===================================================================
##-------------group1
group1_count <- sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter(mst3>0.9| mst7>0.9| mst6>0.9 |  mst9>0.9|mst4>0.9)%>% nrow()
##-------------group2
group2_count <-  sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter( mst8>0.9) %>% nrow()
##-------------group3
group3_count <-  sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter( mst5>0.9|mst10>0.9) %>% nrow()
#----------------------in meta
meta_tmp <-  sample_relative_wide %>%replace(is.na(.), 0) %>% filter(core=="core")%>% filter(species=="L. delbrueckii")  %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% nrow()

##-------------group1
group1_only_frequency <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="L. delbrueckii")  %>% filter(species=="L. delbrueckii")  %>% filter(mst3>0.9| mst7>0.9| mst6>0.9 |  mst9>0.9|mst4>0.9)%>% filter( mst8<=0.9) %>% filter(mst5<=0.9&mst10<=0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% gather(.,sample,Frequency,"Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018") %>% filter(Frequency>0.0) %>% add_column(.,group="L.delbrueckii\ngroup1")
##-------------group2
group2_only_frequency <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="L. delbrueckii")  %>% filter(species=="L. delbrueckii")  %>% filter(mst3<=0.9& mst7<=0.9&mst6<=0.9 &  mst9<=0.9&mst4<=0.9)%>% filter( mst8>0.9) %>% filter( mst5<=0.9&mst10<=0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% gather(.,sample,Frequency,"Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018")  %>% add_column(.,group="L.delbrueckii\ngroup2")
##-------------group3
group3_only_frequency <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="L. delbrueckii")  %>% filter(species=="L. delbrueckii")  %>% filter(mst3<=0.9& mst7<=0.9&mst6<=0.9 &  mst9<=0.9&mst4<=0.9)%>% filter( mst8<=0.9) %>% filter( mst5>0.9|mst10>0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% gather(.,sample,Frequency,"Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018")  %>% filter(Frequency>0.0)%>% add_column(.,group="L.delbrueckii\ngroup3")

##-------------not explained
not_explained_frequency_ldel <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="L. delbrueckii")  %>% filter(species=="L. delbrueckii")  %>% filter(mst3<=0.9& mst7<=0.9&mst6<=0.9 &  mst9<=0.9&mst4<=0.9)%>% filter( mst8<=0.9) %>% filter( mst5<=0.9&mst10<=0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% gather(.,sample,Frequency,"Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018") %>% filter(Frequency>0.0) %>% add_column(.,group="not\nexplained")


all_ldel_frequency <- rbind(group1_only_frequency,group2_only_frequency,not_explained_frequency_ldel,group3_only_frequency)

# MAG_frequency <- 1-(median(mst1_only_frequency$Frequency)+median(mst2_only_frequency$Frequency))
##----------------
##plotting frequency
##----------------

all_ldel_frequency$group = factor(all_ldel_frequency$group, levels=c("L.delbrueckii\ngroup1","L.delbrueckii\ngroup2","L.delbrueckii\ngroup3","not\nexplained"))
table(all_ldel_frequency$group)

ldel_freqeunyPlot <- ggplot(all_ldel_frequency,aes(x=group,y=Frequency))+ geom_violin(aes(color=group,fill=group))+geom_boxplot(aes(alpha = 0.2),outlier.shape=NA)+theme_classic()+
      scale_fill_manual(values=c("skyblue", "pink1", "mediumorchid","grey"))+
      scale_color_manual(values=c("skyblue", "pink1", "mediumorchid","grey"))+
  lims(y=c(0,0.2))+
  labs(y="alternative allele\nfrequeny")+
    theme(legend.position="none",
       # axis.text.x = element_text(size=8),
       axis.title.x = element_blank())
ldel_freqeunyPlot
svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/04_frequeny_ldel.svg",width=6,height=4)
ldel_freqeunyPlot
dev.off()

```

##### strain count
needs previous chunk to run
```{r}
library(robustbase)
##==================================================Streptococcus thermophilus===================================================================
##---------------------------------
##Sample medians calulation per group
##---------------------------------
##-------------mst1
mst1_Median <-  sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter(mst1>0.9 )  %>% filter( mst2<=0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% select("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018") %>% as.matrix %>%colMedians()
##-------------mst2
mst2_Median <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter(mst1<=0.9 )  %>% filter( mst2>0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% select("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018") %>% as.matrix %>%colMedians()
##-------------not explained
not_explained_Median <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="S. thermophilus")  %>% filter(mst1<=0.9 )  %>% filter( mst2<=0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% select("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018") %>% as.matrix %>%colMedians()

##---------------------------------
##Gather data
##---------------------------------
relAbundance_isolates_sterm <- rbind(mst1_Median,mst2_Median,not_explained_Median)
S_thermophilusMAG_rmk202 <- 1-(relAbundance_isolates_sterm[1,]+relAbundance_isolates_sterm[2,])
relAbundance_isolates_sterm <- rbind(mst1_Median,mst2_Median,S_thermophilusMAG_rmk202)
relAbundance_isolates_sterm_long <-  rbind(mst1_Median,mst2_Median,S_thermophilusMAG_rmk202)%>% as.data.frame()%>% add_column(group=rownames(relAbundance_isolates_sterm))%>% gather(.,sample,Median,"Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018") 


##---------------------------------
##stacked barplot
##---------------------------------
table(relAbundance_isolates_sterm_long$group)
relAbundance_isolates_sterm_long$group <- revalue(relAbundance_isolates_sterm_long$group, c("mst1_Median"="S.thermophilus\nmst1", "mst2_Median"="S.thermophilus\nmst2", "S_thermophilusMAG_rmk202"="S.thermophilus\nMAG RMK202"))
relAbundance_isolates_sterm_long$group = factor(relAbundance_isolates_sterm_long$group, levels=rev(c("S.thermophilus\nmst1","S.thermophilus\nmst2","S.thermophilus\nMAG RMK202")))

ggplot(relAbundance_isolates_sterm_long,aes(x=sample,y=Median,group=group,color=group,fill=group))+geom_bar(stat="identity")+theme_classic()+
  scale_color_manual(values=rev(c("light blue","pink","grey")))+
 scale_fill_manual(values=rev(c("light blue","pink","grey")))


##==================================================Lactobacillus delbrueckii===================================================================
library(naniar)

##---------------------------------
##Sample medians calulation per group
##---------------------------------
group1_Median <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="L. delbrueckii")  %>% filter(species=="L. delbrueckii")  %>% filter(mst3>0.9| mst7>0.9| mst6>0.9 |  mst9>0.9|mst4>0.9)%>% filter( mst8<=0.9) %>% filter(mst5<=0.9&mst10<=0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 )%>% select("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018") %>%  replace_with_na_at(.vars = c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018"),condition = ~.x == 0)%>% as.matrix()%>% colMedians(na.rm = TRUE)   %>%replace(is.na(.), 0) 


##-------------group2
group2_Median <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="L. delbrueckii")  %>% filter(species=="L. delbrueckii")  %>% filter(mst3<=0.9& mst7<=0.9&mst6<=0.9 &  mst9<=0.9&mst4<=0.9)%>% filter( mst8>0.9) %>% filter( mst5<=0.9&mst10<=0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% select("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018") %>%  replace_with_na_at(.vars = c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018"),condition = ~.x == 0)%>% as.matrix()%>% colMedians(na.rm = TRUE)   %>%replace(is.na(.), 0) 
##-------------group3
group3_Median <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="L. delbrueckii")  %>% filter(species=="L. delbrueckii")  %>% filter(mst3<=0.9& mst7<=0.9&mst6<=0.9 &  mst9<=0.9&mst4<=0.9)%>% filter( mst8<=0.9) %>% filter( mst5>0.9|mst10>0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% select("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018") %>%  replace_with_na_at(.vars = c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018"),condition = ~.x == 0)%>% as.matrix()%>% colMedians(na.rm = TRUE)   %>%replace(is.na(.), 0) 
##-------------not explained
not_explained_Median_ldel <- sample_relative_wide  %>%replace(is.na(.), 0)  %>% filter(core=="core") %>% filter(species=="L. delbrueckii")  %>% filter(species=="L. delbrueckii")  %>% filter(mst3<=0.9& mst7<=0.9&mst6<=0.9 &  mst9<=0.9&mst4<=0.9)%>% filter( mst8<=0.9) %>% filter( mst5<=0.9&mst10<=0.9) %>% filter( "Lyo\n1996">0.05| "Lyo\n2012">0.05| "Lyo\n2014">0.05| "working\nstock">0.05 | "starter\nculture\n2012">0.05 | "starter\nculture\n2018">0.05 ) %>% select("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018") %>%  replace_with_na_at(.vars = c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018"),condition = ~.x == 0)%>% as.matrix()%>% colMedians(na.rm = TRUE)   %>%replace(is.na(.), 0) 


##---------------------------------
##Gather data
##---------------------------------

relAbundance_isolates_ldel <- rbind(group1_Median,group2_Median,group3_Median,not_explained_Median_ldel)
Ldel_MAG_rmk202 <- 1-(colSums(relAbundance_isolates_ldel))
group1_Median_new <- Ldel_MAG_rmk202+group1_Median
relAbundance_isolates_ldel <- rbind(group1_Median_new,group2_Median,group3_Median,not_explained_Median_ldel)
relAbundance_isolates_ldel_long <- rbind(group1_Median_new,group2_Median,group3_Median,not_explained_Median_ldel) %>% as.data.frame()%>% add_column(group=rownames(relAbundance_isolates_ldel))%>% gather(.,sample,Median,"Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018") 

# colSums(relAbundance_isolates_ldel)

##---------------------------------
##stacked barplot
##---------------------------------

relAbundance_isolates_ldel_long$group <- revalue(relAbundance_isolates_ldel_long$group, c("group1_Median_new"="L.delbrueckii\ngroup1", "group2_Median"="L.delbrueckii\ngroup2", "group3_Median"="L.delbrueckii\ngroup3", "not_explained_Median_ldel"="not\nexplained"))
relAbundance_isolates_ldel_long$group = factor(relAbundance_isolates_ldel_long$group, levels=rev(c("L.delbrueckii\ngroup1","L.delbrueckii\ngroup2","L.delbrueckii\ngroup3","not\nexplained")))

ggplot(relAbundance_isolates_ldel_long,aes(x=sample,y=Median,group=group,color=group,fill=group))+geom_bar(stat="identity")+theme_classic()+
      scale_fill_manual(values=rev(c("skyblue", "pink1", "mediumorchid","grey")))+
      scale_color_manual(values=rev(c("skyblue", "pink1", "mediumorchid","grey")))


```

#####Muller plot strains

good [blog](https://cran.r-project.org/web/packages/ggmuller/vignettes/ggmuller.html)
or [this blog](https://cran.r-project.org/web/packages/MullerPlot/vignettes/MullerPlot.html)


```{r}
library(ggmuller)
library(MullerPlot)
library(viridis)

##==================================================Streptococcus thermophilus===================================================================
library(MullerPlot)

parent_identity_file <- as.matrix(data.frame(names=rownames(relAbundance_isolates_sterm),parents=NA,colors=c("light blue","pink","grey")))
Final_Abundance_Sterm_matrix <- relAbundance_isolates_sterm 
colnames(Final_Abundance_Sterm_matrix) <- 1:6
layout(matrix(c(1,2), nrow = 1), widths = c(0.8, 0.2))
par(mar=c(3, 3, 1, 0))
m.plot <- Muller.plot(attributes = parent_identity_file, population.data = Final_Abundance_Sterm_matrix,
                      data.method = "table", time.interval.method = "equal",mgp=c(2,0.5,0))

##==================================================Lactobacillus delbrueckii===================================================================

library(MullerPlot)

parent_identity_file <- as.matrix(data.frame(names=rownames(relAbundance_isolates_ldel),parents=NA,colors=c("skyblue", "pink1", "mediumorchid","grey")))
Final_Abundance_Ldel_matrix <- relAbundance_isolates_ldel
colnames(Final_Abundance_Ldel_matrix) <- 1:6
layout(matrix(c(1,2), nrow = 1), widths = c(0.8, 0.2))
par(mar=c(3, 3, 1, 0))
m.plot <- Muller.plot(attributes = parent_identity_file, population.data = Final_Abundance_Ldel_matrix,
                      data.method = "table", time.interval.method = "equal",mgp=c(2,0.5,0))


###------------------------------------------------------------
##all Bacteria together
###------------------------------------------------------------


library(readr)
coverage_rmk202 <- read_delim("~/Desktop/Projects/2019_Pilotplan/coverage_rmk202.tsv","\t", escape_double = FALSE, col_names = c("rowNumber","sample","species","Median_Coverage","total_coverage","percent_coverage"),trim_ws = TRUE)

coverage_rmk202$sample <- revalue(coverage_rmk202$sample, c("lyo202_96"="Lyo\n1996","Lyo_202_2012"="Lyo\n2012", "Lyo_202_2014"="Lyo\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018"))

##----------------------percent calculation
coverage_rmk202_percent <- coverage_rmk202 %>% filter(species %in% c("S_thermophilus_RMK202","L_delbrueckii_RMK202")) %>% group_by(., sample) %>% mutate(percent_bacteria = Median_Coverage/sum(Median_Coverage))

##---------------------plot bacterial diversity

coverage_rmk202_percent$sample = factor(coverage_rmk202_percent$sample, levels=c("Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018"))
all_colours <- rev(c("#EB4D4D","#10B552") )


PrelAbundance <-  ggplot( data = coverage_rmk202_percent,aes(y = percent_bacteria, x = sample, group=interaction(species),fill = species))+ geom_bar( stat="identity")+
    labs("",
         x="",
         y="relative abundance")+
    theme_classic()+
   scale_color_viridis(discrete=TRUE)+
   scale_fill_manual(values=all_colours)+
  # scale_fill_viridis(discrete=TRUE)+
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
        # axis.text.x = element_blank(),
          legend.position="right",
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank()
          )

  PrelAbundance
  # svg("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.svg",width=7,height=4)
   # png("~/Desktop/Manuscripts/2019_RMK202/Figures/F1_relativeAbundance.png", width = 1000, height = 800,res=300)
svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/F1_relativeAbundance_speciesLevel.svg",width=7,height=4)

PrelAbundance

dev.off()


##---------------------calculate subspecies divesity

##------------------------------------------------------------------Streptococcus thermophilus
relAbundance_isolates_sterm
coverage_rmk202_percent
relAbundance_isolates_sterm_normalized <- relAbundance_isolates_sterm
for (sampleName in colnames(relAbundance_isolates_sterm)) {
  print(sampleName)
  coverage_rmk202_percent_tmp <- coverage_rmk202_percent %>% filter(sample==sampleName)  %>% filter(species=="S_thermophilus_RMK202") %>% select(percent_bacteria) %>% as.double()
  print(coverage_rmk202_percent_tmp[2])
relAbundance_isolates_sterm_normalized[,sampleName] <- relAbundance_isolates_sterm[,sampleName]*coverage_rmk202_percent_tmp[2]
  
}

##------------------------------------------------------------------Lactobacillus delbrueckii
relAbundance_isolates_ldel
coverage_rmk202_percent
relAbundance_isolates_ldel_normalized <- relAbundance_isolates_ldel
for (sampleName in colnames(relAbundance_isolates_ldel)) {
  print(sampleName)
    coverage_rmk202_percent_tmp <- coverage_rmk202_percent %>% filter(sample==sampleName)  %>% filter(species=="L_delbrueckii_RMK202") %>% select(percent_bacteria) %>% as.double()
  print(coverage_rmk202_percent_tmp[2])
relAbundance_isolates_ldel_normalized[,sampleName] <- relAbundance_isolates_ldel[,sampleName]*coverage_rmk202_percent_tmp[2]
  
}
##------------------------------------------------------------------together

relAbundance_isolates_togther <- rbind(relAbundance_isolates_sterm_normalized,relAbundance_isolates_ldel_normalized) %>% as.data.frame()

# relAbundance_isolates_togther$group <- revalue(rownames(relAbundance_isolates_togther),c("mst1_Median"="S.thermophilus\nmst1", "mst2_Median"="S.thermophilus\nmst2", "S_thermophilusMAG_rmk202"="S.thermophilus\nMAG RMK202","group1_Median_new"="L.delbrueckii\ngroup1", "group2_Median"="L.delbrueckii\ngroup2", "group3_Median"="L.delbrueckii\ngroup3", "not_explained_Median_ldel"="not\nexplained"))
# 
# 
# relAbundance_isolates_togther$group = factor(relAbundance_isolates_sterm_long$group, levels=(c("S.thermophilus\nmst1","S.thermophilus\nmst2","S.thermophilus\nMAG RMK202","L.delbrueckii\ngroup1","L.delbrueckii\ngroup2","L.delbrueckii\ngroup3","not\nexplained")))



parent_identity_file <- as.matrix(data.frame(names=rownames(relAbundance_isolates_togther),parents=NA,colors=c("light blue","pink","grey","skyblue", "pink1", "mediumorchid","grey")))
Final_Abundance_together_matrix <- relAbundance_isolates_togther
colnames(Final_Abundance_together_matrix) <- 1:6
layout(matrix(c(1,2), nrow = 1), widths = c(0.8, 0.2))
par(mar=c(3, 3, 1, 0))
m.plot <- Muller.plot(attributes = parent_identity_file, population.data = Final_Abundance_together_matrix,
                      data.method = "table", time.interval.method = "equal",mgp=c(2,0.5,0))


svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/F1_relativeAbundance_subspeciesLevel.svg",width=7,height=4)

m.plot <- Muller.plot(attributes = parent_identity_file, population.data = Final_Abundance_together_matrix,
                      data.method = "table", time.interval.method = "equal",mgp=c(2,0.5,0))

dev.off()
```


######nucleotide diversity over time

Here, we calculate nucleotide diversity within genes
```{r}

# 
# 
# 
# 
# sterm_genome=1861615
# ldel_genome=2167371
# 
# # sample_relative_safe_final_tsne_safe
# 
# # sample_relative_safe_final_tsne_safe_ldel<- sample_relative_safe_final[grep("L_delbrueckii_RMK202_",sample_relative_safe_final$site),]
# # sample_relative_safe_final_tsne_safe_sterm <- sample_relative_safe_final[grep("S_thermophilus_RMK202",sample_relative_safe_final$site),]
# # nrow(sample_relative_safe_final_tsne_safe_sterm )
# # nrow(sample_relative_safe_final_tsne_safe_ldel_plasmid)
# 
# ##reduce to necessary
# 
# sample_relative_Sterm <- sample_relative_safe_final_meta_cleaned[grep("S_thermophilus_RMK202",sample_relative_safe_final_meta_cleaned$site),] 
# dim(sample_relative_Sterm)
# 
# sample_relative_Ldel <- sample_relative_safe_final_meta_cleaned[grep("L_delbrueckii_RMK202",sample_relative_safe_final_meta_cleaned$site),] 
# dim(sample_relative_Ldel)
# 
# ##calculations sterm
# 
# #samples <- sample_relative_Sterm[,9:ncol(sample_relative_Sterm)]
# samples <- sample_relative_Sterm[,c("RMK202","Lyo_202_2014","Lyo_202_2012","Versand_202","Konserve_202")]
# nrow(samples)
# samples_count <- colSums(samples>0.05,na.rm=TRUE)
# nrow(samples_count)
# 
# nucleotid_div <- data.frame(samples=factor(names(samples_count)),div=100*(samples_count/sterm_genome))
# 
# nucleotid_div$group <- "sterm"
# 
# nucleotid_div$samples_02 = factor(nucleotid_div$samples, levels=c("Lyo_202_2012","Lyo_202_2014","RMK202","Konserve_202","Versand_202"))
# levels(nucleotid_div$samples_02)
# levels(nucleotid_div$samples)
# 
# nucleotid_div_sterm <- nucleotid_div
# 
#  p1 <- ggplot(nucleotid_div,aes(x=samples_02,y=div,group=group))+ geom_point(size=2)+ geom_line()+ #stat_summary(fun.y=sum, geom="line")+
#    labs("",
#          x="",
#          y="nucleotide diversity [%]",
#           title="Streptococcus thermophilus")+
#   theme_classic()+
#  theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
#        #axis.text.x=element_blank(),
#           rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
#           legend.position="none"
#        )
#   p1
#   svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/05_sterm_nucleotideDiv.svg",width=3,height=4)
# p1
# 
# dev.off()
# 
# ##calculations sterm
# 
# # samples <- sample_relative_Ldel[,9:ncol(sample_relative_Ldel)]
# samples <- sample_relative_Ldel[,c("RMK202","Lyo_202_2014","Lyo_202_2012","Versand_202","Konserve_202")]
# samples_count <- colSums(samples>0,na.rm=TRUE)
# samples_count
# nucleotid_div <- data.frame(samples=factor(names(samples_count)),div=100*(samples_count/sterm_genome))
# 
# nucleotid_div$group <- "ldel"
# 
# nucleotid_div$samples_02 = factor(nucleotid_div$samples, levels=c("Lyo_202_2012","Lyo_202_2014","RMK202","Konserve_202","Versand_202"))
# levels(nucleotid_div$samples_02)
# levels(nucleotid_div$samples)
# 
# 
#  p2 <- ggplot(nucleotid_div,aes(x=samples_02,y=div,group=group))+ geom_point(size=2)+ geom_line()+ #stat_summary(fun.y=sum, geom="line")+
#    labs("",
#          x="",
#          y="nucleotide diversity [%]",
#           title="Lactobacillus delbrueckii")+
#   theme_classic()+
#  theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
#        #axis.text.x=element_blank(),
#           rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
#           legend.position="none"
#        )
#   p2
#   
# svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/05_ldel_nucleotideDiv.svg",width=3,height=4)
# p2
# 
# dev.off()
# 
# ##merge
# 
# nucleotid_div_all <- rbind(nucleotid_div_sterm,nucleotid_div)
# 
# all_colours <- rev(c("#EB4D4D","#FAA0A0","#10B552","#36E37B","#6CF5A3","#C5F6FA","#99E9F2") ) 
# 
# 
#  p2 <- ggplot(nucleotid_div_all,aes(x=samples_02,y=div,group=group,fill=group,color=group))+ geom_point(size=3)+ geom_line()+ #stat_summary(fun.y=sum, geom="line")+
#    labs("",
#          x="",
#          y="nucleotide diversity [%]",
#           title="")+
#    scale_color_manual(values=c("#10B552","#EB4D4D"))+
#   theme_classic()+
#  theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
#        #axis.text.x=element_blank(),
#           rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
#           # legend.position="right",
#        legend.justification=c(1,1), legend.position=c(0.7,0.7),
#        legend.title = element_blank()
#        )
#   p2
#   
# svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/05_all_nucleotideDiv.svg",width=3,height=3)
# p2
# 
# dev.off()
# 
# ##merge and prep to plot with PTR
# 
# nucleotid_div_all <- rbind(nucleotid_div_sterm,nucleotid_div)
# 
# all_colours <- rev(c("#EB4D4D","#FAA0A0","#10B552","#36E37B","#6CF5A3","#C5F6FA","#99E9F2") ) 
# 
# 
#  p1 <- ggplot(nucleotid_div_all,aes(x=samples_02,y=div,group=group,fill=group,color=group))+ geom_point(size=3)+ geom_line()+ #stat_summary(fun.y=sum, geom="line")+
#    labs("",
#          x="",
#          y="nucleotide diversity [%]",
#           title="")+
#    scale_color_manual(values=c("#10B552","#EB4D4D"))+
#   theme_classic()+
#  theme(axis.text.x = element_blank(),
#        #axis.text.x=element_blank(),
#           rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
#           legend.position="none"
#        # legend.justification=c(1,1), legend.position=c(0.7,0.7),
#        # legend.title = element_blank()
#        )
#   p1
#   
# ##------------subsetting for only larger than 0.05 allel frequency
#   nrow(nucleotid_div_all)
#   nucleotid_div_subset <- nucleotid_div_all[nucleotid_div_all$div<0.05,]
#   nrow(nucleotid_div_subset)
#   ##nucleotide diversity in one plot
#   
#    ###---prepare axis 
#   ylim.pH <- c(min(nucleotid_div_sterm$div),max(nucleotid_div_sterm$div)) # in this example, precipitation
#   ylim.temp <- c(min(nucleotid_div$div),max(nucleotid_div$div))  # in this example, temperature
# 
#   b <- diff(ylim.pH)/diff(ylim.temp)
#   a <- 0.73
# 
#   
#    pall <- ggplot()+ #stat_summary(fun.y=sum, geom="line")+
#      geom_point(data = nucleotid_div_sterm,aes(y = div, x = samples_02),fill="#EB4D4D",color="#EB4D4D",size=2)+
#      geom_line(data = nucleotid_div_sterm,aes(y = div, x = samples_02,group=group),fill="#EB4D4D",color="#EB4D4D")+
#      geom_point(data = nucleotid_div,aes(y = (a+div*b), x = samples_02),fill="#10B552",color="#10B552",size=2)+
#      geom_line(data = nucleotid_div,aes(y = (a+div*b), x = samples_02,group=group),fill="#10B552",color="#10B552")+
#    labs("",
#          x="",
#          y="nucleotide diversity [%]",
#           title="")+
#   theme_classic()+
#      scale_x_discrete( expand = c(0, 0)) +
#  scale_y_continuous(name = "S. thermophilus\n nucleotide diversity [%]", sec.axis = sec_axis(~ (. - a)/b, name = "L. delbreuckii\n nucleotide diversity [%]"))+
#  theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
#        #axis.text.x=element_blank(),
#           rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
#           legend.position="none",
#        axis.text.y.left = element_text(color = "#EB4D4D"),
#           axis.text.y.right = element_text(color="#10B552"),
#           axis.title.y.left = element_text(color="#EB4D4D"),
#           axis.title.y.right = element_text(color="#10B552"),
#        )
#   pall
#   svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/05_sterm_nucleotideDiv.svg",width=4,height=2.5)
# pall
# dev.off()
#   
###------------------------------------------------------------------------new try
##--------------
##get length of core genome
##-------------
coreGeneNames_sterm_ldel <- read_delim("~/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/coreGeneNames_rmk202_bothLdelSterm.vcf", "\t", escape_double = FALSE, col_names = c("contig","method","annotation","start","end"), trim_ws = TRUE)
# table(coreGeneNames_sterm_ldel$contig)
coreGeneNames_sterm <- coreGeneNames_sterm_ldel %>% filter(contig=="S_thermophilus_RMK202") %>% summarize(GeneLength = end-start) %>% colSums()
coreGeneNames_ldel <- coreGeneNames_sterm_ldel %>% filter(contig=="L_delbrueckii_RMK202") %>% summarize(GeneLength = end-start) %>% colSums() 

##--------------
##Snp count per sample and species
##-------------
colnames(sample_relative_long)
sample_relative_long_Core <- sample_relative_long %>% filter(core=="core" & Snps>0.05) 
table(sample_relative_long_Core$sample)


sample_relative_long_Core <-  gather(sample_relative_wide, sample, Snps, "Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018", factor_key=TRUE,na.rm = TRUE) %>% filter(core=="core" & Snps>0.05)  %>% group_by(sample,species) %>% tally()
sample_relative_long_Core$coreGenomeSize <- if_else(sample_relative_long_Core$species=="S. thermophilus",coreGeneNames_sterm,coreGeneNames_ldel) 
CoreGenome_nucleotide_div <- sample_relative_long_Core %>% mutate(.,nucleotide_diversity=100*(n/coreGenomeSize))


##--------------
##add metagenome coverage to plot
##-------------

library(readr)
coverage_rmk202_setForMerge <- read_delim("~/Desktop/Projects/2019_Pilotplan/coverage_rmk202.tsv","\t", escape_double = FALSE, col_names = c("rowNumber","sample","species","Median_Coverage","total_coverage","percent_coverage"),trim_ws = TRUE)%>% select(sample,species,Median_Coverage) %>% filter(species %in% c("S_thermophilus_RMK202","L_delbrueckii_RMK202")) %>% mutate(species=recode(species, S_thermophilus_RMK202="S. thermophilus",L_delbrueckii_RMK202="L. delbrueckii")) %>% mutate(sample=recode(sample, lyo202_96="Lyo\n1996",Lyo_202_2012="Lyo\n2012", Lyo_202_2014="Lyo\n2014", Konserve_202="working\nstock", RMK202="starter\nculture\n2012", Versand_202="starter\nculture\n2018"))



table(coverage_rmk202_setForMerge$species)

CoreGenome_nucleotide_div <- merge(CoreGenome_nucleotide_div,coverage_rmk202_setForMerge,by=c("sample","species"),all.x = TRUE)

##--------------
##plot
##-------------

  nucleotid_div_sterm <- CoreGenome_nucleotide_div %>% filter(species=="S. thermophilus")
  nucleotid_div_ldel<- CoreGenome_nucleotide_div %>% filter(species!="S. thermophilus")

  ##-------------nucleotide diversity in one plot
  
   ###---prepare axis 
  ylim.pH <- c(min(nucleotid_div_sterm$nucleotide_diversity),max(nucleotid_div_sterm$nucleotide_diversity)) # in this example, precipitation
  ylim.temp <- c(min(nucleotid_div_ldel$nucleotide_diversity),max(nucleotid_div_ldel$nucleotide_diversity))  # in this example, temperature

  b <- diff(ylim.pH)/diff(ylim.temp)
  a <- 0.73

   pall <- ggplot()+ #stat_summary(fun.y=sum, geom="line")+
     geom_point(data = nucleotid_div_sterm,aes(y = nucleotide_diversity, x = sample,size=Median_Coverage),fill="#EB4D4D",color="#EB4D4D")+
     geom_line(data = nucleotid_div_sterm,aes(y = nucleotide_diversity, x = sample,group=species),fill="#EB4D4D",color="#EB4D4D")+
     geom_point(data = nucleotid_div_ldel,aes(y = (a+nucleotide_diversity*b), x = sample,size=Median_Coverage),fill="#10B552",color="#10B552")+
     geom_line(data = nucleotid_div_ldel,aes(y = (a+nucleotide_diversity*b), x = sample,group=species),fill="#10B552",color="#10B552")+
   labs("",
         x="",
         y="nucleotide diversity [%]",
          title="")+
  theme_classic()+
     scale_x_discrete( expand = c(0, 0)) +
 scale_y_continuous(name = "S. thermophilus\n nucleotide diversity [%]", sec.axis = sec_axis(~ (. - a)/b, name = "L. delbreuckii\n nucleotide diversity [%]"))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="left",
       axis.text.y.left = element_text(color = "#EB4D4D"),
          axis.text.y.right = element_text(color="#10B552"),
          axis.title.y.left = element_text(color="#EB4D4D"),
          axis.title.y.right = element_text(color="#10B552"),
       )
  pall
  svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/05_sterm_nucleotideDiv.svg",width=4,height=2.5)
pall
dev.off()
  

```


Finally I have the conclusion that the large majority of the ldel community is exlained with the metagenome assembled genome. The remaining variation is in average very small. 

##### new SNPs

```{r}


# sample_relative_wide %>% filter(.,`Lyo\n1996`==0.00 & `Lyo\n2014`>0.05)

##-----------------------------------------------------------
##cases where Lyo96==0
sample_relative_long_newSNPS <- sample_relative_wide %>% filter(.,`Lyo\n1996`==0.00) %>%  filter(.,`starter\nculture\n2018`>0.05) %>% gather(., sample, Snps, "Lyo\n1996","Lyo\n2012","Lyo\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018", factor_key=TRUE,na.rm = TRUE) %>% filter(.,significance!="MODIFIER") 



# nrow(sample_relative_safe_woSTRAINS)
# sample_relative_safe_woSTRAINS_sub01 <- sample_relative_safe_woSTRAINS[which(sample_relative_safe_woSTRAINS$significance!="MODIFIER"),]
# nrow(sample_relative_safe_woSTRAINS_sub01)
# sample_relative_safe_woSTRAINS_sub02 <- sample_relative_safe_woSTRAINS_sub01[which(sample_relative_safe_woSTRAINS_sub01$core=="core"),]
# nrow(sample_relative_safe_woSTRAINS_sub02)
# sample_relative_safe_woSTRAINS_sub03 <- sample_relative_safe_woSTRAINS_sub02[which(sample_relative_safe_woSTRAINS_sub02$Snps>0.05),]
# nrow(sample_relative_safe_woSTRAINS_sub03)
# table(sample_relative_safe_woSTRAINS_sub03$species)

  ##============
### plot 
   ##============
table(sample_relative_long_newSNPS$geneName)

  p3 <- ggplot(sample_relative_long_newSNPS,aes(x=sample,y=Snps,group=site,color=species,fill=species, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nRepeat_identity:",Repeat_identity,"\ncore:",core,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.8, alpha=.5)+ 
    # facet_grid(day~kessel~species)+
    facet_grid(species~.)+
    # facet_grid(treatment~species)+
    # facet_grid(treatment~species)+
    labs("",
         x="time [h]",
         y="Alternative \nallele\n frequency")+
    #scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    #scale_y_continuous(limits = c(0,0.3))+
    theme_classic()+
   scale_fill_manual(values=all_colours)+
   scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
      #axis.text.x = element_blank(),
          legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          )
  
#   png("~/Desktop/presenations/my_presentations/20190919_VUA/abundance_snps.png", width = 1900, height = 1200,res=300)
# 
p3
# 
# dev.off()



```



##### lost or gained SNPs

```{r}
##-------------
##lost SNPs
##-------------

colnames(meta_sterm)
# meta_sterm[c("Lyo_202_2012", "Lyo_202_2014","Konserve_202","RMK202","Versand_202")][is.na(meta_sterm[c("Lyo_202_2012", "Lyo_202_2014","Konserve_202","RMK202","Versand_202")])] <- 0

sum(is.na(meta_sterm$Versand_202))
sum(meta_sterm$RMK202<0.05)
sum(meta_sterm$Versand_202<0.05 )
sum(meta_sterm$Versand_202<0.05 |meta_sterm$RMK202<0.01)

threshold_01 <- 0.05
threshold_02 <- 0.01


lostSNPs <- meta_sterm[which((meta_sterm$Lyo_202_2012>threshold_01 &meta_sterm$Lyo_202_2014>threshold_01 &meta_sterm$Konserve_202>threshold_01) &(meta_sterm$Versand_202<threshold_02 |meta_sterm$RMK202<threshold_02)),]
# lostSNPs <- meta_sterm[which((meta_sterm$Versand_202<threshold_02 |meta_sterm$RMK202<threshold_02)),]
lostSNPs_long <- gather(lostSNPs, sample, Snps, "Lyo_202_2012":"Versand_202", factor_key=TRUE,na.rm = TRUE) 
lostSNPs_long_02 <- merge(lostSNPs_long,annotation_all,by="site",all.x = TRUE)
lostSNPs_long_02$direction <- "lost"


##plot
  p3 <- ggplot(lostSNPs_long_02,aes(x=sample,y=Snps,group=site, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nEC:",EC,"\nKEGG_ko:",KEGG_ko,"\nCAZy:",CAZy,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.5, alpha=1)+ 
    # facet_grid(day~kessel~species)+
    # facet_grid(species~.)+
    # facet_grid(treatment~species)+
    # facet_grid(treatment~species)+
    labs("",
         x="time [h]",
         y="Alternative \nallele\n frequency")+
    #scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    scale_y_continuous(limits = c(0,1))+
    theme_classic()+
   # scale_fill_manual(values=all_colours)+
   # scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
      #axis.text.x = element_blank(),
          legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          )
  
  p3

##-------------
##gained SNPs
##-------------

threshold_03 <- 0.95
threshold_04 <- 0.98

gainedSNPs <- meta_sterm[which((meta_sterm$Lyo_202_2012<threshold_03 &meta_sterm$Lyo_202_2014<threshold_03 &meta_sterm$Konserve_202<threshold_03) &(meta_sterm$Versand_202>threshold_04 |meta_sterm$RMK202>threshold_04)),]
gainedSNPs_long <- gather(gainedSNPs, sample, Snps, "Lyo_202_2012":"Versand_202", factor_key=TRUE,na.rm = TRUE) 
gainedSNPs_long_02 <- merge(gainedSNPs_long,annotation_all,by="site",all.x = TRUE)
gainedSNPs_long_02$direction <- "gained"


##plothttps://www.sciencedirect.com/science/article/pii/S0023643819311247#!
  p3 <- ggplot(gainedSNPs_long_02,aes(x=sample,y=Snps,group=site, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nEC:",EC,"\nKEGG_ko:",KEGG_ko,"\nCAZy:",CAZy,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.5, alpha=1)+ 
    # facet_grid(day~kessel~species)+
    # facet_grid(species~.)+
    # facet_grid(treatment~species)+
    # facet_grid(treatment~species)+
    labs("",
         x="time [h]",
         y="Alternative \nallele\n frequency")+
    #scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    scale_y_continuous(limits = c(0,1))+
    theme_classic()+
    geom_hline(yintercept = threshold_01,linetype="dashed", color = "red")+
    geom_hline(yintercept = threshold_02,linetype="dashed", color = "green")+
   # scale_fill_manual(values=all_colours)+
   # scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
      #axis.text.x = element_blank(),
          legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          )
  
  p3
  

##-------------
##merge SNPs
##-------------

gained_lost_SNPs_long_02 <- rbind(lostSNPs_long_02,gainedSNPs_long_02)

table(gained_lost_SNPs_long_02$significance)
gained_lost_SNPs_long_02$significance = factor(gained_lost_SNPs_long_02$significance, levels=c("MODERATE","LOW","MODIFIER"))

  p3 <- ggplot(gained_lost_SNPs_long_02,aes(x=sample,y=Snps,group=interaction(significance,site), color=direction,text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nEC:",EC,"\nKEGG_ko:",KEGG_ko,"\nCAZy:",CAZy,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.5, alpha=1,aes(linetype=significance))+ 
    # facet_grid(day~kessel~species)+
    # facet_grid(species~.)+
    # facet_grid(treatment~species)+
    # facet_grid(treatment~species)+
    labs("",
         x="time [h]",
         y="Alternative \nallele\n frequency")+
    #scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    scale_y_continuous(limits = c(0,1))+
    theme_classic()+
    # geom_hline(yintercept = threshold_01,linetype="dashed", color = "red")+
    # geom_hline(yintercept = threshold_02,linetype="dashed", color = "green")+
    # geom_hline(yintercept = threshold_03,linetype="dashed", color = "red")+
    # geom_hline(yintercept = threshold_04,linetype="dashed", color = "green")+
   # scale_fill_manual(values=all_colours)+
   # scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
      #axis.text.x = element_blank(),
          legend.position="right"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          )
  
  p3

###-------------  
##color according to significance
###-------------  

moderate_gained_lost_SNPs_long_02 <- gained_lost_SNPs_long_02[which(gained_lost_SNPs_long_02$significance=="MODERATE"),]
table(moderate_gained_lost_SNPs_long_02$Preferred_name)
moderate_gained_lost_SNPs_long_02$ForCOLOR <- moderate_gained_lost_SNPs_long_02$Preferred_name

NONmoderate_gained_lost_SNPs_long_02 <- gained_lost_SNPs_long_02[which(gained_lost_SNPs_long_02$significance!="MODERATE"),]
NONmoderate_gained_lost_SNPs_long_02$ForCOLOR <- NONmoderate_gained_lost_SNPs_long_02$significance

Renamed_gained_lost_SNPs_long_02 <- rbind(moderate_gained_lost_SNPs_long_02,NONmoderate_gained_lost_SNPs_long_02)

# table(Renamed_gained_lost_SNPs_long_02$ForCOLOR)
Renamed_gained_lost_SNPs_long_02$ForCOLOR = factor(Renamed_gained_lost_SNPs_long_02$ForCOLOR, levels=c("glcU","mntH","ylbM","LOW","MODIFIER"))

all_colours <- c("red","sandybrown","orange","darkgrey","lightgrey")

library(plyr)
Renamed_gained_lost_SNPs_long_02$sample <- revalue(Renamed_gained_lost_SNPs_long_02$sample, c("Lyo_202_2012"="Lyoampulle\n2012", "Lyo_202_2014"="Lyoampulle\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018"))
Renamed_gained_lost_SNPs_long_02$sample = factor(Renamed_gained_lost_SNPs_long_02$sample, levels=c("Lyoampulle\n2012","Lyoampulle\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018"))



  p3 <- ggplot(Renamed_gained_lost_SNPs_long_02,aes(x=sample,y=Snps,group=site, color=ForCOLOR,text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nEC:",EC,"\nKEGG_ko:",KEGG_ko,"\nCAZy:",CAZy,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.5, aes(alpha=significance))+ #,aes(linetype=significance)
    # facet_grid(day~kessel~species)+
    # facet_grid(species~.)+
    # facet_grid(treatment~species)+
    # facet_grid(treatment~species)+
    labs("",
         x="",
         y="Alternative allele frequency")+
    #scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    scale_y_continuous(limits = c(0,1))+
    theme_classic()+
    geom_hline(yintercept = 0.5,linetype="dashed", color = "black")+
    # geom_hline(yintercept = threshold_02,linetype="dashed", color = "green")+
    # geom_hline(yintercept = threshold_03,linetype="dashed", color = "red")+
    # geom_hline(yintercept = threshold_04,linetype="dashed", color = "green")+
    # scale_alpha_discrete(range = c(1,0.5))+
    scale_alpha_manual(values = c(1,0.5,0.5))+
    # scale_alpha_discrete(values = c(1,1,1,0.5,0.5))+
   # scale_fill_manual(values=all_colours)+
   scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
      #axis.text.x = element_blank(),
          legend.position="right"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          )
  
  p3

svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/lost_Gained_sterm_mutations.svg",width=6,height=4)
p3
dev.off()
  
```


(glcU)[https://www.uniprot.org/uniprot/Q5HDV2]
Involved in the uptake of glucose.
(mntH)[https://www.uniprot.org/uniprot/P0A769]
H+-stimulated, divalent metal cation uptake system. Involved in manganese and iron uptake. Can also transport cadmium, cobalt, zinc and to a lesser extent nickel and copper. Involved in response to reactive oxygen
(ylbM)[https://www.uniprot.org/uniprot/L8AHH4]
Catalyzes the formation of N4-acetylcytidine (ac4C) at the wobble position of elongator tRNA(Met), using acetate and ATP as substrates. First activates an acetate ion to form acetyladenylate (Ac-AMP) and then transfers the acetyl group to tRNA to form ac4C34.

###### mutations gained

mutations that are 0 at the beginning (Lyos) are higher afterwards. 


```{r}



##-------------
##gained SNPs
##-------------

colnames(meta_sterm)
# meta_sterm[c("Lyo_202_2012", "Lyo_202_2014","Konserve_202","RMK202","Versand_202")][is.na(meta_sterm[c("Lyo_202_2012", "Lyo_202_2014","Konserve_202","RMK202","Versand_202")])] <- 0

# sum(is.na(meta_sterm$Versand_202))
# sum(meta_sterm$RMK202<0.05)
# sum(meta_sterm$Versand_202<0.05 )
# sum(meta_sterm$Versand_202<0.05 |meta_sterm$RMK202<0.01)

threshold_01 <- 0.05
threshold_02 <- 0.05


gainedSNPs <- meta_sterm[which((meta_sterm$Lyo_202_2012<threshold_01) & (meta_sterm$Lyo_202_2014>threshold_02 &meta_sterm$Konserve_202>threshold_02 &meta_sterm$Versand_202>threshold_02 & meta_sterm$RMK202<threshold_02)),]
gainedSNPs <- meta_sterm[which((meta_sterm$Lyo_202_2012<threshold_01)& (meta_sterm$Lyo_202_2014>threshold_02)),]
nrow(gainedSNPs)
# lostSNPs <- meta_sterm[which((meta_sterm$Versand_202<threshold_02 |meta_sterm$RMK202<threshold_02)),]
gainedSNPs_long <- gather(gainedSNPs, sample, Snps, "Lyo_202_2012":"Versand_202", factor_key=TRUE,na.rm = TRUE) 
gainedSNPs_long_02 <- merge(gainedSNPs_long,annotation_all,by="site",all.x = TRUE)
gainedSNPs_long_02$direction <- "lost"


##plot
  p3 <- ggplot(lostSNPs_long_02,aes(x=sample,y=Snps,group=site, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nEC:",EC,"\nKEGG_ko:",KEGG_ko,"\nCAZy:",CAZy,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.5, alpha=1)+ 
    # facet_grid(day~kessel~species)+
    # facet_grid(species~.)+
    # facet_grid(treatment~species)+
    # facet_grid(treatment~species)+
    labs("",
         x="time [h]",
         y="Alternative \nallele\n frequency")+
    #scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    scale_y_continuous(limits = c(0,1))+
    theme_classic()+
   # scale_fill_manual(values=all_colours)+
   # scale_color_manual(values=all_colours)+
    scale_x_discrete( expand = c(0, 0)) +
    theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
      #axis.text.x = element_blank(),
          legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          )
  
  p3


```

no new mutations detected.



####explain metagenome snps with isolates

good [cluster](https://jmonlong.github.io/Hippocamplus/2018/02/13/tsne-and-clustering/) from tsne blog

```{r}
library(tidyverse)
library(vcfR)
library(readr)
library(grid)
library(gridExtra)
library(dbscan)
library(Rtsne)
library(ggrepel)
library(wesanderson)
##=====================
##sterm
##=====================
# 
# nrow(meta_sterm)
# both_metaIso_sterm <- meta_sterm[onlysecond_sterm %in% onlyfirst_sterm,]
# nrow(both_metaIso_sterm)
# only_meta_sterm <- meta_sterm[!onlysecond_sterm %in% onlyfirst_sterm,]
# nrow(only_meta_sterm)
# # sum(duplicated(only_meta_sterm))
# only_isolates_sterm <- sample_relative_safe_final_iso_cleaned_sterm[!onlyfirst_sterm %in% onlysecond_sterm,]

###-----not explained by isolates


  not_explained_sterm_plot <- gather(only_meta_sterm, sample, allele_frequency, "Lyo_202_2012":"Versand_202", factor_key=TRUE)
  # not_explained_sterm_plot$site <- paste(not_explained_sterm_plot$X.CHROM,not_explained_sterm_plot$POS,not_explained_sterm_plot$REF,not_explained_sterm_plot$ALT,sep="_")

  not_explained_sterm_plot$group <- "not_explained"
  boxplot_not_explained_sterm <- data.frame(name=(only_meta_sterm[,1]),rowSums=rowSums(only_meta_sterm[,-1]),name="not explained\nsterm")
nrow(boxplot_not_explained_sterm)

not_explained_sterm_plot$sample


# nucleotid_div_all$samples = factor(nucleotid_div_all$samples, levels=c("Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202"))
library(plyr)
not_explained_sterm_plot$sample <- revalue(not_explained_sterm_plot$sample, c("Lyo_202_2012"="Lyoampulle\n2012", "Lyo_202_2014"="Lyoampulle\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018"))
not_explained_sterm_plot$sample = factor(not_explained_sterm_plot$sample, levels=c("Lyoampulle\n2012","Lyoampulle\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018"))


###-----explained by both
  
  both_metaIso_sterm_inIsolates$site==both_metaIso_sterm$site
nrow(both_metaIso_sterm)
cutoff <- 0.9
explained_sterm_both <- both_metaIso_sterm[((both_metaIso_sterm_inIsolates$S50>=cutoff)  & (both_metaIso_sterm_inIsolates$S72>=cutoff)) %in% TRUE,]
nrow(explained_sterm_both)
  explained_sterm_both_plot <- gather(explained_sterm_both, sample, allele_frequency, "Lyo_202_2012":"Versand_202", factor_key=TRUE)

explained_sterm_both_plot$sample <- revalue(explained_sterm_both_plot$sample, c("Lyo_202_2012"="Lyoampulle\n2012", "Lyo_202_2014"="Lyoampulle\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018"))
explained_sterm_both_plot$sample = factor(explained_sterm_both_plot$sample, levels=c("Lyoampulle\n2012","Lyoampulle\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018"))

  
boxplot_explained_sterm_both <- data.frame(name=(explained_sterm_both[,1]),rowSums=rowSums(explained_sterm_both[,-1]),name="both explained\nsterm")

# both_metaIso_sterm_inIsolates[((both_metaIso_sterm_inIsolates$S50>=cutoff)  & (both_metaIso_sterm_inIsolates$S72>=cutoff)) %in% TRUE,]


explained_sterm_both_plot$group <- "both"

explained_s50  <-both_metaIso_sterm[((both_metaIso_sterm_inIsolates$S50>=cutoff)  & (both_metaIso_sterm_inIsolates$S72<cutoff)) %in% TRUE,]
dim(explained_s50)
# explained_s50$group <- "S50"
  explained_s50_plot <- gather(explained_s50, sample, allele_frequency, "Lyo_202_2012":"Versand_202", factor_key=TRUE)
nrow(explained_s50)
explained_s50_plot$group <- "S50"


explained_s50_plot$sample <- revalue(explained_s50_plot$sample, c("Lyo_202_2012"="Lyoampulle\n2012", "Lyo_202_2014"="Lyoampulle\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018"))
explained_s50_plot$sample = factor(explained_s50_plot$sample, levels=c("Lyoampulle\n2012","Lyoampulle\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018"))


boxplot_explained_s50 <- data.frame(name=(explained_s50[,1]),rowSums=rowSums(explained_s50[,-1]),name="explained\ns50")

explained_s72  <-  both_metaIso_sterm[((both_metaIso_sterm_inIsolates$S50<cutoff)  & (both_metaIso_sterm_inIsolates$S72>=cutoff)) %in% TRUE,]
dim(explained_s72)
  explained_s72_plot <- gather(explained_s72, sample, allele_frequency, "Lyo_202_2012":"Versand_202", factor_key=TRUE)
factor(explained_s72_plot$sample)
nrow(explained_s72)
explained_s72_plot$group <- "S72"
boxplot_explained_s72 <- data.frame(name=(explained_s72[,1]),rowSums=rowSums(explained_s72[,-1]),name="explained\ns72")


explained_s72_plot$sample <- revalue(explained_s72_plot$sample, c("Lyo_202_2012"="Lyoampulle\n2012", "Lyo_202_2014"="Lyoampulle\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018"))
explained_s72_plot$sample = factor(explained_s72_plot$sample, levels=c("Lyoampulle\n2012","Lyoampulle\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018"))

##=====================
##Ldel
##=====================
# 
# nrow(meta_sterm)
# both_metaIso_sterm <- meta_sterm[onlysecond_sterm %in% onlyfirst_sterm,]
# nrow(both_metaIso_sterm)
# only_meta_sterm <- meta_sterm[!onlysecond_sterm %in% onlyfirst_sterm,]
# nrow(only_meta_sterm)
# # sum(duplicated(only_meta_sterm))
# only_isolates_sterm <- sample_relative_safe_final_iso_cleaned_sterm[!onlyfirst_sterm %in% onlysecond_sterm,]

###-----not explained by isolates


  not_explained_Ldel_plot <- gather(only_meta_ldel, sample, allele_frequency, "Lyo_202_2012":"Versand_202", factor_key=TRUE)
  # not_explained_sterm_plot$site <- paste(not_explained_sterm_plot$X.CHROM,not_explained_sterm_plot$POS,not_explained_sterm_plot$REF,not_explained_sterm_plot$ALT,sep="_")

  not_explained_Ldel_plot$group <- "not_explained"
  boxplot_not_explained_Ldel <- data.frame(name=(only_meta_ldel[,1]),rowSums=rowSums(only_meta_ldel[,-1]),name="not explained\nLdel")
nrow(boxplot_not_explained_Ldel)

table(not_explained_Ldel_plot$sample)


# nucleotid_div_all$samples = factor(nucleotid_div_all$samples, levels=c("Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202"))
library(plyr)
not_explained_Ldel_plot$sample <- revalue(not_explained_Ldel_plot$sample, c("Lyo_202_2012"="Lyoampulle\n2012", "Lyo_202_2014"="Lyoampulle\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018"))
table(not_explained_Ldel_plot$sample)
not_explained_Ldel_plot$sample = factor(not_explained_Ldel_plot$sample, levels=c("Lyoampulle\n2012","Lyoampulle\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018"))


###-----explained by both
  
  both_metaIso_sterm_inIsolates$site==both_metaIso_sterm$site
nrow(both_metaIso_sterm)
cutoff <- 0.9
explained_sterm_both <- both_metaIso_sterm[((both_metaIso_sterm_inIsolates$S50>=cutoff)  & (both_metaIso_sterm_inIsolates$S72>=cutoff)) %in% TRUE,]
nrow(explained_sterm_both)
  explained_sterm_both_plot <- gather(explained_sterm_both, sample, allele_frequency, "Lyo_202_2012":"Versand_202", factor_key=TRUE)

explained_sterm_both_plot$sample <- revalue(explained_sterm_both_plot$sample, c("Lyo_202_2012"="Lyoampulle\n2012", "Lyo_202_2014"="Lyoampulle\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018"))
explained_sterm_both_plot$sample = factor(explained_sterm_both_plot$sample, levels=c("Lyoampulle\n2012","Lyoampulle\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018"))

  
boxplot_explained_sterm_both <- data.frame(name=(explained_sterm_both[,1]),rowSums=rowSums(explained_sterm_both[,-1]),name="both explained\nsterm")

# both_metaIso_sterm_inIsolates[((both_metaIso_sterm_inIsolates$S50>=cutoff)  & (both_metaIso_sterm_inIsolates$S72>=cutoff)) %in% TRUE,]


explained_sterm_both_plot$group <- "both"

explained_s50  <-both_metaIso_sterm[((both_metaIso_sterm_inIsolates$S50>=cutoff)  & (both_metaIso_sterm_inIsolates$S72<cutoff)) %in% TRUE,]
dim(explained_s50)
# explained_s50$group <- "S50"
  explained_s50_plot <- gather(explained_s50, sample, allele_frequency, "Lyo_202_2012":"Versand_202", factor_key=TRUE)
nrow(explained_s50)
explained_s50_plot$group <- "S50"


explained_s50_plot$sample <- revalue(explained_s50_plot$sample, c("Lyo_202_2012"="Lyoampulle\n2012", "Lyo_202_2014"="Lyoampulle\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018"))
explained_s50_plot$sample = factor(explained_s50_plot$sample, levels=c("Lyoampulle\n2012","Lyoampulle\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018"))


boxplot_explained_s50 <- data.frame(name=(explained_s50[,1]),rowSums=rowSums(explained_s50[,-1]),name="explained\ns50")

explained_s72  <-  both_metaIso_sterm[((both_metaIso_sterm_inIsolates$S50<cutoff)  & (both_metaIso_sterm_inIsolates$S72>=cutoff)) %in% TRUE,]
dim(explained_s72)
  explained_s72_plot <- gather(explained_s72, sample, allele_frequency, "Lyo_202_2012":"Versand_202", factor_key=TRUE)
factor(explained_s72_plot$sample)
nrow(explained_s72)
explained_s72_plot$group <- "S72"
boxplot_explained_s72 <- data.frame(name=(explained_s72[,1]),rowSums=rowSums(explained_s72[,-1]),name="explained\ns72")


explained_s72_plot$sample <- revalue(explained_s72_plot$sample, c("Lyo_202_2012"="Lyoampulle\n2012", "Lyo_202_2014"="Lyoampulle\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018"))
explained_s72_plot$sample = factor(explained_s72_plot$sample, levels=c("Lyoampulle\n2012","Lyoampulle\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018"))
```


###### subsetting and effect hisotgramm

Finally, we decided to look at following SNPs:
1. location with higher than 20x coverage 
2. minimum 3x support for SNP
3. minimum 0.05 allele frequency
4. only modifications within genes are used
5. only non-synomous SNVs (moderate and high impact mutations) are looked at
6. SNVs that are not present in all samples are not used for phasing


```{r}
###-------------
##--------merge all
###-------------
not_explained_sterm_plot_nonModi <- merge(not_explained_sterm_plot,annotation_all,by="site",all.x = TRUE)
explained_sterm_both_plot_nonModi <- merge(explained_sterm_both_plot,annotation_all,by="site",all.x = TRUE)
explained_s50_plot_nonModi <- merge(explained_s50_plot,annotation_all,by="site",all.x = TRUE)
explained_s72_plot_nonModi <- merge(explained_s72_plot,annotation_all,by="site",all.x = TRUE)

all_sterm_plot_hist <- rbind(explained_sterm_both_plot_nonModi ,explained_s50_plot_nonModi,explained_s72_plot_nonModi,not_explained_sterm_plot_nonModi)
nrow(all_sterm_plot_hist)


###-------------
##------creat histogramm
###-------------
# cbind(all_sterm_plot$site,all_sterm_plot$group)
table(all_sterm_plot_hist$significance)  
table(all_sterm_plot_hist$group)  

all_sterm_plot_hist$detection <- ifelse(is.na(all_sterm_plot_hist$allele_frequency),"bellow detection\nlimit",ifelse(all_sterm_plot_hist$allele_frequency>0.05,"used","bellow detection\nlimit"))
table(all_sterm_plot_hist$detection)
# readsExplained <- [,c("site","group")]

all_sterm_plot_hist$significance = factor(all_sterm_plot_hist$significance, levels=c('MODIFIER','LOW','MODERATE','HIGH'))
all_sterm_plot_hist$group = factor(all_sterm_plot_hist$group, levels=c('both','S50','S72','not_explained'))
all_sterm_plot_hist$group <- revalue(all_sterm_plot_hist$group, c("S50"="mst1", "S72"="mst2"))



histogramm_SNVs <- ggplot(all_sterm_plot_hist,aes(x=group))+geom_bar(aes(fill = interaction(significance,detection)))+theme_classic()+
  facet_grid(~significance)+
  xlab("") + 
  ylab("variant sites") +
  scale_fill_manual(values=c("grey89","grey89","grey89","grey89","grey","gold1","orange","red"))+   #wes_palette(n=3, name="Moonrise3")[1:3]))+
  theme(legend.position = "right",
        plot.margin=unit(c(0,0,0,2.5),"cm"),
        axis.text.x=element_text(angle=45, hjust=1,size=9))

histogramm_SNVs
 svg("~/Desktop/Manuscripts/2019_RMK202/Figures/S09_SNVs_histogramm",width=8,height=4)
histogramm_SNVs

 dev.off()
 
#  
#   library("tidyverse")
# library("officer")
# library("rvg")

# read_pptx() %>%
#     add_slide(layout = "Title and Content", master = "Office Theme") %>%
#     ph_with_vg(code = print(grid.arrange(p1,p2, p3,p0,ncol=1, nrow=4,heights=c(1.15,1,1,1.55))), type = "body",height = 7, width = 8) %>% 
#     print(target = "~/Desktop/presenations/my_presentations/20190919_VUA/slides_03.pptx")

 
##================
###-------------
##----------subsett only dominant non-synomous mutations
###-------------
##================


###---------------------
# ---------------------subset data1: low abundance reads
###---------------------
 # table(not_explained_sterm_plot_nonModi$group)
not_explained_sterm_plot_nonModi <- not_explained_sterm_plot_nonModi[which(not_explained_sterm_plot_nonModi$allele_frequency>0.05),]
explained_sterm_both_plot_nonModi <- explained_sterm_both_plot_nonModi[which(explained_sterm_both_plot_nonModi$allele_frequency>0.05),]
explained_s50_plot_nonModi$group <- revalue(explained_s50_plot_nonModi$group, c("S50"="mst1"))
explained_s50_plot_nonModi <- explained_s50_plot_nonModi[which(explained_s50_plot_nonModi$allele_frequency>0.05),]
explained_s72_plot_nonModi$group <- revalue(explained_s72_plot_nonModi$group, c( "S72"="mst2"))
explained_s72_plot_nonModi <- explained_s72_plot_nonModi[which(explained_s72_plot_nonModi$allele_frequency>0.05),]


all_sterm_plot <- rbind(explained_sterm_both_plot_nonModi ,explained_s50_plot_nonModi,explained_s72_plot_nonModi,not_explained_sterm_plot_nonModi)
nrow(all_sterm_plot)
all_sterm_plot$group = factor(all_sterm_plot$group, levels=c("both","mst1","mst2","not_explained"))


 
 ##----plooting without modification

  pall_sterm_withModi <- ggplot(all_sterm_plot,aes(x=sample,y=allele_frequency,group=site,color=group,fill=group))+ geom_line(size=0.1, alpha=.1)+ 
       facet_grid(group~.)+
   labs("",
         x="",
         # y="",
          y="all")+
       # scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )

pall_sterm_withModi
#  library(plotly)
# ggp <- ggplotly(pall_sterm_withModi)
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_bacteria_all_split.html")



###---------------------
# ---------------------subset data2: non-complete snps
###---------------------

all_sterm_plot_wide <- spread(all_sterm_plot, sample, allele_frequency)
dim(all_sterm_plot_wide)
all_sterm_plot_wide_woNA <- all_sterm_plot_wide[!is.na(rowSums(all_sterm_plot_wide[,41:45])),]
dim(all_sterm_plot_wide_woNA)
all_sterm_plot_wide_onlyNA <- all_sterm_plot_wide[is.na(rowSums(all_sterm_plot_wide[,41:45])),]
dim(all_sterm_plot_wide_onlyNA)
all_sterm_plot_wide_onlyNA_long <- gather(all_sterm_plot_wide_onlyNA, sample, allele_frequency, colnames(all_sterm_plot_wide_onlyNA)[41:45], factor_key=TRUE,na.rm = TRUE) 

all_sterm_plot_wide_onlyNA_long_nonMOD <- all_sterm_plot_wide_onlyNA_long[all_sterm_plot_wide_onlyNA_long$significance!="MODIFIER",]
all_sterm_plot_wide_onlyNA_long_nonMOD_LOW <- all_sterm_plot_wide_onlyNA_long_nonMOD[all_sterm_plot_wide_onlyNA_long_nonMOD$significance!="LOW",]

 ##----plooting without modification

  pall_sterm_withModi_onlyNA <- ggplot(all_sterm_plot_wide_onlyNA_long_nonMOD_LOW,aes(x=sample,y=allele_frequency,group=site,color=group,fill=group, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nEC:",EC,"\nKEGG_ko:",KEGG_ko,"\nCAZy:",CAZy,"\nCOG:",COG_Functional_Category))) + geom_line(size=0.1, alpha=.1)+ 
       facet_grid(group~.)+
   labs("",
         x="",
         # y="",
          y="all")+
       # scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )

pall_sterm_withModi_onlyNA


 library(plotly)
ggp <- ggplotly(pall_sterm_withModi_onlyNA)
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_bacteria_all_split_woModi_snps that go away.html")



###-------------
##remove modifier muations
###-------------
table(all_sterm_plot$significance)
all_sterm_plot_woModi <- all_sterm_plot[all_sterm_plot$significance!="MODIFIER",]
(table(all_sterm_plot_woModi$group)/5)/(nrow(all_sterm_plot_woModi)/5)



  pall_sterm_woModi <- ggplot(all_sterm_plot_woModi,aes(x=sample,y=allele_frequency,group=site,color=group,fill=group, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nEC:",EC,"\nKEGG_ko:",KEGG_ko,"\nCAZy:",CAZy,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.1, alpha=.5)+ 
       facet_grid(group~.)+
   labs("",
         x="",
         # y="",
          y="all")+
       # scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
pall_sterm_woModi
#  library(plotly)
# ggp <- ggplotly(pall_sterm_woModi)
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_bacteria_all_split_woModi.html")


###-------------
##remove Low effect muations
###-------------
table(all_sterm_plot$significance)
all_sterm_plot_woModi_LOW <- all_sterm_plot_woModi[all_sterm_plot_woModi$significance!="LOW",]
(table(all_sterm_plot_woModi_LOW$group)/5)/(nrow(all_sterm_plot_woModi_LOW)/5)



  pall_sterm_woModi_LOW <- ggplot(all_sterm_plot_woModi_LOW,aes(x=sample,y=allele_frequency,group=site,color=group,fill=group, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nEC:",EC,"\nKEGG_ko:",KEGG_ko,"\nCAZy:",CAZy,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.1, alpha=.5)+ 
       facet_grid(group~.)+
   labs("",
         x="",
         # y="",
          y="all")+
       # scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
pall_sterm_woModi_LOW
#  library(plotly)
# ggp <- ggplotly(pall_sterm_woModi)
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_bacteria_all_split_woModi.html")


###-------------
##only copre
###-------------

coreGeneNames_rmk202 <- read_delim("~/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/coreGeneNames_rmk202_FINAL.txt", "\t", escape_double = FALSE, col_names = c("site","core"), trim_ws = TRUE)
dim(all_sterm_plot)
all_sterm_plot_core <- merge(all_sterm_plot,coreGeneNames_rmk202,by="site",all.x = TRUE,all.y = FALSE)
dim(all_sterm_plot_core)

table(all_sterm_plot_core$core)
all_sterm_plot_core <- all_sterm_plot_core[all_sterm_plot_core$core=="core",]
table(all_sterm_plot_core$core)
(table(all_sterm_plot_core$group)/5)/(nrow(all_sterm_plot_core)/5)



  pall_sterm_CORE <- ggplot(all_sterm_plot_core,aes(x=sample,y=allele_frequency,group=site,color=group,fill=group, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nEC:",EC,"\nKEGG_ko:",KEGG_ko,"\nCAZy:",CAZy,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.1, alpha=.5)+ 
       facet_grid(group~.)+
   labs("",
         x="",
         # y="",
          y="all")+
       # scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
pall_sterm_CORE
#  library(plotly)
# ggp <- ggplotly(pall_sterm_woModi)
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_bacteria_all_split_woModi.html")

###---------
##boxplot
###---------

  pall_sterm_CORE_box <- ggplot(all_sterm_plot_core,aes(x=sample,y=allele_frequency,color=group,fill=group))+ geom_boxplot()+
       # facet_grid(group~.)+
   labs("",
         x="",
         # y="",
          y="all")+
       # scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="right",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
pall_sterm_CORE_box


  pall_sterm_CORE_box <- ggplot(all_sterm_plot_core,aes(x=group,y=allele_frequency,color=sample,fill=sample))+ geom_boxplot()+
       # facet_grid(group~.)+
   labs("",
         x="",
         # y="",
          y="all")+
       # scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="right",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
pall_sterm_CORE_box

pall_sterm_CORE_box_median <- aggregate(.~sample+group, data=all_sterm_plot_core[,c("allele_frequency","sample","group")], median, na.rm=TRUE)
pall_sterm_CORE_box_median

pall_sterm_CORE_box_median_wide <- spread(pall_sterm_CORE_box_median, group, allele_frequency)
pall_sterm_CORE_box_median_wide

Final_Abundance_Sterm_wide <- pall_sterm_CORE_box_median_wide %>% select(sample,mst1,mst2,not_explained)

Final_Abundance_Sterm_wide$MAG_rmk202 <- 1-pall_sterm_CORE_box_median_wide$both
Final_Abundance_Sterm_wide$mst1 <- pall_sterm_CORE_box_median_wide$both*pall_sterm_CORE_box_median_wide$mst1
Final_Abundance_Sterm_wide$mst2 <- pall_sterm_CORE_box_median_wide$both*pall_sterm_CORE_box_median_wide$mst2
Final_Abundance_Sterm_wide$not_explained <- pall_sterm_CORE_box_median_wide$both*pall_sterm_CORE_box_median_wide$not_explained
Final_Abundance_Sterm_wide
Final_Abundance_Sterm_long <- gather(Final_Abundance_Sterm_wide,Identity, Population, "mst1":"MAG_rmk202", factor_key=TRUE,na.rm = TRUE) 
colnames(Final_Abundance_Sterm_long)[1] <- "Generation"
Final_Abundance_Sterm_long$Generation <- revalue(Final_Abundance_Sterm_long$Generation, c("Lyoampulle\n2012"=as.double(1) ,"Lyoampulle\n2014"=as.double(2),"working\nstock"=as.double(3), "starter\nculture\n2012"=as.double(4), "starter\nculture\n2018"=as.double(5)))

Final_Abundance_Sterm_long

```





######S72 substrains


```{r}
library(forcats) 
library(dplyr)
colorsTSNE <- c("#DD8D29","#E2D200","#46ACC8","black","#B40F20")

###------------
###make tsne to isolate strands
###------------

explained_s72_plot_woModi <- explained_s72_plot_nonModi[explained_s72_plot_nonModi$significance!="MODIFIER",]
explained_s72_plot_woModi <- explained_s72_plot_woModi[explained_s72_plot_woModi$significance!="LOW",]
explained_s72_plot_woModi_wide <- spread(explained_s72_plot_woModi, sample, allele_frequency)
dim(explained_s72_plot_woModi_wide)

Nas_in_data <- explained_s72_plot_woModi_wide[is.na(rowSums(explained_s72_plot_woModi_wide[,41:45])),]
fortsne <- explained_s72_plot_woModi_wide[!is.na(rowSums(explained_s72_plot_woModi_wide[,41:45])),]

set.seed(1234567)
tsne <- Rtsne(fortsne[,41:45], dims = 2, perplexity=10, verbose=TRUE, max_iter = 500)
# tsne <- Rtsne(fortsne[,41:45], dims = 2, perplexity=10, verbose=TRUE, max_iter = 500)
tsne_plot <- as.data.frame(tsne$Y)
tsne_plot$site <- fortsne[,1]
tsne_plot$contig <- paste(str_split_fixed(tsne_plot$site, "_", 5)[,1],str_split_fixed(tsne_plot$site, "_", 5)[,2],sep = "_")
# unique(tsne_plot$contig)
tsne_plot$species <- tsne_plot$contig

##------------plot blank
ggplot(tsne_plot,aes(x=V1,y=V2))+geom_point()

##------------clusters


set.seed(123456)
ds.real = dbscan(tsne_plot[,c(1,2)], 4)
tsne_plot$density = factor(ds.real$cluster)
ds.cent = tsne_plot %>% group_by(density) %>% select(V1, 
    V2) %>% summarize_all(mean)
ggplot(tsne_plot, aes(x = V1, y = V2, colour = density)) + 
    geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label = density), 
    data = ds.cent) + guides(colour = FALSE)



##-----------make clustering manual
# tsne_plot$final_clust <- tsne_plot$density
# tsne_plot_final_clust <- tsne_plot  %>% select(site,  final_clust)
# table(tsne_plot_final_clust$final_clust)

tsne_plot_final_clust <- tsne_plot %>% mutate(final_clust = ifelse(density==3,2,
                                                                    ifelse(density==2,3,1)))  %>% select(site,  final_clust)
 tsne_plot <- tsne_plot %>% mutate(final_clust = ifelse(density==3,2,
                                                                    ifelse(density==2,3,1)))
table(tsne_plot_final_clust$final_clust)



##-------sumplementary plot
# unique(tsne_plot_final_clust$final_clust)
tsne_plot$final_clust = factor(tsne_plot$final_clust, levels=c("1","2","3"))
ds.cent = tsne_plot %>% group_by(final_clust) %>% select(V1, 
    V2) %>% summarize_all(mean)
# colors_substrains <- c(wes_palette(n=5, name="FantasticFox1"),"grey60")
# colors_substrains <- c(wes_palette(n=5, name="FantasticFox1")[1:3])
colors_substrains <- c(colorsTSNE[1:3],"grey")

Sterm_S72_suplementaryTsne <- ggplot(tsne_plot, aes(x = V1, y = V2, colour = final_clust)) + 
    geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label =final_clust), 
    data = ds.cent) + guides(colour = FALSE)+scale_color_manual(values = colors_substrains)

Sterm_S72_suplementaryTsne

##---------clustering

explained_s72_plot_nonModi_CLUSTERED <- merge(explained_s72_plot_woModi,tsne_plot_final_clust,by = "site",all=TRUE)
table(explained_s72_plot_nonModi_CLUSTERED$final_clust)
unique(explained_s72_plot_nonModi_CLUSTERED$final_clust)
explained_s72_plot_nonModi_CLUSTERED$final_clust <- fct_explicit_na(as.character(explained_s72_plot_nonModi_CLUSTERED$final_clust), na_level = "not clustered")
unique(explained_s72_plot_nonModi_CLUSTERED$final_clust)

# explained_s72_plot_nonModi_CLUSTERED[is.na(explained_s72_plot_nonModi_CLUSTERED$final_clust),3]
# unique(explained_s72_plot_nonModi_CLUSTERED$final_clust)
# revalue

# explained_s72_plot_nonModi_CLUSTERED
explained_s72_plot_nonModi_CLUSTERED$final_clust = factor(explained_s72_plot_nonModi_CLUSTERED$final_clust, levels=rev(c("1","2","3","not clustered")))

  p_S72_sterm_woModi_clustered <- ggplot(explained_s72_plot_nonModi_CLUSTERED,aes(x=sample,y=allele_frequency,group=site,color=final_clust,fill=final_clust, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nEC:",EC,"\nKEGG_ko:",KEGG_ko,"\nCAZy:",CAZy,"\nCOG:",COG_Functional_Category)))+ geom_line(size=0.1, alpha=.5)+ 
       # facet_grid(final_clust~.)+
   labs("",
         x="",
         # y="",
          y="all")+
       scale_fill_manual(values=rev(colors_substrains))+
      scale_color_manual(values=rev(colors_substrains))+
       scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )

p_S72_sterm_woModi_clustered

ggp <- ggplotly(p_S72_sterm_woModi_clustered)
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_S72_clustered_woModi.html")


##---------------density plot

explained_s72_plot_nonModi_CLUSTERED$POS
explained_s72_plot_nonModi_CLUSTERED_wide <- spread(explained_s72_plot_nonModi_CLUSTERED, sample, allele_frequency)
# explained_s72_plot_nonModi_CLUSTERED_wide[explained_s72_plot_nonModi_CLUSTERED_wide$final_clust==1,]

# explained_s72_plot_nonModi_CLUSTERED_wide$final_clust = factor(explained_s72_plot_nonModi_CLUSTERED_wide$final_clust, levels=c("NA","1","2","3","4"))
explained_s72_plot_nonModi_CLUSTERED_wide$final_clust = factor(explained_s72_plot_nonModi_CLUSTERED_wide$final_clust, levels=rev(c("1","2","3","not clustered")))

  pDensity_S72_sterm_woModi_clustered <- ggplot(explained_s72_plot_nonModi_CLUSTERED_wide,aes(x=POS,group=final_clust,color=final_clust,fill=final_clust, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nEC:",EC,"\nKEGG_ko:",KEGG_ko,"\nCAZy:",CAZy,"\nCOG:",COG_Functional_Category)))+ geom_density( alpha=.5)+ 
       # facet_grid(final_clust~.)+
   labs("",
         x="",
         # y="",
          y="all")+
       # scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="right",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )

pDensity_S72_sterm_woModi_clustered

##----------COG plot


explained_s72_plot_nonModi_CLUSTERED_wide_only1 <- explained_s72_plot_nonModi_CLUSTERED_wide[which(explained_s72_plot_nonModi_CLUSTERED_wide$final_clust==1),]


COG_functional_categories <- read_delim("~/Desktop/ScriptRepository/COG_functional_categories.csv",  ",", escape_double = FALSE, col_names = c("short","long","divided"),trim_ws = TRUE, na = "NA")
explained_s72_plot_nonModi_CLUSTERED_wide

explained_s72_plot_nonModi_CLUSTERED_cogNaming <- merge(explained_s72_plot_nonModi_CLUSTERED_wide,COG_functional_categories,by.x = "COG_Functional_Category",by.y="short",all.x=TRUE)

CogPlotSterm <- ggplot(explained_s72_plot_nonModi_CLUSTERED_cogNaming,aes(x=long,fill=final_clust))+geom_bar(position = "dodge2")+
  theme_classic()+
  labs(x="")+
  theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9),
          # # axis.text.y.left = element_text(color = "blue"),
          # axis.text.y.right = element_text(color="red"),
          # axis.title.y.left = element_text(color="blue"),
          # axis.title.y.right = element_text(color="red"),
          # rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          #legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank(),
          plot.margin=unit(c(0,0,0,2), "cm"),
          axis.text.x.bottom = element_text(size=10),
          legend.position = c(0.02, 0.98),
          legend.justification = c("left", "top")
          )
CogPlotSterm

sort(table(explained_s72_plot_nonModi_CLUSTERED_cogNaming$geneName),decreasing = TRUE)
# 
table(explained_s72_plot_nonModi_CLUSTERED_cogNaming[which(explained_s72_plot_nonModi_CLUSTERED_cogNaming$geneName=="addA"),]$final_clust)
table(explained_s72_plot_nonModi_CLUSTERED_cogNaming[which(explained_s72_plot_nonModi_CLUSTERED_cogNaming$geneName=="mvk"),]$final_clust)
table(explained_s72_plot_nonModi_CLUSTERED_cogNaming[which(explained_s72_plot_nonModi_CLUSTERED_cogNaming$geneName=="dnaG"),]$final_clust)
# table(explained_s72_plot_nonModi_CLUSTERED_cogNaming[which(explained_s72_plot_nonModi_CLUSTERED_cogNaming$geneName=="murA"),]$final_clust)
table(explained_s72_plot_nonModi_CLUSTERED_cogNaming[which(explained_s72_plot_nonModi_CLUSTERED_cogNaming$geneName=="glmS"),]$final_clust)
# table(explained_s72_plot_nonModi_CLUSTERED_cogNaming[which(explained_s72_plot_nonModi_CLUSTERED_cogNaming$geneName=="purH"),]$final_clust)

explained_S72_plot_nonModi_CLUSTERED <- explained_s72_plot_nonModi_CLUSTERED
```

among the most mutate genes are:
[addA](https://www.uniprot.org/uniprot/P23478) a double strand break repair gene
[mvk]
[dnaG](https://www.uniprot.org/uniprot/P0ABS5) RNA polymerase that catalyzes the synthesis of short RNA molecules used as primers for DNA polymerase during DNA replication
[murA](https://www.uniprot.org/uniprot/P0A749) Cell wall formation (PubMed:1512209). Adds enolpyruvyl to UDP-N-acetylglucosamine (PubMed:1512209, PubMed:20392080). Target for the antibiotic fosfomycin.
[glmS](https://www.uniprot.org/uniprot/P17169) Catalyzes the first step in hexosamine metabolism, converting fructose-6P into glucosamine-6P using glutamine as a nitrogen source
[purH]

######S50 substrains

```{r}
library(forcats) 

explained_s72_plot_woModi <- explained_s50_plot_nonModi[explained_s50_plot_nonModi$significance!="MODIFIER",]
explained_s72_plot_woModi <- explained_s72_plot_woModi[explained_s72_plot_woModi$significance!="LOW",]
explained_s72_plot_woModi_wide <- spread(explained_s72_plot_woModi, sample, allele_frequency)
dim(explained_s72_plot_woModi_wide)
fortsne <- explained_s72_plot_woModi_wide[which(!is.na(rowSums(explained_s72_plot_woModi_wide[,41:45]))),]
nas <- explained_s72_plot_woModi_wide[which(is.na(rowSums(explained_s72_plot_woModi_wide[,41:45]))),]


set.seed(123456)
set.seed(1234567)

fortsne <- fortsne[!duplicated(fortsne[,41:45]), ]
tsne <- Rtsne(fortsne[,41:45], dims = 2, perplexity=30, verbose=TRUE, max_iter = 500)
tsne_plot <- as.data.frame(tsne$Y)
tsne_plot$site <- fortsne[,1]
tsne_plot$contig <- paste(str_split_fixed(tsne_plot$site, "_", 5)[,1],str_split_fixed(tsne_plot$site, "_", 5)[,2],sep = "_")

tsne_plot$species <- tsne_plot$contig


##------------plot blank
ggplot(tsne_plot,aes(x=V1,y=V2))+geom_point()

##------------clusters
set.seed(123456)
ds.real = dbscan(tsne_plot[,c(1,2)], 1.5)
tsne_plot$density = factor(ds.real$cluster)
ds.cent = tsne_plot %>% group_by(density) %>% select(V1, 
    V2) %>% summarize_all(mean)
ggplot(tsne_plot, aes(x = V1, y = V2, colour = density)) + 
    geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label = density), 
    data = ds.cent) + guides(colour = FALSE)


##-----------make clustering manual
tsne_plot_final_clust <- tsne_plot %>% mutate(final_clust = ifelse(density==5,3,
                                                                    ifelse(density==7,1,
                                                                            ifelse(density==2,4,
                                                                                    ifelse(density==9,5,
                                                                                            ifelse(density==1,2,
                                                                                                  ifelse(density==10,2,
                                                                                            ifelse(density ==8,2,1))))))))    %>% select(site,  final_clust)
 tsne_plot <- tsne_plot %>% mutate(final_clust = ifelse(density==5,3,
                                                                    ifelse(density==7,1,
                                                                            ifelse(density==2,4,
                                                                                    ifelse(density==9,5,
                                                                                            ifelse(density==1,2,
                                                                                                  ifelse(density==10,2,
                                                                                            ifelse(density ==8,2,1)))))))) 
table(tsne_plot_final_clust$final_clust)

##-------sumplementary plot
# "#DD8D29","#E2D200","#46ACC8","#E58601","#B40F20"

unique(tsne_plot$final_clust)
tsne_plot$final_clust = factor(tsne_plot$final_clust, levels=c("1","2","3","4","5"))
ds.cent = tsne_plot %>% group_by(final_clust) %>% select(V1, 
    V2) %>% summarize_all(mean)
# colors_substrains <- c(wes_palette(n=5, name="FantasticFox1"),"grey60")
# colors_substrains <- c(wes_palette(n=5, name="FantasticFox1")[1:5])
colors_substrains <- c(colorsTSNE[1:5])

Sterm_S50_suplementaryTsne <- ggplot(tsne_plot, aes(x = V1, y = V2, colour = final_clust)) + 
    geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label =final_clust), 
    data = ds.cent) + guides(colour = FALSE)+scale_color_manual(values = colors_substrains)
Sterm_S50_suplementaryTsne


##-----------PLOT

explained_s72_plot_nonModi_CLUSTERED <- merge(explained_s72_plot_woModi,tsne_plot_final_clust,by = "site",all=TRUE)
table(explained_s72_plot_nonModi_CLUSTERED$final_clust)
unique(explained_s72_plot_nonModi_CLUSTERED$final_clust)
explained_s72_plot_nonModi_CLUSTERED$final_clust <- fct_explicit_na(as.character(explained_s72_plot_nonModi_CLUSTERED$final_clust), na_level = "not clustered")
unique(explained_s72_plot_nonModi_CLUSTERED$final_clust)

# revalue

explained_s72_plot_nonModi_CLUSTERED[is.na(explained_s72_plot_nonModi_CLUSTERED$final_clust),]

unique(explained_s72_plot_nonModi_CLUSTERED$final_clust)
explained_s72_plot_nonModi_CLUSTERED$final_clust = factor(explained_s72_plot_nonModi_CLUSTERED$final_clust, levels=rev(c("1","2","3","4","5","not clustered")))

colors_substrains <- c(colorsTSNE[1:5],"grey")


  p_S72_sterm_woModi_clustered <- ggplot(explained_s72_plot_nonModi_CLUSTERED,aes(x=sample,y=allele_frequency,group=site,color=final_clust,fill=final_clust, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nEC:",EC,"\nKEGG_ko:",KEGG_ko,"\nCAZy:",CAZy,"\nCOG:",COG_Functional_Category,"\nCluster:",final_clust)))+ geom_line(size=0.1, alpha=.5)+ 
       # facet_grid(final_clust~.)+
   labs("",
         x="",
         # y="",
          y="alternative allele frequency")+
        scale_fill_manual(values=rev(colors_substrains))+
      scale_color_manual(values=rev(colors_substrains))+
       scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="right",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )

p_S72_sterm_woModi_clustered

png("~/Desktop/presenations/my_presentations/20191007_groupmeeting/figures/S50_phasing_01.png",width=1800,height=1000,res=300)


p_S72_sterm_woModi_clustered+theme(legend.position="none")

dev.off()


ggp <- ggplotly(p_S72_sterm_woModi_clustered)
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_S50_clustered_woModi.html")



##---------------density plot

explained_s72_plot_nonModi_CLUSTERED$POS
explained_s72_plot_nonModi_CLUSTERED_wide <- spread(explained_s72_plot_nonModi_CLUSTERED, sample, allele_frequency)
# explained_s72_plot_nonModi_CLUSTERED_wide[explained_s72_plot_nonModi_CLUSTERED_wide$final_clust==1,]

# explained_s72_plot_nonModi_CLUSTERED_wide$final_clust = factor(explained_s72_plot_nonModi_CLUSTERED_wide$final_clust, levels=c("NA","1","2","3","4"))

  pDensity_S72_sterm_woModi_clustered <- ggplot(explained_s72_plot_nonModi_CLUSTERED_wide,aes(x=POS,group=final_clust,color=final_clust,fill=final_clust, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nEC:",EC,"\nKEGG_ko:",KEGG_ko,"\nCAZy:",CAZy,"\nCOG:",COG_Functional_Category)))+ geom_density( alpha=.5)+ 
       # facet_grid(final_clust~.)+
   labs("",
         x="genomic location",
         # y="",
          y="density")+ 
    scale_fill_manual(values=rev(colors_substrains))+
      scale_color_manual(values=rev(colors_substrains))+
       scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="right",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )

pDensity_S72_sterm_woModi_clustered


png("~/Desktop/presenations/my_presentations/20191007_groupmeeting/figures/S50_phasing_density_01.png",width=1000,height=1000,res=300)


pDensity_S72_sterm_woModi_clustered+theme(legend.position="none")

dev.off()

##----------COG plot


# explained_s72_plot_nonModi_CLUSTERED_wide_only1 <- explained_s72_plot_nonModi_CLUSTERED_wide[which(explained_s72_plot_nonModi_CLUSTERED_wide$final_clust==1),]


COG_functional_categories <- read_delim("~/Desktop/ScriptRepository/COG_functional_categories.csv",  ",", escape_double = FALSE, col_names = c("short","long","divided"),trim_ws = TRUE, na = "NA")
explained_s72_plot_nonModi_CLUSTERED_wide

explained_s72_plot_nonModi_CLUSTERED_cogNaming <- merge(explained_s72_plot_nonModi_CLUSTERED_wide,COG_functional_categories,by.x = "COG_Functional_Category",by.y="short",all.x=TRUE)

CogPlotSterm <- ggplot(explained_s72_plot_nonModi_CLUSTERED_cogNaming,aes(x=long,fill=final_clust))+geom_bar(position = "dodge2")+
  theme_classic()+
  labs(x="")+
  theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9),
          # # axis.text.y.left = element_text(color = "blue"),
          # axis.text.y.right = element_text(color="red"),
          # axis.title.y.left = element_text(color="blue"),
          # axis.title.y.right = element_text(color="red"),
          # rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          #legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank(),
          plot.margin=unit(c(0,0,0,2), "cm"),
          axis.text.x.bottom = element_text(size=10),
          legend.position = c(0.02, 0.98),
          legend.justification = c("left", "top")
          )
CogPlotSterm

sort(table(explained_s72_plot_nonModi_CLUSTERED_cogNaming$geneName),decreasing = TRUE)

# table(explained_s72_plot_nonModi_CLUSTERED_cogNaming[which(explained_s72_plot_nonModi_CLUSTERED_cogNaming$geneName=="cas9"),]$final_clust)
table(explained_s72_plot_nonModi_CLUSTERED_cogNaming[which(explained_s72_plot_nonModi_CLUSTERED_cogNaming$geneName=="csm5"),]$final_clust)
table(explained_s72_plot_nonModi_CLUSTERED_cogNaming[which(explained_s72_plot_nonModi_CLUSTERED_cogNaming$geneName=="gap"),]$final_clust)
table(explained_s72_plot_nonModi_CLUSTERED_cogNaming[which(explained_s72_plot_nonModi_CLUSTERED_cogNaming$geneName=="galM"),]$final_clust)
table(explained_s72_plot_nonModi_CLUSTERED_cogNaming[which(explained_s72_plot_nonModi_CLUSTERED_cogNaming$geneName=="mvaD"),]$final_clust)
table(explained_s72_plot_nonModi_CLUSTERED_cogNaming[which(explained_s72_plot_nonModi_CLUSTERED_cogNaming$geneName=="atpB"),]$final_clust)
table(explained_s72_plot_nonModi_CLUSTERED_cogNaming[which(explained_s72_plot_nonModi_CLUSTERED_cogNaming$geneName=="hisF"),]$final_clust)
# 
# table(explained_s72_plot_nonModi_CLUSTERED_cogNaming[which(explained_s72_plot_nonModi_CLUSTERED_cogNaming$geneName=="lysS"),]$final_clust)
# table(explained_s72_plot_nonModi_CLUSTERED_cogNaming[which(explained_s72_plot_nonModi_CLUSTERED_cogNaming$geneName=="gyrA"),]$final_clust)
explained_S50_plot_nonModi_CLUSTERED <- explained_s72_plot_nonModi_CLUSTERED
```


######not explained

```{r}
# not_explained_sterm_plot_nonModi <- not_explained_sterm_plot_nonModi[not_explained_sterm_plot_nonModi$allele_frequency>0.05,]
explained_s72_plot_woModi <- not_explained_sterm_plot_nonModi[not_explained_sterm_plot_nonModi$significance!="MODIFIER",]
explained_s72_plot_woModi <- explained_s72_plot_woModi[explained_s72_plot_woModi$significance!="LOW",]
explained_s72_plot_woModi_wide <- spread(explained_s72_plot_woModi, sample, allele_frequency)
dim(explained_s72_plot_woModi_wide)
fortsne <- explained_s72_plot_woModi_wide[which(!is.na(rowSums(explained_s72_plot_woModi_wide[,41:45]))),]

set.seed(123456)
fortsne <- fortsne[!duplicated(fortsne[,41:45]), ]
tsne <- Rtsne(fortsne[,41:45], dims = 2, perplexity=30, verbose=TRUE, max_iter = 500)
tsne_plot <- as.data.frame(tsne$Y)
tsne_plot$site <- fortsne[,1]
tsne_plot$contig <- paste(str_split_fixed(tsne_plot$site, "_", 5)[,1],str_split_fixed(tsne_plot$site, "_", 5)[,2],sep = "_")

tsne_plot$species <- tsne_plot$contig


##------------plot blank
ggplot(tsne_plot,aes(x=V1,y=V2))+geom_point()

##------------clusters
set.seed(123456)
ds.real = dbscan(tsne_plot[,c(1,2)], 2)
tsne_plot$density = factor(ds.real$cluster)
ds.cent = tsne_plot %>% group_by(density) %>% select(V1, 
    V2) %>% summarize_all(mean)
ggplot(tsne_plot, aes(x = V1, y = V2, colour = density)) + 
    geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label = density), 
    data = ds.cent) + guides(colour = FALSE)

##-----------make clustering manual
# 
# tsne_plot$final_clust <- tsne_plot$density
# tsne_plot_final_clust <-  tsne_plot %>% select(site,  final_clust)
tsne_plot_final_clust <- tsne_plot %>% mutate(final_clust = ifelse(density==1,1,
                                                                    ifelse(density==2,2,
                                                                           ifelse(density==3,4,
                                                                           ifelse(density==4,3,5)))))  %>% select(site,  final_clust)

tsne_plot <-tsne_plot %>% mutate(final_clust = ifelse(density==1,1,
                                                                    ifelse(density==2,2,
                                                                           ifelse(density==3,4,
                                                                           ifelse(density==4,3,5)))))
table(tsne_plot_final_clust$final_clust)

##---suplemantary

unique(tsne_plot$final_clust)
tsne_plot$final_clust = factor(tsne_plot$final_clust, levels=c("1","2","3","4","5"))
ds.cent = tsne_plot %>% group_by(final_clust) %>% select(V1, 
    V2) %>% summarize_all(mean)
# colors_substrains <- c(wes_palette(n=5, name="FantasticFox1"),"grey60")
# colors_substrains <- c(wes_palette(n=5, name="FantasticFox1")[1:4])
colors_substrains <- c(colorsTSNE[1:5],"grey")

Sterm_notExplained_suplementaryTsne <- ggplot(tsne_plot, aes(x = V1, y = V2, colour = final_clust)) + 
    geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label =final_clust), 
    data = ds.cent) + guides(colour = FALSE)+scale_color_manual(values = colors_substrains)
Sterm_notExplained_suplementaryTsne


##-----------PLOT
library(forcats) 

explained_s72_plot_nonModi_CLUSTERED <- merge(explained_s72_plot_woModi,tsne_plot_final_clust,by = "site",all=TRUE)
table(explained_s72_plot_nonModi_CLUSTERED$final_clust)
# unique(explained_s72_plot_nonModi_CLUSTERED$final_clust)
explained_s72_plot_nonModi_CLUSTERED$final_clust <- fct_explicit_na(as.character(explained_s72_plot_nonModi_CLUSTERED$final_clust), na_level = "not clustered")
# unique(explained_s72_plot_nonModi_CLUSTERED$final_clust)



# explained_s72_plot_nonModi_CLUSTERED
explained_s72_plot_nonModi_CLUSTERED$final_clust = factor(explained_s72_plot_nonModi_CLUSTERED$final_clust, levels=rev(c("1","2","3","4","5","not clustered")))

# nrow(explained_s72_plot_nonModi_CLUSTERED)

  p_S72_sterm_woModi_clustered <- ggplot(explained_s72_plot_nonModi_CLUSTERED,aes(x=sample,y=allele_frequency,group=site,color=final_clust,fill=final_clust, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nEC:",EC,"\nKEGG_ko:",KEGG_ko,"\nCAZy:",CAZy,"\nCOG:",COG_Functional_Category,"\nCluster:",final_clust)))+ geom_line(size=0.1, alpha=.5)+ 
       # facet_grid(final_clust~.)+
   labs("",
         x="",
         # y="",
          y="all")+
       # scale_fill_manual(values="gray41")+
      scale_color_manual(values=rev(colors_substrains))+
       scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,0.1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="right",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )

p_S72_sterm_woModi_clustered


png("~/Desktop/presenations/my_presentations/20191007_groupmeeting/figures/notExplained_phasing_01.png",width=1800,height=1000,res=300)


p_S72_sterm_woModi_clustered+theme(legend.position="none")

dev.off()


ggp <- ggplotly(p_S72_sterm_woModi_clustered)
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_notExplained_clustered_woModi.html")

##---------------density plot

explained_s72_plot_nonModi_CLUSTERED$POS
explained_s72_plot_nonModi_CLUSTERED_wide <- spread(explained_s72_plot_nonModi_CLUSTERED, sample, allele_frequency)
# explained_s72_plot_nonModi_CLUSTERED_wide[explained_s72_plot_nonModi_CLUSTERED_wide$final_clust==1,]

# explained_s72_plot_nonModi_CLUSTERED_wide$final_clust = factor(explained_s72_plot_nonModi_CLUSTERED_wide$final_clust, levels=c("NA","1","2","3","4"))

  pDensity_S72_sterm_woModi_clustered <- ggplot(explained_s72_plot_nonModi_CLUSTERED_wide,aes(x=POS,group=final_clust,color=final_clust,fill=final_clust, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nEC:",EC,"\nKEGG_ko:",KEGG_ko,"\nCAZy:",CAZy,"\nCOG:",COG_Functional_Category)))+ geom_density( alpha=.5)+ 
       # facet_grid(final_clust~.)+
   labs("",
         x="",
         # y="",
          y="all")+
       # scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(0,0.5)+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="right",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )

pDensity_S72_sterm_woModi_clustered

##----------COG plot


# explained_s72_plot_nonModi_CLUSTERED_wide_only1 <- explained_s72_plot_nonModi_CLUSTERED_wide[which(explained_s72_plot_nonModi_CLUSTERED_wide$final_clust==1),]


COG_functional_categories <- read_delim("~/Desktop/ScriptRepository/COG_functional_categories.csv",  ",", escape_double = FALSE, col_names = c("short","long","divided"),trim_ws = TRUE, na = "NA")
explained_s72_plot_nonModi_CLUSTERED_wide

explained_s72_plot_nonModi_CLUSTERED_cogNaming <- merge(explained_s72_plot_nonModi_CLUSTERED_wide,COG_functional_categories,by.x = "COG_Functional_Category",by.y="short",all.x=TRUE)

CogPlotSterm <- ggplot(explained_s72_plot_nonModi_CLUSTERED_cogNaming,aes(x=long,fill=final_clust))+geom_bar(position = "dodge2")+
  theme_classic()+
  labs(x="")+
  theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9),
          # # axis.text.y.left = element_text(color = "blue"),
          # axis.text.y.right = element_text(color="red"),
          # axis.title.y.left = element_text(color="blue"),
          # axis.title.y.right = element_text(color="red"),
          # rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          #legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank(),
          plot.margin=unit(c(0,0,0,2), "cm"),
          axis.text.x.bottom = element_text(size=10),
          legend.position = c(0.02, 0.98),
          legend.justification = c("left", "top")
          )
CogPlotSterm

sort(table(explained_s72_plot_nonModi_CLUSTERED_cogNaming$geneName),decreasing = TRUE)

table(explained_s72_plot_nonModi_CLUSTERED_cogNaming[which(explained_s72_plot_nonModi_CLUSTERED_cogNaming$geneName=="lysS"),]$final_clust)
table(explained_s72_plot_nonModi_CLUSTERED_cogNaming[which(explained_s72_plot_nonModi_CLUSTERED_cogNaming$geneName=="deoD"),]$final_clust)
explained_not_plot_nonModi_CLUSTERED <- explained_s72_plot_nonModi_CLUSTERED
```


######both explained

```{r}
explained_s72_plot_woModi <- explained_sterm_both_plot_nonModi[explained_sterm_both_plot_nonModi$significance!="MODIFIER",]
explained_s72_plot_woModi <- explained_s72_plot_woModi[explained_s72_plot_woModi$significance!="LOW",]
explained_s72_plot_woModi_wide <- spread(explained_s72_plot_woModi, sample, allele_frequency)
dim(explained_s72_plot_woModi_wide)
fortsne <- explained_s72_plot_woModi_wide[which(!is.na(rowSums(explained_s72_plot_woModi_wide[,41:45]))),]

set.seed(123456)

fortsne <- fortsne[!duplicated(fortsne[,41:45]), ]
tsne <- Rtsne(fortsne[,41:45], dims = 2, perplexity=30, verbose=TRUE, max_iter = 500)
tsne_plot <- as.data.frame(tsne$Y)
tsne_plot$site <- fortsne[,1]
tsne_plot$contig <- paste(str_split_fixed(tsne_plot$site, "_", 5)[,1],str_split_fixed(tsne_plot$site, "_", 5)[,2],sep = "_")

tsne_plot$species <- tsne_plot$contig


##------------plot blank
ggplot(tsne_plot,aes(x=V1,y=V2))+geom_point()
##------------clusters
set.seed(123456)
ds.real = dbscan(tsne_plot[,c(1,2)], 3)
tsne_plot$density = factor(ds.real$cluster)
ds.cent = tsne_plot %>% group_by(density) %>% select(V1, 
    V2) %>% summarize_all(mean)
ggplot(tsne_plot, aes(x = V1, y = V2, colour = density)) + 
    geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label = density), 
    data = ds.cent) + guides(colour = FALSE)



##-----------make clustering manual
# tsne_plot$final_clust <- tsne_plot$density
# tsne_plot_final_clust <-  tsne_plot %>% select(site,  final_clust)
tsne_plot_final_clust <- tsne_plot %>% mutate(final_clust = ifelse(density==1,3,
                                                                    ifelse(density==3,2,
                    ifelse(density==7,4,1))))  %>% select(site,  final_clust)

tsne_plot <-tsne_plot %>% mutate(final_clust = ifelse(density==1,3,
                                                                    ifelse(density==3,2,
                    ifelse(density==7,4,1))))

table(tsne_plot_final_clust$final_clust)

##---suplemantary

unique(tsne_plot$final_clust)
tsne_plot$final_clust = factor(tsne_plot$final_clust, levels=c("1","2","3","4"))
ds.cent = tsne_plot %>% group_by(final_clust) %>% select(V1, 
    V2) %>% summarize_all(mean)
# colors_substrains <- c(wes_palette(n=5, name="FantasticFox1"),"grey60")
# colors_substrains <- c(wes_palette(n=5, name="FantasticFox1")[1:3])
colors_substrains <- c(colorsTSNE[1:4],"grey")

Sterm_both_suplementaryTsne <- ggplot(tsne_plot, aes(x = V1, y = V2, colour = final_clust)) + 
    geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label =final_clust), 
    data = ds.cent) + guides(colour = FALSE)+scale_color_manual(values = colors_substrains)
Sterm_both_suplementaryTsne



##-----------PLOT

explained_s72_plot_nonModi_CLUSTERED <- merge(explained_s72_plot_woModi,tsne_plot_final_clust,by = "site",all=TRUE)
table(explained_s72_plot_nonModi_CLUSTERED$final_clust)
unique(explained_s72_plot_nonModi_CLUSTERED$final_clust)
explained_s72_plot_nonModi_CLUSTERED$final_clust <- fct_explicit_na(as.character(explained_s72_plot_nonModi_CLUSTERED$final_clust), na_level = "not clustered")
unique(explained_s72_plot_nonModi_CLUSTERED$final_clust)


# explained_s72_plot_nonModi_CLUSTERED
explained_s72_plot_nonModi_CLUSTERED$final_clust = factor(explained_s72_plot_nonModi_CLUSTERED$final_clust, levels=rev(c("1","2","3","4","not clustered")))

  p_S72_sterm_woModi_clustered <- ggplot(explained_s72_plot_nonModi_CLUSTERED,aes(x=sample,y=allele_frequency,group=site,color=final_clust,fill=final_clust, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nEC:",EC,"\nKEGG_ko:",KEGG_ko,"\nCAZy:",CAZy,"\nCOG:",COG_Functional_Category,"\nCluster:",final_clust)))+ geom_line(size=0.1, alpha=.5)+ 
       # facet_grid(final_clust~.)+
   labs("",
         x="",
         # y="",
          y="all")+
       # scale_fill_manual(values="gray41")+
      scale_color_manual(values=rev(colors_substrains))+
       scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="right",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )

p_S72_sterm_woModi_clustered


ggp <- ggplotly(p_S72_sterm_woModi_clustered)
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_bothExplained_clustered_woModi.html")


##---------------density plot

explained_s72_plot_nonModi_CLUSTERED$POS
explained_s72_plot_nonModi_CLUSTERED_wide <- spread(explained_s72_plot_nonModi_CLUSTERED, sample, allele_frequency)
# explained_s72_plot_nonModi_CLUSTERED_wide[explained_s72_plot_nonModi_CLUSTERED_wide$final_clust==1,]

# explained_s72_plot_nonModi_CLUSTERED_wide$final_clust = factor(explained_s72_plot_nonModi_CLUSTERED_wide$final_clust, levels=c("NA","1","2","3","4"))

  pDensity_S72_sterm_woModi_clustered <- ggplot(explained_s72_plot_nonModi_CLUSTERED_wide,aes(x=POS,group=final_clust,color=final_clust,fill=final_clust, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nEC:",EC,"\nKEGG_ko:",KEGG_ko,"\nCAZy:",CAZy,"\nCOG:",COG_Functional_Category)))+ geom_density( alpha=.5)+ 
       # facet_grid(final_clust~.)+
   labs("",
         x="",
         # y="",
          y="all")+
       # scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="right",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )

pDensity_S72_sterm_woModi_clustered

##----------COG plot


# explained_s72_plot_nonModi_CLUSTERED_wide_only1 <- explained_s72_plot_nonModi_CLUSTERED_wide[which(explained_s72_plot_nonModi_CLUSTERED_wide$final_clust==1),]


COG_functional_categories <- read_delim("~/Desktop/ScriptRepository/COG_functional_categories.csv",  ",", escape_double = FALSE, col_names = c("short","long","divided"),trim_ws = TRUE, na = "NA")
explained_s72_plot_nonModi_CLUSTERED_wide

explained_s72_plot_nonModi_CLUSTERED_cogNaming <- merge(explained_s72_plot_nonModi_CLUSTERED_wide,COG_functional_categories,by.x = "COG_Functional_Category",by.y="short",all.x=TRUE)

CogPlotSterm <- ggplot(explained_s72_plot_nonModi_CLUSTERED_cogNaming,aes(x=long,fill=final_clust))+geom_bar(position = "dodge2")+
  theme_classic()+
  labs(x="")+
  theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9),
          # # axis.text.y.left = element_text(color = "blue"),
          # axis.text.y.right = element_text(color="red"),
          # axis.title.y.left = element_text(color="blue"),
          # axis.title.y.right = element_text(color="red"),
          # rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          #legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank(),
          plot.margin=unit(c(0,0,0,2), "cm"),
          axis.text.x.bottom = element_text(size=10),
          legend.position = c(0.02, 0.98),
          legend.justification = c("left", "top")
          )
CogPlotSterm

sort(table(explained_s72_plot_nonModi_CLUSTERED_cogNaming$geneName),decreasing = TRUE)

table(explained_s72_plot_nonModi_CLUSTERED_cogNaming[which(explained_s72_plot_nonModi_CLUSTERED_cogNaming$geneName=="dinB"),]$final_clust)
table(explained_s72_plot_nonModi_CLUSTERED_cogNaming[which(explained_s72_plot_nonModi_CLUSTERED_cogNaming$geneName=="rlmD"),]$final_clust)

explained_both_plot_nonModi_CLUSTERED <- explained_s72_plot_nonModi_CLUSTERED

```

#####bring all together and plot with tsne cluster
```{r}

##--------colors

library("RColorBrewer")
display.brewer.all()
display.brewer.pal(n = 5, name = 'RdYlBu')
# colors_substrains <- c(brewer.pal(n = 5, name = 'RdYlBu'),"grey60")
colors_substrains <- c(colorsTSNE[1:5],"grey")
library(wesanderson)
# colors_substrains <- c(wes_palette(n=5, name="FantasticFox1"),"grey60")
##----------merge
explained_all_plot_nonModi_CLUSTERED <- rbind(explained_S72_plot_nonModi_CLUSTERED,
explained_S50_plot_nonModi_CLUSTERED,
explained_both_plot_nonModi_CLUSTERED,
explained_not_plot_nonModi_CLUSTERED)
nrow(explained_all_plot_nonModi_CLUSTERED)
##----------plot line

(table(explained_all_plot_nonModi_CLUSTERED$group)/5)/(nrow(explained_all_plot_nonModi_CLUSTERED)/5)*100

unique(explained_all_plot_nonModi_CLUSTERED$group)
unique(explained_all_plot_nonModi_CLUSTERED$final_clust)

explained_all_plot_nonModi_CLUSTERED$group = factor(explained_all_plot_nonModi_CLUSTERED$group, levels=c("both","mst1","mst2","not_explained"))
explained_all_plot_nonModi_CLUSTERED$final_clust = factor(explained_all_plot_nonModi_CLUSTERED$final_clust, levels=rev(c("1","2","3","4","5","not clustered")))

  p_S72_sterm_woModi_clustered <- ggplot(explained_all_plot_nonModi_CLUSTERED,aes(x=sample,y=allele_frequency,group=site,color=final_clust,fill=final_clust, text =paste("effect:", effect,"\nsignficance:",significance,"\ngeneName:",geneName,"\nPreferred Name:",Preferred_name,"\nEC:",EC,"\nKEGG_ko:",KEGG_ko,"\nCAZy:",CAZy,"\nCOG:",COG_Functional_Category,"\nCluster:",final_clust)))+ geom_line(size=0.1, alpha=.5)+ 
       facet_grid(group~.)+
   labs("",
         x="",
         # y="",
          y="alternative allele frequency")+
        scale_fill_manual(values=rev(colors_substrains))+
        scale_color_manual(values=rev(colors_substrains))+
       scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.25,0.5,0.75,1),labels=c(0,0.25,0.5,0.75,1), expand = c(0, 0)) +
  theme_classic()+
          ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=6),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
           legend.position="top",
       legend.title = element_blank(),
       axis.text.y = element_text(size=6),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm"),legend.direction = "horizontal"
       )

p_S72_sterm_woModi_clustered
svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_CLUSTERED_02.svg",width=3,height=4.2)
# png("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_sterm.png",width=1800,height=1600,res=300)


p_S72_sterm_woModi_clustered

dev.off()

ggp <- ggplotly(p_S72_sterm_woModi_clustered)
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_all_clustered_woModi.html")




##----------plot box

# explained_s72_plot_nonModi_CLUSTERED$final_clust = factor(explained_s72_plot_nonModi_CLUSTERED$final_clust, levels=c("1","2","3","NA"))

  p_S72_sterm_woModi_clustered_box <- ggplot(explained_all_plot_nonModi_CLUSTERED,aes(x=final_clust,y=allele_frequency,color=final_clust,fill=final_clust))+ geom_boxplot(alpha=.5)+ 
       facet_grid(group~.)+
   labs("",
         x="",
         # y="",
          y="allele frequency")+
       # scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,1))+
 theme(#axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            # plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )

p_S72_sterm_woModi_clustered_box


ggp <- ggplotly(p_S72_sterm_woModi_clustered_box)
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_all_clustered_woModi.html")


###-----------------
##Tsne plots
###-----------------

p1 <- Sterm_both_suplementaryTsne+labs(title = "explained by both strains")
p2 <- Sterm_S50_suplementaryTsne+labs(title = "explained by mst1")
p3 <- Sterm_S72_suplementaryTsne+labs(title = "explained by mst2")
p4 <- Sterm_notExplained_suplementaryTsne+labs(title = "not-explained")

grid.arrange(p1,p2,p3,p4, ncol=1)

library(grid)
library(gridExtra)
 svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_tsne_all.svg",width=2.5,height=10)

   grid.arrange(p1,p2,p3,p4, ncol=1)

dev.off()
```


```{r}
##plot

  library("tidyverse")
library("officer")
library("rvg")

# read_pptx() %>%
#     add_slide(layout = "Title and Content", master = "Office Theme") %>%
#     ph_with_vg(code = print(grid.arrange(p1,p2, p3,p0,ncol=1, nrow=4,heights=c(1.15,1,1,1.55))), type = "body",height = 7, width = 8) %>% 
#     print(target = "~/Desktop/presenations/my_presentations/20190919_VUA/slides_03.pptx")

  
     # grid.arrange(p0,p1, p2,p3,ncol=1, nrow=4,heights=c(1,1,1,2))
  # pdf("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored.pdf",width=9.8,height=20,paper='special')
# svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored.svg",width=5,height=4)
# png("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_sterm.png",width=1800,height=1600,res=300)
#  svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_sterm.svg",width=4,height=7)
# 
# grid.arrange(p1,p2, p3,p0,ncol=1, nrow=4,heights=c(1.15,1,1,1.55))
# 
# dev.off()

  
  ##============
  #look at special things
  ##============
  ##-------
  ##decreasing snp
  ##-------
library(plotly)
ggp <- ggplotly(p1)
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_sterm_both_decreasing.html")
sample_relative_temp_cleaned[grep("RMK202_1049174_ACAA",sample_relative_temp_cleaned$site),]
sample_relative_temp_cleaned[grep("RMK202_1049189_",sample_relative_temp_cleaned$site),]

  ##-------
  ##increasing snp
  ##-------
library(plotly)
ggp <- ggplotly(p2)
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_sterm_both_increase.html")
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_sterm_both_increase.html")


sample_relative_temp_cleaned[grep("RMK202_1624846",sample_relative_temp_cleaned$site),]
sample_relative_temp_cleaned[grep("RMK202_1049189_",sample_relative_temp_cleaned$site),]

  ##-------
  ##S72
  ##-------
library(plotly)
ggp <- ggplotly(p3)
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_sterm_S72.html")

  ##-------
  ##not explained
  ##-------
library(plotly)
ggp <- ggplotly(p0)
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_sterm_S72.html")
sample_relative_temp_cleaned[grep("RMK202_1353308_",sample_relative_temp_cleaned$site),]


##------------------------
##ldel
##------------------------


##----------START

# both_metaIso_ldel <- meta_ldel[onlysecond_sterm %in% onlyfirst_sterm,]
# nrow(both_metaIso_ldel)
# only_meta_ldel <- meta_ldel[!onlysecond_sterm %in% onlyfirst_sterm,]
# nrow(only_meta_ldel)
# # sum(duplicated(only_meta_sterm))
# only_isolates_ldel <- sample_relative_safe_final_iso_cleaned_ldel[!onlyfirst_sterm %in% onlysecond_sterm,]
# nrow(only_isolates_ldel)

###-----not explained by isolates
  not_explained_ldel_plot <- gather(only_meta_ldel, sample, allele_frequency, "Lyo_202_2012":"Versand_202", factor_key=TRUE)
  # not_explained_sterm_plot$site <- paste(not_explained_sterm_plot$X.CHROM,not_explained_sterm_plot$POS,not_explained_sterm_plot$REF,not_explained_sterm_plot$ALT,sep="_")

  not_explained_ldel_plot$group <- "not_explained"
  
  boxplot_not_explained_ldel <- data.frame(name=(only_meta_ldel[,1]),rowSums=rowSums(only_meta_ldel[,-1]),name="not explained\nldel")

###-----explained by isolates

   explained_isolates_ldel_plot <- gather(both_metaIso_ldel, sample, allele_frequency, "Lyo_202_2012":"Versand_202", factor_key=TRUE)
  # not_explained_sterm_plot$site <- paste(not_explained_sterm_plot$X.CHROM,not_explained_sterm_plot$POS,not_explained_sterm_plot$REF,not_explained_sterm_plot$ALT,sep="_")

  explained_isolates_ldel_plot$group <- "explained_by_isolates"

  boxplot_both_metaIso_ldel <- data.frame(name=(both_metaIso_ldel[,1]),rowSums=rowSums(both_metaIso_ldel[,-1]),name="explained\nldel")

##---------------plot

  p1 <- ggplot(explained_isolates_ldel_plot,aes(x=sample,y=allele_frequency,group=site,color=group,fill=group))+ geom_line(size=0.2, alpha=.2)+ 
      # facet_grid(species~.)+
   labs("",
         x="",
         # y="",
          y="explained by\nisolates")+
       scale_fill_manual(values="black")+
      scale_color_manual(values="black")+
           scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0),limits = c(0,0.2)) +
      scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
    # ylim(0,0.2)+
  theme_classic()+
 theme(#axis.text.x = element_text(angle = 75, hjust = 1,size=9),
       axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
         plot.margin=unit(c(t = 0.5, r = 0.5, b = 0, l = 0.1),"cm")
       )
  
  # p1
  
  
  p0 <- ggplot(not_explained_ldel_plot,aes(x=sample,y=allele_frequency,group=site,color=group,fill=group))+ geom_line(size=0.2, alpha=.2)+ 
      # facet_grid(species~.)+
   labs("",
         x="",
         # y="",
          y="not\nexplained")+
       scale_fill_manual(values="gray41")+
      scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0),limits = c(0,0.2)) +
     scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
    # ylim(0,0.2)+
  theme_classic()+
 theme(#axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
  
  # p0

  
  # pfinal_ldel <-  grid.arrange(p1, p0, ncol=1, nrow=2,heights=c(1,1.5))
   
#    
# library(plotly)
# ggp <- ggplotly(p0)
# 
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_ldel.html")
#   

  # pdf("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored.pdf",width=9.8,height=20,paper='special')
# svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored.svg",width=5,height=4)
# png("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_ldel.png",width=1800,height=1600,res=300)
# 
#    grid.arrange(p1, p0, ncol=1, nrow=2,heights=c(1,1.5))
# 
# dev.off()

##---------------Nucleotide diversity

nucleotid_div_all <- rbind(nucleotid_div_sterm,nucleotid_div)

all_colours <- rev(c("#EB4D4D","#FAA0A0","#10B552","#36E37B","#6CF5A3","#C5F6FA","#99E9F2") ) 

nucleotid_div_all$samples = factor(nucleotid_div_all$samples, levels=c("Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202"))
library(plyr)
nucleotid_div_all$samples_renamed <- revalue(nucleotid_div_all$samples, c("Lyo_202_2012"="Lyoampulle\n2012", "Lyo_202_2014"="Lyoampulle\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018"))
nucleotid_div_all$samples_renamed = factor(nucleotid_div_all$samples_renamed, levels=c("Lyoampulle\n2012","Lyoampulle\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018"))



 pnucdiv <- ggplot(nucleotid_div_all,aes(x=samples_renamed,y=div,group=group,fill=group,color=group))+ geom_point(size=3)+ geom_line()+ #stat_summary(fun.y=sum, geom="line")+
   labs("",
         x="",
         y="nucleotide\ndiversity [%]")+
   scale_color_manual(values=c("#10B552","#EB4D4D"))+
          scale_x_discrete( expand = c(0, 0)) +
   scale_y_continuous(breaks =c(0,0.4,0.8),labels=c(0,0.4,0.8),expand = c(0, 0),limits = c(-0.1,0.9)) +
  theme_classic()+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,vjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
        plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       # legend.justification=c(1,1), legend.position=c(0.7,0.7),
       # legend.title = element_blank()
       )
  pnucdiv
  
#   
#   png("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_ldel.png",width=1800,height=1600,res=300)
# 
#    grid.arrange(p1, p0,pnucdiv, ncol=1, nrow=3,heights=c(1,1,1.5))
# 
# dev.off()

##---------boxplot rowsums of alternative allele frequencies

boxplot_data_rowsums <- rbind(boxplot_explained_sterm_both,boxplot_explained_s50,boxplot_explained_s72,boxplot_not_explained_sterm,boxplot_both_metaIso_ldel, boxplot_not_explained_ldel)



PSummedFrequency <- ggplot(boxplot_data_rowsums,aes(y=rowSums,x=name.1,color=name.1))+geom_boxplot()+
   labs(x="",
         y="summed\nfrequency")+
   #scale_color_manual(values=c("#10B552","#EB4D4D"))+
          #scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.4,0.8),labels=c(0,0.4,0.8),expand = c(0, 0),limits = c(-0.1,0.9)) +
  theme_classic()+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
        plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.38),"cm")
       # legend.justification=c(1,1), legend.position=c(0.7,0.7),
       # legend.title = element_blank()
       )

 PSummedFrequency
  # png("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_ldel.png",width=1800,height=1600,res=300)
 svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_explained_ldel.svg",width=4,height=7)

   grid.arrange(p1, p0,pnucdiv,PSummedFrequency, ncol=1, nrow=4,heights=c(1.3,1,1.8,2))

dev.off()

###t-test


 rbind(boxplot_explained_sterm_both,boxplot_explained_s50,boxplot_explained_s72,boxplot_not_explained_sterm,boxplot_both_metaIso_ldel, boxplot_not_explained_ldel)
#=============
##-ldel
#=============
 # summary(boxplot_both_metaIso_ldel)
 #  summary(boxplot_not_explained_ldel)

ldel_Ttest <- t.test(boxplot_both_metaIso_ldel$rowSums,boxplot_not_explained_ldel$rowSums)
ldel_Ttest$p.value
#=============
#====-sterm
#=============
 # summary(boxplot_both_metaIso_ldel)
 #  summary(boxplot_not_explained_ldel)

lsterm_Ttest <- t.test(boxplot_explained_sterm_both$rowSums,boxplot_explained_s72$rowSums)
lsterm_Ttest$p.value

lsterm_Ttest <- t.test(boxplot_explained_sterm_both$rowSums,boxplot_explained_s72$rowSums)
lsterm_Ttest$p.value

```

######Go-Analysis

```{r}

table(explained_all_plot_nonModi_CLUSTERED$significance)
table(str_split_fixed(explained_all_plot_nonModi_CLUSTERED$PGAP_name, "-", 2)[,2])
explained_all_plot_nonModi_CLUSTERED$geneNameprep <- str_split_fixed(explained_all_plot_nonModi_CLUSTERED$PGAP_name, "-", 2)[,2]

##================
###prepare for topGO analysis
##================

library(readr)
library(Rgraphviz)
library(topGO)

##------allgenes
geneID2GO_ldel <- readMappings(file = "~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog_GO/Lactobacillus_delbrueckii_RMK202_pgap_query_seqs.fa.emapper.annotations") 
geneID2GO_sterm <- readMappings(file = "~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog_GO/Streptococcus_thermophilus_RMK202_pgap_query_seqs.fa.emapper.annotations") 
length(geneID2GO_sterm)

##============
##topGO 
##============
##------without modifier snp genes


geneUniverse <- names(geneID2GO_sterm) 
genesOfInterest <- as.character(explained_all_plot_nonModi_CLUSTERED$geneNameprep)
length(genesOfInterest)


table(genesOfInterest)

genesOfInterest <- genesOfInterest[genesOfInterest %in% geneUniverse]

 geneList <- factor(as.integer(geneUniverse %in% genesOfInterest))
 names(geneList) <- geneUniverse
 
 
 myGOdata <- new("topGOdata", description="My project", ontology="BP", allGenes=geneList,  annot = annFUN.gene2GO, gene2GO = geneID2GO_sterm)
 
 sg <- sigGenes(myGOdata)
 str(sg)
numSigGenes(myGOdata)
resultFisher <- runTest(myGOdata, algorithm="classic", statistic="fisher")
resultFisher

allRes <- GenTable(myGOdata, classicFisher = resultFisher, orderBy = "resultFisher", ranksOf = "classicFisher", topNodes = 200)
allRes

 showSigOfNodes(myGOdata, score(resultFisher), firstSigNodes = 5, useInfo ='all')
 printGraph(myGOdata, resultFisher, firstSigNodes = 5, fn.prefix = "tGO", useInfo = "all", pdfSW = TRUE)
 
 
 length(usedGO(myGOdata))
 
  write.csv(allRes,"~/Desktop/presenations/my_presentations/20191007_groupmeeting/figures/Goterms.csv")
  
  
##------high significance mutations
 
annotation_sterm_snps_HIGH <- explained_all_plot_nonModi_CLUSTERED[explained_all_plot_nonModi_CLUSTERED$significance=="HIGH",]
table(annotation_sterm_snps_HIGH$significance)
# - as.character(explained_all_plot_nonModi_CLUSTERED$)
genesOfInterest <- as.character(annotation_sterm_snps_HIGH$geneNameprep)
genesOfInterest <- genesOfInterest[genesOfInterest %in% geneUniverse]

geneList <- factor(as.integer(geneUniverse %in% genesOfInterest))
names(geneList) <- geneUniverse
 
myGOdata <- new("topGOdata", description="My project", ontology="BP", allGenes=geneList,  annot = annFUN.gene2GO, gene2GO = geneID2GO_sterm)
 
sg <- sigGenes(myGOdata)
numSigGenes(myGOdata)

resultFisher <- runTest(myGOdata, algorithm="classic", statistic="fisher")
resultFisher

allRes <- GenTable(myGOdata, classicFisher = resultFisher, orderBy = "resultFisher", ranksOf = "classicFisher", topNodes = 200)
allRes

  write.csv(allRes,"~/Desktop/presenations/my_presentations/20191007_groupmeeting/figures/Goterms_loseOFfunction.csv")


##------moderate significance mutations
 
annotation_sterm_snps_HIGH <- explained_all_plot_nonModi_CLUSTERED[explained_all_plot_nonModi_CLUSTERED$significance=="MODERATE",]
table(annotation_sterm_snps_HIGH$significance)
# - as.character(explained_all_plot_nonModi_CLUSTERED$)
genesOfInterest <- as.character(annotation_sterm_snps_HIGH$geneNameprep)
genesOfInterest <- genesOfInterest[genesOfInterest %in% geneUniverse]

geneList <- factor(as.integer(geneUniverse %in% genesOfInterest))
names(geneList) <- geneUniverse
 
myGOdata <- new("topGOdata", description="My project", ontology="BP", allGenes=geneList,  annot = annFUN.gene2GO, gene2GO = geneID2GO_sterm)
 
sg <- sigGenes(myGOdata)
numSigGenes(myGOdata)

resultFisher <- runTest(myGOdata, algorithm="classic", statistic="fisher")
resultFisher

allRes <- GenTable(myGOdata, classicFisher = resultFisher, orderBy = "resultFisher", ranksOf = "classicFisher", topNodes = 200)
allRes

  write.csv(allRes,"~/Desktop/presenations/my_presentations/20191007_groupmeeting/figures/Goterms_MODERATE.csv")

```



#####look at interesting genes

```{bash}

##sterm decreasing

grep "1049189" /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata/SnpEff/Streptococcus_thermophilus_RMK202_pgap/variantCallsfreebayes_all_Streptococcus_thermophilus_RMK202_snpEff.vcf 

grep "1049174" /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata/SnpEff/Streptococcus_thermophilus_RMK202_pgap/variantCallsfreebayes_all_Streptococcus_thermophilus_RMK202_snpEff.vcf

grep "pgaptmp_001104" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/*_pgap.gff


##sterm not explained strain all same location 

grep "347968" /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1_final_kneaddata/SnpEff/Streptococcus_thermophilus_RMK202_pgap/variantCallsfreebayes_all_Streptococcus_thermophilus_RMK202_snpEff.vcf 

grep "pgaptmp_000374" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/02_PGAP/ST_singu/*_pgap.gff

```


#####tsne test

```{r}

library(ggplot2)
library(dplyr)
library(magrittr)
library(ggrepel)

set.seed(123456)
N = 5000
D = 100
data.norm = matrix(rnorm(N * D, 2), N)
groups.probs = runif(10)
groups = sample(1:10, N, TRUE, groups.probs/sum(groups.probs))
for (gp in unique(groups)) {
    dev = rep(1, D)
    dev[sample.int(D, 3)] = runif(3, -10, 10)
    data.norm[which(groups == gp), ] = data.norm[which(groups == 
        gp), ] %*% diag(dev)
}
info.norm = tibble(truth = factor(groups))


data.norm <- as.matrix(sample_relative_safe_final_tsne_safe_sterm_woDuplicates[,c(-1,-2)])

pca.norm = prcomp(data.norm)
info.norm %<>% cbind(pca.norm$x[, 1:4])
ggplot(info.norm, aes(x = PC1, y = PC2, colour = truth)) + 
    geom_point(alpha = 0.3) + theme_bw()


library(Rtsne)
tsne.norm = Rtsne(pca.norm$x, pca = FALSE)
tsne_plot <- as.data.frame(tsne.norm$Y)
tsne_plot$site <- sample_relative_safe_final_tsne_safe_sterm_woDuplicates[,1]
tsne_plot$contig <- paste(str_split_fixed(tsne_plot$site, "_", 5)[,1],str_split_fixed(tsne_plot$site, "_", 5)[,2],sep = "_")


tsne_plot$species <- tsne_plot$contig

##plot blank
ggplot(tsne_plot,aes(x=V1,y=V2))+geom_point()



# info.norm %<>% mutate(tsne1 = tsne.norm$Y[, 1], tsne2 = tsne.norm$Y[,2])
# ggplot(info.norm, aes(x = tsne.norm$Y[, 1], y = tsne.norm$Y[,2])) + 
#     geom_point(alpha = 0.3) + theme_bw()

##ward
hc.norm = hclust(dist(tsne_plot), method = "ward.D")
tsne_plot$hclust =factor(cutree(hc.norm, 7))
hc.norm.cent = tsne_plot %>% group_by(hclust) %>% select(V1, 
    V2) %>% summarize_all(mean)
plotTsne <- ggplot(tsne_plot, aes(x = V1, y = V2, colour = hclust)) + 
  facet_grid(species~.)+
    geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label = hclust), 
    data = hc.norm.cent) + guides(colour = FALSE) + 
    ggtitle("Linkage criterion: Ward")
plotTsne

##kmeans
hc.norm = hclust(dist(tsne_plot))
tsne_plot$hclust =factor(cutree(hc.norm, 6))
hc.norm.cent = tsne_plot %>% group_by(hclust) %>% select(V1, 
    V2) %>% summarize_all(mean)
plotTsne <- ggplot(tsne_plot, aes(x = V1, y = V2, colour = hclust)) + 
  facet_grid(species~.)+
    geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label = hclust), 
    data = hc.norm.cent) + guides(colour = FALSE) + 
    ggtitle("Linkage criterion: kmeans")
plotTsne


km.norm = kmeans(tsne_plot, 9, nstart = 100)
tsne_plot$kmeans = factor(km.norm$cluster)

km.cent = info.norm %>% group_by(kmeans) %>% select(tsne1, 
    tsne2) %>% summarize_all(mean)
ggplot(info.norm, aes(x = tsne1, y = tsne2, colour = kmeans)) + 
    geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label = kmeans), 
    data = km.cent) + guides(colour = FALSE) + ggtitle("9 clusters")


```


#####t-sne

Here, I try to cluster the previous snp profiles with t-sne. explained [here](https://www.analyticsvidhya.com/blog/2017/01/t-sne-implementation-r-python/)


```{r}
library(Rtsne)
library(ggrepel)

##---------prep not explained
not_explained_sterm_plot_prep <- not_explained_sterm_plot[,c(-2,-4)]
# not_explained_sterm_plot_prep$sample_renamed <- factor(not_explained_sterm_plot_prep$sample_renamed)
notExplained_sterm_wide <- spread(not_explained_sterm_plot_prep, sample_renamed, allele_frequency)
nrow(notExplained_sterm_wide)



##-------------------sterm

sample_relative_safe_final_tsne_safe_sterm <- notExplained_sterm_wide
# sample_relative_safe_final_tsne_safe_sterm <- sample_relative_safe_final_meta_cleaned_02[grep("S_thermophilus_RMK202",sample_relative_safe_final_meta_cleaned_02$site),]
#    nrow(sample_relative_safe_final_tsne_safe_sterm)

   # sample_relative_safe_final_tsne_safe_sterm[which(rowSums(is.na(sample_relative_safe_final_tsne_safe_sterm[,c("Lyo_202_2012","Lyo_202_2014","RMK202","Konserve_202","Versand_202")]))!=0),]

##duplicate rows
nrow(sample_relative_safe_final_tsne_safe_sterm)
sample_relative_safe_final_tsne_safe_sterm_woDuplicates <- sample_relative_safe_final_tsne_safe_sterm[!duplicated(sample_relative_safe_final_tsne_safe_sterm[,-1]), ]
nrow(sample_relative_safe_final_tsne_safe_sterm_woDuplicates)   
sample_relative_safe_final_tsne_safe_sterm_woDuplicates <- sample_relative_safe_final_tsne_safe_sterm_woDuplicates[!rowSums(is.na(sample_relative_safe_final_tsne_safe_sterm_woDuplicates))>0,]

library(philentropy)
dist_sterm <- distance(as.matrix(sample_relative_safe_final_tsne_safe_sterm_woDuplicates[,c(-1,-2)]), method = "jaccard")

tsne <- Rtsne(dist_sterm, dims = 2, perplexity=30, verbose=TRUE, max_iter = 500)
 

# tsne <- Rtsne(sample_relative_safe_final_tsne_safe_sterm_woDuplicates[,c("Lyoampulle\n2012","Lyoampulle\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018")], dims = 2, perplexity=30, verbose=TRUE, max_iter = 500)
 # tsne <- Rtsne(sample_relative_safe_final_tsne_safe_sterm_woDuplicates[,c("Lyo_202_2012","Lyo_202_2014","RMK202","Konserve_202","Versand_202")], dims = 2, perplexity=30, verbose=TRUE, max_iter = 500)
tsne_plot <- as.data.frame(tsne$Y)
tsne_plot$site <- sample_relative_safe_final_tsne_safe_sterm_woDuplicates[,1]
tsne_plot$contig <- paste(str_split_fixed(tsne_plot$site, "_", 5)[,1],str_split_fixed(tsne_plot$site, "_", 5)[,2],sep = "_")




unique(tsne_plot$contig)
tsne_plot$species <- tsne_plot$contig
# tsne_plot$species <- ifelse(tsne_plot$contig=="contig_1","S.thermophilus",
#                                              ifelse(tsne_plot$contig=="contig_2","L.delbrueckii",
#                                                     ifelse(tsne_plot$contig=="contig_3","Lactococcus phage",
#                                                            ifelse(tsne_plot$contig=="contig_4","Cow",
#                                                                   ifelse(tsne_plot$contig=="contig_5","L.delbrueckii plasmid",
#                                                                         ifelse(tsne_plot$contig=="plasmidSpades_CowMito","Cow mitogenome",NA))))))

unique(tsne_plot$species)

##plot blank
ggplot(tsne_plot,aes(x=V1,y=V2))+geom_point()

##clusters
hc.norm = hclust(dist(tsne_plot[,c(1,2)]))
tsne_plot$hclust = factor(cutree(hc.norm, 7))
hc.norm.cent = tsne_plot %>% group_by(hclust) %>% select(V1, 
    V2) %>% summarize_all(mean)
plotTsne <- ggplot(tsne_plot, aes(x = V1, y = V2, colour = hclust)) + 
  facet_grid(species~.)+
    geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label = hclust), 
    data = hc.norm.cent) + guides(colour = FALSE) + 
    ggtitle("Linkage criterion: Complete")

  # pdf("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_tsnePlot_sterm.pdf",width=5,height=5,paper='special')
  # svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_tsnePlot_sterm.svg",width=4,height=4)
  # png("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_tsnePlot_sterm.png",width=1600,height=1600,res=300)
plotTsne

  
library(plotly)
ggp <- ggplotly(plotTsne)
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_notexplained_sterm_colored_tsne.html")



# dev.off()
tsne_plot_sterm <- tsne_plot
##------------------Ldel
# 
# sample_relative_safe_final_tsne_safe_ldel_plasmid <- sample_relative_safe_final_meta_cleaned_02[grep("L_delbrueckii_RMK202_",sample_relative_safe_final_meta_cleaned_02$site),]
#   
# ##duplicate rows
# nrow(sample_relative_safe_final_tsne_safe_ldel_plasmid)
# sample_relative_safe_final_tsne_safe_ldel_woDuplicates <- sample_relative_safe_final_tsne_safe_ldel_plasmid[!duplicated(sample_relative_safe_final_tsne_safe_ldel_plasmid[,-1]), ]
# nrow(sample_relative_safe_final_tsne_safe_ldel_woDuplicates)   
# sample_relative_safe_final_tsne_safe_ldel_woDuplicates <- sample_relative_safe_final_tsne_safe_ldel_woDuplicates[!rowSums(is.na(sample_relative_safe_final_tsne_safe_ldel_woDuplicates))>0,]
# library(philentropy)
# distance(sample_relative_safe_final_tsne_safe_ldel_woDuplicates[,-1], method = "jaccard")
# 
# 
# tsne <- Rtsne(sample_relative_safe_final_tsne_safe_ldel_woDuplicates[,-1], dims = 2, perplexity=30, verbose=TRUE, max_iter = 500,)
# tsne_plot <- as.data.frame(tsne$Y)
# tsne_plot$site <- sample_relative_safe_final_tsne_safe_ldel_woDuplicates[,1]
# tsne_plot$contig <- paste(str_split_fixed(tsne_plot$site, "_", 5)[,1],str_split_fixed(tsne_plot$site, "_", 5)[,2],sep = "_")
# 
# 
# 
# 
# unique(tsne_plot$contig)
# tsne_plot$species <- tsne_plot$contig
# # tsne_plot$species <- ifelse(tsne_plot$contig=="contig_1","S.thermophilus",
# #                                              ifelse(tsne_plot$contig=="contig_2","L.delbrueckii",
# #                                                     ifelse(tsne_plot$contig=="contig_3","Lactococcus phage",
# #                                                            ifelse(tsne_plot$contig=="contig_4","Cow",
# #                                                                   ifelse(tsne_plot$contig=="contig_5","L.delbrueckii plasmid",
# #                                                                         ifelse(tsne_plot$contig=="plasmidSpades_CowMito","Cow mitogenome",NA))))))
# 
# unique(tsne_plot$species)
# 
# ##plot blank
# ggplot(tsne_plot,aes(x=V1,y=V2))+geom_point()
# 
# ##clusters
# hc.norm = hclust(dist(tsne_plot[,c(1,2)]))
# tsne_plot$hclust = factor(cutree(hc.norm, 20))
# hc.norm.cent = tsne_plot %>% group_by(hclust) %>% select(V1, 
#     V2) %>% summarize_all(mean)
# plotTsne <- ggplot(tsne_plot, aes(x = V1, y = V2, colour = hclust)) + 
#   facet_grid(species~.)+
#     geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label = hclust), 
#     data = hc.norm.cent) + guides(colour = FALSE) + 
#     ggtitle("Linkage criterion: Complete")
#   plotTsne
# 
#   # pdf("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_tsnePlot_ldel.pdf",width=5,height=5,paper='special')
# # svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_tsnePlot_ldel.svg",width=4,height=4)
# png("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_tsnePlot_ldel.png",width=1600,height=1600,res=300)
#   plotTsne
# 
# dev.off()
# tsne_plot_ldel <- tsne_plot
# tsne_plot_all <- rbind(tsne_plot_sterm,tsne_plot_ldel)


```

######add tsne annoation to plot

```{r}

not_explained_sterm_plot$grouping <- "none"
not_explained_sterm_plot$grouping <- tsne_plot_sterm[match(not_explained_sterm_plot$site,tsne_plot_sterm$site),"hclust"]
table(not_explained_sterm_plot$grouping)

# unique(sample_relative_safe_final$sample_02)
# sample_relative_safe_final_tsne_safe_ldel<- sample_relative_safe_final[grep("L_delbrueckii_RMK202_",sample_relative_safe_final$site),]
# sample_relative_safe_final_tsne_safe_sterm <- sample_relative_safe_final[grep("S_thermophilus_RMK202",sample_relative_safe_final$site),]
# 
# nrow(sample_relative_safe_final_tsne_safe_sterm)
# sample_relative_safe_final_reduce <- rbind(sample_relative_safe_final_tsne_safe_ldel,sample_relative_safe_final_tsne_safe_sterm)


  
  pcoloredtsne <- ggplot(not_explained_sterm_plot,aes(x=sample_renamed,y=allele_frequency,group=site,color=grouping,fill=grouping))+ geom_line(size=0.1, alpha=.1)+ 
      # facet_grid(species~.)+
   labs("",
         x="",
         # y="",
          y="not\nexplained")+
       # scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="right",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
  
  pcoloredtsne
  
library(plotly)
ggp <- ggplotly(pcoloredtsne)

htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_notexplained_sterm_colored.html")

  
  
  # pdf("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored.pdf",width=9.8,height=20,paper='special')
svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored.svg",width=5,height=4)
# png("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored.png",width=2000,height=1500,res=300)
  pcoloredtsne

dev.off()


```
######look at tsne grouping locations

```{r}

nrow(sample_relative_safe_final_tsne_safe_ldel)
nrow(sample_relative_safe_final_tsne_safe_sterm)
sample_relative_safe_final_reduce <- rbind(sample_relative_safe_final_tsne_safe_ldel,sample_relative_safe_final_tsne_safe_sterm)

##sterm

ggplot(sample_relative_safe_final_tsne_safe_sterm,aes(x=POS,group=grouping,color=grouping,fill=grouping))+geom_density(alpha=0.4)

##ldel

ldel_group1 <- sample_relative_safe_final_tsne_safe_ldel[sample_relative_safe_final_tsne_safe_ldel$grouping=="1",]
ggplot(ldel_group1,aes(x=POS))+geom_histogram()

ldel_group2 <- sample_relative_safe_final_tsne_safe_ldel[sample_relative_safe_final_tsne_safe_ldel$grouping=="2",]
ggplot(ldel_group2,aes(x=POS))+geom_histogram()

ldel_group3 <- sample_relative_safe_final_tsne_safe_ldel[sample_relative_safe_final_tsne_safe_ldel$grouping=="3",]
ggplot(ldel_group3,aes(x=POS))+geom_histogram()

ggplot(sample_relative_safe_final_tsne_safe_ldel,aes(x=POS,group=grouping,color=grouping,fill=grouping))+geom_density(alpha=0.2)

```





The two snps that decrease and are explained by both strains are modifier mutations of a zinc ABC transporter substrate-binding protein AdcA.


The snps that cannot be explained by the two strain seem to be upstream of a ABC transporter ATP-binding protein. 

#####annote the SCOG with 
```{bash}
##=============
###make a query file
##=============

subgroups=Lactobacillus_delbrueckii_group_subgroup_2

for subgroups in $(echo "Lactobacillus_delbrueckii_group_subgroup_2 Streptococcus_thermophilus_group_subgroup_4")
do
#date_fix=Results_May17
date_fix=$(ls /archiv/Dialact/ReferenceDatabase/20190513/Orthofinder/01_normalgrouping/${subgroups}/ )

echo "##=============================="
echo ${subgroups}
echo "##=============================="


rm /archiv/Dialact/ReferenceDatabase/20190513/Orthofinder/01_normalgrouping/${subgroups}/${date_fix}/blastOGs.fasta

for OGs in $(cat /archiv/Dialact/ReferenceDatabase/20190513/Orthofinder/01_normalgrouping/${subgroups}/${date_fix}/Orthogroups/Orthogroups_SingleCopyOrthologues.txt)
do
#OGs=$(echo ${OG} | cut -d '.' -f 1)

## extract first fasta in multiFasta
awk '/^>/{if(N)exit;++N;} {print;}' /archiv/Dialact/ReferenceDatabase/20190513/Orthofinder/01_normalgrouping/${subgroups}/${date_fix}/Orthogroup_Sequences/${OGs}.fa | sed "s/>/>$OGs /g" >> /archiv/Dialact/ReferenceDatabase/20190513/Orthofinder/01_normalgrouping/${subgroups}/${date_fix}/blastOGs.fasta


done # subgroups
done #OG

##=============
###make blast DB
##=============


threads=37

num=1
for names in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt )
  do
  echo ${num}"/4  :" ${names}
  num=$((num+1))
echo "---------------"
${names}
makeblastdb -max_file_sz 10GB -in /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/03_PROKKA_annotation_polishedGenome/${names}/PROKKA_*.ffn \
 -out /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/03_PROKKA_annotation_polishedGenome/${names}/BlastDB -parse_seqids -dbtype nucl

done

##=============
###tblastn query to DB
##=============
threads=37

num=1
for names in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt )
  do
  echo ${num}"/4  :" ${names}
  num=$((num+1))
echo "---------------"
echo ${names}

for subgroups in $(echo "Lactobacillus_delbrueckii_group_subgroup_2 Streptococcus_thermophilus_group_subgroup_4")
do

rm -r /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/BLASTannotation_OG/${names}/${subgroups}
mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/BLASTannotation_OG/${names}/${subgroups}

date_fix=$(ls /archiv/Dialact/ReferenceDatabase/20190513/Orthofinder/01_normalgrouping/${subgroups}/ )

  tblastn -num_threads ${threads} -max_hsps 1 -max_target_seqs 1 -show_gis \
  -query /archiv/Dialact/ReferenceDatabase/20190513/Orthofinder/01_normalgrouping/${subgroups}/${date_fix}/blastOGs.fasta \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen" \
  -db /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/03_PROKKA_annotation_polishedGenome/${names}/BlastDB \
  -out /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/BLASTannotation_OG/${names}/${subgroups}/${subgroups}_${names}_annotation_blast.txt \
  -evalue 0.01 -word_size 4


done #subgroups
done #name


##=============
###create gff file of OGs
##=============

threads=37

num=1
for names in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt )
  do
  echo ${num}"/4  :" ${names}
  num=$((num+1))
echo "---------------"
echo ${names}

for subgroups in $(echo "Lactobacillus_delbrueckii_group_subgroup_2 Streptococcus_thermophilus_group_subgroup_4")
do

rm /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/BLASTannotation_OG/${names}/${subgroups}/${subgroups}_${names}_annotation_blast.gff

echo "============="
echo ${subgroups}

n=$(wc -l  /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/BLASTannotation_OG/${names}/${subgroups}/${subgroups}_${names}_annotation_blast.txt |awk '{print $1}')

for OG in $(seq 1 $n)
do

OGs=$(sed "${OG}q;d" /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/BLASTannotation_OG/${names}/${subgroups}/${subgroups}_${names}_annotation_blast.txt |awk -F "\t" '{print $1}')
gene=$(sed "${OG}q;d" /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/BLASTannotation_OG/${names}/${subgroups}/${subgroups}_${names}_annotation_blast.txt |awk -F "\t" '{print $2}')

#gene=$(echo "th_K2_8h_00003")

#/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/03_PROKKA_annotation_polishedGenome/${names}/

#grep "IfD=$gene;" /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/03_PROKKA_annotation_polishedGenome/${names}/PROKKA_*.gff | awk -F "\t"

grep "ID=${gene};" /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/03_PROKKA_annotation_polishedGenome/${names}/PROKKA_*.gff |awk -F "--" -v og="$OGs" -v spe="$subgroups" -v gens="$gene" '{print $0";OG="og";grouping="spe";gene="gens}' >> \
                /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/BLASTannotation_OG/${names}/${subgroups}/${subgroups}_${names}_annotation_blast.gff


done #OG
done #subgroups
done #name


##=============
##extracting coverage
##=============




num=1
for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name.txt)
  do
  echo ${num}"/37  :" ${names}
  num=$((num+1))
#  rm -r /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/${names}

#  mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/${names}/bwaMapping2DB/
  
  sampleShort=$(echo ${names}| awk -F "_" 'BEGIN{OFS="_"}{print $1,$2}')
  
   NanoSample=$(grep "$sampleShort" /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt) 
  


  threads=30
mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/raw/
rm /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/raw/${names}_mapping_2_OG_${subgroups}.bed

for subgroups in $(echo "Lactobacillus_delbrueckii_group_subgroup_2 Streptococcus_thermophilus_group_subgroup_4")
do
    #rm -r /home/vincent/Projects/2019_pilotplant/05_mapping2DB/${names}/BacteriaDBmapping
  
    #mkdir -p /home/vincent/Projects/2019_pilotplant/05_mapping2DB/${names}/BacteriaDBmapping
    

echo "============="
echo ${subgroups}



bedtools coverage -bed -a /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/BLASTannotation_OG/${NanoSample}/${subgroups}/${subgroups}_${NanoSample}_annotation_blast.gff -b /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam  > \
  /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/raw/${names}_mapping_2_OG_${subgroups}.bed
  
  
done #subgroup
done #names





###----------------change name of bed file
 
 rm -r /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/tmp/
 rm -r /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/final/
 mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/tmp/
  mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/final/

for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name.txt )
    do
    
       # name=th_K2_4h


Day=$(echo $names |cut -d '_' -f 1)
Kessel=$(echo $names |cut -d '_' -f 2)
time=$(echo $names |cut -d '_' -f 3|sed 's/h//g')

 echo ${num}"/37  :" ${names}
  num=$((num+1))
#  rm -r /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/${names}

#  mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/${names}/bwaMapping2DB/
  
  sampleShort=$(echo ${names}| awk -F "_" 'BEGIN{OFS="_"}{print $1,$2}')
  
   NanoSample=$(grep "$sampleShort" /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt) 
  
 # rm /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/tmp/
  mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/tmp/
  mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/final/


for subgroups in $(echo "Lactobacillus_delbrueckii_group_subgroup_2 Streptococcus_thermophilus_group_subgroup_4")
do

#home/vincent/Projects/2019_pilotplant/05_mapping2DB/${name}/BacteriaDBmapping/${name}2BacteriaDB.bed
   
    cat /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/raw/${names}_mapping_2_OG_${subgroups}.bed | sed -e 's/ID=.*;OG=/OG=/g'|awk -F "[\t;]" -v namess="$name" -v days="$Day" -v Kessels="$Kessel" -v times="$time" 'BEGIN{OFS="\t"}{print $1,$4,$5,$7,$9,$10,$11,$12,$13,$14,$15,days,Kessels,times}' |sed 's/OG=//g' |sed 's/grouping=//g'|sed 's/gene=//g' |sed 's/genome=//g' > /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/tmp/${names}_mapping_2_OG_${subgroups}.tmp
  
#cut -d '_' -f 1,2 /home/vincent/Projects/2019_pilotplant/05_mapping2DB/BacteriaDBmapping/tmp.bed > /home/vincent/Projects/2019_pilotplant/05_mapping2DB/BacteriaDBmapping/tmp.species

#awk -F "[-_]" 'BEGIN{OFS="_"}{print $1,$2}' /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/tmp/${names}_mapping_2_OG_${subgroups}.tmp > /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/tmp/${names}_mapping_2_OG_${subgroups}.species


#paste /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/tmp/${names}_mapping_2_OG_${subgroups}.species /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/tmp/${names}_mapping_2_OG_${subgroups}.tmp > /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/final/${names}_mapping_2_OG_${subgroups}_final.bed

#rm /home/vincent/Projects/2019_pilotplant/05_mapping2DB/BacteriaDBmapping/tmp.bed
#rm /home/vincent/Projects/2019_pilotplant/05_mapping2DB/BacteriaDBmapping/tmp.species

#cut -f 1 /home/vincent/Projects/2019_pilotplant/05_mapping2DB/BacteriaDBmapping/${name}2BacteriaDB_final.bed|sort|uniq -c
#awk '{if($1=="_")print $0}' /home/vincent/Projects/2019_pilotplant/05_mapping2DB/BacteriaDBmapping/${name}2BacteriaDB_final.bed

    done ##subgroups
    done ##names
    
    
##local

scp -r vincent@130.223.51.151:/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/OG_mapping_ALL/tmp/ /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/OG_DB
```


```{r}
source("/home/vincent/Desktop/ScriptRepository/R_functions-master/R_functions/g_legend.R") ##for plotting only legends
library(readr)
library(ggplot2)
library(dplyr)
library(tidyverse)
library(grid)
library(gridExtra)
library(cowplot) ## for arranging plots
library(stringr) ##for split string
library(rlist) ##rbind all dataframe of list
library(viridis) ##colour palette
##===================================
#-------------file import
# startingGff <- read_delim("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/phageDBmapping/di_K2_6h2phageDB_final.bed","\t", escape_double = FALSE, col_names = c("species","chr","start","end","orientation","OG","group","gene","genome","genomeSize","count","length_unmapped","geneLength","percent","name","day","kessel","time"),trim_ws = TRUE)

samples_names <- read_csv("~/Desktop/Projects/2019_Pilotplan/samples_ordered.txt", col_names = FALSE)
listcsv_ldel <- as.vector(paste0("/home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/OG_DB/tmp/",samples_names$X1,"_mapping_2_OG_Lactobacillus_delbrueckii_group_subgroup_2.tmp"))
listcsv_sterm <- as.vector(paste0("/home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/OG_DB/tmp/",samples_names$X1,"_mapping_2_OG_Streptococcus_thermophilus_group_subgroup_4.tmp"))

#read_count <- read_delim("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/04_againstAllWithCow/Mulitqc/table_count_final.txt", 
 #   "\t", escape_double = FALSE, col_names = c("name","MioReads"), 
    # trim_ws = TRUE)
# k <- 1
ldf_ldel <- list()
ldf_sterm <- list()

for (k in 1:length(listcsv_ldel)){
 ldf_ldel[[k]] <- read_delim(listcsv_ldel[[k]],"\t", escape_double = FALSE, col_names = c("chr","start","end","orientation","OG","group","gene","count","length_unmapped","geneLength","percent","day","kessel","time"),trim_ws = TRUE)
 ldf_sterm[[k]] <- read_delim(listcsv_sterm[[k]],"\t", escape_double = FALSE, col_names = c("chr","start","end","orientation","OG","group","gene","count","length_unmapped","geneLength","percent","day","kessel","time"),trim_ws = TRUE)

  
  # ldf_ldel[[k]] <- ldf_ldel[[k]]
}
#ldf[[1]]

# head(ldf[[2]])
# 
# dim(startingGff)
# table(startingGff$count)
# startingGff$geneCoverage <- (startingGff$count*600)/startingGff$geneLength

# ggplot(startingGff,aes(x=group,y=geneCoverage,fill=species,colour=species) )+geom_boxplot()+
#     theme(axis.text.x = element_text(angle = 75, hjust = 1),
#           #legend.position="top",
#           legend.justification=c(1,1), legend.position=c(1,1),
#           legend.title = element_blank()
#           )
##===================================
###---TPKM calculations


# startingGff$RawReads <- unlist(read_count[match(startingGff$name,read_count$name),2])
# # aggregate(startingGff$RawReads, list(startingGff$name), mean)
# startingGff$TPKM <- ((startingGff$count*600)/startingGff$RawReads)/startingGff$geneLength
# summary(startingGff$TPKM)


p <- list()
for (i in  1:length(listcsv_sterm)){
  ldf_ldel[[i]]$geneCoverage  <- (ldf_ldel[[i]]$count*600)/ldf_ldel[[i]]$geneLength
  ldf_ldel[[i]]$name  <- paste0(ldf_ldel[[i]]$day,"_",ldf_ldel[[i]]$kessel,"_",ldf_ldel[[i]]$time,"h")
  # ldf_ldel[[i]]$RawReads <- unlist(read_count[match(ldf_ldel[[i]]$name,read_count$name),"MioReads"])
  # ldf_ldel[[i]]$TPKM <- ((ldf_ldel[[i]]$count)/ldf_ldel[[i]]$RawReads)/ldf_ldel[[i]]$geneLength
  ldf_sterm[[i]]$geneCoverage  <- (ldf_sterm[[i]]$count*600)/ldf_sterm[[i]]$geneLength
  ldf_sterm[[i]]$name  <- paste0(ldf_sterm[[i]]$day,"_",ldf_sterm[[i]]$kessel,"_",ldf_sterm[[i]]$time,"h")

}


##===================================
####-----------------plot only grouping=Streptococcus_thermophilus_group_subgroup_4


sterm_s4_list <- list()
p1 <- list()
p2 <- list()
sterm_s4Plot <- list()
OG_sterm_s4_sum_list <- list()
# OG_Ldel_s2_sum_list <- list()
OG_sterm_s4_quantile_list <- as.data.frame(matrix(0, ncol = 7, nrow = 32))
colnames(OG_sterm_s4_quantile_list) <- c("sample","0%","25%","50%","75%","100%","PTR")

for (i in  1:length(ldf_sterm)){

  sterm_s4_list[[i]] <- ldf_sterm[[i]][ldf_sterm[[i]]$group=="Streptococcus_thermophilus_group_subgroup_4",]

  # p1[[i]] <- ggplot() + geom_bar(aes(y = TPKM, x = reorder(OG, -TPKM), fill = genome), data = sterm_s4_list[[i]], stat="identity")+
  #   labs(title = paste0(unique(sterm_s4_list[[i]]$group), " from ",unique(sterm_s4_list[[i]]$name)),
  #        x="Orthologous groups",
  #        y="")+
  #   theme_classic()+
  #   theme(#axis.text.x = element_text(angle = 75, hjust = 1),
  #         axis.text.x = element_blank(),
  #         plot.title = element_text(size=9,hjust = 0.5),
  #         legend.position="none",
  #         #legend.justification=c(1,1), legend.position=c(1,1),
  #         #legend.title = element_blank()
  #         )

  # OG_sterm_s4_sum_list[[i]] <- aggregate(sterm_s4_list[[i]]$TPKM, list(sterm_s4_list[[i]]$OG), sum)
  OG_sterm_s4_sum_list[[i]] <- aggregate(sterm_s4_list[[i]]$geneCoverage, list(sterm_s4_list[[i]]$OG), sum)

   # OG_Ldel_s2_sum_list[[i]] <- aggregate(Ldel_s2_list[[i]]$TPKM, list(Ldel_s2_list[[i]]$OG), sum)
    OG_sterm_s4_quantile_list[i,1:6] <- c(unique(sterm_s4_list[[i]]$name),quantile(OG_sterm_s4_sum_list[[i]]$x))
OG_sterm_s4_quantile_list[i,7] <- as.numeric(OG_sterm_s4_quantile_list[i,5])/as.numeric(OG_sterm_s4_quantile_list[i,3])
  
  
  p1[[i]] <- ggplot(sterm_s4_list[[i]],aes(x=start,y=geneCoverage,fill=name,colour=name) )+geom_point(size = 0.1)+
    facet_grid(~name)+
    # labs(title = paste0(unique(ldf[[i]]$name)),
    #      x="",
    #      y="")+
    theme(axis.text.x = element_text(angle = 75, hjust = 1),
          #axis.text.x =element_blank(),
          plot.title = element_text(size=20),
          legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          )
  
}

pdf("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/figures/coverageOG_sterm.pdf",width=8,height=16,paper='special') 

# do.call(grid.arrange, c(p, ncol=4))

do.call(plot_grid, c(p1, labels = "TPKM", align = 'h', ncol=4))


dev.off()



  OG_sterm_s4_quantile_list$day <- str_split_fixed(OG_sterm_s4_quantile_list$sample, "_", 3)[,1]
  OG_sterm_s4_quantile_list$kessel <- str_split_fixed(OG_sterm_s4_quantile_list$sample, "_", 3)[,2]
  OG_sterm_s4_quantile_list$time <- str_split_fixed(OG_sterm_s4_quantile_list$sample, "_", 3)[,3]
  OG_sterm_s4_quantile_list$treatment <- paste0(OG_sterm_s4_quantile_list$day,"_",OG_sterm_s4_quantile_list$kessel)
 
  OG_sterm_s4_quantile_list$treatment_f = factor(OG_sterm_s4_quantile_list$treatment, levels=c('di_K1','th_K1','di_K2','th_K2'))
  OG_sterm_s4_quantile_list$time_f = factor(OG_sterm_s4_quantile_list$time, levels=c("0h","2h","4h","6h","8h","10h","12h","24h"))

OG_sterm_s4_quantile_list$species <- "S. thermophilus"

ggplot(OG_sterm_s4_quantile_list,aes(x=time_f,y=PTR,group=treatment_f,color=treatment_f,fill=treatment_f))+geom_line()+geom_point()+facet_grid(~kessel)

##viridis

#   pdf("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/Bacteria/Streptococcus_thermophilus_group_subgroup_4_OGs.pdf",width=20,height=32,paper='special')
# do.call(plot_grid, c(sterm_s4Plot, labels = "S_the_subgroup_4", align = 'h', ncol=4))
# 
# 
# dev.off()

# sum_ldels9 <- quantile(OG_ldels9_sum$x)
# sum_ldels9 <- c(unique(ldel_s9$group),unique(ldel_s9$name),unique(ldel_s9$day),unique(ldel_s9$kessel),unique(ldel_s9$time),sum_ldels9)
# names(sum_ldels9) <- c("subgroup","name","day","kessel","time","min","1st","median","3rd","max")
# sum_ldels9

##===================================
####-----------------plot only grouping=Lactobacillus_delbrueckii_group_subgroup_2


Ldel_s2_list <- list()
p1 <- list()
p2 <- list()
Ldel_s2Plot <- list()
OG_Ldel_s2_sum_list <- list()
OG_Ldel_s2_quantile_list <- as.data.frame(matrix(0, ncol = 7, nrow = 32))
colnames(OG_Ldel_s2_quantile_list) <- c("sample","0%","25%","50%","75%","100%","PTR")
# OG_Ldel_s2_quantile_list <- data.frame()

for (i in  1:length(ldf_ldel)){

  Ldel_s2_list[[i]] <- ldf_ldel[[i]][ldf_ldel[[i]]$group=="Lactobacillus_delbrueckii_group_subgroup_2",]

  # p1[[i]] <- ggplot() + geom_bar(aes(y = TPKM, x = reorder(OG, -TPKM), fill = genome), data = Ldel_s2_list[[i]], stat="identity")+
  #   labs(title = paste0(unique(Ldel_s2_list[[i]]$group), " from ",unique(Ldel_s2_list[[i]]$name)),
  #        x="Orthologous groups",
  #        y="")+
  #   theme_classic()+
  #   theme(#axis.text.x = element_text(angle = 75, hjust = 1),
  #         axis.text.x = element_blank(),
  #         plot.title = element_text(size=9,hjust = 0.5),
  #         legend.position="none"
  #         #legend.justification=c(1,1), legend.position=c(1,1),
  #         #legend.title = element_blank()
  #         )

  # OG_Ldel_s2_sum_list[[i]] <- aggregate(Ldel_s2_list[[i]]$TPKM, list(Ldel_s2_list[[i]]$OG), sum)
  OG_Ldel_s2_sum_list[[i]] <- aggregate(Ldel_s2_list[[i]]$geneCoverage, list(Ldel_s2_list[[i]]$OG), sum)
    OG_Ldel_s2_quantile_list[i,1:6] <- c(unique(Ldel_s2_list[[i]]$name),quantile(OG_Ldel_s2_sum_list[[i]]$x))
OG_Ldel_s2_quantile_list[i,7] <- as.numeric(OG_Ldel_s2_quantile_list[i,5])/as.numeric(OG_Ldel_s2_quantile_list[i,3])
  
  
  p1[[i]] <- ggplot(Ldel_s2_list[[i]],aes(x=start,y=geneCoverage,fill=name,colour=name) )+geom_point()+
    facet_grid(~name)+
    # labs(title = paste0(unique(ldf[[i]]$name)),
    #      x="",
    #      y="")+
    theme(#axis.text.x = element_text(angle = 75, hjust = 1),
          axis.text.x =element_blank(),
          plot.title = element_text(size=20),
          legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          )
  
}


# do.call(plot_grid, c(p1, labels = "TPKM", align = 'h', ncol=4))



pdf("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/figures/coverageOG.pdf",width=8,height=16,paper='special') 

# do.call(grid.arrange, c(p, ncol=4))

do.call(plot_grid, c(p1, labels = "TPKM", align = 'h', ncol=4))


dev.off()




  OG_Ldel_s2_quantile_list$day <- str_split_fixed(OG_Ldel_s2_quantile_list$sample, "_", 3)[,1]
  OG_Ldel_s2_quantile_list$kessel <- str_split_fixed(OG_Ldel_s2_quantile_list$sample, "_", 3)[,2]
  OG_Ldel_s2_quantile_list$time <- str_split_fixed(OG_Ldel_s2_quantile_list$sample, "_", 3)[,3]
  OG_Ldel_s2_quantile_list$treatment <- paste0(OG_Ldel_s2_quantile_list$day,"_",OG_Ldel_s2_quantile_list$kessel)
 
  OG_Ldel_s2_quantile_list$treatment_f = factor(OG_Ldel_s2_quantile_list$treatment, levels=c('di_K1','th_K1','di_K2','th_K2'))
  OG_Ldel_s2_quantile_list$time_f = factor(OG_Ldel_s2_quantile_list$time, levels=c("0h","2h","4h","6h","8h","10h","12h","24h"))

OG_Ldel_s2_quantile_list$species <- "L. delbrueckii"

all_OG <- rbind(OG_Ldel_s2_quantile_list,OG_sterm_s4_quantile_list)


ggplot(OG_Ldel_s2_quantile_list,aes(x=time_f,y=PTR,group=treatment_f,color=treatment_f,fill=treatment_f))+geom_line()+geom_point()

ggplot(all_OG,aes(x=time_f,y=PTR,group=treatment_f,color=treatment_f,fill=treatment_f))+geom_line()+geom_point(aes_string(size=as.numeric(all_OG$`50%`)))+facet_grid(~species~kessel)
# 
# 
# all_OG_ONT <- all_OG
# 
# all_OG_ONT$Method <- "ONT_assembled"

# png("~/Desktop/Projects/2019_Pilotplan/PTR_referenceDB.png",width=1500,height=1200,res=300)
# 
# ggplot(all_OG,aes(x=time_f,y=PTR,group=treatment_f,color=treatment_f,fill=treatment_f))+geom_line()+geom_point(aes_string(size=as.numeric(all_OG$`50%`)))+facet_grid(~species~kessel)+
#       labs("",
#          x="time",
#          y="Peak-to-trough ratio")+
#     theme_classic()+
#    # scale_color_viridis(discrete=TRUE)+
#   # scale_fill_viridis(discrete=TRUE)+
#     theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
#       #axis.text.x = element_blank(),
#           legend.position="right"
#           #legend.justification=c(1,1), legend.position=c(1,1),
#           #legend.title = element_blank()
#           )
# dev.off()


##======================
##only K1_di
##======================


samples_names <- read_csv("~/Desktop/Projects/2019_Pilotplan/samples_ordered.txt", col_names = FALSE)
samples_names_di_K1 <- samples_names[grep("di_K1",samples_names$X1),]

listcsv_ldel_di_K1 <- as.vector(paste0("/home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/OG_DB/tmp/",samples_names_di_K1$X1,"_mapping_2_OG_Lactobacillus_delbrueckii_group_subgroup_2.tmp"))
listcsv_sterm_di_K1 <- as.vector(paste0("/home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/OG_DB/tmp/",samples_names_di_K1$X1,"_mapping_2_OG_Streptococcus_thermophilus_group_subgroup_4.tmp"))



ldf_ldel_di_K1 <- list()
ldf_sterm_di_K1 <- list()

for (k in 1:length(listcsv_ldel_di_K1)){
 ldf_ldel_di_K1[[k]] <- read_delim(listcsv_ldel_di_K1[[k]],"\t", escape_double = FALSE, col_names = c("chr","start","end","orientation","OG","group","gene","count","length_unmapped","geneLength","percent","day","kessel","time"),trim_ws = TRUE)
 ldf_sterm_di_K1[[k]] <- read_delim(listcsv_sterm_di_K1[[k]],"\t", escape_double = FALSE, col_names = c("chr","start","end","orientation","OG","group","gene","count","length_unmapped","geneLength","percent","day","kessel","time"),trim_ws = TRUE)

  
}
##===================================
###---TPKM calculations



p <- list()
for (i in  1:length(ldf_sterm_di_K1)){
  ldf_ldel_di_K1[[i]]$geneCoverage  <- (ldf_ldel_di_K1[[i]]$count*600)/ldf_ldel_di_K1[[i]]$geneLength
  ldf_ldel_di_K1[[i]]$name  <- paste0(ldf_ldel_di_K1[[i]]$day,"_",ldf_ldel_di_K1[[i]]$kessel,"_",ldf_ldel_di_K1[[i]]$time,"h")

  
  ldf_sterm_di_K1[[i]]$geneCoverage  <- (ldf_sterm_di_K1[[i]]$count*600)/ldf_sterm_di_K1[[i]]$geneLength
  ldf_sterm_di_K1[[i]]$name  <- paste0(ldf_sterm_di_K1[[i]]$day,"_",ldf_sterm_di_K1[[i]]$kessel,"_",ldf_sterm_di_K1[[i]]$time,"h")

}


##===================================
####-----------------plot only grouping=Streptococcus_thermophilus_group_subgroup_4


sterm_s4_list <- list()
p1 <- list()
p2 <- list()
sterm_s4Plot <- list()
OG_sterm_s4_sum_list <- list()
# OG_Ldel_s2_sum_list <- list()
OG_sterm_s4_quantile_list <- as.data.frame(matrix(0, ncol = 7, nrow = 32))
colnames(OG_sterm_s4_quantile_list) <- c("sample","0%","25%","50%","75%","100%","PTR")

sterm_s4_list_01 <- list()
sterm_s4_list_02 <- list()

for (i in  1:length(ldf_sterm_di_K1)){

  sterm_s4_list[[i]] <- ldf_sterm_di_K1[[i]][ldf_sterm_di_K1[[i]]$group=="Streptococcus_thermophilus_group_subgroup_4",]

  # OG_sterm_s4_sum_list[[i]] <- aggregate(sterm_s4_list[[i]]$TPKM, list(sterm_s4_list[[i]]$OG), sum)
  OG_sterm_s4_sum_list[[i]] <- aggregate(sterm_s4_list[[i]]$geneCoverage, list(sterm_s4_list[[i]]$OG), sum)

   # OG_Ldel_s2_sum_list[[i]] <- aggregate(Ldel_s2_list[[i]]$TPKM, list(Ldel_s2_list[[i]]$OG), sum)
    OG_sterm_s4_quantile_list[i,1:6] <- c(unique(sterm_s4_list[[i]]$name),quantile(OG_sterm_s4_sum_list[[i]]$x))
OG_sterm_s4_quantile_list[i,7] <- as.numeric(OG_sterm_s4_quantile_list[i,5])/as.numeric(OG_sterm_s4_quantile_list[i,3])
  
  
  p1[[i]] <- ggplot(sterm_s4_list[[i]],aes(x=start,y=geneCoverage,fill=name,colour=name) )+geom_point(size = 0.5)+
    facet_grid(~name)+
    # labs(title = paste0(unique(ldf[[i]]$name)),
    #      x="",
    #      y="")+
    theme(axis.text.x = element_text(angle = 75, hjust = 1),
          #axis.text.x =element_blank(),
          plot.title = element_text(size=20),
          legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          )
  
sterm_s4_list_01[[i]] <- sterm_s4_list[[i]][which(sterm_s4_list[[i]]$start<922302),]
sterm_s4_list_02[[i]] <- sterm_s4_list[[i]][which(sterm_s4_list[[i]]$start>922302),]

}

pdf("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/figures/coverageOG_sterm_di_K1.pdf",width=3,height=24,paper='special') 

# do.call(grid.arrange, c(p, ncol=4))

do.call(plot_grid, c(p1, labels = "TPKM", align = 'h', ncol=1))


dev.off()

##linear regression

# summary(sterm_s4_list[[1]]$start)

sterm_s4_list_01[[i]] <- sterm_s4_list[[i]][which(sterm_s4_list[[i]]$start<922302),]
sterm_s4_list_02[[i]] <- sterm_s4_list[[i]][which(sterm_s4_list[[i]]$start>922302),]

linearMod <- lm(sterm_s4_list_01[[1]]$geneCoverage ~ sterm_s4_list_01[[1]]$start)
print(linearMod)

linearMod <- lm(sterm_s4_list_02[[1]]$geneCoverage ~ sterm_s4_list_02[[1]]$start)
print(linearMod)


linearMod <- lm(dist ~ speed, data=sterm_s4_list)  # build linear regression model on full data
print(linearMod)

```



## ANI Figure

add hierachial clustering and order accordingly. This is trivial in the heatmap2 function but non-trivial for ggplot heatmpa. [here](https://jcoliver.github.io/learn-r/008-ggplot-dendrograms-and-heatmaps.html) is a blog post how to do it. 

```{r}
###=================
##Sterm
###=================

##--------------
##snps
##-------------

colnames(annotation_sterm_snps_01)
sterm_snps_core <- annotation_sterm_snps_01
table(sterm_snps_core$X.CHROM.x)
sterm_snps_core <- sterm_snps_core[sterm_snps_core$X.CHROM.x=="S_thermophilus_RMK202",]
table(sterm_snps_core$X.CHROM.x)

sterm_snps_core$Sterm_rmk202 <- TRUE
sterm_snps_core$mst1 <- sterm_snps_core$S50<0.9
sterm_snps_core$mst2 <- sterm_snps_core$S72<0.9

sterm_snps_core <- sterm_snps_core %>% select(site,core,Sterm_rmk202,mst1,mst2)
sterm_snps_core[c("mst1", "mst2")][is.na(sterm_snps_core[c("mst1", "mst2")])] <- TRUE

##--------------
##get length
##-------------
coreGeneNames_sterm_ldel <- read_delim("~/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/coreGeneNames_rmk202_bothLdelSterm.vcf", "\t", escape_double = FALSE, col_names = c("contig","method","annotation","start","end"), trim_ws = TRUE)
# table(coreGeneNames_sterm_ldel$contig)
coreGeneNames_sterm <- coreGeneNames_sterm_ldel[coreGeneNames_sterm_ldel$contig=="S_thermophilus_RMK202",]
# table(coreGeneNames_sterm$contig)
CoreGeneSize_sterm <- sum(coreGeneNames_sterm$end-coreGeneNames_sterm$start)
# 
##--------------
##plot
##-------------
library(stringr)
library(readr)
library(ggplot2)
library(reshape2)

table(sterm_snps_core$core)
sterm_snps_CORE <- sterm_snps_core[which(sterm_snps_core$core=="core"),]
table(sterm_snps_CORE$core)



##------------
##prep
##------------


colnames(sterm_snps_core)[3:5]
names_normal <- colnames(sterm_snps_core)[3:5]
count <- 1 ## triangle above
count <- 3 ; names <- rev(names_normal) ## triangle bellow


sterm_curated_rev_names <- data.frame()
# for (i in 1:length(names)) {  ## triangle above
for (i in rev(1:length(names))) { ## triangle bellow

   rowName <- names[i]
    print(rowName)
    
  for(a in rev(1:length(names))){
    
    colNames <- names[a]
    print(colNames)
    
    


    countSNPs <- sum(sterm_snps_CORE[,rowName]!=sterm_snps_CORE[,colNames])
    ANI_core <- 100*((CoreGeneSize_sterm-countSNPs)/CoreGeneSize_sterm)
    tmp <- data.frame("genome1"=rowName,"genome2"=colNames,"SNPs"=countSNPs,"ANI"=ANI_core)
    sterm_curated_rev_names <- rbind(sterm_curated_rev_names,tmp)
    

  }

# count <- count +1  ## triangle above
count <- count -1  ## triangle bellow

}
sterm_curated_rev_names


##------------
##make dendro of all snps sites
##------------
library(ggdendro)

sterm_matrix <- as.matrix(t(sterm_snps_CORE[, -c(1:2)]))
rownames(sterm_matrix) <- names_normal
otter.dendro <- as.dendrogram(hclust(d = dist(x = sterm_matrix)))


# Create dendro
dendro.plot <- ggdendrogram(data = otter.dendro, rotate = FALSE)
# Preview the plot
print(dendro.plot)

labels(otter.dendro)

svg("~/Desktop/Projects/2018_culturomics/fastani/STERM_SNPs_DENDROGRAM.svg",width=2.5,height=3)
print(dendro.plot)
dev.off()

sterm_curated_rev_names
##reorder according to dendro
levels(sterm_curated_rev_names$genome1)
sterm_curated_rev_names$genome1 = factor(sterm_curated_rev_names$genome1, levels=labels(otter.dendro))
levels(sterm_curated_rev_names$genome1)
levels(sterm_curated_rev_names$genome2)
sterm_curated_rev_names$genome2 = factor(sterm_curated_rev_names$genome2, levels=labels(otter.dendro))
levels(sterm_curated_rev_names$genome2)

# creat Heatmap
# sterm_curated$X3 <- round(sterm_curated$X3,digits = 2)

##-----------
##change names
##-----------
sterm_curated_rev_names$genome1 <- revalue(sterm_curated_rev_names$genome1, c("Sterm_rmk202"="MAG RMK202"))
sterm_curated_rev_names$genome2 <- revalue(sterm_curated_rev_names$genome2, c("Sterm_rmk202"="MAG RMK202"))


##------------
##for SNP legend
##------------

ggheatmap <- ggplot(sterm_curated_rev_names, aes(genome1, genome2, fill = SNPs))+
 geom_tile(color = "white",size=1.1)+
 #scale_fill_gradient2(low = "blue", high = "red", mid = "white", 
  # midpoint = 99.5, limit = c(98.9,100), space = "Lab", 
  # name="ANI") +
  scale_fill_distiller(name = "SNPs", palette = "Blues", direction = -1)+
  # scale_fill_distiller(name = "ANI", palette = "Reds", direction = 1)+
  theme_minimal()+ # minimal theme
 theme(axis.text.y = element_text(size = 12),
      axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1))+
 coord_fixed()
# Print the heatmap
print(ggheatmap)

##------------
##SNPs
##------------


plotFinal_sterm <- ggheatmap + 
geom_text(aes(genome1, genome2, label = SNPs), color = "black", size = 2.5) +
    labs(title=paste0("# core genes :",nrow(coreGeneNames_sterm)," (",CoreGeneSize_sterm,"bp)"))+
  scale_y_discrete(position = 'right')+
theme(
  axis.title.x = element_blank(),
  axis.title.y = element_blank(),
  # axis.text.y.left = element_text(angle = 75, hjust = 1,size=9),
  panel.grid.major = element_blank(),
  # scale_y_date(position = "right"),
  panel.border = element_blank(),
  panel.background = element_blank(),
  axis.ticks = element_blank(),
  # legend.justification = c(1, 0),
  legend.position = "right",
  legend.text = element_text(angle = 0),
  legend.direction = "vertical")+
  guides(fill = guide_colorbar(barwidth = 1, barheight = 7,
                title.position = "top", title.hjust = 0.5))
plotFinal_sterm
##save heatmap
svg("~/Desktop/Projects/2018_culturomics/fastani/Sterm_SNPs_core_absolute_final.svg",width=5.5,height=4)
plotFinal_sterm#+theme(legend.position = "none")
dev.off()


##------------
##ANI 2
##------------
library(viridis)
# creat Heatmap
sterm_curated_rev_names$ANI_2 <- round(sterm_curated_rev_names$ANI,digits = 2)
min(sterm_curated_rev_names$ANI)
ggheatmap <- ggplot(sterm_curated_rev_names, aes(genome1, genome2, fill = ANI))+
 geom_tile(color = "white",size=1.1)+
 # scale_fill_gradient2(low = "red", high = "red4",
 # midpoint = 99.995, limit = c(99.99,100), space = "Lab",
 # name="ANI") +
  # scale_fill_viridis(discrete = FALSE, option = "B",direction = -1)+
  scale_fill_distiller(name = "ANI", palette = "Reds", direction = 1)+
  theme_minimal()+ # minimal theme
 theme(axis.text.y = element_text(size = 12),
      axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1))+
 coord_fixed()
# Print the heatmap
print(ggheatmap)

plotFinal_sterm <- ggheatmap + 
geom_text(aes(genome1, genome2, label = ANI_2), color = "black", size = 2.5) +
  scale_y_discrete(position = 'right')+
  labs(title=paste0("# core genes :",nrow(coreGeneNames_sterm)," (",CoreGeneSize_sterm,"bp)"))+
theme(
  axis.title.x = element_blank(),
  axis.title.y = element_blank(),
  # axis.text = element_blank(),
  panel.grid.major = element_blank(),
  panel.border = element_blank(),
  panel.background = element_blank(),
  axis.ticks = element_blank(),
  # legend.justification = c(1, 0),
  legend.position = "right",
  legend.text = element_text(angle = 0),
  legend.direction = "vertical")+
  guides(fill = guide_colorbar(barwidth = 1, barheight = 7,
                title.position = "top", title.hjust = 0.5))
plotFinal_sterm
##save heatmap
svg("~/Desktop/Projects/2018_culturomics/fastani/Sterm_SNPs_core_ANI.svg",width=5.5,height=4)
plotFinal_sterm#+theme(legend.position = "none")
dev.off()


```


```{r}
###=================
##Ldel
###=================
library(plyr)
##--------------
##snps
##-------------

colnames(annotation_sterm_snps_01)
sterm_snps_core <- annotation_sterm_snps_01
table(sterm_snps_core$X.CHROM.x)
sterm_snps_core <- sterm_snps_core[sterm_snps_core$X.CHROM.x=="L_delbrueckii_RMK202",]
table(sterm_snps_core$X.CHROM.x)

sterm_snps_core$Ldel_rmk202 <- TRUE
sterm_snps_core$mst3 <- sterm_snps_core$L70<0.9
sterm_snps_core$mst10 <- sterm_snps_core$L44<0.9
sterm_snps_core$mst7 <- sterm_snps_core$L108<0.9
sterm_snps_core$mst8 <- sterm_snps_core$L99<0.9
sterm_snps_core$mst6 <- sterm_snps_core$L104<0.9
sterm_snps_core$mst5 <- sterm_snps_core$L80<0.9
sterm_snps_core$mst9 <- sterm_snps_core$L35<0.9
sterm_snps_core$mst4 <- sterm_snps_core$L71<0.9
sterm_snps_core$mst6 <- sterm_snps_core$L104<0.9

sterm_snps_core <- sterm_snps_core %>% select(site,core,Ldel_rmk202,mst3, mst4,mst5, mst6,mst7, mst8,mst9, mst10)
sterm_snps_core[c("mst3", "mst4","mst5", "mst6","mst7", "mst8","mst9", "mst10")][is.na(sterm_snps_core[c("mst3", "mst4","mst5", "mst6","mst7", "mst8","mst9", "mst10")])] <- TRUE


##--------------
##get length
##-------------
coreGeneNames_sterm_ldel <- read_delim("~/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/coreGeneNames_rmk202_bothLdelSterm.vcf", "\t", escape_double = FALSE, col_names = c("contig","method","annotation","start","end"), trim_ws = TRUE)
# table(coreGeneNames_sterm_ldel$contig)
coreGeneNames_sterm <- coreGeneNames_sterm_ldel[coreGeneNames_sterm_ldel$contig=="L_delbrueckii_RMK202",]
# table(coreGeneNames_sterm$contig)
CoreGeneSize_sterm <- sum(coreGeneNames_sterm$end-coreGeneNames_sterm$start)



##--------------
##plot
##-------------
library(stringr)
library(readr)
library(ggplot2)
library(reshape2)

table(sterm_snps_core$core)
sterm_snps_CORE <- sterm_snps_core[which(sterm_snps_core$core=="core"),]
table(sterm_snps_CORE$core)



##------------
##prep
##------------
colnames(sterm_snps_core)
colnames(sterm_snps_core)[3:11]
names_normal <- colnames(sterm_snps_core)[3:11]
# count <- 1 ## triangle above
count <- 9 ; names <- rev(names_normal) ## triangle bellow

sterm_curated_rev_names <- data.frame()
# for (i in 1:length(names)) {  ## triangle above
for (i in rev(1:length(names))) { ## triangle bellow

   rowName <- names[i]
    print(rowName)
    
  for(a in rev(1:length(names))){
    
    colNames <- names[a]
    print(colNames)
    
    


    countSNPs <- sum(sterm_snps_CORE[,rowName]!=sterm_snps_CORE[,colNames])
    ANI_core <- 100*((CoreGeneSize_sterm-countSNPs)/CoreGeneSize_sterm)
    tmp <- data.frame("genome1"=rowName,"genome2"=colNames,"SNPs"=countSNPs,"ANI"=ANI_core)
    sterm_curated_rev_names <- rbind(sterm_curated_rev_names,tmp)
    

  }

# count <- count +1  ## triangle above
count <- count -1  ## triangle bellow

}
sterm_curated_rev_names

##------------
##make dendro of all snps sites
##------------##------------
##make dendro of all snps sites


library(ggdendro)

sterm_matrix <- as.matrix(t(sterm_snps_CORE[, -c(1:2)]))
rownames(sterm_matrix) <- names_normal
otter.dendro <- as.dendrogram(hclust(d = dist(x = sterm_matrix)))


# Create dendro
dendro.plot <- ggdendrogram(data = otter.dendro, rotate = FALSE)
# Preview the plot
print(dendro.plot)

labels(otter.dendro)

svg("~/Desktop/Projects/2018_culturomics/fastani/Ldel_SNPs_DENDROGRAM.svg",width=2.5,height=3)
print(dendro.plot)
dev.off()

sterm_curated_rev_names
##reorder according to dendro
levels(sterm_curated_rev_names$genome1)
sterm_curated_rev_names$genome1 = factor(sterm_curated_rev_names$genome1, levels=labels(otter.dendro))
levels(sterm_curated_rev_names$genome1)
levels(sterm_curated_rev_names$genome2)
sterm_curated_rev_names$genome2 = factor(sterm_curated_rev_names$genome2, levels=labels(otter.dendro))
levels(sterm_curated_rev_names$genome2)

# creat Heatmap
# sterm_curated$X3 <- round(sterm_curated$X3,digits = 2)
##-----------
##change names
##-----------
sterm_curated_rev_names$genome1 <- revalue(sterm_curated_rev_names$genome1, c("Ldel_rmk202"="MAG RMK202"))
sterm_curated_rev_names$genome2 <- revalue(sterm_curated_rev_names$genome2, c("Ldel_rmk202"="MAG RMK202"))

# ##------------
# ##for SNP legend
# ##------------

ggheatmap <- ggplot(sterm_curated_rev_names, aes(genome1, genome2, fill = SNPs))+
 geom_tile(color = "white",size=1.1)+
 #scale_fill_gradient2(low = "blue", high = "red", mid = "white",
  # midpoint = 99.5, limit = c(98.9,100), space = "Lab",
  # name="ANI") +
  scale_fill_distiller(name = "SNPs", palette = "Blues", direction = -1)+
  # scale_fill_distiller(name = "ANI", palette = "Reds", direction = 1)+
  theme_minimal()+ # minimal theme
 theme(axis.text.y = element_text(size = 12),
      axis.text.x = element_text(angle = 45, vjust = 1,
    size = 12, hjust = 1))+
 coord_fixed()
# Print the heatmap
print(ggheatmap)

##------------
##SNPs
##------------


plotFinal_sterm <- ggheatmap + 
geom_text(aes(genome1, genome2, label = SNPs), color = "black", size = 2.5) +
    labs(title=paste0("# core genes :",nrow(coreGeneNames_sterm)," (",CoreGeneSize_sterm,"bp)"))+
  scale_y_discrete(position = 'right')+
theme(
  axis.title.x = element_blank(),
  axis.title.y = element_blank(),
  # axis.text.y.left = element_text(angle = 75, hjust = 1,size=9),
  panel.grid.major = element_blank(),
  # scale_y_date(position = "right"),
  panel.border = element_blank(),
  panel.background = element_blank(),
  axis.ticks = element_blank(),
  # legend.justification = c(1, 0),
  legend.position = "right",
  legend.text = element_text(angle = 0),
  legend.direction = "vertical")+
  guides(fill = guide_colorbar(barwidth = 1, barheight = 7,
                title.position = "top", title.hjust = 0.5))
plotFinal_sterm
##save heatmap
svg("~/Desktop/Projects/2018_culturomics/fastani/Ldel_SNPs_core_absolute_final.svg",width=5.5,height=4)
plotFinal_sterm#+theme(legend.position = "none")
dev.off()


##------------
##ANI 2
##------------


library(viridis)
# creat Heatmap
sterm_curated_rev_names$ANI_2 <- round(sterm_curated_rev_names$ANI,digits = 2)
min(sterm_curated_rev_names$ANI)
ggheatmap <- ggplot(sterm_curated_rev_names, aes(genome1, genome2, fill = ANI))+
 geom_tile(color = "white",size=1.1)+
 # scale_fill_gradient2(low = "red", high = "red4",
 # midpoint = 99.995, limit = c(99.99,100), space = "Lab",
 # name="ANI") +
  # scale_fill_viridis(discrete = FALSE, option = "B",direction = -1)+
  scale_fill_distiller(name = "ANI", palette = "Reds", direction = 1)+
  theme_minimal()+ # minimal theme
 theme(axis.text.y = element_text(size = 12),
      axis.text.x = element_text(angle = 45, vjust = 1, 
    size = 12, hjust = 1))+
 coord_fixed()
# Print the heatmap
print(ggheatmap)


plotFinal_sterm <- ggheatmap + 
geom_text(aes(genome1, genome2, label = ANI_2), color = "black", size = 2.5) +
  scale_y_discrete(position = 'right')+
  labs(title=paste0("# core genes :",nrow(coreGeneNames_sterm)," (",CoreGeneSize_sterm,"bp)"))+
theme(
  axis.title.x = element_blank(),
  axis.title.y = element_blank(),
  # axis.text = element_blank(),
  panel.grid.major = element_blank(),
  panel.border = element_blank(),
  panel.background = element_blank(),
  axis.ticks = element_blank(),
  # legend.justification = c(1, 0),
  legend.position = "right",
  legend.text = element_text(angle = 0),
  legend.direction = "vertical")+
  guides(fill = guide_colorbar(barwidth = 1, barheight = 7,
                title.position = "top", title.hjust = 0.5))
plotFinal_sterm
##save heatmap
svg("~/Desktop/Projects/2018_culturomics/fastani/Ldel_SNPs_core_ANI.svg",width=5.5,height=4)
plotFinal_sterm#+theme(legend.position = "none")
dev.off()

```




##SNPeff analysis supplementary


rAnalysis

```{r}


##==========snpEff !!!!do after snpEff chunk!!!
#Ldel
snpEff_Ldel <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/effects/variantCallsfreebayes_all_Lactobacillus_delbrueckii_RMK202-withOld_merged_forR_unmodified.csv", "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("count","effect","impact"));snpEff_Ldel$species <- "L.delbrueckii"

#Sterm
snpEff_Sterm <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/SnpEff/effects/variantCallsfreebayes_all_Streptococcus_thermophilus_RMK202-withOld_merged_forR_unmodified.csv", "\t", escape_double = FALSE, trim_ws = TRUE,col_names = c("count","effect","impact"));snpEff_Sterm$species <- "S. thermophilus"

snpEff <- rbind(snpEff_Ldel,snpEff_Sterm)

snpEff$location <- ifelse(snpEff$impact=="MODIFIER","intragenic","genic")

# ##seperate
# snpEff_Sterm <- snpEff[which(snpEff$species=="Sterm"),]
# snpEff_Ldel <- snpEff[which(snpEff$species=="Ldel"),]

##==========plot abundance 



#--- genomic or intergenomic

table(snpEff$sample)

# aggregate(startingGff$RawReads, list(startingGff$name), mean)

interPlot <- ggplot(data=snpEff, aes(x=location, weight=count,fill=location)) +
  facet_grid(species~., scales="free")+
  geom_bar()+
  # geom_bar(stat="identity", position = "dodge")+
labs(y="count",
     x="genomic location")+
scale_fill_manual(values=c("black","grey"))+ 
  theme_classic()+
theme(#axis.text.y=element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "none")


interPlot
 svg("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/snpEff/genomicLocation_snps.svg",width=3,height=5)
interPlot
 dev.off()
 

source("/home/vincent/Desktop/ScriptRepository/R_functions-master/R_functions/g_legend.R")
library(sparklyr)
##==========change order

##-Sterm

snpEff_new <- snpEff


factor(snpEff_new$impact)
snpEff_new$impact <- factor(snpEff_new$impact, levels=(c("MODIFIER","LOW","MODERATE","HIGH")))
factor(snpEff_new$impact)



snpEff_new[match(levels(factor(snpEff_new$effect)),snpEff_new$effect),c(2,3)][match(c("MODIFIER","LOW","MODERATE","HIGH"),snpEff_new[match(levels(factor(snpEff_new$effect)),snpEff_new$effect),c(2,3)]$impact),]

orderEffect <- snpEff_new[match(levels(factor(snpEff_new$effect)),snpEff_new$effect),c(2,3)][( snpEff_new[match(levels(factor(snpEff_new$effect)),snpEff_new$effect),c(2,3)]$impact == "MODIFIER"),]
orderEffect <- rbind(orderEffect,snpEff_new[match(levels(factor(snpEff_new$effect)),snpEff_new$effect),c(2,3)][( snpEff_new[match(levels(factor(snpEff_new$effect)),snpEff_new$effect),c(2,3)]$impact == "LOW"),])
orderEffect <- rbind(orderEffect,snpEff_new[match(levels(factor(snpEff_new$effect)),snpEff_new$effect),c(2,3)][( snpEff_new[match(levels(factor(snpEff_new$effect)),snpEff_new$effect),c(2,3)]$impact == "MODERATE"),])
orderEffect <- rbind(orderEffect,snpEff_new[match(levels(factor(snpEff_new$effect)),snpEff_new$effect),c(2,3)][( snpEff_new[match(levels(factor(snpEff_new$effect)),snpEff_new$effect),c(2,3)]$impact == "HIGH"),])

orderEffect_true <- droplevels(factor(orderEffect$effect))

factor(snpEff_new$effect)
snpEff_new$effect <- factor(snpEff_new$effect, levels=(orderEffect_true))
factor(snpEff_new$effect)


##==========plot insertion sequence per mutation type
##---Sterm


all_impact <- ggplot(data=snpEff_new, aes(x=effect, y=count,fill=impact)) +
  facet_grid(species~., scales="free")+
  geom_bar(stat="identity", position = "dodge")+
# ggtitle(paste0("SNV sites in Sterm n= " ,(sum(snpEff_Sterm$count))," ( " , round(100*sum(snpEff_Sterm$count)/StermContigSize,digits=2), " % )")) +
xlab("effect") + 
ylab("variant sites") +
scale_fill_manual(values=c("grey","gold1","orange","red"))+   #wes_palette(n=3, name="Moonrise3")[1:3]))+
# scale_fill_manual(values=c("red","gold1","orange","grey"))+   #wes_palette(n=3, name="Moonrise3")[1:3]))+
  theme_classic()+
  theme(legend.position = "none",
        plot.margin=unit(c(0,0,0,2.5),"cm"),
        axis.text.x=element_text(angle=45, hjust=1,size=9))

all_impact

 svg("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/snpEff/inter_genomic_snp_impact.svg",width=5,height=5.5)
all_impact
 dev.off()
 

all_impact <- ggplot(data=snpEff_new, aes(x=impact,weight=count,fill=impact)) +
  facet_grid(species~., scales="free")+
  geom_bar()+
# ggtitle(paste0("SNV sites in Sterm n= " ,(sum(snpEff_Sterm$count))," ( " , round(100*sum(snpEff_Sterm$count)/StermContigSize,digits=2), " % )")) +
xlab("impact") + 
ylab("variant sites") +
scale_fill_manual(values=c("grey","gold1","orange","red"))+   #wes_palette(n=3, name="Moonrise3")[1:3]))+
# scale_fill_manual(values=c("red","gold1","orange","grey"))+   #wes_palette(n=3, name="Moonrise3")[1:3]))+
  theme_classic()+
  theme(legend.position = "none",
        plot.margin=unit(c(0,0,0,2.5),"cm"),
        axis.text.x=element_text(angle=45, hjust=1,size=9))

all_impact

 svg("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/snpEff/inter_genomic_snp_impact_02.svg",width=5,height=5.5)
all_impact
 dev.off()
 
###---------------------------high impact
 
snpEff_new_high <- snpEff_new[which(snpEff_new$impact=="HIGH"),]
 

high_impact <- ggplot(data=snpEff_new_high, aes(x=effect, y=count,fill=impact)) +
  facet_grid(species~., scales="free")+
  geom_bar(stat="identity", position = "dodge")+
# ggtitle(paste0("SNV sites in Sterm n= " ,(sum(snpEff_Sterm$count))," ( " , round(100*sum(snpEff_Sterm$count)/StermContigSize,digits=2), " % )")) +
xlab("effect") + 
ylab("variant sites") +
scale_fill_manual(values=c("red"))+   #wes_palette(n=3, name="Moonrise3")[1:3]))+
    theme_classic()+
  theme(legend.position = "none",
        plot.margin=unit(c(0,0,0,2.5),"cm"),
        axis.text.x=element_text(angle=45, hjust=1,size=10))


high_impact

 svg("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/snpEff/inter_genomic_snp_impact_high.svg",width=5,height=5.5)
high_impact
 dev.off()

```


eggnog analysis with (COG)[http://clovr.org/docs/clusters-of-orthologous-groups-cogs/]categories
```{r}
library(readr)

##==========data import data prep
COG_functional_categories <- read_delim("~/Desktop/ScriptRepository/COG_functional_categories.csv",  ",", escape_double = FALSE, col_names = c("short","long","divided"),trim_ws = TRUE)

cogFunctionsHighImapactSNVs <- read_delim("~/Desktop/Projects/2018_culturomics/MetaSNV/cogFunctionsHighImapactSNVs.csv", "\t", escape_double = FALSE, trim_ws = TRUE)

cogFunctionsHighImapactSNVs$long <- NA

for (i in 1:nrow(cogFunctionsHighImapactSNVs)) {
  cogFunctionsHighImapactSNVs[i,"long"] <- COG_functional_categories[which(COG_functional_categories$short==as.character(cogFunctionsHighImapactSNVs[i,"function"])),"long"]
  
}

library(ggplot2)


source("/home/vincent/Desktop/ScriptRepository/R_functions-master/R_functions/g_legend.R")

##==========change order

cogFunctionsHighImapactSNVs_Sterm <- cogFunctionsHighImapactSNVs[which(cogFunctionsHighImapactSNVs$species=="Sterm"),]
cogFunctionsHighImapactSNVs_Ldel <- cogFunctionsHighImapactSNVs[which(cogFunctionsHighImapactSNVs$species=="Ldel"),]

colnames(cogFunctionsHighImapactSNVs_Sterm)[2] <- "functionss"
colnames(cogFunctionsHighImapactSNVs_Ldel)[2] <- "functionss"


##-Sterm

cogFunctionsHighImapactSNVs_Sterm_new <- cogFunctionsHighImapactSNVs_Sterm

factor(cogFunctionsHighImapactSNVs_Sterm_new$'functionss')
cogFunctionsHighImapactSNVs_Sterm_new$'functionss' <- factor(cogFunctionsHighImapactSNVs_Sterm_new$'functionss', levels=(c("L","S","M","V","T","K","F","E","G","P","O","C","H","unknown")))
factor(cogFunctionsHighImapactSNVs_Sterm_new$'functionss')

cogFunctionsHighImapactSNVs_Sterm_new$'long' <- gsub(" and ","\n",cogFunctionsHighImapactSNVs_Sterm_new$'long') %>% gsub(", ","\n",.) %>% gsub("/e","\ne",.)

factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')

cogFunctionsHighImapactSNVs_Sterm_new$'long' <- factor(cogFunctionsHighImapactSNVs_Sterm_new$'long', levels=(c(as.character(factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')[8]),as.character(factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')[12]),as.character(factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')[9]),as.character(factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')[14]),as.character(factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')[13]),as.character(factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')[7]),as.character(factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')[4]),as.character(factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')[3]),as.character(factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')[5]),as.character(factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')[11]),as.character(factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')[10]),as.character(factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')[2]),as.character(factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')[6]),as.character(factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')[1]))))

factor(cogFunctionsHighImapactSNVs_Sterm_new$'long')

##-Ldel

cogFunctionsHighImapactSNVs_Ldel_new <- cogFunctionsHighImapactSNVs_Ldel

factor(cogFunctionsHighImapactSNVs_Ldel_new$'functionss')
cogFunctionsHighImapactSNVs_Ldel_new$'functionss' <- factor(cogFunctionsHighImapactSNVs_Ldel_new$'functionss', levels=(c("L","E","K","I")))
factor(cogFunctionsHighImapactSNVs_Ldel_new$'functionss')

cogFunctionsHighImapactSNVs_Ldel_new$'long' <- gsub(" and ","\n",cogFunctionsHighImapactSNVs_Ldel_new$'long')

factor(cogFunctionsHighImapactSNVs_Ldel_new$'long')

cogFunctionsHighImapactSNVs_Ldel_new$'long' <- factor(cogFunctionsHighImapactSNVs_Ldel_new$'long', levels=(c(as.character(factor(cogFunctionsHighImapactSNVs_Ldel_new$'long')[4]),as.character(factor(cogFunctionsHighImapactSNVs_Ldel_new$'long')[1]),as.character(factor(cogFunctionsHighImapactSNVs_Ldel_new$'long')[3]),as.character(factor(cogFunctionsHighImapactSNVs_Ldel_new$'long')[2]))))

factor(cogFunctionsHighImapactSNVs_Ldel_new$'long')



##==========plot insertion sequence per mutation type
##---Sterm

Sterm_COG <- ggplot(data=cogFunctionsHighImapactSNVs_Sterm_new, aes(x=long, y=count,fill=long)) +
  #facet_grid(~species)+
  geom_bar(stat="identity")+
ggtitle(paste0("COG categories of high impact SNV in Sterm n= " ,(sum(cogFunctionsHighImapactSNVs_Sterm_new$count)))) +
xlab("genomic location") + 
ylab("variant sites") +
#scale_fill_manual(values=c("red","gold1","orange","grey"))+   #wes_palette(n=3, name="Moonrise3")[1:3]))+
theme(#axis.text.y=element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank(),
        #axis.text.x=element_blank(),
        axis.title.x = element_blank(),
        #axis.text.x=element_blank(),
        legend.position = "none",
        axis.line = element_line(colour = "black"),
        panel.background = element_blank(),
        panel.grid.major.x = element_blank() ,
        axis.text.x=element_text(angle=45, hjust=1),
        plot.margin=unit(c(0,0,0,0), "cm"),
        panel.border = element_rect(colour = "black", fill=NA, size=1),
        #panel.background = element_rect(fill = "white", colour = "grey50"),
        panel.grid.major =element_line(colour = "grey90"),panel.grid.major.y = element_blank())

#---Ldel


Ldel_COG <- ggplot(data=cogFunctionsHighImapactSNVs_Ldel_new, aes(x=long, y=count,fill=long)) +
  #facet_grid(~species)+
  geom_bar(stat="identity")+
ggtitle(paste0("Ldel n= " ,(sum(cogFunctionsHighImapactSNVs_Ldel_new$count)))) +
xlab("genomic location") + 
ylab("variant sites") +
scale_x_discrete(labels= as.character(cogFunctionsHighImapactSNVs_Ldel_new$long))+
#scale_fill_manual(values=c("red","gold1","orange","grey"))+   #wes_palette(n=3, name="Moonrise3")[1:3]))+
theme(#axis.text.y=element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_blank(),
        #axis.text.x=element_blank(),
        axis.title.x = element_blank(),
        #axis.text.x=element_blank(),
        legend.position = "none",
        axis.line = element_line(colour = "black"),
        panel.background = element_blank(),
        panel.grid.major.x = element_blank() ,
        axis.text.x=element_text(angle=45, hjust=1),
        plot.margin=unit(c(0,0,0,0), "cm"),
        panel.border = element_rect(colour = "black", fill=NA, size=1),
        #panel.background = element_rect(fill = "white", colour = "grey50"),
        panel.grid.major =element_line(colour = "grey90"),panel.grid.major.y = element_blank())

Sterm_COG
Ldel_COG 

 grid.arrange(Sterm_COG, Ldel_COG, 
         ncol=2, nrow=1, widths=c(4,1.5))

svg("~/Desktop/Projects/2018_culturomics/MetaSNV/genomicLocation_short_Highimpact_COG.svg",width=10,height=4.5)

 grid.arrange(Sterm_COG, Ldel_COG, 
         ncol=2, nrow=1, widths=c(4,1.5))
 dev.off()
 



```



####PTR calculation



own pipeline
```{bash}


##=============
##bowtie2 mapping
##=============

num=1
for NanoSample in $(cat /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt )
  do
  echo ${num}"/37  :" ${names}
  num=$((num+1))
  echo "======================================"
bowtie2-build /archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${NanoSample}/08_eightFreebayes/freebayesOuput/_eightFreebayes.fasta /archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${NanoSample}/08_eightFreebayes/freebayesOuput/_eightFreebayes


done
threads=37

num=1
for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name.txt )
  do
  echo ${num}"/37  :" ${names}
  num=$((num+1))
  #rm -r /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/${names}

  mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_bowtie2/${names}/bwaMapping2DB/
  
  #gunzip /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R1_val_1.fq.gz
  #gunzip /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R2_val_2.fq.gz
  
  sampleShort=$(echo ${names}| awk -F "_" 'BEGIN{OFS="_"}{print $1,$2}')
  
   NanoSample=$(grep "$sampleShort" /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt) 
  
  #/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${NanoSample}/assembly/assembly.fasta
  

  
bowtie2 -p ${threads} -x /archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/${NanoSample}/08_eightFreebayes/freebayesOuput/_eightFreebayes  -1 /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R1_val_1.fq -2 /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R2_val_2.fq | samtools sort -@${threads} -O SAM -o /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_bowtie2/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.sam - 

#samtools index /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam

 # gzip /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R1_val_1.fq
#  gzip /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R2_val_2.fq
  
  
  done 

##================
#irep
##================



threads=37

num=1
for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name.txt | grep "th_K1_")
  do
  echo ${num}"/37  :" ${names}
  num=$((num+1))

  sampleShort=$(echo ${names}| awk -F "_" 'BEGIN{OFS="_"}{print $1,$2}')
  
   NanoSample=$(grep "$sampleShort" /archiv/Projects/2019_Nano_Meta/nanoporeSamples.txt) 
  
  mkdir -p /archiv/Projects/2019_Nano_Meta/04_polishing/02_PTR_irep/all_th_K1/
  

iRep -f /archiv/Projects/2019_Nano_Meta/04_polishing/01_demultiplexed_Flye_uncirclator/th_K1_8h/08_eightFreebayes/freebayesOuput/genomes/contig_16.fasta  -s /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_bowtie2/${names}/bwaMapping2DB/*.sam -o /archiv/Projects/2019_Nano_Meta/04_polishing/02_PTR_irep/all_th_K1/${names}_PTR.tsv 

done

```



(irep)[https://github.com/christophertbrown/iRep] of all indiviual and then put into excel manually

```{r}

library(readr)
library(tidyverse)
library(dplyr)
library(tidyr)

irep_K1_PTR <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/PTR_irep/irep_K1_PTR.csv", "\t", escape_double = FALSE, trim_ws = TRUE)
irep_K1_PTR <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/PTR_irep/irep_K1_PTR.csv", ",", escape_double = FALSE, trim_ws = TRUE)



  irep_K1_PTR$day <- str_split_fixed(irep_K1_PTR$sample, "_", 3)[,1]
  irep_K1_PTR$kessel <- str_split_fixed(irep_K1_PTR$sample, "_", 3)[,2]
  irep_K1_PTR$time <- str_split_fixed(irep_K1_PTR$sample, "_", 3)[,3]
  irep_K1_PTR$treatment <- paste0(irep_K1_PTR$day,"_",irep_K1_PTR$kessel)
 
#  di_K1_long$treatment_f = factor(di_K1_long$treatment, levels=c('di_K1','th_K1','di_K2','th_K2'))
  irep_K1_PTR$time_f = factor(irep_K1_PTR$time, levels=c("0h","2h","4h","6h","8h","10h","12h","24h"))

  colours <-  c("#10B552","#EB4D4D") #c("#FAA0A0","#FCC2C2","#EB4D4D","#D42828") #c("#FAA0A0","#FAA0A0","#FAA0A0","#FAA0A0")
# Ldel_colours <- c("#6CF5A3","#B3FCD1","#10B552","#088A3C") 
  irep_K1_PTR <- irep_K1_PTR[which(irep_K1_PTR$day=="di"),]
  
  
    irep_K1_PTR$time_f_02 <- as.character(gsub("h","",irep_K1_PTR_poster$time_f) )
  irep_K1_PTR$time_f_02 = as.numeric(irep_K1_PTR_poster$time_f_02, levels=c("0","2","4","6","8","10","12","24"))

  
 ptr_01 <-  ggplot(irep_K1_PTR,aes(x=time_f,y=PTR,size=coverage,group=genome,color=genome,fill=genome))+geom_point()+geom_line()+facet_grid(~treatment)+
    labs("",
         x="time",
         y="Peak-to-trough")+
    theme_classic()+
    #scale_color_viridis(discrete=TRUE)+
    scale_fill_manual(values=colours)+
    scale_color_manual(values=colours)

  # scale_fill_viridis(discrete=TRUE)+
    # theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
      #axis.text.x = element_blank(),
          # legend.position="right"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          # )
  ptr_01
svg("~/Desktop/presenations/my_presentations/20190708_groupmeeting/figures/PTR.svg",width=6,height=3.2)
ptr_01
 dev.off()

 
 ##absolute count model
 
 total_samples
 
unique(total_samples$group)

 
 
 Ldel_abs <- total_samples[which(total_samples$group=="Lactobacillus_delbrueckii_group_subgroup_2"),] %>% filter(., kessel=="K1") 
 sterm_abs <- total_samples[which(total_samples$group=="Streptococcus_thermophilus_group_subgroup_4"),] %>% filter(., kessel=="K1") 
 
 Ldel_abs$genome <- "L. delbrueckii"
 sterm_abs$genome <- "S. thermophilus"

   sterm_abs$time_f_02 <- as.character(gsub("h","",sterm_abs$time_f) )
   Ldel_abs$time_f_02 <- as.character(gsub("h","",Ldel_abs$time_f) )

 

   sterm_abs_new <- subset(sterm_abs, select=c("name","genome","day","kessel","time_f","time_f_02","treatment","absCount"))
  Ldel_abs_new <- subset(Ldel_abs, select=c("name","genome","day","kessel","time_f","time_f_02","treatment","absCount"))

  all_abs_new <- rbind(Ldel_abs_new,sterm_abs_new)
  write.csv(all_abs_new,"~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/PTR_irep/PTR_fromR.csv")

  ##change in R
 
 
 # irep_K1_PTR_new <- read_delim("~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/PTR_irep/PTR_fromR.csv", ",", escape_double = FALSE, trim_ws = TRUE)

 # ggplot(irep_K1_PTR_new,aes(x=absCount,y=expecCount))+geom_point()
 
 
  ##-------plot for poster
  irep_K1_PTR_poster <- irep_K1_PTR[which(irep_K1_PTR$day=="di"),]
  
    irep_K1_PTR_poster$time_f_02 <- as.character(gsub("h","",irep_K1_PTR_poster$time_f) )
  irep_K1_PTR_poster$time_f_02 = as.numeric(irep_K1_PTR_poster$time_f_02, levels=c("0","2","4","6","8","10","12","24"))

 ptr_01 <-  ggplot(irep_K1_PTR_poster,aes(x=time_f_02,y=PTR,size=coverage,group=genome,color=genome,fill=genome))+geom_point()+geom_line()+
    labs("",
         x="time [h]",
         y="Peak-to-trough")+
   scale_x_continuous(breaks =c(0,2,4,6,8,10,12,24),labels=c(0,2,4,6,8,10,12,24))+
    theme_classic()+
    #scale_color_viridis(discrete=TRUE)+
    scale_fill_manual(values=colours)+
    scale_color_manual(values=colours)

  # scale_fill_viridis(discrete=TRUE)+
    # theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
      #axis.text.x = element_blank(),
          # legend.position="right"
          #legend.justification=c(1,1), legend.position=c(1,1),
          #legend.title = element_blank()
          # )
  ptr_01
svg("~/Desktop/posters/2019_ISCB/figures/PTR_poster.svg",width=9,height=4.7)
ptr_01
 dev.off()
  
```


##PTR with segmented



```{r}
library(readr)
library(segmented)
library(plotly)
library(ggplot2)
library(gridExtra)
library(cowplot)

coverage <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/1AB/mapping2SCOG_reference/coverage_2_referenceGenome_allSamples.bed","\t", escape_double = FALSE, col_names = c("contig","annotation","gene","start","end","unknown1","orientation","unknown2","id","count","length_unmapped","geneLength","percent","name"), trim_ws = TRUE)


  coverage <- read_delim("/home/vincent/Desktop/Projects/2019_Pilotplan/09_annoation/all_Bacteria_allSamples.txt","\t", escape_double = FALSE, col_names = c("chr","type","sort","start","end","unknown2","orientation","unknown","description","count","length_mapped","geneLength","covered","sample"),trim_ws = TRUE)


coverage$coverage <- (coverage$count*600)/coverage$geneLength

#coverage$species <- ifelse(coverage$contig=="contig_1","S. thermophilus","L. delbrueckii")
coverage <- coverage[which(coverage$end-coverage$start>600),]


##-------------Ldel


psterm  <- list()
pldel  <- list()
namesss = NULL
PTRss= NULL
#coverage_sample <- list
# p <- list()
for (i in  1:length(unique(coverage$sample))){
  
  
  sample <- unique(coverage$sample)[i]
  coverage_sample  <- coverage[which(coverage$sample==sample),]
  coverage_sample_Ldel <- coverage_sample[which(coverage_sample$chr=="L_delbrueckii_RMK202"),]
  mean_ldel <- max(coverage_sample_Ldel$end)/2

  y <- as.double(coverage_sample_Ldel$coverage)
  x <- as.double(coverage_sample_Ldel$start)

  lin.mod <- lm(y~x)
summary( lin.mod)
  segmented.mod <- segmented(lin.mod, seg.Z = ~x, psi=mean_ldel)
  midpoint <- segmented.mod$psi[2]
  maxpoint <- max(coverage_sample_Ldel$end)
  summary(segmented.mod)
  slope_01 <- segmented.mod$coefficients[2]
  slope_02 <- segmented.mod$coefficients[3]+segmented.mod$coefficients[2]
  intercept_01 <- segmented.mod$coefficients[1]
  break1 <- segmented.mod$psi[[3]]
  intercept_02 <-  intercept_01 + (slope_01* midpoint) - (slope_02* midpoint) 
  
  minValue_01 <- intercept_01 + (slope_01* midpoint)
  minValue_02 <- intercept_02 + (slope_02* midpoint)
  
  maxValue_01 <- intercept_01
  maxValue_02 <- intercept_02 + (slope_02* maxpoint)
  
  maxValue_total <- max(maxValue_01,maxValue_02)
  minValue_total <- min(minValue_01,minValue_02)
  PTR <- maxValue_total/ minValue_total
  
    namesss = append(namesss, as.character(sample))
  PTRss = append(PTRss, PTR)
  
  fit <- numeric(length(x)) * NA
  fit[complete.cases(rowSums(cbind(y, x)))] <- broken.line(segmented.mod)$fit
  data1 <- data.frame(x = x, y = y, fit = fit,geneID=coverage_sample_Ldel$description,geneStart=coverage_sample_Ldel$start,geneSize=coverage_sample_Ldel$end-coverage_sample_Ldel$start,Species=coverage_sample_Ldel$chr,Name=coverage_sample_Ldel$sample)

pldel[[i]] <- ggplot(data1, aes(x=x,y=y,label=geneID,label2=geneStart,label3=Name))+
  geom_point(aes(size=geneSize,color=Species))+
  annotate(geom="text", x=500000, y=max(data1$y)-(0.2*max(data1$y)), label=paste0(sample," (PTR= ",round(PTR,digits = 2),")"),color="black")+
  geom_line(aes(x = x, y = fit),size=2, color = 'grey')+
  labs(x="genome location",y="coverage")+
  scale_color_manual(values="#10B552")+
  theme_bw()+
  theme(legend.position="none")


}
dldel = data.frame(namesss,PTRss,species="Lactobacillus delbrueckii")


# htmlwidgets::saveWidget(pldel, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/PTR_ldel.html")

# pdf("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/1AB/mapping2SCOG_reference/mapping2ONTreference_ldel.pdf",width=16,height=32,paper='special') 
svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/PTR_ldel.svg",width=20,height=4)
do.call(plot_grid, c(pldel, align = 'h', ncol=5))
dev.off()


##-------------Sterm
# table(coverage_sample$chr)
psterm  <- list()
pldel  <- list()
namesss = NULL
PTRss= NULL
# PTR_total <- data.frame(c("name","PTR"))
#coverage_sample <- list
# p <- list()
for (i in  1:length(unique(coverage$sample))){
  
  sample <- unique(coverage$sample)[i]
  coverage_sample  <- coverage[which(coverage$sample==sample),]
  coverage_sample_Ldel <- coverage_sample[which(coverage_sample$chr=="S_thermophilus_RMK202"),]
  mean_ldel <- max(coverage_sample_Ldel$end)/2

  y <- as.double(coverage_sample_Ldel$coverage)
  x <- as.double(coverage_sample_Ldel$start)

  lin.mod <- lm(y~x)
summary( lin.mod)
  segmented.mod <- segmented(lin.mod, seg.Z = ~x, psi=mean_ldel)
  midpoint <- segmented.mod$psi[2]
  maxpoint <- max(coverage_sample_Ldel$end)
  summary(segmented.mod)
  slope_01 <- segmented.mod$coefficients[2]
  slope_02 <- segmented.mod$coefficients[3]+segmented.mod$coefficients[2]
  intercept_01 <- segmented.mod$coefficients[1]
  break1 <- segmented.mod$psi[[3]]
  intercept_02 <-  intercept_01 + (slope_01* midpoint) - (slope_02* midpoint) 
  
  minValue_01 <- intercept_01 + (slope_01* midpoint)
  minValue_02 <- intercept_02 + (slope_02* midpoint)
  
  maxValue_01 <- intercept_01
  maxValue_02 <- intercept_02 + (slope_02* maxpoint)
  
  maxValue_total <- max(maxValue_01,maxValue_02)
  minValue_total <- min(minValue_01,minValue_02)
  PTR <- maxValue_total/ minValue_total
  
  # PTR[[i,]] <- c(as.character(sample),PTR)
  namesss = append(namesss, as.character(sample))
  PTRss = append(PTRss, PTR)

  fit <- numeric(length(x)) * NA
  fit[complete.cases(rowSums(cbind(y, x)))] <- broken.line(segmented.mod)$fit
  data1 <- data.frame(x = x, y = y, fit = fit,geneID=coverage_sample_Ldel$description,geneStart=coverage_sample_Ldel$start,geneSize=coverage_sample_Ldel$end-coverage_sample_Ldel$start,Species=coverage_sample_Ldel$chr,Name=coverage_sample_Ldel$sample)

pldel[[i]] <- ggplot(data1, aes(x=x,y=y,label=geneID,label2=geneStart,label3=Name))+
  geom_point(aes(size=geneSize,color=Species))+
  annotate(geom="text", x=500000, y=max(data1$y)-(0.2*max(data1$y)), label=paste0(sample," (PTR= ",round(PTR,digits = 2),")"),color="black")+
  geom_line(aes(x = x, y = fit),size=2, color = 'grey')+
  labs(x="genome location",y="coverage")+
  scale_color_manual(values="#10B552")+
  theme_bw()+
  theme(legend.position="none")


}

d = data.frame(namesss,PTRss,species="Streptococcus thermophilus")
PTR_all <- rbind(dldel,d)

svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/PTR_sterm.svg",width=20,height=4)
do.call(plot_grid, c(pldel, align = 'h', ncol=5))
dev.off()


##--------------PTR plot

PTR_all$namesss = factor(PTR_all$namesss, levels=c("Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202"))


ggplot(PTR_all,aes(x=namesss,y=PTRss,))


all_colours <- rev(c("#EB4D4D","#FAA0A0","#10B552","#36E37B","#6CF5A3","#C5F6FA","#99E9F2") ) 


 pPTR <- ggplot(PTR_all,aes(x=namesss,y=PTRss,group=species,fill=species,color=species))+ geom_point(size=3)+ geom_line()+ #stat_summary(fun.y=sum, geom="line")+
   labs("",
         x="",
         y="Peak-to-trough ratio",
          title="")+
   scale_color_manual(values=c("#10B552","#EB4D4D"))+
  theme_classic()+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
           legend.position="none"
       #egend.justification=c(1,1), legend.position=c(0.9,0.9),
       #legend.title = element_blank()
       )
  pPTR
  
svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/06_PTR_relAbundance.svg",width=3.5,height=8)
grid.arrange(PrelAbundance,PrelAbundance+theme(legend.position = "none"),pPTR,ncol=1, heights=c(3,3,3))

dev.off()



```


##Mapping colony picking strains to ONT-Ref

In the following we want to map colony picked strains to the ONT-REF.

```{bash}

#rm /archiv/Projects/2019_Nano_Meta/colonyPickingIsolates/bwaMapping2ONT_K1/log/multiqc_logfile.txt

threads=3

for id in $(cut -f 2 /archiv/Projects/2018_Culturomics/02_script/isolates_names_changed.txt)
  do
  echo ==========================
   echo ${id}
   echo ==========================
   
   mkdir -p /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${id}/

    
#gunzip -c /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/gz/${id}/${id}_R1_val_1.fq.gz  > \
#  /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${id}/${id}_R1_val_1.fq
# gunzip -c /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/gz/${id}/${id}_R2_val_2.fq.gz > \
#  /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${id}/${id}_R2_val_2.fq

mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${id}/bwaMapping2DB/

#bwa mem -t ${threads}  /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/_eightFreebayes.fasta \
#/archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/gz/${id}/${id}_R1_val_1.fq.gz /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${id}/${id}_R2_val_2.fq |\
#samtools sort -@8 -O BAM \
#-o /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${id}/bwaMapping2DB/${id}_rawIllumina_bwamem_sorted.bam - 

#qualimap bamqc -bam /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${id}/bwaMapping2DB/${id}_rawIllumina_bwamem_sorted.bam \
#        -outdir /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${id}/bwaMapping2DB/bamqc --java-mem-size=40G

#mkdir -p /archiv/Projects/2019_Nano_Meta/colonyPickingIsolates/bwaMapping2ONT_K1/log

echo "/archiv/Projects/2019_Nano_Meta/colonyPickingIsolates/bwaMapping2ONT_K1/"${id}"/bamqc/raw_data_qualimapReport/" >> /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/log/multiqc_logfile.txt
   
#samtools index /archiv/Projects/2019_Nano_Meta/colonyPickingIsolates/bwaMapping2ONT_K1/${id}/rawIllumina_${id}_bwamem_Assembly_ONTK1_sorted.bam


##===========SNV calling
mkdir -p /archiv/Projects/2019_Nano_Meta/colonyPickingIsolates/SNV_freebayes/${id}

#freebayes -F 0.5 --min-coverage 4 -p 1 -f \
#      /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/_eightFreebayes.fasta  \
#      /archiv/Projects/2019_Nano_Meta/colonyPickingIsolates/bwaMapping2ONT_K1/${id}/rawIllumina_${id}_bwamem_Assembly_ONTK1_sorted.bam > \
#      /archiv/Projects/2019_Nano_Meta/colonyPickingIsolates/SNV_freebayes/${id}//${id}_variantCallsfreebayes.vcf

##===========assembly of non-assembled


#samtools view -b -f 4 /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${id}/bwaMapping2DB/${id}_rawIllumina_bwamem_sorted.bam > /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${id}/bwaMapping2DB/${id}_rawIllumina_bwamem_sorted_unmapped.bam

#bedtools bamtofastq -i /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${id}/bwaMapping2DB/${id}_rawIllumina_bwamem_sorted_unmapped.bam -fq /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${id}/bwaMapping2DB/${id}_rawIllumina_bwamem_sorted_unmapped.fq
     
rm -r /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${id}/bwaMapping2DB/${id}_rawIllumina_bwamem_sorted_unmapped.bam
mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${id}/spadesAssembly_unmapped/assembly


/home/vincent/miniconda2/pkgs/spades-3.13.0-0/bin/spades.py --plasmid -t 35 -m 100 \
  -s /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${id}/bwaMapping2DB/${id}_rawIllumina_bwamem_sorted_unmapped.fq \
   -o /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${id}/spadesAssembly_unmapped/assembly


##----remove short contigs (<500bp)


awk '/^>/ {printf("\n%s\n",$0);next; } { printf("%s",$0);}  END {printf("\n");}' < /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${id}/spadesAssembly_unmapped/assembly/scaffolds.fasta |sed '/^[[:space:]]*$/d' | awk '/^>/ { getline seq } length(seq) >500 { print $0 "\n" seq }' > /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${id}/spadesAssembly_unmapped/assembly/scaffolds_min500bp.fasta



    done




##MULTIQC
rm -r /home/vincent/Projects/2019_pilotplant/02_mapping2referenceDB/02_againstAllCowMito/MultiQC/
mkdir -p /home/vincent/Projects/2019_pilotplant/02_mapping2referenceDB/02_againstAllCowMito/MultiQC/

multiqc --cl_config "qualimap_config: { total_reads: TRUE }" /archiv/logs/multiQC_config.txt --file-list /home/vincent/Projects/2019_pilotplant/02_mapping2referenceDB/02_againstAllCowMito/log/multiqc_logfile.txt \
  -o /home/vincent/Projects/2019_pilotplant/02_mapping2referenceDB/02_againstAllCowMito/MultiQC/










```


##Mapping old RMK samples

In the following we want to map colony picked strains to the ONT-REF.

```{bash}

##----------
##add old RMK202 samples
##----------


threads=3

for names in $(echo "Konserve_202 Versand_202 RMK202")
  do
  echo ==========================
   echo ${names}
   echo ==========================
   
#   mkdir -p /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${id}/

    
# gunzip -c /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/gz/${id}/${id}_R1_val_1.fq.gz  > \
#  /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${id}/${id}_R1_val_1.fq
# gunzip -c /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/gz/${id}/${id}_R2_val_2.fq.gz > \
#  /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${id}/${id}_R2_val_2.fq
 # rm -r /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}

 mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/
  
 # mkdir -p /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${names}/
  
 # gunzip -c /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/gz/${names}/${names}_R1*.fq.gz > /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${names}/${names}_R1.fq
#  gunzip -c /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/gz/${names}/${names}_R2*.fq.gz > /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${names}/${names}_R2.fq
  
  #/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_demultiplexed/${NanoSample}/assembly/assembly.fasta
  
#  bwa mem -t ${threads} /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/_eightFreebayes.fasta /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${names}/${names}_R1.fq /archiv/Projects/2018_Culturomics/01_rawData/trimmed_Galore/fq/${names}/${names}_R2.fq | samtools sort -@${threads} -O BAM -o /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam - 

#rm /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

##-----Qualimap

mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB//bamqc

#qualimap bamqc -bam /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam -outdir /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB//bamqc --java-mem-size=80G


mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/log
#echo "/home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/"${names}"/bwaMapping2DB//bamqc" >> /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/log/multiqc_logfile.txt
   

#samtools index /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam

##===========SNV calling
#mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/SNV_freebayes/

#freebayes -F 0.5 --min-coverage 4 -p 1 -f \
#      /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/_eightFreebayes.fasta  \
#      /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > \
#     /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/SNV_freebayes/${names}_variantCallsfreebayes.vcf

##===========assembly of non-assembled


#samtools view -b -f 4 /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam > /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

#bedtools bamtofastq -i /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam -fq /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq

rm /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.bam

mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/spadesAssembly_unmapped/assembly/

/home/vincent/miniconda2/pkgs/spades-3.13.0-0/bin/spades.py --plasmid -t 35 -m 100 \
  -s /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted_unmapped.fq \
  -o /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/spadesAssembly_unmapped/assembly


##----remove short contigs (<500bp)


awk '/^>/ {printf("\n%s\n",$0);next; } { printf("%s",$0);}  END {printf("\n");}' < /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/spadesAssembly_unmapped/assembly/scaffolds.fasta |sed '/^[[:space:]]*$/d' | awk '/^>/ { getline seq } length(seq) >500 { print $0 "\n" seq }' > /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/spadesAssembly_unmapped/assembly/scaffolds_min500bp.fasta




    done

##----blastn


mkdir -p /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/spadesAssembly_unmapped/blastn
  
  blastn -num_threads ${threads} -max_hsps 1 -max_target_seqs 1 -task megablast -show_gis \
  -query /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/spadesAssembly_unmapped/assembly/scaffolds_min500bp.fasta \
  -outfmt "6 qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore stitle slen" \
  -db /archiv/BLAST/blast_db/20190204/phage_nucleotide/phage_nucleotide_cow \
  -out /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/spadesAssembly_unmapped/blastn/blastAssemblyaginstScaffolds.txt \
  -evalue 0.01 -word_size 64
  



##MULTIQC
rm -r /home/vincent/Projects/2019_pilotplant/02_mapping2referenceDB/02_againstAllCowMito/MultiQC/
mkdir -p /home/vincent/Projects/2019_pilotplant/02_mapping2referenceDB/02_againstAllCowMito/MultiQC/

multiqc --cl_config "qualimap_config: { total_reads: TRUE }" /archiv/logs/multiQC_config.txt --file-list /home/vincent/Projects/2019_pilotplant/02_mapping2referenceDB/02_againstAllCowMito/log/multiqc_logfile.txt \
  -o /home/vincent/Projects/2019_pilotplant/02_mapping2referenceDB/02_againstAllCowMito/MultiQC/










```

####Map to Clostridium botulinum strain MAP 5
I have now seen mulitple times that I mapped to  Clostridium botulinum. 
Here, I want to see what is going on. Therefore I map to the whole genome of  Clostridium botulinum strain MAP 5.

```{bash}

 Clostridium botulinum strain MAP 5


  date=20181226
  #species=Streptococcus_thermophilus

mkdir -p /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/genome/

  curl 'ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt' >  /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/genome/all_refseq.txt

 grep "Clostridium botulinum" -i /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/genome/all_refseq.txt | grep "MAP 5" 

 grep "Clostridium botulinum" -i /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/genome/all_refseq.txt | grep "MAP 5"


grep "Clostridium botulinum" -i /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/genome/all_refseq.txt | grep "MAP 5" |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.fna.gz|' >    \
    /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/genome/download.txt
    cd /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/
    wget -i /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/genome/download.txt



/home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/mapping
    
##-------------------mapping

threads=37
names=th_K1_12h

num=1
for names in $(cut -f 2 /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name_K1.txt |grep -v "Lyo")
  do
  echo ${num}"/37  :" ${names}
  num=$((num+1))

  mkdir -p /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/mapping/${names}/
  
  bwa mem -t ${threads} /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/GCF_003017265.1_ASM301726v1_genomic.fna /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R1_val_1.fq /archiv/Projects/2019_pilotplant/01_rawData/02_mergedfastq/trimmed_Galore/gz/${names}_R2_val_2.fq | samtools sort -@${threads} -O BAM -o /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/mapping/${names}/${names}_rawIllumina_bwamem_sorted.bam - 

##-----Qualimap

mkdir -p /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/mapping/${names}/bamqc 

qualimap bamqc -bam /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/mapping/${names}/${names}_rawIllumina_bwamem_sorted.bam -outdir /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/mapping/${names}/bamqc --java-mem-size=80G


 mkdir -p /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/log/
echo "/home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/mapping/"${names}"/"${names}"_rawIllumina_bwamem_sorted.bam" >> /home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/log/multiqc_logfile.txt
       
done

mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/filesTest
names=th_K1_12h

scp -r vincent@130.223.51.151:/home/vincent/Projects/2019_pilotplant/contamination_check/clostridium_bot/mapping/${names}/bamqc/ /home/vincent/Desktop/Projects/2019_Pilotplan/filesTest



```

#Structural variation detection 
In order to detect structural variation over time and in the isolates see if the single copy orthologous genes have a weird coverage ordering. 

#####Prokka
We first annotate the CDS for all the genomes

```{bash}

##===============
###split MAG into genome
##===============

###get PGAP annotations


scp /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/LD_singu/Lactobacillus_delbrueckii_RMK202_pgap.faa vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/

scp /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/Streptococcus_thermophilus_RMK202_pgap.faa vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/faa/



###multifasta2fasta

mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta
for i in $(grep ">" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/_eightFreebayes.fasta |sed 's/>//g' |cut -d ' ' -f 1)
do
samtools faidx /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/_eightFreebayes.fasta ${i} > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/${i}.fasta
done

##===============
###annotate the genomes
##===============

mkdir -p /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/PROKKA_singleGenomes/{contig_1,contig_2}

##sterm
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/PROKKA_singleGenomes/contig_1/ --force --usegenus --genus streptococcus --locustag Sterm_K1 --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/contig_1.fasta

##Ldel 
prokka --outdir /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/PROKKA_singleGenomes/contig_2/ --force --usegenus --genus lactobacillus --locustag Ldel_K1 --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/contig_2.fasta


##===============
###gather all faa files
##===============

##Sterm

mkdir -p /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/{Sterm,Ldel}
mkdir -p /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/faa


#cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/PROKKA_singleGenomes/contig_1/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/faa/contig_1_meta.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/faa/S72.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/faa/S50.faa


scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/faa/* /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/CDS/

##Ldel

mkdir -p /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa


#cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/PROKKA_singleGenomes/contig_2/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/contig_2_meta.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L104//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L104.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L108//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L108.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L35//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L35.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L39//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L39.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L44//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L44.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L70//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L70.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L71//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L71.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L99//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L99.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L80//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L80.faa


scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/* /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/CDS/


scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/contig_5.fasta /home/vincent/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/1AB/mapping2SCOG_reference/

scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/contig_1.fasta /home/vincent/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/1AB/mapping2SCOG_reference/


```

#####Contigs

Here, I describe the assembled contigs in K1:

contig_1: Sterm
contig_2: Ldel
contig_3: Lactococcus phage
contig_4: Cow
contig_5: ldel plasmid



#####orthofinder
identify SCOG with orthofinder of all picked isolates and the metagenome assembled genome.

```{bash}


##sterm

rm -r /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder

    orthofinder -f /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/faa/ \
      -t 37 \
      -o /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/ \
      -a 37



##Ldel

rm -r /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder

    orthofinder -f /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/ \
      -t 37 \
      -o /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/ \
      -a 37


```

####extract single copy locations

```{bash}
##===============
##sterm
##===============
mkdir -p /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/extraction_of_OGs/
##-------extract OG names
#rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/Orthogroups/test.txt
rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/extraction_of_OGs/OG_names_sterm.txt
for OG in $(cat /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/Orthogroups/Orthogroups_SingleCopyOrthologues.txt)
do

grep "$OG" /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/Orthogroups/Orthogroups.txt |cut -d ' ' -f 4 >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/extraction_of_OGs/OG_names_sterm.txt


done #OG
wc -l /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/extraction_of_OGs/OG_names_sterm.txt

##-------getting gff of OGs 
#rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/Orthogroups/test.txt
rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/extraction_of_OGs/OG_genes_sterm.gff
for OG in $(cat /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/extraction_of_OGs/OG_names_sterm.txt)
do

grep "ID=${OG}_gene" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/PROKKA_singleGenomes/contig_1/*.gff >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/extraction_of_OGs/OG_genes_sterm.gff


done #OG
wc -l /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/extraction_of_OGs/OG_genes_sterm.gff
#wc -l /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/Orthogroups/test.txt

sort -k4 -n /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/extraction_of_OGs/OG_genes_sterm.gff > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/extraction_of_OGs/OG_genes_sterm_sorted.gff

##===============
##Ldel
##===============
mkdir -p /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/extraction_of_OGs/
##-------extract OG names
#rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/Orthogroups/test.txt
rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/extraction_of_OGs/OG_names_ldel.txt
for OG in $(cat /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/Orthogroups/Orthogroups_SingleCopyOrthologues.txt)
do

grep "$OG" /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/Orthogroups/Orthogroups.txt |cut -d ' ' -f 11 >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/extraction_of_OGs/OG_names_ldel.txt


done #OG
wc -l /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/extraction_of_OGs/OG_names_ldel.txt

##-------getting gff of OGs 
#rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/Orthogroups/test.txt
rm /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/extraction_of_OGs/OG_genes_ldel.gff
for OG in $(cat /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/extraction_of_OGs/OG_names_ldel.txt)
do

grep "ID=${OG}_gene" /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/PROKKA_singleGenomes/contig_2/*.gff >> /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/extraction_of_OGs/OG_genes_ldel.gff


done #OG
wc -l /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/extraction_of_OGs/OG_genes_ldel.gff
#wc -l /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/Orthogroups/test.txt

sort -k4 -n /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/extraction_of_OGs/OG_genes_ldel.gff > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/extraction_of_OGs/OG_genes_ldel_sorted.gff


```

#### Coverage at all SCOG

```{bash}

mkdir -p /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/merging_gffs

cat /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/orthofinder/Results_Jul18/extraction_of_OGs/OG_genes_sterm_sorted.gff /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/orthofinder/Results_Jul18/extraction_of_OGs/OG_genes_ldel_sorted.gff > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/merging_gffs/sterm_ldel_SCOG.gff

##===============
##extracting coverage
##===============


  num=1
  
  for names in $(cat /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name_allOfExperiment.txt)
do

echo ${num}"/37  :" ${names}
    num=$((num+1))
    
    #names=th_K1_12h
    #rm -r /home/vincent/Projects/2019_pilotplant/05_mapping2DB/${names}/BacteriaDBmapping
  
    mkdir -p /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coverage/
 

bedtools coverage -bed -a /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/merging_gffs/sterm_ldel_SCOG.gff -b /home/vincent/Projects/2019_pilotplant/04_mapping2Nanopre/02_againstpolished_single_K1/${names}/bwaMapping2DB/${names}_rawIllumina_bwamem_sorted.bam  > \
  /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coverage/${names}2Bacteria.bed 
  

done

##============
## Add sample name to file
##============

  num=1
  
  for names in $(cat /archiv/Projects/2019_pilotplant/01_rawData/01_rawfastq/change_name_allOfExperiment.txt)
do

echo ${num}"/37  :" ${names}
    num=$((num+1))
    
    #names=th_K1_12h
    #rm -r /home/vincent/Projects/2019_pilotplant/05_mapping2DB/${names}/BacteriaDBmapping
  
    mkdir -p /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coverage_final/
 
awk -F "\t" -v name="$names"  '{OFS="\t"} {print $0, name}' /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coverage/${names}2Bacteria.bed  > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coverage_final/${names}2Bacteria.bed
  

done

mkdir -p /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coverage_allToghther

cat /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coverage_final/*.bed > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coverage_allToghther/coverage_2_referenceGenome_allSamples.bed


###Bring local

scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/coverage_allToghther/coverage_2_referenceGenome_allSamples.bed /home/vincent/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/1AB/mapping2SCOG_reference/

```

analysing the SCOG mapping in R

with a piecewise regression: Piecewise regression comes about when you have ‘breakpoints’, where there are clearly two different linear relationships in the data with a sudden, sharp change in directionality

```{r}
library(readr)
library(segmented)
library(plotly)
library(ggplot2)
library(gridExtra)
library(cowplot)

coverage <- read_delim("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/1AB/mapping2SCOG_reference/coverage_2_referenceGenome_allSamples.bed","\t", escape_double = FALSE, col_names = c("contig","annotation","gene","start","end","unknown1","orientation","unknown2","id","count","length_unmapped","geneLength","percent","name"), trim_ws = TRUE)

coverage$coverage <- (coverage$count*600)/coverage$geneLength

coverage$species <- ifelse(coverage$contig=="contig_1","S. thermophilus","L. delbrueckii")
coverage <- coverage[which(coverage$end-coverage$start>600),]




coverage_th_K1_6h <- coverage[which(coverage$name=="di_K1_24h"),]


##-----Ldel
coverage_th_K1_6h_Ldel <- coverage_th_K1_6h[which(coverage_th_K1_6h$species=="L. delbrueckii"),]


mean_ldel <- max(coverage_th_K1_6h$end)/2

y <- as.double(coverage_th_K1_6h_Ldel$coverage)
x <- as.double(coverage_th_K1_6h_Ldel$start)

lin.mod <- lm(y~x)
# summary(lin.mod)

segmented.mod <- segmented(lin.mod, seg.Z = ~x, psi=1000000)
# summary(segmented.mod)

fit <- numeric(length(x)) * NA
fit[complete.cases(rowSums(cbind(y, x)))] <- broken.line(segmented.mod)$fit
data1 <- data.frame(x = x, y = y, fit = fit,geneID=coverage_th_K1_6h_Ldel$id,geneStart=coverage_th_K1_6h_Ldel$start,geneSize=coverage_th_K1_6h_Ldel$end-coverage_th_K1_6h_Ldel$start,Species=coverage_th_K1_6h_Ldel$species)

pldel <- ggplot(data1, aes(x=x,y=y,label=geneID,label2=geneStart))+
  geom_point(aes(size=geneSize,color=Species))+
  geom_line(aes(x = x, y = fit),size=2, color = 'grey')+
  labs(x=" ",y="coverage",title =unique(coverage_th_K1_6h_Ldel$name) )+
  scale_color_manual(values="#10B552")+
  theme(legend.position="none",axis.title.x=element_blank())


##-----sterm
coverage_th_K1_6h_Ldel <- coverage_th_K1_6h[which(coverage_th_K1_6h$species=="S. thermophilus"),]


mean_ldel <- max(coverage_th_K1_6h$end)/2

y <- as.double(coverage_th_K1_6h_Ldel$coverage)
x <- as.double(coverage_th_K1_6h_Ldel$start)

lin.mod <- lm(y~x)
# summary(lin.mod)

segmented.mod <- segmented(lin.mod, seg.Z = ~x, psi=1000000)
# summary(segmented.mod)

fit <- numeric(length(x)) * NA
fit[complete.cases(rowSums(cbind(y, x)))] <- broken.line(segmented.mod)$fit
data1 <- data.frame(x = x, y = y, fit = fit,geneID=coverage_th_K1_6h_Ldel$id,geneStart=coverage_th_K1_6h_Ldel$start,geneSize=coverage_th_K1_6h_Ldel$end-coverage_th_K1_6h_Ldel$start,Species=coverage_th_K1_6h_Ldel$species)

psterm <- ggplot(data1, aes(x=x,y=y,label=geneID,label2=geneStart))+
  geom_point(aes(size=geneSize,color=Species))+
  geom_line(aes(x = x, y = fit),size=2, color = 'grey')+
  labs(x="genome location",y="coverage")+
  scale_color_manual(values="#EB4D4D")+
  theme(legend.position="none")


# p

# pboth <- grid.arrange(pldel,psterm)

# pboth <- grid.arrange(pboth,pboth)

# ggp <- ggplotly(pboth)
# plot(x,y, pch=16)
# plot(segmented.mod,col="grey",lwd=3, add=T)
# 
# ggp

  # colours <-  c("#10B552","#EB4D4D")
  
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/1AB/mapping2SCOG_reference/L44_coveragePlot.html")

psterm  <- list()
pldel  <- list()
#coverage_sample <- list
# p <- list()
for (i in  1:length(unique(coverage$name))){
  
  
  sample <- unique(coverage$name)[i]
  coverage_sample  <- coverage[which(coverage$name==sample),]
  coverage_sample_Ldel <- coverage_sample[which(coverage_sample$species=="L. delbrueckii"),]
  mean_ldel <- max(coverage_sample_Ldel$end)/2

  y <- as.double(coverage_sample_Ldel$coverage)
  x <- as.double(coverage_sample_Ldel$start)

  lin.mod <- lm(y~x)
summary( lin.mod)
  segmented.mod <- segmented(lin.mod, seg.Z = ~x, psi=mean_ldel)
  midpoint <- segmented.mod$psi[2]
  maxpoint <- max(coverage_sample_Ldel$end)
  summary(segmented.mod)
  slope_01 <- segmented.mod$coefficients[2]
  slope_02 <- segmented.mod$coefficients[3]+segmented.mod$coefficients[2]
  intercept_01 <- segmented.mod$coefficients[1]
  break1 <- segmented.mod$psi[[3]]
  intercept_02 <-  intercept_01 + (slope_01* midpoint) - (slope_02* midpoint) 
  
  minValue_01 <- intercept_01 + (slope_01* midpoint)
  minValue_02 <- intercept_02 + (slope_02* midpoint)
  
  maxValue_01 <- intercept_01
  maxValue_02 <- intercept_02 + (slope_02* maxpoint)
  
  maxValue_total <- max(maxValue_01,maxValue_02)
  minValue_total <- min(minValue_01,minValue_02)
  PTR <- maxValue_total/ minValue_total
  
  
  
  fit <- numeric(length(x)) * NA
  fit[complete.cases(rowSums(cbind(y, x)))] <- broken.line(segmented.mod)$fit
  data1 <- data.frame(x = x, y = y, fit = fit,geneID=coverage_sample_Ldel$id,geneStart=coverage_sample_Ldel$start,geneSize=coverage_sample_Ldel$end-coverage_sample_Ldel$start,Species=coverage_sample_Ldel$species,Name=coverage_sample_Ldel$name)

pldel[[i]] <- ggplot(data1, aes(x=x,y=y,label=geneID,label2=geneStart,label3=Name))+
  geom_point(aes(size=geneSize,color=Species))+
  annotate(geom="text", x=500000, y=max(data1$y)-(0.2*max(data1$y)), label=paste0(sample," (PTR= ",round(PTR,digits = 2),")"),color="black")+
  geom_line(aes(x = x, y = fit),size=2, color = 'grey')+
  labs(x="genome location",y="coverage")+
  scale_color_manual(values="#10B552")+
  theme_bw()+
  theme(legend.position="none")






# ggplot(data1, aes(x=x,y=y,label=geneID,label2=geneStart))+
#   geom_point(aes(size=geneSize,color=Species))+ geom_abline(intercept =  intercept_01, slope =  slope_01, 
#                      aes(colour = "first part"), show_guide = TRUE)+ geom_abline(intercept =  intercept_02, slope =  slope_02, 
#                      aes(colour = "first part"), show_guide = TRUE)


}



 
#subplot(pldel, nrows = length(pldel)/4)
#subplot(pldel, nrows = length(pldel)/4, shareX = TRUE, titleX = FALSE)

htmlwidgets::saveWidget(subplot(pldel, nrows = length(pldel)/4), "~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/1AB/mapping2SCOG_reference/L44_coveragePlot.html")

pdf("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/1AB/mapping2SCOG_reference/mapping2ONTreference_ldel.pdf",width=16,height=32,paper='special') 

# do.call(grid.arrange, c(p, ncol=4))

# do.call(plot_grid, c(pldel, labels = "TPKM", align = 'h', ncol=4))
#do.call(plot_grid, c(psterm, labels = "TPKM", align = 'h', ncol=4))
do.call(grid.arrange, c(psterm, ncol=4))


dev.off()

do.call(grid.arrange, c(pldel, ncol=4))

do.call(grid.arrange, c(pldel, ncol=4))



psterm  <- list()
pldel  <- list()
#coverage_sample <- list
# p <- list()
for (i in  1:length(unique(coverage$name))){
  
  
  sample <- unique(coverage$name)[i]
  coverage_sample  <- coverage[which(coverage$name==sample),]
  coverage_sample_Ldel <- coverage_sample[which(coverage_sample$species=="S. thermophilus"),]
  mean_ldel <- max(coverage_sample_Ldel$end)/2

  y <- as.double(coverage_sample_Ldel$coverage)
  x <- as.double(coverage_sample_Ldel$start)

  lin.mod <- lm(y~x)
summary( lin.mod)
  segmented.mod <- segmented(lin.mod, seg.Z = ~x, psi=mean_ldel)
  midpoint <- segmented.mod$psi[2]
  maxpoint <- max(coverage_sample_Ldel$end)
  summary(segmented.mod)
  slope_01 <- segmented.mod$coefficients[2]
  slope_02 <- segmented.mod$coefficients[3]+segmented.mod$coefficients[2]
  intercept_01 <- segmented.mod$coefficients[1]
  break1 <- segmented.mod$psi[[3]]
  intercept_02 <-  intercept_01 + (slope_01* midpoint) - (slope_02* midpoint) 
  
  minValue_01 <- intercept_01 + (slope_01* midpoint)
  minValue_02 <- intercept_02 + (slope_02* midpoint)
  
  maxValue_01 <- intercept_01
  maxValue_02 <- intercept_02 + (slope_02* maxpoint)
  
  maxValue_total <- max(maxValue_01,maxValue_02)
  minValue_total <- min(minValue_01,minValue_02)
  PTR <- maxValue_total/ minValue_total
  
  
  
  fit <- numeric(length(x)) * NA
  fit[complete.cases(rowSums(cbind(y, x)))] <- broken.line(segmented.mod)$fit
  data1 <- data.frame(x = x, y = y, fit = fit,geneID=coverage_sample_Ldel$id,geneStart=coverage_sample_Ldel$start,geneSize=coverage_sample_Ldel$end-coverage_sample_Ldel$start,Species=coverage_sample_Ldel$species,Name=coverage_sample_Ldel$name)



psterm[[i]] <- ggplot(data1, aes(x=x,y=y,label=geneID,label2=geneStart,label3=Name))+
  geom_point(aes(size=geneSize,color=Species))+
  annotate(geom="text", x=500000, y=max(data1$y)-(0.2*max(data1$y)), label=paste0(sample," (PTR= ",round(PTR,digits = 2),")"),color="black")+
  geom_line(aes(x = x, y = fit),size=2, color = 'grey')+
  labs(x="genome location",y="coverage")+
  scale_color_manual(values="#EB4D4D")+
  theme_bw()+
  theme(legend.position="none")




# ggplot(data1, aes(x=x,y=y,label=geneID,label2=geneStart))+
#   geom_point(aes(size=geneSize,color=Species))+ geom_abline(intercept =  intercept_01, slope =  slope_01, 
#                      aes(colour = "first part"), show_guide = TRUE)+ geom_abline(intercept =  intercept_02, slope =  slope_02, 
#                      aes(colour = "first part"), show_guide = TRUE)


}

pdf("~/Desktop/Projects/2019_Nano_Meta/02_flye_Assembly/01_raw_undeplexed/1AB/mapping2SCOG_reference/mapping2ONTreference_sterm.pdf",width=16,height=32,paper='special') 

# do.call(grid.arrange, c(p, ncol=4))

# do.call(plot_grid, c(pldel, labels = "TPKM", align = 'h', ncol=4))
do.call(plot_grid, c(psterm, labels = "TPKM", align = 'h', ncol=4))
#do.call(grid.arrange, c(psterm, ncol=4))


dev.off()

```

plottly for multiple plots [subplots](https://plotly-r.com/arranging-views.html)


## Pandora for Strains



#####Roary

First, run roary to identify all orthologous genes. 
A good [tutorial](https://github.com/microgenomics/tutorials/blob/master/pangenome.md)


```{bash}

##-----sterm
sudo rm -r /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/
mkdir -p /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/strains

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Streptococcus_thermophilus_RMK202/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/strains/Streptococcus_thermophilus_RMK202.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/strains/S72.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/strains/S50.gff

sudo roary -f /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/ -p 5 -e -n -v /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/strains/*.gff

cd /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/_1564994606/

sudo mkdir -p /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/_1564994606//ffn/

sudo cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Streptococcus_thermophilus_RMK202/*.ffn > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/_1564994606/ffn/Streptococcus_thermophilus_RMK202.ffn
sudo cat /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/*.ffn > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/_1564994606/ffn/S72.ffn
sudo cat /archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/*.ffn > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/_1564994606/ffn/S50.ffn

python3 /home/vincent/apps/DBGenerator.py ffn

mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Sterm
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/_1564994606//pangenoma_locus.csv /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/

##=====================
##-----Ldel
##=====================
sudo rm -r /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/
mkdir -p /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/


cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_delbrueckii_RMK202/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/Lactobacillus_delbrueckii_RMK202.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L104//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L104.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L108//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L108.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L35//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L35.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L39//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L39.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L44//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L44.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L70//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L70.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L71//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L71.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L99//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L99.gff
cat /archiv/Projects/2018_Culturomics/06_Spades/L80//Prokka/20190203/*.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L80.gff


sudo roary -f /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/ -e -n -v /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/*.gff



cd /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/_1564994796/

sudo mkdir -p /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/_1564994796/ffn/


cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_delbrueckii_RMK202/*.ffn > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/Lactobacillus_delbrueckii_RMK202.ffn
cat /archiv/Projects/2018_Culturomics/06_Spades/L104//Prokka/20190203/*.ffn > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L104.ffn
cat /archiv/Projects/2018_Culturomics/06_Spades/L108//Prokka/20190203/*.ffn > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L108.ffn
cat /archiv/Projects/2018_Culturomics/06_Spades/L35//Prokka/20190203/*.ffn > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L35.ffn
cat /archiv/Projects/2018_Culturomics/06_Spades/L39//Prokka/20190203/*.ffn > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L39.ffn
cat /archiv/Projects/2018_Culturomics/06_Spades/L44//Prokka/20190203/*.ffn > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L44.ffn
cat /archiv/Projects/2018_Culturomics/06_Spades/L70//Prokka/20190203/*.ffn > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L70.ffn
cat /archiv/Projects/2018_Culturomics/06_Spades/L71//Prokka/20190203/*.ffn > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L71.ffn
cat /archiv/Projects/2018_Culturomics/06_Spades/L99//Prokka/20190203/*.ffn > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L99.ffn
cat /archiv/Projects/2018_Culturomics/06_Spades/L80//Prokka/20190203/*.ffn > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/L80.ffn


python3 /home/vincent/apps/DBGenerator.py ffn



mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Ldel
scp vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/_1564994796/pangenoma_locus.csv /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Ldel


```


create gene family files from roary output.
The following files are promising:
gene_presence_absence.csv : contains total genes in rows and column $15,$16,$17 different genomes


```{bash}

##======
##sterm
##======

##----------------
##merge all ffn files
##----------------
mkdir -p /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/ffn

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/PROKKA_singleGenomes/contig_1/*.ffn \
  /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/*.ffn \
  /archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/*.ffn > \
  /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/ffn/merged_sterm.ffn

##----------------
##prepare orthologous gene file
##----------------

mkdir -p /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned

sed '1d' /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_1564466937/gene_presence_absence.csv >  /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/gene_presence_absence_woHeade.csv
rowwss=$(wc -l /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/gene_presence_absence_woHeade.csv | cut -d  ' ' -f 1)

row=2
genomes=15

rm -r /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/geneOrthologous/
mkdir -p /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/geneOrthologous/

for row in $(seq 1 $rowwss)
do

    rowsss=$(sed -n "${row}p" /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/gene_presence_absence_woHeade.csv) 
    geneName=$(echo ${rowsss} |cut -d ',' -f 1 |sed 's/"//g')

    rm /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/geneOrthologous/${geneName}.ffn

for genomes in $(echo "15 16 17")
do

  gene=$(echo ${rowsss} |cut -d ',' -f ${genomes} |sed 's/"//g' |sed 's/[[:space:]]*$//')
  gene=$(echo ${rowsss} |awk -F '","' -v name="$genomes"  '{print $name}' |sed 's/"//g' |sed 's/[[:space:]]*$//')

  
  
  samtools faidx /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/ffn/merged_sterm.ffn ${gene} >> /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/geneOrthologous/${geneName}.ffn


done #genomes


done ##row


```



#####Unique gene analysis
analysis of roary output
```{r}
library(readr)
library(ggplot2)


##=========
##Sterm
##=========

  uniquees <-read.delim("/home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/pangenoma_locus.csv",sep="|",header = FALSE)
plotData <- table(table(uniquees$V1))
data <- data.frame(genomes=c(1,2,3),genes=as.vector(plotData))

plotsterm <- ggplot(data,aes(x=genomes,y=genes))+geom_bar(stat="identity")+
  theme_classic()

svg("/home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/pangenoma_locus_plot.svg",width=3,height=2)
plotsterm 
 dev.off()
 
 
uniquegenes <- names(table(uniquees$V1)[which((table(uniquees$V1)==1))])
 
     # uniques_who <- uniquees[which(uniquegenes,uniquees$V1),]
      uniques_who <- uniquees[which(uniquees$V1  %in% uniquegenes),]
     uniques_who$sample <- str_split_fixed(uniques_who$V2, "_", 3)[,1]
     uniques_who[uniques_who$sample=="st","sample"] <- "Streptococcus_thermophilus_RMK202"
     
plotData <- table(uniques_who$sample)
plotData
sum(plotData)

write.table(uniques_who,"/home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/pangenoma_locus_plot.csv",sep = "\t",quote = FALSE,col.names = FALSE)

data <- data.frame(genomes=c("s50","S72","MAG"),genes=as.vector(plotData))


plotsterm <- ggplot(data,aes(x=genomes,y=genes))+geom_bar(stat="identity")+
  theme_classic()+
  labs(x="",
       y="unique genes")

plotsterm 
svg("/home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/pangenoma_locus_plot_uniques.svg",width=3,height=2)
plotsterm 
 dev.off()


 
##=========
##Ldel
##=========

  uniquees <-read.delim("/home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Ldel//pangenoma_locus.csv",sep="|",header = FALSE)
plotData <- table(table(uniquees$V1))
data <- data.frame(genomes=1:10,genes=as.vector(plotData))

plotsterm <- ggplot(data,aes(x=genomes,y=genes))+geom_bar(stat="identity")+
  theme_classic()+
  scale_x_continuous(breaks = 1:10,labels=1:10)
plotsterm 
svg("/home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Ldel/pangenoma_locus_plot.svg",width=3,height=2)
plotsterm 
 dev.off()
 
 
uniquegenes <- names(table(uniquees$V1)[which((table(uniquees$V1)==1))])
 
     # uniques_who <- uniquees[which(uniquegenes,uniquees$V1),]
      uniques_who <- uniquees[which(uniquees$V1  %in% uniquegenes),]
     uniques_who$sample <- str_split_fixed(uniques_who$V2, "_", 3)[,1]
     uniques_who[uniques_who$sample=="ld","sample"] <- "Lactobacillus_delbrueckii_RMK202"
     
plotData <- table(uniques_who$sample)
plotData
sum(plotData)
write.table(uniques_who,"/home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Ldel/pangenoma_locus_plot.csv",sep = "\t",quote = FALSE,col.names = FALSE)

data <- data.frame(genomes=c("L104","L108","L35","L39","L44","L70","L71","L80","L99","MAG"),genes=as.vector(plotData))


plotsterm <- ggplot(data,aes(x=genomes,y=genes))+geom_bar(stat="identity")+
  theme_classic()+
  labs(x="",
       y="unique genes")+
theme(axis.text.x=element_text(angle=45, hjust=1))

plotsterm 
svg("/home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Ldel/pangenoma_locus_plot_uniques.svg",width=3,height=2)
plotsterm 
 dev.off()
 
```

make a unique genes file for eggnog
```{bash}

##=========
##transfer to bigmachine
##=========
##-------------
##Sterm
##-------------

scp /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Sterm/pangenoma_locus_plot.csv vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/
scp -r /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/ vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/


##-------------
##Ldel
##-------------

scp /home/vincent/Desktop/Projects/2019_Pilotplan/10_roary/Ldel/pangenoma_locus_plot.csv vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/

##=========
##getting annotations
##=========
##-------------
##Sterm
##-------------
row=2
genome=S50
gene=group_156

rowwss=$(wc -l /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/pangenoma_locus_plot.csv | cut -d  ' ' -f 1)
rm -r /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/uniqueGenes/
mkdir -p /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/uniqueGenes/

for row in $(seq 1 $rowwss)
do

   genome=$(sed -n "${row}p" /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/pangenoma_locus_plot.csv |awk '{print $4}') 
   gene=$(sed -n "${row}p" /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/pangenoma_locus_plot.csv |awk '{print $2}') 
   
   
   geneName=$(grep "${gene}:"  /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/_1564994606/clustered_proteins | sed "s/${gene}: //g" |sed -e's/[[:space:]]*$//')

   grep "ID=${geneName};" /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/strains/${genome}.gff | awk -F "\t" -v genomeee="$genome"  'BEGIN{OFS="\t"}{print $0 , genomeee}' >> /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/uniqueGenes/${genome}_uniques_Prokka.gff
  
 grep "${geneName}" /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/eggnog/${genome}_query_seqs.fa.emapper.annotations | awk -F "\t" -v genomeee="$genome"  'BEGIN{OFS="\t"}{print $0 , genomeee}' >> /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/uniqueGenes/${genome}_uniques_eggnog.gff
  
done

cat /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/uniqueGenes/*eggnog.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/uniqueGenes/sterm_merged.eggnog.txt
cat /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/uniqueGenes/*Prokka.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/uniqueGenes/sterm_merged.Prokka.txt
##-------------
##Ldel
##-------------

row=2
genome=S50
gene=group_156

rowwss=$(wc -l /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/pangenoma_locus_plot.csv | cut -d  ' ' -f 1)
rm -r /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/uniqueGenes/
mkdir -p /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/uniqueGenes/

for row in $(seq 1 $rowwss)
do

   genome=$(sed -n "${row}p" /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/pangenoma_locus_plot.csv |awk '{print $4}') 
   gene=$(sed -n "${row}p" /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/pangenoma_locus_plot.csv |awk '{print $2}') 
   
   
   geneName=$(grep "${gene}:"  /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/_1564994796/clustered_proteins | sed "s/${gene}: //g" |sed -e's/[[:space:]]*$//')

   grep "ID=${geneName};" /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/strains/${genome}.gff | awk -F "\t" -v genomeee="$genome"  'BEGIN{OFS="\t"}{print $0 , genomeee}' >> /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/uniqueGenes/${genome}_uniques_Prokka.gff
  
 grep "${geneName}" /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/eggnog/${genome}_query_seqs.fa.emapper.annotations | awk -F "\t" -v genomeee="$genome"  'BEGIN{OFS="\t"}{print $0 , genomeee}' >> /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/uniqueGenes/${genome}_uniques_eggnog.gff
  
done

cat /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/uniqueGenes/*eggnog.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/uniqueGenes/Ldel_merged.eggnog.txt
cat /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/uniqueGenes/*Prokka.gff > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/uniqueGenes/Ldel_merged.Prokka.txt


##=========
##getting local
##=========
mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/unique_sterm
scp -r  vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm/uniqueGenes/ /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/unique_sterm

mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/unique_ldel
scp -r  vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/Ldel/uniqueGenes/ /home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/unique_ldel

```

[eggnog documentation](https://github.com/eggnogdb/eggnog-mapper/wiki/eggNOG-mapper-v2)

```{r}

library(thacklr)
library(readr)
library(dplyr)
library(ggplot2)

COG_functional_categories <- read_delim("~/Desktop/ScriptRepository/COG_functional_categories.csv",  ",", escape_double = FALSE, col_names = c("short","long","divided"),trim_ws = TRUE, na = "NA")

sterm_all_unique <- read_delim("/home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/unique_sterm/uniqueGenes/sterm_merged.eggnog.txt", "\t", escape_double = FALSE, trim_ws = TRUE, na = "NA",col_names = c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description","genome")) 


#unique(sterm_all_unique$COG_Functional_Category)[match(COG_functional_categories$short , unique(sterm_all_unique$COG_Functional_Category))]
unique(sterm_all_unique$COG_Functional_Category)[which(!unique(sterm_all_unique$COG_Functional_Category) %in% COG_functional_categories$short )]

# table(sterm_all_unique$COG_Functional_Category)

sterm_all_unique$longCOG <- NA
for (i in 1:nrow(sterm_all_unique)) {
  sterm_all_unique[i,"longCOG"] <- COG_functional_categories[which(COG_functional_categories$short==as.character(sterm_all_unique[i,"COG_Functional_Category"])),"long"]
  
}



CogPlotSterm <- ggplot(sterm_all_unique,aes(x=longCOG,fill=genome))+geom_bar(position = "dodge2")+
  theme_classic()+
  labs(x="")+
  theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9),
          # # axis.text.y.left = element_text(color = "blue"),
          # axis.text.y.right = element_text(color="red"),
          # axis.title.y.left = element_text(color="blue"),
          # axis.title.y.right = element_text(color="red"),
          # rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          #legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank(),
          plot.margin=unit(c(0,0,0,2), "cm"),
          axis.text.x.bottom = element_text(size=10),
          legend.position = c(0.02, 0.98),
          legend.justification = c("left", "top")
          )
CogPlotSterm
   svg("~/Desktop/presenations/my_presentations/20190807_committeeMeeting/figures/cogPlot.svg",width=7,height=7)
CogPlotSterm 
 dev.off()

 sterm_all_unique$species ="S. thermophilus"
 sterm_uniques <- sterm_all_unique
##-------------
##Ldel
##-------------

 
sterm_all_unique <- read_delim("/home/vincent/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/unique_ldel/uniqueGenes/Ldel_merged.eggnog.txt", "\t", escape_double = FALSE, trim_ws = TRUE, na = "NA",col_names = c("query_name","seed_eggNOG_ortholog","seed_ortholog_evalue","seed_ortholog_score","best_tax_level","Preferred_name","GOs","EC","KEGG_ko","KEGG_Pathway","KEGG_Module","KEGG_Reaction","KEGG_rclass","BRITE","KEGG_TC","CAZy","BiGG_Reaction","tax_scope","eggNOG OGs","bestOG","COG_Functional_Category","eggNOG free text description","genome")) 


#unique(sterm_all_unique$COG_Functional_Category)[match(COG_functional_categories$short , unique(sterm_all_unique$COG_Functional_Category))]
unique(sterm_all_unique$COG_Functional_Category)[which(!unique(sterm_all_unique$COG_Functional_Category) %in% COG_functional_categories$short )]

# table(sterm_all_unique$COG_Functional_Category)

sterm_all_unique$longCOG <- NA
for (i in 1:nrow(sterm_all_unique)) {
  sterm_all_unique[i,"longCOG"] <- COG_functional_categories[which(COG_functional_categories$short==as.character(sterm_all_unique[i,"COG_Functional_Category"])),"long"]
  
}



CogPlotSterm <- ggplot(sterm_all_unique,aes(x=longCOG,fill=genome))+geom_bar(position = "dodge2")+
  theme_classic()+
  labs(x="")+
  theme(axis.text.x = element_text(angle = 60, hjust = 1,size=9),
          # # axis.text.y.left = element_text(color = "blue"),
          # axis.text.y.right = element_text(color="red"),
          # axis.title.y.left = element_text(color="blue"),
          # axis.title.y.right = element_text(color="red"),
          # rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          #legend.position="none"
          #legend.justification=c(1,1), legend.position=c(1,1),
          legend.title = element_blank(),
          plot.margin=unit(c(0,0,0,2), "cm"),
          axis.text.x.bottom = element_text(size=10),
          legend.position = c(0.02, 0.99),
          legend.justification = c("left", "top")
          )
CogPlotSterm
   svg("~/Desktop/presenations/my_presentations/20190807_committeeMeeting/figures/cogPlot_ldel.svg",width=7,height=5)
CogPlotSterm 
 dev.off()

 
 sterm_all_unique$species ="L. delbrueckii"
 Ldel_uniques <- sterm_all_unique
 
 
##-------------
##Merge and plot
##-------------
```




#####Multiseqalignment for all strains

first I need to creat a Multi-Sequence-Alignment of all isolated strains. 
I do this with (Mugsy)[http://mugsy.sourceforge.net/]
or mafft

```{bash}

geneName=yfkN_3

mkdir -p /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/MSA/

for gene in $(ls /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/geneOrthologous/)
do

geneName=$(echo ${gene}| cut -d '.' -f 1)

rm /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/MSA/${geneName}.msa


mafft --auto /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/geneOrthologous/${geneName}.ffn > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/MSA/${geneName}.msa


done

##----------
##remove empty files
##----------

find /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/MSA/ -size 0 -delete

##----------
##make index file
##----------

mkdir -p /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/log

awk 'BEGIN{OFS="\t"}{print "sample_id","infile"}' > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/log/index.file

for gene in $(ls /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/MSA/)
do

geneName=$(echo ${gene}| cut -d '.' -f 1)


echo "/archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/MSA/" | awk -v name="$geneName"  'BEGIN{OFS="\t"}{print name,$2name".msa"}' > /archiv/Projects/2019_Nano_Meta/07_StrainAnalysis/01_roaryStrains/sterm_cleaned/log/index.file


done

```


```{bash}

#wget https://sourceforge.net/projects/mugsy/files/latest/download/mugsy_x86-64-v1r2.2.tgz

###------------Sterm

mkdir -p /archiv/Projects/2019_Nano_Meta/06_MSA_Mugsy/Sterm

#export MUGSY_INSTALL=/home/vincent/apps/mugsy/mugsy_x86-64-v1r2.2/

cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/contig_1.fasta > /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/MAG_contig_1.fasta
cat /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/*.fna > /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/S72.fna
cat /archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/*.fna > /archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/S50.fna


/home/vincent/apps/mugsy/mugsy_x86-64-v1r2.2/mugsy --directory /archiv/Projects/2019_Nano_Meta/06_MSA_Mugsy/Sterm --prefix mugsy_sterm_msa /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/08_eightFreebayes/freebayesOuput/splitFasta/MAG_contig_1.fasta \
/archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/S72.fna \
/archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/S50.fna

mkdir -p /archiv/Projects/2018_Culturomics/genomes/
cp /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/S72.fna /archiv/Projects/2018_Culturomics/genomes/
cp /archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/S50.fna /archiv/Projects/2018_Culturomics/genomes/

###------------Ldel


cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/PROKKA_singleGenomes/contig_1/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/faa/contig_1_meta.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/S72/Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/faa/S72.faa \
cat /archiv/Projects/2018_Culturomics/06_Spades/S50/Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Sterm/faa/S50.faa


##Ldel

mkdir -p /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa


cat /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL/PROKKA_singleGenomes/contig_2/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/contig_2_meta.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L104//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L104.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L108//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L108.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L35//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L35.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L39//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L39.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L44//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L44.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L70//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L70.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L71//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L71.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L99//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L99.faa
cat /archiv/Projects/2018_Culturomics/06_Spades/L80//Prokka/20190203/*.faa > /archiv/Projects/2019_Nano_Meta/05_Orthofinder_withIsolates/Ldel/faa/L80.faa



cp /archiv/Projects/2018_Culturomics/06_Spades/L104//Prokka/20190203/*.fna /archiv/Projects/2018_Culturomics/genomes/L104.fna
cp /archiv/Projects/2018_Culturomics/06_Spades/L108//Prokka/20190203/*.fna /archiv/Projects/2018_Culturomics/genomes/L108.fna
cp /archiv/Projects/2018_Culturomics/06_Spades/L35//Prokka/20190203/*.fna /archiv/Projects/2018_Culturomics/genomes/L35.fna
cp /archiv/Projects/2018_Culturomics/06_Spades/L39//Prokka/20190203/*.fna /archiv/Projects/2018_Culturomics/genomes/L39.fna
cp /archiv/Projects/2018_Culturomics/06_Spades/L44//Prokka/20190203/*.fna /archiv/Projects/2018_Culturomics/genomes/L44.fna
cp /archiv/Projects/2018_Culturomics/06_Spades/L70//Prokka/20190203/*.fna /archiv/Projects/2018_Culturomics/genomes/L70.fna
cp /archiv/Projects/2018_Culturomics/06_Spades/L71//Prokka/20190203/*.fna /archiv/Projects/2018_Culturomics/genomes/L71.fna
cp /archiv/Projects/2018_Culturomics/06_Spades/L99//Prokka/20190203/*.fna /archiv/Projects/2018_Culturomics/genomes/L99.fna
cp /archiv/Projects/2018_Culturomics/06_Spades/L80//Prokka/20190203/*.fna /archiv/Projects/2018_Culturomics/genomes/L80.fna


cp /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_delbrueckii_RMK202/*.fna /archiv/Projects/2018_Culturomics/genomes/Lactobacillus_delbrueckii_RMK202.fna

cp /archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Streptococcus_thermophilus_RMK202//*.fna /archiv/Projects/2018_Culturomics/genomes/Streptococcus_thermophilus_RMK202.fna


mkdir -p /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/{afterAssembly,startaligned}
scp -r vincent@130.223.51.151:/archiv/Projects/2018_Culturomics/genomes/ /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/afterAssembly
```



#####make_prg

Before running pandora we need to create a pangenome graph with (make prg)[https://github.com/rmcolq/make_prg]

```{bash}
nextflow run /home/vincent/apps/make_prg/make_prg_nexflow.nf /archiv/Projects/2019_Nano_Meta/06_MSA_Mugsy/Sterm/mugsy_sterm_msa.maf

scp vincent@130.223.51.151:/home/vincent/apps/make_prg/pytest.error /

```


##Circos plot

```{bash}
home=/home/vincent/bin/apps/circos-0.69-6/

##karyotyp
${home}/data/karyotype/Ldel_rmk202.txt

##example
cp etc/repeat_nwc1_sterm_01.conf etc/ldel_rmk202_circos.conf

#example run
bin/circos -conf etc/ldel_rmk202_circos.conf ; firefox ./circos.svg &


##links
scp -r vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/links_ldel_curated.txt /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/links_ldel.txt

data/nwc_rmk/links_ldel.txt

##CDS
scp -r vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Lactobacillus_delbrueckii_RMK202/PROKKA_08052019.gff /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/

grep "^#" -v /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/PROKKA_08052019.gff |awk ' { if ($3=="CDS") print $0 } ' |awk 'BEGIN{OFS="\t"} { if ($7=="+") print $1,$4,$5,"color=green";else print $1,$4,$5,"color=black" } ' > /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/cds_ldel.txt

##rRNA

  grep "^#" -v /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/PROKKA_08052019.gff |awk ' { if ($3=="rRNA") print $0 } ' |awk 'BEGIN{OFS="\t"} { if ($7=="+") print $1,$4,$5,"color=green";else print $1,$4,$5,"color=black" } ' > /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/rRNA_ldel.txt

##eggnog

awk -F "\t" ' { if ($21=="L") print $1 } ' ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Lactobacillus_delbrueckii_RMK202_query_seqs.fa.emapper.annotations > /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/Lgenes_names_ldel.txt

rm -r /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/Lgenes_names_ldel_final.txt
for genes in $(cat /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/Lgenes_names_ldel.txt)
do
grep "ID=${genes};" /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/PROKKA_08052019.gff | awk 'BEGIN{OFS="\t"}{print $1,$4,$5}' >> /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/Lgenes_names_ldel_final.txt

done

###transposase from PGAP

grep "transposase" -i ~/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/LD_singu/annot.gff | awk -F "\t" 'BEGIN{OFS="\t"}{print $1,$4,$5}' > /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/transposase_ldel.txt


##======================
##Sterm
##======================

home=/home/vincent/bin/apps/circos-0.69-6/

##karyotyp
${home}/data/karyotype/sterm_rmk202.txt

##example
cp etc/ldel_rmk202_circos.conf etc/sterm_rmk202_circos.conf

#example run
bin/circos -conf etc/sterm_rmk202_circos.conf ; firefox ./circos.svg &


##links
scp -r vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/nucmer/deltaFiltered/links_sterm_curated.txt /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/links_sterm.txt

data/nwc_rmk/links_ldel.txt

##CDS
scp -r vincent@130.223.51.151:/archiv/Projects/2019_Nano_Meta/02_flye_Assembly/02_raw_undeplexed_plasmids/1AB/Polishing_FINAL_stiched/single/01_PROKKA/Streptococcus_thermophilus_RMK202/PROKKA_08052019.gff /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/

grep "^#" -v /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/PROKKA_08052019.gff |awk ' { if ($3=="CDS") print $0 } ' |awk 'BEGIN{OFS="\t"} { if ($7=="+") print $1,$4,$5,"color=green";else print $1,$4,$5,"color=black" } ' > /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/cds_sterm.txt

##rRNA

  grep "^#" -v /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/PROKKA_08052019.gff |awk ' { if ($3=="rRNA") print $0 } ' |awk 'BEGIN{OFS="\t"} { if ($7=="+") print $1,$4,$5,"color=green";else print $1,$4,$5,"color=black" } ' > /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/rRNA_sterm.txt

##eggnog

awk -F "\t" ' { if ($21=="L") print $1 } ' ~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog/Streptococcus_thermophilus_RMK202_query_seqs.fa.emapper.annotations > /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/Lgenes_names_sterm.txt

rm -r /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/Lgenes_names_sterm_final.txt
for genes in $(cat /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/Lgenes_names_sterm.txt)
do
grep "ID=${genes};" /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/PROKKA_08052019.gff | awk 'BEGIN{OFS="\t"}{print $1,$4,$5}' >> /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/Lgenes_names_sterm_final.txt

done

###transposase from PGAP

grep "transposase" -i ~/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/annot.gff | awk -F "\t" 'BEGIN{OFS="\t"}{print $1,$4,$5}' > /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/transposase_sterm.txt


```

##PGAP

```{bash}
##======
##Scanning genomes
##======

##-----------
##delbrueckii_plasmid_RMK202_01
##-----------

grep "_plasmid" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/LD_singu/annot.gff |awk -F "\t" ' { if ($3=="CDS") print $0 } '
#23
grep "_plasmid" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/LD_singu/annot.gff |awk -F "\t" ' { if ($3=="CDS") print $0 } '|grep "hypothetical" -v
#15


#-----------
##L_delbrueckii_RMK202
##-----------

grep "Pseudo Genes (total)              ::" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/LD_singu/annot.gbk |head -1| sed 's/Pseudo Genes (total)              :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk 'BEGIN{OFS="\t"}{print "L_delbrueckii_RMK202",$0}'

```

#####GENOME prep

```{bash}
mkdir -p /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/{log,Ldel,Sterm,Lplant}

curl 'ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/bacteria/assembly_summary.txt' >  /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/log/all_refseq.txt
##Lactobacillus delbrueckii subsp. lactis

grep "Lactobacillus delbrueckii subsp. lactis" -i /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/log/all_refseq.txt 



grep "Lactobacillus delbrueckii subsp. lactis" -i /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/log/all_refseq.txt |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >    \
    //home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/log/ldel_download.txt
    cd //home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/Ldel
    wget -i /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/log/ldel_download.txt


##Lactobacillus plantarum

short=Lplant
mkdir -p /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${Lplant}

grep "Lactobacillus plantarum" -i /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/log/all_refseq.txt |head -20



grep "Lactobacillus plantarum" -i /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/log/all_refseq.txt  |head -20 |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >    \
    //home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/log/${short}_download.txt
    cd //home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}
    wget -i /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/log/${short}_download.txt

##Streptococcus thermophilus

short=sterm
mkdir -p /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}

#grep "Streptococcus thermophilus" -i /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/log/all_refseq.txt 
grep "Streptococcus thermophilus" -i /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/log/all_refseq.txt |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome") print $0}' |wc -l



grep "Streptococcus thermophilus" -i /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/log/all_refseq.txt |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome") print $0}' |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >    \
    //home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/log/${short}_download.txt
    cd //home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}
    wget -i /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/log/${short}_download.txt

##Streptococcus salivarius

short=Ssal
mkdir -p /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}

#grep "Streptococcus thermophilus" -i /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/log/all_refseq.txt 
grep "Streptococcus salivarius" -i /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/log/all_refseq.txt |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome") print $0}' |wc -l



grep "Streptococcus salivarius" -i /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/log/all_refseq.txt |awk -F "\t" 'BEGIN{OFS="\t"} {if ($12=="Complete Genome") print $0}' |cut -f 20 | sed -r 's|(ftp://ftp.ncbi.nlm.nih.gov/genomes/all/.+/)(GCF_.+)|\1\2\/\2_genomic.gbff.gz|' >    \
    //home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/log/${short}_download.txt
    cd //home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}
    wget -i /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/log/${short}_download.txt

```


#####pseudogene
Here, I want to create a pseudo gene comparison

```{bash}
short=Ldel
genus=Lactobacillus
what=pseudogene

##============
##pseudogenes
##============
mkdir -p /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}
##--------------------------
##Ldel
rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |head -1000 | grep "Pseudo Genes (total)              ::" |head -1| sed 's/Pseudo Genes (total)              :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk -v genomesss="$shortGenome" -v speciesss="$short" 'BEGIN{OFS="\t"}{print genomesss , speciesss , "NCBI",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/pseudogene/${short}_${what}.txt

done
grep "Pseudo Genes (total)              ::" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/LD_singu/annot.gbk |head -1|sed 's/Pseudo Genes (total)              :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk  'BEGIN{OFS="\t"}{print "L_delbrueckii_RMK202" ,"Ldel", "rmk202",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/pseudogene/${short}_${what}.txt
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt

##--------------------------
##Lplant
short=Lplant

rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |head -1000 | grep "Pseudo Genes (total)              ::" |head -1| sed 's/Pseudo Genes (total)              :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk -v genomesss="$shortGenome" -v speciesss="$short" 'BEGIN{OFS="\t"}{print genomesss , speciesss , "NCBI",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/pseudogene/${short}_${what}.txt

done
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt

###=====================-----------------------------------------
##Sterm
###=====================-----------------------------------------
short=sterm
genus=Streptococcus
what=pseudogene

##============
##pseudogenes
##============
mkdir -p /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}
##--------------------------
##Ldel
rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |head -1000 | grep "Pseudo Genes (total)              ::" |head -1| sed 's/Pseudo Genes (total)              :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk -v genomesss="$shortGenome" -v speciesss="$short" 'BEGIN{OFS="\t"}{print genomesss , speciesss , "NCBI",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt

done
grep "Pseudo Genes (total)              ::" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/annot.gbk |head -1|sed 's/Pseudo Genes (total)              :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk  'BEGIN{OFS="\t"}{print "S_thermophilus_RMK202" ,"sterm", "rmk202",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt

##--------------------------
##Lplant
short=Ssal

rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |head -1000 | grep "Pseudo Genes (total)              ::" |head -1| sed 's/Pseudo Genes (total)              :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk -v genomesss="$shortGenome" -v speciesss="$short" 'BEGIN{OFS="\t"}{print genomesss , speciesss , "NCBI",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/pseudogene/${short}_${what}.txt

done

cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/Streptococcus_${what}.txt /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/Lactobacillus_${what}.txt > /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/all_${what}.txt

```


#####rRNA

```{bash}

##============
##pseudogenes
##============
what=rRNA
short=Ldel
genus=Lactobacillus
mkdir -p /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}
##--------------------------
##Ldel
rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |head -1000 | grep "Genes (RNA)                       :: " |head -1| sed 's/Genes (RNA)                       :: //g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk -v genomesss="$shortGenome" -v speciesss="$short" 'BEGIN{OFS="\t"}{print genomesss , speciesss , "NCBI",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt

done
grep "Genes (RNA)                       :: " /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/LD_singu/annot.gbk |head -1|sed 's/Genes (RNA)                       :: //g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk  'BEGIN{OFS="\t"}{print "L_delbrueckii_RMK202" ,"Ldel", "rmk202",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt

##--------------------------
##Lplant
short=Lplant

rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |head -1000 | grep "Genes (RNA)                       :: " |head -1| sed 's/Genes (RNA)                       :: //g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk -v genomesss="$shortGenome" -v speciesss="$short" 'BEGIN{OFS="\t"}{print genomesss , speciesss , "NCBI",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt

done
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt



###=====================-----------------------------------------
##Sterm
###=====================-----------------------------------------
short=sterm
genus=Streptococcus
what=rRNA

##============
##rRNA
##============
mkdir -p /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}
##--------------------------
##Ldel
rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |head -1000 | grep "Genes (RNA)                       ::" |head -1| sed 's/Genes (RNA)                       :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk -v genomesss="$shortGenome" -v speciesss="$short" 'BEGIN{OFS="\t"}{print genomesss , speciesss , "NCBI",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt

done
grep "Genes (RNA)                       ::" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/annot.gbk |head -1|sed 's/Genes (RNA)                       :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk  'BEGIN{OFS="\t"}{print "S_thermophilus_RMK202" ,"sterm", "rmk202",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt

##--------------------------
##Lplant
short=Ssal

rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |head -1000 | grep "Genes (RNA)                       ::" |head -1| sed 's/Genes (RNA)                       :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk -v genomesss="$shortGenome" -v speciesss="$short" 'BEGIN{OFS="\t"}{print genomesss , speciesss , "NCBI",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt

done

cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/Streptococcus_${what}.txt /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/Lactobacillus_${what}.txt > /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/all_${what}.txt

```

#####genes

```{bash}

##============
##genes
##============
what=genes
short=Ldel
genus=Lactobacillus

mkdir -p /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}
##--------------------------
##Ldel
rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |head -1000 | grep "Genes (total)                     :: " |head -1| sed 's/Genes (total)                     :: //g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk -v genomesss="$shortGenome" -v speciesss="$short" 'BEGIN{OFS="\t"}{print genomesss , speciesss , "NCBI",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt

done
grep "Genes (total)                     :: " /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/LD_singu/annot.gbk |head -1|sed 's/Genes (total)                     :: //g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk  'BEGIN{OFS="\t"}{print "L_delbrueckii_RMK202" ,"Ldel", "rmk202",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt

cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt

##--------------------------
##Lplant
short=Lplant

rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |head -1000 | grep "Genes (total)                     :: " |head -1| sed 's/Genes (total)                     :: //g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk -v genomesss="$shortGenome" -v speciesss="$short" 'BEGIN{OFS="\t"}{print genomesss , speciesss , "NCBI",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt

done
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt

###=====================-----------------------------------------
##Sterm
###=====================-----------------------------------------
short=sterm
genus=Streptococcus
what=genes

##============
##genes
##============
mkdir -p /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}
##--------------------------
##Ldel
rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |head -1000 | grep "Genes (total)                     ::" |head -1| sed 's/Genes (total)                     :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk -v genomesss="$shortGenome" -v speciesss="$short" 'BEGIN{OFS="\t"}{print genomesss , speciesss , "NCBI",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt

done
grep "Genes (total)                     ::" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/annot.gbk |head -1|sed 's/Genes (total)                     :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk  'BEGIN{OFS="\t"}{print "S_thermophilus_RMK202" ,"sterm", "rmk202",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt

##--------------------------
##Lplant
short=Ssal

rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |head -1000 | grep "Genes (total)                     ::" |head -1| sed 's/Genes (total)                     :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk -v genomesss="$shortGenome" -v speciesss="$short" 'BEGIN{OFS="\t"}{print genomesss , speciesss , "NCBI",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt

done

cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/Streptococcus_${what}.txt /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/Lactobacillus_${what}.txt > /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/all_${what}.txt


```


#####CRISPR

```{bash}

##============
##CRISPR
##============
what=CRISPR
short=Ldel
genus=Lactobacillus
mkdir -p /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}
##--------------------------
##Ldel
rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |head -2000 | grep "CRISPR Arrays                     ::" |head -1| sed 's/CRISPR Arrays                     :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk -v genomesss="$shortGenome" -v speciesss="$short" 'BEGIN{OFS="\t"}{print genomesss , speciesss , "NCBI",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt

done
grep "CRISPR Arrays                     ::" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/LD_singu/annot.gbk |head -1|sed 's/CRISPR Arrays                     :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk  'BEGIN{OFS="\t"}{print "L_delbrueckii_RMK202" ,"Ldel", "rmk202",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt

##--------------------------
##Lplant
short=Lplant

rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |head -2000 | grep "CRISPR Arrays                     ::" |head -1| sed 's/CRISPR Arrays                     :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk -v genomesss="$shortGenome" -v speciesss="$short" 'BEGIN{OFS="\t"}{if ($1==" ") print genomesss , speciesss , "NCBI","0" ; else print genomesss , speciesss , "NCBI",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt

done
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt


###=====================-----------------------------------------
##Sterm
###=====================-----------------------------------------
short=sterm
genus=Streptococcus
what=CRISPR

##============
##CRISPR
##============
mkdir -p /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}
##--------------------------
##Ldel
rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |head -1000 | grep "CRISPR Arrays                     ::" |head -1| sed 's/CRISPR Arrays                     :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk -v genomesss="$shortGenome" -v speciesss="$short" 'BEGIN{OFS="\t"}{print genomesss , speciesss , "NCBI",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt

done
grep "CRISPR Arrays                     ::" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/annot.gbk |head -1|sed 's/CRISPR Arrays                     :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk  'BEGIN{OFS="\t"}{print "S_thermophilus_RMK202" ,"sterm", "rmk202",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt

##--------------------------
##Lplant
short=Ssal

rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |head -1000 | grep "CRISPR Arrays                     ::" |head -1| sed 's/CRISPR Arrays                     :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk -v genomesss="$shortGenome" -v speciesss="$short" 'BEGIN{OFS="\t"}{print genomesss , speciesss , "NCBI",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt

done

cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/Streptococcus_${what}.txt /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/Lactobacillus_${what}.txt > /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/all_${what}.txt
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/all_${what}.txt


```

#####tRNAs

```{bash}

##============
##tRNAs
##============
what=tRNAs
short=Ldel
genus=Lactobacillus
mkdir -p /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}
##--------------------------
##Ldel
rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |head -2000 | grep "tRNAs                             ::" |head -1| sed 's/tRNAs                             :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk -v genomesss="$shortGenome" -v speciesss="$short" 'BEGIN{OFS="\t"}{print genomesss , speciesss , "NCBI",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt

done
grep "tRNAs                             ::" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/LD_singu/annot.gbk |head -1|sed 's/tRNAs                             :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk  'BEGIN{OFS="\t"}{print "L_delbrueckii_RMK202" ,"Ldel", "rmk202",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt

##--------------------------
##Lplant
short=Lplant

rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |head -2000 | grep "tRNAs                             ::" |head -1| sed 's/tRNAs                             :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk -v genomesss="$shortGenome" -v speciesss="$short" 'BEGIN{OFS="\t"}{if ($1==" ") print genomesss , speciesss , "NCBI","0" ; else print genomesss , speciesss , "NCBI",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt

done
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt

###=====================-----------------------------------------
##Sterm
###=====================-----------------------------------------
short=sterm
genus=Streptococcus
what=tRNAs

##============
##tRNAs
##============
mkdir -p /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}
##--------------------------
##Ldel
rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |head -1000 | grep "tRNAs                             ::" |head -1| sed 's/tRNAs                             :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk -v genomesss="$shortGenome" -v speciesss="$short" 'BEGIN{OFS="\t"}{print genomesss , speciesss , "NCBI",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt

done
grep "tRNAs                             ::" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/annot.gbk |head -1|sed 's/tRNAs                             :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk  'BEGIN{OFS="\t"}{print "S_thermophilus_RMK202" ,"sterm", "rmk202",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt

##--------------------------
##Lplant
short=Ssal

rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |head -1000 | grep "tRNAs                             ::" |head -1| sed 's/tRNAs                             :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk -v genomesss="$shortGenome" -v speciesss="$short" 'BEGIN{OFS="\t"}{print genomesss , speciesss , "NCBI",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt

done

cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/Streptococcus_${what}.txt /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/Lactobacillus_${what}.txt > /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/all_${what}.txt



```

#####ncRNAs

```{bash}

##============
##ncRNAs
##============
what=ncRNAs
short=Ldel
genus=Lactobacillus
mkdir -p /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}
##--------------------------
##Ldel
rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |head -2000 | grep "ncRNAs                            ::" |head -1| sed 's/ncRNAs                            :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk -v genomesss="$shortGenome" -v speciesss="$short" 'BEGIN{OFS="\t"}{print genomesss , speciesss , "NCBI",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt

done
grep "ncRNAs                            ::" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/LD_singu/annot.gbk |head -1|sed 's/ncRNAs                            :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk  'BEGIN{OFS="\t"}{print "L_delbrueckii_RMK202" ,"Ldel", "rmk202",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt

##--------------------------
##Lplant
short=Lplant

rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |head -2000 | grep "ncRNAs                            ::" |head -1| sed 's/ncRNAs                            :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk -v genomesss="$shortGenome" -v speciesss="$short" 'BEGIN{OFS="\t"}{if ($1==" ") print genomesss , speciesss , "NCBI","0" ; else print genomesss , speciesss , "NCBI",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt

done
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt

###=====================-----------------------------------------
##Sterm
###=====================-----------------------------------------
short=sterm
genus=Streptococcus
what=ncRNAs

##============
##ncRNAs
##============
mkdir -p /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}
##--------------------------
##Ldel
rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |head -1000 | grep "ncRNAs                            ::" |head -1| sed 's/ncRNAs                            :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk -v genomesss="$shortGenome" -v speciesss="$short" 'BEGIN{OFS="\t"}{print genomesss , speciesss , "NCBI",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt

done
grep "ncRNAs                            ::" /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/annot.gbk |head -1|sed 's/ncRNAs                            :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk  'BEGIN{OFS="\t"}{print "S_thermophilus_RMK202" ,"sterm", "rmk202",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt

##--------------------------
##Lplant
short=Ssal

rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |head -1000 | grep "ncRNAs                            ::" |head -1| sed 's/ncRNAs                            :://g'| sed -e 's/^[ \t]*//' |sed -e 's/*[ \t]$//' |awk -v genomesss="$shortGenome" -v speciesss="$short" 'BEGIN{OFS="\t"}{print genomesss , speciesss , "NCBI",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt

done

cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/Streptococcus_${what}.txt /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/Lactobacillus_${what}.txt > /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/all_${what}.txt
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/all_${what}.txt

```

#####transposase

```{bash}

##============
##transposase
##============
what=transposase
short=Ldel
genus=Lactobacillus
mkdir -p /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}
##--------------------------
##Ldel
rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |grep "transposase" -c |awk -v genomesss="$shortGenome" -v speciesss="$short" 'BEGIN{OFS="\t"}{print genomesss , speciesss , "NCBI",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt

done
grep "transposase" -c /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/LD_singu/annot.gbk  |awk  'BEGIN{OFS="\t"}{print "L_delbrueckii_RMK202" ,"Ldel", "rmk202",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt

##--------------------------
##Lplant
short=Lplant

rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |grep "transposase" -c  |awk -v genomesss="$shortGenome" -v speciesss="$short" 'BEGIN{OFS="\t"}{if ($1==" ") print genomesss , speciesss , "NCBI","0" ; else print genomesss , speciesss , "NCBI",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt

done
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt

###=====================-----------------------------------------
##Sterm
###=====================-----------------------------------------
short=sterm
genus=Streptococcus
what=transposase

##============
##transposase
##============
mkdir -p /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}
##--------------------------
##Ldel
rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |grep "transposase" -c |awk -v genomesss="$shortGenome" -v speciesss="$short" 'BEGIN{OFS="\t"}{print genomesss , speciesss , "NCBI",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt

done
grep "transposase" -c /home/vincent/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/annot.gbk  |awk  'BEGIN{OFS="\t"}{print "S_thermophilus_RMK202" ,"sterm", "rmk202",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt > /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt

##--------------------------
##Lplant
short=Ssal

rm /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt
for genomes in $(ls /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short})
do
  shortGenome=$(echo ${genomes} |cut -d '.' -f 1)
  zcat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/genome/${short}/${genomes} |grep "transposase" -c  |awk -v genomesss="$shortGenome" -v speciesss="$short" 'BEGIN{OFS="\t"}{if ($1==" ") print genomesss , speciesss , "NCBI","0" ; else print genomesss , speciesss , "NCBI",$0}' >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt

done
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${short}_${what}.txt >> /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/${genus}_${what}.txt


cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/Lactobacillus_${what}.txt /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/Streptococcus_${what}.txt > /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/all_${what}.txt
cat /home/vincent/Projects/2019_pilotplant/comparativeGenomics/${what}/all_${what}.txt

```

####rAnalysis
```{r}
library(readr)
source("/home/vincent/Desktop/ScriptRepository/R_functions-master/R_functions/g_legend.R") ##for plotting only legends

##=========================================
##---Pseudo

Lactobacillus_pseudogene <- read_delim("~/Projects/2019_pilotplant/comparativeGenomics/pseudogene/all_pseudogene.txt",  "\t", escape_double = FALSE, col_names =c("genome","species","location","count"),  trim_ws = TRUE)

pseudoPlot <- ggplot(Lactobacillus_pseudogene,aes(x=species,y=count))+geom_boxplot()+geom_dotplot(aes(color=location,fill=location),binaxis='y', stackdir='center', dotsize=1)+
  labs(x="",
       y="pseudogene")+
  theme_classic()+
  theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
        #legend.position = c(0.02, 0.98),
          # legend.justification = c("left", "top"),
          # legend.title = element_blank()
          legend.position="none"        
          )
pseudoPlot

##=========================================
##---rRNA

Lactobacillus_rRNA <- read_delim("~/Projects/2019_pilotplant/comparativeGenomics/rRNA/all_rRNA.txt",  "\t", escape_double = FALSE, col_names =c("genome","species","location","count"),  trim_ws = TRUE)

rRNAPlot <- ggplot(Lactobacillus_rRNA,aes(x=species,y=count))+geom_boxplot()+geom_dotplot(aes(color=location,fill=location),binaxis='y', stackdir='center', dotsize=1)+
  labs(x="",
       y="rRNA")+
  theme_classic()+
  theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
        #legend.position = c(0.02, 0.98),
          # legend.justification = c("left", "top"),
          # legend.title = element_blank()
        legend.position="none"
          )
rRNAPlot

##=========================================
##---ncRNAs

Lactobacillus_ncRNAs <- read_delim("~/Projects/2019_pilotplant/comparativeGenomics/ncRNAs/all_ncRNAs.txt",  "\t", escape_double = FALSE, col_names =c("genome","species","location","count"),  trim_ws = TRUE)

ncRNAsPlot <- ggplot(Lactobacillus_ncRNAs,aes(x=species,y=count))+geom_boxplot()+geom_dotplot(aes(color=location,fill=location),binaxis='y', stackdir='center', dotsize=1)+
  labs(x="",
       y="ncRNAs")+
  theme_classic()+
  theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
        legend.position = c(0.02, 0.98),
          legend.justification = c("left", "top"),
          legend.title = element_blank()
          )
ncRNAsPlot

##=========================================
##---genes

Lactobacillus_genes <- read_delim("~/Projects/2019_pilotplant/comparativeGenomics/genes/all_genes.txt",  "\t", escape_double = FALSE, col_names =c("genome","species","location","count"),  trim_ws = TRUE)

genesPlot <- ggplot(Lactobacillus_genes,aes(x=species,y=count))+geom_boxplot()+geom_dotplot(aes(color=location,fill=location),binaxis='y', stackdir='center', dotsize=1)+
  labs(x="",
       y="genes")+
  theme_classic()+
  theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
        legend.position = c(0.02, 0.98),
          legend.justification = c("left", "top"),
          legend.title = element_blank()
          )
genesPlot

##=========================================
##---CRISPR

Lactobacillus_CRISPR <- read_delim("~/Projects/2019_pilotplant/comparativeGenomics/CRISPR/all_CRISPR.txt",  "\t", escape_double = FALSE, col_names =c("genome","species","location","count"),  trim_ws = TRUE)

CRISPRPlot <- ggplot(Lactobacillus_CRISPR,aes(x=species,y=count))+geom_boxplot()+geom_dotplot(aes(color=location,fill=location),binaxis='y', stackdir='center', dotsize=1)+
  labs(x="",
       y="CRISPR")+
  theme_classic()+
  theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
        #legend.position = c(0.02, 0.98),
          # legend.justification = c("left", "top"),
          # legend.title = element_blank()
        legend.position="none"
          )
CRISPRPlot


##=========================================
##---transposase

Lactobacillus_transposase <- read_delim("~/Projects/2019_pilotplant/comparativeGenomics/transposase/all_transposase.txt",  "\t", escape_double = FALSE, col_names =c("genome","species","location","count"),  trim_ws = TRUE)

transposasePlot <- ggplot(Lactobacillus_transposase,aes(x=species,y=count))+geom_boxplot()+geom_dotplot(aes(color=location,fill=location),binaxis='y', stackdir='center', dotsize=1)+
  labs(x="",
       y="transposase")+
  theme_classic()+
  theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
        #legend.position = c(0.02, 0.98),
          # legend.justification = c("left", "top"),
          # legend.title = element_blank()
        legend.position="none"
          )
transposasePlot

##=========================================
##---tRNAs

Lactobacillus_tRNAs <- read_delim("~/Projects/2019_pilotplant/comparativeGenomics/tRNAs/all_tRNAs.txt",  "\t", escape_double = FALSE, col_names =c("genome","species","location","count"),  trim_ws = TRUE)

tRNAsPlot <- ggplot(Lactobacillus_transposase,aes(x=species,y=count))+geom_boxplot()+geom_dotplot(aes(color=location,fill=location),binaxis='y', stackdir='center', dotsize=1)+
  labs(x="",
       y="tRNAs")+
  theme_classic()+
  theme(axis.text.x = element_text(angle = 75, hjust = 1,size=9),
        #legend.position = c(0.02, 0.98),
          # legend.justification = c("left", "top"),
          # legend.title = element_blank()
        legend.position="none"
          )
tRNAsPlot

##=========================================
##---arrange

legend <- g_legend(genesPlot)
grid.arrange(genesPlot+theme(legend.position="none"),rRNAPlot,tRNAsPlot,legend,pseudoPlot,transposasePlot,CRISPRPlot , ncol=4)


svg("~/Desktop/presenations/my_presentations/20190807_committeeMeeting/figures/ldelGenomes.svg",width=10,height=5)
grid.arrange(genesPlot+theme(legend.position="none"),rRNAPlot,tRNAsPlot,legend,pseudoPlot,transposasePlot,CRISPRPlot , ncol=4)

dev.off()
```


##Circos plot for CRISPR

Here I do three CIRCOS plots to show the CRISPRs and finally put them together with the phage to show where they actually match onto the phage

Now we go step by step what we actually need:

#####Prokka annoation and pilerCR of start aligned fasta

```{bash}
###-----------------
###---------move files on big machine
###-----------------
mkdir -p /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/{S50,S72}

#/home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/S50/alignment3/S50.fna.fas
scp /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/S50/alignment3/S50.fna.fas vincent@130.223.51.151:/archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/

scp /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/s72_03/alignment2/S72.fna.fas vincent@130.223.51.151:/archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/


###-----------------
###--------PROKKA annotation
###-----------------

rm -r mkdir -p /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/PROKKA
    mkdir -p /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/PROKKA
    
prokka --outdir /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/PROKKA --force --usegenus --genus Streptococcus --locustag mst1 --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/S50.fna.fas
  
rm -r mkdir -p /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/PROKKA
    mkdir -p /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/PROKKA
    
prokka --outdir /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/PROKKA --force --usegenus --genus Streptococcus --locustag mst2 --proteins proteins.faa --evalue 0.001 --addgenes /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/S72.fna.fas


###-----------------
###--------pilerCR
###-----------------

rm -r mkdir -p /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR
    mkdir -p /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR

~/apps/PilerCR/pilercr1.06/pilercr -in /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/S50.fna.fas \
    -out /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out -noinfo\
    -seq /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/pilerCR/pilarCR_out.fasta

rm -r mkdir -p /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR
    mkdir -p /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR

~/apps/PilerCR/pilercr1.06/pilercr -in /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/S72.fna.fas \
    -out /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out -noinfo\
    -seq /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/pilerCR/pilarCR_out.fasta

###-----------------
###---------move files back local
###-----------------
mkdir -p /archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/{S50,S72}

#/home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/S50/alignment3/S50.fna.fas
scp  -r vincent@130.223.51.151:/archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S50/ /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/S50/alignment3/

scp -r vincent@130.223.51.151:/archiv/Projects/2018_Culturomics/12_CRISPR_isolates/CIRCOS/S72/ /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/s72_03/alignment2/


```


#####2. genome file

```{bash}

home=/home/vincent/bin/apps/circos-0.69-6/

##karyotyp
${home}/data/karyotype/sterm_rmk202.txt


mkdir -p /home/vincent/bin/apps/circos-0.69-6/data/rmk202/


###------------
##meta
###------------
grep "CRISPR" -i ~/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/annot.gff | awk -F "\t" 'BEGIN{OFS="\t"}{print $1,$4,$5}' > /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/transposase_sterm.txt

less  ~/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/*.gbk

grep "CRISPR Arrays" -i ~/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/*.gbk
grep "CRISPR" -i ~/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/*.gbk


##repeat region --> look in pilerCR how many repeats
grep "CRISPR;rpt_type" -i ~/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/annot.gff | awk -F "\t" 'BEGIN{OFS="\t"}{print $1,$4-2000,$5+2000}' > /home/vincent/bin/apps/circos-0.69-6/data/rmk202/metaRMK202_CRISPRarray.txt

grep "CRISPR;rpt_type" -i ~/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/annot.gff | awk -F "\t" 'BEGIN{OFS="\t"}{print $1,$4-20000,$5}' > /home/vincent/bin/apps/circos-0.69-6/data/rmk202/metaRMK202_CRISPRarray_large.txt

##cas-gene endonuclease
grep "endonuclease Cas" -i ~/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/annot.gff | awk -F "[\t;]" 'BEGIN{OFS="\t"}{print $1,$4,$5+20000,$13}' | sed 's/gene_//g' > /home/vincent/bin/apps/circos-0.69-6/data/rmk202/metaRMK202_Cas.txt
#grep "endonuclease Cas" -i ~/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/annot.gff | awk -F "[\t;]" 'BEGIN{OFS="\t"}{print $1,$4,$5,$13}' | sed 's/gene_//g' > /home/vincent/bin/apps/circos-0.69-6/data/rmk202/metaRMK202_Cas.txt

##crispr-associated gene (csm)
grep "protein Csm" -i ~/Desktop/Projects/2019_Pilotplan/11_PGAP/vincent_2/ST_singu/annot.gff | awk -F "[\t;]" 'BEGIN{OFS="\t"}{print $1,$4,$5+20000,$13}' | sed 's/gene_//g' >> /home/vincent/bin/apps/circos-0.69-6/data/rmk202/metaRMK202_Csm.txt

cat /home/vincent/bin/apps/circos-0.69-6/data/rmk202/metaRMK202_Cas.txt /home/vincent/bin/apps/circos-0.69-6/data/rmk202/metaRMK202_Csm.txt

#example run
cd /home/vincent/bin/apps/circos-0.69-6/
cp etc/sterm_rmk202_circos.conf etc/sterm_rmk202_metagenome_circos.conf
bin/circos -conf etc/sterm_rmk202_metagenome_circos.conf ; firefox ./circos.svg &
bin/circos -conf etc/sterm_rmk202_metagenome_circos.conf ; cp ./circos.svg ./meta_rmk202_circos.svg; firefox ./meta_rmk202_circos.svg &



##--------------CRISPR repeats

grep "SUMMARY BY SIMILARITY" -A 15 /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm//S_thermophilus_RMK202.pilarCR_out |grep "S_thermophilus_R"| sed -e 's/^[ \t]*//' |awk -F " " 'BEGIN{OFS="\t"}{print "S_thermophilus_RMK202",$3-2000,$3+$4+20000,$5,$6,$7,$8,$9,"meta_rmk202"}' > /home/vincent/bin/apps/circos-0.69-6/data/rmk202/metaRMK202_CRISPR_Repeat_long.txt

###------------
##mst 1
###------------


##--------------genome mapping

less /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/S50/alignment3/S50.fna_contigs.tab
grep "^contig" /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/S50/alignment3/S50.fna_contigs.tab | awk -F "[\t]" 'BEGIN{OFS="\t"}{print "S_thermophilus_RMK202",$5,$6}' >  /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst1_genome.txt

##--------------CRISPR repeats

grep "SUMMARY BY SIMILARITY" -A 15 /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/S50/alignment3/S50/pilerCR/pilarCR_out |grep "NODE" > /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst1_infoRepeat.txt
id=NODE_20_length_4

rm /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst1_CRISPR_Repeat.txt
rm /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst1_CRISPR_Repeat_long.txt
for id in $(grep "SUMMARY BY SIMILARITY" -A 15 /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/S50/alignment3/S50/pilerCR/pilarCR_out |grep "NODE"| sed -e 's/^[ \t]*//'|awk -F " " 'BEGIN{OFS="\t"}{print $2}')
  do
  ##get location of rmk202 meta
  baselocation=$(grep "${id}" /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/S50/alignment3/S50.fna_contigs.tab |awk -F "\t" 'BEGIN{OFS="\t"}{print $5}')
  ###get new locations
   grep "${id}" /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst1_infoRepeat.txt | sed -e 's/^[ \t]*//' |awk -F " " -v loc="$baselocation"  'BEGIN{OFS="\t"}{print "S_thermophilus_RMK202",loc+$3-2000,loc+$3+$4+2000}' >> /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst1_CRISPR_Repeat.txt
    grep "${id}" /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst1_infoRepeat.txt | sed -e 's/^[ \t]*//' |awk -F " " -v loc="$baselocation"  'BEGIN{OFS="\t"}{print "S_thermophilus_RMK202",loc+$3-2000,loc+$3+$4+20000,$5,$6,$7,$8,$9,"mst1"}' >> /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst1_CRISPR_Repeat_long.txt
  done


##--------------cas
rm /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst1_Cas.txt
for id in $(grep "SUMMARY BY SIMILARITY" -A 15 /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/S50/alignment3/S50/pilerCR/pilarCR_out |grep "NODE"| sed -e 's/^[ \t]*//'|awk -F " " 'BEGIN{OFS="\t"}{print $2}')
  do
  ##get location of rmk202 meta
  baselocation=$(grep "${id}" /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/S50/alignment3/S50.fna_contigs.tab |awk -F "\t" 'BEGIN{OFS="\t"}{print $5}')
  ###get new locations

grep "CRISPR" -i /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/S50/alignment3/S50/PROKKA/PROKKA_09272019.gff | sed 's/;eC//g'|grep "${id}"| awk -F "[\t;]" -v loc="$baselocation"  'BEGIN{OFS="\t"}{print "S_thermophilus_RMK202",loc+$4,loc+$5+20000,$12}' | grep -v "locus_tag=mst1_01056"| sed 's/gene=//g' >> \
/home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst1_Cas.txt
done

#example run
cd /home/vincent/bin/apps/circos-0.69-6/
cp etc/sterm_rmk202_metagenome_circos.conf etc/sterm_rmk202_mst1_circos.conf
bin/circos -conf etc/sterm_rmk202_mst1_circos.conf ; cp ./circos.svg ./mst1_circos.svg; firefox ./mst1_circos.svg &



###------------
##mst 2
###------------


##--------------genome mapping

less  /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/s72_03/alignment2/S72.fna_contigs.tab
grep "^contig" /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/s72_03/alignment2/S72.fna_contigs.tab | awk -F "[\t]" 'BEGIN{OFS="\t"}{print "S_thermophilus_RMK202",$5,$6}'|sort|uniq >  /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst2_genome.txt

##--------------CRISPR repeats

grep "SUMMARY BY SIMILARITY" -A 15 /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/s72_03/alignment2/S72/pilerCR/pilarCR_out |grep "NODE" > /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst2_infoRepeat.txt
id=NODE_20_length_4

rm /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst2_CRISPR_Repeat.txt
rm /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst2_CRISPR_Repeat_long.txt
for id in $(grep "SUMMARY BY SIMILARITY" -A 15 /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/s72_03/alignment2/S72/pilerCR/pilarCR_out |grep "NODE"| sed -e 's/^[ \t]*//'|awk -F " " 'BEGIN{OFS="\t"}{print $2}')
  do
  ##get location of rmk202 meta
  baselocation=$(grep "${id}" /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/s72_03/alignment2/S72.fna_contigs.tab |awk -F "\t" 'BEGIN{OFS="\t"}{print $5}')
  ###get new locations
   grep "${id}" /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst2_infoRepeat.txt | sed -e 's/^[ \t]*//' |awk -F " " -v loc="$baselocation"  'BEGIN{OFS="\t"}{print "S_thermophilus_RMK202",loc+$3-2000,loc+$3+$4+2000}' >> /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst2_CRISPR_Repeat.txt
    grep "${id}" /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst2_infoRepeat.txt | sed -e 's/^[ \t]*//' |awk -F " " -v loc="$baselocation"  'BEGIN{OFS="\t"}{print "S_thermophilus_RMK202",loc+$3-2000,loc+$3+$4+20000,$5,$6,$7,$8,$9,"mst2"}' >> /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst2_CRISPR_Repeat_long.txt
  done


##--------------cas
rm /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst2_Cas.txt
for id in $(grep "SUMMARY BY SIMILARITY" -A 15 /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/s72_03/alignment2/S72/pilerCR/pilarCR_out |grep "NODE"| sed -e 's/^[ \t]*//'|awk -F " " 'BEGIN{OFS="\t"}{print $2}')
  do
  ##get location of rmk202 meta
  baselocation=$(grep "${id}" /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/s72_03/alignment2/S72.fna_contigs.tab |awk -F "\t" 'BEGIN{OFS="\t"}{print $5}')
  ###get new locations

grep "CRISPR" -i /home/vincent/Desktop/Projects/2019_Pilotplan/assembledGenomes/startaligned/sterm/s72_03/alignment2/S72/PROKKA/PROKKA_09272019.gff | sed 's/;eC//g'|grep "${id}"| awk -F "[\t;]" -v loc="$baselocation"  'BEGIN{OFS="\t"}{print "S_thermophilus_RMK202",loc+$4,loc+$5+20000,$12}' | grep -v "locus_tag=mst2_01056"| sed 's/gene=//g' >> \
/home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst2_Cas.txt
done

#example run
cd /home/vincent/bin/apps/circos-0.69-6/
cp etc/sterm_rmk202_mst1_circos.conf etc/sterm_rmk202_mst2_circos.conf
bin/circos -conf etc/sterm_rmk202_mst2_circos.conf ; cp ./circos.svg ./mst2_circos.svg; firefox ./mst2_circos.svg  &


grep "CRISPR" -i /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/PROKKA_08052019.gff
grep "CRISPR" -i /home/vincent/bin/apps/circos-0.69-6/data/nwc_rmk/PROKKA_08052019.gbk




##===============
###=======meta analysis
##===============


##----------CRISPR_array

cat /home/vincent/bin/apps/circos-0.69-6/data/rmk202/metaRMK202_CRISPR_Repeat_long.txt \
  /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst1_CRISPR_Repeat_long.txt \
  /home/vincent/bin/apps/circos-0.69-6/data/rmk202/mst2_CRISPR_Repeat_long.txt >   /home/vincent/bin/apps/circos-0.69-6/data/rmk202/all_CRISPR_Repeat_long.txt


##----------CRISPR_array CD-HIT
mkdir -p /home/vincent/bin/apps/circos-0.69-6/data/rmk202/CRISPRspacers
cut -f 7,8 /home/vincent/bin/apps/circos-0.69-6/data/rmk202/all_CRISPR_Repeat_long.txt | sed 's/+\t/>CRISPR_spacer\n/g' > /home/vincent/bin/apps/circos-0.69-6/data/rmk202/CRISPRspacers/spacers.fasta


cd-hit -i /home/vincent/bin/apps/circos-0.69-6/data/rmk202/CRISPRspacers/spacers.fasta \
    -o q -c 0.9 -M 30000 -T 37
  
##----------mafft

cp /home/vincent/bin/apps/circos-0.69-6/data/rmk202/CRISPRspacers/spacers.fasta /home/vincent/bin/apps/circos-0.69-6/data/rmk202/CRISPRspacers/spacers_array1.fasta
cp /home/vincent/bin/apps/circos-0.69-6/data/rmk202/CRISPRspacers/spacers.fasta /home/vincent/bin/apps/circos-0.69-6/data/rmk202/CRISPRspacers/spacers_array2.fasta
cp /home/vincent/bin/apps/circos-0.69-6/data/rmk202/CRISPRspacers/spacers.fasta /home/vincent/bin/apps/circos-0.69-6/data/rmk202/CRISPRspacers/spacers_array3.fasta

mafft --clustalout --auto /home/vincent/bin/apps/circos-0.69-6/data/rmk202/CRISPRspacers/spacers_array1.fasta > /home/vincent/bin/apps/circos-0.69-6/data/rmk202/CRISPRspacers/spacers_array1.msa

##----------Cas


```

```{r}
library(knitr)
library(readr)
spacers_array1 <- read_table2("/home/vincent/bin/apps/circos-0.69-6/data/rmk202/CRISPRspacers/spacers_array1.msa",  col_names = FALSE, skip = 1)
kable(spacers_array1)
```

##Phage uniprot annotation

```{bash}

mkdir -p /archiv/Projects/2019_pilotplant/02_annotation/phage_uniprot

##---bring to bigmachine
scp /home/vincent/Desktop/Projects/2019_Pilotplan/09_annoation/rastServer/Streptococcus_thermophilus_ViSo-RMK202_b.faa vincent@130.223.51.151:/archiv/Projects/2019_pilotplant/02_annotation/phage_uniprot/

##---bring to bigmachine

diamond blastp --threads 32 --max-target-seqs 1 --db /archiv/TREMBL_database/20190710/uniprot_trembl.dmnd --query /archiv/Projects/2019_pilotplant/02_annotation/phage_uniprot/Streptococcus_thermophilus_ViSo-RMK202_b.faa --out /archiv/Projects/2019_pilotplant/02_annotation/phage_uniprot/Streptococcus_thermophilus_ViSo-RMK202_b.txt


##---bring local
scp -r vincent@130.223.51.151:/archiv/Projects/2018_Culturomics/12_CRISPR_isolates/cdhit/20190312/Streptococcus_thermophilus/clustering_results.txt ~/Desktop/Projects/2018_culturomics/CRISPR/

```


##Appendix

######KEGG annotation
Currently I use the following annotation forms in inform myself:
- CazY for carbohydrate metabolic enzymes (e.g. lipolyse)
- KEGG_enzymes(EC) for enymes (e.g. Hydrolase(very broad--> no specific pathway))
- KEGG_pathway for most genes (e.g. metabolic, transporters,...) [help](https://www.genome.jp/kegg-bin/get_htext#A1)
- KEGG_ko for Transposons

I will the annotation in the order listed above. 


```{r}
library(KEGGREST)

annotation_sterm_snps_high <- annotation_sterm_snps_01[annotation_sterm_snps_01$significance=="HIGH",]
annotation_sterm_snps_high


##number of unananoted genes
table(!is.na(annotation_sterm_snps_high$KEGG_ko))

##===========
###KEGG ko
##===========
annotation_sterm_snps_high_withko <- annotation_sterm_snps_high[!is.na(annotation_sterm_snps_high$KEGG_ko),]
annotation_sterm_snps_high_woko <- annotation_sterm_snps_high[is.na(annotation_sterm_snps_high$KEGG_ko),]

annotation_sterm_snps_high_kegg_02 <- separate_rows(annotation_sterm_snps_high_withko,KEGG_ko,sep = ",",)

# count_keggs <- (sort(table(annotation_sterm_snps_high_kegg_02$KEGG_ko),decreasing = TRUE)) %>% gsub("ko:","",.)
# keggss <- names(sort(table(annotation_sterm_snps_high_kegg_02$KEGG_ko),decreasing = TRUE)) %>% gsub("ko:","",.)
# count_keggs <- (annotation_sterm_snps_high_kegg_02$KEGG_ko) %>% gsub("ko:","",.)

keggss <- (annotation_sterm_snps_high_kegg_02$KEGG_ko) %>% gsub("ko:","",.)


keggnames <- data.frame()
i<- 10
for (i in 1:length(keggss)) {
  keggsss <- keggGet(keggss[i])[[1]]
def_i <- keggsss$DEFINITION  
pathway_contains <- is.null(keggsss$PATHWAY)
  print(paste0(def_i," AND ",keggss[i]))
 keggnames[i,1:3] <- c(keggss[i],def_i,pathway_contains) 
  
}

colnames(keggnames) <- c("KEGG","DEFINITION","pathway_contains")
# keggnames$count <- count_keggs
keggnames$KEGG_original <- gsub("K","ko:K",keggnames$KEGG)
keggnames$GeneName <- separate_rows(annotation_sterm_snps_high_withko,KEGG_ko,sep = ",",)$finalName

###transposase associated KEGG
keggnames[grep("transposase",keggnames$DEFINITION),]


sort(table(keggnames$DEFINITION),decreasing = TRUE)
sort(table(keggnames$pathway_contains),decreasing = TRUE)
sort(table(keggnames$GeneName),decreasing = TRUE)



##===========
###ENYME (EC-NUMBER)
##===========
annotation_sterm_snps_high_withEC <- annotation_sterm_snps_high[!is.na(annotation_sterm_snps_high$EC),]
annotation_sterm_snps_high_woEC <- annotation_sterm_snps_high[is.na(annotation_sterm_snps_high$EC),]

annotation_sterm_snps_high_withEC_02 <- separate_rows(annotation_sterm_snps_high_withEC,EC,sep = ",",)$EC

enzymenames <- data.frame()
 i<- 1
for (i in 1:length(annotation_sterm_snps_high_withEC_02)) {
  keggsss <- keggGet(annotation_sterm_snps_high_withEC_02[i])[[1]]
NAME_i <- keggsss$NAME[1]
def_i <- keggsss$CLASS[1]
def_i_lowest <- keggsss$CLASS[length(keggsss$CLASS)]
enzymenames[i,1:4] <- c(annotation_sterm_snps_high_withEC_02[i],NAME_i,def_i,def_i_lowest) 
  
}

colnames(enzymenames) <- c("EC","NAME","CLASS","lowestCLASS")
enzymenames$GeneName <- separate_rows(annotation_sterm_snps_high_withEC,EC,sep = ",",)$finalName

sort(table(enzymenames$CLASS),decreasing = TRUE)
sort(table(enzymenames$EC),decreasing = TRUE)
sort(table(enzymenames$NAME),decreasing = TRUE)
sort(table(enzymenames$lowestCLASS),decreasing = TRUE)
##===========
###Pathway
##===========
listDatabases()
table(!is.na(annotation_sterm_snps_high$KEGG_Pathway))
annotation_sterm_snps_high_withPathWay <- annotation_sterm_snps_high[!is.na(annotation_sterm_snps_high$KEGG_Pathway),]
annotation_sterm_snps_high_woPathWay <- annotation_sterm_snps_high[is.na(annotation_sterm_snps_high$KEGG_Pathway),]

annotation_sterm_snps_high_kegg_02 <- separate_rows(annotation_sterm_snps_high_withPathWay,KEGG_Pathway,sep = ",",)

nrow(annotation_sterm_snps_high_withPathWay)
nrow(annotation_sterm_snps_high_kegg_02)

keggss <- annotation_sterm_snps_high_kegg_02$KEGG_Pathway

Pathwaynames <- data.frame()
i<- 1
for (i in 1:length(keggss)) {
  keggsss <- keggGet(keggss[i])[[1]]
  name_i <- keggsss$NAME
def_i <- ifelse(is.null(keggsss$CLASS),NA,keggsss$CLASS)
pathway_contains <- ifelse(is.null(keggsss$PATHWAY_MAP),NA,as.vector(keggsss$PATHWAY_MAP))
  print(paste0(name_i ," ||| ",def_i," ||| ",pathway_contains," ||| ",keggss[i]))
 Pathwaynames[i,1:4] <- c(keggss[i],keggsss$NAME,def_i,pathway_contains) 
  
}

colnames(Pathwaynames) <- c("KEGG","NAME","CLASS","pathway")
Pathwaynames$GeneName <- separate_rows(annotation_sterm_snps_high_withPathWay,KEGG_Pathway,sep = ",",)$finalName

table(Pathwaynames[grep("Metabolism",Pathwaynames$CLASS),]$CLASS)
Pathwaynames[(Pathwaynames$CLASS=="Metabolism; Carbohydrate metabolism")==TRUE,]
sort(table(Pathwaynames[grep("Metabolism; Carbohydrate metabolism",Pathwaynames$CLASS),"NAME"]),decreasing = TRUE)
sort(table(Pathwaynames[grep("Environmental Information Processing; Membrane transport",Pathwaynames$CLASS),"NAME"]),decreasing = TRUE)
sort(table(Pathwaynames[grep("Metabolism; Xenobiotics biodegradation and metabolism",Pathwaynames$CLASS),"NAME"]),decreasing = TRUE) ##not needed?
sort(table(Pathwaynames[grep("Metabolism; Nucleotide metabolism",Pathwaynames$CLASS),"NAME"]),decreasing = TRUE)


namess_extract <- Pathwaynames[grep("Pyrimidine metabolism",Pathwaynames$NAME),"GeneName"]

Pathwaynames[Pathwaynames$GeneName %in% namess_extract,]
keggnames[keggnames$GeneName %in% namess_extract,]
 

# sort(table(Pathwaynames$pathway),decreasing = TRUE)
sort(table(Pathwaynames$CLASS),decreasing = TRUE)
sort(table(Pathwaynames$NAME),decreasing = TRUE)

##===========
###CazY
##===========

annotation_sterm_snps_high[!is.na(annotation_sterm_snps_high$CAZy),]

```



######Go-Analysis

```{r}

##================
###prepare for topGO analysis
##================

library(readr)
library(Rgraphviz)
library(topGO)

##------allgenes
geneID2GO_ldel <- readMappings(file = "~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog_GO/Lactobacillus_delbrueckii_RMK202_pgap_query_seqs.fa.emapper.annotations") 
geneID2GO_sterm <- readMappings(file = "~/Desktop/Projects/2019_Pilotplan/04_mapping2ONT/02_againstpolished_single_K1/gfa/AssemblyGraphs/newGenomestich/annotated/eggnog_GO/Streptococcus_thermophilus_RMK202_pgap_query_seqs.fa.emapper.annotations") 
length(geneID2GO_sterm)

##============
##topGO 
##============
##------without modifier snp genes


geneUniverse <- names(geneID2GO_sterm) 
genesOfInterest <- as.character(annotation_sterm_snps_nonModifier$finalName)
length(genesOfInterest)


table(genesOfInterest)

genesOfInterest <- genesOfInterest[genesOfInterest %in% geneUniverse]

 geneList <- factor(as.integer(geneUniverse %in% genesOfInterest))
 names(geneList) <- geneUniverse
 
 
 myGOdata <- new("topGOdata", description="My project", ontology="BP", allGenes=geneList,  annot = annFUN.gene2GO, gene2GO = geneID2GO)
 
 sg <- sigGenes(myGOdata)
 str(sg)
numSigGenes(myGOdata)
resultFisher <- runTest(myGOdata, algorithm="classic", statistic="fisher")
resultFisher

allRes <- GenTable(myGOdata, classicFisher = resultFisher, orderBy = "resultFisher", ranksOf = "classicFisher", topNodes = 200)
allRes

 showSigOfNodes(myGOdata, score(resultFisher), firstSigNodes = 5, useInfo ='all')
 printGraph(myGOdata, resultFisher, firstSigNodes = 5, fn.prefix = "tGO", useInfo = "all", pdfSW = TRUE)
 
 
 length(usedGO(myGOdata))
 
 ##------high and modereate significance mutations
 
annotation_sterm_snps_MODERATE_HIGH <- annotation_sterm_snps_01[annotation_sterm_snps_01$significance=="MODERATE" | annotation_sterm_snps_01$significance=="HIGH",]
table(annotation_sterm_snps_MODERATE_HIGH$significance)
genesOfInterest <- as.character(annotation_sterm_snps_MODERATE_HIGH$finalName)
genesOfInterest <- genesOfInterest[genesOfInterest %in% geneUniverse]

geneList <- factor(as.integer(geneUniverse %in% genesOfInterest))
names(geneList) <- geneUniverse
 
myGOdata <- new("topGOdata", description="My project", ontology="BP", allGenes=geneList,  annot = annFUN.gene2GO, gene2GO = geneID2GO)
 
sg <- sigGenes(myGOdata)
numSigGenes(myGOdata)

resultFisher <- runTest(myGOdata, algorithm="classic", statistic="fisher")
resultFisher

allRes <- GenTable(myGOdata, classicFisher = resultFisher, orderBy = "resultFisher", ranksOf = "classicFisher", topNodes = 200)
allRes

##------high significance mutations
 
annotation_sterm_snps_HIGH <- annotation_sterm_snps_01[annotation_sterm_snps_01$significance=="HIGH",]
table(annotation_sterm_snps_HIGH$significance)
genesOfInterest <- as.character(annotation_sterm_snps_HIGH$finalName)
genesOfInterest <- genesOfInterest[genesOfInterest %in% geneUniverse]

geneList <- factor(as.integer(geneUniverse %in% genesOfInterest))
names(geneList) <- geneUniverse
 
myGOdata <- new("topGOdata", description="My project", ontology="BP", allGenes=geneList,  annot = annFUN.gene2GO, gene2GO = geneID2GO)
 
sg <- sigGenes(myGOdata)
numSigGenes(myGOdata)

resultFisher <- runTest(myGOdata, algorithm="classic", statistic="fisher")
resultFisher

allRes <- GenTable(myGOdata, classicFisher = resultFisher, orderBy = "resultFisher", ranksOf = "classicFisher", topNodes = 200)
allRes


```

##### disentengle Ldel

```{r}



###-----------
##plot all SNPs that occur in one sample
###-----------

###---------change name
table(sample_relative_wide$X.CHROM)
sample_relative_wide_renamed <- sample_relative_wide
# colnames(sample_relative_wide)
colnames(sample_relative_wide_renamed)


###----------not explained by isolates or reference
colMeans(only_meta_ldel[2:6],na.rm = TRUE) 
# meta_ldel
meta_ldel_notExplained <- gather(only_meta_ldel, sample, Snps, "Lyo_202_2012":"Versand_202", factor_key=TRUE,na.rm = TRUE) 
meta_ldel_notExplained$grouping <- "not Explained\nbyGenomes"
colMeans(only_meta_ldel[2:6],na.rm = TRUE) ##EXPLAINED by unknown

###----------explained by isolates

Ldel_L99_snps <- sample_relative_wide_renamed[which((sample_relative_wide_renamed$mst8>0.9) %in% TRUE),c("site","Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202")]
colMeans(Ldel_L99_snps[2:6],na.rm = TRUE)
Ldel_mst8_snps_Explained <- gather(Ldel_L99_snps, sample, Snps, "Lyo_202_2012":"Versand_202", factor_key=TRUE,na.rm = TRUE) 
Ldel_mst8_snps_Explained$grouping <- "mst8"

Ldel_L80_snps <- sample_relative_wide_renamed[which((sample_relative_wide_renamed$mst5>0.9) %in% TRUE),c("site","Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202")];Ldel_L80_snps
colMeans(Ldel_L80_snps[2:6],na.rm = TRUE)
Ldel_mst5_snps_Explained <- gather(Ldel_L80_snps, sample, Snps, "Lyo_202_2012":"Versand_202", factor_key=TRUE,na.rm = TRUE) 
Ldel_mst5_snps_Explained$grouping <- "mst5";Ldel_mst5_snps_Explained

Ldel_L71_snps <- sample_relative_wide_renamed[which((sample_relative_wide_renamed$mst4>0.9) %in% TRUE),c("site","Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202")];Ldel_L71_snps
colMeans(Ldel_L71_snps[2:6],na.rm = TRUE)
Ldel_mst4_snps_Explained <- gather(Ldel_L71_snps, sample, Snps, "Lyo_202_2012":"Versand_202", factor_key=TRUE,na.rm = TRUE) 
Ldel_mst4_snps_Explained$grouping <- "mst4";Ldel_mst4_snps_Explained

Ldel_L70_snps <- sample_relative_wide_renamed[which((sample_relative_wide_renamed$mst3>0.9) %in% TRUE),c("site","Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202")];Ldel_L70_snps
colMeans(Ldel_L70_snps[2:6],na.rm = TRUE)
Ldel_mst3_snps_Explained <- gather(Ldel_L70_snps, sample, Snps, "Lyo_202_2012":"Versand_202", factor_key=TRUE,na.rm = TRUE) 
Ldel_mst3_snps_Explained$grouping <- "mst3";Ldel_mst3_snps_Explained

Ldel_L108_snps <- sample_relative_wide_renamed[which((sample_relative_wide_renamed$mst7>0.9) %in% TRUE),c("site","Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202")];Ldel_L108_snps
colMeans(Ldel_L108_snps[2:6],na.rm = TRUE)
Ldel_mst7_snps_Explained <- gather(Ldel_L108_snps, sample, Snps, "Lyo_202_2012":"Versand_202", factor_key=TRUE,na.rm = TRUE) 
Ldel_mst7_snps_Explained$grouping <- "mst7";Ldel_mst7_snps_Explained

Ldel_L35_snps <- sample_relative_wide_renamed[which((sample_relative_wide_renamed$mst9>0.9) %in% TRUE),c("site","Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202")];Ldel_L35_snps
colMeans(Ldel_L35_snps[2:6],na.rm = TRUE)
Ldel_mst9_snps_Explained <- gather(Ldel_L35_snps, sample, Snps, "Lyo_202_2012":"Versand_202", factor_key=TRUE,na.rm = TRUE) 
Ldel_mst9_snps_Explained$grouping <- "mst9";Ldel_mst9_snps_Explained

Ldel_L104_snps <- sample_relative_wide_renamed[which((sample_relative_wide_renamed$mst6>0.9) %in% TRUE),c("site","Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202")];Ldel_L104_snps
colMeans(Ldel_L104_snps[2:6],na.rm = TRUE)
Ldel_mst6_snps_Explained <- gather(Ldel_L104_snps, sample, Snps, "Lyo_202_2012":"Versand_202", factor_key=TRUE,na.rm = TRUE) 
Ldel_mst6_snps_Explained$grouping <- "mst6";Ldel_mst6_snps_Explained

Ldel_L44_snps <- sample_relative_wide_renamed[which((sample_relative_wide_renamed$mst10>0.9) %in% TRUE),c("site","Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202")];Ldel_L44_snps
colMeans(Ldel_L44_snps[2:6],na.rm = TRUE)
Ldel_mst10_snps_Explained <- gather(Ldel_L44_snps, sample, Snps, "Lyo_202_2012":"Versand_202", factor_key=TRUE,na.rm = TRUE) 
Ldel_mst10_snps_Explained$grouping <- "mst10";Ldel_mst10_snps_Explained


###----------merge different samples

alltogether_SNPs <- rbind(meta_ldel_notExplained,Ldel_mst3_snps_Explained,Ldel_mst4_snps_Explained,Ldel_mst5_snps_Explained,Ldel_mst6_snps_Explained,Ldel_mst7_snps_Explained,Ldel_mst8_snps_Explained,Ldel_mst9_snps_Explained,Ldel_mst10_snps_Explained)

table(alltogether_SNPs$grouping)
alltogether_SNPs$grouping = factor(alltogether_SNPs$grouping, levels=c("not Explained\nbyGenomes","mst3","mst7","mst6","mst9","mst4","mst8","mst5","mst10"))

###----------boxplot


  pall_sterm_CORE_box <- ggplot(alltogether_SNPs,aes(x=grouping,y=Snps,color=sample,fill=sample))+ geom_boxplot()+
       # facet_grid(group~.)+
   labs("",
         x="",
         # y="",
          y="all")+
       # scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="right",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
pall_sterm_CORE_box

###----------look at SNP frequency 
# table(sample_relative_wide_renamed$X.CHROM)
sample_relative_wide_renamed_Ldel <- sample_relative_wide_renamed[which(sample_relative_wide_renamed$X.CHROM=="L_delbrueckii_RMK202"),]
sample_relative_wide_frequency <- sample_relative_wide_renamed_Ldel %>% select(site,mst3,mst7,mst6,mst9,mst4,mst8,mst5,mst10) %>% gather(., sample, Snps, "mst3","mst7","mst6","mst9","mst4","mst8","mst5","mst10", factor_key=TRUE,na.rm = TRUE) 

ggplot(sample_relative_wide_frequency,aes(x=Snps))+geom_histogram()+facet_grid(sample~.,scales = "free")+
  lims(x=c(0.1,1))+
  theme_classic()

###----------frequeny in Isolate vs. frequency in meta
meta_vs_iso <- data.frame()

for (meta in c("Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202")) {
  
  
  for (iso in c("mst3","mst7","mst6","mst9","mst4","mst8","mst5","mst10")) {
    
    grouping <- ifelse(iso %in% c("mst3","mst7","mst6","mst9","mst4"),"group1","group2")
    
    tmp <- data.frame(site=sample_relative_wide_renamed_Ldel$site,metaName=meta,isoName=iso,groupingName=grouping,metaSNPs=sample_relative_wide_renamed_Ldel[,meta],isoSNPs=sample_relative_wide_renamed_Ldel[,iso])
    meta_vs_iso <- rbind(meta_vs_iso,tmp)
    
  }

}##meta

ggplot(meta_vs_iso,aes(x=isoSNPs,y=metaSNPs,color=isoName))+geom_point()+theme_classic()
ggplot(meta_vs_iso,aes(x=isoSNPs,y=metaSNPs,color=groupingName))+geom_point()+theme_classic()
ggplot(meta_vs_iso,aes(x=isoSNPs,y=metaSNPs,color=metaName))+geom_point()+theme_classic()


##make NA to zero
meta_vs_iso_cleaned <- meta_vs_iso
# meta_vs_iso_cleaned <- meta_vs_iso[!is.na(meta_vs_iso$metaSNPs),]
# meta_vs_iso_cleaned <- meta_vs_iso_cleaned[!is.na(meta_vs_iso_cleaned$isoSNPs),]

meta_vs_iso_cleaned[c("metaSNPs", "isoSNPs")][is.na(meta_vs_iso_cleaned[c("metaSNPs", "isoSNPs")])] <- 0


meta_vs_iso_cleaned <- meta_vs_iso_cleaned[which(meta_vs_iso_cleaned$metaSNPs>"0" |meta_vs_iso_cleaned$isoSNPs>"0"),]

# ggplot(meta_vs_iso_cleaned)+stat_density2d(aes(x=isoSNPs,y=metaSNPs))+theme_classic()
# ggplot(meta_vs_iso_cleaned,aes(x=isoSNPs,y=metaSNPs))+geom_hex(binwidth = c(0.02, 0.02),aes(fill=log(..count..)))+theme_classic()
# ggplot(meta_vs_iso_cleaned,aes(x=isoSNPs,y=metaSNPs))+geom_hex(binwidth = c(0.02, 0.02))+theme_classic()
ggplot(meta_vs_iso_cleaned,aes(x=isoSNPs,y=metaSNPs))+geom_hex(binwidth = c(0.03, 0.03),aes(fill=log(..count..)))+theme_classic()

###----------summed frequeny in Isolate vs. frequency in meta

sample_relative_wide_renamed_Ldel_rowsumsIsolates <- sample_relative_wide_renamed_Ldel %>% select(site,Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202)
  sample_relative_wide_renamed_Ldel_rowsumsIsolates$rowSumsIsolates <- rowSums(sample_relative_wide_renamed_Ldel[,c("mst3","mst7","mst6","mst9","mst4","mst8","mst5","mst10")],na.rm = TRUE)
  sample_relative_wide_renamed_Ldel_rowsumsIsolates_long <- gather(sample_relative_wide_renamed_Ldel_rowsumsIsolates, sample, Snps, "Lyo_202_2012":"Versand_202", factor_key=TRUE,na.rm = TRUE) 

 
sample_relative_wide_renamed_Ldel_rowsumsIsolates_long[c("rowSumsIsolates", "Snps")][is.na(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long[c("rowSumsIsolates", "Snps")])] <- 0


sample_relative_wide_renamed_Ldel_rowsumsIsolates_long <- sample_relative_wide_renamed_Ldel_rowsumsIsolates_long[which(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long$rowSumsIsolates>"0" |sample_relative_wide_renamed_Ldel_rowsumsIsolates_long$Snps>"0"),]

ggplot(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long,aes(x=rowSumsIsolates,y=Snps))+geom_hex(binwidth = c(0.3, 0.02),aes(fill=log(..count..)))+theme_classic()
ggplot(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long,aes(x=rowSumsIsolates,y=Snps))+geom_point()+theme_classic()


###----------tsne

library(Rtsne)
library(ggrepel)
library(philentropy)

sample_relative_wide_renamed_Ldel_onlyIsolates <- sample_relative_wide_renamed_Ldel %>% select(mst3,mst7,mst6,mst9,mst4,mst8,mst5,mst10)
sample_relative_wide_renamed_Ldel_onlyIsolates[c("mst3","mst7","mst6","mst9","mst4","mst8","mst5","mst10")][is.na(sample_relative_wide_renamed_Ldel_onlyIsolates[c("mst3","mst7","mst6","mst9","mst4","mst8","mst5","mst10")])] <- 0
annotation_prep <- sample_relative_wide_renamed_Ldel_onlyIsolates[!duplicated(sample_relative_wide_renamed_Ldel_onlyIsolates), ]

dist_sterm <- distance(as.matrix(annotation_prep), method = "jaccard")

tsne <- Rtsne(dist_sterm, dims = 2, perplexity=30, verbose=TRUE, max_iter = 500)
 

# tsne <- Rtsne(sample_relative_safe_final_tsne_safe_sterm_woDuplicates[,c("Lyoampulle\n2012","Lyoampulle\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018")], dims = 2, perplexity=30, verbose=TRUE, max_iter = 500)
 # tsne <- Rtsne(sample_relative_safe_final_tsne_safe_sterm_woDuplicates[,c("Lyo_202_2012","Lyo_202_2014","RMK202","Konserve_202","Versand_202")], dims = 2, perplexity=30, verbose=TRUE, max_iter = 500)
tsne_plot <- as.data.frame(tsne$Y)
tsne_plot$site <- sample_relative_wide_renamed_Ldel[!duplicated(sample_relative_wide_renamed_Ldel_onlyIsolates), 3]

sample_relative_wide_renamed_Ldel_onlyIsolates_02 <- sample_relative_wide_renamed_Ldel %>% select(site,mst3,mst7,mst6,mst9,mst4,mst8,mst5,mst10)

tsne_plot <- merge(tsne_plot,sample_relative_wide_renamed_Ldel_onlyIsolates_02,by="site",all.x = TRUE)


##plot blank
ggplot(tsne_plot,aes(x=V1,y=V2))+geom_point()


##------------clusters
# set.seed(123456)
# ds.real = dbscan(tsne_plot[,c(1,2)], 1.5)
# tsne_plot$density = factor(ds.real$cluster)
# ds.cent = tsne_plot %>% group_by(density) %>% select(V1, 
#     V2) %>% summarize_all(mean)
# ggplot(tsne_plot, aes(x = V1, y = V2, colour = density)) + 
#     geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label = density), 
#     data = ds.cent) + guides(colour = FALSE)


##clusters
hc.norm = hclust(dist(tsne_plot[,c(1,2)]))
tsne_plot$hclust = factor(cutree(hc.norm, 7))
hc.norm.cent = tsne_plot %>% group_by(hclust) %>% select(V1, 
    V2) %>% summarize_all(mean)
plotTsne <- ggplot(tsne_plot, aes(x = V1, y = V2, color = hclust, text =paste("effect:", site,"\nmst3:",mst3,"\nmst7:",mst7,"\nmst6:",mst6,"\nmst9:",mst9,"\nmst4:",mst4,"\nmst8:",mst8,"\nmst5:",mst5,"\nmst10:",mst10))) + 
  # facet_grid(species~.)+
    geom_point(alpha = 0.3) + theme_bw() +# geom_label_repel(aes(label = hclust), data = hc.norm.cent) +
  guides(colour = FALSE) + 
    ggtitle("Linkage criterion: Complete")

  # pdf("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_tsnePlot_sterm.pdf",width=5,height=5,paper='special')
  # svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_tsnePlot_sterm.svg",width=4,height=4)
  # png("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_tsnePlot_sterm.png",width=1600,height=1600,res=300)
plotTsne

library(plotly)
ggp <- ggplotly(plotTsne)
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_notexplained_ldel_colored_tsne_isolates.html")

##--------------
##make a line graph
##--------------

  pall_sterm_CORE_box <- ggplot(alltogether_SNPs,aes(x=grouping,y=Snps,color=sample,fill=sample))+ geom_boxplot()+
       # facet_grid(group~.)+
   labs("",
         x="",
         # y="",
          y="all")+
       # scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       scale_x_discrete( expand = c(0, 0)) +
  scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
  theme_classic()+
          # ylim(c(0,1))+
 theme(axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       #axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="right",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
pall_sterm_CORE_box

nrow(alltogether_SNPs)
alltogether_SNPs_reduced <- alltogether_SNPs[which(alltogether_SNPs$Snps!=0),]
nrow(alltogether_SNPs_reduced)

  ggplot(alltogether_SNPs_reduced,aes(x=grouping,y=Snps,group=site))+ geom_line(size=0.2, alpha=.2)+ 
       facet_grid(sample~.)+
   labs("",
         x="",
         # y="",
          y="not\nexplained")+
      #  scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       # scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0),limits = c(0,0.2)) +
     scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
    # ylim(0,0.2)+
  theme_classic()+
 theme(#axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       # axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
  
  
  ##-------------
  ##plot uniques
  ##-------------
  length(alltogether_SNPs$site)
  length(unique(alltogether_SNPs$site))
  
  
  sum(sample_relative_wide$site %in% unique(alltogether_SNPs$site))
  sample_relative_wide_subset_wide <- sample_relative_wide[which(sample_relative_wide$site %in% unique(alltogether_SNPs$site)),]

sample_relative_wide_subset_long <- gather(sample_relative_wide_subset_wide, sample, Snps, c("L104","L108","L35","L39","L44","L70","L71","L80","L99"), factor_key=TRUE,na.rm = TRUE) 
nrow(sample_relative_wide_subset_long)

# sample_relative_wide_subset_long$sample"L70"="mst3", "L44"="mst10", "L108"="mst7", "L99"="mst8", "L104"="mst6", "L80"="mst5", "L35"="mst9", "L71"="mst4", "L104"="mst6","Lactobacillus_delbrueckii_RMK202"="RMK202"

sample_relative_wide_subset_long$sample <- revalue(sample_relative_wide_subset_long$sample, c("L70"="mst3", "L44"="mst10", "L108"="mst7", "L99"="mst8", "L104"="mst6", "L80"="mst5", "L35"="mst9", "L71"="mst4", "L104"="mst6","Lactobacillus_delbrueckii_RMK202"="RMK202"))

sample_relative_wide_subset_long[sample_relative_wide_subset_long$Snps<0.8,"Snps"] <- 0

sample_relative_wide_subset_long$sample = factor(sample_relative_wide_subset_long$sample, levels=c("mst3","mst7","mst6","mst9","mst4","mst8","mst5","mst10"))

  ggplot(sample_relative_wide_subset_long,aes(x=sample,y=Snps,group=site))+ geom_line(size=0.2, alpha=.2)+ 
       # facet_grid(sample~.)+
   labs("",
         x="",
         # y="",
          y="not\nexplained")+
      #  scale_fill_manual(values="gray41")+
      # scale_color_manual(values="gray41")+
       # scale_x_discrete( expand = c(0, 0)) +
  # scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0),limits = c(0,0.2)) +
     scale_y_continuous(breaks =c(0,0.5,1),labels=c(0,0.5,1), expand = c(0, 0)) +
    # ylim(0,0.2)+
  theme_classic()+
 theme(#axis.text.x = element_text(angle = 75, hjust = 1,size=8),
       # axis.text.x=element_blank(),
          rect = element_rect(fill = "transparent"), # all rectangles      #axis.text.x = element_blank(),
          legend.position="none",
       axis.text.y = element_text(size=8),
       axis.title.y = element_text(size=9),
            plot.margin=unit(c(t = 0, r = 0.5, b = 0, l = 0.1),"cm")
       )
  
  ###----------tsne this

library(Rtsne)
library(ggrepel)
library(philentropy)

  

  
sample_relative_wide_subset_wide_set <- sample_relative_wide_subset_wide
sample_relative_wide_subset_wide_set[is.na(sample_relative_wide_subset_wide_set)] <- 0

colnames(sample_relative_wide_subset_wide_set) <- revalue(colnames(sample_relative_wide_subset_wide_set), c("L70"="mst3", "L44"="mst10", "L108"="mst7", "L99"="mst8", "L80"="mst5", "L35"="mst9", "L71"="mst4", "L104"="mst6"))  


sample_relative_wide_subset_wide_preped <- sample_relative_wide_subset_wide_set[,c("mst3","mst7","mst6","mst9","mst4","mst8","mst5","mst10")]
sample_relative_wide_subset_wide_preped[duplicated(sample_relative_wide_subset_wide_preped), ]

nrow(sample_relative_wide_subset_wide_preped)
annotation_prep <- sample_relative_wide_subset_wide_preped[!duplicated(sample_relative_wide_subset_wide_preped), ]
nrow(annotation_prep)

annotation_prep[is.na(annotation_prep)] <- 0

dist_sterm <- distance(as.matrix(annotation_prep), method = "jaccard")

tsne <- Rtsne(dist_sterm, dims = 2, perplexity=30, verbose=TRUE, max_iter = 500)
 

# tsne <- Rtsne(sample_relative_safe_final_tsne_safe_sterm_woDuplicates[,c("Lyoampulle\n2012","Lyoampulle\n2014","working\nstock","starter\nculture\n2012","starter\nculture\n2018")], dims = 2, perplexity=30, verbose=TRUE, max_iter = 500)
 # tsne <- Rtsne(sample_relative_safe_final_tsne_safe_sterm_woDuplicates[,c("Lyo_202_2012","Lyo_202_2014","RMK202","Konserve_202","Versand_202")], dims = 2, perplexity=30, verbose=TRUE, max_iter = 500)
tsne_plot <- as.data.frame(tsne$Y)
tsne_plot$site <- sample_relative_wide_subset_wide_set[!duplicated(sample_relative_wide_subset_wide_preped), "site"]

sample_relative_wide_renamed_Ldel_onlyIsolates_02 <- sample_relative_wide_subset_wide_set %>% select(site,mst3,mst7,mst6,mst9,mst4,mst8,mst5,mst10)
# sample_relative_wide_renamed_Ldel_onlyIsolates_02 <- sample_relative_wide_subset_wide_set %>% select(site,L104,L108,L35,L39,L44,L70,L71,L80,L99)

tsne_plot <- merge(tsne_plot,sample_relative_wide_renamed_Ldel_onlyIsolates_02,by="site",all.x = TRUE)


##plot blank
ggplot(tsne_plot,aes(x=V1,y=V2))+geom_point()


##------------clusters
# set.seed(123456)
# ds.real = dbscan(tsne_plot[,c(1,2)], 1.5)
# tsne_plot$density = factor(ds.real$cluster)
# ds.cent = tsne_plot %>% group_by(density) %>% select(V1, 
#     V2) %>% summarize_all(mean)
# ggplot(tsne_plot, aes(x = V1, y = V2, colour = density)) + 
#     geom_point(alpha = 0.3) + theme_bw() + geom_label_repel(aes(label = density), 
#     data = ds.cent) + guides(colour = FALSE)


##clusters
hc.norm = hclust(dist(tsne_plot[,c(1,2)]))
tsne_plot$hclust = factor(cutree(hc.norm, 7))
hc.norm.cent = tsne_plot %>% group_by(hclust) %>% select(V1, 
    V2) %>% summarize_all(mean)
plotTsne <- ggplot(tsne_plot, aes(x = V1, y = V2, color = hclust, text =paste("effect:", site,"\nmst3:",mst3,"\nmst7:",mst7,"\nmst6:",mst6,"\nmst9:",mst9,"\nmst4:",mst4,"\nmst8:",mst8,"\nmst5:",mst5,"\nmst10:",mst10))) + 
  # facet_grid(species~.)+
    geom_point(alpha = 0.3) + theme_bw() +# geom_label_repel(aes(label = hclust), data = hc.norm.cent) +
  guides(colour = FALSE) + 
    ggtitle("Linkage criterion: Complete")

  # pdf("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_tsnePlot_sterm.pdf",width=5,height=5,paper='special')
  # svg("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_tsnePlot_sterm.svg",width=4,height=4)
  # png("~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_tsnePlot_sterm.png",width=1600,height=1600,res=300)
plotTsne


library(plotly)
ggp <- ggplotly(plotTsne)
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_notexplained_ldel_colored_tsne_isolates_02.html")


##-------------
##Unique for Strains
##-------------

sample_relative_wide_strains <- sample_relative_wide
sample_relative_wide_strains[is.na(sample_relative_wide_strains)] <- 0

colnames(sample_relative_wide_strains) <- revalue(colnames(sample_relative_wide_strains), c("L70"="mst3", "L44"="mst10", "L108"="mst7", "L99"="mst8", "L80"="mst5", "L35"="mst9", "L71"="mst4", "L104"="mst6"))  


colMeans(sample_relative_wide_strains[which(sample_relative_wide_strains$site %in% inAll_Ldel$site),c("Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202")])

Ldel_mst8_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8>0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_mst8_snps

Ldel_mst5_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5>0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_mst5_snps

Ldel_mst4_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4>0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_mst4_snps

Ldel_mst3_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3>0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_mst3_snps

Ldel_mst7_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7>0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_mst7_snps

Ldel_mst9_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9>0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_mst9_snps

Ldel_L39_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39>0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_L39_snps

Ldel_mst6_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6>0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_mst6_snps

Ldel_mst10_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10>0.9) %in% TRUE),];Ldel_mst10_snps



##-------------
##Unique for all strains in one of three subgroups
##-------------
library(robustbase)
sample_relative_wide_strains <- sample_relative_wide
sample_relative_wide_strains[is.na(sample_relative_wide_strains)] <- 0

colnames(sample_relative_wide_strains) <- revalue(colnames(sample_relative_wide_strains), c("L70"="mst3", "L44"="mst10", "L108"="mst7", "L99"="mst8", "L80"="mst5", "L35"="mst9", "L71"="mst4", "L104"="mst6"))  

colMeans(sample_relative_wide_strains[which(sample_relative_wide_strains$site %in% inAll_Ldel$site),c("Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202")])

Ldel_subcluster_01 <- sample_relative_wide_strains[which(((sample_relative_wide_strains$mst8<0.1 & sample_relative_wide_strains$mst5<0.1&sample_relative_wide_strains$mst10<0.1) &(sample_relative_wide_strains$mst4>0.9& sample_relative_wide_strains$mst3>0.9& sample_relative_wide_strains$mst7>0.9& sample_relative_wide_strains$mst9>0.9 & sample_relative_wide_strains$mst6>0.9)) %in% TRUE),];Ldel_subcluster_01;Ldel_subcluster_01 %>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% as.matrix() %>% colMedians()

Ldel_subcluster_02 <- sample_relative_wide_strains[which(((sample_relative_wide_strains$mst8>0.9) & (sample_relative_wide_strains$mst5<0.1&sample_relative_wide_strains$mst10<0.1 & sample_relative_wide_strains$mst4<0.1& sample_relative_wide_strains$mst3<0.1& sample_relative_wide_strains$mst7<0.1& sample_relative_wide_strains$mst9<0.1 & sample_relative_wide_strains$mst6<0.1)) %in% TRUE),];Ldel_subcluster_02;Ldel_subcluster_02 %>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% as.matrix() %>% colMedians()

Ldel_subcluster_03 <- sample_relative_wide_strains[which(((sample_relative_wide_strains$mst8<0.1) & (sample_relative_wide_strains$mst5>0.9&sample_relative_wide_strains$mst10>0.9) & (sample_relative_wide_strains$mst4<0.1& sample_relative_wide_strains$mst3<0.1& sample_relative_wide_strains$mst7<0.1& sample_relative_wide_strains$mst9<0.1 & sample_relative_wide_strains$mst6<0.1)) %in% TRUE),];Ldel_subcluster_03;Ldel_subcluster_03 %>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% as.matrix() %>% colMedians()

Ldel_subcluster_largeCulster_02_03 <- sample_relative_wide_strains[which(((sample_relative_wide_strains$mst8>0.9 & sample_relative_wide_strains$mst5>0.9&sample_relative_wide_strains$mst10>0.9) & (sample_relative_wide_strains$mst4<0.1& sample_relative_wide_strains$mst3<0.1& sample_relative_wide_strains$mst7<0.1& sample_relative_wide_strains$mst9<0.1 & sample_relative_wide_strains$mst6<0.1)) %in% TRUE),];Ldel_subcluster_largeCulster_02_03;Ldel_subcluster_largeCulster_02_03 %>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% as.matrix() %>% colMedians();Ldel_subcluster_largeCulster_02_03 %>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% gather(.) %>% ggplot(.,aes(x=key,y=value))+geom_boxplot()


Ldel_mst5_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5>0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_mst5_snps

Ldel_mst4_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4>0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_mst4_snps

Ldel_mst3_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3>0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_mst3_snps

Ldel_mst7_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7>0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_mst7_snps

Ldel_mst9_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9>0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_mst9_snps

Ldel_L39_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39>0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_L39_snps

Ldel_mst6_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6>0.9& sample_relative_wide_strains$mst10<0.9) %in% TRUE),];Ldel_mst6_snps

Ldel_mst10_snps <- sample_relative_wide_strains[which((sample_relative_wide_strains$mst8<0.9 & sample_relative_wide_strains$mst5<0.9& sample_relative_wide_strains$mst4<0.9& sample_relative_wide_strains$mst3<0.9& sample_relative_wide_strains$mst7<0.9& sample_relative_wide_strains$mst9<0.9& sample_relative_wide_strains$L39<0.9& sample_relative_wide_strains$mst6<0.9& sample_relative_wide_strains$mst10>0.9) %in% TRUE),];Ldel_mst10_snps

###----------summed frequeny in Isolate vs. frequency in meta

sample_relative_wide_renamed_Ldel_rowsumsIsolates <- sample_relative_wide_renamed_Ldel %>% select(site,Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202)
  sample_relative_wide_renamed_Ldel_rowsumsIsolates$rowSumsIsolates <- rowSums(sample_relative_wide_renamed_Ldel[,c("mst3","mst7","mst6","mst9","mst4","mst8","mst5","mst10")],na.rm = TRUE)
  sample_relative_wide_renamed_Ldel_rowsumsIsolates_long <- gather(sample_relative_wide_renamed_Ldel_rowsumsIsolates, sample, Snps, "Lyo_202_2012":"Versand_202", factor_key=TRUE,na.rm = TRUE) 

 
sample_relative_wide_renamed_Ldel_rowsumsIsolates_long[c("rowSumsIsolates", "Snps")][is.na(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long[c("rowSumsIsolates", "Snps")])] <- 0


sample_relative_wide_renamed_Ldel_rowsumsIsolates_long <- sample_relative_wide_renamed_Ldel_rowsumsIsolates_long[which(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long$rowSumsIsolates>"0" |sample_relative_wide_renamed_Ldel_rowsumsIsolates_long$Snps>"0"),]

ggplot(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long,aes(x=rowSumsIsolates,y=Snps))+geom_hex(binwidth = c(0.3, 0.02),aes(fill=log(..count..)))+theme_classic()
ggplot(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long,aes(x=rowSumsIsolates,y=Snps))+geom_point()+theme_classic()


sample_relative_wide_renamed_Ldel_rowsumsIsolates_long_subset <- sample_relative_wide_renamed_Ldel_rowsumsIsolates_long[which(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long$rowSumsIsolates>1),]
ggplot(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long_subset,aes(x=rowSumsIsolates,y=Snps))+geom_point()+theme_classic()


sample_relative_wide_renamed_Ldel_rowsumsIsolates_long_subset <- sample_relative_wide_renamed_Ldel_rowsumsIsolates_long_subset[which(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long_subset$Snps>0.2),]
ggplot(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long_subset,aes(x=rowSumsIsolates,y=Snps))+geom_point()+theme_classic()

length(unique(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long_subset$site))

sample_relative_wide_strains_ldel_interesting <- sample_relative_wide_strains_ldel[which(sample_relative_wide_strains_ldel$site %in% unique(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long_subset$site)),]


sample_relative_wide_renamed_Ldel_rowsumsIsolates_long_subset_PLOT <- merge(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long_subset,sample_relative_wide_strains_ldel,by="site",all.x = TRUE)
PLOT1 <- ggplot(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long_subset_PLOT,aes(x=rowSumsIsolates,y=Snps, text =paste("effect:", site,"\nmst3:",mst3,"\nmst7:",mst7,"\nmst6:",mst6,"\nmst9:",mst9,"\nmst4:",mst4,"\nmst8:",mst8,"\nmst5:",mst5,"\nmst10:",mst10)))+geom_point()+theme_classic()

PLOT1

library(plotly)
ggp <- ggplotly(PLOT1)
htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_notexplained_ldel_colored_tsne_isolates_03.html")


##---------------
##subset snps to the ones that only occure in the metagenome
##---------------

keep <- sample_relative_wide_strains_ldel%>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% rowSums() >0 

sample_relative_wide_strains_ldel%>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% as.matrix() %>% colMedians()
sample_relative_wide_strains_ldel%>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% gather(.) %>% ggplot(.,aes(x=key,y=value))+geom_boxplot()

# sample_relative_wide_strains_ldel_kept <- sample_relative_wide_strains_ldel[which(keep),]
# sample_relative_wide_strains_ldel_kept%>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% as.matrix() %>% colMedians()
sample_relative_wide_strains_ldel_kept%>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% gather(.) %>% ggplot(.,aes(x=key,y=value))+geom_boxplot()
sample_relative_wide_strains_ldel_kept_long <- sample_relative_wide_strains_ldel_kept %>% gather(.,sample, snps,c("Lyo_202_2012","Lyo_202_2014","Konserve_202",RMK202,"Versand_202")) 
sample_relative_wide_renamed_Ldel_rowsumsIsolates_long_subset_PLOT <- merge(sample_relative_wide_strains_ldel_kept_long,sample_relative_wide_strains_ldel,by="site",all.x = TRUE)

nrow(sample_relative_wide_strains_ldel_kept)
library(plyr)
# Renamed_gained_lost_SNPs_long_02$sample <- revalue(Renamed_gained_lost_SNPs_long_02$sample, c("Lyo_202_2012"="Lyoampulle\n2012", "Lyo_202_2014"="Lyoampulle\n2014", "Konserve_202"="working\nstock", "RMK202"="starter\nculture\n2012", "Versand_202"="starter\nculture\n2018"))
Renamed_gained_lost_SNPs_long_02$sample = factor(Renamed_gained_lost_SNPs_long_02$sample, levels=c("Lyo_202_2012","Lyo_202_2014","Konserve_202","RMK202","Versand_202"))




# plot1 <- ggplot(sample_relative_wide_renamed_Ldel_rowsumsIsolates_long_subset_PLOT,aes(x=sample,y=snps, text =paste("effect:", site,"\nmst3:",mst3,"\nmst7:",mst7,"\nmst6:",mst6,"\nmst9:",mst9,"\nmst4:",mst4,"\nmst8:",mst8,"\nmst5:",mst5,"\nmst10:",mst10)))+geom_boxplot()
# plot1
# library(plotly)
# ggp <- ggplotly(plot1)
# htmlwidgets::saveWidget(ggp, "~/Desktop/Projects/2019_Pilotplan/02_mapping2referenceDB/05_2Phage_Bact_DB/plots/snpCalling/01_snpCalling_freebayes_colored_notexplained_ldel_colored_tsne_isolates_04.html")

##-------------
##Unique for strains in one of three subgroups
##-------------

Ldel_subcluster_cluster_01_parts <- sample_relative_wide_strains_ldel_kept[which(((sample_relative_wide_strains_ldel_kept$mst8<0.1 & sample_relative_wide_strains_ldel_kept$mst5<0.1 & sample_relative_wide_strains_ldel_kept$mst10<0.1) & (sample_relative_wide_strains_ldel_kept$mst4>0.9 | sample_relative_wide_strains_ldel_kept$mst3>0.9 | sample_relative_wide_strains_ldel_kept$mst7>0.9 | sample_relative_wide_strains_ldel_kept$mst9>0.9 | sample_relative_wide_strains_ldel_kept$mst6>0.9)) %in% TRUE),];Ldel_subcluster_cluster_01_parts;Ldel_subcluster_cluster_01_parts %>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% as.matrix() %>% colMedians();Ldel_subcluster_cluster_01_parts %>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% gather(.) %>% ggplot(.,aes(x=key,y=value))+geom_boxplot()
cluster1_wo_mag <- Ldel_subcluster_cluster_01_parts %>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% as.matrix() %>% colMedians()%>% t()  %>% data.frame() 
cluster1_wo_mag$group <- "cluster1_woMAG"

Ldel_subcluster_cluster_023_parts <- sample_relative_wide_strains_ldel_kept[which(((sample_relative_wide_strains_ldel_kept$mst8>0.9 | sample_relative_wide_strains_ldel_kept$mst5>0.9|sample_relative_wide_strains_ldel_kept$mst10>0.9) & (sample_relative_wide_strains_ldel_kept$mst4<0.1& sample_relative_wide_strains_ldel_kept$mst3<0.1& sample_relative_wide_strains_ldel_kept$mst7<0.1& sample_relative_wide_strains_ldel_kept$mst9<0.1 & sample_relative_wide_strains_ldel_kept$mst6<0.1)) %in% TRUE),];Ldel_subcluster_cluster_023_parts;Ldel_subcluster_cluster_023_parts %>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% as.matrix() %>% colMedians();Ldel_subcluster_cluster_023_parts %>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% gather(.) %>% ggplot(.,aes(x=key,y=value))+geom_boxplot()
cluster2_wo_mag <- Ldel_subcluster_cluster_023_parts %>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% as.matrix() %>% colMedians()%>% t()  %>% data.frame() 
cluster2_wo_mag$group <- "cluster2_all"


##not explained
table(sample_relative_wide_strains$X.CHROM)
sample_relative_wide_strains_ldel <- sample_relative_wide_strains[which(sample_relative_wide_strains$X.CHROM=="L_delbrueckii_RMK202"),]

Ldel_subcluster_cluster_non <- sample_relative_wide_strains_ldel_kept[which(((sample_relative_wide_strains_ldel_kept$mst8<0.1& sample_relative_wide_strains_ldel_kept$mst5<=0.9&sample_relative_wide_strains_ldel_kept$mst10<=0.9& sample_relative_wide_strains_ldel_kept$mst4<=0.9& sample_relative_wide_strains_ldel_kept$mst3<=0.9& sample_relative_wide_strains_ldel_kept$mst7<=0.9& sample_relative_wide_strains_ldel_kept$mst9<=0.9 & sample_relative_wide_strains_ldel_kept$mst6<=0.9)) %in% TRUE),];Ldel_subcluster_cluster_non;Ldel_subcluster_cluster_non %>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% as.matrix() %>% colMedians();Ldel_subcluster_cluster_non %>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% gather(.) %>% ggplot(.,aes(x=key,y=value))+geom_boxplot()

unexplained <- Ldel_subcluster_cluster_non %>% select(Lyo_202_2012,Lyo_202_2014,Konserve_202,RMK202,Versand_202) %>% as.matrix() %>% colMedians()%>% t()  %>% data.frame() 
unexplained$group <- "unexplainsed"

##cluster one with mag

merged_medians <- rbind(cluster1_wo_mag,cluster2_wo_mag,unexplained)
MAG_COUNT <- 1-colSums(merged_medians[1:5]) %>% data.frame() %>% t()  %>% data.frame() 
cluster_1_wMag <- MAG_COUNT+cluster1_wo_mag[1:5]

cluster_1_wMag$group <- "cluster1 with MAG"


merged_medians <- rbind(cluster_1_wMag,cluster2_wo_mag,unexplained)

```


######rarefaction

```{r}
library(vegan)
library(diveRsity)
install.packages("hierfstat")
library(hierfstat)

data(gtrunchier)
result<- allelic.richness(gtrunchier[,-1])
gtrunchier
dim(gtrunchier)
plot(result$Ar)
dim(result$Ar)


rarefy(x, sample, se = FALSE, MARGIN = 1)
sample_relative_temp_withONT

install.packages("adiv")
library("adiv")

dim(sample_relative_safe_final_meta_cleaned_02)
forRarefaction <- t(samples)
dim(forRarefaction)
# rare_Rao(forRarefaction, dis, sim = TRUE, resampling = 999, formula = c("QE", "EDI"))
rare_Rao(forRarefaction,dis=NULL)
is.na(forRarefaction)


data(aviurba, package="ade4")
distances<-dist.ktab(ktab.list.df(list(aviurba$traits)), type = "N")
abundances<- aviurba$fau
dim(aviurba$fau)
rare_Rao(abundances, distances, sim = TRUE, resampling = 100)
rare_Rao(abundances)
```


